<!DOCTYPE html>
<html>











  <head>
    <meta charset="utf-8">
    <title>Options for using Galaxy</title>
    
        <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Galaxy is a scientific workflow, data integration, and da..." />
    <meta property="og:title" content="Galaxy Training: Options for using Galaxy" />
    <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da..." />
    <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    <script type="application/ld+json">
      


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:09 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Options for using Galaxy",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Introduction to Galaxy Analyses",
    "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
    "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
  },
  "courseCode": "introduction / options-for-using-galaxy / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'Options for using Galaxy' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/introduction/tutorials/options-for-using-galaxy/slides.html",
  "description": "The questions this  addresses are:\n - Which Galaxy instance should I use?\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Alaa Elghamry"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Alaa Elghamry"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
    }
  ]
}
    </script>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/introduction" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/main/topics/introduction/tutorials/options-for-using-galaxy/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" class="cover-logo"/>


# Options for using Galaxy


<div class="contributors-line">

<a href="/training-material/hall-of-fame/tnabtaf/" class="contributor-badge contributor-tnabtaf"><img src="https://avatars.githubusercontent.com/tnabtaf?s=27" alt="Avatar">Dave Clements</a>


<a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="https://avatars.githubusercontent.com/abretaud?s=27" alt="Avatar">Anthony Bretaudeau</a>


<a href="/training-material/hall-of-fame/pajanne/" class="contributor-badge contributor-pajanne"><img src="https://avatars.githubusercontent.com/pajanne?s=27" alt="Avatar">Anne Pajon</a>


<a href="/training-material/hall-of-fame/nsoranzo/" class="contributor-badge contributor-nsoranzo"><img src="https://avatars.githubusercontent.com/nsoranzo?s=27" alt="Avatar">Nicola Soranzo</a>


<a href="/training-material/hall-of-fame/alaaelghamry/" class="contributor-badge contributor-alaaelghamry"><img src="https://avatars.githubusercontent.com/alaaelghamry?s=27" alt="Avatar">Alaa Elghamry</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>


</div>


<div class="footnote" style="bottom: 5.5em;"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Updated: Nov 9, 2021</div>

<div class="footnote" style="bottom: 4em;"><a href="/training-material/videos/watch.html?v=/introduction/tutorials/options-for-using-galaxy/slides"><i class="far fa-play-circle" aria-hidden="true"></i><span class="visually-hidden">video-slides</span> View video slides for this lecture</a></div>

<div class="footnote" style="bottom: 2.5em;"><i class="fas fa-file-alt" aria-hidden="true"></i><span class="visually-hidden">text-document</span><a href="slides-plain.html"> Plain-text slides</a></div>
<div class="footnote" style="bottom: 1em;"><strong>Tip: </strong>press <kbd>P</kbd> to view the presenter notes</div>

???
Presenter notes contain extra information which might be useful if you intend to use these slides for teaching.

Press `P` again to switch presenter notes off

Press `C` to create a new window where the same presentation will be displayed.
This window is linked to the main window. Changing slides on one will cause the
slide to change on the other.

Useful when presenting.



---







### &lt;i class=&quot;far fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- Which Galaxy instance should I use?


---




# Options for using Galaxy

---

### Options for using Galaxy

Galaxy is available in many ways, each with different configurations and capabilities

Here are some options

???

Many workshops use Galaxy instances that go away once the workshop is done.
This deck lists options for using Galaxy outside of a training context.

---

### usegalaxy.org

- [usegalaxy.org](https://usegalaxy.org/) is the Galaxy Project's public server
- Hosted at the [Texas Advanced Computing Center (TACC)](https://www.tacc.utexas.edu/)
- Lots of
  - tools
  - reference genomes
  - compute power
  - storage
- But also very busy
  - Averages 250,000 jobs *per month* in 2016
  - A single resource cannot scale to meet the demand of the entire world, nor can it support all wanted tools, genomes, ...

???
- usegalaxy.org is the public server, it's hosted at the Texas Advanced Computing Center.
- It has lots of tools, reference genomes, computer power, and free storage also.
- But it's also very busy, in 2016 it had an average of 250,000 jobs per month, that sometimes could slow down the process
- The world of data analysis is huge and full of tools , genomes , and resources
-  A single resource cannot scale to meet the demand of the entire world
---

### Public Galaxy Servers

[.image-25[![GalaxEast a powerful omics galaxy](../../images/public_server_galaxeast.png)]](https://galaxyproject.org/public-galaxy-servers/#galaxeast)
[.image-25[![80+ galaxy servers and counting](../../images/public_servers_odometer_count.png)]](https://galaxyproject.org/public-galaxy-servers/)
[.image-25[![GigaGalaxy's logo mentioning integrated publishing of workflows](../../images/public_server_gigagalaxy.png)]](https://galaxyproject.org/public-galaxy-servers/#gigagalaxy)
[.image-25[![Prove your point, genome statistics made simple with the Genomic Hyperbrowser](../../images/public_server_genomichyperbrowser.png)]](https://hyperbrowser.uio.no/hb/)
[.image-25[![Unreadable tiny text with a head the Lanuage Application Grid](../../images/public_server_lappsgrid.png)]](https://galaxy.lappsgrid.org/)

??? 
- There are currently more than 80 public Galaxy servers, and the number keeps growing.
- for example, GalaxyEast, an open and powerful galaxy instance for integrative omics data analysis
- the Genomic Hyperbrowser to Manage and Analyze collections of genome-wide datasets.
- Galaxy Platform Directory which  lists platforms where you can use or deploy your own Galaxy Server with minimal effort.

---

### Public Galaxy Servers

1. General purpose genomics servers
1. Domain specific servers&lt;br /&gt;
   Can be specific to:
   - *Research methods*: ChIP-Seq, RNA-Seq, repeats, ...
   - *Organisms*: Pathogens, phage, rice, poplars (!), ...
   - *Non-genomic domains*: image analysis, social science, natural language processing, ...
1. Tool servers
   - Host specific tools to make them easy to access and run

Complete list: [bit.ly/gxyServers ](https://bit.ly/gxyServers)

???

- There are three types of servers
- First is general-purpose genomics servers.
- The second is domain-specific servers, which can be specific to: Research methods, Organisms or Non-genomic domains
- the third type is Tool servers , the tool servers host specific tools to make them easy to access and run

---

### Semi-public Galaxy services

Access based on geography or community membership

| Geography  | Resource |
| ---------- | -------- |
| Australia  | [Genomics Virtual Lab (GVL)](https://launch.genome.edu.au/launch)  |
| Canada | [GenAP](https://www.genap.ca/) |
| Norway | [Norwegian e-Infrastructure for Life Sciences (NeLS)](https://nels.bioinfo.no/) |
| US | [Jetstream](https://galaxyproject.org/cloud/jetstream/) |

Complete list: [bit.ly/gxysemipublic ](https://bit.ly/gxysemipublic)

???
- There are some semi-public galaxy servers for which access is restricted according to the user's geographic location or community membership
- In Australia there is the Genomics Virtual Lab
- In Canada there is GenAP
- In Norway there is Norwegian e-Infrastructure for Life Sciences
- In the US there is Jetstream

---

.pull-left[
### Public clouds via CloudLaunch

.left[[CloudLaunch](https://launch.usegalaxy.org/) currently only supports:]
- Jetstream (covered above)
- **Amazon Web Services** (AWS)
  - You need to create an AWS account (which requires a
    credit card) and then provide your public and secret keys to CloudLaunch

]

.pull-right[
[.image-75[![Illegible screenshot of Galaxy Cloud Launch with a form for entering data.](../../images/cloudlaunch_empty.png)]](https://launch.usegalaxy.org/)
]

???
- In addition to servers there is also Public clouds via Galaxy CloudLaunch
- Unfortunetly it currently only supports Jetstream and Amazon Web Services
- You first need to create an Amazon Web Services account (which requires a credit card) and then provide your
 public and secret keys to CloudLaunch

---

.pull-left[
### CloudLaunch / CloudMan instances

- Comes with 100's of tools and many pre-defined reference genomes
- CloudLaunch starts a CloudMan-based Galaxy instance
  - **CloudMan** is a cloud management tool that moves much / all of the cloud management work into a Graphical user interface
]

.pull-right[
[.image-75[![Screenshot of a Galaxy Cloud Man instance](../../../../shared/images/cloudman_based_server_landing_page.png)]](https://wiki.galaxyproject.org/CloudMan)
]

???
- CloudLaunch Comes with hundredss of tools and many pre-defined reference genomes.
- CloudLaunch starts a Galaxy instance called CloudMan.
- CloudMan is a cloud management tool that moves so much of the cloud management work into a Graphical user interface.
---

### CloudMan

[.image-25[![CloudMan interface with two compute nodes shown busy.](../../../../shared/images/cloudman_management_console_showing_autoscaled_w_1_worker.png)]](https://wiki.galaxyproject.org/CloudMan)

CloudMan enables you to statically or dynamically scale compute power

- This example shows a Galaxy Server with a head node and one worker node, and that is configured to scale up to 5 worker nodes, if demand justifies it
- Cloud instances can be incredibly cost effective, but only if you are actually using them when they are up
  - Paying to have a cloud instance up for two weeks that you only use for 8 hours is not cost effective
- **It is very important to shut down and remove everything when you are done**

???
- CloudMan enables you to statically or dynamically scale compute power
- Cloud instances can be incredibly cost effective only if you are actually using them when they are up.
- you can't have a cloud instance up for two weeks when you only use for 8 hours 
- It is very important to shut down and remove everything when you are done
---

### Docker (and maybe other appliances?)

Stuff about docker...

???
- Galaxy could be running in Docker 
---

### Run your own Galaxy locally

- Galaxy is [open source software](https://getgalaxy.org/) and can be installed on local compute infrastructure, from lab servers to institutional compute clusters
- Installing Galaxy locally is relatively easy, but
  - the initial install does not include reference genomes and only has a few tools
  - installing tools and genomes, setting up authentication, and connecting to institutional compute resources all takes work
- There are hundreds of local Galaxy installs around the world
  - Installing tools and genomes has become much easier in recent years, and can now often be done with the Galaxy Admin GUI
  - Authentication and connecting to institutional compute resources is still heavy lifting

???
- Galaxy is open source software which means it's free to use locally 
- it can be installed on local compute infrastructure, from lab servers to institutional compute clusters.
- Installing Galaxy locally is relatively easy, but you have to consider this
- the initial install does not include reference genomes and only has a few tools
- installing tools and genomes, setting up authentication, and connecting to institutional compute resources all takes work
- installing has become much easier and can now be done with the Galaxy Admin GUI
- Authentication and connecting to institutional compute resources is still heavy lifting
---

### Commercial support

- You can buy a preconfigured Galaxy server from [BioTeam](https://bioteam.net/) or [NABE&lt;sup&gt;3&lt;/sup&gt; International](http://www.nabe-intl.co.jp/seq/takerugalaxy.html)
- [Globus Genomics](http://globusgenomics.org/) provides cloud-based Galaxy servers
- [Commercially provided consulting and training](https://galaxyproject.org/support/commercial/) are also available

???
- Commercial solutions are also available, for example:
- You can buy a preconfigured Galaxy server from BioTeam
- Globus Genomics also provides cloud-based Galaxy servers
---

### Which Galaxy instance to use?

- Choose based on
  - size of datasets, available storage, backup
  - data security
  - computational requirements
  - tools installed
???
- In conclusion, choosing which Galaxy instance to use depends on multiple factors like: size of datasets, available storage, backup
- you also need to consider data security, computational requirements, and the tools installed










---

## Thank You!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://training.galaxyproject.org) and all the contributors!


<div class="contributors-line">

<a href="/training-material/hall-of-fame/tnabtaf/" class="contributor-badge contributor-tnabtaf"><img src="https://avatars.githubusercontent.com/tnabtaf?s=27" alt="Avatar">Dave Clements</a>


<a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="https://avatars.githubusercontent.com/abretaud?s=27" alt="Avatar">Anthony Bretaudeau</a>


<a href="/training-material/hall-of-fame/pajanne/" class="contributor-badge contributor-pajanne"><img src="https://avatars.githubusercontent.com/pajanne?s=27" alt="Avatar">Anne Pajon</a>


<a href="/training-material/hall-of-fame/nsoranzo/" class="contributor-badge contributor-nsoranzo"><img src="https://avatars.githubusercontent.com/nsoranzo?s=27" alt="Avatar">Nicola Soranzo</a>


<a href="/training-material/hall-of-fame/alaaelghamry/" class="contributor-badge contributor-alaaelghamry"><img src="https://avatars.githubusercontent.com/alaaelghamry?s=27" alt="Avatar">Alaa Elghamry</a>


</div>



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>


<a rel="license" href="https://creativecommons.org/licenses/by/4.0/">
This material is licensed under the Creative Commons Attribution 4.0 International License</a>.


    </textarea>
	<script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false}, ratio: '16:9'});
      var hljs = remark.highlighter.engine;
      var snippets = document.querySelectorAll('code.remark-code');
        [].forEach.call(snippets,function(snippet){
          snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });
      var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.parentElement;
      }});
    </script>

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('body').prepend("<div style='text-align: center'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>

  </body>
</html>
