<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>De novo transcriptome assembly, annotation, and differential expression analysis</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: De novo transcriptome assembly, annotation, and different..." />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/full-de-novo/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/full-de-novo/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "De novo transcriptome assembly, annotation, and differential expression analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / full-de-novo / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'De novo transcriptome assembly, annotation, and differential expression analysis' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html",
  "timeRequired": "PT3H",
  "description": "The questions this  addresses are:\n - Which biological questions are addressed by the tutorial?\n - Which bioinformatics techniques are important to know for this type of data?\n\n\\nThe objectives are:\n - The learning objectives are the goals of the tutorial\n - They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course\n - They are single sentences describing what a learner should be able to do once they have completed the tutorial\n - You can use Bloom's Taxonomy to write effective learning objectives\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Gildas Le Corguillé"
    },
    {
      "@type": "Person",
      "name": "Erwan Corre"
    },
    {
      "@type": "Person",
      "name": "Xi Liu"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Gildas Le Corguillé"
    },
    {
      "@type": "Person",
      "name": "Erwan Corre"
    },
    {
      "@type": "Person",
      "name": "Xi Liu"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
</script>

<section class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>De novo transcriptome assembly, annotation, and differential expression analysis</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="https://avatars.githubusercontent.com/abretaud?s=27" alt="Avatar">Anthony Bretaudeau</a>


<a href="/training-material/hall-of-fame/lecorguille/" class="contributor-badge contributor-lecorguille"><img src="https://avatars.githubusercontent.com/lecorguille?s=27" alt="Avatar">Gildas Le Corguillé</a>


<a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="https://avatars.githubusercontent.com/r1corre?s=27" alt="Avatar">Erwan Corre</a>


<a href="/training-material/hall-of-fame/xiliu/" class="contributor-badge contributor-xiliu"><img src="https://avatars.githubusercontent.com/xiliu?s=27" alt="Avatar">Xi Liu</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which biological questions are addressed by the tutorial?</p>
</li>
        
        <li><p>Which bioinformatics techniques are important to know for this type of data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>The learning objectives are the goals of the tutorial</p>
</li>
        
        <li><p>They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course</p>
</li>
        
        <li><p>They are single sentences describing what a learner should be able to do once they have completed the tutorial</p>
</li>
        
        <li><p>You can use Bloom’s Taxonomy to write effective learning objectives</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3541678">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/full-de-novo/workflows/" title="Workflows" alt="De novo transcriptome assembly, annotation, and differential expression analysis workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 12, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p>As a result of the development of novel sequencing technologies, the years between 2008 and 2012 saw a large drop in the cost of sequencing. Per megabase and genome, the cost dropped to 1/100,000th and 1/10,000th of the price, respectively. Prior to this, only transcriptomes of organisms that were of broad interest and utility to scientific research were sequenced; however, these developed in 2010s high-throughput sequencing (also called next-generation sequencing) technologies are both cost- and labor- effective, and the range of organisms studied via these methods is expanding.</p>

<p>Examining non-model organisms can provide novel insights into the mechanisms underlying the “diversity of fascinating morphological innovations” that have enabled the abundance of life on planet Earth. In animals and plants, the “innovations” that cannot be examined in common model organisms include mimicry, mutualism, parasitism, and asexual reproduction. De novo transcriptome assembly is often the preferred method to studying non-model organisms, since it is cheaper and easier than building a genome, and reference-based methods are not possible without an existing genome. The transcriptomes of these organisms can thus reveal novel proteins and their isoforms that are implicated in such unique biological phenomena.</p>

<p><a href="https://en.wikipedia.org/wiki/De_novo_transcriptome_assembly">(source)</a></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#read-cleaning-20-minutes" id="markdown-toc-read-cleaning-20-minutes">Read cleaning (20 minutes)</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#read-cleaning-with-trimmomatic" id="markdown-toc-read-cleaning-with-trimmomatic">Read cleaning with <strong>Trimmomatic</strong></a></li>
      <li><a href="#quality-control-after-cleaning" id="markdown-toc-quality-control-after-cleaning">Quality control after cleaning</a></li>
    </ol>
  </li>
  <li><a href="#assembly-120-minutes---computing" id="markdown-toc-assembly-120-minutes---computing">Assembly (120 minutes - computing)</a>    <ol>
      <li><a href="#assembly-with-trinity" id="markdown-toc-assembly-with-trinity">Assembly with <strong>Trinity</strong></a></li>
    </ol>
  </li>
  <li><a href="#assembly-assessment--cleaning" id="markdown-toc-assembly-assessment--cleaning">Assembly assessment / cleaning</a>    <ol>
      <li><a href="#checking-of-the-assembly-statistics" id="markdown-toc-checking-of-the-assembly-statistics">Checking of the assembly statistics</a></li>
      <li><a href="#remapping-on-the-raw-transcriptome" id="markdown-toc-remapping-on-the-raw-transcriptome">Re<span class="notranslate">mapping</span> on the raw transcriptome</a></li>
      <li><a href="#merge-the-mapping-tables-and-compute-normalizations" id="markdown-toc-merge-the-mapping-tables-and-compute-normalizations">Merge the <span class="notranslate">mapping</span> tables and compute normalizations</a></li>
      <li><a href="#compute-contig-ex90n50-statistic-and-ex90-transcript-count" id="markdown-toc-compute-contig-ex90n50-statistic-and-ex90-transcript-count">Compute contig Ex90N50 statistic and Ex90 transcript count</a></li>
      <li><a href="#transcriptome-annotation-completeness" id="markdown-toc-transcriptome-annotation-completeness">Transcriptome annotation completeness</a></li>
      <li><a href="#filter-low-expression-transcripts" id="markdown-toc-filter-low-expression-transcripts">Filter low <span class="notranslate">expression</span> transcripts</a></li>
      <li><a href="#checking-of-the-assembly-statistics-after-cleaning" id="markdown-toc-checking-of-the-assembly-statistics-after-cleaning">Checking of the assembly statistics after cleaning</a></li>
    </ol>
  </li>
  <li><a href="#annotation" id="markdown-toc-annotation">Annotation</a>    <ol>
      <li><a href="#generate-gene-to-transcript-map" id="markdown-toc-generate-gene-to-transcript-map">Generate gene to transcript map</a></li>
      <li><a href="#peptide-prediction" id="markdown-toc-peptide-prediction">Peptide prediction</a></li>
      <li><a href="#similarity-search" id="markdown-toc-similarity-search">Similarity search</a></li>
      <li><a href="#find-signal-peptides" id="markdown-toc-find-signal-peptides">Find signal peptides</a></li>
      <li><a href="#find-transmembrane-domains" id="markdown-toc-find-transmembrane-domains">Find transmembrane domains</a></li>
      <li><a href="#search-again-profile-database" id="markdown-toc-search-again-profile-database">Search again profile database</a></li>
      <li><a href="#transcriptome-annotation-using-trinotate" id="markdown-toc-transcriptome-annotation-using-trinotate">Transcriptome annotation using <strong>Trinotate</strong></a></li>
    </ol>
  </li>
  <li><a href="#differential-expression-de-analysis" id="markdown-toc-differential-expression-de-analysis">Differential <span class="notranslate">Expression</span> (DE) Analysis</a>    <ol>
      <li><a href="#remapping-on-the-filtered-transcriptome-using" id="markdown-toc-remapping-on-the-filtered-transcriptome-using">Re<span class="notranslate">mapping</span> on the filtered transcriptome using</a></li>
      <li><a href="#merge-the-mapping-tables-and-compute-a-tmm-normalization" id="markdown-toc-merge-the-mapping-tables-and-compute-a-tmm-normalization">Merge the <span class="notranslate">mapping</span> tables and compute a TMM normalization</a></li>
      <li><a href="#rnaseq-samples-quality-check" id="markdown-toc-rnaseq-samples-quality-check"><span class="notranslate">RNA</span>Seq samples quality check</a></li>
      <li><a href="#differential-expression-analysis" id="markdown-toc-differential-expression-analysis">Differential <span class="notranslate">expression</span> analysis</a></li>
      <li><a href="#extract-and-cluster-differentially-expressed-transcripts" id="markdown-toc-extract-and-cluster-differentially-expressed-transcripts">Extract and cluster differentially expressed transcripts</a></li>
      <li><a href="#partition-genes-into-expression-clusters" id="markdown-toc-partition-genes-into-expression-clusters">Partition genes into <span class="notranslate">expression</span> clusters</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="read-cleaning-20-minutes">Read cleaning (20 minutes)</h1>

<p>Known sequencing biases:</p>
<ul>
  <li>Unknown nucleotides (Ns)</li>
  <li>Bad quality nucleotides</li>
  <li>Hexamers biases (Illumina. Now corrected ?)</li>
</ul>

<p>Why do we need to correct those?</p>
<ul>
  <li>To remove a lot of sequencing errors (detrimental to the vast majority of assemblers)</li>
  <li>Because most de-bruijn graph based assemblers can’t handle unknown nucleotides</li>
</ul>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Import the 12 <code class="language-plaintext highlighter-rouge">fq.gz</code> into a <code class="language-plaintext highlighter-rouge">List of Pairs</code> collection named <code class="language-plaintext highlighter-rouge">fastq_raw</code>
      <ul>
        <li>Option 1: from a shared data library (ask your instructor)</li>
        <li>
          <p>Option 2: from Zenodo using the URLs given below</p>

          <p><a href="https://doi.org/10.5281/zenodo.3541678"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.3541678.svg" alt="DOI. " loading="lazy" /></a></p>
        </li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3541678/files/A1_left.fq.gz
https://zenodo.org/record/3541678/files/A1_right.fq.gz
https://zenodo.org/record/3541678/files/A2_left.fq.gz
https://zenodo.org/record/3541678/files/A2_right.fq.gz
https://zenodo.org/record/3541678/files/A3_left.fq.gz
https://zenodo.org/record/3541678/files/A3_right.fq.gz
https://zenodo.org/record/3541678/files/B1_left.fq.gz
https://zenodo.org/record/3541678/files/B1_right.fq.gz
https://zenodo.org/record/3541678/files/B2_left.fq.gz
https://zenodo.org/record/3541678/files/B2_right.fq.gz
https://zenodo.org/record/3541678/files/B3_left.fq.gz
https://zenodo.org/record/3541678/files/B3_right.fq.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>    <p>Click on <strong>Collection</strong> on the top</p>  </li>  <li>    <p>Click on <strong>Collection Type</strong> and select <code class="language-plaintext highlighter-rouge">List of Pairs</code></p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li>    <p>Click on <strong>Build</strong> when available</p>  </li>  <li>    <p>Enter a name for the collection</p>    <ul>      <li>fastq_raw</li>    </ul>  </li>  <li>Click on <strong>Create list</strong> (and wait a bit)</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>Rename the datasets</li>
    <li>
      <p>Check that the datatype</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">datatypes</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Add to each database a tag corresponding to …</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="quality-control">Quality control</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">fastq_raw</code> (collection)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<!-- ## Quality control with **MultiQC** - step 2/2

> ### <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description
>
> 1. **MultiQC** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
>    - In *"Results"*:
>        - <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> *"Insert Results"*
>            - *"Which tool was used generate logs?"*: `FastQC`
>                - In *"FastQC output"*:
>                    - *"Type of FastQC output?"*: `Raw data`
>                    - *"FastQC output"*: `data XX, data XX, and others (flattened)`
>
>    > ### <i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment
>    >
>    > We agree that it's not comfortable. The wrapper of MultiQC must be improved
>    {: .comment}
>
{: .hands_on} -->

<h2 id="read-cleaning-with-trimmomatic">Read cleaning with <strong>Trimmomatic</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Single-end or paired-end <span class="notranslate">reads</span>?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end (as collection)</code></li>
        <li><em>“Select FASTQ dataset collection with R1/R2 pair”</em>: <code class="language-plaintext highlighter-rouge">fastq_raw</code></li>
        <li><em>“Perform initial ILLUMINACLIP step?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Adapter sequences to use”</em>: <code class="language-plaintext highlighter-rouge">TruSeq3 (additional seqs) (paired-ended, for MiSeq and HiSeq)</code></li>
        <li>In <em>“Trimmomatic Operation”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off end of a read, if below a threshold quality (TRAILING)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off start of a read, if below a threshold quality (LEADING)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Sliding window trimming (SLIDINGWINDOW)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Drop <span class="notranslate">reads</span> with average quality lower than a specific level (AVGQUAL)</code>
                  <ul>
                    <li><em>“Minimum length of <span class="notranslate">reads</span> to be kept”</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Drop <span class="notranslate">reads</span> below a specified length (MINLEN)</code>
                  <ul>
                    <li><em>“Minimum length of <span class="notranslate">reads</span> to be kept”</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Output trimmomatic log messages?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Dataset Collection
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Trimmomatic on collection XX: paired</code> -&gt; <code class="language-plaintext highlighter-rouge">fastqc_cleaned</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>You can check the Trimmomatic log files to get the number of read before and after the cleaning</p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Input Read Pairs: 10000
Both Surviving: 8804 (88.04%)
Forward Only Surviving: 491 (4.91%)
Reverse Only Surviving: 456 (4.56%) Dropped: 249 (2.49%)
</code></pre></div>        </div>
      </blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a collection</h3>  <ol>  <li>Click on the collection</li>  <li>Click on the name of the collection at the top</li>  <li>Change the name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="quality-control-after-cleaning">Quality control after cleaning</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">fastqc_cleaned</code> (collection)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="assembly-120-minutes---computing">Assembly (120 minutes - computing)</h1>

<h2 id="assembly-with-trinity">Assembly with <strong>Trinity</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-3"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Trinity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Are you pooling sequence datasets?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Paired or Single-end data?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end collection</code>
              <ul>
                <li><em>“Strand specific data”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Run in silico normalization of <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>In <em>“Additional Options”</em>:
          <ul>
            <li><em>“Use the genome guided mode?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Trinity output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Trinity on data 52, data 51, and others: Assembled Transcripts</code> -&gt; <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="assembly-assessment--cleaning">Assembly assessment / cleaning</h1>

<h2 id="checking-of-the-assembly-statistics">Checking of the assembly statistics</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-4"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Trinity Statistics</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Trinity assembly”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>This step, even with this toy dataset, will take around 2 hours</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="remapping-on-the-raw-transcriptome">Re<span class="notranslate">mapping</span> on the raw transcriptome</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-5"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Align <span class="notranslate">reads</span> and estimate abundance</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Transcripts”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>“Paired or Single-end data?”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><em>“Left/Forward strand <span class="notranslate">reads</span>”</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">left</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Right/Reverse strand <span class="notranslate">reads</span>”</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">right</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Strand specific data”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Abundance estimation method”</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
        <li>In <em>“Additional Options”</em>:
          <ul>
            <li><em>“Trinity assembly?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the 6 <code class="language-plaintext highlighter-rouge">* isoforms counts</code> :(
      <ul>
        <li>Check in the information panel (<strong>i</strong> icon) the lineage of your file (ex: <code class="language-plaintext highlighter-rouge">A1_left.fq.gz</code> … )</li>
        <li>Rename the datasets: <code class="language-plaintext highlighter-rouge">A1_raw</code>, <code class="language-plaintext highlighter-rouge">A2_raw</code>, <code class="language-plaintext highlighter-rouge">A3_raw</code>, <code class="language-plaintext highlighter-rouge">B1_raw</code>, <code class="language-plaintext highlighter-rouge">B2_raw</code>, <code class="language-plaintext highlighter-rouge">B3_raw</code>.</li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge"><span class="notranslate">Mapping</span> rate</code></p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[2019-11-14 15:44:21.500] [jointLog] [info] <span class="notranslate">Mapping</span> rate = 44.4358%
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>At this stage, you can now delete some useless datasets</p>
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Trimmomatic on collection XX: unpaired</code></li>
          <li><code class="language-plaintext highlighter-rouge">Align <span class="notranslate">reads</span> and estimate abundance on *: genes counts</code>
Note that the dataset are just hidden. You can delete them permanently and make some room in the history options (the little wheel icon)</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="merge-the-mapping-tables-and-compute-normalizations">Merge the <span class="notranslate">mapping</span> tables and compute normalizations</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-6"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Build <span class="notranslate">expression</span> matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Abundance estimates”</em>: <code class="language-plaintext highlighter-rouge">A1_raw</code>, <code class="language-plaintext highlighter-rouge">A2_raw</code>, <code class="language-plaintext highlighter-rouge">A3_raw</code>, <code class="language-plaintext highlighter-rouge">B1_raw</code>, <code class="language-plaintext highlighter-rouge">B2_raw</code>, <code class="language-plaintext highlighter-rouge">B3_raw</code></li>
        <li><em>“Abundance estimation method”</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What are the three tables?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">estimated <span class="notranslate">RNA</span>-Seq fragment isoform counts (raw counts)</code>`</li>
      <li><code class="language-plaintext highlighter-rouge">matrix of isoform TPM <span class="notranslate">expression</span> values (not cross-sample normalized)</code></li>
      <li><code class="language-plaintext highlighter-rouge">matrix of TMM-normalized <span class="notranslate">expression</span> values</code></li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="compute-contig-ex90n50-statistic-and-ex90-transcript-count">Compute contig Ex90N50 statistic and Ex90 transcript count</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-7"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Compute contig Ex90N50 statistic and Ex90 transcript count</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“<span class="notranslate">Expression</span> matrix”</em>: <code class="language-plaintext highlighter-rouge">Build <span class="notranslate">expression</span> matrix: matrix of TMM-normalized <span class="notranslate">expression</span> values</code></li>
        <li><em>“Transcripts”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>
    </li>
    <li>Click on the visulization icon on the dataset <code class="language-plaintext highlighter-rouge">Compute contig Ex90N50 statistic and Ex90 transcript count: ExN50 statistics</code>
      <ol>
        <li><strong>Scatterplot - Creates a 2D-scatterplot from tabular datapoints</strong></li>
        <li><em>“X Column”</em>: select the Columns <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Y Column”</em>: select the Columns <code class="language-plaintext highlighter-rouge">2</code></li>
      </ol>
    </li>
  </ol>

</blockquote>

<h3 id="what-we-get">What we get</h3>
<p><img src="../../images/full-de-novo/ExN50_plot_toy_dataset.png" alt="ExN50_plot_toy_dataset. " loading="lazy" /></p>

<h3 id="what-we-should-get-with-a-real-dataset">What we should get with a real dataset</h3>
<p><img src="../../images/full-de-novo/ExN50_plot.png" alt="ExN50_plot. " loading="lazy" />
<a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Contig-Nx-and-ExN50-stats">(source)</a></p>

<h2 id="transcriptome-annotation-completeness">Transcriptome annotation completeness</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-8"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Sequence to analyse”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>“Mode”</em>: <code class="language-plaintext highlighter-rouge">transcriptome</code></li>
        <li><em>“Lineage”</em>: <code class="language-plaintext highlighter-rouge">eukaryota_odb9</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/full-de-novo/rnaseq_samples_quality_check.png" alt="RNASeq samples quality check Graphs. " loading="lazy" /></p>

<h2 id="filter-low-expression-transcripts">Filter low <span class="notranslate">expression</span> transcripts</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-9"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Filter low <span class="notranslate">expression</span> transcripts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Trinity assembly”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>“<span class="notranslate">Expression</span> matrix”</em>: <code class="language-plaintext highlighter-rouge">Build <span class="notranslate">expression</span> matrix: matrix of isoform TPM <span class="notranslate">expression</span> values (not cross-sample normalized)</code></li>
        <li><em>“Minimum <span class="notranslate">expression</span> level required across any sample”</em>: <code class="language-plaintext highlighter-rouge">1.0</code></li>
        <li><em>“Isoform filtering method”</em>: <code class="language-plaintext highlighter-rouge">Keep all isoforms above a minimum percent of dominant <span class="notranslate">expression</span></code>
          <ul>
            <li><em>“Minimum percent of dominant isoform <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge">Retained</code> rate</p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> Retained 2096 / 2102 = 99.71% of total transcripts.
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>
    </li>
    <li><strong>Rename</strong> the output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Filter low <span class="notranslate">expression</span> transcripts on data 42 and data 14: filtered low <span class="notranslate">expression</span> transcripts</code> -&gt; <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="checking-of-the-assembly-statistics-after-cleaning">Checking of the assembly statistics after cleaning</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-10"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Trinity Statistics</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Trinity assembly”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="annotation">Annotation</h1>
<h2 id="generate-gene-to-transcript-map">Generate gene to transcript map</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-11"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Generate gene to transcript map</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Trinity assembly”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="peptide-prediction">Peptide prediction</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-12"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>TransDecoder</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Transcripts”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li>In <em>“Training Options”</em>:
          <ul>
            <li><em>“Select the training method”</em>: <code class="language-plaintext highlighter-rouge">Train with the top longest ORFs</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="similarity-search">Similarity search</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-13"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“What do you want to align?”</em>: <code class="language-plaintext highlighter-rouge">Align amino acid query sequences (blastp)</code></li>
        <li><em>“Input query file in FASTA or FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
        <li><em>“Select a reference database”</em>: <code class="language-plaintext highlighter-rouge">Uniprot Swissprot</code></li>
        <li><em>“Format of output file”</em>: <code class="language-plaintext highlighter-rouge">BLAST Tabular</code></li>
        <li>In <em>“Method to restrict the number of hits?”</em>: <code class="language-plaintext highlighter-rouge">Maximum number of target sequences</code>
          <ul>
            <li><em>“The maximum number of target sequence per query to report alignments for”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Diamond output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Diamond on data XXX</code> -&gt; <code class="language-plaintext highlighter-rouge">Diamond (blastp)</code></li>
      </ul>
    </li>
    <li><strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“What do you want to align?”</em>: <code class="language-plaintext highlighter-rouge">Align <span class="notranslate">DNA</span> query sequences (blastx)</code></li>
        <li><em>“Input query file in FASTA or FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>“Select a reference database”</em>: <code class="language-plaintext highlighter-rouge">Uniprot Swissprot</code></li>
        <li><em>“Format of output file”</em>: <code class="language-plaintext highlighter-rouge">BLAST Tabular</code></li>
        <li>In <em>“Method to restrict the number of hits?”</em>: <code class="language-plaintext highlighter-rouge">Maximum number of target sequences</code>
          <ul>
            <li><em>“The maximum number of target sequence per query to report alignments for”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Diamond output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Diamond on data XXX</code> -&gt; <code class="language-plaintext highlighter-rouge">Diamond (blastx)</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Note that you can both use <strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> or the <strong>NCBI BLAST+ blastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>NCBI BLAST+ blast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="find-signal-peptides">Find signal peptides</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-14"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>SignalP 3.0</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Fasta file of protein sequences”</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="find-transmembrane-domains">Find transmembrane domains</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-15"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>TMHMM 2.0</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“FASTA file of protein sequences”</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="search-again-profile-database">Search again profile database</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-16"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>hmmscan</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Sequence file”</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="transcriptome-annotation-using-trinotate">Transcriptome annotation using <strong>Trinotate</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-17"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Trinotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Transcripts”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>“Peptides”</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
        <li><em>“Genes to transcripts map”</em>: <code class="language-plaintext highlighter-rouge">Generate gene to transcript map on data XXX: Genes to transcripts map</code></li>
        <li><em>“BLASTP: Peptides vs Uniprot.SwissProt”</em>: <code class="language-plaintext highlighter-rouge">Diamond (blastp)</code></li>
        <li><em>“BLASTX: Transcripts vs Uniprot.SwissProt”</em>: <code class="language-plaintext highlighter-rouge">Diamond (blastx)</code></li>
        <li><em>“HMMER hmmscan: Peptides vs PFAM”</em>: <code class="language-plaintext highlighter-rouge">Table of per-domain hits from HMM matches of TransDecoder on data XXX: pep against the profile database</code></li>
        <li><em>“TMHMM on Peptides”</em>: <code class="language-plaintext highlighter-rouge">TMHMM results</code></li>
        <li><em>“SignalP on Peptides”</em>: <code class="language-plaintext highlighter-rouge">SignalP euk results</code></li>
        <li><em>“Let <span class="notranslate">Galaxy</span> downloading the Trinotate Pre-generated Resource SQLite database”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="differential-expression-de-analysis">Differential <span class="notranslate">Expression</span> (DE) Analysis</h1>

<h2 id="remapping-on-the-filtered-transcriptome-using">Re<span class="notranslate">mapping</span> on the filtered transcriptome using</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-18"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Align <span class="notranslate">reads</span> and estimate abundance</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Transcripts”</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>“Paired or Single-end data?”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><em>“Left/Forward strand <span class="notranslate">reads</span>”</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">left</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Right/Reverse strand <span class="notranslate">reads</span>”</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">right</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Strand specific data”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Abundance estimation method”</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
        <li>In <em>“Additional Options”</em>:
          <ul>
            <li><em>“Trinity assembly?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the 6 <code class="language-plaintext highlighter-rouge">* isoforms counts</code> :(
      <ul>
        <li>Check in the information panel (<strong>i</strong> icon) the lineage of your file (ex: <code class="language-plaintext highlighter-rouge">A1_left.fq.gz</code> … )</li>
        <li>Rename the datasets: <code class="language-plaintext highlighter-rouge">A1</code>, <code class="language-plaintext highlighter-rouge">A2</code>, <code class="language-plaintext highlighter-rouge">A3</code>, <code class="language-plaintext highlighter-rouge">B1</code>, <code class="language-plaintext highlighter-rouge">B2</code>, <code class="language-plaintext highlighter-rouge">B3</code>.</li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-6"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge"><span class="notranslate">Mapping</span> rate</code></p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[2019-11-14 15:44:21.500] [jointLog] [info] <span class="notranslate">Mapping</span> rate = 44.4358%
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-7"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>At this stage, you can now delete some useless datasets</p>
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Align <span class="notranslate">reads</span> and estimate abundance on *: genes counts</code>
Note that the dataset are just hidden. You can delete them permanently and make some room in the history options (the little wheel icon)</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="merge-the-mapping-tables-and-compute-a-tmm-normalization">Merge the <span class="notranslate">mapping</span> tables and compute a TMM normalization</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-19"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Build <span class="notranslate">expression</span> matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Abundance estimates”</em>: <code class="language-plaintext highlighter-rouge">A1</code>, <code class="language-plaintext highlighter-rouge">A2</code>, <code class="language-plaintext highlighter-rouge">A3</code>, <code class="language-plaintext highlighter-rouge">B1</code>, <code class="language-plaintext highlighter-rouge">B2</code>, <code class="language-plaintext highlighter-rouge">B3</code></li>
        <li><em>“Abundance estimation method”</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
      </ul>
    </li>
    <li><strong>Describe samples and replicates</strong>  <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Samples”</em>
          <ul>
            <li><em>“1: Samples”</em>:
              <ul>
                <li><em>“Full sample name”</em>: <code class="language-plaintext highlighter-rouge">A1</code></li>
                <li><em>“Condition”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
              </ul>
            </li>
            <li><em>“2: Samples”</em>:
              <ul>
                <li><em>“Full sample name”</em>: <code class="language-plaintext highlighter-rouge">A2</code></li>
                <li><em>“Condition”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
              </ul>
            </li>
            <li>…:</li>
            <li><em>“6: Samples”</em>:
              <ul>
                <li><em>“Full sample name”</em>: <code class="language-plaintext highlighter-rouge">B3</code></li>
                <li><em>“Condition”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="rnaseq-samples-quality-check"><span class="notranslate">RNA</span>Seq samples quality check</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-20"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>
  <ol>
    <li><strong><span class="notranslate">RNA</span>Seq samples quality check</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“<span class="notranslate">Expression</span> matrix”</em>: <code class="language-plaintext highlighter-rouge">Build <span class="notranslate">expression</span> matrix: estimated <span class="notranslate">RNA</span>-Seq fragment isoform counts (raw counts)</code></li>
        <li><em>“Samples description”</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="differential-expression-analysis">Differential <span class="notranslate">expression</span> analysis</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-21"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>
  <ol>
    <li><strong>Differential <span class="notranslate">expression</span> analysis</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“<span class="notranslate">Expression</span> matrix”</em>: <code class="language-plaintext highlighter-rouge">Build <span class="notranslate">expression</span> matrix: estimated <span class="notranslate">RNA</span>-Seq fragment isoform counts (raw counts)</code></li>
        <li><em>“Sample description”</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code> (the last one)</li>
        <li><em>“Differential analysis method”</em>: <code class="language-plaintext highlighter-rouge">DESeq2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="extract-and-cluster-differentially-expressed-transcripts">Extract and cluster differentially expressed transcripts</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-22"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Extract and cluster differentially expressed transcripts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“Additional Options”</em>:
          <ul>
            <li><em>“<span class="notranslate">Expression</span> matrix”</em>: <code class="language-plaintext highlighter-rouge">Build <span class="notranslate">expression</span> matrix: estimated <span class="notranslate">RNA</span>-Seq fragment isoform counts (raw counts)</code></li>
            <li><em>“Sample description”</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code></li>
            <li><em>“Differential <span class="notranslate">expression</span> results”</em>: <code class="language-plaintext highlighter-rouge">Differential <span class="notranslate">expression</span> results on data XXX and data XXX</code></li>
            <li><em>“p-value cutoff for FDR”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li><em>“Run GO enrichment analysis”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-8"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p><em>“p-value cutoff for FDR”</em>: <code class="language-plaintext highlighter-rouge">1</code>
Don’t do this at home! It’s because we have a Toy Dataset. The cutoff should be around <code class="language-plaintext highlighter-rouge">0.001</code></p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="partition-genes-into-expression-clusters">Partition genes into <span class="notranslate">expression</span> clusters</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-23"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Partition genes into <span class="notranslate">expression</span> clusters</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“RData file”</em>: <code class="language-plaintext highlighter-rouge">Extract and cluster differentially expressed transcripts: RData file</code></li>
        <li><em>“Method for partitioning genes into clusters”</em>: <code class="language-plaintext highlighter-rouge">Cut tree based on x percent of max(height) of tree</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Sum up the tutorial and the key takeaways here. We encourage adding an overview image of the
pipeline used.</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>The take-home messages</p>
</li>
                        
                        <li><p>They will appear at the end of the tutorial</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                

                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=De novo transcriptome assembly, annotation, and differential expression analysis (Transcriptomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Anthony Bretaudeau, Gildas Le Corguillé, Erwan Corre, Xi Liu, 2021 <b>De novo transcriptome assembly, annotation, and differential expression analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-full-de-novo,
author = "Anthony Bretaudeau and Gildas Le Corguillé and Erwan Corre and Xi Liu",
title = "De novo transcriptome assembly, annotation, and differential expression analysis (Galaxy Training Materials)",
year = "2021",
month = "03",
day = "12"
url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
