<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>De novo transcriptome assembly, annotation, and differential expression analysis</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703895990' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703895990' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="De novo transcriptome assembly, annotation, and differential expression analysis">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-18 09:34:29 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Gildas Le CorguillÃ©">
<meta name="DC.creator" content="Erwan Corre">
<meta name="DC.creator" content="Xi Liu">

        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: De novo transcriptome assembly, annotation, and different...">
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis.">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-18 09:34:29 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "De novo transcriptome assembly, annotation, and differential expression analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Under development",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'De novo transcriptome assembly, annotation, and differential expression analysis' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html",
  "timeRequired": "PT3H",
  "teaches": "The learning objectives are the goals of the tutorial\n - They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course\n - They are single sentences describing what a learner should be able to do once they have completed the tutorial\n - You can use Bloom's Taxonomy to write effective learning objectives",
  "keywords": "transcriptomics",
  "description": "The questions this  addresses are:\n - Which biological questions are addressed by the tutorial?\n - Which bioinformatics techniques are important to know for this type of data?\n\n\nThe objectives are:\n - The learning objectives are the goals of the tutorial\n - They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course\n - They are single sentences describing what a learner should be able to do once they have completed the tutorial\n - You can use Bloom's Taxonomy to write effective learning objectives\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "name": "Anthony Bretaudeau",
      "image": "https://avatars.githubusercontent.com/abretaud",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0914-2470",
      "orcid": "https://orcid.org/0000-0003-0914-2470"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/lecorguille/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/lecorguille/",
      "name": "Gildas Le CorguillÃ©",
      "image": "https://avatars.githubusercontent.com/lecorguille",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "name": "Erwan Corre",
      "image": "https://avatars.githubusercontent.com/r1corre",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6354-2278",
      "orcid": "https://orcid.org/0000-0001-6354-2278"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/xiliu/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/xiliu/",
      "name": "Xi Liu",
      "image": "https://avatars.githubusercontent.com/xiliu",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703895990'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/full-de-novo/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>De novo transcriptome assembly, annotation, and differential expression analysis</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/lecorguille/" class="contributor-badge contributor-lecorguille"><img src="https://avatars.githubusercontent.com/lecorguille?s=36" alt="Avatar" width="36" height="36">Gildas Le CorguillÃ©</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/r1corre?s=36" alt="Avatar" width="36" height="36">Erwan Corre</a><a href="/training-material/hall-of-fame/xiliu/" class="contributor-badge contributor-xiliu"><img src="https://avatars.githubusercontent.com/xiliu?s=36" alt="Avatar" width="36" height="36">Xi Liu</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which biological questions are addressed by the tutorial?</p>
</li>
        
        <li><p>Which bioinformatics techniques are important to know for this type of data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>The learning objectives are the goals of the tutorial</p>
</li>
        
        <li><p>They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course</p>
</li>
        
        <li><p>They are single sentences describing what a learner should be able to do once they have completed the tutorial</p>
</li>
        
        <li><p>You can use Bloomâs Taxonomy to write effective learning objectives</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3541678">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/transcriptomics/tutorials/full-de-novo/workflows/" title="De novo transcriptome assembly, annotation, and differential expression analysis workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/full-de-novo/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">â</abbr> â­ï¸
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">â</abbr> â­ï¸
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 18, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00290">https://gxy.io/GTN:T00290</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p>As a result of the development of novel sequencing technologies, the years between 2008 and 2012 saw a large drop in the cost of sequencing. Per megabase and genome, the cost dropped to 1/100,000th and 1/10,000th of the price, respectively. Prior to this, only transcriptomes of organisms that were of broad interest and utility to scientific research were sequenced; however, these developed in 2010s high-throughput sequencing (also called next-generation sequencing) technologies are both cost- and labor- effective, and the range of organisms studied via these methods is expanding.</p>

<p>Examining non-model organisms can provide novel insights into the mechanisms underlying the âdiversity of fascinating morphological innovationsâ that have enabled the abundance of life on planet Earth. In animals and plants, the âinnovationsâ that cannot be examined in common model organisms include mimicry, mutualism, parasitism, and asexual reproduction. De novo transcriptome assembly is often the preferred method to studying non-model organisms, since it is cheaper and easier than building a genome, and reference-based methods are not possible without an existing genome. The transcriptomes of these organisms can thus reveal novel proteins and their isoforms that are implicated in such unique biological phenomena.</p>

<p><a href="https://en.wikipedia.org/wiki/De_novo_transcriptome_assembly">(source)</a></p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#read-cleaning-20-minutes" id="markdown-toc-read-cleaning-20-minutes">Read cleaning (20 minutes)</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#read-cleaning-with-trimmomatic" id="markdown-toc-read-cleaning-with-trimmomatic">Read cleaning with <strong>Trimmomatic</strong></a></li>
      <li><a href="#quality-control-after-cleaning" id="markdown-toc-quality-control-after-cleaning">Quality control after cleaning</a></li>
    </ol>
  </li>
  <li><a href="#assembly-120-minutes---computing" id="markdown-toc-assembly-120-minutes---computing">Assembly (120 minutes - computing)</a>    <ol>
      <li><a href="#assembly-with-trinity" id="markdown-toc-assembly-with-trinity">Assembly with <strong>Trinity</strong></a></li>
    </ol>
  </li>
  <li><a href="#assembly-assessment--cleaning" id="markdown-toc-assembly-assessment--cleaning">Assembly assessment / cleaning</a>    <ol>
      <li><a href="#checking-of-the-assembly-statistics" id="markdown-toc-checking-of-the-assembly-statistics">Checking of the assembly statistics</a></li>
      <li><a href="#remapping-on-the-raw-transcriptome" id="markdown-toc-remapping-on-the-raw-transcriptome">Remapping on the raw transcriptome</a></li>
      <li><a href="#merge-the-mapping-tables-and-compute-normalizations" id="markdown-toc-merge-the-mapping-tables-and-compute-normalizations">Merge the mapping tables and compute normalizations</a></li>
      <li><a href="#compute-contig-ex90n50-statistic-and-ex90-transcript-count" id="markdown-toc-compute-contig-ex90n50-statistic-and-ex90-transcript-count">Compute contig Ex90N50 statistic and Ex90 transcript count</a></li>
      <li><a href="#transcriptome-annotation-completeness" id="markdown-toc-transcriptome-annotation-completeness">Transcriptome annotation completeness</a></li>
      <li><a href="#filter-low-expression-transcripts" id="markdown-toc-filter-low-expression-transcripts">Filter low expression transcripts</a></li>
      <li><a href="#checking-of-the-assembly-statistics-after-cleaning" id="markdown-toc-checking-of-the-assembly-statistics-after-cleaning">Checking of the assembly statistics after cleaning</a></li>
    </ol>
  </li>
  <li><a href="#annotation" id="markdown-toc-annotation">Annotation</a>    <ol>
      <li><a href="#generate-gene-to-transcript-map" id="markdown-toc-generate-gene-to-transcript-map">Generate gene to transcript map</a></li>
      <li><a href="#peptide-prediction" id="markdown-toc-peptide-prediction">Peptide prediction</a></li>
      <li><a href="#similarity-search" id="markdown-toc-similarity-search">Similarity search</a></li>
      <li><a href="#find-signal-peptides" id="markdown-toc-find-signal-peptides">Find signal peptides</a></li>
      <li><a href="#find-transmembrane-domains" id="markdown-toc-find-transmembrane-domains">Find transmembrane domains</a></li>
      <li><a href="#search-again-profile-database" id="markdown-toc-search-again-profile-database">Search again profile database</a></li>
      <li><a href="#transcriptome-annotation-using-trinotate" id="markdown-toc-transcriptome-annotation-using-trinotate">Transcriptome annotation using <strong>Trinotate</strong></a></li>
    </ol>
  </li>
  <li><a href="#differential-expression-de-analysis" id="markdown-toc-differential-expression-de-analysis">Differential Expression (DE) Analysis</a>    <ol>
      <li><a href="#remapping-on-the-filtered-transcriptome-using" id="markdown-toc-remapping-on-the-filtered-transcriptome-using">Remapping on the filtered transcriptome using</a></li>
      <li><a href="#merge-the-mapping-tables-and-compute-a-tmm-normalization" id="markdown-toc-merge-the-mapping-tables-and-compute-a-tmm-normalization">Merge the mapping tables and compute a TMM normalization</a></li>
      <li><a href="#rnaseq-samples-quality-check" id="markdown-toc-rnaseq-samples-quality-check">RNASeq samples quality check</a></li>
      <li><a href="#differential-expression-analysis" id="markdown-toc-differential-expression-analysis">Differential expression analysis</a></li>
      <li><a href="#extract-and-cluster-differentially-expressed-transcripts" id="markdown-toc-extract-and-cluster-differentially-expressed-transcripts">Extract and cluster differentially expressed transcripts</a></li>
      <li><a href="#partition-genes-into-expression-clusters" id="markdown-toc-partition-genes-into-expression-clusters">Partition genes into expression clusters</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="read-cleaning-20-minutes">Read cleaning (20 minutes)</h1>

<p>Known sequencing biases:</p>
<ul>
  <li>Unknown nucleotides (Ns)</li>
  <li>Bad quality nucleotides</li>
  <li>Hexamers biases (Illumina. Now corrected ?)</li>
</ul>

<p>Why do we need to correct those?</p>
<ul>
  <li>To remove a lot of sequencing errors (detrimental to the vast majority of assemblers)</li>
  <li>Because most de-bruijn graph based assemblers canât handle unknown nucleotides</li>
</ul>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Import the 12 <code class="language-plaintext highlighter-rouge">fq.gz</code> into a <code class="language-plaintext highlighter-rouge">List of Pairs</code> collection named <code class="language-plaintext highlighter-rouge">fastq_raw</code>
      <ul>
        <li>Option 1: from a shared data library (ask your instructor)</li>
        <li>
          <p>Option 2: from Zenodo using the URLs given below</p>

          <p><a href="https://doi.org/10.5281/zenodo.3541678"><a href="https://zenodo.org/badge/DOI/10.5281/zenodo.3541678.svg" rel="noopener noreferrer"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.3541678.svg" alt="DOI. " loading="lazy" /></a></a></p>
        </li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3541678/files/A1_left.fq.gz
https://zenodo.org/record/3541678/files/A1_right.fq.gz
https://zenodo.org/record/3541678/files/A2_left.fq.gz
https://zenodo.org/record/3541678/files/A2_right.fq.gz
https://zenodo.org/record/3541678/files/A3_left.fq.gz
https://zenodo.org/record/3541678/files/A3_right.fq.gz
https://zenodo.org/record/3541678/files/B1_left.fq.gz
https://zenodo.org/record/3541678/files/B1_right.fq.gz
https://zenodo.org/record/3541678/files/B2_left.fq.gz
https://zenodo.org/record/3541678/files/B2_right.fq.gz
https://zenodo.org/record/3541678/files/B3_left.fq.gz
https://zenodo.org/record/3541678/files/B3_right.fq.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>     <p>Click on <strong>Collection</strong> on the top</p>   </li>   <li>     <p>Click on <strong>Collection Type</strong> and select <code class="language-plaintext highlighter-rouge">List of Pairs</code></p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li>     <p>Click on <strong>Build</strong> when available</p>   </li>   <li>     <p>Enter a name for the collection</p>     <ul>       <li>fastq_raw</li>     </ul>   </li>   <li>Click on <strong>Create list</strong> (and wait a bit)</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material â&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>âSelect historyâ</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Rename the datasets</li>
    <li>
      <p>Check that the datatype</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing the datatype<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the lower part <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong>, select <code class="language-plaintext highlighter-rouge">datatypes</code>     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Add to each database a tag corresponding to â¦</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Adding a tag<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</li>     </ul>   </li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h2 id="quality-control">Quality control</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âShort read data from your current historyâ</em>: <code class="language-plaintext highlighter-rouge">fastq_raw</code> (collection)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<!-- ## Quality control with **MultiQC** - step 2/2

> <div class="box-title hands-on-title" id="hands-on-task-description-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>
>
> 1. **MultiQC** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
>    - In *"Results"*:
>        - <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> *"Insert Results"*
>            - *"Which tool was used generate logs?"*: `FastQC`
>                - In *"FastQC output"*:
>                    - *"Type of FastQC output?"*: `Raw data`
>                    - *"FastQC output"*: `data XX, data XX, and others (flattened)`
>
>    > <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
>    >
>    > We agree that it's not comfortable. The wrapper of MultiQC must be improved
>    {: .comment}
>
{: .hands_on} -->

<h2 id="read-cleaning-with-trimmomatic">Read cleaning with <strong>Trimmomatic</strong></h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âSingle-end or paired-end reads?â</em>: <code class="language-plaintext highlighter-rouge">Paired-end (as collection)</code></li>
        <li><em>âSelect FASTQ dataset collection with R1/R2 pairâ</em>: <code class="language-plaintext highlighter-rouge">fastq_raw</code></li>
        <li><em>âPerform initial ILLUMINACLIP step?â</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>âAdapter sequences to useâ</em>: <code class="language-plaintext highlighter-rouge">TruSeq3 (additional seqs) (paired-ended, for MiSeq and HiSeq)</code></li>
        <li>In <em>âTrimmomatic Operationâ</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Trimmomatic Operationâ</em>
              <ul>
                <li><em>âSelect Trimmomatic operation to performâ</em>: <code class="language-plaintext highlighter-rouge">Cut bases off end of a read, if below a threshold quality (TRAILING)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Trimmomatic Operationâ</em>
              <ul>
                <li><em>âSelect Trimmomatic operation to performâ</em>: <code class="language-plaintext highlighter-rouge">Cut bases off start of a read, if below a threshold quality (LEADING)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Trimmomatic Operationâ</em>
              <ul>
                <li><em>âSelect Trimmomatic operation to performâ</em>: <code class="language-plaintext highlighter-rouge">Sliding window trimming (SLIDINGWINDOW)</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Trimmomatic Operationâ</em>
              <ul>
                <li><em>âSelect Trimmomatic operation to performâ</em>: <code class="language-plaintext highlighter-rouge">Drop reads with average quality lower than a specific level (AVGQUAL)</code>
                  <ul>
                    <li><em>âMinimum length of reads to be keptâ</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Trimmomatic Operationâ</em>
              <ul>
                <li><em>âSelect Trimmomatic operation to performâ</em>: <code class="language-plaintext highlighter-rouge">Drop reads below a specified length (MINLEN)</code>
                  <ul>
                    <li><em>âMinimum length of reads to be keptâ</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>âOutput trimmomatic log messages?â</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Dataset Collection
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Trimmomatic on collection XX: paired</code> -&gt; <code class="language-plaintext highlighter-rouge">fastqc_cleaned</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>You can check the Trimmomatic log files to get the number of read before and after the cleaning</p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Input Read Pairs: 10000
Both Surviving: 8804 (88.04%)
Forward Only Surviving: 491 (4.91%)
Reverse Only Surviving: 456 (4.56%) Dropped: 249 (2.49%)
</code></pre></div>        </div>
      </blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection</li>   <li>Click on the name of the collection at the top</li>   <li>Change the name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h2 id="quality-control-after-cleaning">Quality control after cleaning</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-3"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âShort read data from your current historyâ</em>: <code class="language-plaintext highlighter-rouge">fastqc_cleaned</code> (collection)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="assembly-120-minutes---computing">Assembly (120 minutes - computing)</h1>

<h2 id="assembly-with-trinity">Assembly with <strong>Trinity</strong></h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-4"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Trinity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âAre you pooling sequence datasets?â</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>âPaired or Single-end data?â</em>: <code class="language-plaintext highlighter-rouge">Paired-end collection</code>
              <ul>
                <li><em>âStrand specific dataâ</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>âRun in silico normalization of readsâ</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>In <em>âAdditional Optionsâ</em>:
          <ul>
            <li><em>âUse the genome guided mode?â</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Trinity output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Trinity on data 52, data 51, and others: Assembled Transcripts</code> -&gt; <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="assembly-assessment--cleaning">Assembly assessment / cleaning</h1>

<h2 id="checking-of-the-assembly-statistics">Checking of the assembly statistics</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-5"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Trinity Statistics</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTrinity assemblyâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>This step, even with this toy dataset, will take around 2 hours</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="remapping-on-the-raw-transcriptome">Remapping on the raw transcriptome</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-6"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Align reads and estimate abundance</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTranscriptsâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>âPaired or Single-end data?â</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><em>âLeft/Forward strand readsâ</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">left</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>âRight/Reverse strand readsâ</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">right</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>âStrand specific dataâ</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>âAbundance estimation methodâ</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
        <li>In <em>âAdditional Optionsâ</em>:
          <ul>
            <li><em>âTrinity assembly?â</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the 6 <code class="language-plaintext highlighter-rouge">* isoforms counts</code> :(
      <ul>
        <li>Check in the information panel (<strong>i</strong> icon) the lineage of your file (ex: <code class="language-plaintext highlighter-rouge">A1_left.fq.gz</code> â¦ )</li>
        <li>Rename the datasets: <code class="language-plaintext highlighter-rouge">A1_raw</code>, <code class="language-plaintext highlighter-rouge">A2_raw</code>, <code class="language-plaintext highlighter-rouge">A3_raw</code>, <code class="language-plaintext highlighter-rouge">B1_raw</code>, <code class="language-plaintext highlighter-rouge">B2_raw</code>, <code class="language-plaintext highlighter-rouge">B3_raw</code>.</li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge">Mapping rate</code></p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[2019-11-14 15:44:21.500] [jointLog] [info] Mapping rate = 44.4358%
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>At this stage, you can now delete some useless datasets</p>
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Trimmomatic on collection XX: unpaired</code></li>
          <li><code class="language-plaintext highlighter-rouge">Align reads and estimate abundance on *: genes counts</code>
Note that the dataset are just hidden. You can delete them permanently and make some room in the history options (the little wheel icon)</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="merge-the-mapping-tables-and-compute-normalizations">Merge the mapping tables and compute normalizations</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-7"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Build expression matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âAbundance estimatesâ</em>: <code class="language-plaintext highlighter-rouge">A1_raw</code>, <code class="language-plaintext highlighter-rouge">A2_raw</code>, <code class="language-plaintext highlighter-rouge">A3_raw</code>, <code class="language-plaintext highlighter-rouge">B1_raw</code>, <code class="language-plaintext highlighter-rouge">B2_raw</code>, <code class="language-plaintext highlighter-rouge">B3_raw</code></li>
        <li><em>âAbundance estimation methodâ</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What are the three tables?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">estimated RNA-Seq fragment isoform counts (raw counts)</code>`</li>
      <li><code class="language-plaintext highlighter-rouge">matrix of isoform TPM expression values (not cross-sample normalized)</code></li>
      <li><code class="language-plaintext highlighter-rouge">matrix of TMM-normalized expression values</code></li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="compute-contig-ex90n50-statistic-and-ex90-transcript-count">Compute contig Ex90N50 statistic and Ex90 transcript count</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-8"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Compute contig Ex90N50 statistic and Ex90 transcript count</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âExpression matrixâ</em>: <code class="language-plaintext highlighter-rouge">Build expression matrix: matrix of TMM-normalized expression values</code></li>
        <li><em>âTranscriptsâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
      </ul>
    </li>
    <li>Click on the visulization icon on the dataset <code class="language-plaintext highlighter-rouge">Compute contig Ex90N50 statistic and Ex90 transcript count: ExN50 statistics</code>
      <ol>
        <li><strong>Scatterplot - Creates a 2D-scatterplot from tabular datapoints</strong></li>
        <li><em>âX Columnâ</em>: select the Columns <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>âY Columnâ</em>: select the Columns <code class="language-plaintext highlighter-rouge">2</code></li>
      </ol>
    </li>
  </ol>

</blockquote>

<h3 id="what-we-get">What we get</h3>
<p><a href="../../images/full-de-novo/ExN50_plot_toy_dataset.png" rel="noopener noreferrer"><img src="../../images/full-de-novo/ExN50_plot_toy_dataset.png" alt="ExN50_plot_toy_dataset. " width="482" height="453" loading="lazy" /></a></p>

<h3 id="what-we-should-get-with-a-real-dataset">What we should get with a real dataset</h3>
<p><a href="../../images/full-de-novo/ExN50_plot.png" rel="noopener noreferrer"><img src="../../images/full-de-novo/ExN50_plot.png" alt="ExN50_plot. " width="500" height="423" loading="lazy" /></a>
<a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki/Transcriptome-Contig-Nx-and-ExN50-stats">(source)</a></p>

<h2 id="transcriptome-annotation-completeness">Transcriptome annotation completeness</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-9"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âSequence to analyseâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>âModeâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome</code></li>
        <li><em>âLineageâ</em>: <code class="language-plaintext highlighter-rouge">eukaryota_odb9</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><a href="../../images/full-de-novo/rnaseq_samples_quality_check.png" rel="noopener noreferrer"><img src="../../images/full-de-novo/rnaseq_samples_quality_check.png" alt="RNASeq samples quality check Graphs. " width="1000" height="1000" loading="lazy" /></a></p>

<h2 id="filter-low-expression-transcripts">Filter low expression transcripts</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-10"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Filter low expression transcripts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTrinity assemblyâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_raw.fasta</code></li>
        <li><em>âExpression matrixâ</em>: <code class="language-plaintext highlighter-rouge">Build expression matrix: matrix of isoform TPM expression values (not cross-sample normalized)</code></li>
        <li><em>âMinimum expression level required across any sampleâ</em>: <code class="language-plaintext highlighter-rouge">1.0</code></li>
        <li><em>âIsoform filtering methodâ</em>: <code class="language-plaintext highlighter-rouge">Keep all isoforms above a minimum percent of dominant expression</code>
          <ul>
            <li><em>âMinimum percent of dominant isoform expressionâ</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-5"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge">Retained</code> rate</p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> Retained 2096 / 2102 = 99.71% of total transcripts.
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>
    </li>
    <li><strong>Rename</strong> the output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Filter low expression transcripts on data 42 and data 14: filtered low expression transcripts</code> -&gt; <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="checking-of-the-assembly-statistics-after-cleaning">Checking of the assembly statistics after cleaning</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-11"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Trinity Statistics</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTrinity assemblyâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="annotation">Annotation</h1>
<h2 id="generate-gene-to-transcript-map">Generate gene to transcript map</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-12"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Generate gene to transcript map</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTrinity assemblyâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="peptide-prediction">Peptide prediction</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-13"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>TransDecoder</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTranscriptsâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li>In <em>âTraining Optionsâ</em>:
          <ul>
            <li><em>âSelect the training methodâ</em>: <code class="language-plaintext highlighter-rouge">Train with the top longest ORFs</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="similarity-search">Similarity search</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-14"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âWhat do you want to align?â</em>: <code class="language-plaintext highlighter-rouge">Align amino acid query sequences (blastp)</code></li>
        <li><em>âInput query file in FASTA or FASTQ formatâ</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
        <li><em>âSelect a reference databaseâ</em>: <code class="language-plaintext highlighter-rouge">Uniprot Swissprot</code></li>
        <li><em>âFormat of output fileâ</em>: <code class="language-plaintext highlighter-rouge">BLAST Tabular</code></li>
        <li>In <em>âMethod to restrict the number of hits?â</em>: <code class="language-plaintext highlighter-rouge">Maximum number of target sequences</code>
          <ul>
            <li><em>âThe maximum number of target sequence per query to report alignments forâ</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Diamond output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Diamond on data XXX</code> -&gt; <code class="language-plaintext highlighter-rouge">Diamond (blastp)</code></li>
      </ul>
    </li>
    <li><strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âWhat do you want to align?â</em>: <code class="language-plaintext highlighter-rouge">Align DNA query sequences (blastx)</code></li>
        <li><em>âInput query file in FASTA or FASTQ formatâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>âSelect a reference databaseâ</em>: <code class="language-plaintext highlighter-rouge">Uniprot Swissprot</code></li>
        <li><em>âFormat of output fileâ</em>: <code class="language-plaintext highlighter-rouge">BLAST Tabular</code></li>
        <li>In <em>âMethod to restrict the number of hits?â</em>: <code class="language-plaintext highlighter-rouge">Maximum number of target sequences</code>
          <ul>
            <li><em>âThe maximum number of target sequence per query to report alignments forâ</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the Diamond output
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Diamond on data XXX</code> -&gt; <code class="language-plaintext highlighter-rouge">Diamond (blastx)</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-6"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>Note that you can both use <strong>Diamond</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> or the <strong>NCBI BLAST+ blastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>NCBI BLAST+ blast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="find-signal-peptides">Find signal peptides</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-15"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>SignalP 3.0</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âFasta file of protein sequencesâ</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="find-transmembrane-domains">Find transmembrane domains</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-16"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>TMHMM 2.0</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âFASTA file of protein sequencesâ</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="search-again-profile-database">Search again profile database</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-17"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>hmmscan</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âSequence fileâ</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="transcriptome-annotation-using-trinotate">Transcriptome annotation using <strong>Trinotate</strong></h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-18"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Trinotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTranscriptsâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>âPeptidesâ</em>: <code class="language-plaintext highlighter-rouge">TransDecoder on data XXX: pep</code></li>
        <li><em>âGenes to transcripts mapâ</em>: <code class="language-plaintext highlighter-rouge">Generate gene to transcript map on data XXX: Genes to transcripts map</code></li>
        <li><em>âBLASTP: Peptides vs Uniprot.SwissProtâ</em>: <code class="language-plaintext highlighter-rouge">Diamond (blastp)</code></li>
        <li><em>âBLASTX: Transcripts vs Uniprot.SwissProtâ</em>: <code class="language-plaintext highlighter-rouge">Diamond (blastx)</code></li>
        <li><em>âHMMER hmmscan: Peptides vs PFAMâ</em>: <code class="language-plaintext highlighter-rouge">Table of per-domain hits from HMM matches of TransDecoder on data XXX: pep against the profile database</code></li>
        <li><em>âTMHMM on Peptidesâ</em>: <code class="language-plaintext highlighter-rouge">TMHMM results</code></li>
        <li><em>âSignalP on Peptidesâ</em>: <code class="language-plaintext highlighter-rouge">SignalP euk results</code></li>
        <li><em>âLet Galaxy downloading the Trinotate Pre-generated Resource SQLite databaseâ</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="differential-expression-de-analysis">Differential Expression (DE) Analysis</h1>

<h2 id="remapping-on-the-filtered-transcriptome-using">Remapping on the filtered transcriptome using</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-19"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Align reads and estimate abundance</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âTranscriptsâ</em>: <code class="language-plaintext highlighter-rouge">transcriptome_filtered.fasta</code></li>
        <li><em>âPaired or Single-end data?â</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><em>âLeft/Forward strand readsâ</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">left</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>âRight/Reverse strand readsâ</em> -&gt; <code class="language-plaintext highlighter-rouge">Multiple datasets</code>
              <ul>
                <li>Click on the <em>Folder</em> button at the right
                  <ul>
                    <li><em>Type to Search</em>: <code class="language-plaintext highlighter-rouge">right</code></li>
                    <li>Select the 6 <code class="language-plaintext highlighter-rouge">Trimmomatic on ..._left.fq.gz</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>âStrand specific dataâ</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>âAbundance estimation methodâ</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
        <li>In <em>âAdditional Optionsâ</em>:
          <ul>
            <li><em>âTrinity assembly?â</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> the 6 <code class="language-plaintext highlighter-rouge">* isoforms counts</code> :(
      <ul>
        <li>Check in the information panel (<strong>i</strong> icon) the lineage of your file (ex: <code class="language-plaintext highlighter-rouge">A1_left.fq.gz</code> â¦ )</li>
        <li>Rename the datasets: <code class="language-plaintext highlighter-rouge">A1</code>, <code class="language-plaintext highlighter-rouge">A2</code>, <code class="language-plaintext highlighter-rouge">A3</code>, <code class="language-plaintext highlighter-rouge">B1</code>, <code class="language-plaintext highlighter-rouge">B2</code>, <code class="language-plaintext highlighter-rouge">B3</code>.</li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-7"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>If you check at the Standard Error messages of your outputs. You can get the <code class="language-plaintext highlighter-rouge">Mapping rate</code></p>
        <ol>
          <li>Click on one dataset</li>
          <li>Click on the little <strong>i</strong> icon</li>
          <li>Click on <em>Tool Standard Error:	stderr</em>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[2019-11-14 15:44:21.500] [jointLog] [info] Mapping rate = 44.4358%
</code></pre></div>            </div>
          </li>
        </ol>
      </blockquote>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-8"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>At this stage, you can now delete some useless datasets</p>
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Align reads and estimate abundance on *: genes counts</code>
Note that the dataset are just hidden. You can delete them permanently and make some room in the history options (the little wheel icon)</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="merge-the-mapping-tables-and-compute-a-tmm-normalization">Merge the mapping tables and compute a TMM normalization</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-20"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Build expression matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âAbundance estimatesâ</em>: <code class="language-plaintext highlighter-rouge">A1</code>, <code class="language-plaintext highlighter-rouge">A2</code>, <code class="language-plaintext highlighter-rouge">A3</code>, <code class="language-plaintext highlighter-rouge">B1</code>, <code class="language-plaintext highlighter-rouge">B2</code>, <code class="language-plaintext highlighter-rouge">B3</code></li>
        <li><em>âAbundance estimation methodâ</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
      </ul>
    </li>
    <li><strong>Describe samples and replicates</strong>  <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âSamplesâ</em>
          <ul>
            <li><em>â1: Samplesâ</em>:
              <ul>
                <li><em>âFull sample nameâ</em>: <code class="language-plaintext highlighter-rouge">A1</code></li>
                <li><em>âConditionâ</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
              </ul>
            </li>
            <li><em>â2: Samplesâ</em>:
              <ul>
                <li><em>âFull sample nameâ</em>: <code class="language-plaintext highlighter-rouge">A2</code></li>
                <li><em>âConditionâ</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
              </ul>
            </li>
            <li>â¦:</li>
            <li><em>â6: Samplesâ</em>:
              <ul>
                <li><em>âFull sample nameâ</em>: <code class="language-plaintext highlighter-rouge">B3</code></li>
                <li><em>âConditionâ</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="rnaseq-samples-quality-check">RNASeq samples quality check</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-21"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>
  <ol>
    <li><strong>RNASeq samples quality check</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âExpression matrixâ</em>: <code class="language-plaintext highlighter-rouge">Build expression matrix: estimated RNA-Seq fragment isoform counts (raw counts)</code></li>
        <li><em>âSamples descriptionâ</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="differential-expression-analysis">Differential expression analysis</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-22"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>
  <ol>
    <li><strong>Differential expression analysis</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âExpression matrixâ</em>: <code class="language-plaintext highlighter-rouge">Build expression matrix: estimated RNA-Seq fragment isoform counts (raw counts)</code></li>
        <li><em>âSample descriptionâ</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code> (the last one)</li>
        <li><em>âDifferential analysis methodâ</em>: <code class="language-plaintext highlighter-rouge">DESeq2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="extract-and-cluster-differentially-expressed-transcripts">Extract and cluster differentially expressed transcripts</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-23"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Extract and cluster differentially expressed transcripts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>âAdditional Optionsâ</em>:
          <ul>
            <li><em>âExpression matrixâ</em>: <code class="language-plaintext highlighter-rouge">Build expression matrix: estimated RNA-Seq fragment isoform counts (raw counts)</code></li>
            <li><em>âSample descriptionâ</em>: <code class="language-plaintext highlighter-rouge">Describe samples</code></li>
            <li><em>âDifferential expression resultsâ</em>: <code class="language-plaintext highlighter-rouge">Differential expression results on data XXX and data XXX</code></li>
            <li><em>âp-value cutoff for FDRâ</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li><em>âRun GO enrichment analysisâ</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-9"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p><em>âp-value cutoff for FDRâ</em>: <code class="language-plaintext highlighter-rouge">1</code>
Donât do this at home! Itâs because we have a Toy Dataset. The cutoff should be around <code class="language-plaintext highlighter-rouge">0.001</code></p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="partition-genes-into-expression-clusters">Partition genes into expression clusters</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description-24"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><strong>Partition genes into expression clusters</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>âRData fileâ</em>: <code class="language-plaintext highlighter-rouge">Extract and cluster differentially expressed transcripts: RData file</code></li>
        <li><em>âMethod for partitioning genes into clustersâ</em>: <code class="language-plaintext highlighter-rouge">Cut tree based on x percent of max(height) of tree</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Sum up the tutorial and the key takeaways here. We encourage adding an overview image of the
pipeline used.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>The take-home messages</p>
</li>
                        
                        <li><p>They will appear at the end of the tutorial</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=De novo transcriptome assembly, annotation, and differential expression analysis (Transcriptomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Anthony Bretaudeau, Gildas Le CorguillÃ©, Erwan Corre, Xi Liu,  <b>De novo transcriptome assembly, annotation, and differential expression analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{transcriptomics-full-de-novo,
author = "Anthony Bretaudeau and Gildas Le CorguillÃ© and Erwan Corre and Xi Liu",
	title = "De novo transcriptome assembly, annotation, and differential expression analysis (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine FÃ¶ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and BjÃ¶rn GrÃ¼ning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/transcriptomics/tutorials/full-de-novo/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: fastqc
  owner: devteam
  revisions: f2e8552cf1d0
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: describe_samples
  owner: iuc
  revisions: '0693ccd171c4'
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: e4574a4d09d8
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: transdecoder
  owner: iuc
  revisions: 0db979fead3a
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity
  owner: iuc
  revisions: c9cfec002f71
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_abundance_estimates_to_matrix
  owner: iuc
  revisions: d6a479fd4281
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_align_and_estimate_abundance
  owner: iuc
  revisions: 478f36effca1
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_analyze_diff_expr
  owner: iuc
  revisions: fab23c3b5258
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_contig_exn50_statistic
  owner: iuc
  revisions: 34a018cbee9c
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_define_clusters_by_cutting_tree
  owner: iuc
  revisions: 33406f5a445c
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_filter_low_expr_transcripts
  owner: iuc
  revisions: e406ac71165d
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_run_de_analysis
  owner: iuc
  revisions: 1d689e4e1caf
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinity_samples_qccheck
  owner: iuc
  revisions: 9871104ea630
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trinotate
  owner: iuc
  revisions: 6fd16fad465d
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trimmomatic
  owner: pjbriggs
  revisions: f8a9a5eaca8a
  tool_panel_section_label: Transcriptomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00290">https://gxy.io/GTN:T00290</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/full-de-novo/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703895990'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>