<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Genome assembly quality control.</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-quality-control/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1705105666' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1705105666' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Genome assembly quality control.">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-01-13 00:27:47 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Alexandre Cormier">
<meta name="DC.creator" content="StÃ©phanie Robin">
<meta name="DC.creator" content="Erwan Corre">
<meta name="DC.creator" content="Laura Leroi">
<meta name="DC.creator" content="Christophe Klopp">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Genome assembly quality control.">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1705105666'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/assembly-quality-control/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/assembly-quality-control/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/assembly/tutorials/assembly-quality-control/slides.html">View markdown source on GitHub</a>
	<h1>Genome assembly quality control.</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/alexcorm/" class="contributor-badge contributor-alexcorm"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/alexcorm?s=36" alt="Avatar" width="36" height="36">Alexandre Cormier</a><a href="/training-material/hall-of-fame/stephanierobin/" class="contributor-badge contributor-stephanierobin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/stephanierobin?s=36" alt="Avatar" width="36" height="36">StÃ©phanie Robin</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/r1corre?s=36" alt="Avatar" width="36" height="36">Erwan Corre</a><a href="/training-material/hall-of-fame/lleroi/" class="contributor-badge contributor-lleroi"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/lleroi?s=36" alt="Avatar" width="36" height="36">Laura Leroi</a><a href="/training-material/hall-of-fame/chklopp/" class="contributor-badge contributor-chklopp"><img src="https://avatars.githubusercontent.com/chklopp?s=36" alt="Avatar" width="36" height="36">Christophe Klopp</a>
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>Assembly: Is my genome assembly ready for scaffolding?</p>
</li>
		
		<li><p>Annotation: Is my genome assembly ready for structural annotation?</p>
</li>
		
		</ul>
		

		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 13, 2024 </div>

	<hr />


	<h1 id="genome-assembly-quality-control-or-the-3c">Genome assembly quality control, or âthe 3Câ</h1>

<p><br />
<br /></p>

<p>.image-100[
  <img src="../../images/genome-qc/3C.png" alt="The 3C for genome assembly quality control" />
]</p>

<hr />

<p>.image-30[
  <img src="../../images/genome-qc/contiguity.png" alt="Illustration of genome assembly contiguity" />
]</p>

<hr />

<h3 id="contiguity"><strong>C</strong>ontiguity</h3>

<p>.pull-left[</p>

<p><strong>Desire</strong></p>
<ul>
  <li>Fewer sequences</li>
  <li>Longer sequences
]</li>
</ul>

<p>.pull-right[
<strong>Metrics</strong>:</p>
<ul>
  <li>Number of sequences</li>
  <li>Average sequences length</li>
  <li>Median sequences length</li>
  <li>Minimum and maximum sequences length</li>
  <li>N50, NG50, L50, LG50</li>
  <li>GC content</li>
  <li>Number and proportion of bases that are N</li>
</ul>

<p>]</p>

<p><strong>Sequences</strong>, i.e. a set of contigs and/or scaffolds</p>

<hr />

<h3 id="n50--l50">N50 &amp; L50</h3>

<p>.left[
<strong>N50</strong>: given a set of sequences of varying lengths, the N50 is defined as <strong>the length L of the shortest contig</strong> for which <strong>longer and equal length contigs cover at least 50% of the assembly</strong>.</p>

<p><strong>L50</strong>: given a set of sequences of varying lengths, the L50 is defined as <strong>count of smallest number of sequences</strong> whose <strong>length sum makes up 50% of the assembly</strong>.
]</p>

<p><strong>N50 describes a sequence length whereas L50 describes a number of sequences.</strong></p>

<hr />

<h3 id="n50--l50-1">N50 &amp; L50</h3>

<p>.pull-left[
Example:</p>
<ul>
  <li>Genome size = 100</li>
  <li>Sequence sorted by size list L = (25, 10, 10, 8 , 7, 7 , 6 , 5, 5, 5, 5, 3, 2, 2 ) = 100</li>
  <li>50% of the total length is contained within sequences of at least 8bp: 25 + 10 + 10 + 8 â¥ 50
]</li>
</ul>

<p>.image-100[
  <img src="../../images/genome-qc/N50.png" alt="Schematic explanation of N50" />
]</p>

<p><strong>N50 = 8</strong> and <strong>L50 = 4</strong></p>

<p>.footnote[Alhakami, H., Mirebrahim, H., &amp; Lonardi, S. (2017). A comparative evaluation of genome assembly reconciliation tools. Genome biology, 18(1), 1-14.]</p>

<hr />

<h3 id="n50--l50-2">N50 &amp; L50</h3>

<p>However, the theses statistics may not reflect some assembly improvements.
If we connect two sequences longer than N50 or connect two sequences shorter than N50, N50 is not changed. N50 is only improved if we connect a sequence shorter than N50 and a sequence longer than N50.</p>

<p>.image-40[
<img src="../../images/genome-qc/N50-limitation.png" alt="Schematic explanation of N50 and it limits" />
]</p>

<hr />

<h3 id="nx-curve">Nx curve</h3>

<p>Â« 50 Â» is a single point on the Nx curve. The entire Nx curve in fact gives us a better sense of contiguity.</p>

<p>.pull-left[
<img src="../../images/genome-qc/Nx-curve.png" alt="Example of Nx graph for Drosophila assemblies" />
]</p>

<p>.pull-right[
<img src="../../images/genome-qc/cumulative-length.png" alt="Example of cumulative sequence length graph for Drosophila assemblies" />
]</p>

<hr />

<h3 id="quast---a-tool-to-evaluate-genome-assemblies">QUAST - A tool to evaluate genome assemblies</h3>

<ul>
  <li><strong>QUAST</strong>: for genome assemblies.</li>
  <li><strong>MetaQUAST</strong>: for metagenomic datasets.</li>
  <li><strong>QUAST-LG</strong>: for large genomes (e.g., mammalians).</li>
  <li><strong>rnaQUAST</strong>: for RNAseq.</li>
  <li><strong>Icarus</strong>: an interactive visualizer for these tools.</li>
</ul>

<p>It also includes:</p>
<ul>
  <li>Reads mapping (mi-assemblies evaluation).</li>
  <li>Kmer representation (KMC)</li>
  <li>Structural prediction modules (GeneMark,  GlimmerHMM, Barrnap and BUSCO).</li>
  <li>For metagenomics dataset: MetaGeneMark, Krona tools, BLAST, and SILVA 16S rRNA database.</li>
</ul>

<hr />

<p>.image-30[
  <img src="../../images/genome-qc/completeness.png" alt="Illustration of genome assembly completeness" />
]</p>

<hr />

<h3 id="types-of-completeness">Types of completeness</h3>

<ul>
  <li>Assembly size</li>
  <li>Known vs. unknown nucleotides</li>
  <li>âCoreâ genes</li>
  <li>Assembly kmer content</li>
  <li>Reads mapping and assembly coverage</li>
</ul>

<hr />

<h3 id="assembly-size-vs-estimated">Assembly size vs estimated</h3>

<p>Proportion of the original genome represented by the assembly:</p>

<p>.image-100[
<img src="../../images/genome-qc/assembly-size-qc-formula.png" alt="Formula to estimate assembly size completeness" />
]</p>

<p>â*â itâs an estimation, so not perfect. See <img src="/training-material/topics/assembly/tutorials/get-started-genome-assembly/slides.html" alt="An introduction to get started in genome assembly and annotation" /> to find methods to determine the genome size.</p>

<hr />

<h3 id="known-vs-unknown-nucleotides">Known vs. unknown nucleotides</h3>

<p>Proportion of A, T, G, C versus N (unknown nucleotide).We expect an assembly without unknown nucleotides (N).</p>

<hr />

<h3 id="core-genes">âCoreâ genes</h3>

<p>.pull-left[
Quantitative assessment of genome assembly based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs.
.image-70[
<img src="../../images/genome-qc/busco-formula.png" alt="Formula to estimate assembly completeness for core genes" />
]
]
.pull-right[
.image-70[
<img src="../../images/genome-qc/BUSCO-scores.png" alt="Example of BUSCO plot for Nosema species (Microsporidia) " />
]]</p>

<p>.footnote[Tips: Reference databases are constructed using known genomes. Species with few/no close genomes available can have very bad scores.]</p>

<hr />

<h3 id="core-genes-evaluation-software">Core genes evaluation software</h3>

<p><strong>BUSCO</strong>: Assessing genome assembly and annotation completeness with <strong>B</strong>enchmarking <strong>U</strong>niversal <strong>S</strong>ingle-<strong>C</strong>opy <strong>O</strong>rthologs</p>

<p>.pull-left[
.image-70[
<img src="../../images/genome-qc/busco_sampling_universality.png" alt="Explanation of orthology classification for creation of BUSCO sets - Universality" />
]]</p>

<p>.pull-right[
.image-60[
<img src="../../images/genome-qc/busco_sampling_duplicability.png" alt="Explanation of orthology classification for creation of BUSCO sets - Duplicability" />
]]</p>

<p>Eukaryota: 255 single copy from 70 species; Arthropoda: 1013 single copy from 90 species; Fungi: 758 single copy from 549 species</p>

<p>.footnote[Waterhouse, R. M., Zdobnov, E. M. &amp; Kriventseva, E. V. Correlating Traits of Gene Retention, Sequence Divergence, Duplicability and Essentiality in Vertebrates, Arthropods, and Fungi. Genome Biol Evol 3, 75â86 (2011).]</p>

<hr />

<h3 id="busco-limitations">BUSCO limitations</h3>

<p><strong>The value of the BUSCO is only as good as its reference database.</strong>
<br /></p>

<p>Example with BUSCO Eukaryotic set:
.pull-left[
.image-70[
<img src="../../images/genome-qc/BUSCO_limitations_A.png" alt="Limitation of BUSCO orthoDB for eukaOrthoDB - part A" />
]]
.pull-right[
.image-70[
<img src="../../images/genome-qc/BUSCO_limitations_B.png" alt="Limitation of BUSCO orthoDB for eukaOrthoDB - part B" />
]]</p>

<p>.footnote[Saary, P., Mitchell, A. L. &amp; Finn, R. D. Estimating the quality of eukaryotic genomes recovered from metagenomic analysis with EukCC. Genome Biol 21, 244 (2020).]</p>

<hr />

<h3 id="busco-limitations-1">BUSCO limitations</h3>

<p>The use of transcriptome alignment of a closely related species or a de novo RNA-Seq assembly of the same species can be another proxy to assess the completeness of the assembly and adress BUSCO limitations.</p>

<hr />

<h3 id="assembly-kmer-content">Assembly kmer content</h3>

<p>The aim is to check assembly coherence against the content within reads that were used to produce the assembly. Basically, how many elements of each frequency on the readâs spectrum ended up being not included in the assembly, included once, included twice etc.
<br />
<br /></p>

<ul>
  <li>Merqury or KAT</li>
  <li>Histogram is build with read kmer content.</li>
</ul>

<hr />

<h3 id="k-mer-spectrum-plots">K-mer spectrum plots</h3>

<p>.pull-left[
.image-70[
<img src="../../images/genome-qc/kmer-spectrum-reads.png" alt="How to read kmer spectrum of reads" />
]]
.pull-right[
.image-70[
<img src="../../images/genome-qc/kmer-spectrum-assembly.png" alt="How to read kmer spectrum of an assembly using reads" />
]]</p>

<p>.footnote[Rhie, A., Walenz, B. P., Koren, S. &amp; Phillippy, A. M. Merqury: reference-free quality, completeness, and phasing assessment for genome assemblies. Genome Biol 21, 245 (2020).]</p>

<hr />

<h3 id="assembly-kmer-content---homozygous-genomes">Assembly kmer content - Homozygous genomes</h3>

<p>.image-70[
<img src="../../images/genome-qc/kmer-spectra-merqury-homozygous-good.png" alt="Merqury spectra-cn plot for S.cerevesiae S288C assembly" />
]</p>

<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Good kmer representation of reads in the assembly</p>

<hr />

<h3 id="assembly-kmer-content---homozygous-genomes-1">Assembly kmer content - Homozygous genomes</h3>

<p>.image-70[
<img src="../../images/genome-qc/kmer-spectra-merqury-homozygous-bad.png" alt="Merqury spectra-cn plot for S.cerevesiae S288C assembly in case of missing contigs" />
]</p>

<p><i class="fa fa-thumbs-down"></i> Bad kmer representation of reads in the assembly</p>

<hr />

<h3 id="assembly-kmer-content---heterozygous-genomes">Assembly kmer content - Heterozygous genomes</h3>
<p><br /></p>

<p>.image-70[
<img src="../../images/genome-qc/kmer-spectra-merqury-heterozygous-good.png" alt="Merqury spectra-cn plot for highly heterozygous assembly and for which haplotig were purged or collapsed." />
]</p>

<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Good kmer representation of reads in the assembly</p>

<p>The lost content (the black peak) represents the half of the heterozygous content that is lost when bubbles are collapsed.</p>

<hr />

<h3 id="assembly-kmer-content---heterozygous-genomes-1">Assembly kmer content - Heterozygous genomes</h3>

<p>.image-80[
<img src="../../images/genome-qc/kmer-spectra-merqury-c.gigas-hap2.png" alt="Merqury spectra-cn plot for second haplotype of C. gigas." />
]</p>

<p><i class="fa fa-thumbs-down"></i> Bad kmer representation of reads in the assembly</p>

<hr />

<h3 id="reads-mapping-and-assembly-coverage">Reads mapping and assembly coverage</h3>
<p><br /></p>

<ul>
  <li><strong>Proportion of mapped vs. unmapped reads</strong> i.e. proportion of missing parts in the assembly</li>
  <li><strong>Coverage in terms of redundancy (A)</strong>: number of reads that align to, or âcover,â a known reference.</li>
  <li><strong>Coverage in terms of the percentage coverage of a reference by reads (B)</strong>: E.g. if 90% of a reference is covered by reads (and 10% not) it is a 90% coverage.</li>
</ul>

<p>.image-60[
  <img src="../../images/genome-qc/read-coverage.gif" alt="Illustration of genome assembly correctness" />
]</p>

<hr />

<p>.image-30[
  <img src="../../images/genome-qc/correctness.png" alt="Illustration of genome assembly correctness" />
]</p>

<hr />

<h3 id="mistakes-into-the-assembly">Mistakes into the assembly</h3>

<p>** Proportion of the assembly that is free from mistakes**
<br />
<br /></p>
<ul>
  <li>Indels / SNPs</li>
  <li>Mis-joins</li>
  <li>Repeat compressions</li>
  <li>Unnecessary duplications</li>
  <li>Rearrangements
<br />
<br />
<strong>â Align back reads to the assembly and check for inconsistencies</strong></li>
</ul>

<hr />

<h3 id="snp--indels-errors">SNP / indels errors</h3>

<p>.image-80[
<img src="../../images/genome-qc/snp-indels_errors.png" alt="Illustration of SNP errors in a genome assembly." />
]</p>

<hr />

<h3 id="other-mis-assemblies">Other mis-assemblies</h3>

<p>.image-80[
<img src="../../images/genome-qc/misassemblies_1.png" alt="Illustration of rearrangements inversions assembly errors." />
]</p>

<hr />

<h3 id="other-mis-assemblies-1">Other mis-assemblies</h3>

<p>.image-80[
<img src="../../images/genome-qc/misassemblies_2.png" alt="Illustration of collapsed and expanded repeats assembly errors." />
]</p>

<hr />

<h3 id="switch-and-hamming-errors-phased-assemblies">Switch and hamming errors (phased assemblies)</h3>

<p>.pull-left[
.image-70[
<img src="../../images/genome-qc/switch-hamming-errors.png" alt="Illustration of switch and hamming errors in phased genome assembly." />
]
In red, heterozygous locus from second haplotype. In blue, heterozygous locus from first haplotype.
]</p>

<p>.pull-right[</p>
<ul>
  <li><strong>Switch error</strong>: a change from one parental allele to another parental allele on a contig. This terminology has been used for measuring reference-based phasing accuracy for two decades. A haplotig is supposed to have no switch errors.</li>
  <li><strong>Yak hamming error</strong>: an allele not on the most supported haplotype of a contig. Its main purpose is to test how close a contig is to a haplotig. The yak definition is not widely accepted. The hamming error rate is arguably less important in practice.
]</li>
</ul>

<p>.footnote[http://lh3.github.io/2021/04/17/concepts-in-phased-assemblies]</p>

<hr />

<h1 id="evaluation-against-reference-genome">Evaluation against reference genome</h1>

<p>.image-30[
<img src="../../images/genome-qc/dotplot-dgenies.png" alt="Example of a dot plot between 2 genomes." />
]</p>

<hr />

<p>.pull-left[
.left[
Dot plots are widely used to quickly compare 2 sequence sets. They provide a synthetic overview of:</p>
<ul>
  <li>Similarity</li>
  <li>Specificity</li>
  <li>Highlighting repetitions, breaks and inversions.
]</li>
</ul>

<p><br /></p>

<p>.image-50[
<img src="../../images/genome-qc/dotplot-dgenies.png" alt="Example of a dot plot between 2 genomes." />
]
]</p>

<p>.pull-right[
.left[
A non-exhaustive list of tools for making dot plots:</p>
<ul>
  <li>MUMmer dotplot</li>
  <li>Chromeister</li>
  <li>D-genies (not yet available into Galaxy)
]</li>
</ul>

<p><br />
<br />
<br /></p>

<p>.image-50[
<img src="../../images/dotplot.png" alt="Interpret Dot Plots here based on great explanation by Michael Schatz." />
]
]</p>

<hr />

<h1 id="tips">Tips</h1>

<ul>
  <li>
    <p>The quality of an assembly is often validated by using other data from the same individual or from other individuals (RNA-Seq alignment, Hi-C alignment, DNA-Seq alignment,â¦).</p>
  </li>
  <li>
    <p>The positions of the telomeric repeats in the chromosome assemblies are also of interesting to evaluate the correctness.</p>
  </li>
  <li>
    <p>The identification of organelles (mitochondria, chloroplast,â¦) can also inform us about the quality of the assembly in terms of completness. However, the structure of the organelles may lead the assembler to think that they are repeats and he discards them.</p>
  </li>
  <li>
    <p>In the case of diploid organisms, one of the classical problems of assemblies is the conservation of the two haplotypes. We obtains particular BUSCO / kmer / assembly size metrics that can be corrected by removing, âpurgingâ, the haplotigs.</p>
  </li>
</ul>



	<hr />


<h2>Key Points</h2>

<ul>

<li>We learned that it is essential to control the quality of an assembly</li>

<li>We learned that there are several quality criteria and tools to enable this assessment</li>

<li>Certain quality criteria are expected at the time of publication</li>

</ul>







<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/assembly-quality-control/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1705105666'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
