<!DOCTYPE html>
<html lang="en">


    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Genome assembly quality control.</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@0c9d252243153e0d408d3eb6b0478dfb6ae973ec",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-quality-control/slides.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Genome assembly quality control.">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-03-20 10:41:17 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Alexandre Cormier">
<meta name="DC.creator" content="StÃ©phanie Robin">
<meta name="DC.creator" content="Erwan Corre">
<meta name="DC.creator" content="Laura Leroi">
<meta name="DC.creator" content="Christophe Klopp">

        
        
        
        
        <meta name="description" content="DNA sequence data has become an indispensable tool for Mo..." />
        <meta property="og:site_name" content="Galaxy Training Network" />
        <meta property="og:title" content="Galaxy Training: Genome assembly quality control." />
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Mo..." />
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-03-20 10:41:17 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Genome assembly quality control.",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
  },
  "learningResourceType": "slides",
  "name": "Slides for 'Genome assembly quality control.' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-quality-control/slides.html",
  "keywords": "assembly",
  "description": "The questions this  addresses are:\n - Assembly: Is my genome assembly ready for scaffolding?\n - Annotation: Is my genome assembly ready for structural annotation?\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "name": "Anthony Bretaudeau",
      "image": "https://avatars.githubusercontent.com/abretaud",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0914-2470",
      "orcid": "https://orcid.org/0000-0003-0914-2470"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/alexcorm/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/alexcorm/",
      "name": "Alexandre Cormier",
      "image": "https://avatars.githubusercontent.com/alexcorm",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7775-8413",
      "orcid": "https://orcid.org/0000-0002-7775-8413"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/stephanierobin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/stephanierobin/",
      "name": "StÃ©phanie Robin",
      "image": "https://avatars.githubusercontent.com/stephanierobin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-7379-9173",
      "orcid": "https://orcid.org/0000-0001-7379-9173"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "name": "Erwan Corre",
      "image": "https://avatars.githubusercontent.com/r1corre",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6354-2278",
      "orcid": "https://orcid.org/0000-0001-6354-2278"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/lleroi/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/lleroi/",
      "name": "Laura Leroi",
      "image": "https://avatars.githubusercontent.com/lleroi",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3748-4179",
      "orcid": "https://orcid.org/0000-0003-3748-4179"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/chklopp/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/chklopp/",
      "name": "Christophe Klopp",
      "image": "https://avatars.githubusercontent.com/chklopp",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0196",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0196",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0196"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [

  ],
  "abstract": "# Genome assembly quality control, or \"the 3C\""
}
        </script>
    </head>
  <body data-brightness="auto" data-contrast="auto">


    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/assembly" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/assembly-quality-control/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---



<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" class="cover-logo"/>


# Genome assembly quality control.


<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/alexcorm/" class="contributor-badge contributor-alexcorm"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/alexcorm?s=36" alt="Avatar" width="36" height="36">Alexandre Cormier</a><a href="/training-material/hall-of-fame/stephanierobin/" class="contributor-badge contributor-stephanierobin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/stephanierobin?s=36" alt="Avatar" width="36" height="36">StÃ©phanie Robin</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/r1corre?s=36" alt="Avatar" width="36" height="36">Erwan Corre</a><a href="/training-material/hall-of-fame/lleroi/" class="contributor-badge contributor-lleroi"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/lleroi?s=36" alt="Avatar" width="36" height="36">Laura Leroi</a><a href="/training-material/hall-of-fame/chklopp/" class="contributor-badge contributor-chklopp"><img src="https://avatars.githubusercontent.com/chklopp?s=36" alt="Avatar" width="36" height="36">Christophe Klopp</a>
	</div>

</div>


<!-- modified date -->
<div class="footnote" style="bottom: 8em;">
  <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Updated:  &nbsp;
  <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:S00030">gxy.io/GTN:S00030</a>
</div>


<!-- other slide formats (video and plain-text) -->
<div class="footnote" style="bottom: 6em;">


  <i class="fas fa-file-alt" aria-hidden="true"></i><span class="visually-hidden">text-document</span><a href="slides-plain.html"> Plain-text slides</a>
</div>


<!-- usage tips -->
<div class="footnote" style="bottom: 2em;">
    <strong>Tip: </strong>press <kbd>P</kbd> to view the presenter notes
    | <i class="fa fa-arrows" aria-hidden="true"></i><span class="visually-hidden">arrow-keys</span> Use arrow keys to move between slides

</div>

???
Presenter notes contain extra information which might be useful if you intend to use these slides for teaching.

Press `P` again to switch presenter notes off

Press `C` to create a new window where the same presentation will be displayed.
This window is linked to the main window. Changing slides on one will cause the
slide to change on the other.

Useful when presenting.



---

## Requirements

Before diving into this slide deck, we recommend you to have a look at:



    
        
        
            
        
- [Introduction to Galaxy Analyses](/training-material/topics/introduction)
    

    
        
        
            
        
- [Sequence analysis](/training-material/topics/sequence-analysis)
    - Quality Control: [<i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides](/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html) - [<i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on](/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html)
    






---







### &lt;i class=&quot;far fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- Assembly: Is my genome assembly ready for scaffolding?

- Annotation: Is my genome assembly ready for structural annotation?


---




# Genome assembly quality control, or &quot;the 3C&quot;


&lt;br&gt;
&lt;br&gt;

.image-100[
  ![The 3C for genome assembly quality control](../../images/genome-qc/3C.png)
]

---

.image-30[
  ![Illustration of genome assembly contiguity](../../images/genome-qc/contiguity.png)
]

---

### **C**ontiguity

.pull-left[

**Desire**
- Fewer sequences
- Longer sequences
]

.pull-right[
**Metrics**:
- Number of sequences
- Average sequences length
- Median sequences length
- Minimum and maximum sequences length
- N50, NG50, L50, LG50
- GC content
- Number and proportion of bases that are N

]

**Sequences**, i.e. a set of contigs and/or scaffolds

---

### N50 &amp; L50

.left[
**N50**: given a set of sequences of varying lengths, the N50 is defined as **the length L of the shortest contig** for which **longer and equal length contigs cover at least 50% of the assembly**.

**L50**: given a set of sequences of varying lengths, the L50 is defined as **count of smallest number of sequences** whose **length sum makes up 50% of the assembly**.
]

**N50 describes a sequence length whereas L50 describes a number of sequences.**

---

### N50 &amp; L50

.pull-left[
Example:
- Genome size = 100
- Sequence sorted by size list L = (25, 10, 10, 8 , 7, 7 , 6 , 5, 5, 5, 5, 3, 2, 2 ) = 100
- 50% of the total length is contained within sequences of at least 8bp: 25 + 10 + 10 + 8 â¥ 50
]

.image-100[
  ![Schematic explanation of N50](../../images/genome-qc/N50.png)
]

**N50 = 8** and **L50 = 4**

.footnote[Alhakami, H., Mirebrahim, H., &amp; Lonardi, S. (2017). A comparative evaluation of genome assembly reconciliation tools. Genome biology, 18(1), 1-14.]

---

### N50 &amp; L50

However, the theses statistics may not reflect some assembly improvements.
If we connect two sequences longer than N50 or connect two sequences shorter than N50, N50 is not changed. N50 is only improved if we connect a sequence shorter than N50 and a sequence longer than N50.

.image-40[
![Schematic explanation of N50 and it limits](../../images/genome-qc/N50-limitation.png)
]

---

### Nx curve

Â« 50 Â» is a single point on the Nx curve. The entire Nx curve in fact gives us a better sense of contiguity.

.pull-left[
![Example of Nx graph for Drosophila assemblies](../../images/genome-qc/Nx-curve.png)
]

.pull-right[
![Example of cumulative sequence length graph for Drosophila assemblies](../../images/genome-qc/cumulative-length.png)
]

---

### QUAST - A tool to evaluate genome assemblies

- **QUAST**: for genome assemblies.
- **MetaQUAST**: for metagenomic datasets.
- **QUAST-LG**: for large genomes (e.g., mammalians).
- **rnaQUAST**: for RNAseq.
- **Icarus**: an interactive visualizer for these tools.


It also includes:
- Reads mapping (mi-assemblies evaluation).
- Kmer representation (KMC)
- Structural prediction modules (GeneMark,  GlimmerHMM, Barrnap and BUSCO).
- For metagenomics dataset: MetaGeneMark, Krona tools, BLAST, and SILVA 16S rRNA database.

---

.image-30[
  ![Illustration of genome assembly completeness](../../images/genome-qc/completeness.png)
]

---

### Types of completeness

- Assembly size
- Known vs. unknown nucleotides
- &quot;Core&quot; genes
- Assembly kmer content
- Reads mapping and assembly coverage

---

### Assembly size vs estimated

Proportion of the original genome represented by the assembly:

.image-100[
![Formula to estimate assembly size completeness](../../images/genome-qc/assembly-size-qc-formula.png)
]

&quot;*&quot; itâs an estimation, so not perfect. See ![An introduction to get started in genome assembly and annotation](/training-material/topics/assembly/tutorials/get-started-genome-assembly/slides.html) to find methods to determine the genome size.

---

### Known vs. unknown nucleotides

Proportion of A, T, G, C versus N (unknown nucleotide).We expect an assembly without unknown nucleotides (N).

---

### &quot;Core&quot; genes


.pull-left[
Quantitative assessment of genome assembly based on evolutionarily informed expectations of gene content from near-universal single-copy orthologs.
.image-70[
![Formula to estimate assembly completeness for core genes](../../images/genome-qc/busco-formula.png)
]
]
.pull-right[
.image-70[
![Example of BUSCO plot for Nosema species (Microsporidia) ](../../images/genome-qc/BUSCO-scores.png)
]]

.footnote[Tips: Reference databases are constructed using known genomes. Species with few/no close genomes available can have very bad scores.]

---

### Core genes evaluation software

**BUSCO**: Assessing genome assembly and annotation completeness with **B**enchmarking **U**niversal **S**ingle-**C**opy **O**rthologs

.pull-left[
.image-70[
![Explanation of orthology classification for creation of BUSCO sets - Universality](../../images/genome-qc/busco_sampling_universality.png)
]]

.pull-right[
.image-60[
![Explanation of orthology classification for creation of BUSCO sets - Duplicability](../../images/genome-qc/busco_sampling_duplicability.png)
]]

Eukaryota: 255 single copy from 70 species; Arthropoda: 1013 single copy from 90 species; Fungi: 758 single copy from 549 species

.footnote[Waterhouse, R. M., Zdobnov, E. M. &amp; Kriventseva, E. V. Correlating Traits of Gene Retention, Sequence Divergence, Duplicability and Essentiality in Vertebrates, Arthropods, and Fungi. Genome Biol Evol 3, 75â86 (2011).]

---

### BUSCO limitations

**The value of the BUSCO is only as good as its reference database.**
&lt;br&gt;

Example with BUSCO Eukaryotic set:
.pull-left[
.image-70[
![Limitation of BUSCO orthoDB for eukaOrthoDB - part A](../../images/genome-qc/BUSCO_limitations_A.png)
]]
.pull-right[
.image-70[
![Limitation of BUSCO orthoDB for eukaOrthoDB - part B](../../images/genome-qc/BUSCO_limitations_B.png)
]]


.footnote[Saary, P., Mitchell, A. L. &amp; Finn, R. D. Estimating the quality of eukaryotic genomes recovered from metagenomic analysis with EukCC. Genome Biol 21, 244 (2020).]

---

### BUSCO limitations

The use of transcriptome alignment of a closely related species or a de novo RNA-Seq assembly of the same species can be another proxy to assess the completeness of the assembly and adress BUSCO limitations.

---

### Assembly kmer content

The aim is to check assembly coherence against the content within reads that were used to produce the assembly. Basically, how many elements of each frequency on the readâs spectrum ended up being not included in the assembly, included once, included twice etc.
&lt;br&gt;
&lt;br&gt;

- Merqury or KAT
- Histogram is build with read kmer content.

---

### K-mer spectrum plots

.pull-left[
.image-70[
![How to read kmer spectrum of reads](../../images/genome-qc/kmer-spectrum-reads.png)
]]
.pull-right[
.image-70[
![How to read kmer spectrum of an assembly using reads](../../images/genome-qc/kmer-spectrum-assembly.png)
]]

.footnote[Rhie, A., Walenz, B. P., Koren, S. &amp; Phillippy, A. M. Merqury: reference-free quality, completeness, and phasing assessment for genome assemblies. Genome Biol 21, 245 (2020).]

---

### Assembly kmer content - Homozygous genomes

.image-70[
![Merqury spectra-cn plot for S.cerevesiae S288C assembly](../../images/genome-qc/kmer-spectra-merqury-homozygous-good.png)
]

&lt;i class=&quot;far fa-thumbs-up&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;congratulations&lt;/span&gt; Good kmer representation of reads in the assembly

---

### Assembly kmer content - Homozygous genomes

.image-70[
![Merqury spectra-cn plot for S.cerevesiae S288C assembly in case of missing contigs](../../images/genome-qc/kmer-spectra-merqury-homozygous-bad.png)
]

&lt;i class=&quot;fa fa-thumbs-down&quot;&gt;&lt;/i&gt; Bad kmer representation of reads in the assembly

---

### Assembly kmer content - Heterozygous genomes
&lt;br&gt;

.image-70[
![Merqury spectra-cn plot for highly heterozygous assembly and for which haplotig were purged or collapsed.](../../images/genome-qc/kmer-spectra-merqury-heterozygous-good.png)
]

&lt;i class=&quot;far fa-thumbs-up&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;congratulations&lt;/span&gt; Good kmer representation of reads in the assembly

The lost content (the black peak) represents the half of the heterozygous content that is lost when bubbles are collapsed.

---

### Assembly kmer content - Heterozygous genomes

.image-80[
![Merqury spectra-cn plot for second haplotype of C. gigas.](../../images/genome-qc/kmer-spectra-merqury-c.gigas-hap2.png)
]

&lt;i class=&quot;fa fa-thumbs-down&quot;&gt;&lt;/i&gt; Bad kmer representation of reads in the assembly

---

### Reads mapping and assembly coverage
&lt;br&gt;

- **Proportion of mapped vs. unmapped reads** i.e. proportion of missing parts in the assembly
- **Coverage in terms of redundancy (A)**: number of reads that align to, or &quot;cover,&quot; a known reference. 
- **Coverage in terms of the percentage coverage of a reference by reads (B)**: E.g. if 90% of a reference is covered by reads (and 10% not) it is a 90% coverage.

.image-60[
  ![Illustration of genome assembly correctness](../../images/genome-qc/read-coverage.gif)
]

---

.image-30[
  ![Illustration of genome assembly correctness](../../images/genome-qc/correctness.png)
]

---

### Mistakes into the assembly

** Proportion of the assembly that is free from mistakes**
&lt;br&gt;
&lt;br&gt;
- Indels / SNPs
- Mis-joins
- Repeat compressions
- Unnecessary duplications
- Rearrangements
&lt;br&gt;
&lt;br&gt;
**â Align back reads to the assembly and check for inconsistencies**

---

### SNP / indels errors

.image-80[
![Illustration of SNP errors in a genome assembly.](../../images/genome-qc/snp-indels_errors.png)
]

---

### Other mis-assemblies

.image-80[
![Illustration of rearrangements inversions assembly errors.](../../images/genome-qc/misassemblies_1.png)
]

---

### Other mis-assemblies

.image-80[
![Illustration of collapsed and expanded repeats assembly errors.](../../images/genome-qc/misassemblies_2.png)
]

---

### Switch and hamming errors (phased assemblies)

.pull-left[
.image-70[
![Illustration of switch and hamming errors in phased genome assembly.](../../images/genome-qc/switch-hamming-errors.png)
]
In red, heterozygous locus from second haplotype. In blue, heterozygous locus from first haplotype.
]

.pull-right[
- **Switch error**: a change from one parental allele to another parental allele on a contig. This terminology has been used for measuring reference-based phasing accuracy for two decades. A haplotig is supposed to have no switch errors.
- **Yak hamming error**: an allele not on the most supported haplotype of a contig. Its main purpose is to test how close a contig is to a haplotig. The yak definition is not widely accepted. The hamming error rate is arguably less important in practice.
]

.footnote[http://lh3.github.io/2021/04/17/concepts-in-phased-assemblies]

---

# Evaluation against reference genome


.image-30[
![Example of a dot plot between 2 genomes.](../../images/genome-qc/dotplot-dgenies.png)
]

---

.pull-left[
.left[
Dot plots are widely used to quickly compare 2 sequence sets. They provide a synthetic overview of:
- Similarity
- Specificity
- Highlighting repetitions, breaks and inversions.
]

&lt;br&gt;

.image-50[
![Example of a dot plot between 2 genomes.](../../images/genome-qc/dotplot-dgenies.png)
]
]

.pull-right[
.left[
A non-exhaustive list of tools for making dot plots:
- MUMmer dotplot
- Chromeister
- D-genies (not yet available into Galaxy)
]

&lt;br&gt;
&lt;br&gt;
&lt;br&gt;

.image-50[
![Interpret Dot Plots here based on great explanation by Michael Schatz.](../../images/dotplot.png)
]
]

---

# Tips

- The quality of an assembly is often validated by using other data from the same individual or from other individuals (RNA-Seq alignment, Hi-C alignment, DNA-Seq alignment,...).

- The positions of the telomeric repeats in the chromosome assemblies are also of interesting to evaluate the correctness.

- The identification of organelles (mitochondria, chloroplast,...) can also inform us about the quality of the assembly in terms of completness. However, the structure of the organelles may lead the assembler to think that they are repeats and he discards them.

- In the case of diploid organisms, one of the classical problems of assemblies is the conservation of the two haplotypes. We obtains particular BUSCO / kmer / assembly size metrics that can be corrected by removing, &quot;purging&quot;, the haplotigs.





---
### &lt;i class=&quot;fas fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- We learned that it is essential to control the quality of an assembly

- We learned that there are several quality criteria and tools to enable this assessment

- Certain quality criteria are expected at the time of publication







---

## Thank You!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://training.galaxyproject.org) and all the contributors!

<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/alexcorm/" class="contributor-badge contributor-alexcorm"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/alexcorm?s=36" alt="Avatar" width="36" height="36">Alexandre Cormier</a><a href="/training-material/hall-of-fame/stephanierobin/" class="contributor-badge contributor-stephanierobin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/stephanierobin?s=36" alt="Avatar" width="36" height="36">StÃ©phanie Robin</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/r1corre?s=36" alt="Avatar" width="36" height="36">Erwan Corre</a><a href="/training-material/hall-of-fame/lleroi/" class="contributor-badge contributor-lleroi"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/lleroi?s=36" alt="Avatar" width="36" height="36">Laura Leroi</a><a href="/training-material/hall-of-fame/chklopp/" class="contributor-badge contributor-chklopp"><img src="https://avatars.githubusercontent.com/chklopp?s=36" alt="Avatar" width="36" height="36">Christophe Klopp</a>
	</div>

</div>


<div style="display: flex;flex-direction: row;align-items: center;justify-content: center;">

<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


</div>

<a rel="license" href="https://creativecommons.org/licenses/by/4.0/">
This material is licensed under the Creative Commons Attribution 4.0 International License</a>.


    </textarea>
	<script src="/training-material/assets/js/remark-latest.min.js"></script>
    <script src="/training-material/assets/js/theme.js"></script>
    <script src="/training-material/assets/js/clipboard.min.js"></script>
    <script>
      var slideshow = remark.create({navigation: {scroll: false}, ratio: '16:9'});
      var hljs = remark.highlighter.engine;
      var snippets = document.querySelectorAll('code.remark-code');
        [].forEach.call(snippets,function(snippet){
          snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });
      var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.parentElement;
      }});
    </script>

    <script>
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('body').prepend("<div style='text-align: center'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
        <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>

    </body>
</html>
