<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>RNA-Seq analysis with AskOmics Interactive Tool</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: RNA-Seq analysis with AskOmics Interactive Tool" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "RNA-Seq analysis with AskOmics Interactive Tool",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / rna-seq-analysis-with-askomics-it / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'RNA-Seq analysis with AskOmics Interactive Tool' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - How to integrate RNA-Seq results with other datasets using AskOmics?\n - How to query these datasets to answer a biological question?\n - How to exploit a distant SPARQL endpoint to retrieve information?\n\n\\nThe objectives are:\n - Launch an AskOmics Interactive Tool\n - Integrate RNA-Seq and reference datasets into AskOmics\n - Create a complex query to get over-expressed genes, their location on the reference genome, and check if they are included in a known QTL\n - Complete the query to get the human homologs and their location using neXtProt database\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html",
      "name": "2: RNA-seq counts to genes",
      "description": "Hands-on for '2: RNA-seq counts to genes' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Xavier Garnier"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Anne Siegel"
    },
    {
      "@type": "Person",
      "name": "Olivier Dameron"
    },
    {
      "@type": "Person",
      "name": "Mateo Boudet"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Xavier Garnier"
    },
    {
      "@type": "Person",
      "name": "Anthony Bretaudeau"
    },
    {
      "@type": "Person",
      "name": "Anne Siegel"
    },
    {
      "@type": "Person",
      "name": "Olivier Dameron"
    },
    {
      "@type": "Person",
      "name": "Mateo Boudet"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
</script>

<section class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>RNA-Seq analysis with AskOmics Interactive Tool</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/xgaia/" class="contributor-badge contributor-xgaia"><img src="https://avatars.githubusercontent.com/xgaia?s=27" alt="Avatar">Xavier Garnier</a>


<a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="https://avatars.githubusercontent.com/abretaud?s=27" alt="Avatar">Anthony Bretaudeau</a>


<a href="/training-material/hall-of-fame/annesiegel/" class="contributor-badge contributor-annesiegel"><img src="https://avatars.githubusercontent.com/annesiegel?s=27" alt="Avatar">Anne Siegel</a>


<a href="/training-material/hall-of-fame/odameron/" class="contributor-badge contributor-odameron"><img src="https://avatars.githubusercontent.com/odameron?s=27" alt="Avatar">Olivier Dameron</a>


<a href="/training-material/hall-of-fame/mboudet/" class="contributor-badge contributor-mboudet"><img src="https://avatars.githubusercontent.com/mboudet?s=27" alt="Avatar">Mateo Boudet</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to integrate RNA-Seq results with other datasets using AskOmics?</p>
</li>
        
        <li><p>How to query these datasets to answer a biological question?</p>
</li>
        
        <li><p>How to exploit a distant SPARQL endpoint to retrieve information?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Launch an AskOmics Interactive Tool</p>
</li>
        
        <li><p>Integrate RNA-Seq and reference datasets into AskOmics</p>
</li>
        
        <li><p>Create a complex query to get over-expressed genes, their location on the reference genome, and check if they are included in a known QTL</p>
</li>
        
        <li><p>Complete the query to get the human homologs and their location using neXtProt database</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> 2: RNA-seq counts to genes:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3925863">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 21, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<!-- AskOmics intro -->
<p>AskOmics is a web application for data integration and querying using the Semantic Web technologies. It helps users to convert multiple data sources (CSV/TSV files, GFF and <span class="notranslate">BED</span> annotation) into “RDF triples” and store them in a specific kind of database: an “RDF triplestore”. Under this form, data can then be queried using a specific language: “SPARQL”. AskOmics hides the complexity of these technologies and allows to perform complex queries using a user-friendly interface.</p>

<!-- AskOmics for RNA-Seq -->
<p>AskOmics comes in useful for cross-referencing results datasets with various reference data. For example, in <span class="notranslate">RNA</span>-Seq studies, we often need to filter the results on the fold change and the p-value, to get the most significant differentially expressed genes. If you are studying a particular phenotype and already know the position of some QTL associated to this phenotype, you would then want to find the positions of the differentially expressed genes and determine which gene is located within one of those QTL. Finally, you would want to know if these genes have human homologs, and use the neXtProt database to get the location of the proteins coded by the homologs. The whole process involves several tools to parse and manipulate the different data format, and to map datasets on each other. AskOmics offer a solution to 1) automatically convert the multiple formats to RDF, 2) use a user-friendly interface to perform complex SPARQL queries on the RDF datasets to find the genes you are interested in, and 3) connect exte<span class="notranslate">rna</span>l SPARQL databases and link exte<span class="notranslate">rna</span>l data with your own.</p>

<!-- The data -->
<p>In this tutorial, we will use results from a differential <span class="notranslate">expression</span> analysis. This file is provided for you below. You could also generate the file yourself, by following the <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><span class="notranslate">RNA</span>-Seq counts to gene tutorial</a>. The file used here was generated from limma-voom but you could use a file from any <span class="notranslate">RNA</span>-seq differential <span class="notranslate">expression</span> tool, such as edgeR or DESeq2, as long as it contains the required columns (see below).</p>

<p>The differentially expressed results will be linked to the official mouse genome annotation, in general feature format (GFF). The file provided is a subset of the mouse annotation (GRCm38.p6) obtained from <a href="http://www.ensembl.org/Mus_musculus/Info/Index">Ensembl</a>.</p>

<p>We will use a file containing quantitative trait loci (QTL) information, to find if our differentially expressed genes are located inside a known QTL. This file is a subset of a query performed on <a href="http://www.informatics.jax.org">Mouse Genome Informatics</a>.</p>

<p>A file containing all homologies between mouse and human will be used to get the human homolog genes. This file is provided by <a href="http://www.informatics.jax.org/downloads/reports/index.html#homology">MGI</a>.</p>

<p>In the differentially expressed file, and the homologs file, gene are described by a symbol (e.g. Pwgrq10). However, in the annotation file and neXtProt database, gene are represented by Ensembl id (e.g. ENSMUSG00000025969). To link the 2 datasets, we will need a file to map the gene symbol with Ensembl id. This file provided for you was previously generated with an AskOmics query on the mouse annotation file and the homolog file.</p>

<p>To link the human gene with neXtProt database, we will use the RDF abstraction of neXtProt. This file was obtained using the <a href="https://github.com/askomics/abstractor">Abstractor</a> tool.</p>

<blockquote class="details">
  <h3 id="details-abstraction"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Abstraction</h3>
  <p>During the integration step, AskOmics builds an RDF description of the data: the <strong>abstraction</strong>. This abstraction is used to explore the data and build the query.
AskOmics can also integrate <strong>abstraction</strong> of distant endpoint. Abstraction are obtained using <a href="https://github.com/askomics/abstractor">abstractor</a>, a python package to generate RDF abstractions from distant endpoints.
The query builder interface is used to create a path through the <strong>abstraction</strong> of each ressources. The path is converted to a SPARQL query that is sent to the multiple SPARQL endpoint.</p>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#preparing-the-inputs" id="markdown-toc-preparing-the-inputs">Preparing the inputs</a>    <ol>
      <li><a href="#import-data" id="markdown-toc-import-data">Import data</a></li>
    </ol>
  </li>
  <li><a href="#upload-inputs-into-askomics" id="markdown-toc-upload-inputs-into-askomics">Upload inputs into AskOmics</a>    <ol>
      <li><a href="#launch-askomics-interactive-tool" id="markdown-toc-launch-askomics-interactive-tool">Launch AskOmics Interactive Tool</a></li>
    </ol>
  </li>
  <li><a href="#integrate-input-files-into-askomics" id="markdown-toc-integrate-input-files-into-askomics">Integrate input files into AskOmics</a>    <ol>
      <li><a href="#integrate-gff-files" id="markdown-toc-integrate-gff-files">Integrate GFF files</a></li>
      <li><a href="#integration-of-tabular-tsv-files" id="markdown-toc-integration-of-tabular-tsv-files">Integration of tabular (TSV) files</a></li>
      <li><a href="#integration-of-rdf-files" id="markdown-toc-integration-of-rdf-files">Integration of RDF files</a></li>
      <li><a href="#monitor-integration" id="markdown-toc-monitor-integration">Monitor integration</a></li>
    </ol>
  </li>
  <li><a href="#query" id="markdown-toc-query">Query</a>    <ol>
      <li><a href="#query-builder-overview" id="markdown-toc-query-builder-overview">Query builder overview</a></li>
      <li><a href="#results-management" id="markdown-toc-results-management">Results management</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="preparing-the-inputs">Preparing the inputs</h1>

<p>We will use four files for this analysis:</p>

<ul>
  <li><strong>Differentially expressed results file</strong> (genes in rows, and 4 required columns: identifier (ENTREZID), gene symbol (SYMBOL), log fold change (logFC) and adjusted P values (adj.P.Val))</li>
  <li><strong>Reference genome annotation file</strong> (in GFF format)</li>
  <li><strong>QTL file</strong> (QTL in rows, with 5 required columns: identifier, chromosome, start, end and name)</li>
  <li><strong>Homolog file</strong> (TSV of 13 columns including homolog id, organism name and gene symbol)</li>
  <li><strong>Correspondence file between gene symbol and Ensembl id</strong> (TSV of 3 columns: symbol, the corresponding Ensembl id (mouse and human)</li>
  <li><strong>neXtProt abstraction</strong> (RDF data description of neXtProt database in turtle format)</li>
</ul>

<h2 id="import-data">Import data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this <span class="notranslate">RNA</span>-seq exercise e.g. <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>-seq AskOmics</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
    <li>
      <p>Import the files.</p>

      <p>To import the files, there are two options:</p>
      <ul>
        <li>Option 1: From a shared data library if available (ask your instructor)</li>
        <li>Option 2: From <a href="https://zenodo.org/record/3950862">Zenodo</a></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>

      <ul>
        <li>
          <p>You can paste the links below into the <strong>Paste/Fetch</strong> box:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2529117/files/limma-voom_luminalpregnant-luminallactate
https://zenodo.org/record/3950862/files/Mus_musculus.GRCm38.98.subset.gff3
https://zenodo.org/record/3950862/files/Symbol.tsv
https://zenodo.org/record/3950862/files/MGIBatchReport_Qtl_Subset.txt
https://zenodo.org/record/3950862/files/HOM_MouseHumanSequence.rpt
https://zenodo.org/record/3950862/files/nextprot_abstraction.ttl
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>Rename the files using the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> (pencil) icon.
      <ul>
        <li>limma-voom_luminalpregnant-luminallactate to <code class="language-plaintext highlighter-rouge">DE results</code></li>
        <li>Mus_musculus.GRCm38.98.subset.gff3 to <code class="language-plaintext highlighter-rouge">Mus musculus annotation</code></li>
        <li>Symbol.tsv to <code class="language-plaintext highlighter-rouge">Gene Symbols</code></li>
        <li>MGIBatchReport_Qtl_Subset.txt to <code class="language-plaintext highlighter-rouge">QTL</code></li>
        <li>HOM_MouseHumanSequence.rpt to <code class="language-plaintext highlighter-rouge">Homolog groups</code></li>
        <li>nextprot_asbtraction.ttl to <code class="language-plaintext highlighter-rouge">neXtProt abstraction</code></li>
      </ul>
    </li>
    <li>Check every datatype.
      <ul>
        <li>DE results: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li>Mus musculus annotation: <code class="language-plaintext highlighter-rouge">gff</code></li>
        <li>Gene Symbol: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li>QTL: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li>Homolog groups: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li>neXtprot abstraction: <code class="language-plaintext highlighter-rouge">ttl</code></li>
      </ul>
    </li>
  </ol>

  <p>If the datatypes are wrong, please change it.</p>

  <!--SNIPPET-->
  <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select  your desired datatype</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
</blockquote>

<p>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon and take a look at the uploaded files.</p>

<p>Two step are necessary to get our data converted into RDF triples. The first step is to upload the <span class="notranslate">Galaxy</span> datasets into the AskOmics server. The second step is to integrate the uploaded data into the RDF triplestore.</p>

<h1 id="upload-inputs-into-askomics">Upload inputs into AskOmics</h1>

<p>We will first launch an AskOmics interactive tool, and upload the data into it.</p>

<h2 id="launch-askomics-interactive-tool">Launch AskOmics Interactive Tool</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-launch-askomics-it"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Launch AskOmics IT</h3>
  <ol>
    <li><strong>AskOmics</strong> a visual SPARQL query builder <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to launch the Interactive Tool
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Datasets to load into AskOmics”</em>: <code class="language-plaintext highlighter-rouge">DE results</code>, <code class="language-plaintext highlighter-rouge">Mus musculus annotation</code>, <code class="language-plaintext highlighter-rouge">Gene Symbols</code>, <code class="language-plaintext highlighter-rouge">QTL</code>, <code class="language-plaintext highlighter-rouge">Homolog groups</code> and <code class="language-plaintext highlighter-rouge">neXtProt abstraction</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Wait a few seconds (or minutes if computing resources are busy) for AskOmics to be ready to use. A <strong>view</strong> link should appear in the confirmation box just after clicking on the <strong>Execute</strong> button.</p>

<p>AskOmics is an <em>Interactive tool</em>. It means that when you launch it, it will stay in running state (yellow background) in your History. As long as it stays in this running state, you can access it by looking in the “User” &gt; “Active Interactive Tools” menu (click on its name to view it). When you no longer need it, you can stop it by deleting it from your history, or using the “Stop” button in the “User” &gt; “Active Interactive Tools” page.</p>

<p>Keep in mind that as long as this tool runs, it uses computing resources, so don’t forget to stop it when you no longer have use for it.</p>

<p>Once the AskOmics Interactive Tool is ready, you should see a start page looking like this:</p>

<figure id="figure-1"><img src="images/home.png" alt="home page. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> AskOmics home page</figcaption></figure>

<p>You can see that there is no data available yet. It’s because data needs to be integrated: it is the next step.</p>

<h1 id="integrate-input-files-into-askomics">Integrate input files into AskOmics</h1>

<p>AskOmics conversion into RDF is called <em>integration</em>.</p>

<p>On the <strong>Files</strong> page (link at the top of the page), you will see the files you uploaded from <span class="notranslate">Galaxy</span>. We will now integrate all these files.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate data</h3>
  <ol>
    <li>Got to the <strong>Files</strong> page</li>
    <li>Select all the input files</li>
    <li>Click on the <strong>Integrate</strong> button</li>
  </ol>
</blockquote>

<figure id="figure-2"><img src="images/files.png" alt="files page. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> AskOmics files page</figcaption></figure>

<p>You will land on the <strong>Integrate</strong> page that shows a preview of the data present in each selected file, depending of its data type.</p>

<h2 id="integrate-gff-files">Integrate GFF files</h2>

<p>The GFF preview shows the entities that the file contains. We need to select the entities we want to be integrated.</p>

<p>The <code class="language-plaintext highlighter-rouge">Mus musculus annotation</code> file we’re using contains gene and m<span class="notranslate">RNA</span> entities, and we will need both in the rest of the tutorial.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-mus-musculus-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">Mus musculus annotation</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">Mus musculus annotation (preview)</code></li>
    <li>Select <code class="language-plaintext highlighter-rouge">gene</code> and <code class="language-plaintext highlighter-rouge">m<span class="notranslate">RNA</span></code></li>
    <li>Click on the <strong>Integrate (private dataset)</strong> button</li>
  </ol>
  <figure id="figure-3"><img src="images/mus_musculus_annotation_preview.png" alt="mus musculus annotation preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Mus musculus annotation preview</figcaption></figure>
</blockquote>

<h2 id="integration-of-tabular-tsv-files">Integration of tabular (TSV) files</h2>

<p>The TSV preview shows an HTML table representing the first lines of the TSV file. During integration, AskOmics will convert the file using the header.</p>

<!-- First col: entity, then, attribute -->
<p>The first column of a TSV file will be the <em>entity</em> name. Other columns of the file will be <em>attributes</em> of the <em>entity</em>. <em>Labels</em> of the <em>entity</em> and <em>attributes</em> will be set by the header. Each <em>label</em> can be edited by clicking on it.</p>

<!-- Attribute types -->
<p>Entity and attributes can have special types. The types are defined with the select box below the header. An <em>entity</em> can be a <em>start entity</em> or an <em>entity</em>. A <em>start entity</em> means that the entity may be used to start a query on the AskOmics homepage.</p>

<p>Attributes can take the following types:</p>
<ul>
  <li>Numeric: if all the values of the column are numeric</li>
  <li>Text: if all the values are strings</li>
  <li>Category: if there is a limited number of repeated values (e.g. ‘green’, ‘yellow’ and ‘red’, each one found in multiple lines)</li>
</ul>

<p>If the entity describes a locatable element on a genome:</p>
<ul>
  <li>Reference: chromosome</li>
  <li>Strand: strand</li>
  <li>Start: start position</li>
  <li>End: end position</li>
</ul>

<!-- Relation -->
<p>A column can also represent a relation between the <em>entity</em> to another. In this case, the header have to be named <code class="language-plaintext highlighter-rouge">relationName@TargetedEntity</code> and the type <em>Directed</em> or <em>Symetric</em> relation. A <em>Directed</em> relation is a relation from this entity to the targeted one (e.g. A is B’s father, but B is not A’s father). A <em>Symetric relation</em> is a relation that works in both directions (e.g. A loves B, and B loves A).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-de-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">DE results</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">DE results (preview)</code></li>
    <li>Edit attribute names and types:
      <ul>
        <li>change <code class="language-plaintext highlighter-rouge">ENTREZ ID</code> to <code class="language-plaintext highlighter-rouge">Differential <span class="notranslate">Expression</span></code> and set type to <em>start entity</em></li>
        <li>change <code class="language-plaintext highlighter-rouge">SYMBOL</code> to <code class="language-plaintext highlighter-rouge">linkedTo@Gene Symbol</code> and set type to <em>Symetric relation</em></li>
        <li>change <code class="language-plaintext highlighter-rouge">GENENAME</code> to <code class="language-plaintext highlighter-rouge">name</code> and set type to <em>text</em></li>
        <li>Keep the other column names and set their types to <em>numeric</em></li>
      </ul>
    </li>
    <li><strong>Integrate (private dataset)</strong></li>
  </ol>
  <figure id="figure-4"><img src="images/de_results_preview.png" alt="De results preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> DE results preview</figcaption></figure>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-gene-symbols"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">Gene symbols</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">Gene symbols (preview)</code></li>
    <li>Edit attribute names and types:
      <ul>
        <li>change <code class="language-plaintext highlighter-rouge">symbol</code> to <code class="language-plaintext highlighter-rouge">Gene Symbol</code> and set type to <em>entity</em></li>
        <li>Set <code class="language-plaintext highlighter-rouge">to mouse genes@gene</code> type to <em>Symetric relation</em></li>
      </ul>
    </li>
    <li>Click on the <strong>Integrate (private dataset)</strong> button</li>
  </ol>
  <figure id="figure-5"><img src="images/symbol_to_ensembl_preview.png" alt="Gene symbols preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Symbol to Ensembl preview</figcaption></figure>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-qtl"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">QTL</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">QTL (preview)</code></li>
    <li>Edit attribute names and types:
      <ul>
        <li>change <code class="language-plaintext highlighter-rouge">Input</code> to <code class="language-plaintext highlighter-rouge">QTL</code> and set type to <em>start entity</em></li>
        <li>set <code class="language-plaintext highlighter-rouge">Chr</code> type to <em>Reference</em></li>
        <li>set <code class="language-plaintext highlighter-rouge">Start</code> type to <em>Start</em></li>
        <li>set <code class="language-plaintext highlighter-rouge">End</code> type to <em>End</em></li>
      </ul>
    </li>
    <li>Click on the <strong>Integrate (private dataset)</strong> button</li>
  </ol>
  <figure id="figure-6"><img src="images/qtl_preview.png" alt="QTL preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> QTL preview</figcaption></figure>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-homolog-groups"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">Homolog groups</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">Homolog groups (preview)</code></li>
    <li>Edit attribute names and types:
      <ul>
        <li>change <code class="language-plaintext highlighter-rouge">HomoloGene ID</code> to <code class="language-plaintext highlighter-rouge">Homolog Group</code> and set type to <em>start entity</em></li>
        <li>set <code class="language-plaintext highlighter-rouge">Common Organism Name</code> type to <em>category</em></li>
        <li>change <code class="language-plaintext highlighter-rouge">Symbol</code> to <code class="language-plaintext highlighter-rouge">linkedTo@Gene Symbol</code> and set type to <em>Directed relation</em></li>
        <li>Keep the other column names and set their types to <em>text</em></li>
      </ul>
    </li>
    <li>Click on the <strong>Integrate (private dataset)</strong> button</li>
  </ol>
  <figure id="figure-7"><img src="images/homolog_groups.png" alt="Homolog groups preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Homolog groups preview</figcaption></figure>
</blockquote>

<h2 id="integration-of-rdf-files">Integration of RDF files</h2>

<p>The last dataset we want to integrate is the neXtProt abstraction. This file contains some RDF data that instructs AskOmics how to communicate with a remote RDF database containing <a href="https://www.nextprot.org/">neXtProt</a> data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-integrate-nextprot-abstraction"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Integrate <code class="language-plaintext highlighter-rouge">neXtProt abstraction</code></h3>
  <ol>
    <li>Search for <code class="language-plaintext highlighter-rouge">neXtProt abstraction (preview)</code></li>
    <li>Check that <strong>Distant endpoint</strong> is set to <code class="language-plaintext highlighter-rouge">https://sparql.nextprot.org/sparql</code> in <strong>advanced options</strong></li>
    <li>Click on the <strong>Integrate (private dataset)</strong> button</li>
  </ol>
  <figure id="figure-8"><img src="images/nextprot_abstraction_preview.png" alt="neXtProt abstraction preview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> neXtProt abstraction preview</figcaption></figure>
</blockquote>

<h2 id="monitor-integration">Monitor integration</h2>

<p>Integration can take some time depending on the file size. The <strong>Datasets</strong> page shows the progress.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-track-integration-progress"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: track integration progress</h3>
  <ol>
    <li>Go to the <strong>Dataset</strong> page</li>
    <li>Wait for all datasets to be in <em>success</em> state</li>
  </ol>
  <figure id="figure-9"><img src="images/datasets.png" alt="dataset. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Datasets table</figcaption></figure>
</blockquote>

<h1 id="query">Query</h1>

<p>Once all the data of interest is integrated (converted to RDF), its time to query them. Querying RDF data is done by using the SPARQL language. Fortunately, AskOmics provides a user-friendly interface to build SPARQL queries without having to learn the SPARQL language.</p>

<h2 id="query-builder-overview">Query builder overview</h2>

<h3 id="simple-query">Simple query</h3>

<p>The first step to build a query is to choose a start point for the query.</p>

<figure id="figure-10"><img src="images/ask.png" alt="ask. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> ask page</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-start-a-query"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Start a query</h3>
  <ol>
    <li>Go to the <strong>Ask!</strong> page</li>
    <li>Select the <em>Differential <span class="notranslate">Expression</span></em> entity</li>
    <li><strong>Start!</strong></li>
  </ol>
</blockquote>

<p>Once the start entity is chosen, the query builder is displayed.</p>

<p>The query builder is composed of a graph. Nodes represents <em>entities</em> and links represents <em>relations</em> between entities. The selected entity is surrounded by a red circle. Links and other entities are dotted and lighter because there are not instantiated.</p>

<figure id="figure-11"><img src="images/query_builder.png" alt="query builder. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> Query builder, Differential <span class="notranslate">Expression</span> is the selected entity, Gene Symbol is a suggested entity</figcaption></figure>

<p>On the right, attributes of the selected entity are displayed as attribute boxes. Each box has an eye icon: an opened eye means the attribute will be displayed on the results.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ask-for-all-differential-expression-and-display-some-attributes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Ask for all Differential <span class="notranslate">Expression</span> and display some attributes</h3>
  <ol>
    <li>Display <code class="language-plaintext highlighter-rouge">logFC</code> and <code class="language-plaintext highlighter-rouge">adj.P.val</code> by clicking on the eye icon</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
  <figure id="figure-12"><img src="images/preview_results.png" alt="preview results. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> Results preview</figcaption></figure>
</blockquote>

<p><strong>Run &amp; preview</strong> launch the query with a limit of 30 rows returned. We use this button to get an idea of the results returned.</p>

<h3 id="filter-on-attributes">Filter on attributes</h3>

<p>Next query will search for all over-expressed genes. Genes are considered over-expressed if the log fold change is &gt; 2. We are only interested by significant results (Adj P value ≤ 0.05)</p>

<p>Back to the query builder,</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-attributes-to-get-significant-over-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter attributes to get significant over-expressed genes</h3>
  <ol>
    <li>Filter <code class="language-plaintext highlighter-rouge">logFC</code> with <code class="language-plaintext highlighter-rouge">&gt; 2</code></li>
    <li>Filter <code class="language-plaintext highlighter-rouge">adj.P.val</code> with <code class="language-plaintext highlighter-rouge">≤ 0.05</code></li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
  <figure id="figure-13"><img src="images/preview_results_filtered.png" alt="preview filtered results. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Filtered results preview</figcaption></figure>
</blockquote>

<p>The preview shows only significantly over-expressed genes.</p>

<h3 id="filter-on-relations">Filter on relations</h3>

<p>Now that we have found our genes of interest, we will link these genes to the reference genome to get information about their location.</p>

<p>To constraint on relation, we have to click on suggested nodes, linked to our entity of interest.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-link-de-results-to-gene-using-gene-symbol"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Link <code class="language-plaintext highlighter-rouge">De results</code> to <code class="language-plaintext highlighter-rouge">gene</code> using <code class="language-plaintext highlighter-rouge">Gene Symbol</code></h3>
  <ol>
    <li>First, hide <code class="language-plaintext highlighter-rouge">Label</code>, <code class="language-plaintext highlighter-rouge">logFC</code> and <code class="language-plaintext highlighter-rouge">adj.P.val</code> of <code class="language-plaintext highlighter-rouge">Differential <span class="notranslate">Expression</span></code> using the eye icon</li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">Gene Symbol</code> by clicking on the suggested node, and hide his <code class="language-plaintext highlighter-rouge">Label</code> using the eye icon</li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">gene</code> by clicking on the <code class="language-plaintext highlighter-rouge">gene</code> node</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
  <figure id="figure-14"><img src="images/gene_query.png" alt="gene query. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> Gene query</figcaption></figure>
</blockquote>

<p>Results now show the Ensembl id of our over-expressed genes. We have now access to all the information about the <code class="language-plaintext highlighter-rouge">gene</code> entity contained in the GFF file. For example, we can filter on chromosome and display chromosome and strand to get information about gene location.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-gene"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter <code class="language-plaintext highlighter-rouge">gene</code></h3>
  <ol>
    <li>Show <code class="language-plaintext highlighter-rouge">reference</code> and <code class="language-plaintext highlighter-rouge">strand</code> using the eye icon</li>
    <li>Filter <code class="language-plaintext highlighter-rouge">reference</code> by selecting <code class="language-plaintext highlighter-rouge">X</code> chromosome</li>
    <li>Filter <code class="language-plaintext highlighter-rouge">strand</code> by selecting <code class="language-plaintext highlighter-rouge">+</code> strand</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
  <figure id="figure-15"><img src="images/gene_filter.png" alt="gene filter. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> Gene filtering</figcaption></figure>
</blockquote>

<h3 id="query-on-the-position-of-elements-on-the-genome">Query on the position of elements on the genome.</h3>

<p>AskOmics is able to perform special queries between entities that are locatable. These queries are:</p>

<ul>
  <li>Entities overlapping another one</li>
  <li>Entities included in another entity</li>
</ul>

<blockquote class="details">
  <h3 id="details-faldo-ontology"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> FALDO ontology</h3>
  <p>The <a href="https://bioportal.bioontology.org/ontologies/FALDO">FALDO</a> ontology describes sequence feature positions and regions. AskOmics uses FALDO ontology to represent entity positions. GFF are using FALDO, as well as TSV entities with chromosome, strand, start and end.</p>
</blockquote>

<p>On the query builder interface, locatable entities are represented with a green circle and relations based on location are represented as green arrow.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-gene-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter <code class="language-plaintext highlighter-rouge">gene</code></h3>
  <ol>
    <li>First, remove the reference filter (unselect <code class="language-plaintext highlighter-rouge">X</code> using <code class="language-plaintext highlighter-rouge">ctrl</code>+<code class="language-plaintext highlighter-rouge">click</code>)</li>
    <li>Remove the strand filter (unselect <code class="language-plaintext highlighter-rouge">+</code> using <code class="language-plaintext highlighter-rouge">ctrl</code>+<code class="language-plaintext highlighter-rouge">click</code>)</li>
    <li>Hide <code class="language-plaintext highlighter-rouge">reference</code> <code class="language-plaintext highlighter-rouge">strand</code> using the eye</li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">QTL</code></li>
    <li>Click on the link between <code class="language-plaintext highlighter-rouge">gene</code> and <code class="language-plaintext highlighter-rouge">QTL</code> to edit the relation</li>
    <li>Check that the relation is <code class="language-plaintext highlighter-rouge">gene</code> <code class="language-plaintext highlighter-rouge">included in</code> <code class="language-plaintext highlighter-rouge">QTL</code> <code class="language-plaintext highlighter-rouge">on the same reference</code> with <code class="language-plaintext highlighter-rouge">strict</code> ticked</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
  <figure id="figure-16"><img src="images/query_builder_position.png" alt="FALDO query. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 16:</span> FALDO query</figcaption></figure>
</blockquote>

<p>To go further, we can filter on <code class="language-plaintext highlighter-rouge">QTL</code> to refine the results.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-gene-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter <code class="language-plaintext highlighter-rouge">gene</code></h3>
  <ol>
    <li>Go back to the <code class="language-plaintext highlighter-rouge">QTL</code> node</li>
    <li>Show the <code class="language-plaintext highlighter-rouge">Name</code> attribute using the eye icon</li>
    <li>Filter the name with a <code class="language-plaintext highlighter-rouge">regexp</code> with <code class="language-plaintext highlighter-rouge">growth</code></li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
</blockquote>

<p>From now, our query is “All Genes that are over-expressed (logFC &gt; 2 and FDR ≤ 0.05) and located on a QTL that is related to growth”. We can save this results with the <em>Run &amp; save</em> button.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-save-a-result"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Save a result</h3>
  <ol>
    <li><strong>Run &amp; save</strong></li>
  </ol>
</blockquote>

<h3 id="use-nextprot-distant-data-to-refine-results">Use neXtProt distant data to refine results</h3>

<p><a href="https://www.nextprot.org">neXtProt</a> is a comprehensive human-centric discovery platform, offering its users a seamless integration of and navigation through protein-related data. It offer a <a href="https://sparql.nextprot.org/sparql">SPARQL endpoint</a> that can be interrogated with AskOmics.</p>

<p>Since we added the neXtProt abstraction into our AskOmics instance, we can link our data to neXtProt.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-find-human-homolog-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Find human homolog genes</h3>
  <ol>
    <li>Go back to the <code class="language-plaintext highlighter-rouge">gene</code> node</li>
    <li>instantiate <code class="language-plaintext highlighter-rouge">Gene Symbol</code> and hide his <code class="language-plaintext highlighter-rouge">Label</code></li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">Homolog Group</code>, hide his label and filter his <code class="language-plaintext highlighter-rouge">Common Organism Name</code> with <code class="language-plaintext highlighter-rouge">human</code></li>
  </ol>
  <figure id="figure-17"><img src="images/filter_organism_name.png" alt="Organism name filter. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 17:</span> Organism name filter</figcaption></figure>
  <ol>
    <li>From <code class="language-plaintext highlighter-rouge">Homolog Group</code>, instantiate another <code class="language-plaintext highlighter-rouge">Gene Symbol</code> and hide his <code class="language-plaintext highlighter-rouge">Label</code></li>
    <li>Finally, follow the <code class="language-plaintext highlighter-rouge">to neXtProt Gene</code> link and instantiate <code class="language-plaintext highlighter-rouge">Gene</code> (with a capital G)</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
</blockquote>

<p>The query we’ve just built asks for the human homologs of our over-expressed genes. We use the <code class="language-plaintext highlighter-rouge">Gene Symbol</code> to get information from the neXtProt database. AskOmics converts the query into small SPARQL subqueries and send them to the local database and to the remote neXtProt endpoint.</p>

<p>Now we are linked to the neXtProt database, we can obtain information about the proteins encoded by these genes, as well as their location in the cell.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-protein-and-their-location"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the protein and their location</h3>
  <ol>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">Entry</code></li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">Isoform</code> and hide the <code class="language-plaintext highlighter-rouge">Label</code></li>
    <li>Many nodes are connected to <code class="language-plaintext highlighter-rouge">Isoform</code>. Use the <em>Filter links</em> field to filter nodes linked with a link named <code class="language-plaintext highlighter-rouge">location</code></li>
  </ol>
  <figure id="figure-18"><img src="images/filter_links.png" alt="Filter links. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 18:</span> Filter links</figcaption></figure>
  <ol>
    <li>Instantiate the <code class="language-plaintext highlighter-rouge">Subcellular Location</code> node and hide <code class="language-plaintext highlighter-rouge">Uri</code></li>
    <li>Use the <em>Filter node</em> field to filter nodes with “location” in their name</li>
    <li>Instantiate <code class="language-plaintext highlighter-rouge">Uniprot subcellular Location CV</code> (you can use the node filter to clear up the screen)</li>
    <li><strong>Run &amp; preview</strong></li>
  </ol>
</blockquote>

<p>Finally, our query is “All genes that are over-expressed and located on a QTL that is related to growth, their human homologs and the location of the proteins coded by this genes”. We will save it to the results.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-save-a-result-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Save a result</h3>
  <ol>
    <li><strong>Run &amp; save</strong></li>
  </ol>
</blockquote>

<h2 id="results-management">Results management</h2>

<p>The results page displays the saved queries. Queries are sorted by creation date. At the end of the table, action buttons can be used to preview the result, download or send it to <span class="notranslate">Galaxy</span> history.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-edit-query-name"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Edit query name</h3>
  <ol>
    <li>Go to the <em>Results</em> page</li>
    <li>Use the <em>Preview</em> button to check the result</li>
    <li>Click on the name to rename the two query with <code class="language-plaintext highlighter-rouge">Over-expressed genes on a growth QTL</code> and <code class="language-plaintext highlighter-rouge">Over-expressed genes on a growth QTL, their human homologs and protein location</code> (press enter key to validate)</li>
  </ol>
</blockquote>

<figure id="figure-19"><img src="images/results_table.png" alt="preview table. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 19:</span> Results table</figcaption></figure>

<p>The <strong>Action</strong> column contain button to perform certain action:</p>

<ul>
  <li>Preview: show a results preview on the bottom of the table</li>
  <li>Download: Download the results (TSV file)</li>
  <li>Edit: Edit the query with the query builder</li>
  <li>SPARQL: edit the query with a SPARQL editor for advanced users</li>
  <li>Send results to <span class="notranslate">Galaxy</span>: send the results (TSV file) to the most recently used <span class="notranslate">Galaxy</span> history</li>
  <li>Send query to <span class="notranslate">Galaxy</span>: send the query representation (json file) to the most recently used <span class="notranslate">Galaxy</span> history</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-send-results-to-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Send results to <span class="notranslate">Galaxy</span></h3>
  <ol>
    <li>Click on <strong>Send results to <span class="notranslate">Galaxy</span></strong> on each query to send them to the last used <span class="notranslate">Galaxy</span> history</li>
    <li>Get back to <span class="notranslate">galaxy</span> and wait for the dataset (reload if needed)</li>
  </ol>
</blockquote>

<figure id="figure-20"><img src="images/galaxy_history_results.png" alt="history results. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 20:</span> <span class="notranslate">Galaxy</span> history with the two results</figcaption></figure>

<p>Now that you have used AskOmics to generate this final tabular file, you can continue analysing it with other <span class="notranslate">Galaxy</span> tools.
If you are done, don’t forget to close the AskOmics instance by going to the “User” &gt; “Active Interactive Tools” page.</p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we have seen how to use AskOmics Interactive Tool. We launch the tools with a set of input files, then we have integrated these files into RDF and finally, we built complex queries over this local datasets and neXtProt to answer a biological question.</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>AskOmics helps to integrate multiple data sources into an RDF database without prior knowledge in the Semantic Web</p>
</li>
                        
                        <li><p>It can also be connected to external SPARQL endpoints, such as <a href="https://www.nextprot.org">neXtProt</a></p>
</li>
                        
                        <li><p>AskOmics helps to perform complex SPARQL queries without knowing SPARQL, using a user-friendly interface</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                

                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=RNA-Seq analysis with AskOmics Interactive Tool (Transcriptomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Xavier Garnier, Anthony Bretaudeau, Anne Siegel, Olivier Dameron, Mateo Boudet, 2021 <b>RNA-Seq analysis with AskOmics Interactive Tool (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-rna-seq-analysis-with-askomics-it,
author = "Xavier Garnier and Anthony Bretaudeau and Anne Siegel and Olivier Dameron and Mateo Boudet",
title = "RNA-Seq analysis with AskOmics Interactive Tool (Galaxy Training Materials)",
year = "2021",
month = "05",
day = "21"
url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
