<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1702168203' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1702168203' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Hi-C analysis of Drosophila melanogaster cells using HiCExplorer">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-11-28 09:55:50 +0000">
<meta name="DC.creator" content="Joachim Wolff">
<meta name="DC.creator" content="Fidel Ramirez">
<meta name="DC.creator" content="Vivek Bhardwaj">
<meta name="DC.creator" content="Ekaterina Polkh">

        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Hi-C analysis of Drosophila melanogaster cells using HiCE...">
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2022-11-28 09:55:50 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Hi-C analysis of Drosophila melanogaster cells using HiCExplorer' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html",
  "timeRequired": "PT1H",
  "keywords": "epigenetics",
  "description": "The questions this  addresses are:\n - Why is a Hi-C analysis useful?\n - What is 'chromosome conformation capture'?\n - What are main steps in order to generate and plot a Hi-C contact matrix?\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/joachimwolff/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/joachimwolff/",
      "name": "Joachim Wolff",
      "image": "https://avatars.githubusercontent.com/joachimwolff",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "name": "Fidel Ramirez",
      "image": "https://avatars.githubusercontent.com/fidelram",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "name": "Vivek Bhardwaj",
      "image": "https://avatars.githubusercontent.com/vivekbhr",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/polkhe/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/polkhe/",
      "name": "Ekaterina Polkh",
      "image": "https://avatars.githubusercontent.com/polkhe",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1702168203'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Epigenetics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/epigenetics/faqs/" title="Check our FAQs for the Epigenetics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/hicexplorer/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-epigenetics">
    <h1 data-toc-skip>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/joachimwolff/" class="contributor-badge contributor-joachimwolff"><img src="https://avatars.githubusercontent.com/joachimwolff?s=36" alt="Avatar" width="36" height="36">Joachim Wolff</a><a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge contributor-fidelram"><img src="https://avatars.githubusercontent.com/fidelram?s=36" alt="Avatar" width="36" height="36">Fidel Ramirez</a><a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge contributor-vivekbhr"><img src="https://avatars.githubusercontent.com/vivekbhr?s=36" alt="Avatar" width="36" height="36">Vivek Bhardwaj</a><a href="/training-material/hall-of-fame/polkhe/" class="contributor-badge contributor-polkhe"><img src="https://avatars.githubusercontent.com/polkhe?s=36" alt="Avatar" width="36" height="36">Ekaterina Polkh</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Why is a Hi-C analysis useful?</p>
</li>
        
        <li><p>What is ‘chromosome conformation capture’?</p>
</li>
        
        <li><p>What are main steps in order to generate and plot a Hi-C contact matrix?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.1183661">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/epigenetics/tutorials/hicexplorer/workflows/" title="Hi-C analysis of Drosophila melanogaster cells using HiCExplorer workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/hicexplorer/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 28, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00141">https://gxy.io/GTN:T00141</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>In this HiCExplorer tutorial we will generate and plot a Hi-C contact matrix.
For this the following steps are necessary to be performed:</p>
<ol>
  <li>Map the Hi-C reads to the reference genome</li>
  <li>Creation of a Hi-C matrix</li>
  <li>Plotting the Hi-C matrix</li>
  <li>Correction of Hi-C matrix</li>
  <li>TAD Calling</li>
  <li>A/B compartments computation</li>
  <li>pyGenomeTracks visualization</li>
  <li>Loop detection</li>
</ol>

<p>After a corrected Hi-C matrix is created other tools can be used to visualize it, call TADS or compare it with other matrices.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#reads-mapping" id="markdown-toc-reads-mapping">Reads mapping</a></li>
  <li><a href="#creation-of-a-hi-c-matrix" id="markdown-toc-creation-of-a-hi-c-matrix">Creation of a Hi-C matrix</a></li>
  <li><a href="#plotting-the-hi-c-matrix" id="markdown-toc-plotting-the-hi-c-matrix">Plotting the Hi-C matrix</a></li>
  <li><a href="#correction-of-hi-c-matrix" id="markdown-toc-correction-of-hi-c-matrix">Correction of Hi-C matrix</a></li>
  <li><a href="#tad-calling" id="markdown-toc-tad-calling">TAD calling</a></li>
  <li><a href="#ab-compartments-computation" id="markdown-toc-ab-compartments-computation">A/B compartments computation</a></li>
  <li><a href="#integrating-hi-c-and-other-data" id="markdown-toc-integrating-hi-c-and-other-data">Integrating Hi-C and other data</a></li>
  <li><a href="#loop-detection" id="markdown-toc-loop-detection">Loop detection</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import from <a href="https://doi.org/10.5281/zenodo.1183661">Zenodo</a>.</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data via links<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>
            <p>Select <strong>Paste/Fetch Data</strong></p>

            <p>Below are the links to the read files that can be copied and pasted in the upload manager.</p>

            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1183661/files/HiC_S2_1p_10min_lowU_R1.fastq.gz
https://zenodo.org/record/1183661/files/HiC_S2_1p_10min_lowU_R2.fastq.gz
</code></pre></div>            </div>
          </li>
          <li>Paste the links into the text field</li>
          <li>Press <strong>Start</strong></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Rename the data set to something meaningful, e.g. <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code> and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code>.
By default, when data is imported via its link, Galaxy names it with its URL.</p>
    </li>
  </ol>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-get-data-from-public-sources"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Get data from public sources</div>
    <p>HiCExplorer needs as input the forward and reverse strand of a pair end read which are mapped independently. A usual start point for a typical analysis is the given GSE number of a publication, e.g. GSE63525 for Rao 2014. To get the actual data, go to <a href="https://www.ncbi.nlm.nih.gov/geo/">NCBI</a> and search for the <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63525">GSE number</a>. In the section ‘Samples’ the GSM numbers of all samples are given. Select the correct one for you, and go to the <a href="https://www.ebi.ac.uk/ena">European Nucleotide Archive</a> and enter the GSM number. Select a matching result e.g. <a href="https://www.ebi.ac.uk/ena/data/view/SRX764936">SRX764936</a> and download the data given in the row ‘FASTQ files (FTP)’ the forward and reverse strand.
It is important to have the forward and reverse strand individual as a FASTQ file and to map it individually, HiCExplorer can not work with interleaved files.</p>
  </blockquote>

</blockquote>

<h1 id="reads-mapping">Reads mapping</h1>

<p>Mates have to be mapped individually to avoid mapper specific heuristics designed for standard paired-end libraries.</p>

<p>We have used the HiCExplorer successfully with bwa, bowtie2 and hisat2. In this tutorial we will be using <strong>Map with BWA-MEM</strong> tool. It is important to remember to:</p>
<ul>
  <li>use local mapping, in contrast to end-to-end. A fraction of Hi-C reads are chimeric and will not map end-to-end thus, local mapping is important to increase the number of mapped reads</li>
  <li>tune the aligner parameters to penalize deletions and insertions. This is important to avoid aligned reads with gaps if they happen to be chimeric.</li>
  <li>If bowtie2 or hisat2 are used, <code class="language-plaintext highlighter-rouge">--reorder</code> option and as a file format <code class="language-plaintext highlighter-rouge">bam_native</code> needs to be used. Regular <code class="language-plaintext highlighter-rouge">bam</code> files are sorted by Galaxy and can not be used as an input for HiCExplorer.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping reads</div>

  <ol>
    <li><strong>Map with Bowtie</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run Bowtie on both strands <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code> and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code> with:
      <ul>
        <li>“Is this single or paired library” to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>Set multiple data sets</li>
        <li>“FASTQ file” to <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code>and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code></li>
        <li>“Will you select a reference genome from your history or use a built-in index?” to <code class="language-plaintext highlighter-rouge">Use a built-in index</code></li>
        <li>“Select a reference genome” to <code class="language-plaintext highlighter-rouge">dm3</code></li>
        <li>“Do you want to tweak SAM/BAM Options?” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>“Reorder output to reflect order of the input file” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Rename the output of the tool according to the corresponding files: <code class="language-plaintext highlighter-rouge">R1.bam</code> and <code class="language-plaintext highlighter-rouge">R2.bam</code></li>
  </ol>

</blockquote>

<h1 id="creation-of-a-hi-c-matrix">Creation of a Hi-C matrix</h1>

<p>Once the reads have been mapped the Hi-C matrix can be built.</p>

<p>For this step we will use <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicBuildMatrix.html#hicbuildmatrix">hicBuildMatrix</a> tool, which builds the matrix of read counts over the bins in the genome, considering the sites around the given restriction site.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-hicbuildmatrix"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: hicBuildMatrix</div>

  <ol>
    <li><strong>hicBuildMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicBuildMatrix on the <code class="language-plaintext highlighter-rouge">R1.bam</code> and <code class="language-plaintext highlighter-rouge">R2.bam</code> from previous step with modifying the following parameters:
      <ul>
        <li>“1: Sam/Bam files to process” to <code class="language-plaintext highlighter-rouge">R1.bam</code></li>
        <li>“2: Sam/Bam files to process” to <code class="language-plaintext highlighter-rouge">R2.bam</code></li>
        <li>“Choose to use a restriction cut file or a bin size” to <code class="language-plaintext highlighter-rouge">Bin size</code></li>
        <li>“Bin size in bp” to <code class="language-plaintext highlighter-rouge">10000</code></li>
        <li>
          <p>“Sequence of the restriction site” to <code class="language-plaintext highlighter-rouge">GATC</code></p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

            <p><em>hicBuildMatrix</em> creates two files, a bam file containing only the valid Hi-C read pairs and a matrix containing the Hi-C contacts at the given resolution. The bam file is useful to check the quality of the Hi-C library on the genome browser. A good Hi-C library should contain piles of reads near the restriction fragment sites. In the QC folder a html file is saved with plots containing useful information for the quality control of the Hi-C sample like the number of valid pairs, duplicated pairs, self-ligations etc. Usually, only 25%-40% of the reads are valid and used to build the Hi-C matrix mostly because of the reads that are on repetitive regions that need to be discarded.</p>
          </blockquote>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

            <p>Normally 25% of the total reads are selected. The output matrices have counts for the genomic regions. The extension of output matrix files is .h5.
A quality report is created in e.g. <code class="language-plaintext highlighter-rouge">hicMatrix/R1_10kb_QC</code>, have a look at the report hicQC.html.</p>
          </blockquote>
        </li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code>.</li>
  </ol>

</blockquote>

<h1 id="plotting-the-hi-c-matrix">Plotting the Hi-C matrix</h1>

<p>A 10kb bin matrix is too large to plot, it’s better to reduce the resolution. We usually run out of memory for a 1 kb or a 10 kb bin matrix and the time to plot it is very long (minutes instead of seconds). In order to reduce the resolution we use the tool <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicMergeMatrixBins.html#hicmergematrixbins">hicMergeMatrixBins</a>.</p>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicMergeMatrixBins.html#hicmergematrixbins">hicMergeMatrixBins</a> merges the bins into larger bins of given number (specified by –numBins). We will merge 100 bins in the original (uncorrected) matrix and then correct it. The new bin size is going to be 10.000 bp * 100 = 1.000.000 bp = 1 Mb</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-hicmergematrixbins"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: hicMergeMatrixBins</div>

  <ol>
    <li><strong>hicMergeMatrixBins</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicMergeMatrixBins on the output from previous step setting the following parameter:
      <ul>
        <li>“Number of bins to merge” to <code class="language-plaintext highlighter-rouge">100</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output to <code class="language-plaintext highlighter-rouge">1 MB contact matrix</code>.</p>
    </li>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on the output from hicMergeMatrixBins <code class="language-plaintext highlighter-rouge">1 MB contact matrix</code> adjusting the parameters:
      <ul>
        <li>“Plot title” to <code class="language-plaintext highlighter-rouge">Hi-C matrix for dm3</code></li>
        <li>“Remove masked bins from the matrix” to <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>“Plot the log1p of the matrix values: <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>“Chromosomes to include (and order to plot in)” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-log1p"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-log1p" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: log1p<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>Because of the large differences in counts found in the matrix, it is better to plot the counts using the <em>–log1p</em> option.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The resulting plot of the 1 Mb contact matrix should look like:
<a href="../../images/plotMatrix.png" rel="noopener noreferrer"><img src="../../images/plotMatrix.png" alt="TAD plot. " width="576" height="504" loading="lazy" /></a></p>

<h1 id="correction-of-hi-c-matrix">Correction of Hi-C matrix</h1>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicCorrectMatrix.html#hiccorrectmatrix">hicCorrectMatrix</a> corrects the matrix counts in an iterative manner. For correcting the matrix, it’s important to remove the unassembled scaffolds (e.g. NT_) and keep only chromosomes, as scaffolds create problems with matrix correction. Therefore we use the chromosome names (chr2R, chr2L, chr3R, chr3L, chrX) here.</p>

<p>Matrix correction works in two steps: first a histogram containing the sum of contact per bin (row sum) is produced. This plot needs to be inspected to decide the best threshold for removing bins with lower number of reads. The second steps removes the low scoring bins and does the correction.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-matrix-diagnostic"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Matrix diagnostic</div>

  <ol>
    <li><strong>hicCorrectMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicCorrectMatrix on the output from hicBuildMatrix <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code> adjusting the parameters:
      <ul>
        <li>“Range restriction (in bp)” to <code class="language-plaintext highlighter-rouge">Diagnostic plot</code></li>
        <li>“Chromosomes to include (and order to plot in)” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output of the program prints a threshold suggestion that is usually accurate but is better to revise the histogram plot. The threshold is visualized in the plot as a black vertical line.</p>

<p>In our case the distribution describes the counts per bin of a genomic distance. To remove all bins with a z-score threshold less / more than X means to remove all bins which have less / more counts than X of mean of their specific distribution in units of the standard deviation. Looking at the distribution, we can select the value of -1.6 (lower end) and 1.8 (upper end) to remove. This is given by the –filterThreshold option in hicCorrectMatrix set to ‘correct matrix’ mode.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-matrix-correction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Matrix correction</div>

  <ol>
    <li><strong>hicCorrectMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicCorrectMatrix on the original matrix <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code> adjusting the parameters:
      <ul>
        <li>“Range restriction (in bp)” to <code class="language-plaintext highlighter-rouge">Correct matrix</code></li>
        <li>“Normalize each chromosome separately” to <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>“Remove bins of low coverage” to <code class="language-plaintext highlighter-rouge">-1.6</code></li>
        <li>“Remove bins of large coverage” to <code class="language-plaintext highlighter-rouge">1.8</code></li>
        <li>“Chromosomes to include (and order to plot in)” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>”+ Insert Chromosomes to include (and order to plot in):” to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
    <li>Rename the corrected matrix to <code class="language-plaintext highlighter-rouge">10 kb corrected contact matrix</code>.</li>
  </ol>

</blockquote>

<p>It can happen that the correction stops with:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ERROR:iterative correction:*Error* matrix correction produced extremely large values.
This is often caused by bins of low counts. Use a more stringent filtering of bins.
</code></pre></div></div>

<p>This can be solved by a more stringent z-score values for the filter threshold or by a look at the plotted matrix. For example, chromosomes with 0 reads in its bins can be excluded from the correction by not defining it for the set of chromosomes that should be corrected (parameter ‘Include chromosomes’).</p>

<h3 id="plotting-the-corrected-hi-c-matrix">Plotting the corrected Hi-C matrix</h3>

<p>We can now plot chromosome 2L with the corrected matrix.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-the-corrected-hi-c-matrix"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plotting the corrected Hi-C matrix</div>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">10 kb corrected contact matrix</code> adjusting the parameters:
      <ul>
        <li>“Plot title” to <code class="language-plaintext highlighter-rouge">Hi-C matrix for dm3</code></li>
        <li>“Plot per chromosome” to <code class="language-plaintext highlighter-rouge">False</code></li>
        <li>“Plot only this region” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>“Plot the log1p of the matrix values” to <code class="language-plaintext highlighter-rouge">True</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="load-new-data">Load new data</h3>

<p>The steps so far would have led to long run times if real data would have been used. We therefore prepared a new matrix for you, <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code>. Please load it into your history.</p>

<h1 id="tad-calling">TAD calling</h1>

<p>“The partitioning of chromosomes into topologically associating domains (TADs) is an emerging concept that is reshaping our understanding of gene regulation in the context of physical organization of the genome” <a href="https://doi.org/10.1101/115063">Ramirez et al. 2017</a>.</p>

<p>TAD calling works in two steps: First HiCExplorer computes a TAD-separation score based on a z-score matrix for all bins. Then those bins having a local minimum of the TAD-separation score are evaluated with respect to the surrounding bins to assign a p-value. Then a cutoff is applied to select the bins more likely to be TAD boundaries.</p>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicFindTADs.html#hicfindtads">hicFindTADs</a> tries to identify sensible parameters but those can be change to identify more stringent set of boundaries.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-finding-tads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Finding TADs</div>

  <ol>
    <li><strong>hicFindTADs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicFindTADs on <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code> adjusting the parameters:
      <ul>
        <li>“Minimum window length (in bp) to be considered to the left and to the right of each Hi-C bin.” to <code class="language-plaintext highlighter-rouge">30000</code></li>
        <li>“Maximum window length (in bp) to be considered to the left and to the right of each Hi-C bin.” to <code class="language-plaintext highlighter-rouge">100000</code></li>
        <li>“Step size when moving from minDepth to maxDepth” to <code class="language-plaintext highlighter-rouge">10000</code></li>
        <li>“Multiple Testing Corrections” to <code class="language-plaintext highlighter-rouge">False discovery rate</code></li>
        <li>“q-value” to <code class="language-plaintext highlighter-rouge">0.05</code></li>
        <li>“Minimum threshold of the difference between the TAD-separation score of a putative boundary and the mean of the TAD-sep. score of surrounding bins.” to <code class="language-plaintext highlighter-rouge">0.001</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>As an output we get the boundaries, domains and scores separated files. We will use in the plot later only the TAD-score file.</p>

<h1 id="ab-compartments-computation">A/B compartments computation</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-computing-a-b-compartments"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Computing A / B compartments</div>

  <ol>
    <li><strong>hicPCA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPCA adjusting the parameters:
      <ul>
        <li>“Matrix to compute on” to <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code></li>
        <li>“Output file format” to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>“Return internally used Pearson matrix” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-the-pearson-matrix-and-pca-track"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plotting the pearson matrix and PCA track</div>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">pearson_matrix from PCA computation</code> adjusting the parameters:
      <ul>
        <li>“Plot title” to <code class="language-plaintext highlighter-rouge">Pearson matrix and PC1</code></li>
        <li>“Chromosomes to include” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>“Color map to use for the heatmap” to <code class="language-plaintext highlighter-rouge">gist_heat</code></li>
        <li>“Datatype of eigenvector file” to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>“Eigenvector file” to <code class="language-plaintext highlighter-rouge">hicPCA on [...] PC1</code></li>
      </ul>
    </li>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">pearson_matrix from PCA computation</code> adjusting the parameters:
      <ul>
        <li>“Plot title” to <code class="language-plaintext highlighter-rouge">Pearson matrix and PC2</code></li>
        <li>“Chromosomes to include” to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>“Color map to use for the heatmap” to <code class="language-plaintext highlighter-rouge">gist_heat</code></li>
        <li>“Datatype of eigenvector file” to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>“Eigenvector file” to <code class="language-plaintext highlighter-rouge">hicPCA on [...] PC2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><a href="../../images/pearson_pc1.png" rel="noopener noreferrer"><img src="../../images/pearson_pc1.png" alt="Pearson PC1. " width="1200" height="1275" loading="lazy" /></a></p>

<p><a href="../../images/pearson_pc2.png" rel="noopener noreferrer"><img src="../../images/pearson_pc2.png" alt="Pearson PC2. " width="1200" height="1275" loading="lazy" /></a></p>

<p>The first principal component correlates with the chromosome arms, while the second component correlates with A/B compartments.</p>

<h1 id="integrating-hi-c-and-other-data">Integrating Hi-C and other data</h1>

<p>We can plot the TADs for a given chromosomal region. For this we will use <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicPlotTADs.html">pyGenomeTracks</a>.</p>

<p>For the next step we need additional data tracks. Please load <code class="language-plaintext highlighter-rouge">dm3_genes.bed</code>, <code class="language-plaintext highlighter-rouge">H3K27me3.bw</code>, <code class="language-plaintext highlighter-rouge">H3K36me3.bw</code> and <code class="language-plaintext highlighter-rouge">H4K16ac.bw</code> to your history.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-tads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plotting TADs</div>

  <ol>
    <li><strong>pyGenomeTracks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run pyGenomeTracks adjusting the parameters:
      <ul>
        <li>“Region of the genome to limit the operation” to <code class="language-plaintext highlighter-rouge">chr2L:14500000-16500000</code></li>
        <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">TAD visualization</code>
          <ul>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">HiC dm3 chr2L:14500000-16500000</code></li>
            <li>“Matrix to compute on.” to the corrected matrix from hicCorrectMatrix step</li>
            <li>“Depth” to <code class="language-plaintext highlighter-rouge">750000</code></li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">4</code></li>
            <li>“Boundaries file” to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD domains</code></li>
            <li>“Show x labels” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bedgraph matrix track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">TAD separation score</code></li>
            <li>“Track file bedgraph format” to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD information in bm file</code></li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">4</code></li>
            <li>“Set type to lines” to ‘True’</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">PC1</code></li>
            <li>“Track file bigwig format” the first computed <code class="language-plaintext highlighter-rouge">hicPCA</code> result</li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">PC2</code></li>
            <li>“Track file bigwig format” the second computed <code class="language-plaintext highlighter-rouge">hicPCA</code> result</li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">H3K36me3</code></li>
            <li>“Track file bigwig format” to <code class="language-plaintext highlighter-rouge">H3K36me3</code></li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">H3K27me3</code></li>
            <li>“Track file bigwig format” to <code class="language-plaintext highlighter-rouge">H3K27me3</code></li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">H4K16ac</code></li>
            <li>“Track file bigwig format” to <code class="language-plaintext highlighter-rouge">H4K16ac</code></li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Gene track</code></li>
            <li>“Plot title” to <code class="language-plaintext highlighter-rouge">dm3 genes</code></li>
            <li>“Track file bedgraph format” the imported .bed file</li>
            <li>“Height” to <code class="language-plaintext highlighter-rouge">3</code></li>
            <li>“Type” to <code class="language-plaintext highlighter-rouge">genes</code></li>
            <li>“Gene rows” to <code class="language-plaintext highlighter-rouge">15</code></li>
            <li>“Color of track” to a color of your choice</li>
          </ul>
        </li>
        <li>“+Insert Include tracks in your plot”
          <ul>
            <li>“Choose style of the track” to <code class="language-plaintext highlighter-rouge">Vlines track</code></li>
            <li>“Track file bed format” to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD domains</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The resulting image should look like this one:
<a href="../../images/plotTADs.png" rel="noopener noreferrer"><img src="../../images/plotTADs.png" alt="TAD plot. " width="1133" height="502" loading="lazy" /></a></p>

<h1 id="loop-detection">Loop detection</h1>

<p>In Hi-C data, the term <code class="language-plaintext highlighter-rouge">loop</code> refers to a 3D structure which represents enhancer-promoter, gene, architectural or polycomb-mediated interactions. These interactions have the characteristics to be enriched in a single region compared to the local background. These loops are also called long-range interactions with an expected maximum distance of 2 MB (see <a href="https://doi.org/10.1016/j.cell.2014.11.021">Rao et al. 2014</a>).</p>

<p><a href="../../images/loops_bonev_cavalli.png" rel="noopener noreferrer"><img src="../../images/loops_bonev_cavalli.png" alt="Loops visualization. " width="1666" height="580" loading="lazy" /></a></p>

<p>To compute loops, we have to import a new data set from the shared library to our history: <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code> or download it via the <a href="ftp://cooler.csail.mit.edu/coolers/hg19/Rao2014-GM12878-MboI-allreps-filtered.10kb.cool">FTP cool files server</a>. (FTP links seems to be blocked, therefore the full URL: <code class="language-plaintext highlighter-rouge">ftp://cooler.csail.mit.edu/coolers/hg19/Rao2014-GM12878-MboI-allreps-filtered.10kb.cool</code>)</p>

<p>This dataset is from the human cell GM12878, mapped to hg19 and of 10 kb resolution. We use a new file because to detect loop structures the read coverage is required to be in the hunderts of million; this was not the case for the previous used drosophila dataset.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-matrix-information"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Matrix information</div>

  <ol>
    <li><strong>hicInfo</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicInfo adjusting the parameters:
      <ul>
        <li>“Select” <code class="language-plaintext highlighter-rouge">Multiple datasets</code></li>
        <li>“Matrix to compute on” to <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code> and <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>We now investigate the result of hicInfo and see that the new imported file is having 1.2 billion non-zero elements, while the drosophila Hi-C interaction matrix has around 12 million non-zero elements.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-computing-loops"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Computing loops</div>

  <ol>
    <li><strong>hicDetectLoops</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicDetectLoops adjusting the parameters:
      <ul>
        <li>“Matrix to compute on” to <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
        <li>“Peak width” to <code class="language-plaintext highlighter-rouge">6</code></li>
        <li>“Window size” to <code class="language-plaintext highlighter-rouge">10</code></li>
        <li>“P-value preselection” to <code class="language-plaintext highlighter-rouge">0.01</code></li>
        <li>“P-value” to <code class="language-plaintext highlighter-rouge">0.01</code></li>
        <li>“Chromosomes to include” to <code class="language-plaintext highlighter-rouge">chr1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The detection of the loops is based on a pre-selection of interactions, a p-value given a continuous negative binomial distribution over all interactions of a relative distance is computed. All interactions are filtered with a threshold (<code class="language-plaintext highlighter-rouge">p-value preselection</code>) to retrieve loop candidates. In a second step, the selected peak candidate is compared against its background using a Wilcoxon rank-sum test.</p>

<p>As an output we get a loop file containing the positions of both anchor points of the loop and the p-value of the used statistical test.</p>

<p><a href="../../images/loops_result.png" rel="noopener noreferrer"><img src="../../images/loops_result.png" alt="Loops computed_file. " width="638" height="485" loading="lazy" /></a></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-of-loops"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plotting of loops</div>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix adjusting the parameters:
      <ul>
        <li>“Matrix to compute on” to <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
        <li>“Plot title” to <code class="language-plaintext highlighter-rouge">Loops</code></li>
        <li>“Plot only this region” to <code class="language-plaintext highlighter-rouge">chr1:18000000-22000000</code></li>
        <li>“Plot the log1p of the matrix values” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>“Add detected loops” to <code class="language-plaintext highlighter-rouge">Computed loops</code> of the previous step</li>
        <li>“DPI for image” to <code class="language-plaintext highlighter-rouge">300</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><a href="../../images/loops_plot.png" rel="noopener noreferrer"><img src="../../images/loops_plot.png" alt="Loops result_plot. " width="2400" height="2100" loading="lazy" /></a></p>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we used HiCExplorer to analyze drosophila melanogaster cells. We mapped chimeric reads and created a contact matrix, to reduce noise this contact matrix was normalized. We showed how to visualize a contact matrix and how we can investigate topological associating domains and relate them to additional data like gene tracks. Moreover, we used a human Hi-C interaction matrix to compute loop structures.</p>

<p>To improve your learned skills we offer an additional tutorial based on mouse stem cells: <a href="http://hicexplorer.readthedocs.io/en/latest/content/example_usage.html">following work</a>.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Hi-C helps to investigate the 3D structure of the genome and to uncover folding principles of chromatin.</p>
</li>
                        
                        <li><p>In order to build a Hi-C contact matrix the reads have to be mapped to the reference genome.</p>
</li>
                        
                        <li><p>Based on a contact matrix open and closed chromatin (A/B compartments) and topological associating domains (TADs) can be computed.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/epigenetics/faqs/">FAQ page for the Epigenetics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Epigenetics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh,  <b>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{epigenetics-hicexplorer,
author = "Joachim Wolff and Fidel Ramirez and Vivek Bhardwaj and Ekaterina Polkh",
	title = "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/epigenetics/tutorials/hicexplorer/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: hicexplorer_hicbuildmatrix
  owner: bgruening
  revisions: 707f691c974c
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hiccorrectmatrix
  owner: bgruening
  revisions: f7d344dacfeb
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hicfindtads
  owner: bgruening
  revisions: 43332e51a712
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hicmergematrixbins
  owner: bgruening
  revisions: 8926550196ce
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hicpca
  owner: bgruening
  revisions: ea50f7bba657
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hicplotmatrix
  owner: bgruening
  revisions: d398f5373c70
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hicexplorer_hicplottads
  owner: bgruening
  revisions: 94fff1a6a899
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00141">https://gxy.io/GTN:T00141</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/hicexplorer/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1702168203'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>