<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Designing plasmids encoding predicted pathways by using the BASIC assembly method</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1707006564' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1707006564' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Designing plasmids encoding predicted pathways by using the BASIC assembly method">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-17 12:24:12 +0000">
<meta name="DC.creator" content="Kenza Bazi-Kabbaj">
<meta name="DC.creator" content="Thomas Duigou">
<meta name="DC.creator" content="Joan HÃ©risson">
<meta name="DC.creator" content="Guillaume Gricourt">
<meta name="DC.creator" content="Ioana Popescu">
<meta name="DC.creator" content="Jean-Loup Faulon">

        
        
        
        
        <meta name="description" content="Synthetic Biology is the engineering of biology : the del...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Designing plasmids encoding predicted pathways by using t...">
        <meta property="og:description" content="Synthetic Biology is the engineering of biology : the del...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-17 12:24:12 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Designing plasmids encoding predicted pathways by using the BASIC assembly method",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Synthetic Biology",
    "description": "Synthetic Biology is the engineering of biology : the deliberate (re)design and construction of novel biological and biologically based parts, devices and systems to perform new functions for useful purposes, that draws on principles elucidated from biology and engineering.\nBy applying these principles to living systems, Synthetic Biology overcomes mimicry and optimisation-led research and introduces a rationale and systematic approach to the construction and (re)design.\nSynthetic Biology is at the intersection of engineering, bioscience, chemistry, and information technology.\nThe goal of synthetic biology is to extend or modify the behavior of organisms and engineer them to perform new tasks. \nIn this section, the tutorials shows how to design metabolic pathways for producing the desired chemical targets.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/synthetic-biology/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Designing plasmids encoding predicted pathways by using the BASIC assembly method' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html",
  "timeRequired": "PT15M",
  "teaches": "Annotate the SBML pathway using Selenzy tool.\n - Design plasmids encoding predicted pathways using rpbasicdesign tool.\n - Generate dnabot scripts enabling the automatic build of plasmids.",
  "keywords": "synthetic-biology",
  "description": "The questions this  addresses are:\n - How to design plasmids encoding predicted pathways using BASIC assembly method ?\n - How to generate scripts enabling the automatic build of plasmids ?\n\n\nThe objectives are:\n - Annotate the SBML pathway using Selenzy tool.\n - Design plasmids encoding predicted pathways using rpbasicdesign tool.\n - Generate dnabot scripts enabling the automatic build of plasmids.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/kenza12/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/kenza12/",
      "name": "Kenza Bazi-Kabbaj",
      "image": "https://avatars.githubusercontent.com/kenza12",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-4319-5616",
      "orcid": "https://orcid.org/0000-0002-4319-5616"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/tduigou/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/tduigou/",
      "name": "Thomas Duigou",
      "image": "https://avatars.githubusercontent.com/tduigou",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-2649-2950",
      "orcid": "https://orcid.org/0000-0002-2649-2950"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/breakthewall/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/breakthewall/",
      "name": "Joan HÃ©risson",
      "image": "https://avatars.githubusercontent.com/breakthewall",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9741-0847",
      "orcid": "https://orcid.org/0000-0001-9741-0847"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/guillaume-gricourt/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/guillaume-gricourt/",
      "name": "Guillaume Gricourt",
      "image": "https://avatars.githubusercontent.com/guillaume-gricourt",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0143-5535",
      "orcid": "https://orcid.org/0000-0003-0143-5535"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/ioanagry/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/ioanagry/",
      "name": "Ioana Popescu",
      "image": "https://avatars.githubusercontent.com/ioanagry",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-2547-8935",
      "orcid": "https://orcid.org/0000-0003-2547-8935"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/jfaulon/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/jfaulon/",
      "name": "Jean-Loup Faulon",
      "image": "https://avatars.githubusercontent.com/jfaulon",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4274-2953",
      "orcid": "https://orcid.org/0000-0003-4274-2953"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Synthetic Biology",
      "description": "Synthetic Biology is the engineering of biology : the deliberate (re)design and construction of novel biological and biologically based parts, devices and systems to perform new functions for useful purposes, that draws on principles elucidated from biology and engineering.\nBy applying these principles to living systems, Synthetic Biology overcomes mimicry and optimisation-led research and introduces a rationale and systematic approach to the construction and (re)design.\nSynthetic Biology is at the intersection of engineering, bioscience, chemistry, and information technology.\nThe goal of synthetic biology is to extend or modify the behavior of organisms and engineer them to perform new tasks. \nIn this section, the tutorials shows how to design metabolic pathways for producing the desired chemical targets.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/synthetic-biology/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3895",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3895",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3895"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1707006564'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/synthetic-biology" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Synthetic Biology
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/synthetic-biology/faqs/" title="Check our FAQs for the Synthetic Biology topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-synthetic-biology">
    <h1 data-toc-skip>Designing plasmids encoding predicted pathways by using the BASIC assembly method</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/kenza12/" class="contributor-badge contributor-kenza12"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/kenza12?s=36" alt="Avatar" width="36" height="36">Kenza Bazi-Kabbaj</a><a href="/training-material/hall-of-fame/tduigou/" class="contributor-badge contributor-tduigou"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/tduigou?s=36" alt="Avatar" width="36" height="36">Thomas Duigou</a><a href="/training-material/hall-of-fame/breakthewall/" class="contributor-badge contributor-breakthewall"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/breakthewall?s=36" alt="Avatar" width="36" height="36">Joan HÃ©risson</a><a href="/training-material/hall-of-fame/guillaume-gricourt/" class="contributor-badge contributor-guillaume-gricourt"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/guillaume-gricourt?s=36" alt="Avatar" width="36" height="36">Guillaume Gricourt</a><a href="/training-material/hall-of-fame/ioanagry/" class="contributor-badge contributor-ioanagry"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/ioanagry?s=36" alt="Avatar" width="36" height="36">Ioana Popescu</a><a href="/training-material/hall-of-fame/jfaulon/" class="contributor-badge contributor-jfaulon"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/jfaulon?s=36" alt="Avatar" width="36" height="36">Jean-Loup Faulon</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to design plasmids encoding predicted pathways using BASIC assembly method ?</p>
</li>
        
        <li><p>How to generate scripts enabling the automatic build of plasmids ?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Annotate the SBML pathway using Selenzy tool.</p>
</li>
        
        <li><p>Design plasmids encoding predicted pathways using rpbasicdesign tool.</p>
</li>
        
        <li><p>Generate dnabot scripts enabling the automatic build of plasmids.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 15 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/6123887">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/" title="Designing plasmids encoding predicted pathways by using the BASIC assembly method workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        
        
        
          
          
          

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/synthetic-biology/basic_assembly_analysis/demo" title="View GTN Video Library for these Slides">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Demo (May 2022)
            </a>
        </li>
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 17, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00273">https://gxy.io/GTN:T00273</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Synthetic biology is a novel engineering discipline which requires computational tools for the design of metabolic pathways for the production of chemicals such as SynBioCAD portal which is the first Galaxy set of tools for synthetic biology and metabolic engineering (<span class="citation"><a href="#Hrisson2022">HÃ©risson <i>et al.</i> 2022</a></span>).</p>

<p>In this tutorial, we will use a set of tools from the <strong>Genetic Design - BASIC Assembly Workflow</strong> (https://galaxy-synbiocad.org) which will enable you to design plasmids implementing metabolic pathways for the bioproduction of lycopene in <em>E.coli</em> (one of the preferred host cell for microbial biochemicals production).</p>

<p>Lycopene is a potent antioxidant and has been widely used in the fields of pharmaceuticals, nutraceuticals, and cosmetics. Itâs widely found in fruits including tomato, watermelon, guava, and papaya but the extraction method of lycopene from these natural sources is expensive, complicated and cannot match the large market demand.</p>

<p>To address this demand, synthetic biology and metabolic engineering have been employed to develop microbial cell factories (e.g. <em>E.coli</em> strains) for lycopene production.</p>

<p>To design plasmids encoding lycopene bioproducing pathways, we will use the BASIC assembly method (as described in <span class="citation"><a href="#Storch2015">Storch <i>et al.</i> 2015</a></span>) which relies on orthogonal linkers and type IIs restriction enzyme cleavage to provide a robust and accurate assembly of DNA parts into plasmid constructs. From these construct definition, the workflow will generate scripts enabling the automatic build of the plasmids as well as the transformation of strains using an Opentrons liquid handler robot. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench.</p>

<p>The workflow scheme we will use is shown below. First, we will run the steps of this workflow individually so as not to neglect the understanding of the intermediate steps as well. Then, we will run the workflow automatically so that it itself retrieves the outputs from the previous step and gives them as input to the next tool.</p>

<p><a href="../../images/basic_assembly_workflow.png" rel="noopener noreferrer"><img src="../../images/basic_assembly_workflow.png" alt="Genetic Design - BASIC Assembly Workflow. " width="975" height="573" loading="lazy" /></a></p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#find-enzymes-using-selenzyme" id="markdown-toc-find-enzymes-using-selenzyme">Find enzymes using <strong>Selenzyme</strong></a></li>
  <li><a href="#generate-potential-constructs-from-sbml-file-using-basicdesign" id="markdown-toc-generate-potential-constructs-from-sbml-file-using-basicdesign">Generate potential constructs from SBML file using <strong>BasicDesign</strong></a></li>
  <li><a href="#generating-python-scripts-using-dna-bot" id="markdown-toc-generating-python-scripts-using-dna-bot">Generating python scripts using <strong>DNA-Bot</strong></a></li>
  <li><a href="#run-the-genetic-design---basic-assembly-workflow" id="markdown-toc-run-the-genetic-design---basic-assembly-workflow">Run the <strong>Genetic Design - BASIC Assembly Workflow</strong></a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>First we need to upload and prepare the following inputs to analyze:</p>

<ul>
  <li>
    <p>One SBML (Systems Biology Markup Language) file modeling a heterologous pathway producing lycopene such as those produced by the Pathway Analysis Workflow (https://galaxy-synbiocad.org).</p>
  </li>
  <li>
    <p>The <code class="language-plaintext highlighter-rouge">parts_for_lycopene.csv</code> file listing the parts to be used (linkers, backbone and promoters) in the constructions.</p>
  </li>
  <li>
    <p>Two YAML files providing two examples of settings, i.e. providing the identifiers of the laboratory equipment and the parameters to be used in the Opentrons scripts.</p>
  </li>
</ul>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create a new history for this tutorial named <em>Genetic Design - BASIC Assembly Analysis</em>.</li>
    <li>
      <p>Import the input files from <a href="https://zenodo.org/record/6123887">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/dnabot_london_settings.yaml
https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/dnabot_paris_settings.yaml
https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/parts_for_lycopene.csv
https://zenodo.org/api/files/f58fe126-0ca2-4ca5-bc4b-0063f3a8951c/rp_002_0011.xml
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the datasets if you want to shorten the file names.
Rename <code class="language-plaintext highlighter-rouge">rp_002_0011.xml</code> to <code class="language-plaintext highlighter-rouge">pathway.xml</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="find-enzymes-using-selenzyme">Find enzymes using <strong>Selenzyme</strong></h1>

<p>At first, a pathway generated by the <strong>Pathway Analysis workflow</strong> (https://galaxy-synbiocad.org) is provided as input to the Selenzyme tool (<span class="citation"><a href="#Carbonell2018">Carbonell <i>et al.</i> 2018</a></span>). Selenzyme searches for enzymes corresponding to each reaction of the pathway. It performs a reaction similarity search in the reference reaction database <a href="https://www.metanetx.org">MetaNetX</a> and outputs an updated SBML file annotated with the enzyme UniProt IDs.
The tool provides several scores that can be combined in order to define an overall score. Scores are given for reaction similarity, conservation based on a multiple sequence alignment, phylogenetic distance between source organism and host, and additional scores calculated from sequence properties as shown in the example bellow.</p>

<p><a href="../../images/selenzy_tool.png" rel="noopener noreferrer"><img src="../../images/selenzy_tool.png" alt="Selenzyme Results. " width="1836" height="695" loading="lazy" /></a></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-annotate-enzymes-with-uniprot-ids-from-a-sbml-pathway"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Annotate enzymes with Uniprot IDs from a SBML pathway</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/tduigou/selenzy/selenzy-wrapper/0.2.0" title="Selenzyme tool" aria-role="link"><strong>Selenzyme</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/tduigou/selenzy/selenzy-wrapper/0.2.0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âPathway (SBML)â</em>: Select <code class="language-plaintext highlighter-rouge">pathway.xml</code> from your current history</li>
        <li>In <em>âAdvanced Optionsâ</em>:
          <ul>
            <li><em>âHost taxon IDâ</em>: Leave the default value <code class="language-plaintext highlighter-rouge">83333</code>. This stands for using E. coli as the chassis host.</li>
            <li><em>âComma separated taxon IDs of output enzyme sequencesâ</em>: <em>enter <code class="language-plaintext highlighter-rouge">553</code> which is the taxon ID of the Pantoea ananatis strain from which we want to extact enzymes</em></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>This tool will produce annotated pathway with UniProt IDs in XML and CSV format.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How are identified the enzyme in the SBML file?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Enzyme are identified by their UniProt ID</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="generate-potential-constructs-from-sbml-file-using-basicdesign">Generate potential constructs from SBML file using <strong>BasicDesign</strong></h1>

<p>Second, the BasicDesign tool extracts enzyme IDs contained in the SBML (previously produced by Selenzyme) to generate genetic constructs compliant with the BASIC assembly approach. It uses as input an SBML file annotated with enzyme IDs for each reaction, and optionally one or several CSV files listing by their IDs the linkers, the promoters and the backbone used (<code class="language-plaintext highlighter-rouge">parts_for_lycopene.csv</code>). Example, below:</p>

<p><em>id,type,sequence,comment</em><br />
<em>L1,neutral linker,,</em><br />
<em>L2,neutral linker,,</em><br />
<em>L3,neutral linker,,</em></p>

<p>For <strong>linkers</strong>, the type annotation should be one of <em>neutral linker</em>, <em>methylated linker</em>, <em>peptide fusion linker</em> or <em>RBS linker</em>. For <strong>user parts</strong>, type should be one of <em>backbone</em> or <em>constitutive promoter</em>. Other type will raise a warning and will be omited.</p>

<p>BasicDesign converts the SBML file into CSV files describing the DNA-parts to be included into each construct (in an operon format, i.e. with only one promoter) and enumerate possible combinations of promoters, RBSs and enzymes into constructs. Depending on the numbers of enzymes per reaction, of RBSs and promoters available, and whether or not to perform CDS permutation within the operon, the number of constructs may vary.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-genetic-constructs-from-the-previously-annotated-sbml-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate genetic constructs from the previously annotated SBML file</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/tduigou/rpbasicdesign/rpbasicdesign/0.3.4" title="BasicDesign tool" aria-role="link"><strong>BasicDesign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/tduigou/rpbasicdesign/rpbasicdesign/0.3.4</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>ârpSBML fileâ: <code class="language-plaintext highlighter-rouge">uniprot_ids</code> (output of <strong>Selenzyme</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in xml format)</em></li>
        <li>In <em>âAdvanced Optionsâ</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âLinkers and user partsâ: Select <code class="language-plaintext highlighter-rouge">parts_for_lycopene.csv</code> from your current history in csv format</em></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>This tool will output a CSV constructs file (listing the constructs to be built), where each row corresponds to one construct and consists of a sequence of BASIC linker and DNA part IDs. The two other outputted CSV files provide the plate coordinates of (i) the BASIC linkers and (ii) the DNA-parts that the user will need to provide. Additionally, one SBOL (Synthetic Biology Open Language) file is produced for each construct generated. These files can be downloaded and visualized using online tools such as <a href="https://visbol.org/">VisBOL</a>.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many constructs did you get ?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>88 construct designs were generated in CSV and SBOL format.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="generating-python-scripts-using-dna-bot">Generating python scripts using <strong>DNA-Bot</strong></h1>

<p>In the last step, the DNA-Bot tool (<span class="citation"><a href="#Storch2020">Storch <i>et al.</i> 2020</a></span>) reads the list of constructs (previously produced by BasicDesign) and the DNA-parts position on the source plates and generates a set of python scripts to drive an Opentrons liquid handling robot for building the the plasmids. Optional parameters can be set by the user to define the plastic labwares to be used, and set protocol parameters such as washing or incubation times for purification step (<code class="language-plaintext highlighter-rouge">dnabot_paris_settings.yaml</code>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-dna-bot-python-scripts"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate DNA Bot python scripts</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/tduigou/dnabot/dnabot/3.1.0" title="DNA-Bot tool" aria-role="link"><strong>DNA-Bot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/tduigou/dnabot/dnabot/3.1.0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âSource Constructâ</em>: <code class="language-plaintext highlighter-rouge">Constructs</code> (output of <strong>BasicDesign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>âPlate filesâ</em>: Select these two files: <code class="language-plaintext highlighter-rouge">User parts plate</code> (output of <strong>BasicDesign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>), <code class="language-plaintext highlighter-rouge">Biolegio plate</code> (output of <strong>BasicDesign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>) with <strong>Multiple datasets</strong>.</li>
        <li>In <em>âAdvanced Optionsâ</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âYaml file providing labware IDs and parameter to be used (Optionally)â</em>: Select either <code class="language-plaintext highlighter-rouge">dnabot_paris_settings.yaml</code> or <code class="language-plaintext highlighter-rouge">dnabot_london_settings.yaml</code> from your history)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Select multiple datasets<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
  <p><!--END_SNIPPET--></p>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

    <p>This tool will output DNA-Bot scripts in tar format. You need to download it and decompress the archive. After downloading these scripts onto a computer connected to an Opentrons, one can perform the automated construction of the plasmids at the bench. Additional metadata meaningful to keep track of parameters are also outputted by the tool.
<a href="../../images/dnabot_outputs.png" rel="noopener noreferrer"><img src="../../images/dnabot_outputs.png" alt="Dnabot Outputs. " width="470" height="611" loading="lazy" /></a>
<a href="../../images/BASIC_ASSEMBLY_WF.png" rel="noopener noreferrer"><img src="../../images/BASIC_ASSEMBLY_WF.png" alt="BASIC assembly workflow. " width="1849" height="944" loading="lazy" /></a></p>
  </blockquote>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Looking at the scripts names, can you figure out the main steps of DNA-Bot ?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>DNA-Bot outputs python scripts that implement the 4 assembly steps: clip reactions, purification, assembly and strain transformation. In short, the <strong>Clip reactions</strong> step prepares the mixes for the ligation of the individual DNA parts with the linkers; the <strong>Purification</strong> step purifies the linker-ligated DNA parts using magnetic beads and the Opentrons magnetic module; the <strong>Assembly</strong> step mixes the DNA purified parts to build the final constructs; while the <strong>Transformation</strong> step transforms the chassis micro-organism with the plasmid and inoculates onto agar.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="run-the-genetic-design---basic-assembly-workflow">Run the <strong>Genetic Design - BASIC Assembly Workflow</strong></h1>

<p>In this section, you can run the Genetic Design - BASIC Assembly Workflow more easily and fastly following these instructions:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-execute-the-entire-workflow-in-one-go"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Execute the entire workflow in one go.</div>

  <ol>
    <li>
      <p>Import the workflow into Galaxy</p>

      <!--SNIPPET-->
      <blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and Launching a workflow</div>   <div class="show-when-galaxy-proxy-active"> Click here to run the  <span class="workflow" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/synthetic-biology-basic_assembly_analysis/versions/genetic_design_basic_assembly"><strong>Genetic Design - Basic Assembly Workflow</strong> <i class="fas fa-share-alt" aria-hidden="true"></i></span> workflow. </div>   <p class="hide-when-galaxy-proxy-active">Or copy the URL (e.g. via right-click) of <a href="https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/workflows/Genetic_Design_BASIC_Assembly.ga">this workflow</a>, and import the workflow into Galaxy</p>   <div class="hide-when-galaxy-proxy-active"> <!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>âArchived Workflow URLâ</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>âArchived Workflow Fileâ</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote> <!--END_SNIPPET--> </div> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see <strong>Genetic Design - Basic Assembly Workflow</strong></li>
    <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>
    <li>Provide the workflow with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âPathway (SBML)â:</em> Select <code class="language-plaintext highlighter-rouge">pathway.xml</code> from your current history</li>
        <li><em>âHost taxon IDâ</em>: Leave the default value <code class="language-plaintext highlighter-rouge">83333</code>. This stands for using E. coli as the chassis host.</li>
        <li><em>âComma separated taxon IDs of output enzyme sequencesâ</em>: enter <code class="language-plaintext highlighter-rouge">553</code> which is the taxon ID of the Pantoea ananatis strain from which we want to extact enzymes</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âLinkers and user partsâ: Select <code class="language-plaintext highlighter-rouge">parts_for_lycopene.csv</code> from your current history in csv format</em></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>âDNA-Bot settings (Optionally)â</em>: Select either <code class="language-plaintext highlighter-rouge">dnabot_paris_settings.yaml</code> or <code class="language-plaintext highlighter-rouge">dnabot_london_settings.yaml</code> from your history)</li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>All the outputs will be automatically generated and identical to the previous ones.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Knowing the reaction is not sufficient to implement a pathway, one needs to find out which enzyme is capable of catalyzing the reactions. For this, Selenzyme is used to search for enzymes using reactions as query. As a result, UniProt IDs of enzymes are stored as annotation in SBML files.</p>
</li>
                        
                        <li><p>Expressing pathways into cells involved not only gene encoding enzymes, but also additional genetic parts such as libraries of promoter(s), RBSs, terminators, â¦ From these, multiple combinations of parts are possible, and each one could produce different results. Several methods exist to assemble genetic parts together into plasmids, the BASIC Assembly approach is one of them. The BasicDesign tool generates combinations of the genetic parts that are compliant with the BASIC Assembly patterns.</p>
</li>
                        
                        <li><p>Automatizing the construction of compound producing strains necessitate to translate protocols into instructions that could be understable by lab robots. To do this, DNA-Bot reads the list of constructs and the DNA parts information and generates python instructions for the automated build of the genetic constructs using OpenTrons liquid handling robots.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/synthetic-biology/faqs/">FAQ page for the Synthetic Biology topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Storch2015">Storch, M., A. Casini, B. Mackrow, T. Fleming, H. Trewhitt <i>et al.</i>, 2015 <b>BASIC: A New Biopart Assembly Standard for Idempotent Cloning Provides Accurate,  Single-Tier DNA Assembly for Synthetic Biology</b>. ACS Synthetic Biology 4: 781â787. <a href="https://doi.org/10.1021/sb500356d">10.1021/sb500356d</a></li>
<li id="Carbonell2018">Carbonell, P., J. Wong, N. Swainston, E. Takano, N. J. Turner <i>et al.</i>, 2018 <b>Selenzyme: enzyme selection tool for pathway design</b> (O. Stegle, Ed.). Bioinformatics 34: 2153â2154. <a href="https://doi.org/10.1093/bioinformatics/bty065">10.1093/bioinformatics/bty065</a></li>
<li id="Storch2020">Storch, M., M. C. Haines, and G. S. Baldwin, 2020 <b>DNA-BOT: a low-cost,  automated DNA assembly platform for synthetic biology</b>. Synthetic Biology 5: <a href="https://doi.org/10.1093/synbio/ysaa010">10.1093/synbio/ysaa010</a></li>
<li id="Hrisson2022">HÃ©risson, J., T. Duigou, M. du Lac, K. Bazi-Kabbaj, M. S. Azad <i>et al.</i>, 2022 <b>Galaxy-SynBioCAD: Automated Pipeline for Synthetic Biology Design and Engineering</b>. <a href="https://doi.org/10.1101/2022.02.23.481618">10.1101/2022.02.23.481618</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Designing plasmids encoding predicted pathways by using the BASIC assembly method (Synthetic Biology)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Kenza Bazi-Kabbaj, Thomas Duigou, Joan HÃ©risson, Guillaume Gricourt, Ioana Popescu, Jean-Loup Faulon,  <b>Designing plasmids encoding predicted pathways by using the BASIC assembly method (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html">https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{synthetic-biology-basic_assembly_analysis,
author = "Kenza Bazi-Kabbaj and Thomas Duigou and Joan HÃ©risson and Guillaume Gricourt and Ioana Popescu and Jean-Loup Faulon",
	title = "Designing plasmids encoding predicted pathways by using the BASIC assembly method (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine FÃ¶ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and BjÃ¶rn GrÃ¼ning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: dnabot
  owner: tduigou
  revisions: b450154dc448
  tool_panel_section_label: Synthetic Biology
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: rpbasicdesign
  owner: tduigou
  revisions: de9f53630349
  tool_panel_section_label: Synthetic Biology
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: selenzy
  owner: tduigou
  revisions: 34a9d136a5bf
  tool_panel_section_label: Synthetic Biology
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00273">https://gxy.io/GTN:T00273</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1707006564'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>