<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy 101 for everyone</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:title" content="Galaxy Training: Galaxy 101 for everyone" />
        <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/introduction" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Introduction to Galaxy Analyses
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/introduction/faqs/" title="Check our FAQs for the Introduction to Galaxy Analyses topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Galaxy 101 for everyone",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Introduction to Galaxy Analyses",
    "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
    "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
  },
  "courseCode": "introduction / galaxy-intro-101-everyone / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Galaxy 101 for everyone' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html",
  "timeRequired": "PT1H30M",
  "description": "The questions this  addresses are:\n - What are the differences between the Iris species?\n\n\\nThe objectives are:\n - Familiarize yourself with the basics of Galaxy\n - Learn how to obtain data from external sources\n - Learn how to tag datasets\n - Learn how to run tools\n - Learn how histories work\n - Learn how to create a workflow\n - Learn how to share your work\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anne Fouilloux"
    },
    {
      "@type": "Person",
      "name": "Nadia Goué"
    },
    {
      "@type": "Person",
      "name": "Christopher Barnett"
    },
    {
      "@type": "Person",
      "name": "Michele Maroni"
    },
    {
      "@type": "Person",
      "name": "Olha Nahorna"
    },
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anne Fouilloux"
    },
    {
      "@type": "Person",
      "name": "Nadia Goué"
    },
    {
      "@type": "Person",
      "name": "Christopher Barnett"
    },
    {
      "@type": "Person",
      "name": "Michele Maroni"
    },
    {
      "@type": "Person",
      "name": "Olha Nahorna"
    },
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
    }
  ],
  "skillLevel": "Introductory"
}
</script>

<section class="tutorial topic-introduction">
    <h1 data-toc-skip>Galaxy 101 for everyone</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/annefou/" class="contributor-badge contributor-annefou"><img src="https://avatars.githubusercontent.com/annefou?s=27" alt="Avatar">Anne Fouilloux</a>


<a href="/training-material/hall-of-fame/nagoue/" class="contributor-badge contributor-nagoue"><img src="https://avatars.githubusercontent.com/nagoue?s=27" alt="Avatar">Nadia Goué</a>


<a href="/training-material/hall-of-fame/chrisbarnettster/" class="contributor-badge contributor-chrisbarnettster"><img src="https://avatars.githubusercontent.com/chrisbarnettster?s=27" alt="Avatar">Christopher Barnett</a>


<a href="/training-material/hall-of-fame/michelemaroni89/" class="contributor-badge contributor-michelemaroni89"><img src="https://avatars.githubusercontent.com/michelemaroni89?s=27" alt="Avatar">Michele Maroni</a>


<a href="/training-material/hall-of-fame/olanag1/" class="contributor-badge contributor-olanag1"><img src="https://avatars.githubusercontent.com/olanag1?s=27" alt="Avatar">Olha Nahorna</a>


<a href="/training-material/hall-of-fame/tnabtaf/" class="contributor-badge contributor-tnabtaf"><img src="https://avatars.githubusercontent.com/tnabtaf?s=27" alt="Avatar">Dave Clements</a>


<a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=27" alt="Avatar">Saskia Hiltemann</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the differences between the Iris species?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Familiarize yourself with the basics of Galaxy</p>
</li>
        
        <li><p>Learn how to obtain data from external sources</p>
</li>
        
        <li><p>Learn how to tag datasets</p>
</li>
        
        <li><p>Learn how to run tools</p>
</li>
        
        <li><p>Learn how histories work</p>
</li>
        
        <li><p>Learn how to create a workflow</p>
</li>
        
        <li><p>Learn how to share your work</p>
</li>
        
        </ul>

        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour 30 minutes</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/introduction/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/1319069/files/iris.csv">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/workflows/" title="Workflows" alt="Galaxy 101 for everyone workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/introduction/galaxy-intro-101-everyone/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/introduction/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://climate.usegalaxy.eu/" title="Climate Science Workbench">
            Climate Science Workbench
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://ecology.usegalaxy.eu/" title="Galaxy for Ecology">
            Galaxy for Ecology
        </a>
        
    
        
        <a class="dropdown-item" href="https://galaxy.mesocentre.uca.fr" title="Galaxy@AuBi">
            Galaxy@AuBi
        </a>
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://hicexplorer.usegalaxy.eu" title="HiCExplorer">
            HiCExplorer
        </a>
        
    
        
        <a class="dropdown-item" href="https://humancellatlas.usegalaxy.eu/" title="Human Cell Atlas">
            Human Cell Atlas
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://metagenomics.usegalaxy.eu/" title="Metagenomics">
            Metagenomics
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="http://galaxy.migale.inra.fr/" title="Migale">
            Migale
        </a>
        
    
        
        <a class="dropdown-item" href="https://nanopore.usegalaxy.eu/" title="NanoGalaxy">
            NanoGalaxy
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://plants.usegalaxy.eu/" title="Plant Anaylsis Workbench">
            Plant Anaylsis Workbench
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/" title="Single Cell Omics workbench">
            Single Cell Omics workbench
        </a>
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 16, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>This practical aims at familiarizing you with the <span class="notranslate">Galaxy</span> user interface.
It will teach you how to perform basic tasks such as importing data, running tools, working with histories, creating workflows and sharing your work.
Not everyone has the same background and that’s ok!</p>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#what-does-galaxy-look-like" id="markdown-toc-what-does-galaxy-look-like">What does <span class="notranslate">Galaxy</span> look like?</a></li>
  <li><a href="#create-a-history" id="markdown-toc-create-a-history">Create a history</a>    <ol>
      <li><a href="#upload-the-iris-dataset" id="markdown-toc-upload-the-iris-dataset">Upload the Iris dataset</a></li>
    </ol>
  </li>
  <li><a href="#pre-processing" id="markdown-toc-pre-processing">Pre-processing</a>    <ol>
      <li><a href="#convert-format" id="markdown-toc-convert-format">Convert format</a></li>
      <li><a href="#remove-header" id="markdown-toc-remove-header">Remove header</a></li>
    </ol>
  </li>
  <li><a href="#data-analysis-what-does-the-dataset-contain" id="markdown-toc-data-analysis-what-does-the-dataset-contain">Data Analysis: What does the dataset contain?</a>    <ol>
      <li><a href="#how-many-different-species-are-in-the-dataset" id="markdown-toc-how-many-different-species-are-in-the-dataset">How many different species are in the dataset?</a></li>
      <li><a href="#how-many-samples-by-species-are-in-the-dataset" id="markdown-toc-how-many-samples-by-species-are-in-the-dataset">How many samples by species are in the dataset?</a></li>
    </ol>
  </li>
  <li><a href="#analysis-how-to-differentiate-the-different-iris-species" id="markdown-toc-analysis-how-to-differentiate-the-different-iris-species">Analysis: How to differentiate the different Iris species?</a>    <ol>
      <li><a href="#generate-summary-and-descriptive-statistics-with" id="markdown-toc-generate-summary-and-descriptive-statistics-with">Generate summary and descriptive statistics with</a></li>
      <li><a href="#visualize-iris-dataset-features-with-two-dimensional-scatterplots" id="markdown-toc-visualize-iris-dataset-features-with-two-dimensional-scatterplots">Visualize Iris dataset features with two-dimensional scatterplots</a></li>
    </ol>
  </li>
  <li><a href="#galaxy-management" id="markdown-toc-galaxy-management"><span class="notranslate">Galaxy</span> management</a>    <ol>
      <li><a href="#convert-your-analysis-history-into-a-workflow" id="markdown-toc-convert-your-analysis-history-into-a-workflow">Convert your analysis history into a <span class="notranslate">workflow</span></a></li>
      <li><a href="#the-workflow-editor" id="markdown-toc-the-workflow-editor">The <span class="notranslate">workflow</span> editor</a></li>
      <li><a href="#run-workflow-on-different-data" id="markdown-toc-run-workflow-on-different-data">Run <span class="notranslate">workflow</span> on different data</a></li>
      <li><a href="#share-your-work" id="markdown-toc-share-your-work">Share your work</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-background"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Background</h3>
  <p>The Iris flower data set, also known as Fisher’s or Anderson’s Iris data set, is a multivariate dataset introduced by the British statistician and biologist Ronald Fisher in his 1936 paper (<span class="citation"><a href="#Fisher1936">Fisher 1936</a></span>).
Each row of the table represents an iris flower sample, describing its species and the dimensions in centimeters of its botanical parts, the sepals and petals.
You can find more detailed information about this dataset on its dedicated <a href="https://en.wikipedia.org/wiki/Iris_flower_data_set">Wikipedia page</a>.</p>
</blockquote>

<h2 id="what-does-galaxy-look-like">What does <span class="notranslate">Galaxy</span> look like?</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-log-in-or-register"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Log in or register</h3>
  <ol>
    <li>Open your favorite browser (Chrome/Chromium, Safari, or Firefox, but not Internet Explorer/Edge!)</li>
    <li>Browse to a <a href="https://galaxyproject.org/use/"><span class="notranslate">Galaxy</span> instance</a> of your choice</li>
    <li>Choose <em>Login or Register</em> from the navigation bar at the top of the page</li>
    <li>
      <p>If you have previously registered an account with this particular instance of <span class="notranslate">Galaxy</span> (user accounts are <em>not</em> shared between public servers!), proceed by logging in with your registered <em>public name</em>, or email address, and your password.</p>

      <p>If you need to create a new account, click on <em>Register here</em> instead.</p>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-different-galaxy-servers"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Different <span class="notranslate">Galaxy</span> servers</h3>
    <p>The particular <span class="notranslate">Galaxy</span> server that you are using may look slightly different than the one shown in this training.
<span class="notranslate">Galaxy</span> instance administrators can choose the exact version of <span class="notranslate">Galaxy</span> they would like to offer and can customize its look and feel to some extent.
The basic functionality will be rather similar across instances, so don’t worry!</p>
  </blockquote>

</blockquote>

<p>The <span class="notranslate">Galaxy</span> interface consists of three main parts:</p>

<ol>
  <li>The available tools are listed on the left</li>
  <li>Your analysis history is recorded on the right</li>
  <li>The central panel will let you run analyses and view outputs</li>
</ol>

<p><img src="/training-material/shared/images/galaxy_interface.png" alt="Galaxy interface screenshot showing history panel on the right, tools panel on the left, and main panel at the center. " loading="lazy" /></p>

<h1 id="create-a-history">Create a history</h1>

<p><span class="notranslate">Galaxy</span> allows you to create analysis histories. A <strong>history</strong> can be thought of as an electronic experimental lab book; it keeps track of all the tools and parameters you used in your analysis. From such a history, a <strong><span class="notranslate">workflow</span></strong> can be extracted; this <span class="notranslate">workflow</span> can be used to easily repeat the analysis on different data.</p>

<p>Think of a <span class="notranslate">workflow</span> as a cooking recipe with a list of ingredients (datasets) and a set of instructions
(pipeline of operations) that describes how to prepare or make something (such as a plot, or a new dataset).
The order of operations is important as very often the next operation takes as input the result of the previous operations. For instance, when baking
a cake, you would first sift the flour and then mix it with eggs as it would be impossible to sift the flour afterward.
That is what we call a pipeline. To make a full meal, we may need to combine multiple recipes (pipelines) together.</p>

<p>The finalized pipelines can be generalized as a <span class="notranslate">workflow</span>. If we use cooking as an analogy, a <span class="notranslate">workflow</span> could represent an entire menu with all the recipes for each meal.
In other words, using a <span class="notranslate">workflow</span> makes it possible to apply the same procedure to a different dataset, just by changing the input.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-history"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create history</h3>

  <ol>
    <li>
      <p>Make sure you start from an empty analysis history.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p><strong>Rename your history</strong> to be meaningful and easy to find. For instance, you can choose <strong><span class="notranslate">Galaxy</span> 101 for everyone</strong> as the name of your new history.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="upload-the-iris-dataset">Upload the Iris dataset</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p><span class="tool" data-tool="upload1" title="Tested with upload1"><strong>Import</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span>  the file <code class="language-plaintext highlighter-rouge">iris.csv</code> from <a href="https://zenodo.org/record/1319069/files/iris.csv">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1319069/files/iris.csv
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>Check the <strong>datatype</strong>
      <ul>
        <li>Click on the history item to expand it to get more information.</li>
        <li>The datatype of the iris dataset should be <code class="language-plaintext highlighter-rouge">csv</code>.</li>
        <li><strong>Change</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the datatype <em>if</em> it is different than <code class="language-plaintext highlighter-rouge">csv</code>.
          <ul>
            <li>Option 1: Datatypes can be <strong>autodetected</strong></li>
            <li>Option 2: Datatypes can be <strong>manually set</strong></li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-detecting-the-datatype-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Detecting the datatype (file format)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Click the <strong>Detect datatype</strong> button to have <span class="notranslate">Galaxy</span> try to autodetect it.</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">csv</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Add an <code class="language-plaintext highlighter-rouge">#iris</code> tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> to the dataset</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>

      <p>Make sure the tag starts with a hash symbol (<code class="language-plaintext highlighter-rouge">#</code>), which will make the tag stick not only to this dataset, but also to any results derived from it.
This will help you make sense of your history.</p>
    </li>
  </ol>

</blockquote>

<h1 id="pre-processing">Pre-processing</h1>

<p>Often, one or more data pre-processing step(s) may be required to proceed with the analysis.
In our case, the tools we will use require tab-separated input data and assume there is no header line. Since our data is comma-separated and has a header line, we will have to perform the following pre-processing steps to prepare it for the actual analysis:</p>

<ul>
  <li>Format conversion</li>
  <li>Header removal</li>
</ul>

<h2 id="convert-format">Convert format</h2>

<p>First, we will convert the file from comma-separated to tab-separated format. <span class="notranslate">Galaxy</span> has built-in format converters we can use for this.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-converting-dataset-format"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Converting dataset format</h3>

  <ol>
    <li>
      <p><strong>Convert</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the CSV file (comma-separated values) to tabular format (tsv; tab-separated values)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-converting-the-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Converting the file format</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <strong>Convert</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">Convert CSV to Tabular</code></li>  <li>Click the <strong>Convert datatype</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the resulting dataset to <code class="language-plaintext highlighter-rouge">iris tabular</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p><strong>View</strong> the generated file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon</p>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>How many header lines does our file have?</p>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The file has one header line, it contains the column names.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="remove-header">Remove header</h2>

<p>Now it is time to run your first tool! We saw in the previous step that our file has 1 header line. This line does not contain any data, but the names of each column. We will now remove that line from our file before moving on to our analysis.</p>

<blockquote class="comment">
  <h3 id="comment-tip-finding-your-tool"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tip: Finding your tool</h3>

  <p>Different <span class="notranslate">Galaxy</span> servers may have tools available under different sections, therefore it is often useful to use the <strong>search bar</strong> at the top of the tool panel to find your tool.</p>

  <p>Additionally different servers may have multiple, similarly named tools which accomplish similar functions. When following tutorials, you should use precisely the tools that they describe. For real analyses, however, you will need to search among the various options to find the one that works for you.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-removing-header"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Removing header</h3>

  <ol>
    <li><span class="tool" data-tool="Remove beginning1" title="Tested with Remove beginning1"><strong>Remove Beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Remove beginning1</span></span>  with the following parameters:
      <ul>
        <li><em>Remove first</em>: <code class="language-plaintext highlighter-rouge">1</code> (to remove the first line only)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from”</em>: select the <strong>iris tabular</strong> file from your history</li>
        <li>Click <strong>Execute</strong></li>
      </ul>

      <blockquote class="tip">
        <h3 id="comment-tip-search-for-the-tool"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tip: search for the tool</h3>

        <p>Use the <strong>tools search box</strong> at the top of the tool panel to find <strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>
      </blockquote>

      <p><img src="../../images/101_foreveryone_remove_beginning.png" alt="Settings for the `Remove beginning` tool. " loading="lazy" /></p>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris clean</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Click on the new history item to expand it</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>Which tags are present on this resulting dataset? (You may have to refresh the history panel to see the tags)</li>
          <li>How many samples (lines) does our dataset contain?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>
              <p>The output of <strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is also tagged with the label <code class="language-plaintext highlighter-rouge">iris</code>. Tags beginning with a hashtag (<code class="language-plaintext highlighter-rouge">#</code>) will propagate; they will appear on any datasets derived from your original tagged file.</p>
            </li>
            <li>
              <p>There are 150 lines in our file (we can see this under the file name when we have expanded the history item). This means we have 150 samples.</p>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the contents of the resulting file.
      <ul>
        <li>You should see that the header line is now no longer present.</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="data-analysis-what-does-the-dataset-contain">Data Analysis: What does the dataset contain?</h1>

<p>Now we are going to inspect the dataset using simple tools in order to get used to the <span class="notranslate">Galaxy</span> interface and answer basic questions.</p>

<h2 id="how-many-different-species-are-in-the-dataset">How many different species are in the dataset?</h2>

<p>In order to answer this question, we will have to look at column 5 of our file, and count how many different values (species) appear there. There are several ways we could do this in <span class="notranslate">Galaxy</span>. One approach might be to first extract this column from the file, and then count how many unique lines the file contains. Let’s do it!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-species"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract species</h3>

  <ol>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  columns from a table with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c5</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">iris clean</code> dataset</li>
      </ul>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris species column</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the resulting file</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0"><strong>Unique</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.1.0</span></span>  occurrences of each record  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to scan for unique values”</em>: <code class="language-plaintext highlighter-rouge">iris species column</code> (the output from <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris species</code></p>
    </li>
    <li>
      <p><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the resulting file</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many different species are in the dataset?</li>
          <li>What are the different Iris species?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>There are 3 species.</li>
            <li>The 3 different Iris species are:
              <ul>
                <li>setosa</li>
                <li>versicolor</li>
                <li>virginica</li>
              </ul>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>Now we have our answer! There are 3 different Iris species in our file.</p>

<p>Like we mentioned before, there are often multiple ways to reach your answer in <span class="notranslate">Galaxy</span>. For example, we could have done this with just a single tool, <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> as well.
(This tool also can be searched for by the term “grouping” )</p>

<blockquote class="question">
  <h3 id="hands_on-exercise-grouping-dataset"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Exercise: Grouping dataset</h3>

  <ol>
    <li>Try answering this question (how many Iris species are in the file?) again, using a different approach:
      <ul>
        <li>Tool: <strong>Group</strong> data by a column and perform aggregate operation on other columns <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Input dataset: <code class="language-plaintext highlighter-rouge">iris clean</code> dataset to answer the same question.</li>
      </ul>
    </li>
    <li>
      <p>Did you get the same answer as before?</p>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris species group</code></li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li><span class="tool" data-tool="Grouping1" title="Tested with Grouping1"><strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grouping1</span></span>    with the following parameters:
        <ul>
          <li><em>“Select data”</em> select <code class="language-plaintext highlighter-rouge">iris clean</code> dataset</li>
          <li><em>“Group by column”</em>: <code class="language-plaintext highlighter-rouge">Column: 5</code></li>
        </ul>
      </li>
      <li>This approach should give the same answer. There are often multiple ways to do a task in <span class="notranslate">Galaxy</span>, which way you choose is up to you!</li>
    </ol>
  </blockquote>

</blockquote>

<h2 id="how-many-samples-by-species-are-in-the-dataset">How many samples by species are in the dataset?</h2>

<p>Now that we know that there are 3 different species in our dataset, our next objective is determining how many samples of each species we have. To answer this, we need to look at column 5 again, but instead of just determining how many unique values there are, we need to count how many times each of them occurs.</p>

<p>You may have noticed there were more parameters in the <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool that we did not use. Let’s have a closer look and see if any of them might help us answer this question.</p>

<blockquote class="comment">
  <h3 id="comment-tool-help"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tool Help</h3>
  <p>To find out more about how a tool works, look at the help text (below the <strong>Execute</strong> button).</p>

  <p>Look at the tool help for the <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Do you see any parameters that could help answer this question?</p>
</blockquote>

<p>Looking at the tool help for <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, we see that we can also perform aggregate operations such as mean, median, sum, max, min, count (and more). Counting sounds just like what we need, let’s try it!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-grouping-dataset-and-adding-information"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Grouping dataset and adding information</h3>

  <ol>
    <li><strong>Re-run</strong> <i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> the <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select data”</em>: <code class="language-plaintext highlighter-rouge">iris clean</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Group by column”</em>: <code class="language-plaintext highlighter-rouge">Column: 5</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert operation”</em>
          <ul>
            <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Count</code></li>
            <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-re-running-a-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Re-running a tool</h3>  <ol>  <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>  <li><strong>Click</strong> re-run <i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> the tool</li></ol>  <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p></blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">iris samples per species group</code></p>
    </li>
    <li>
      <p><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the resulting file.</p>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
        <p>How many samples per species are in the dataset?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>We have 50 samples per species:</p>

          <table>
            <thead>
              <tr>
                <th>1</th>
                <th>2</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>setosa</td>
                <td>50</td>
              </tr>
              <tr>
                <td>versicolor</td>
                <td>50</td>
              </tr>
              <tr>
                <td>virginica</td>
                <td>50</td>
              </tr>
            </tbody>
          </table>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="analysis-how-to-differentiate-the-different-iris-species">Analysis: How to differentiate the different Iris species?</h1>

<p>Our objective is to find what distinguishes the different Iris species (Figure 1). We know that we have <strong>3</strong> species of iris flowers, with
<strong>50</strong> samples for each:</p>
<ul>
  <li>setosa</li>
  <li>versicolor</li>
  <li>virginica</li>
</ul>

<p>These species look very much alike as shown on the figure below.</p>

<figure id="figure-1"><img src="../../images/iris_flowers.png" alt="Three species of Iris flowers. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Three species of Iris flowers (Image attributions: versicolor by Danielle Langlois licensed under CC BY-SA 3.0, retrieved from <a href="https://commons.wikimedia.org/wiki/File:Iris_versicolor_3.jpg">WikiMedia</a>; virginica by Christer Johansson licensed under CC BY-SA 3.0, retrieved from <a href="https://commons.wikimedia.org/wiki/File:IMG_7911-Iris_virginica.jpg">WikiMedia</a>; setosa by and used with permission of Sonja Keohane, retrieved from <a href="http://www.twofrog.com/irissetosa.html">www.twofrog.com</a>)</figcaption></figure>

<p>And our objective is to find out whether the features we have been given for each species can help us to highlight the differences between the 3 species.</p>

<p>In our dataset, we have the following features measured for each sample:</p>
<ul>
  <li>Petal length</li>
  <li>Petal width</li>
  <li>Sepal length</li>
  <li>Sepal width</li>
</ul>

<blockquote class="comment">
  <h3 id="comment-petal-and-sepal"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> petal and sepal</h3>
  <p>The image below shows you what the terms sepal and petal mean.</p>
  <figure id="figure-2"><img src="../../images/iris_sepal_petal.png" alt="Iris flower image showing Petal and Sepal. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Sepal and petal of Iris flowers</figcaption></figure>
</blockquote>

<h2 id="generate-summary-and-descriptive-statistics-with">Generate summary and descriptive statistics with</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-mean-and-sample-standard-deviation-of-iris-flower-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the mean and sample standard deviation of Iris flower features</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0"><strong>Datamash</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.1.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tabular dataset”</em>: <code class="language-plaintext highlighter-rouge">iris tabular</code></li>
        <li><em>“Group by fields”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
        <li><em>“Input file has a header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Print header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Sort input”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>“Print all fields from input file”: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Ignore case when grouping”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>In <em>“Operation to perform on each group”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Sample Standard deviation</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c2</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Sample Standard deviation</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c2</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c3</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Sample Standard deviation</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c3</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation to perform on each group”</em>
              <ul>
                <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Sample Standard deviation</code></li>
                <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the dataset to <code class="language-plaintext highlighter-rouge">iris summary and statistics</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the generated file</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>Can we differentiate the different Iris flower species?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <ol>
        <li>From the results, we can see that the average <em>Iris setosa</em> petal length is lower than 1.5 with a relatively small standard deviation (&lt;0.2).
The same can be observed for <em>Iris setosa</em> petal widths. These numbers are much smaller (width and length) than <em>Iris versicolor</em> and <em>Iris virginica</em> petals.
We can then use these characteristics to differentiate <em>Iris setosa</em> from the two other species (<em>I. versicolor</em> and <em>I. virginica</em>). On the other hand,
we cannot easily differentiate <em>Iris Versicolor</em> from <em>Iris Virginica</em>. Further analysis is necessary.</li>
      </ol>

    </blockquote>

  </blockquote>

</blockquote>

<h2 id="visualize-iris-dataset-features-with-two-dimensional-scatterplots">Visualize Iris dataset features with two-dimensional scatterplots</h2>

<p>Let’s visualize the Iris dataset to see how the features depend on each other, and
check whether we can spot any immediate patterns.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-iris-feature-pairs-in-two-dimensions"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot iris feature pairs in two dimensions</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+galaxy1"><strong>Scatterplot w ggplot2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+<span class="notranslate">galaxy</span>1</span></span>    with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tabular dataset”</em>: <strong>iris clean</strong></li>
        <li><em>“Column to plot on x-axis”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Column to plot on y-axis”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
        <li><em>“Plot title”</em>: <code class="language-plaintext highlighter-rouge">Sepal length as a function of sepal width</code></li>
        <li><em>“Label for x axis”</em>: <code class="language-plaintext highlighter-rouge">Sepal length</code></li>
        <li><em>“Label for y axis”</em>: <code class="language-plaintext highlighter-rouge">Sepal width</code></li>
        <li>In <em>“Advanced Options”</em>:
          <ul>
            <li><em>“Data point options”</em>: <code class="language-plaintext highlighter-rouge">User defined point options</code>
              <ul>
                <li><em>“relative size of points”</em>: <code class="language-plaintext highlighter-rouge">2.0</code></li>
              </ul>
            </li>
            <li><em>“Plotting multiple groups”</em>: <code class="language-plaintext highlighter-rouge">Plot multiple groups of data on one plot</code>
              <ul>
                <li><em>“column differentiating the different groups”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                <li><em>“Color schemes to differentiate your groups”</em>: <code class="language-plaintext highlighter-rouge">Set 2 - predefined color pallete</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>:
          <ul>
            <li><em>Additional output format</em>: <code class="language-plaintext highlighter-rouge">PDF</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the resulting plot:</p>

      <p><img src="../../images/101_foreveryone_scatter.png" alt="Contents of the `Group` output dataset. " loading="lazy" /></p>
    </li>
    <li>Rename the dataset to <code class="language-plaintext highlighter-rouge">iris sepal scatterplot</code></li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>What does this scatter plot tell us about Iris species?</li>
      <li>Make a new scatter plot, this time with <em>Petal</em> length versus <em>Petal</em> width.</li>
      <li>
        <p>Can we differentiate between the three Iris species?</p>

        <blockquote class="tip">
          <h3 id="tip-tip-re-running-the-ggplot-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Re-running the ggplot tool</h3>
          <p>Instead of clicking on <strong>Scatterplot w ggplot2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> again, it is possible to recall the previous scatterplot parameters by clicking on re-run button and updating the parameters we wish to modify.</p>
          <blockquote class="notranslate hands_on">
            <h3 id="hands_on-hands-on-re-run-the-tool"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Re-run the tool</h3>
            <ol>
              <li>Click on the <i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> icon (<strong>Run this job again</strong>) for the output dataset of <strong>Scatterplot w ggplot2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
<img src="../../images/rerun_scatterplot.png" alt="rerun. " loading="lazy" /></li>
            </ol>

            <p>This brings up the tool interface in the central panel with the parameters set to the values used previously to generate this dataset.</p>
          </blockquote>
        </blockquote>
      </li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <ol>
        <li>We get similar results than with Summary and statistics: <em>Iris setosa</em> can clearly be distinguished from <em>Iris versicolor</em> and
<em>Iris virginica</em>. We can also see that sepal width and length are not sufficient features to differentiate <em>Iris versicolor</em> from <em>Iris
virginica</em>.</li>
        <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+galaxy1"><strong>Scatterplot w ggplot2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tabular dataset”</em>: <code class="language-plaintext highlighter-rouge">iris clean</code></li>
            <li><em>“Column to plot on x-axis”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>“Column to plot on y-axis”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
            <li><em>“Plot title”</em>: <code class="language-plaintext highlighter-rouge">Petal length as a function of petal width</code></li>
            <li><em>“Label for x axis”</em>: <code class="language-plaintext highlighter-rouge">Petal length</code></li>
            <li><em>“Label for y axis”</em>: <code class="language-plaintext highlighter-rouge">Petal width</code></li>
            <li>In <em>“Advanced Options”</em>:
              <ul>
                <li><em>“Data point options”</em>: <code class="language-plaintext highlighter-rouge">User defined point options</code>
                  <ul>
                    <li><em>“relative size of points”</em>: <code class="language-plaintext highlighter-rouge">2.0</code></li>
                  </ul>
                </li>
                <li><em>“Plotting multiple groups”</em>: <code class="language-plaintext highlighter-rouge">Plot multiple groups of data on one plot</code>
                  <ul>
                    <li><em>“column differentiating the different groups”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                    <li><em>“Color schemes to differentiate your groups”</em>: <code class="language-plaintext highlighter-rouge">Set 2 - predefined color pallete</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Your new output dataset will look something like this:</li>
      </ol>

      <p><img src="../../images/101_foreveryone_scatter_petal.png" alt="Contents of the `Group` output dataset. " loading="lazy" /></p>

      <p>We can better differentiate between the 3 Iris species but for some samples the petal length versus width is still insufficient
to differentiate <em>Iris versicolor</em> from <em>Iris virginica</em>. And as before, <em>Iris setosa</em> can easily be distinguished from the two other species.</p>
    </blockquote>
  </blockquote>
</blockquote>

<h1 id="galaxy-management"><span class="notranslate">Galaxy</span> management</h1>

<h2 id="convert-your-analysis-history-into-a-workflow">Convert your analysis history into a <span class="notranslate">workflow</span></h2>

<p>When you look carefully at your history, you can see that it contains all the steps of our analysis, from the beginning to the end. By building this history we have actually built a complete record of our analysis with <span class="notranslate">Galaxy</span> preserving all parameter settings applied at every step. But when you receive new data, or a new report is requested, it would be tedious to do each step over again. Wouldn’t it be nice to just convert this history into a <span class="notranslate">workflow</span> that we will be able to execute again and again?</p>

<p><span class="notranslate">Galaxy</span> makes this very easy with the <code class="language-plaintext highlighter-rouge">Extract <span class="notranslate">workflow</span></code> option. This means any time you want to build a <span class="notranslate">workflow</span>, you can just perform the steps once manually, and then convert it to a <span class="notranslate">workflow</span>, so that next time it will be a lot less work to do the same analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract <span class="notranslate">workflow</span></h3>

  <ol>
    <li>
      <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> button.</p>

      <p>This will make the creation of the <span class="notranslate">workflow</span> easier.</p>
    </li>
    <li>
      <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract <span class="notranslate">workflow</span></strong>.</p>

      <p><img src="../../images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu. " loading="lazy" /></p>

      <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the <span class="notranslate">workflow</span>.</p>

      <p><img src="../../images/101_foreveryone_workflow_create.png" alt="Selection of steps for `Extract Workflow` from history. " loading="lazy" /></p>
    </li>
    <li>
      <p>Replace the <strong><span class="notranslate">Workflow</span> name</strong> to something more descriptive, for example: <code class="language-plaintext highlighter-rouge">Exploring Iris dataset with statistics and scatterplots</code>.</p>
    </li>
    <li>
      <p>If there are any steps that shouldn’t be included in the <span class="notranslate">workflow</span>, you can <strong>uncheck</strong> them in the first column of boxes.</p>
    </li>
    <li>
      <p>Click on the <strong>Create <span class="notranslate">Workflow</span></strong> button near the top.</p>

      <p>You will get a message that the <span class="notranslate">workflow</span> was created. But where did it go?</p>

      <p><img src="../../images/101_foreveryone_workflow.png" alt="`Where workflows go` list. " loading="lazy" /></p>
    </li>
    <li>
      <p>Click on <strong><span class="notranslate">Workflow</span></strong> in the top menu of <span class="notranslate">Galaxy</span>.</p>
      <ul>
        <li>Here you have a list of all your workflows.</li>
        <li>Your newly created <span class="notranslate">workflow</span> should be listed at the top:</li>
      </ul>

      <p><img src="../../images/101_foreveryone_workflow_list.png" alt="`Your workflows` list. " loading="lazy" /></p>
    </li>
  </ol>
</blockquote>

<h2 id="the-workflow-editor">The <span class="notranslate">workflow</span> editor</h2>

<blockquote class="comment">
  <h3 id="comment-tip-problems-creating-your-workflow"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tip: Problems creating your <span class="notranslate">workflow</span>?</h3>
  <p>If you had problems extracting your <span class="notranslate">workflow</span> in the previous step, we provide a working copy for you <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/workflows/main_workflow.ga">here</a>,
which you can import to <span class="notranslate">Galaxy</span> and use for the next sections (see below how to import a <span class="notranslate">workflow</span> to <span class="notranslate">Galaxy</span>).</p>

  <!--SNIPPET-->
  <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>“Archived <span class="notranslate">Workflow</span> URL”</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>“Archived <span class="notranslate">Workflow</span> File”</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>

</blockquote>

<p>We can examine the <span class="notranslate">workflow</span> in <span class="notranslate">Galaxy</span>’s <span class="notranslate">workflow</span> editor. Here you can view/change the parameter settings of each step, add and remove tools, and connect an output from one tool to the input of another, all in an easy and graphical manner. You can also use this editor to build workflows from scratch.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-editing-our-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Editing our <span class="notranslate">workflow</span></h3>

  <ol>
    <li>Open the <strong><span class="notranslate">workflow</span> editor</strong>
      <ul>
        <li>Click on the dropdown menu <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-dropdown</span> (triangle icon) to the right of your <span class="notranslate">workflow</span> name.</li>
      </ul>
    </li>
    <li>Select <strong>Edit</strong> to launch the <span class="notranslate">workflow</span> editor.
      <ul>
        <li>You should see something like this:</li>
      </ul>

      <p><img src="../../images/101_foreveryone_workflow_editor.png" alt="Workflow editor. " loading="lazy" /></p>

      <ul>
        <li>When you click on a <span class="notranslate">workflow</span> step, you will get a view of all the parameter settings for that tool on the right-hand side of your screen (the <strong>Details</strong> section)</li>
        <li>You can also change the parameter settings of your <span class="notranslate">workflow</span> here, and also do more advanced configuration.</li>
      </ul>
    </li>
    <li><strong>Hiding intermediate outputs</strong>
      <ul>
        <li>We can tell <span class="notranslate">Galaxy</span> which outputs of a <span class="notranslate">workflow</span> are important and should be shown in our history when we run it, and which can be hidden.</li>
        <li>By default, all outputs will be shown</li>
        <li>Click the <strong>checkbox</strong> next to the outputs to mark them as important:
          <ul>
            <li><code class="language-plaintext highlighter-rouge">outfile</code> in  <strong>Unique</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><code class="language-plaintext highlighter-rouge">out_file1</code> in <strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step
              <ul>
                <li>This should be the Group tool where we performed the counting, you can check which one that is by clicking on it and looking at the parameter settings in the <strong>Details</strong> box on the right.</li>
              </ul>
            </li>
            <li><code class="language-plaintext highlighter-rouge">png</code> in both <strong>Scatterplot w ggplot2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> steps</li>
          </ul>
        </li>
        <li>Now, when we run the <span class="notranslate">workflow</span>, we will only see these final outputs
          <ul>
            <li>i.e. the two dataset with species, the dataset with number of samples by species and the two scatterplots.</li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-hiding-intermediate-steps"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Hiding intermediate steps</h3>  <p>When a <span class="notranslate">workflow</span> is executed, the user is usually primarily interested in the final product and not in all intermediate steps.By default all the outputs of a <span class="notranslate">workflow</span> will be shown, but we can explicitly tell <span class="notranslate">Galaxy</span> which outputs to show and which to hide for a given <span class="notranslate">workflow</span>.This behaviour is controlled by the little checkbox in front of every output dataset:</p>  <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p></blockquote>
    </li>
    <li><strong>Renaming output datasets</strong>
      <ul>
        <li>When we performed the analysis manually, we often renamed output datasets to something more meaningful</li>
        <li>We can do the same in a <span class="notranslate">workflow</span> (see the tip box below)</li>
        <li>Let’s rename the outputs we marked as important with the checkbox (and thus do not hide) to more meaningful names:
          <ul>
            <li><strong>Unique</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, output <code class="language-plaintext highlighter-rouge">outfile</code>: rename to <code class="language-plaintext highlighter-rouge">categories</code> tool</li>
            <li><strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, output <code class="language-plaintext highlighter-rouge">out_file1</code>: rename to <code class="language-plaintext highlighter-rouge">samples per category</code></li>
            <li>Rename the scatterplot outputs as well, remember to choose a generic name, since we can now also run this on data other than iris plants.</li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-workflow-outputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming <span class="notranslate">workflow</span> outputs</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.    <ul>      <li>        <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>        <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>      </li>    </ul>  </li></ol></blockquote>
    </li>
    <li>
      <p><strong>Save your <span class="notranslate">workflow</span></strong> (important!) by clicking on the <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span> icon at the top right of the screen.</p>
    </li>
    <li><strong>Return</strong> to the analysis view by clicking on the Home icon <i class="fas fa-home" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-home</span> (or <strong>Analyze Data</strong> on older <span class="notranslate">Galaxy</span> versions) at the top menu bar.</li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>We could <strong>validate</strong> our newly built <span class="notranslate">workflow</span> by running it on the same input datasets that we used at the start of this tutorial, in order to make sure we do obtain the same results.</p>
</blockquote>

<h2 id="run-workflow-on-different-data">Run <span class="notranslate">workflow</span> on different data</h2>

<p>Now that we have built our <span class="notranslate">workflow</span>, let’s use it on some different data. For example, let us explore the <code class="language-plaintext highlighter-rouge">diamonds</code> R dataset with it.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-a-new-history-and-upload-a-new-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create a new history and upload a new data</h3>

  <ol>
    <li>
      <p>Create a <strong>new history</strong> and give it a name.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p><span class="tool" data-tool="upload1" title="Tested with upload1"><strong>Import</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span>  the file <code class="language-plaintext highlighter-rouge">diamonds.csv</code> from <a href="https://zenodo.org/record/3540705/files/diamonds.csv">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3540705/files/diamonds.csv
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset to <code class="language-plaintext highlighter-rouge">diamonds</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Add a propagating tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> (e.g. <code class="language-plaintext highlighter-rouge">#diamonds</code>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
  </ol>
</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">diamonds</code> dataset comes from the well-known <a href="https://ggplot2.tidyverse.org/reference/diamonds.html">ggplot2</a> package developed by Hadley Wickham and was initially collected from the Diamond Search Engine in 2008.
The original dataset consists of 53940 specimen of diamonds, for which it lists the prices and various properties.
For this training, we have created a simpler dataset from the original, in which only the five columns relating to the price and the so-called 4 Cs (carat, cut, color and clarity) of diamond characteristics have been retained.</p>

<blockquote class="comment">
  <h3 id="comment-the-4-cs-of-diamond-grading"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> The 4 Cs of diamond grading</h3>
  <p>According to the <a href="https://4cs.gia.edu/wp-content/uploads/2013/03/All-Scales.jpg">GIA’s (Gemological Institute of America) diamond grading system</a></p>
  <ul>
    <li><strong>Carat</strong> refers to the weight of the diamond when measured on a scale</li>
    <li><strong>Cut</strong> refers to the quality of the cut and can take the grades <em>Fair</em>, <em>Good</em>, <em>Very Good</em>, <em>Premium</em> and <em>Ideal</em></li>
    <li><strong>Color</strong> describes the overall tint, or lack thereof, of the diamond from colorless/white to yellow and is given on a letter scale ranging from D to Z (D being the best, known as colorless).</li>
    <li><strong>Clarity</strong> describes the amount and location of naturally occuring “inclusions” found in nearly all diamonds on a scale of eleven grades ranging from <em>Flawless</em> (the ideal situation) to <em>I3</em> (Included level 3, the worst quality).</li>
  </ul>
</blockquote>

<p>As a further simplification, our training dataset has the qualities in the <em>color</em> and <em>clarity</em> columns re-encoded as integer values (1-23 for color qualities D-Z, and 1-11 for the clarity levels from <em>Flawless</em> to <em>I3</em>).
With this adjustment, we can reuse our <span class="notranslate">workflow</span> on the data, and analyze and visualize it following the same steps as we took for the Iris dataset.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run <span class="notranslate">workflow</span></h3>

  <p>To analyze the diamonds price/4 Cs dataset by reusing our <span class="notranslate">workflow</span>:</p>

  <ol>
    <li>Open the <strong><span class="notranslate">workflow</span> menu</strong> (top menu bar).
      <ul>
        <li>Find the <span class="notranslate">workflow</span> you made in the previous section,</li>
        <li>Select the option <code class="language-plaintext highlighter-rouge">Run</code>.</li>
        <li>The central panel will change to allow you to configure and launch the <span class="notranslate">workflow</span>.</li>
      </ul>
    </li>
    <li>
      <p>Select the <code class="language-plaintext highlighter-rouge">diamonds</code> dataset as the input dataset.</p>
    </li>
    <li>
      <p>Customize the first scatter plot:</p>

      <p>This step is preconfigured to plot column 1 along the x and column 2 along the y axis, while grouping by column 5.
This is fine and will result in <em>price</em> getting plotted against <em>carat</em> with grouping by <em>cut</em>, but you would want to adjust the plot title and axis labels accordingly:</p>

      <ul>
        <li>Change <em>“Plot title”</em> to <code class="language-plaintext highlighter-rouge">Diamond price as a function of carat with cut as a factor</code></li>
        <li>Change <em>“Label for x axis”</em> to <code class="language-plaintext highlighter-rouge">Weight of the diamond (carat)</code></li>
        <li>Change <em>“Label for y axis”</em> to <code class="language-plaintext highlighter-rouge">Price (US dollars)</code></li>
      </ul>

      <p><img src="../../images/customize_diamond_plot.png" alt="Customize scatter plot. " loading="lazy" /></p>
    </li>
    <li>
      <p>Customize the second scatter plot.</p>

      <p>This one is preconfigured to plot column 3 along the x and column 4 along the y axis, which, for our new data, would plot <em>color</em> as a function of <em>clarity</em>. However, we would rather want to stick to plotting <em>price</em> against weight in <em>carat</em> as in the first plot, but group by <em>clarity</em> instead of by <em>cut</em> this time, so:</p>

      <ul>
        <li>Change <em>“Column to plot on x-axis”</em> to <code class="language-plaintext highlighter-rouge">1</code></li>
        <li>Change <em>“Column to plot on y-axis”</em> to <code class="language-plaintext highlighter-rouge">2</code></li>
        <li>Change <em>“Plot title”</em> to <code class="language-plaintext highlighter-rouge">Diamond price as a function of carat with clarity as a factor</code></li>
        <li>Change <em>“Label for x axis”</em> to <code class="language-plaintext highlighter-rouge">Weight of the diamond (carat)</code></li>
        <li>Change <em>“Label for y axis”</em> to <code class="language-plaintext highlighter-rouge">Price (US dollars)</code></li>
        <li>And finally in <em>“Advanced Options”</em> change <em>“column differentiating the different groups”</em> to <code class="language-plaintext highlighter-rouge">4</code> (clarity).</li>
      </ul>
    </li>
    <li>
      <p>Click <code class="language-plaintext highlighter-rouge">Run <span class="notranslate">workflow</span></code>.</p>
    </li>
    <li>
      <p>Once the <span class="notranslate">workflow</span> has started, you will initially be able to see all its steps, but the unimportant intermediates will disappear after they complete successfully:</p>

      <blockquote class="question">
        <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
        <ol>
          <li>How many cut category are there in the Diamond dataset ?</li>
          <li>How many samples are there in each cut category ?</li>
          <li>What do you notice about the relationship between price and carat ?</li>
          <li>Based on the plot showing Price vs. Carat with Clarity as a factor, do you think clarity accounts for some of the variance in price? Why ?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>There are 5 different cut categories:
              <ul>
                <li>Fair</li>
                <li>Good</li>
                <li>Ideal</li>
                <li>Premium</li>
                <li>Very Good</li>
              </ul>
            </li>
            <li>
              <p>We have the following number of samples in each cut category:</p>

              <table>
                <thead>
                  <tr>
                    <th>1</th>
                    <th>2</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>Fair</td>
                    <td>1610</td>
                  </tr>
                  <tr>
                    <td>Good</td>
                    <td>4906</td>
                  </tr>
                  <tr>
                    <td>Ideal</td>
                    <td>21551</td>
                  </tr>
                  <tr>
                    <td>Premium</td>
                    <td>13791</td>
                  </tr>
                  <tr>
                    <td>Very Good</td>
                    <td>12082</td>
                  </tr>
                </tbody>
              </table>
            </li>
            <li>
              <p>Using any of the scatter plots we made, we can see an obvious positive (non-linear) relationship between both variables: as carat size increases, price also increases. There is also very clear discrete values that carat size takes on, which are those vertical strips on the graph.</p>

              <p><img src="../../images/price_by_carat_and_clarity.png" alt="Price by Carat and Clarity. " loading="lazy" /></p>
            </li>
            <li>Holding carat weight constant, we see on the scatter plot shown above that diamonds with lower clarity are almost always cheaper than diamonds with better clarity: diamonds that are “Inte<span class="notranslate">rna</span>lly Flawless” are the most expensive whereas “I1” are the least expensive clarity types. So clarity explains a lot of the variance found in price!</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="share-your-work">Share your work</h2>

<p>One of the most important features of <span class="notranslate">Galaxy</span> comes at the end of an analysis. When you have published striking findings, it is important that other researchers are able to reproduce your in-silico experiment. <span class="notranslate">Galaxy</span> enables users to easily share their workflows and histories with others.</p>

<p>To share a history, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon in the history panel and select <code class="language-plaintext highlighter-rouge">Share or Publish</code>. On this page you can do 3 things:</p>

<ol>
  <li><strong>Make History Accessible via Link</strong>. This generates a link that you can give out to others. Anybody with this link will be able to view your history.</li>
  <li><strong>Make History Accessible and Publish</strong>. This will not only create a link, but will also publish your history. This means your history will be listed under <code class="language-plaintext highlighter-rouge">Shared Data → Histories</code> in the top menu.</li>
  <li><strong>Share with a user</strong>. This will share the history only with specific users on the <span class="notranslate">Galaxy</span> instance.</li>
</ol>

<blockquote class="comment">
  <h3 id="comment-permissions"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Permissions</h3>
  <p>Different servers have different default permission settings. Some servers create all of your datasets completely private to you, while others make them accessible if you know the secret ID.</p>

  <p>Be sure to select <strong>Also make all objects within the History accessible</strong> whenever you make a history accessible via link, otherwise whomever you send your link to might not be able to see your history.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-share-history"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Share history</h3>

  <ol>
    <li>Share your history with your neighbour.</li>
    <li>Find the history shared by your neighbour. Histories shared with specific users can be accessed by those users under their top masthead “User” menu under <code class="language-plaintext highlighter-rouge">Histories shared with me</code>.</li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Well done! You have just performed your first analysis in <span class="notranslate">Galaxy</span>. Additionally you can share your results and methods with others.</p>



                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Galaxy provides an easy-to-use graphical user interface for often complex command-line tools</p>
</li>
                        
                        <li><p>Galaxy keeps a full record of your analysis in a history</p>
</li>
                        
                        <li><p>Workflows enable you to repeat your analysis on different data</p>
</li>
                        
                        <li><p>Galaxy can connect to external sources for data import and visualization purposes</p>
</li>
                        
                        <li><p>Galaxy provides ways to share your results and methods with others</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/introduction/faqs/">FAQ page for the Introduction to Galaxy Analyses topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Fisher1936">Fisher, R. A., 1936 <b>The use of multiple measurements in taxonomic problems</b>. Annals of Eugenics 7: 179–188. <a href="https://doi.org/10.1111/j.1469-1809.1936.tb02137.x">10.1111/j.1469-1809.1936.tb02137.x</a> <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1809.1936.tb02137.x">https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1809.1936.tb02137.x</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Galaxy 101 for everyone (Introduction to Galaxy Analyses)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Anne Fouilloux, Nadia Goué, Christopher Barnett, Michele Maroni, Olha Nahorna, Dave Clements, Saskia Hiltemann, 2021 <b>Galaxy 101 for everyone (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{introduction-galaxy-intro-101-everyone,
author = "Anne Fouilloux and Nadia Goué and Christopher Barnett and Michele Maroni and Olha Nahorna and Dave Clements and Saskia Hiltemann",
title = "Galaxy 101 for everyone (Galaxy Training Materials)",
year = "2021",
month = "11",
day = "16"
url = "\url{https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
