<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:title" content="Galaxy Training" />
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/all-faq.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/all-faq.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/all-faq.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        



<section class="tutorial faqs">
    <h1 data-toc-skip>Frequently Asked Questions</h1>
<div class="contributors-line">Authors: 
    </div>
    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h2 id=""> <a href="#">  </a></h2>
<hr/>



<h3 class="faq-area"> Which icons are available to use in my tutorial? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-icons-are-available-to-use-in-my-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which icons are available to use in my tutorial?</h3>  <p>To use icons in your tutorial, take the name of the icon, ‚Äòdetails‚Äô in this example, and write something like this in your tutorial:</p>  <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{% icon details %}</code></pre></div></div>  <p>The following icons are currently available:</p>  <div class="row">	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bullhorn" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>announcement</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>code-in</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-laptop-code" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>code-out</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>cofest</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>comment</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>congratulations</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>curriculum</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>details</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>docker_image</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-envelope" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>email</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>exchange</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>event</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>feedback</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-barchart</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-bug</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-chart-select-data</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-clear</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-columns</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-cross</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-dropdown</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-eye</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-gear</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-home" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-home</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-info</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-library</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-pencil</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-refresh</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-rulebuilder-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-save</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-scratchbook</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-selector</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-star</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-tags</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-toggle</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-upload</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-arrows-alt-h" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-connection</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-new</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>-wf-report-download</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-external-link-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">galaxy</span>_instance</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>github</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>gitter</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>gtn-theme</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>hall-of-fame</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>hands_on</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>help</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-comment" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>history-annotate</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>history-share</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>instances</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>interactive_tour</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>keypoints</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>language</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>last_modification</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>level</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>license</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-linkedin" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>linkedin</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>new-history</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-file-code" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>notebook</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>objectives</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-orcid" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>orcid</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-check</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-collection</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-file</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-files</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-repeat</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-select</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>param-text</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>question</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>references</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>requirements</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-rss" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>rss-feed</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>search</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>slides</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>solution</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-sticky-note" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>sticky-note</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>time</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-file-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>text-document</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tip</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tool</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>topic</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>trophy</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>tutorial</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fab fa-twitter" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>twitter</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>warning</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-chevron-circle-right" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>wf-input</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span>-runtime-toggle</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span></div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div><span class="notranslate">workflow</span>-run</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>video</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-play-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>video-slides</div>	</div>	<div class="col-md-2 col-sm-3" style="text-align: center">		<div style="font-size: 400%"><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">icon[0]</span></div>		<div>zenodo_link</div>	</div></div></blockquote>





 <br><br> 
<h2 id="DESeq2"> <a href="#DESeq2"> Deseq2 </a></h2>
<hr/>



<h3 class="faq-area"> Regarding DESeq2, in the tutorial you used the normalised count table. Some people use VST normalised counts or rlog normalised counts for visualisation (heatmaps),  would you recommend it ? And second question, regarding the heatmap2, I think this depends on the data you analyse but do you have any advise on how to select the clustering method and the distance method? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-regarding-deseq2-in-the-tutorial-you-used-the-normalised-count-table-some-people-use-vst-normalised-counts-or-rlog-normalised-counts-for-visualisation-heatmaps--would-you-recommend-it--and-second-question-regarding-the-heatmap2-i-think-this-depends-on-the-data-you-analyse-but-do-you-have-any-advise-on-how-to-select-the-clustering-method-and-the-distance-method"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Regarding DESeq2, in the tutorial you used the normalised count table. Some people use VST normalised counts or rlog normalised counts for visualisation (heatmaps),  would you recommend it ? And second question, regarding the heatmap2, I think this depends on the data you analyse but do you have any advise on how to select the clustering method and the distance method?</h3>  <p>this depends on what you would like to do with the table. The DESeq2 wrapper in <span class="notranslate">Galaxy</span> can output all of these, and there is a nice discussion in the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#count-data-transformations">DESeq2 vignette</a> about this topic.</p></blockquote>





 <br><br> 
<h2 id="De novo transcriptome reconstruction with RNA-Seq"> <a href="#De novo transcriptome reconstruction with RNA-Seq"> De novo transcriptome reconstruction with <span class="notranslate">rna</span>-seq </a></h2>
<hr/>



<h3 class="faq-area"> I‚Äôm using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why? <a href="/training-material/topics/transcriptomics/faqs/transcript.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-im-using-the-same-training-data-tools-and-parameters-as-the-tutorial-but-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I‚Äôm using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states. Why?</h3>  <p>This is okay! Many aspects of the tutorial can potentially affect the exact results you obtain. For example, the reference genome version used and versions of tools. It‚Äôs less important to get the exact results shown in the tutorial, and more important to understand the concepts so you can apply them to your own data.</p></blockquote>





 <br><br> 
<h2 id="RNA-seq"> <a href="#RNA-seq"> <span class="notranslate">Rna</span>-seq </a></h2>
<hr/>



<h3 class="faq-area"> I have <span class="notranslate">RNA</span>seq data for pilot experiment for differential <span class="notranslate">expression</span> in host associated bacterium. One dataset is obtained from bacterial culture, but the other comes from bacteria obtained from the host (plant). I expect strong contamination of the second sample with host <span class="notranslate">RNA</span> <span class="notranslate">reads</span>. Should I filter out <span class="notranslate">reads</span> from the host before performing the analysis (if so, what tools I could use for that), or could I just ignore the contamination (since I will use the bacterial genome to map the <span class="notranslate">reads</span>, it will disregard any host associated <span class="notranslate">reads</span>)? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnaseq_data.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-have-rnaseq-data-for-pilot-experiment-for-differential-expression-in-host-associated-bacterium-one-dataset-is-obtained-from-bacterial-culture-but-the-other-comes-from-bacteria-obtained-from-the-host-plant-i-expect-strong-contamination-of-the-second-sample-with-host-rna-reads-should-i-filter-out-reads-from-the-host-before-performing-the-analysis-if-so-what-tools-i-could-use-for-that-or-could-i-just-ignore-the-contamination-since-i-will-use-the-bacterial-genome-to-map-the-reads-it-will-disregard-any-host-associated-reads"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I have <span class="notranslate">RNA</span>seq data for pilot experiment for differential <span class="notranslate">expression</span> in host associated bacterium. One dataset is obtained from bacterial culture, but the other comes from bacteria obtained from the host (plant). I expect strong contamination of the second sample with host <span class="notranslate">RNA</span> <span class="notranslate">reads</span>. Should I filter out <span class="notranslate">reads</span> from the host before performing the analysis (if so, what tools I could use for that), or could I just ignore the contamination (since I will use the bacterial genome to map the <span class="notranslate">reads</span>, it will disregard any host associated <span class="notranslate">reads</span>)?</h3>  <p>You could map both sets of <span class="notranslate">reads</span> to the reference genome. You are right - no pre-filtering required as the host <span class="notranslate">reads</span> shouldn‚Äôt map to the ref and will be excluded.</p></blockquote>





 <br><br> 
<h2 id="analysis"> <a href="#analysis"> Analysis </a></h2>
<hr/>



<h3 class="faq-area"> Do I need to create collections to run MaxQuant analysis or can I use single sample inputs? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Collections are not necessary to run MaxQuant but they make the history more clean and easier to navigate. The multiple datasets options allows to select multiple files that are not part of a collection and will give the same result as with a collection as input.</p>






<h3 class="faq-area"> How can I include custom modifications into MaxQuant in <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Unfortunately the inclusion of custom modifications is not possible by the user because it requires profound changes in the underlying code. Please let us know the modification you need by creating a new issue: https://github.com/<span class="notranslate">galaxy</span>proteomics/tools-<span class="notranslate">galaxy</span>p/issues entitled MaxQuant new modification request.</p>






<h3 class="faq-area"> Which isobaric labeled quantification methods does MaxQuant in <span class="notranslate">Galaxy</span> support? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>The current MaxQuant version supports: iTRAQ 4 and 8 plex; TMT 2,6,8,10,11 plex; iodoTMT6plex. Includion of TMT16 plex is in preparation.</p>






<h3 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p>






<h3 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p>






<h3 class="faq-area"> Results may vary <a href="/training-material/faqs/galaxy/analysis_results_may_vary.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>






<h3 class="faq-area"> Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h3>
When you get a red dataset in your history, it means something went wrong. But how can you find out what it was? And how can you report errors?
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-troubleshooting-errors"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Troubleshooting errors</h3>  <p>When someting goes wrong in <span class="notranslate">Galaxy</span>, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your <span class="notranslate">Galaxy</span> errors.</p>  <ol>  <li><strong>Expand</strong> the red history dataset by clicking on it.    <ul>      <li>Sometime you can already see an error message here</li>    </ul>  </li>  <li>    <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></p>  </li>  <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:    <ul>      <li>Expand the history item</li>      <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>      <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:        <ul>          <li>Tool Standard Output</li>          <li>Tool Standard Error</li>        </ul>      </li>    </ul>  </li>  <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.    <ul>      <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></li>      <li>Write down any information you think might help solve the problem        <ul>          <li>See <a href="/training-material/faqs/gtn/issues_galaxy.html">this FAQ</a> on how to write good bug reports</li>        </ul>      </li>      <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> Report</strong> button</li>      <li>In the meantime, you can ask for help in the <a href="https://gitter.im/galaxyproject/Lobby"><span class="notranslate">Galaxy</span> Gitter Channel</a> or the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a>, or you can browse the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help Forum</a> to see if others have encountered the same problem before.</li>    </ul>  </li></ol></blockquote>






<h3 class="faq-area"> Creating an account <a href="/training-material/faqs/galaxy/creating_account.html"><i class="fas fa-external-link-alt"></i></a> </h3>
To get access to all features of a <span class="notranslate">Galaxy</span> instance, you need to create an account.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-an-account"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating an account</h3>  <p>To create an account:</p>  <ol>  <li>Click <strong>Login or Register</strong></li>  <li>At the bottom of the form click <strong>Register here</strong></li>  <li>Fill the form and click <strong>Create</strong></li></ol></blockquote>






<h3 class="faq-area"> I got slightly different numbers than were in the tutorial <a href="/training-material/topics/introduction/faqs/ucsc.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-got-slightly-different-numbers-than-were-in-the-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I got slightly different numbers than were in the tutorial</h3>  <p>This tutorial uses UCSC which is constantly updating it‚Äôs data! As a result it gets outdated very quickly before we can update it :( But it‚Äôs ok! It‚Äôs expected here to get different numbers.</p></blockquote>





 <br><br> 
<h2 id="ansible"> <a href="#ansible"> Ansible </a></h2>
<hr/>



<h3 class="faq-area"> Operating system compatibility <a href="/training-material/topics/admin/faqs/admin-testing.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-operating-system-compatibility"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Operating system compatibility</h3>  <p>These Ansible roles and training materials were last tested on Centos 7 and Ubuntu 18.04, but will probably work on other RHEL and Debian variants.</p>  <p>The roles that are used in these training are currently used by <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.*</code>, and other, servers in maintaining their infrastructure. (<a href="https://github.com/galaxyproject/infrastructure-playbook/">US</a>, <a href="https://github.com/usegalaxy-eu/infrastructure-playbook">EU</a>, both are running CentOS 7)</p>  <p>If you have an issue running these trainings on your OS flavour, please report <a href="https://github.com/galaxyproject/training-material/issues/new">the issue</a> in the training material and we can see if it is possible to solve.</p></blockquote>






<h3 class="faq-area"> Running Ansible on your remote machine <a href="/training-material/topics/admin/faqs/ansible_local.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-ansible-on-your-remote-machine"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running Ansible on your remote machine</h3>  <p>It is possible to have ansible installed on the remote machine and run it there, not just from your local machine connecting to the remote machine.</p>  <p>Your hosts file will need to use <code class="language-plaintext highlighter-rouge">localhost</code>,  and whenever you run playbooks with <code class="language-plaintext highlighter-rouge">ansible-playbook -i hosts playbook.yml</code>, you will need to add <code class="language-plaintext highlighter-rouge">-c local</code> to your command.</p>  <p>Be <strong>certain</strong> that the playbook that you‚Äôre writing on the remote machine is stored somewhere safe, like your user home directory, or backed up on your local machine. The cloud can be unreliable and things can disappear at any time.</p></blockquote>





 <br><br> 
<h2 id="collections"> <a href="#collections"> Collections </a></h2>
<hr/>



<h3 class="faq-area"> Adding a tag to a collection <a href="/training-material/faqs/galaxy/collections_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag-to-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag to a collection</h3>  <ul>  <li>Click on the collection</li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code> in the <code class="language-plaintext highlighter-rouge">Add tags</code> field</p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Press <kbd>Enter</kbd></li>  <li>Check that the tag is appearing below the collection name</li></ul></blockquote>






<h3 class="faq-area"> Creating a dataset collection <a href="/training-material/faqs/galaxy/collections_build_list.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a dataset collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>    <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>  </li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>






<h3 class="faq-area"> Creating a paired collection <a href="/training-material/faqs/galaxy/collections_build_list_paired.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-paired-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a paired collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build List of Dataset Pairs</strong></p>  </li>  <li>Change the text of <em>unpaired forward</em> to a common selector for the forward <span class="notranslate">reads</span></li>  <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse <span class="notranslate">reads</span></li>  <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>






<h3 class="faq-area"> Renaming a collection <a href="/training-material/faqs/galaxy/collections_rename.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a collection</h3>  <ol>  <li>Click on the collection</li>  <li>Click on the name of the collection at the top</li>  <li>Change the name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>





 <br><br> 
<h2 id="contributing"> <a href="#contributing"> Contributing </a></h2>
<hr/>



<h3 class="faq-area"> How to Contribute to <span class="notranslate">Galaxy</span> <a href="/training-material/topics/dev/faqs/contributing.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Contributing to <span class="notranslate">Galaxy</span> is a multi-step proces, this will guide you through it.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-to-contribute-to-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How to Contribute to <span class="notranslate">Galaxy</span></h3>  <p>To contribute to <span class="notranslate">galaxy</span>, a GitHub account is required. Changes are proposed via a <a href="https://docs.github.com/en/github/collaborating-with-pull-requests">pull request</a>. This allows the project maintainers to review the changes and suggest improvements.</p>  <p>The general steps are as follows:</p>  <ol>  <li>Fork the <span class="notranslate">Galaxy</span> repository</li>  <li>Clone your fork</li>  <li>Make changes in a new branch</li>  <li>Commit your changes, push branch to your fork</li>  <li>Open a pull request for this branch in the <span class="notranslate">upstream</span> <span class="notranslate">Galaxy</span> repository</li></ol>  <blockquote class="details">  <h3 data-toc-skip id="details-git-github-and-galaxy-core"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Git, Github, and <span class="notranslate">Galaxy</span> Core</h3>  <p>For a lot more information about Git branching and managing a repository on Githubsee the <a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a>tutorial.</p>  <p>The <a href="/training-material/topics/dev/tutorials/architecture/slides.html"><span class="notranslate">Galaxy</span> Core Architecture slides</a> have a lot of import <span class="notranslate">Galaxy</span> core related information related to branches,project management, and contributing to <span class="notranslate">Galaxy</span> - under the Project Management section of the slides.</p></blockquote></blockquote>





 <br><br> 
<h2 id="contributors"> <a href="#contributors"> Contributors </a></h2>
<hr/>



<h3 class="faq-area"> How does the GTN implement the "Ten simple rules for collaborative lesson development" <a href="/training-material/faqs/gtn/collaborative_development.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>The GTN framwework is inherently collaborative and community-driven, and comprises a growing number of contributors with expertise in a wide range of scientific and technical domains. Given this highly collaborative nature of a community with very different skill sets, the GTN framework has evolved over the years to facilitate the contribution and maintenance of the tutorials. We aim to adhere to best-practice guidelines for collaborative lesson development described in <span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>. The structure of the tutorials and repository has been made modular with unified syntax and use of snippets enabling easy access for authors to add common tips and tricks new users might need to know. This system allows for easy updating of all tutorials, if there is a change in tools or interface. More generally, we continually strive to lower contribution barriers for content creators by providing a framework that is easy to use for training developers regardless of their level of knowledge of the underlying technical framework.</p><p>Implementation of the ‚ÄúTen simple rules for collaborative lesson development‚Äù (<span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>) in the training materia:</p><table>  <thead>    <tr>      <th>Rules</th>      <th>Implementation in the GTN framework</th>    </tr>  </thead>  <tbody>    <tr>      <td>Clarify audience</td>      <td>Tutorial metadata includes level indicators (introductory, intermediate, advanced) and a list of prerequisite tutorials as recommended prior knowledge. This information is rendered at the top of each tutorial.</td>    </tr>    <tr>      <td>Make lessons modular</td>      <td>Development of small tutorials linked together via learning paths</td>    </tr>    <tr>      <td>Teach best practice lesson development</td>      <td>We maintain the topic <a href="/training-material/topics/contributing/"><em>Contributing to the <span class="notranslate">Galaxy</span> Training Material</em></a> including numerous tutorials describing how to create new content. Furthermore, quarterly online collaboration fest (CoFests) are organized, where contributors can get direct support. Development of a Train the Trainer program and a mentoring program for instructors, in which lesson development is taught</td>    </tr>    <tr>      <td>Encourage and empower contributors</td>      <td>Involve them in reviews. Mentor them. Encourage them to become maintainers.</td>    </tr>    <tr>      <td>Build community around lessons</td>      <td>Quarterly online collaboration fest (CoFests) and Community calls. Chat on our Gitter/Matrix channel.</td>    </tr>    <tr>      <td>Publish periodically and recognize contributions</td>      <td>Author listed on tutorials. Hall of fame listing all contributors. Full tutorial citation at the end of the tutorial. Tweet about new or updated tutorials. List of new or updated tutorials in <span class="notranslate">Galaxy</span> Community newsletter. Soon: publication of tutorials via article</td>    </tr>    <tr>      <td>Evaluate lessons at several scales</td>      <td>Tutorial change (Pull Request) review. Embedded feedback form in tutorials for trainee feedback. Instructor feedback. Automatic <span class="notranslate">workflow</span> testing</td>    </tr>    <tr>      <td>Reduce, re-use, recycle</td>      <td>Sharing content between tutorials, specially using snippets. Development of small modular tutorials linked by learning paths</td>    </tr>    <tr>      <td>Link to other resources</td>      <td>Links to original paper, documentation, exte<span class="notranslate">rna</span>l tutorials and other material</td>    </tr>    <tr>      <td>You can‚Äôt please everyone</td>      <td>but we can try (several different <span class="notranslate">Galaxy</span> introduction tutorials for different audience). Aim to clearly state what the tutorial does and does not cover, at the start.</td>    </tr>  </tbody></table>






<h3 class="faq-area"> Thanks! <a href="/training-material/faqs/gtn/contributors_appreciated.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>First off, thanks for your interest in contributing to the <span class="notranslate">Galaxy</span> training materials!</p><p>Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.</p><p>We will address your issues and/or assess your change proposal as promptly as we can, and help you become a member of our community. You can also check our <a href="/training-material/topics/contributing/">tutorials</a> for more details.</p>






<h3 class="faq-area"> How can I get started with contributing? <a href="/training-material/faqs/gtn/contributors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project.</p><h4 id="1-checking-existing-tutorials">1. Checking existing tutorials</h4><p>A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?</p><p>Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.</p><p><em>Basic</em></p><ul>  <li><strong>Test</strong> the tutorial on a running <span class="notranslate">Galaxy</span> instance    <ul>      <li>For example <a href="https://usegalaxy.org"><span class="notranslate">Galaxy</span> Main</a>, <a href="https://usegalaxy.eu"><span class="notranslate">Galaxy</span> Europe</a>, or <a href="https://usegalaxy.org.au"><span class="notranslate">Galaxy</span> Australia</a></li>      <li>Report any issues you run into</li>    </ul>  </li>  <li><strong>Language</strong> editing    <ul>      <li>Fix spelling and grammar mistakes</li>      <li>Simplify the English (to make it more accessible)</li>    </ul>  </li></ul><p><em>Intermediate</em></p><ul>  <li><strong>Metadata</strong>    <ul>      <li>Are the objectives, keypoints and time estimate filled in?</li>      <li>Do they fit with the contents of the tutorial?</li>    </ul>  </li>  <li><strong>Content</strong>    <ul>      <li>Is there enough background information provided in the introduction section and throughout the tutorial?</li>      <li><strong>Question boxes</strong>        <ul>          <li>Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)</li>          <li>Check that answers are still up-to-date</li>        </ul>      </li>      <li><strong>Screenshots and Videos</strong>        <ul>          <li>Make sure there is also a textual description of the image/video contents</li>          <li>Does the screenshot add value to the tutorial or can it be removed?</li>        </ul>      </li>    </ul>  </li></ul><p><em>Advanced</em></p><ul>  <li><strong>Workflows</strong>    <ul>      <li>Add a <span class="notranslate">workflow</span> definition file <code class="language-plaintext highlighter-rouge">.ga</code> if none is present</li>      <li>Check that the existing <span class="notranslate">workflow</span> is up-to-date with the tutorial contents</li>      <li>Enable <a href="https://github.com/usegalaxy-eu/workflow-testing"><span class="notranslate">workflow</span> testing</a></li>    </ul>  </li>  <li><strong>Tours</strong>    <ul>      <li>Add a tour if none exists</li>      <li>Run the existing tour and check that it is up-to-date with the tutorial contents</li>    </ul>  </li>  <li><strong>Datasets</strong>    <ul>      <li>Check that all datasets used in the tutorial are present in Zenodo</li>      <li>Add a data-library.yaml file if none exists</li>    </ul>  </li></ul><h4 id="2-reviewing-pull-requests">2. Reviewing pull requests</h4><p>Another great way to help out the project is by reviewing <a href="https://github.com/galaxyproject/training-material/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc">open pull requests</a>. You can use the above checklist as a guide for your review. Some documentation about how to add your review in the GitHub interface can be found <a href="https://help.github.com/articles/about-pull-request-reviews/">here</a></p>






<h3 class="faq-area"> How can I contribute in "advanced" mode? <a href="/training-material/faqs/gtn/contributors_git_advanced.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Most of the content is written in <a href="https://guides.github.com/features/mastering-markdown/">GitHub Flavored Markdown</a> with some metadata (or variables) found in <a href="http://yaml.org/">YAML</a> files. Everything is stored on our <a href="https://github.com/galaxyproject/training-material">GitHub repository</a>. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (<em>e.g.</em> <code class="language-plaintext highlighter-rouge">transcriptomics</code> directory contains the material related to transcriptomic analysis). Each topic has the following <a name="structure">structure</a>:</p><p><img src="/training-material/shared/images/repo_organization.png" alt="Structure of the repository" width="400px" /></p><ul>  <li>a metadata file in YAML format</li>  <li>a directory with the topic introduction slide deck in Markdown with introductions to the topic</li>  <li>    <p>a directory with the tutorials:</p>    <p>Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:</p>    <ul>      <li>a tutorial file written in Markdown with hands-on</li>      <li>an optional slides file in Markdown with slides to support the tutorial</li>      <li>a directory with <span class="notranslate">Galaxy</span> Interactive Tours to reproduce the tutorial</li>      <li>a directory with workflows extracted from the tutoria</li>      <li>a YAML file with the links to the input data needed for the tutorial</li>      <li>a YAML file with the description of needed tools to run the tutorial</li>    </ul>  </li>  <li>a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).</li></ul><p>To manage changes, we use <a href="https://guides.github.com/introduction/flow/">GitHub flow</a> based on Pull Requests (check our <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">tutorial</a>):</p><ol>  <li><a href="https://help.github.com/articles/fork-a-repo/">Create a fork</a> of this repository on GitHub</li>  <li>Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (<code class="language-plaintext highlighter-rouge">git submodule init</code> and <code class="language-plaintext highlighter-rouge">git submodule update</code>)</li>  <li>Create a new branch in your local copy for each significant change</li>  <li>Commit the changes in that branch</li>  <li>Push that branch to your fork on GitHub</li>  <li>Submit a pull request from that branch to the <a href="https://github.com/galaxyproject/training-material">original repository</a></li>  <li>If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically</li>  <li>Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.</li></ol><p>Globally, the process of development of new content is open and transparent:</p><ol>  <li>Creation of a branch derived from the main branch of the GitHub repository</li>  <li>Initialization of a new directory for the tutorial</li>  <li>Filling of the metadata with title, questions, learning objectives, etc</li>  <li>Generation of the input dataset for the tutorial</li>  <li>Filling of the tutorial content</li>  <li>Extraction of the workflows of the tutorial</li>  <li>Automatic extraction of the required tools to populate the tool file</li>  <li>Automatic annotation of the public <span class="notranslate">Galaxy</span> servers</li>  <li>Generation of an interactive tour for the tutorial with the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tourbuilder</a> web-browser extension</li>  <li>Upload of the datasets to Zenodo and addition of the links in the data library file.</li>  <li>Once ready, opening a Pull Request</li>  <li>Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI</li>  <li>Review of the content by several other instructors via discussions</li>  <li>After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI</li>  <li>Regeneration of the website and publication on <a href="/training-material/">https://training.<span class="notranslate">galaxy</span>project.org/training-material/</a></li>  <li>Generation of PDF artifacts of the tutorials and slides and upload on the FTP server</li>  <li>Population of <a href="https://tess.elixir-europe.org/">TeSS</a>, the ELIXIR‚Äôs Training Portal, via the metadata</li></ol><p><img src="/training-material/shared/images/development_process.png" alt="Development process" /></p><p>To learn how to add new content, check out our <a href="/training-material/topics/contributing/">series of tutorials on creating new content</a>:</p><ol>   <li><a href="/training-material/topics/contributing/slides/introduction.html">Overview of the <span class="notranslate">Galaxy</span> Training Material</a></li>       <li><a href="/training-material/topics/contributing/tutorials/slides-with-video/tutorial.html">Adding auto-generated video to your slides</a></li>          <li><a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a></li>          <li><a href="/training-material/topics/contributing/tutorials/github-interface-contribution/tutorial.html">Contributing with GitHub via its interface</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">Creating a new tutorial</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">Creating content in Markdown</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-tours/tutorial.html">Creating Interactive <span class="notranslate">Galaxy</span> Tours</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html">Creating Slides</a></li>           <li><a href="/training-material/topics/contributing/tutorials/generating-pdf/tutorial.html">Generating PDF artefacts of the website</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-topic/tutorial.html">Including a new topic</a></li>          <li><a href="/training-material/topics/contributing/tutorials/running-jekyll/tutorial.html">Running the GTN website locally</a></li>          <li><a href="/training-material/topics/contributing/tutorials/gitpod/tutorial.html">Running the GTN website online using GitPod</a></li>          <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html">Tools, Data, and Workflows for tutorials</a></li>          <li><a href="/training-material/topics/contributing/tutorials/admin-knitting/tutorial.html">Updating diffs in admin training</a></li>      </ol><p>We also strongly recommend you read and follow <a href="https://carpentries.org/">The Carpentries</a> recommendations on <a href="https://carpentries.github.io/lesson-example/01-design/">lesson design</a> and <a href="https://carpentries.github.io/instructor-training/15-lesson-study/">lesson writing</a> if you plan to add or change some training materials, and also to check the <a href="#structure">structure of the training material</a>.</p>






<h3 class="faq-area"> How can I fix mistakes or expand an existing tutorial using the GitHub interface? <a href="/training-material/faqs/gtn/contributors_github_interface.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Check our tutorial to learn how to use the GitHub interface (soon‚Ä¶)</p>






<h3 class="faq-area"> How can I give feedback? <a href="/training-material/faqs/gtn/contributors_giving_feedback.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>At the end of each tutorial, there is a link to <a href="https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit">a feedback form</a>. We use this information to improve our tutorials.</p><p>For global feedbacks, you can open an <a href="https://github.com/galaxyproject/training-material/issues/new">issue on GitHub</a>, write us on <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or send us an <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>.</p>






<h3 class="faq-area"> What can I do to help the project? <a href="/training-material/faqs/gtn/contributors_helping_project.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>In <a href="https://github.com/galaxyproject/training-material/issues">issues</a>, you will find lists of issues to fix and features to implement (with the <a href="https://github.com/galaxyproject/training-material/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Anewcomer-friendly">‚Äúnewcomer-friendly‚Äù</a> label for example). Feel free to work on them!</p>






<h3 class="faq-area"> How can I report mistakes or errors? <a href="/training-material/faqs/gtn/contributors_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>The easiest way to start contributing is to <a href="https://github.com/galaxyproject/training-material/issues/new">file an issue</a> to tell us about a problem such as a typo, spelling mistake, or a factual error. You can then introduce yourself and meet some of our community members.</p>






<h3 class="faq-area"> How can I test an Interactive Tour? <a href="/training-material/faqs/gtn/contributors_tours.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Perhaps you‚Äôve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tour builder</a> browser extension.</p><ol>  <li>Install the Tour Builder extension to your browser (<a href="https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg">Chrome Web Store</a>, <a href="https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/">Firefox add-on</a>).</li>  <li>Navigate to a <span class="notranslate">Galaxy</span> instance supporting the tutorial. To find which <span class="notranslate">Galaxy</span> instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the use<span class="notranslate">galaxy</span>.* instances (<a href="https://usegalaxy.org"><span class="notranslate">Galaxy</span> Main</a>, <a href="https://usegalaxy.eu"><span class="notranslate">Galaxy</span> Europe</a>, or <a href="https://usegalaxy.org.au"><span class="notranslate">Galaxy</span> Australia</a>) is usually a good bet.</li>  <li>Start the Tour Builder plugin by clicking on the icon in your browser menu bar</li>  <li>Copy the contents of the <code class="language-plaintext highlighter-rouge">tour.yaml</code> file into the Tour builder editor window</li>  <li>Click <code class="language-plaintext highlighter-rouge">Save</code> and then <code class="language-plaintext highlighter-rouge">Run</code></li></ol>






<h3 class="faq-area"> How can I create new content without dealing with git? <a href="/training-material/faqs/gtn/contributors_without_git.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you feel uncomfortable with using the git and the <a href="https://guides.github.com/introduction/flow/">GitHub flow</a>, you can write a new tutorial with any text editor and then contact us (via <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>). We will work together to integrate the new content.</p>






<h3 class="faq-area"> How does the GTN ensure our training materials are FAIR? <a href="/training-material/faqs/gtn/fair_training.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>This infrastructure has been developed in accordance with the FAIR (Findable, Accessible, Interoperable, Reusable) principles for training materials <span class="citation"><a href="#Garcia2020">Garcia <i>et al.</i> 2020</a></span>. Following these principles enables trainers and trainees to find, reuse, adapt, and improve the available tutorials.</p><table>  <thead>    <tr>      <th>10 Simple Rules</th>      <th>Implementation in GTN framework</th>    </tr>  </thead>  <tbody>    <tr>      <td>Plan to share your training materials online</td>      <td>Online <a href="https://training.galaxyproject.org/">training material portfolio</a>, managed via <a href="https://github.com/galaxyproject/training-material">a public GitHub repository</a></td>    </tr>    <tr>      <td>Improve findability of your training materials by properly describing them</td>      <td>Rich metadata associated with each tutorial that are visible and accessible via schema.org on each tutorial webpage.</td>    </tr>    <tr>      <td>Give your training materials a unique identity</td>      <td>URL persistency with redirection in case of renaming of tutorials. Data used for tutorials stored on Zenodo and associated with a Digital Object Identifiers (DOI)</td>    </tr>    <tr>      <td>Register your training materials online</td>      <td>Tutorials automatically registered on TeSS, the <a href="https://tess.elixir-europe.org/">ELIXIR‚Äôs Training e-Support System</a></td>    </tr>    <tr>      <td>If appropriate, define access rules for your training materials</td>      <td>Online and free to use without registration</td>    </tr>    <tr>      <td>Use an interoperable format for your training materials</td>      <td>Content of the tutorials and slides written in Markdown. Metadata associated with tutorials stored in YAML, and workflows in JSON. All of this metadata is available <a href="https://training.galaxyproject.org/training-material/api/">from the GTN‚Äôs API</a></td>    </tr>    <tr>      <td>Make your training materials (re-)usable for trainers</td>      <td>Online. Rich metadata associated with each tutorial: title, contributor details, license, description, learning outcomes, audience, requirements, tags/keywords, duration, date of last revision. Strong technical support for each tutorial: <span class="notranslate">workflow</span>, data on Zenodo and also available as data libraries on Use<span class="notranslate">Galaxy</span>.*, tools installable via the <span class="notranslate">Galaxy</span> Tool Shed, list of possible <span class="notranslate">Galaxy</span> instances with the needed tools.</td>    </tr>    <tr>      <td>Make your training materials (re-)usable for trainees</td>      <td>Online and easy to follow hands-on tutorials. Rich metadata with ‚ÄúSpecific, Measurable, Attainable, Realistic and Time bound‚Äù (SMART) learning outcomes following Bloom‚Äôs taxonomy. Requirements and follow-up tutorials to build learning path. List of <span class="notranslate">Galaxy</span> instances offering needed tools, data on Zenodo and also available as data libraries on Use<span class="notranslate">Galaxy</span>.*. Support chat embedded in tutorial pages.</td>    </tr>    <tr>      <td>Make your training materials contribution friendly and citable</td>      <td>Open and collaborative infrastructure with contribution guidelines, a CONTRIBUTING file and a chat. Details to cite tutorials and give credit to contributors available at the end of each tutorial.</td>    </tr>    <tr>      <td>Keep your training materials up-to-date</td>      <td>Open, collaborative and transparent peer-review and curation process. Short time between updates.</td>    </tr>  </tbody></table>






<h3 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/gtn/issues_galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p><p><strong>What to include</strong></p><ol>  <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>  <li>Which server(s) did you run those commands on?</li>  <li>What account/use<span class="notranslate">rna</span>me did you use?</li>  <li>Where did it go wrong?</li>  <li>What were the stdout/stderr of the tool that failed? Include the text.</li>  <li>Did you try any workarounds? What results did those produce?</li>  <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>  <li>If there are job IDs, please include them as text so administrators don‚Äôt have to manually transcribe the job ID in your picture.</li></ol><p>It makes the process of answering ‚Äòbug reports‚Äô much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p><p><strong>What does a GOOD bug report look like?</strong></p><p>The people who provide support for <span class="notranslate">Galaxy</span> are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p><blockquote class="quote">  <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My use<span class="notranslate">rna</span>me is jane-doe.</p>  <p>Here is everything that I know:</p>  <ul>    <li>The dataset is green, the job did not fail</li>    <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>    <li>I have read it but I do not understand what X/Y means.</li>    <li>The job ID from the output information page is 123123abdef.</li>    <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>  </ul>  <p>Could you help me?</p></blockquote>





 <br><br> 
<h2 id="data upload"> <a href="#data upload"> Data upload </a></h2>
<hr/>



<h3 class="faq-area"> Importing data from a data library <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>






<h3 class="faq-area"> Importing via links <a href="/training-material/faqs/galaxy/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>






<h3 class="faq-area"> Importaci√≥n por medio de enlaces <a href="/training-material/faqs/galaxy-es/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importaci√≥n-por-medio-de-enlaces"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importaci√≥n por medio de enlaces</h3>  <ul>  <li>Copia los enlaces</li>  <li>    <p>Abre el manejador de carga de datos de <span class="notranslate">Galaxy</span> (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> en la parte superior derecha del panel de herramientas)</p>  </li>  <li>Selecciona ‚ÄòPegar/Traer datos‚Äô <strong>Paste/Fetch Data</strong></li>  <li>    <p>Copia los enlaces en el campo de textos</p>  </li>  <li>    <p>Presiona ‚ÄòIniciar‚Äô <strong>Start</strong></p>  </li>  <li>    <p><strong>Close</strong> Cierra la ventana.</p>  </li>  <li><span class="notranslate">Galaxy</span> utiliza los URLs como nombres de forma predeterminada , as√≠ que los tendr√°s que cambiar a algunos que sean m√°s √∫tiles o informativos.the window</li></ul></blockquote>





 <br><br> 
<h2 id="datasets"> <a href="#datasets"> Datasets </a></h2>
<hr/>



<h3 class="faq-area"> Adding a custom database/build (dbkey) <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> may have several reference genomes built-in, but you can also create your own.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-custom-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a custom database/build (dbkey)</h3>  <ul>  <li>In the top menu bar, go to the <em>User</em>, and select <em>Custom Builds</em></li>  <li>Choose a name for your reference build</li>  <li>Choose a dbkey for your reference build</li>  <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>  <li>Under <strong>FASTA-file</strong>, select your fasta file</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>






<h3 class="faq-area"> Upload fastqsanger datasets via links <a href="/training-material/faqs/galaxy/dataset_upload_fastqsanger_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-fastqsanger-datasets-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload fastqsanger datasets via links</h3>  <ol>  <li>    <p>Click on <strong>Upload Data</strong> on the top of the left panel:</p>    <p><img src="/training-material/faqs/galaxy/images/upload_data_button.png" alt="UploadDataButton" /></p>  </li>  <li>    <p>Click on <strong>Paste/Fetch</strong>:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_button.png" alt="PasteFetchButton" /></p>  </li>  <li>    <p>Paste URL into text box that would appear:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_modal.png" alt="PasteFetchModal" /></p>  </li>  <li>    <p>Set <strong>Type (set all)</strong> to <code class="language-plaintext highlighter-rouge">fastqsanger</code> or, if your data is compressed as in URLs above (they have <code class="language-plaintext highlighter-rouge">.gz</code> extensions), to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_set_data_type.png" alt="ChangeTypeDropDown" />:</p>  </li></ol></blockquote>






<h3 class="faq-area"> Adding a tag <a href="/training-material/faqs/galaxy/datasets_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Tags can help you to better organize your history and track datasets.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>






<h3 class="faq-area"> Changing the datatype <a href="/training-material/faqs/galaxy/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select  your desired datatype</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>






<h3 class="faq-area"> Cambiar el tipo de datos <a href="/training-material/faqs/galaxy-es/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-cambiar-el-tipo-de-datos"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Cambiar el tipo de datos</h3>  <ul>  <li>Selecciona sobre el <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>icono del l√°piz</strong> para editar los atributos del conjunto de datos</li>  <li>Selecciona en la pesta√±a <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> en la parte superior del panel central</li>  <li>Selecciona  tu tipo de datos</li>  <li>Da clic en el bot√≥n <strong>Change datatype</strong></li></ul></blockquote>






<h3 class="faq-area"> Changing database/build (dbkey) <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html"><i class="fas fa-external-link-alt"></i></a> </h3>
You can tell <span class="notranslate">Galaxy</span> which dbkey (e.g. reference genome) your dataset is associated with. This may be used by tools to automatically use the correct settings.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing database/build (dbkey)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Database/Build</strong> field</li>  <li>Select your desired database key from the dropdown list</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>






<h3 class="faq-area"> Converting the file format <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Some datasets can be transformed into a different format. <span class="notranslate">Galaxy</span> has some built-in file conversion options depending on the type of data you have.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-converting-the-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Converting the file format</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <strong>Convert</strong> tab on the top</li>  <li>Select the appropriate datatype from the list</li>  <li>Click the <strong>Convert datatype</strong> button</li></ul></blockquote>






<h3 class="faq-area"> Creating a new file <a href="/training-material/faqs/galaxy/datasets_create_new_file.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> allows you to create new files from the upload menu. You can supply the contents of the file.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new file</h3>  <ul>  <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>Paste the file contents into the text field</li>  <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li></ul></blockquote>






<h3 class="faq-area"> Detecting the datatype (file format) <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-detecting-the-datatype-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Detecting the datatype (file format)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Click the <strong>Detect datatype</strong> button to have <span class="notranslate">Galaxy</span> try to autodetect it.</li></ul></blockquote>






<h3 class="faq-area"> Renaming a dataset <a href="/training-material/faqs/galaxy/datasets_rename.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>






<h3 class="faq-area"> Upload few files (1-10) <a href="/training-material/faqs/galaxy/datasets_upload.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-few-files-1-10"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload few files (1-10)</h3>  <ol>  <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>  <li>Click on <strong>Choose local file</strong> and select the files or drop the files in the <strong>Drop files here</strong> part</li>  <li>Click on <strong>Start</strong></li>  <li>Click on <strong>Close</strong></li></ol></blockquote>






<h3 class="faq-area"> Upload many files (>10) via FTP <a href="/training-material/faqs/galaxy/datasets_upload_ftp.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-many-files-10-via-ftp"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload many files (&gt;10) via FTP</h3>  <ol>  <li>    <p>Make sure to have an FTP client installed</p>    <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>  </li>  <li>Establish FTP connection to the <span class="notranslate">Galaxy</span> server    <ol>      <li>Provide the <span class="notranslate">Galaxy</span> server‚Äôs FTP server name (e.g. <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.org</code>, <code class="language-plaintext highlighter-rouge">ftp.use<span class="notranslate">galaxy</span>.eu</code>)</li>      <li>Provide the <strong>use<span class="notranslate">rna</span>me</strong> (usually the email address) and the <strong>password</strong> on the <span class="notranslate">Galaxy</span> server</li>      <li>Connect</li>    </ol>  </li>  <li>    <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>    <p>The FTP transfer will start. We need to wait until they are done.</p>  </li>  <li>Open the Upload menu on the <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Choose FTP file</strong> on the bottom</li>  <li>Select files to import into the history</li>  <li>Click on <strong>Start</strong></li></ol></blockquote>





 <br><br> 
<h2 id="estimation of strandedness"> <a href="#estimation of strandedness"> Estimation of strandedness </a></h2>
<hr/>



<h3 class="faq-area"> Infer experiment, is it ever used in practice? I mean, most often you are aware if the <span class="notranslate">RNA</span>-seq data is stranded or not in the first place, right, because you sequenced it yourself or ordered it from a company. <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-infer-experiment-is-it-ever-used-in-practice-i-mean-most-often-you-are-aware-if-the-rna-seq-data-is-stranded-or-not-in-the-first-place-right-because-you-sequenced-it-yourself-or-ordered-it-from-a-company"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Infer experiment, is it ever used in practice? I mean, most often you are aware if the <span class="notranslate">RNA</span>-seq data is stranded or not in the first place, right, because you sequenced it yourself or ordered it from a company.</h3>  <p>This can happen in cases where you get the data from someone else, and they don‚Äôt know.</p></blockquote>






<h3 class="faq-area"> I am trying to check the strandedness of my libraries and I get unequal numbers in the infer experiments, but in the IGV it looks like it is unstranded. What does this mean? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-am-trying-to-check-the-strandedness-of-my-libraries-and-i-get-unequal-numbers-in-the-infer-experiments-but-in-the-igv-it-looks-like-it-is-unstranded-what-does-this-mean"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I am trying to check the strandedness of my libraries and I get unequal numbers in the infer experiments, but in the IGV it looks like it is unstranded. What does this mean?</h3>  <p>It‚Äôs also often the case that elimination of the second strand is not perfect, and there are genuine cases of bidirectional transcription in the genome. 70 / 30 % as in your report is not a good result for a stranded library. You can treat this as a stranded library in your analysis, but for instance you couldn‚Äôt make the conclusion that a given gene is actually transcribed from the reverse strand. Likely that the library preparation didn‚Äôt work perfectly. This can depend on many factors, one is that you need to completely digest your <span class="notranslate">DNA</span> using a high quality <span class="notranslate">DNa</span>se before doing the reverse transcription.</p></blockquote>





 <br><br> 
<h2 id="extract tool"> <a href="#extract tool"> Extract tool </a></h2>
<hr/>



<h3 class="faq-area"> Are UMIs not actually unique? <a href="/training-material/topics/transcriptomics/faqs/umi.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-are-umis-not-actually-unique"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Are UMIs not actually unique?</h3>  <p>Not strictly, but unique enough. The distribution of UMIs should ideally be uniform so that the chance of any two same UMIs capturing the same transcript (via different amplicons) is small. As barcodes have increased in size, the number of UMIs has also increased allowing for UMIs to reach more or less the same numbers of transcripts.</p></blockquote>





 <br><br> 
<h2 id="features"> <a href="#features"> Features </a></h2>
<hr/>



<h3 class="faq-area"> Using the Scratchbook to view multiple datasets <a href="/training-material/faqs/galaxy/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-the-scratchbook-to-view-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using the Scratchbook to view multiple datasets</h3>  <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in <span class="notranslate">Galaxy</span>:</p>  <ol>  <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> on the top menu bar.    <ul>      <li>You should see a little checkmark on the icon now</li>    </ul>  </li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> to view the output    <ul>      <li>You should see the output in a window overlayed over <span class="notranslate">Galaxy</span></li>      <li>You can resize this window by dragging the bottom-right corner</li>    </ul>  </li>  <li><strong>Click</strong> outside the file to exit the Scratchbook</li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a second dataset from your history    <ul>      <li>You should now see a second window with the new dataset</li>      <li>This makes it easier to compare the two outputs</li>    </ul>  </li>  <li>Repeat this for as many files as you would like to compare</li>  <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> by clicking on the icon again</li></ol></blockquote>






<h3 class="faq-area"> Uso del cuaderno de apuntes para ver varios conjuntos de datos <a href="/training-material/faqs/galaxy-es/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-uso-del-cuaderno-de-apuntes-para-ver-varios-conjuntos-de-datos"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Uso del cuaderno de apuntes para ver varios conjuntos de datos</h3>  <p>Si deseas ver dos o m√°s conjuntos de datos al mismo tiempo, puedes usar la funci√≥n <strong>Scratchbook</strong> en <span class="notranslate">Galaxy</span>:    1. <strong>Haz clic</strong> en el icono <em>Scratchbook</em> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> en la barra de men√∫ superior.       - Deber√≠a aparecer ver una peque√±a marca de verificaci√≥n en el icono    2. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> un conjunto de datos haciendo clic en el icono de ojo <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> para ver el resultado.       - Deber√≠as ver la salida en una ventana emergente sobre <span class="notranslate">Galaxy</span>       - Puedes cambiar el tama√±o de esta ventana arrastrando la esquina inferior derecha    3. <strong>Haz clic</strong> fuera del archivo para salir del Scratchbook    4. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> un segundo conjunto de datos de tu historial       - Ahora deber√≠as poder ver una segunda ventana con el nuevo conjunto de datos       - Esto hace que sea m√°s f√°cil comparar las dos salidas.    5. Repite estos pasos para todos los archivos que desees comparar.    6. Puedes desactivar <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> haciendo clic en el icono nuevamente.</p></blockquote>






<h3 class="faq-area"> Why not use Excel? <a href="/training-material/faqs/galaxy/why_not_excel.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Excel is a fantastic tool and a great place to build simple analysis models, but when it comes to scaling, <span class="notranslate">Galaxy</span> wins every time.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-not-use-excel"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why not use Excel?</h3>  <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>  <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel won‚Äôt help you here. It doesn‚Äôt track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>  <p><span class="notranslate">Galaxy</span>, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable <span class="notranslate">workflow</span> from your analysis that others (or yourself) can use on other datasets.</p>  <p>Another challenge with sp<span class="notranslate">reads</span>heet programs is that they don‚Äôt scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but you‚Äôll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p></blockquote>





 <br><br> 
<h2 id="further reading"> <a href="#further reading"> Further reading </a></h2>
<hr/>



<h3 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p>






<h3 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p>





 <br><br> 
<h2 id="galaxy admin interface"> <a href="#galaxy admin interface"> <span class="notranslate">Galaxy</span> admin interface </a></h2>
<hr/>



<h3 class="faq-area"> Install tools via the Admin UI <a href="/training-material/topics/admin/faqs/install_tool.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-install-tools-via-the-admin-ui"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Install tools via the Admin UI</h3>  <ol>  <li>Open <span class="notranslate">Galaxy</span> in your browser and type `` in the tool search box on the left. If ‚Äú‚Äù is among the search results, you can skip the following steps.</li>  <li>Access the Admin menu from the top bar (you need to be logged-in with an email specified in the <code class="language-plaintext highlighter-rouge">admin_users</code> setting)</li>  <li>Click ‚ÄúInstall and Uninstall‚Äù, which can be found on the left, under ‚ÄúTool Management‚Äù</li>  <li>Enter `` in the search interface</li>  <li>Click on the first hit, having <code class="language-plaintext highlighter-rouge">devteam</code> as owner</li>  <li>Click the ‚ÄúInstall‚Äù button for the latest revision</li>  <li>Enter ‚Äú‚Äù as the target section and click ‚ÄúOK‚Äù.</li></ol></blockquote>





 <br><br> 
<h2 id="gene"> <a href="#gene"> Gene </a></h2>
<hr/>



<h3 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/transcriptomics/faqs/dimension_reduction.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</h3>  <p>The actual data has tens of thousands of genes, and so tens of thousands of variables to consider. Even after selecting for the most variable genes and the most high quality genes, we can still be left with &gt; 1000 genes. Performing clustering on a dataset with 1000s of variables is possible, but computationally expensive. It is therefore better to perform dimension reduction to reduce the number of variables to a latent representation of these variables. These latent variables are ideally more than 10 but less than 50 to capture the variability in the data to perform clustering upon.</p></blockquote>






<h3 class="faq-area"> What exactly is a ‚ÄòGene profile‚Äô? <a href="/training-material/topics/transcriptomics/faqs/gene_profile.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-exactly-is-a-gene-profile"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What exactly is a ‚ÄòGene profile‚Äô?</h3>  <p>Think of it like a fingerprint that some cells exhibit and others don‚Äôt. It‚Äôs a small collection of genes which are up or down regulated in relation to one another. Their differences are not absolute, but relative. So if CellA has 100 counts of Gene1 and 50 counts of Gene2, this creates a relation of 2:1 between Gene1 and Gene2. If CellB has a 20 counts of Gene1 and 10 counts of Gene2, then they share the same relation. If CellA and CellB share other relations with other genes than this might be enough to say that they share a Gene profile, and will therefore likely cluster together as they describe the same cell type.</p></blockquote>






<h3 class="faq-area"> Why do we only consider highly variable genes? <a href="/training-material/topics/transcriptomics/faqs/variable_genes.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-only-consider-highly-variable-genes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we only consider highly variable genes?</h3>  <p>The non-variable genes are likely housekeeping genes, which are expressed everywhere and are not so useful for distinguishing one cell type from another. However background genes are important to the analysis and are used to generate a background baseline model for measuring the variability of the other genes.</p></blockquote>





 <br><br> 
<h2 id="historias"> <a href="#historias"> Historias </a></h2>
<hr/>



<h3 class="faq-area"> Para la creaci√≥n de un historial nuevo <a href="/training-material/faqs/galaxy-es/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Los historiales son una parte importante de <span class="notranslate">Galaxy</span>, la mayor√≠a de la gente utiliza un historial para cada an√°lisis nuevo. Aseg√∫rate siempre de darle buenos nombres a tus historiales, de tal forma que despu√©s puedas encontrar f√°cilmente tus resultados.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-para-la-creaci√≥n-de-un-historial-nuevo"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Para la creaci√≥n de un historial nuevo</h3>  <p>Haz click sobre el icono <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> en la parte superior del panel de historiales.</p></blockquote>






<h3 class="faq-area"> Cambiando el nombre de un historial <a href="/training-material/faqs/galaxy-es/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-cambiando-el-nombre-de-un-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Cambiando el nombre de un historial</h3>  <ol>  <li>Haz clic sobre  <strong>Unnamed history</strong> (o el nombre que tenga el historial sobre el que est√°s trabajando) (<strong>Haz clic para cambiar el nombre del historial</strong>) en la parte superior de tu panel de historial</li>  <li>Escribe el nombre nuevo</li>  <li>Pulsa <kbd>Enter</kbd></li></ol></blockquote>





 <br><br> 
<h2 id="histories"> <a href="#histories"> Histories </a></h2>
<hr/>



<h3 class="faq-area"> Copy a dataset to a new history <a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Sometimes you may want to use a dataset in multiple histories. You do not need to re-upload the data, but you can copy datasets from one history to another.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-copy-a-dataset-to-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Copy a dataset to a new history</h3>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Click on <strong>Copy Dataset</strong></li>  <li>    <p>Select the desired files</p>  </li>  <li>    <p>Give a relevant name to the ‚ÄúNew history‚Äù</p>  </li>  <li>Click on the new history name in the green box that have just appear to switch to this history</li></ol></blockquote>






<h3 class="faq-area"> Creating a new history <a href="/training-material/faqs/galaxy/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Histories are an important part of <span class="notranslate">Galaxy</span>, most people use a new history for every new analysis. Always make sure to give your histories good names, so you can easily find your results back later.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>






<h3 class="faq-area"> Import an history <a href="/training-material/faqs/galaxy/histories_import.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-import-an-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Import an history</h3>  <ol>  <li>Open the link to the shared history</li>  <li>Click on the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>Import history</em> button on the top right</li>  <li>Enter a title for the new history</li>  <li>Click on <strong>Import</strong></li></ol></blockquote>






<h3 class="faq-area"> Renaming a history <a href="/training-material/faqs/galaxy/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>






<h3 class="faq-area"> Searching your history <a href="/training-material/faqs/galaxy/histories_search.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-searching-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Searching your history</h3>  <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>  <ol>  <li>    <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>    <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>  </li>  <li><strong>Type</strong> a search term in this box    <ul>      <li>For example a tool name, or sample name</li>    </ul>  </li>  <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-clear</span> next to the search box</li></ol></blockquote>






<h3 class="faq-area"> Sharing your History <a href="/training-material/faqs/galaxy/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h3>
You can share your work in <span class="notranslate">Galaxy</span>. There are various ways you can give access one of your histories to other users.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sharing-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sharing your History</h3>  <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>  <ol>  <li><strong>Share via link</strong>    <ul>      <li>Open the <strong>History Options</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> menu (gear icon) at the top of your history panel        <ul>          <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History accessible</strong></li>          <li>A <strong>Share Link</strong> will appear that you give to others</li>        </ul>      </li>      <li>Anybody who has this link can view and copy your history</li>    </ul>  </li>  <li><strong>Publish your history</strong>    <ul>      <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>      <li>Anybody on this <span class="notranslate">Galaxy</span> server will see your history listed under the <em>Shared Data</em> menu</li>    </ul>  </li>  <li><strong>Share only with another user</strong>.    <ul>      <li>Click the <strong>Share with a user</strong> button at the bottom</li>      <li>Enter an email address for the user you want to share with</li>      <li>Your history will be shared only with this user.</li>    </ul>  </li>  <li><strong>Finding histories others have shared with me</strong>    <ul>      <li>Click on <strong>User</strong> menu on the top bar</li>      <li>Select <strong>Histories shared with me</strong></li>      <li>Here you will see all the histories others have shared with you directly</li>    </ul>  </li></ol>  <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <em>History options</em> icon in your history and clicking <em>Copy</em></p></blockquote>






<h3 class="faq-area"> Compartiendo un historial <a href="/training-material/faqs/galaxy-es/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h3>
Puedes compartir tu trabajo en <span class="notranslate">Galaxy</span>. Hay varias formas de dar acceso a tus historiales a otros usuarios.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-compartiendo-un-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Compartiendo un historial</h3>  <p>Compartir tu historial permite a otros importar y acceder a los conjuntos de datos, par√°metros y pasos de tu historial.</p>  <ol>  <li><strong>Compartir a trav√©s de un enlace</strong>    <ul>      <li>Abre el men√∫ <strong>Opciones de historial</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (icono de engranaje) en la parte superior del panel de historial        <ul>          <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Hacer que el historial sea accesible</strong></li>          <li>Aparecer√° un <strong>Compartir enlace</strong> que puedes dar a otros usuarios.</li>        </ul>      </li>      <li>Cualquiera que tenga este enlace puede ver y copiar tu historial.</li>    </ul>  </li>  <li><strong>Publica tu historial</strong>    <ul>      <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Hacer que el historial est√© disponible p√∫blicamente en Historias publicadas</strong></li>      <li>Cualquiera en este servidor <span class="notranslate">Galaxy</span> podr√° ver tu historial en el men√∫ <em>Datos compartidos</em></li>    </ul>  </li>  <li><strong>Comparte solo con otro usuario</strong>.    <ul>      <li>Haz clic en el bot√≥n <strong>Compartir con un usuario</strong> en la parte inferior</li>      <li>Ingresa una direcci√≥n de correo electr√≥nico del usuario con el que deseas compartir</li>      <li>Tu historial se compartir√° solo con este usuario.</li>    </ul>  </li>  <li><strong>Encontrar historiales que otros han compartido conmigo</strong>    <ul>      <li>Haz clic en el men√∫ <strong>Usuario</strong> en la barra superior</li>      <li>Selecciona <strong>Historiales compartidos conmigo</strong></li>      <li>Aqu√≠ ver√°s todos los historiales que otros han compartido contigo directamente** Nota: ** Si deseas realizar cambios en tu historial sin afectar la versi√≥n compartida, crea una copia mediante al √≠cono <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <em>Opciones de historial</em> en tu historial y haciendo clic en <em>Copiar</em></li>    </ul>  </li></ol></blockquote>






<h3 class="faq-area"> Undeleting history <a href="/training-material/faqs/galaxy/histories_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h3>
undelete your deleted histories
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-undeleting-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Undeleting history</h3>  <ul>  <li>Click on <strong>User</strong> then select <strong>Histories</strong></li>  <li>Click on <strong>Advanced search</strong> on the top left side below  <strong>Saved Histories</strong></li>  <li>On <strong>Status</strong> click <strong>Deleted</strong></li>  <li>Select the history you want to undelete using the <strong>checkbox</strong> on the left side</li>  <li>Click <strong>Undelete</strong> button below the deleted histories</li></ul></blockquote>





 <br><br> 
<h2 id="history"> <a href="#history"> History </a></h2>
<hr/>



<h3 class="faq-area"> My jobs are not running / I cannot see the history overview menu <a href="/training-material/topics/introduction/faqs/login.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-jobs-are-not-running--i-cannot-see-the-history-overview-menu"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My jobs are not running / I cannot see the history overview menu</h3>  <p>Please make sure you are logged in. At the top menu bar, you should see a section labeled ‚ÄúUser‚Äù. If you see ‚ÄúLogin/Register‚Äù here you are not logged in.</p></blockquote>





 <br><br> 
<h2 id="igv"> <a href="#igv"> Igv </a></h2>
<hr/>



<h3 class="faq-area"> Add genome and annotations to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-genome-and-annotations-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add genome and annotations to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Expand the FASTA dataset with the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>  <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>  <li>    <p>Close the window</p>    <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>  </li>  <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The annotation track should appear. Be careful that all files have the same genome ID</p>  </li></ol></blockquote>






<h3 class="faq-area"> Add genome and annotations to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/visualisation/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-genome-and-annotations-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add genome and annotations to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Expand the FASTA dataset with the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>  <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>  <li>    <p>Close the window</p>    <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>  </li>  <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The annotation track should appear. Be careful that all files have the same genome ID</p>  </li></ol></blockquote>






<h3 class="faq-area"> Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/assembly/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-mapped-reads-track-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Check if the reference genome is available on the IGV instance</li>  <li>Expand the BAM dataset with the mapped <span class="notranslate">reads</span> in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The mapped <span class="notranslate">reads</span> track should appear. Be sure that all files have the same genome ID</p>  </li></ol></blockquote>






<h3 class="faq-area"> Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span> <a href="/training-material/topics/visualisation/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-add-mapped-reads-track-to-igv-from-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Add Mapped <span class="notranslate">reads</span> track to IGV from <span class="notranslate">Galaxy</span></h3>  <ol>  <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>  <li>Launch IGV on your computer</li>  <li>Check if the reference genome is available on the IGV instance</li>  <li>Expand the BAM dataset with the mapped <span class="notranslate">reads</span> in the history</li>  <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>  <li>    <p>Switch to the IGV instance</p>    <p>The mapped <span class="notranslate">reads</span> track should appear. Be sure that all files have the same genome ID</p>  </li></ol></blockquote>





 <br><br> 
<h2 id="instructors"> <a href="#instructors"> Instructors </a></h2>
<hr/>



<h3 class="faq-area"> What are the best practices for teaching with <span class="notranslate">Galaxy</span>? <a href="/training-material/faqs/gtn/instructors_best_practices.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>We started to collect some best practices for instructors inside our <a href="/training-material/topics/instructors/">Good practices slides</a></p>






<h3 class="faq-area"> What <span class="notranslate">Galaxy</span> instance should I use for my training? <a href="/training-material/faqs/gtn/instructors_choosing_server.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>To teach the hands-on tutorials you need a <span class="notranslate">Galaxy</span> server to run the examples on.</p><p>Each tutorial is annotated with the information on which <a href="https://galaxyproject.org/public-galaxy-servers/">public <span class="notranslate">Galaxy</span> servers</a> it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server‚Äôs admins to confirm that think the server can handle the workload for a workshop. For example, the <a href="https://usegalaxy.eu/">use<span class="notranslate">galaxy</span>.eu</a></p><p>If your organization/consortia/community has its own <span class="notranslate">Galaxy</span> server, then you may  want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same you your participants will be using after the training. They‚Äôll also be able to revisit any analysis they did during the training. If you pursue this option you‚Äôll need to work with your organization‚Äôs <span class="notranslate">Galaxy</span> Admins to confirm that</p><ul>  <li>the server can support a room full of people all doing the same analysis at the same time.</li>  <li>all tools and reference datasets needed in the tutorial are locally installed.  To learn how to setup a <span class="notranslate">Galaxy</span> instance for a tutorial, you can follow our <a href="/training-material/topics/instructors/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</li>  <li>all participants will be able to create/use accounts on the system.</li></ul><p>Some training topics have a Docker image that can be installed and run on all participants‚Äô laptops.  These images contain <span class="notranslate">Galaxy</span> instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.</p><p>Finally, you can also run your tutorials on cloud-based infrastructures.  <span class="notranslate">Galaxy</span> is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://launch.genome.edu.au/launch">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.</p>






<h3 class="faq-area"> How do I get help? <a href="/training-material/faqs/gtn/instructors_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the <span class="notranslate">Galaxy</span> Training Network <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>. Anyone can view the discussion, but you‚Äôll need to login (using your GitHub or Twitter account) to add to the discussion.</p><p>If you have questions about <span class="notranslate">Galaxy</span> in general (that are not training-centric) then there are <a href="https://galaxyproject.org/support/">several support options</a>.</p>






<h3 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/instructors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.</p>





 <br><br> 
<h2 id="interactive tools"> <a href="#interactive tools"> Interactive tools </a></h2>
<hr/>



<h3 class="faq-area"> Launch JupyterLab <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch JupyterLab</h3>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-jupyterlab-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch JupyterLab in <span class="notranslate">Galaxy</span></h3>  <p>Currently JupyterLab in <span class="notranslate">Galaxy</span> is available on <a href="https://live.usegalaxy.eu">Live.use<span class="notranslate">Galaxy</span>.eu</a>, <a href="https://usegalaxy.org">use<span class="notranslate">galaxy</span>.org</a> and <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a>.</p>  <blockquote class="notranslate hands_on">    <h3 data-toc-skip id="hands_on-hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run JupyterLab</h3>    <ol>      <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Tested with interactive_tool_jupyter_notebook"><strong>Interactive Jupyter Notebook</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_jupyter_notebook</span></span> :</li>      <li>Click Execute</li>      <li>The tool will start running and will stay running permanently</li>      <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>      <li>Click on your JupyterLab instance</li>    </ol>  </blockquote></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-try-jupyterlab-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch Try JupyterLab if not available on <span class="notranslate">Galaxy</span></h3>  <p>If JupyterLab is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>  </ol></blockquote></blockquote>






<h3 class="faq-area"> Open a Terminal in Jupyter <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-open-a-terminal-in-jupyter"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Open a Terminal in Jupyter</h3>  <p>This tutorial will let you accomplish almost everything from this view, running code in the cells below directly in the training material. You can choose between running the code here, or opening up a terminal tab in which to run it.Here are some instructions for how to do this on various environments.</p>  <p><strong>Jupyter</strong> on Use<span class="notranslate">Galaxy</span>.* and <strong>MyBinder.org</strong></p>  <ol>  <li>    <p>Use the File ‚Üí New ‚Üí Terminal menu to launch a terminal.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-launch.png" alt="screenshot of jupyterlab showing the File menu expanded to show new and terminal option." /></p>  </li>  <li>    <p>Disable ‚ÄúSimple‚Äù mode in the bottom left hand corner, if it activated.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-simple.png" alt="screenshot of jupyterlab showing a toggle labelled simple" /></p>  </li>  <li>    <p>Drag one of the terminal or notebook tabs to the side to have the training materials and terminal side-by-side</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-final.png" alt="screenshot of jupyterlab with notebook and terminal side-by-side." /></p>  </li></ol>  <p><strong>CoCalc</strong></p>  <ol>  <li>    <p>Use the <em>Split View</em> functionality of cocalc to split your view into two portions.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-split.png" alt="screenshot of cocalc button to split views" /></p>  </li>  <li>    <p>Change the view of one panel to a terminal</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-terminal.png" alt="screenshot of cocalc swapping view port to that of a terminal" /></p>  </li></ol></blockquote>






<h3 class="faq-area"> Open interactive tool <a href="/training-material/faqs/galaxy/interactive_tools_open.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-open-interactive-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Open interactive tool</h3>  <ol>  <li>Go to User &gt; Active InteractiveTools</li>  <li>Wait for the  to be running (Job Info)</li>  <li>Click on</li></ol></blockquote>






<h3 class="faq-area"> Knitting RMarkdown documents in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-knitting-rmarkdown-documents-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Knitting RMarkdown documents in RStudio</h3>  <p>One of the other nice features of RMarkdown documents is making lovely presentation-quality worthy documents. You can take, for example, a tutorial and produce a nice report like output as HTML, PDF, or <code class="language-plaintext highlighter-rouge">.doc</code> document that can easily be shared with colleagues or students.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-meta-output.png" alt="Screenshot of the metadata with html_notebook and word_document being visible and a number of options controlling their output. TOC, standing for table of contents, has been set to true for both." /></p>  <p>Now you‚Äôre <strong>ready to preview the document</strong>:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of preview dropdown with options like preview, knit to html, knit to pdf, knit to word" /></p>  <p><strong>Click Preview</strong>. A window will popup with a preview of the rendered verison of this document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of rendered document with the table of contents on left, title is in a large font, and there are coloured boxes similar to GTN tutorials offering tips and more information" /></p>  <p>The preview is really similar to the GTN rendering, no cells have been executed, and no output is embedded yet in the preview document. But if you have run cells (e.g. the first few loading a library and previewing the <code class="language-plaintext highlighter-rouge">msleep</code> dataset:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview-output.png" alt="screenshot of the rendered document with a fancy table browser embedded as well as the output of each step" /></p>  <p><strong>When you‚Äôre ready to distribute</strong> the document, you can instead use the <code class="language-plaintext highlighter-rouge">Knit</code> button. This runs every cell in the entire document fresh, and then compiles the outputs together with the rendered markdown to produce a nice result file as HTML, PDF, or Word document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-knit.png" alt="screenshot of the console with 'chunks' being knitted together" /></p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-pdf--word-require-a-latex-installation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: PDF + Word require a LaTeX installation</h3>  <p>You might need to install additional packages to compile the PDF and Word document versions</p></blockquote>  <p>And at the end you can see a pretty document rendered with all of the output of every step along the way. This is a fantastic way to e.g. distribute read-only lesson materials to students, if you feel they might struggle with using an RMarkdown document, or just want to read the output without <em>doing</em> it themselves.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-pdf.png" alt="screenshot of a PDF document showing the end of the tutorial where a pretty plot has been rendered and there is some text for conclusions and citations" /></p></blockquote>






<h3 class="faq-area"> Launch RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch RStudio</h3>  <p>Depending on which server you are using, you may be able to run RStudio directly in <span class="notranslate">Galaxy</span>. If that is not available, RStudio Cloud can be an alte<span class="notranslate">rna</span>tive.</p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>  <p>Currently RStudio in <span class="notranslate">Galaxy</span> is only available on <a href="https://usegalaxy.eu">Use<span class="notranslate">Galaxy</span>.eu</a> and <a href="https://usegalaxy.org">Use<span class="notranslate">Galaxy</span>.org</a></p>  <ol>    <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>    <li>Click Execute</li>    <li>The tool will start running and will stay running permanently</li>    <li>Click on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started.</li>  </ol></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>  <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>    <li>Create a new project</li>  </ol></blockquote></blockquote>






<h3 class="faq-area"> Learning with RMarkdown in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-learning-with-rmarkdown-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Learning with RMarkdown in RStudio</h3>  <p>Learning with RMarkdown is a bit different than you might be used to. Instead of copying and pasting code from the GTN into a document you‚Äôll instead be able to run the code directly as it was written, inside RStudio! You can now focus just on the code and reading within RStudio.</p>  <ol>  <li>    <p><strong>Load the notebook</strong> if you have not already, following the tip box at the top of the tutorial</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-download-file.png" alt="Screenshot of the Console in RStudio. There are three lines visible of not-yet-run R code with the download.file statements which were included in the setup tip box." /></p>  </li>  <li>    <p><strong>Open it</strong> by clicking on the <code class="language-plaintext highlighter-rouge">.Rmd</code> file in the file browser (bottom right)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-file-menu.png" alt="Screenshot of Files tab in RStudio, here there are three files listed, a data-science-r-dplyr.Rmd file, a css and a bib file." /></p>  </li>  <li>    <p>The RMarkdown document will appear in the document viewer (top left)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-document-view.png" alt="Screenshot of an open document in RStudio. There is some yaml metadata above the tutorial showing the title of the tutorial." /></p>  </li></ol>  <p>You‚Äôre now ready to view the RMarkdown notebook! Each notebook starts with a lot of metadata about how to build the notebook for viewing, but you can ignore this for now and <strong>scroll down to the content of the tutorial</strong>.</p>  <p>You‚Äôll see codeblocks scattered throughout the text, and these are all runnable snippets that appear like this in the document:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-run-cell.png" alt="Screenshot of the RMarkdown document in the viewer, a cell is visible between markdown text reading library tidyverse. It is slightly more grey than the background region, and it has a run button at the right of the cell in a contextual menu." /></p>  <p>And you have a few options for how to run them:</p>  <ol>  <li>Click the green arrow</li>  <li><kbd>ctrl+enter</kbd></li>  <li>    <p>Using the menu at the top to run all</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-run-all.png" alt="Screenshot of the run dropdown menu in R, the first item is run selected lines showing the mentioned shortcut above, the second is run next chunk, and then it also mentions a 'run all chunks below' and 'restart r and run all chunks' option." /></p>  </li></ol>  <p>When you run cells, the output will appear below in the Console. RStudio essentially copies the code from the RMarkdown document, to the console, and runs it, just as if you had typed it out yourself!</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-cell-output.png" alt="Screenshot of a run cell, its output is included below in the RMarkdown document and the same output is visible below in the console. It shows a log of loading the tidyverse library." /></p>  <p>One of the best features of RMarkdown documents is that they include a very nice table browser which makes previewing results a lot easier! Instead of needing to use <code class="language-plaintext highlighter-rouge">head</code> every time to preview the result, you get an interactive table browser for any step which outputs a table.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-table.png" alt="Screenshot of the table browser. Below a code chunk is a large white area with two images, the first reading 'r console' and the second reading 'tbl_df'. The tbl_df is highlighted like it is active. Below that is a pretty-printed table with bold column headers like name and genus and so on. At the right of the table is a small arrow indicating you can switch to seeing more columns than just the initial three. At the bottom of the table is 1-10 of 83 rows written, and buttons for switching between each page of results." /></p></blockquote>






<h3 class="faq-area"> Stop RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Stop RStudio</h3>  <p>When you have finished your R analysis, it‚Äôs time to stop RStudio.</p>  <ol>  <li>First, save your work into <span class="notranslate">Galaxy</span>, to ensure reproducibility:    <ol>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>    </ol>  </li>  <li>Once you have saved your data, you can proceed in 2 different ways:    <ul>      <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a ‚Äúin progress state‚Äù, so yellow, OR</li>      <li>Clicking on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the ‚ÄúStop‚Äù button at the bottom.</li>    </ul>  </li></ol></blockquote>





 <br><br> 
<h2 id="introduction"> <a href="#introduction"> Introduction </a></h2>
<hr/>



<h3 class="faq-area"> What is <span class="notranslate">Galaxy</span>? <a href="/training-material/faqs/gtn/galaxy_what_is_it.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p><a href="https://galaxyproject.org/"><span class="notranslate">Galaxy</span></a> is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.</p>






<h3 class="faq-area"> What is this website? <a href="/training-material/faqs/gtn/gtn-what-is-it.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>This website is a collection of hands-on tutorials that are designed to be interactive and are built around <span class="notranslate">Galaxy</span>:</p><p><img src="/training-material/shared/images/interactive_training.png" alt="Interactive training" title="The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance" /></p><p>This material is developed and maintained by the <a href="https://galaxyproject.org/">worldwide <span class="notranslate">Galaxy</span> community</a>. You can learn more about this effort by reading our <a href="https://doi.org/10.1016/j.cels.2018.05.012">article</a>.</p>






<h3 class="faq-area"> How can I advertise the training materials on my posters? <a href="/training-material/faqs/gtn/gtn_advertise.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>We provide some QR codes and logos in the <a href="https://github.com/galaxyproject/training-material/tree/main/assets/images">images folder</a>.</p>






<h3 class="faq-area"> What audiences are the tutorials for? <a href="/training-material/faqs/gtn/gtn_audience.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>There are two distinct audiences for these materials.</p><ol>  <li><strong>Self-paced individual learners.</strong> These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.</li>  <li><strong>Instructors.</strong> They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.</li></ol>






<h3 class="faq-area"> How can I cite the GTN? <a href="/training-material/faqs/gtn/gtn_citing.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>We wrote an <a href="https://doi.org/10.1016/j.cels.2018.05.012">article</a> about our efforts.</p><p>To cite individual tutorials, please find citation information at the end of the tutorial.</p>






<h3 class="faq-area"> How is the content licensed? <a href="/training-material/faqs/gtn/gtn_license.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>The content of this website is licensed under the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 License</a>.</p>






<h3 class="faq-area"> What are the tutorials for? <a href="/training-material/faqs/gtn/gtn_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>These tutorials can be used for learning and teaching how to use <span class="notranslate">Galaxy</span> for general data analysis, and for learning/teaching specific domains such as assembly and differential gene <span class="notranslate">expression</span> analysis with <span class="notranslate">RNA</span>-Seq data.</p>





 <br><br> 
<h2 id="learners"> <a href="#learners"> Learners </a></h2>
<hr/>



<h3 class="faq-area"> How can I get help? <a href="/training-material/faqs/gtn/learners_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you have questions about this training material, you can reach us using the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>.  You‚Äôll need a <a href="https://github.com/">GitHub</a> or <a href="https://twitter.com/">Twitter</a> account to post questions.  If you have questions about <span class="notranslate">Galaxy</span> outside the context of training, see the <a href="https://galaxyproject.org/support/"><span class="notranslate">Galaxy</span> Support page</a>.</p>






<h3 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/learners_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>If you are new to <span class="notranslate">Galaxy</span> then start with one of the introductory topics.  These introduce you to concepts that are useful in <span class="notranslate">Galaxy</span>, no matter what domain you are doing analysis in.</p><p>If you are already familiar with <span class="notranslate">Galaxy</span> basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.</p><p>If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in <span class="notranslate">Galaxy</span>, then many tutorials include <em>Instructions for the impatient</em> sections.</p>






<h3 class="faq-area"> Where can I run the hands-on tutorials? <a href="/training-material/faqs/gtn/learners_running_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>To run the hands-on tutorials you need a <span class="notranslate">Galaxy</span> server to run them on.</p><p>Each tutorial is annotated with information about which <a href="https://galaxyproject.org/public-galaxy-servers/">public <span class="notranslate">Galaxy</span> servers</a> it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.</p><p>If your organization/consortia/community has its own <span class="notranslate">Galaxy</span> server, then you may  want to run tutorials on that. You will need to confirm that all necessary tools and reference genomes are available on your server and possible install missing tools and data. To learn how to do that, you can follow our <a href="/training-material/topics/instructors/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</p><p>Some topics have a <a href="https://www.docker.com/">Docker</a> image that can be installed and run on participants‚Äô laptops.  These Docker images contain <span class="notranslate">Galaxy</span> instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to <a href="https://docs.docker.com/install/">install Docker</a>.</p><p>Finally, you can also run your tutorials on cloud-based infrastructures.  <span class="notranslate">Galaxy</span> is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://www.gvl.org.au/">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.</p><p>If you are already familiar with, and have an account on <a href="https://aws.amazon.com/">Amazon Web Services</a> then you can also launch a <span class="notranslate">Galaxy</span> server there using <a href="https://launch.usegalaxy.org/">CloudLaunch</a>.</p>






<h3 class="faq-area"> How do I use this material? <a href="/training-material/faqs/gtn/learners_using_materials.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.</p><p>Most of your learning will happen in the next step - the hands-on tutorials.  This is where you‚Äôll become familiar with using the <span class="notranslate">Galaxy</span> interface and experiment with different ways to use <span class="notranslate">Galaxy</span> and the tools in <span class="notranslate">Galaxy</span>.</p>





 <br><br> 
<h2 id="mapping"> <a href="#mapping"> <span class="notranslate">Mapping</span> </a></h2>
<hr/>



<h3 class="faq-area"> Is it possible to visualize the <span class="notranslate">RNA</span> STAR bam file using the JBrowse tool? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-is-it-possible-to-visualize-the-rna-star-bam-file-using-the-jbrowse-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Is it possible to visualize the <span class="notranslate">RNA</span> STAR bam file using the JBrowse tool?</h3>  <p>Yes, that should work.</p></blockquote>






<h3 class="faq-area"> I am using the <span class="notranslate">RNA</span> STAR tool to map my <span class="notranslate">reads</span> on the reference genome. while filling in the details for , I have to fill in "Length of the genomic sequence around annotated junctions", which apparently has to be 36. I'm lost for a moment why this is 36, what is meant by it, and why is it relevant? Does anyone have any ideas? Why should it be the length of the <span class="notranslate">reads</span> -1? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-am-using-the-rna-star-tool-to-map-my-reads-on-the-reference-genome-while-filling-in-the-details-for--i-have-to-fill-in-length-of-the-genomic-sequence-around-annotated-junctions-which-apparently-has-to-be-36-im-lost-for-a-moment-why-this-is-36-what-is-meant-by-it-and-why-is-it-relevant-does-anyone-have-any-ideas-why-should-it-be-the-length-of-the-reads--1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I am using the <span class="notranslate">RNA</span> STAR tool to map my <span class="notranslate">reads</span> on the reference genome. while filling in the details for , I have to fill in ‚ÄúLength of the genomic sequence around annotated junctions‚Äù, which apparently has to be 36. I‚Äôm lost for a moment why this is 36, what is meant by it, and why is it relevant? Does anyone have any ideas? Why should it be the length of the <span class="notranslate">reads</span> -1?</h3>  <p><span class="notranslate">RNA</span> STAR is using the gene model to create the database of splice junctions, and that these don‚Äôt ‚Äúneed‚Äù to have a length longer than the <span class="notranslate">reads</span> (37bp).</p></blockquote>





 <br><br> 
<h2 id="markdown"> <a href="#markdown"> Markdown </a></h2>
<hr/>



<h3 class="faq-area"> How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>? <a href="/training-material/topics/contributing/faqs/tutorial_from_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-create-a-tutorial-skeleton-from-a-galaxy-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>?</h3>  <p>There are two ways to do this:</p>  <ol>  <li>Use <a href="https://planemo.readthedocs.io/en/latest/">planemo</a> on your local machine. Please see the tutorial named <a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">‚ÄúCreating a new tutorial‚Äù</a> for detailed instructions.</li>  <li>Use our <a href="https://ptdk.herokuapp.com">web service</a></li></ol></blockquote>






<h3 class="faq-area"> How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-can-i-create-a-tutorial-skeleton-from-a-galaxy-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How can I create a tutorial skeleton from a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span>?</h3>  <p>There are two ways to do this:</p>  <ol>  <li>Use <a href="https://planemo.readthedocs.io/en/latest/">planemo</a> on your local machine. Please see the tutorial named <a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">‚ÄúCreating a new tutorial‚Äù</a> for detailed instructions.</li>  <li>Use our <a href="https://ptdk.herokuapp.com">web service</a></li></ol></blockquote>





 <br><br> 
<h2 id="metaproteomics"> <a href="#metaproteomics"> Metaproteomics </a></h2>
<hr/>



<h3 class="faq-area"> How does one compare metaproteomics measurements from two experimental conditions? <a href="/training-material/topics/proteomics/faqs/comparing_measurement.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-one-compare-metaproteomics-measurements-from-two-experimental-conditions"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does one compare metaproteomics measurements from two experimental conditions?</h3>  <p>For comparing taxonomy composition or functional content of two conditions in metaproteomics or metatranscriptomics studies, users are recommended to use metaQuantome. GTN tutorials for metaQuantome are available <a href="/training-material/topics/proteomics/">in the proteomics topic.</a></p></blockquote>






<h3 class="faq-area"> How does one convert RAW files to MGF peak lists within <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/faqs/convert_to_MGF.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-how-does-one-convert-raw-files-to-mgf-peak-lists-within-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: How does one convert RAW files to MGF peak lists within <span class="notranslate">Galaxy</span>?</h3>  <p><span class="notranslate">Galaxy</span> has implemented msconvert tool so that RAW files from Thermo instruments can be converted into MGF or mzML formats.</p></blockquote>






<h3 class="faq-area"> I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches? <a href="/training-material/topics/proteomics/faqs/converting_FASTA.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-have-fastq-files-from-metagenomics-or-metatranscriptomics-datasets-how-can-i-convert-them-into-a-protein-fasta-file-for-metaproteomics-searches"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches?</h3>  <p><span class="notranslate">Galaxy</span> has a tool named Sixgill that can be used to convert the nucleic acid sequences to ‚Äòmetapeptide‚Äô sequences. There are other options available within <span class="notranslate">Galaxy</span> such as the <span class="notranslate">Galaxy</span>Graph approach and Metagenome Binning, Assembly and Annotation <span class="notranslate">Workflow</span>. Please <a href="https://help.galaxyproject.org/">contact us</a>, if you need assistance.</p></blockquote>






<h3 class="faq-area"> What software tools are available to determine taxonomic composition from mass spectrometry data? <a href="/training-material/topics/proteomics/faqs/determine_taxonomic_composition.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-software-tools-are-available-to-determine-taxonomic-composition-from-mass-spectrometry-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What software tools are available to determine taxonomic composition from mass spectrometry data?</h3>  <p>Within the <span class="notranslate">Galaxy</span> framework we recommend the use of Unipept software that uses NCBI taxonomy and UniProt databases to detect unique peptides for taxonomy. Other software tools such as MetaTryp 2.0 (PMID: 32897080) can also be used to determine the taxonomic composition of the metaproteomics datasets.</p></blockquote>






<h3 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/proteomics/faqs/functional_analysis_tools.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</h3>  <p>Within the <span class="notranslate">Galaxy</span> framework we recommend the use of Unipept software that uses UniProt databases and annotation to detect proteins (EC terms) and functional groups such as GO Ontology and InterPro terms. Other software tools such as <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">EggNOG Mapper</a> are also available within the <span class="notranslate">Galaxy</span> platform. Other software such as MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA), ProPHAnE also generate functional outputs.</p></blockquote>






<h3 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/faqs/reusing_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Can I use these workflows on datasets generated from our laboratory?</h3>  <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p></blockquote>






<h3 class="faq-area"> Which search algorithms are recommended for searching the metaproteomics data? <a href="/training-material/topics/proteomics/faqs/search_algorithms.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-search-algorithms-are-recommended-for-searching-the-metaproteomics-data"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which search algorithms are recommended for searching the metaproteomics data?</h3>  <p>SearchGUI supports search using nine search algorithms (X! Tandem. MS-GF+. OMSSA, Comet, Tide, MyriMatch, MS_Amanda, DirecTag and Novor). For this tutorial, we have used the first two search algorithms in the list. In our hands, the first four search algorithms have given us the most optimal results.</p></blockquote>






<h3 class="faq-area"> I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset? <a href="/training-material/topics/proteomics/faqs/search_strategies.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-have-a-really-large-search-database-what-search-strategies-do-you-recommend-for-searching-my-mass-spectrometry-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset?</h3>  <p>Readers are encouraged to use the database sectioning approach described by <a href="https://pubmed.ncbi.nlm.nih.gov/32396365/">Praveen Kumar et al</a> and available within <span class="notranslate">Galaxy</span>. Readers are also encouraged to consider other approaches such as MetaNovo (not yet available in <span class="notranslate">Galaxy</span>). In absence of any database or taxonomic information about the microbiome dataset, other methods such as COMPIL 2.0  and De novo search methods can also be considered.</p></blockquote>






<h3 class="faq-area"> What other methods are available to study the functional state of the microbiome within <span class="notranslate">Galaxy</span>? <a href="/training-material/topics/proteomics/faqs/studying_microbiome_state.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-other-methods-are-available-to-study-the-functional-state-of-the-microbiome-within-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What other methods are available to study the functional state of the microbiome within <span class="notranslate">Galaxy</span>?</h3>  <p>Other software such as EggNOG Mapper, MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA) and  ProPHAnE also generate <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">functional outputs</a>.</p></blockquote>






<h3 class="faq-area"> Which version of SearchGUI and PeptideShaker shall I use for this tutorial? <a href="/training-material/topics/proteomics/faqs/tools_versions.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-which-version-of-searchgui-and-peptideshaker-shall-i-use-for-this-tutorial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Which version of SearchGUI and PeptideShaker shall I use for this tutorial?</h3>  <p>We highly recommend the usage of SearchGUI <span class="notranslate">Galaxy</span> version 3.3.10.1 and PeptideShaker version <span class="notranslate">Galaxy</span> Version 1.16.36.3. The newer versions of SearchGUI and PeptideShaker have not yet been tested for this <span class="notranslate">workflow</span>.</p></blockquote>





 <br><br> 
<h2 id="other"> <a href="#other"> Other </a></h2>
<hr/>



<h3 class="faq-area"> Are there any upcoming events focused on <span class="notranslate">Galaxy</span> Training? <a href="/training-material/faqs/gtn/galaxy_events.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>Yes, always! Have a look at the <a href="https://galaxyproject.org/events/"><span class="notranslate">Galaxy</span> Community Events Calendar</a> for what coming up right now.</p>






<h3 class="faq-area"> Sustainability of the training-material and metadata <a href="/training-material/faqs/gtn/gtn_sustainability.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><p>This repository is hosted on <a href="https://github.com/">GitHub</a> using git as a <a href="https://en.wikipedia.org/wiki/Distributed_version_control">DVCS</a>. Therefore the community is hosting backups of thisrepository in a decentralised way. The repository is self-contained and contains all needed content and all metadata.In addition we mirror snapshops of this repo on <a href="http://zenodo.org">Zenodo</a>.</p>





 <br><br> 
<h2 id="recommended browser"> <a href="#recommended browser"> Recommended browser </a></h2>
<hr/>



<h3 class="faq-area"> UCSC - I fetched data from a remote website but now I‚Äôm logged out of <span class="notranslate">Galaxy</span> and my data is gone? <a href="/training-material/topics/introduction/faqs/fetched_data.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-ucsc---i-fetched-data-from-a-remote-website-but-now-im-logged-out-of-galaxy-and-my-data-is-gone"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: UCSC - I fetched data from a remote website but now I‚Äôm logged out of <span class="notranslate">Galaxy</span> and my data is gone?</h3>  <p>This is a known bug with Chrome + <span class="notranslate">Galaxy</span>, we‚Äôre working on it <a href="https://github.com/galaxyproject/galaxy/issues/11374"><span class="notranslate">galaxy</span>project/<span class="notranslate">galaxy</span>#11374</a>. For now we can recommend using Firefox (known to work) or trying another browser.</p></blockquote>





 <br><br> 
<h2 id="scRNA"> <a href="#scRNA"> Sc<span class="notranslate">rna</span> </a></h2>
<hr/>



<h3 class="faq-area"> Why is amplification more of an issue in sc<span class="notranslate">RNA</span>-seq than <span class="notranslate">RNA</span>-seq? <a href="/training-material/topics/transcriptomics/faqs/amplification.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why is amplification more of an issue in sc<span class="notranslate">RNA</span>-seq than <span class="notranslate">RNA</span>-seq?</h3>  <p>Due to the extremely small amount of starting material, the initial amplification is likely to be uneven due to the first cycle of amplified products being overrepresented in the second cycle of amplification leading to further bias. In Bulk <span class="notranslate">RNA</span>-seq, the larger selection of <span class="notranslate">RNA</span> molecules to amplify, evens out the odds that any one transcript will be amplified more than others.</p></blockquote>






<h3 class="faq-area"> Can <span class="notranslate">RNA</span>-seq techniques be applied to sc<span class="notranslate">RNA</span>-seq? <a href="/training-material/topics/transcriptomics/faqs/techniques.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-can-rna-seq-techniques-be-applied-to-scrna-seq"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Can <span class="notranslate">RNA</span>-seq techniques be applied to sc<span class="notranslate">RNA</span>-seq?</h3>  <p>The short answer is ‚Äòno, but yes‚Äô. At the beginning this was impossible due to the over-prevalence of dropout events (‚Äúzeroes‚Äù) in the data complicating the normalisation techniques, but this is not so much of a problem any more with newer methods.</p></blockquote>





 <br><br> 
<h2 id="sequencing"> <a href="#sequencing"> Sequencing </a></h2>
<hr/>



<h3 class="faq-area"> Illumina MiSeq sequencing <a href="/training-material/faqs/galaxy/sequencing_illumina_miseq.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Illumina MiSeq sequencing</h3>  <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the namesuggests, fluorescent labels are measured for every base that bind at aspecific moment at a specific place on a flow cell. These flow cells arecovered with oligos (small single strand <span class="notranslate">DNA</span> strands). In the librarypreparation the <span class="notranslate">DNA</span> strands are cut into small <span class="notranslate">DNA</span> fragments (differs perkit/device) and specific pieces of <span class="notranslate">DNA</span> (adapters) are added, which arecomplementary to the oligos. Using bridge amplification large amounts ofclusters of these <span class="notranslate">DNA</span> fragments are made. The reverse string is washed away,making the clusters single stranded. Fluorescent bases are added one by one,which emit a specific light for different bases when added. This is happeningfor whole clusters, so this light can be detected and this data is basecalled(translation from light to a nucleotide) to a nucleotide sequence (Read). Forevery base a quality score is determined and also saved per read. Thisprocess is repeated for the reverse strand on the same place on the flowcell, so the forward and reverse <span class="notranslate">reads</span> are from the same <span class="notranslate">DNA</span> strand. Theforward and reversed <span class="notranslate">reads</span> are linked together and should always be processedtogether!</p>  <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p></blockquote>






<h3 class="faq-area"> Nanopore sequencing <a href="/training-material/faqs/galaxy/sequencing_nanopore.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Nanopore sequencing</h3>  <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>  <ol>  <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>  <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>  <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li></ol>  <p><img src="/training-material/topics/metagenomics/images/plasmid-metagenomics-nanopore/sequence_method.jpg" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://nanoporetech.com/sites/default/files/s3/white-papers/WGS_Assembly_white_paper.pdf?submissionGuid=40a7546b-9e51-42e7-bde9-b5ddef3c3512&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>  <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change inelectrical current is measured over the membrane of a flow cell. Whennucleotides pass the pores in the flow cell the current change is translated(basecalled) to nucleotides by a basecaller. A schematic overview is given inthe picture above.</p>  <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software ispresent on the devices. With basecalling the electrical signals are translatedto bases (A,T,G,C) with a quality score per base. The sequenced <span class="notranslate">DNA</span> strand willbe basecalled and this will form one read. Multiple <span class="notranslate">reads</span> will be stored in afastq file.</p></blockquote>





 <br><br> 
<h2 id="tools"> <a href="#tools"> Tools </a></h2>
<hr/>



<h3 class="faq-area"> Re-running a tool <a href="/training-material/faqs/galaxy/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-re-running-a-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Re-running a tool</h3>  <ol>  <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>  <li><strong>Click</strong> re-run <i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> the tool</li></ol>  <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p></blockquote>






<h3 class="faq-area"> Para volver a ejecutar una herramienta <a href="/training-material/faqs/galaxy-es/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-para-volver-a-ejecutar-una-herramienta"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Para volver a ejecutar una herramienta</h3>  <ol>  <li><strong>Expande</strong> uno de los conjuntos de datos de la salida de la herramienta haciendo clic sobre √©l</li>  <li><strong>Selecciona</strong> volver a ejecutar <i class="fas fa-sync-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> de la herramienta</li></ol>  <p>Esto es de utilidad si quieres volver a correr la herramienta variando ligeramente los valores de los par√°metros, o si deseas verificar la configuraci√≥n de par√°metros que utilizaste.</p></blockquote>






<h3 class="faq-area"> Selecting a datast collection as input <a href="/training-material/faqs/galaxy/tools_select_collection.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>






<h3 class="faq-area"> Select multiple datasets <a href="/training-material/faqs/galaxy/tools_select_multiple_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>





 <br><br> 
<h2 id="transcriptome analysis of Arabidopsis thaliana"> <a href="#transcriptome analysis of Arabidopsis thaliana"> Transcriptome analysis of arabidopsis thaliana </a></h2>
<hr/>



<h3 class="faq-area"> Could I use a different p-adj value for filtering differentially expressed genes? <a href="/training-material/topics/transcriptomics/faqs/p_adj.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-could-i-use-a-different-p-adj-value-for-filtering-differentially-expressed-genes"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Could I use a different p-adj value for filtering differentially expressed genes?</h3>  <p>Yes, you can modify this value, to perform a more rigorous analysis, or extend the range of genes selected. A higher p-value will significantly increase the number of genes selected, at the expense of including possible false positives.</p></blockquote>






<h3 class="faq-area"> Could I use a different tool for performing the quantification step? <a href="/training-material/topics/transcriptomics/faqs/quantification.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-could-i-use-a-different-tool-for-performing-the-quantification-step"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Could I use a different tool for performing the quantification step?</h3>  <p>There are some alte<span class="notranslate">rna</span>tives to Salmon for reference transcriptome-based <span class="notranslate">RNA</span> quantification. Kallisto and Sailfish use a similar approach, known as pseudoalignment.</p></blockquote>





 <br><br> 
<h2 id="user interface"> <a href="#user interface"> User interface </a></h2>
<hr/>



<h3 class="faq-area"> I can‚Äôt find the ‚ÄúAnalyze Data‚Äù button <a href="/training-material/topics/introduction/faqs/analyze_data.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-i-cant-find-the-analyze-data-button"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: I can‚Äôt find the ‚ÄúAnalyze Data‚Äù button</h3>  <p>The <span class="notranslate">Galaxy</span> interface has changed a bit recently, ‚ÄúAnalyze Data‚Äù was always the home button, and now looks like a home icon.</p></blockquote>






<h3 class="faq-area"> My <span class="notranslate">Galaxy</span> looks different than in the video <a href="/training-material/topics/introduction/faqs/galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-galaxy-looks-different-than-in-the-video"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My <span class="notranslate">Galaxy</span> looks different than in the video</h3>  <p><span class="notranslate">Galaxy</span> gets frequent updates, different servers will be running different versions. This is nothing to worry about, just let us know if you can‚Äôt find how to perform a task in your <span class="notranslate">Galaxy</span>.</p></blockquote>





 <br><br> 
<h2 id="user preferences"> <a href="#user preferences"> User preferences </a></h2>
<hr/>



<h3 class="faq-area"> Getting your API key <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-getting-your-api-key"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Getting your API key</h3>  <ol>  <li>In your browser, open your <span class="notranslate">Galaxy</span> homepage</li>  <li>Log in, or register a new account, if it‚Äôs the first time you‚Äôre logging in</li>  <li>Go to <code class="language-plaintext highlighter-rouge">User -&gt; Preferences</code> in the top menu bar, then click on <code class="language-plaintext highlighter-rouge">Manage API key</code></li>  <li>If there is no current API key available, click on <code class="language-plaintext highlighter-rouge">Create a new key</code> to generate it</li>  <li>Copy your API key to somewhere convenient, you will need it throughout this tutorial</li></ol></blockquote>





 <br><br> 
<h2 id="utilities"> <a href="#utilities"> Utilities </a></h2>
<hr/>



<h3 class="faq-area"> Got lost along the way? <a href="/training-material/topics/admin/faqs/missed-something.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-got-lost-along-the-way"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Got lost along the way?</h3>  <p>If you missed any steps, you can compare against the <a href="https://github.com/hexylena/git-gat/tree/step-">reference files</a>, or see what changed since <a href="https://github.com/hexylena/git-gat/compare/step--1...step-#files_bucket">the previous tutorial</a>.</p></blockquote>





 <br><br> 
<h2 id="workflows"> <a href="#workflows"> Workflows </a></h2>
<hr/>



<h3 class="faq-area"> Annotate a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_annotate.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-annotate-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Annotate a <span class="notranslate">workflow</span></h3>  <ul>  <li>Open the <span class="notranslate">workflow</span> editor for the <span class="notranslate">workflow</span></li>  <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>Edit Attributes</strong> on the top right</li>  <li>Write a description of the worklow in the <strong>Annotation</strong> box</li>  <li>Add a tag (which will help to search for the <span class="notranslate">workflow</span>) in the <strong>Tags</strong> section</li></ul></blockquote>






<h3 class="faq-area"> Creating a new <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h3>
You can create a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> from scratch in the <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> editor.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new <span class="notranslate">workflow</span></h3>  <ol>  <li>Click <strong><span class="notranslate">Workflow</span></strong> on the top bar</li>  <li>Click the new <span class="notranslate">workflow</span> <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-new</span> button</li>  <li>Give it a clear and memorable name</li>  <li>Clicking <strong>Save</strong> will take you directly into the <span class="notranslate">workflow</span> editor for that <span class="notranslate">workflow</span></li>  <li>Need more help? Please see the <em>How to make a <span class="notranslate">workflow</span> subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li></ol></blockquote>






<h3 class="faq-area"> Opening the <span class="notranslate">workflow</span> editor <a href="/training-material/faqs/galaxy/workflows_edit.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-opening-the-workflow-editor"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Opening the <span class="notranslate">workflow</span> editor</h3>  <ol>  <li>Click on the name of the imported <span class="notranslate">workflow</span><img src="/training-material/topics/galaxy-interface/images/edit_workflow.png" alt="Workflow drop down menu showing Edit option" title="When you click on the name of a workflow it will open a drop down menu whose first option is Edit. Select the Edit option to edit the workflow." /></li>  <li>Select the Edit <span class="notranslate">workflow</span> to open the <span class="notranslate">workflow</span> in the <span class="notranslate">workflow</span> editor</li></ol></blockquote>






<h3 class="faq-area"> Extracting a <span class="notranslate">workflow</span> from your history <a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> can automatically create a <span class="notranslate">workflow</span> based on the analysis you have performed in a history. This means that once you have done an analysis manually once, you can easily extract a <span class="notranslate">workflow</span> to repeat it on different data.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extracting-a-workflow-from-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extracting a <span class="notranslate">workflow</span> from your history</h3>  <ol>  <li>    <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> button.</p>    <p>This will make the creation of the <span class="notranslate">workflow</span> easier.</p>  </li>  <li>    <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract <span class="notranslate">workflow</span></strong>.</p>    <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>    <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the <span class="notranslate">workflow</span>.</p>  </li>  <li>    <p>Replace the <strong><span class="notranslate">Workflow</span> name</strong> to something more descriptive.</p>  </li>  <li>    <p><strong>Rename</strong> each <span class="notranslate">workflow</span> input in the boxes at the top of the second column.</p>  </li>  <li>    <p>If there are any steps that shouldn‚Äôt be included in the <span class="notranslate">workflow</span>, you can <strong>uncheck</strong> them in the first column of boxes.</p>  </li>  <li>    <p>Click on the <strong>Create <span class="notranslate">Workflow</span></strong> button near the top.</p>    <p>You will get a message that the <span class="notranslate">workflow</span> was created.</p>  </li></ol></blockquote>






<h3 class="faq-area"> Extraer un flujo de trabajo de tu historial <a href="/training-material/faqs/galaxy-es/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h3>
<span class="notranslate">Galaxy</span> puede crear autom√°ticamente un flujo de trabajo basado en un an√°lisis almacenado en tu historial. Esto significa que una vez que hayas realizado un an√°lisis manualmente, puedes extraer f√°cilmente un flujo de trabajo para repetirlo con diferentes datos.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extraer-un-flujo-de-trabajo-de-tu-historial"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extraer un flujo de trabajo de tu historial</h3>  <ol>  <li>Elimina cualquier trabajo fallido o no deseado de tu historial.</li>  <li>Haz clic en <strong>Opciones de historial</strong> (icono de engranaje <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span>) en la parte superior del panel de historial.</li>  <li>Selecciona <strong>Extraer flujo de trabajo</strong></li>  <li>Verifica los pasos, ingresa un nombre para tu flujo de trabajo y presiona el bot√≥n <strong>Crear flujo de trabajo</strong>.</li></ol></blockquote>






<h3 class="faq-area"> Hiding intermediate steps <a href="/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-hiding-intermediate-steps"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Hiding intermediate steps</h3>  <p>When a <span class="notranslate">workflow</span> is executed, the user is usually primarily interested in the final product and not in all intermediate steps.By default all the outputs of a <span class="notranslate">workflow</span> will be shown, but we can explicitly tell <span class="notranslate">Galaxy</span> which outputs to show and which to hide for a given <span class="notranslate">workflow</span>.This behaviour is controlled by the little checkbox in front of every output dataset:</p>  <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p></blockquote>






<h3 class="faq-area"> Importing a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_import.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> URL‚Äù</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> File‚Äù</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>






<h3 class="faq-area"> Importing a <span class="notranslate">workflow</span> using the search <a href="/training-material/faqs/galaxy/workflows_import_search.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow-using-the-search"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span> using the search</h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>Import</strong> icon at the top-right of the screen</li>  <li>    <p>Click on <strong>search form</strong> in <strong>Import a <span class="notranslate">Workflow</span> from Configured GA4GH Tool Registry Servers (e.g. Dockstore)</strong></p>  </li>  <li>    <p>Select the relevant TRS Server</p>  </li>  <li>    <p>Type the query</p>  </li>  <li>    <p>Expand the correct <span class="notranslate">workflow</span></p>  </li>  <li>    <p>Click on the wanted version</p>    <p>The <span class="notranslate">workflow</span> will be imported in your workflows</p>  </li></ul></blockquote>






<h3 class="faq-area"> Setting parameters at run-time <a href="/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-setting-parameters-at-run-time"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Setting parameters at run-time</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the parameter you want users to provide every time they run the <span class="notranslate">workflow</span></li>  <li>Click on the arrow in front of the name <i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-runtime-toggle</span> to toggle to <em>set at runtime</em></li></ol></blockquote>






<h3 class="faq-area"> Make a <span class="notranslate">workflow</span> public <a href="/training-material/faqs/galaxy/workflows_publish.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-make-a-workflow-public"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Make a <span class="notranslate">workflow</span> public</h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows</li>  <li>Click on the interesting <span class="notranslate">workflow</span></li>  <li>Click on <strong>Share</strong></li>  <li>Clik on *Make <span class="notranslate">Workflow</span> Accessible and Publish**</li></ul></blockquote>






<h3 class="faq-area"> Renaming <span class="notranslate">workflow</span> outputs <a href="/training-material/faqs/galaxy/workflows_rename_output.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-workflow-outputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming <span class="notranslate">workflow</span> outputs</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.    <ul>      <li>        <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>        <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>      </li>    </ul>  </li></ol></blockquote>






<h3 class="faq-area"> Viewing a <span class="notranslate">workflow</span> report <a href="/training-material/faqs/galaxy/workflows_report_view.html"><i class="fas fa-external-link-alt"></i></a> </h3>
When creating a <span class="notranslate">workflow</span> in <span class="notranslate">Galaxy</span>, you can also define an output report page that should be created. Here you can display certain outputs of the pipeline (e.g. output files, tables, images, etc.) and other information about the run.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-viewing-a-workflow-report"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Viewing a <span class="notranslate">workflow</span> report</h3>  <ul>  <li>Go to <strong>User</strong> on the top menu bar of <span class="notranslate">Galaxy</span>.</li>  <li>Click on <strong><span class="notranslate">Workflow</span> invocations</strong>    <ul>      <li>Here you will find a list of all the workflows you have run</li>    </ul>  </li>  <li>Click on the name of a <span class="notranslate">workflow</span> invocation to expand it<img src="/training-material/faqs/galaxy/images/workflow_invocations_menu.png" alt="workflow invocations list" /></li>  <li>Click on <strong>View Report</strong> to go to the <span class="notranslate">workflow</span> report page</li>  <li><strong>Note:</strong> The report can also be downloaded in PDF format by clicking on the <i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-report-download</span> icon.</li></ul></blockquote>






<h3 class="faq-area"> Running a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_run.html"><i class="fas fa-external-link-alt"></i></a> </h3>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>






<h2 id="bibliography">References</h2>
<ol class="bibliography"><li id="Devenyi_2018">Devenyi, G. A., R. Emonet, R. M. Harris, K. L. Hertweck, D. Irving <i>et al.</i>, 2018 <b>Ten simple rules for collaborative lesson development</b> (S. Markel, Ed.). PLOS Computational Biology 14: e1005963. <a href="https://doi.org/10.1371/journal.pcbi.1005963">10.1371/jou<span class="notranslate">rna</span>l.pcbi.1005963</a></li>
<li id="Garcia2020">Garcia, L., B. Batut, M. L. Burke, M. Kuzak, F. Psomopoulos <i>et al.</i>, 2020 <b>Ten simple rules for making training materials FAIR</b> (S. Markel, Ed.). PLOS Computational Biology 16: e1007854. <a href="https://doi.org/10.1371/journal.pcbi.1007854">10.1371/jou<span class="notranslate">rna</span>l.pcbi.1007854</a></li></ol>


            </div>
        </div>
    </div>
</section>

        </div>
        
            <footer>
    <div class="container">
        <br><br>
        <p>
            The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
            provides researchers with online training materials, connects them with local trainers, and helps promoting open data analysis practices worldwide.
        </p>
        <p>
The content of the tutorials is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>. The website and infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>.
</p>
        <p>This is revision <a href="https://github.com/galaxyproject/training-material/commit/86ba6f30c3c8e76d2a8f07c2a333a13a48cabfc5">86ba6f30c3c8e76d2a8f07c2a333a13a48cabfc5</a></p>
    </div>
</footer>

        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
