<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Search Tutorials</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:title" content="Galaxy Training: Search Tutorials" />
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/search.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/search.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/search.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        <section>
  <h1>Search Tutorials</h1>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/lunr.js/2.3.9/lunr.min.js" integrity="sha512-4xUl/d6D6THrAnXAwGajXkoWaeMNwEKK4iNfq5DotEbLPAfk6FSxSP3ydNxqDgCw1c/0Z1Jg6L8h2j+++9BZmg==" crossorigin="anonymous"></script>

<div id="search-container">
	<input type="text" id="search-input" placeholder=" search..." class="nicer" /><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">search</span>
	<div class="search-results row" id="results-container"></div>
</div>

<!-- Configuration -->
<script>

var tutorials = {
  "/topics/visualisation/slides/introduction.html": {
    "topic": "Visualisation",
    "title": "Visualisations in Galaxy",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/visualisation/slides/introduction.html"
  },
  "/topics/visualisation/tutorials/jbrowse/tutorial.html": {
    "topic": "Visualisation",
    "title": "Genomic Data Visualisation with JBrowse",
    "description": null,
    "question": [
      "How can I visualise features or blast data?",
      "How can I visualise sequencing data in a workflow"
    ],
    "objectives": [
      "Build several visualisations in JBrowse",
      "Have basic familiarity with moving around JBrowse, and loading several data tracks"
    ],
    "tags": [

    ],
    "contributors": "Helena Rasche, Saskia Hiltemann, Erasmus+ Programme",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html"
  },
  "/topics/visualisation/tutorials/circos/tutorial.html": {
    "topic": "Visualisation",
    "title": "Visualisation with Circos",
    "description": null,
    "question": [
      "What can the Circos Galaxy tool be used for?",
      "How can I visualise common genomic datasets using Circos?"
    ],
    "objectives": [
      "Create a number of Circos plots using the Galaxy tool",
      "Familiarise yourself with the various different track types"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, Helena Rasche, Crist√≥bal Gallardo",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/visualisation/tutorials/circos/tutorial.html"
  },
  "/topics/epigenetics/slides/introduction-chip-seq.html": {
    "topic": "Epigenetics",
    "title": "Introduction to ChIP-Seq data analysis",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/epigenetics/slides/introduction-chip-seq.html"
  },
  "/topics/epigenetics/slides/introduction-dna-methylation.html": {
    "topic": "Epigenetics",
    "title": "Introduction to DNA Methylation data analysis",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Joachim Wolff, Yvan Le Bras",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/epigenetics/slides/introduction-dna-methylation.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/tutorial.html": {
    "topic": "Epigenetics",
    "title": "ATAC-Seq data analysis",
    "description": null,
    "question": [
      "Which DNA regions are accessible in the human lymphoblastoid cell line GM12878?",
      "How to analyse and visualise ATAC-Seq data?"
    ],
    "objectives": [
      "Apply appropriate analysis and quality control steps for ATAC-Seq",
      "Generate a heatmap of transcription start site accessibility",
      "Visualise peaks for specific regions"
    ],
    "tags": [

    ],
    "contributors": "Lucille Delisle, Maria Doyle, Florian Heyl",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/methylation-seq/tutorial.html": {
    "topic": "Epigenetics",
    "title": "DNA Methylation data analysis",
    "description": null,
    "question": [
      "What is methylation and why it cannot be recognised by a normal NGS procedure?",
      "Can a different methylation influence the expression of a gene? How?",
      "Which tools you can use to analyse methylation data?"
    ],
    "objectives": [
      "Learn how to analyse methylation data",
      "Get a first intuition what are common pitfalls."
    ],
    "tags": [

    ],
    "contributors": "Joachim Wolff, Devon Ryan",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html": {
    "topic": "Epigenetics",
    "title": "Formation of the Super-Structures on the Inactive X",
    "description": null,
    "question": [
      "Histone modification?",
      "How is a raw set of ChIP-seq data processed and analyzed?",
      "Where are the enriched regions for H3K27me3, H3K4me3 and CTCF on the chrX?"
    ],
    "objectives": [
      "Inspect the read quality",
      "Trim low quality bases",
      "Map reads on a reference genome",
      "Assess the quality of a ChIP-seq experiment",
      "Extract coverage files",
      "Call enriched regions or peaks"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?ChIP-seq\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">ChIP-seq</a>"
    ],
    "contributors": "Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez, Leily Rabbani",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html"
  },
  "/topics/epigenetics/tutorials/hicexplorer/tutorial.html": {
    "topic": "Epigenetics",
    "title": "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer",
    "description": null,
    "question": [
      "Why is a Hi-C analysis useful?",
      "What is 'chromosome conformation capture'?",
      "What are main steps in order to generate and plot a Hi-C contact matrix?"
    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html"
  },
  "/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html": {
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the Estrogen receptor",
    "description": null,
    "question": [
      "How is raw ChIP-seq data processed and analyzed?",
      "What are the binding sites of the Estrogen receptor?"
    ],
    "objectives": [
      "Inspect read quality with FastQC",
      "Map reads with Bowtie2",
      "Assess the quality of an ChIP-seq experiments",
      "Extract coverage files",
      "Call enriched regions or peaks"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?ChIP-seq\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">ChIP-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html": {
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)",
    "description": null,
    "question": [
      "How is raw ChIP-seq data processed and analyzed?",
      "What are the binding sites of TAL1?",
      "Which genes are regulated by TAL1?"
    ],
    "objectives": [
      "Inspect read quality with FastQC",
      "Perform read trimming with Trimmomatic",
      "Align trimmed reads with BWA",
      "Assess quality and reproducibility of experiments",
      "Identify TAL1 binding sites with MACS2",
      "Determine unique/common TAL1 binding sites from G1E and Megakaryocytes",
      "Identify unique/common TAL1 peaks occupying gene promoters",
      "Visually inspect TAL1 peaks with Trackster"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?ChIP-seq\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">ChIP-seq</a>"
    ],
    "contributors": "Mallory Freeberg, Mo Heydarian, Vivek Bhardwaj, Joachim Wolff, Anika Erxleben",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/ewas-suite/tutorial.html": {
    "topic": "Epigenetics",
    "title": "Infinium Human Methylation BeadChip",
    "description": null,
    "question": [
      "Which DNA regions and positions are diffrentialy methylated in pre MAPKi treatment and post MAPKi resistance Melanomas GSE65183?",
      "How to analyse and visualise Infinium Human Methylation BeadChip's?"
    ],
    "objectives": [
      "Learn how to perform reproducible Infinium Human Methylation BeadChip analysis",
      "Visualise differentially methylated positions using UCSC browser"
    ],
    "tags": [

    ],
    "contributors": "Katarzyna Murat, Krzysztof Poterlowicz",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/tutorial.html": {
    "topic": "Sequence analysis",
    "title": "Mapping",
    "description": null,
    "question": [
      "What is mapping?",
      "What two things are crucial for a correct mapping?",
      "What is BAM?"
    ],
    "objectives": [
      "You will learn what mapping is",
      "A genome browser is shown that helps you to understand your data"
    ],
    "tags": [

    ],
    "contributors": "Joachim Wolff, B√©r√©nice Batut, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/tutorial.html": {
    "topic": "Sequence analysis",
    "title": "Quality Control",
    "description": null,
    "question": [
      "How to perform quality control of NGS raw data?",
      "What are the quality parameters to check for a dataset?",
      "How to improve the quality of a dataset?"
    ],
    "objectives": [
      "Assess short reads FASTQ quality using FASTQE üß¨üòé and FastQC",
      "Assess long reads FASTQ quality using Nanoplot and PycoQC",
      "Perform quality correction with Cutadapt (short reads)",
      "Summarise quality metrics MultiQC",
      "Process single-end and paired-end data"
    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Maria Doyle, Alexandre Cormier, Anthony Bretaudeau, Laura Leroi, Erwan Corre, St√©phanie Robin, Erasmus+ Programme",
    "level": "Introductory",
    "time_estimation": "1H30M",
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html": {
    "topic": "Sequence analysis",
    "title": "Removal of human reads from SARS-CoV-2 sequencing data",
    "description": null,
    "question": [
      "How can you remove traces of human sequence data from a sequenced viral sample?"
    ],
    "objectives": [
      "Obtain viral (SARS-CoV-2) sequencing data with contaminating human reads from public sources",
      "Organize the data into a collection",
      "Preprocess and map the data against the human reference genome",
      "Eliminate reads/read pairs that map to the human genome from the original data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Wolfgang Maier",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html"
  },
  "/topics/introduction/slides/introduction.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Galaxy",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Andrea Bagnacani, B√©r√©nice Batut, Saskia Hiltemann, Anne Pajon, Nicola Soranzo, Helena Rasche, Christopher Barnett, Michele Maroni, Anne Fouilloux, Nadia Gou√©, Olha Nahorna, Dave Clements",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/introduction/slides/introduction.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-short/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "A short introduction to Galaxy",
    "description": null,
    "question": [
      "How to get started in Galaxy"
    ],
    "objectives": [
      "Learn how to upload a file",
      "Learn how to use a tool",
      "Learn how to view results",
      "Learn how to view histories",
      "Learn how to extract and run a workflow"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?espa√±ol\" style=\"background-color: hsl(184, 80.0%, 85%); color: #333; border: 1px solid hsl(184, 80.0%, 45%)\">espa√±ol</a>"
    ],
    "contributors": "Anna Syme, Nicola Soranzo",
    "level": "Introductory",
    "time_estimation": "40m",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "From peaks to genes",
    "description": null,
    "question": [
      "How to use Galaxy?",
      "How to get from peak regions to a list of gene names?"
    ],
    "objectives": [
      "Familiarize yourself with the basics of Galaxy",
      "Learn how to obtain data from external sources",
      "Learn how to run tools",
      "Learn how histories work",
      "Learn how to create a workflow",
      "Learn how to share your work"
    ],
    "tags": [

    ],
    "contributors": "Anne Pajon, Clemens Blank, B√©r√©nice Batut, Bj√∂rn Gr√ºning, Nicola Soranzo, Dilmurat Yusuf, Sarah Peter, Helena Rasche",
    "level": "Introductory",
    "time_estimation": "3H",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy 101",
    "description": null,
    "question": [
      "Which coding exon has the highest number of single nucleotide polymorphisms (SNPs) on human chromosome 22?"
    ],
    "objectives": [
      "Familiarize yourself with the basics of Galaxy",
      "Learn how to obtain data from external sources",
      "Learn how to run tools",
      "Learn how histories work",
      "Learn how to create a workflow",
      "Learn how to share your work"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, Nicola Soranzo, Clemens Blank, Anton Nekrutenko, Bj√∂rn Gr√ºning, Anne Pajon, Helena Rasche",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy 101 for everyone",
    "description": null,
    "question": [
      "What are the differences between the Iris species?"
    ],
    "objectives": [
      "Familiarize yourself with the basics of Galaxy",
      "Learn how to obtain data from external sources",
      "Learn how to tag datasets",
      "Learn how to run tools",
      "Learn how histories work",
      "Learn how to create a workflow",
      "Learn how to share your work"
    ],
    "tags": [

    ],
    "contributors": "Anne Fouilloux, Nadia Gou√©, Christopher Barnett, Michele Maroni, Olha Nahorna, Dave Clements, Saskia Hiltemann",
    "level": "Introductory",
    "time_estimation": "1H30M",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-reproduce/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "How to reproduce published Galaxy analyses",
    "description": null,
    "question": [
      "How to reproduce published Galaxy results (workflows and histories)"
    ],
    "objectives": [
      "Learn how to load published data into Galaxy",
      "Learn how to run a published Galaxy workflow",
      "Learn how histories can be inspected and re-used."
    ],
    "tags": [

    ],
    "contributors": "Melanie F√∂ll, Anne Fouilloux",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html"
  },
  "/topics/introduction/tutorials/igv-introduction/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "IGV Introduction",
    "description": null,
    "question": null,
    "objectives": null,
    "tags": [

    ],
    "contributors": "Anton Nekrutenko, Joachim Wolff, Anne Pajon",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Genomics and Galaxy",
    "description": null,
    "question": [
      "Do genes on opposite strands ever overlap?  If so, how often?"
    ],
    "objectives": [
      "Learn some of the basic concepts of genomics and bioinformatics",
      "Familiarize yourself with the basics of Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Dave Clements, Crist√≥bal Gallardo",
    "level": "Introductory",
    "time_estimation": "2H30M",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "NGS data logistics",
    "description": null,
    "question": [
      "How to manipulate and process NGS data"
    ],
    "objectives": [
      "Understand most common types of NGS-related datatypes",
      "Learn about how Galaxy handles NGS data using Illumina data derived from patients infected with SARS-CoV-2"
    ],
    "tags": [

    ],
    "contributors": "Anton Nekrutenko, Marius van den Beek, Dave Clements, Daniel Blankenberg",
    "level": null,
    "time_estimation": "1H30M",
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html"
  },
  "/topics/introduction/tutorials/options-for-using-galaxy/slides.html": {
    "topic": "Introduction to Galaxy Analyses",
    "title": "Options for using Galaxy",
    "description": null,
    "question": [
      "Which Galaxy instance should I use?"
    ],
    "objectives": null,
    "tags": [

    ],
    "contributors": "Dave Clements, Anthony Bretaudeau, Anne Pajon, Nicola Soranzo, Alaa Elghamry",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/introduction/tutorials/options-for-using-galaxy/slides.html"
  },
  "/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Age prediction using machine learning",
    "description": null,
    "question": [
      "How to use machine learning to create predictive models from biological datasets (RNA-seq and DNA methylation)?"
    ],
    "objectives": [
      "Learn ageing biomarkers from RNA-seq and DNA methylation datasets",
      "Apply regression based machine learning algorithms",
      "Learn feature selection and hyperparameter optimisation"
    ],
    "tags": [

    ],
    "contributors": "Ekaterina Polkh, Anup Kumar",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html"
  },
  "/topics/statistics/tutorials/machinelearning/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Basics of machine learning",
    "description": null,
    "question": [
      "What is machine learning?",
      "Why is it useful?",
      "What are its different approaches?"
    ],
    "objectives": [
      "Provide the basics of machine learning and its variants.",
      "Learn how to do classification using the training and test data.",
      "Learn how to use Galaxy's machine learning tools."
    ],
    "tags": [

    ],
    "contributors": "Anup Kumar",
    "level": null,
    "time_estimation": "30M",
    "url": "/training-material/topics/statistics/tutorials/machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/classification_machinelearning/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Classification in Machine Learning",
    "description": null,
    "question": [
      "What is classification and how we can use classification techniques?"
    ],
    "objectives": [
      "Learn classification background",
      "Learn what a quantitative structure-analysis relationship (QSAR) model is and how it can be constructed in Galaxy",
      "Learn to apply logistic regression, k-nearest neighbors, support verctor machines, random forests and bagging algorithms",
      "Learn how visualizations can be used to analyze the classification results"
    ],
    "tags": [

    ],
    "contributors": "Alireza Khanteymoori, Anup Kumar, Simon Bray",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/clustering_machinelearning/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Clustering in Machine Learning",
    "description": null,
    "question": [
      "How to use clustering algorithms to categorize data in different clusters"
    ],
    "objectives": [
      "Learn clustering background",
      "Learn hierarchical clustering algorithm",
      "Learn k-means clustering algorithm",
      "Learn DBSCAN clustering algorithm",
      "Apply clustering algorithms to different datasets",
      "Learn how to visualize clusters"
    ],
    "tags": [

    ],
    "contributors": "Alireza Khanteymoori, Anup Kumar",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/FNN/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 1) - Feedforward neural networks (FNN)",
    "description": null,
    "question": [
      "What is a feedforward neural network (FNN)?",
      "What are some applications of FNN?"
    ],
    "objectives": [
      "Understand the inspiration for neural networks",
      "Learn various activation functions, and classification/regression problems solved by neural networks",
      "Discuss various cost/loss functions and the backpropagation algorithm",
      "Learn how to create a neural network using Galaxy's deep learning tools",
      "Solve a simple regression problem, car purchase price prediction, via FNN in Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Kaivan Kamali",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/FNN/tutorial.html"
  },
  "/topics/statistics/tutorials/RNN/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 2) - Recurrent neural networks (RNN)",
    "description": null,
    "question": [
      "What is a recurrent neural network (RNN)?",
      "What are some applications of RNN?"
    ],
    "objectives": [
      "Understand the difference between feedforward neural networks (FNN) and RNN",
      "Learn various RNN types and architectures",
      "Learn how to create a neural network using Galaxy's deep learning tools",
      "Solve a sentiment analysis problem on IMDB movie review dataset using RNN in Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Kaivan Kamali",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/RNN/tutorial.html"
  },
  "/topics/statistics/tutorials/CNN/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 3) - Convolutional neural networks (CNN)",
    "description": null,
    "question": [
      "What is a convolutional neural network (CNN)?",
      "What are some applications of CNN?"
    ],
    "objectives": [
      "Understand the inspiration behind CNN and learn the CNN architecture",
      "Learn the convolution operation and its parameters",
      "Learn how to create a CNN using Galaxy's deep learning tools",
      "Solve an image classification problem on MNIST digit classification dataset using CNN in Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Kaivan Kamali",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/CNN/tutorial.html"
  },
  "/topics/statistics/tutorials/fruit_360/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Image classification in Galaxy with fruit 360 dataset",
    "description": null,
    "question": [
      "How to solve an image classification problem using convolutional neural network (CNN)?"
    ],
    "objectives": [
      "Learn how to create a CNN using Galaxy's deep learning tools",
      "Solve an image classification problem on fruit 360 dataset using CNN in Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Kaivan Kamali",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/fruit_360/tutorial.html"
  },
  "/topics/statistics/tutorials/iwtomics/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Interval-Wise Testing for omics data",
    "description": null,
    "question": [
      "How to visualize high-resolution omics data in different groups of genomic regions?",
      "How to evaluate differences in high-resolution omics data between groups of genomic regions?",
      "How to detect locations and scales at which the significant effects unfold?"
    ],
    "objectives": [
      "Pre-process high-resolution omics data",
      "Create pointwise boxplots corresponding to different groups of genomic regions",
      "Perform the Interval-Wise Testing between two groups of genomic regions",
      "Visualize and inspect test results",
      "Detect relevant locations and scales"
    ],
    "tags": [

    ],
    "contributors": "Marzia A Cremona, Fabio Cumbo",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/statistics/tutorials/iwtomics/tutorial.html"
  },
  "/topics/statistics/tutorials/intro_deep_learning/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Introduction to deep learning",
    "description": null,
    "question": [
      "What are deep learning and neural networks?",
      "Why is it useful?",
      "How to create a neural network architecture for classification?"
    ],
    "objectives": [
      "Learn basic principles of deep learning",
      "Learn about how to create an end-to-end neural network architecture",
      "Learn about Galaxy deep learning tools",
      "Learn how to interpret predictions"
    ],
    "tags": [

    ],
    "contributors": "Anup Kumar, Alireza Khanteymoori",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html"
  },
  "/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Introduction to Machine Learning using R",
    "description": null,
    "question": [
      "What are the main categories in Machine Learning algorithms?",
      "How can I perform exploratory data analysis?",
      "What are the main part of a clustering process?",
      "How can a create a decision tree?",
      "How can I assess a linear regression model?"
    ],
    "objectives": [
      "Understand the ML taxonomy and the commonly used machine learning algorithms for analysing -omics data",
      "Understand differences between ML algorithms categories and to which kind of problem they can be applied",
      "Understand different applications of ML in different -omics studies",
      "Use some basic, widely used R packages for ML",
      "Interpret and visualize the results obtained from ML analyses on omics datasets",
      "Apply the ML techniques to analyse their own datasets"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Fotis E. Psomopoulos, Erasmus+ Programme",
    "level": "Intermediate",
    "time_estimation": "3H",
    "url": "/training-material/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html"
  },
  "/topics/statistics/tutorials/classification_regression/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Machine learning: classification and regression",
    "description": null,
    "question": [
      "what are classification and regression techniques?",
      "How they can be used for prediction?",
      "How visualizations can be used to analyze predictions?"
    ],
    "objectives": [
      "Explain the types of supervised machine learning - classification and regression.",
      "Learn how to make predictions using the training and test dataset.",
      "Visualize the predictions."
    ],
    "tags": [

    ],
    "contributors": "Anup Kumar, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/statistics/tutorials/classification_regression/tutorial.html"
  },
  "/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis",
    "description": null,
    "question": [
      "How to predict aberrant pathway activity in The Cancer Genome Atlas (TCGA) using Machine learning approaches?"
    ],
    "objectives": [
      "Learn to predict aberrant pathway activity using RNA-Seq data, mutational status and copy number variation data from TCGA.",
      "Apply logistic regression based machine learning algorithms on TCGA data."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Machine learning\" href=\"/training-material/search?Machine learning\" style=\"background-color: hsl(269, 80.0%, 85%); color: #333; border: 1px solid hsl(269, 80.0%, 45%)\">Machine learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Pan-cancer\" href=\"/training-material/search?Pan-cancer\" style=\"background-color: hsl(185, 80.0%, 85%); color: #333; border: 1px solid hsl(185, 80.0%, 45%)\">Pan-cancer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer biomarkers\" href=\"/training-material/search?cancer biomarkers\" style=\"background-color: hsl(143, 80.0%, 85%); color: #333; border: 1px solid hsl(143, 80.0%, 45%)\">cancer biomarkers</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged oncogenes and tumor suppressor genes\" href=\"/training-material/search?oncogenes and tumor suppressor genes\" style=\"background-color: hsl(358, 80.0%, 85%); color: #333; border: 1px solid hsl(358, 80.0%, 45%)\">oncogenes and tumor suppressor genes</a>"
    ],
    "contributors": "Vijay, Daniel Blankenberg",
    "level": null,
    "time_estimation": "1H30M",
    "url": "/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html"
  },
  "/topics/statistics/tutorials/regression_machinelearning/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Regression in Machine Learning",
    "description": null,
    "question": [
      "How to use regression techniques to create predictive models from biological datasets?"
    ],
    "objectives": [
      "Learn regression background",
      "Apply regression based machine learning algorithms",
      "Learn ageing biomarkers and predict age from DNA methylation datasets",
      "Learn how visualizations can be used to analyze predictions"
    ],
    "tags": [

    ],
    "contributors": "Alireza Khanteymoori, Anup Kumar, Simon Bray",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/text-mining_simtext/tutorial.html": {
    "topic": "Statistics and machine learning",
    "title": "Text-mining with the SimText toolset",
    "description": null,
    "question": [
      "How can I automatically collect PubMed data for a set of biomedical entities such as genes?",
      "How can I analyze similarities among biomedical entities based on PubMed data on large-scale?"
    ],
    "objectives": [
      "Learn how to use the SimText toolset",
      "Upload a table with biomedical entities in Galaxy",
      "Retrieve PubMed data for each of the biomedical entities",
      "Extract biomedical terms from the PubMed data for each biomedical entity",
      "Analyze the similarity among the biomedical entities based on the extracted data in an interactive app"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Marie Gramm, Dennis Lal group, Daniel Blankenberg",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html"
  },
  "/topics/transcriptomics/slides/introduction.html": {
    "topic": "Transcriptomics",
    "title": "Introduction to Transcriptomics",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Anika Erxleben, Markus Wolfien",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/transcriptomics/slides/introduction.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "1: RNA-Seq reads to counts",
    "description": null,
    "question": [
      "How to convert RNA-seq reads into counts?",
      "How to perform quality control (QC) of RNA-seq reads?",
      "How to do this analysis efficiently in Galaxy?"
    ],
    "objectives": [
      "Learn how RNA-seq reads are converted into counts",
      "Understand QC steps that can be performed on RNA-seq reads",
      "Generate interactive reports to summarise QC information with MultiQC",
      "Use the Galaxy Rule-based Uploader to import FASTQs from URLs",
      "Make use of Galaxy Collections for a tidy analysis",
      "Create a Galaxy Workflow that converts RNA-seq reads into counts"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?collections\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?mouse\" style=\"background-color: hsl(164, 80.0%, 85%); color: #333; border: 1px solid hsl(164, 80.0%, 45%)\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?QC\" style=\"background-color: hsl(12, 80.0%, 85%); color: #333; border: 1px solid hsl(12, 80.0%, 45%)\">QC</a>"
    ],
    "contributors": "Maria Doyle, Belinda Phipson, Harriet Dashnow",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "2: RNA-seq counts to genes",
    "description": null,
    "question": [
      "What are the differentially expressed genes in the mammary gland of pregnant versus lactating mice?",
      "How to analyze RNA count data using limma-voom?",
      "How to perform quality control (QC) of RNA-seq count data?"
    ],
    "objectives": [
      "Analysis of RNA-seq count data using limma-voom",
      "QC of count data",
      "Visualisation and interactive exploration of count data",
      "Identification of differentially expressed genes"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged limma-voom\" href=\"/training-material/search?limma-voom\" style=\"background-color: hsl(213, 80.0%, 85%); color: #333; border: 1px solid hsl(213, 80.0%, 45%)\">limma-voom</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?mouse\" style=\"background-color: hsl(164, 80.0%, 85%); color: #333; border: 1px solid hsl(164, 80.0%, 45%)\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?QC\" style=\"background-color: hsl(12, 80.0%, 85%); color: #333; border: 1px solid hsl(12, 80.0%, 45%)\">QC</a>"
    ],
    "contributors": "Maria Doyle, Belinda Phipson, Jovana Maksimovic, Anna Trigos, Matt Ritchie, Harriet Dashnow, Shian Su, Charity Law",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "3: RNA-seq genes to pathways",
    "description": null,
    "question": [
      "What are the differentially expressed pathways in the mammary gland of pregnant versus lactating mice?"
    ],
    "objectives": [
      "Identification of differentially expressed pathways"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?mouse\" style=\"background-color: hsl(164, 80.0%, 85%); color: #333; border: 1px solid hsl(164, 80.0%, 45%)\">mouse</a>"
    ],
    "contributors": "Maria Doyle, Belinda Phipson",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-intro/slides.html": {
    "topic": "Transcriptomics",
    "title": "An introduction to scRNA-seq data analysis",
    "description": null,
    "question": [
      "How are samples compared?",
      "How are cells captured?",
      "How does bulk RNA-seq differ from scRNA-seq?",
      "Why is clustering important?"
    ],
    "objectives": [
      "To understand the pitfalls in scRNA-seq sequencing and amplification, and how they are overcome.",
      "Know the types of variation in an analysis and how to control for them.",
      "Grasp what dimension reduction is, and how it might be performed.",
      "Be familiarised with the main types of clustering techniques and when to use them."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?espa√±ol\" style=\"background-color: hsl(184, 80.0%, 85%); color: #333; border: 1px solid hsl(184, 80.0%, 45%)\">espa√±ol</a>"
    ],
    "contributors": "Mehmet Tekman",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-intro/slides.html"
  },
  "/topics/transcriptomics/tutorials/scrna-plant/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Analysis of plant scRNA-Seq Data with Scanpy",
    "description": null,
    "question": [
      "Can we reclaim cell markers using a different analysis method?",
      "Are highly variable genes paramount to the analysis?"
    ],
    "objectives": [
      "Perform filtering, dimensionality reduction, and clustering",
      "Generate a DotPlot emulating the original paper using a different analysis tool",
      "Determine robust clusters across scRNA-seq pipelines"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?plants\" style=\"background-color: hsl(121, 80.0%, 85%); color: #333; border: 1px solid hsl(121, 80.0%, 45%)\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?paper-replication\" style=\"background-color: hsl(190, 80.0%, 85%); color: #333; border: 1px solid hsl(190, 80.0%, 45%)\">paper-replication</a>"
    ],
    "contributors": "Mehmet Tekman, Beatriz Serrano-Solano, Crist√≥bal Gallardo, Pavankumar Videm",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-plant/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/clipseq/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "CLIP-Seq data analysis from pre-processing to motif detection",
    "description": null,
    "question": [
      "How is raw CLIP-Seq data processed and analysed?",
      "How do I find binding motifs and targets for a protein (e.g., RBFOX2)?"
    ],
    "objectives": [
      "Remove Adapters, Barcodes and Unique Molecular Identifiers (UMIs) from the reads",
      "Align trimmed reads with STAR",
      "De-duplicate the read library",
      "Inspect the read mapping and de-duplication quality",
      "Perform peak calling with PEAKachu",
      "Analyse the peaks and find potential binding motifs and targets",
      "Check the quality of the peak calling"
    ],
    "tags": [

    ],
    "contributors": "Florian Heyl, Daniel Maticzka, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "6h",
    "url": "/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Clustering 3K PBMCs with Scanpy",
    "description": null,
    "question": [
      "What are the steps to prepare single-cell RNA-Seq data for clustering?",
      "How to cluster cells in single-cell RNA-Seq data?",
      "How cell type annotation can be assigned to cell clusters?"
    ],
    "objectives": [
      "Describe an AnnData object to store single-cell data",
      "Explain the preprocessing steps for single-cell data",
      "Evaluate quality of single-cell data and apply steps to select and filter cells and genes based on QC",
      "Execute data normalization and scaling",
      "Identify highly variable genes",
      "Construct and run a dimensionality reduction using Principal Component Analysis",
      "Perform a graph-based clustering for cells",
      "Identify marker genes for the clusters",
      "Construct and run a cell type annotation for the clusters"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?10x\" style=\"background-color: hsl(3, 80.0%, 85%); color: #333; border: 1px solid hsl(3, 80.0%, 45%)\">10x</a>"
    ],
    "contributors": "B√©r√©nice Batut, Hans-Rudolf Hotz, Mehmet Tekman",
    "level": null,
    "time_estimation": "5H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/full-de-novo/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "De novo transcriptome assembly, annotation, and differential expression analysis",
    "description": null,
    "question": [
      "Which biological questions are addressed by the tutorial?",
      "Which bioinformatics techniques are important to know for this type of data?"
    ],
    "objectives": [
      "The learning objectives are the goals of the tutorial",
      "They will be informed by your audience and will communicate to them and to yourself what you should focus on during the course",
      "They are single sentences describing what a learner should be able to do once they have completed the tutorial",
      "You can use Bloom's Taxonomy to write effective learning objectives"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Anthony Bretaudeau, Gildas Le Corguill√©, Erwan Corre, Xi Liu",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "De novo transcriptome reconstruction with RNA-Seq",
    "description": null,
    "question": [
      "What genes are differentially expressed between G1E cells and megakaryocytes?",
      "How can we generate a transcriptome de novo from RNA sequencing data?"
    ],
    "objectives": [
      "Analysis of RNA sequencing data using a reference genome",
      "Reconstruction of transcripts without reference transcriptome (de novo)",
      "Analysis of differentially expressed genes"
    ],
    "tags": [

    ],
    "contributors": "Mallory Freeberg, Mo Heydarian",
    "level": null,
    "time_estimation": "6h",
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/srna/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Differential abundance testing of small RNAs",
    "description": null,
    "question": [
      "What small RNAs are expressed?",
      "What RNA features have significantly different numbers of small RNAs targeting them between two conditions?"
    ],
    "objectives": [
      "Process small RNA-seq datasets to determine quality and reproducibility.",
      "Filter out contaminants (e.g. rRNA reads) in small RNA-seq datasets.",
      "Differentiate between subclasses of small RNAs based on their characteristics.",
      "Identify differently abundant small RNAs and their targets."
    ],
    "tags": [

    ],
    "contributors": "Mallory Freeberg",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/transcriptomics/tutorials/srna/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-raceid/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Downstream Single-cell RNA analysis with RaceID",
    "description": null,
    "question": [
      "What is normalisation and why is it necessary?",
      "How many types of unwanted variation are there?",
      "How are biological phenotypes clustered?",
      "What is the difference between PCA and tSNE?",
      "What is the difference between cell trajectory and cell fate?"
    ],
    "objectives": [
      "Filtering, normalising, and clustering cells in a matrix",
      "Assessing the quality of individual clusters",
      "Inferring cell type lineages",
      "Examining gene expression",
      "Determining the top most expressive genes per cluster",
      "Correcting for unwanted variation"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>"
    ],
    "contributors": "Mehmet Tekman, Alex Ostrovsky",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-raceid/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-seq-basic-pipeline/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Filter, Plot and Explore Single-cell RNA-seq Data",
    "description": null,
    "question": [
      "Is my single cell dataset a quality dataset?",
      "How do I generate and annotate cell clusters?",
      "How do I pick thresholds and parameters in my analysis? What's a \"reasonable\" number, and will the world collapse if I pick the wrong one?"
    ],
    "objectives": [
      "Interpret quality control plots to direct parameter decisions",
      "Repeat analysis from matrix to clustering",
      "Identify decision-making points",
      "Appraise data outputs and decisions",
      "Explain why single cell analysis is an iterative (i.e. the first plots you generate are not final, but rather you go back and re-analyse your data repeatedly) process"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?10x\" style=\"background-color: hsl(3, 80.0%, 85%); color: #333; border: 1px solid hsl(3, 80.0%, 45%)\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?paper-replication\" style=\"background-color: hsl(190, 80.0%, 85%); color: #333; border: 1px solid hsl(190, 80.0%, 45%)\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?espa√±ol\" style=\"background-color: hsl(184, 80.0%, 85%); color: #333; border: 1px solid hsl(184, 80.0%, 45%)\">espa√±ol</a>"
    ],
    "contributors": "Wendi Bacon",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-seq-basic-pipeline/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/droplet-quantification-preprocessing/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Generating a single cell matrix using Alevin",
    "description": null,
    "question": [
      "I have some single cell FASTQ files I want to analyse. Where do I start?"
    ],
    "objectives": [
      "Repeat matrix generation for any droplet-based single cell sequencing data",
      "Apply data combination and metadata editing for particular experimental designs",
      "Interpret quality control (QC) plots to make informed decisions on cell thresholds",
      "Find relevant information in GTF files for the particulars of their study, and include this in data matrix metadata"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?10x\" style=\"background-color: hsl(3, 80.0%, 85%); color: #333; border: 1px solid hsl(3, 80.0%, 45%)\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?paper-replication\" style=\"background-color: hsl(190, 80.0%, 85%); color: #333; border: 1px solid hsl(190, 80.0%, 45%)\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?espa√±ol\" style=\"background-color: hsl(184, 80.0%, 85%); color: #333; border: 1px solid hsl(184, 80.0%, 45%)\">espa√±ol</a>"
    ],
    "contributors": "Wendi Bacon, Jonathan Manning",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/transcriptomics/tutorials/droplet-quantification-preprocessing/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/goenrichment/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "GO Enrichment Analysis",
    "description": null,
    "question": [
      "How can I functionally interpret a list of genes of interest that I obtained from my experiment?"
    ],
    "objectives": [
      "How to perform a GO Enrichment Analysis",
      "How to interpret and simplify the results"
    ],
    "tags": [

    ],
    "contributors": "IGC Bioinformatics Unit, Maria Doyle",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Network analysis with Heinz",
    "description": null,
    "question": [
      "Which pathways are potentially contributing to dental caries?"
    ],
    "objectives": [
      "Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?metatranscriptomics\" style=\"background-color: hsl(294, 80.0%, 85%); color: #333; border: 1px solid hsl(294, 80.0%, 45%)\">metatranscriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged network analysis\" href=\"/training-material/search?network analysis\" style=\"background-color: hsl(170, 80.0%, 85%); color: #333; border: 1px solid hsl(170, 80.0%, 45%)\">network analysis</a>"
    ],
    "contributors": "Chao (Cico) Zhang",
    "level": null,
    "time_estimation": "1h30m",
    "url": "/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html": {
    "topic": "Transcriptomics",
    "title": "Plates, Batches, and Barcodes",
    "description": null,
    "question": [
      "What is a batch when it comes to single cells?",
      "What‚Äôs the difference between a barcode and an index?",
      "What is batch effect and how do you prevent it?",
      "What is a lane?"
    ],
    "objectives": [
      "How to set up plates to prevent batch effect",
      "Proper naming conventions when dealing with scRNA-seq samples"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>"
    ],
    "contributors": "Wendi Bacon, Mehmet Tekman, Alex Ostrovsky",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html"
  },
  "/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Pre-processing of 10X Single-Cell RNA Datasets",
    "description": null,
    "question": [
      "What is 10X?",
      "What is STARsolo and what is Cell Ranger?",
      "What are BCL and MTX files?",
      "What is an HDF5 file, and why is it important?"
    ],
    "objectives": [
      "Demultiplex single-cell FASTQ data from 10X Genomics",
      "Learn about transparent matrix formats",
      "Understand the importance of high and low quality cells"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?10x\" style=\"background-color: hsl(3, 80.0%, 85%); color: #333; border: 1px solid hsl(3, 80.0%, 45%)\">10x</a>"
    ],
    "contributors": "Mehmet Tekman, Hans-Rudolf Hotz, Daniel Blankenberg, Wendi Bacon",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Pre-processing of Single-Cell RNA Data",
    "description": null,
    "question": [
      "What is single-cell?",
      "How do we process a batch?",
      "How do we process multiple batches?",
      "How do we check for cross-contamination?"
    ],
    "objectives": [
      "Demultiplex FASTQ data via UMI-tools",
      "Understand and validate the extraction of barcodes",
      "Obtain an overview of general alignment and quantification techniques",
      "Generate a count matrix for downstream single-cell RNA analysis"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>"
    ],
    "contributors": "Mehmet Tekman, B√©r√©nice Batut, Anika Erxleben, Wolfgang Maier",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Reference-based RNA-Seq data analysis",
    "description": null,
    "question": [
      "What are the steps to process RNA-Seq data?",
      "How to identify differentially expressed genes across multiple experimental conditions?",
      "What are the biological functions impacted by the differential expression of genes?"
    ],
    "objectives": [
      "Check a sequence quality report generated by FastQC for RNA-Seq data",
      "Explain the principle and specificity of mapping of RNA-Seq data to an eukaryotic reference genome",
      "Select and run a state of the art mapping tool for RNA-Seq data",
      "Evaluate the quality of mapping results",
      "Describe the process to estimate the library strandness",
      "Estimate the number of reads per genes",
      "Explain the count normalization to perform before sample comparison",
      "Construct and run a differential gene expression analysis",
      "Analyze the DESeq2 output to identify, annotate and visualize differentially expressed genes",
      "Perform a gene ontology enrichment analysis",
      "Perform and visualize an enrichment analysis for KEGG pathways"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?bulk\" style=\"background-color: hsl(119, 80.0%, 85%); color: #333; border: 1px solid hsl(119, 80.0%, 45%)\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rna-seq\" href=\"/training-material/search?rna-seq\" style=\"background-color: hsl(37, 80.0%, 85%); color: #333; border: 1px solid hsl(37, 80.0%, 45%)\">rna-seq</a>"
    ],
    "contributors": "B√©r√©nice Batut, Mallory Freeberg, Mo Heydarian, Anika Erxleben, Pavankumar Videm, Clemens Blank, Maria Doyle, Nicola Soranzo, Peter van Heusden",
    "level": null,
    "time_estimation": "8h",
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Reference-based RNAseq data analysis (long)",
    "description": null,
    "question": [
      "How to perform analysis of RNAseq data when reference genome is available?"
    ],
    "objectives": [
      "Learns basic concepts of RNAseq analysis"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Anton Nekrutenko",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "RNA Seq Counts to Viz in R",
    "description": null,
    "question": [
      "How can I create neat visualizations of the data?"
    ],
    "objectives": [
      "Describe the role of data, aesthetics, geoms, and layers in `ggplot` functions.",
      "Customize plot scales, titles, subtitles, themes, fonts, layout, and orientation."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "B√©r√©nice Batut, Fotis E. Psomopoulos, Toby Hodges, Erasmus+ Programme",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-interactome/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "RNA-RNA interactome data analysis",
    "description": null,
    "question": [
      "What are the difficulties in mapping chimeric reads from RNA interactome data?",
      "How multi mapping is a big problem in these datasets?",
      "How to filter for meaningful results from large analysis output files?"
    ],
    "objectives": [
      "Quality control and data preparation",
      "Mapping chimeric reads",
      "Quantification of the mapped loci",
      "Visualization and filtering of results"
    ],
    "tags": [

    ],
    "contributors": "Pavankumar Videm",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "RNA-Seq analysis with AskOmics Interactive Tool",
    "description": null,
    "question": [
      "How to integrate RNA-Seq results with other datasets using AskOmics?",
      "How to query these datasets to answer a biological question?",
      "How to exploit a distant SPARQL endpoint to retrieve information?"
    ],
    "objectives": [
      "Launch an AskOmics Interactive Tool",
      "Integrate RNA-Seq and reference datasets into AskOmics",
      "Create a complex query to get over-expressed genes, their location on the reference genome, and check if they are included in a known QTL",
      "Complete the query to get the human homologs and their location using neXtProt database"
    ],
    "tags": [

    ],
    "contributors": "Xavier Garnier, Anthony Bretaudeau, Anne Siegel, Olivier Dameron, Mateo Boudet",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-scater-qc/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Single-cell quality control with scater",
    "description": null,
    "question": [
      "How to ensure the quality of single-cell RNA-seq data?",
      "What are the confounding factors that may affect the interpretation of downstream analyses?"
    ],
    "objectives": [
      "Examine scRNA-seq data for low-quality cells",
      "Visualise data with various types of plots",
      "Filtering low-quality cells with the appropriate approach"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>"
    ],
    "contributors": "Graham Etherington, Nicola Soranzo",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-scater-qc/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Small Non-coding RNA Clustering using BlockClust",
    "description": null,
    "question": [
      "What do the read profiles of small non-coding RNAs represent?",
      "How to cluster the read profiles based on some intrinsic features?"
    ],
    "objectives": [
      "Difference between mRNA-seq and smallRNA-seq",
      "Unsupervised grouping of the adjacent reads into read profiles",
      "Look and learn what the shapes of the read profiles represent",
      "Clustering of the read profiles by machine learning algorithm"
    ],
    "tags": [

    ],
    "contributors": "Pavankumar Videm",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-JUPYTER-trajectories/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Trajectory Analysis using Python (Jupyter Notebook) in Galaxy",
    "description": null,
    "question": [
      "How can I infer lineage relationships between single cells based on their RNA, without a time series?"
    ],
    "objectives": [
      "Execute multiple plotting methods designed to maintain lineage relationships between cells",
      "Interpret these plots"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?10x\" style=\"background-color: hsl(3, 80.0%, 85%); color: #333; border: 1px solid hsl(3, 80.0%, 45%)\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?paper-replication\" style=\"background-color: hsl(190, 80.0%, 85%); color: #333; border: 1px solid hsl(190, 80.0%, 45%)\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-JUPYTER-trajectories/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/scrna-umis/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Understanding Barcodes",
    "description": null,
    "question": [
      "What are barcodes?",
      "What is their purpose?",
      "How are barcodes formatted?"
    ],
    "objectives": [
      "Demultiplex FASTQ data via UMI-tools",
      "Understand and validate the extraction of barcodes"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?single-cell\" style=\"background-color: hsl(179, 80.0%, 85%); color: #333; border: 1px solid hsl(179, 80.0%, 45%)\">single-cell</a>"
    ],
    "contributors": "Mehmet Tekman",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with CummeRbund",
    "description": null,
    "question": [
      "How are RNA-Seq results stored?",
      "Why are visualization techniques needed?",
      "How to select genes for visualizing meaningful results of differential gene expression analysis?"
    ],
    "objectives": [
      "Manage RNA-Seq results",
      "Extract genes for producing differential gene expression analysis visualizations",
      "Visualize meaningful information"
    ],
    "tags": [

    ],
    "contributors": "Andrea Bagnacani",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with heatmap2",
    "description": null,
    "question": [
      "How to generate heatmaps from RNA-seq data?"
    ],
    "objectives": [
      "Create heatmaps of RNA-seq data"
    ],
    "tags": [

    ],
    "contributors": "Maria Doyle",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot",
    "description": null,
    "question": [
      "How to generate a volcano plot from RNA-seq data?"
    ],
    "objectives": [
      "Create a volcano plot of RNA-seq data to visualize significant genes"
    ],
    "tags": [

    ],
    "contributors": "Maria Doyle",
    "level": "Introductory",
    "time_estimation": "30m",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot in R",
    "description": null,
    "question": [
      "How to customise Volcano plot output in R?"
    ],
    "objectives": [
      "Learn how to use R to edit Volcano plot colours, points, labels and categories"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Maria Doyle",
    "level": "Intermediate",
    "time_estimation": "1H",
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html": {
    "topic": "Transcriptomics",
    "title": "Whole transcriptome analysis of Arabidopsis thaliana",
    "description": null,
    "question": [
      "Which miRNAs are upregulated in response to brassinosteroids?",
      "Which genes are potential target of brassinosteroid-induced miRNAs?"
    ],
    "objectives": [
      "Perform miRNA differential expression analysis",
      "Understand the quasi-mapping-based Salmon method for quantifying the expression of transcripts using RNA-Seq data",
      "Idenfity potential miRNAs involved in brassinosteroid-mediated regulation networks"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged miRNA\" href=\"/training-material/search?miRNA\" style=\"background-color: hsl(213, 80.0%, 85%); color: #333; border: 1px solid hsl(213, 80.0%, 45%)\">miRNA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?plants\" style=\"background-color: hsl(121, 80.0%, 85%); color: #333; border: 1px solid hsl(121, 80.0%, 45%)\">plants</a>"
    ],
    "contributors": "Crist√≥bal Gallardo, Pavankumar Videm, Beatriz Serrano-Solano",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html"
  },
  "/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html": {
    "topic": "Ecology",
    "title": "Compute and analyze biodiversity metrics with PAMPA toolsuite",
    "description": null,
    "question": [
      "How to evaluate properly populations and communities biological state with abundance data?",
      "How does trawl exploited populations of Baltic sea, Southern Atlantic and Scotland are doing over time?",
      "How to compute and analyze biodiversity metrics from abundance data?"
    ],
    "objectives": [
      "Upload data from DATRAS portal of ICES",
      "Pre-process population data with Galaxy",
      "Learning how to use an ecological analysis workflow from raw data to graphical representations",
      "Learning how to construct a Generalized Linear (Mixed) Model from a usual ecological question",
      "Learning how to interpret a Generalized Linear (Mixed) Model"
    ],
    "tags": [

    ],
    "contributors": "Coline Royaux, Yvan Le Bras",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html": {
    "topic": "Ecology",
    "title": "RAD-Seq de-novo data analysis",
    "description": null,
    "question": [
      "How to analyze RAD sequencing data without a reference genome for a population genomics study?"
    ],
    "objectives": [
      "Analysis of RAD sequencing data without a reference genome",
      "SNP calling from RAD sequencing data",
      "Calculate population genomics statistics from RAD sequencing data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?RAD-seq\" style=\"background-color: hsl(74, 80.0%, 85%); color: #333; border: 1px solid hsl(74, 80.0%, 45%)\">RAD-seq</a>"
    ],
    "contributors": "Yvan Le Bras",
    "level": null,
    "time_estimation": "8h",
    "url": "/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html": {
    "topic": "Ecology",
    "title": "RAD-Seq Reference-based data analysis",
    "description": null,
    "question": [
      "How to analyze RAD sequencing data using a reference genome for a population genomics study?"
    ],
    "objectives": [
      "Analysis of RAD sequencing data using a reference genome",
      "SNP calling from RAD sequencing data",
      "Calculate population genomics statistics from RAD sequencing data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?RAD-seq\" style=\"background-color: hsl(74, 80.0%, 85%); color: #333; border: 1px solid hsl(74, 80.0%, 45%)\">RAD-seq</a>"
    ],
    "contributors": "Yvan Le Bras",
    "level": null,
    "time_estimation": "8h",
    "url": "/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html": {
    "topic": "Ecology",
    "title": "RAD-Seq to construct genetic maps",
    "description": null,
    "question": [
      "How to analyze RAD sequencing data for a genetic map study?"
    ],
    "objectives": [
      "SNP calling from RAD sequencing data",
      "Find and correct haplotypes",
      "Create input files for genetic map building software"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?RAD-seq\" style=\"background-color: hsl(74, 80.0%, 85%); color: #333; border: 1px solid hsl(74, 80.0%, 45%)\">RAD-seq</a>"
    ],
    "contributors": "Yvan Le Bras",
    "level": null,
    "time_estimation": "8h",
    "url": "/training-material/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/regionalGAM/tutorial.html": {
    "topic": "Ecology",
    "title": "Regional GAM",
    "description": null,
    "question": [
      "What are abundance and trends of a butterfly species?"
    ],
    "objectives": [
      "Obtain and filter/manipulate occurrence data",
      "Compute and visualize phenology of a species through the years",
      "Compute temporal abundance trends"
    ],
    "tags": [

    ],
    "contributors": "Clara Urfer, Elisa Michon, Yvan Le Bras, B√©r√©nice Batut, Simon Benateau",
    "level": null,
    "time_estimation": "2h30m",
    "url": "/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html"
  },
  "/topics/ecology/tutorials/species-distribution-modeling/tutorial.html": {
    "topic": "Ecology",
    "title": "Species distribution modeling",
    "description": null,
    "question": [
      "How to build & evaluate a niche model, visualize and project it ?"
    ],
    "objectives": [
      "Find and download occurrences data from GBIF",
      "Find and download environmental data",
      "Process both occurrences and environmental",
      "Partition occurrence data",
      "Model a theoretical ecological niche and predict species distribution in a future climate scenario by using SDM"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Elisa Michon, Yvan Le Bras, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Analysis of molecular dynamics simulations",
    "description": null,
    "question": [
      "Which analysis tools are available?"
    ],
    "objectives": [
      "Learn which analysis tools are available.",
      "Analyse a protein and discuss the meaning behind each analysis."
    ],
    "tags": [

    ],
    "contributors": "Christopher Barnett, Tharindu Senapathi, Simon Bray",
    "level": "Intermediate",
    "time_estimation": "1H",
    "url": "/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "High Throughput Molecular Dynamics and Analysis",
    "description": null,
    "question": [
      "How are protein-ligand systems parameterized for molecular dynamics simulation?",
      "What kind of analysis can be carried out on molecular trajectories?",
      "How can high-throughput MD be used to study multiple ligands?"
    ],
    "objectives": [
      "Learn about force-fields and MD parameterization",
      "Learn how to conduct MD simulation and analysis for a protein-ligand system",
      "Understand how different molecular interactions contribute to the binding affinity of various ligands for the Hsp90 protein."
    ],
    "tags": [

    ],
    "contributors": "Simon Bray, Tharindu Senapathi, Christopher Barnett, Bj√∂rn Gr√ºning",
    "level": "Advanced",
    "time_estimation": "3H",
    "url": "/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Protein-ligand docking",
    "description": null,
    "question": [
      "What is cheminformatics?",
      "What is protein-ligand docking?",
      "How can I perform a simple docking workflow in Galaxy?"
    ],
    "objectives": [
      "Create a small compound library using the ChEMBL database",
      "Dock a variety of ligands to the active site of the Hsp90 protein"
    ],
    "tags": [

    ],
    "contributors": "Simon Bray",
    "level": "Intermediate",
    "time_estimation": "3H",
    "url": "/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using GROMACS",
    "description": null,
    "question": [
      "How do I use the GROMACS engine in Galaxy?",
      "What is the correct procedure for performing a simple molecular dynamics simulation of a protein?"
    ],
    "objectives": [
      "Learn about the GROMACS engine provided via Galaxy.",
      "Understand the structure of the molecular dynamics workflow.",
      "Prepare a short (1 ns) trajectory for a simulation of a protein."
    ],
    "tags": [

    ],
    "contributors": "Simon Bray",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using NAMD",
    "description": null,
    "question": [
      "How do I use the NAMD engine in Galaxy?",
      "What is the correct procedure for performing a simple molecular dynamics simulation of a protein?"
    ],
    "objectives": [
      "Learn about the NAMD engine provided in BRIDGE"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Christopher Barnett, Tharindu Senapathi, Simon Bray",
    "level": "Intermediate",
    "time_estimation": "3H",
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Setting up molecular systems",
    "description": null,
    "question": [
      "How to get started modelling a protein and a ligand?"
    ],
    "objectives": [
      "learn about the Protein Data Bank",
      "learn how to set up up a model protein and ligand system (with CHARMM-GUI)",
      "learn how to upload the system to Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Christopher Barnett, Simon Bray",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html": {
    "topic": "Computational chemistry",
    "title": "Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring",
    "description": null,
    "question": [
      "How can candidate ligands be generated and docked to a protein in Galaxy?",
      "How can the poses of the docked ligands be evaluated?",
      "How can a workflow for drug virtual screening be constructed in Galaxy?"
    ],
    "objectives": [
      "Understand how Galaxy was used to perform docking and pose scoring on the SARS-CoV-2 main protease (MPro).",
      "Replicate the study on a (very) small scale",
      "Gain familiarity with the docking and scoring techniques involved."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Simon Bray",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Automating Galaxy workflows using the command line",
    "description": null,
    "question": [
      "How can I schedule and run tens or hundreds of Galaxy workflows easily?",
      "How can I automate my analyses when large amounts of data are being produced daily?"
    ],
    "objectives": [
      "Learn to use the `planemo run` subcommand to run workflows from the command line.",
      "Be able to write simple shell scripts for running multiple workflows concurrently or sequentially.",
      "Learn how to use Pangolin to assign annotated variants to lineages."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?workflows\" style=\"background-color: hsl(41, 80.0%, 85%); color: #333; border: 1px solid hsl(41, 80.0%, 45%)\">workflows</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged variant-analysis\" href=\"/training-material/search?variant-analysis\" style=\"background-color: hsl(159, 80.0%, 85%); color: #333; border: 1px solid hsl(159, 80.0%, 45%)\">variant-analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Simon Bray, Wolfgang Maier",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Creating, Editing and Importing Galaxy Workflows",
    "description": null,
    "question": [
      "How can you construct Galaxy workflows from scratch?",
      "How can you label outputs?",
      "How can you include workflows in workflows?",
      "How can you tag workflows?",
      "How can you manage tool versions?",
      "How can you manage workflow versions?"
    ],
    "objectives": [
      "Understand key aspects of workflows",
      "Create clean, non-repetitive workflows"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?workflows\" style=\"background-color: hsl(41, 80.0%, 85%); color: #333; border: 1px solid hsl(41, 80.0%, 45%)\">workflows</a>"
    ],
    "contributors": "Marius van den Beek",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Downloading and Deleting Data in Galaxy",
    "description": null,
    "question": [
      "How can I efficiently download my data from Galaxy once I've completed my analyses?",
      "How do I delete old data to make more room in my quota?"
    ],
    "objectives": [
      "Have a greater knowledge of how Galaxy handles data downloads and deletions",
      "Be able to successfully download and delete their own data"
    ],
    "tags": [

    ],
    "contributors": "Simon Gladman, Helena Rasche",
    "level": "Introductory",
    "time_estimation": "20M",
    "url": "/training-material/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Extracting Workflows from Histories",
    "description": null,
    "question": [
      "What is a workflow?",
      "How can I create a workflow based on my analysis history?"
    ],
    "objectives": [
      "Learn how to extract a workflow from a Galaxy history",
      "Learn how to change a workflow using the workflow editor"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?workflows\" style=\"background-color: hsl(41, 80.0%, 85%); color: #333; border: 1px solid hsl(41, 80.0%, 45%)\">workflows</a>"
    ],
    "contributors": "Dave Clements",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Galaxy workflows in Dockstore",
    "description": null,
    "question": [

    ],
    "objectives": [
      "What is Dockstore?",
      "Why use Dockstore?",
      "How to find workflows on Dockstore?",
      "How to launch workflows from Dockstore?",
      "Getting started with Galaxy workflows on Dockstore",
      "Register your Galaxy workflow on Dockstore"
    ],
    "tags": [

    ],
    "contributors": "Avani Khadilkar, Elizabeth Sheets, Charles Overbeck, Enis Afgan",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html"
  },
  "/topics/galaxy-interface/tutorials/get-data/slides.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Getting data into Galaxy",
    "description": null,
    "question": [
      "How do I get my data into Galaxy?",
      "How do I get public data into Galaxy?"
    ],
    "objectives": null,
    "tags": [

    ],
    "contributors": "Anne Pajon, Dave Clements, Maria Doyle, Saskia Hiltemann",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/galaxy-interface/tutorials/get-data/slides.html"
  },
  "/topics/galaxy-interface/tutorials/group-tags/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Group tags for complex experimental designs",
    "description": null,
    "question": [
      "What are group tags?",
      "How can I use group tags to perform multi-factor analyses with collections"
    ],
    "objectives": [
      "Learn how to set group tags",
      "Learn how to select group tags in tools"
    ],
    "tags": [

    ],
    "contributors": "Marius van den Beek",
    "level": null,
    "time_estimation": "10m",
    "url": "/training-material/topics/galaxy-interface/tutorials/group-tags/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/intermine/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "InterMine integration with Galaxy",
    "description": null,
    "question": [
      "How to export your query results from your InterMine of choice to Galaxy?",
      "How to export a list of identifiers from Galaxy to your InterMine of choice?"
    ],
    "objectives": [
      "Learn how to import/export data from/to InterMine instances",
      "Understand the InterMine Interchange Dataset"
    ],
    "tags": [

    ],
    "contributors": "Daniela Butano, Yo Yehudi",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/galaxy-interface/tutorials/intermine/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "JupyterLab in Galaxy",
    "description": null,
    "question": [
      "How can I manipulate data using JupyterLab in Galaxy?",
      "How can I start a notebook in JupyterLab?",
      "How can I import/export dataset from/to my history to/from the notebook?",
      "How can I save my notebook to my history?"
    ],
    "objectives": [
      "Launch JupyterLab in Galaxy",
      "Start a notebook",
      "Import libraries",
      "Use get() to import datasets from your history to the notebook",
      "Use put() to export datasets from the notebook to your history",
      "Save your notebook into your history"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Anne Fouilloux",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Multisample Analysis",
    "description": null,
    "question": null,
    "objectives": null,
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?collections\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">collections</a>"
    ],
    "contributors": "Anton Nekrutenko, Anne Pajon",
    "level": "Advanced",
    "time_estimation": "1h",
    "url": "/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/name-tags/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Name tags for following complex histories",
    "description": null,
    "question": [
      "What are name tags or hash tags?",
      "How can I use them to keep track of my data?"
    ],
    "objectives": [
      "Learn how to set name tags",
      "Learn how they are propagated"
    ],
    "tags": [

    ],
    "contributors": "Helena Rasche",
    "level": null,
    "time_estimation": "20m",
    "url": "/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/rstudio/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "RStudio in Galaxy",
    "description": null,
    "question": [
      "How can I manipulate data using R in Galaxy?"
    ],
    "objectives": [
      "Launch RStudio in Galaxy"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "B√©r√©nice Batut, Fotis E. Psomopoulos, Toby Hodges, Erasmus+ Programme",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader",
    "description": null,
    "question": [
      "How to use the rule based uploader to create complex collections"
    ],
    "objectives": [
      "Learn about the Rule Based Uploader"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?collections\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?tags\" style=\"background-color: hsl(353, 80.0%, 85%); color: #333; border: 1px solid hsl(353, 80.0%, 45%)\">tags</a>"
    ],
    "contributors": "John Chilton, Helena Rasche",
    "level": "Intermediate",
    "time_estimation": "20m",
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader: Advanced",
    "description": null,
    "question": [
      "How to use the rule based uploader to create complex collections"
    ],
    "objectives": [
      "Learn even more about the Rule Based Uploader"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?collections\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?tags\" style=\"background-color: hsl(353, 80.0%, 85%); color: #333; border: 1px solid hsl(353, 80.0%, 45%)\">tags</a>"
    ],
    "contributors": "John Chilton, Helena Rasche",
    "level": "Advanced",
    "time_estimation": "20m",
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/search/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Searching Your History",
    "description": null,
    "question": [
      "How can you find your datasets?",
      "How to search by file type?",
      "What filters are available to you?"
    ],
    "objectives": [
      "Feel confident in searching your history"
    ],
    "tags": [

    ],
    "contributors": "Helena Rasche, Anton Nekrutenko",
    "level": "Introductory",
    "time_estimation": "5m",
    "url": "/training-material/topics/galaxy-interface/tutorials/search/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis",
    "description": null,
    "question": [
      "How can I search SRA SARS-CoV-2 metadata from within Galaxy?",
      "How can I import SRA aligned read files and extract the data in my format of choice?",
      "How can I import vcf files into Galaxy that have been generated for these Runs?"
    ],
    "objectives": [
      "Learn about SRA aligned read format and vcf files for Runs containing SARS-CoV-2 content",
      "Understand how to search the metadata for these Runs to find your dataset of interest and then import that data in your preferred format"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ncbi\" href=\"/training-material/search?ncbi\" style=\"background-color: hsl(351, 80.0%, 85%); color: #333; border: 1px solid hsl(351, 80.0%, 45%)\">ncbi</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Jon Trow, Adelaide Rhodes",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Submitting raw sequencing reads to ENA",
    "description": null,
    "question": [
      "How do you submit raw viral sequence reads to the European Nucleotide Archive?"
    ],
    "objectives": [
      "Add ENA Webin credentials to your Galaxy user information",
      "Use Galaxy's 'ENA upload tool' to interactively generate metadata",
      "Use a metadata template to upload bulk metadata",
      "Submit raw sequencing reads and metadata to ENA's test server"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Miguel Roncoroni",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Understanding Galaxy history system",
    "description": null,
    "question": [
      "How do Galaxy histories work?"
    ],
    "objectives": [
      "Gain understanding on navigating and manipulating histories"
    ],
    "tags": [

    ],
    "contributors": "Anton Nekrutenko, Martin ƒåech, Helena Rasche",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/history/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Use Jupyter notebooks in Galaxy",
    "description": null,
    "question": [
      "How to use a Jupyter Notebook in Galaxy"
    ],
    "objectives": [
      "Learn about the Jupyter Interactive Environment"
    ],
    "tags": [

    ],
    "contributors": "Delphine Lariviere",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/collections/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using dataset collections",
    "description": null,
    "question": [
      "How to manipulate large numbers of datasets at once?"
    ],
    "objectives": [
      "Understand and master dataset collections"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?collections\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">collections</a>"
    ],
    "contributors": "Anton Nekrutenko",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html": {
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using Workflow Parameters",
    "description": null,
    "question": [
      "What are Workflow Parameters",
      "How can I define and use Workflow Parameters",
      "How can I read Parameters from Datasets"
    ],
    "objectives": [
      "Learn how to use Workflow Parameters to improve your Workflows"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?workflows\" style=\"background-color: hsl(41, 80.0%, 85%); color: #333; border: 1px solid hsl(41, 80.0%, 45%)\">workflows</a>"
    ],
    "contributors": "Marius van den Beek, Helena Rasche",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html"
  },
  "/topics/dev/slides/introduction.html": {
    "topic": "Development in Galaxy",
    "title": "Galaxy from a developer point of view",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/dev/slides/introduction.html"
  },
  "/topics/dev/tutorials/core-contributing/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Contributing a New Feature to Galaxy Core",
    "description": null,
    "question": [
      "How can I develop extensions to Galaxy data model?",
      "How can I implement new API functionality within Galaxy?",
      "How can I extend the Galaxy user interface with VueJS components?"
    ],
    "objectives": null,
    "tags": [

    ],
    "contributors": "John Chilton",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/dev/tutorials/core-contributing/tutorial.html"
  },
  "/topics/dev/tutorials/bioblend-dev/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Contributing to BioBlend as a developer",
    "description": null,
    "question": [
      "How to get started making contributions to BioBlend?"
    ],
    "objectives": [
      "Learn the basics behind BioBlend and the Galaxy API from a developer perspective.",
      "Learn how to implement a simple method in BioBlend.",
      "Learn how to run the BioBlend test suite."
    ],
    "tags": [

    ],
    "contributors": "Erik Schill, Simon Bray",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/dev/tutorials/bioblend-dev/tutorial.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Creating Galaxy tools from Conda Through Deployment",
    "description": null,
    "question": [
      "How do you create a conda package used as a dependency in a Galaxy tool wrapper?",
      "How do you create a Galaxy tool wrapper and all of its requirements?",
      "How do you test a Galaxy tool wrapper after it's written?",
      "How do you deploy a Galaxy tool wrapper?",
      "How to do you make a Galaxy tool wrapper publicly available?"
    ],
    "objectives": [
      "Learn how to make and deploy a conda package",
      "Understand how to write a Galaxy tool wrapper",
      "Be able to use a tool wrapper locally",
      "Be able to test the tool wrapper",
      "Learn how to add tools to the Galaxy toolshed and make them available to any user"
    ],
    "tags": [

    ],
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html"
  },
  "/topics/dev/tutorials/data-source-integration/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Data source integration",
    "description": null,
    "question": [
      "How can I write a tool that can import data into Galaxy from an external database?",
      "What are \"data sources\" and how do they function?",
      "Is there any ready-to-use example?"
    ],
    "objectives": null,
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Gianmauro Cuccuru, Helena Rasche",
    "level": null,
    "time_estimation": "10M",
    "url": "/training-material/topics/dev/tutorials/data-source-integration/tutorial.html"
  },
  "/topics/dev/tutorials/debugging/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Debugging Galaxy",
    "description": null,
    "question": [
      "How do I debug Galaxy?"
    ],
    "objectives": [
      "Fix a broken branch",
      "Interpret the results of failed tests on GitHub",
      "Run individual tests locally",
      "Fix errors identified by failing tests",
      "Debug simple runtime errors using the Python debugger",
      "Write tests exposing the identified bug"
    ],
    "tags": [

    ],
    "contributors": "Assunta DeSanto, Sergey Golitsynskiy, John Chilton",
    "level": null,
    "time_estimation": "4h",
    "url": "/training-material/topics/dev/tutorials/debugging/tutorial.html"
  },
  "/topics/dev/tutorials/architecture/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Galaxy Code Architecture",
    "description": null,
    "question": [
      "How is the Galaxy code structured?",
      "What do the various other projects related to Galaxy do?",
      "What happens when I start Galaxy?"
    ],
    "objectives": [
      "Explore various aspects of the Galaxy codebase.",
      "Understand the various top-level files and modules in Galaxy.",
      "Understand how dependencies work in Galaxy's frontend and backend."
    ],
    "tags": [

    ],
    "contributors": "John Chilton, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "150m",
    "url": "/training-material/topics/dev/tutorials/architecture/slides.html"
  },
  "/topics/dev/tutorials/interactive-environments/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Environments",
    "description": null,
    "question": [
      "What are Galaxy Interactive Environments (GIEs)?",
      "How to enable GIEs in Galaxy?",
      "How to develop your own GIE?"
    ],
    "objectives": [
      "Implement a Hello-World Galaxy Interactive Environment"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Saskia Hiltemann, Bj√∂rn Gr√ºning, Helena Rasche",
    "level": null,
    "time_estimation": "90m",
    "url": "/training-material/topics/dev/tutorials/interactive-environments/slides.html"
  },
  "/topics/dev/tutorials/interactive-tours/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Tours",
    "description": null,
    "question": [
      "What is a Galaxy Interactive Tour?",
      "How to create a Galaxy Interactive Tour?"
    ],
    "objectives": [
      "Discover what is a Galaxy Interactive Tour",
      "Be able to create a Galaxy Interactive Tour",
      "Be able to add a Galaxy Interactive Tour in a Galaxy instance"
    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/dev/tutorials/interactive-tours/slides.html"
  },
  "/topics/dev/tutorials/webhooks/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Galaxy Webhooks",
    "description": null,
    "question": [
      "What are Galaxy Webhooks?",
      "How to create them?"
    ],
    "objectives": [
      "Discover what Galaxy Webhooks are",
      "Be able to create Webhooks",
      "Be able to add a Galaxy Webhook in a Galaxy instance"
    ],
    "tags": [

    ],
    "contributors": "Clemens Blank, Martin ƒåech, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/dev/tutorials/webhooks/tutorial.html"
  },
  "/topics/dev/tutorials/visualization-generic/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "Generic plugins",
    "description": null,
    "question": [
      "How can visualization plugins benefit science?"
    ],
    "objectives": [
      "Implement a first Galaxy visualization",
      "Understand the client side vs. server side principle"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, Youri Hoogstrate",
    "level": null,
    "time_estimation": "90m",
    "url": "/training-material/topics/dev/tutorials/visualization-generic/tutorial.html"
  },
  "/topics/dev/tutorials/visualization-charts/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "JavaScript plugins",
    "description": null,
    "question": [
      "How can I make a custom visualization plugin for Galaxy?"
    ],
    "objectives": [
      "Learn how to add JavaScript plugins to Galaxy using the Charts visualization framework"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, Youri Hoogstrate, Bj√∂rn Gr√ºning, Aysam Guerler, Dannon Baker",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/dev/tutorials/visualization-charts/tutorial.html"
  },
  "/topics/dev/tutorials/conda_sys/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Prerequisites for building software/conda packages",
    "description": null,
    "question": [
      "What does 'installing a software' means on a Linux architecture?",
      "Why my compilations always fail?",
      "How to solve common compiling and installation issues?"
    ],
    "objectives": [
      "Learn how to compile and install tools using standard procedures.",
      "Learn needed tricks to write conda build.sh files."
    ],
    "tags": [

    ],
    "contributors": "Anthony Bretaudeau, Cyril Monjeaud",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/dev/tutorials/conda_sys/slides.html"
  },
  "/topics/dev/tutorials/bioblend-api/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Scripting Galaxy using the API and BioBlend",
    "description": null,
    "question": [
      "What is a REST API?",
      "How to interact with Galaxy programmatically?",
      "Why and when should I use BioBlend?"
    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Olivia Doppelt-Azeroual, Fabien Mareuil, Nicola Soranzo, Dannon Baker",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/dev/tutorials/bioblend-api/slides.html"
  },
  "/topics/dev/tutorials/conda/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Conda",
    "description": null,
    "question": [
      "How can I connect tools to applications and libraries?",
      "What are the advantages of declaring dependencies for my tool?",
      "What are Conda and Bioconda?",
      "What are Conda recipes and environments?",
      "How do I find and use existing Conda recipes?",
      "How do I develop Conda recipes for use within Galaxy tools?"
    ],
    "objectives": [
      "Learn to use existing Conda recipes to enable best practice tool dependency management in Galaxy.",
      "Learn the basics of building Conda recipes and contributing to Bioconda.",
      "Learn to use Planemo to assist in developing Galaxy tools from existing and new Conda recipes."
    ],
    "tags": [

    ],
    "contributors": "Nicola Soranzo, John Chilton, Bj√∂rn Gr√ºning, Herv√© M√©nager",
    "level": null,
    "time_estimation": "75m",
    "url": "/training-material/topics/dev/tutorials/conda/slides.html"
  },
  "/topics/dev/tutorials/containers/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Containers",
    "description": null,
    "question": [
      "What are the advantages of running my Galaxy tool inside of a container?",
      "How does Galaxy find a container to run my tool in?",
      "What are BioContainers and how are they related to Galaxy?"
    ],
    "objectives": [
      "Explore the differences between containerizing Galaxy and tool execution.",
      "Discuss the advantages of containerizing tools.",
      "Learn to build best practice tools ready to be containerized."
    ],
    "tags": [

    ],
    "contributors": "John Chilton, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "45m",
    "url": "/training-material/topics/dev/tutorials/containers/slides.html"
  },
  "/topics/dev/tutorials/tool-integration/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Tool development and integration into Galaxy",
    "description": null,
    "question": [
      "What is a tool for Galaxy?",
      "How to write a best-practice tool?",
      "How to deal with the tool environment?"
    ],
    "objectives": [
      "Learn what a tool is and its structure",
      "Use the Planemo utilities to develop a tool",
      "Deal with the dependencies",
      "Write functional tests",
      "Make a tool ready for publishing in a ToolShed"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Anthony Bretaudeau, John Chilton, Nicola Soranzo, Bj√∂rn Gr√ºning, Gildas Le Corguill√©",
    "level": null,
    "time_estimation": "6h",
    "url": "/training-material/topics/dev/tutorials/tool-integration/slides.html"
  },
  "/topics/dev/tutorials/toolshed/slides.html": {
    "topic": "Development in Galaxy",
    "title": "Tool Shed: sharing Galaxy tools",
    "description": null,
    "question": [
      "What is a Tool Shed?",
      "How to install tools and workflows from a Tool Shed into a Galaxy instance?",
      "What are the Tool Shed repository types?",
      "How to publish with Planemo?"
    ],
    "objectives": [
      "Discover what is a Tool Shed",
      "Be able to install tools and workflows from a Tool Shed into a Galaxy instance",
      "Be able to publish tools with Planemo"
    ],
    "tags": [

    ],
    "contributors": "Anthony Bretaudeau, Loraine Gu√©guen, St√©phanie Legras, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/dev/tutorials/toolshed/slides.html"
  },
  "/topics/dev/tutorials/tool-generators-advanced/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From More Complex Scripts",
    "description": null,
    "question": [
      "What else can I do with the ToolFactory?"
    ],
    "objectives": [
      "Further develop your ToolFactory Skills"
    ],
    "tags": [

    ],
    "contributors": "Ross Lazarus",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/dev/tutorials/tool-generators-advanced/tutorial.html"
  },
  "/topics/dev/tutorials/tool-generators/tutorial.html": {
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From Simple Scripts",
    "description": null,
    "question": [
      "What options exist for new-to-Galaxy developers to convert functioning command line scripts into Galaxy tools?",
      "Can any command line script I've written be wrapped as a Galaxy tool?",
      "Can I make a tool from a script developed in a Galaxy Interactive Environment notebook?",
      "How can I get the ToolFactory working locally, since you tell me it should never be exposed on a public server?",
      "What is the difference between a hand-crafted tool and a ToolFactory generated one?"
    ],
    "objectives": [
      "Learn why you might want to use the ToolFactory Appliance",
      "Watch a video demonstration and explore the generated code - Hello Galaxy Training Network!",
      "Run it locally in Docker.",
      "Install and explore the simple examples provided.",
      "Modify and re-generate them to see how the changes affect the tool",
      "Generate new simple Galaxy tools using your own scripts"
    ],
    "tags": [

    ],
    "contributors": "Ross Lazarus, Helena Rasche",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/dev/tutorials/tool-generators/tutorial.html"
  },
  "/topics/genome-editing/tutorials/crispr-screen/tutorial.html": {
    "topic": "Genome Editing",
    "title": "CRISPR screen analysis",
    "description": null,
    "question": [
      "What are the steps to process CRISPR screen data?",
      "How to identify differentially enriched guides across multiple experimental conditions?"
    ],
    "objectives": [
      "Check quality of raw reads",
      "Trim sequencing adapters",
      "Count guide sequences in samples",
      "Evaluate the quality of count results",
      "Test for differential enrichment of guides across conditions"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Maria Doyle, Kenji Fujihara, Twishi Gulati",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/genome-editing/tutorials/crispr-screen/tutorial.html"
  },
  "/topics/metabolomics/slides/introduction.html": {
    "topic": "Metabolomics",
    "title": "Introduction to Metabolomics",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Gildas Le Corguill√©, C√©cile Canlet",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/metabolomics/slides/introduction.html"
  },
  "/topics/metabolomics/tutorials/gcms/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry : GC-MS analysis with metaMS package",
    "description": null,
    "question": [
      "What are the main steps of GC-MS datas processing for metabolomic analysis ?",
      "How te be able to annotate the maximum of unknowns using Galaxy ?"
    ],
    "objectives": [
      "To be sure you have already comprehend the diversity of MS pre-processing analysis.",
      "To learn the principal functions of metaMS package through Galaxy.",
      "To evaluate the potential of this new GC-MS workflow for GC-MS metabolomic analysis."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Julien Saint-Vanne",
    "level": "Introductory",
    "time_estimation": "2H",
    "url": "/training-material/topics/metabolomics/tutorials/gcms/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Examining the spatial distribution of analytes",
    "description": null,
    "question": [
      "What is the typical mass range of VOCs?",
      "In which compartments of the chilli are the measured VOCs located?",
      "In which compartment of the chilli is the capsaicin located?"
    ],
    "objectives": [
      "Plot average mass spectra and overlaid mass spectra of single spectra.",
      "Filter MSI data for a specific m/z range.",
      "Automatic generation of many m/z images.",
      "Overlay the distribution of several m/z in one image."
    ],
    "tags": [

    ],
    "contributors": "Melanie F√∂ll, Maren Stillger",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Finding differential analytes",
    "description": null,
    "question": [
      "Can N-linked glycans from FFPE tissues be detected by MALDI imaging?",
      "Can potential N-linked glycans be identified by an additional LC-MS/MS experiment?",
      "Do specific kidney compartments have different N-linked glycan compositions?"
    ],
    "objectives": [
      "Combining MSI datasets while using the information about each subfile in further steps.",
      "Preprocessing raw MSI data.",
      "Performing supervised and unsupervised statistical analysis."
    ],
    "tags": [

    ],
    "contributors": "Melanie F√∂ll",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS analysis",
    "description": null,
    "question": [
      "What are the main steps of untargeted LC-MS data processing for metabolomic analysis?",
      "How to conduct metabolomic data analysis from preprocessing to annotation using Galaxy?"
    ],
    "objectives": [
      "To comprehend the diversity of LC-MS metabolomic data analysis.",
      "To get familiar with the main steps constituting a metabolomic workflow for untargeted LC-MS analysis.",
      "To evaluate the potential of a workflow approach when dealing with LC-MS metabolomic data."
    ],
    "tags": [

    ],
    "contributors": "M√©lanie Petera, Gildas Le Corguill√©, Jean-Fran√ßois Martin, Yann Guitton, Workflow4Metabolomics core team",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/metabolomics/tutorials/lcms/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS data processing",
    "description": null,
    "question": [
      "Why should one consider performing data processing steps when dealing with Metabolomics data?",
      "How to conduct LCMS-based untargeted metabolomic data processing using Galaxy?"
    ],
    "objectives": [
      "To comprehend the key role of data processing",
      "To comprehend the diversity of steps necessary to perform untargeted LC-MS metabolomic data processing.",
      "To be familiar with the \"identify then perform\" approach necessary to deploy relevant data processing strategies."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "M√©lanie Petera, Workflow4Metabolomics core team",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html": {
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS preprocessing with XCMS",
    "description": null,
    "question": [
      "What are the main steps of untargeted LC-MS data preprocessing for metabolomic analyses?",
      "How to conduct metabolomic data preprocessing using Galaxy?"
    ],
    "objectives": [
      "To comprehend the diversity of steps necessary to perform untargeted LC-MS metabolomic preprocessing.",
      "To get familiar with the way to use XCMS-based Galaxy modules dedicated to LC-MS data preprocessing.",
      "To evaluate the potential of a workflow approach when dealing with LC-MS preprocessing."
    ],
    "tags": [

    ],
    "contributors": "M√©lanie Petera, Jean-Fran√ßois Martin, Gildas Le Corguill√©, Workflow4Metabolomics core team",
    "level": "Intermediate",
    "time_estimation": "3h",
    "url": "/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html"
  },
  "/topics/climate/slides/introduction.html": {
    "topic": "Climate",
    "title": "Introduction to climate data",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Anne Fouilloux, Jean Iaquinta",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/climate/slides/introduction.html"
  },
  "/topics/climate/tutorials/fates/tutorial.html": {
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES)",
    "description": null,
    "question": [
      "How to run CLM-FATES with the CLM-FATES Galaxy tool?",
      "How to upload input data for running CLM-FATES?",
      "How to customize your runs?",
      "How to analyze your model outputs?",
      "How to create a workflow?",
      "How to share your workflow?"
    ],
    "objectives": [
      "Setting up a CLM-FATES case.",
      "Customizing your run.",
      "Interactive visualization with Panoply.",
      "Automating your analyzes and visualisations of your CLM-FATES case.",
      "Creating multi-case scenarios.",
      "Composing, executing and publishing CML-FATES workflow."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Anne Fouilloux, Hui Tang",
    "level": null,
    "time_estimation": "4H",
    "url": "/training-material/topics/climate/tutorials/fates/tutorial.html"
  },
  "/topics/climate/tutorials/fates-jupyterlab/tutorial.html": {
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab",
    "description": null,
    "question": [
      "Why and when using Galaxy Climate JupyterLab for CLM-FATES?",
      "How to start Galaxy Climate JupyterLab in Galaxy?",
      "How to upload input data for running CLM-FATES?",
      "How to create CLM-FATES case in Galaxy Climate JupyterLab?",
      "How to customize your run?",
      "How to analyze your model outputs?",
      "How to save your model results into a Galaxy history?",
      "How to share your results?"
    ],
    "objectives": [
      "Motivation for using the Galaxy Climate JupyterLab for CLM-FATES.",
      "Setting up CLM-FATES case with Galaxy Climate JupyterLab.",
      "Running CLM-FATES in Galaxy for single-point locations where in-situ measurements are available.",
      "Analyzing CLM-FATES results.",
      "Sharing CLM-FATES simulations.",
      "Composing, executing and publishing the corresponding Jupyter notebooks."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Anne Fouilloux",
    "level": null,
    "time_estimation": "6H",
    "url": "/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html"
  },
  "/topics/climate/tutorials/climate-101/tutorial.html": {
    "topic": "Climate",
    "title": "Getting your hands-on climate data",
    "description": null,
    "question": [
      "What is climate?",
      "What type of data is available?"
    ],
    "objectives": [
      "Learn about the terminology",
      "Learn about the different source of climate data",
      "Learn about climate observations, reanalysis, climate predictions and climate projections"
    ],
    "tags": [

    ],
    "contributors": "Anne Fouilloux",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/climate/tutorials/climate-101/tutorial.html"
  },
  "/topics/climate/tutorials/panoply/tutorial.html": {
    "topic": "Climate",
    "title": "Visualize Climate data with Panoply netCDF viewer",
    "description": null,
    "question": [
      "How to start Panoply interactive environment in Galaxy?",
      "How to inspect netCDF data?",
      "How to make a plot with Panoply?",
      "Where to save your plots in Panoply?",
      "How to customize plots in Panoply?",
      "How to generate animation in Panoply?"
    ],
    "objectives": [
      "Learn to use Panoply in Galaxy to visuallize netCDF geo-referenced data",
      "Learn how Panoply plots are exported to Galaxy"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Anne Fouilloux",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/climate/tutorials/panoply/tutorial.html"
  },
  "/topics/admin/slides/introduction.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy from an administrator's point of view",
    "description": null,
    "question": [
      "What options to deploy Galaxy do I have?",
      "Which platforms are supported by Galaxy?",
      "What requirements does Galaxy have?"
    ],
    "objectives": [
      "Learn about different options about Galaxy deployment.",
      "Make an educated decision about your preferred deployment model."
    ],
    "tags": [

    ],
    "contributors": "Valentin Marcon, Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "level": null,
    "time_estimation": "20m",
    "url": "/training-material/topics/admin/slides/introduction.html"
  },
  "/topics/admin/tutorials/advanced-galaxy-customisation/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Advanced customisation of a Galaxy instance",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Bj√∂rn Gr√ºning, Helena Rasche, Martin ƒåech",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/advanced-galaxy-customisation/slides.html"
  },
  "/topics/admin/tutorials/ansible/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Ansible",
    "description": null,
    "question": [
      "Why Ansible?",
      "How and when to use Ansible?",
      "How to write a role?",
      "How to leverage community build roles?"
    ],
    "objectives": [
      "Learn Ansible basics",
      "Write a simple role",
      "Install a role from Ansible Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "level": null,
    "time_estimation": "60m",
    "url": "/training-material/topics/admin/tutorials/ansible/tutorial.html"
  },
  "/topics/admin/tutorials/jenkins/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Automation with Jenkins",
    "description": null,
    "question": [
      "What sort of tasks should be automated?",
      "What are my options for automation?",
      "How can I automate repetitive tasks?"
    ],
    "objectives": [
      "Setup Jenkins",
      "Setup a simple job",
      "Automate running of the Galaxy playbook",
      "Secure Jenkins"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged automation\" href=\"/training-material/search?automation\" style=\"background-color: hsl(278, 80.0%, 85%); color: #333; border: 1px solid hsl(278, 80.0%, 45%)\">automation</a>"
    ],
    "contributors": "Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/jenkins/tutorial.html"
  },
  "/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Connecting Galaxy to a compute cluster",
    "description": null,
    "question": [
      "How to connect Galaxy to a compute cluster?",
      "What are job metrics?",
      "What sort of information can I collect?",
      "Where can I find this information?"
    ],
    "objectives": [
      "Be familiar with the basics of installing, configuring, and using Slurm",
      "Understand all components of the Galaxy job running stack",
      "Understand how the `job_conf.xml` file controls Galaxy's jobs subsystem",
      "Have a strong understanding of Galaxy job destinations",
      "Understand the purpose and function of Galaxy job metrics"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?jobs\" style=\"background-color: hsl(134, 80.0%, 85%); color: #333; border: 1px solid hsl(134, 80.0%, 45%)\">jobs</a>"
    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html"
  },
  "/topics/admin/tutorials/systemd-supervisor/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Controlling Galaxy with systemd or Supervisor",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Helena Rasche",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/systemd-supervisor/slides.html"
  },
  "/topics/admin/tutorials/data-library/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Data Libraries",
    "description": null,
    "question": [
      "How do data libraries work?",
      "What are they good for?",
      "How can I use them?",
      "How can I setup permissions for them?"
    ],
    "objectives": [
      "Setup a data library",
      "Manage permissions",
      "Import data from disk"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?storage\" style=\"background-color: hsl(325, 80.0%, 85%); color: #333; border: 1px solid hsl(325, 80.0%, 45%)\">storage</a>"
    ],
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/data-library/tutorial.html"
  },
  "/topics/admin/tutorials/terraform/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Deploying a compute cluster in OpenStack via Terraform",
    "description": null,
    "question": [
      "What is Terraform?",
      "In which situations is it good/bad?",
      "How to use it for managing your VM cluster"
    ],
    "objectives": [
      "Learn Terraform basics",
      "Launch a VM with Terraform",
      "Launch and tear down a cluster with Terraform"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged terraform\" href=\"/training-material/search?terraform\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">terraform</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?deploying\" style=\"background-color: hsl(22, 80.0%, 85%); color: #333; border: 1px solid hsl(22, 80.0%, 45%)\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cloud\" href=\"/training-material/search?cloud\" style=\"background-color: hsl(11, 80.0%, 85%); color: #333; border: 1px solid hsl(11, 80.0%, 45%)\">cloud</a>"
    ],
    "contributors": "Helena Rasche",
    "level": null,
    "time_estimation": "60m",
    "url": "/training-material/topics/admin/tutorials/terraform/tutorial.html"
  },
  "/topics/admin/tutorials/object-store/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Distributed Object Storage",
    "description": null,
    "question": [
      "How does Galaxy locate data?",
      "How can I have Galaxy use multiple storage locations?"
    ],
    "objectives": [
      "Setup Galaxy with both the Hierarachical and Distributed Object Storages"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?storage\" style=\"background-color: hsl(325, 80.0%, 85%); color: #333; border: 1px solid hsl(325, 80.0%, 45%)\">storage</a>"
    ],
    "contributors": "Nate Coraor, Helena Rasche, Gianmauro Cuccuru",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/object-store/tutorial.html"
  },
  "/topics/admin/tutorials/galaxy-docker/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Docker and Galaxy",
    "description": null,
    "question": [
      "Why Docker? What is it?",
      "How to use Docker?",
      "How to integrate Galaxy in Docker to facilitate its deployment?"
    ],
    "objectives": [
      "Docker basics",
      "Galaxy Docker image (usage)",
      "Galaxy Docker (internals)",
      "Galaxy flavours"
    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/galaxy-docker/slides.html"
  },
  "/topics/admin/tutorials/empathy/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Empathy",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/empathy/slides.html"
  },
  "/topics/admin/tutorials/ftp/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Enable upload via FTP",
    "description": null,
    "question": [
      "How can I setup FTP to be easy for my users?",
      "Can I authenticate ftp users with Galaxy credentials?"
    ],
    "objectives": [
      "Configure galaxy and install a FTP server.",
      "Use an Ansible playbook for this."
    ],
    "tags": [

    ],
    "contributors": "Lucille Delisle",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/ftp/tutorial.html"
  },
  "/topics/admin/tutorials/external-auth/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "External Authentication",
    "description": null,
    "question": [
      "How can I connect Galaxy with CAS, SAML, etc."
    ],
    "objectives": [
      "be familiar with configuring Galaxy to use an upstream (proxy) authentication provider",
      "be able to log in to your Galaxy server with a file-configured user."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged authentication\" href=\"/training-material/search?authentication\" style=\"background-color: hsl(291, 80.0%, 85%); color: #333; border: 1px solid hsl(291, 80.0%, 45%)\">authentication</a>"
    ],
    "contributors": "Nate Coraor, Nicola Soranzo, Helena Rasche",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/external-auth/tutorial.html"
  },
  "/topics/admin/tutorials/database/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Galactic Database",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Martin ƒåech, Nicola Soranzo, Nate Coraor, Helena Rasche",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/database/slides.html"
  },
  "/topics/admin/tutorials/database-schema/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Database schema",
    "description": null,
    "question": [
      "Running a production Galaxy server, you some times end up in with a situation, where you manually need to interact with the Galaxy database: how do you do that",
      "How to extract usage information, which can not be gathered using the given report tools",
      "How to move from MySQL to PostgreSQL",
      "Is there ever a need to manually change the contents of a table"
    ],
    "objectives": [
      "Learn some of the design concepts of the Galaxy database",
      "Extract information from the Galaxy database",
      "Get to know SchemaSpy"
    ],
    "tags": [

    ],
    "contributors": "Hans-Rudolf Hotz, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/admin/tutorials/database-schema/tutorial.html"
  },
  "/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation on Kubernetes",
    "description": null,
    "question": [
      "How do I deploy Galaxy on Kubernetes using Helm?",
      "How can I create a simple replica of usegalaxy.org?"
    ],
    "objectives": [
      "Have an understanding of how to use Galaxy's Helm chart",
      "Be able to use Helm to install different flavors of Galaxy for different purposes"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?kubernetes\" style=\"background-color: hsl(354, 80.0%, 85%); color: #333; border: 1px solid hsl(354, 80.0%, 45%)\">kubernetes</a>"
    ],
    "contributors": "Pablo Moreno, Enis Afgan, Nuwan Goonasekera, Alex Mahmoud, Sergey Golitsynskiy",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/ansible-galaxy/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation with Ansible",
    "description": null,
    "question": [
      "How does the Galaxy Ansible module work internally?",
      "How can I install a Galaxy server with Ansible"
    ],
    "objectives": [
      "Have an understanding of how Galaxy's Ansible roles are structured and interact with one another",
      "Be able to use an Ansible playbook to install different flavors of Galaxy for different purposes"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?deploying\" style=\"background-color: hsl(22, 80.0%, 85%); color: #333; border: 1px solid hsl(22, 80.0%, 45%)\">deploying</a>"
    ],
    "contributors": "Helena Rasche, Nate Coraor, Simon Gladman, Saskia Hiltemann, Nicola Soranzo",
    "level": null,
    "time_estimation": "2h30m",
    "url": "/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/interactive-tools/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Interactive Tools",
    "description": null,
    "question": null,
    "objectives": [
      "Understand what Galaxy Interactive Tools are and how they work",
      "Be aware of the security implications of Interactive Tools",
      "Have a basic understanding of the Interactive Tools (GxIT/GIE) Proxy, its purpose, and configuration",
      "Be familiar with wildcard SSL certificates and how to get them from Let's Encrypt",
      "Configure your Galaxy to serve Interactive Tools using an Ansible Playbook",
      "Start, run, and use an Interactive Tool"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?ansible\" style=\"background-color: hsl(331, 80.0%, 85%); color: #333; border: 1px solid hsl(331, 80.0%, 45%)\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?interactive-tools\" style=\"background-color: hsl(29, 80.0%, 85%); color: #333; border: 1px solid hsl(29, 80.0%, 45%)\">interactive-tools</a>"
    ],
    "contributors": "Nate Coraor, Simon Gladman, Helena Rasche, Anthony Bretaudeau",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/admin/tutorials/interactive-tools/tutorial.html"
  },
  "/topics/admin/tutorials/general-monitoring/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/general-monitoring/slides.html"
  },
  "/topics/admin/tutorials/gxadmin/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with gxadmin",
    "description": null,
    "question": [
      "What is gxadmin",
      "What can it do?",
      "How to write a query?"
    ],
    "objectives": [
      "Learn gxadmin basics",
      "See some queries and learn how they help debug production issues"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?monitoring\" style=\"background-color: hsl(30, 80.0%, 85%); color: #333; border: 1px solid hsl(30, 80.0%, 45%)\">monitoring</a>"
    ],
    "contributors": "Helena Rasche",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/gxadmin/tutorial.html"
  },
  "/topics/admin/tutorials/reports/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Reports",
    "description": null,
    "question": [
      "How to monitor a Galaxy service with the Reports application?"
    ],
    "objectives": [
      "Setup and start the Galaxy reports app."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?monitoring\" style=\"background-color: hsl(30, 80.0%, 85%); color: #333; border: 1px solid hsl(30, 80.0%, 45%)\">monitoring</a>"
    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/reports/tutorial.html"
  },
  "/topics/admin/tutorials/monitoring/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Telegraf and Grafana",
    "description": null,
    "question": [
      "How to monitor Galaxy with Telegraf",
      "How do I set up InfluxDB",
      "How can I make graphs in Grafana?",
      "How can I best alert on important metrics?"
    ],
    "objectives": [
      "Setup InfluxDB",
      "Setup Telegraf",
      "Setup Grafana",
      "Create several charts"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?monitoring\" style=\"background-color: hsl(30, 80.0%, 85%); color: #333; border: 1px solid hsl(30, 80.0%, 45%)\">monitoring</a>"
    ],
    "contributors": "Helena Rasche",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/admin/tutorials/monitoring/tutorial.html"
  },
  "/topics/admin/tutorials/cloudbursting/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy on the Cloud",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Simon Gladman",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/cloudbursting/slides.html"
  },
  "/topics/admin/tutorials/tool-management/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Tool Management with Ephemeris",
    "description": null,
    "question": [
      "How to install, update, and maintain Galaxy tools?",
      "How to extract a list of tools from a workflow or Galaxy instance?"
    ],
    "objectives": [
      "Learn about Ephemeris",
      "Extract a list of tools from a workflow",
      "Install these tools on a given Galaxy"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tools\" href=\"/training-material/search?tools\" style=\"background-color: hsl(344, 80.0%, 85%); color: #333; border: 1px solid hsl(344, 80.0%, 45%)\">tools</a>"
    ],
    "contributors": "Martin ƒåech, Helena Rasche, Nicola Soranzo",
    "level": null,
    "time_estimation": "45m",
    "url": "/training-material/topics/admin/tutorials/tool-management/tutorial.html"
  },
  "/topics/admin/tutorials/troubleshooting/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Galaxy Troubleshooting",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Martin ƒåech, Nate Coraor",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/troubleshooting/slides.html"
  },
  "/topics/admin/tutorials/production/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Gearing towards production",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Daniel Blankenberg, Abdulrahman Azab, Martin ƒåech",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/production/slides.html"
  },
  "/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Managing Galaxy on Kubernetes",
    "description": null,
    "question": [
      "How do I change Galaxy configs?",
      "How can I upgrade to a new version?",
      "How do I rollback my changes?",
      "How do I scale Galaxy?"
    ],
    "objectives": [
      "Have an understanding of how to modify Galaxy configuration",
      "Be able to upgrade and scale galaxy"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?kubernetes\" style=\"background-color: hsl(354, 80.0%, 85%); color: #333; border: 1px solid hsl(354, 80.0%, 45%)\">kubernetes</a>"
    ],
    "contributors": "Nuwan Goonasekera, Enis Afgan, Alex Mahmoud, Pablo Moreno, Sergey Golitsynskiy",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/job-destinations/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Mapping Jobs to Destinations",
    "description": null,
    "question": [
      "How can I configure job dependent resources, like cores, memory for my DRM?",
      "How can I map jobs to resources and destinations"
    ],
    "objectives": [
      "Know how to map tools to job destinations",
      "Be able to use the dynamic job runner to make arbitrary destination mappings",
      "Understand the job resource selector config and dynamic rule creation",
      "The various ways in which tools can be mapped to destinations, both statically and dynamically",
      "How to write a dynamic tool destination (DTD)",
      "How to write a dynamic python function destination",
      "How to use the job resource parameter selection feature"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?jobs\" style=\"background-color: hsl(134, 80.0%, 85%); color: #333; border: 1px solid hsl(134, 80.0%, 45%)\">jobs</a>"
    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/admin/tutorials/job-destinations/tutorial.html"
  },
  "/topics/admin/tutorials/cvmfs/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS",
    "description": null,
    "question": null,
    "objectives": [
      "Have an understanding of what CVMFS is and how it works",
      "Install and configure the CVMFS client on a linux machine and mount the Galaxy reference data repository",
      "Configure your Galaxy to use these reference genomes and indices",
      "Use an Ansible playbook for all of the above."
    ],
    "tags": [

    ],
    "contributors": "Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/cvmfs/tutorial.html"
  },
  "/topics/admin/tutorials/cvmfs-manual/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS without Ansible",
    "description": null,
    "question": null,
    "objectives": [
      "Have an understanding of what CVMFS is and how it works",
      "Install and configure the CVMFS client on a linux machine and mount the Galaxy reference data repository",
      "Configure your Galaxy to use these reference genomes and indices"
    ],
    "tags": [

    ],
    "contributors": "Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/cvmfs-manual/tutorial.html"
  },
  "/topics/admin/tutorials/reference-genomes/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Reference Genomes in Galaxy",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Daniel Blankenberg, Simon Gladman",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/reference-genomes/slides.html"
  },
  "/topics/admin/tutorials/pulsar/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Running Jobs on Remote Resources with Pulsar",
    "description": null,
    "question": [
      "How does pulsar work?",
      "How can I deploy it?"
    ],
    "objectives": [
      "Have an understanding of what Pulsar is and how it works",
      "Install and configure a RabbitMQ message queueing server",
      "Install and configure a Pulsar server on a remote linux machine",
      "Be able to get Galaxy to send jobs to a remote Pulsar server"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?jobs\" style=\"background-color: hsl(134, 80.0%, 85%); color: #333; border: 1px solid hsl(134, 80.0%, 45%)\">jobs</a>"
    ],
    "contributors": "Nate Coraor, Simon Gladman, Marius van den Beek, Helena Rasche, Gianmauro Cuccuru",
    "level": null,
    "time_estimation": "60m",
    "url": "/training-material/topics/admin/tutorials/pulsar/tutorial.html"
  },
  "/topics/admin/tutorials/maintenance/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Server Maintenance and Backups",
    "description": null,
    "question": [
      "How do I maintain a Galaxy server?",
      "What happens if I lose everything?"
    ],
    "objectives": [
      "Learn about different maintenance steps",
      "Learn what to back up and how to recover"
    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/maintenance/slides.html"
  },
  "/topics/admin/tutorials/stuff/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "Server: Other",
    "description": null,
    "question": [
      "How to monitor a Galaxy service?"
    ],
    "objectives": [
      "Learn about different monitoring strategies.",
      "Setup and start the Galaxy reports app."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/stuff/slides.html"
  },
  "/topics/admin/tutorials/tiaas/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Training Infrastructure as a Service (TIaaS)",
    "description": null,
    "question": [
      "How to deploy EU's TIaaS"
    ],
    "objectives": [
      "Setup TIaaS",
      "Request and manage trainings",
      "Join a training"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged training\" href=\"/training-material/search?training\" style=\"background-color: hsl(315, 80.0%, 85%); color: #333; border: 1px solid hsl(315, 80.0%, 45%)\">training</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?jobs\" style=\"background-color: hsl(134, 80.0%, 85%); color: #333; border: 1px solid hsl(134, 80.0%, 45%)\">jobs</a>"
    ],
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "level": null,
    "time_estimation": "30m",
    "url": "/training-material/topics/admin/tutorials/tiaas/tutorial.html"
  },
  "/topics/admin/tutorials/upgrading/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Upgrading Galaxy",
    "description": null,
    "question": [
      "How can I find which version of Galaxy I'm running?",
      "How do I upgrade the Galaxy server to the latest version?"
    ],
    "objectives": [
      "Understand how to find the current Galaxy server version you are using",
      "How to find the latest release version",
      "How to backup the Galaxy server database and why it's a good idea",
      "How to upgrade the Galaxy server to a new version"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?ansible\" style=\"background-color: hsl(331, 80.0%, 85%); color: #333; border: 1px solid hsl(331, 80.0%, 45%)\">ansible</a>"
    ],
    "contributors": "Simon Gladman",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/upgrading/tutorial.html"
  },
  "/topics/admin/tutorials/singularity/tutorial.html": {
    "topic": "Galaxy Server administration",
    "title": "Use Singularity containers for running Galaxy jobs",
    "description": null,
    "question": null,
    "objectives": [
      "Configure your Galaxy to use Singularity and BioContainers for running jobs"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?jobs\" style=\"background-color: hsl(134, 80.0%, 85%); color: #333; border: 1px solid hsl(134, 80.0%, 45%)\">jobs</a>"
    ],
    "contributors": "Torfinn Nome, Marius van den Beek, Matthias Bernt, Helena Rasche",
    "level": null,
    "time_estimation": "1h",
    "url": "/training-material/topics/admin/tutorials/singularity/tutorial.html"
  },
  "/topics/admin/tutorials/users-groups-quotas/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "User, Group and Quota managment",
    "description": null,
    "question": [
      "How does Galaxy manage users and groups?",
      "How can I assign Quotas to specific users/groups?"
    ],
    "objectives": [
      "Learn the Galaxy user/group management and assign Quotas.",
      "Understand the Role Based Access Control (RBAC) of Galaxy."
    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "level": null,
    "time_estimation": "15m",
    "url": "/training-material/topics/admin/tutorials/users-groups-quotas/slides.html"
  },
  "/topics/admin/tutorials/uwsgi/slides.html": {
    "topic": "Galaxy Server administration",
    "title": "uWSGI",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Nate Coraor, Simon Gladman",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/admin/tutorials/uwsgi/slides.html"
  },
  "/topics/genome-annotation/slides/introduction.html": {
    "topic": "Genome Annotation",
    "title": "Introduction to Genome Annotation",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Anthony Bretaudeau, Helena Rasche",
    "level": "Intermediate",
    "time_estimation": null,
    "url": "/training-material/topics/genome-annotation/slides/introduction.html"
  },
  "/topics/genome-annotation/tutorials/tnseq/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Essential genes detection with Transposon insertion sequencing",
    "description": null,
    "question": [
      "What is Transposon insertion Sequencing?",
      "How to get TA sites Coverage ? ",
      "How to predict essential genes ?"
    ],
    "objectives": [
      "Understand the read structure of TnSeq analyses",
      "Predict Essential genes with Transit",
      "Compare gene essentiality in control sample and an different experimental condition"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?bacteria\" style=\"background-color: hsl(55, 80.0%, 85%); color: #333; border: 1px solid hsl(55, 80.0%, 45%)\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tnseq\" href=\"/training-material/search?tnseq\" style=\"background-color: hsl(19, 80.0%, 85%); color: #333; border: 1px solid hsl(19, 80.0%, 45%)\">tnseq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged essential genes\" href=\"/training-material/search?essential genes\" style=\"background-color: hsl(6, 80.0%, 85%); color: #333; border: 1px solid hsl(6, 80.0%, 45%)\">essential genes</a>"
    ],
    "contributors": "Delphine Lariviere, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "7H",
    "url": "/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "From small to large-scale genome comparison",
    "description": null,
    "question": [
      "How can we run pairwise genome comparisons using Galaxy?",
      "How can we run massive chromosome comparisons in Galaxy?",
      "How can we quickly visualize genome comparisons in Galaxy?"
    ],
    "objectives": [
      "Learn the basics of pairwise sequence comparison",
      "Learn how to run different tools in Galaxy to perform sequence comparison at fine and coarse-grained levels",
      "Learn how to post-process your sequence comparisons"
    ],
    "tags": [

    ],
    "contributors": "Esteban Perez-Wohlfeil",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/genome-annotation/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Genome Annotation",
    "description": null,
    "question": null,
    "objectives": null,
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Anika Erxleben, Bj√∂rn Gr√ºning",
    "level": "Introductory",
    "time_estimation": "2H",
    "url": "/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/funannotate/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Genome annotation with Funannotate",
    "description": null,
    "question": [
      "How to annotate an eukaryotic genome with Funannotate?",
      "How to perform functional annotation?",
      "How to evaluate and visualize annotated genomic features?",
      "How to format the annotation for submission at NCBI?"
    ],
    "objectives": [
      "Load genome into Galaxy",
      "Annotate genome with Funannotate",
      "Perform functional annotation using EggNOG-mapper and InterProScan",
      "Evaluate annotation quality with BUSCO",
      "View annotations in JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?eukaryote\" style=\"background-color: hsl(124, 80.0%, 85%); color: #333; border: 1px solid hsl(124, 80.0%, 45%)\">eukaryote</a>"
    ],
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Laura Leroi, Erwan Corre, St√©phanie Robin, Erasmus+ Programme",
    "level": "Intermediate",
    "time_estimation": "8h",
    "url": "/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Genome annotation with Maker",
    "description": null,
    "question": [
      "How to annotate an eukaryotic genome?",
      "How to evaluate and visualize annotated genomic features?"
    ],
    "objectives": [
      "Load genome into Galaxy",
      "Annotate genome with Maker",
      "Evaluate annotation quality with BUSCO",
      "View annotations in JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?eukaryote\" style=\"background-color: hsl(124, 80.0%, 85%); color: #333; border: 1px solid hsl(124, 80.0%, 45%)\">eukaryote</a>"
    ],
    "contributors": "Anthony Bretaudeau",
    "level": "Advanced",
    "time_estimation": "4h",
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Genome annotation with Maker (short)",
    "description": null,
    "question": [
      "How to annotate an eukaryotic genome?",
      "How to evaluate and visualize annotated genomic features?"
    ],
    "objectives": [
      "Load genome into Galaxy",
      "Annotate genome with Maker",
      "Evaluate annotation quality with BUSCO",
      "View annotations in JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?eukaryote\" style=\"background-color: hsl(124, 80.0%, 85%); color: #333; border: 1px solid hsl(124, 80.0%, 45%)\">eukaryote</a>"
    ],
    "contributors": "Anthony Bretaudeau",
    "level": "Intermediate",
    "time_estimation": "2h",
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Genome annotation with Prokka",
    "description": null,
    "question": [
      "How can we annotate a bacterial genome?",
      "How can we visualize annotated genomic features?"
    ],
    "objectives": [
      "Load genome into Galaxy",
      "Annotate genome with Prokka",
      "View annotations in JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anna Syme, Torsten Seemann, Simon Gladman",
    "level": "Introductory",
    "time_estimation": "1h",
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/repeatmasker/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Masking repeats with RepeatMasker",
    "description": null,
    "question": [
      "How to mask repeats in a genome?",
      "What is the difference between hard and soft masking?"
    ],
    "objectives": [
      "Use RepeatMasker to soft mask a newly assmbled genome"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?eukaryote\" style=\"background-color: hsl(124, 80.0%, 85%); color: #333; border: 1px solid hsl(124, 80.0%, 45%)\">eukaryote</a>"
    ],
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Laura Leroi, Erwan Corre, St√©phanie Robin, Erasmus+ Programme",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/apollo/tutorial.html": {
    "topic": "Genome Annotation",
    "title": "Refining Genome Annotations with Apollo",
    "description": null,
    "question": [
      "How to visualize your genome after automated annotations have been performed?",
      "How to manually annotate genome after automated annotations have been performed?",
      "How to evaluate and visualize annotated genomic features?",
      "How do I collaborate when doing genome annotation?"
    ],
    "objectives": [
      "Load a genome into Galaxy",
      "View annotations in JBrowse",
      "Learn how to load JBrowse data into Apollo",
      "Learn how to manually refine genome annotations within Apollo",
      "Export refined genome annotations"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anthony Bretaudeau, Helena Rasche, Nathan Dunn, Mateo Boudet, Erasmus+ Programme",
    "level": "Intermediate",
    "time_estimation": "3h",
    "url": "/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html"
  },
  "/topics/imaging/tutorials/hela-screen-analysis/tutorial.html": {
    "topic": "Imaging",
    "title": "Analyse HeLa fluorescence siRNA screen",
    "description": null,
    "question": [
      "How do I analyze a HeLa fluorescence siRNA screen?",
      "How do I segment cell nuclei?",
      "How do I extract features from segmentations?",
      "How do I filter segmentations by morphological features?",
      "How do I apply a feature extraction workflow to a screen?",
      "How do I visualize feature extraction results?"
    ],
    "objectives": [
      "How to segment cell nuclei in Galaxy.",
      "How to extract features from segmentations in Galaxy.",
      "How to filter segmentations by morphological features in Galaxy.",
      "How to extract features from an imaging screen in Galaxy.",
      "How to analyse extracted features from an imaging screen in Galaxy."
    ],
    "tags": [

    ],
    "contributors": "Thomas Wollmann",
    "level": "Intermediate",
    "time_estimation": "1H",
    "url": "/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html"
  },
  "/topics/imaging/tutorials/imaging-introduction/tutorial.html": {
    "topic": "Imaging",
    "title": "Introduction to image analysis using Galaxy",
    "description": null,
    "question": [
      "How do I use Galaxy with imaging data?",
      "How do I convert images using Galaxy?",
      "How do I display images in Galaxy?",
      "How do I filter images in Galaxy?",
      "How do I segment simple images in Galaxy?"
    ],
    "objectives": [
      "How to handle images in Galaxy.",
      "How to perform basic image processing in Galaxy."
    ],
    "tags": [

    ],
    "contributors": "Thomas Wollmann, Saskia Hiltemann",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html"
  },
  "/topics/imaging/tutorials/tutorial-CP/tutorial.html": {
    "topic": "Imaging",
    "title": "Nucleoli segmentation and feature extraction using CellProfiler",
    "description": null,
    "question": [
      "How do I run an image analysis pipeline on public data using CellProfiler?",
      "How do I analyse the DNA channel of fluorescence siRNA screens?",
      "How do I download public image data into my history?",
      "How do I segment and label cell nuclei?",
      "How do I segment nucleoli (as the absence of DNA)?",
      "How do I combine nuclei and nucleoli into one segmentation mask?",
      "How do I extract the background of an image?",
      "How do I relate the nucleoli to their parent nucleus?",
      "How do I measure the image and object features?",
      "How do I measure the image quality?"
    ],
    "objectives": [
      "How to download images from a public image repository.",
      "How to segment cell nuclei using CellProfiler in Galaxy.",
      "How to segment cell nucleoli using CellProfiler in Galaxy.",
      "How to extract features for images, nuclei and nucleoli."
    ],
    "tags": [

    ],
    "contributors": "Beatriz Serrano-Solano, Jean-Karim H√©rich√©",
    "level": null,
    "time_estimation": "4H",
    "url": "/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html"
  },
  "/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html": {
    "topic": "Imaging",
    "title": "Object tracking using CellProfiler",
    "description": null,
    "question": [
      "How to segment and track objects in fluorescence time-lapse microscopy images?"
    ],
    "objectives": [
      "Segment fluorescent objects using CellProfiler in Galaxy",
      "Track objects over multiple frames using CellProfiler in Galaxy"
    ],
    "tags": [

    ],
    "contributors": "Yi Sun, Beatriz Serrano-Solano, Jean-Karim H√©rich√©",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html"
  },
  "/topics/metagenomics/slides/introduction.html": {
    "topic": "Metagenomics",
    "title": "Introduction to metagenomics",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/metagenomics/slides/introduction.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html": {
    "topic": "Metagenomics",
    "title": "16S Microbial Analysis with mothur (extended)",
    "description": null,
    "question": [
      "What is the effect of normal variation in the gut microbiome on host health?"
    ],
    "objectives": [
      "Analyze of 16S rRNA sequencing data using the mothur toolsuite in Galaxy",
      "Using a mock community to assess the error rate of your sequencing experiment",
      "Visualize sample diversity using Krona and Phinch"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Dave Clements",
    "level": null,
    "time_estimation": "6h",
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html": {
    "topic": "Metagenomics",
    "title": "16S Microbial Analysis with mothur (short)",
    "description": null,
    "question": [
      "What is the effect of normal variation in the gut microbiome on host health?"
    ],
    "objectives": [
      "Analyze of 16S rRNA sequencing data using the mothur toolsuite in Galaxy",
      "Using a mock community to assess the error rate of your sequencing experiment",
      "Visualize sample diversity using Krona and Phinch"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Dave Clements",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html"
  },
  "/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html": {
    "topic": "Metagenomics",
    "title": "16S Microbial analysis with Nanopore data",
    "description": null,
    "question": [
      "How can we analyse the health status of the soil?",
      "How do plants modify the composition of microbial communities?"
    ],
    "objectives": [
      "Use Nanopore data for studying soil metagenomics",
      "Analyze and preprocess Nanopore reads",
      "Use Kraken2 to assign a taxonomic labels"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?nanopore\" style=\"background-color: hsl(239, 80.0%, 85%); color: #333; border: 1px solid hsl(239, 80.0%, 45%)\">nanopore</a>"
    ],
    "contributors": "Crist√≥bal Gallardo",
    "level": null,
    "time_estimation": "2H",
    "url": "/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html"
  },
  "/topics/metagenomics/tutorials/general-tutorial/tutorial.html": {
    "topic": "Metagenomics",
    "title": "Analyses of metagenomics data - The global picture",
    "description": null,
    "question": [
      "How to analyze metagenomics data?",
      "What information can be extracted of metagenomics data?",
      "What is the difference between amplicon and shotgun data?",
      "What are the difference in the analyses of amplicon and shotgun data?"
    ],
    "objectives": [
      "Choosing the best approach to analyze metagenomics data",
      "Selection of tools to analyze amplicon data or shotgun data",
      "Visualisation of a community structure"
    ],
    "tags": [

    ],
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut",
    "level": null,
    "time_estimation": "2H30M",
    "url": "/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html"
  },
  "/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html": {
    "topic": "Metagenomics",
    "title": "Antibiotic resistance detection",
    "description": null,
    "question": [
      "How do I assemble a genome with Nanopore data?",
      "How do I get more information about the structure of the genomes?",
      "How do I get more information about the antimicrobial resistance genes?"
    ],
    "objectives": [
      "Perform Quality control on your reads",
      "Assemble a genome with Minimap2/Miniasm/Racon",
      "Determine the structure of the genome(s)",
      "Scan for antimicrobial resistance genes with Staramr"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?nanopore\" style=\"background-color: hsl(239, 80.0%, 85%); color: #333; border: 1px solid hsl(239, 80.0%, 45%)\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plasmids\" href=\"/training-material/search?plasmids\" style=\"background-color: hsl(156, 80.0%, 85%); color: #333; border: 1px solid hsl(156, 80.0%, 45%)\">plasmids</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?assembly\" style=\"background-color: hsl(334, 80.0%, 85%); color: #333; border: 1px solid hsl(334, 80.0%, 45%)\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?amr\" style=\"background-color: hsl(255, 80.0%, 85%); color: #333; border: 1px solid hsl(255, 80.0%, 45%)\">amr</a>"
    ],
    "contributors": "Willem de Koning, Saskia Hiltemann",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html": {
    "topic": "Metagenomics",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data",
    "description": null,
    "question": [
      "How to analyze metatranscriptomics data?",
      "What information can be extracted of metatranscriptomics data?",
      "How to assign taxa and function to the identified sequences?"
    ],
    "objectives": [
      "Choose the best approach to analyze metatranscriptomics data",
      "Understand the functional microbiome characterization using metatranscriptomic results",
      "Understand where metatranscriptomics fits in 'multi-omic' analysis of microbiomes",
      "Visualise a community structure"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?metatranscriptomics\" style=\"background-color: hsl(294, 80.0%, 85%); color: #333; border: 1px solid hsl(294, 80.0%, 45%)\">metatranscriptomics</a>"
    ],
    "contributors": "Pratik Jagtap, Subina Mehta, Ray Sajulga, B√©r√©nice Batut, Emma Leith, Praveen Kumar, Saskia Hiltemann",
    "level": "Introductory",
    "time_estimation": "5H",
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics-short/tutorial.html": {
    "topic": "Metagenomics",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data (short)",
    "description": null,
    "question": [
      "How to analyze metatranscriptomics data?\"",
      "What information can be extracted of metatranscriptomics data?",
      "How to assign taxa and function to the identified sequences?"
    ],
    "objectives": [
      "Choose the best approach to analyze metatranscriptomics data",
      "Understand the functional microbiome characterization using metatranscriptomic results",
      "Understand where metatranscriptomics fits in 'multi-omic' analysis of microbiomes",
      "Visualise a community structure"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?metatranscriptomics\" style=\"background-color: hsl(294, 80.0%, 85%); color: #333; border: 1px solid hsl(294, 80.0%, 45%)\">metatranscriptomics</a>"
    ],
    "contributors": "Pratik Jagtap, Subina Mehta, Ray Sajulga, B√©r√©nice Batut, Emma Leith, Praveen Kumar, Saskia Hiltemann",
    "level": "Introductory",
    "time_estimation": "3H",
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics-short/tutorial.html"
  },
  "/topics/assembly/tutorials/general-introduction/tutorial.html": {
    "topic": "Assembly",
    "title": "An Introduction to Genome Assembly",
    "description": null,
    "question": [
      "How do we perform a very basic genome assembly from short read data?"
    ],
    "objectives": [
      "assemble some paired end reads using Velvet",
      "examine the output of the assembly."
    ],
    "tags": [

    ],
    "contributors": "Simon Gladman",
    "level": "Introductory",
    "time_estimation": "30m",
    "url": "/training-material/topics/assembly/tutorials/general-introduction/tutorial.html"
  },
  "/topics/assembly/tutorials/get-started-genome-assembly/slides.html": {
    "topic": "Assembly",
    "title": "An introduction to get started in genome assembly and annotation",
    "description": null,
    "question": [
      "What are the guidelines before starting a Genome Assembly and Annotation project?"
    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, St√©phanie Robin, Erwan Corre, Laura Leroi, Erasmus+ Programme",
    "level": "Introductory",
    "time_estimation": null,
    "url": "/training-material/topics/assembly/tutorials/get-started-genome-assembly/slides.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/tutorial.html": {
    "topic": "Assembly",
    "title": "Chloroplast genome assembly",
    "description": null,
    "question": [
      "How can we assemble a chloroplast genome?"
    ],
    "objectives": [
      "Assemble a chloroplast genome from long reads",
      "Polish the assembly with short reads",
      "Annotate the assembly and view",
      "Map reads to the assembly and view"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?plants\" style=\"background-color: hsl(121, 80.0%, 85%); color: #333; border: 1px solid hsl(121, 80.0%, 45%)\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?nanopore\" style=\"background-color: hsl(239, 80.0%, 85%); color: #333; border: 1px solid hsl(239, 80.0%, 45%)\">nanopore</a>"
    ],
    "contributors": "Anna Syme",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html": {
    "topic": "Assembly",
    "title": "De Bruijn Graph Assembly",
    "description": null,
    "question": [
      "What are the factors that affect genome assembly?",
      "How does Genome assembly work?"
    ],
    "objectives": [
      "Perform an optimised Velvet assembly with the Velvet Optimiser",
      "Compare this assembly with those we did in the basic tutorial",
      "Perform an assembly using the SPAdes assembler."
    ],
    "tags": [

    ],
    "contributors": "Simon Gladman, Helena Rasche, Saskia Hiltemann",
    "level": "Introductory",
    "time_estimation": "2h",
    "url": "/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/algorithms-introduction/slides.html": {
    "topic": "Assembly",
    "title": "Deeper look into Genome Assembly algorithms",
    "description": null,
    "question": [
      "What are the main types of assembly algorithms?",
      "How do they perform with short and long reads?",
      "How to improve assemblies using other technologies?"
    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Anthony Bretaudeau, Simon Gladman, Anton Nekrutenko, Delphine Lariviere",
    "level": "Intermediate",
    "time_estimation": null,
    "url": "/training-material/topics/assembly/tutorials/algorithms-introduction/slides.html"
  },
  "/topics/assembly/tutorials/mrsa-illumina/tutorial.html": {
    "topic": "Assembly",
    "title": "Genome Assembly of MRSA using Illumina MiSeq Data",
    "description": null,
    "question": [
      "How to check the quality of the MiSeq data?",
      "How to perform an assembly with MiSeq data?",
      "What species do I have and what is its sequence type?",
      "Which resistance genes are on my genome?",
      "Where are the genes located on my genome?"
    ],
    "objectives": [
      "Assess your data on quality and quantity",
      "Assemble a genome",
      "Assess your assembly quality",
      "Assess presence of antimicrobial resistance genes",
      "Perform a species identification and MLST typing",
      "Search for resistance genes on the assembly",
      "Find a gene on your genome using Prokka + JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?illumina\" style=\"background-color: hsl(52, 80.0%, 85%); color: #333; border: 1px solid hsl(52, 80.0%, 45%)\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?assembly\" style=\"background-color: hsl(334, 80.0%, 85%); color: #333; border: 1px solid hsl(334, 80.0%, 45%)\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?amr\" style=\"background-color: hsl(255, 80.0%, 85%); color: #333; border: 1px solid hsl(255, 80.0%, 45%)\">amr</a>"
    ],
    "contributors": "Bazante Sanders, Miaomiao Zhou, Saskia Hiltemann, Helena Rasche, Avans Hogeschool",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html"
  },
  "/topics/assembly/tutorials/mrsa-nanopore/tutorial.html": {
    "topic": "Assembly",
    "title": "Genome Assembly of MRSA using Oxford Nanopore MinION Data",
    "description": null,
    "question": [
      "How to check the quality of the MinION data?",
      "How to perform an assembly with MinION data?",
      "What species do I have and what is its sequence type?",
      "Which resistance genes are on my genome?",
      "Where are the genes located on my genome?"
    ],
    "objectives": [
      "Assess your data on quality and quantity",
      "Assemble a genome",
      "Assess your assembly quality",
      "Assess presence of antimicrobial resistance genes",
      "Perform a species identification and MLST typing",
      "Search for resistance genes on the assembly",
      "Find a gene on your genome using Prokka + JBrowse"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?nanopore\" style=\"background-color: hsl(239, 80.0%, 85%); color: #333; border: 1px solid hsl(239, 80.0%, 45%)\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?assembly\" style=\"background-color: hsl(334, 80.0%, 85%); color: #333; border: 1px solid hsl(334, 80.0%, 45%)\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?amr\" style=\"background-color: hsl(255, 80.0%, 85%); color: #333; border: 1px solid hsl(255, 80.0%, 45%)\">amr</a>"
    ],
    "contributors": "Bazante Sanders, Miaomiao Zhou, Saskia Hiltemann, Helena Rasche, Avans Hogeschool",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html"
  },
  "/topics/assembly/tutorials/flye-assembly/tutorial.html": {
    "topic": "Assembly",
    "title": "Genome assembly using PacBio data",
    "description": null,
    "question": [
      "How to perform a genome assembly with PacBio data ?",
      "How to check assembly quality ?"
    ],
    "objectives": [
      "Assemble a Genome with PacBio data",
      "Assess assembly quality"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?assembly\" style=\"background-color: hsl(334, 80.0%, 85%); color: #333; border: 1px solid hsl(334, 80.0%, 45%)\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?pacbio\" style=\"background-color: hsl(337, 80.0%, 85%); color: #333; border: 1px solid hsl(337, 80.0%, 45%)\">pacbio</a>"
    ],
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Erwan Corre, Laura Leroi, St√©phanie Robin, Erasmus+ Programme",
    "level": "Intermediate",
    "time_estimation": "6h",
    "url": "/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/ecoli_comparison/tutorial.html": {
    "topic": "Assembly",
    "title": "Making sense of a newly assembled genome",
    "description": null,
    "question": [
      "I just assembled a genome. How does it compare with already sequenced genomes?",
      "How do I find rearranged, inserted, or deleted regions?"
    ],
    "objectives": [
      "Identification of the most closely related genome to my new assembly",
      "Perform sequence comparison to locate rearrangements",
      "Identify genes located in deletions"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anton Nekrutenko, Delphine Lariviere, Helena Rasche",
    "level": null,
    "time_estimation": "4h",
    "url": "/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html"
  },
  "/topics/assembly/tutorials/unicycler-assembly/tutorial.html": {
    "topic": "Assembly",
    "title": "Unicycler Assembly",
    "description": null,
    "question": [
      "I have short reads and long reads. How do I assemble a genome?"
    ],
    "objectives": [
      "Perform Quality Control on your reads",
      "Perform a Small genome Assembly with Unicycler",
      "Evaluate the Quality of the Assembly with Quast",
      "Annotate the assembly with Prokka"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anton Nekrutenko, Delphine Lariviere, Simon Gladman",
    "level": "Introductory",
    "time_estimation": "4h",
    "url": "/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html": {
    "topic": "Assembly",
    "title": "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads",
    "description": null,
    "question": [
      "How can a genome of interest be assembled against a background of contaminating reads from other genomes?",
      "How can sequencing data from public sources be turned into assembly-ready polished datasets?"
    ],
    "objectives": [
      "Obtain viral (SARS-CoV-2) sequencing data with contaminating human reads from public sources",
      "Organize the data into collections and check its quality",
      "Detect and remove human reads",
      "Assemble retained reads and explore the results"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Wolfgang Maier",
    "level": "Intermediate",
    "time_estimation": "4h",
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html"
  },
  "/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html": {
    "topic": "Assembly",
    "title": "VGP assembly pipeline",
    "description": null,
    "question": [
      "what combination of tools can produce the highest quality assembly of vertebrate genomes?",
      "How can we evaluate how good it is?"
    ],
    "objectives": [
      "Learn the tools necessary to perform a de novo assembly of a vertebrate genome",
      "Evaluate the quality of the assembly"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Delphine Lariviere, Alex Ostrovsky, Crist√≥bal Gallardo, Anna Syme, Linelle Abueg, Brandon Pickett",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html"
  },
  "/topics/proteomics/slides/introduction.html": {
    "topic": "Proteomics",
    "title": "Introduction to proteomics, protein identification, quantification and statistical modelling",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "Melanie F√∂ll",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/proteomics/slides/introduction.html"
  },
  "/topics/proteomics/tutorials/proteome_annotation/tutorial.html": {
    "topic": "Proteomics",
    "title": "Annotating a protein list identified by LC-MS/MS experiments",
    "description": null,
    "question": [
      "How to filter out technical contaminants?",
      "How to check for tissue-specificity?",
      "How to perform enrichment analysis?",
      "How to map your protein list to pathways (Reactome)?",
      "How to compare your proteome with other studies?"
    ],
    "objectives": [
      "Execute a complete annotation pipeline of a protein list identified by LC-MS/MS experiments"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?human\" style=\"background-color: hsl(16, 80.0%, 85%); color: #333; border: 1px solid hsl(16, 80.0%, 45%)\">human</a>"
    ],
    "contributors": "Valentin Loux, Florence Combes, David Christiany, Yves Vandenbrouck",
    "level": null,
    "time_estimation": "1H",
    "url": "/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/tutorial.html": {
    "topic": "Proteomics",
    "title": "Biomarker candidate identification",
    "description": null,
    "question": [
      "How to mine public databases to retrieve info?",
      "How to build a selection strategy by applying successive biochemical/cellular criteria to a list of gene/protein?",
      "How to select biomarkers candidates using experimental information (transcriptomics & proteomics) and annotation from public databases?"
    ],
    "objectives": [
      "Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?human\" style=\"background-color: hsl(16, 80.0%, 85%); color: #333; border: 1px solid hsl(16, 80.0%, 45%)\">human</a>"
    ],
    "contributors": "Florence Combes, David Christiany, Valentin Loux, Yves Vandenbrouck",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html"
  },
  "/topics/proteomics/tutorials/ntails/tutorial.html": {
    "topic": "Proteomics",
    "title": "Detection and quantitation of N-termini (degradomics) via N-TAILS",
    "description": null,
    "question": [
      "How can protein N-termini be enriched for LC-MS/MS?",
      "How to analyze the LC-MS/MS data?"
    ],
    "objectives": [
      "Run an N-TAILS data analysis."
    ],
    "tags": [

    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/proteomics/tutorials/ntails/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html": {
    "topic": "Proteomics",
    "title": "DIA Analysis using OpenSwathWorkflow",
    "description": null,
    "question": [
      "How to analyze data independent acquisition (DIA) mass spectrometry (MS) data using OpenSwathWorkflow?",
      "How to detect different Spike-in amounts of Ecoli in a HEK-Ecoli Benchmark DIA dataset?"
    ],
    "objectives": [
      "Analysis of HEK-Ecoli Spike-in DIA data in Galaxy",
      "Understanding DIA data principles and characteristics",
      "Using OpenSwathworkflow to analyze HEK-Ecoli Spike-in DIA data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?DIA\" style=\"background-color: hsl(215, 80.0%, 85%); color: #333; border: 1px solid hsl(215, 80.0%, 45%)\">DIA</a>"
    ],
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/tutorial.html": {
    "topic": "Proteomics",
    "title": "EncyclopeDIA",
    "description": null,
    "question": [
      "How to perform quantitative analysis using DIA data with the help of EncyclopeDIA?",
      "How to perform quantitation with or without Chromatogram Libraries?"
    ],
    "objectives": [
      "Understand the difference between DDA and DIA methods",
      "Performing quantitative analysis using DIA data",
      "Understand the purpose and benefits of using a Chromatogram Library for detection of peptides",
      "You can use Bloom's Taxonomy to write effective learning objectives"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?DIA\" style=\"background-color: hsl(215, 80.0%, 85%); color: #333; border: 1px solid hsl(215, 80.0%, 45%)\">DIA</a>"
    ],
    "contributors": "Emma Leith, Subina Mehta, James Johnson, Pratik Jagtap, Timothy J. Griffin",
    "level": null,
    "time_estimation": "6H",
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-label-free/tutorial.html": {
    "topic": "Proteomics",
    "title": "Label-free data analysis using MaxQuant",
    "description": null,
    "question": [
      "How to perform label-free analysis in Maxquant?",
      "Which are the most abundant proteins in serum?",
      "How successful was the depletion of those in our experiment?"
    ],
    "objectives": [
      "Analysis of human serum proteome samples with label-free quantification in MaxQuant"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?label-free\" style=\"background-color: hsl(260, 80.0%, 85%); color: #333; border: 1px solid hsl(260, 80.0%, 45%)\">label-free</a>"
    ],
    "contributors": "Melanie F√∂ll, Matthias Fahrner",
    "level": "Introductory",
    "time_estimation": "1H",
    "url": "/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html"
  },
  "/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html": {
    "topic": "Proteomics",
    "title": "Label-free versus Labelled - How to Choose Your Quantitation Method",
    "description": null,
    "question": [
      "What are benefits and drawbacks of different quantitation methods?",
      "How to choose which quantitation method bests suits my need?"
    ],
    "objectives": [
      "Choose your quantitation method."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Introductory",
    "time_estimation": "15m",
    "url": "/training-material/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html": {
    "topic": "Proteomics",
    "title": "Library Generation for DIA Analysis",
    "description": null,
    "question": [
      "How does the peptide identification work in data independent acquisition (DIA) mass spectrometry (MS) data?",
      "What is a spectral library and how to generate a spectral library using Maxquant, diapysef and OpenSwath tools?"
    ],
    "objectives": [
      "Generating a spectral library from data dependent acquisition (DDA) MS data",
      "Understanding DIA data principles and characteristics",
      "Optimizing and refining a spectral library for the analysis of DIA data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?DIA\" style=\"background-color: hsl(215, 80.0%, 85%); color: #333; border: 1px solid hsl(215, 80.0%, 45%)\">DIA</a>"
    ],
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "level": "Intermediate",
    "time_estimation": "1H",
    "url": "/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html": {
    "topic": "Proteomics",
    "title": "Machine Learning Modeling of Anticancer Peptides",
    "description": null,
    "question": [
      "Which machine learning (ML) algorithm is superior in classifying anticancer peptides (ACPs) and non-anticancer peptides (non-ACPs)?"
    ],
    "objectives": [
      "Learn, how to calculate peptide descriptor",
      "Learn, how to create training data set from features?",
      "Assessment of best ML algorithm in predicting anticancer peptide"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ML\" href=\"/training-material/search?ML\" style=\"background-color: hsl(277, 80.0%, 85%); color: #333; border: 1px solid hsl(277, 80.0%, 45%)\">ML</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer\" href=\"/training-material/search?cancer\" style=\"background-color: hsl(335, 80.0%, 85%); color: #333; border: 1px solid hsl(335, 80.0%, 45%)\">cancer</a>"
    ],
    "contributors": "Jayadev Joshi, Daniel Blankenberg",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html"
  },
  "/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html": {
    "topic": "Proteomics",
    "title": "Mass spectrometry imaging: Loading and exploring MSI data",
    "description": null,
    "question": [
      "Did the tryptic digestions work?",
      "Is there an intensity gradient in the data?",
      "How good is the m/z accuracy?"
    ],
    "objectives": [
      "How to upload an imzML file into Galaxy",
      "Getting an overview of the MSI data with the MSI quality control tool",
      "How to export MSI data as tabular files"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?mouse\" style=\"background-color: hsl(164, 80.0%, 85%); color: #333; border: 1px solid hsl(164, 80.0%, 45%)\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged imaging\" href=\"/training-material/search?imaging\" style=\"background-color: hsl(330, 80.0%, 85%); color: #333; border: 1px solid hsl(330, 80.0%, 45%)\">imaging</a>"
    ],
    "contributors": "Melanie F√∂ll, Bj√∂rn Gr√ºning",
    "level": "Introductory",
    "time_estimation": "45m",
    "url": "/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html": {
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of label-free data",
    "description": null,
    "question": [
      "How to perform label-free shotgun (DDA) data analysis in MaxQuant and MSstats?",
      "Which proteins are differentially abundant in the two types of cutaneous squamous cell carcinomas?"
    ],
    "objectives": [
      "Learn how to use MaxQuant and MSstats for the analysis of label-free shotgun (DDA) data"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?label-free\" style=\"background-color: hsl(260, 80.0%, 85%); color: #333; border: 1px solid hsl(260, 80.0%, 45%)\">label-free</a>"
    ],
    "contributors": "Melanie F√∂ll, Matthias Fahrner",
    "level": "Intermediate",
    "time_estimation": "2H",
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html": {
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of TMT data",
    "description": null,
    "question": [
      "How to analyze tandem-mass-tag (TMT) labelled shotgun (DDA) data analysis in MaxQuant and MSstats?",
      "Which proteins are differentially abundant between KMT9Œ± knock-down and control cells?"
    ],
    "objectives": [
      "Learn how to use MaxQuant and MSstats for the analysis of TMT labelled shotgun (DDA) data",
      "Learn how to create an experimental design template for fractionated data in MaxQuant",
      "Learn how to create an annotation file for MSstatsTMT"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged TMT\" href=\"/training-material/search?TMT\" style=\"background-color: hsl(83, 80.0%, 85%); color: #333; border: 1px solid hsl(83, 80.0%, 45%)\">TMT</a>"
    ],
    "contributors": "Klemens Fr√∂hlich, Melanie F√∂ll, Matthias Fahrner",
    "level": null,
    "time_estimation": "3H",
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/tutorial.html": {
    "topic": "Proteomics",
    "title": "Metaproteomics tutorial",
    "description": null,
    "question": [
      "How can I match metaproteomic mass spectrometry data to peptide sequences derived from shotgun metagenomic data?",
      "How can I perform taxonomy analysis and visualize metaproteomics data?",
      "How can I perform functional analysis on this metaproteomics data?"
    ],
    "objectives": [
      "A taxonomy and functional analysis of metaproteomic mass spectrometry data."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?microbiome\" style=\"background-color: hsl(26, 80.0%, 85%); color: #333; border: 1px solid hsl(26, 80.0%, 45%)\">microbiome</a>"
    ],
    "contributors": "Timothy J. Griffin, Pratik Jagtap, James Johnson, Clemens Blank, Subina Mehta",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html": {
    "topic": "Proteomics",
    "title": "metaQuantome 1: Data creation",
    "description": null,
    "question": [
      "How do I perform functional and taxonomy analysis on metaproteomics data?",
      "How can I perform quantitation on metaproteomics data?",
      "How do I create inputs that can be used in metaquantome to examine differentially expressed proteins?"
    ],
    "objectives": [
      "A taxonomy, functional and quantitational analysis of metaproteomic mass spectrometry data."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?microbiome\" style=\"background-color: hsl(26, 80.0%, 85%); color: #333; border: 1px solid hsl(26, 80.0%, 45%)\">microbiome</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-function/tutorial.html": {
    "topic": "Proteomics",
    "title": "metaQuantome 2: Function",
    "description": null,
    "question": [
      "How do I look at differentially expressed proteins?",
      "How do I get top 5 or 10 function present in the sample?",
      "How can I visualize my results?"
    ],
    "objectives": [
      "A functional and quantitational analysis of metaproteomic mass spectrometry data."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?microbiome\" style=\"background-color: hsl(26, 80.0%, 85%); color: #333; border: 1px solid hsl(26, 80.0%, 45%)\">microbiome</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane, Praveen Kumar",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html": {
    "topic": "Proteomics",
    "title": "metaQuantome 3: Taxonomy",
    "description": null,
    "question": [
      "How do I look at differentially expressed taxa?",
      "Can we get top 5 or 10 taxa present in the sample?",
      "How can I visualize my results?"
    ],
    "objectives": [
      "A taxonomy and quantitational analysis of metaproteomic mass spectrometry data."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?microbiome\" style=\"background-color: hsl(26, 80.0%, 85%); color: #333; border: 1px solid hsl(26, 80.0%, 45%)\">microbiome</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane, Praveen Kumar",
    "level": "Intermediate",
    "time_estimation": "1h",
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-oms/tutorial.html": {
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using OpenMS tools",
    "description": null,
    "question": [
      "How to convert LC-MS/MS raw files?",
      "How to identify peptides?",
      "How to identify proteins?",
      "How to evaluate the results?"
    ],
    "objectives": [
      "Protein identification from LC-MS/MS raw files."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Advanced",
    "time_estimation": "45m",
    "url": "/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html": {
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using SearchGUI and PeptideShaker",
    "description": null,
    "question": [
      "How to convert LC-MS/MS raw files?",
      "How to identify peptides?",
      "How to identify proteins?",
      "How to evaluate the results?"
    ],
    "objectives": [
      "Protein identification from LC-MS/MS raw files."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Introductory",
    "time_estimation": "45m",
    "url": "/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-quant-sil/tutorial.html": {
    "topic": "Proteomics",
    "title": "Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
    "description": null,
    "question": [
      "What are MS1 features?",
      "How to quantify based on MS1 features?",
      "How to map MS1 features to MS2 identifications?",
      "How to evaluate and optimize the results?"
    ],
    "objectives": [
      "MS1 feature quantitation and mapping of quantitations to peptide and protein IDs."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SILAC\" href=\"/training-material/search?SILAC\" style=\"background-color: hsl(92, 80.0%, 85%); color: #333; border: 1px solid hsl(92, 80.0%, 45%)\">SILAC</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning, Matthias Fahrner",
    "level": "Advanced",
    "time_estimation": "1h",
    "url": "/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html"
  },
  "/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html": {
    "topic": "Proteomics",
    "title": "Peptide Library Data Analysis",
    "description": null,
    "question": [
      "How to utilize quantitative properties of amino acids and peptide sequence to analyse peptide data?"
    ],
    "objectives": [
      "Calculate descriptors",
      "Quantitative analysis of peptide sequence properties"
    ],
    "tags": [

    ],
    "contributors": "Jayadev Joshi, Daniel Blankenberg",
    "level": "Intermediate",
    "time_estimation": "20m",
    "url": "/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/database-handling/tutorial.html": {
    "topic": "Proteomics",
    "title": "Protein FASTA Database Handling",
    "description": null,
    "question": [
      "How to download protein FASTA databases of a certain organism?",
      "How to download a contaminant database?",
      "How to create a decoy database?",
      "How to combine databases?"
    ],
    "objectives": [
      "Creation of a protein FASTA database ready for use with database search algorithms."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?DDA\" style=\"background-color: hsl(205, 80.0%, 85%); color: #333; border: 1px solid hsl(205, 80.0%, 45%)\">DDA</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Introductory",
    "time_estimation": "30m",
    "url": "/training-material/topics/proteomics/tutorials/database-handling/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html": {
    "topic": "Proteomics",
    "title": "Proteogenomics 1: Database Creation",
    "description": null,
    "question": [
      "How to create a customized Protein Database from RNAseq data?"
    ],
    "objectives": [
      "Generating a customized Protein sequence database with variants, indels, splice junctions and known sequences."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?proteogenomics\" style=\"background-color: hsl(140, 80.0%, 85%); color: #333; border: 1px solid hsl(140, 80.0%, 45%)\">proteogenomics</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html": {
    "topic": "Proteomics",
    "title": "Proteogenomics 2: Database Search",
    "description": null,
    "question": [
      "How to identify variant proteoforms in MS data by searching with the customized Protein database?"
    ],
    "objectives": [
      "A proteogenomic data analysis of mass spectrometry data to identify and visualize variant peptides."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?proteogenomics\" style=\"background-color: hsl(140, 80.0%, 85%); color: #333; border: 1px solid hsl(140, 80.0%, 45%)\">proteogenomics</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "level": "Intermediate",
    "time_estimation": "15m",
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html": {
    "topic": "Proteomics",
    "title": "Proteogenomics 3: Novel peptide analysis",
    "description": null,
    "question": [
      "How to verify the spectra of novel proteoforms?",
      "How to assign genomic allocation to these novel proteoforms?"
    ],
    "objectives": [
      "How to assign and visualize the genomic localization of these identified novel proteoforms?"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?proteogenomics\" style=\"background-color: hsl(140, 80.0%, 85%); color: #333; border: 1px solid hsl(140, 80.0%, 45%)\">proteogenomics</a>"
    ],
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/secretome-prediction/tutorial.html": {
    "topic": "Proteomics",
    "title": "Secretome Prediction",
    "description": null,
    "question": [
      "How to predict cellular protein localization based upon GO-terms?",
      "How to combine multiple localization predictions?"
    ],
    "objectives": [
      "Predict proteins in the cellular secretome by using GO-terms.",
      "Predict proteins in the cellular secretome by using WolfPSORT.",
      "Combine the results of both predictions."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?human\" style=\"background-color: hsl(16, 80.0%, 85%); color: #333; border: 1px solid hsl(16, 80.0%, 45%)\">human</a>"
    ],
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "level": "Intermediate",
    "time_estimation": "30m",
    "url": "/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html": {
    "topic": "Proteomics",
    "title": "Statistical analysis of DIA data",
    "description": null,
    "question": [
      "How to perform statistical analysis on DIA mass spectrometry data?",
      "How to detect and quantify differentially abundant proteins in a HEK-Ecoli Benchmark DIA datatset?"
    ],
    "objectives": [
      "Statistical analysis of a HEK-Ecoli Benchmark DIA dataset",
      "Understanding statistical approaches in proteomic analysis",
      "Using MSstats to find significantly dysregulated proteins in a Spike-in dataset"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?DIA\" style=\"background-color: hsl(215, 80.0%, 85%); color: #333; border: 1px solid hsl(215, 80.0%, 45%)\">DIA</a>"
    ],
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "level": "Intermediate",
    "time_estimation": "1H",
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html"
  },
  "/topics/variant-analysis/slides/introduction.html": {
    "topic": "Variant Analysis",
    "title": "Introduction to Variant analysis",
    "description": null,
    "question": [

    ],
    "objectives": [

    ],
    "tags": [

    ],
    "contributors": "B√©r√©nice Batut, Yvan Le Bras",
    "level": null,
    "time_estimation": null,
    "url": "/training-material/topics/variant-analysis/slides/introduction.html"
  },
  "/topics/variant-analysis/tutorials/dip/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Calling variants in diploid systems",
    "description": null,
    "question": [
      "How to find variable sites in diploid genomes?"
    ],
    "objectives": [
      "Identification of genetic variations using variant calling"
    ],
    "tags": [

    ],
    "contributors": "Anton Nekrutenko, Nicola Soranzo",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/variant-analysis/tutorials/dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/non-dip/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Calling variants in non-diploid systems",
    "description": null,
    "question": [
      "How does frequency of mitochondrial polymorphisms change from mother to child?"
    ],
    "objectives": [
      "Using Galaxy's main site we will see how to call variants in bacteria, viruses, and organelles."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anton Nekrutenko, Alex Ostrovsky",
    "level": null,
    "time_estimation": "1h30m",
    "url": "/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/dunovo/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Calling very rare variants",
    "description": null,
    "question": [
      "What frequency of variants is so low that it is obscured by sequencing error rate?",
      "What are the different types of consensus sequences produced from duplex sequencing?"
    ],
    "objectives": [
      "Processing raw duplex sequencing data into consensus sequences",
      "Find rare variants without relying on diploid assumptions"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?work-in-progress\" style=\"background-color: hsl(246, 80.0%, 85%); color: #333; border: 1px solid hsl(246, 80.0%, 45%)\">work-in-progress</a>"
    ],
    "contributors": "Anton Nekrutenko, Nick Stoler",
    "level": null,
    "time_estimation": "3h",
    "url": "/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/exome-seq/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Exome sequencing data analysis for diagnosing a genetic disease",
    "description": null,
    "question": [
      "How do you identify genetic variants in samples based on exome sequencing data?",
      "How do you, among the set of detected variants, identify candidate causative variants for a given phenotype/disease?"
    ],
    "objectives": [
      "Jointly call variants and genotypes for a family trio from whole-exome sequencing data",
      "Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them"
    ],
    "tags": [

    ],
    "contributors": "Wolfgang Maier, B√©r√©nice Batut, Torsten Houwaart, Anika Erxleben, Bj√∂rn Gr√ºning",
    "level": null,
    "time_estimation": "5h",
    "url": "/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis",
    "description": null,
    "question": [
      "Learn how to get and use data from the Sequence Read Archive in Galaxy."
    ],
    "objectives": [
      "Understand how Galaxy and the Sequence Read Archive interact.",
      "Be able to go from Galaxy to the Short Reach Archive, query SRA, use the SRA Run Selector to send selected metadata to Galaxy, and then import sequence data from SRA into Galaxy."
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Marius van den Beek, Dave Clements, Daniel Blankenberg, Anton Nekrutenko",
    "level": null,
    "time_estimation": "45m",
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/somatic-variants/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Identification of somatic and germline variants from tumor and normal sample pairs",
    "description": null,
    "question": [
      "What are the specific challenges in somatic variant calling that set it apart from regular diploid variant calling?",
      "How can you call variants and classify them according to their presence/absence in/from tumor and normal tissue of the same individual?",
      "How can you annotate variants and affected genes with prior knowledge from human genetic and cancer-specific databases to generate clinically relevant reports?"
    ],
    "objectives": [
      "Call variants and their somatic status from whole-exome sequencing data",
      "Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases",
      "Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians"
    ],
    "tags": [

    ],
    "contributors": "Wolfgang Maier",
    "level": null,
    "time_estimation": "7h",
    "url": "/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "M. tuberculosis Variant Analysis",
    "description": null,
    "question": [
      "How do we detect differences between a set of reads from *M. tuberculosis* (Mtb) and the Mtb reference genome"
    ],
    "objectives": [
      "How should we filter those variants",
      "How can we predict drug resistance from those variants",
      "How do we annotate those variants"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Peter van Heusden, Simon Gladman, Thoba Lose",
    "level": "Intermediate",
    "time_estimation": "2h",
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Mapping and molecular identification of phenotype-causing mutations",
    "description": null,
    "question": [
      "What is mapping-by-sequencing?",
      "How can it help you identify the causative mutation in phenotypic mutants isolated from genetic screens?"
    ],
    "objectives": [
      "Use joint variant calling and extraction to facilitate variant comparison across samples",
      "Perform variant linkage analyses for phenotypically selected recombinant progeny",
      "Filter, annotate and report lists of variants"
    ],
    "tags": [

    ],
    "contributors": "Wolfgang Maier",
    "level": null,
    "time_estimation": "2h",
    "url": "/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/microbial-variants/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Microbial Variant Calling",
    "description": null,
    "question": [
      "How do we detect differences between a set of reads from a microorganism and a reference genome"
    ],
    "objectives": [
      "Find variants between a reference genome and a set of reads",
      "Visualise the SNP in context of the reads aligned to the genome",
      "Determine the effect of those variants on genomic features",
      "Understand if the SNP is potentially affecting the phenotype"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?prokaryote\" style=\"background-color: hsl(271, 80.0%, 85%); color: #333; border: 1px solid hsl(271, 80.0%, 45%)\">prokaryote</a>"
    ],
    "contributors": "Anna Syme, Simon Gladman, Torsten Seemann",
    "level": null,
    "time_estimation": "45m",
    "url": "/training-material/topics/variant-analysis/tutorials/microbial-variants/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html": {
    "topic": "Variant Analysis",
    "title": "Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data",
    "description": null,
    "question": [
      "How can we extract annotated allelic variants in SARS-Cov-2 sequences in Galaxy?",
      "Which tools and workflows can we use to identify SARS-CoV-2 lineages in Galaxy?"
    ],
    "objectives": [
      "Repeat SARS-CoV-2 data preparation",
      "Select and run workflow to extract annotated allelic variants from FASTQ files",
      "Run workflow to summarize and generate report for previously called allelic variants",
      "Interpret summaries for annotated allelic variants",
      "Run workflow to extract consensus sequences",
      "Select and run tools to assign clades/lineages"
    ],
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?covid19\" style=\"background-color: hsl(332, 80.0%, 85%); color: #333; border: 1px solid hsl(332, 80.0%, 45%)\">covid19</a>"
    ],
    "contributors": "Wolfgang Maier, B√©r√©nice Batut",
    "level": "Intermediate",
    "time_estimation": "3H",
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html"
  }
};

function search(idx, q){
	if(q.length > 2){
        var results_partial = idx.search(`*${q}*`),
            results_exact = idx.search(`${q}`),
            results_fuzzy = idx.search(`${q}~3`);

        thereMap  = Object.assign({}, ...results_partial.map((x) => ({[x.ref]: x.score})));

        results_exact.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score + 4){
                    thereMap[x.ref] = x.score + 4
                }
            } else {
                    thereMap[x.ref] = x.score + 4
            }
        })
        results_fuzzy.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score - 2){
                    thereMap[x.ref] = x.score - 2
                }
            } else {
                    thereMap[x.ref] = x.score - 2
            }
        })

        combined_results = Object.getOwnPropertyNames(thereMap);
        combined_results.sort((a, b) => {
            if (thereMap[a] > thereMap[b]) {
                return -1;
            }
            if (thereMap[a] < thereMap[b]) {
                return 1;
            }
            return 0;
        });

		var results_final = combined_results.map(x => {
			return tutorials['/' + x.replaceAll(".md", ".html")];
		}).filter(x => x !== undefined);

        $("#results-container").html(results_final.map(x => `
        <div class='col-sm-6'>
        <div class='card'>
        <div class='card-body'>
          <h5 class='card-title'>${x.title}</h5>
          <h6 class='card-subtitle text-muted'>${x.topic}</h6>
          <p>${x.tags.join(' ')}</p>
          <p>${x.contributors}</p>
          <a class='btn btn-primary' href='${x.url}'>View Tutorial</a>
          </div>
          </div>
          </div>
                    `));
	}
}

fetch('/training-material/search.json')
	.then(response => response.json())
	.then(data => {
		var idx = lunr.Index.load(data);

		var  params = (new URL(document.location)).searchParams;
		paramQuery = params.get('query');
		if(paramQuery){
			document.getElementById('search-input').value = paramQuery;
			search(idx, paramQuery);
		}

		$("#search-input").on("change keyup paste", function(){
			search(idx, $("#search-input").val());
		})
});
</script>



</section>

        </div>
        
            <footer>
    <div class="container">
        <br><br>
        <p>
            The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
            provides researchers with online training materials, connects them with local trainers, and helps promoting open data analysis practices worldwide.
        </p>
        <p>
The content of the tutorials is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>. The website and infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>.
</p>
        <p>This is revision <a href="https://github.com/galaxyproject/training-material/commit/86ba6f30c3c8e76d2a8f07c2a333a13a48cabfc5">86ba6f30c3c8e76d2a8f07c2a333a13a48cabfc5</a></p>
    </div>
</footer>

        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
