<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Introduction to Genome Annotation</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/introduction/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1706660815' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1706660815' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Introduction to Genome Annotation">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-01-31 00:26:57 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Helena Rasche">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Introduction to Genome Annotation">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1706660815'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/genome-annotation/tutorials/introduction/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/genome-annotation/tutorials/introduction/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/genome-annotation/tutorials/introduction/slides.html">View markdown source on GitHub</a>
	<h1>Introduction to Genome Annotation</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hexylena?s=36" alt="Avatar" width="36" height="36">Helena Rasche</a>
	</div>

</div>


		

		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 31, 2024 </div>

	<hr />


	<h1 id="genome-annotation">Genome Annotation</h1>

<p><strong>Structural Annotation</strong></p>

<p>Positions of genomic features along the genome</p>

<p><strong>Functional Annotation</strong></p>

<p>Assigning functions to those features</p>

<p><span>Speaker Notes</span></p>

<p>Two parts, structural and function. Structural can come from ab-initio predictions or structural data. Functional annotation often comes from analysis of protein domains or in rare cases from experimental data.</p>

<hr />

<h2 id="structural-annotation">Structural Annotation</h2>

<p>Types of elements:</p>

<ul>
  <li>genes</li>
  <li>regulatory regions</li>
  <li>ncRNA</li>
  <li>repeat elements</li>
  <li>pseudogenes and paralogs</li>
</ul>

<hr />

<h3 id="structural-annotation-why">Structural Annotation: Why?</h3>

<p><img src="../../images/intro-structural-annotation.png" alt="Screenshot of apollo, with genome sequence at the top, and several gene models shown below. These gene models are coding sequences + protein sequences." /></p>

<hr />

<h3 id="structural-annotation-why-1">Structural Annotation: Why?</h3>

<p>Locate your favorite gene + see what’s next to it</p>

<p>Basis for other analysis, e.g.:</p>

<ul>
  <li>Transcriptomic data (count reads mapping inside exons)</li>
  <li>Variants detection (SNP, indels, …) and their effects</li>
  <li>Epigenomic (ChIPSeq, FAIRESeq, …)</li>
</ul>

<p>Compare with other species</p>

<ul>
  <li>Presence/absence/mutations of genes</li>
  <li>Family reduction or expansion</li>
  <li>Structural variants</li>
</ul>

<hr />

<h3 id="prokaryotic-genes">Prokaryotic Genes</h3>

<div>
.pull-left[
Promoter:

- -35 Region
- TATA Box
- Initiation site (TSS)
]
.pull-right[
![Cartoon of a promoter region with a -35 region box reading TTGACA, a -10 region TATAAT, and that makes up the promoter. Then at +1 is the TSS or transcription start site.](../../images/intro-prok1.png)
]
</div>

<div>
.pull-left[
Operons:

- Promoter
- Some genes
- A terminator
]
.pull-right.image-90[
![A cartoon of gene with promoter, gene 1, gene 2, and gene 3 boxes before a terminator. This produces a polycistronic mRNA and that is cut up into protein 1, 2, and 3](../../images/intro-prok2.png)
]
</div>

<p><span>Speaker Notes</span></p>

<p>Prokaryotic genes often have a well conserved structure, with a promoter, one or a few genes and a terminator.</p>

<hr />

<h3 id="eukaryotic-genes">Eukaryotic Genes</h3>

<p><img src="../../images/intro-eukaryotic-genes.png" alt="A cartoon of a eukaryotic gene with a UTR, Exon 1, an intro, Exon 2, another intro, and Exon 3 before a final UTR shown along a line representing DNA. The process of Transcription extracts the region between and including UTRs, exons, and introns. Splicing produces the next product which is just the UTRs and 3 exons, before translation into a protein." /></p>

<p><span>Speaker Notes</span></p>

<p>Things are a little more complicated for eukaryotic: splicing</p>

<hr />

<h2 id="automatic-structural-annotation">Automatic Structural Annotation</h2>

<p>Very difficult problem</p>

<ul>
  <li>Short, variable, unspecific motifs</li>
  <li>Need data to support predictions</li>
</ul>

<p><img src="../../images/intro-tss.png" alt="The previous cartoon of a eukaryotic gene with TSS pointing to the UTR, the ATG codon at exon 1, the three exons, and a final UTR. The letters GT appear at the end of exons 1 and 2, and AG appear at the beginning of Exon 2 and 3, indicating where it will be spliced." /></p>

<hr />

<h3 id="evidence">Evidence</h3>

<p>Multiple pieces of evidence</p>

<ul>
  <li>Alignment of RNASeq reads</li>
  <li>Alignment of EST or transcripts (same species or closely related species)</li>
  <li>Alignment of proteins (closely related species)</li>
</ul>

<p><img src="../../images/intro-evidences.png" alt="Screenshot of genome browser with a new gene at the top, a row with protein/mRNA from other species used as evidence, and a final row with RNASeq coverage indicating mapped reads." /></p>

<p><em>But</em> data unavailable for novel or very distant genes, or unexpressed genes</p>

<hr />

<h3 id="ab-initio-gene-calling"><em>Ab initio</em> Gene Calling</h3>

<p>.pull-left[
Predictions using:</p>
<ul>
  <li>Genome sequence</li>
  <li>Statistical model (specific to organism)</li>
</ul>

<p>Models:</p>
<ul>
  <li>Training on the best evidence-based gene calls</li>
  <li>“Best” = strong evidence, highly conserved</li>
  <li>Training can be iterative:
    <ul>
      <li>train, predict, select best genes, retrain, etc
]</li>
    </ul>
  </li>
</ul>

<p>.pull-right.image-90[
<img src="../../images/intro-ab-initio.png" alt="Cartoon of ab initio prediction with several rows, genome, coding potential, atg and stop codons, splice sites, then these rows are duplicated for the reverse strand. Examples are listed like genefinder, augustus, glimmer, snap, and fgenesh." />
]</p>

<hr />

<h3 id="data-reconcilliation">Data Reconcilliation</h3>

<p>.pull-left[</p>

<ul>
  <li>Integration of evidence and <em>ab initio</em> predictions</li>
  <li>“Consensus” of multiple sources</li>
  <li>Automated pipelines
    <ul>
      <li><a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html"><strong>Maker</strong></a>, <strong>Braker</strong>, <a href="/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html"><strong>Funannotate</strong></a>, <strong>Pasa</strong>, <strong>Prokka</strong>, …</li>
      <li>Align evidences (or use pre-aligned)</li>
      <li>Run <em>ab initio</em> predictors</li>
      <li>Reconciliate gene models
]</li>
    </ul>
  </li>
</ul>

<p>.pull-right.image-90[
<img src="../../images/intro-consensus.png" alt="Multiple gene models are shown labelled current evidence, which is complemented by three more gene models from ab initio prediction. Below a final gene model is constructed and labelled current assembly." />
]</p>

<blockquote>
  <blockquote>
    <blockquote>
      <blockquote>
        <blockquote>
          <blockquote>
            <blockquote>
              <p>7259342d100 (Remove slides folder and move introduction decks):topics/genome-annotation/tutorials/introduction/slides.html
<small>
Source: <a href="http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/MAKER_Tutorial_for_WGS_Assembly_and_Annotation_Winter_School_2018">Maker documentation</a>
</small>
]</p>
            </blockquote>
          </blockquote>
        </blockquote>
      </blockquote>
    </blockquote>
  </blockquote>
</blockquote>

<hr />

<h3 id="evaluation-of-annotation-metrics">Evaluation of annotation: metrics</h3>

<ul>
  <li>Number of genes</li>
  <li>Average number of exons</li>
  <li>Average gene length</li>
  <li>Average protein length</li>
  <li>…</li>
</ul>

<hr />

<h3 id="evaluation-of-annotation-busco">Evaluation of annotation: BUSCO</h3>

<h4 id="benchmarking-universal-single-copy-orthologs">Benchmarking Universal Single-Copy Orthologs</h4>

<ul>
  <li>Sets of genes having single-copy orthologs in all species of a clade (insects, plants, bacteria, …)</li>
  <li>Genes supposed to be vital for the species</li>
  <li>Expected to be found in a good annotation</li>
  <li>Results:
    <ul>
      <li>Found genes</li>
      <li>Fragmented genes</li>
      <li>Duplicated genes</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="visualisation-of-results">Visualisation of Results</h3>

<p>Genome Browsers (JBrowse, UCSC, …)</p>

<p><img src="../../images/intro-viz.png" alt="Screenshot of Apollo showing a track list on left, and then several rows of evidence like maker annotation and blastn showing gene models, followed by my bam file." /></p>

<p><small>
<strong>JBrowse</strong>: <a href="/training-material/topics/visualisation/tutorials/jbrowse/slides.html">Slides</a> - <a href="/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html">Tutorial</a>
</small></p>

<hr />

<h3 id="repeated-elements">Repeated Elements</h3>

<ul>
  <li>Transposons, Retrotransposons, low complexity regions</li>
  <li>Disrupt gene calling</li>
  <li>Prediction pipelines:
    <ul>
      <li><a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">RepeatMasker</a></li>
      <li>RepeatModeler</li>
      <li>REPET</li>
    </ul>
  </li>
  <li>Databases of repeated elements
    <ul>
      <li>Can be used by pipelines</li>
      <li>Dfam</li>
      <li>RepBase (non free)</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="exotic-elements">Exotic Elements</h3>

<ul>
  <li>tRNA, rRNA, ncRNA, …</li>
  <li>Dedicated tools for prediction
    <ul>
      <li>Aragorn</li>
      <li>tRNAscan</li>
      <li>…</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="summary">Summary</h3>

<ul>
  <li>Difficult problem</li>
  <li>Automated pipelines</li>
  <li>Need for evidences</li>
  <li>Never perfect
    <ul>
      <li>Missing/incomplete genes</li>
      <li>Split/fused genes</li>
      <li>Pseudogenes</li>
    </ul>
  </li>
</ul>

<hr />

<h2 id="manual-annotation">Manual Annotation</h2>

<ul>
  <li>Recruit experts of some gene families</li>
  <li>Manual curation of their favorite genes
    <ul>
      <li>Better annotation</li>
      <li>Things to say in the genome paper</li>
    </ul>
  </li>
  <li>Limits
    <ul>
      <li>There aren’t experts for all genes</li>
      <li>They can only annotate what is in the sequence</li>
      <li>Poor assembly ⇒ Poor annotation</li>
      <li>We need a user-friendly environment</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="editors">Editors</h3>

<p>Apollo (based upon JBrowse), Artemis, others</p>

<p><img src="../../images/intro-apollo.png" alt="Screenshot of apollo with dozens of tracks in the list. Several gene model predictions are shown as well as a few bigwig XY density from illumina sequencing data." /></p>

<p>Check out the Apollo tutorials for more details: <a href="/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html">prokaryotes</a> - <a href="/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html">eukaryotes</a></p>

<hr />

<h3 id="steps">Steps</h3>

<p>.pull-left[
Annotations steps</p>

<ul>
  <li>Check structure (exons, introns, start, stop, utr, …)</li>
  <li>Search for isoforms</li>
  <li>Ensure consistent naming conventions</li>
  <li>Add functional annotations (based on homologies with other species)
]</li>
</ul>

<p>.pull-right[
<img src="../../images/intro-steps.png" alt="A report showing 15 errors like gene having no alternate alleles or missing a symbol, or missing a human readable name. 2 warnings about model structures are shown, and the other 119 genes are shown as valid." />
]</p>

<hr />

<h2 id="functional-annotation">Functional Annotation</h2>

<p>Collection of information on the function of identified genes</p>
<ul>
  <li>Biological function</li>
  <li>Regulation, expression, …</li>
</ul>

<p>Data Sources</p>
<ul>
  <li>Wet lab experiments (reliable but long and expensive)</li>
  <li>Manual assignment (cf Apollo)</li>
  <li><strong>Automatic assignment</strong></li>
</ul>

<hr />

<h3 id="methods">Methods</h3>

<ul>
  <li>Similarity search / homology</li>
  <li>Pattern search</li>
  <li>Orthologies</li>
  <li>Comparison against databases:
    <ul>
      <li>GenBank, NR: sequence databanks</li>
      <li>InterPro: pattern databank (active sites, protein families, peptide signal …)</li>
      <li>EggNOG: databank of orthology relationships + functional annotation</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="blast">Blast</h3>

<ul>
  <li>Blast/Diamond against NR
    <ul>
      <li>For each protein (or CDS) of the annotation</li>
      <li>Find the best xx hits</li>
    </ul>
  </li>
  <li>Huge database, good chances to have a match</li>
  <li>Risk:
    <ul>
      <li>Spread of “putative xx protein”</li>
      <li>Spread of low-evidence annotations</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="interproscan">InterProScan</h3>

<ul>
  <li>For each protein (or CDS) of the annotation
    <ul>
      <li>Search for all InterPro patterns</li>
    </ul>
  </li>
  <li>Many motifs</li>
  <li>Some of them manually curated</li>
  <li>Gene Ontology Terms available for domains</li>
</ul>

<hr />

<h3 id="eggnog">EggNOG</h3>

<ul>
  <li>EggNOG: &gt;4M known orthology groups in &gt;5000 organisms</li>
  <li>Tool using it: <strong>EggNOG-mapper</strong></li>
  <li>For each protein of the annotation
    <ul>
      <li>Search for matches with known orthology groups</li>
    </ul>
  </li>
  <li>Assign corresponding gene name and functional annotation</li>
</ul>

<hr />

<h3 id="gene-ontology">Gene Ontology</h3>

<p>.pull-left[
Controlled vocabulary to describe:</p>
<ul>
  <li>molecular function</li>
  <li>biological process</li>
  <li>cellular component
    <ul>
      <li>e.g.: <code class="language-plaintext highlighter-rouge">GO:0044430</code> = cytoskeletal part</li>
    </ul>
  </li>
</ul>

<p><img src="../../images/intro-go.png" alt="A tree browser showing the top level groups (MF, CC, BP) followed by a drill down into molecular function with several groups like acetylcholine receptor activator activity, inhibitor, and regulator activity. antioxidant activity, binding, etc. Each group shows large numbers of children." />
]</p>

<p>.pull-right[
<img src="../../images/intro-go-source.png" alt="A stacked bar chart with species along the bottom and the number of annotations. The split is experimental and  non-experimental annotations which is split ~15% experimental/85% non, for most species. Human has the highest with 200k experimental/250k non-experimental." />
]</p>

<hr />

<h3 id="gene-ontology-1">Gene Ontology</h3>

<p>Various sources</p>

<ul>
  <li>InterProScan
    <ul>
      <li>Assigned from matches with annotated motifs</li>
    </ul>
  </li>
  <li>EggNOG-mapper
    <ul>
      <li>Assigned from matches with annotated orthology groups</li>
    </ul>
  </li>
  <li>Blast2GO
    <ul>
      <li>Based on Blast (and InterProScan) results</li>
      <li>For each protein (or CDS) of the annotation, tag with GO terms</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="orthology">Orthology</h3>

<ul>
  <li>For each annotated gene
    <ul>
      <li>Search of orthologous genes in related species</li>
      <li>Search for paralogues</li>
    </ul>
  </li>
  <li>Bioinformatics method:
    <ul>
      <li>Blast all against all transcripts</li>
      <li>Filtering the best hits</li>
      <li>Clustering</li>
      <li>OrthoFinder, OrthoMCL, …</li>
    </ul>
  </li>
</ul>

<p><img src="../../images/intro-ortho.png" alt="Diagram of orthology with species a and species B as two circles. Inside A is a1, 2, and 3 labelled as in-paralogs. Inside B is b1/b2 also in-paralogs. A1 and B1 are orthologs, and the rest of the potential connections between A and B are labelled co-orthologs." /></p>

<hr />

<h3 id="visualisation">Visualisation</h3>

<ul>
  <li>Genomic databases (NCBI, FlyBase, etc.)</li>
  <li>Other sites (Tripal sites)
    <ul>
      <li>reference data (assembly, annotation, …)</li>
      <li>interfaces to visualize this data</li>
      <li>interfaces for querying (e.g. <a href="https://bipaa.genouest.org">bipaa.genouest.org</a>)</li>
    </ul>
  </li>
</ul>

<hr />

<h3 id="comparing-annotations">Comparing annotations</h3>

<ul>
  <li>Needed to choose between different results on a same genome sequence</li>
  <li>Compare general statistics
    <ul>
      <li>Number of genes</li>
      <li>Average number of exons</li>
      <li>Average gene length</li>
      <li>Average protein length</li>
      <li>…</li>
    </ul>
  </li>
  <li>Compare gene content
    <ul>
      <li>Alignment of gene structures</li>
      <li>Functional annotation
        <ul>
          <li>over/underrepresentation of functions</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>Tools: <strong>AEGeAN</strong>, <strong>Funannotate compare</strong></li>
</ul>

<hr />

<h2 id="genome-annotation-1">Genome Annotation</h2>

<ul>
  <li>Difficult problem</li>
  <li>Automatic:
    <ul>
      <li>Structural:</li>
      <li>EST/RNA-Seq data provides good evidence
        <ul>
          <li><em>Ab initio</em> methods are improving</li>
        </ul>
      </li>
      <li>Functional:
        <ul>
          <li>Concrete evidence cost-prohibitive to obtain</li>
        </ul>
      </li>
      <li>Various sources of automatic assignment
        <ul>
          <li>Risks of automatically spreading “putative” annotations</li>
        </ul>
      </li>
    </ul>
  </li>
  <li>Manual:
    <ul>
      <li>Slow</li>
      <li>Requires experts and evidences</li>
    </ul>
  </li>
</ul>


	<hr />








<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/genome-annotation/tutorials/introduction/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1706660815'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
