<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Reference Data with CVMFS</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Resources related to configuration and maintenance of Gal..." />
        <meta property="og:title" content="Galaxy Training: Reference Data with CVMFS" />
        <meta property="og:description" content="Resources related to configuration and maintenance of Gal..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/admin" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Galaxy Server administration
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/admin/tutorials/cvmfs/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/admin/faqs/" title="Check our FAQs for the Galaxy Server administration topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/admin/tutorials/cvmfs/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/admin/tutorials/cvmfs/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'galaxyproject/admins'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Reference Data with CVMFS",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Galaxy Server administration",
    "description": "Resources related to configuration and maintenance of Galaxy servers",
    "url": "https://training.galaxyproject.org//training-material/topics/admin/"
  },
  "courseCode": "admin / cvmfs / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Reference Data with CVMFS' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/admin/tutorials/cvmfs/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The objectives are:\n - Have an understanding of what CVMFS is and how it works\n - Install and configure the CVMFS client on a linux machine and mount the Galaxy reference data repository\n - Configure your Galaxy to use these reference genomes and indices\n - Use an Ansible playbook for all of the above.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/admin/tutorials/ansible/slides.html",
      "name": "Ansible",
      "description": "Slides for 'Ansible' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/admin/tutorials/ansible/tutorial.html",
      "name": "Ansible",
      "description": "Hands-on for 'Ansible' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/admin/tutorials/ansible-galaxy/slides.html",
      "name": "Galaxy Installation with Ansible",
      "description": "Slides for 'Galaxy Installation with Ansible' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html",
      "name": "Galaxy Installation with Ansible",
      "description": "Hands-on for 'Galaxy Installation with Ansible' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Simon Gladman"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Simon Gladman"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Server administration",
      "description": "Resources related to configuration and maintenance of Galaxy servers",
      "url": "https://training.galaxyproject.org//training-material/topics/admin/"
    }
  ]
}
</script>

<section class="tutorial topic-admin">
    <h1 data-toc-skip>Reference Data with CVMFS</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/slugger70/" class="contributor-badge contributor-slugger70"><img src="https://avatars.githubusercontent.com/slugger70?s=27" alt="Avatar">Simon Gladman</a>


<a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="https://avatars.githubusercontent.com/hexylena?s=27" alt="Avatar">Helena Rasche</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Have an understanding of what CVMFS is and how it works</p>
</li>
        
        <li><p>Install and configure the CVMFS client on a linux machine and mount the Galaxy reference data repository</p>
</li>
        
        <li><p>Configure your Galaxy to use these reference genomes and indices</p>
</li>
        
        <li><p>Use an Ansible playbook for all of the above.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
        
    <li>
    
        
        
        <a href="/training-material/topics/admin">Galaxy Server administration</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Ansible:
                            
                                <a href="/training-material/topics/admin/tutorials/ansible/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/admin/tutorials/ansible/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Galaxy Installation with Ansible:
                            
                                <a href="/training-material/topics/admin/tutorials/ansible-galaxy/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/admin/tutorials/cvmfs/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            

            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/admin/cvmfs/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
          
          
          

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/admin/cvmfs/slides" title="View GTN Video Library for these Slides">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture (February 2021)
            </a>
        </li>
        
        
    </ul>
   </li>
   



            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 22, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <blockquote class="spoken" data-visual="gtn" data-target="#top-navbar">
  <p>These words come from a transcript of Simon Gladman teaching this course. He
is a bioinformatician at the University of Melbourne in Australia and also
one of the administrators of <span class="notranslate">Galaxy</span> Australia.</p>

  <p>Hello everybody and welcome back to the <span class="notranslate">Galaxy</span>
administrators course. in this session we‚Äôre going to be talking about
reference data in <span class="notranslate">Galaxy</span> using CVMFS.</p>

  <p>Hopefully by now you‚Äôve all seen the video of
the slides - if not I suggest you go and do
look at those first as that explains how reference
data works in <span class="notranslate">Galaxy</span> and also what CVMFS does to
help us with that reference data.</p>

  <p>Some of the requirements from that we hope you‚Äôve
already completed: <span class="notranslate">Galaxy</span> server installation
with Ansible and hopefully you
understand what Ansible is and how
it‚Äôs used.</p>
</blockquote>

<h1 class="no_toc" id="overview">Overview</h1>

<p>The CernVM-FS is a distributed filesystem perfectly designed for sharing readonly data across the globe. We use it in the <a href="https://galaxyproject.org"><span class="notranslate">Galaxy</span> Project</a> for sharing things that a lot of <span class="notranslate">Galaxy</span> servers need. Namely:</p>
<ul>
  <li><strong>Reference Data</strong>
    <ul>
      <li>Genome sequences for hundreds of useful species.</li>
      <li>Indices for the genome sequences</li>
      <li>Various bioinformatic tool indices for the available genomes</li>
    </ul>
  </li>
  <li><strong>Tool containers</strong>
    <ul>
      <li><a href="https://www.sylabs.io/">Singularity</a> containers of everything stored in <a href="https://biocontainers.pro/">Biocontainers</a> (A bioinformatic tool container repository.) You get these for free every time you build a <a href="https://bioconda.github.io/">Bioconda</a> recipe/package for a tool.</li>
    </ul>
  </li>
  <li>Others too..</li>
</ul>

<p>From the Cern website:</p>

<blockquote class="quote" id="cvmfs-quote">
  <p>The CernVM File System provides a scalable, reliable and low-maintenance software distribution service. It was developed to assist High Energy Physics (HEP) collaborations to deploy software on the worldwide-distributed computing infrastructure used to run data processing applications. CernVM-FS is implemented as a POSIX read-only file system in user space (a FUSE module). Files and directories are hosted on standard web servers and mounted in the universal namespace /cvmfs.‚Äù</p>

  <p>‚Äì <a href="https://cernvm.cern.ch/portal/filesystem">https://cernvm.cern.ch/portal/filesystem</a></p>
</blockquote>

<blockquote class="spoken" data-visual="gtn" data-target="#cvmfs-quote">
  <p>All right, just a quick recap. CVMFS or
Cern-VMFS is a distributed file system perfectly
designed for sharing read-only data across the
globe and we use it extensively in the <span class="notranslate">Galaxy</span>
project for sharing things that
a lot of <span class="notranslate">Galaxy</span> servers need.
Namely all the reference data; so the genome
sequences for all the different genomes that
we need to think about in in <span class="notranslate">Galaxy</span> and
bioinformatics. Things like a human genome,
mouse genome, etc., etc. And we need a lot of
indices for the genome sequences and lots of
tool indices for all those genomes. And we also
have a repository that contains tool containers.
So singularity containers for all of the
bioinformatics tools that we might want to use.
There‚Äôs a tutorial idea already done this week
or coming up soon this week that explains how to
use those singularity containers within <span class="notranslate">Galaxy</span> as
well. So we‚Äôre just going to get on with things.</p>
</blockquote>

<p>A slideshow presentation on this subject can be found <a href="slides.html">here</a>. More details on the use<span class="notranslate">galaxy</span>.org (<span class="notranslate">Galaxy</span> Main‚Äôs) reference data setup and CVMFS system can be found <a href="https://galaxyproject.org/admin/reference-data-repo/#usegalaxyorg-reference-data">here</a></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

<ol id="markdown-toc">
  <li><a href="#ansible-cvmfs-and-galaxy" id="markdown-toc-ansible-cvmfs-and-galaxy">Ansible-CVMFS and <span class="notranslate">Galaxy</span></a>    <ol>
      <li><a href="#installing-and-configuring" id="markdown-toc-installing-and-configuring">Installing and Configuring</a></li>
      <li><a href="#exploring-the-cvmfs-installation" id="markdown-toc-exploring-the-cvmfs-installation">Exploring the CVMFS Installation</a></li>
      <li><a href="#configuring-galaxy-to-use-the-cvmfs-references" id="markdown-toc-configuring-galaxy-to-use-the-cvmfs-references">Configuring <span class="notranslate">Galaxy</span> to use the CVMFS references.</a></li>
    </ol>
  </li>
  <li><a href="#common-production-questions" id="markdown-toc-common-production-questions">Common Production Questions</a></li>
  <li><a href="#other-aspects" id="markdown-toc-other-aspects">Other Aspects</a>    <ol>
      <li><a href="#development" id="markdown-toc-development">Development</a></li>
      <li><a href="#automation" id="markdown-toc-automation">Automation</a></li>
      <li><a href="#access-control" id="markdown-toc-access-control">Access Control</a></li>
      <li><a href="#proxying-recap" id="markdown-toc-proxying-recap">Proxying Recap</a></li>
      <li><a href="#plant-data" id="markdown-toc-plant-data">Plant Data</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<blockquote class="spoken" data-visual="gtn" data-target="#agenda">
  <p>The agenda we‚Äôre going to follow today is: We‚Äôre going to install and
configure <span class="notranslate">Galaxy</span> CVMFS reference data using ansible. We‚Äôre going to explore
the CVMFS installation and then we‚Äôre going to configure <span class="notranslate">Galaxy</span> to use it.</p>
</blockquote>

<h1 id="ansible-cvmfs-and-galaxy">Ansible-CVMFS and <span class="notranslate">Galaxy</span></h1>

<blockquote class="spoken" data-visual="gtn" data-target="#ansible-cvmfs-and-galaxy">
  <p>There are a few different repositories that um <span class="notranslate">Galaxy</span> project has created and
shared with everybody. The first one is the reference data and indices and
this is in data.<span class="notranslate">galaxy</span>project.org. And then we have another one called
singularity.<span class="notranslate">galaxy</span>project.org.</p>

  <p>If you want to have a look at what‚Äôs in there you can click on the ‚Äúdata
cache‚Äù, and this link here will show you all the things that are inside this
data.<span class="notranslate">galaxy</span>project.org</p>
</blockquote>

<p>The <span class="notranslate">Galaxy</span> project supports a few CVMFS repositories.</p>

<table>
  <thead>
    <tr>
      <th>Repository</th>
      <th>Repository Address</th>
      <th>Contents</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Reference Data and Indices</td>
      <td><code class="language-plaintext highlighter-rouge">data.<span class="notranslate">galaxy</span>project.org</code></td>
      <td>Genome sequences and their tool indices, <span class="notranslate">Galaxy</span> <code class="language-plaintext highlighter-rouge">.loc</code> files for them as well</td>
    </tr>
    <tr>
      <td>Singularity Containers</td>
      <td><code class="language-plaintext highlighter-rouge">singularity.<span class="notranslate">galaxy</span>project.org</code></td>
      <td>Singularity containers for everything in Biocontainers for use in <span class="notranslate">Galaxy</span> systems</td>
    </tr>
    <tr>
      <td><span class="notranslate">Galaxy</span> Main Configuration</td>
      <td><code class="language-plaintext highlighter-rouge">main.<span class="notranslate">galaxy</span>project.org</code></td>
      <td>The configuration files etc for <span class="notranslate">Galaxy</span> Main (use<span class="notranslate">galaxy</span>.org)</td>
    </tr>
  </tbody>
</table>

<p>You can browse the contents of <code class="language-plaintext highlighter-rouge">data.<span class="notranslate">galaxy</span>project.org</code> at the <a href="http://datacache.galaxyproject.org/">datacache</a>.</p>

<h2 id="installing-and-configuring">Installing and Configuring</h2>

<p>Luckily for us, the <span class="notranslate">Galaxy</span> Project has a lot of experience with using and configuring CVMFS and we are going to leverage off that. To get CVMFS working on our <span class="notranslate">Galaxy</span> server, we will use the Ansible role for CVMFS written by the <span class="notranslate">Galaxy</span> Project. Firstly, we need to install the role and then write a playbook for using it.</p>

<p>If the terms ‚ÄúAnsible‚Äù, ‚Äúrole‚Äù and ‚Äúplaybook‚Äù mean nothing to you, please checkout <a href="/training-material/topics/admin/tutorials/ansible/slides.html">the Ansible introduction slides</a> and <a href="/training-material/topics/admin/tutorials/ansible/tutorial.html">the Ansible introduction tutorial</a></p>

<!--SNIPPET-->
<blockquote class="tip">  <h3 data-toc-skip="" id="-tip-running-ansible-on-your-remote-machine"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running Ansible on your remote machine</h3>  <p>It is possible to have ansible installed on the remote machine and run it there, not just from your local machine connecting to the remote machine.</p>  <p>Your hosts file will need to use <code class="language-plaintext highlighter-rouge">localhost</code>,  and whenever you run playbooks with <code class="language-plaintext highlighter-rouge">ansible-playbook -i hosts playbook.yml</code>, you will need to add <code class="language-plaintext highlighter-rouge">-c local</code> to your command.</p>  <p>Be <strong>certain</strong> that the playbook that you‚Äôre writing on the remote machine is stored somewhere safe, like your user home directory, or backed up on your local machine. The cloud can be unreliable and things can disappear at any time.</p></blockquote>

<blockquote class="spoken" data-visual="gtn" data-target="#installing-and-configuring">
  <p>Okay, so now we‚Äôre going to move on to installing and configuring <span class="notranslate">Galaxy</span>‚Äôs
CVMFS reference data with Ansible. We are going to do some Ansible here and
we‚Äôre going to install CVMFS onto our <span class="notranslate">Galaxy</span> server. Hopefully you all have
access to a <span class="notranslate">Galaxy</span> server.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target="/" data-action="goto">
  <p>Here is mine. And on it I am an admin user and I have access to the admin
page. This was done as part of the installation of <span class="notranslate">Galaxy</span> um and hopefully
you‚Äôve installed a tool.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target=".search-input input" data-action="fill" data-value="bwa-mem">
  <p>I have bwa and bwa-mem installed under <span class="notranslate">Mapping</span>.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target="/?tool_id=bwa_mem" data-action="goto">
  <p>So if I click on that you can see
here that I have bwa-mem but there are no options available for reference
genomes, so we want to fix that. We want to connect to all of <span class="notranslate">Galaxy</span>‚Äôs
pre-built references and so we‚Äôre going to use <span class="notranslate">Galaxy</span>‚Äôs CVMFS system to let
our own Galaxies connect and get access to all the pre-built caches and
everything we already have.</p>
</blockquote>

<blockquote class="spoken" data-visual="gtn" data-target="#hands_on-hands-on-installing-cvmfs-with-ansible">
  <p>Okay, so let‚Äôs get started. If we go back to our
tutorial here, it says that we need to install a CVMFS role into our
requirements.yml and then add it to our Ansible.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-installing-cvmfs-with-ansible"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Installing CVMFS with Ansible</h3>

  <ol>
    <li>
      <p>In your working directory, add the CVMFS role to your <code class="language-plaintext highlighter-rouge">requirements.yml</code></p>

      <blockquote class="spoken" data-visual="terminal" data-cmd="whoami; hostname -f">
        <p>Well the first thing I need to do is log into my <span class="notranslate">Galaxy</span> machine in the
terminal.</p>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-cmd="ls -al">
        <p>Have a look at he contents of this directory and I‚Äôll go into <span class="notranslate">galaxy</span> and here we have all of the Ansible scripts that hopefully everybody already has.</p>
      </blockquote>

      <div data-commit="Add requirement" data-ref="add-req" class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gd">--- a/requirements.yml
</span><span class="gi">+++ b/requirements.yml
</span><span class="p">@@ -16,3 +16,5 @@</span>
   version: 048c4f178077d05c1e67ae8d9893809aac9ab3b7
 - src: gantsign.golang
   version: 2.6.3
<span class="gi">+- src: <span class="notranslate">galaxy</span>project.cvmfs
+  version: 0.2.13
</span>   
</code></pre></div>      </div>

      <blockquote class="spoken" data-visual="terminal" data-ref="add-req">
        <p>Okay, so the first thing I‚Äôm going to do is I‚Äôm going to add the CVMFS
role to the requirements.yml.
Edit requirements.yml and we need to add this to the bottom of that file. Copy. Paste. And save it.</p>
      </blockquote>
    </li>
    <li>
      <p>Install the role with:</p>

      <blockquote class="code-in">
        <h3 id="code-in-input-bash"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Input: Bash</h3>
        <div data-cmd="true" data-ref="req-install" class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ansible-<span class="notranslate">galaxy</span> <span class="nb">install</span> <span class="nt">-p</span> roles <span class="nt">-r</span> requirements.yml
</code></pre></div>        </div>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-ref="req-install">
        <p>And now install the role into our local Ansible scripts using the
ansible-<span class="notranslate">galaxy</span> command. And as you can see, it‚Äôs downloading the CVMFS
role.</p>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-cmd="ls roles/">
        <p>And if we look into roles now you can see that we have <span class="notranslate">galaxy</span>project.cmfs</p>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-cmd="clear">
        <p>Right, clear the screen.</p>
      </blockquote>
    </li>
    <li>
      <p>The variables available in this role are:</p>

      <blockquote class="spoken" data-visual="gtn" data-target="#variables-table">
        <p>Okay, now what we need to do is we need to run this role and that will
install the CVMFS client onto our <span class="notranslate">Galaxy</span> server. So the first thing we
need to do is edit our group files <span class="notranslate">galaxy</span>servers file. There are a bunch
of different variables that we can set. We can set the cvmfs role to be
a client or stratum zero, stratum one server. URLs - so where we‚Äôre
getting all this data from, and if you remember in the slideshow, the
data can come from Europe or America or Australia. Which repositories
we want to have installed. And this is an important one - the quota
limit. This is basically saying that CVMFS will cache some data on your
local machine and the quota limit is the maximum size of that cache.</p>
      </blockquote>

      <table id="variables-table">
        <thead>
          <tr>
            <th>Variable</th>
            <th>Type</th>
            <th>Description</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td><code class="language-plaintext highlighter-rouge">cvmfs_role</code></td>
            <td>string</td>
            <td>Type of CVMFS host: <code class="language-plaintext highlighter-rouge">client</code>, <code class="language-plaintext highlighter-rouge">stratum0</code>, <code class="language-plaintext highlighter-rouge">stratum1</code>, or <code class="language-plaintext highlighter-rouge">localproxy</code>. Controls what packages are installed and what configuration is performed.</td>
          </tr>
          <tr>
            <td><code class="language-plaintext highlighter-rouge">cvmfs_keys</code></td>
            <td>list of dicts</td>
            <td>Keys to install on hosts of all types.</td>
          </tr>
          <tr>
            <td><code class="language-plaintext highlighter-rouge">cvmfs_server_urls</code></td>
            <td>list of dicts</td>
            <td>CVMFS server URLs, the value of <code class="language-plaintext highlighter-rouge">CVMFS_SERVER_URL</code> in <code class="language-plaintext highlighter-rouge">/etc/cvmfs/domain.d/&lt;domain&gt;.conf</code>.</td>
          </tr>
          <tr>
            <td><code class="language-plaintext highlighter-rouge">cvmfs_repositories</code></td>
            <td>list of dicts</td>
            <td>CVMFS repository configurations, <code class="language-plaintext highlighter-rouge">CVMFS_REPOSITORIES</code> in <code class="language-plaintext highlighter-rouge">/etc/cvmfs/default.local</code> plus additional settings in <code class="language-plaintext highlighter-rouge">/etc/cvmfs/repositories.d/&lt;repository&gt;/{client,server}.conf</code>.</td>
          </tr>
          <tr>
            <td><code class="language-plaintext highlighter-rouge">cvmfs_quota_limit</code></td>
            <td>integer in MB</td>
            <td>Size of CVMFS client cache. Default is <code class="language-plaintext highlighter-rouge">4000</code>.</td>
          </tr>
        </tbody>
      </table>

      <p>But, luckily for us, the <span class="notranslate">Galaxy</span> Project CVMFS role has a lot of defaults for these variables which we can use by just setting <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span>_cvmfs_repos_enabled</code> to <code class="language-plaintext highlighter-rouge">config-repo</code>. We will also set the <code class="language-plaintext highlighter-rouge">cvmfs_quota_limit</code> to something sensible (500MB) as we have relatively small disks on our instances. In a production setup, you should size this appropriately for the client.</p>

      <blockquote class="tip">
        <h3 id="tip-what-is-a-good-size-for-this"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> What is a good size for this?</h3>
        <p>In production Use<span class="notranslate">Galaxy</span>.org.au uses 100GB, different sites have different needs and you can make your cache smaller depending on your usage. E.g. if your users only use one dataset from the reference data (e.g. just hg38) then perhaps you don‚Äôt need such a large cache.</p>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-cmd="ls group_vars/">
        <p>Okay but instead of just modifying <span class="notranslate">galaxy</span>servers.yml and adding in some
of these variables - instead what we‚Äôre going to do here is - we‚Äôre
going to create a new group file called all.yml.</p>

        <p>Because one of the things that we may want to do in the future - is we
may want to create other machines. We might want to have worker nodes
for our <span class="notranslate">Galaxy</span> cluster; or we may want to have other machines that we
want to be able to create using these Ansible scripts that also have the
CVMFS role installed. And instead of reproducing these variables in each
of the group var files for those particular machines, we can create a
special group vars file called all.yml. And whatever we put in there
will be automatically available to all machines that we create with
Ansible from this directory. Hopefully that makes a bit of sense and
we‚Äôre going to use that a bit later on if you come along to the pulsar
tutorial where we will also be installing CVMFS on another machine - on
a remote machine to run Pulsar.</p>
      </blockquote>

      <p>Add the following lines to your <code class="language-plaintext highlighter-rouge">group_vars/all.yml</code> file, creating it if it doesn‚Äôt exist:</p>

      <div data-commit="Configure CVMFS variables" data-ref="vars-all" class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gd">--- /dev/null
</span><span class="gi">+++ b/group_vars/all.yml
</span><span class="p">@@ -0,0 +1,4 @@</span>
<span class="gi">+# CVMFS vars
+cvmfs_role: client
+<span class="notranslate">galaxy</span>_cvmfs_repos_enabled: config-repo
+cvmfs_quota_limit: 500
</span>   
</code></pre></div>      </div>

      <blockquote class="spoken" data-visual="terminal" data-ref="vars-all">
        <p>So what we‚Äôre going to do, is we‚Äôre going to create a new file called
groupvars/all.yml and we‚Äôre going to put some of these CVMFS variables
inside it. So i‚Äôll just copy that. Okay, groupvars all dot yaml. And
I‚Äôll paste this in. So basically the CVMFS role we want this machine to
have is client. Which means that we just want it to be able to access
all of our CVMFS reference data. And then we want we‚Äôre going to set
this one here to say that, ‚Äúyes we want to set this up for <span class="notranslate">Galaxy</span>.‚Äù And
the config repo is the one that tells CVMFS how to set everything else
up. And then this is the other important one - the CVMFS quota limit.
We‚Äôre setting to 500 megabytes and that‚Äôs just so we don‚Äôt fill the root
disk of these machines. So I‚Äôll save that.</p>
      </blockquote>

      <blockquote class="tip">
        <h3 id="tip-why-allyml"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Why all.yml?</h3>
        <p>We‚Äôve integrated the cvmfs and pulsar tutorials better, such that CVMFS will be used for Pulsar as well, this configuration will be needed on all of our machines. This mirrors real life where you want CVMFS on every node that does computation.</p>
      </blockquote>
    </li>
    <li>
      <p>Add the new role to the list of roles under the <code class="language-plaintext highlighter-rouge">roles</code> key in your playbook, <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span>.yml</code>:</p>

      <div data-commit="Add role to playbook" data-ref="pb" class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gd">--- a/<span class="notranslate">galaxy</span>.yml
</span><span class="gi">+++ b/<span class="notranslate">galaxy</span>.yml
</span><span class="p">@@ -21,3 +21,4 @@</span>
       become: true
       become_user: "{{ <span class="notranslate">galaxy</span>_user.name }}"
     - <span class="notranslate">galaxy</span>project.nginx
<span class="gi">+    - <span class="notranslate">galaxy</span>project.cvmfs
</span>   
</code></pre></div>      </div>

      <blockquote class="spoken" data-visual="terminal" data-ref="pb">
        <p>And now we need to add the role - we need to add the role to our <span class="notranslate">Galaxy</span>
playbook. So we edit <span class="notranslate">galaxy</span>.yml which is our playbook and we just need to
add the <span class="notranslate">galaxy</span>project.cvmfs to the bottom of this. <span class="notranslate">galaxy</span>project.cvmfs.
And that‚Äôs pretty much it.</p>
      </blockquote>
    </li>
    <li>
      <p>Run the playbook</p>

      <blockquote class="code-in">
        <h3 id="code-in-input-bash-1"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Input: Bash</h3>
        <div data-cmd="true" data-ref="run-pb1" class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ansible-playbook <span class="notranslate">galaxy</span>.yml
</code></pre></div>        </div>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-ref="run-pb1">
  <p>Okay now we just run the playbook. ansible-playbook. We want to run the
<span class="notranslate">Galaxy</span> one. So. All right, we‚Äôre about to get to the CVMFS repo here. We are
now it‚Äôs installing the um the apt package of CVMFS. It‚Äôs going to get
that out of this special um Cern apt repository. Okay, it‚Äôs installing
it. Hopefully it won‚Äôt take too long. Okay, now setting up the
repositories. And it‚Äôs done.</p>
</blockquote>

<p>Congratulations, you‚Äôve set up CVMFS.</p>

<h2 id="exploring-the-cvmfs-installation">Exploring the CVMFS Installation</h2>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd /">
  <p>Okay, that was completed. We‚Äôve now installed CVMFS client onto our machine
and we‚Äôve told it to go looking for certain repositories. Now to get access
to them. We‚Äôll see what‚Äôs in them. uh They‚Äôll be located at slash cvmfs. So
under the cvmfs directory in your root directory. So, we can go to that. cd
/</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="ls -al">
  <p>Do an ll. You can see here there‚Äôs a directory here called cvmfs.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd /cvmfs">
  <p>So we‚Äôll go in there and have a look and see what‚Äôs in there.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="ls -al">
  <p>So let‚Äôs have a look. Oh there‚Äôs nothing in there!
Well actually, what what‚Äôs going to happen is; as soon as we go looking for
something in this directory, so autofs will automatically mount the
particular thing we‚Äôre looking for. And so what we‚Äôre going to do here is I‚Äôm
going to go: cd data.<span class="notranslate">galaxy</span>project.org, because I know that‚Äôs one of the one
of the repositories that should have been installed.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd data.galaxyproject.org">
  <p>And when I do that, autofs is automatically going to mount it for me on the
fly. Like that.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="ls -al">
  <p>And now I‚Äôve cd‚Äôd into it and if I do an ll, you can see here I‚Äôve got some
things in here now. I‚Äôve got byhand and managed.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd byhand; ls">
  <p>If I go into byhand you can see here that I have quite a lot of different
 genomes and their tool indices.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="ls -al sacCer2/">
  <p>So if I‚Äôm going to, say, sacCer2, I can see in here there are bowtie index,
the bwa index, the uh the picard index, the sam index; a whole bunch of
other different things - including the original sequences etc. So yeah,
quite a lot of data and we just have access to that on the fly. And then as
soon as we try and look at any of these files, what will happen is CVMFS
will automatically cache it to the local disk within within that 500
megabyte cache that we uh we set up earlier. This is really cool. And we can
tell <span class="notranslate">Galaxy</span> to look at all of this data and use it as its reference data.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-exploring-cvmfs"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Exploring CVMFS</h3>

  <ol>
    <li>
      <p>SSH into your machine</p>
    </li>
    <li>
      <p>Change directory into <code class="language-plaintext highlighter-rouge">/cvmfs/</code> and list the files in that folder</p>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>What do you see?</p>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>You should see nothing, as CVMFS uses <code class="language-plaintext highlighter-rouge">autofs</code> in order to mount paths only upon request.</p>

        </blockquote>

      </blockquote>
    </li>
    <li>
      <p>Change directory into <code class="language-plaintext highlighter-rouge">/cvmfs/data.<span class="notranslate">galaxy</span>project.org/</code>.</p>

      <blockquote class="code-in">
        <h3 id="code-in-input-bash-2"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Input: Bash</h3>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /cvmfs/data.<span class="notranslate">galaxy</span>project.org/
ls
ls byhand
ls managed
</code></pre></div>        </div>
      </blockquote>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>What do you see now?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>You‚Äôll see <code class="language-plaintext highlighter-rouge">.loc</code> files, genomes and indices.
AutoFS only mounts the files when they‚Äôre accessed, so it appears like there is no folder there.</p>
        </blockquote>

      </blockquote>

      <p>And just like that we all have access to all the reference genomes and associated tool indices thanks to the <span class="notranslate">Galaxy</span> Project, IDC, and Nate‚Äôs hard work!</p>
    </li>
  </ol>

</blockquote>

<h2 id="configuring-galaxy-to-use-the-cvmfs-references">Configuring <span class="notranslate">Galaxy</span> to use the CVMFS references.</h2>

<p>Now that we have mounted the CVMFS repository we need to tell <span class="notranslate">Galaxy</span> how to find it and use it.</p>

<p>There are two primary directories in the reference data repository:</p>

<table>
  <thead>
    <tr>
      <th>Directory</th>
      <th>Contents</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="language-plaintext highlighter-rouge">/managed</code></td>
      <td>Data generated with <span class="notranslate">Galaxy</span> Data Managers, organized by data table (index format), then by genome build.</td>
    </tr>
    <tr>
      <td><code class="language-plaintext highlighter-rouge">/byhand</code></td>
      <td>Data generated prior to the existence/use of Data Managers, manually curated. (For legacy reasons, this directory is shared as <code class="language-plaintext highlighter-rouge">/indexes</code> on the HTTP and rsync servers.)</td>
    </tr>
  </tbody>
</table>

<p>These directories have somewhat different structures:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">/managed</code> is organized by index type, then by genome build (<span class="notranslate">Galaxy</span> dbkey)</li>
  <li><code class="language-plaintext highlighter-rouge">/byhand</code> is organzied by genome build, then by index type</li>
</ul>

<p>Both directories contain a location subdirectory, and each of these contain a <code class="language-plaintext highlighter-rouge">tool_data_table_conf.xml</code> file:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">/managed/location/tool_data_table_conf.xml</code></li>
  <li><code class="language-plaintext highlighter-rouge">/byhand/location/tool_data_table_conf.xml</code></li>
</ul>

<p><span class="notranslate">Galaxy</span> consumes these <code class="language-plaintext highlighter-rouge">tool_data_table_conf.xml</code> files and the <code class="language-plaintext highlighter-rouge">.loc</code> ‚Äúlocation‚Äù files they reference. The paths contained in these files are valid if the data is mounted via CVMFS.</p>

<p>Examples of data include:</p>

<ul>
  <li>twoBit (<code class="language-plaintext highlighter-rouge">.2bit</code>) and FASTA (<code class="language-plaintext highlighter-rouge">.fa</code>) sequence files</li>
  <li>Bowtie 2 and BWA indexes</li>
  <li>Multiple Alignment Format (<code class="language-plaintext highlighter-rouge">.maf</code>) files</li>
  <li>SAMTools FASTA indexes (<code class="language-plaintext highlighter-rouge">.fai</code>)</li>
</ul>

<p id="spoken-7">Now all we need to do is tell <span class="notranslate">Galaxy</span> how to find it! This tutorial assumes that you have run the tutorial in the requirements, <a href="/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html"><span class="notranslate">Galaxy</span> Installation with Ansible</a>. The hands-on below will use the <span class="notranslate">Galaxy</span> Project Ansible role to configure everything.</p>

<blockquote class="spoken" data-visual="gtn" data-target="#spoken-7">
  <p>So that‚Äôs what we‚Äôre going to do now. Okay. So now we‚Äôre going to try and
configure <span class="notranslate">Galaxy</span> to use this CVMFS data. And and to have it so that we can run
things like bwa and bwa mem and run them against the human genome or the bee
genome or the mouse genome and take advantage of the fact that a lot of other
people in the <span class="notranslate">Galaxy</span> community have done a lot of work for reference data for
us already.</p>

  <p>So the way to do this is we‚Äôre going to edit the groupvars <span class="notranslate">galaxy</span>servers
file and we‚Äôre going to add a variable called tool_data_table_config_path.
And then we‚Äôre going to point it to the two files that are in - there‚Äôs one
in byhand and one in managed.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd /cvmfs/data.galaxyproject.org/byhand;  ls location;">
  <p>If we go into byhand you can see here. And then we‚Äôre going to location. As
you can see in here, are all the lock files, but you‚Äôll also see there‚Äôs an
xml file here called tool_data_table_conf xml and we‚Äôre going to point <span class="notranslate">Galaxy</span>
at this file and there‚Äôs another one in the same position in managed.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd ../; ls managed/location">
  <p>And you can see here there‚Äôs another one in managed there. And so we‚Äôre going to
add both of these files to our <span class="notranslate">Galaxy</span> configuration and then <span class="notranslate">Galaxy</span> will be
able to use all of the data contained within this repository.</p>
</blockquote>

<blockquote class="spoken" data-visual="terminal" data-cmd="cd ~/galaxy/">
  <p>Okay, so we‚Äôll go back to our Ansible directory.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-configuring-galaxy-to-use-cvmfs"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Configuring <span class="notranslate">Galaxy</span> to use CVMFS</h3>

  <ol>
    <li>
      <p>Edit the <code class="language-plaintext highlighter-rouge">group_vars/<span class="notranslate">galaxy</span>servers.yml</code> file and add a <code class="language-plaintext highlighter-rouge">tool_data_table_config_path</code> entry under the <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span></code> key of the <code class="language-plaintext highlighter-rouge"><span class="notranslate">galaxy</span>_config</code> section in the <code class="language-plaintext highlighter-rouge">group_vars/<span class="notranslate">galaxy</span>servers.yml</code> file. This new entry should be a list containing the paths to both <code class="language-plaintext highlighter-rouge">tool_data_table_conf.xml</code> files referenced above.</p>

      <div data-commit="Add tool_data_table_config_path to group variables" data-ref="gvconf" class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gd">--- a/group_vars/<span class="notranslate">galaxy</span>servers.yml
</span><span class="gi">+++ b/group_vars/<span class="notranslate">galaxy</span>servers.yml
</span><span class="p">@@ -29,6 +29,7 @@</span> miniconda_manage_dependencies: false
    
 <span class="notranslate">galaxy</span>_config:
   <span class="notranslate">galaxy</span>:
<span class="gi">+    tool_data_table_config_path: /cvmfs/data.<span class="notranslate">galaxy</span>project.org/byhand/location/tool_data_table_conf.xml,/cvmfs/data.<span class="notranslate">galaxy</span>project.org/managed/location/tool_data_table_conf.xml
</span>     dependency_resolvers_config_file: "{{ <span class="notranslate">galaxy</span>_config_dir }}/dependency_resolvers_conf.xml"
     containers_resolvers_config_file: "{{ <span class="notranslate">galaxy</span>_config_dir }}/container_resolvers_conf.xml"
     brand: "üß¨üî¨üöÄ"
   
</code></pre></div>      </div>

      <blockquote class="spoken" data-visual="terminal" data-ref="gvconf">
        <p>This time I‚Äôm going to edit the groupvars <span class="notranslate">galaxy</span>servers.yml file. And in
our <span class="notranslate">Galaxy</span> section. Which is here. Which is here. At the bottom of that
I‚Äôm going to add a variable called tool_data_table_config_path. And I‚Äôm
going to point it to the locations that we um I showed you before. um I
can‚Äôt remember what they are off the top of my head but luckily they‚Äôre
inside this solution box and so I will just copy them. And paste. And as
you can see pointing to /cvmfs/data.<span class="notranslate">galaxy</span>project.org/byhand/location
and then that tool data table conf xml file. And then we have a list
here and we separate it by commas and then we point it to the second
one. Right, so we save this file.</p>
      </blockquote>
    </li>
    <li>
      <p>Re-run the playbook</p>

      <blockquote class="code-in">
        <h3 id="code-in-input-bash-3"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Input: Bash</h3>
        <div data-cmd="true" data-ref="pb-run2" class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ansible-playbook <span class="notranslate">galaxy</span>.yml
</code></pre></div>        </div>
      </blockquote>

      <blockquote class="spoken" data-visual="terminal" data-ref="pb-run2">
        <p>So this time what we‚Äôre going to do, all we‚Äôre doing is making a minor change
to the <span class="notranslate">Galaxy</span> yaml file in the <span class="notranslate">Galaxy</span> config to add that one line and then
we‚Äôre going to restart <span class="notranslate">Galaxy</span>. And yeah, <span class="notranslate">Galaxy</span> will suddenly automatically
have access to all of that data. So you can see here we‚Äôve changed the <span class="notranslate">Galaxy</span>
configuration file. And then <span class="notranslate">Galaxy</span> is now restarting and it‚Äôs done.</p>
      </blockquote>
    </li>
    <li>
      <p>Install the BWA-MEM tool, if needed.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-install-tools-via-the-admin-ui"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Install tools via the Admin UI</h3>  <ol>  <li>Open <span class="notranslate">Galaxy</span> in your browser and type <code class="language-plaintext highlighter-rouge">bwa</code> in the tool search box on the left. If ‚ÄúMap with BWA-MEM‚Äù is among the search results, you can skip the following steps.</li>  <li>Access the Admin menu from the top bar (you need to be logged-in with an email specified in the <code class="language-plaintext highlighter-rouge">admin_users</code> setting)</li>  <li>Click ‚ÄúInstall and Uninstall‚Äù, which can be found on the left, under ‚ÄúTool Management‚Äù</li>  <li>Enter <code class="language-plaintext highlighter-rouge">bwa</code> in the search interface</li>  <li>Click on the first hit, having <code class="language-plaintext highlighter-rouge">devteam</code> as owner</li>  <li>Click the ‚ÄúInstall‚Äù button for the latest revision</li>  <li>Enter ‚Äú<span class="notranslate">Mapping</span>‚Äù as the target section and click ‚ÄúOK‚Äù.</li></ol></blockquote>
    </li>
    <li>
      <p>In your <span class="notranslate">Galaxy</span> server, open the <strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool. Now check that there are a lot more reference genomes available for use!</p>

      <p><img src="../../images/available_genomes.png" alt="available_genomes.png. " loading="lazy" /></p>
    </li>
    <li>
      <p>Login to <span class="notranslate">Galaxy</span> as the admin user, and go to <strong>Admin ‚Üí Data Tables ‚Üí bwa_mem indexes</strong></p>

      <p><img src="../../images/bwa_mem_indices.png" alt="bwa_mem_indices.png. " loading="lazy" /></p>
    </li>
  </ol>

</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target="#analysis a" data-action="click">
  <p>Okay let‚Äôs go and have a look at our <span class="notranslate">Galaxy</span> server and see if bwa can
suddenly see all of those - that stuff. All right so we‚Äôre back on our <span class="notranslate">Galaxy</span>
server. I‚Äôll click on Analyze Data to just to reload the page.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target=".search-input input" data-action="click">
  <p>I‚Äôll go back to <span class="notranslate">Mapping</span></p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target=".search-input input" data-action="fill" data-value="bwa-mem">
  <p>and load bwa-mem.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target="/?tool_id=bwa_mem" data-action="goto">
  <p>And then suddenly, instead of
having no options available, you can see here we‚Äôve got the Bee genome.</p>
</blockquote>

<blockquote class="spoken" data-visual="galaxy" data-target="s2id_field-uid-12_select a" data-action="click">
  <p>Now click on that. Oh look at that, there are lots and lots and lots of
 available genomes now including: lots of human, mouse, rat, yeast, all sorts
 of things. And in fact if you want to see the list of all the different
 available genomes now, that we have available to us.</p>

  <p>If you go to admin. We go to data tables over here. You can see here that we
 have um a couple of data tables for managed and for all fasta. So if we
 click on that one, you can see that we have a lot of genomes available now
 in the all fasta data table that <span class="notranslate">Galaxy</span> can get access to. If we go back to
 the data tables again, and go down to bwa indexes or bwa mem indexes here.
 You can see we have access to a lot of pre-built indexes for bwa for all of
 these different genomes. That is pretty powerful.</p>

  <p>So what did that take us? Maybe 30 minutes? And uh suddenly our <span class="notranslate">Galaxy</span>
 server has access to all the uh the data the reference data and the tool
 indices that the community have built over a number of years and it‚Äôs super
 simple.</p>
</blockquote>

<blockquote class="hidden">
  <div data-test="true" class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1.sh
</code></pre></div>  </div>
</blockquote>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-got-lost-along-the-way"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Got lost along the way?</h3>  <p>If you missed any steps, you can compare against the <a href="https://github.com/hexylena/git-gat/tree/step-4">reference files</a>, or see what changed since <a href="https://github.com/hexylena/git-gat/compare/step-3...step-4#files_bucket">the previous tutorial</a>.</p></blockquote>

<h1 id="common-production-questions">Common Production Questions</h1>

<blockquote class="question">
  <h3 id="question-for-the-most-used-datasets-for-ex-hg38-could-we-have-a-local-copy-or-would-that-be-irrelevant"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> For the most used datasets (for ex. hg38) could we have a local copy, or would that be irrelevant?</h3>
  <p>This would be irrelevant, the most used datasets will stay in the cache. CVMFS uses a Least Recently Used (LRU) cache (see their <a href="https://cvmfs.readthedocs.io/en/latest/cpt-details.html#disk-cache">docs</a>), so whenever it runs out of space, it will remove the least recently used file. If you have a file that is very commonly used, it will remain in the cache.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-could-you-explain-how-to-calculate-a-good-cache-space"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Could you explain how to calculate a good cache space?</h3>
  <p>Here are two approaches, there are others:</p>
  <ol>
    <li>Allocate some cache, see how it is, make it larger if it is fully used + users complain of speed.</li>
    <li>Enable reference data, and collect a week or two of data, analyse which reference datasets are being used, and allocate enough space for all of them.</li>
  </ol>

  <p>Essentially you just need data on how your users will behave and what reference data they want, combined with ‚Äúwhen will they accept a wait period‚Äù to determine how much space you must allocate.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-if-i-use-a-cluster-will-i-need-to-configure-this-fs-in-each-node-given-that-the-folder-is-at--directly"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> If I use a cluster, will I need to configure this FS in each node (given that the folder is at / directly)?</h3>
  <p>Yes. Often admins with a cluster keep a smaller cache local to each compute node, and then setup a Squid proxy to hold the most commonly accessed data on a machine with more storage. E.g. each compute node could have 10-50GB of CVMFS storage while you might setup a Squid proxy with 200-300 GB of storage that will store everything your site uses.</p>
</blockquote>

<h1 id="other-aspects">Other Aspects</h1>

<blockquote class="spoken" data-visual="gtn" data-target="#other-aspects">
  <p>Right, we‚Äôll go back to our tutorial. um Yeah. Just finally, just before we
finish up. um If we are developing a new tool and you want to add a reference
genome or a different index just give us - drop us a line on Gitter and we‚Äôll
be able to add it into our - into the reference data for the community. um
We‚Äôre looking at automating the process of building all of this material
using data managers and ephemeris. And we‚Äôre working with a group of people
called the IDC, which is the Intergalactic Data Commission, which is a funny
name for everyone in <span class="notranslate">Galaxy</span> - in the <span class="notranslate">Galaxy</span> community who likes reference
data. And we‚Äôre looking at making a community controlled resource that will
be semi-automatic. One of the other things that you can do is have automatic
fallbacks. So, if say, you‚Äôre in Australia and you‚Äôre hooked up to the
Australian mirror of the CVMFS repository and the Australian mirror dies, the
CVMFS client is smart enough to automatically go to the next closest one and
so you won‚Äôt lose anything. If you‚Äôre interested in looking at plant data
there‚Äôs a link here for that.</p>
</blockquote>

<h2 id="development">Development</h2>

<p>If you are developing a new tool, and want to add a reference genome, we recommend you <a href="https://gitter.im/galaxy-iuc/iuc">talk to us on Gitter</a>. You can also look at one of the tools that uses reference data, and try and copy from that. If you‚Äôre developing the location files completely new, you need to write the data manager.</p>

<h2 id="automation">Automation</h2>

<p>You can automate the process of installing and setting up data managers and data with ephemeris. We‚Äôre working in the <a href="https://github.com/galaxyproject/idc">IDC</a> to democratise this CVMFS repository, and make this a community-controlled resource. You can also look here for ideas on automating your data management.</p>

<h2 id="access-control">Access Control</h2>

<p>It is not easily possible to filter access to reference data depending on the user‚Äôs role or group.</p>

<p>You could set up a tool per user/group, <a href="https://galaxyproject.org/admin/config/access-control/">secure access to running this tool</a>, and then allow this private tool to access a private tool data table. But you will not get tool updates, you will have to copy and edit this tool every time it gets updated. Or write more advanced job control rules to reject specific jobs which use specific datasets.</p>

<h2 id="proxying-recap">Proxying Recap</h2>

<p>The client talks directly to the stratum 1 (or to a proxy), and manages the data, and exposes it to the user. The proxy stores an opaque cache, that can‚Äôt really be used, except as a proxy to a stratum 1.</p>

<h2 id="plant-data">Plant Data</h2>

<p>If you are working with plants, you can find separate reference data here: <a href="https://github.com/frederikcoppens/galaxy_data_management">frederikcoppens/<span class="notranslate">galaxy</span>_data_management</a></p>

<blockquote class="spoken" data-visual="gtn" data-target="#feedback">
  <p>And finally if you could please click on this link here and give us some
feedback on how you think the tutorial went, whether it was useful, if you
enjoyed it or um if you have any criticisms, could you please put them in
here as well. And if you end up using this to build a <span class="notranslate">Galaxy</span> server and you
publish that <span class="notranslate">Galaxy</span> server um could you cite the tutorial for us please. That
would be make a big difference to us. All right, thank you very much and I
hope you enjoyed it and hopefully I‚Äôll get to meet some of you in person one
day soon at a <span class="notranslate">Galaxy</span> conference. Thank you and goodbye.</p>
</blockquote>


                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/admin/faqs/">FAQ page for the Galaxy Server administration topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Reference Data with CVMFS (Galaxy Server administration)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Simon Gladman, Helena Rasche, 2021 <b>Reference Data with CVMFS (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/admin/tutorials/cvmfs/tutorial.html">https://training.galaxyproject.org/training-material/topics/admin/tutorials/cvmfs/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{admin-cvmfs,
author = "Simon Gladman and Helena Rasche",
title = "Reference Data with CVMFS (Galaxy Training Materials)",
year = "2021",
month = "06",
day = "22"
url = "\url{https://training.galaxyproject.org/training-material/topics/admin/tutorials/cvmfs/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
