<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Bioinformatics Data Types and Databases</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/online-resources-gene/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1699662338' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1699662338' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Bioinformatics Data Types and Databases">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-11 00:25:39 +0000">
<meta name="DC.creator" content="Lisanna Paladin">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Bioinformatics Data Types and Databases">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1699662338'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/data-science/tutorials/online-resources-gene/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/data-science/tutorials/online-resources-gene/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/data-science/tutorials/online-resources-gene/slides.html">View markdown source on GitHub</a>
	<h1>Bioinformatics Data Types and Databases</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/lisanna/" class="contributor-badge contributor-lisanna"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/lisanna?s=36" alt="Avatar" width="36" height="36">Lisanna Paladin</a>
		</td>
	</tr>
	


	

	

	

	

	
</table>


	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>What are some of the main resources to explore bioinformatics information?</p>
</li>
		
		<li><p>How is this information represented in file formats?</p>
</li>
		
		<li><p>What type of information do these file formats convey?</p>
</li>
		
		</ul>
		

		
		<h2>Objectives</h2>
		<ul>
		
		<li><p>Understand that the biological data is multi-layered</p>
</li>
		
		<li><p>Identify multiple sources of information in biology</p>
</li>
		
		<li><p>Describe how this different types of information are conveyed through different file formats</p>
</li>
		
		</ul>
		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Nov 11, 2023 </div>

	<hr />


	<h1 id="background">Background</h1>

<ul>
  <li>Need: digitally store biological data</li>
  <li>All biological data could be (and initially was) included in simple text files</li>
  <li>Yet, significant limitations:
    <ul>
      <li>Not structured, hence not programmatically accessible</li>
      <li>Impossible to distinguish data (e.g. gene sequence) from metadata (e.g. annotations about location, quality, function, etc.)</li>
    </ul>
  </li>
</ul>

<p><span>Speaker Notes</span></p>

<p>In this presentation, we’ll look into the history of biological data.
Initially, all type of data was approached using simple text files, but this quickly became limiting.
Indeed, unstructured text files are not programmatically accessible and in such files it is impossible to distinguish data from metadata.
It’s important to understand these limitations as they set the stage for the development of more advanced storage methods.</p>

<hr />

<h2 id="different-information-in-different-file-formats">Different information in different file formats</h2>

<ul>
  <li>In the years, different file formats have been developed to store different types of data with the relevant metadata fields</li>
  <li>E.g. for a biological sequence
    <ul>
      <li>From the simplest, text-like file (FASTA)</li>
      <li>To more complex formats which include genomic features and quality annotation</li>
    </ul>
  </li>
  <li>Different file formats not only to represent different levels of complexity but also different types of information</li>
  <li>E.g. about a protein
    <ul>
      <li>From a text-like file to store the sequence (FASTA)</li>
      <li>To a tabular file to store the exact coordinates of each atom in the structure, hence comvey the 3D arrangement</li>
    </ul>
  </li>
</ul>

<p><span>Speaker Notes</span></p>

<p>As time progressed, the need for more structured and accessible data storage became apparent.
Various file formats were developed to accommodate different types of biological data.
We’ll explore some of these formats, ranging from simple text-like files for storing sequences to more complex ones
that include annotations, 3D structures, and genomic features.</p>

<hr />

<h2 id="different-information-in-different-databases">Different information in different databases</h2>

<ul>
  <li>Consequently, different resources evolved not only to store, but also represent/visualise this varied information</li>
  <li>These resources often have a database storing data and a web interface that allows to navigate it</li>
  <li>They usually represent different levels of complexity of one specific type of biological entity
    <ul>
      <li>E.g. A database of protein sequences and their annotation (sequence variability, genomic location, effect of mutations, etc.)</li>
      <li>E.g. A database of protein structures and their annotation (3D coordinates, flexibility, methods used to resolve the structure, etc.)</li>
    </ul>
  </li>
</ul>

<p><span>Speaker Notes</span></p>

<p>In parallel, different biological resources emerged, each designed to handle specific types of data and complexity.
These resources often consist of databases with associated web interfaces, enabling users to navigate and visualize the data effectively.</p>

<hr />

<h3 id="definition-of-a-biological-databaseresources">Definition of a biological database/resources</h3>

<ul>
  <li>The <a href="https://www.oxfordjournals.org/nar/database/c/">NAR Database Issue</a> collects publications of established databases in the field</li>
  <li>Collection of data (and metadata) in the related format
    <ul>
      <li>structured</li>
      <li>searchable (indexed)</li>
      <li>updated periodically</li>
      <li>entries mapped to unique identifiers, and cross-referenced</li>
    </ul>
  </li>
  <li>Includes associated software necessary for DB access, update, search, visualisation (web)</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>Biological databases play a crucial role in housing and organizing biological data.
The NAR Database Issue collects publications about established databases in the field.
Requirements to be featured in this issues are to have a structured nature, searchability, regular updates, and cross-referencing capabilities.
These databases also offer software tools for accessing, updating, and visualizing the data they contain.</p>

<hr />

<h2 id="some-history">Some history</h2>

<p>.pull-left[</p>
<ul>
  <li>1953: 3D structure of DNA (Watson, Crick, Franklin, Wilkins)</li>
  <li>1956: first protein sequence, insulin (51 AA)</li>
  <li>1965: first whole nucleic acid sequence, tRNA from yeast</li>
  <li>1966: Atlas of protein sequences and structures, by Margaret Dayhoff, printed book</li>
  <li>1972: first complete protein-coding gene, coat protein from a bacteriophage</li>
  <li>1976: same Lab, its complete genome</li>
  <li>1971: Protein Data Bank (PDB)</li>
  <li>1980-87: the European Molecular Biology Laboratory (EMBL) Nucleotide Sequence Database; GenBank from the National Center for Biotechnology Information (NCBI); and the DNA Databank of Japan (DDJ)</li>
  <li>1986: SwissProt was created by Rolf Apweiler
]</li>
</ul>

<p>.pull-right[
<img src="http://cdn.rcsb.org/rcsb-pdb/v2/about-us/early.png" alt="The dataset of PDB structures in 1973 included only 9 proteins illustrated in this image" />
]</p>

<p><span>Speaker Notes</span></p>

<p>The source of information for this slide, which includes a short early history of biological data formats and databases evolution, is the paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4727787/
Understanding the historical context of biological data storage helps us appreciate the progress made in the field.</p>
<ul>
  <li>1953: Watson and Crick famously solved the three-dimensional structure of DNA in 1953, working from crystallographic data produced by Rosalind Franklin and Maurice Wilkins</li>
  <li>1956: Fred Sanger obtained the first protein sequence, of insulin (51 AA)</li>
  <li>1965: Robert Holley and colleagues were able to produce the first whole nucleic acid sequence, that of alanine tRNA from Saccharomyces cerevisiae</li>
  <li>1966: Atlas of protein sequences and structures, by Margaret Dayhoff, a printed book including multiple
    <blockquote>
      <p>This book is a compilation of known protein sequences. The major ones listed are for cytochrome C and for hemoglobin alpha and beta chains.</p>
    </blockquote>
  </li>
  <li>1972: Walter Fiers’ laboratory was able to produce the first complete protein-coding gene sequence in 1972, that of the coat protein of bacteriophage MS2</li>
  <li>1976: same Lab, its complete genome</li>
  <li>1971: Protein Data Bank (PDB)</li>
  <li>1980-87: the European Molecular Biology Laboratory (EMBL) Nucleotide Sequence Database; GenBank from the National Center for Biotechnology Information (NCBI); and the DNA Databank of Japan (DDJ)</li>
  <li>1986: SwissProt was created by Rolf Apweiler</li>
</ul>

<hr />

<h1 id="examples-of-biological-databases">Examples of biological databases</h1>

<ul>
  <li>SwissProt + TrEMBL = UniProtKB</li>
  <li>PDB</li>
  <li>GenBank</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>Prominent biological databases that have significantly contributed to our understanding of biological entities
are for example UniProtKB, PDB, and GenBank. We will discuss their importance and the types of data they store.</p>

<hr />

<h3 id="uniprotkb">UniProtKB</h3>

<p>.pull-left[</p>
<ul>
  <li>Swiss-Prot: Manually curated / annotated Sequence Database</li>
  <li>TrEMBL: Database of EMBL nucleotide translated sequences, automatically annotated</li>
</ul>

<p>The two databases are merged into the UniProt Knowledge Base, including information of different types about proteins.
]</p>

<p>.pull-right[
<img src="./images/UniProt_proteins.png" alt="The UniProtKB, at the time of creation of these slides, includes 596793 manually curated entries (Reviewed) in Swiss-Prot and 248272897 Unreviewed entries in TrEMBL" />
]</p>

<p><span>Speaker Notes</span></p>

<p>UniProtKB is a comprehensive resource that brings together data from both Swiss-Prot and TrEMBL databases.
We’ll explore how these databases are merged to create a unified knowledge base about proteins, encompassing a wide array of information.</p>

<hr />

<h3 id="pdb">PDB</h3>

<p>.pull-left[
Protein Data Bank (PDB) archive of 3D structure data for biological molecules (proteins, DNA, RNA).</p>

<p>Currently includes &gt; 1TB of structure data, archived world-wide.
]</p>

<p>.pull-right[
<img src="http://cdn.rcsb.org/rcsb-pdb/v2/about-us/wwpdb.png" alt="The wwPDB project maintains a single PDB archive distributed in the USA, Europe and Japan, and freely and publicly available to the global community" />
]</p>

<p><span>Speaker Notes</span></p>

<p>The Protein Data Bank, or PDB, is a vital repository for 3D structure data of biological molecules.
We’ll delve into the significance of PDB, its role in advancing structural biology, and the substantial volume of data it currently archives.</p>

<hr />

<h3 id="genbank">GenBank</h3>

<p>.pull-left[
An annotated collection of all publicly available DNA sequences, which comprises the DNA DataBank of Japan (DDBJ), the European Nucleotide Archive (ENA), and GenBank at NCBI
]</p>

<p>.pull-right[
<img src="https://www.researchgate.net/publication/343364994/figure/fig2/AS:919700666073090@1596285134479/Growth-of-GenBank-sequences-and-NCBI-web-users-through-2019-Figure-borrowed-from-the.png" alt="A graph showing that both the number of GenBank sequences and the number of NCBI web users has been constantly growing from 1989 to 2019, reaching more than 200 millions sequences and 6 millions users." />
]</p>

<p><span>Speaker Notes</span></p>

<p>GenBank stands as a critical resource for DNA sequences. It collaborates with other databases,
such as DDBJ and ENA, to provide a comprehensive collection of publicly available DNA sequences.</p>

<hr />

<h2 id="biological-knowledge">Biological knowledge</h2>

<p>.pull-left[
Understanding about biological entities comes from crossing the information from/to these different resources and formats
]</p>

<p>.pull-right[
<img src="./images/merged-info.png" alt="New knowledge comes from merging and crossing different levels of information about a protein, the schema mentions: the sequence (plain, conservation), structure, genomic information (conservation, location, regulation), function." />
]</p>

<p><span>Speaker Notes</span></p>

<p>An intricate web of information exists around biological entities, and understanding them involves merging
insights from various resources. A big part of some bioinformaticians’ job is to integrate information from different
databases and formats to gain a holistic understanding of biological entities.</p>

<hr />

<h2 id="features-of-biological-databases">Features of biological databases</h2>

<ul>
  <li>Data heterogeneity</li>
  <li>High volume of data</li>
  <li>Large scale data integration</li>
  <li>Data sharing / user visualisation and navigation</li>
  <li>Uncertainty / data quality measure needed</li>
  <li>Dynamic and subject to change</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>Biological databases are characterized by a range of features that reflect the complexity of biological data.
Biological databases face the challenges of handling data heterogeneity, ensuring data quality, and accommodating the dynamic nature of biological information.</p>

<hr />

<h2 id="possible-classifications-of-biological-databases">Possible classifications of biological databases</h2>

<p>.pull-left[</p>
<ul>
  <li>Data type</li>
  <li>Data access</li>
  <li>Data source</li>
  <li>…
]</li>
</ul>

<p>–</p>

<p>.pull-right[
<strong>Data type</strong></p>

<ul>
  <li>Genome database</li>
  <li>Sequence database</li>
  <li>Structure database</li>
  <li>Pathway database</li>
  <li>Disease database</li>
  <li>…
]</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>Classifying biological databases helps us categorize and understand their diverse nature. There might be various
ways of classifying databases, such as by data type, data access, and data source.</p>

<p>The world of biological data is rich with different file formats designed to accommodate diverse types of information,
including those for sequences, alignments, features/annotations, and protein structures.</p>

<hr />

<h2 id="possible-classifications-of-biological-databases-1">Possible classifications of biological databases</h2>

<p>.pull-left[</p>
<ul>
  <li>Data type</li>
  <li>Data access</li>
  <li>Data source</li>
  <li>…
]</li>
</ul>

<p>.pull-right[
<strong>Data access</strong></p>

<ul>
  <li>Publicly available (browsing, downloading)</li>
  <li>Freely accessible and reusable under a license</li>
  <li>License open to certain usages (e.g. academic)</li>
  <li>Proprietary / commercial</li>
  <li>Restricted to certain people / institutions</li>
  <li>…
]</li>
</ul>

<p><span>Speaker Notes</span></p>

<hr />

<h2 id="possible-classifications-of-biological-databases-2">Possible classifications of biological databases</h2>

<p>.pull-left[</p>
<ul>
  <li>Data type</li>
  <li>Data access</li>
  <li>Data source</li>
  <li>…
]</li>
</ul>

<p>.pull-right[
<strong>Data source</strong></p>

<ul>
  <li>Primary databases (GenBank, PDB)</li>
  <li>Secondary databases: analysed/aggregated results of the primary ones (UniProtKB)</li>
  <li>Composite database: non-redundant / filtered data (SwissProt)</li>
  <li>…
]</li>
</ul>

<hr />

<h1 id="biological-file-formats">Biological file formats</h1>

<ul>
  <li>Sequence formats</li>
  <li>Alignment formats</li>
  <li>Features/annotations formats</li>
  <li>Structure formats</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>In the following tutorials, we’ll explore some of the most commonly used biological file formats in detail.
We’ll provide examples and explanations for each format, helping you understand how they store and represent different types of biological data.</p>

<hr />

<h2 id="sequence-formats">Sequence formats</h2>

<p><strong>FASTA</strong></p>

<p>File extensions: file.fa, file.fasta, file.fsa</p>

<p>Example:</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gt">&gt;XR_002086427.1 Candida albicans SC5314 uncharacterized ncRNA (SCR1), ncRNA</span>

TGGCTGTGATGGCTTTTAGCGGAAGCGCGCTGTTCGCGTACCTGCTGTTTGTTGAAAATTTAAGAGCAAAGTGTCCGGCTCGATCCCTGCGAATTGAATTCTGAACGCTAGAGTAATCAGTGTCTTTCAAGTTCTGGTAATGTTTAGCATAACCACTGGAGGGAAGCAATTCAGCACAGTAATGCTAATCGTGGTGGAGGCGAATCCGGATGGCACCTTGTTTGTTGATAAATAGTGCGGTATCTAGTGTTGCAACTCTATTTTT
</code></pre></div></div>

<p><span>Speaker Notes</span></p>

<p>Fasta format is a simple way of representing nucleotide or amino acid sequences of nucleic acids and proteins.
This is a very basic format with two minimum lines. First line referred as comment line starts with ‘&gt;’ and gives
basic information about sequence. There is no set format for comment line. Any other line that starts with ‘;’ will
be ignored. Lines with ‘;’ are not a common feature of fasta files. After comment line, sequence of nucleic acid or
protein is included in standard one letter code. Any tabulators, spaces, asterisks etc in sequence will be ignored.</p>

<hr />

<h2 id="sequence-formats-1">Sequence formats</h2>

<p><strong>FASTQ</strong></p>

<p>File extensions: ile.fastq, file.sanfastq, file.fq</p>

<p>Example:</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@K00188:208:HFLNGBBXX:3:1101:1428:1508 2:N:0:CTTGTA
ATAATAGGATCCCTTTTCCTGGAGCTGCCTTTAGGTAATGTAGTATCTNATNGACTGNCNCCANANGGCTAAAGT
<span class="p">+</span>
AAAFFJJJJJJJJJJJJJJJJJFJJFJJJJJFJJJJJJJJJJJJJJJJ#FJ#JJJJF#F#FJJ#F#JJJFJJJJJ
</code></pre></div></div>

<p><span>Speaker Notes</span></p>

<p>Fastq format was developed by Sanger institute in order to group together sequence and its quality scores (Q: phred quality score). In fastq files each entry is associated with 4 lines.</p>

<ul>
  <li>Line 1 begins with a ‘@‘ character and is a sequence identifier and an optional description.</li>
  <li>Line 2 Sequence in standard one letter code.</li>
  <li>Line 3 begins with a ‘+‘ character and is optionally followed by the same sequence identifier (and any additional description) again.</li>
  <li>Line 4 encodes the quality values for the sequence in Line 2, and must contain the same number of symbols as letters in the sequence.</li>
</ul>

<hr />

<h2 id="alignment-formats">Alignment formats</h2>

<p><strong>SAM (Sequence Alignment Map)</strong></p>

<p>File extensions: file.sam</p>

<p>Example:</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1:497:R:-272+13M17D24M	113	1	497	37	37M	15	100338662	0	CGGGTCTGACCTGAGGAGAACTGTGCTCCGCCTTCAG	0;==-==9;&gt;&gt;&gt;&gt;&gt;=&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;=&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;	XT:A:U	NM:i:0	SM:i:37	AM:i:0	X0:i:1	X1:i:0	XM:i:0	XO:i:0	XG:i:0	MD:Z:37
19:20389:F:275+18M2D19M	99	1	17644	0	37M	=	17919	314	TATGACTGCTAATAATACCTACACATGTTAGAACCAT	&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;<span class="nt">&lt;</span><span class="err">&lt;</span><span class="nt">&gt;</span>&gt;&gt;<span class="nt">&lt;</span><span class="err">&lt;</span><span class="nt">&gt;</span>&gt;4::&gt;&gt;:&lt;9	RG:Z:UM0098:1	XT:A:R	NM:i:0	SM:i:0	AM:i:0	X0:i:4	X1:i:0	XM:i:0	XO:i:0	XG:i:0	MD:Z:37
19:20389:F:275+18M2D19M	147	1	17919	0	18M2D19M	=	17644	-314	GTAGTACCAACTGTAAGTCCTTATCTTCATACTTTGT	;44999;499<span class="nt">&lt;</span><span class="err">8&lt;8&lt;&lt;&lt;8&lt;&lt;</span><span class="nt">&gt;&lt;</span><span class="err">&lt;&lt;&lt;</span><span class="nt">&gt;&lt;</span><span class="err">7&lt;;&lt;&lt;&lt;</span><span class="nt">&gt;</span>&gt;&lt;&lt;	XT:A:R	NM:i:2	SM:i:0	AM:i:0	X0:i:4	X1:i:0	XM:i:0	XO:i:1	XG:i:2	MD:Z:18^CA19
9:21597+10M2I25M:R:-209	83	1	21678	0	8M2I27M	=	21469	-244	CACCACATCACATATACCAAGCCTGGCTGTGTCTTCT	<span class="nt">&lt;</span><span class="err">;9&lt;&lt;5</span><span class="nt">&gt;&lt;</span><span class="err">&lt;&lt;&lt;</span><span class="nt">&gt;&lt;</span><span class="err">&lt;&lt;</span><span class="nt">&gt;</span>&gt;<span class="nt">&lt;</span><span class="err">&lt;</span><span class="nt">&gt;&lt;&gt;</span>&gt;<span class="nt">&lt;</span><span class="err">9</span><span class="nt">&gt;</span>&gt;<span class="nt">&lt;&gt;</span>&gt;&gt;9&gt;&gt;&gt;<span class="nt">&lt;&gt;</span>	XT:A:R	NM:i:2	SM:i:0	AM:i:0	X0:i:5	X1:i:0	XM:i:0	XO:i:1	XG:i:2	MD:Z:35
</code></pre></div></div>

<p><span>Speaker Notes</span></p>

<p>The SAM Format is a text format for storing sequence data in a series of tab delimited ASCII columns. Most often it is
generated as a human readable version of its sister BAM format, which stores the same data in a compressed, indexed, binary form.</p>

<p>SAM format files are generated following mapping of the reads to reference sequence. It is TAB-delimited text format
with header and a body. Header lines start with ‘@’ while alignment lines do not. Header hold generic information on
SAM file along with version information, if the file is sorted, information on reference sequence, etc. The alignment
records constitute the body of the file. Each alignment line/record has 11 mandatory fields describing essential alignment information.</p>

<hr />

<h2 id="alignment-formats-1">Alignment formats</h2>

<p><strong>BAM (Binary Alignment/Map)</strong></p>

<p>File extensions: file.bam</p>

<p>A BAM file is the compressed binary version of the Sequence Alignment/Map (SAM).</p>

<p><span>Speaker Notes</span></p>

<p>a compact and indexable representation of nucleotide sequence alignments. The data between SAM and BAM is exactly
same. Being Binary BAM files are small in size and ideal to store alignment files. Require samtools to view the file.</p>

<hr />

<h2 id="featuresannotations-formats">Features/annotations formats</h2>

<p><strong>VCF (Variant Calling Format/File)</strong></p>

<p>File extensions: file.vcf</p>

<p>Example:</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gu">##fileformat=VCFv4.2</span>
<span class="gu">##fileDate=20090805</span>
<span class="gu">##source=myImputationProgramV3.1</span>
<span class="gu">##reference=file:///seq/references/1000GenomesPilot-NCBI36.fasta</span>
<span class="gu">##contig=&lt;ID=20,length=62435964,assembly=B36,md5=f126cdf8a6e0c7f379d618ff66beb2da,species="Homo sapiens",taxonomy=x&gt;</span>
<span class="gu">##phasing=partial</span>
<span class="gu">##INFO=&lt;ID=NS,Number=1,Type=Integer,Description="Number of Samples With Data"&gt;</span>
<span class="gu">##INFO=&lt;ID=DP,Number=1,Type=Integer,Description="Total Depth"&gt;</span>
<span class="gu">##INFO=&lt;ID=AF,Number=A,Type=Float,Description="Allele Frequency"&gt;</span>
...
20 14370 rs6054257 G A 29 PASS NS=3;DP=14;AF=0.5;DB;H2 GT:GQ:DP:HQ 0|0:48:1:51,51 1|0:48:8:51,51 1/1:43:5:.,.
20 17330 . T A 3 q10 NS=3;DP=11;AF=0.017 GT:GQ:DP:HQ 0|0:49:3:58,50 0|1:3:5:65,3 0/0:41:3
20 1110696 rs6040355 A G,T 67 PASS NS=2;DP=10;AF=0.333,0.667;AA=T;DB GT:GQ:DP:HQ 1|2:21:6:23,27 2|1:2:0:18,2 2/2:35:4
20 1230237 . T . 47 PASS NS=3;DP=13;AA=T GT:GQ:DP:HQ 0|0:54:7:56,60 0|0:48:4:51,51 0/0:61:2
20 1234567 microsat1 GTC G,GTCT 50 PASS NS=3;DP=9;AA=G GT:GQ:DP 0/1:35:4 0/2:17:2 1/1:40:3
</code></pre></div></div>

<p><span>Speaker Notes</span></p>

<p>VCF is a text file format with a header (information VCF version, sample etc) and data lines constitute the body of file.</p>

<hr />

<h2 id="featuresannotations-formats-1">Features/annotations formats</h2>

<p><strong>GFF (General Feature Format or Gene Finding Format)</strong></p>

<p>File extensions: file.gff2, file. gff3, file.gff</p>

<p>Example (GFF2):</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>browser position chr22:10000000-10025000

browser hide all

track name=regulatory description="TeleGene(tm) Regulatory Regions"

visibility=2

chr22 TeleGene enhancer 10000000 10001000 500 + . touch1

chr22 TeleGene promoter 10010000 10010100 900 + . touch1

chr22 TeleGene promoter 10020000 10025000 800 - . touch2
</code></pre></div></div>

<p><span>Speaker Notes</span></p>

<p>GFF (General Feature Format or Gene Finding Format). GFF can be used for any kind of feature (Transcripts, exon,
intron, promoter, 3’ UTR, repeatitive elements etc) associated with the sequence, whereas GTF is primarily for
genes/transcripts.  GFF3 is the latest version and an improvement over GFF2 format. However, many databases are
still not equipped to handle GFF3 version. The differences will be explained later in text.</p>

<p>The GFF format has 9 mandatory columns and they are TAB separated.</p>
<ul>
  <li>Col. 1 Reference Sequence</li>
  <li>Col. 2 Source</li>
  <li>Col. 3 Feature</li>
  <li>Col. 4 Start</li>
  <li>Col. 5 End</li>
  <li>Col. 6 Score</li>
  <li>Col. 7 Strand</li>
  <li>Col. 8 Frame (GFF2 and GTF) or Phase (GFF3)</li>
  <li>Col. 9  Attribute or Group field</li>
</ul>

<hr />

<h2 id="featuresannotations-formats-2">Features/annotations formats</h2>

<p><strong>BED (Browser Extensible Data)</strong></p>

<p>The BED (Browser Extensible Data) file format includes information about sequences that can be visualized in a genome
browser; a feature called an annotation track. BED files are tabs-delimited and include 12 fields (columns) of data.</p>

<p>Example of fields: name of chromosome or scaffold, starting position in the chromosome, the ending position…</p>

<hr />

<h2 id="featuresannotations-formats-3">Features/annotations formats</h2>

<p><strong>PSI-MI</strong></p>

<p>The PSI MI format is a data exchange format for molecular interactions.</p>

<p>Example of fields: interaction detection method, biological role, experimental features, location of the interaction, …</p>

<hr />

<h2 id="featuresannotations-formats-4">Features/annotations formats</h2>

<p><strong>PED</strong></p>

<p>File extensions: file.ped</p>

<p>PED is a file format for pedigree analysis, which creates a familial relationship between different samples.</p>

<hr />

<h2 id="structure-formats">Structure formats</h2>

<p><strong>PDB (Protein Data Bank formats)</strong></p>

<p>File extensions: file.pdb</p>

<p>PDB file formats contain atomic coordinates and are used for storing 3D protein structures by the Protein Data Bank.</p>

<p>Example:</p>

<div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>COMPND    UNNAMED
AUTHOR    GENERATED BY OPEN BABEL 2.3.2
ATOM      1  N   ALA A   1       0.000   0.000   0.000  1.00  0.00           N
ATOM      2  CA  ALA A   1       1.456   0.000   0.000  1.00  0.00           C
ATOM      3  C   ALA A   1       1.930   0.000   1.463  1.00  0.00           C
ATOM      4  O   ALA A   1       1.160   0.000   2.421  1.00  0.00           O
...
CONECT  101   98
CONECT  102   94  103
CONECT  103  102
MASTER        0    0    0    0    0    0    0    0  103    0  103    0
END
</code></pre></div></div>

<hr />

<h2 id="other-formats">Other formats</h2>

<p><strong>CSV</strong></p>

<p>CSV (.csv file format) files stands for comma separated value and is a text file, where each line is a row and
columns are delimited with a comma. It can store different types of sequencing data and can be opened using common
spreadsheet programs.</p>

<p><strong>JSON</strong></p>

<p>JSON (JavaScript Object Notation) is a common file format for many other industries, but is used in a growing number
of bioinformatics applications and web resources.</p>

<p><strong>And the list of generic file formats goes on…</strong></p>

<hr />

<h2 id="why-are-there-so-many-different-types">Why Are There So Many Different Types?</h2>

<p>The many different ways of generating and using biological data have given rise to the diversity previously described.
These file formats have their own specific use cases depending on:</p>

<ul>
  <li>Compatibility with specific software</li>
  <li>Data processing, parsing, and human readability needs</li>
  <li>Efficiency for storage</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>In conclusion, the multitude of biological file formats arises from the diverse needs and characteristics of biological data.</p>


	<hr />


<h2>Key Points</h2>

<ul>

<li>Biological data is multi-layered. E.g. the information about one gene can actually regard multiple different biological entities: the variability of its sequence, the derived protein, the diseases associated etc.</li>

<li>Consequently, several different sources of information can be identified and used to describe a biological entity, as well as several different file formats.</li>

</ul>







<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!



<img src="/training-material//assets/images/BioNT_Logo.png" alt="page logo" style="height: 100px;"/>



This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





<div markdown="1">
<h2 id="funding">Funding</h2>
<p>These individuals or organisations provided funding support for the development of this resource</p>

<div class="row">

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="http://biont-training.eu/assets/img/BioNT_Logo_Layout_2106_blue_Version_2_cropped.png" alt="Logo">
		
		<a href="/training-material/hall-of-fame/biont/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		<p>Co-funded by the European Union</p>

	</div>

</div>
</div>




        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/data-science/tutorials/online-resources-gene/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1699662338'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
