<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Submitting sequence data to ENA</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703895990' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703895990' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Submitting sequence data to ENA">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-17 17:21:19 +0000">
<meta name="DC.creator" content="Miguel Roncoroni">
<meta name="DC.creator" content="Bert Droesbeke">

        
        
        
        
        <meta name="description" content="A collection of microtutorials explaining various feature...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Submitting sequence data to ENA">
        <meta property="og:description" content="A collection of microtutorials explaining various feature...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-17 17:21:19 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Submitting sequence data to ENA",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Using Galaxy and Managing your Data",
    "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
    "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Submitting sequence data to ENA' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html",
  "timeRequired": "PT1H",
  "teaches": "Submit raw sequencing reads and metadata to ENA's test server\n - Submit consensus sequence and metadata to ENA's test server",
  "keywords": "galaxy-interface, covid19",
  "description": "The questions this  addresses are:\n - How do you submit raw sequence reads and assembled genomes to the European Nucleotide Archive?\n\n\nThe objectives are:\n - Submit raw sequencing reads and metadata to ENA's test server\n - Submit consensus sequence and metadata to ENA's test server\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html",
      "name": "Removal of human reads from SARS-CoV-2 sequencing data",
      "description": "Hands-on for 'Removal of human reads from SARS-CoV-2 sequencing data' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html",
      "name": "Using dataset collections",
      "description": "Hands-on for 'Using dataset collections' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/roncoronimiguel/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/roncoronimiguel/",
      "name": "Miguel Roncoroni",
      "image": "https://avatars.githubusercontent.com/roncoronimiguel",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bedroesb/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bedroesb/",
      "name": "Bert Droesbeke",
      "image": "https://avatars.githubusercontent.com/bedroesb",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0522-5674",
      "orcid": "https://orcid.org/0000-0003-0522-5674"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Using Galaxy and Managing your Data",
      "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "name": "covid19"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703895990'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/galaxy-interface" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Using Galaxy and Managing your Data
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/galaxy-interface/faqs/" title="Check our FAQs for the Using Galaxy and Managing your Data topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-galaxy-interface">
    <h1 data-toc-skip>Submitting sequence data to ENA</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/roncoronimiguel/" class="contributor-badge contributor-roncoronimiguel"><img src="https://avatars.githubusercontent.com/roncoronimiguel?s=36" alt="Avatar" width="36" height="36">Miguel Roncoroni</a><a href="/training-material/hall-of-fame/bedroesb/" class="contributor-badge contributor-bedroesb"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bedroesb?s=36" alt="Avatar" width="36" height="36">Bert Droesbeke</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do you submit raw sequence reads and assembled genomes to the European Nucleotide Archive?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Submit raw sequencing reads and metadata to ENA’s test server</p>
</li>
        
        <li><p>Submit consensus sequence and metadata to ENA’s test server</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
        
    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Removal of human reads from SARS-CoV-2 sequencing data:
                            
                            
                                
                                     <a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/galaxy-interface">Using Galaxy and Managing your Data</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Using dataset collections:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.5163611">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/workflows/" title="Submitting sequence data to ENA workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 17, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00159">https://gxy.io/GTN:T00159</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p>Raw reads contain valuable information, such as coverage depth and quality scores, that is lost in a consensus sequence.
Submission of raw reads to public repositories allows reuse of data and reproducibility of analysis and enables discovery of minor allelic variants and intrahost variation, for example during the recent COVID-19 pandemic (<span class="citation"><a href="#Maier2021">Maier <i>et al.</i> 2021</a></span>).</p>

<p>The European Nucleotide Archive is an Open and FAIR repository of nucleotide data. As part of the International Nucleotide Sequence Database Collaboration (INSDC), ENA also indexes data from the NCBI and DDBJ <span class="citation"><a href="#Arita2020">Arita <i>et al.</i> 2020</a></span>. Data submitted to ENA must be accompanied by sufficient metadata. You can learn more from this <a href="/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/slides.html">introductory slide deck</a> or directly from <a href="https://ena-docs.readthedocs.io/en/latest/submit/general-guide/metadata.html">ENA</a>.</p>

<p>In this tutorial we will show you how to use Galaxy’s ‘ENA Upload tool’ to submit raw sequencing reads, consensus sequences and their associated metadata to ENA <span class="citation"><a href="#Roncoroni2021">Roncoroni <i>et al.</i> 2021</a></span>. You will learn to add your ENA Webin credentials to Galaxy, input metadata interactively or via a metadata template and submit the reads to ENA (test) server using Galaxy’s ‘ENA upload tool’.
Specifically, we will use one ONT sequencing file to demonstrate interactive metadata input and two sets of PE Illumina reads to demonstrate how to use the ENA metadata template. Finally, we will submit consensus sequences to ENA using ‘Submit consensus sequence to ENA’ tool.</p>

<p>Data will be submitted to ENA’s test server and will not be public.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-nature-of-the-input-data"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Nature of the input data</div>
  <p>We will use data derived from sequencing data of bronchoalveolar lavage fluid (BALF) samples obtained from early COVID-19 patients in China as our input data.
Human traces have been removed in <a href="/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html">Galaxy</a>.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#adding-ena-webin-credentials-to-your-galaxy-user-information" id="markdown-toc-adding-ena-webin-credentials-to-your-galaxy-user-information">Adding ENA Webin credentials to your Galaxy user information</a></li>
  <li><a href="#submitting-raw-sequence-data-reads-to-the-ena" id="markdown-toc-submitting-raw-sequence-data-reads-to-the-ena">Submitting raw sequence data (reads) to the ENA</a>    <ol>
      <li><a href="#option-1-submitting-to-ena-using-interactive-metadata-generator" id="markdown-toc-option-1-submitting-to-ena-using-interactive-metadata-generator">Option 1: submitting to ENA using interactive metadata generator</a></li>
      <li><a href="#option-2-submitting-to-ena-using-a-metadata-template" id="markdown-toc-option-2-submitting-to-ena-using-a-metadata-template">Option 2: submitting to ENA using a metadata template</a></li>
    </ol>
  </li>
  <li><a href="#submitting-consensus-sequences-to-ena" id="markdown-toc-submitting-consensus-sequences-to-ena">Submitting consensus sequences to ENA</a></li>
</ol>

</blockquote>

<h1 id="adding-ena-webin-credentials-to-your-galaxy-user-information">Adding ENA Webin credentials to your Galaxy user information</h1>

<p>In order to submit data to ENA, you need to have a valid Webin account. If you don’t have one already you can register for one <a href="https://www.ebi.ac.uk/ena/submit/sra/#registration">on ENA’s Registration Page</a>. Webin credentials need to be included in your Galaxy user information before you can use the ENA Upload tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-add-webin-credentials-to-your-galaxy-user-information"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Add Webin credentials to your Galaxy user information</div>

  <ol>
    <li>If you have not already done so, log in to usegalaxy.eu</li>
    <li>Navigate to <em>“User”</em> &gt; <em>“Preferences”</em> on the top menu
      <ul>
        <li>Click on <i class="fa fa-user" aria-hidden="true"></i> <strong>Manage Information</strong></li>
      </ul>
      <ul>
        <li>Scroll down to <em>“Your ENA Webin account details”</em> and fill in your <strong><em>ENA Webin ID</em></strong> and <strong><em>Password</em></strong>
<a href="../../images/upload-data-to-ena/ENA-credentials.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/upload-data-to-ena/ENA-credentials.png" alt="ENA Webin Account details in Galaxy. " width="212" height="165" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> ENA Webin Account details&lt;/figcaption&gt;</figure></a></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h1 id="submitting-raw-sequence-data-reads-to-the-ena">Submitting raw sequence data (reads) to the ENA</h1>
<h2 id="option-1-submitting-to-ena-using-interactive-metadata-generator">Option 1: submitting to ENA using interactive metadata generator</h2>

<p>In this first example, you will submit one ONT sequence file using the interactive metadata forms from the ENA Upload tool. This method is only convenient for small submissions. For bulk submissions, we recommend you use the metadata template described below in Option 2.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Upload the ONT data from Zenodo via URLs</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>The URL for our example data is this:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6912963/files/SRR10902284_ONT.fq.gz
</code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>Once the data is uploaded, we fill the metadata using  the ENA Upload tool. Interactive metadata forms are nested to fit <a href="https://ena-docs.readthedocs.io/en/latest/submit/general-guide/metadata.html">ENA’s metadata model</a>. Briefly, you add Samples to a Study, Experiments to Samples and Runs to Experiments. The interactive metadata form does only two two <a href="https://www.ebi.ac.uk/ena/browser/checklists">ENA Sample Checklists</a>, the basic minimal sample metadata and the ENA virus pathogen reporting standard checklist. Switch between the basic template and the virus pathogen one under the “Does your submission contains viral samples?” question. If you wish to include additional metadata from a sample checklist, please use Option 2 below.</p>

<p>We recommend always submitting to the test server before submitting to the public one.
After you confirm that all the data and metadata looks ok, you can go ahead and submit to the public ENA server.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-add-metadata-interactively-and-submit-a-single-sequence-to-ena"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: add metadata interactively and submit a single sequence to ENA</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ena_upload/ena_upload/0.6.1" title="ENA Upload tool tool" aria-role="link"><strong>ENA Upload tool</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/ena_upload/ena_upload/0.6.1</span></span>:
      <ul>
        <li><em>“Action to execute”</em>: <code class="language-plaintext highlighter-rouge">Add new data</code></li>
        <li>Under “Testing options”:
          <ul>
            <li><em>“Submit to test ENA server?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
            <li><em>“Print the tables but do not submit the datasets”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
          </ul>
        </li>
        <li><em>“Would you like to submit pregenerated table files or interactively define the input structures?”</em>: <code class="language-plaintext highlighter-rouge">Interactive generation of the study structure</code></li>
        <li><em>“Add .fastq.(gz.bz2) extension to the Galaxy dataset names to match the ones described in the input tables?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Does your submission contains viral samples?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
    <li>Fill all metadata boxes and make sure that:
      <ul>
        <li><em>”“</em>:</li>
        <li><em>“Please select the type of study”</em>: <code class="language-plaintext highlighter-rouge">Whole Genome Sequencing</code></li>
        <li><em>“Enter the species of the sample”</em>: <code class="language-plaintext highlighter-rouge">Severe acute respiratory syndrome coronavirus 2</code></li>
        <li><em>“Enter the taxonomic ID corresponding to the sample species”</em>: <code class="language-plaintext highlighter-rouge">2697049</code></li>
        <li><em>“Host common name”</em>: <code class="language-plaintext highlighter-rouge">human</code></li>
        <li><em>“Host subject id”</em>: avoid using ID that can be use to trace samples back to patients</li>
        <li><em>“Host scientific name”</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens</code></li>
        <li><em>“Library strategy”</em>: <code class="language-plaintext highlighter-rouge">RNA-Seq</code></li>
        <li><em>“Select library source”</em>: <code class="language-plaintext highlighter-rouge">METAGENOMIC</code></li>
        <li><em>“Library selection”</em>: <code class="language-plaintext highlighter-rouge">RANDOM</code></li>
        <li><em>“Library layout”</em>: <code class="language-plaintext highlighter-rouge">SINGLE</code></li>
        <li><em>“Select the sequencing platform used”</em>: <code class="language-plaintext highlighter-rouge">Oxford Nanopore</code></li>
        <li><em>“Instrument model”</em>: <code class="language-plaintext highlighter-rouge">minION</code></li>
        <li><em>“Runs executed within this experiment”</em>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“File(s) associated with this run”</em>: <code class="language-plaintext highlighter-rouge">SRR10902284_ONT.fq.gz</code></li>
          </ul>
        </li>
        <li><em>“Affiliation center”</em>: your institution</li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-do-not-include-personal-identifiable-data"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Do not include personal identifiable data</div>
  <p>In some cases, some information is requested by ENA that may classify as personal or could be used to identify persons (e.g. ‘host ID’ for checklist ERC000033). Make sure that you do not publish any personal metadata that infringes privacy protection regulations in your jurisdiction.</p>
</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-submit-to-the-test-server-first"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Submit to the test server first</div>
  <p>Make sure  <em>“Submit to test ENA server?”</em>: <code class="language-plaintext highlighter-rouge">yes</code>. Otherwise your data will be submitted to the public server.</p>
</blockquote>

<p>Four metadata tables (Study, Sample, Experiment and Run), and a metadata ticket with submission information are generated. You can confirm a successful submission at ENA <a href="https://wwwdev.ebi.ac.uk/ena/submit/webin/">test server</a> (or the <a href="https://www.ebi.ac.uk/ena/submit/sra">public server</a>, if you chose it).</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-metadata-ticket"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-metadata-ticket" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Metadata ticket<span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Upon successful submission, a metadata ticket is generated. This contains information of the submission, including parseable metadata. Importantly, it contains Study, Sample, Run and Experiment accession numbers. The former two you will use later to link the consensus sequence to the raw data.</p>
</blockquote>

<h2 id="option-2-submitting-to-ena-using-a-metadata-template">Option 2: submitting to ENA using a metadata template</h2>

<p>For larger submissions, interactive metadata input can be tedious and not practical. In the second example, you will submit two sets of Illumina PE sequence files and input metadata using a template spreadsheet. This template contains all fields for ENA sample checklist <em>‘ERC000033 - ENA virus pathogen reporting standard checklist’</em>. Tabular (.tsv and .xlsx) metadata templates for all sample checklist can be found in <a href="https://github.com/ELIXIR-Belgium/ENA-metadata-templates">this repository</a>.
For this tutorial, we provide you with a pre-filled template and encourage you to explore it.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-upload-and-inspect-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Upload and inspect data</div>

  <ol>
    <li>
      <p>Upload the ONT data from Zenodo via URLs:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6912963/files/SRR10903401_1.fastq.gz
https://zenodo.org/record/6912963/files/SRR10903401_2.fastq.gz
https://zenodo.org/record/6912963/files/SRR10903402_1.fastq.gz
https://zenodo.org/record/6912963/files/SRR10903402_2.fastq.gz
https://zenodo.org/record/6912963/files/metadata_template_ERC000033_mock_complete.xlsx
</code></pre></div>      </div>
    </li>
    <li>
      <p>Arrange the data into a paired dataset collection</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-paired-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-paired-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a paired collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build List of Dataset Pairs</strong></p>   </li>   <li>Change the text of <em>unpaired forward</em> to a common selector for the forward reads</li>   <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse reads</li>   <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>For the example datasets this means:</p>
      <ul>
        <li>You need to tell Galaxy about the suffix for your forward and reverse reads, respectively:
          <ul>
            <li>set the text of <em>unpaired forward</em> to: <code class="language-plaintext highlighter-rouge">_1.fastq.gz</code></li>
            <li>set the text of <em>unpaired reverse</em> to: <code class="language-plaintext highlighter-rouge">_2.fastq.gz</code></li>
            <li>click: <code class="language-plaintext highlighter-rouge">Auto-pair</code></li>
          </ul>

          <p>All datasets should now be moved to the <em>paired section</em> of the dialog, and the middle column there should show that only the sample accession numbers, <em>i.e.</em> <code class="language-plaintext highlighter-rouge">SRR10903401</code> and <code class="language-plaintext highlighter-rouge">SRR10903402</code>, will be used as the pair names.</p>
        </li>
      </ul>

      <blockquote class="warning">
        <div class="box-title warning-title" id="warning-paired-collection-names"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Paired collection names</div>
        <p>It is very important that the paired collection names contain no suffix (e.g. _1, _R1, etc.) or file extensions (.fastq, .fastq.gz). The submission tool will add these at runtime and leaving them in the paired collection names will cause a mismatch with the filenames in the metadata table.</p>
      </blockquote>
      <ul>
        <li>Make sure <em>Hide original elements</em> is checked to obtain a cleaned-up history after building the collection.</li>
        <li>Give your collection a name</li>
        <li>Click <em>Create Collection</em></li>
      </ul>
    </li>
    <li>
      <p>Inspect the <code class="language-plaintext highlighter-rouge">GTN_tutorial_mock_metadata.xlsx</code> (filled-in template) file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon</p>

      <p>https://github.com/enasequence/webin-cli</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How many metadata sheets are there?</li>
          <li>How is the ‘Sample’ section in the template different from that in the interactive metadata input?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>There are four metadata sheets, one per metadata object (Study, Sample, Experiment, Run)</li>
            <li>The Sample section is more extensive in the template spreadsheet, because it contains all the fields from ENA ERC000033 sample checklist, as well as all ‘Recommended’ and ‘Optional’ fields in the other sections.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>As before, the submission is done to the test server before submitting to the public one.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-a-metadata-template-and-submit-multiple-sequences-to-ena"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: use a metadata template and submit multiple sequences to ENA</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ena_upload/ena_upload/0.6.1" title="ENA Upload tool tool" aria-role="link"><strong>ENA Upload tool</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/ena_upload/ena_upload/0.6.1</span></span>:
      <ul>
        <li><em>“Action to execute”</em>: <code class="language-plaintext highlighter-rouge">Add new data</code></li>
        <li>Under “Testing options”:
          <ul>
            <li><em>“Submit to test ENA server?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
            <li><em>“Print the tables but do not submit the datasets”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
          </ul>
        </li>
        <li><em>“Would you like to submit pregenerated table files or interactively define the input structures?”</em>: <code class="language-plaintext highlighter-rouge">User generated metadata tables based on Excel templates</code></li>
        <li><em>“Select the metadata checklist”</em>: <code class="language-plaintext highlighter-rouge">ENA virus pathogen reporting standard checklist (ERC000033)</code></li>
        <li><em>“Select Excel (xlsx) file based on template”</em>: <code class="language-plaintext highlighter-rouge">metadata_template_ERC000033_mock_complete.xlsx</code></li>
        <li><em>“Select runs input format”</em>: <code class="language-plaintext highlighter-rouge">Input from a paired collection</code></li>
        <li><em>“List of paired-end runs files”</em>: select the collection you made above during data upload</li>
        <li><em>“Affiliation center”</em>: your institution</li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-do-not-include-personal-identifiable-data-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Do not include personal identifiable data</div>
  <p>In some cases, some information is requested by ENA that may classify as personal or could be used to identify persons (e.g. ‘host ID’ for checklist ERC000033). Make sure that you do not publish any personal metadata that infringes privacy protection regulations in your jurisdiction.</p>
</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-submit-to-the-test-server-first-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Submit to the test server first</div>
  <p>Make sure  <em>“Submit to test ENA server?”</em>: <code class="language-plaintext highlighter-rouge">yes</code>. Otherwise your data will be submitted to the public server.</p>
</blockquote>

<p>Four metadata tables (Study, Sample, Experiment and Run), and a metadata ticket with submission information are generated. You can confirm a successful submission at ENA <a href="https://wwwdev.ebi.ac.uk/ena/submit/webin/">test server</a> (or the <a href="https://www.ebi.ac.uk/ena/submit/sra">public server</a>, if you chose it).</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-metadata-ticket-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-metadata-ticket-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Metadata ticket<span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Upon succesful submission, a metadata ticket is generated. This contains information of the submission, including parseable metadata. Importantly, it contains Study, Sample, Run and Experiment accession numbers. The former two you will use later to link the consensus sequence to the raw data.</p>
</blockquote>

<h1 id="submitting-consensus-sequences-to-ena">Submitting consensus sequences to ENA</h1>

<p>We produced consensus sequences for the Illumina data from Option 2 above following <a href="https://workflowhub.eu/workflows/113?version=4">SARS-CoV-2-PE-Illumina-WGS-variant-calling</a>, <a href="https://workflowhub.eu/workflows/109?version=5">SARS-CoV-2-variation-reporting</a> and <a href="https://workflowhub.eu/workflows/138?version=4">COVID-19-consensus-construction</a> workflows.</p>

<p>In this step we will submit one consensus sequence to ENA.
We will link it to the reads submitted in the first step using the accession numbers given to those submissions. Galaxy’s ‘ENA Upload tool’ captures and stores metadata on a metadata ticket.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-submit-consensus-sequences-to-ena"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: submit consensus sequences to ENA</div>

  <ol>
    <li>Upload the consensus sequence to Galaxy from Zenodo via URLs:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/6912963/files/SRR10903401.fasta
</code></pre></div>      </div>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Open the ‘ENA submsission receipt’ and find the Study and Sample accession numbers from the raw data submission.</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ieguinoa/ena_webin_cli/ena_webin_cli/7d751b5943b0" title="Submit consensus sequence to ENA tool" aria-role="link"><strong>Submit consensus sequence to ENA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ieguinoa/ena_webin_cli/ena_webin_cli/7d751b5943b0</span></span>:
      <ul>
        <li><em>“Submit to test ENA server?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li><em>“Validate files and metadata but do not submit”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li>Fill the assembly metadata. For our assembly:
          <ul>
            <li><em>“Assembly type”</em>: <code class="language-plaintext highlighter-rouge">Clone</code></li>
            <li><em>“Assembly program”</em>: <code class="language-plaintext highlighter-rouge">BWA-MEM</code></li>
            <li><em>“Molecule type”</em>: <code class="language-plaintext highlighter-rouge">genomic RNA</code></li>
            <li><em>“Coverage”</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
            <li><em>“Select the method to load study and sample metadata”</em>: <code class="language-plaintext highlighter-rouge">Fill in required metadata</code></li>
            <li><em>“Assembly name”</em>: give a name to your assembly.</li>
            <li><em>“Study accession”</em>: <code class="language-plaintext highlighter-rouge">ERP139884</code> (you can find the Study accession number from your raw data submission metadata ticket)</li>
            <li><em>“Sample accession”</em>: <code class="language-plaintext highlighter-rouge">ERS12519941</code> (you can find the Sample accession number from your raw data submission metadata ticket)</li>
            <li><em>“Sequencing platform”</em>: <code class="language-plaintext highlighter-rouge">Illumina</code></li>
          </ul>
        </li>
        <li>Select the consensus sequence assembly file from your history: <code class="language-plaintext highlighter-rouge">SRR10903401.fasta</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output are a list of manifest (in our example one manifest) and a ‘submission log’. The manifest file is the metadata required by <a href="https://github.com/enasequence/webin-cli">ENA’s Webin CLI tool</a>. The ‘submission log’ can be useful to troubleshoot failed submissions.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Use Galaxy’s ‘ENA Upload tool’ to submit raw reads to ENA</p>
</li>
                        
                        <li><p>Use Galaxy’s ‘Submit consensus sequence to ENA’ tool</p>
</li>
                        
                        <li><p>You need to include your ENA Webin credentials in Galaxy</p>
</li>
                        
                        <li><p>For small submission use ‘ENA Upload tool’ interactive metadata forms feature</p>
</li>
                        
                        <li><p>For bulk submissions use a spreadsheet metadata template</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/galaxy-interface/faqs/">FAQ page for the Using Galaxy and Managing your Data topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Arita2020">Arita, M., I. Karsch-Mizrachi, and G. Cochrane, 2020 <b>The international nucleotide sequence database collaboration</b>. Nucleic Acids Research 49: D121–D124. <a href="https://doi.org/10.1093/nar/gkaa967">10.1093/nar/gkaa967</a></li>
<li id="Maier2021">Maier, W., S. Bray, M. van den Beek, D. Bouvier, N. Coraor <i>et al.</i>, 2021 <b>Freely accessible ready to use global infrastructure for SARS-CoV-2 monitoring</b>. <a href="https://doi.org/10.1101/2021.03.25.437046">10.1101/2021.03.25.437046</a></li>
<li id="Roncoroni2021">Roncoroni, M., B. Droesbeke, I. Eguinoa, K. D. Ruyck, F. D’Anna <i>et al.</i>, 2021 <b>A SARS-CoV-2 sequence submission tool for the European Nucleotide Archive</b> (Z. Lu, Ed.). Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/btab421">10.1093/bioinformatics/btab421</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Submitting sequence data to ENA (Using Galaxy and Managing your Data)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Miguel Roncoroni, Bert Droesbeke,  <b>Submitting sequence data to ENA (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html">https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{galaxy-interface-upload-data-to-ena,
author = "Miguel Roncoroni and Bert Droesbeke",
	title = "Submitting sequence data to ENA (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: ena_upload
  owner: iuc
  revisions: 9e2df763086c
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ena_upload
  owner: iuc
  revisions: a62c4a11a67d
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00159">https://gxy.io/GTN:T00159</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703895990'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>