<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Analyses of metagenomics data - The global picture</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1697502352' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1697502352' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Analyses of metagenomics data - The global picture">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-10-20 16:51:14 +0000">
<meta name="DC.creator" content="Saskia Hiltemann">
<meta name="DC.creator" content="Bérénice Batut">

        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Analyses of metagenomics data - The global picture">
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2022-10-20 16:51:14 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Analyses of metagenomics data - The global picture",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Analyses of metagenomics data - The global picture' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html",
  "timeRequired": "PT2H30M",
  "teaches": "Choosing the best approach to analyze metagenomics data\n - Selection of tools to analyze amplicon data or shotgun data\n - Visualisation of a community structure",
  "keywords": "metagenomics",
  "description": "The questions this  addresses are:\n - How to analyze metagenomics data?\n - What information can be extracted of metagenomics data?\n - What is the difference between amplicon and shotgun data?\n - What are the difference in the analyses of amplicon and shotgun data?\n\n\nThe objectives are:\n - Choosing the best approach to analyze metagenomics data\n - Selection of tools to analyze amplicon data or shotgun data\n - Visualisation of a community structure\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "name": "Saskia Hiltemann",
      "image": "https://avatars.githubusercontent.com/shiltemann",
      "description": "Researcher at Erasmus Medical Center",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3803-468X",
      "orcid": "https://orcid.org/0000-0003-3803-468X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1697502352'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/general-tutorial/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Analyses of metagenomics data - The global picture</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Avatar" width="36" height="36">Saskia Hiltemann</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Avatar" width="36" height="36">Bérénice Batut</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to analyze metagenomics data?</p>
</li>
        
        <li><p>What information can be extracted of metagenomics data?</p>
</li>
        
        <li><p>What is the difference between amplicon and shotgun data?</p>
</li>
        
        <li><p>What are the difference in the analyses of amplicon and shotgun data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Choosing the best approach to analyze metagenomics data</p>
</li>
        
        <li><p>Selection of tools to analyze amplicon data or shotgun data</p>
</li>
        
        <li><p>Visualisation of a community structure</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours 30 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.815875">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/metagenomics/tutorials/general-tutorial/workflows/" title="Analyses of metagenomics data - The global picture workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/general-tutorial/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Oct 20, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00201">https://gxy.io/GTN:T00201</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>In metagenomics, information about micro-organisms in an environment can be extracted with two main techniques:</p>

<ul>
  <li>Amplicon sequencing, which sequences only the rRNA or ribosomal DNA of organisms</li>
  <li>Shotgun sequencing, which sequences full genomes of the micro-organisms in the environment</li>
</ul>

<p>In this tutorial, we will introduce the two main types of analyses with their general principles and differences. For a more in-depth look at these analyses, we recommend our detailed tutorials on each analysis.</p>

<p>We will use two datasets (one amplicon and one shotgun) from the same <a href="https://www.ebi.ac.uk/metagenomics/projects/SRP016633">project on the Argentinean agricultural pampean soils</a>. In this project, three different geographic regions that are under different types of land uses and two soil types (bulk and rhizospheric) were analyzed using shotgun and amplicon sequencing. We will focus on data from the Argentina Anguil and Pampas Bulk Soil (the original study included one more geographical regions <span class="citation"><a href="#Rascovan2013">Rascovan <i>et al.</i> 2013</a></span>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#amplicon-data" id="markdown-toc-amplicon-data">Amplicon data</a>    <ol>
      <li><a href="#importing-the-data" id="markdown-toc-importing-the-data">Importing the data</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a></li>
      <li><a href="#sequence-alignment" id="markdown-toc-sequence-alignment">Sequence Alignment</a></li>
      <li><a href="#extraction-of-taxonomic-information" id="markdown-toc-extraction-of-taxonomic-information">Extraction of taxonomic information</a></li>
      <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
    </ol>
  </li>
  <li><a href="#shotgun-metagenomics-data" id="markdown-toc-shotgun-metagenomics-data">Shotgun metagenomics data</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#extraction-of-taxonomic-information-1" id="markdown-toc-extraction-of-taxonomic-information-1">Extraction of taxonomic information</a></li>
      <li><a href="#extraction-of-functional-information" id="markdown-toc-extraction-of-functional-information">Extraction of functional information</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="amplicon-data">Amplicon data</h1>

<p>Amplicon sequencing is a highly targeted approach for analyzing genetic variation in specific genomic regions.
In the metagenomics fields, amplicon sequencing refers to capture and sequence of rRNA data in a sample.
It can be 16S for bacteria or archea or 18S for eukaryotes.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-background-the-16s-ribosomal-rna-gene"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Background: The 16S ribosomal RNA gene</div>
  <p><a href="../../images/16S_gene.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/16S_gene.png" alt="The 16S ribosomal RNA gene. " width="1374" height="927" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> The 16S ribosomal RNA gene. Credit: https://www.slideshare.net/beiko/ccbc-tutorial-beiko&lt;/figcaption&gt;</figure></a></p>

  <p>The 16S rRNA gene has several properties that make it ideally suited to our purposes</p>

  <ol>
    <li>Present in all living organisms</li>
    <li>Highly conserved + highly variable regions</li>
    <li>Huge reference databases</li>
  </ol>

  <p><a href="../../images/16S_variableregions.jpg" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/16S_variableregions.jpg" alt="Variable regions. " width="1200" height="626" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Variable regions of the 16S rRNA&lt;/figcaption&gt;</figure></a></p>

  <p>The highly conserved regions make it easy to target the gene across different organisms, while the highly variable regions allow us to distinguish between different species.</p>

</blockquote>

<p>With amplicon data, we can determine the micro-organisms from which the sequences in our sample are coming from. This is called taxonomic assignation.
We try to assign sequences to taxons and then classify or extract the taxonomy in our sample.</p>

<p>In this analysis, we will use the <a href="https://mothur.org">mothur tool suite</a>, but only a small portion of its tools and possibilities.
To learn more in detail about how to use this tool, check out the full <a href="../mothur-miseq-sop/tutorial.html">mothur tutorial</a>.</p>

<h2 id="importing-the-data">Importing the data</h2>

<p>Our datasets come from a soil samples in two different Argentinian locations, for which the 16S rDNA V4 region
has been sequenced using 454 GS FLX Titanium. For the tutorial, the original fastq data has been down sampled and converted to fasta. The original data are available at EBI Metagenomics under the following run numbers:</p>

<ul>
  <li>Pampa soil: <a href="https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS353016/runs/SRR531818/results/versions/2.0">SRR531818</a></li>
  <li>Anguil soil: <a href="https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS386929/runs/SRR651839/results/versions/2.0">SRR651839</a></li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Import the following files from <a href="https://zenodo.org/record/815875">Zenodo</a>, or from the shared data library (in <code class="language-plaintext highlighter-rouge">GTN - Material -&gt; metagenomics -&gt; Analyses of metagenomics data - The global picture</code>):
      <ul>
        <li><code class="language-plaintext highlighter-rouge">SRR531818_pampa.fasta</code></li>
        <li><code class="language-plaintext highlighter-rouge">SRR651839_anguil.fasta</code></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/815875/files/SRR531818_pampa.fasta
https://zenodo.org/record/815875/files/SRR651839_anguil.fasta
</code></pre></div>      </div>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data via links<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Go into “Shared data” (top panel) then “Data libraries”</li>
          <li>Click on “Training data” and then “Analyses of metagenomics data”</li>
          <li>Select interesting file</li>
          <li>Click on “Import selected datasets into history”</li>
          <li>Import in a new history</li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name, so rename them.</p>
    </li>
  </ol>

</blockquote>

<!--

Anguil Soil: https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS386929
Pampa Soil https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS353016/runs/SRR531818/results/versions/2.0

Project's data: https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS353016/runs/SRR531818/results/versions/2.0
Project's pipeline: https://www.ebi.ac.uk/metagenomics/pipelines/2.0
Project's QC results: https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS386929/runs/SRR651839/results/versions/2.0
-->

<h3 id="preparing-datasets">Preparing datasets</h3>

<p>We will perform a multisample analysis with mothur. To do this, we will merge all reads into a single file,
and create a <em>group file</em>, indicating which reads belong to which samples.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-prepare-multisample-analysis"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Prepare multisample analysis</div>

  <ol>
    <li><strong>Merge.files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Merge” to <code class="language-plaintext highlighter-rouge">fasta files</code></li>
        <li>“Inputs” to the two sample fasta files</li>
      </ul>
    </li>
    <li><strong>Make.group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Method to create group file” to <code class="language-plaintext highlighter-rouge">Manually specify fasta files and group names</code></li>
        <li>“Additional”: Add two elements to this repeat
          <ul>
            <li>First element
              <ul>
                <li>“fasta - Fasta to group” to <code class="language-plaintext highlighter-rouge">SRR531818_pampa</code> file</li>
                <li>“group - Group name” to <code class="language-plaintext highlighter-rouge">pampa</code></li>
              </ul>
            </li>
            <li>Second element (click on “Insert Additional”)
              <ul>
                <li>“fasta - Fasta to group” to <code class="language-plaintext highlighter-rouge">SRR651839_anguil</code> file</li>
                <li>“group - Group name” to <code class="language-plaintext highlighter-rouge">anguil</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-note"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Note</div>

  <p>Because we only have a small number of samples, we used the manual specification. If you have hundreds of samples this would quickly become bothersome. The solution? use a collection! To read more about collections in Galaxy, please see <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">dedicated collections tutorial</a>.</p>
</blockquote>

<p>Have a look at the group file. It is a very simple file, it contains two columns: the first contains the read names, the second contains the group (sample) name, in our case <code class="language-plaintext highlighter-rouge">pampa</code> or <code class="language-plaintext highlighter-rouge">anguil</code>.</p>

<h3 id="optimization-of-files-for-computation">Optimization of files for computation</h3>

<p>Because we are sequencing many of the same organisms, we anticipate that many of our sequences are
duplicates of each other. Because it’s computationally wasteful to align the same thing a bazillion
times, we’ll unique our sequences using the <code class="language-plaintext highlighter-rouge">Unique.seqs</code> command:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-duplicate-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove duplicate sequences</div>

  <ol>
    <li><strong>Unique.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” to the merged fasta file</li>
        <li>“output format” to <code class="language-plaintext highlighter-rouge">Name File</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>How many sequences were unique? How many duplicates were removed?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>19,502 unique sequences and 498 duplicates.</p>

          <p>This can be determined from the number of lines in the fasta (or names) output, compared to the
number of lines in the fasta file before this step.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>This <code class="language-plaintext highlighter-rouge">Unique.seqs</code> tool outputs two files: a fasta file containing only the unique sequences, and a <em>names file</em>.
The names file consists of two columns, the first contains the sequence names for each of the unique sequences, and the second column contains all other sequence names that are identical to the representative sequence in the first column.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>name          representatives
read_name1    read_name2,read_name,read_name5,read_name11
read_name4    read_name6,read_name,read_name10
read_name7    read_name8
...
</code></pre></div></div>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-count-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Count sequences</div>

  <ol>
    <li><strong>Count.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“name” to the name file from <code class="language-plaintext highlighter-rouge">Unique.seqs</code></li>
        <li>“Use a group file” to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>“group” to the group file from <code class="language-plaintext highlighter-rouge">Make.group</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">Count.seqs</code> file keeps track of the number of sequences represented by each unique representative across multiple samples. We will pass this file to many of the following tools to be used or updated as needed.</p>

<h2 id="quality-control">Quality Control</h2>

<p>The first step in any analysis should be to check and improve the quality of our data.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>For more information on the topic of quality control, please see our <a href="/training-material/topics/sequence-analysis/index.html">dedicated training materials</a>.</p>
</blockquote>

<p>First, let’s get a feel for our data:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-summarize-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Summarize data</div>

  <ol>
    <li><strong>Summary.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” parameter to the fasta from <code class="language-plaintext highlighter-rouge">Unique.seqs</code></li>
        <li>“count” to count table from <code class="language-plaintext highlighter-rouge">Count.seqs</code></li>
        <li>“output logfile?” to <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">summary</code> output files give information per read. The <code class="language-plaintext highlighter-rouge">logfile</code> outputs also contain some summary
statistics:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>              Start    End        NBases     Ambigs   Polymer  NumSeqs
Minimum:      1        80         80         0        3        1
2.5%-tile:    1        104        104        0        3        501
25%-tile:     1        242        242        0        4        5001
Median:       1        245        245        0        4        10001
75%-tile:     1        245        245        0        4        15001
97.5%-tile:   1        247        247        0        6        19501
Maximum:      1        275        275        2        31       20000
Mean:         1        237.519    237.519    0.00495  4.24965
# of unique seqs:   19502
total # of seqs:    20000
</code></pre></div></div>

<p>This tells us that we have a total of 19,502 unique sequences, representing 20,000 total sequences that vary in length between 80 and 275 bases. Also, note that at least some of our sequences had some ambiguous base calls.
Furthermore, at least one read had a homopolymer stretch of 31 bases, this is likely an error so we would like to filter such reads out as well.</p>

<p>If you are thinking that 20,000 is an oddly round number, you are correct; we downsampled the original datasets to 10,000 reads per sample for this tutorial to reduce the amount of time the analysis steps will take.</p>

<p>We can filter our dataset on length, base quality, and maximum homopolymer length using the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> tool</p>

<p>The following tool will remove any sequences with ambiguous bases (<code class="language-plaintext highlighter-rouge">maxambig</code> parameter), homopolymer stretches of 9 or more bases (<code class="language-plaintext highlighter-rouge">maxhomop</code> parameter) and any reads longer than 275 bp or shorter than 225 bp.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-reads-based-on-quality-and-length"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter reads based on quality and length</div>

  <ol>
    <li><strong>Screen.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” to the fasta file from <code class="language-plaintext highlighter-rouge">Unique.seqs</code></li>
        <li>“minlength” parameter to <code class="language-plaintext highlighter-rouge">225</code></li>
        <li>“maxlength” parameter to <code class="language-plaintext highlighter-rouge">275</code></li>
        <li>“maxambig” parameter to <code class="language-plaintext highlighter-rouge">0</code></li>
        <li>“maxhomop” parameter to <code class="language-plaintext highlighter-rouge">8</code></li>
        <li>“count” to the count file from <code class="language-plaintext highlighter-rouge">Count.seqs</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many reads were removed in this screening step? (Hint: run the <code class="language-plaintext highlighter-rouge">Summary.seqs</code> tool again)</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>1,804.</p>

    <p>This can be determined by looking at the number of lines in bad.accnos output of screen.seqs step or by comparing the total number of seqs between of the summary.seqs log before and after this screening step</p>
  </blockquote>
</blockquote>

<h2 id="sequence-alignment">Sequence Alignment</h2>

<p>Aligning our sequences to a reference helps improve OTU assignment [<a href="https://www.ncbi.nlm.nih.gov/pubmed/23018771">Schloss et. al.</a>], so we will now align our sequences to an alignment of the V4 variable region of the 16S rRNA. This alignment has been created as described in [<a href="https://mothur.org/wiki/MiSeq_SOP">mothur’s MiSeq SOP</a>] from the Silva reference database.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-align-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Align sequences</div>

  <ol>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">silva.v4.fasta</code> file in your history</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/815875/files/silva.v4.fasta
</code></pre></div>      </div>
    </li>
    <li><strong>Align.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” to the <code class="language-plaintext highlighter-rouge">good.fasta</code> output from <code class="language-plaintext highlighter-rouge">Screen.seqs</code></li>
        <li>“Select Reference Template from” to <code class="language-plaintext highlighter-rouge">Your history</code></li>
        <li>“reference” to the <code class="language-plaintext highlighter-rouge">silva.v4.fasta</code> reference file</li>
        <li>“flip” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step may take a few minutes, please be patient.</p>
    </li>
    <li><strong>Summary.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” parameter to the aligned output from <code class="language-plaintext highlighter-rouge">Align.seqs</code></li>
        <li>“count” parameter to count_table output from <code class="language-plaintext highlighter-rouge">Screen.seqs</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>To get an idea of the quality of the alignment, we can view the log output from the summary step:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>        Start   End NBases  Ambigs  Polymer NumSeqs
Minimum:    2391    10674   9   0   2   1
2.5%-tile:  3080    12071   234 0   4   455
25%-tile:   3080    13424   244 0   4   4545
Median:     3080    13424   245 0   4   9090
75%-tile:   3080    13424   245 0   4   13634
97.5%-tile: 3082    13424   246 0   6   17724
Maximum:    13396   13425   267 0   7   18178
Mean:   3080.6  13380   244.212 0   4.27946
# of unique seqs:   17698
total # of seqs:    18178
</code></pre></div></div>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many sequences have been aligned?</li>
    <li>Between which positions most of the reads are aligned to this references?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>17,698 are aligned</li>
      <li>From this we can see that most of our reads align nicely to positions <code class="language-plaintext highlighter-rouge">3080-13424</code> on this reference. This corresponds exactly to the V4 target region of the 16S gene.</li>
    </ol>
  </blockquote>
</blockquote>

<p>To make sure that everything overlaps the same region we’ll re-run <code class="language-plaintext highlighter-rouge">Screen.seqs</code> to get sequences that start at or before position 3,080 and end at or after position 13,424.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-poorly-aligned-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove poorly aligned sequences</div>

  <ol>
    <li><strong>Screen.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” to the aligned fasta file</li>
        <li>“start” to <code class="language-plaintext highlighter-rouge">3080</code></li>
        <li>“end” to <code class="language-plaintext highlighter-rouge">13424</code></li>
        <li>“count” to the group file created by the previous run of <code class="language-plaintext highlighter-rouge">Screen.seqs</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>How many sequences were removed in this step?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>4,579 sequences were removed. This is the number of lines in the bad.accnos output.</p>
  </blockquote>
</blockquote>

<p>Now we know our sequences overlap the same alignment coordinates, we want to make sure they <em>only</em> overlap that region. So we’ll filter the sequences to remove the overhangs at both ends. In addition, there are many columns in the alignment that only contain external gap characters (<em>i.e.</em> “.”), while columns containing only internal gap characters (<em>i.e.</em>, “-“) are not considered. These can be pulled out without losing any information. We’ll do all this with <code class="language-plaintext highlighter-rouge">Filter.seqs</code>:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter sequences</div>

  <ol>
    <li><strong>Filter.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta”” to <code class="language-plaintext highlighter-rouge">good.fasta</code> output from <code class="language-plaintext highlighter-rouge">Screen.seqs</code></li>
        <li>“trump” to <code class="language-plaintext highlighter-rouge">.</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="extraction-of-taxonomic-information">Extraction of taxonomic information</h2>

<p>The main questions when analyzing amplicon data are: Which micro-organisms are present in an environmental samples? And in what proportion? What is the structure of the community of the micro-organisms?</p>

<p>The idea is to take the sequences and assign them to a taxon. To do that, we group (or cluster) sequences based on their similarity to defined Operational Taxonomic Units (OTUs): groups of similar sequences that can be treated as a single “genus” or “species” (depending on the clustering threshold)</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-background-operational-taxonomic-units-otus"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Background: Operational Taxonomic Units (OTUs)</div>

  <p>In 16S metagenomics approaches, OTUs are clusters of similar sequence variants of the 16S rDNA marker gene sequence. Each of these clusters is intended to represent a taxonomic unit of a bacterial species or genus depending on the sequence similarity threshold. Typically, OTU clusters are defined by a 97% identity threshold of the 16S gene sequence variants at genus level. 98% or 99% identity is suggested for species separation.</p>

  <p><a href="../../images/otu.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/otu.png" alt="OTU and cluster with 97% identity threshold. " width="662" height="479" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> OTU and cluster with 97% identity threshold&lt;/figcaption&gt;</figure></a></p>

  <p><a href="../../images/OTU_graph.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/OTU_graph.png" alt="OTU graph. " width="512" height="536" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Cladogram of operational taxonomic units (OTUs). Credit: Danzeisen et al. 2013, 10.7717/peerj.237&lt;/figcaption&gt;</figure></a></p>

</blockquote>

<p>The first thing we want to do is to further de-noise our sequences from potential sequencing errors, by pre-clustering the sequences using the <code class="language-plaintext highlighter-rouge">Pre.cluster</code> command, allowing for up to 2 differences between sequences. This command will split the sequences by group and then sort them by abundance, then go from most abundant to least and identify sequences that differ by no more than 2 nucleotides from on another. If this is the case, then they get merged. We generally recommend allowing 1 difference for every 100 basepairs of sequence:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-perform-preliminary-clustering-of-sequences-and-remove-undesired-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Perform preliminary clustering of sequences and remove undesired sequences</div>

  <ol>
    <li><strong>Pre.cluster</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“fasta” to the fasta output from the last <code class="language-plaintext highlighter-rouge">Filter.seqs</code> run</li>
        <li>“name file or count table” to the count table from the last <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step</li>
        <li>“diffs” to 2</li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>How many unique sequences are we left with after this clustering of highly similar sequences?</p>
    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>10,398</p>

      <p>This is the number of lines in the fasta output</p>
    </blockquote>
  </blockquote>

</blockquote>

<!-- optional additional QC: chimera.uchime -->
<p>We would like to classify the sequences using a training set, which is again is provided on [<a href="https://mothur.org/wiki/MiSeq_SOP">mothur’s MiSeq SOP</a>].</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-classify-the-sequences-into-phylotypes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Classify the sequences into phylotypes</div>

  <ol>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">trainset16_022016.pds.fasta</code> and <code class="language-plaintext highlighter-rouge">trainset16_022016.pds.tax</code> in your history</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/815875/files/trainset16_022016.pds.fasta
https://zenodo.org/record/815875/files/trainset16_022016.pds.tax
</code></pre></div>      </div>
    </li>
    <li>
      <p><strong>Classify.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters</p>
      <ul>
        <li>“fasta” to the fasta output from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
        <li>“Select Reference Template from” to <code class="language-plaintext highlighter-rouge">History</code></li>
        <li>“reference” to <code class="language-plaintext highlighter-rouge">trainset16_022016.pds.fasta</code> from your history</li>
        <li>“Select Taxonomy from” to <code class="language-plaintext highlighter-rouge">History</code></li>
        <li>“taxonomy” to <code class="language-plaintext highlighter-rouge">trainset16_022016.pds.tax</code> from your history</li>
        <li>“count” to the count table from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
      </ul>
    </li>
  </ol>

  <p>This step may take a couple of minutes, now may be a good time to grab a cup of tea/coffee.</p>

</blockquote>

<p>Have a look at the taxonomy output.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>name    taxonomy
SRR651839.9109-HXY9DLL01BSHQO-2 Bacteria(99);Verrucomicrobia(99);Spartobacteria(99);unclassified;unclassified;unclassified;
SRR651839.11437-HXY9DLL01AMIPI-2    Bacteria(100);Verrucomicrobia(97);Spartobacteria(97);unclassified;unclassified;unclassified;
SRR651839.15884-HXY9DLL01BDG2F-2    Bacteria(99);unclassified;unclassified;unclassified;unclassified;unclassified;
SRR651839.12048-HXY9DLL01DSOWS-2    Bacteria(100);Verrucomicrobia(100);Spartobacteria(100);unclassified;unclassified;unclassified;
SRR651839.9410-HXY9DLL01BGRG2-2 Bacteria(100);Proteobacteria(100);Alphaproteobacteria(100);Rhizobiales(100);Bradyrhizobiaceae(100);unclassified;
SRR651839.9029-HXY9DLL01E4W9Z-2 Bacteria(100);Verrucomicrobia(100);Spartobacteria(100);unclassified;unclassified;unclassified;
SRR651839.6283-HXY9DLL01CHX25-2 Bacteria(100);Acidobacteria(100);Acidobacteria_Gp4(100);Gp4(100);unclassified;unclassified;
SRR651839.3134-HXY9DLL01DOM67-2 Bacteria(100);Acidobacteria(100);Acidobacteria_Gp6(100);Gp6(100);unclassified;unclassified;
SRR651839.13044-HXY9DLL01ETN7L-2    Bacteria(100);Proteobacteria(100);Alphaproteobacteria(100);Rhizobiales(100);Bradyrhizobiaceae(100);unclassified;
SRR531818.61708-G88ZSJI01AVPPR-2    Bacteria(100);Acidobacteria(99);Acidobacteria_Gp6(99);Gp6(99);unclassified;unclassified;
</code></pre></div></div>

<p>You will see that every read now has a classification.</p>

<p>The next step is to use this information to determine the abundances of the different found taxa. This consists of three steps:</p>
<ol>
  <li>first, all individual sequences are classified, and get assigned a confidence score (0-100%)</li>
  <li>next, sequences are grouped at 97% identity threshold (not using taxonomy info)</li>
  <li>finally, for each cluster, a consensus classification is determined based on the classification of the individual sequences and taking their confidence scores into account</li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-assign-sequences-to-otus"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assign sequences to OTUs</div>

  <ol>
    <li><strong>Cluster.split</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Split by” to <code class="language-plaintext highlighter-rouge">Classification using fasta</code></li>
        <li>“fasta” to the fasta output from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
        <li>“taxonomy” to the taxonomy output from <code class="language-plaintext highlighter-rouge">Classify.seqs</code></li>
        <li>“count” to the count table output from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
        <li>“Clustering method” to <code class="language-plaintext highlighter-rouge">Average Neighbour</code></li>
        <li>“cutoff” to <code class="language-plaintext highlighter-rouge">0.15</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We obtain a table with the columns being the different identified OTUs, the rows the different distances and the cells the ids of the sequences identified for these OTUs for the different distances.</p>

<p>Next we want to know how many sequences are in each OTU from each group with a distance of 0.03 (97% similarity). We can do this using the <code class="language-plaintext highlighter-rouge">Make.shared</code> command with the 0.03 cutoff level:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-estimate-otu-abundance"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Estimate OTU abundance</div>

  <ol>
    <li><strong>Make.shared</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Select input type” to <code class="language-plaintext highlighter-rouge">OTU list</code></li>
        <li>“list” to list output from <code class="language-plaintext highlighter-rouge">Cluster.split</code></li>
        <li>“name file or count table” to the count table from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
        <li>“label” to <code class="language-plaintext highlighter-rouge">0.03</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We probably also want to know the taxonomy for each of our OTUs. We can get the consensus taxonomy for each OTU using the <code class="language-plaintext highlighter-rouge">Classify.otu</code> command:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-classify-the-otus"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Classify the OTUs</div>

  <ol>
    <li><strong>Classify.otu</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“list” to output from <code class="language-plaintext highlighter-rouge">Cluster.split</code></li>
        <li>“count” to the count table from <code class="language-plaintext highlighter-rouge">Pre.cluster</code></li>
        <li>“Select Taxonomy from” to <code class="language-plaintext highlighter-rouge">History</code></li>
        <li>“taxonomy” to the taxonomy output from <code class="language-plaintext highlighter-rouge">Classify.seqs</code></li>
        <li>“label” to <code class="language-plaintext highlighter-rouge">0.03</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many OTUs with taxonomic assignation are found for the Anguil sample? And for the Pampa sample?</li>
    <li>What is the annotation of first OTU and its size?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>2,195 for Anguil and 2,472 for Pampa (“tax.summary”)</li>
      <li>Otu00001 is associated to 929 sequences and to Bacteria (kingdom), Verrucomicrobia (phylum), Spartobacteria (class) in “taxonomy” file</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="visualization">Visualization</h2>

<p>We have now determined our OTUs and classified them, but looking at a long text file is not very informative.
Let’s visualize our data using Krona:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-krona"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Krona</div>

  <p>First we convert our mothur taxonomy file to a format compatible with Krona</p>

  <ul>
    <li><strong>Taxonomy-to-Krona</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Taxonomy file” to the taxonomy output from Classify.otu (note: this is a collection input)</li>
      </ul>
    </li>
    <li><strong>Krona pie chart</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“Type of input” to <code class="language-plaintext highlighter-rouge">Tabular</code></li>
        <li>“Input file” to taxonomy output from Taxonomy-to-Krona (collection)</li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>The result is an HTML file with an interactive visualization, for instance try clicking
on one of the rings in the image or playing around with some of the settings.</p>

<iframe id="krona" src="krona.html" frameborder="0" width="100%" height="900px"> <a href="../../images/krona.png" rel="noopener noreferrer"><img src="../../images/krona.png" alt="Krona. " loading="lazy" /></a> </iframe>

<p>This produced a single plot for both your samples, but what if you want to compare
the two samples?</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-per-sample-krona-plots"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Per-sample Krona plots</div>

  <ol>
    <li>
      <p><strong>Classify.otu</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <p>Hit the rerun button on the <code class="language-plaintext highlighter-rouge">Classify.otu</code> job in your history and see if you can find settings that will give you per-sample taxonomy data</p>
    </li>
    <li>
      <p><strong>Krona</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <p>Now use this new output collection to create per-sample Krona plots</p>
    </li>
  </ol>

</blockquote>

<p>In this new Krona output you can switch between the combined plot and the per-sample plots via the selector in the top-left corner.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Which soil sample had a higher percentage of Acidobacteria, anguil or pampa? what were the respective percentages?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The anguil sample had a higher proportion of Acidobacteria. The exact percentages can be found by looking at the pie charts at the
top right-hand corner after clicking on the label Acidobacteria. For anguil the percentage is 36%, for the pampa sample it is 26%.</p>

    <p><a href="../../images/krona-multisample.png" rel="noopener noreferrer"><img src="../../images/krona-multisample.png" alt="krona plot with acidobactaria highlighted. " width="929" height="921" loading="lazy" /></a></p>
  </blockquote>
</blockquote>

<p>To further explore the community structure, we can visualize it with dedicated tools such as Phinch.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualization-of-the-community-structure-with-phinch"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualization of the community structure with Phinch</div>

  <ol>
    <li><strong>Make.biom</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li>“shared” to <code class="language-plaintext highlighter-rouge">Make.shared</code> output</li>
        <li>“constaxonomy” to taxonomy output from the first run of <code class="language-plaintext highlighter-rouge">Classify.otu</code> (collection)</li>
      </ul>
    </li>
    <li>
      <p>Expand the dataset and click on the “view biom at phinch” link</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>If this link is not present on your Galaxy, you can download the generated BIOM file and upload directly to Phinch server at <a href="http://phinch.org">http://phinch.org</a>.</p>
      </blockquote>
    </li>
    <li>Play with the several interactive visualisations:</li>
  </ol>

  <p><a href="../../../../shared/images/phinch_overviewpage.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../../../shared/images/phinch_overviewpage.png" alt="Phinch website interface. " width="1088" height="780" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Phinch visualizations&lt;/figcaption&gt;</figure></a></p>

</blockquote>

<p>Once we have information about the community structure (OTUs with taxonomic structure), we can do more analysis on it: estimation of the diversity of micro-organism, comparison of diversity between samples, analysis of populations, … We will not go into detail of such analyses here but you can follow our tutorials on amplicon data analyses to learn more about them.</p>

<h1 id="shotgun-metagenomics-data">Shotgun metagenomics data</h1>

<p>In the previous section, we saw how to analyze amplicon data to extract the community structure. Such information can also be extracted from shotgun metagenomic data.</p>

<p>In shotgun data, full genomes of the micro-organisms in the environment are sequenced (not only the 16S or 18S). We can then have access to the rRNA (only a small part of the genomes), but also to the other genes of the micro-organisms. Using this information, we can try to answer questions such as “What are the micro-organisms doing?” in addition to the question “What micro-organisms are present?”.</p>

<p>In this second part, we will use a metagenomic sample of the Pampas Soil (<a href="https://www.ebi.ac.uk/metagenomics/projects/SRP016633/samples/SRS372043/runs/SRR606451/results/versions/2.0">SRR606451</a>).</p>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create a new history</li>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">SRR606451_pampa</code> Fasta file from <a href="http://zenodo.org/record/815875">Zenodo</a> or from the data library (in “Analyses of metagenomics data”)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/815875/files/SRR606451_pampa.fasta
</code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<h2 id="extraction-of-taxonomic-information-1">Extraction of taxonomic information</h2>

<p>As for amplicon data, we can extract taxonomic and community structure information from shotgun data. Different approaches can be used:</p>

<ul>
  <li>
    <p>Same approach as for amplicon data with identification and classification of OTUs</p>

    <p>Such an approach requires a first step of sequence sorting to extract only the 16S and 18S sequences, then using the same tools as for amplicon data. However, rRNA sequences represent a low proportion (&lt; 1%) of the shotgun sequences so such an approach is not the most statistically supported</p>
  </li>
  <li>
    <p>Assignation of taxonomy on the whole sequences using databases with marker genes</p>
  </li>
</ul>

<p>In this tutorial, we use the second approach with MetaPhlAn2. This tools is using a database of ~1M unique clade-specific marker genes (not only the rRNA genes) identified from ~17,000 reference (bacterial, archeal, viral and eukaryotic) genomes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-taxonomic-assignation-with-metaphlan2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Taxonomic assignation with MetaPhlAn2</div>

  <ol>
    <li><strong>MetaPhlAN2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Input file” to the imported file</li>
        <li>“Database with clade-specific marker genes” to <code class="language-plaintext highlighter-rouge">locally cached</code></li>
        <li>“Cached database with clade-specific marker genes” to <code class="language-plaintext highlighter-rouge">MetaPhlAn2 clade-specific marker genes</code></li>
      </ul>
    </li>
  </ol>

  <p>This step may take a couple of minutes.</p>
</blockquote>

<p>3 files are generated:</p>

<ul>
  <li>
    <p>A tabular file with the community structure</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  #SampleID   Metaphlan2_Analysis
  k__Bacteria 100.0
  k__Bacteria|p__Proteobacteria   86.20712
  k__Bacteria|p__Actinobacteria   13.79288
  k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria    86.20712
  k__Bacteria|p__Actinobacteria|c__Actinobacteria 13.79288
</code></pre></div>    </div>

    <p>Each line contains a taxa and its relative abundance found for our sample. The file starts with high level taxa (kingdom: <code class="language-plaintext highlighter-rouge">k__</code>) and go to more precise taxa.</p>
  </li>
  <li>
    <p>A BIOM file with the same information as the previous file but in BIOM format</p>

    <p>It can be used then by mothur and other tools requiring community structure information in BIOM format</p>
  </li>
  <li>
    <p>A SAM file with the results of the mapping of the sequences on the reference database</p>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What is the most precise level we have access to with MetaPhlAn2?</li>
    <li>What are the two orders found in our sample?</li>
    <li>What is the most abundant family in our sample?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We have access to species level</li>
      <li>Pseudomonadales and Solirubrobacterales are found in our sample</li>
      <li>The most abundant family is Pseudomonadaceae with 86.21 % of the assigned sequences</li>
    </ol>
  </blockquote>
</blockquote>

<p>Even if the output of MetaPhlAn2 is bit easier to parse than the BIOM file, we want to visualize and explore the community structure with KRONA</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-interactive-visualization-with-krona"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Interactive visualization with KRONA</div>

  <ol>
    <li><strong>Format MetaPhlAn2 output for Krona</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Input file” to <code class="language-plaintext highlighter-rouge">Community profile</code> output of <code class="language-plaintext highlighter-rouge">MetaPhlAn2</code></li>
      </ul>
    </li>
    <li><strong>KRONA pie chart</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“What is the type of your input data” as <code class="language-plaintext highlighter-rouge">MetaPhlan</code></li>
        <li>“Input file” to the output of <code class="language-plaintext highlighter-rouge">Format MetaPhlAn2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="extraction-of-functional-information">Extraction of functional information</h2>

<p>We would now like to answer the question “What are the micro-organisms doing?” or “Which functions are performed by the micro-organisms in the environment?”.</p>

<p>In the shotgun data, we have access to the gene sequences from the full genome. We use that to identify the genes, associate them with a function, build pathways, etc., to investigate the functional part of the community.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-metabolism-function-identification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Metabolism function identification</div>

  <ol>
    <li><strong>HUMAnN2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Input sequence file” to the imported sequence file</li>
        <li>“Use of a custom taxonomic profile” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>“Taxonomic profile file” to <code class="language-plaintext highlighter-rouge">Community profile</code> output of <code class="language-plaintext highlighter-rouge">MetaPhlAn2</code></li>
        <li>“Nucleotide database” to <code class="language-plaintext highlighter-rouge">Locally cached</code></li>
        <li>“Nucleotide database” to <code class="language-plaintext highlighter-rouge">Full</code></li>
        <li>“Protein database” to <code class="language-plaintext highlighter-rouge">Locally cached</code></li>
        <li>“Protein database” to <code class="language-plaintext highlighter-rouge">Full UniRef50</code></li>
        <li>“Search for uniref50 or uniref90 gene families?” to <code class="language-plaintext highlighter-rouge">uniref50</code></li>
        <li>“Database to use for pathway computations” to <code class="language-plaintext highlighter-rouge">MetaCyc</code></li>
        <li>“Advanced Options”</li>
        <li>“Remove stratification from output” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step is long so we generated the output for you!</p>
    </li>
    <li>
      <p>Import the 3 files whose the name is starting with “humann2”</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/815875/files/humann2_gene_families_abundance.tsv
https://zenodo.org/record/815875/files/humann2_pathways_abundance.tsv
https://zenodo.org/record/815875/files/humann2_pathways_coverage.tsv
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>HUMAnN2 generates 3 files</p>

<ul>
  <li>
    <p>A file with the abundance of gene families</p>

    <p>Gene family abundance is reported in RPK (reads per kilobase) units to normalize for gene length. It reflects the relative gene (or transcript) copy number in the community.</p>

    <p>The “UNMAPPED” value is the total number of reads which remain unmapped after both alignment steps (nucleotide and translated search). Since other gene features in the table are quantified in RPK units, “UNMAPPED” can be interpreted as a single unknown gene of length 1 kilobase recruiting all reads that failed to map to known sequences.</p>
  </li>
  <li>
    <p>A file with the coverage of pathways</p>

    <p>Pathway coverage provides an alternative description of the presence (1) and absence (0) of pathways in a community, independent of their quantitative abundance.</p>
  </li>
  <li>
    <p>A file with the abundance of pathways</p>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many gene families and pathways have been identified?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>44 gene families but no pathways are identified</p>
  </blockquote>
</blockquote>

<p>The RPK for the gene families are quite difficult to interpret in term of relative abundance. We decide then to normalize the values</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-normalize-the-gene-family-abundances"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Normalize the gene family abundances</div>

  <ol>
    <li><strong>Renormalize a HUMAnN2 generated table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Gene/pathway table” to the gene family table generated with <code class="language-plaintext highlighter-rouge">HUMAnN2</code></li>
        <li>“Normalization scheme” to <code class="language-plaintext highlighter-rouge">Relative abundance</code></li>
        <li>“Normalization level” to <code class="language-plaintext highlighter-rouge">Normalization of all levels by community total</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>What percentage of sequences has not be assigned to a gene family?</li>
      <li>What is the most abundant gene family?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>55% of the sequences have not be assigned to a gene family</li>
        <li>The most abundant gene family with 25% of sequences is a putative secreted protein</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<p>With the previous analyses, we investigate “Which micro-organims are present in my sample?” and “What function are performed by the micro-organisms in my sample?”. We can go further in these analyses (for example, with a combination of functional and taxonomic results). We did not detail that in this tutorial but you can find more analyses in our tutorials on shotgun metagenomic data analyses.</p>

<h1 id="conclusion">Conclusion</h1>

<p>We can summarize the analyses with amplicon and shotgun metagenomic data:</p>

<p><a href="../../images/general-tutorial-scheme.png" rel="noopener noreferrer"><img src="../../images/general-tutorial-scheme.png" alt="Scheme to sum up the analysis. " width="2878" height="3192" loading="lazy" /></a></p>

<p>Both analyses are quite complex! However, in this tutorial, we only showed simple cases of metagenomics data analysis with subset of real data.</p>

<p>Check our other tutorials to learn more in detail of how to analyze metagenomics data.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>With amplicon data, we can extract information about the studied community structure</p>
</li>
                        
                        <li><p>With shotgun data, we can extract information about the studied community structure and also the functions realised by the community</p>
</li>
                        
                        <li><p>The tools used to analyze amplicon and shotgun data are different, except for the visualisation</p>
</li>
                        
                        <li><p>Metagenomics data analyses are complex and time-consuming</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Rascovan2013">Rascovan, N., B. Carbonetto, S. Revale, M. D. Reinert, R. Alvarez <i>et al.</i>, 2013 <b>The PAMPA datasets: a metagenomic survey of microbial communities in Argentinean pampean soils</b>. Microbiome 1: <a href="https://doi.org/10.1186/2049-2618-1-21">10.1186/2049-2618-1-21</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Analyses of metagenomics data - The global picture (Metagenomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Saskia Hiltemann, Bérénice Batut,  <b>Analyses of metagenomics data - The global picture (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metagenomics-general-tutorial,
author = "Saskia Hiltemann and Bérénice Batut",
	title = "Analyses of metagenomics data - The global picture (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/metagenomics/tutorials/general-tutorial/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: taxonomy_krona_chart
  owner: crs4
  revisions: 73486cc6170e
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: humann2
  owner: iuc
  revisions: 1ab06263e083
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: humann2_regroup_table
  owner: iuc
  revisions: a1747df2bc21
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: humann2_renorm_table
  owner: iuc
  revisions: ba4dfbdc1295
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: metaphlan2
  owner: iuc
  revisions: 8c82c4d90cc6
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: metaphlan2krona
  owner: iuc
  revisions: ebb74774457e
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_align_seqs
  owner: iuc
  revisions: 2692b3107533
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_classify_otu
  owner: iuc
  revisions: e664787de3ab
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_classify_seqs
  owner: iuc
  revisions: cc9c7b550c29
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_cluster_split
  owner: iuc
  revisions: a29ac7993cb0
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_count_seqs
  owner: iuc
  revisions: ecbd9ed733e6
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_filter_seqs
  owner: iuc
  revisions: bf20a30c263a
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_make_biom
  owner: iuc
  revisions: 5dc4d49a238a
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_make_group
  owner: iuc
  revisions: 8c964d119da8
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_make_shared
  owner: iuc
  revisions: 83dabeeadb55
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_merge_files
  owner: iuc
  revisions: a70b25374a23
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_pre_cluster
  owner: iuc
  revisions: 58d0fe3cc0f2
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_screen_seqs
  owner: iuc
  revisions: 8743aecd26a9
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_summary_seqs
  owner: iuc
  revisions: e4a4cb8b6bbf
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_unique_seqs
  owner: iuc
  revisions: 1f72d9a3d09f
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: krona_text
  owner: saskia-hiltemann
  revisions: b14f1444e464
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00201">https://gxy.io/GTN:T00201</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/general-tutorial/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1697502352'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>