<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Genome Assembly of MRSA using Illumina MiSeq Data</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="DNA sequence data has become an indispensable tool for Mo..." />
        <meta property="og:title" content="Galaxy Training: Genome Assembly of MRSA using Illumina MiSeq Data" />
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Mo..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Assembly
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/mrsa-illumina/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/assembly/faqs/" title="Check our FAQs for the Assembly topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/mrsa-illumina/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:14 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Genome Assembly of MRSA using Illumina MiSeq Data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
  },
  "courseCode": "assembly / mrsa-illumina / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Genome Assembly of MRSA using Illumina MiSeq Data' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html",
  "timeRequired": "PT2H",
  "keywords": "illumina, assembly, amr",
  "description": "The questions this  addresses are:\n - How to check the quality of the MiSeq data?\n - How to perform an assembly with MiSeq data?\n - What species do I have and what is its sequence type?\n - Which resistance genes are on my genome?\n - Where are the genes located on my genome?\n\n\\nThe objectives are:\n - Assess your data on quality and quantity\n - Assemble a genome\n - Assess your assembly quality\n - Assess presence of antimicrobial resistance genes\n - Perform a species identification and MLST typing\n - Search for resistance genes on the assembly\n - Find a gene on your genome using Prokka + JBrowse\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Bazante Sanders"
    },
    {
      "@type": "Person",
      "name": "Miaomiao Zhou"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    },
    {
      "@type": "Person",
      "name": "Avans Hogeschool"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Bazante Sanders"
    },
    {
      "@type": "Person",
      "name": "Miaomiao Zhou"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    },
    {
      "@type": "Person",
      "name": "Avans Hogeschool"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
    }
  ]
}
</script>

<section class="tutorial topic-assembly">
    <h1 data-toc-skip>Genome Assembly of MRSA using Illumina MiSeq Data</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/bazante1/" class="contributor-badge contributor-bazante1"><img src="https://avatars.githubusercontent.com/bazante1?s=27" alt="Avatar">Bazante Sanders</a>


<a href="/training-material/hall-of-fame/miaomiaozhou88/" class="contributor-badge contributor-miaomiaozhou88"><img src="https://avatars.githubusercontent.com/miaomiaozhou88?s=27" alt="Avatar">Miaomiao Zhou</a>


<a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=27" alt="Avatar">Saskia Hiltemann</a>


<a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="https://avatars.githubusercontent.com/hexylena?s=27" alt="Avatar">Helena Rasche</a>


<a href="/training-material/hall-of-fame/avans-atgm/" class="contributor-badge contributor-avans-atgm"><img src="https://avatars.githubusercontent.com/avans-atgm?s=27" alt="Avatar">Avans Hogeschool</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to check the quality of the MiSeq data?</p>
</li>
        
        <li><p>How to perform an assembly with MiSeq data?</p>
</li>
        
        <li><p>What species do I have and what is its sequence type?</p>
</li>
        
        <li><p>Which resistance genes are on my genome?</p>
</li>
        
        <li><p>Where are the genes located on my genome?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Assess your data on quality and quantity</p>
</li>
        
        <li><p>Assemble a genome</p>
</li>
        
        <li><p>Assess your assembly quality</p>
</li>
        
        <li><p>Assess presence of antimicrobial resistance genes</p>
</li>
        
        <li><p>Perform a species identification and MLST typing</p>
</li>
        
        <li><p>Search for resistance genes on the assembly</p>
</li>
        
        <li><p>Find a gene on your genome using Prokka + JBrowse</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4534098">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/assembly/tutorials/mrsa-illumina/workflows/" title="Workflows" alt="Genome Assembly of MRSA using Illumina MiSeq Data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/assembly/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Dec 14, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In this training you’re going to make an assembly of data produced by
“Complete Genome Sequences of Eight Methicillin-Resistant
<em>Staphylococcus aureus</em> Strains Isolated from Patients in
Japan” from <span class="citation"><a href="#Hikichi_2019">Hikichi <i>et al.</i> 2019</a></span> which describes:</p>

<blockquote>
  <p>Methicillin-resistant <em>Staphylococcus aureus</em> (MRSA) is a major pathogen
causing nosocomial infections, and the clinical manifestations of MRSA
range from asymptomatic colonization of the nasal mucosa to soft tissue
infection to fulminant invasive disease. Here, we report the complete
genome sequences of eight MRSA strains isolated from patients in Japan.</p>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#background" id="markdown-toc-background">Background</a></li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a></li>
  <li><a href="#assembly" id="markdown-toc-assembly">Assembly</a>    <ol>
      <li><a href="#genome-assembly-evaluation" id="markdown-toc-genome-assembly-evaluation">Genome Assembly Evaluation</a></li>
    </ol>
  </li>
  <li><a href="#identification-of-amr-genes" id="markdown-toc-identification-of-amr-genes">Identification of AMR Genes</a>    <ol>
      <li><a href="#card-database" id="markdown-toc-card-database">CARD database</a></li>
      <li><a href="#gene-annotation-using-prokka" id="markdown-toc-gene-annotation-using-prokka">Gene annotation using Prokka</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="background">Background</h1>

<p>Sequencing (determining of <span class="notranslate">DNA</span>/<span class="notranslate">RNA</span> nucleotide sequence) is used all over
the world for all kinds of analysis. The product of these sequencers are
<span class="notranslate">reads</span>, which are sequences of detected nucleotides. Depending on the
technique these have specific lengths (30-500bp) or using Oxford
Nanopore Technologies sequencing have much longer variable lengths.</p>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Illumina MiSeq sequencing</h3>  <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the namesuggests, fluorescent labels are measured for every base that bind at aspecific moment at a specific place on a flow cell. These flow cells arecovered with oligos (small single strand <span class="notranslate">DNA</span> strands). In the librarypreparation the <span class="notranslate">DNA</span> strands are cut into small <span class="notranslate">DNA</span> fragments (differs perkit/device) and specific pieces of <span class="notranslate">DNA</span> (adapters) are added, which arecomplementary to the oligos. Using bridge amplification large amounts ofclusters of these <span class="notranslate">DNA</span> fragments are made. The reverse string is washed away,making the clusters single stranded. Fluorescent bases are added one by one,which emit a specific light for different bases when added. This is happeningfor whole clusters, so this light can be detected and this data is basecalled(translation from light to a nucleotide) to a nucleotide sequence (Read). Forevery base a quality score is determined and also saved per read. Thisprocess is repeated for the reverse strand on the same place on the flowcell, so the forward and reverse <span class="notranslate">reads</span> are from the same <span class="notranslate">DNA</span> strand. Theforward and reversed <span class="notranslate">reads</span> are linked together and should always be processedtogether!</p>  <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p></blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p><span class="tool" data-tool="upload1" title="Tested with upload1"><strong>Import</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span>  the files from <a href="https://zenodo.org/record/4534098">Zenodo</a> or from the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4534098/files/DRR187559_1.fastqsanger.bz2
https://zenodo.org/record/4534098/files/DRR187559_2.fastqsanger.bz2
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>
      <p>Convert the datatype of this output to uncompress it</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-converting-the-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Converting the file format</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <strong>Convert</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">Convert compressed to uncompressed</code></li>  <li>Click the <strong>Convert datatype</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Rename the uncompressed dataset to just the sequence run ID: <code class="language-plaintext highlighter-rouge">DRR187559_1</code> and <code class="language-plaintext highlighter-rouge">DRR187559_2</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field  to <code class="language-plaintext highlighter-rouge">DRR187559_1</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Tag the dataset <code class="language-plaintext highlighter-rouge">#unfiltered</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
    <li>
      <p><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the renamed file</p>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <ol>
          <li>What are the 4 main features of each read in a fastq file.</li>
          <li>What does the <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code> mean in your filenames?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>
              <p>The following:</p>

              <ul>
                <li>A <code class="language-plaintext highlighter-rouge">@</code> followed by a name and sometimes information of the read</li>
                <li>A nucleotide sequence</li>
                <li>A <code class="language-plaintext highlighter-rouge">+</code> (optional followed by the name)</li>
                <li>The quality score per base of nucleotide sequence (Each symbol
represents a quality score, which will be explained later)</li>
              </ul>
            </li>
            <li>
              <p>Forward and reverse <span class="notranslate">reads</span>, by convention, are labelled <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code>, but they might also be <code class="language-plaintext highlighter-rouge">_f</code>/<code class="language-plaintext highlighter-rouge">_r</code> or <code class="language-plaintext highlighter-rouge">_r1</code>/<code class="language-plaintext highlighter-rouge">_r2</code>.</p>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="quality-control">Quality Control</h1>

<p>When assessing the fastq files all bases had their own quality (or Phred score)
represented by symbols. You can read more in our dedicated <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control
Tutorial</a>.</p>

<p>To assess the quality by hand would be too much work. That’s why tools like
<a href="https://github.com/wdecoster/NanoPlot">NanoPlot</a> or
<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> are made,
which will generate a summary and plots of the data statistics. NanoPlot is
mainly used for long-read data, like ONT and PACBIO and FastQC for short read,
like Illumina and Sanger.</p>

<p>Before doing any assembly, the first questions you should ask about your input
<span class="notranslate">reads</span> include:</p>

<ul>
  <li>What is the coverage of my genome?</li>
  <li>How good are my <span class="notranslate">reads</span>?</li>
  <li>Do I need to ask/perform for a new sequencing run?</li>
  <li>Is it suitable for the analysis I need to do?</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-qc--filtering"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: QC &amp; Filtering</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">DRR187559_1</code> and <code class="language-plaintext highlighter-rouge">DRR187559_2</code></li>
      </ul>
    </li>
    <li>Examine the output “WebPage” files</li>
  </ol>

</blockquote>

<table>
  <thead>
    <tr>
      <th>DRR187559_1</th>
      <th>DRR187559_2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/mrsa/fastqc-1.png" alt="FastQC plot showing reads that mostly stay in the green. " loading="lazy" /></td>
      <td><img src="../../images/mrsa/fastqc-2.png" alt="Same as previous plot, but the beginning of the reads are slightly better quality. " loading="lazy" /></td>
    </tr>
  </tbody>
</table>

<p>FastQC combines all quality statistics from all separate <span class="notranslate">reads</span> and combines them in plots. An important plot is the Per base sequence quality. Here you have the <span class="notranslate">reads</span> sequence length on the x-axes against the quality score (Phred-score) on the y-axis. The y-axis is divided in three sections: Green = good quality, Orange = mediocre quality and red = bad quality. For Illumina data it is normal that the first few bases are of some lower quality and how longer the <span class="notranslate">reads</span> get the worse the quality becomes. This is often due to signal decay or phasing during the sequencing run.</p>

<p>For each position, a boxplot is drawn with:</p>

<ul>
  <li>the median value, represented by the central red line</li>
  <li>the inter-quartile range (25-75%), represented by the yellow box</li>
  <li>the 10% and 90% values in the upper and lower whiskers</li>
  <li>the mean quality, represented by the blue line</li>
</ul>

<p>Depending on the analysis it could be possible that a certain quality or length
is needed. In this case we’re going to trim the data using Trimmomatic. Here we
will trim the start (LEADING) and end (TRAILING)of the <span class="notranslate">reads</span> if those fall
below a quality score of 20. Different trimming tools have different algorithms
for deciding when to cut but trimmomatic will cut based on the quality csore of
one base alone. Trimmomatic starts from each end, and as long as the base is
below 20, it will be cut until it reaches one greater than 20. A sliding window
trimming will be performed where if the average quality of 4 bases drops below
20, the read will be truncated there. And last we will also filter for <span class="notranslate">reads</span>
which are at least 30 bases long, anything shorter will complicate the assembly
and we will drop.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assessing-the-data-quality-of-the-trimmed-reads-and-comparing-to-the-input-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assessing the data quality of the trimmed <span class="notranslate">reads</span> and comparing to the input <span class="notranslate">reads</span></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.38.1" title="Tested with toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.38.1"><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.38.1</span></span>   with the following parameters:
      <ul>
        <li><em>“Single-end or paired-end <span class="notranslate">reads</span>?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end (two separate input files)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R1/first of pair)”</em>: <code class="language-plaintext highlighter-rouge">DRR187559_1 uncompressed</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R2/second of pair)”</em>: <code class="language-plaintext highlighter-rouge">DRR187559_2 uncompressed</code></li>
          </ul>
        </li>
        <li><em>“Perform initial ILLUMINACLIP step?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>In <em>“Trimmomatic Operation”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off the start of a read, if below a threshold quality (LEADING)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off the end of a read, if below a threshold quality (TRAILING)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Sliding window trimming (SLIDINGWINDOW)</code>
                  <ul>
                    <li><em>“Number of bases to average across”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
                    <li><em>“Average quality required”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Drop <span class="notranslate">reads</span> below a specified length (MINLEN)</code>
                  <ul>
                    <li><em>“Minimum length of <span class="notranslate">reads</span> to be kept”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <p>This produces two sets of output files, the R1/R2 paired, and R1/R2 unpaired.</p>

      <ul>
        <li>The first first two are the paired <span class="notranslate">reads</span>, the <span class="notranslate">reads</span> in each of those files match up.</li>
        <li>The second two are the unpaired <span class="notranslate">reads</span> where one of the two <span class="notranslate">reads</span> was discarded.</li>
      </ul>
    </li>
    <li>
      <p>Edit the tags of the trimmomatic outputs. Remove the <code class="language-plaintext highlighter-rouge">#unfiltered</code> tag and add a new tag, <code class="language-plaintext highlighter-rouge">#filtered</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_1 (R1 Paired)</code> and <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_2 (R2 Paired)</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9</span></span>  with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Results”</em>
              <ul>
                <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code>
                  <ul>
                    <li>In <em>“FastQC output”</em>:
                      <ul>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert FastQC output”</em>
                          <ul>
                            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: select <strong>all</strong> of the <code class="language-plaintext highlighter-rouge">FastQC on data ... RawData</code> files. You should have 4; 2 FastQCs from before trimming, 2 from after trimming.</li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>MultiQC is a tool to combine multiple outputs in one clear and easy to read overview. It generates plots to easily visualize the different quality statistics of all your fastq files.</p>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Looking at your <code class="language-plaintext highlighter-rouge">MultiQC on data ....: Webpage</code> output, answer the following questions:</p>

  <ol>
    <li>How did the duplicate <span class="notranslate">reads</span> change from before to after?</li>
    <li>How did the average read length change?</li>
    <li>Did trimming improve the mean quality scores?</li>
    <li>Did trimming affect the GC content?</li>
    <li>Is this data ok to assemble? Do we need to re-sequence it?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>There were slight decreases in duplicate <span class="notranslate">reads</span> (19.8% →  18%, 21.2% →  20.5%)</li>
      <li>Read lengths went down more significantly; 191 to 152 bp, and 221 to 192 bp. These are still probably fine for assembly but you’ll see that MultiQC has marked the 152 bp result in yellow.</li>
      <li>Yes, definitely. It is most visible at the end of the <span class="notranslate">reads</span>, they stay completely in the “green zone”, unlike the untrimmed data which falls into the yellow.</li>
      <li>No, it did not. If it had, that would be unexpected.</li>
      <li>This data looks OK. The number of shorter <span class="notranslate">reads</span> in R1 is not optimal but it should assemble something. Probably not the entire, closed genomes but something.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="assembly">Assembly</h1>

<p>When the quality of the <span class="notranslate">reads</span> is determined and the data is filtered and/or trimmed (like we did with trimmomatic)
an assembly can be made.</p>

<p>There are many tools that create assembly for short-read data, but in this
tutorial Shovill will be used. Shovill is a SPAdes based genome assembler,
improved to work faster and only for smaller (bacterial) genomes.</p>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assembly-using-shovill"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assembly using Shovill</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+galaxy0"><strong>Shovill</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/shovill/shovill/1.1.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>“Input <span class="notranslate">reads</span> type, collection or single library”</em>: <code class="language-plaintext highlighter-rouge">Paired End</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Forward <span class="notranslate">reads</span> (R1)”</em>: <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_1 uncompressed (R1 paired)</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reverse <span class="notranslate">reads</span> (R2)”</em>: <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_2 uncompressed (R2 paired)</code></li>
          </ul>
        </li>
        <li>In <em>“Advanced options”</em>:
          <ul>
            <li><em>“Estimated genome size”</em>: <code class="language-plaintext highlighter-rouge">2914567</code></li>
            <li><em>“Minimum contig length”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>View <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the <code class="language-plaintext highlighter-rouge">Contigs</code> file:</p>

      <blockquote class="question">
        <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
        <ol>
          <li>How big is the first contig?</li>
          <li>What is the coverage of your biggest (first) contig?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>Both of these can be found in the header line of the Contigs produced by Shovill</p>

          <p><em>NOTE: The results can differ from this example, because Shovill can differ a bit per assembly</em></p>

          <ol>
            <li>419691 bp</li>
            <li>14.1</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2"><strong>Bandage Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:</p>

      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Graphical Fragment Assembly”</em>: <code class="language-plaintext highlighter-rouge">Shovill on ... Contig Graph</code></li>
      </ul>
    </li>
    <li>
      <p>View <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the assembly graph image</p>

      <p><img src="../../images/mrsa/bandage-illumina.jpg" alt="Bandage output showing a mess of a genome graph. " loading="lazy" /></p>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
  <p>First read <a href="https://github.com/rrwick/Bandage/wiki/Simple-example">this page in the Bandage wiki</a> to help understand what the graph means.</p>

  <ol>
    <li>What do you think of this assembly? Is it useful? Is it good enough?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>This is a very messy assembly, with a lot of potential paths through the sequence. We cannot feel confident in the output .fasta file (as it is much smaller than the expected 2.9Mbp). In real life we might consider doing a hybrid assembly with Nanopore or other long read data to help resolve these issues.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="genome-assembly-evaluation">Genome Assembly Evaluation</h2>

<p><a href="http://quast.bioinf.spbau.ru/">Quast</a> (<span class="citation"><a href="#Gurevich2013">Gurevich <i>et al.</i> 2013</a></span>)
is a tool providing quality metrics for assemblies, and can also be used
to compare multiple assemblies. The tool can also take an optional
reference file as input, and will provide complementary metrics. QUAST
stands for QUality ASsessment Tool. With later updates gene annotation
also possible with QUAST.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control-of-assembly-using-quast"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality Control of assembly using Quast</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>“Use customized names for the input files?”</em>: <code class="language-plaintext highlighter-rouge">No, use dataset names</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs/scaffolds file”</em>: <code class="language-plaintext highlighter-rouge">Shovill on ... Contigs</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>View <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the HTML report from QUAST</p>

      <p>The Quast tool outputs assembly metrics as an html file with metrics and
graphs. The image below looks exceptionally boring. This is a good
thing, because each corner means one contig. A contig is the contiguous
sequence made by combining all separate <span class="notranslate">reads</span> in the assembly</p>

      <p><img src="../../images/mrsa/quast-illumina.png" alt="Image showing the HTML output of quast including a table over conting information and a cumulative length graph with the contigs." loading="lazy" /></p>
    </li>
  </ol>

</blockquote>

<p>This fasta file contains 32 contigs, meaning the chromosome is separated over multiple contigs. These contigs can also contain (parts of) plasmids.</p>

<blockquote class="question">
  <h3 id="question-question-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>What is you GC content?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The GC content for our assembly was 32.77% (for comparison <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5174738/">MRSA Isolate HC1335 Strain</a> GC% is 32.89%). This means the length and GC% of the assembly could be good!</li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="identification-of-amr-genes">Identification of AMR Genes</h1>

<p>Because we are working with a MRSA we are curious to see which resistance genes are located on the genome or on the plasmid. To determine whether the contigs contain antimicrobial resistance (AMR) genes <a href="https://github.com/phac-nml/staramr">staramr</a> can be used  <strong>Staramr</strong> scans bacterial genome contigs against both the <strong>ResFinder</strong> (<span class="citation"><a href="#Zankari2012">Zankari <i>et al.</i> 2012</a></span>), <strong>PointFinder</strong> (<span class="citation"><a href="#Zankari2017">Zankari <i>et al.</i> 2017</a></span>), and <strong>PlasmidFinder</strong> (<span class="citation"><a href="#Carattoli2014">Carattoli <i>et al.</i> 2014</a></span>) databases (used by the ResFinder webservice) and compiles a summary report of detected antimicrobial resistance genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-staramr"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run staramr</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+galaxy2"><strong>staramr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“genomes”</em>: <code class="language-plaintext highlighter-rouge">Shovill on data ... Contigs</code></li>
      </ul>

      <p>There are 7 different output files produced by <strong>staramr</strong> tool:</p>

      <table>
        <thead>
          <tr>
            <th>File</th>
            <th>Contents</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>summary.tsv</td>
            <td>A summary of all detected AMR genes/mutations in each genome, one genome per line.</td>
          </tr>
          <tr>
            <td>detailed_summary.tsv</td>
            <td>A detailed summary of all detected AMR genes/mutations of each genome, one line per feature and multiple lines per genome.</td>
          </tr>
          <tr>
            <td>resfinder.tsv</td>
            <td>A tabular file of each AMR gene and additional BLAST information from the <strong>ResFinder</strong> database, one gene per line.</td>
          </tr>
          <tr>
            <td>Plasmidfinder.tsv</td>
            <td>A tabular file of each plastid sequences with additional BLAST information from the <strong>PlasmidFinder</strong> database, one sequence per line.</td>
          </tr>
          <tr>
            <td>settings.txt</td>
            <td>The command-line, database versions, and other settings used to run staramr.</td>
          </tr>
          <tr>
            <td>mlst.tsv</td>
            <td>A tabular file of the found loci per genome with its specified MLST scheme.</td>
          </tr>
          <tr>
            <td>results.xlsx</td>
            <td>An Excel sp<span class="notranslate">reads</span>heet containing the previous 4 files as separate worksheets.</td>
          </tr>
        </tbody>
      </table>
    </li>
    <li>View <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the detailed_summary.tsv file
      <ul>
        <li>In this example the ST-typing could not be obtained. Multi-locus sequence type (MLST) is based on specific locus/alleles, which is sometimes hard to determine with error rich sequence data (like NanoPore).</li>
        <li>For the plasmid and resistance results the identity, overlap, length and the location on the contig can be found here.</li>
        <li>Multiple rep sequences are located on the second contig. (See “plasmid typing for gram-positive bacteria” <span class="citation"><a href="#Lozano_2012">Lozano <i>et al.</i> 2012</a></span> for more information)</li>
        <li>Multiple resistance genes can be found on both contig 1 and contig 2.</li>
        <li>In the last column there are “Accession” numbers. These are references to NCBI, and you can search for these numbers there. E.g. <a href="https://www.ncbi.nlm.nih.gov/nuccore/M13771">M13771</a></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="card-database">CARD database</h2>

<p>To get more information about these antibiotic resistant genes, you can check the <a href="https://card.mcmaster.ca">CARD database</a> (<strong>C</strong>omprehensive <strong>A</strong>ntibiotic <strong>R</strong>esistance <strong>D</strong>atabase) (<span class="citation"><a href="#Jia2016">Jia <i>et al.</i> 2016</a></span>)</p>

<figure id="figure-1"><img src="../../images/mrsa/card.png" alt="Screenshot of mecA sequence in CARD database with lots of metadata. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Screenshot of the CARD database interface. CARD gives information about the antibiotic resistance genes, as well as links to relevant publications.</figcaption></figure>

<p>CARD can be very helpful to check all the resistance genes and check if
it is logical to find the resistance gene in a specific bacteria.</p>

<blockquote class="question">
  <h3 id="question-question-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>To what family does <a href="https://card.mcmaster.ca/ontology/36911">mecA</a> belong?</li>
    <li>Do you expect to find this gene in this MRSA strain and why?</li>
    <li>Is the accession number of the entry related to the accession reported by staramr?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li><a href="https://card.mcmaster.ca/ontology/37589">Methicillin resistant PBP2</a></li>
      <li>The strain we use is a Methicillin(multi) resistant Staphylococcus aureus. As <code class="language-plaintext highlighter-rouge">mecA</code> has a perfect resistome mach with <em>S. aureus</em>, and the AMR Gene Family is methicillin resistant PBP2, we expect to see mecA in MRSA.</li>
      <li>No, these are completely unrelated. Unfortunately this is a <strong>very</strong> common issue in bioinformatics. Everyone builds their own numbering system for entries in their database (usually calling them ‘accessions’), and then someone else needs to build a service to link these databases.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="gene-annotation-using-prokka">Gene annotation using Prokka</h2>

<p><a href="https://github.com/tseemann/prokka/blob/master/README.md">Prokka</a> is a
tool software tool to rapidly annotate bacterial, archaeal and viral
genomes. Prokka will be used on your own made genome (assembly). Prokka
will try to annotate the bacteria based on related species and starting
codons can be chosen or default of the species can be used.</p>

<p><a href="https://jbrowse.org/docs/tutorial.html">JBrowse</a> is used to visualize
your genome file and merge multiple outputs.<br />
In this case you will use your assembly as your reference and the output
from prokka as an information track.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-annotating-the-genome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Annotating the Genome</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/crs4/prokka/prokka/1.14.5+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/crs4/prokka/prokka/1.14.5+galaxy0"><strong>Prokka</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/crs4/prokka/prokka/1.14.5+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Contigs to annotate”</em>: <code class="language-plaintext highlighter-rouge">Shovill on data ... Contigs</code></li>
        <li><em>“Genus name (–genus)”</em>: <code class="language-plaintext highlighter-rouge">staphylococcus </code></li>
        <li><em>“Species name (–species)”</em>: <code class="language-plaintext highlighter-rouge">aureus</code></li>
        <li><em>“Kingdom (–kingdom)”</em>: <code class="language-plaintext highlighter-rouge">Bacteria</code></li>
        <li><em>“Additional outputs”</em>: Select only the “Annotation in GFF3 format contaianing both sequences and annotations”</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">staramr on data .. detailed_summary.tsv</code></li>
        <li><em>“that”</em>: Matching</li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">[0-9]+\.[0-9]+\t</code></li>
      </ul>

      <p>This will select lines with a decimal value (###.##) followed by a tab character, the column separator in <span class="notranslate">Galaxy</span>. As a result, any lines without an identity value will be filtered out.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/tbl2gff3/tbl2gff3/1.2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/tbl2gff3/tbl2gff3/1.2"><strong>Table to GFF3</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/tbl2gff3/tbl2gff3/1.2</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Table”</em>: the output of the above <strong>Select lines</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step.</li>
        <li><em>“Record ID column or value”</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
        <li><em>“Feature start column or value”</em>: <code class="language-plaintext highlighter-rouge">9</code></li>
        <li><em>“Feature end column or value”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
        <li><em>“Feature score column or value”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
        <li><em>“Feature source column or value”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Qualifiers”</em>
          <ul>
            <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">Name</code></li>
            <li><em>“Qualifier value column or raw text”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Qualifiers”</em>
          <ul>
            <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">phenotype</code></li>
            <li><em>“Qualifier value column or raw text”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Qualifiers”</em>
          <ul>
            <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">accession</code></li>
            <li><em>“Qualifier value column or raw text”</em>: <code class="language-plaintext highlighter-rouge">11</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.3.4.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.3.4.3+galaxy0"><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.3.4.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>“Is this single or paired library”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA/Q file #1”</em>: <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_1 uncompressed (R1 paired)</code> (output of <strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA/Q file #2”</em>: <code class="language-plaintext highlighter-rouge">Trimmomatic on DRR187559_2 uncompressed (R2 paired)</code> (output of <strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select reference genome”</em>: <code class="language-plaintext highlighter-rouge">contigs</code> (output of <strong>Shovill</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">1: Default setting only</code> - You should have a look at the non-default parameters and try to understand them, but they don’t need to be changed currently.</li>
        <li><em>“Save the bowtie2 <span class="notranslate">mapping</span> statistics to the history”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.9+galaxy0"><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>“Reference genome to display”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Shovill on ...: Contigs</code> (output of <strong>Shovill assembly</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>“Genetic Code”</em>: <code class="language-plaintext highlighter-rouge">11. The Bacterial, Archaeal and Plant Plastid Code</code></li>
        <li>In <em>“Track Group”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Track Group”</em>
              <ul>
                <li><em>“Track Category”</em>: <code class="language-plaintext highlighter-rouge">Prokka</code></li>
                <li>In <em>“Annotation Track”</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Annotation Track”</em>
                      <ul>
                        <li><em>“Track Type”</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/<span class="notranslate">BED</span> Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GFF/GFF3/<span class="notranslate">BED</span> Track Data”</em>: <code class="language-plaintext highlighter-rouge">Prokka on data ...: gff</code> (output of <strong>Prokka</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                            <li><em>“JBrowse Track Type [Advanced]”</em>: <code class="language-plaintext highlighter-rouge">Neat Canvas Features</code></li>
                            <li><em>“Track Visibility”</em>: <code class="language-plaintext highlighter-rouge">On for new users</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Track Group”</em>
              <ul>
                <li><em>“Track Category”</em>: <code class="language-plaintext highlighter-rouge">AMR</code></li>
                <li>In <em>“Annotation Track”</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Annotation Track”</em>
                      <ul>
                        <li><em>“Track Type”</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/<span class="notranslate">BED</span> Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GFF/GFF3/<span class="notranslate">BED</span> Track Data”</em>: <code class="language-plaintext highlighter-rouge">Table to GFF3 on ...</code>, the output of the table to gff3 step</li>
                            <li><em>“JBrowse Track Type [Advanced]”</em>: <code class="language-plaintext highlighter-rouge">Neat Canvas Features</code></li>
                            <li><em>“Track Visibility”</em>: <code class="language-plaintext highlighter-rouge">On for new users</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Track Group”</em>
              <ul>
                <li><em>“Track Category”</em>: <code class="language-plaintext highlighter-rouge">Sequencing</code></li>
                <li>In <em>“Annotation Track”</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Annotation Track”</em>
                      <ul>
                        <li><em>“Track Type”</em>: <code class="language-plaintext highlighter-rouge">BAM Pileups</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM Track Data”</em>: Bowtie2’s output</li>
                            <li><em>“Autogenerate SNP Track”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>View the output of JBrowse</li>
  </ol>

</blockquote>

<p>In the output of the JBrowse you can view the mapped <span class="notranslate">reads</span> and the found genes against the reference genome. With the search tools you can easily find genes of interest. JBrowse can handle many inputs and can be very useful. Using the bowtie2 <span class="notranslate">mapping</span> output, low coverage regions can be detected. This SNP detection can also give a clear view of where the data was less reliable or where variations were located.</p>

<p>If it takes too long to build the JBrowse instance, you can view an embedded one here. (<strong>Warning</strong>: feature name search will not work.)</p>

<!--SNIPPET-->
<iframe id="embedded_jbrowse" src="/training-material/assets/jbrowse/index.html?loc=contig00018:5488..27391&amp;data=/training-material/topics/assembly/tutorials/mrsa-illumina/data&amp;trackList=0" border="0" width="100%" height="500px"></iframe>
<p><a href="/training-material/assets/jbrowse/index.html?loc=contig00018:5488..27391&amp;data=/training-material/topics/assembly/tutorials/mrsa-illumina/data" target="_blank" style="padding: 0.5rem 1rem; background-color: #dfe5f9;border-bottom-left-radius: 8px; border-bottom-right-radius: 8px;">Open JBrowse in a new tab</a></p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Illumina assemblies can be done with Shovill, but the assemblies are not completely resolved</p>
</li>
                        
                        <li><p>Annotation with Prokka is very easy</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/assembly/faqs/">FAQ page for the Assembly topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Lozano_2012">Lozano, C., Garcı́a-Migura Lourdes, C. Aspiroz, M. Zarazaga, C. Torres <i>et al.</i>, 2012 <b>Expansion of a Plasmid Classification System for Gram-Positive Bacteria and Determination of the Diversity of Plasmids in Staphylococcus aureus Strains of Human, Animal, and Food Origins</b>. Applied and Environmental Microbiology 78: 5948–5955. <a href="https://doi.org/10.1128/aem.00870-12">10.1128/aem.00870-12</a></li>
<li id="Zankari2012">Zankari, E., H. Hasman, S. Cosentino, M. Vestergaard, S. Rasmussen <i>et al.</i>, 2012 <b>Identification of acquired antimicrobial resistance genes</b>. Journal of Antimicrobial Chemotherapy 67: 2640–2644. <a href="https://doi.org/10.1093/jac/dks261">10.1093/jac/dks261</a></li>
<li id="Gurevich2013">Gurevich, A., V. Saveliev, N. Vyahhi, and G. Tesler, 2013 <b>QUAST: quality assessment tool for genome assemblies</b>. Bioinformatics 29: 1072–1075. <a href="https://doi.org/10.1093/bioinformatics/btt086">10.1093/bioinformatics/btt086</a></li>
<li id="Carattoli2014">Carattoli, A., E. Zankari, Garcı́a-Fernández Aurora, M. V. Larsen, O. Lund <i>et al.</i>, 2014 <b>In SilicoDetection and Typing of Plasmids using PlasmidFinder and Plasmid Multilocus Sequence Typing</b>. Antimicrobial Agents and Chemotherapy 58: 3895–3903. <a href="https://doi.org/10.1128/aac.02412-14">10.1128/aac.02412-14</a></li>
<li id="Jia2016">Jia, B., A. R. Raphenya, B. Alcock, N. Waglechner, P. Guo <i>et al.</i>, 2016 <b>CARD 2017: expansion and model-centric curation of the comprehensive antibiotic resistance database</b>. Nucleic Acids Research 45: D566–D573. <a href="https://doi.org/10.1093/nar/gkw1004">10.1093/nar/gkw1004</a></li>
<li id="Zankari2017">Zankari, E., R. Allesøe, K. G. Joensen, L. M. Cavaco, O. Lund <i>et al.</i>, 2017 <b>PointFinder: a novel web tool for WGS-based detection of antimicrobial resistance associated with chromosomal point mutations in bacterial pathogens</b>. Journal of Antimicrobial Chemotherapy 72: 2764–2768. <a href="https://doi.org/10.1093/jac/dkx217">10.1093/jac/dkx217</a></li>
<li id="Hikichi_2019">Hikichi, M., M. Nagao, K. Murase, C. Aikawa, T. Nozawa <i>et al.</i>, 2019 <b>Complete Genome Sequences of Eight Methicillin-Resistant Staphylococcus aureus Strains Isolated from Patients in Japan</b> (I. L. G. Newton, Ed.). Microbiology Resource Announcements 8: <a href="https://doi.org/10.1128/mra.01212-19">10.1128/mra.01212-19</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Genome Assembly of MRSA using Illumina MiSeq Data (Assembly)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Bazante Sanders, Miaomiao Zhou, Saskia Hiltemann, Helena Rasche, Avans Hogeschool, 2021 <b>Genome Assembly of MRSA using Illumina MiSeq Data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{assembly-mrsa-illumina,
author = "Bazante Sanders and Miaomiao Zhou and Saskia Hiltemann and Helena Rasche and Avans Hogeschool",
title = "Genome Assembly of MRSA using Illumina MiSeq Data (Galaxy Training Materials)",
year = "2021",
month = "12",
day = "14"
url = "\url{https://training.galaxyproject.org/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                
                <blockquote class="agenda follow-up">
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/visualisation">Visualisation</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Genomic Data Visualisation with JBrowse:
                            
                                <a href="/training-material/topics/visualisation/tutorials/jbrowse/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/galaxy-interface">Using Galaxy and Managing your Data</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Extracting Workflows from Histories:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
