<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>16S Microbial analysis with Nanopore data</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1704933032' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1704933032' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="16S Microbial analysis with Nanopore data">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-18 09:34:29 +0000">
<meta name="DC.creator" content="Cristóbal Gallardo">

        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: 16S Microbial analysis with Nanopore data">
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-18 09:34:29 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "16S Microbial analysis with Nanopore data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for '16S Microbial analysis with Nanopore data' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html",
  "timeRequired": "PT2H",
  "teaches": "Use Nanopore data for studying soil metagenomics\n - Analyze and preprocess Nanopore reads\n - Use Kraken2 to assign a taxonomic labels",
  "keywords": "metagenomics, nanopore",
  "description": "The questions this  addresses are:\n - How can we analyse the health status of the soil?\n - How do plants modify the composition of microbial communities?\n\n\nThe objectives are:\n - Use Nanopore data for studying soil metagenomics\n - Analyze and preprocess Nanopore reads\n - Use Kraken2 to assign a taxonomic labels\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "name": "Cristóbal Gallardo",
      "image": "https://avatars.githubusercontent.com/gallardoalba",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5752-2155",
      "orcid": "https://orcid.org/0000-0002-5752-2155"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "nanopore"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1704933032'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-metagenomics">
    <h1 data-toc-skip>16S Microbial analysis with Nanopore data</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/gallardoalba?s=36" alt="Avatar" width="36" height="36">Cristóbal Gallardo</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can we analyse the health status of the soil?</p>
</li>
        
        <li><p>How do plants modify the composition of microbial communities?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Use Nanopore data for studying soil metagenomics</p>
</li>
        
        <li><p>Analyze and preprocess Nanopore reads</p>
</li>
        
        <li><p>Use Kraken2 to assign a taxonomic labels</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4274812">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/" title="16S Microbial analysis with Nanopore data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://galaxytrakr.org/" title="">
			GalaxyTrakr
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 18, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00207">https://gxy.io/GTN:T00207</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p><strong>Healthy soils</strong> are an essential element in maintaining the planet’s ecological balance. That is why their protection must be considered a priority in order to guarantee the well-being of humanity. The alteration of microbial populations often precedes changes in the physical and chemical properties of soils, so monitoring their condition can serve to predict their future evolution, allowing to develop strategies to <strong>mitigate ecosystem damage</strong>.</p>

<p>Advances in sequencing technologies have opened up the possibility of using the study of taxa present in bacterial communities as indicators of soil condition. For such an approach to be possible, variations in microbial populations need to be less affected by spatial factors than by human-derived alterations, which has been confirmed by various investigations (<span class="citation"><a href="#Hermans2016">Hermans <i>et al.</i> 2016</a></span>, <span class="citation"><a href="#Fierer2006">Fierer and Jackson 2006</a></span>).</p>

<p>Among the environmental variables that have shown the greatest correlation with alterations in the composition of bacterial communities are soil pH, the carbon-nitrogen ratio (C:N), and the co-concentrations of Olsen P (a measure of plant available phosphorus), aluminium and copper (Fig. 1).</p>

<p><a href="../../images/metagenomics-nanopore/bioindicators.jpg" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/bioindicators.jpg" alt="Bacterial bioindicators. " width="1280" height="894" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Relationship between bacterial community composition and soil parameters. The radius of each circle represents the taxon abundance that was accounted for by each soil variable, based on adjusted R-squared values from distance-based multivariate multiple regression analyses. The univariate relationship between the abundance of each taxon and soil variables, calculated using Pearson’s correlation coefficient, is represented by the color of the circle. Phyla or classes are ordered according to overall abundance in all the samples, from most abundant (top) to least abundant (bottom). Image published originally in <span class="citation"><a href="#Hermans2016">Hermans <i>et al.</i> 2016</a></span>.&lt;/figcaption&gt;</figure></a></p>

<p>On the other hand, many species of microorganisms establish complex <strong>symbiotic relationships</strong> with plant organisms. The fraction of the substrate that is directly influenced by root secretions and associated microorganism is known as <strong>rhizosphere</strong>. Thus, for example, bacteria of the genus <em>Bacillus</em>, <em>Pseudomonas</em> or <em>Burkholderia</em> appear associated with the plant roots, protecting them from pathogenic microorganisms.</p>

<p><a href="../../images/metagenomics-nanopore/bacteria_community.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/bacteria_community.png" alt="Bacterial communities. " width="1280" height="720" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Representation of the relationships between root microbiomes and plant organisms.&lt;/figcaption&gt;</figure></a></p>

<p>In this tutorial, we will use sequencing data obtained through the <strong>MinION sequencer</strong> (Oxford Nanopore Technologies) with two objectives: 1) evaluate the health status of soil samples and 2) study how microbial populations are modified by their interaction with plant roots. This example was inspired by <span class="citation"><a href="#Brown2017">Brown <i>et al.</i> 2017</a></span>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#background-on-data" id="markdown-toc-background-on-data">Background on data</a>    <ol>
      <li><a href="#the-datasets" id="markdown-toc-the-datasets">The datasets</a></li>
    </ol>
  </li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#assess-datasets-quality" id="markdown-toc-assess-datasets-quality">Assess datasets quality</a>    <ol>
      <li><a href="#quality-control-using-fastqc-and-multiqc" id="markdown-toc-quality-control-using-fastqc-and-multiqc">Quality control using FastQC and MultiQC</a></li>
    </ol>
  </li>
  <li><a href="#improve-the-dataset-quality" id="markdown-toc-improve-the-dataset-quality">Improve the dataset quality</a>    <ol>
      <li><a href="#adapter-and-chimera-removal-with-porechop" id="markdown-toc-adapter-and-chimera-removal-with-porechop">Adapter and chimera removal with porechop</a></li>
      <li><a href="#filter-sequences-with-fastp" id="markdown-toc-filter-sequences-with-fastp">Filter sequences with fastp</a></li>
    </ol>
  </li>
  <li><a href="#re-evaluate-datasets-quality" id="markdown-toc-re-evaluate-datasets-quality">Re-evaluate datasets quality</a></li>
  <li><a href="#assign-taxonomic-classifications" id="markdown-toc-assign-taxonomic-classifications">Assign taxonomic classifications</a>    <ol>
      <li><a href="#taxonomic-classification-with-kraken2" id="markdown-toc-taxonomic-classification-with-kraken2">Taxonomic classification with Kraken2</a></li>
    </ol>
  </li>
  <li><a href="#analyze-taxonomic-assigment" id="markdown-toc-analyze-taxonomic-assigment">Analyze taxonomic assigment</a>    <ol>
      <li><a href="#visualize-the-taxonomical-classification-with-krona" id="markdown-toc-visualize-the-taxonomical-classification-with-krona">Visualize the taxonomical classification with Krona</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="background-on-data">Background on data</h1>

<p>Next, we will introduce some details about the datasets that we are going to use to perform the analysis.</p>

<h2 id="the-datasets">The datasets</h2>

<p>In this example, we will use a dataset originally hosted in the <strong>NCBI SRA database</strong>, with the accession number <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP194577">SRP194577</a>. To obtain the samples, the DNA was extracted using the Zymo Research Kit, followed by <strong>PCR amplification of 16S rRNA genes</strong>. Sequencing was carried out on a Nanopore Minion device. We are going to use four datasets, corresponding to two experimental conditions:</p>

<ul>
  <li>Soil:
    <ul>
      <li>Surface sample (0-5 cm): <code class="language-plaintext highlighter-rouge">bulk_top.fastq.gz</code></li>
      <li>Deep sample (10-15 cm): <code class="language-plaintext highlighter-rouge">bulk_bottom.fastq.gz</code></li>
    </ul>
  </li>
  <li>Rhizosphere:
    <ul>
      <li>Surface sample (0-5 cm): <code class="language-plaintext highlighter-rouge">rhizosphere_top.fastq.gz</code></li>
      <li>Deep sample (10-15 cm): <code class="language-plaintext highlighter-rouge">rhizosphere_bottom.fastq.gz</code></li>
    </ul>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Why do we sequence the 16S rRNA genes for analyzing microbial communities?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>There are several reasons to use these genes as taxonomic markers. First, they are constituent parts of ribosomes, a piece of highly-conserved biological machinery, which makes them universally distributed. On the other hand, these genes present regions with varying degrees of sequence variability, which allows them to be used as a species-specific signature.</p>
  </blockquote>
</blockquote>

<h1 id="get-data">Get data</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Collection</strong> tab</li>
        <li>Click of the <strong>Paste/Fetch</strong> button</li>
        <li>Paste the Zenodo links and press <kbd>Start</kbd> and <kbd>Build</kbd></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> https://zenodo.org/record/4274812/files/bulk_bottom.fastq.gz
 https://zenodo.org/record/4274812/files/bulk_top.fastq.gz
 https://zenodo.org/record/4274812/files/rhizosphere_bottom.fastq.gz
 https://zenodo.org/record/4274812/files/rhizosphere_top.fastq.gz
</code></pre></div>      </div>
      <ul>
        <li>Assign a name to the new collection: <code class="language-plaintext highlighter-rouge">soil collection</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="assess-datasets-quality">Assess datasets quality</h1>

<p>Before starting to work on our datasets, it is necessary to assess their quality. This is an essential step if we aim to obtain a <strong>meaningful downstream analysis</strong>.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-fastq-sequence-quality-format"><button class="gtn-boxify-button details" type="button" aria-controls="details-fastq-sequence-quality-format" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: FASTQ sequence quality format<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>FASTQ format is a text-based format for storing both a biological sequence and its corresponding quality scores. Both the sequence letter and the quality score are encoded with a single ASCII character for brevity. You can find more info in the <a href="https://en.wikipedia.org/wiki/FASTQ_format">Wikipedia article</a>.</p>

</blockquote>

<h2 id="quality-control-using-fastqc-and-multiqc">Quality control using FastQC and MultiQC</h2>

<p><strong>FastQC</strong> is one of the most widely used tools to <strong>check the quality</strong> of the samples generated by High Throughput Sequencing (HTS) technologies.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">soil collection</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC unprocessed: Raw</code> and <code class="language-plaintext highlighter-rouge">FastQC unprocessed: Web</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1</span></span> with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>
              <p><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>:</p>

              <ul>
                <li><code class="language-plaintext highlighter-rouge">FastQC unprocessed: Raw</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<p><strong>FastQC</strong> provides information on various parameters, such as the range of quality values across all bases at each position. <strong>MultiQC</strong> allows summarizing the output of different outputs from <strong>FastQC</strong>.</p>

<p><a href="../../images/metagenomics-nanopore/fastqc_sequence_length_distribution_plot.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/fastqc_sequence_length_distribution_plot.png" alt="FastQC plots length. " width="800" height="533" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Sequence length distribution&lt;/figcaption&gt;</figure></a></p>

<p>In the first place, we are going to analyse the sequence length distribution of the different datasets.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Can you explain the sequence length distribution plot?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The main peak, around 1700 bp, corresponds approximately to the length of the gene coding for 16S rRNA. There is also a secondary peak around 200 bp, which may be due to <strong>truncated amplifications</strong>, or as a result of <strong>non-specific hybridization of primers</strong> used for PCR.</p>

  </blockquote>
</blockquote>

<p><a href="../../images/metagenomics-nanopore/fastqc_per_base_sequence_quality_plot.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/fastqc_per_base_sequence_quality_plot.png" alt="FastQC plots quality. " width="800" height="533" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Per base sequence quality&lt;/figcaption&gt;</figure></a></p>

<p>If we examine figure 3, we can see that up to 3000 bp the quality of our sequencing data is around a <strong>Phred score</strong> of 12, which is a relatively low value compared to other sequencing technologies. This is explained because Nanopore reads poses <strong>high error rates in the basecalled reads</strong> (10% as compared to 1% for Illumina). However, Nanopore sequencing generates <strong>very long reads</strong> (in theory only limited by the mechanisms of extraction of the genetic material), enabling the sequencing of the complete 16S rRNA gene, which makes it possible to identify bacterial taxa at higher resolution.</p>

<p><a href="../../images/metagenomics-nanopore/fastqc_per_sequence_gc_content_plot.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/fastqc_per_sequence_gc_content_plot.png" alt="FasTQC plots CG. " width="800" height="533" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Per sequence GC content&lt;/figcaption&gt;</figure></a></p>

<p>Let’s have a look at figure 4, which shows the content of the GC by sequence.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What can cause GC content to show a bimodal peak?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>There are several possible causes for the presence of twin peaks in the GC content. One possible explanation is the presence of adapters in the sequence. Another possible cause is some kind of contamination, such as chimeras.</p>

  </blockquote>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>For more information on the topic, please see <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">the quality control tutorial</a>.</p>
</blockquote>

<h1 id="improve-the-dataset-quality">Improve the dataset quality</h1>

<p>In order to improve the quality of our data, we will use two tools presented above, <strong>porechop</strong> and <strong>fastp</strong>.</p>

<h2 id="adapter-and-chimera-removal-with-porechop">Adapter and chimera removal with porechop</h2>

<p>Nanopore sequencing technology requires to ligate adapters to both ends of genomic material to facilitate the strand capture and loading of a processive enzyme at the 5’end, boosting the effectiveness of the sequencing process. Adapter sequences should be removed because they <strong>can interfere with aligment of reads</strong> to 16S rRNA gene reference database, for which we will use the <strong>porechop</strong> tool (<span class="citation"><a href="#rrwick2017">Wick 2017</a></span>).</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>If you are interested in Nanopore sequecing technology, you can find more information in <span class="citation"><a href="#Jain2016">Jain <i>et al.</i> 2016</a></span>.</p>

</blockquote>

<p>On the other hand, <strong>chimeric sequences</strong> are considered a contaminant and should be removed because they can result in artificial inflation of the microbial diversity. With <strong>Porechop</strong> you can eliminate them.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-adapters-with-porechop"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove adapters with porechop</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3" title="Porechop tool" aria-role="link"><strong>Porechop</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3</span></span> with the following parameters:
      <ul>
        <li><em>“Input FASTA/FASTQ”</em>: <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <code class="language-plaintext highlighter-rouge">soil collection</code></li>
        <li><em>“Output format for the reads”</em>: <code class="language-plaintext highlighter-rouge">fastq</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">soil collection trimmed</code></li>
  </ol>

</blockquote>

<h2 id="filter-sequences-with-fastp">Filter sequences with fastp</h2>

<p>To increase the specificity of the analysis, we will select the reads with lengths between 1000 bp and 2000 bp, which are more informative from a taxonomic point of view, because they include both preserved and hypervariable regions of the 16S rRNA gene. In addition, sequences will be filtered on a minimum average read quality score of 9, according to the recommendations from <span class="citation"><a href="#Nygaard2020">Nygaard <i>et al.</i> 2020</a></span>. This stage will be carried out through the use of <strong>fastp</strong> (<span class="citation"><a href="#Chen_2018">Chen <i>et al.</i> 2018</a></span>), an open-source tool designed to process FASTQ files.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-sequence-with-fastp"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter sequence with fastp</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0" title="fastp tool" aria-role="link"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0</span></span> with the following parameters:
      <ul>
        <li><em>“Single-end or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">soil collection trimmed</code>
          <ul>
            <li>In <em>“Adapter Trimming Options”</em>:
              <ul>
                <li><em>“Disable adapter trimming”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>:
          <ul>
            <li>In <em>“Quality filtering options”</em>:
              <ul>
                <li><em>“Qualified quality phred”</em>: <code class="language-plaintext highlighter-rouge">9</code></li>
              </ul>
            </li>
            <li>In <em>“Length filtering options”</em>:
              <ul>
                <li><em>“Length required”</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
                <li><em>“Maximum length”</em>: <code class="language-plaintext highlighter-rouge">2000</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>:
          <ul>
            <li><em>“PolyG tail trimming”</em>: <code class="language-plaintext highlighter-rouge">Disable polyG tail trimming</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
  </ol>

</blockquote>

<h1 id="re-evaluate-datasets-quality">Re-evaluate datasets quality</h1>

<p>After processing the sequences, we are going to analyze them again using <strong>FastQC</strong> and <strong>MultiQC</strong> to see if we have managed to correct the anomalies that we had detected.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC processed: Raw</code> and <code class="language-plaintext highlighter-rouge">FastQC processed: Web</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1</span></span> with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>
              <p><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>:</p>

              <ul>
                <li><code class="language-plaintext highlighter-rouge">FastQC processed: Raw</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<p>We can verify that after processing the samples, the GC content presents a unimodal distribution, which indicates that the anomalies in the sequences have been successfully eliminated (Figure 6).</p>

<p><a href="../../images/metagenomics-nanopore/fasqc_per_sequence_content_plot_clean.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/metagenomics-nanopore/fasqc_per_sequence_content_plot_clean.png" alt="FasTQC plots GC. " width="746" height="475" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Per sequence GC content in processed samples&lt;/figcaption&gt;</figure></a></p>

<h1 id="assign-taxonomic-classifications">Assign taxonomic classifications</h1>

<p>One of the key steps in metagenomic data analysis is to identify the taxon to which the individual reads belong. Taxonomic classification tools are based on microbial genome databases to identify the origin of each sequence.</p>

<h2 id="taxonomic-classification-with-kraken2">Taxonomic classification with Kraken2</h2>

<p>To perform the taxonomic classification we will use <strong>Kraken2</strong> (<span class="citation"><a href="#Wood2019">Wood <i>et al.</i> 2019</a></span>). This tool uses the minimizer method to sample the k-mers (all the read’s subsequences of length <em>k</em>) in a deterministic fashion in order to reduce memory constumption and processing time. In addition, it masks low-complexity sequences from reference sequences by using <strong>dustmasker</strong>.</p>

<!--SNIPPET-->
<blockquote class="details">   <div class="box-title details-title" id="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification"><button class="gtn-boxify-button details" type="button" aria-controls="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> Details: Kraken2 and the k-mer approach for taxonomy classification<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In the \(k\)-mer approach for taxonomy classification, we use a database containing DNA sequences of genomes whose taxonomy we already know. On a computer, the genome sequences are broken into short pieces of length \(k\) (called \(k\)-mers), usually 30bp.</p>   <p><strong>Kraken</strong> examines the \(k\)-mers within the query sequence, searches for them in the database, looks for where these are placed within the taxonomy tree inside the database, makes the classification with the most probable position, then maps \(k\)-mers to the lowest common ancestor (LCA) of all genomes known to contain the given \(k\)-mer.</p>   <p><img src="/training-material/topics/metagenomics/faqs/images/kmers-kraken.jpg" alt="Kraken2" title="Kraken sequence classification algorithm. To classify a sequence, each k-mer in the sequence is mapped to the lowest common ancestor (LCA, i.e. the lowest node) of the genomes that contain that k-mer in the database. The taxa associated with the sequence's k-mers, as well as the taxa's ancestors, form a pruned subtree of the general taxonomy tree, which is used for classification. In the classification tree, each node has a weight equal to the number of k-mers in the sequence associated with the node's taxon. Each root-to-leaf (RTL) path in the classification tree is scored by adding all weights in the path, and the maximal RTL path in the classification tree is the classification path (nodes highlighted in yellow). The leaf of this classification path (the orange, leftmost leaf in the classification tree) is the classification used for the query sequence. Source: &lt;span class=&quot;citation&quot;&gt;&lt;a href=&quot;#Wood2014&quot;&gt;Wood and Salzberg 2014&lt;/a&gt;&lt;/span&gt;" /></p>   <p><strong>Kraken2</strong> uses a compact hash table, a probabilistic data structure that allows for faster queries and lower memory requirements. It applies a spaced seed mask of <em>s</em> spaces to the minimizer and calculates a compact hash code, which is then used as a search query in its compact hash table; the lowest common ancestor (LCA) taxon associated with the compact hash code is then assigned to the k-mer.</p>   <p>You can find more information about the <strong>Kraken2</strong> algorithm in the paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0"><em>Improved metagenomic analysis with Kraken 2</em></a>.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<p>For this tutorial, we will use the <strong>SILVA database</strong> (<span class="citation"><a href="#Quast2012">Quast <i>et al.</i> 2012</a></span>). It includes over 3.2 million 16S rRNA sequences from the <em>Bacteria</em>, <em>Archaea</em> and <em>Eukaryota</em> domains.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-assign-taxonomic-labels-with-kraken2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assign taxonomic labels with Kraken2</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0" title="Kraken2 tool" aria-role="link"><strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0</span></span> with the following parameters:
      <ul>
        <li><em>“Single or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
        <li><em>“Print scientific names instead of just taxids”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Confidence”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
        <li>In <em>“Create Report”</em>:
          <ul>
            <li><em>“Print a report with aggregrate counts/clade to file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Format report output like Kraken 1’s kraken-mpa-report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Select a Kraken2 database”</em>: <code class="language-plaintext highlighter-rouge">Silva (Created: 2020-06-24T164526Z, kmer-len=35, minimizer-len=31, minimizer-spaces=6)</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>A confidence score of 0.1 means that at least 10% of the k-mers should match entries in the database. This value can be reduced if a less restrictive taxonomic assignation is desired.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="analyze-taxonomic-assigment">Analyze taxonomic assigment</h1>

<p>Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the <strong>Krona pie chart</strong> tool (<span class="citation"><a href="#Ondov_2011">Ondov <i>et al.</i> 2011</a></span>). But before that, we need to adjust the format of the data output from <strong>Kraken2</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-adjust-dataset-format"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Adjust dataset format</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_reverse/datamash_reverse/1.1.0" title="Reverse tool" aria-role="link"><strong>Reverse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/datamash_reverse/datamash_reverse/1.1.0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Report: Kraken2 on collection</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2" title="Replace Text tool" aria-role="link"><strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Reverse on collection</code></li>
        <li>In <em>“Replacement”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Replacement”</em>
              <ul>
                <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">\|</code></li>
                <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">\t</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Remove beginning1" title="Remove beginning tool" aria-role="link"><strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Remove beginning1</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Replace Text on collection</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="visualize-the-taxonomical-classification-with-krona">Visualize the taxonomical classification with Krona</h2>

<p><strong>Krona</strong> allows hierarchical data to be explored with zooming, multi-layered pie charts. With this tool, we can easily visualize the composition of the bacterial communities and compare how the populations of microorganisms are modified according to the conditions of the environment.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualize-metagenomics-analysis-results"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualize metagenomics analysis results</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1" title="Krona pie chart tool" aria-role="link"><strong>Krona pie chart</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1</span></span> with the following parameters:
      <ul>
        <li><em>“What is the type of your input data”</em>: <code class="language-plaintext highlighter-rouge">Tabular</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Remove beginning on collection</code></li>
        <li><em>“Provide a name for the basal rank”</em>: <code class="language-plaintext highlighter-rouge">Bacteria</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Let’s take a look at the result. Using the search bar we can check if certain taxa are present.</p>

<iframe id="krona" src="krona_all.html" frameborder="0" width="100%" height="900px"> <a href="../../images/metagenomics-nanopore/krona.png" rel="noopener noreferrer"><img src="../../images/metagenomics-nanopore/krona.png" alt="Krona. " width="1325" height="874" loading="lazy" /></a> </iframe>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What can we say about the health status of the soil samples?</li>
    <li>Are there significant differences in the bacterial population structure between soil samples and rhizosphere samples?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The low presence of <em>Alphaproteobacteria</em> (including members of the order <em>Rhizobiales</em>), as well as the abundance of <em>Bacteroidetes</em> and <em>Gammaproteobacteria</em>, suggest that the soil is highly exposed to phosphorus. This mineral is strongly associated with the anthropogenic activity as it is an important component of agrochemicals.</li>
      <li>We can observe that there are important differences in the structure of the bacterial populations between the soil and rhizosphere samples. Particularly significant is the increase in phylum <em>Planctomycetes</em>, which are usually abundant in the rhizosphere.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we used MinION Nanopore sequencing data to study the health status of soil samples and the structure of bacterial populations. The results indicate that the soil suffers some degree of erosion as a result of their exposure to agrochemicals. We have also been able to study how the composition of microbial communities is modified in presence of plant organisms.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>We learned to use MinION Nanopore data for analyzing the health status of the soil</p>
</li>
                        
                        <li><p>We preprocessed Nanopore sequences in order to improve their quality</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Fierer2006">Fierer, N., and R. B. Jackson, 2006 <b>The diversity and biogeography of soil bacterial communities</b>. Proceedings of the National Academy of Sciences 103: 626–631. <a href="https://doi.org/10.1073/pnas.0507535103">10.1073/pnas.0507535103</a></li>
<li id="Ondov_2011">Ondov, B. D., N. H. Bergman, and A. M. Phillippy, 2011 <b>Interactive metagenomic visualization in a Web browser</b>. BMC Bioinformatics 12: <a href="https://doi.org/10.1186/1471-2105-12-385">10.1186/1471-2105-12-385</a></li>
<li id="Quast2012">Quast, C., E. Pruesse, P. Yilmaz, J. Gerken, T. Schweer <i>et al.</i>, 2012 <b>The SILVA ribosomal RNA gene database project: improved data processing and web-based tools</b>. Nucleic Acids Research 41: D590–D596. <a href="https://doi.org/10.1093/nar/gks1219">10.1093/nar/gks1219</a></li>
<li id="Wood2014">Wood, D. E., and S. L. Salzberg, 2014 <b>Kraken: ultrafast metagenomic sequence classification using exact alignments</b>. Genome Biology 15: R46. <a href="https://doi.org/10.1186/gb-2014-15-3-r46">10.1186/gb-2014-15-3-r46</a></li>
<li id="Hermans2016">Hermans, S. M., H. L. Buckley, B. S. Case, F. Curran-Cournane, M. Taylor <i>et al.</i>, 2016 <b>Bacteria as Emerging Indicators of Soil Condition</b> (F. E. Loeffler, Ed.). Applied and Environmental Microbiology 83: <a href="https://doi.org/10.1128/aem.02826-16">10.1128/aem.02826-16</a></li>
<li id="Jain2016">Jain, M., H. E. Olsen, B. Paten, and M. Akeson, 2016 <b>The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</b>. Genome Biology 17: <a href="https://doi.org/10.1186/s13059-016-1103-0">10.1186/s13059-016-1103-0</a></li>
<li id="Brown2017">Brown, B. L., M. Watson, S. S. Minot, M. C. Rivera, and R. B. Franklin, 2017 <b>MinION™ nanopore sequencing of environmental metagenomes: a synthetic approach</b>. GigaScience 6: <a href="https://doi.org/10.1093/gigascience/gix007">10.1093/gigascience/gix007</a></li>
<li id="rrwick2017">Wick, R., 2017 <b>Porechop</b>. <i>GitHub</i>. <a href="https://github.com/rrwick/Porechop">https://github.com/rrwick/Porechop</a></li>
<li id="Chen_2018">Chen, S., Y. Zhou, Y. Chen, and J. Gu, 2018 <b>fastp: an ultra-fast all-in-one FASTQ preprocessor</b>. <a href="https://doi.org/10.1101/274100">10.1101/274100</a></li>
<li id="Wood2019">Wood, D. E., J. Lu, and B. Langmead, 2019 <b>Improved metagenomic analysis with Kraken 2</b>. Genome Biology 20: <a href="https://doi.org/10.1186/s13059-019-1891-0">10.1186/s13059-019-1891-0</a></li>
<li id="Nygaard2020">Nygaard, A. B., H. S. Tunsjø, R. Meisal, and C. Charnock, 2020 <b>A preliminary study on the potential of Nanopore MinION and Illumina MiSeq 16S rRNA gene sequencing to characterize building-dust microbiomes</b>. Scientific Reports 10: 3209. <a href="https://doi.org/10.1038/s41598-020-59771-0">10.1038/s41598-020-59771-0</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=16S Microbial analysis with Nanopore data (Metagenomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Cristóbal Gallardo,  <b>16S Microbial analysis with Nanopore data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metagenomics-nanopore-16S-metagenomics,
author = "Cristóbal Gallardo",
	title = "16S Microbial analysis with Nanopore data (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: taxonomy_krona_chart
  owner: crs4
  revisions: 1334cb4c6b68
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: e7b2202befea
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: datamash_reverse
  owner: iuc
  revisions: 0f1724dd59d2
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastp
  owner: iuc
  revisions: dbf9c561ef29
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: kraken2
  owner: iuc
  revisions: 328c607150ff
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: 5e33b465d8d5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: porechop
  owner: iuc
  revisions: 93d623d9979c
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00207">https://gxy.io/GTN:T00207</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1704933032'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>