<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Automated Cell Annotation</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case-cell-annotation/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703464120' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703464120' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Automated Cell Annotation">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-12-25 00:28:43 +0000">
<meta name="DC.creator" content="Morgan Howells">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Automated Cell Annotation">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703464120'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/single-cell/tutorials/scrna-case-cell-annotation/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-case-cell-annotation/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/single-cell/tutorials/scrna-case-cell-annotation/slides.html">View markdown source on GitHub</a>
	<h1>Automated Cell Annotation</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/hexhowells/" class="contributor-badge contributor-hexhowells"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hexhowells?s=36" alt="Avatar" width="36" height="36">Morgan Howells</a>
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>What is automated cell annotation?</p>
</li>
		
		<li><p>How is automated cell annotation performed?</p>
</li>
		
		<li><p>What methods are available?</p>
</li>
		
		<li><p>What is the best method to choose for your data?</p>
</li>
		
		</ul>
		

		
		<h2>Objectives</h2>
		<ul>
		
		<li><p>Learn the different methods underlying automated cell annotation</p>
</li>
		
		<li><p>Understand the data needed to annotate cells</p>
</li>
		
		<li><p>Be able to choose the most appropriate method for your data</p>
</li>
		
		<li><p>Recognise the common issues and be able to resolve them</p>
</li>
		
		</ul>
		

		
		<h2>Requirements</h2>
		<ul>
		
		
    <li>
    
        
        
        <a href="/training-material/topics/single-cell">Single Cell</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> An introduction to scRNA-seq data analysis:
                            
                                <a href="/training-material/topics/single-cell/tutorials/scrna-intro/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

		</ul>
		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Dec 25, 2023 </div>

	<hr />


	<h3 id="what-is-cell-annotation">What is Cell Annotation?</h3>

<ul>
  <li>
    <p>Categorising cells into cell type categories based on transcriptomic data</p>
  </li>
  <li>
    <p>Can be done manually or automatically</p>
  </li>
  <li>
    <p>Is possible due to single-cell sequencing technology</p>
  </li>
</ul>

<p><img src="../../images/scrna-cell-annotation/cell-annotation.png" alt="Basic pipeline for automated cell annotation" /></p>

<p><span>Speaker Notes</span></p>

<p>Automated cell annotation uses gene expression data to classify unknown cells (often referred to as queries) into discrete “cell type” categories.</p>

<p>This process can be done manually or automatically using a wide range of algorithms and statistical methods</p>

<hr />

<h3 id="why-is-it-important">Why is it Important?</h3>

<ul>
  <li>
    <p>Able to process and analyse single cell data much faster than manual analysis</p>
  </li>
  <li>
    <p>Can produce results more consistently allowing for reproducibility of results</p>
  </li>
  <li>
    <p>Can uncover complex patterns in large amounts of data</p>
  </li>
</ul>

<p><span>Speaker Notes</span></p>

<p>Manually annotating cells can be time consuming and require domain expertise, with the advent of single-cell RNA sequencing (scRNA-seq) large amounts of individual cell data can be processed, on order to annotate this large data automated methods need to be produced.</p>

<p>Additionally these automated tools can help produce more consistent results and allow reproducibility as they are open and usable by anybody.</p>

<p>As mentioned earlier determining cell types can be complex with thousands of cell types and subtypes and large amounts of potential data to process, therefore automated analysis makes interpreting this data and uncovering complex patterns much easier.</p>

<hr />

<h3 id="understanding-our-data">Understanding our Data</h3>

<p>.pull-right[</p>

<p><img src="../../images/scrna-cell-annotation/data-object.png" alt="Diagram showing the different sections of a AnnData object" /></p>

<p>]</p>

<p>.pull-left[</p>

<p>What data should we use to classify cells?</p>

<ul>
  <li>
    <p>Cell annotation uses sc-RNA seq data</p>
  </li>
  <li>
    <p>Gene expressions are stored in a gene expression matrix (X)</p>
  </li>
</ul>

<p>Common data types:</p>

<ul>
  <li>AnnData</li>
  <li>SingleCellExperiment</li>
  <li>CellDataSet</li>
  <li>Loom</li>
  <li>Seurat</li>
</ul>

<p>]</p>

<p><span>Speaker Notes</span></p>

<p>Cells are classified using gene expression data derived from single-cell RNA sequencing, this data tells us which genes are being actively expressed in a cell at a given time.</p>

<p>The expression data is stored in a gene expression matrix where each row represents genes and each column represents individual cells, in the diagram this matrix is represented by the green box, all other boxes represent different types of metadata.</p>

<p>There are many different formats for storing this data and metadata, since different tools use different data types converting between them is a common task.</p>

<hr />

<h3 id="challenges">Challenges</h3>

<ul>
  <li>
    <p>Data is often noisy due to amplification techniques, varying sequencing depths, and errors in the reads</p>
  </li>
  <li>
    <p>Cell type definitions are inherently subjective and may be suboptimal</p>
  </li>
  <li>
    <p>Dealing with unknown cell types due to undiscovered classifications</p>
  </li>
</ul>

<p><span>Speaker Notes</span></p>

<p>There are some challenges that need to be faced in order to perform automated cell annotation:</p>

<ul>
  <li>
    <p>Due to the process of reading the gene expression data from cells there can be a lot of noise introduced, algorithms should be robust to this noisy data to ensure accurate predictions are made. Additionally manually pre-processing the data (e.g. filtering poor quality reads) can help make the data more clean.</p>
  </li>
  <li>
    <p>Humans like to place things into categories, even when there are no clear definitions/separations, therefore the process of deciding the different cell types can be tricky and requires some subjective decisions to be made. Due to this the categories used may not be perfect and could affect the performance of these algorithms.</p>
  </li>
  <li>
    <p>Finally not all cell types are known and defined and we may end up running into them during our analysis, therefore it is important that our tools can handle undefined cell categories.</p>
  </li>
</ul>

<hr />

<h3 id="manual-cell-annotation">Manual Cell Annotation</h3>

<ul>
  <li>
    <p>Generate a cluster map and manually annotate each clusters</p>
  </li>
  <li>
    <p>Requires the researchers to find known marker genes in the data</p>
  </li>
</ul>

<p><img src="../../images/scrna-cell-annotation/manual-annotation.png" alt="Various cell cluster diagrams showing the expression values of various marker genes" /></p>

<p><span>Speaker Notes</span></p>

<p>In order to manually annotate cell data you would first need to reduce the number of dimensions in your data so that you can produce cluster maps, this is done using a reduction method (PCA, tSNE, UMAP, etc)</p>

<p>Now that your data has been separated into distinct clusters you now need to manually annotate each individual cluster. This is typically done using known marker genes and their associated cell types.</p>

<p>The figure shows a graph for each marker gene we are interested in, it can be seen that each marker gene is only present in certain areas of our data, ideally these areas will be individual clusters.</p>

<hr />

<h3 id="feature-selection">Feature Selection</h3>

<p>.pull-right[</p>

<p><img src="../../images/scrna-cell-annotation/gene-features.png" alt="Diagram showing a distribution of gene expressions - outlining genes further away from the mean being better features" /></p>

<p>]</p>

<p>.pull-left[</p>

<ul>
  <li>
    <p>The process of identifying unique gene expressions that can be used for identifying cell types</p>
  </li>
  <li>
    <p>Genes are selected based on how differently they are expressed across different cell types</p>
  </li>
</ul>

<p>]</p>

<p><span>Speaker Notes</span></p>

<p>Similar to manual annotation we need to use certain genes as indicators for determining each cell type (since certain gene profiles will correlate more strongly with certain cell types), these are referred to as features/marker genes.</p>

<p>The best gene expressions to use as features will be the ones that vary a lot across different cell type samples as certain gene expressions can be used as a ‘fingerprint’ for that certain cell type.</p>

<p>Features can be selected using various automated methods, some common methods include:</p>
<ul>
  <li>M3Drop</li>
  <li>ScMarker</li>
  <li>Highly variable genes (HVGs)</li>
</ul>

<p>Many tools allow you to choose how many features to use, performance will differ depending on the tool but as a general rule too many features will mean the useful information is too sparse and it will be difficult to find patterns, whereas too few features will lose a lot of useful information. It is best to first use the number of features recommended for the tool and make adjustments using that as a baseline.</p>

<hr />

<h3 id="different-classification-methods">Different Classification Methods</h3>

<p><img src="../../images/scrna-cell-annotation/cell-annotation-overview.png" alt="Overview of the three different types of methods (marker gene database, correlation-based, supervised classification)" /></p>

<p><span>Speaker Notes</span></p>

<p>There are three primary types of methods that can be applied to classifying cell types</p>

<ul>
  <li>Correlation based</li>
  <li>Cluster annotation using marker genes</li>
  <li>Supervised classification</li>
</ul>

<hr />

<h3 id="correlation-based-methods">Correlation-Based Methods</h3>

<ul>
  <li>
    <p>Compares the query with a reference dataset to find the most similar cells</p>
  </li>
  <li>
    <p>Measures similarity using different distance metric scores</p>
  </li>
  <li>
    <p>Can be applied at cell levels and cluster levels</p>
  </li>
</ul>

<p>.pull-right[</p>

<p><img src="../../images/scrna-cell-annotation/correlation-figure.png" alt="Figure showing a basic correlation-based pipeline" /></p>

<p>]</p>

<p><span>Speaker Notes</span></p>

<p>Correlation-based methods take the query and compares it to every cell/cluster in the reference dataset in an attempt to figure out which cells/clusters the query is most similar to. If the query has a strong similarity to a cell/cluster then it is probable that they share the same cell type classification, this idea is the basis of this method.</p>

<p>Similarity between the reference and the query is computed using various distance metric scores (cosine similarity, Pearson correlation, Spearman correlation, etc) and operates on gene expression data.</p>

<p>By using feature selection we can improve the efficiency of comparing each cell/cluster since there is less gene data to process. Additionally the reference data is stored as an n-dimensional feature space (where each gene is a separate dimension) which allows for the identification of clusters as well as faster searching using algorithms such as approximate nearest neighbour.</p>

<p>This method can be applied to entire clusters or to individual cells in the reference data. Applying the query to entire clusters can help reduce computation time but lacks accuracy, comparing the query to each individual cell yields much better accuracy with the expense of time.</p>

<hr />

<h3 id="cluster-annotation-with-marker-genes">Cluster Annotation with Marker Genes</h3>

<ul>
  <li>
    <p>Uses a known database of marker genes to identify cell types</p>
  </li>
  <li>
    <p>Requires feature selection for comparison with the reference data</p>
  </li>
  <li>
    <p>Can be limited by the datasets available</p>
  </li>
</ul>

<p>.pull-right[</p>

<p><img src="../../images/scrna-cell-annotation/marker-figure.png" alt="Figure showing a basic marker gene database pipeline" /></p>

<p>]</p>

<p><span>Speaker Notes</span></p>

<p>Cluster annotation methods rely on known marker genes for identifying cell types. These genes are hand-selected features that can be used to uniquely identify different cell types. Using a database of marker genes, query data can be compared against the database to determine the most likely matching cell types.</p>

<p>In order to compare the query with known marker gene databases features need to be extracted from the query data, this can be done automatically using the methods described in the feature selection slide.</p>

<p>This method benefits from utilising a known database of cell type markers, these can be published and reused with different methods which allows for easy replication of results as well as allowing researchers to compare algorithms since the reference data remains constant.</p>

<p>This method has some downsides however, it can take a lot of prior manual labor and expertise in order to create an initial database of marker genes. The databases that are available can also limit which cell types can be annotated and may not contain the best marker genes.</p>

<hr />

<h3 id="supervised-classification">Supervised Classification</h3>

<p>.pull-left[</p>

<ul>
  <li>
    <p>Trains a statistical classifier on reference data to predict cell types</p>
  </li>
  <li>
    <p>Requires clean reference data for good quality models</p>
  </li>
  <li>
    <p>Is able to handle higher dimensional and more noisy data</p>
  </li>
</ul>

<p><img src="../../images/scrna-cell-annotation/svd-figure.png" alt="Diagram of a support vector machine" /></p>

<p>]</p>

<p>.pull-right[</p>

<p><img src="../../images/scrna-cell-annotation/random-forrest-figure.png" alt="Diagram of a random forrest classifier" /></p>

<p>]</p>

<p><span>Speaker Notes</span></p>

<p>Supervised classification trains a statistical model using some labelled reference data in order to learn a mapping between input data (gene expression values) and cell type categories.</p>

<p>There are many algorithms that can be used to train a classifier such as a Support Vector Machine, Artificial Neural Network, Random Forest Classifier, etc. However all these methods operate using some form of statistical analysis.</p>

<p>This method benefits from not requiring any human curated marker genes and is typically able to operate with more noisy data and can handle higher dimensionality data (which typically means being able to use more feature genes in the classification process).</p>

<p>However it should be noted that in order to train a classifier for identifying cell types a good reference dataset of labeled data is required. The size and accuracy of this data will affect the quality of the trained classifier, it can also be time consuming to train a accurate classifier for use in prediction. Additionally in order to use more reference data or add more classifications after training, you will need to either fine-tune or completely re-train the model using the additional data/classifications.</p>

<hr />

<p>Comparison of Methods</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Approach</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Advantages</th>
      <th style="text-align: center">Limitations</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">Correlation based</td>
      <td style="text-align: center">Compares gene expression patterns between a query dataset and a reference dataset, using measures such as Pearson or Spearman correlation coefficients</td>
      <td style="text-align: center">Comprehensive annotation, flexibility with multiple references and data merging. Can be applied at cell and cluster level. Simple and fast to compute.</td>
      <td style="text-align: center">Performance decreases with the number of features used. Potential bias from reference selection.</td>
    </tr>
    <tr>
      <td style="text-align: center">Cluster annotation with marker genes</td>
      <td style="text-align: center">Matches the expression patterns of specific marker genes to reference cell types</td>
      <td style="text-align: center">Use of marker gene databases provides a comprehensive collection of known cell type markers.</td>
      <td style="text-align: center">Uncertain annotation if query data is not clean. Relies on human annotated known cell type markers for annotation.</td>
    </tr>
    <tr>
      <td style="text-align: center">Supervised Classification</td>
      <td style="text-align: center">Machine learning approach that learns the relationship between genes and cell types.</td>
      <td style="text-align: center">Robust to data noise and batch effects, higher accuracy with appropriate training. Can handle high dimensional data.</td>
      <td style="text-align: center">Training step is required which can be computationally intensive. Need to re-train or finetune an existing model to add more reference data.</td>
    </tr>
  </tbody>
</table>

<p><span>Speaker Notes</span></p>

<p>Here is an overview table presenting the information covered in the previous slides.</p>

<p>This can be used as a quick reference for what methods are available, an overview of how they work, and some things to consider when deciding which method is right for you.</p>

<hr />

<h3 id="visualisation">Visualisation</h3>

<ul>
  <li>
    <p>Quick method to check the results of the annotation</p>
  </li>
  <li>
    <p>Can be used to debug any issues with the tool’s predictions</p>
  </li>
  <li>
    <p>Visualisation is available through many cell annotation tools</p>
  </li>
</ul>

<p><img src="../../images/scrna-cell-annotation/visualisation.png" alt="Figure showing multiple clusters with descriptions of how they have failed" /></p>

<p><span>Speaker Notes</span></p>

<p>Visualisation can be an important step in classifying cell types, after processing the query data and generating predicted cell type labels it can be good to validate the predictions.</p>

<p>The easiest way to validate the predicted results is to plot the data on a graph and visually inspect the clusters, quickly inspecting the graph can help identify any errors in the classification process and determine what could be causing the issue.</p>

<p>The most common method to produce visualisations is to reduce the dimensionality of the data using methods such as UMAP, tSNE, PCA and plot the output on a 2-dimensional graph, each cluster can then be labelled with the predicted cell types.</p>

<p>Many tools for automated cell annotation offer the ability to plot the predicted data.</p>

<hr />

<h3 id="common-issues">Common Issues</h3>

<ul>
  <li>Bad reference data
    <ul>
      <li>A lot of noise in the data that hasn’t been filtered</li>
      <li>Pre-annotated cell labels are not correct</li>
    </ul>
  </li>
  <li>Missing cell types
    <ul>
      <li>Tools can only classify cell types that are available in the reference data!</li>
    </ul>
  </li>
  <li>
    <p>Poor quality or missing markers</p>
  </li>
  <li>Some tools perform poorly on certain data</li>
</ul>

<p><span>Speaker Notes</span></p>

<p>When performing cell annotation with any tool you may encounter some issues where the annotations are largely wrong or unassigned, if this happens then there are some common things that you should check:</p>

<ul>
  <li>
    <p>Ensure the reference data is clean: if the reference data is not labelled correctly or contains a lot of noise then this will affect the annotation tool (the tool will only ever be as good as the reference data!). In order to resolve this you may want to filter out poor quality data and possibly visualise the annotated data to check that the labels look correct.</p>
  </li>
  <li>
    <p>Additionally it is important that the reference data matches the query data that you will be using, if the reference data used is from mouse cells and the queries are from human cells then the annotation tool will not contain the relevant knowledge to classify the cells.</p>
  </li>
  <li>
    <p>If the tool being used relies on gene marker data it is important that the markers are good quality.</p>
  </li>
  <li>
    <p>Different tools will perform better on certain data, it may be a good idea to do some research of the tool being used to ensure that it doesn’t perform poorly on the data you are using (if it does then using a different tool may be the best option!).</p>
  </li>
  <li>
    <p>Finally many tools will have adjustable parameters that you can change (such as how many features to use for the input) it may be a good idea to play around with these values to see if it produces better results.</p>
  </li>
</ul>


	<hr />


<h2>Key Points</h2>

<ul>

<li>Automated cell annotation is a useful tool for automatically assigning cell type labels to cell data.</li>

<li>There are various algorithms and methods that can be used for conducting cell annotation with each method having its own strengths and weaknesses.</li>

<li>Automated cell annotation isn't perfect, when analysing results it's important to identify common issues that could occur.</li>

</ul>







<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/single-cell/tutorials/scrna-case-cell-annotation/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703464120'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
