<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Essential genes detection with Transposon insertion sequencing</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703809381' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703809381' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Essential genes detection with Transposon insertion sequencing">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-10-18 14:23:35 +0000">
<meta name="DC.creator" content="Delphine Lariviere">
<meta name="DC.creator" content="Bérénice Batut">

        
        
        
        
        <meta name="description" content="Genome annotation is a multi-level process that includes ...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Essential genes detection with Transposon insertion seque...">
        <meta property="og:description" content="Genome annotation is a multi-level process that includes ...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2022-10-18 14:23:35 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Essential genes detection with Transposon insertion sequencing",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Genome Annotation",
    "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Essential genes detection with Transposon insertion sequencing' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html",
  "timeRequired": "PT7H",
  "teaches": "Understand the read structure of TnSeq analyses\n - Predict Essential genes with Transit\n - Compare gene essentiality in control sample and an different experimental condition",
  "keywords": "genome-annotation, bacteria, tnseq, essential genes",
  "description": "The questions this  addresses are:\n - What is Transposon insertion Sequencing?\n - How to get TA sites Coverage ? \n - How to predict essential genes ?\n\n\nThe objectives are:\n - Understand the read structure of TnSeq analyses\n - Predict Essential genes with Transit\n - Compare gene essentiality in control sample and an different experimental condition\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/delphine-l/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/delphine-l/",
      "name": "Delphine Lariviere",
      "image": "https://avatars.githubusercontent.com/delphine-l",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6421-3484",
      "orcid": "https://orcid.org/0000-0001-6421-3484"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Genome Annotation",
      "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0362",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0362",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0362"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "bacteria"
    },
    {
      "@type": "Thing",
      "name": "tnseq"
    },
    {
      "@type": "Thing",
      "name": "essential genes"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703809381'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/genome-annotation" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Genome Annotation
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/genome-annotation/faqs/" title="Check our FAQs for the Genome Annotation topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/genome-annotation/tutorials/tnseq/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-genome-annotation">
    <h1 data-toc-skip>Essential genes detection with Transposon insertion sequencing</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/delphine-l/" class="contributor-badge contributor-delphine-l"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/delphine-l?s=36" alt="Avatar" width="36" height="36">Delphine Lariviere</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Avatar" width="36" height="36">Bérénice Batut</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What is Transposon insertion Sequencing?</p>
</li>
        
        <li><p>How to get TA sites Coverage ?</p>
</li>
        
        <li><p>How to predict essential genes ?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Understand the read structure of TnSeq analyses</p>
</li>
        
        <li><p>Predict Essential genes with Transit</p>
</li>
        
        <li><p>Compare gene essentiality in control sample and an different experimental condition</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 7 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/genome-annotation/tutorials/tnseq/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.2579335">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/genome-annotation/tutorials/tnseq/workflows/" title="Essential genes detection with Transposon insertion sequencing workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/genome-annotation/tutorials/tnseq/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Oct 18, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00179">https://gxy.io/GTN:T00179</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>In microbiology, identifying links between genotype and phenotype is key to understand bacteria growth and virulence mechanisms, and to identify targets for drugs and vaccines. These analysis are limited by the lack of bacterial genome annotations (<em>e.g.</em> 30% of genes for <em>S. pneumoniae</em> are of unknown function) and by the fact that genotypes often arose from complex composant interactions.</p>

<h2 id="transposon-insertion-sequencing">Transposon insertion Sequencing</h2>

<p>Transposon insertion sequencing is a technique used to functionally annotate bacterial genomes. In this technique, the genome is saturated by insertions of transposons. Transposons are highly regulated, discrete DNA segments that can relocate within the genome. They have a large influence on gene expression and can be used to determine the function of genes.</p>

<p>When a transposon inserts itself in a gene, the gene’s function will be disrupted, affecting the fitness (growth) of the bacteria. We can then manipulate transposons for use in insertional mutagenesis, <em>i.e.</em> creation of mutations of DNA by the addition of transposons. The genomes can be then sequenced to locate the transposon insertion site and the function affected by a transposon insertion can be linked to the disrupted gene.</p>

<p><a href="../../images/tnseq/principle_tnseq.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/principle_tnseq.png" alt="Illustration of tnseq Method. " width="1407" height="1090" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Transposon insertion sequencing method (from <a href="#Chao2016"> Chao <i>et al.</i> 2016</a>)&lt;/figcaption&gt;</figure></a></p>

<ol>
  <li>
    <p><strong>Data production</strong> (<strong>a</strong> in the previous image)</p>

    <p>An initial population of genomes is mutated so that the genome of the bacteria is saturated with transposon insertions. A library is called <em>saturated</em> if in the genomes across the whole population of bacteria, each potential insertion site has at least one insertion. The population is then divided into several media containing different growth conditions, to identify the impact of the insertions on the bacteria growth. After growth, the regions flanking the insertion are amplified and sequenced, allowing to determine the location of the insertion.</p>
  </li>
  <li>
    <p><strong>Analysis</strong> (<strong>b</strong> in the previous image)</p>

    <p>The sequences are aligned to the reference genome to identify the location of the regions flanking the insertions. The resulting data will show a discrete repartition of reads on each site. If a gene present several insertions, like the two leftmost genes in <em>Condition A</em>, it means that its disruption has little or no impact to the bacterial growth. On the other hand, when a gene shows no insertions at all, like the rightmost gene in <em>Condition A</em>, is means that any disruption in this gene killed the bacteria, meaning it is a gene essential to bacteria survival. If the library is sufficiently saturated, there is a clear threshold between essential and non-essential genes when you analyze the insertion rate per gene.</p>
  </li>
</ol>

<p>Two type of transposon insertion methods exist:</p>
<ul>
  <li>Gene disruption, where we analyze only the disruptions. This will covered by of this tutorial</li>
  <li>Regulatory element insertion, where different promoters are inserted by the transposon, and we analyze the change in gene expression in addition to the disruption. This method will be the subject of another tutorial.</li>
</ul>

<h2 id="building-a-tnseq-library">Building a TnSeq library</h2>

<p>Different types of transposons can be used depending of the goal of your analysis.</p>

<ul>
  <li>Randomly pooled transposon
    <ul>
      <li>
        <p>Mariner-based transposons, common and stable transposons which target the “TA” dinucleotides</p>

        <p>The TA are distributed relatively evenly along genome. The Mariner-based transposons can be inserted to impact statistically every gene, with in average more than 30 insertions site per kb. With the low insertion bias, it is easy to build saturated libraries. But local variations means less loci and less statistical power.</p>
      </li>
      <li>
        <p>Tn5-based vectors, which insert at random sites</p>

        <p>This transposon do not require any target sequences. It is useful for specie where it is difficult to build mariner based transposons. But it has a preference for high GC content, causing insertion bias</p>
      </li>
    </ul>
  </li>
  <li>
    <p>Defined sequence transposon</p>

    <p>It can be used to study interactions in pathways of interest, but also more precise targeting (small genes, pathways) for specific analyses</p>
  </li>
</ul>

<p>Independently of the transposon choice we need to be careful about the library complexity. With a large complex library, multiple insertion can be found in every potential locus. The higher density of insertion, the greater precision in identifying limits of regions of interest. If the density of the library is too low, some genes might by chance not be disrupted and mistaken for essential. The advantage of a target specific transposon, like the mariner, in opposition of a Tn5-based transposon inserting randomly, it that the limited number of insertion sites makes it easier to build high complexity libraries.</p>

<p>After selection of the type of transposon, we need to modify it to allow insertion site amplification and sequencing to get a library fitting the transposon insertion. Biases could be introduced during the process due to uneven fragment sizes. To avoid that, we can introduce a Type I restriction site to cleave DNA downstream of transposon, and get uniform fragment sizes and therefore avoid a bias in the representation of the insertions.</p>

<p>As we just want to identify the TA site affected by an insertion, we only need the location of the start of the reads and not a good coverage of the entire genome. Long reads are then not so important. On the other hand, a minimum transposon length of 16 bp is necessary for precise mapping on the genome <span class="citation"><a href="#Kwon2015">Kwon <i>et al.</i> 2015</a></span>. We can therefore not use the BsmFI restriction site (11 to 12 bp) but MmeI.</p>

<p><strong>In this tutorial, we are using mariner transposon targeting TA sequences, in ordered to target the whole genome uniformly,</strong> with two specific regions used to specifically sequence the region upstream of the insertion <span class="citation"><a href="#Santiago2015">Santiago <i>et al.</i> 2015</a></span></p>

<p><a href="../../images/tnseq/tranposon_structure.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/tranposon_structure.png" alt="Structure of the transposon containing several parcodes and adapters. " width="1284" height="1045" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Structure of the transposon containing several parcodes and adapters (from <a href="#Santiago2015"> Santiago <i>et al.</i> 2015</a>)&lt;/figcaption&gt;</figure></a></p>

<p>The transposon inserts itself at TA site at the ITR junctions. These ITR junctions have been modified to include a Mme1 restriction site and a NotI restriction site which cut 21 bp upstream the restriction site. These two site are the 5’ and 3’ limits to the genomic DNA we want to sequence.</p>

<ol>
  <li>
    <p>After digestion by NotI restriction enzyme, the fragments are attached to biotinylated adaptors that link to NotI restriction site. The attached fragment are then digested by MMeI at a site upstream, where an Illumina primer is then linked. The sequencing is then done, adding Illumina adaptors and an additional barcode to the read for multiplexed sequencing.</p>
  </li>
  <li>
    <p>An insertion can sometimes be composed of one or more copies of the transposon (multimer). There is therefore a risk to select plasmid backbone sequence. To solve this problem, an additional NotI has been added in the backbone to create different length construct, that can later be filtrated. Different promoters are added with an additional 3 bp barcode to analyze differential expression impact. This type of complex analysis will be covered in a follow-up tutorial.</p>
  </li>
</ol>

<p>Because of this complex transposon structure, the reads obtained after sequencing contain a lot of adapters and foreign sequences used to insert and target the transposon. Several steps of preprocessing are then need to extract only the transposon sequence before finding its location on the genome.</p>

<h2 id="tnseq-analysis">Tnseq analysis</h2>

<p>Once the genomic sequences are extracted from the initial reads (<em>i.e.</em> remove non genomic sequences from the reads), they need to be located each on the genome to link them to a TA site and genes. To do that we map them to a reference genome, link them to a specific insertion site, and then count the number of insertion for each TA site and identify essential genes of regions.</p>

<p>We will apply this approach in this tutorial using a subset of TnSeq reads from <span class="citation"><a href="#Santiago2015">Santiago <i>et al.</i> 2015</a></span>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a>    <ol>
      <li><a href="#transposon-insertion-sequencing" id="markdown-toc-transposon-insertion-sequencing">Transposon insertion Sequencing</a></li>
      <li><a href="#building-a-tnseq-library" id="markdown-toc-building-a-tnseq-library">Building a TnSeq library</a></li>
      <li><a href="#tnseq-analysis" id="markdown-toc-tnseq-analysis">Tnseq analysis</a></li>
    </ol>
  </li>
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#remove-all-non-genomic-sequences-from-the-sequenced-reads" id="markdown-toc-remove-all-non-genomic-sequences-from-the-sequenced-reads">Remove all non genomic sequences from the sequenced reads</a>    <ol>
      <li><a href="#separate-reads-by-experimental-conditions" id="markdown-toc-separate-reads-by-experimental-conditions">Separate reads by experimental conditions</a></li>
      <li><a href="#remove-adapter-sequence" id="markdown-toc-remove-adapter-sequence">Remove Adapter sequence</a></li>
      <li><a href="#separate-reads-from-different-transposon-constructs" id="markdown-toc-separate-reads-from-different-transposon-constructs">Separate reads from different transposon constructs</a></li>
      <li><a href="#remove-the-remaining-transposon-sequence" id="markdown-toc-remove-the-remaining-transposon-sequence">Remove the remaining transposon sequence</a></li>
    </ol>
  </li>
  <li><a href="#count-the-number-of-insertion-per-ta-sites" id="markdown-toc-count-the-number-of-insertion-per-ta-sites">Count the number of insertion per TA sites</a>    <ol>
      <li><a href="#align-the-reads-to-a-reference-genome" id="markdown-toc-align-the-reads-to-a-reference-genome">Align the reads to a reference genome</a></li>
      <li><a href="#compute-coverage-of-the-genome" id="markdown-toc-compute-coverage-of-the-genome">Compute coverage of the genome</a></li>
      <li><a href="#identify-ta-sites-positions" id="markdown-toc-identify-ta-sites-positions">Identify TA sites positions</a></li>
      <li><a href="#merge-overall-coverage-and-positions-of-ta-sites-to-get-the-coverage-of-each-ta-sites" id="markdown-toc-merge-overall-coverage-and-positions-of-ta-sites-to-get-the-coverage-of-each-ta-sites">Merge overall coverage and positions of TA sites to get the coverage of each TA sites</a></li>
    </ol>
  </li>
  <li><a href="#predicting-essential-genes-with-transit" id="markdown-toc-predicting-essential-genes-with-transit">Predicting Essential Genes with Transit</a>    <ol>
      <li><a href="#transit" id="markdown-toc-transit">Transit</a></li>
      <li><a href="#predict-the-essentiality-of-genes" id="markdown-toc-predict-the-essentiality-of-genes">Predict the essentiality of genes</a></li>
      <li><a href="#compare-the-essential-genes-between-two-conditions" id="markdown-toc-compare-the-essential-genes-between-two-conditions">Compare the essential genes between two conditions</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<p>Let’s start with uploading the data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import the data</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Import from <a href="https://zenodo.org/record/2579335">Zenodo</a> or a data library (ask your instructor):
      <ul>
        <li>FASTQ file with the Tnseq reads: <code class="language-plaintext highlighter-rouge">Tnseq-Tutorial-reads.fastqsanger.gz</code></li>
        <li>Set of barcodes to separate reads from different experimental conditions: <code class="language-plaintext highlighter-rouge">condition_barcodes.fasta</code></li>
        <li>Set of barcodes to separate reads from different transposon constructs: <code class="language-plaintext highlighter-rouge">construct_barcodes.fasta</code></li>
        <li>Genome file for <em>Staphylococcus aureus</em>: <code class="language-plaintext highlighter-rouge">staph_aur.fasta</code></li>
        <li>Annotation file for <em>Staphylococcus aureus</em>: <code class="language-plaintext highlighter-rouge">staph_aur.gff3</code></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2579335/files/Tnseq-Tutorial-reads.fastqsanger.gz
https://zenodo.org/record/2579335/files/condition_barcodes.fasta
https://zenodo.org/record/2579335/files/construct_barcodes.fasta
https://zenodo.org/record/2579335/files/staph_aur.fasta
https://zenodo.org/record/2579335/files/staph_aur.gff3
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the files</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="remove-all-non-genomic-sequences-from-the-sequenced-reads">Remove all non genomic sequences from the sequenced reads</h1>

<p>Because of the experimental design for transposon insertion sequencing, the raw reads contain a lot of adapters and foreign sequences used to insert and target the transposon. To obtain the core reads that contain only genomic sequence, the reads have to go through several steps (using the <strong>Cutadapt</strong> tool) to remove them and divide the reads per experimental condition and type of transposon:</p>

<p><a href="../../images/tnseq/preprocessing.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/preprocessing.png" alt="Workflow of data pre-processing. " width="920" height="424" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Data pre-processing&lt;/figcaption&gt;</figure></a></p>

<ol>
  <li>We separate the reads of each experimental condition based on a 8 bp barcode at the beginning of each read. These barcodes were added to be able to pool different conditions together before the transposon insertion and sequencing.</li>
  <li>The tail of each set of read is then removed. It immediately follows the 3 bp barcode specific to transposon constructs, and contains illumina adapter sequence and downstream</li>
  <li>To be sure all our reads have been trimmed correctly we filter out the reads too large.</li>
  <li>We then separate the reads per transposon construct</li>
  <li>We remove the remaining transposon sequence containing MmeI.</li>
</ol>

<h2 id="separate-reads-by-experimental-conditions">Separate reads by experimental conditions</h2>

<p>First we divide the initial data set by experimental conditions using the 8 bp barcode added during the Illumina multiplexing protocol.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-condition-barcodes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect condition barcodes</div>

  <ol>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">condition_barcodes</code> file</li>
  </ol>

</blockquote>

<p>This fasta file contains the barcodes for each condition:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;control
^CTCAGAAG
&gt;condition
^GACGTCAT
</code></pre></div></div>

<p>We can see 2 barcodes there: one for control and one for condition.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-barcode-symbols-used-by-b-cutadapt-b"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Barcode symbols used by <b>Cutadapt</b></div>

  <p>The <code class="language-plaintext highlighter-rouge">^</code> at the beginning of the sequence means we want to anchor the barcode at the beginning of the read. To know more about the symbols used by <strong>Cutadapt</strong>, you check the <a href="https://cutadapt.readthedocs.io/en/stable/guide.html#adapter-types"><strong>Cutadapt</strong> manual</a>.</p>
</blockquote>

<p>We would like now to split our Tnseq reads in <code class="language-plaintext highlighter-rouge">Tnseq-Tutorial-reads</code> given the barcodes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-split-reads-by-condition-barcodes-using-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Split reads by condition barcodes using Cutadapt</div>

  <ol>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ/A file”</em>: <code class="language-plaintext highlighter-rouge">Tnseq-Tutorial-reads</code></li>
            <li>In <em>“Read 1 Options”</em>
              <ul>
                <li>In <em>“5’ (Front) Adapters”</em>
                  <ul>
                    <li>Click on <em>“Insert 5’ (Front) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">File From History</code>
                          <ul>
                            <li><em>“Choose file containing 5’ adapters”</em>: <code class="language-plaintext highlighter-rouge">condition_barcodes</code> file</li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Adapter Options”</em>
          <ul>
            <li><em>“Maximum error rate”</em>: <code class="language-plaintext highlighter-rouge">0.15</code> (to allow 1 mismatch)</li>
            <li><em>“Match times”</em>: <code class="language-plaintext highlighter-rouge">3</code> (to cover cases where barcodes are attached several times)</li>
          </ul>
        </li>
        <li>In <em>“Output Selector”</em>, select
          <ul>
            <li><em>“Report”</em></li>
            <li><em>“Multiple output”</em> (to separate the reads into one file per condition)</li>
            <li><em>“Untrimmed reads”</em> (to write reads that do not contain the adapter to a separate file)</li>
          </ul>
        </li>
      </ul>

      <p>The output is a collection of the different conditions datasets, here control and condition, and a report text file.</p>
    </li>
    <li>Inspect the report text generated by <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What percentage of reads has been trimmed for the adapter?</li>
    <li>How many reads have been trimmed for each condition?</li>
    <li>What should we change if our barcodes were at the end of the reads ?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>100% of the reads were trimmed (line <code class="language-plaintext highlighter-rouge">Reads with adapters</code>)</li>
      <li>2,847,018 for control (<code class="language-plaintext highlighter-rouge">CTCAGAAG</code> barcode) and 2,862,108 for condition (<code class="language-plaintext highlighter-rouge">GACGTCAT</code> barcode)</li>
      <li>We should use the option <em>“Insert 3’ (End) Adapters”</em> and anchored them at the end of the read with the symbol <code class="language-plaintext highlighter-rouge">$</code> in our fasta file containing the barcodes.</li>
    </ol>
  </blockquote>

</blockquote>

<h2 id="remove-adapter-sequence">Remove Adapter sequence</h2>

<p>Our reads are now divided by condition. We need to trim their tail containing the Illumina adapters, used for initiating the sequencing (<code class="language-plaintext highlighter-rouge">CGTTATGGCACGC</code>, here). To do so, we remove the adapter and everything downstream, using the end adapter option of <strong>Cutadapt</strong> and not anchor the sequence anywhere. To eliminate reads that might not have been trimmed because of too many mismatches or other reasons, we filter the reads by size, given the known approximate size of the remaining sequences (<strong>how is it computed??</strong>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-adapter-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove Adapter with Cutadapt</div>

  <ol>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to remove adapters with:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: collection output of the previous <strong>Cutadapt</strong></li>
            <li>In <em>“Read 1 Options”</em>
              <ul>
                <li>In <em>“3’ (End) Adapters”</em>
                  <ul>
                    <li>Click on <em>“Insert 3’ (End) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom Sequence</code>
                          <ul>
                            <li><em>“Enter custom 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">CGTTATGGCACGC</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Adapter Options”</em>
          <ul>
            <li><em>“Match times”</em>: <code class="language-plaintext highlighter-rouge">3</code> (to cover cases where barcodes are attached several times)</li>
          </ul>
        </li>
        <li>In <em>“Output Selector”</em>, select
          <ul>
            <li><em>“Report”</em></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>What are the outputs at this step?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>The outputs are two collections: one containing the reads in both conditions, and one containing the <strong>Cutadapt</strong> reports for each condition.</p>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Inspect the generated report files</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>What percentage of the reads contained the adapter?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>More than 99% of the reads contained the adapter in both conditions (line <code class="language-plaintext highlighter-rouge">Reads with adapters</code> in the reports)</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to filter reads based on length with:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: collection output of the previous <strong>Cutadapt</strong></li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>
          <ul>
            <li><em>“Minimum length”</em> : <code class="language-plaintext highlighter-rouge">64</code></li>
            <li><em>“Maximum length”</em> : <code class="language-plaintext highlighter-rouge">70</code></li>
          </ul>
        </li>
        <li>In <em>“Output Selector”</em>, select
          <ul>
            <li><em>“Report”</em></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated report files</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>How many reads where discarded after filtering?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>Less than 2% of the reads were discarded in both conditions (lines <code class="language-plaintext highlighter-rouge">Reads that were too short</code> and <code class="language-plaintext highlighter-rouge">Reads that were too long</code> in the reports)</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We can see that is both samples the reads have pass the filtering at more than 98%. If this percentage is very low, it means the previous trimming steps is incomplete or faulty.</p>

<h2 id="separate-reads-from-different-transposon-constructs">Separate reads from different transposon constructs</h2>

<p>The constructs used in this experiment contain different strengths and directions of promoters. We use the different constructs as replicates, so we need now to separate the reads based on the construct specific 3 bp barcodes.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>In addition of disrupting a gene at the location of the insertion, such constructs can modify the expression of either upstream or downstream regions. The analysis of such modification will be studied in another training material, but for now we consider that the construct does not impact the essentiality analysis.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-construct-barcodes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect construct barcodes</div>

  <ol>
    <li>
      <p>Inspect the <code class="language-plaintext highlighter-rouge">construct_barcodes</code> file</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>What does the <code class="language-plaintext highlighter-rouge">$</code> mean in the barcode sequence file?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>It means the barcode is anchored at the end of the reads.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-barcode-split-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Barcode split with Cutadapt</div>

  <ol>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: collection output of the previous <strong>Cutadapt</strong></li>
            <li>In <em>“Read 1 Options”</em>
              <ul>
                <li>In <em>“3’ (End) Adapters”</em>
                  <ul>
                    <li>Click on <em>“Insert 3’ (End) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em> : <code class="language-plaintext highlighter-rouge">File From History</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Choose file containing 5’ adapters”</em>: <code class="language-plaintext highlighter-rouge">construct_barcodes</code> file</li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Adapter Options”</em>
          <ul>
            <li><em>“Match times”</em>: <code class="language-plaintext highlighter-rouge">3</code> (to cover cases where barcodes are attached several times)</li>
          </ul>
        </li>
        <li>In <em>“Output Selector”</em>, select
          <ul>
            <li><em>“Report”</em></li>
            <li><em>“Multiple output”</em> (to separate the reads into one file per condition)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the report files</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Are the reads equally divided between constructs?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>When you look at the reports, you can see that most of the reads have been assigned to the <em>blunt</em> construct (<code class="language-plaintext highlighter-rouge">TAC</code> barcode): the blunt construct is the control and does not contain any promoters. This means that there is less negative selective pressure on blunt than the other ones, that have affected flanking region in addition to the disrupted gene at the insertion site. This won’t be a problem here as the tool we use for the essentiality prediction consider the sum of reads in the replicates.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>You can notice that the output of this split is a <em>nested collection</em>, a collection of collection.</p>

<h2 id="remove-the-remaining-transposon-sequence">Remove the remaining transposon sequence</h2>

<p>The last remaining transposon sequence in the reads is the linker with the MmeI restriction site (<code class="language-plaintext highlighter-rouge">ACAGGTTGGATGATAAGTCCCCGGTCTATATTGAGAGTAACTACATTT</code>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-linker-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Remove Linker with Cutadapt</div>

  <ol>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: collection output of the previous <strong>Cutadapt</strong></li>
            <li>In <em>“Read 1 Options”</em>
              <ul>
                <li>In <em>“3’ (End) Adapters”</em>
                  <ul>
                    <li>Click on <em>“Insert 3’ (End) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom Sequence</code>
                          <ul>
                            <li><em>“Enter custom 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">ACAGGTTGGATGATAAGTCCCCGGTCTATATTGAGAGTAACTACATTT</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Adapter Options”</em>
          <ul>
            <li><em>“Maximum error rate”</em>: <code class="language-plaintext highlighter-rouge">0.15</code></li>
          </ul>
        </li>
        <li>In <em>“Output Selector”</em>, select
          <ul>
            <li><em>“Report”</em></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the report files and check that the majority of the read have been trimmed</li>
  </ol>
</blockquote>

<p>Now that we isolated the genomic sequences from the initial reads, we want to align them to count how many insertion have been retained at each TA site.</p>

<h1 id="count-the-number-of-insertion-per-ta-sites">Count the number of insertion per TA sites</h1>

<h2 id="align-the-reads-to-a-reference-genome">Align the reads to a reference genome</h2>

<p>To identify the location of each TA site to the count them, the first step is to map the reads on the reference genome. We use the <strong>Bowtie</strong>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>We could also use <strong>Bowtie2</strong> with an end-to-end option, but <strong>Bowtie</strong> is more suitable for very short reads like ours (16-17 bp).</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-map-reads-with-bowtie"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Map reads with Bowtie</div>

  <ol>
    <li><strong>Map with Bowtie for Illumina</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use one from the history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the reference genome”</em>: <code class="language-plaintext highlighter-rouge">staph_aur.fasta</code> file.</li>
          </ul>
        </li>
        <li><em>“Is this library mate-paired?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ file”</em>: collection output of the last <strong>Cutadapt</strong></li>
            <li><em>“Bowtie settings to use”</em>: <code class="language-plaintext highlighter-rouge">Full parameters list</code></li>
            <li><em>“Skip the first n reads (-s)”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
            <li><em>“Maximum number of mismatches permitted in the seed (-n)”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
            <li><em>“Seed length (-l)”</em>: <code class="language-plaintext highlighter-rouge">17</code></li>
            <li><em>“Whether or not to make Bowtie guarantee that reported singleton alignments are ‘best’ in terms of stratum and in terms of the quality values at the mismatched positions (–best)”</em>: <code class="language-plaintext highlighter-rouge">Use best</code></li>
            <li><em>“Whether or not to try as hard as possible to find valid alignments when they exist (-y)”</em> : <code class="language-plaintext highlighter-rouge">Try Hard</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>Why are we strictly enforcing no mismatch mapping?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>Our reads being very short, the smallest size needs precise mapping, allowing even one mismatch would risk having reads mapping in wrong positions.</p>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Rename your collection for better clarity</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection</li>   <li>Click on the name of the collection at the top</li>   <li>Change the name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h2 id="compute-coverage-of-the-genome">Compute coverage of the genome</h2>

<p>We have now our reads mapped on the reference genome. We can compute the genome coverage, <em>i.e.</em> how much the nucleotides of the genome are covered by reads. This information will be later crossed with our TA sites position.</p>

<p>In our case, the sequenced reads cover the  5’ region flanking region of the TA site where the transposon inserted. We should not then extract the coverage across the whole reads, as it could cover several TA sites, but only the coverage at the 3’ end of the read:</p>

<p><a href="../../images/tnseq/Map_cov.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/Map_cov.png" alt="A read align to the genome with its 3' end covering half of the TA site. " width="935" height="222" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Mapping read and TA site coverage&lt;/figcaption&gt;</figure></a></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-genome-coverage"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Compute genome coverage</div>

  <ol>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM/CRAM file”</em>: collection output of the last <strong>Map with Bowtie for Illumina</strong></li>
        <li><em>“Bin size in bases”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Scaling/Normalization method”</em>: <code class="language-plaintext highlighter-rouge">Do not normalize or scale</code></li>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bedgraph</code></li>
        <li><em>“Show advanced options”</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>“Ignore missing data?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (to get only region with read counts)</li>
            <li><em>“Offset inside each alignment to use for the signal location”</em>: <code class="language-plaintext highlighter-rouge">-1</code> (to read the signal of the coverage only at the 3’ end of the read)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="identify-ta-sites-positions">Identify TA sites positions</h2>

<p>In order to get the coverage on each TA site we need to prepare a file containing the position of each TA site.</p>

<p>As you can see on the <a href="#figure-4">previous figure</a>, the read can cover one side or the other of the TA depending on the direction of insertion of the transposon. Depending on the direction of insertion the coverage will be counted on the leftmost position of the TA site or on the rightmost. As we are not considering strand separately in this analyses, we will consider both count as attached to the leftmost base of the TA site. To do that we will create two list of TA site positions, listing the 5’ end of each TA site for forward and reverse strand, and then merge them to get a global count per TA site.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-ta-sites-coordinates"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get TA sites coordinates</div>

  <ol>
    <li><strong>Nucleotide subsequence search</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Nucleotide sequence to be searched”</em>: <code class="language-plaintext highlighter-rouge">staph_aur.fasta</code></li>
        <li><em>“Search with a”</em>: <code class="language-plaintext highlighter-rouge">user defined pattern</code></li>
        <li><em>“Search pattern”</em>: <code class="language-plaintext highlighter-rouge">TA</code></li>
        <li><em>“Search pattern on”</em>: <code class="language-plaintext highlighter-rouge">forward strand</code></li>
      </ul>
    </li>
    <li>Inspect the generated file</li>
  </ol>
</blockquote>

<p><strong>Nucleotide subsequence search</strong> generate one BED file with the coordinates of each TA on the genome:</p>
<ol>
  <li>Chrom</li>
  <li>Start</li>
  <li>End</li>
  <li>Name</li>
  <li>Strand</li>
</ol>

<p>The only information we need here are the positions of the 5’ end of each TA site for each strand.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-forward-and-reverse-strand-positions"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get forward and reverse strand positions</div>

  <ol>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li>Cut columns : <code class="language-plaintext highlighter-rouge">c2</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of <strong>Nucleotide subsequence search</strong></li>
      </ul>
    </li>
    <li>
      <p>Add the <code class="language-plaintext highlighter-rouge">#forward</code> tag to the output</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Adding a tag<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</li>     </ul>   </li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li>Cut columns : <code class="language-plaintext highlighter-rouge">c1,c3</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of <strong>Nucleotide subsequence search</strong></li>
      </ul>
    </li>
    <li><strong>Compute an expression on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to shift the end position of TA sites  by 1 and get the start of the TA site on the reverse strand with:
      <ul>
        <li><em>Add expression”</em>: <code class="language-plaintext highlighter-rouge">c2-1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“as a new column to”</em>: output of <strong>cut</strong></li>
        <li><em>“Round result?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li>Cut columns : <code class="language-plaintext highlighter-rouge">c3</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of <strong>Compute</strong></li>
      </ul>
    </li>
    <li>Add the <code class="language-plaintext highlighter-rouge">#reverse</code> tag to the output</li>
  </ol>

</blockquote>

<h2 id="merge-overall-coverage-and-positions-of-ta-sites-to-get-the-coverage-of-each-ta-sites">Merge overall coverage and positions of TA sites to get the coverage of each TA sites</h2>

<p>We have now 2 files containing the coordinates of our TA sites for both strands. We should cross them with the coverage files to get the coverage on both sides of each TA site.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-coverage-of-ta-sites"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Get coverage of TA sites</div>

  <ol>
    <li><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>1st file”</em>: output of <strong>bamCoverage</strong></li>
        <li><em>“Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column 2</code> (to select the position of the end of the read)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“2nd File”</em>: output of last <strong>cut</strong> with the <code class="language-plaintext highlighter-rouge">forward</code> tags</li>
        <li><em>“Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>“Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable lines from 2st file. (-a 2)</code> (to get all TA sites)</li>
        <li><em>“First line is a header line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Value to put in unpaired (empty) fields”</em>: <code class="language-plaintext highlighter-rouge">0</code> (to assign a count of 0 to TA sites not covered in the bamCoverage file)</li>
      </ul>
    </li>
    <li>
      <p>Add <code class="language-plaintext highlighter-rouge">#forward</code> tag to the output collections</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-to-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-to-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Adding a tag to a collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection in your history to view it</li>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name at the top of the history panel</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs any tools using this dataset.</li>     </ul>   </li>   <li>Click <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span></li>   <li>Check that the tag appears below the collection name</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li>Cut columns : <code class="language-plaintext highlighter-rouge">c1,c4</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of<strong>Join two files</strong> with the <code class="language-plaintext highlighter-rouge">forward</code> tag</li>
      </ul>
    </li>
    <li>Repeat the same steps for <code class="language-plaintext highlighter-rouge">reverse</code></li>
  </ol>

</blockquote>

<p>We now have a read count for each nucleotide at the different TA sites. The insertions on the forward strand files are in a different direction than those on the reverse strand file. In our case, we are only interested in studying the gene disruption, and therefore we do not want to consider the directions separately. We can then combine the forward and reverse counts together to get a total count per TA site. In order to do that we need to assign the count of both strand to the same nucleotide, by shifting by one position the count on the reverse strand.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-total-count-per-ta-site"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get total count per TA site</div>

  <ol>
    <li><strong>Compute an expression on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to shift the positions of reverse strand counts with:
      <ul>
        <li><em>Add expression”</em>: <code class="language-plaintext highlighter-rouge">c1-1</code> (to shift the position by 1)</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“as a new column to”</em>: <strong>Cut</strong> output collection with the tags <code class="language-plaintext highlighter-rouge">reverse</code></li>
        <li><em>“Round result?”</em> : <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c3,c2</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>File to cut”</em>: output of the previous <strong>Compute</strong></li>
      </ul>
    </li>
    <li><strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sort Query”</em>: the newest <code class="language-plaintext highlighter-rouge">reverse</code> collection</li>
        <li><em>“Number of header lines”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li>In <em>“Column selections”</em>
          <ul>
            <li><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code> (to sort on positions)</li>
            <li><em>“in”</em>: <code class="language-plaintext highlighter-rouge">Ascending Order</code></li>
            <li><em>“Flavor”</em>: <code class="language-plaintext highlighter-rouge">Fast numeric sort (-n)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat the <strong>Sort</strong> with the newest <code class="language-plaintext highlighter-rouge">forward</code> collection</p>
    </li>
    <li><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to merge the two collections and get the forward and reverse count in two different columns with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>1st file”</em> : the <code class="language-plaintext highlighter-rouge">forward</code> collection from <strong>Sort</strong></li>
        <li><em>“Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column 1</code> to join on positions</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“2nd File”</em> : the <code class="language-plaintext highlighter-rouge">reverse</code> collection from <strong>Sort</strong></li>
        <li><em>“Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>“Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file</code> (to get all TA sites)</li>
        <li><em>“First line is a header line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Value to put in unpaired (empty) fields”</em>: <code class="language-plaintext highlighter-rouge">.</code></li>
      </ul>
    </li>
    <li><strong>Compute an expression on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to get the total read count with:
      <ul>
        <li><em>Add expression”</em>: <code class="language-plaintext highlighter-rouge">c2+c3</code> to get the total count</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“as a new column to”</em>: output of <strong>Join two files</strong> with the tags <code class="language-plaintext highlighter-rouge">reverse</code> and <code class="language-plaintext highlighter-rouge">forward</code></li>
        <li><em>“Round result?”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
    </li>
    <li><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c4</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>File to cut”</em>:output of the last <strong>Compute</strong></li>
      </ul>
    </li>
    <li><strong>Sort data in ascending or descending order</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Sort Query”</em>: output of the last <strong>Cut</strong></li>
        <li><em>“Number of header lines”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li>In <em>“Column selections”</em>
          <ul>
            <li><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code> (to sort on positions)</li>
            <li><em>“in”</em>: <code class="language-plaintext highlighter-rouge">Ascending Order</code></li>
            <li><em>“Flavor”</em>: <code class="language-plaintext highlighter-rouge">Fast numeric sort (-n)</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Take a look at the output files:</p>

  <ol>
    <li>What are the data contained in the two columns?</li>
    <li>What do the 5 first lines of the blunt sample in the treatment with the condition look like?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The first column contain the position of the TA site, and the second column the number of insertions observed.</li>
      <li>Your five first lines should look like this :
        <ul>
          <li>One column with the positions : <code class="language-plaintext highlighter-rouge">4,10,16,42,79</code></li>
          <li>One column with the number of insertions : <code class="language-plaintext highlighter-rouge">0,3,20,0,11</code></li>
        </ul>
      </li>
    </ol>

    <p>If your counts are only zeros in one of the blunt sample, you might have made a mistake in merging the counts to the TA sites.</p>

  </blockquote>
</blockquote>

<h1 id="predicting-essential-genes-with-transit">Predicting Essential Genes with Transit</h1>

<p>Now that we have the counts of insertions per TA site, we can use them to predict gene essentiality. Several methods exist to identify essential genes of regions. They can be divided in two major categories:</p>

<p><a href="../../images/tnseq/type_of_analyses.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/type_of_analyses.png" alt="Different types of TnSeq Analyses. " width="867" height="711" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Methods of TnSeq Analyses (from <a href="#Chao2016"> Chao <i>et al.</i> 2016</a>)&lt;/figcaption&gt;</figure></a></p>

<ol>
  <li>
    <p><strong>Annotation dependent method</strong></p>

    <p>The total read count an/or percentage of disrupted site are computed per annotated regions. The values are then compared to the rest of the genome to classify the genes into the categories <em>essential</em> or <em>non-essential</em>.</p>
  </li>
  <li>
    <p><strong>Annotation independent method</strong></p>

    <p>The total read count and/or disrupted sites are computed independently of annotated regions. One of these methods is using a sliding window. Each window is then classified into the categories <em>essential</em> or <em>non-essential</em>. After the windows have been classified, they are linked annotations, and the genes/regions can be classified as <em>essential</em>, <em>non-essential</em>, or <em>domain essential</em> according to the classification of the windows they cover. The same classification can be done using HMM based methods instead of sliding windows. In that case, each insertion site will be predicted as <em>essential</em> or <em>non essential</em> <span class="citation"><a href="#Chao2016">Chao <i>et al.</i> 2016</a></span>).</p>
  </li>
</ol>

<p>To predict the essential genes in our datasets, we will use the Transit tool <span class="citation"><a href="#dejesus2015transit">DeJesus <i>et al.</i> 2015</a></span>.</p>

<h2 id="transit">Transit</h2>
<p>Transit is a software that can be used to analyse TnSeq Data. It is compatible with Mariner and Tn5 transposon. In total, 3 methods are available to assess gene essentiality in one sample.</p>

<h4 id="gumbel-method">Gumbel method</h4>

<p>The Gumbel method performs a gene by gene analysis of essentiality for Mariner data based on the longest consecutive sequence of TA site without insertions in a gene. This allows to identify essential domains regardless of insertion at other location of the gene.</p>

<p>The distribution governing the longest sequence of consecutive non insertion is  characterized by a Gumbel distribution. This distribution is often used to model the distribution of maximums (or minimums) of a variable (See the <a href="https://en.wikipedia.org/wiki/Gumbel_distribution">Wikipedia article on Gumbel distribution </a>). The Gumbel distribution is used as a likelihood of non-essentiality, as the longest run of non-insertions should follow what is expected given the global distribution of non-insertion runs. The essential genes, whose runs of empty sites are longer than expected, are modeled through a normalized sigmoid function. This function reflect that any gene can be essential, except the gene were the run of non insertions are too smell to represent a domain.</p>

<p>The total distribution of the maximum run of non-insertion per gene is therefore represented y a bimodal distribution composed of a Gumbel distribution of non-essential genes and a normalized sigmoid function of essential ones. The parameters of these distributions are estimated through a Metropolis–Hastings (MH) sampling procedure <span class="citation"><a href="#dejesus2013bayesian">DeJesus <i>et al.</i> 2013</a></span>).</p>

<p>Using these two distribution, the posterior probability of each distribution is calculated for each gene. Some genes can be classified as “Unclear” if one probability is not winning over the other. Some other can be classified as “Small” if the space of TA sites covered by the gene is insufficient to categorize it (See the <a href="https://transit.readthedocs.io/en/latest/transit_methods.html#gumbel">Transit Manual for the Gumbel method</a>).</p>

<h4 id="hmm-method">HMM method</h4>

<p>The HMM method performs a whole genome essentiality analysis for Mariner data. This approach uses the clustering of TA sites along the genome to identify essential regions, and then apply results to the annotation to identify genes containing essential regions. The HMM method provides a classification for each TA site into 4 states : Essential, Non-Essential, Growth Advantage, and Growth Defect.</p>

<p>This method require a well-saturated library and is sensitive to sparse datasets <span class="citation"><a href="#dejesus2013hidden">DeJesus and Ioerger 2013</a></span>.</p>

<h4 id="tn5gaps-method">Tn5Gaps method</h4>

<p>The Tn5Gaps method is a method dedicated to the identification of essential genes in studies using Tn5 transposons. The analysis is performed on the whole genome to identify regions of non insertion overlapping with genes. It is based on a Gumbel analysis method <span class="citation"><a href="#griffin2011high">Griffin <i>et al.</i> 2011</a></span> and adapted to Tn5 transposon specificity. The main difference comes from the fact that Tn5 transposon can insert everywhere, thus
creating libraries with lower insertion rates. The difference from the Gumbel method described above is that the run of non insertion are computer on the whole genome instead of individual genes. The longest run of non insertion considered is not the longest within the gene, but the longest one overlapping the gene.</p>

<h2 id="predict-the-essentiality-of-genes">Predict the essentiality of genes</h2>

<p>In our case, we will use the Gumbel method, because we are using Mariner data and our data look pretty sparse. Gumbel is therefore the safe choice compared to HMM which is more sensitive to sparse data.</p>

<p>In order to use transit, we need to create a an annotation file in the <code class="language-plaintext highlighter-rouge">prot_table</code> format, specific to the Transit tool. It can be created this file from a GFF3 from GenBank like the one we uploaded earlier.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-hands-on-create-annotation-file-in-prot-table-format"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Hands-on : Create annotation file in prot_table format</div>
  <ol>
    <li>Check that the format of <code class="language-plaintext highlighter-rouge">staph_aur.gff3</code> file is <code class="language-plaintext highlighter-rouge">gff3</code> and not <code class="language-plaintext highlighter-rouge">gff</code></li>
    <li>
      <p>Change the datatype if needed</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing the datatype<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the lower part <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong>, select <code class="language-plaintext highlighter-rouge">gff3</code>     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><strong>Convert GFF3 to prot_table for TRANSIT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>GenBank GFF file”</em>: the <code class="language-plaintext highlighter-rouge">staph_aur.gff3</code> file</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Now that we have prepared the annotation file, we can use the count per TA site to predict essential genes using Transit tool, with some customized parameters:</p>

<ul>
  <li>The number of sites with a single read could be artefactual. We ignore then counts lower than 2</li>
  <li>A disrupted site that would be very close to the border of the gene may not actually disturb the gene function, and therefore not be an actual signal of disruption. We ignore then insertion near the extremities of the genes.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-predict-gene-essentiality-with-transit"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Predict gene essentiality with Transit</div>

  <ol>
    <li><strong>Gumbel</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>“Operation mode”</em>: <code class="language-plaintext highlighter-rouge">Replicates</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input .wig files”</em>: output of the latest <strong>Sort</strong></li>
          </ul>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input annotation”</em>: output of the latest <strong>Convert</strong></li>
        <li><em>“Smallest read-count to consider”</em>: <code class="language-plaintext highlighter-rouge">2</code> (to ignore single count insertions)</li>
        <li><em>“Ignore TAs occurring at given fraction of the N terminus.”</em>: <code class="language-plaintext highlighter-rouge">0.1</code> (to ignore 10% of the insertion at the N-terminus extremity of the gene)</li>
        <li><em>“Ignore TAs occurring at given fraction of the C terminus.”</em>: <code class="language-plaintext highlighter-rouge">0.1</code> (to ignore 10% of the insertion at the C-terminus extremity of the gene)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output of Transit is a tabulated file containing the following columns:</p>
<ul>
  <li>Gene ID</li>
  <li>Name of the gene</li>
  <li>Gene description</li>
  <li>Number of Transposon Insertions Observed within the Gene</li>
  <li>Total Number of TA sites within the Gene</li>
  <li>Length of the Maximum Run of Non-Insertions observed (in number of TA sites)</li>
  <li>Span of nucleotides for the Maximum Run of Non-Insertions</li>
  <li>Posterior Probability of Essentiality</li>
  <li>Essentiality call for the gene
    <ul>
      <li>E=Essential</li>
      <li>U=Uncertain</li>
      <li>NE=Non-Essential</li>
      <li>S=too short</li>
    </ul>
  </li>
</ul>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-more-details-about-transit"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More details about Transit</div>
  <p>You can find more information on <a href="https://transit.readthedocs.io/en/latest/transit_methods.html">Transit manual page</a></p>
</blockquote>

<p>We can obtain the list of genes predicted as essential by filtering on the essentiality call.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-hands-on-get-table-of-essential-genes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Hands-on : Get table of essential genes</div>

  <ol>
    <li><strong>Filter data on any column using simple expressions</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Filter”</em>: output of the last <strong>Gumbel</strong></li>
        <li><em>With following condition”</em>: <code class="language-plaintext highlighter-rouge">c9=='E'</code> (to select essential genes)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many genes are essential in each conditions?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>359 genes are essential in control, and 381 in condition.</p>
  </blockquote>

</blockquote>

<h2 id="compare-the-essential-genes-between-two-conditions">Compare the essential genes between two conditions</h2>

<p>Now let’s compare the results between out two conditions.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-separate-files-from-the-collection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Separate Files from the collection</div>

  <ol>
    <li><strong>Extract Dataset from a list</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to extract 1st file in the collection with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Input List”</em>: output of the last <strong>Filter</strong></li>
        <li><em>How should a dataset be selected?”</em>: <code class="language-plaintext highlighter-rouge">Select by index</code>
          <ul>
            <li><em>Element index:”</em>: <code class="language-plaintext highlighter-rouge">0</code> to select the first dataset</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Add the <code class="language-plaintext highlighter-rouge">#condition</code> tag</p>
    </li>
    <li><strong>Extract Dataset from a list</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to extract 2nd file in the collection with:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Input List”</em>: output of the last <strong>Filter</strong></li>
        <li><em>How should a dataset be selected?”</em>: <code class="language-plaintext highlighter-rouge">Select by index</code>
          <ul>
            <li><em>Element index:”</em>: <code class="language-plaintext highlighter-rouge">1</code> to select the second dataset</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add the <code class="language-plaintext highlighter-rouge">#control</code> tag</li>
  </ol>
</blockquote>

<p>We would like now to get the list of essential gene in both conditions</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-gene-essential-in-both-conditions"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get gene essential in both conditions</div>

  <ol>
    <li><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>1st file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">control</code> tag</li>
        <li><em>Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>2nd file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">condition</code> tag</li>
        <li><em>Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><em>Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st and 2nd files</code> (to get common essential genes)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Let’s now get the list of essential gene that are different between the two conditions.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-gene-essential-only-in-control"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get gene essential only in control</div>

  <ol>
    <li><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>1st file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">control</code> tag</li>
        <li><em>Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>2nd file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">condition</code> tag</li>
        <li><em>Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><em>Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">1st but not 2nd</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-hands-on-get-gene-essential-only-in-condition"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Hands-on : Get gene essential only in condition</div>

  <ol>
    <li><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>1st file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">control</code> tag</li>
        <li><em>Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><em>2nd file”</em>: output of <strong>Extract Dataset</strong> with <code class="language-plaintext highlighter-rouge">condition</code> tag</li>
        <li><em>Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column : 1</code> (to compare on gene ID)</li>
        <li><em>Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">2nd but not 1st</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many genes are essential in both conditions?</li>
    <li>How many genes are essential only in control?</li>
    <li>How many genes are essential only in condition?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>320 genes are essential in both conditions.</li>
      <li>39 genes are essential only in control.</li>
      <li>61 genes are essential only in condition.</li>
    </ol>
  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial, we learned how to predict essential genes in different conditions. Although it seems to be a long process, keep in mind that the steps allowing to get the list of TA site does not need to be repeated for each analysis as long as you study the same genome.
For repetitive analyses, the steps are fewer : mapping, coverage, attribution to TA sites, merging of strands and Transit analysis.</p>

<p>This tutorial focused on simple gene essentiality prediction in individual samples. You could be interested in intergenic regions as well, or comparing many samples together. Other methods exist in transit for these goals, that follow the same workflow. It is important to look at your data and to perform quality control to evaluate the coverage and sparsity of your data. This will influence not only the choice of method (HMM does not manage sparse data well), but the choice of parameters for your analysis.</p>

<p>Once you get this list of gene, it is interesting to verify manually the genes classified as essentials. In order to do that, you can visualize the read mapped on your genome by displaying the bam files in IGV.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-add-genome-and-annotations-to-igv-from-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-genome-and-annotations-to-igv-from-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Add genome and annotations to IGV from Galaxy<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Expand the FASTA dataset with the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>   <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>   <li>     <p>Close the window</p>     <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>   </li>   <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The annotation track should appear. Be careful that all files have the same genome ID</p>   </li> </ol> </blockquote>
<p><!--END_SNIPPET--></p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-add-mapped-reads-track-to-igv-from-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-mapped-reads-track-to-igv-from-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Add Mapped reads track to IGV from Galaxy<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Check if the reference genome is available on the IGV instance</li>   <li>Expand the BAM dataset with the mapped reads in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the reads into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The mapped reads track should appear. Be sure that all files have the same genome ID</p>   </li> </ol> </blockquote>
<p><!--END_SNIPPET--></p>

<p>When looking at the read mapping at the location of predicted essential genes, you may encounter several situation: Some gene will have no read at all mapping to them, some will have an empty region, while other will have no clearly defined empty region but a very low count of reads, possibly indicating a growth defect induced by the gene disruption.</p>

<p><a href="../../images/tnseq/empty_gene.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/empty_gene.png" alt="Example of Essential gene with no reads mapped. " width="1424" height="1040" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Example of Essential gene with no reads mapped&lt;/figcaption&gt;</figure></a>
<a href="../../images/tnseq/low_coverage_gene.png" rel="noopener noreferrer"><figure id="figure-7" style="max-width: 90%; margin:auto;"><img src="../../images/tnseq/low_coverage_gene.png" alt="Example of Essential gene with depleted coverage. " width="1230" height="1094" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Example of Essential gene with depleted coverage&lt;/figcaption&gt;</figure></a></p>

<p>Once you have curated and confirm the genes that appear essential, comparing the annotation of essential genes with the condition of growth will allow you a better understanding of certain mechanism of interest, like the gene involved in resistance to stress for example.</p>

<p>A comparison with published results of essentiality in your organism will be useful but don’t expect to find exactly the same results. The conditions of growth differ from one experiment to another, and can highly influence the detection of essential genes.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>TnSeq is a useful analysis to detect essential genes and conditional essential genes</p>
</li>
                        
                        <li><p>The saturation of the library must be sufficient so that every gene must have been impacted at least once</p>
</li>
                        
                        <li><p>Be sure you have removed every transposon sequence before starting your analysis</p>
</li>
                        
                        <li><p>Verify the results manually do separate domain essential and essential genes</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/genome-annotation/faqs/">FAQ page for the Genome Annotation topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="griffin2011high">Griffin, J. E., J. D. Gawronski, M. A. DeJesus, T. R. Ioerger, B. J. Akerley <i>et al.</i>, 2011 <b>High-resolution phenotypic profiling defines genes essential for mycobacterial growth and cholesterol catabolism</b>. PLoS pathogens 7: e1002251. <a href="https://doi.org/10.1371/journal.ppat.1002251">10.1371/journal.ppat.1002251</a></li>
<li id="dejesus2013hidden">DeJesus, M. A., and T. R. Ioerger, 2013 <b>A hidden Markov model for identifying essential and growth-defect regions in bacterial genomes from transposon insertion sequencing data</b>. BMC bioinformatics 14: 303. <a href="https://doi.org/10.1186/1471-2105-14-303">10.1186/1471-2105-14-303</a></li>
<li id="dejesus2013bayesian">DeJesus, M. A., Y. J. Zhang, C. M. Sassetti, E. J. Rubin, J. C. Sacchettini <i>et al.</i>, 2013 <b>Bayesian analysis of gene essentiality based on sequencing of transposon insertion libraries</b>. Bioinformatics 29: 695–703. <a href="https://doi.org/10.1093/bioinformatics/btt043">10.1093/bioinformatics/btt043</a></li>
<li id="dejesus2015transit">DeJesus, M. A., C. Ambadipudi, R. Baker, C. Sassetti, and T. R. Ioerger, 2015 <b>TRANSIT-a software tool for Himar1 TnSeq analysis</b>. PLoS computational biology 11: e1004401. <a href="https://doi.org/10.1371/journal.pcbi.1004401">10.1371/journal.pcbi.1004401</a></li>
<li id="Kwon2015">Kwon, Y. M., S. C. Ricke, and R. K. Mandal, 2015 <b>Transposon sequencing: methods and expanding applications</b>. Applied Microbiology and Biotechnology 100: 31–43. <a href="https://doi.org/10.1007/s00253-015-7037-8">10.1007/s00253-015-7037-8</a></li>
<li id="Santiago2015">Santiago, M., L. M. Matano, S. H. Moussa, M. S. Gilmore, S. Walker <i>et al.</i>, 2015 <b>A new platform for ultra-high density Staphylococcus aureus transposon libraries</b>. BMC Genomics 16: <a href="https://doi.org/10.1186/s12864-015-1361-3">10.1186/s12864-015-1361-3</a></li>
<li id="Chao2016">Chao, M. C., S. Abel, B. M. Davis, and M. K. Waldor, 2016 <b>The design and analysis of transposon insertion sequencing experiments</b>. Nature Reviews Microbiology 14: 119–128. <a href="https://doi.org/10.1038/nrmicro.2015.7">10.1038/nrmicro.2015.7</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Essential genes detection with Transposon insertion sequencing (Genome Annotation)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Delphine Lariviere, Bérénice Batut,  <b>Essential genes detection with Transposon insertion sequencing (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html">https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{genome-annotation-tnseq,
author = "Delphine Lariviere and Bérénice Batut",
	title = "Essential genes detection with Transposon insertion sequencing (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/genome-annotation/tutorials/tnseq/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: deeptools_bam_coverage
  owner: bgruening
  revisions: bb1e4f63e0e6
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: find_subsequences
  owner: bgruening
  revisions: d882a0a75759
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bowtie_wrappers
  owner: devteam
  revisions: b46e7d48076a
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: column_maker
  owner: devteam
  revisions: be25c075ed54
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gff_to_prot
  owner: iuc
  revisions: 07d7991bf205
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: transit_gumbel
  owner: iuc
  revisions: f50807720df1
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: cutadapt
  owner: lparsons
  revisions: 49370cb85f0f
  tool_panel_section_label: Genome Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00179">https://gxy.io/GTN:T00179</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/genome-annotation/tutorials/tnseq/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703809381'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>