<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1697502352' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1697502352' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-01-24 14:21:57 +0000">
<meta name="DC.creator" content="Jon Trow">
<meta name="DC.creator" content="Adelaide Rhodes">

        
        
        
        
        <meta name="description" content="A collection of microtutorials explaining various feature...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis">
        <meta property="og:description" content="A collection of microtutorials explaining various feature...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-01-24 14:21:57 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Using Galaxy and Managing your Data",
    "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
    "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html",
  "timeRequired": "PT30M",
  "teaches": "Learn about SRA aligned read format and vcf files for Runs containing SARS-CoV-2 content\n - Understand how to search the metadata for these Runs to find your dataset of interest and then import that data in your preferred format",
  "keywords": "galaxy-interface, ncbi, covid19",
  "description": "The questions this  addresses are:\n - How can I search SRA SARS-CoV-2 metadata from within Galaxy?\n - How can I import SRA aligned read files and extract the data in my format of choice?\n - How can I import vcf files into Galaxy that have been generated for these Runs?\n\n\nThe objectives are:\n - Learn about SRA aligned read format and vcf files for Runs containing SARS-CoV-2 content\n - Understand how to search the metadata for these Runs to find your dataset of interest and then import that data in your preferred format\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html",
      "name": "Rule Based Uploader",
      "description": "Hands-on for 'Rule Based Uploader' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/jontrow/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/jontrow/",
      "name": "Jon Trow",
      "image": "https://avatars.githubusercontent.com/jontrow",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/RareSeas/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/RareSeas/",
      "name": "Adelaide Rhodes",
      "image": "https://avatars.githubusercontent.com/RareSeas",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Using Galaxy and Managing your Data",
      "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "ncbi"
    },
    {
      "@type": "Thing",
      "name": "covid19"
    }
  ],
  "abstract": "# Background"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1697502352'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/galaxy-interface" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Using Galaxy and Managing your Data
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/galaxy-interface/faqs/" title="Check our FAQs for the Using Galaxy and Managing your Data topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-galaxy-interface">
    <h1 data-toc-skip>SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/jontrow/" class="contributor-badge contributor-jontrow"><img src="https://avatars.githubusercontent.com/jontrow?s=36" alt="Avatar" width="36" height="36">Jon Trow</a><a href="/training-material/hall-of-fame/RareSeas/" class="contributor-badge contributor-RareSeas"><img src="https://avatars.githubusercontent.com/RareSeas?s=36" alt="Avatar" width="36" height="36">Adelaide Rhodes</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can I search SRA SARS-CoV-2 metadata from within Galaxy?</p>
</li>
        
        <li><p>How can I import SRA aligned read files and extract the data in my format of choice?</p>
</li>
        
        <li><p>How can I import vcf files into Galaxy that have been generated for these Runs?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn about SRA aligned read format and vcf files for Runs containing SARS-CoV-2 content</p>
</li>
        
        <li><p>Understand how to search the metadata for these Runs to find your dataset of interest and then import that data in your preferred format</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
        
    <li>
    
        
        
        <a href="/training-material/topics/galaxy-interface">Using Galaxy and Managing your Data</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Rule Based Uploader:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 30 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/galaxy-interface/ncbi-sarf/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
          
          
          

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/galaxy-interface/ncbi-sarf/slides" title="View GTN Video Library for these Slides">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture (February 2021)
            </a>
        </li>
        
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
    

    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jan 24, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00155">https://gxy.io/GTN:T00155</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="background">Background</h1>

<p>Traditionally, after a list of run accessions has been filtered on the NCBI website, the accessions are used to download and extract fastq using the SRA toolkit to enter into the next steps of the workflow. A newer compressed data type, generated from raw submitted data containing SARS-CoV-2 sequence, is also accessible to Galaxy users from SRA in the Cloud.</p>

<p>SRA Aligned Read Format (SARF) provides further output options beyond basic fastq format, for example:</p>

<ol>
  <li>contigs created from the raw reads in the run</li>
  <li>reads aligned back to the contigs</li>
  <li>reads with placeholder quality scores</li>
  <li>VCF files can also be downloaded for these records relative to the SARS-CoV-2 RefSeq record</li>
</ol>

<ul>
  <li>These formats can speed up workflows such as assembly and variant calling.</li>
  <li>This data format is still referenced by the Run accession and accessed using the SRA toolkit.</li>
  <li>This workshop describes the SARF data objects along with associated searchable metadata, and demonstrates a few ways to enter them into traditional workflows.</li>
</ul>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#background" id="markdown-toc-background">Background</a></li>
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a>    <ol>
      <li><a href="#sra-aligned-read-format" id="markdown-toc-sra-aligned-read-format">SRA Aligned Read Format</a></li>
    </ol>
  </li>
  <li><a href="#workflow-diagram" id="markdown-toc-workflow-diagram">Workflow Diagram</a></li>
  <li><a href="#finding-sra-sars-cov-2-runs-of-interest" id="markdown-toc-finding-sra-sars-cov-2-runs-of-interest">Finding SRA SARS-CoV-2 Runs of Interest</a></li>
  <li><a href="#query-sarf-metadata" id="markdown-toc-query-sarf-metadata">Query SARF Metadata</a></li>
  <li><a href="#importing-sarfs-of-interest" id="markdown-toc-importing-sarfs-of-interest">Importing SARFs of Interest</a></li>
  <li><a href="#importing-vcfs-for-sars-cov-2-runs" id="markdown-toc-importing-vcfs-for-sars-cov-2-runs">Importing VCFs for SARS-Cov-2 Runs</a></li>
  <li><a href="#feedback-for-ncbi" id="markdown-toc-feedback-for-ncbi">Feedback for NCBI</a></li>
  <li><a href="#other-ncbi-resources" id="markdown-toc-other-ncbi-resources">Other NCBI Resources</a></li>
  <li><a href="#acknowledgements" id="markdown-toc-acknowledgements">Acknowledgements</a></li>
  <li><a href="#affiliations" id="markdown-toc-affiliations">Affiliations</a></li>
</ol>

</blockquote>

<h1 id="introduction">Introduction</h1>

<p>The aim of this tutorial is to introduce you to some of SRA’s new SARS-CoV-2 cloud resources and data formats, then show you how to filter for Runs of interest to you and access that data in your format of choice in Galaxy to use in your analysis pipeline.</p>

<h2 id="sra-aligned-read-format">SRA Aligned Read Format</h2>

<p>All data submitted to SRA is scanned with our SARS-CoV-2 Detection Tool which uses a Kmer-based approach to identify Runs with Coronaviridae content. The initial scope of the project is limited to those runs deposited in SRA with at least 100 hits for SARS-CoV-2 via the SARS-CoV-2 Detection Tool, a read length of at least 75, and generated using the Illumina platform.</p>

<ol>
  <li>
    <p>For these Runs, Saute was used to assemble contigs via guided assembly, with the SARS-CoV-2 refseq genomic sequence (NC_045512.2) used as the guide.</p>
  </li>
  <li>
    <p>If contigs were successfully assembled, reads were mapped back to the contigs and coverage calculated. These contigs with the reads mapped back and with quality scores removed (to keep the object size small) are the aligned read format files.</p>
  </li>
  <li>
    <p>The SRA toolkit can be used to dump just the contigs in fasta format, the reads aligned to the contigs in sam format or the raw reads in fastq format with placeholder quality scores.</p>
  </li>
  <li>
    <p>The contigs were also assessed via megablast against the nucleotide blast database and the results made available for search.</p>
  </li>
  <li>
    <p>In addition, to support investigation of viral evolution during the pandemic and after the introduction of vaccines, variants are identified relative to the SARS-CoV-2 RefSeq record for each processed run using BCFTools.</p>
  </li>
</ol>

<p>The SRA aligned reads, the VCF files, the results of these analyses (such as BLAST and VIGOR3 annotation), and the associated BioSample and sequencing library metadata are available for free access from cloud providers.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>These data can be dumped in <code class="language-plaintext highlighter-rouge">sam</code> format using the <code class="language-plaintext highlighter-rouge">sam-dump</code> tool in the SRA Toolkit, but this function doesn’t work within Galaxy yet.
We hope to include that functionality in a future update.</p>

</blockquote>

<h1 id="workflow-diagram">Workflow Diagram</h1>

<p><a href="../../images/SARF_Workflow_Overview.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/SARF_Workflow_Overview.png" alt="Workflow Diagram for this Tutorial. " width="823" height="464" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Workflow Diagram for this Tutorial&lt;/figcaption&gt;</figure></a></p>

<h1 id="finding-sra-sars-cov-2-runs-of-interest">Finding SRA SARS-CoV-2 Runs of Interest</h1>

<p>Metadata for SARS-CoV-2 submissions to the SRA includes submitted sample and library information, BLAST results, descriptive contig statistics, and variation and annotation information.  These metadata are updated daily and made available to query in the cloud using <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-bigquery/">Google’s BigQuery</a> or <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-athena/">Amazon’s Athena</a> services. However, the raw underlying information is also provided as a group of json files that can be <strong>downloaded for free from the Open Data Platform</strong> without logging in to the cloud. These json files can be imported to Galaxy and queried there to find <strong>Runs of interest</strong>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>Some of these tables include complex data fields (array of values) that don’t have a clean analogue in a classic SQL database or table and these can’t be easily queried in Galaxy currently. If you require access to <a href="https://www.ncbi.nlm.nih.gov/sra/docs/aligned-metadata-tables/">cloud tables</a> or fields not available in Galaxy we recommend accessing those natively in <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-bigquery/">BigQuery</a> or <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-athena/">Athena</a>.</p>

</blockquote>

<p>We will import the JSON files into Galaxy to query them directory, however the files are split up for efficient querying in the cloud and updated daily, so we first need to get the most up-to-date list of files so we can import those to Galaxy. We’ll just be using a couple of tables in this training, but the other tables can be imported in the same way, using the index files below.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>These metadata files are updated daily around 5:30pm EST. If you try to access the data around this time but encounter an error, trying again a short while later should resolve the issue. <a href="https://www.thetimezoneconverter.com">Time Zone Converter</a></p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-loading-sra-aligned-read-format-sarf-object-metadata-urls-into-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Loading SRA Aligned Read Format (SARF) Object Metadata URLs into Galaxy</div>

  <p>This step needs to be repeated at the beginning of an analysis to refresh the metadata to the latest daily version.</p>

  <ol>
    <li>
      <p>Go to your Galaxy instance of choice such as one of the <a href="https://usegalaxy.org/">usegalaxy.org</a>, <a href="https://usegalaxy.eu">usegalaxy.eu</a>, <a href="https://usegalaxy.org.au">usegalaxy.org.au</a> or any other.</p>
    </li>
    <li>
      <p>Create a new history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename your history, <em>e.g.</em> “NCBI SARF”</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><strong>Click</strong> the upload icon toward the top left corner.</p>

      <p>By default the familiar simple upload dialog should appear. This dialog has more advanced options as different tabs across the top of this dialog though.</p>
    </li>
    <li>
      <p><strong>Click</strong> <code class="language-plaintext highlighter-rouge">Rule-based</code> as shown below.</p>

      <p><a href="../../images/rules/rules_example_1_1_rules_landing.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/rules/rules_example_1_1_rules_landing.png" alt="screenshot of the rule based uploader modal, it is empty. " width="2560" height="732" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Be sure that you are uploading data as Datasets, and that it will be loaded from a ‘Pasted Table’&lt;/figcaption&gt;</figure></a></p>

      <ol>
        <li>
          <p>Copy/<strong>paste</strong> the URLs into the provided box:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/contigs.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/annotated_variations.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/blastn.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/metadata.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/peptides.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/tax_analysis.filelist
https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/variations.filelist
</code></pre></div>          </div>
        </li>
        <li>
          <p>Click <strong>Build</strong></p>
        </li>
        <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Basename of Path of URL</code>
          <ul>
            <li><em>“From Column?”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
          </ul>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>
              <ul>
                <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
              </ul>
            </li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name</code> (not name tag!)
              <ul>
                <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p><strong>Click</strong> <code class="language-plaintext highlighter-rouge">Apply</code>. You should see a table with two columns, the left being the URL column, and the right being the Name column with just the filename.</p>

          <p>You are now ready to start the upload.</p>
        </li>
        <li><strong>Click</strong> the <code class="language-plaintext highlighter-rouge">Upload</code> button</li>
      </ol>
    </li>
  </ol>

</blockquote>

<p>With that you should have 7 different files in your history. If you examine the files you’ll see that they each have a list of filenames like the following table, with either today or yesterday’s date in the filename, depending on your timezone offset from NCBI’s offices. [Time zone converter](</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>2021-05-27.000000000000.json.gz
2021-05-27.000000000001.json.gz
2021-05-27.000000000002.json.gz
2021-05-27.000000000003.json.gz
2021-05-27.000000000004.json.gz
2021-05-27.000000000005.json.gz
2021-05-27.000000000006.json.gz
2021-05-27.000000000007.json.gz
</code></pre></div></div>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-loading-sra-aligned-read-format-sarf-contig-metadata-into-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Loading SRA Aligned Read Format (SARF) Contig Metadata into Galaxy</div>

  <p>Next we will convert this list of filenames to the HTTP URLs for easy import into Galaxy.</p>

  <ol>
    <li><strong>Open</strong> the <code class="language-plaintext highlighter-rouge">Rule-based</code> upload tab again, but this time:
      <ol>
        <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
        <li><em>“Load tabular data from”</em>: <code class="language-plaintext highlighter-rouge">History Dataset</code></li>
        <li>
          <p><em>“Select dataset to load”</em>: <code class="language-plaintext highlighter-rouge">contigs.filelist</code></p>
        </li>
        <li>
          <p><strong>Click</strong> <code class="language-plaintext highlighter-rouge">Build</code> to bring up the rule builder.</p>
        </li>
        <li>
          <p>Make the following changes in the Rule Builder</p>

          <ul>
            <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Fixed Value</code>
              <ul>
                <li><em>“Value”</em>: <code class="language-plaintext highlighter-rouge">https://storage.googleapis.com/nih-sequence-read-archive/SARS_COV_2/contigs/</code></li>
                <li>Apply</li>
              </ul>
            </li>
            <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Concatenate Columns</code>
              <ul>
                <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                <li>Apply</li>
              </ul>
            </li>
            <li>From <strong>Rules</strong> menu, select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li><code class="language-plaintext highlighter-rouge">Add Definition</code>, <code class="language-plaintext highlighter-rouge">URL</code>, Select Column <code class="language-plaintext highlighter-rouge">C</code></li>
                <li><code class="language-plaintext highlighter-rouge">Add Definition</code>, <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>, Column <code class="language-plaintext highlighter-rouge">A</code></li>
                <li>Apply</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p>Name the collection <code class="language-plaintext highlighter-rouge">contigs.json</code></p>
        </li>
        <li>Click the <code class="language-plaintext highlighter-rouge">Upload</code> button.</li>
      </ol>

      <p><strong>Once those download jobs have all turned green in the history list, we’ll concatenate these into a single file.</strong></p>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0" title="Concatenate datasets tool" aria-role="link"><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0</span></span> tail to head (cat) files with the following parameters:</p>

      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Datasets to Concatenate”</em>: <code class="language-plaintext highlighter-rouge">contigs.json</code> collection</li>
        <li>Click Run Tool</li>
      </ul>
    </li>
    <li>
      <p>Rename this history item to <code class="language-plaintext highlighter-rouge">contigs.single.json</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field  to <code class="language-plaintext highlighter-rouge">contigs.single.json</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/jq/jq/1.0" title="JQ tool" aria-role="link"><strong>JQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/jq/jq/1.0</span></span> process JSON files with the following parameters:</p>

      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“JSON Input”</em>: <code class="language-plaintext highlighter-rouge">contigs.single.json</code></li>
        <li><em>“jq filter”</em>: <code class="language-plaintext highlighter-rouge">[.[]]</code></li>
        <li><em>“Convert output to tabular”</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>Click Run Tool</li>
      </ul>
    </li>
    <li>
      <p>Rename this file <code class="language-plaintext highlighter-rouge">contigs.tsv</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field  to <code class="language-plaintext highlighter-rouge">contigs.tsv</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>The conversion of the json files to tabular format takes some time, this is a good time to go make some tea.</p>

</blockquote>

<h1 id="query-sarf-metadata">Query SARF Metadata</h1>

<p>Now that a table has been generated, we will query the table to find the runs of interest.  It is a good idea to save the table for future queries on the same dataset.  Rerunning the import steps above without filtration will provide a different set of metadata each day.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-query-the-sra-metadata-table-using-sqlite"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Query the SRA Metadata Table using SQLite</div>

  <p>Next we’ll query this metadata using the <code class="language-plaintext highlighter-rouge">Query tabular</code> tool to get a list of all Runs containing contigs of greater than 20,000 nucleotides and average coverage of at least 100X.</p>

  <ol>
    <li>
      <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Query Tabular using sqlite sql tool" aria-role="link"><strong>Query Tabular using sqlite sql</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0</span></span> with the following parameters:</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-can-t-find-it"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-t-find-it" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Can't find it?<span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>If you’re not using the <a href="https://docs.galaxyproject.org/en/latest/admin/special_topics/gtn.html">GTN-in-Galaxy view</a>, you can search for ‘sql’ to find it.</p>
      </blockquote>

      <ul>
        <li>In <em>“Database Table”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Database Table”</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Dataset for Table”</em>: <code class="language-plaintext highlighter-rouge">contigs.tsv</code></li>
                <li>In <em>“Table Options”</em>
                  <ul>
                    <li>In <em>“Table Options”</em>
                      <ul>
                        <li><em>“specify name for table”</em>: <code class="language-plaintext highlighter-rouge">SARS_contigs</code></li>
                        <li><em>“Specify Column Names”</em>: <code class="language-plaintext highlighter-rouge">name,run,coverage,tax_id,hits,length,md5</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p><em>“SQL Query to generate tabular output”</em>:</p>

          <div class="language-sql highlighter-rouge"><div class="highlight"><pre class="highlight"><code> <span class="k">SELECT</span> <span class="k">DISTINCT</span> <span class="n">run</span>
 <span class="k">FROM</span> <span class="n">SARS_contigs</span>
 <span class="k">WHERE</span> <span class="k">length</span> <span class="o">&gt;</span> <span class="mi">20000</span> <span class="k">AND</span> <span class="n">coverage</span> <span class="o">&gt;</span> <span class="mi">100</span> <span class="k">AND</span> <span class="n">run</span> <span class="k">like</span> <span class="s1">'%SRR%'</span>
 <span class="k">ORDER</span> <span class="k">BY</span> <span class="n">run</span> <span class="k">ASC</span>
 <span class="k">LIMIT</span> <span class="mi">10</span>
</code></pre></div>          </div>
        </li>
        <li><em>“include query result column headers”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-save-sqlite-database-for-future-queries"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-save-sqlite-database-for-future-queries" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Save SQlite Database for future queries<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>If you plan to do multiple queries on the same SQL database or want to skip preprocessing the metadata for future work, it may be useful to set</p>

        <ul>
          <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Save the sqlite database in your history”</em> to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        </ul>

      </blockquote>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong> and rename the output file to <code class="language-plaintext highlighter-rouge">Run_list</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field  to <code class="language-plaintext highlighter-rouge">Run_list</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-need-fastq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-need-fastq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Need fastq?<span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>Use <code class="language-plaintext highlighter-rouge">Run_list</code> to bring in Fastq files with Submitted Quality Scores (+BQS format)</p>
      </blockquote>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-column-headers-for-the-other-metadata-tables"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-column-headers-for-the-other-metadata-tables" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Column Headers for the Other Metadata Tables<span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>We are not going to bring in the other metadata tables in this tutorial. Here is a list of column headers for contigs and the other tables.
You can find full definitions for these columns here:</p>

        <p><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud-based-examples/">https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud-based-examples/</a></p>

        <p><a href="https://www.ncbi.nlm.nih.gov/sra/docs/aligned-metadata-tables/">https://www.ncbi.nlm.nih.gov/sra/docs/aligned-metadata-tables/</a></p>

        <p><strong>contigs</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>name,run,coverage,tax_id,hits,length,md5
</code></pre></div>        </div>
        <p><strong>annotated_variations</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>run,chrom,pos,id,ref,alt,qual,filter,info,format,sample_a,ac,an,bqb,dp,dp4,dp4_1,dp4_2,dp4_3,dp4_4,idv,imf,mq,mq0f,mqb,mqsb,rpb,sgb,vdb,g_gt,g_pl,g_pl_1,g_pl_2,g_dp,g_ad,g_ad_1,g_ad_2,protein_position,ref_codon,alt_codon,ref_aa,alt_aa,protein_name,protein_length,variation
</code></pre></div>        </div>
        <p><strong>blastn</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>acc,qacc,staxid,sacc,slen,length,bitscore,score,pident,sskingdom,evalue,ssciname
</code></pre></div>        </div>
        <p><strong>metadata</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>acc,assay_type,center_name,consent,experiment,sample_name,instrument,librarylayout,libraryselection,librarysource,platform,sample_acc,biosample,organism,sra_study,releasedate,bioproject,mbytes,loaddate,avgspotlen,mbases,insertsize,library_name,biosamplemodel_sam,collection_date_sam,geo_loc_name_country_calc,geo_loc_name_country_continent_calc,ena_first_public_run,ena_last_update_run,sample_name_sam,datastore_filetype,datastore_provider,datastore_region,attributes,jattr
</code></pre></div>        </div>
        <p><strong>peptides</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>name,contig,mat_peptide,run,location,gene,product,ref_db,ref_id,sequence
</code></pre></div>        </div>
        <p><strong>tax_analysis</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>run,contig,tax_id,rank,name,total_count,self_count,ilevel,ileft,iright
</code></pre></div>        </div>
        <p><strong>variations</strong></p>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>run,chrom,pos,id,ref,alt,qual,filter,info
</code></pre></div>        </div>

      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-download-fastq-with-quality-scores"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-download-fastq-with-quality-scores" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Download Fastq with Quality Scores<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you would like to dump the raw, underlying data in fastq format with the original quality scores, you can stop here and use the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.11.0+galaxy0" title="Faster Download and Extract Reads in FASTQ tool" aria-role="link"><strong>Faster Download and Extract Reads in FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.11.0+galaxy0</span></span> tool with the following parameters:</p>

  <ul>
    <li><em>“input type”</em>: <code class="language-plaintext highlighter-rouge">list of SRA accessions, one per line</code></li>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“sra accession list”</em>: <code class="language-plaintext highlighter-rouge">Run_list</code></li>
  </ul>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-importing-a-list-of-srr-from-athena-or-bigquery"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-list-of-srr-from-athena-or-bigquery" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing a list of SRR from Athena or BigQuery<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you opted to conduct your metadata search in the cloud using AWS Athena or GCP BigQuery instead of importing the json file to Galaxy, you can save a list of your Run accessions from that search result and import that file as the <code class="language-plaintext highlighter-rouge">Run_list</code> to proceed with the rest of this tutorial.</p>
</blockquote>

<h1 id="importing-sarfs-of-interest">Importing SARFs of Interest</h1>

<p>Now that we have assembled a list of Runs that have contigs we are interested in, we’ll construct the path to the <strong>SARFS</strong> in the cloud and import those to Galaxy so we can work with them.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-importing-sarfs-of-interest"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Importing SARFs of Interest</div>

  <ol>
    <li>
      <p>Upload Data</p>

      <ol>
        <li><strong>Open</strong> the <code class="language-plaintext highlighter-rouge">Rule-based</code> upload tab again, but this time:
          <ul>
            <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
            <li><em>“Load tabular data from”</em>: <code class="language-plaintext highlighter-rouge">History Dataset</code></li>
            <li><em>“Select dataset to load”</em>: <code class="language-plaintext highlighter-rouge">Run_list</code></li>
          </ul>
        </li>
        <li>
          <p><strong>Click</strong> <code class="language-plaintext highlighter-rouge">Build</code> to bring up the rule builder.</p>
        </li>
        <li>
          <p>Make the following changes in the Rule Builder</p>

          <ul>
            <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Using a Regular Expression</code>
              <ul>
                <li>Check <em>“Create column from expression replacement”</em></li>
                <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">(.*)</code></li>
                <li><em>“Replacement Expression”</em>: <code class="language-plaintext highlighter-rouge">https://sra-pub-sars-cov2.s3.amazonaws.com/RAO/\1/\1.realign</code></li>
                <li>Apply</li>
              </ul>
            </li>
            <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>
                  <ul>
                    <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                  </ul>
                </li>
                <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier</code>
                  <ul>
                    <li><em>“List Identifier”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                    <li>Apply</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p>Name the output collection <code class="language-plaintext highlighter-rouge">sarf_path</code> before clicking <strong>Upload</strong></p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

            <p>Please note that there can be some lag in availability of SARF/VCF files in the cloud (particularly for newly submitted data). So it’s possible to get a download error for a file that isn’t yet present in the cloud. In these cases waiting ~24 hours will generally resolve the issue and allow you to access the file.</p>

          </blockquote>
        </li>
      </ol>
    </li>
    <li>
      <p><strong>Now we will use the SRA toolkit to retrieve the contigs in fasta format</strong></p>

      <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fastq_dump/2.11.0+galaxy0" title="Download and Extract Reads in FASTA/Q format from NCBI SRA tool" aria-role="link"><strong>Download and Extract Reads in FASTA/Q format from NCBI SRA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fastq_dump/2.11.0+galaxy0</span></span> with the following parameters:</p>

      <ul>
        <li><em>“Select input type”</em>: <code class="language-plaintext highlighter-rouge">SRA Archive in current history</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span><em>“sra archive”</em>: <code class="language-plaintext highlighter-rouge">sarf_path</code></li>
        <li>In “Advanced Options”:
          <ul>
            <li><em>“Table name within cSRA object”</em>: <code class="language-plaintext highlighter-rouge">REFERENCE</code></li>
          </ul>
        </li>
        <li>Click Run Tool</li>
      </ul>

      <p>The resulting dataset includes the contigs generated from these runs with placeholder <code class="language-plaintext highlighter-rouge">?</code> for quality scores</p>

      <ul>
        <li>Rename this collection to <code class="language-plaintext highlighter-rouge">sarf_contigs</code></li>
      </ul>

      <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5" title="Fastq to Fasta converter tool" aria-role="link"><strong>Fastq to Fasta converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.1.5</span></span></p>

      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span><em>“FASTQ file to convert”</em>: <code class="language-plaintext highlighter-rouge">sarf_contigs</code> (note: in the video this did not get renamed)</li>
        <li>Click Run Tool</li>
      </ul>

      <p>The resulting dataset includes the contigs generated from these Runs in fasta format</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-fastq-format-option"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-fastq-format-option" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Fastq format option<span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>If you prefer to dump the raw reads in fastq format with placeholder quality scores, leave the <code class="language-plaintext highlighter-rouge">Table name within cSRA object</code> field blank.</p>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="importing-vcfs-for-sars-cov-2-runs">Importing VCFs for SARS-Cov-2 Runs</h1>

<p>This example starts with the same <code class="language-plaintext highlighter-rouge">Run_list</code> generated for importing SARFs.</p>

<p>A <code class="language-plaintext highlighter-rouge">Run_list</code> could also be imported after querying metadata in the cloud using  <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-bigquery/">Google’s BigQuery</a> or <a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-athena/">Amazon’s Athena</a> services. Metadata about these runs includes submitted sample and library information, BLAST results, descriptive contig statistics, and variation and annotations. See the tutorial video for a short demo on how to search and download <code class="language-plaintext highlighter-rouge">Run_list</code> from the cloud.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-importing-vcfs-of-interest"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Importing VCFs of Interest</div>

  <ol>
    <li>
      <p>Upload Data</p>

      <ol>
        <li><strong>Open</strong> the <code class="language-plaintext highlighter-rouge">Rule-based</code> upload tab again, but this time:
          <ul>
            <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
            <li><em>“Load tabular data from”</em>: <code class="language-plaintext highlighter-rouge">History Dataset</code></li>
            <li><em>“Select dataset to load”</em>: <code class="language-plaintext highlighter-rouge">Run_list</code></li>
          </ul>
        </li>
        <li>
          <p><strong>Click</strong> <code class="language-plaintext highlighter-rouge">Build</code> to bring up the rule builder.</p>
        </li>
        <li>
          <p>Make the following changes in the Rule Builder</p>

          <ul>
            <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Using a Regular Expression</code>
              <ul>
                <li>Check <em>“Create column from expression replacement”</em></li>
                <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">(.*)</code></li>
                <li><em>“Replacement Expression”</em>: <code class="language-plaintext highlighter-rouge">https://sra-pub-sars-cov2.s3.amazonaws.com/VCF/\1/\1.vcf</code></li>
                <li>Apply</li>
              </ul>
            </li>
            <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                <li><em>“List Identifier”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                <li>Apply</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>
          <p>Name the output collection <code class="language-plaintext highlighter-rouge">VCFs</code> before clicking <strong>Upload</strong></p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-5"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

            <p>Please note that there can be some lag in availability of SARF/VCF files in the cloud (particularly for newly submitted data). So it’s possible to get a download error for a file that isn’t yet present in the cloud. In these cases waiting ~24 hours will generally resolve the issue and allow you to access the file.</p>

          </blockquote>
        </li>
      </ol>
    </li>
    <li>
      <p><strong>Use VCFs in Another Galaxy Tool</strong>. Once you have imported the VCF files, you can use them in your standard pipeline- here we will annotate them with SnpEff.</p>

      <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19" title="SnpEff eff: annotate variants for SARS-CoV-2 tool" aria-role="link"><strong>SnpEff eff: annotate variants for SARS-CoV-2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19</span></span>  with the following parameters:</p>

      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequence changes (SNPs, MNPs, InDels)”</em>: the <code class="language-plaintext highlighter-rouge">VCFs</code> collection we just created</li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-6"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>Please note that there are 2 Snepff tools, please choose the one for SARS-CoV-2</p>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="feedback-for-ncbi">Feedback for NCBI</h1>

<p>If you enjoyed this tutorial, please consider filling out this feedback link for NCBI: <a href="https://nlmenterprise.co1.qualtrics.com/jfe/form/SV_0jQct4IQOgfYYaq">https://nlmenterprise.co1.qualtrics.com/jfe/form/SV_0jQct4IQOgfYYaq</a></p>

<p>Note: the survey will stay open until July 31, 2021 (it may say it expires June 30, but we will extend the deadline at the end of the month)</p>

<p>(There is another survey link below for Galaxy).</p>

<h1 id="other-ncbi-resources">Other NCBI Resources</h1>
<ul>
  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud/">Getting started with SRA in the Cloud</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/sars-cov-2/">NCBI SARS-CoV-2 Resources</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-taxonomy-analysis-tool/">STAT tool</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-aligned-read-format/">SRA Aligned Read Format</a></li>
  <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sars-cov-2-variant-calling/">VCF generation</a></li>
</ul>

<p>If you have questions or feedback you can email the SRA helpdesk: <a href="mailto:sra@ncbi.nlm.nih.gov">sra@ncbi.nlm.nih.gov</a></p>

<h1 id="acknowledgements">Acknowledgements</h1>

<p>The authors would like to acknowledge the SRA Product Team at NCBI and the Community Outreach staff at Galaxy for their assistance on this tutorial.</p>

<h1 id="affiliations">Affiliations</h1>

<p>Adelaide Rhodes &amp; Jon Trow - Computercraft assigned to NCBI/NLM/NIH</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>NCBI Publishes datasets in the cloud that you can easily process with Galaxy</p>
</li>
                        
                        <li><p>The Rule Based Uploader simplifies processing and downloading large numbers of files</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/galaxy-interface/faqs/">FAQ page for the Using Galaxy and Managing your Data topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis (Using Galaxy and Managing your Data)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Jon Trow, Adelaide Rhodes,  <b>SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html">https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{galaxy-interface-ncbi-sarf,
author = "Jon Trow and Adelaide Rhodes",
	title = "SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: ddf54b12c295
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqtofasta
  owner: devteam
  revisions: 297962e79f39
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: jq
  owner: iuc
  revisions: 5ff75eb1a893
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: query_tabular
  owner: iuc
  revisions: 33d61c89fb8d
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff_sars_cov_2
  owner: iuc
  revisions: 2a3a00c1fa0a
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sra_tools
  owner: iuc
  revisions: 69ebb7f46e45
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sra_tools
  owner: iuc
  revisions: 83c7d564b128
  tool_panel_section_label: Using Galaxy and Managing your Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00155">https://gxy.io/GTN:T00155</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1697502352'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>