<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Quality Control</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1697329661' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1697329661' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Quality Control">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-09-19 22:36:51 +0000">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Maria Doyle">
<meta name="DC.creator" content="Alexandre Cormier">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Laura Leroi">
<meta name="DC.creator" content="Erwan Corre">
<meta name="DC.creator" content="Stéphanie Robin">
<meta name="DC.creator" content="Erasmus+ Programme">
<meta name="DC.creator" content="Cameron Hyde">

        
        
        
        
        <meta name="description" content="Analyses of sequences">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Quality Control">
        <meta property="og:description" content="Analyses of sequences">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-09-19 22:36:51 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Quality Control",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Sequence analysis",
    "description": "Analyses of sequences",
    "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Quality Control' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
  "timeRequired": "PT1H30M",
  "teaches": "Assess short reads FASTQ quality using FASTQE 🧬😎 and FastQC\n - Assess long reads FASTQ quality using Nanoplot and PycoQC\n - Perform quality correction with Cutadapt (short reads)\n - Summarise quality metrics MultiQC\n - Process single-end and paired-end data",
  "keywords": "sequence-analysis",
  "description": "The questions this  addresses are:\n - How to perform quality control of NGS raw data?\n - What are the quality parameters to check for a dataset?\n - How to improve the quality of a dataset?\n\n\nThe objectives are:\n - Assess short reads FASTQ quality using FASTQE 🧬😎 and FastQC\n - Assess long reads FASTQ quality using Nanoplot and PycoQC\n - Perform quality correction with Cutadapt (short reads)\n - Summarise quality metrics MultiQC\n - Process single-end and paired-end data\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/mblue9/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/mblue9/",
      "name": "Maria Doyle",
      "image": "https://avatars.githubusercontent.com/mblue9",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4847-8436",
      "orcid": "https://orcid.org/0000-0003-4847-8436"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/alexcorm/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/alexcorm/",
      "name": "Alexandre Cormier",
      "image": "https://avatars.githubusercontent.com/alexcorm",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7775-8413",
      "orcid": "https://orcid.org/0000-0002-7775-8413"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/abretaud/",
      "name": "Anthony Bretaudeau",
      "image": "https://avatars.githubusercontent.com/abretaud",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0914-2470",
      "orcid": "https://orcid.org/0000-0003-0914-2470"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/lleroi/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/lleroi/",
      "name": "Laura Leroi",
      "image": "https://avatars.githubusercontent.com/lleroi",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3748-4179",
      "orcid": "https://orcid.org/0000-0003-3748-4179"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/r1corre/",
      "name": "Erwan Corre",
      "image": "https://avatars.githubusercontent.com/r1corre",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6354-2278",
      "orcid": "https://orcid.org/0000-0001-6354-2278"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/stephanierobin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/stephanierobin/",
      "name": "Stéphanie Robin",
      "image": "https://avatars.githubusercontent.com/stephanierobin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-7379-9173",
      "orcid": "https://orcid.org/0000-0001-7379-9173"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/erasmusplus/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/erasmusplus/",
      "name": "Erasmus+ Programme",
      "image": "https://avatars.githubusercontent.com/erasmusplus",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/neoformit/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/neoformit/",
      "name": "Cameron Hyde",
      "image": "https://avatars.githubusercontent.com/neoformit",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5913-9766",
      "orcid": "https://orcid.org/0000-0002-5913-9766"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Sequence analysis",
      "description": "Analyses of sequences",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0080",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0080",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0080"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1697329661'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Sequence analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/sequence-analysis/faqs/" title="Check our FAQs for the Sequence analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/sequence-analysis/tutorials/quality-control/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-sequence-analysis">
    <h1 data-toc-skip>Quality Control</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Avatar" width="36" height="36">Bérénice Batut</a><a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/mblue9?s=36" alt="Avatar" width="36" height="36">Maria Doyle</a><a href="/training-material/hall-of-fame/alexcorm/" class="contributor-badge contributor-alexcorm"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/alexcorm?s=36" alt="Avatar" width="36" height="36">Alexandre Cormier</a><a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Avatar" width="36" height="36">Anthony Bretaudeau</a><a href="/training-material/hall-of-fame/lleroi/" class="contributor-badge contributor-lleroi"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/lleroi?s=36" alt="Avatar" width="36" height="36">Laura Leroi</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/r1corre?s=36" alt="Avatar" width="36" height="36">Erwan Corre</a><a href="/training-material/hall-of-fame/stephanierobin/" class="contributor-badge contributor-stephanierobin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/stephanierobin?s=36" alt="Avatar" width="36" height="36">Stéphanie Robin</a><a href="/training-material/hall-of-fame/erasmusplus/" class="contributor-badge contributor-erasmusplus"><img src="https://www.erasmusplus.nl/assets/images/logo.png" alt="Avatar" width="36" height="36">Erasmus+ Programme</a><a href="/training-material/hall-of-fame/neoformit/" class="contributor-badge contributor-neoformit"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/neoformit?s=36" alt="Avatar" width="36" height="36">Cameron Hyde</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to perform quality control of NGS raw data?</p>
</li>
        
        <li><p>What are the quality parameters to check for a dataset?</p>
</li>
        
        <li><p>How to improve the quality of a dataset?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Assess short reads FASTQ quality using FASTQE 🧬😎 and FastQC</p>
</li>
        
        <li><p>Assess long reads FASTQ quality using Nanoplot and PycoQC</p>
</li>
        
        <li><p>Perform quality correction with Cutadapt (short reads)</p>
</li>
        
        <li><p>Summarise quality metrics MultiQC</p>
</li>
        
        <li><p>Process single-end and paired-end data</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour 30 minutes</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.61771">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/sequence-analysis/tutorials/quality-control/workflows/" title="Quality Control workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/sequence-analysis/quality-control/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (May 2023)
            </a>
        </li>
        
        
          
          
          

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/sequence-analysis/quality-control/slides" title="View GTN Video Library for these Slides">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture (February 2021)
            </a>
        </li>
        
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Sep 19, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00239">https://gxy.io/GTN:T00239</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>During sequencing, the nucleotide bases in a DNA or RNA sample (library) are determined by the sequencer. For each fragment in the library, a sequence is generated, also called a <strong>read</strong>, which is simply a succession of nucleotides.</p>

<p>Modern sequencing technologies can generate a massive number of sequence reads in a single experiment. However, no sequencing technology is perfect, and each instrument will generate different types and amount of errors, such as incorrect nucleotides being called. These wrongly called bases are due to the technical limitations of each sequencing platform.</p>

<p>Therefore, it is necessary to understand, identify and exclude error-types that may impact the interpretation of downstream analysis.
Sequence quality control is therefore an essential first step in your analysis. Catching errors early saves time later on.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#inspect-a-raw-sequence-file" id="markdown-toc-inspect-a-raw-sequence-file">Inspect a raw sequence file</a></li>
  <li><a href="#assess-quality-with-fastqe----short-reads-only" id="markdown-toc-assess-quality-with-fastqe----short-reads-only">Assess quality with FASTQE 🧬😎 - short reads only</a></li>
  <li><a href="#assess-quality-with-fastqc---short--long-reads" id="markdown-toc-assess-quality-with-fastqc---short--long-reads">Assess quality with FastQC - short &amp; long reads</a>    <ol>
      <li><a href="#per-base-sequence-quality" id="markdown-toc-per-base-sequence-quality">Per base sequence quality</a></li>
      <li><a href="#per-sequence-quality-scores" id="markdown-toc-per-sequence-quality-scores">Per sequence quality scores</a></li>
      <li><a href="#per-base-sequence-content" id="markdown-toc-per-base-sequence-content">Per base sequence content</a></li>
      <li><a href="#per-sequence-gc-content" id="markdown-toc-per-sequence-gc-content">Per sequence GC content</a></li>
      <li><a href="#sequence-duplication-levels" id="markdown-toc-sequence-duplication-levels">Sequence Duplication Levels</a></li>
      <li><a href="#over-represented-sequences" id="markdown-toc-over-represented-sequences">Over-represented sequences</a></li>
    </ol>
  </li>
  <li><a href="#trim-and-filter---short-reads" id="markdown-toc-trim-and-filter---short-reads">Trim and filter - short reads</a></li>
  <li><a href="#processing-multiple-datasets" id="markdown-toc-processing-multiple-datasets">Processing multiple datasets</a>    <ol>
      <li><a href="#process-paired-end-data" id="markdown-toc-process-paired-end-data">Process paired-end data</a></li>
    </ol>
  </li>
  <li><a href="#assess-quality-with-nanoplot---long-reads-only" id="markdown-toc-assess-quality-with-nanoplot---long-reads-only">Assess quality with Nanoplot - Long reads only</a>    <ol>
      <li><a href="#histogram-of-read-lengths" id="markdown-toc-histogram-of-read-lengths">Histogram of read lengths</a></li>
      <li><a href="#read-lengths-vs-average-read-quality-plot-using-dots" id="markdown-toc-read-lengths-vs-average-read-quality-plot-using-dots">Read lengths vs Average read quality plot using dots</a></li>
    </ol>
  </li>
  <li><a href="#assess-quality-with-pycoqc---nanopore-only" id="markdown-toc-assess-quality-with-pycoqc---nanopore-only">Assess quality with PycoQC - Nanopore only</a>    <ol>
      <li><a href="#basecalled-reads-length" id="markdown-toc-basecalled-reads-length">Basecalled reads length</a></li>
      <li><a href="#basecalled-reads-length-vs-reads-phred-quality" id="markdown-toc-basecalled-reads-length-vs-reads-phred-quality">Basecalled reads length vs reads PHRED quality</a></li>
      <li><a href="#output-over-experiment-time" id="markdown-toc-output-over-experiment-time">Output over experiment time</a></li>
      <li><a href="#channel-activity-over-time" id="markdown-toc-channel-activity-over-time">Channel activity over time</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="inspect-a-raw-sequence-file">Inspect a raw sequence file</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the file <code class="language-plaintext highlighter-rouge">female_oral2.fastq-4143.gz</code> from <a href="https://zenodo.org/record/3977236">Zenodo</a> or from the data library (ask your instructor)
This is a microbiome sample from a snake <span class="citation"><a href="#StJacques2021">Jacques <i>et al.</i> 2021</a></span>.</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3977236/files/female_oral2.fastq-4143.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the imported dataset to <code class="language-plaintext highlighter-rouge">Reads</code>.</p>
    </li>
  </ol>
</blockquote>

<p>We just imported a file into Galaxy. This file is similar to the data we could get directly from a sequencing facility: a <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ file</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-the-fastq-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect the FASTQ file</div>

  <ol>
    <li>Inspect the file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon</li>
  </ol>

</blockquote>

<p>Although it looks complicated (and maybe it is), the FASTQ format is easy to understand with a little decoding.</p>

<p>Each read, representing a fragment of the library, is encoded by 4 lines:</p>

<table>
  <thead>
    <tr>
      <th>Line</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Always begins with <code class="language-plaintext highlighter-rouge">@</code> followed by the information about the read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>The actual nucleic sequence</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Always begins with a <code class="language-plaintext highlighter-rouge">+</code> and contains sometimes the same info in line 1</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Has a string of characters which represent the quality scores associated with each base of the nucleic sequence; must have the same number of characters as line 2</td>
    </tr>
  </tbody>
</table>

<p>So for example, the first sequence in our file is:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M00970:337:000000000-BR5KF:1:1102:17745:1557 1:N:0:CGCAGAAC+ACAGAGTT
GTGCCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCCGAGCCCACGAGACCGAAGAACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAGAAGCAAATGACGATTCAAGAAAGAAAAAAACACAGAATACTAACAATAAGTCATAAACATCATCAACATAAAAAAGGAAATACACTTACAACACATATCAATATCTAAAATAAATGATCAGCACACAACATGACGATTACCACACATGTGTACTACAAGTCAACTA
+
GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGFGGGGGGAFFGGFGGGGGGGGFGGGGGGGGGGGGGGFGGG+38+35*311*6,,31=******441+++0+0++0+*1*2++2++0*+*2*02*/***1*+++0+0++38++00++++++++++0+0+2++*+*+*+*+*****+0**+0**+***+)*.***1**//*)***)/)*)))*)))*),)0(((-((((-.(4(,,))).,(())))))).)))))))-))-(
</code></pre></div></div>

<p>It means that the fragment named <code class="language-plaintext highlighter-rouge">@M00970</code> corresponds to the DNA sequence <code class="language-plaintext highlighter-rouge">GTGCCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCCGAGCCCACGAGACCGAAGAACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAGAAGCAAATGACGATTCAAGAAAGAAAAAAACACAGAATACTAACAATAAGTCATAAACATCATCAACATAAAAAAGGAAATACACTTACAACACATATCAATATCTAAAATAAATGATCAGCACACAACATGACGATTACCACACATGTGTACTACAAGTCAACTA</code> and this sequence has been sequenced with a quality <code class="language-plaintext highlighter-rouge">GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGFGGGGGGAFFGGFGGGGGGGGFGGGGGGGGGGGGGGFGGG+38+35*311*6,,31=******441+++0+0++0+*1*2++2++0*+*2*02*/***1*+++0+0++38++00++++++++++0+0+2++*+*+*+*+*****+0**+0**+***+)*.***1**//*)***)/)*)))*)))*),)0(((-((((-.(4(,,))).,(())))))).)))))))-))-(</code>.</p>

<!--SNIPPET-->
<blockquote class="details">   <div class="box-title details-title" id="details-quality-scores"><button class="gtn-boxify-button details" type="button" aria-controls="details-quality-scores" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> Details: Quality Scores<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>But what does this quality score mean?</p>   <p>The quality score for each sequence is a string of characters, one for each base of the nucleotide sequence, used to characterize the probability of misidentification of each base. The score is encoded using the ASCII character table (with <a href="https://en.wikipedia.org/wiki/FASTQ_format#Encoding">some historical differences</a>):</p>   <p>To save space, the sequencer records an <a href="http://drive5.com/usearch/manual/quality_score.html">ASCII character</a> to represent scores 0-42. For example 10 corresponds to “+” and 40 corresponds to “I”. FastQC knows how to translate this. This is often called “Phred” scoring.</p>   <p><img src="/training-material/topics/sequence-analysis/faqs/images/fastq-quality-encoding.png" alt="Encoding of the quality score with ASCII characters for different Phred encoding. The ascii code sequence is shown at the top with symbols for 33 to 64, upper case letters, more symbols, and then lowercase letters. Sanger maps from 33 to 73 while solexa is shifted, starting at 59 and going to 104. Illumina 1.3 starts at 54 and goes to 104, Illumina 1.5 is shifted three scores to the right but still ends at 104. Illumina 1.8+ goes back to the Sanger except one single score wider. Illumina" /></p>   <p>So there is an ASCII character associated with each nucleotide, representing its <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred quality score</a>, the probability of an incorrect base call:</p>   <table>   <thead>     <tr>       <th>Phred Quality Score</th>       <th>Probability of incorrect base call</th>       <th>Base call accuracy</th>     </tr>   </thead>   <tbody>     <tr>       <td>10</td>       <td>1 in 10</td>       <td>90%</td>     </tr>     <tr>       <td>20</td>       <td>1 in 100</td>       <td>99%</td>     </tr>     <tr>       <td>30</td>       <td>1 in 1000</td>       <td>99.9%</td>     </tr>     <tr>       <td>40</td>       <td>1 in 10,000</td>       <td>99.99%</td>     </tr>     <tr>       <td>50</td>       <td>1 in 100,000</td>       <td>99.999%</td>     </tr>     <tr>       <td>60</td>       <td>1 in 1,000,000</td>       <td>99.9999%</td>     </tr>   </tbody> </table>   <p>What does 0-42 represent? These numbers, when plugged into a formula, tell us the probability of an error for that base. This is the formula, where Q is our quality score (0-42) and P is the probability of an error:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Q = -10 log10(P)<br /></code></pre></div></div>   <p>Using this formula, we can calculate that a quality score of 40 means only 0.00010 probability of an error!</p> </blockquote>
<p><!--END_SNIPPET--></p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Which ASCII character corresponds to the worst Phred score for Illumina 1.8+?</li>
    <li>What is the Phred quality score of the 3rd nucleotide of the 1st sequence?</li>
    <li>What is the accuracy of this 3rd nucleotide?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The worst Phred score is the smallest one, so 0. For Illumina 1.8+, it corresponds to the <code class="language-plaintext highlighter-rouge">!</code> character.</li>
      <li>The 3rd nucleotide of the 1st sequence has a ASCII character <code class="language-plaintext highlighter-rouge">G</code>, which correspond to a score of 38.</li>
      <li>The corresponding nucleotide <code class="language-plaintext highlighter-rouge">G</code> has an accuracy of almost 99.99%</li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>The current lllumina (1.8+) uses Sanger format (Phred+33). If you are working with older datasets you may encounter the older scoring schemes. <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, a tool we will use later in this tutorial, can be used to try to determine what type of quality encoding is used (through assessing the range of Phred values seen in the FASTQ).</p>
</blockquote>

<p>When looking at the file in Galaxy, it looks like most the nucleotides have a high score (<code class="language-plaintext highlighter-rouge">G</code> corresponding to a score 38). Is it true for all sequences? And along the full sequence length?</p>

<h1 id="assess-quality-with-fastqe----short-reads-only">Assess quality with FASTQE 🧬😎 - short reads only</h1>

<p>To take a look at sequence quality along all sequences, we can use <a href="https://fastqe.com/">FASTQE</a>. It is an open-source tool that provides a simple and fun way to quality control raw sequence data and print them as emoji. You can use it to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2" title="FASTQE tool" aria-role="link"><strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2</span></span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQ data”</em>: <code class="language-plaintext highlighter-rouge">Reads</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Score types to show”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<p>Rather than looking at quality scores for each individual read, FASTQE looks at quality collectively across all reads within a sample and can calculate the mean for each nucleotide position along the length of the reads. Below shows the mean values for this dataset.</p>

<p><a href="../../images/quality-control/fastqe-mean-before.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/fastqe-mean-before.png" alt="FASTQE before. " width="817" height="207" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> FASTQE mean scores&lt;/figcaption&gt;</figure></a></p>

<p>You can see the score for each <a href="https://github.com/fastqe/fastqe#scale">emoji in fastqe’s documentation</a>. The emojis below, with Phred scores less than 20, are the ones we hope we don’t see much.</p>

<table>
  <thead>
    <tr>
      <th>Phred Quality Score</th>
      <th>ASCII code</th>
      <th>Emoji</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>!</td>
      <td>🚫</td>
    </tr>
    <tr>
      <td>1</td>
      <td>”</td>
      <td>❌</td>
    </tr>
    <tr>
      <td>2</td>
      <td>#</td>
      <td>👺</td>
    </tr>
    <tr>
      <td>3</td>
      <td>$</td>
      <td>💔</td>
    </tr>
    <tr>
      <td>4</td>
      <td>%</td>
      <td>🙅</td>
    </tr>
    <tr>
      <td>5</td>
      <td>&amp;</td>
      <td>👾</td>
    </tr>
    <tr>
      <td>6</td>
      <td>’</td>
      <td>👿</td>
    </tr>
    <tr>
      <td>7</td>
      <td>(</td>
      <td>💀</td>
    </tr>
    <tr>
      <td>8</td>
      <td>)</td>
      <td>👻</td>
    </tr>
    <tr>
      <td>9</td>
      <td>*</td>
      <td>🙈</td>
    </tr>
    <tr>
      <td>10</td>
      <td>+</td>
      <td>🙉</td>
    </tr>
    <tr>
      <td>11</td>
      <td>,</td>
      <td>🙊</td>
    </tr>
    <tr>
      <td>12</td>
      <td>-</td>
      <td>🐵</td>
    </tr>
    <tr>
      <td>13</td>
      <td>.</td>
      <td>😿</td>
    </tr>
    <tr>
      <td>14</td>
      <td>/</td>
      <td>😾</td>
    </tr>
    <tr>
      <td>15</td>
      <td>0</td>
      <td>🙀</td>
    </tr>
    <tr>
      <td>16</td>
      <td>1</td>
      <td>💣</td>
    </tr>
    <tr>
      <td>17</td>
      <td>2</td>
      <td>🔥</td>
    </tr>
    <tr>
      <td>18</td>
      <td>3</td>
      <td>😡</td>
    </tr>
    <tr>
      <td>19</td>
      <td>4</td>
      <td>💩</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What is the lowest mean score in this dataset?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The lowest score in this dataset is 😿 13.</p>
  </blockquote>
</blockquote>

<h1 id="assess-quality-with-fastqc---short--long-reads">Assess quality with FastQC - short &amp; long reads</h1>

<p>An additional or alternative way we can check sequence quality is with <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>. It provides a modular set of analyses which you can use to check whether your data has any problems of which you should be aware before doing any further analysis.  We can use it, for example, to assess whether there are known adapters present in the data. We’ll run it on the FASTQ file.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FASTQC tool" aria-role="link"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Raw read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">Reads</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Which Phred encoding is used in the FASTQ file for these sequences?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The Phred scores are encoded using <code class="language-plaintext highlighter-rouge">Sanger / Illumina 1.9</code> (<code class="language-plaintext highlighter-rouge">Encoding</code> in the top table).</p>
  </blockquote>
</blockquote>

<h2 id="per-base-sequence-quality">Per base sequence quality</h2>

<p>With FastQC we can use the per base sequence quality plot to check the base quality of the reads, similar to what we did with FASTQE.</p>

<p><a href="../../images/quality-control/per_base_sequence_quality-before.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_base_sequence_quality-before.png" alt="Per base sequence quality. " width="1005" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Per base sequence quality&lt;/figcaption&gt;</figure></a></p>

<p>On the x-axis are the base position in the read. In this example, the sample contains reads that are up to 296 bp long.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-non-uniform-x-axis"><button class="gtn-boxify-button details" type="button" aria-controls="details-non-uniform-x-axis" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Non uniform x-axis<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The x-axis is not always uniform. When you have long reads, some binning is applied to keep things compact. We can see that in our sample. It starts out with individual 1-10 bases. After that, bases are binned across a window a certain number of bases wide. Data binning means grouping and is a data pre-processing technique used to reduce the effects of minor observation errors. The number of base positions binned together depends on the length of the read. With reads &gt;50bp, the latter part of the plot will report aggregate statistics for 5bp windows. Shorter reads will have smaller windows and longer reads larger windows. Binning can be removed when running FastQC by setting the paramter “Disable grouping of bases for reads &gt;50bp” to Yes.</p>
</blockquote>

<p>For each position, a boxplot is drawn with:</p>

<ul>
  <li>the median value, represented by the central red line</li>
  <li>the inter-quartile range (25-75%), represented by the yellow box</li>
  <li>the 10% and 90% values in the upper and lower whiskers</li>
  <li>the mean quality, represented by the blue line</li>
</ul>

<p>The y-axis shows the quality scores. The higher the score, the better the base call. The background of the graph divides the y-axis into very good quality scores (green), scores of reasonable quality (orange), and reads of poor quality (red).</p>

<p>It is normal with all Illumina sequencers for the median quality score to start out lower over the first 5-7 bases and to then rise. The quality of reads on most platforms will drop at the end of the read. This is often due to signal decay or phasing during the sequencing run. The recent developments in chemistry applied to sequencing has improved this somewhat, but reads are now longer than ever.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-signal-decay-and-phasing"><button class="gtn-boxify-button details" type="button" aria-controls="details-signal-decay-and-phasing" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Signal decay and phasing<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>Signal decay</li>
  </ul>

  <p>The fluorescent signal intensity decays with each cycle of the sequencing process. Due to the degrading fluorophores, a proportion of the strands in the cluster are not being elongated. The proportion of the signal being emitted continues to decrease with each cycle, yielding to a decrease of quality scores at the 3’ end of the read.</p>

  <ul>
    <li>Phasing</li>
  </ul>

  <p>The signal starts to blur with the increase of number of cycles because the cluster looses synchronicity. As the cycles progress, some strands get random failures of nucleotides to incorporate due to:</p>

  <ul>
    <li>Incomplete removal of the 3’ terminators and fluorophores</li>
    <li>Incorporation of nucleotides without effective 3’ terminators</li>
  </ul>

  <p>This leads to a decrease in quality scores at the 3’ end of the read.</p>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-sequence-quality-profiles"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-sequence-quality-profiles" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other sequence quality profiles<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>These are some per base sequence quality profiles that can indicate issues with the sequencing.</p>

  <ul>
    <li>
      <p>Overclustering</p>

      <p>Sequencing facilities can overcluster the flow cells. It results in small distances between clusters and an overlap in the signals. Two clusters can be interpreted as a single cluster with mixed fluorescent signals being detected, decreasing signal purity. It generates lower quality scores across the entire read.</p>
    </li>
    <li>
      <p>Instrumentation breakdown</p>

      <p>Some issues can occasionally happen with the sequencing instruments during a run. Any sudden drop in quality or a large percentage of low quality reads across the read could indicate a problem at the facility. Some examples of such issues:</p>

      <ul>
        <li>
          <p>Manifold burst</p>

          <p><a href="../../images/quality-control/per_base_sequence_quality_manifold_burst.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_quality_manifold_burst.png" alt="Manifold burst. " width="737" height="551" loading="lazy" /></a></p>
        </li>
        <li>
          <p>Cycles loss</p>

          <p><a href="../../images/quality-control/per_base_sequence_quality_cycle_loss.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_quality_cycle_loss.png" alt="Cycles loss. " width="1010" height="764" loading="lazy" /></a></p>
        </li>
        <li>
          <p>Read 2 failure</p>

          <p><a href="../../images/quality-control/per_base_sequence_quality_read2_failure.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_quality_read2_failure.png" alt="Cycles loss. " width="535" height="379" loading="lazy" /></a></p>
        </li>
      </ul>

      <p>With such data, the sequencing facility should be contacted for discussion. Often, a resequencing then is needed (and from our experience also offered by the company).</p>
    </li>
  </ul>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How does the mean quality score change along the sequence?</li>
    <li>Is this tendency seen in all sequences?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The mean quality score (blue line) drops about midway though these sequences. It is common for the mean quality to drop towards the end of the sequences, as the sequencers are incorporating more incorrect nucleotides at the end. However, in this sample there is a very large drop in quality from the middle onwards.</li>
      <li>The box plots are getting wider from position ~100.  It means a lot of sequences have their score dropping from the middle of the sequence. After 100 nucleotides, more than 10% of the sequences have scores below 20.</li>
    </ol>

  </blockquote>
</blockquote>

<p>When the median quality is below a Phred score of ~20, we should consider trimming away bad quality bases from the sequence. We will explain that process in the Trim and filter section.</p>

<h3 id="adapter-content">Adapter Content</h3>

<p><a href="../../images/quality-control/adapter_content-before.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/adapter_content-before.png" alt="Adapter Content. " width="975" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Adapter Content&lt;/figcaption&gt;</figure></a></p>

<p>The plot shows the cumulative percentage of reads with the different adapter sequences at each position. Once an adapter sequence is seen in a read it  is counted as being present right through to the end of the read so the percentage increases with the read length. FastQC can detect some adapters by default (e.g. Illumina, Nextera), for others we could provide a contaminants file as an input to the FastQC tool.</p>

<p>Ideally Illumina sequence data should not have any adapter sequence present. But with long reads, some of the library inserts are shorter than the read length resulting in read-through to the adapter at the 3’ end of the read. This microbiome sample has relatively long reads and we can see Nextera dapater has been detected.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-adapter-content-profiles"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-adapter-content-profiles" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other adapter content profiles<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Adapter content may also be detected with RNA-Seq libraries where the distribution of library insert sizes is varied and likely to include some short inserts.</p>

  <p><a href="../../images/quality-control/adapter_content_rna_seq.png" rel="noopener noreferrer"><img src="../../images/quality-control/adapter_content_rna_seq.png" alt="Adapter Content. " width="1110" height="600" loading="lazy" /></a></p>

</blockquote>

<p>We can run an trimming tool such as Cutadapt to remove this adapter. We will explain that process in the filter and trim section.</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-take-a-shortcut"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-take-a-shortcut" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Take a shortcut<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The following sections go into detail about some of the other plots generated by FastQC. Note that some plots/modules may give warnings but be normal
for the type of data you’re working with, as discussed below and <a href="https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/">in the FASTQC FAQ</a>.
The other plots give us information to more deeply understand the quality of the data, and to see if changes could be made in the lab to get higher-quality data in the future.
These sections are <strong>optional</strong>, and if you would like to skip these you can:</p>
  <ul>
    <li>Jump straight to the <a href="#trim-and-filter---short-reads">next section</a> to learn about trimming paired-end data</li>
  </ul>
</blockquote>

<h3 id="per-tile-sequence-quality">Per tile sequence quality</h3>

<p>This plot enables you to look at the quality scores from each tile across all of your bases to see if there was a loss in quality associated with only one part of the flowcell. The plot shows the deviation from the average quality for each flowcell tile. The hotter colours indicate that reads in the given tile have worse qualities for that position than reads in other tiles. With this sample, you can see that certain tiles show consistently poor quality, especially from ~100bp onwards. A good plot should be blue all over.</p>

<p><a href="../../images/quality-control/per_tile_sequence_quality-before.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_tile_sequence_quality-before.png" alt="Per tile sequence quality. " width="1005" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Per tile sequence quality&lt;/figcaption&gt;</figure></a></p>

<p>This plot will only appear for Illumina library which retains its original sequence identifiers. Encoded in these is the flowcell tile from which each read came.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-tile-quality-profiles"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-tile-quality-profiles" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other tile quality profiles<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>In some cases, the chemicals used during sequencing becoming a bit exhausted over the time and the last tiles got worst chemicals which makes the sequencing reactions a bit error-prone. The “Per tile sequence quality” graph will then have some horizontal lines like this:</p>

  <p><a href="../../images/quality-control/per_tile_sequence_quality_horizontal_lines.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_tile_sequence_quality_horizontal_lines.png" alt="Per tile sequence quality with horizontal lines. " width="1515" height="600" loading="lazy" /></a></p>

</blockquote>

<h2 id="per-sequence-quality-scores">Per sequence quality scores</h2>

<p>It plots the average quality score over the full length of all reads on the x-axis and gives the total number of reads with this score on the y-axis:</p>

<p><a href="../../images/quality-control/per_sequence_quality_scores-before.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_sequence_quality_scores-before.png" alt="Per sequence quality scores. " width="800" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Per sequence quality scores&lt;/figcaption&gt;</figure></a></p>

<p>The distribution of average read quality should be tight peak in the upper range of the plot. It can also report if a subset of the sequences have universally low quality values: it can happen because some sequences are poorly imaged (on the edge of the field of view etc), however these should represent only a small percentage of the total sequences.</p>

<h2 id="per-base-sequence-content">Per base sequence content</h2>

<p><a href="../../images/quality-control/per_base_sequence_content-before.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_base_sequence_content-before.png" alt="Per base sequence content. " width="1005" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Per base sequence content for a DNA library&lt;/figcaption&gt;</figure></a></p>

<p>“Per Base Sequence Content” plots the percentage of each of the four nucleotides (T, C, A, G) at each position across all reads in the input sequence file. As for the per base sequence quality, the x-axis is non-uniform.</p>

<p>In a random library we would expect that there would be little to no difference between the four bases. The proportion of each of the four bases should remain relatively constant over the length of the read with <code class="language-plaintext highlighter-rouge">%A=%T</code> and <code class="language-plaintext highlighter-rouge">%G=%C</code>, and the lines in this plot should run parallel with each other. This is amplicon data, where 16S DNA is PCR amplified and sequenced, so we’d expect this plot to have some bias and not show a random distribution.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-biases-by-library-type"><button class="gtn-boxify-button details" type="button" aria-controls="details-biases-by-library-type" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Biases by library type<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>It’s worth noting that some library types will always produce biased sequence composition, normally at the start of the read. Libraries produced by priming using random hexamers (including nearly all RNA-Seq libraries), and those which were fragmented using transposases, will contain an intrinsic bias in the positions at which reads start (the first 10-12 bases). This bias does not involve a specific sequence, but instead provides enrichment of a number of different K-mers at the 5’ end of the reads. Whilst this is a true technical bias, it isn’t something which can be corrected by trimming and in most cases doesn’t seem to adversely affect the downstream analysis. It will, however, produce a warning or error in this module.</p>

  <p><a href="../../images/quality-control/per_base_sequence_content_rnaseq.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_content_rnaseq.png" alt="Per base sequence content for RNA-seq data. " width="800" height="600" loading="lazy" /></a></p>

  <p>ChIP-seq data can also encounter read start sequence biases in this plot if fragmenting with transposases. With bisulphite converted data, e.g. HiC data, a separation of G from C and A from T is expected:</p>

  <p><a href="../../images/quality-control/per_base_sequence_content_bisulphite.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_content_bisulphite.png" alt="Per base sequence content for Bisulphite data. " width="1305" height="600" loading="lazy" /></a></p>

  <p>At the end, there is an overall shift in the sequence composition. If the shift correlates with a loss of sequencing quality, it can be suspected that miscalls are made with a more even sequence bias than bisulphite converted libraries.  Trimming the sequences fixed this problem, but if this hadn’t been done it would have had a dramatic effect on the methylation calls which were made.</p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Why is there a warning for the per-base sequence content graphs?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>In the beginning of sequences, the sequence content per base is not really good and the percentages are not equal, as expected for 16S amplicon data.</p>

  </blockquote>
</blockquote>

<h2 id="per-sequence-gc-content">Per sequence GC content</h2>

<p><a href="../../images/quality-control/per_sequence_gc_content-before.png" rel="noopener noreferrer"><figure id="figure-7" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_sequence_gc_content-before.png" alt="Per sequence GC content. " width="800" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Per sequence GC content&lt;/figcaption&gt;</figure></a></p>

<p>This plot displays the number of reads vs. percentage of bases G and C per read. It is compared to a theoretical distribution assuming an uniform GC content for all reads, expected for whole genome shotgun sequencing, where the central peak corresponds to the overall GC content of the underlying genome. Since the GC content of the genome is not known, the modal GC content is calculated from the observed data and used to build a reference distribution.</p>

<p>An unusually-shaped distribution could indicate a contaminated library or some other kind of biased subset. A shifted normal distribution indicates some systematic bias, which is independent of base position. If there is a systematic bias which creates a shifted normal distribution then this won’t be flagged as an error by the module since it doesn’t know what your genome’s GC content should be.</p>

<p>But there are also other situations in which an unusually-shaped distribution may occur. For example, with RNA sequencing there may be a greater or lesser distribution of mean GC content among transcripts causing the observed plot to be wider or narrower than an ideal normal distribution.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Why is there a fail for the per sequence GC content graphs?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>There are multiple peaks. This can be indicative of unexpected contamination, such as adapter, rRNA or overrepresented sequences. Or it may be normal if it is amplicon data or you have highly abundant RNA-seq transcripts.</p>
  </blockquote>
</blockquote>

<h3 id="sequence-length-distribution">Sequence length distribution</h3>

<p>This plot shows the distribution of fragment sizes in the file which was analysed. In many cases this will produce a simple plot showing a peak only at one size, but for variable length FASTQ files this will show the relative amounts of each different size of sequence fragment. Our plot shows variable length as we trimmed the data. The biggest peak is at 296bp but there is a second large peak at ~100bp. So even though our sequences range up to 296bp in length, a lot of the good-quality sequences are shorter. This corresponds with the drop we saw in the sequence quality at ~100bp and the red stripes starting at this position in the per tile sequence quality plot.</p>

<p><a href="../../images/quality-control/sequence_length_distribution-before.png" rel="noopener noreferrer"><figure id="figure-8" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/sequence_length_distribution-before.png" alt="Sequence length distribution. " width="800" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Sequence length distribution&lt;/figcaption&gt;</figure></a></p>

<p>Some high-throughput sequencers generate sequence fragments of uniform length, but others can contain reads of widely varying lengths. Even within uniform length libraries some pipelines will trim sequences to remove poor quality base calls from the end or the first \(n\) bases if they match the first \(n\) bases of the adapter up to 90% (by default), with sometimes \(n = 1\).</p>

<h2 id="sequence-duplication-levels">Sequence Duplication Levels</h2>

<p>The graph shows in blue the percentage of reads of a given sequence in the file which are present a given number of times in the file:</p>

<p><a href="../../images/quality-control/sequence_duplication_levels-before.png" rel="noopener noreferrer"><figure id="figure-9" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/sequence_duplication_levels-before.png" alt="Sequence Duplication Levels. " width="800" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 9</strong>:</span> Sequence Duplication Levels&lt;/figcaption&gt;</figure></a></p>

<p>In a diverse library most sequences will occur only once in the final set. A low level of duplication may indicate a very high level of coverage of the target sequence, but a high level of duplication is more likely to indicate some kind of enrichment bias.</p>

<p>Two sources of duplicate reads can be found:</p>
<ul>
  <li>
    <p>PCR duplication in which library fragments have been over-represented due to biased PCR enrichment</p>

    <p>It is a concern because PCR duplicates misrepresent the true proportion of sequences in the input.</p>
  </li>
  <li>
    <p>Truly over-represented sequences such as very abundant transcripts in an RNA-Seq library or in amplicon data (like this sample)</p>

    <p>It is an expected case and not of concern because it does faithfully represent the input.</p>
  </li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-details-about-duplication"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-details-about-duplication" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: More details about duplication<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>FastQC counts the degree of duplication for every sequence in a library and creates a plot showing the relative number of sequences with different degrees of duplication. There are two lines on the plot:</p>
  <ul>
    <li>Blue line: distribution of the duplication levels for the full sequence set</li>
    <li>Red line: distribution for the de-duplicated sequences with the proportions of the deduplicated set which come from different duplication levels in the original data.</li>
  </ul>

  <p>For whole genome shotgun data it is expected that nearly 100% of your reads will be unique (appearing only 1 time in the sequence data). Most sequences should fall into the far left of the plot in both the red and blue lines. This indicates a highly diverse library that was not over sequenced. If the sequencing depth is extremely high (e.g. &gt; 100x the size of the genome) some inevitable sequence duplication can appear: there are in theory only a finite number of completely unique sequence reads which can be obtained from any given input DNA sample.</p>

  <p>More specific enrichments of subsets, or the presence of low complexity contaminants will tend to produce spikes towards the right of the plot. These high duplication peaks will most often appear in the blue trace as they make up a high proportion of the original library, but usually disappear in the red trace as they make up an insignificant proportion of the deduplicated set. If peaks persist in the red trace then this suggests that there are a large number of different highly duplicated sequences which might indicate either a contaminant set or a very severe technical duplication.</p>

  <p>It is usually the case for RNA sequencing where there is some very highly abundant transcripts and some lowly abundant. It is expected that duplicate reads will be observed for high abundance transcripts:</p>

  <p><a href="../../images/quality-control/sequence_duplication_levels_rna_seq.png" rel="noopener noreferrer"><img src="../../images/quality-control/sequence_duplication_levels_rna_seq.png" alt="Sequence Duplication Levels for RNA-seq. " width="800" height="600" loading="lazy" /></a></p>

</blockquote>

<h2 id="over-represented-sequences">Over-represented sequences</h2>

<p>A normal high-throughput library will contain a diverse set of sequences, with no individual sequence making up a tiny fraction of the whole. Finding that a single sequence is very over-represented in the set either means that it is highly biologically significant, or indicates that the library is contaminated, or not as diverse as expected.</p>

<p>FastQC lists all of the sequence which make up more than 0.1% of the total. For each over-represented sequence FastQC will look for matches in a database of common contaminants and will report the best hit it finds. Hits must be at least 20bp in length and have no more than 1 mismatch. Finding a hit doesn’t necessarily mean that this is the source of the contamination, but may point you in the right direction. It’s also worth pointing out that many adapter sequences are very similar to each other so you may get a hit reported which isn’t technically correct, but which has a very similar sequence to the actual match.</p>

<p>RNA sequencing data may have some transcripts that are so abundant that they register as over-represented sequence. With DNA sequencing data no single sequence should be present at a high enough frequency to be listed, but we can sometimes see a small percentage of adapter reads.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How could we find out what the overrepreseented sequences are?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>We can BLAST overrepresented sequences to see what they are. In this case, if we take the top overrepresented sequence</p>
    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;overrep_seq1
GTGTCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCC
</code></pre></div>    </div>
    <p>and use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">blastn</a> against the default Nucleotide (nr/nt) database we don’t get any hits. But if we use <a href="https://www.ncbi.nlm.nih.gov/tools/vecscreen/">VecScreen</a> we see it is the Nextera adapter.
<a href="../../images/quality-control/vecscreen-nextera.png" rel="noopener noreferrer"><figure id="figure-10" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/vecscreen-nextera.png" alt="VecScreen. " width="976" height="310" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 10</strong>:</span> Nextera adapter&lt;/figcaption&gt;</figure></a></p>
  </blockquote>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-details-about-other-fastqc-plots"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-details-about-other-fastqc-plots" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: More details about other FastQC plots<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <h4 id="per-base-n-content">Per base N content</h4>

  <p><a href="../../images/quality-control/per_base_n_content-before.png" rel="noopener noreferrer"><figure id="figure-11" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_base_n_content-before.png" alt="Per base N content. " width="1005" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 11</strong>:</span> Per base N content&lt;/figcaption&gt;</figure></a></p>

  <p>If a sequencer is unable to make a base call with sufficient confidence, it will write an “N” instead of a conventional base call. This plot displays the percentage of base calls at each position or bin for which an N was called.</p>

  <p>It’s not unusual to see a very high proportion of Ns appearing in a sequence, especially near the end of a sequence. But this curve should never rises noticeably above zero. If it does this indicates a problem occurred during the sequencing run. In the example below, an error caused the instrument to be unable to call a base for approximately 20% of the reads at position 29:</p>

  <p><a href="../../images/quality-control/per_base_n_content_error.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_n_content_error.png" alt="Per base N content. " width="800" height="600" loading="lazy" /></a></p>

  <h4 id="kmer-content">Kmer Content</h4>

  <p>This plot not output by default. As stated in the tool form, if you want this module it needs to be enabled using a custom Submodule and limits file. With this module, FastQC does a generic analysis of all of the short nucleotide sequences of length k (kmer, with k = 7 by default) starting at each position along the read in the library to find those which do not have an even coverage through the length of your reads. Any given kmer should be evenly represented across the length of the read.</p>

  <p>FastQC will report the list of kmers which appear at specific positions with a greater frequency than expected. This can be due to different sources of bias in the library, including the presence of read-through adapter sequences building up on the end of the sequences. The presence of any overrepresented sequences in the library (such as adapter dimers) causes the kmer plot to be dominated by the kmer from these sequences. Any biased kmer due to other interesting biases may be then diluted and not easy to see.</p>

  <p>The following example is from a high-quality DNA-Seq library. The biased kmers nearby the start of the read likely are due to slight sequence dependent efficiency of DNA shearing or a result of random priming:</p>

  <p><a href="../../images/quality-control/kmer_content.png" rel="noopener noreferrer"><figure id="figure-12" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/kmer_content.png" alt="Kmer Content. " width="800" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 12</strong>:</span> Kmer content&lt;/figcaption&gt;</figure></a></p>

  <p>This module can be very difficult to interpret. The adapter content plot and overrepesented sequences table are easier to interpret and may give you enough information without needing this plot. RNA-seq libraries may have highly represented kmers that are derived from highly expressed sequences. To learn more about this plot, please check the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html">FastQC Kmer Content documentation</a>.</p>

</blockquote>

<p>We tried to explain here there different FastQC reports and some use cases. More about this and also some common next-generation sequencing problems can be found on <a href="https://sequencing.qcfail.com/">QCFAIL.com</a></p>

<blockquote class="details">
  <div class="box-title details-title" id="details-specific-problem-for-alternate-library-types"><button class="gtn-boxify-button details" type="button" aria-controls="details-specific-problem-for-alternate-library-types" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Specific problem for alternate library types<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <h4 id="smallmicro-rna">Small/micro RNA</h4>

  <p>In small RNA libraries, we typically have a relatively small set of unique, short sequences. Small RNA libraries are not randomly sheared before adding sequencing adapters to their ends: all the reads for specific classes of microRNAs will be identical. It will result in:</p>

  <ul>
    <li>Extremely biased per base sequence content</li>
    <li>Extremely narrow distribution of GC content</li>
    <li>Very high sequence duplication levels</li>
    <li>Abundance of overrepresented sequences</li>
    <li>Read-through into adapters</li>
  </ul>

  <h4 id="amplicon">Amplicon</h4>

  <p>Amplicon libraries are prepared by PCR amplification of a specific target. For example, the V4 hypervariable region of the bacterial 16S rRNA gene. All reads from this type of library are expected to be nearly identical. It will result in:</p>

  <ul>
    <li>Extremely biased per base sequence content</li>
    <li>Extremely narrow distribution of GC content</li>
    <li>Very high sequence duplication levels</li>
    <li>Abundance of overrepresented sequences</li>
  </ul>

  <h4 id="bisulfite-or-methylation-sequencing">Bisulfite or Methylation sequencing</h4>

  <p>With Bisulfite or methylation sequencing, the majority of the cytosine (C) bases are converted to thymine (T). It will result in:</p>

  <ul>
    <li>Biased per base sequence content</li>
    <li>Biased per sequence GC content</li>
  </ul>

  <h4 id="adapter-dimer-contamination">Adapter dimer contamination</h4>

  <p>Any library type may contain a very small percentage of adapter dimer (i.e. no insert) fragments. They are more likely to be found in amplicon libraries constructed entirely by PCR (by formation of PCR primer-dimers) than in DNA-Seq or RNA-Seq libraries constructed by adapter ligation. If a sufficient fraction of the library is adapter dimer it will become noticeable in the FastQC report:</p>

  <ul>
    <li>Drop in per base sequence quality after base 60</li>
    <li>Possible bi-modal distribution of per sequence quality scores</li>
    <li>Distinct pattern observed in per bases sequence content up to base 60</li>
    <li>Spike in per sequence GC content</li>
    <li>Overrepresented sequence matching adapter</li>
    <li>Adapter content &gt; 0% starting at base 1</li>
  </ul>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-bad-quality-sequences"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Bad quality sequences</div>
  <p>If the quality of the reads is not good, we should always first check what is wrong and think about it: it may come from the type of sequencing or what we sequenced (high quantity of overrepresented sequences in transcriptomics data, biased percentage of bases in HiC data).</p>

  <p>You can also ask the sequencing facility about it, especially if the quality is really bad: the quality treatments can not solve everything. If too many bad quality bases are cut away, the corresponding reads then will be filtered out and you lose them.</p>
</blockquote>

<h1 id="trim-and-filter---short-reads">Trim and filter - short reads</h1>

<p>The quality drops in the middle of these sequences. This could cause bias in downstream analyses with these potentially incorrectly called nucleotides. Sequences must be treated to reduce bias in downstream analysis. Trimming can help to increase the number of reads the aligner or assembler are able to succesfully use, reducing the number of reads that are unmapped or unassembled. In general, quality treatments include:</p>

<ol>
  <li>Trimming/cutting/masking sequences
    <ul>
      <li>from low quality score regions</li>
      <li>beginning/end of sequence</li>
      <li>removing adapters</li>
    </ul>
  </li>
  <li>Filtering of sequences
    <ul>
      <li>with low mean quality score</li>
      <li>too short</li>
      <li>with too many ambiguous (N) bases</li>
    </ul>
  </li>
</ol>

<p>To accomplish this task we will use <a href="https://cutadapt.readthedocs.io/en/stable/guide.html">Cutadapt</a> <span class="citation"><a href="#marcel2011cutadapt">Marcel 2011</a></span>, a tool that enhances sequence quality by automating adapter trimming as well as quality control.  We will:</p>

<ul>
  <li>Trim low-quality bases from the ends. Quality trimming is done before any adapter trimming. We will set the quality threshold as 20, a commonly used threshold, see more <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531872-Phred-scaled-quality-scores">in GATK’s Phred Score FAQ</a>.</li>
  <li>Trim adapter with Cutadapt. For that we need to supply the sequence of the adapter. In this sample, Nextera is the adapter that was detected. We can find the sequence of the Nextera adapter on the <a href="https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html">Illumina website here</a> <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCT</code>. We will trim that sequence from the 3’ end of the reads.</li>
  <li>Filter out sequences with length &lt; 20 after trimming</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-improvement-of-sequence-quality"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Improvement of sequence quality</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy2" title="Cutadapt tool" aria-role="link"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy2</span></span> with the following parameters
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li>
              <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reads in FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">Reads</code> (Input dataset)</p>

              <blockquote class="tip">
                <div class="box-title tip-title" id="tip-files-not-selectable"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-files-not-selectable" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Files not selectable?<span class="fold-unfold fa fa-minus-square"></span></button></div>
                <p>If your FASTQ file cannot be selected, you might check whether the format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>). You can edit the data type by clicking on the pencil symbol.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>In <em>“Read 1 Options”</em>:
          <ul>
            <li><em>“Insert 3’ (End) Adapters”</em>:
              <ul>
                <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code></li>
                <li><em>“Enter custom 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCT</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>
          <ul>
            <li><em>“Minimum length”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>
          <ul>
            <li><em>“Quality cutoff”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Outputs selector”</em>: <code class="language-plaintext highlighter-rouge">Report</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">Report</code>)</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What % reads contain adapter?</li>
          <li>What % reads have been trimmed because of bad quality?</li>
          <li>What % reads have been removed because they were too short?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>56.8% reads contain adapter (<code class="language-plaintext highlighter-rouge">Reads with adapters:</code>)</li>
            <li>35.1% reads have been trimmed because of bad quality (<code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>)</li>
            <li>0 % reads were removed because they were too short</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-trimming-with-cutadapt"><button class="gtn-boxify-button details" type="button" aria-controls="details-trimming-with-cutadapt" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Trimming with Cutadapt<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>One of the biggest advantage of Cutadapt compared to other trimming tools (e.g. TrimGalore!) is that it has a good <a href="https://cutadapt.readthedocs.io">documentation</a> explaining how the tool works in detail.</p>

  <p>Cutadapt quality trimming algorithm consists of three simple steps:</p>

  <ol>
    <li>Subtract the chosen threshold value from the quality value of each position</li>
    <li>Compute a partial sum of these differences from the end of the sequence to each position
(as long as the partial sum is negative)</li>
    <li>Cut at the minimum value of the partial sum</li>
  </ol>

  <p>In the following example, we assume that the 3’ end is to be quality-trimmed with a threshold of 10 and we have the following quality values</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>42 40 26 27 8 7 11 4 2 3
</code></pre></div>  </div>

  <ol>
    <li>
      <p>Subtract the threshold</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 32 30 16 17 -2 -3 1 -6 -8 -7
</code></pre></div>      </div>
    </li>
    <li>
      <p>Add up the numbers, starting from the 3’ end (partial sums) and stop early if the sum is greater than zero</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> (70) (38) 8 -8 -25 -23 -20, -21 -15 -7
</code></pre></div>      </div>

      <p>The numbers in parentheses are not computed (because 8 is greater than zero), but shown here for completeness.</p>
    </li>
    <li>
      <p>Choose the position of the minimum (<code class="language-plaintext highlighter-rouge">-25</code>) as the trimming position</p>
    </li>
  </ol>

  <p>Therefore, the read is trimmed to the first four bases, which have quality values</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>42 40 26 27
</code></pre></div>  </div>

  <p>Note that therefore, positions with a quality value larger than the chosen threshold are also removed if they are embedded in regions with lower quality (the partial sum is decreasing if the quality values are smaller than the threshold). The advantage of this procedure is that it is robust against a small number of positions with a quality higher than the threshold.</p>

  <p>Alternatives to this procedure would be:</p>

  <ul>
    <li>Cut after the first position with a quality smaller than the threshold</li>
    <li>
      <p>Sliding window approach</p>

      <p>The sliding window approach checks that the average quality of each sequence window of specified length is larger than the threshold. Note that in contrast to cutadapt’s approach, this approach has one more parameter and the robustness depends of the length of the window (in combination with the quality threshold). Both approaches are implemented in Trimmomatic.</p>
    </li>
  </ul>
</blockquote>

<p>We can examine our trimmed data with FASTQE and/or FastQC.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-checking-quality-after-trimming"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Checking quality after trimming</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2" title="FASTQE tool" aria-role="link"><strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2</span></span>: Re-run <strong>FASTQE</strong> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQ data”</em>: <code class="language-plaintext highlighter-rouge">Cutadapt Read 1 Output</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Score types to show”</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the new FASTQE report</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Compare the FASTQE output to the previous one before trimming above. Has sequence quality been improved?</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-using-the-scratchbook-to-view-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-the-scratchbook-to-view-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Using the Scratchbook to view multiple datasets<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in Galaxy:</p>   <ol>   <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> on the top menu bar.     <ul>       <li>You should see a little checkmark on the icon now</li>     </ul>   </li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> to view the output     <ul>       <li>You should see the output in a window overlayed over Galaxy</li>       <li>You can resize this window by dragging the bottom-right corner</li>     </ul>   </li>   <li><strong>Click</strong> outside the file to exit the Scratchbook</li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a second dataset from your history     <ul>       <li>You should now see a second window with the new dataset</li>       <li>This makes it easier to compare the two outputs</li>     </ul>   </li>   <li>Repeat this for as many files as you would like to compare</li>   <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> by clicking on the icon again</li> </ol> </blockquote>
        <p><!--END_SNIPPET--></p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>Yes, the quality score emojis look better (happier) now.</p>

          <p><a href="../../images/quality-control/fastqe-mean-before.png" rel="noopener noreferrer"><figure id="figure-13" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/fastqe-mean-before.png" alt="FASTQE before. " width="817" height="207" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 13</strong>:</span> Before trimming&lt;/figcaption&gt;</figure></a></p>

          <p><a href="../../images/quality-control/fastqe-mean-after.png" rel="noopener noreferrer"><figure id="figure-14" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/fastqe-mean-after.png" alt="FASTQE after. " width="1420" height="344" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 14</strong>:</span> After trimming&lt;/figcaption&gt;</figure></a></p>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>With FASTQE we can see we improved the quality of the bases in the dataset.</p>

<p>We can also, or instead, check the quality-controlled data with FastQC.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-checking-quality-after-trimming-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Checking quality after trimming</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FASTQC tool" aria-role="link"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">Cutadapt Read 1 Output</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>Does the per base sequence quality look better?</li>
    <li>Is the adapter gone?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>
        <p>Yes. The vast majority of the bases have a quality score above 20 now.
<a href="../../images/quality-control/per_base_sequence_quality-after.png" rel="noopener noreferrer"><figure id="figure-15" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/per_base_sequence_quality-after.png" alt="Per base sequence quality. " width="1005" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Per base sequence quality&lt;/figcaption&gt;</figure></a></p>
      </li>
      <li>
        <p>Yes. No adapter is detected now.
<a href="../../images/quality-control/adapter_content-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/adapter_content-after.png" alt="Adapter Content. " width="960" height="600" loading="lazy" /></a></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>With FastQC we can see we improved the quality of the bases in the dataset and removed the adapter.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-fastqc-plots-after-trimming"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-fastqc-plots-after-trimming" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other FastQC plots after trimming<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><a href="../../images/quality-control/per_tile_sequence_quality-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_tile_sequence_quality-after.png" alt="Per tile sequence quality. " width="1005" height="600" loading="lazy" /></a>
We have some red stripes as we’ve trimmed those regions from the reads.</p>

  <p><a href="../../images/quality-control/per_sequence_quality_scores-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_sequence_quality_scores-after.png" alt="Per sequence quality scores. " width="800" height="600" loading="lazy" /></a>
We now have one peak of high quality instead of one high and one lower quality that we had previously.</p>

  <p><a href="../../images/quality-control/per_base_sequence_content-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_sequence_content-after.png" alt="Per base sequence content. " width="1005" height="600" loading="lazy" /></a>
We don’t have equal representation of the bases as before as this is amplicon data.</p>

  <p><a href="../../images/quality-control/per_sequence_gc_content-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_sequence_gc_content-after.png" alt="Per sequence GC content. " width="800" height="600" loading="lazy" /></a>
We now have a single main GC peak due to removing the adapter.</p>

  <p><a href="../../images/quality-control/per_base_n_content-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/per_base_n_content-after.png" alt="Per base N content. " width="1005" height="600" loading="lazy" /></a>
This is the same as before as we don’t have any Ns in these reads.</p>

  <p><a href="../../images/quality-control/sequence_length_distribution-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/sequence_length_distribution-after.png" alt="Sequence length distribution. " width="800" height="600" loading="lazy" /></a>
We now have multiple peaks and a range of lengths, instead of the single peak with had before trimming when all sequences were the same length.</p>

  <p><a href="../../images/quality-control/sequence_duplication_levels-after.png" rel="noopener noreferrer"><img src="../../images/quality-control/sequence_duplication_levels-after.png" alt="Sequence Duplication Levels. " width="800" height="600" loading="lazy" /></a></p>
  <blockquote class="question">
    <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>What does the top overrepresented sequence <code class="language-plaintext highlighter-rouge">GTGTCAGCCGCCGCGGTAGTCCGACGTGG</code> correspond to?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>If we take the top overrepresented sequence</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;overrep_seq1_after
GTGTCAGCCGCCGCGGTAGTCCGACGTGG
</code></pre></div>      </div>
      <p>and use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">blastn</a> against the default Nucleotide (nr/nt) database we see the top hits are to 16S rRNA genes. This makes sense as this is 16S amplicon data, where the 16S gene is PCR amplified.</p>
    </blockquote>
  </blockquote>
</blockquote>

<h1 id="processing-multiple-datasets">Processing multiple datasets</h1>

<h2 id="process-paired-end-data">Process paired-end data</h2>

<p>With paired-end sequencing, the fragments are sequenced from both sides. This approach results in two reads per fragment, with the first read in forward orientation and the second read in reverse-complement orientation. With this technique, we have the advantage to get more information about each DNA fragment compared to reads sequenced by only single-end sequencing:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    ------&gt;                       [single-end]

    ----------------------------- [fragment]

    ------&gt;               &lt;------ [paired-end]
</code></pre></div></div>
<p>The distance between both reads is known and therefore is additional information that can improve read mapping.</p>

<p>Paired-end sequencing generates 2 FASTQ files:</p>
<ul>
  <li>One file with the sequences corresponding to <strong>forward</strong> orientation of all the fragments</li>
  <li>One file with the sequences corresponding to <strong>reverse</strong> orientation of all the fragments</li>
</ul>

<p>Usually we recognize these two files which belong to one sample by the name which has the same identifier for the reads but a different extension, e.g. <code class="language-plaintext highlighter-rouge">sampleA_R1.fastq</code> for the forward reads and <code class="language-plaintext highlighter-rouge">sampleA_R2.fastq</code> for the reverse reads. It can also be <code class="language-plaintext highlighter-rouge">_f</code> or <code class="language-plaintext highlighter-rouge">_1</code> for the forward reads and <code class="language-plaintext highlighter-rouge">_r</code> or <code class="language-plaintext highlighter-rouge">_2</code> for the reverse reads.</p>

<p>The data we analyzed in the previous step was single-end data so we will import a paired-end RNA-seq dataset to use. We will run FastQC and aggregate the two reports with MultiQC <span class="citation"><a href="#ewels2016multiqc">Ewels <i>et al.</i> 2016</a></span>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-assessing-the-quality-of-paired-end-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assessing the quality of paired-end reads</div>

  <ol>
    <li>
      <p>Import the paired-end reads <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_1.fastq</code> and <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_2.fastq</code> from <a href="https://zenodo.org/record/61771">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_1.fastq
https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_2.fastq
</code></pre></div>      </div>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FASTQC tool" aria-role="link"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span> with both datasets:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Raw read data from your current history”</em>: both the uploaded datasets.</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Select multiple datasets<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1</span></span> with the following parameters to aggregate the FastQC reports of both forward and reverse reads
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code> files (output of both <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the webpage output from MultiQC.</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What do you think about the quality of the sequences?</li>
    <li>What should we do?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The quality of the sequences seems worse for the reverse reads than for the forward reads:
        <ul>
          <li>Per Sequence Quality Scores: distribution more on the left, i.e. a lower mean quality of the sequences</li>
          <li>Per base sequence quality: less smooth curve and stronger decrease at the end with a mean value below 28</li>
          <li>Per Base Sequence Content: stronger bias at the beginning and no clear distinction between C-G and A-T groups</li>
        </ul>

        <p>The other indicators (adapters, duplication levels, etc) are similar.</p>
      </li>
      <li>We should trim the end of the sequences and filter them with <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
    </ol>

  </blockquote>
</blockquote>

<p>With paired-end reads the average quality scores for forward reads will almost always be higher than for reverse reads.</p>

<p>After trimming, reverse reads will be shorter because of their quality and then will be eliminated during the filtering step. If one of the reverse reads is removed, its corresponding forward read should be removed too. Otherwise we will get different number of reads in both files and in different order, and order is important for the next steps. Therefore <strong>it is important to treat the forward and reverse reads together for trimming and filtering</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-improving-the-quality-of-paired-end-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Improving the quality of paired-end data</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy2" title="Cutadapt tool" aria-role="link"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy2</span></span> with the following parameters
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ/A file #1”</em>: <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_1.fastq</code> (Input dataset)</li>
            <li>
              <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTQ/A file #2”</em>: <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_2.fastq</code> (Input dataset)</p>

              <p>The order is important here!</p>
            </li>
            <li>
              <p>In <em>Read 1 Options</em> or <em>Read 2 Options</em></p>

              <p>No adapters were found in these datasets. When you process your own data and you know which adapter sequences were used during library preparation, you should provide their sequences here.</p>
            </li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>
          <ul>
            <li><em>“Minimum length”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>
          <ul>
            <li><em>“Quality cutoff”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">Report</code>)</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-12"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How many basepairs has been removed from the reads because of bad quality?</li>
          <li>How many sequence pairs have been removed because they were too short?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-12"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-12" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>44,164 bp (<code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>) for the forward reads and 138,638 bp for the reverse reads.</li>
            <li>1,376 sequences have been removed because at least one read was shorter than the length cutoff (322 when only the forward reads were analyzed).</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>In addition to the report, Cutadapt generates 2 files:</p>
<ul>
  <li>Read 1 with the trimmed and filtered forward reads</li>
  <li>Read 2 with the trimmed and filtered reverse reads</li>
</ul>

<p>These datasets can be used for the downstream analysis, e.g. mapping.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-13"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What kind of alignment is used for finding adapters in reads?</li>
    <li>What is the criterion to choose the best adapter alignment?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-13"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-13" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Semi-global alignment, i.e., only the overlapping part of the read and the adapter sequence is used for scoring.</li>
      <li>An alignment with maximum overlap is computed that has the smallest number of mismatches and indels.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="assess-quality-with-nanoplot---long-reads-only">Assess quality with Nanoplot - Long reads only</h1>

<p>In case of long reads, we can check sequence quality with <a href="https://github.com/wdecoster/NanoPlot/">Nanoplot</a> (<span class="citation"><a href="#10.1093/bioinformatics/bty149">De Coster <i>et al.</i> 2018</a></span>). It provides basic statistics with nice plots for a fast quality control overview.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check-of-long-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check of long reads</div>
  <ol>
    <li>
      <p>Create a new history for this part and give it a proper name</p>
    </li>
    <li>
      <p>Import the PacBio HiFi reads <code class="language-plaintext highlighter-rouge">m64011_190830_220126.Q20.subsample.fastq.gz</code> from <a href="https://zenodo.org/record/5730295">Zenodo</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/api/files/ff9aa6e3-3d69-451f-9798-7ea69b475989/m64011_190830_220126.Q20.subsample.fastq.gz
</code></pre></div>      </div>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.41.0+galaxy0" title="Nanoplot tool" aria-role="link"><strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.41.0+galaxy0</span></span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“files”</em>: <code class="language-plaintext highlighter-rouge">m64011_190830_220126.Q20.subsample.fastq.gz</code></li>
        <li><em>“Options for customizing the plots created”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify the bivariate format of the plots.”</em>: <code class="language-plaintext highlighter-rouge">dot</code>, <code class="language-plaintext highlighter-rouge">kde</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Show the N50 mark in the read length histogram.”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-14"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What is the mean Qscore ?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-14"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-14" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The Qscore is around Q32.
In case of PacBio CLR and Nanopore, it’s around Q12 and close to Q31 for Illumina (NovaSeq 6000).
<a href="../../images/quality-control/qscore-illumina-pacbio-nanopore.png" rel="noopener noreferrer"><figure id="figure-16" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/qscore-illumina-pacbio-nanopore.png" alt="Plot of Qscore between Illumina, PacBio and Nanopore. " width="638" height="456" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 16</strong>:</span> Comparison of Qscore between Illumina, PacBio and Nanopore&lt;/figcaption&gt;</figure></a></p>

    <p>Definition: Qscores is the average per-base error probability, expressed on the log (Phred) scale</p>
  </blockquote>

  <p>What is the median, mean and N50?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-15"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-15" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The median, the mean read length and the N50 as well are close to 18,000bp.
For PacBio HiFi reads, the majority of the reads are generally near this value as the library preparation includes a size selection step.
For other technologies like PacBio CLR and Nanopore, it is larger and mostly depends on the quality of your DNA extraction.</p>
  </blockquote>
</blockquote>

<h2 id="histogram-of-read-lengths">Histogram of read lengths</h2>

<p>This plot shows the distribution of fragment sizes in the file that was analyzed.
Unlike most of Illumina runs, long reads have a variable length and this will show the relative amounts of each different size of sequence fragment.
In this example, the distribution of read length is centered near 18kbp but the results can be very different depending of your experiment.</p>

<p><a href="../../images/quality-control/HistogramReadlength.png" rel="noopener noreferrer"><figure id="figure-17" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/HistogramReadlength.png" alt="Histogram of read lengths. " width="657" height="450" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 17</strong>:</span> Histogram of read length&lt;/figcaption&gt;</figure></a></p>

<h2 id="read-lengths-vs-average-read-quality-plot-using-dots">Read lengths vs Average read quality plot using dots</h2>

<p>This plot shows the distribution of fragment sizes according to the Qscore in the file which was analysed.
In general, there is no link between read length and read quality but this representation allows to visualize both information into a single plot and detect possible aberrations.
In runs with a lot of short reads the shorter reads are sometimes of lower quality than the rest.</p>

<p><a href="../../images/quality-control/LengthvsQualityScatterPlot_dot.png" rel="noopener noreferrer"><figure id="figure-18" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/LengthvsQualityScatterPlot_dot.png" alt="Read lengths vs Average read quality plot using dots. " width="631" height="450" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 18</strong>:</span> Histogram of read length&lt;/figcaption&gt;</figure></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-15"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Looking at “Read lengths vs Average read quality plot using dots plot”. Did you notice something unusual with the Qscore? Can you explain it?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-16"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-16" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>There is no reads under Q20.
The qualification for HiFi reads is:</p>
    <ul>
      <li>A minimal number of 3 subreads</li>
      <li>A read Qscore &gt;=20
<a href="../../images/quality-control/pacbio-css-hifi-sequencing.png" rel="noopener noreferrer"><figure id="figure-19" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/pacbio-css-hifi-sequencing.png" alt="PacBio HiFi sequencing. " width="636" height="636" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 19</strong>:</span> PacBio HiFi sequencing&lt;/figcaption&gt;</figure></a></li>
    </ul>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-try-it-on"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Try it on!</div>
  <p>Do the quality control with <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on <code class="language-plaintext highlighter-rouge">m64011_190830_220126.Q20.subsample.fastq.gz</code> and compare the results!</p>
</blockquote>

<h1 id="assess-quality-with-pycoqc---nanopore-only">Assess quality with PycoQC - Nanopore only</h1>

<p><a href="https://github.com/tleonardi/pycoQC">PycoQC</a> (<span class="citation"><a href="#Leger2019">Leger and Leonardi 2019</a></span>) is a data visualisation and quality control tool for nanopore data. In contrast to FastQC/Nanoplot it needs a specific sequencing_summary.txt file generated by Oxford nanopore basecallers such as Guppy or the older albacore basecaller.</p>

<p>One of the strengths of PycoQC is that it is interactive and highly customizable, e.g., plots can be cropped, you can zoom in and out, sub-select areas and export figures.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-check-of-nanopore-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality check of Nanopore reads</div>
  <ol>
    <li>
      <p>Create a new history for this part and give it a proper name</p>
    </li>
    <li>
      <p>Import the nanopore reads <code class="language-plaintext highlighter-rouge">nanopore_basecalled-guppy.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">sequencing_summary.txt</code> from <a href="https://zenodo.org/record/5730295">Zenodo</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/api/files/ff9aa6e3-3d69-451f-9798-7ea69b475989/nanopore_basecalled-guppy.fastq.gz
https://zenodo.org/api/files/ff9aa6e3-3d69-451f-9798-7ea69b475989/sequencing_summary.txt
</code></pre></div>      </div>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pycoqc/pycoqc/2.5.2+galaxy0" title="PycoQC tool" aria-role="link"><strong>PycoQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/pycoqc/pycoqc/2.5.2+galaxy0</span></span> with the following parameters</p>

      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“A sequencing_summary file “</em>: <code class="language-plaintext highlighter-rouge">sequencing_summary.txt</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the webpage output from PycoQC</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-16"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many reads do you have in total?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-17"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-17" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>~270k reads in total (see the Basecall summary table, “All reads”)
For most of basecalling profiles, Guppy will assign reads as “Pass” if the read Qscore is at least equal to 7.</p>
  </blockquote>

  <p>What is the median, minimum and maximum read length, what is the N50?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-18"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-18" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The median read length and the N50 can be found for all as well as for all passed reads, i.e., reads that passed Guppy quality settings (Qscore &gt;= 7), in the basecall summary table.
For the minimum (195bp) and maximum (256kbp) read lengths, it can be found with the read lengths plot.</p>
  </blockquote>
</blockquote>

<h2 id="basecalled-reads-length">Basecalled reads length</h2>

<p>As for FastQC and Nanoplot, this plot shows the distribution of fragment sizes in the file that was analyzed.
As for PacBio CLR/HiFi, long reads have a variable length and this will show the relative amounts of each different size of sequence fragment.
In this example, the distribution of read length is quite dispersed with a minimum read length for the passed reads around 200bp and a maximum length ~150,000bp.</p>

<p><a href="../../images/quality-control/basecalled_reads_length-pycoqc.png" rel="noopener noreferrer"><figure id="figure-20" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/basecalled_reads_length-pycoqc.png" alt="Basecalled reads length. " width="974" height="500" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 20</strong>:</span> Basecalled reads length&lt;/figcaption&gt;</figure></a></p>

<h3 id="basecalled-reads-phred-quality">Basecalled reads PHRED quality</h3>

<p>This plot shows the distribution of the Qscores (Q) for each read. This score aims to give a global quality score for each read.
The exact definition of Qscores is: the average per-base error probability, expressed on the log (Phred) scale.
In case of Nanopore data, the distribution is generally centered around 10 or 12.
For old runs, the distribution can be lower, as basecalling models are less precise than recent models.</p>

<p><a href="../../images/quality-control/basecalled_reads_PHRED_quality-pycoqc.png" rel="noopener noreferrer"><figure id="figure-21" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/basecalled_reads_PHRED_quality-pycoqc.png" alt="Basecalled reads PHRED quality. " width="974" height="500" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 21</strong>:</span> Basecalled reads PHRED quality&lt;/figcaption&gt;</figure></a></p>

<h2 id="basecalled-reads-length-vs-reads-phred-quality">Basecalled reads length vs reads PHRED quality</h2>

<blockquote class="question">
  <div class="box-title question-title" id="question-17"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>What do the mean quality and the quality distribution of the run look like?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-19"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-19" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The majority of the reads have a Qscore between 8 and 11 which is standard for Nanopore data.
Beware that for the same data, the basecaller used (Albacor, Guppy, Bonito), the model (fast, hac, sup) and the tool version can give different results.</p>
  </blockquote>
</blockquote>

<p>As for NanoPlot, this representation give a 2D visualisation of read Qscore according to the length.</p>

<p><a href="../../images/quality-control/basecalled_reads_length_vs_reads_PHRED_quality-pycoqc.png" rel="noopener noreferrer"><figure id="figure-22" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/basecalled_reads_length_vs_reads_PHRED_quality-pycoqc.png" alt="Basecalled reads length vs reads PHRED quality. " width="974" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 22</strong>:</span> Basecalled reads length vs reads PHRED quality&lt;/figcaption&gt;</figure></a></p>

<h2 id="output-over-experiment-time">Output over experiment time</h2>

<p>This representation gives information about sequenced reads over the time for a single run:</p>

<ul>
  <li>Each pic indicates a new loading of the flow cell (3 + the first load).</li>
  <li>The contribution in total reads for each “refuel”.</li>
  <li>The production of reads is decreasing over time:
    <ul>
      <li>Most of the material (DNA/RNA) is sequenced</li>
      <li>Saturation of pores</li>
      <li>Material/pores degradation</li>
      <li>…</li>
    </ul>
  </li>
</ul>

<p>In this example, the contribution of each refueling is very low, and it can be considered as a bad run.
The “Cummulative” plot area (light blue) indicates that 50% of all reads and almost 50% of all bases were produced in the first 5h of the 25h experiment.
Although it is normal that yield decreases over time a decrease like this is not a good sign.</p>

<p><a href="../../images/quality-control/output_over_experiment_time-pycoqc.png" rel="noopener noreferrer"><figure id="figure-23" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/output_over_experiment_time-pycoqc.png" alt="Output over experiment time. " width="974" height="500" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 23</strong>:</span> Output over experiment time&lt;/figcaption&gt;</figure></a></p>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-quot-output-over-experiment-time-quot-profile"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-quot-output-over-experiment-time-quot-profile" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other &quot;Output over experiment time&quot; profile<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>In this example, the data production over the time only slightly decreased over the 12h with a continuous increasing of cumulative data.
This absence of a decreasing curve at the end of the run indicate that there is still biological material on the flow cell. The run was ended before all was sequenced.
It’s an excellent run, even can be considered as exceptional.</p>

  <p><a href="../../images/quality-control/output_over_experiment_time-pycoqc-good.png" rel="noopener noreferrer"><img src="../../images/quality-control/output_over_experiment_time-pycoqc-good.png" alt="Output over experiment time good profile. " width="974" height="500" loading="lazy" /></a></p>

</blockquote>

<h3 id="read-length-over-experiment-time">Read length over experiment time</h3>

<blockquote class="question">
  <div class="box-title question-title" id="question-18"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Did the read length change over time? What could the reason be?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-20"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-20" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>In the current example the read length increases over the time of the sequencing run.
One explanation is that the adapter density is higher for lots of short fragments and therefore the chance of a shorter fragment to attach to a pore is higher. Also, shorter molecules may move faster over the chip.
Over time, however, the shorter fragments are becoming rarer and thus more long fragments attach to pores and are sequenced.</p>
  </blockquote>
</blockquote>

<p>The read length over experiment time should be stable.
It can slightly increase over the time as short fragments tend to be over-sequenced at the beginning and are less present over the time.</p>

<p><a href="../../images/quality-control/read_length_over_experiment_time-pycoqc.png" rel="noopener noreferrer"><figure id="figure-24" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/read_length_over_experiment_time-pycoqc.png" alt="Read length over experiment time. " width="974" height="500" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 24</strong>:</span> Read length over experiment time&lt;/figcaption&gt;</figure></a></p>

<h2 id="channel-activity-over-time">Channel activity over time</h2>

<p>It gives an overview of available pores, pore usage during the experiment, inactive pores and shows if the loading of the flow cell is good (almost all pores are used).
In this case, the vast majority of channels/pores are inactive (white) throughout the sequencing run, so the run can be considered as bad.</p>

<p>You would hope for a plot that it is dark near the X-axis, and with higher Y-values (increasing time) doesn’t get too light/white.
Depending if you chose “Reads” or “Bases” on the left the colour indicates either number of bases or reads per time interval</p>

<p><a href="../../images/quality-control/channel_activity_over_time-pycoqc.png" rel="noopener noreferrer"><figure id="figure-25" style="max-width: 90%; margin:auto;"><img src="../../images/quality-control/channel_activity_over_time-pycoqc.png" alt="Channel activity over time. " width="974" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 25</strong>:</span> Channel activity over time&lt;/figcaption&gt;</figure></a></p>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-quot-channel-activity-over-time-quot-profile"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-quot-channel-activity-over-time-quot-profile" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Other &quot;Channel activity over time&quot; profile<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>In this example, almost all pores are active all along the run (yellow/red profile) which indicate an excellent run.</p>

  <p><a href="../../images/quality-control/channel_activity_over_time-pycoqc-good.png" rel="noopener noreferrer"><img src="../../images/quality-control/channel_activity_over_time-pycoqc-good.png" alt="Channel activity over time good profile. " width="974" height="600" loading="lazy" /></a></p>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-try-it-out"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Try it out!</div>
  <p>Do the quality control with <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and/or <strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on <code class="language-plaintext highlighter-rouge">nanopore_basecalled-guppy.fastq.gz</code> and compare the results!</p>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we checked the quality of FASTQ files to ensure that their data looks good before inferring any further information.
This step is the usual first step for analyses such as RNA-Seq, ChIP-Seq, or any other OMIC analysis relying on NGS data.
Quality control steps are similar for any type of sequencing data:</p>

<ul>
  <li>Quality assessment with tools like:
    <ul>
      <li><em>Short Reads</em>: <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2" title="FASTQE tool" aria-role="link"><strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2</span></span></li>
      <li><em>Short+Long</em>: <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FASTQC tool" aria-role="link"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span></li>
      <li><em>Long Reads</em>:  <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.41.0+galaxy0" title="Nanoplot tool" aria-role="link"><strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.41.0+galaxy0</span></span></li>
      <li><em>Nanopore only</em>: <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pycoqc/pycoqc/2.5.2+galaxy0" title="PycoQC tool" aria-role="link"><strong>PycoQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/pycoqc/pycoqc/2.5.2+galaxy0</span></span></li>
    </ul>
  </li>
  <li>Trimming and filtering for <strong>short reads</strong> with a tool like <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
</ul>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Perform quality control on every dataset before running any other bioinformatics analysis</p>
</li>
                        
                        <li><p>Assess the quality metrics and improve quality if necessary</p>
</li>
                        
                        <li><p>Check the impact of the quality control</p>
</li>
                        
                        <li><p>Different tools are available to provide additional quality metrics</p>
</li>
                        
                        <li><p>For paired-end reads analyze the forward and reverse reads together</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/sequence-analysis/faqs/">FAQ page for the Sequence analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="marcel2011cutadapt">Marcel, M., 2011 <b>Cutadapt removes adapter sequences from high-throughput sequencing reads</b>. EMBnet.journal 17: 10–12. <a href="http://journal.embnet.org/index.php/embnetjournal/article/view/200">http://journal.embnet.org/index.php/embnetjournal/article/view/200</a></li>
<li id="ewels2016multiqc">Ewels, P., M. Magnusson, S. Lundin, and M. Käller, 2016 <b>MultiQC: summarize analysis results for multiple tools and samples in a single report</b>. Bioinformatics 32: 3047–3048. <a href="https://academic.oup.com/bioinformatics/article/32/19/3047/2196507">https://academic.oup.com/bioinformatics/article/32/19/3047/2196507</a></li>
<li id="10.1093/bioinformatics/bty149">De Coster, W., S. D’Hert, D. T. Schultz, M. Cruts, and C. Van Broeckhoven, 2018 <b>NanoPack: visualizing and processing long-read sequencing data</b>. Bioinformatics 34: 2666–2669. <a href="https://doi.org/10.1093/bioinformatics/bty149">10.1093/bioinformatics/bty149</a></li>
<li id="Leger2019">Leger, A., and T. Leonardi, 2019 <b>pycoQC, interactive quality control for Oxford Nanopore Sequencing </b>. Journal of Open Source Software 4: 1236. <a href="https://doi.org/10.21105/joss.01236">10.21105/joss.01236</a></li>
<li id="StJacques2021">Jacques, R. M. S., W. M. Maza, S. D. Robertson, A. Lonsdale, C. S. Murray <i>et al.</i>, 2021 <b>A Fun Introductory Command Line Lesson: Next Generation Sequencing Quality Analysis with Emoji!</b> CourseSource 8: <a href="https://doi.org/10.24918/cs.2021.17">10.24918/cs.2021.17</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Quality Control (Sequence analysis)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Bérénice Batut, Maria Doyle, Alexandre Cormier, Anthony Bretaudeau, Laura Leroi, Erwan Corre, Stéphanie Robin, Erasmus+ Programme, Cameron Hyde,  <b>Quality Control (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{sequence-analysis-quality-control,
author = "Bérénice Batut and Maria Doyle and Alexandre Cormier and Anthony Bretaudeau and Laura Leroi and Erwan Corre and Stéphanie Robin and Erasmus+ Programme and Cameron Hyde",
	title = "Quality Control (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/sequence-analysis/tutorials/quality-control/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: fastqc
  owner: devteam
  revisions: e7b2202befea
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 3d0c7bdf12f5
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqe
  owner: iuc
  revisions: 89d299153577
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: 75c93c70d094
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: nanoplot
  owner: iuc
  revisions: 0f1c34698076
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pycoqc
  owner: iuc
  revisions: dc1500f42073
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: cutadapt
  owner: lparsons
  revisions: 660cffd8d92a
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: cutadapt
  owner: lparsons
  revisions: 44e2cc14f75b
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00239">https://gxy.io/GTN:T00239</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/sequence-analysis/tutorials/quality-control/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1697329661'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>