<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Somatic Variant Discovery from WES Data Using Control-FREEC</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1706401745' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1706401745' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Somatic Variant Discovery from WES Data Using Control-FREEC">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-24 08:35:58 +0000">
<meta name="DC.creator" content="Khaled Jum'ah">
<meta name="DC.creator" content="Katarzyna Kamieniecka">
<meta name="DC.creator" content="Wolfgang Maier">
<meta name="DC.creator" content="David Salgado">
<meta name="DC.creator" content="Krzysztof Poterlowicz">

        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Somatic Variant Discovery from WES Data Using Control-FREEC">
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-24 08:35:58 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Somatic Variant Discovery from WES Data Using Control-FREEC",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Somatic Variant Discovery from WES Data Using Control-FREEC' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html",
  "timeRequired": "PT3H",
  "teaches": "Use Control-Freec for hCNV Identefication in tumor tissue.\n - Visualise the detected hCNVs in specific chromosomes.",
  "keywords": "variant-analysis",
  "description": "The questions this  addresses are:\n - What are the specific challenges in locating human Copy Number Variances (hCNVs)?\n - How to preprocess the sequenced reads for hCNVs detection?\n - How can you detect the hCNVs in/from tumor and normal tissue of the same individual?\n - How can you visualise the hCNVs' findings and compare them for specific regions?\n\n\nThe objectives are:\n - Use Control-Freec for hCNV Identefication in tumor tissue.\n - Visualise the detected hCNVs in specific chromosomes.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/Khaled196/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/Khaled196/",
      "name": "Khaled Jum'ah",
      "image": "https://avatars.githubusercontent.com/Khaled196",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "name": "Katarzyna Kamieniecka",
      "image": "https://avatars.githubusercontent.com/kkamieniecka",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0009-0004-2454-5950",
      "orcid": "https://orcid.org/0009-0004-2454-5950"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/d-salgado/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/d-salgado/",
      "name": "David Salgado",
      "image": "https://avatars.githubusercontent.com/d-salgado",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "name": "Krzysztof Poterlowicz",
      "image": "https://avatars.githubusercontent.com/poterlowicz-lab",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6173-5674",
      "orcid": "https://orcid.org/0000-0001-6173-5674"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1706401745'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Somatic Variant Discovery from WES Data Using Control-FREEC</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/Khaled196/" class="contributor-badge contributor-Khaled196"><img src="https://avatars.githubusercontent.com/Khaled196?s=36" alt="Avatar" width="36" height="36">Khaled Jum'ah</a><a href="/training-material/hall-of-fame/kkamieniecka/" class="contributor-badge contributor-kkamieniecka"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/kkamieniecka?s=36" alt="Avatar" width="36" height="36">Katarzyna Kamieniecka</a><a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Avatar" width="36" height="36">Wolfgang Maier</a><a href="/training-material/hall-of-fame/d-salgado/" class="contributor-badge contributor-d-salgado"><img src="https://avatars.githubusercontent.com/d-salgado?s=36" alt="Avatar" width="36" height="36">David Salgado</a><a href="/training-material/hall-of-fame/poterlowicz-lab/" class="contributor-badge contributor-poterlowicz-lab"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/poterlowicz-lab?s=36" alt="Avatar" width="36" height="36">Krzysztof Poterlowicz</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the specific challenges in locating human Copy Number Variances (hCNVs)?</p>
</li>
        
        <li><p>How to preprocess the sequenced reads for hCNVs detection?</p>
</li>
        
        <li><p>How can you detect the hCNVs in/from tumor and normal tissue of the same individual?</p>
</li>
        
        <li><p>How can you visualise the hCNVs’ findings and compare them for specific regions?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Use Control-Freec for hCNV Identefication in tumor tissue.</p>
</li>
        
        <li><p>Visualise the detected hCNVs in specific chromosomes.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.2582555">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/workflows/" title="Somatic Variant Discovery from WES Data Using Control-FREEC workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 24, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00317">https://gxy.io/GTN:T00317</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Human Copy Number Variations (hCNVs) are the result of structural genomic rearrangements that result in the duplication or deletion of DNA segments. These changes contribute significantly to human genetic variability, diseases, and somatic genome variations in cancer and other diseases <span class="citation"><a href="#Nam2015">Nam <i>et al.</i> 2015</a></span>. hCNVs can be routinely investigated by genomic hybridisation and sequencing technologies
. There is a range of software tools that can be used to identify and quantify hCNVs. Unfortunately, locating hCNVs is still a challenge in standardising formats for data representation and exchange. Furthermore, the sensitivity, specificity, reproducibility, and reusability of hCNV detection and analysis research software varies. As a result, there is a need for the adoption of community-developed standards for data discovery and exchange. To address that ELIXIR developed <span class="citation"><a href="#Beacon">Beacon</a></span> protocol and <span class="citation"><a href="#GA4GH">for Genomics and Health</a></span> standards, as well as mechanisms for annotating, benchmarking, creating reproducible and sharable tools and workflows, such as <span class="citation"><a href="#WorkflowHub">WorkflowHub</a></span>, Galaxy, and <span class="citation"><a href="#OpenEBench">ELIXIR</a></span>, and, most importantly, accessible training resources and infrastructure.</p>

<p>This tutorial is a modification of a Galaxy Training Network tutorial <a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">Somatic variant calling</a> tutorial to provide training on how to preprocess, identify and visualise hCNV regions using Control-FreeC tool using tumor/normal samples pairs.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control-and-mapping-of-ngs-reads" id="markdown-toc-quality-control-and-mapping-of-ngs-reads">Quality control and mapping of NGS reads</a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a></li>
      <li><a href="#read-trimming-and-filtering" id="markdown-toc-read-trimming-and-filtering">Read trimming and filtering</a></li>
      <li><a href="#read-mapping" id="markdown-toc-read-mapping">Read Mapping</a></li>
    </ol>
  </li>
  <li><a href="#copy-number-variation-detection-hcnv" id="markdown-toc-copy-number-variation-detection-hcnv">Copy Number Variation detection (hCNV).</a>    <ol>
      <li><a href="#mapped-reads-filtering" id="markdown-toc-mapped-reads-filtering">Mapped reads filtering</a></li>
      <li><a href="#homogenize-mapped-reads" id="markdown-toc-homogenize-mapped-reads">Homogenize mapped reads</a></li>
      <li><a href="#filtrate-indels" id="markdown-toc-filtrate-indels">Filtrate indels</a></li>
      <li><a href="#control_freec-for-hcnv-detection" id="markdown-toc-control_freec-for-hcnv-detection">Control_FREEC for hCNV detection</a></li>
      <li><a href="#visualise-detected-hcnvs" id="markdown-toc-visualise-detected-hcnvs">Visualise detected hCNVs</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>
<p>First, start with uploading and preparing the input data to analyze. The sequencing reads used in this analysis are from real-world data from a cancer patient’s tumor and normal tissue samples. For the sake of an acceptable speed of the analysis, the original data has been downsampled though to include only the reads from human chromosomes 5, 12 and 17.</p>

<table>
  <thead>
    <tr>
      <th>Name</th>
      <th>Format</th>
      <th>Origin</th>
      <th>Encoding</th>
      <th>Sequence length</th>
      <th>Total Sequences</th>
      <th>Chromosome</th>
      <th>Data size (MB)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>SLGFSK-N_231335_r1_chr5_12_17</td>
      <td>fastq</td>
      <td>Normal tissue</td>
      <td>Sanger / Illumina 1.9</td>
      <td>101</td>
      <td>10602766</td>
      <td>5, 12 and 7</td>
      <td>530.4 MB</td>
    </tr>
    <tr>
      <td>SLGFSK-N_231335_r2_chr5_12_17</td>
      <td>fastq</td>
      <td>Normal tissue</td>
      <td>Sanger / Illumina 1.9</td>
      <td>101</td>
      <td>10602766</td>
      <td>5, 12 and 7</td>
      <td>582.0 MB</td>
    </tr>
    <tr>
      <td>SLGFSK-T_231336_r1_chr5_12_17</td>
      <td>fastq</td>
      <td>Cancer tissue</td>
      <td>Sanger / Illumina 1.9</td>
      <td>101</td>
      <td>16293448</td>
      <td>5, 12 and 7</td>
      <td>811.3 MB</td>
    </tr>
    <tr>
      <td>SLGFSK-T_231336_r2_chr5_12_17</td>
      <td>fastq</td>
      <td>Cancer tissue</td>
      <td>Sanger / Illumina 1.9</td>
      <td>101</td>
      <td>16293448</td>
      <td>5, 12 and 7</td>
      <td>868.7 MB</td>
    </tr>
  </tbody>
</table>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>For this tutorial, make a new history.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the data files from
<a href="https://zenodo.org/record/2582555">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz
</code></pre></div>      </div>
      <p>This will download four sequenced files ordered as: 
  The first two files are for the forward and the reverse reads for the sample <strong>normal</strong>
  tissue sequence.
  The other two belong to the <strong>tumor</strong> reads.</p>

      <p>In some cases the same dataset can be found in the Galaxy shared data library. 
Ask the instructor for more details about this.</p>

      <p>The dat aset can also be downloaded a local storage.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Make sure to upload the sequences in fastaq format. Look at the history and
check if the created datasets have their data types assigned correctly with two reads for
the tumor tissues and two reads for the normal tissues. If not, fix any
missing or wrong data type assignments.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing the datatype<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the lower part <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong>, select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Give the data meaningful names and tags to facilitate analysis.</p>

      <p>When uploading data from a link, Galaxy names the files after the link address.
It might be useful to change or modify the name to something more meaningful.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>This tutorial has a set of shared steps performed on the data. To track the
data in the history, it is recommended to tag the datasets by attaching a meaningful tag ‘#’
to them. The tagging will automatically be attached to any file generated
from the original tagged dataset.
<em>e.g.</em>, <code class="language-plaintext highlighter-rouge">#normal</code> for normal tissue datasets (with <code class="language-plaintext highlighter-rouge">-N_</code> in the name) and
<em>e.g.</em>, <code class="language-plaintext highlighter-rouge">#tumor</code> for tumor dataset (with <code class="language-plaintext highlighter-rouge">-T_</code> in the name).</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Adding a tag<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</li>     </ul>   </li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="quality-control-and-mapping-of-ngs-reads">Quality control and mapping of NGS reads</h1>
<p>The data was obtained following a series of laboratory procedures, including DNA preparation, extraction, and sequencing, which means there is a possibility of errors occurring during those steps, which could affect data quality. To address that, it is necessary to test the quality of the fastq reads. The data quality needs to be within an acceptable range before looking for hCNVs. The low-quality data can lead us to false results. To detect low-quality data,  preprocessing step is required to trim or discard the low-quality reads before proceeding with the mapping and hCNV detection steps.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-more-on-quality-control-and-mapping"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More on quality control and mapping</div>
  <p>To read more about quality control this is tutorial on Galaxy training network 
<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"> Quality Control</a> 
For mapping <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">Mapping</a></p>
</blockquote>

<h2 id="quality-control">Quality Control</h2>
<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control-of-the-input-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control of the input datasets</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1</span></span> on the fastq datasets
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from the current history”</em>: all 4 FASTQ  datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Select multiple datasets<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>This job will generate eight new datasets to the history. To
parse the quality results view the html report of each dataset.</p>

      <p>For the next step use the raw data fidings from FastQC.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0</span></span> to aggregate the raw <strong>FastQC</strong> data of all four input datasets into one report
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: all four <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <em>Webpage</em> output produced by the tool</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>
            <ol>
              <li>How do you feel about the sequence’s overall quality? 
<a href="../../images/Control_Freec_fastqc_per_base_sequence_quality_plot_before.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_fastqc_per_base_sequence_quality_plot_before.png" alt="fastqbefore. " width="1200" height="800" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Sequence quality per base generated by FastQC before end trimming.&lt;/figcaption&gt;</figure></a>
<a href="../../images/Control_Freec_fastqc_per_sequence_gc_content_plot_before.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_fastqc_per_sequence_gc_content_plot_before.png" alt="GCbefore. " width="1200" height="800" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Sequence quality per Sequence GC content generated by FastQC before end trimming.&lt;/figcaption&gt;</figure></a></li>
            </ol>
          </li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>
              <p>The forwards and reversed reads show good 
quality, , with no major issues discovered
discovered during the preparation process..</p>
            </li>
            <li>
              <p>TThe GC content plots for the samples’ forward and reverse reads 
show an unusual bimodal distribution.</p>

              <p>The unnormal distribution of the GC content of reads from a sample is 
usually interpreted as a sign of possible contamination.
However, we are dealing with sequencing data from captured exomes,
which means that the reads do not represent random sequences from a  genome.
They rather represent an arbitrary selection. Indeed, the samples were 
prepared using Agilent’s SureSelect V5 technology for exome enrichment, 
and bimodal GC content distributions have been identified as a hallmark 
of that capture method
for example, see Fig. 4C in
<span class="citation"><a href="#Meienberg2015">Meienberg <i>et al.</i> 2015</a></span>.</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="read-trimming-and-filtering">Read trimming and filtering</h2>
<p>As previously demonstrated, The data have relatively high-quality sequenced reads.
However, the aim is to detect clear reads for hCNVs and will use a trimming step to see if the analysis can be improved.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read trimming and filtering of the normal tissue reads</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5" title="Trimmomatic tool" aria-role="link"><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5</span></span> to trim and filter the normal tissue reads
      <ul>
        <li>
          <p><em>“Single-end or paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end (two separate
input files)</code></p>

          <p>This makes the tool treat the forward and reverse reads simultaneously.</p>

          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R1/first of pair)”</em>: the
forward reads (r1) dataset of the normal tissue sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R2/second of pair)”</em>: the
reverse reads (r2) dataset of the normal tissue sample</li>
          </ul>
        </li>
        <li><em>“Perform initial ILLUMINACLIP step?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Select standard adapter sequences or provide custom?”</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
              <ul>
                <li><em>“Adapter sequences to use”</em>: <code class="language-plaintext highlighter-rouge">TruSeq3 (paired-ended, for MiSeq and HiSeq)</code></li>
              </ul>
            </li>
            <li><em>“Maximum mismatch count which will still allow a full match to be
  performed”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>“How accurate the match between the two ‘adapter ligated’ reads must
  be for PE palindrome read alignment”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
            <li><em>“How accurate the match between any adapter etc. sequence must be
  against a read”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>“Minimum length of adapter that needs to be detected (PE specific/
  palindrome mode)”</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
            <li><em>“Always keep both reads (PE specific/palindrome mode)?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>

          <p>These parameters are used to cut ILLUMINA-specific adapter sequences
 from the reads.</p>
        </li>
        <li>In <em>“Trimmomatic Operation”</em>
          <ul>
            <li>In <em>“1: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut the specified number
 of bases from the start of the read (HEADCROP)</code>
                  <ul>
                    <li><em>“Number of bases to remove from the start of the read”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> “Insert Trimmomatic Operation”*</li>
            <li>In <em>“2: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off the end of
 a read, if below a threshold quality (TRAILING)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> “Insert Trimmomatic Operation”*</li>
            <li>In <em>“3: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Drop reads below a
 specified length (MINLEN)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This step will creates four files in the history. The sizes of those two files vary depending on the original data quality and trimming intensity. The first two files are for mated forward and reverse reads, respectively.
The other two are for unmated reads as a result of excessive trimming.
However, because of the high average data quality, there was no need to perform excessive trimming by selecting the previous three trimming conditions, so those files should be empty. Those files can be heden to keep the history cleaner.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-keeping-attention-on-the-data-type"><button class="gtn-boxify-button details" type="button" aria-controls="details-keeping-attention-on-the-data-type" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Keeping attention on the data type<span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Track whether the reads are paired or unpaired, and remember to include them in any tool to be used.
The reason is that there are some tools, such as read mappers,
that expect reads to be in a specific order and having unmapped reads can result in significant.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-trimming-and-filtering-of-the-tumor-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read trimming and filtering of the tumor tissue reads</div>
  <ol>
    <li>repeat the previous step for <strong>tumor tissue</strong> reads following the same steps as above.</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-exercise-quality-control-of-the-polished-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Exercise: Quality control of the polished datasets</div>
  <p>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1</span></span> and <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0</span></span> like before,
but using the four trimmed datasets produced by Trimmomatic as input.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>Is there any difference between the reads before and after the trimming?
<a href="../../images/Control_Freec_fastqc_per_base_sequence_quality_plot_after.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_fastqc_per_base_sequence_quality_plot_after.png" alt="fastqafter. " width="1200" height="800" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Sequence quality per base generated by FastQC after end trimming. The x-axis on the grapgh is fore the read length and the The y-axis shows the quality scores. The higher the score the better the base call. The background of the graph divides the y axis into very good quality calls (green), calls of reasonable quality (orange), and calls of poor quality (red).&lt;/figcaption&gt;</figure></a>
<a href="../../images/Control_Freec_fastqc_per_sequence_gc_content_plot_afrer.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_fastqc_per_sequence_gc_content_plot_afrer.png" alt="GCafter. " width="1200" height="800" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Sequence quality per Sequence GC content generated by FastQC after end trimming the plot shows an unusually shaped distribution that could indicate a contaminated library or some other kinds of biased subset&lt;/figcaption&gt;</figure></a></p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>The quality of the data is good, so 
trimming them didn’t lead to dramatic changes. However,
we can point out that some of the adapters were removed.</p>
    </blockquote>
  </blockquote>
</blockquote>

<h2 id="read-mapping">Read Mapping</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read Mapping</div>
  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Map with BWA-MEM tool" aria-role="link"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span> to map the reads from the <strong>normal tissue</strong> sample to the reference genome
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>“Will you select a reference genome from your history or use a
built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the trimmed
forward reads (r1) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the trimmed
reverse reads (r2) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">Not available.</code></li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">Not available.</code></li>
              </ul>
            </li>
            <li><em>“Platform/technology used to produce the reads (PL)”</em>: <code class="language-plaintext highlighter-rouge">ILLUNINA</code></li>
            <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">Simple illumina mode</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-more-on-read-group-identifiers-and-sample-names"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More on read group identifiers and sample names</div>
        <p>In general, we can choose our own ID and SM values, but the ID should
unambiguously identify the sequencing run that produced the reads,
while the SM value should identify the biological sample.</p>
      </blockquote>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Map with BWA-MEM tool" aria-role="link"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span> to map the reads from the <strong>tumor tissue</strong> sample
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <p>Adjust these settings as before in case of using imported reference
genome.</p>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the trimmed
forward reads (r1) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the reverse
reads (r2) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">Not available.</code></li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">Not available.</code></li>
              </ul>
            </li>
            <li><em>“Platform/technology used to produce the reads (PL)”</em>: <code class="language-plaintext highlighter-rouge">ILLUNINA</code></li>
            <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">Simple illumina mode</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Name the created list as <strong>Mapping-lsit</strong></p>

<h1 id="copy-number-variation-detection-hcnv">Copy Number Variation detection (hCNV).</h1>

<p>After the mapping step, the data are ready to the hCNV detection step. This tutorial focuses on the <strong>Control-FreeC</strong> tool as a method for hCNV identification:</p>

<ul>
  <li>Identifies variant alleles in tumor/normal pair samples.</li>
  <li>Visualize the hCNV using the Circos tool.</li>
</ul>

<h2 id="mapped-reads-filtering">Mapped reads filtering</h2>

<p>The remaining data preprocessing until the Control-FreeC step is the same for Normal and Tumor reads.  Create a data collection to include those two files.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filtrate-the-mapped-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filtrate the mapped reads</div>
  <ol>
    <li>Use <span class="tool" data-tool="__BUILD_LIST__" title="Build list tool" aria-role="link"><strong>Build list</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>BUILD_LIST</strong></span></span> to creat a list from the maped reads of the <strong>normal tissue</strong> and <strong>tumor tissue</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset”</em>: <code class="language-plaintext highlighter-rouge">Insert dataset</code>
          <ul>
            <li><em>“Input dataset”</em>: <code class="language-plaintext highlighter-rouge">The output of map with BWA-MEM for normal tissue</code></li>
          </ul>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset”</em>: <code class="language-plaintext highlighter-rouge">Insert dataset</code>
          <ul>
            <li><em>“Input dataset”</em>: <code class="language-plaintext highlighter-rouge">The output of map with BWA-MEM for cancer tissue</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0" title="Create text file tool" aria-role="link"><strong>Create text file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.1.0</span></span>  with the following parameters
      <ul>
        <li><em>“Characters to insert”</em>: <code class="language-plaintext highlighter-rouge">normal reads</code></li>
        <li><em>“Specify the number of iterations by”</em>: <code class="language-plaintext highlighter-rouge">User defined number</code></li>
        <li><em>“How many times?”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> “Insert selection”*</li>
      </ul>
      <ul>
        <li><em>“Characters to insert”</em>: <code class="language-plaintext highlighter-rouge">tumor reads</code></li>
        <li><em>“Specify the number of iterations by”</em>: <code class="language-plaintext highlighter-rouge">User defined number</code></li>
        <li><em>“How many times?”</em>: <code class="language-plaintext highlighter-rouge">1</code>
    - This will create a text file with only two lines normal reads and tumor reads</li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="__RELABEL_FROM_FILE__" title="Relabel identifiers tool" aria-role="link"><strong>Relabel identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>RELABEL_FROM_FILE</strong></span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection”</em>: the creat a list from <code class="language-plaintext highlighter-rouge">Build lsit tool</code>.</li>
        <li><em>“How should the new labels be specified?”</em>: <code class="language-plaintext highlighter-rouge">Using lines in a simple text file.</code></li>
        <li><em>“New Identifiers”</em>: The outcom from <code class="language-plaintext highlighter-rouge">Create txt file tool</code></li>
        <li><em>“Ensure strict mapping”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
      </ul>
    </li>
  </ol>

  <ul>
    <li>Make sure to use the same files order for the both of <strong>Build list</strong> tool and <strong>Creat text file</strong>.</li>
  </ul>
</blockquote>

<p>It is essential to filtrate the reads before the mapping step.
It is also required to filtrate the reads after. The preprocessing step
works by filtrating the mapped reads by removing the low-quality regions
and the duplicated reads. This step is needed to reduce the running time
and in results interpretation.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-filtration-and-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data filtration and Remove duplicates</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3" title="Samtools view tool" aria-role="link"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM/BAM/CRAM dataset”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Relabel identifiers</code> dataset cpllection.</li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
        <li>In<em>“Configure filters:”</em>:
    - In <em>“1: Configure filters:”</em>
          <ul>
            <li><em>“Filter by quality”</em>: <code class="language-plaintext highlighter-rouge">1</code>
              <ul>
                <li>In <em>“2: Configure filters:”</em></li>
              </ul>
            </li>
            <li><em>“Exclude reads with any of the following flags set”</em>: <code class="language-plaintext highlighter-rouge">Read is unmapped</code> <code class="language-plaintext highlighter-rouge">Mate  is unmapped</code></li>
          </ul>
        </li>
        <li><em>Produce extra dataset with dropped reads?”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">BAM (-b)</code></li>
        <li><em>“Reference data”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-selecting-a-dataset-collection-as-input"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-selecting-a-dataset-collection-as-input" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Selecting a dataset collection as input<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in  front of the input parameter you want to supply the collection to.</li>   <li>Select the collection you want to use from the list</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1" title="RmDup tool" aria-role="link"><strong>RmDup</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM File”</em>:  The outpot of  <code class="language-plaintext highlighter-rouge">Samtools view</code></li>
        <li><em>“Is this paired-end or single end data”</em>: <code class="language-plaintext highlighter-rouge">BAM is paired-end</code></li>
        <li><em>“Treat as single-end”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="homogenize-mapped-reads">Homogenize mapped reads</h2>

<p>To detect hCNVs expression accurately. The reads must go through the lift alignment process. Lift Alignment works by shifting reads that contain indels to the left of a reference genome until they can not be shifted anymore. As a result, it will only extract reads with indels, with no false reads (reads that mismatch with the reference genome other than the indels).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-homogenize-the-positional-distributed-indels"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Homogenize the positional distributed indels</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1" title="BamLeftAlign tool" aria-role="link"><strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1</span></span>  with the following parameters
      <ul>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Locally cached </code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select alignment file in BAM format”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">tool RmDup</code>.</li>
        <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">hg19</code></li>
        <li><em>“Maximum number of iterations”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2" title="Samtools calmd tool" aria-role="link"><strong>Samtools calmd</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM file to recalculate”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">BamLeftAlign</code>.</li>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code></li>
        <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">hg19</code></li>
        <li><em>“Do you also want BAQ (Base Alignment Quality) scores to be calculated?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Additional options”</em>: <code class="language-plaintext highlighter-rouge">Advanced options</code></li>
        <li><em>“Change identical bases to ‘=’“</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
        <li><em>“Coefficient to cap mapping quality of poorly mapped reads”</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>After the Homogenizing step, it is now to extract the reads which hold indels from the mapped reads.</p>

<h2 id="filtrate-indels">Filtrate indels</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filtrate-the-indels-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filtrate the indels reads</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3" title="Samtools view tool" aria-role="link"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM file to recalculate”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Samtools CalMD</code>.</li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">Just the input header (-H)</code></li>
        <li><em>“What would you like to have reported?”</em>: <code class="language-plaintext highlighter-rouge">The header in ...</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">SAM</code></li>
        <li><em>“Reference data”</em>: <code class="language-plaintext highlighter-rouge">No, see help (-output-fmt-option no_ref)</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="Grep1" title="Select tool" aria-role="link"><strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select lines from”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Samtools view</code>.</li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">^@SQ\tSN:(chr[0-9]+|chrM|chrX|chrY)\tLN:[0-9]+</code></li>
        <li><em>“Keep header line”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2" title="Replace Text in entire line tool" aria-role="link"><strong>Replace Text in entire line</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select lines from”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Select</code>.</li>
        <li>In <em>“Replacements:”</em>:</li>
      </ul>
      <ul>
        <li>In <em>“1: Replacements:”</em>
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^@SQ\tSN:(chr[0-9]+|chrM|chrX|chrY)\tLN:([0-9]+)</code></li>
          </ul>
        </li>
        <li>In <em>“2: Replacements:”</em>
          <ul>
            <li><em>“Replace with:”</em>: <code class="language-plaintext highlighter-rouge">\1\t0\t\2</code></li>
          </ul>
        </li>
      </ul>

      <p>This step will generate a data collection folder with two files inside. Change the datatype for the files inside it into <strong>BED format</strong>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing the datatype<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the lower part <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong>, select  your desired datatype     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3" title="Samtools view tool" aria-role="link"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM/BAM/CRAM data set”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">CalMD</code>.</li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
        <li>In<em>“Configure filters:”</em>:</li>
      </ul>
      <ul>
        <li>In <em>“1: Configure filters:”</em>
          <ul>
            <li><em>“Filter by quality”</em>: <code class="language-plaintext highlighter-rouge">255</code>
              <ul>
                <li><em>What would you like to have reported?”</em>: <code class="language-plaintext highlighter-rouge">Reads dropped during filtering and subsampling</code></li>
                <li><em>Produce extra dataset with dropped reads?”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
                <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">BAM (-b)</code></li>
                <li><em>“Reference data”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3" title="Samtools view tool" aria-role="link"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.9+galaxy3</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM/BAM/CRAM data set”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Samtools view</code>.</li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
        <li>In<em>“Configure filters:”</em>:</li>
      </ul>
      <ul>
        <li>In <em>“1: Configure filters:”</em>
          <ul>
            <li><em>“Filter by regions:”</em>: <code class="language-plaintext highlighter-rouge">Regions from BED file</code></li>
          </ul>
        </li>
        <li>In <em>“2: Configure filters:”</em>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Filter by intervals in a bed file:”</em>: The outpot of  ` Replace Text` BED format.</li>
          </ul>
        </li>
        <li>In <em>“3: Configure filters:”</em>
          <ul>
            <li><em>“Filter by readgroup”</em>: <code class="language-plaintext highlighter-rouge">NO</code></li>
          </ul>
        </li>
        <li>In <em>“4: Configure filters:”</em>
          <ul>
            <li><em>“Filter by quality”</em>: <code class="language-plaintext highlighter-rouge">1</code>
              <ul>
                <li><em>Produce extra dataset with dropped reads?”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
                <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">BAM (-b)</code></li>
                <li><em>“Reference data”</em>:<code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-files-form-list"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Extract files form list</div>
  <p>extract the files from the list to handel them separitly</p>
  <ol>
    <li>Run <span class="tool" data-tool="__EXTRACT_DATASET__" title="Extract Dataset tool" aria-role="link"><strong>Extract Dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>EXTRACT_DATASET</strong></span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input List”</em>: <code class="language-plaintext highlighter-rouge">The outpot of Samtools view</code>.</li>
        <li><em>“How should a dataset be selected?”</em>: <code class="language-plaintext highlighter-rouge">Select by element identifier</code></li>
        <li><em>“Element identifier:”</em>: <code class="language-plaintext highlighter-rouge">tumor reads</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="__EXTRACT_DATASET__" title="Extract Dataset tool" aria-role="link"><strong>Extract Dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>EXTRACT_DATASET</strong></span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input List”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Samtools view</code>.</li>
        <li><em>“How should a dataset be selected?”</em>: <code class="language-plaintext highlighter-rouge">Select by element identifier</code></li>
        <li><em>“Element identifier:”</em>: <code class="language-plaintext highlighter-rouge">normal reads</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="control_freec-for-hcnv-detection">Control_FREEC for hCNV detection</h2>
<p>The data are now ready to detect hCNV. Control-FREEC detects copy-number alterations and allelic imbalances (including loss of heterozygosity; LOH) by automatically computing, normalising, and segmenting copy number profile and beta allele frequency (BAF) profile, and then calling copy number alterations and LOH. Control-FREEC differentiates between somatic and germline variants. Based on those profiles. The control reads display the gene status for each segment.
<a href="../../images/Control_Freec_output.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_output.png" alt="Controlfreec. " width="1350" height="442" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> 1. Control-FREEC calculates copy number and BAF profiles and detects regions of copy number gain/loss and LOH regions in chromosomes 5,12 17. Gains, losses and LOH are shown in red, blue and light blue, respectively.&lt;/figcaption&gt;</figure></a></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-more-on-control-freec-and-hcnvs-detection"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More on control_FREEC and hCNVs detection</div>
  <p>Control-freec works by:</p>
  <ol>
    <li>Annotating genomic changes and heterozygosity loss in the sample dataset.</li>
    <li>Distinguishes between germline and somatic variants by creating copy number profile and BAF profile.</li>
    <li>Employs the information in those profiles to detect copy number changes in sample reads.</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Can you expect the essential factors that afflict hCNVs detection?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Coverage bias in reads 
Changes in reading mobility and GC content may favour the duplication of specific reads over others.</li>
      <li>Bias in reading alignment
Because normal cells have higher read coverage than allelic reads during alignment, they may be classified as noise findings.</li>
      <li>Normal cell contamination 
The presence of normal cells within tumour cells can have an impact on the construction of a tumour genome’s copy number profile.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-detection-of-copy-number-changes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Detection of copy-number changes</div>
  <ol>
    <li>
      <p>Import the DED file for the captured reagions from <a href="https://doi.org/10.5281/zenodo.5697358">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/5697358/files/capture_targets_chr5_12_17.bed
</code></pre></div>      </div>
    </li>
    <li>
      <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/control_freec/control_freec/11.6+galaxy1" title="Control-FREEC tool" aria-role="link"><strong>Control-FREEC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/control_freec/control_freec/11.6+galaxy1</span></span>  with the following parameters</p>
      <ul>
        <li><em>“Select the sequencing method of the input file(s)?”</em>: <code class="language-plaintext highlighter-rouge">whole-exome sequencing (WES)</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sample file”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Extract Dataset</code> (<strong>cancer reads</strong>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Control file”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Extract Dataset</code> (<strong>normal reads</strong>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BED file with capture regions”</em>: <code class="language-plaintext highlighter-rouge">chr 5, 7, 12 capture reagions **Input file**</code></li>
        <li><em>“Format of reads”</em>: <code class="language-plaintext highlighter-rouge">Illumina paired-end (FR)</code></li>
      </ul>
    </li>
  </ol>

  <ul>
    <li>In<em>“Advanced WES settings:”</em>:
      <ul>
        <li>In <em>“1: Advanced WES settings:”</em>
          <ul>
            <li><em>“Degree of polynomial:”</em>: <code class="language-plaintext highlighter-rouge">control-read-count-based normalization, WES (1)</code></li>
          </ul>
        </li>
        <li>In <em>“2:Advanced WES settings:”</em>
          <ul>
            <li><em>“Read Count (RC) correction for GC-content bias and low mappability:”</em>: <code class="language-plaintext highlighter-rouge">normalize the sample and the control RC using GC-content and then calculate the ratio "Sample RC/contol RC" (1)</code></li>
          </ul>
        </li>
        <li>In <em>“3: Advanced WES settings:”</em>
          <ul>
            <li><em>“Minimal number of consecutive windows to call a CNA”</em>: <code class="language-plaintext highlighter-rouge">WES (3)</code></li>
          </ul>
        </li>
        <li>In <em>“4: Advanced WES settings:”</em>
          <ul>
            <li><em>“Segmentation of normalized profiles (break point)”</em>: <code class="language-plaintext highlighter-rouge">1.2</code></li>
          </ul>
        </li>
        <li>In <em>“5: Advanced WES settings:”</em>
          <ul>
            <li><em>“Desired behavior in the ambiguous regions”-1” “</em>: <code class="language-plaintext highlighter-rouge">make a separate fragment of this "unknown" region and do not assign any copy number to this region at all (4)</code></li>
          </ul>
        </li>
        <li>In <em>“6: Advanced WES settings:”</em>
          <ul>
            <li><em>“Adjust sample contamination?”</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
          </ul>
        </li>
        <li>In <em>“7: Advanced WES settings:”</em>
          <ul>
            <li><em>“Sample contamination by normal cells”</em>: <code class="language-plaintext highlighter-rouge">0.30000000000000004</code></li>
          </ul>
        </li>
        <li>In <em>“8: Advanced WES settings:”</em>
          <ul>
            <li><em>“Intercept of polynomial”</em>: <code class="language-plaintext highlighter-rouge">with GC-content (1)</code></li>
          </ul>
        </li>
        <li>In <em>“9: Advanced WES settings:”</em>
          <ul>
            <li><em>“Sample sex”</em>: <code class="language-plaintext highlighter-rouge">XX</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code></li>
    <li><em>“Reference genome”</em>: <code class="language-plaintext highlighter-rouge">hg19</code></li>
    <li>In<em>“Outputs:”</em>:
      <ul>
        <li>In <em>“1: Outputs:”</em>
          <ul>
            <li><em>“BedGraph Output for UCSC genome browser”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
          </ul>
        </li>
        <li>In <em>“2 :Outputs:”</em>
          <ul>
            <li><em>“Visualize normalized copy number profile with predicted CNAs”</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
          </ul>
        </li>
        <li>In <em>“3: Outputs:”</em>
          <ul>
            <li><em>“2D data track file for Circos”</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>

  <blockquote class="question">
    <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>In your opinion, what are the challenges in hCNVs detection?</p>
    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>The bias in Reads coverage changes in reads mobility and GC content can lead to 
favouring the duplication of specific reads over others.</li>
        <li>The bias in reads alignment. 
Normal cells’ reads coverage is higher than the allelic reads, for 
that the allelic reads some times expressed as noise findings</li>
        <li>Contamination with normal cells 
The availability of normal cells within tumor cells can
effect on the construction of the copy number profile of a 
tumor genome</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<h2 id="visualise-detected-hcnvs">Visualise detected hCNVs</h2>

<p>Cicros demonstrates the relationship and the positions of different objects with an appealing,
high quality and illustrative multilayers circular plot. Circos gives the user flexibility to
present the link between their data at a high rate by providing the ability to control the features 
and elements in creating the plot. Circos visualise the genomic alterations in genome structure and 
the relationships between the genomic intervals <a href="https://genome.cshlp.org/content/19/9/1639.short">Krzywinski, Schein et al. 2009</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualise-the-hcnv-findings"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualise the hCNV findings</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/circos/circos/0.69.8+galaxy7" title="Circos tool" aria-role="link"><strong>Circos</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/circos/circos/0.69.8+galaxy7</span></span>  with the following parameters
      <ul>
        <li><em>“Reference Genome Source”</em>: <code class="language-plaintext highlighter-rouge">Custom Karyotype</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sample file”</em>: The outpot of  <code class="language-plaintext highlighter-rouge">Output dataset out_chr_sorted_circos from control freec</code></li>
      </ul>
    </li>
  </ol>

  <ul>
    <li>In<em>“Ideogram:”</em>:
      <ul>
        <li>In <em>“1: Ideogram:”</em>
          <ul>
            <li><em>Spacing Between Ideograms (in chromosome units)”</em>: <code class="language-plaintext highlighter-rouge">3.0</code></li>
          </ul>
        </li>
        <li>In <em>“2: Ideogram:”</em>
          <ul>
            <li><em>“Radius”</em>: <code class="language-plaintext highlighter-rouge">0.8</code></li>
          </ul>
        </li>
        <li>In <em>“3: Ideogram:”</em>
          <ul>
            <li><em>“Thickness”</em>: <code class="language-plaintext highlighter-rouge">45.0</code></li>
          </ul>
        </li>
        <li>In <em>“4: Ideogram:”</em>
          <ul>
            <li>In<em>“Labels:”</em>:
              <ul>
                <li>In<em>“1: Labels:”</em>
                  <ul>
                    <li><em>“Radius”</em>: <code class="language-plaintext highlighter-rouge">0.01</code></li>
                  </ul>
                </li>
                <li>In<em>“2: Labels:”</em>
                  <ul>
                    <li><em>“Label Font Size”</em>: <code class="language-plaintext highlighter-rouge">40</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“5: Ideogram:”</em>
          <ul>
            <li>In<em>“Cytogenic Bands:”</em>:
              <ul>
                <li>In<em>“1: Cytogenic Bands:”</em>
                  <ul>
                    <li><em>“Band Stroke Color”</em>: <code class="language-plaintext highlighter-rouge">Black</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>In<em>“2D Data Tracks:”</em>:
      <ul>
        <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> “Insert 2D Data Plot”*
          <ul>
            <li>In<em>“2D Data Plots:”</em>:
              <ul>
                <li>In <em>“1: 2D Data Plots:”</em>
                  <ul>
                    <li><em>“Outside Radius”</em>: <code class="language-plaintext highlighter-rouge">0.95</code></li>
                  </ul>
                </li>
                <li>In <em>“2: 2D Data Plots:”</em>
                  <ul>
                    <li><em>“Plot Type”</em>: <code class="language-plaintext highlighter-rouge">Scatter</code></li>
                  </ul>
                </li>
                <li>In <em>“3: 2D Data Plots:”</em>
                  <ul>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Scatter Plot Data Source”</em>: <code class="language-plaintext highlighter-rouge">Output dataset 'out_ratio_log2_circos' from Control-FreeC</code></li>
                  </ul>
                </li>
                <li>In <em>“4: 2D Data Plots:”</em>
                  <ul>
                    <li>In<em>“Plot Format Specific Options:”</em>:
                      <ul>
                        <li>In <em>“1: Plot Format Specific Options:”</em>
                          <ul>
                            <li><em>“Glyph”</em>: <code class="language-plaintext highlighter-rouge">Circle</code></li>
                          </ul>
                        </li>
                        <li>In <em>“2: Plot Format Specific Options:”</em>
                          <ul>
                            <li><em>“Glyph Size”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
                          </ul>
                        </li>
                        <li>In <em>“3: Plot Format Specific Options:”</em>
                          <ul>
                            <li><em>“Fill Color”</em>: <code class="language-plaintext highlighter-rouge">Gray</code></li>
                          </ul>
                        </li>
                        <li>In <em>“4: Plot Format Specific Options:”</em>
                          <ul>
                            <li><em>“Stroke Color”</em>: <code class="language-plaintext highlighter-rouge">Black</code></li>
                          </ul>
                        </li>
                        <li>In <em>“5: Plot Format Specific Options:”</em>
                          <ul>
                            <li><em>“Stroke Thickness”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>In <em>“5: 2D Data Plots:”</em>
              <ul>
                <li>In<em>“Rules:”</em>:
                  <ul>
                    <li>In <em>“1: Rules:”</em>
                      <ul>
                        <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Rule”</em>:
                          <ul>
                            <li>In<em>“Rule 1”</em>
                              <ul>
                                <li>In <em>“1: Rule 1”</em>
                                  <ul>
                                    <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Conditions to Apply”</em>
                                      <ul>
                                        <li>In<em>” Conditions to Applies”</em>
                                          <ul>
                                            <li>In <em>“1: Conditions to Applies”</em>
                                              <ul>
                                                <li><em>“Condition”</em>: <code class="language-plaintext highlighter-rouge">Based on value (ONLY for scatter/histogram/heatmap/line)</code></li>
                                              </ul>
                                            </li>
                                            <li>In <em>“2: Conditions to Applies”</em>
                                              <ul>
                                                <li><em>“Points above this value”</em>: <code class="language-plaintext highlighter-rouge">0.0</code></li>
                                              </ul>
                                            </li>
                                          </ul>
                                        </li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                                <li>In <em>“2: Rule 1”</em>
                                  <ul>
                                    <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Actions to Apply”</em>
                                      <ul>
                                        <li>In<em>“Actions to Applies”</em>
                                          <ul>
                                            <li>In <em>“1: Actions to Applies”</em>
                                              <ul>
                                                <li><em>“Action”</em>: <code class="language-plaintext highlighter-rouge">Change Fill Color for all points</code></li>
                                              </ul>
                                            </li>
                                            <li>In <em>“2: Actions to Applies”</em>
                                              <ul>
                                                <li><em>“Fill Color”</em>: <code class="language-plaintext highlighter-rouge">Red</code></li>
                                              </ul>
                                            </li>
                                          </ul>
                                        </li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                                <li>In <em>“3: Rule 1”</em>
                                  <ul>
                                    <li><em>“Continue flow”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li>In <em>“2: Rules:”</em>
                      <ul>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Rule”</em>:
                          <ul>
                            <li>In<em>“Rule 2”</em>
                              <ul>
                                <li>In <em>“1: Rule 2”</em>
                                  <ul>
                                    <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Conditions to Apply”</em>:
                                      <ul>
                                        <li>In<em>” Conditions to Applies”</em>
                                          <ul>
                                            <li>In <em>“1: Conditions to Applies”</em>
                                              <ul>
                                                <li><em>“Condition”</em>: <code class="language-plaintext highlighter-rouge">Based on value (ONLY for scatter/histogram/heatmap/line)</code></li>
                                              </ul>
                                            </li>
                                            <li>In <em>“2: Conditions to Applies”</em>
                                              <ul>
                                                <li><em>“Points below this value”</em>: <code class="language-plaintext highlighter-rouge">0.0</code></li>
                                              </ul>
                                            </li>
                                          </ul>
                                        </li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                                <li>In <em>“2: Rule 2”</em>
                                  <ul>
                                    <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Actions to ApplyInsert Actions to Apply”</em>:
                                      <ul>
                                        <li>In<em>“Actions to Applies”</em>
                                          <ul>
                                            <li>In <em>“1: Actions to Applies”</em>
                                              <ul>
                                                <li><em>“Action”</em>: <code class="language-plaintext highlighter-rouge">Change Fill Color for all points</code></li>
                                              </ul>
                                            </li>
                                            <li>In <em>“2: Actions to Applies”</em>
                                              <ul>
                                                <li><em>“Fill Color”</em>: <code class="language-plaintext highlighter-rouge">Blue</code></li>
                                              </ul>
                                            </li>
                                          </ul>
                                        </li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                            <li>In <em>“3: Rule 2”</em>
                              <ul>
                                <li><em>“Continue flow”</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>In <em>“6: 2D Data Plots:”</em>
                  <ul>
                    <li>In<em>“Axes:”</em>:
                      <ul>
                        <li>In “1: Axes:”
                          <ul>
                            <li><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> <em>“Insert Conditions to Apply”</em></li>
                            <li>In<em>“Axis 1:”</em>:
                              <ul>
                                <li>In <em>“1: Axis 1”</em>
                                  <ul>
                                    <li><em>“Radial Position”</em>: <code class="language-plaintext highlighter-rouge">Absolute position (values match data values)</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“2: Axis 2”</em>
                                  <ul>
                                    <li><em>“Spacing”</em>: <code class="language-plaintext highlighter-rouge">1.0</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“3: Axis 1”</em>
                                  <ul>
                                    <li><em>“y0”</em>: <code class="language-plaintext highlighter-rouge">-4.0</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“4: Axis 1”</em>
                                  <ul>
                                    <li><em>“y1”</em>: <code class="language-plaintext highlighter-rouge">4.0</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“5: Axis 1”</em>
                                  <ul>
                                    <li><em>“Color”</em>: <code class="language-plaintext highlighter-rouge">Gray</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“6: Axis 1”</em>
                                  <ul>
                                    <li><em>“Color Transparency”</em>: <code class="language-plaintext highlighter-rouge">1.0</code></li>
                                  </ul>
                                </li>
                                <li>In <em>“7: Axis 1”</em>
                                  <ul>
                                    <li><em>“Thickness”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                        <li>In “2: Axes:”
                          <ul>
                            <li><em>“When to show”</em>: <code class="language-plaintext highlighter-rouge">Always</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>In<em>“Limits:”</em>:
      <ul>
        <li>In <em>“1: Limits:”</em>
          <ul>
            <li><em>“Maximum number of links to draw”</em>: <code class="language-plaintext highlighter-rouge">2500000</code></li>
          </ul>
        </li>
        <li>In <em>“2: Limits:”</em>
          <ul>
            <li><em>“Maximum number of points per track”</em>: <code class="language-plaintext highlighter-rouge">2500000</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Can you interpret generated plot from the Circos tool?
<a href="../../images/Control_Freec-%5BCircos_Plot%5D.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec-%5BCircos_Plot%5D.png" alt="Cicros. " loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> 	This circos plot shows of genetic alterations in chromosomes 5,12 and 17 tutmer tissues. The outermost circle represents the targeted chromosomes and inner circle for deletion related and amplificated related variations&lt;/figcaption&gt;</figure></a>
<a href="../../images/Control_Freec_Cicros_zoomed.png" rel="noopener noreferrer"><figure id="figure-7" style="max-width: 90%; margin:auto;"><img src="../../images/Control_Freec_Cicros_zoomed.png" alt="Cicros. " width="886" height="819" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Zoomed hCNV regions detected by Control_FREEC using Cicros&lt;/figcaption&gt;</figure></a></p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The outermost circle represents the targeted chromosomes. 
The inner circle
shows the copy number changes in those regions.</p>

    <p>The inner circle has two parts. The deletion-related variations are the red dots directed to the centre 
(below the centre line), while the amplificated variations are the green dots pointed out of the center (toward chromosomes).</p>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial, we introduced Contol-FreeC as an alternative tool for detecting hCNVs and highlighted the steps for preparing reads and analysis.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Follow best practices for read mapping, quality control and mapped reads postprocessing to minimize false-positive hCNVs.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Meienberg2015">Meienberg, J., K. Zerjavic, I. Keller, M. Okoniewski, A. Patrignani <i>et al.</i>, 2015 <b>New insights into the performance of human whole-exome capture platforms</b>. Nucleic Acids Research 43: e76–e76. <a href="https://doi.org/10.1093/nar/gkv216">10.1093/nar/gkv216</a></li>
<li id="Nam2015">Nam, J.-Y., N. K. D. Kim, S. C. Kim, J.-G. Joung, R. Xi <i>et al.</i>, 2015 <b>Evaluation of somatic copy number estimation tools for whole-exome sequencing data</b>. Briefings in Bioinformatics 17: 185–192. <a href="https://doi.org/10.1093/bib/bbv055">10.1093/bib/bbv055</a></li>
<li id="Beacon">Beacon, E. L. I. X. I. R. <b>A Driver Project of the Global Alliance for Genomics and Health GA4GH and supported through ELIXIR</b>. <a href="https://beacon-project.io/">https://beacon-project.io/</a></li>
<li id="OpenEBench">ELIXIR <b>The ELIXIR gateway to benchmarking communities, software monitoring, and quality metrics for life sciences tools and workflows</b>. <a href="https://openebench.bsc.es/">https://openebench.bsc.es/</a></li>
<li id="WorkflowHub">WorkflowHub <b>WorkflowHub is a registry for describing, sharing and publishing scientific computational workflows</b>. <a href="https://workflowhub.eu/">https://workflowhub.eu/</a></li>
<li id="GA4GH">Genomics, G. A. for, and Health <b>GA4GH Work Streams develop standards and tools that are founded on the Framework for Responsible Sharing of Genomic and Health-Related Data. Their work is designed to enable international genomic data sharing based on the specific needs of clinical and research Driver Projects — real-world genomic data initiatives sourced from around the globe</b>. <a href="https://www.ga4gh.org/">https://www.ga4gh.org/</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Somatic Variant Discovery from WES Data Using Control-FREEC (Variant Analysis)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Khaled Jum'ah, Katarzyna Kamieniecka, Wolfgang Maier, David Salgado, Krzysztof Poterlowicz,  <b>Somatic Variant Discovery from WES Data Using Control-FREEC (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-somatic-variant-discovery,
author = "Khaled Jum'ah and Katarzyna Kamieniecka and Wolfgang Maier and David Salgado and Krzysztof Poterlowicz",
	title = "Somatic Variant Discovery from WES Data Using Control-FREEC (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa
  owner: devteam
  revisions: 3fe632431b68
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa
  owner: devteam
  revisions: e188dc7a68e6
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: e7b2202befea
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 3d0c7bdf12f5
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: freebayes
  owner: devteam
  revisions: 57def2d7c093
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_calmd
  owner: devteam
  revisions: 06dc50b7b711
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_rmdup
  owner: devteam
  revisions: 586f9e1cdb2b
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: circos
  owner: iuc
  revisions: df7356989ac1
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: circos
  owner: iuc
  revisions: 31a35811dda6
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: control_freec
  owner: iuc
  revisions: 2c6349fb175c
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: control_freec
  owner: iuc
  revisions: f923c54a17ee
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: 9a913cdee30e
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: bf675f34b056
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_view
  owner: iuc
  revisions: b72793637686
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trimmomatic
  owner: pjbriggs
  revisions: dfa082f84068
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trimmomatic
  owner: pjbriggs
  revisions: d94aff5ee623
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00317">https://gxy.io/GTN:T00317</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1706401745'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>