<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Introduction to metatranscriptomics</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metatranscriptomics/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1700958544' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1700958544' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Introduction to metatranscriptomics">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-26 00:29:07 +0000">
<meta name="DC.creator" content="Subina Mehta">
<meta name="DC.creator" content="Pratik Jagtap">
<meta name="DC.creator" content="Saskia Hiltemann">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Introduction to metatranscriptomics">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1700958544'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/metagenomics/tutorials/metatranscriptomics/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/metatranscriptomics/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/metagenomics/tutorials/metatranscriptomics/slides.html">View markdown source on GitHub</a>
	<h1>Introduction to metatranscriptomics</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/subinamehta/" class="contributor-badge contributor-subinamehta"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/subinamehta?s=36" alt="Avatar" width="36" height="36">Subina Mehta</a><a href="/training-material/hall-of-fame/pratikdjagtap/" class="contributor-badge contributor-pratikdjagtap"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pratikdjagtap?s=36" alt="Avatar" width="36" height="36">Pratik Jagtap</a><a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Avatar" width="36" height="36">Saskia Hiltemann</a>
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>How to analyze metatranscriptomics data?</p>
</li>
		
		<li><p>What information can be extracted of metatranscriptomics data?</p>
</li>
		
		<li><p>How to assign taxa and function to the identified sequences?</p>
</li>
		
		</ul>
		

		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Nov 26, 2023 </div>

	<hr />


	<h2 id="why-study-the-microbiome">Why study the microbiome?</h2>

<p>.pull-left[</p>

<ul>
  <li>Health care research</li>
  <li>Humans are full of microorganisms</li>
  <li>Skin, gut, oral cavity, nasal cavity, eyes, ..</li>
  <li>Affects health, drug efficacy, etc</li>
</ul>

<p>]
.pull-right[</p>

<p>.image-100[ <img src="../../images/human_microbiome.png" alt="Image of a human with various pie charts pointing to various regions of the body where microbe populations live" /> ]</p>

<p>]</p>

<ul>
  <li>Sometimes referred to as your <strong>second genome</strong></li>
  <li>~10 times more cells than you</li>
  <li>~100 times more genes than you</li>
  <li>~1000s different species</li>
</ul>

<hr />

<h2 id="why-study-the-microbiome-1">Why study the microbiome?</h2>

<ul>
  <li>Environmental studies
    <ul>
      <li>Microbes in the soil affect plants and animals</li>
      <li>Improve agriculture</li>
    </ul>
  </li>
</ul>

<p>.image-75[ <img src="../../images/environmentalstudies.png" alt="Rhizodeposition: image of a tree converting sun and co2 into fixed carbon used as food for soil microbes." /> ]</p>

<hr />

<h2 id="meta--omics">Meta- Omics</h2>

<p><img src="../../images/metatranscriptomics/meta-omics.png" alt="meta-momics diagram" /></p>

<hr />

<h2 id="this-tutorial-asaim-pipeline">This Tutorial: ASaiM pipeline</h2>

<p>.pull-left[</p>

<ul>
  <li>Quality Control
    <ul>
      <li>Assess Quality</li>
      <li>Trim and Filter raw reads</li>
      <li>Filter ribosomal RNA (rRNA)</li>
    </ul>
  </li>
  <li>Community profiling (Who?)
    <ul>
      <li>Determine composition of sample</li>
      <li>Visualisation</li>
    </ul>
  </li>
  <li>Functional Analysis (What?)</li>
</ul>

<p>]</p>

<p>.pull-right[
.image-90[<img src="../../images/asaim-wf.png" alt="ASiaiM diagram" />]
]</p>

<p>.footnote[Batut et al Gigascience. 2018 7(6) doi: 10.1093/gigascience/giy057]</p>

<p><span>Speaker Notes</span></p>

<p>For this short tutorial, while the workflow is running, these slides can be useful
to explain the tools that are being run in that section. After explaining the tools,
the workflows should be far enough along to start showing the results</p>

<hr />

<h2 id="input-cellulose-degradation-in-a-biogas-reactor">Input: Cellulose Degradation in a Biogas Reactor</h2>

<p><img src="../../images/metatranscriptomics/experimental_setup.png" alt="Workflow graph showing biogas reactor extract being transferred to cellulose and incubated. Time series samples are taken and run through a mass spectrometer and genomic sequencer." /></p>

<p><span>Speaker Notes</span></p>

<p>A 100 µl aliquot of an enriched community from a biogas reactor was transferred to 27 anaerobic bottles containing a rich medium and 10g/L of cellulose as sole carbon source and incubated at 65 °C.</p>

<p>Three bottles were collected at 9 different time points (0, 8, 13, 18, 23, 28, 33, 38 and 43 h) and processed in triplicates. Metatranscriptomic analysis was performed on all time points. Metaproteomics analysis on 4 data points.</p>

<hr />

<h2 id="input-format-fastq-files">Input Format: FastQ Files</h2>

<ul>
  <li>Four lines per read</li>
</ul>

<p><img src="../../../sequence-analysis/images/quality-control/fastq_fig.jpg" alt="Image of a fastq file with label on the first line, sequence on the second, + on the third, and quality scores on the fourth as ascii chars. A callout shows that Base=T, quality=colon, and that means a score of 25." /></p>

<p><span>Speaker Notes</span></p>

<ul>
  <li>Four lines per read
    <ul>
      <li><code class="language-plaintext highlighter-rouge">@</code> + identifier on first line, just like fasta</li>
      <li>sequence</li>
      <li><code class="language-plaintext highlighter-rouge">+</code></li>
      <li>quality score characters</li>
    </ul>
  </li>
</ul>

<p>Segue: so what do the quality chars mean?</p>

<hr />

<h2 id="fastq-quality-score">FastQ: Quality score</h2>

<ul>
  <li>
    <p>Each character denotes a different Phred score</p>

    <p><img src="../../../sequence-analysis/images/quality_score_encoding.png" alt="Encoding of the quality score with ASCII characters for different Phred encoding. The ascii code sequence is shown at the top with symbols for 33 to 64, upper case letters, more symbols, and then lowercase letters. Sanger maps from 33 to 73 while solexa is shifted, starting at 59 and going to 104. Illumina 1.3 starts at 54 and goes to 104, Illumina 1.5 is shifted three scores to the right but still ends at 104. Illumina 1.8+ goes back to the Sanger except one single score wider. Illumina" /></p>
  </li>
  <li>
    <p>Phred scores are logarithmic</p>
  </li>
</ul>

<p>.small[
Phred Quality Score | Probability of incorrect base call | Base call accuracy
— | — | —
10 | 1 in 10 | 90%
20 | 1 in 100 | 99%
30 | 1 in 1000 | 99.9%
40 | 1 in 10,000 | 99.99%
…
]</p>

<p><span>Speaker Notes</span></p>

<ul>
  <li>Logarithmic scale</li>
  <li>Different flavours of encoding exist</li>
</ul>

<hr />

<h2 id="preprocessing">Preprocessing</h2>

<p>In this tutorial we start with some preprocessing steps</p>

<p><img src="../../images/metatranscriptomics/workflow_qc.png" alt="preprocessing workflow" /></p>

<hr />

<h2 id="preprocessing-tools">Preprocessing: Tools</h2>

<p>In this tutorial we start with some preprocessing steps</p>

<p><img src="../../images/metatranscriptomics/workflow_qc_tools.png" alt="preprocessing workflow" /></p>

<p><span>Speaker Notes</span>
| Step | Tools |
|:——|——-|
|Quality Control reports         | FastQC <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and MultiQC <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> |
|Trimming and Filtering | Cutadapt <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> |
|Filter ribosomal RNA | SortMeRNA <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> |
|Interlace FastQ files | FastQ interlacer <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> |</p>

<hr />

<h2 id="quality-reports-fastqc">Quality Reports: FastQC</h2>

<ul>
  <li>Generate a web report with quality metrics of your FastQ file</li>
</ul>

<p><img src="../../../sequence-analysis/images/fastqc-report.png" alt="Screenshot of FastQC report, showing the table of contents with green checks on nearly every result, and the base statistics and per-base sequence quality graphs shown." /></p>

<p>.footnote[see also our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">dedicated QC tutorial</a>  ]</p>

<hr />

<h2 id="quality-reports-fastqc-1">Quality Reports: FastQC</h2>

<ul>
  <li>Many different QC plots</li>
  <li>Example: Per-base sequence Quality plot</li>
</ul>

<p><img src="../../../sequence-analysis/images/per_base_sequence_quality_good.png" alt="Fastqc quality score plot, most results are in the green region but the box portion of the box and whisker plot start to dip into the yellow, medium quality (less than 30) region near 34+ base position in read. The whiskers begin extending to the red region (less than 20) by base 31 and get progressively worse." /></p>

<p>.footnote[explanation of different plots: <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">dedicated QC tutorial</a>  ]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>in the per-base sequence quality plot, a boxplot of the base quality (y-axis) per position in the read (x-axis) is drawn</li>
  <li>often you might observe a drop in quality towards the end of the reads, and may consider trimming ends</li>
  <li>this example is very good</li>
</ul>

<hr />

<h2 id="quality-reports-fastqc-2">Quality Reports: FastQC</h2>

<ul>
  <li>Many more plots</li>
  <li>See <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">QC tutorial</a> for more information</li>
</ul>

<p><img src="../../../sequence-analysis/images/quality-control/all_plots.png" alt="Montage of several different fastq reports showing sequence quality graphs, and a numb er of other line graphs." /></p>

<hr />

<h2 id="quality-reports-multiqc">Quality Reports: MultiQC</h2>

<p>.pull-left[</p>
<ul>
  <li>
    <p>Combine multiple FastQC reports into one report</p>
  </li>
  <li>
    <p>Also for outputs of other tools</p>
  </li>
  <li>
    <p>Great when sequencing large numbers of samples
]</p>
  </li>
</ul>

<p>.pull-right[
<img src="../../images/metatranscriptomics/multiqc.png" alt="Multiqc's report showing an aggregation of multiple samples. An overview at the top provides context for the 4 samples, and a sequence quality histogram shows 4 samples with similarly behaving quality scores" />
]</p>

<hr />

<h2 id="read-trimming-and-filtering-cutadapt">Read Trimming and Filtering: Cutadapt</h2>

<ul>
  <li>Trim low-quality bases from reads</li>
  <li>Filter reads based on length, mean quality score, ..</li>
  <li>
    <p>Remove adapters/primers</p>

    <p>.image-75[<img src="../../../sequence-analysis/images/quality-control/trimming.png" alt="adapter trimming" />]</p>
  </li>
  <li>Many tools: CutAdapt <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, TrimGalore <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, Trimmomatic <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> ..</li>
</ul>

<p><span>Speaker Notes</span>
These are some examples of ways to trim and filter data, but many more are possible and depend on your experiment what is necessary</p>

<hr />

<h2 id="sortmerna">SortMeRNA</h2>

<p>.pull-left[</p>

<ul>
  <li>
    <p>Most RNA sequences will be ribosomal RNA (rRNA)</p>
  </li>
  <li>
    <p>Great for taxonomical assignment (who is there?)</p>
  </li>
  <li>
    <p>Not informative for functional analysis (What are they doing?)</p>
  </li>
  <li>
    <p>Filter out rRNA before doing functional analysis
]</p>
  </li>
</ul>

<p>.pull-right[
.image-90[<img src="../../images/metatranscriptomics/sortmerna.png" alt="SortMeRNA" />]
]</p>

<hr />

<h2 id="fastq-interlacer">FastQ interlacer</h2>

<ul>
  <li>Paired-end data often comes in two separate FastQ files</li>
  <li>One file with <em>forward</em> reads, one with <em>reverse</em> reads</li>
</ul>

<p><br /></p>

<p><img src="../../../sequence-analysis/images/mapping/pairedend_fastq.png" alt="paired end deinterlaced file" /></p>

<hr />

<h2 id="fastq-interlacer-1">FastQ interlacer</h2>

<ul>
  <li>Some tools require a single <strong>interlaced</strong> FastQ file</li>
  <li>Galaxy has tools for <strong>interlacing</strong> and <strong>deinterlacing</strong> FastQ files</li>
</ul>

<p><br /></p>

<p><img src="../../../sequence-analysis/images/mapping/pairedend_interlaced.png" alt="paired end interlaced file" /></p>

<p><span>Speaker Notes</span>
forward and reverse files are ‘zipped’ together into a single file</p>

<hr />

<h2 id="community-profile">Community Profile</h2>

<ul>
  <li>We want to identify which organisms are present in our sample, and their relative abundances</li>
</ul>

<p><img src="../../images/microbial-community.png" alt="Cartoon of several differently coloured and shaped microbes in a circle." /></p>

<ul>
  <li><strong>MetaPhlan2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for identification</li>
  <li><strong>Krona</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>Graphlan</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for visualisation</li>
</ul>

<hr />

<h2 id="metaphlan2-tool">MetaPhlan2 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></h2>

<ul>
  <li>Estimates the presence and relative abundance of microbial cells</li>
  <li>Maps reads against a set of marker sequences</li>
  <li>Caveat: this tool is designed for DNA-seq
    <ul>
      <li>Be careful interpreting abundances when using this tool with transcriptomics data</li>
    </ul>
  </li>
</ul>

<p>.footnote[Nat Methods. 2012 Jun 10;9(8):811-4. doi: 10.1038/nmeth.2066.]</p>

<p><span>Speaker Notes</span></p>

<p>About the caveat: The theoretical problem is that we quantify species abundance by averaging the coverage of marker genes. Marker genes are supposed to be at the same coverage as they are single copy genes from the same genome, but this is not true for their transcripts. So MetaPhlAn2 on metatranscriptomics gives an idea about the average transcriptional rate of a given species. So it <em>can</em> be used with caution…</p>

<hr />

<h2 id="krona-tool">Krona <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></h2>

<ul>
  <li>Visualization of community composition, interactive plot</li>
</ul>

<iframe id="krona" src="krona.html" frameborder="0" width="80%" height="600px"> ![Krona at bacteria level](../../images/metatranscriptomics/krona_bacteria.png) </iframe>

<hr />

<h2 id="graphlan-tool">Graphlan <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></h2>

<ul>
  <li><em>Cladogram</em> visualisation</li>
</ul>

<p><img src="../../images/metatranscriptomics/graphlan_general.png" alt="Colourful cladogram which begins from the center and expands outward with the lineage of the samples. Each sector of the chart is coloured differently for each group of genus and spieces. E.g. streptococcus streptococcaceae has three different leafs of the cladogram tree." /></p>

<hr />

<h2 id="genus-abundance">Genus Abundance</h2>

<ul>
  <li>Tutorial: one timepoint</li>
  <li>Over multiple timepoints:</li>
</ul>

<p><img src="../../images/metatranscriptomics/genus_abundance.png" alt="stacked bar chart with timepoints along the x axis and genus abundance as a percentage along the y axis. Each of the 7 time samples consists mostly of Coprothermobacter and Clostridium." /></p>

<hr />

<h2 id="functional-analysis">Functional Analysis</h2>

<ul>
  <li>Pathways</li>
  <li>Gene Ontology
    <ul>
      <li>Biological process</li>
      <li>Molecular function</li>
      <li>Cellular component</li>
    </ul>
  </li>
  <li>Gene Family</li>
</ul>

<hr />

<h2 id="workflow">Workflow</h2>

<p><img src="../../images/metatranscriptomics/workflow_functional.png" alt="functional analysis workflow schematic" /></p>

<p><span>Speaker Notes</span>
HUMAnN2</p>
<ul>
  <li>next generation</li>
  <li>HMP Unified Metabolic Analysis Network</li>
  <li>developed by Huttenhower lab</li>
  <li>itself a workflow/pipeline</li>
  <li>basically answering the question about what the microbial community is capable of?</li>
</ul>

<hr />

<h2 id="humann2-tool">HUMAnN2 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></h2>

<ul>
  <li>Profiles presence/absence and abundance of microbial community</li>
  <li>
    <p>Efficiently characterizes microbial metabolic pathways</p>
  </li>
  <li><strong>Input</strong>
    <ul>
      <li>Interlaced non-rRNA reads</li>
      <li>Taxonomic profile (MetaPhlAn2 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    </ul>
  </li>
  <li><strong>Output</strong>
    <ul>
      <li>Gene families and their abundance</li>
      <li>Pathways and their coverage</li>
      <li>Pathways and their abundance</li>
    </ul>
  </li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>contains 5 parts -&gt; non rRNA reads, MetaPhlAn taxonomy, NCBI nucleotide db, Uniref 50/90 protein db, MetaCyc/Unipathway.</li>
  <li>Show the Galaxy wrapper</li>
</ul>

<hr />

<h2 id="humann2-tiered-search">HUMAnN2 Tiered Search</h2>

<hr />
<p>class: top</p>

<p>.left-column70[</p>

<p><br /></p>

<ul>
  <li>Meta-omic sequences (DNA/RNA)</li>
</ul>

<p>]</p>

<p>.right-column30[
.image-60[<img src="../../images/metatranscriptomics/humann2_tiered1.png" alt="Cartoon of several reads coloured into four groups, Species 1, Species 2, Unclassified, Novel." />]
]</p>

<p>–</p>

<p>.left-column70[</p>

<p><br /></p>

<ul>
  <li>Initial screen through <strong>MetaPhlAn2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: known microbial species
    <ul>
      <li>Database: merging pangenomes of identified species</li>
    </ul>
  </li>
</ul>

<p>]</p>

<p>.right-column30[
.image-60[<img src="../../images/metatranscriptomics/humann2_tiered2.png" alt="Four bins labelled 1 (red), 2 (blue), 3, 4 with reads from the top cartoon show piles of 1 an 2 with NO signs over 3 and 4." />]
]</p>

<p>–</p>

<p>.left-column70[</p>

<p><br /></p>
<ul>
  <li>Nucleotide-level mapping against database</li>
</ul>

<p>]</p>

<p>.right-column30[
.image-60[<img src="../../images/metatranscriptomics/humann2_tiered3.png" alt="Regions x 1 y in red and x 2 y in blue are shown, the pangenomes of each of the red and blue species are shown. Reads map to most segments of the pangenome." />]
]</p>

<p>–</p>

<p>.left-column70[
<br /></p>
<ul>
  <li>Unaligned reads searched against proteinDB (Uniref) through accelerated translated search</li>
</ul>

<p>]</p>

<p>.right-column30[
.image-75[<img src="../../images/metatranscriptomics/humann2_tiered4.png" alt="Reads are shown matching against portions of protein sequences of X, Y, Z" />]
]</p>

<p><span>Speaker Notes</span>
Takes non rRNA reads + MetaPhlAn2 gives list of abundant organism, then it does Nucleotide level pangenome mapping with Bowtie and uses CHocophlAN db
giving unmapped and organims specific gene hits, the unmapped reads are further searched against accelerated translated protein database
the protein hits are tehn combined with gene hits and metacyc to give the output.</p>

<hr />

<p><strong>Result:</strong> Gene family and pathway abundances</p>

<table>
  <tbody>
    <tr>
      <td>.image-40[![A table with two columns, Feature on left and RPK on right. GeneX has an RPK of 8. GeneX</td>
      <td>Species1 has an RPK of 2, species2 and unclassified are listed with an RPK of 3.](../../images/metatranscriptomics/humann2_tiered5.png)]</td>
    </tr>
  </tbody>
</table>

<hr />

<h2 id="gene-families-abundances">Gene Families Abundances</h2>

<p><img src="../../images/metatranscriptomics/gene_family_abundance.png" alt="Screenshot of a table in Galaxy with Gene Family on left and humann2 abundance RPKs on right." /></p>

<p>RPK (reads per kilobase) = sum of alignment scores</p>

<p><span>Speaker Notes</span>
Gene families: groups of evolutionary related protein that perform similar function
Pathway: sum over genes catalyzing the reaction
Pathway coverage: presence/absence
RPK relative gene copy number : is computed as the sum of all alignments scores over a particular gene family
UNMAPPED: total number of reads that remained unmapped even after both alignment steps
UNINTEGRATED: no pathway detected.</p>

<hr />

<h2 id="gene-families-to-functional-annotation">Gene Families to Functional Annotation</h2>

<p><img src="../../images/metatranscriptomics/results_functional_annotation.png" alt="Humann2 regroup table is the left node in a flow chart with UniRef50. Multiple lines are drawn to an unlabelled right node that lists metacyc, kegg, pfam, EC, GO, informative GO, slim GO." /></p>

<p><span>Speaker Notes</span>
Gene familes are too large depending on the complexity thus to simplify users can regroup gene families using grouping tool, can download mapping files.
HUMAnN2 regroups Uniref 50/90 values to Go terms to get a broad overview.</p>

<hr />

<h2 id="group-abundances">Group Abundances</h2>

<p><img src="../../images/metatranscriptomics/results_group_abundances.png" alt="humann2 regroup table, lines from uniref50 to GO. group humann2 to GO slim terms shows a similar graphic, lines from uniref50 to slim GO." /></p>

<p><span>Speaker Notes</span>
Group abundances converts GO terms to Go slim (subset of GO terms) into Mol function, biological process and cellular components.</p>

<hr />

<h2 id="gene-families-to-functional-annotation-1">Gene Families to Functional Annotation</h2>

<p><img src="../../images/metatranscriptomics/results_gene_family.png" alt="Table from Galaxy shown with gene family and RPK" /></p>

<p><img src="../../images/metatranscriptomics/results_gene_family_function.png" alt="group Human2 to GO slim terms with lines from uniref50 to slim GO and boxes of Molecular Function, biological process, and cellular component below Slim GO" /></p>

<p><img src="../../images/metatranscriptomics/results_go.png" alt="Another galaxy table screenshot with GO id, GO name, and abundance." /></p>

<hr />
<p>class: top</p>

<h2 id="output">Output</h2>

<p>.left-column30[
<br />
Molecular Function</p>

<p>]</p>

<p>.right-column70[
.image-90[<img src="../../images/metatranscriptomics/results_molecular_function.png" alt="Table in Galaxy with GO ID, name, abundance." />]
]
–</p>

<p>.left-column70[
.image-90[<img src="../../images/metatranscriptomics/results_biological_process.png" alt="Basically the same table as above." />]
]</p>

<p>.right-column30[
<br />
Biological Process</p>

<p>]</p>

<p>–</p>

<p>.right-column70[
.image-90[<img src="../../images/metatranscriptomics/results_cellular_component.png" alt="Again the same columns in a table. None of the specific data is legible or important." />]
]</p>

<p>.left-column30[</p>

<p><br /><br />
Cellular Component</p>

<p>]</p>

<p><span>Speaker Notes</span>
g is genus
s is species level
—</p>

<h2 id="unpack-pathway-abundances-to-show-genes-included">Unpack pathway abundances to show genes included</h2>

<ul>
  <li>Renormalize the gene and pathway abundances in copies per million or relative abundance</li>
  <li>This tool unpacks the pathways abundance by including gene families</li>
</ul>

<p><img src="../../images/metatranscriptomics/results_unpack.png" alt="output file unpack pathway tool" /></p>

<hr />

<h2 id="function-cellulose-degradation">Function: Cellulose Degradation</h2>

<ul>
  <li>Quantitative analysis of gene family outputs from HUMAnN2 shows upregulation of cellulase</li>
</ul>

<p>.image-75[<img src="../../images/metatranscriptomics/results_cellulose_degradation.png" alt="line chart shown cellulase abundance decreasing from 80 copies per million to 40 as time goes from 13 to 43. Cellulose 1,4 beta cellobiosidase starts at 140 cpm and dps at hour 23 to 120 before increasing to 200 by the end of the graph" />]</p>

<p><span>Speaker Notes</span>
explain about datasets first
cellulose 1,4 beta-cellulobiosidase responsible for hydrolysis of cellulose
Gene encoding for the cellulose-binding domain
protein shows an initial decrease and subsequent
increase during cellulose degradation.
—</p>

<h2 id="functions-associated-with-a-selected-taxon">Functions associated with a selected taxon</h2>

<p>.image-75[<img src="../../images/metatranscriptomics/results_functions_taxon.png" alt="Stacked bar chart with a lot of organisms as left axis (abundance, copies per million) and time on the bottom. It is labelled Coprothermobacter: Functional Pathways" />]</p>

<p><span>Speaker Notes</span>
In gene abundance, Coprothermobacter and Clostridium were observed to be the
most abundant.
In this figure we are looking at Coprothermobacter only-&gt;Glycolysis is observed to be the most abundant functional pathway across time points in
Coprothermobacter</p>

<hr />

<h2 id="taxa-associated-with-a-selected-function">Taxa associated with a selected function</h2>

<p>.image-75[<img src="../../images/metatranscriptomics/results_taxa_function.png" alt="Bar chart titled Adenosine ribonucleotides de novo biosynthesis with time in hours as x axis, and Genus abundance (copies per million). Coprothermobacter and Clostridium decrease from ~2000 combined copies per million to ~800, in approximately equal amounts." />]</p>

<p><span>Speaker Notes</span>
This figure shows the contribution of genera to adenosine ribonucleotides denovo biosynthesis across time points.
it shows during ATP synthesis, we see clostridium and coprothermobacter in abundance.
—</p>

<h1 id="tabular-outputs-from-asaim-workflow">Tabular Outputs from ASaIM Workflow</h1>

<ul>
  <li>Taxonomy (Who?)
    <ul>
      <li>Kingdom, phylum, class, order, family, genus, species, strain</li>
    </ul>
  </li>
  <li>Function (What?)
    <ul>
      <li>Pathways</li>
      <li>Gene Ontology
        <ul>
          <li>Biological Process</li>
          <li>Molecular Function</li>
          <li>Cellular Component</li>
        </ul>
      </li>
      <li>Gene Family</li>
    </ul>
  </li>
</ul>



	<hr />








<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/metagenomics/tutorials/metatranscriptomics/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1700958544'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
