<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Metatranscriptomics analysis using microbiome RNA-seq data</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:title" content="Galaxy Training: Metatranscriptomics analysis using microbiome RNA-seq data" />
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/metatranscriptomics/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:20 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Metatranscriptomics analysis using microbiome RNA-seq data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
  },
  "courseCode": "metagenomics / metatranscriptomics / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Metatranscriptomics analysis using microbiome RNA-seq data' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html",
  "timeRequired": "PT5H",
  "keywords": "metatranscriptomics",
  "description": "The questions this  addresses are:\n - How to analyze metatranscriptomics data?\n - What information can be extracted of metatranscriptomics data?\n - How to assign taxa and function to the identified sequences?\n\n\\nThe objectives are:\n - Choose the best approach to analyze metatranscriptomics data\n - Understand the functional microbiome characterization using metatranscriptomic results\n - Understand where metatranscriptomics fits in 'multi-omic' analysis of microbiomes\n - Visualise a community structure\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Emma Leith"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Emma Leith"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ],
  "skillLevel": "Introductory"
}
</script>

<section class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Metatranscriptomics analysis using microbiome RNA-seq data</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/pratikdjagtap/" class="contributor-badge contributor-pratikdjagtap"><img src="https://avatars.githubusercontent.com/pratikdjagtap?s=27" alt="Avatar">Pratik Jagtap</a>


<a href="/training-material/hall-of-fame/subinamehta/" class="contributor-badge contributor-subinamehta"><img src="https://avatars.githubusercontent.com/subinamehta?s=27" alt="Avatar">Subina Mehta</a>


<a href="/training-material/hall-of-fame/jraysajulga/" class="contributor-badge contributor-jraysajulga"><img src="https://avatars.githubusercontent.com/jraysajulga?s=27" alt="Avatar">Ray Sajulga</a>


<a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="https://avatars.githubusercontent.com/bebatut?s=27" alt="Avatar">Bérénice Batut</a>


<a href="/training-material/hall-of-fame/emmaleith/" class="contributor-badge contributor-emmaleith"><img src="https://avatars.githubusercontent.com/emmaleith?s=27" alt="Avatar">Emma Leith</a>


<a href="/training-material/hall-of-fame/pravs3683/" class="contributor-badge contributor-pravs3683"><img src="https://avatars.githubusercontent.com/pravs3683?s=27" alt="Avatar">Praveen Kumar</a>


<a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=27" alt="Avatar">Saskia Hiltemann</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to analyze metatranscriptomics data?</p>
</li>
        
        <li><p>What information can be extracted of metatranscriptomics data?</p>
</li>
        
        <li><p>How to assign taxa and function to the identified sequences?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Choose the best approach to analyze metatranscriptomics data</p>
</li>
        
        <li><p>Understand the functional microbiome characterization using metatranscriptomic results</p>
</li>
        
        <li><p>Understand where metatranscriptomics fits in ‘multi-omic’ analysis of microbiomes</p>
</li>
        
        <li><p>Visualise a community structure</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 5 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4776250">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/metatranscriptomics/workflows/" title="Workflows" alt="Metatranscriptomics analysis using microbiome RNA-seq data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        
        
        

        <li>

            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/metagenomics/metatranscriptomics" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture & Tutorial (February 2021)
            </a>
        </li>
        
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/metagenomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.mesocentre.uca.fr" title="Galaxy@AuBi">
            Galaxy@AuBi
        </a>
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 20, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                
<h1 id="overview">Overview</h1>

<p>In this tutorial we will perform a metatranscriptomics analysis based on the ASAIM <span class="notranslate">workflow</span> (<span class="citation"><a href="#batut2018asaim">Batut <i>et al.</i> 2018</a></span>), using data from <span class="citation"><a href="#kunath2018proteins">Kunath <i>et al.</i> 2018</a></span>.</p>

<blockquote class="comment">
  <h3 id="comment-note-two-versions-of-this-tutorial"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: Two versions of this tutorial</h3>

  <p>Because this tutorial consists of many steps, we have made two versions of it, one long and one short.</p>

  <p>This is the <strong>extended version</strong>. We will run every tool manually and discuss the results in detail.
If you would like to run through the tutorial a bit quicker and use workflows to run groups of
analysis steps (e.g. data cleaning) at once, please see the <a href="/training-material/topics/metagenomics/tutorials/metatranscriptomics-short/tutorial.html">shorter version of this tutorial</a></p>

  <p>You can also <strong>switch</strong> between the long and short version at the start of any section.</p>
</blockquote>

<h1 class="no_toc" id="introduction">Introduction</h1>

<p>Microbiomes play a critical role in host health, disease, and the environment. The study of microbiota and microbial communities has been facilitated by the evolution of technologies, specifically the sequencing techniques. We can now study the microbiome dynamics by investigating the <span class="notranslate">DNA</span> content (metagenomics), <span class="notranslate">RNA</span> <span class="notranslate">expression</span> (metatranscriptomics), protein <span class="notranslate">expression</span> (metaproteomics) or small molecules (metabolomics):</p>

<p><img src="../../images/metatranscriptomics/Why_MT.png" alt="Multiomics" /></p>

<p>New generations of sequencing platforms coupled to numerous bioinformatics tools have led to a spectacular technological progress in metagenomics and metatranscriptomics to investigate complex microorganism communities. These techniques are giving insight into taxonomic profiles and genomic components of microbial communities. Metagenomics is packed with information about the present taxonomies in a microbiome, but do not tell much about important functions. That is where metatranscriptomics and metaproteomics play a big part.</p>

<p><img src="../../images/metatranscriptomics/meta-omics.png" alt="meta-momics diagram" /></p>

<p>In this tutorial, we will focus on <strong>metatranscriptomics</strong>.</p>

<p>Metatranscriptomics analysis enables understanding of how the microbiome responds to the environment by studying the functional analysis of genes expressed by the microbiome. It can also estimate the taxonomic composition of the microbial population. It provides scientists with the confirmation of predicted open‐reading frames (ORFs) and potential identification of novel sites of transcription and/or translation from microbial genomes. Metatranscriptomics can enable more complete generation of protein sequences databases for metaproteomics.</p>

<p>To illustrate how to analyze metatranscriptomics data, we will use data from time-series analysis of a microbial community inside a bioreactor (<span class="citation"><a href="#kunath2018proteins">Kunath <i>et al.</i> 2018</a></span>). They generated metatranscriptomics data for 3 replicates over 7 time points. <span class="notranslate">RNA</span>s were enriched by r<span class="notranslate">RNA</span> depletion and treated with <span class="notranslate">DNA</span>se and library was prepared with the TruSeq stranded <span class="notranslate">RNA</span> sample preparation, which included the production of a c<span class="notranslate">DNA</span> library.</p>

<p>In this tutorial, we focus on biological replicate A of the 1st time point. In a follow-up tutorial we will illustrate how compare the results over the different time points and replicates. The input files used here are trimmed version of the original file for the purpose of saving time and resources.</p>

<p>To analyze the data, we will follow the ASaiM <span class="notranslate">workflow</span> and explain it step by step. ASaiM (<span class="citation"><a href="#batut2018asaim">Batut <i>et al.</i> 2018</a></span>) is an open-source <span class="notranslate">Galaxy</span>-based <span class="notranslate">workflow</span> that enables microbiome analyses. Its <span class="notranslate">workflow</span> offers a streamlined <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> for users to explore metagenomic/metatranscriptomic data in a reproducible and transparent environment. The ASaiM <span class="notranslate">workflow</span> has been updated by the <strong><span class="notranslate">Galaxy</span>P</strong> team (University of Minnesota) to perform metatranscriptomics analysis of large microbial datasets.</p>

<p>The <span class="notranslate">workflow</span> described in this tutorial takes in paired-end datasets of raw shotgun sequences (in FastQ format) as an input and proceeds to:</p>

<ol>
  <li>Preprocess</li>
  <li>Extract and analyze the community structure (taxonomic information)</li>
  <li>Extract and analyze the community functions (functional information)</li>
  <li>Combine taxonomic and functional information to offer insights into taxonomic contribution to a function or functions expressed by a particular taxonomy.</li>
</ol>

<p>A graphical representation of the ASaiM <span class="notranslate">workflow</span> which we will be using today is given below:</p>

<p><img src="../../images/asaim-wf.png" alt="ASaiM diagram" width="75%" /></p>

<blockquote class="comment">
  <h3 id="comment-workflow-also-applicable-to-metagenomics-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> <span class="notranslate">Workflow</span> also applicable to metagenomics data</h3>
  <p>The approach with the tools described here can also apply to metagenomics data. What will change are the quality control profiles and proportion of r<span class="notranslate">RNA</span> sequences.</p>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#overview" id="markdown-toc-overview">Overview</a></li>
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#preprocessing" id="markdown-toc-preprocessing">Preprocessing</a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#ribosomal-rna-fragments-filtering" id="markdown-toc-ribosomal-rna-fragments-filtering">Ribosomal <span class="notranslate">RNA</span> fragments filtering</a></li>
      <li><a href="#interlace-forward-and-reverse-reads" id="markdown-toc-interlace-forward-and-reverse-reads">Interlace forward and reverse <span class="notranslate">reads</span></a></li>
    </ol>
  </li>
  <li><a href="#extraction-of-the-community-profile" id="markdown-toc-extraction-of-the-community-profile">Extraction of the community profile</a>    <ol>
      <li><a href="#community-structure-visualization" id="markdown-toc-community-structure-visualization">Community structure visualization</a></li>
    </ol>
  </li>
  <li><a href="#extract-the-functional-information" id="markdown-toc-extract-the-functional-information">Extract the functional information</a>    <ol>
      <li><a href="#normalize-the-abundances" id="markdown-toc-normalize-the-abundances">Normalize the abundances</a></li>
      <li><a href="#identify-the-gene-families-involved-in-the-pathways" id="markdown-toc-identify-the-gene-families-involved-in-the-pathways">Identify the gene families involved in the pathways</a></li>
      <li><a href="#group-gene-families-into-go-terms" id="markdown-toc-group-gene-families-into-go-terms">Group gene families into GO terms</a></li>
    </ol>
  </li>
  <li><a href="#combine-taxonomic-and-functional-information" id="markdown-toc-combine-taxonomic-and-functional-information">Combine taxonomic and functional information</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
    <li>
      <p><span class="tool" data-tool="upload1" title="Tested with upload1"><strong>Import</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span>  <code class="language-plaintext highlighter-rouge">T1A_forward</code> and <code class="language-plaintext highlighter-rouge">T1A_reverse</code> from <a href="https://zenodo.org/record/4776250">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4776250/files/T1A_forward.fastqsanger
https://zenodo.org/record/4776250/files/T1A_reverse.fastqsanger
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>

      <p>As default, <span class="notranslate">Galaxy</span> takes the link as name, so rename them.</p>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the files to <code class="language-plaintext highlighter-rouge">T1A_forward</code> and <code class="language-plaintext highlighter-rouge">T1A_reverse</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">fastqsanger</code> (e.g. <strong>not</strong> <code class="language-plaintext highlighter-rouge">fastq</code>). If it is not, please change the datatype to <code class="language-plaintext highlighter-rouge">fastqsanger</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="preprocessing">Preprocessing</h1>
<p><a href="../metatranscriptomics-short/tutorial.html#preprocessing"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">exchange</span> Switch to short tutorial</a></p>

<h2 id="quality-control">Quality control</h2>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data.</p>

<p>Sequence quality control is therefore an essential first step in your analysis.
In this tutorial we use similar tools as described in the tutorial <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control”</a>:</p>
<ul>
  <li><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> generates a web report that will aid you in assessing the quality of your data</li>
  <li><a href="https://multiqc.info/">MultiQC</a> combines multiple FastQC reports into a single overview report</li>
  <li><a href="https://cutadapt.readthedocs.io/en/stable/guide.html">Cutadapt</a> for trimming and filtering</li>
</ul>

<p><!-- only in extended tutorial --></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: both <code class="language-plaintext highlighter-rouge">T1A_forward</code> and <code class="language-plaintext highlighter-rouge">T1A_reverse</code> datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>
    </li>
    <li>
      <p>Inspect the webpage output of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <code class="language-plaintext highlighter-rouge">T1A_forward</code> dataset</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>What is the read length?</p>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The read length is 151 bp.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  with the following parameters to aggregate the FastQC reports:
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: both <code class="language-plaintext highlighter-rouge">Raw data</code> files (outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the webpage output from MultiQC for each FASTQ</li>
  </ol>

</blockquote>

<p>For more information about how to interpret the plots generated by FastQC and MultiQC, please see <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#assess-quality-with-fastqc---short--long-reads">this section</a> in our dedicated Quality Control Tutorial.</p>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Inspect the webpage output from MultiQC</p>

  <ol>
    <li>How many sequences does each file have?</li>
    <li>How is the quality score over the <span class="notranslate">reads</span>? And the mean score?</li>
    <li>Is there any bias in base content?</li>
    <li>How is the GC content?</li>
    <li>Are there any unindentified bases?</li>
    <li>Are there duplicated sequences?</li>
    <li>Are there over-represented sequences?</li>
    <li>Are there still some adapters left?</li>
    <li>What should we do next?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Both files have 260,554 sequences</li>
      <li>
        <p>The “Per base sequence quality” is globally good: the quality stays around 40 over the <span class="notranslate">reads</span>, with just a slight decrease at the end (but still higher than 35)</p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_base_sequence_quality_plot.png" alt="Sequence Quality" /></p>

        <p>The reverse <span class="notranslate">reads</span> have a slight worst quality than the forward, a usual case in Illumina sequencing.</p>

        <p>The distribution of the mean quality score is almost at the maximum for the forward and reverse <span class="notranslate">reads</span>:</p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_sequence_quality_scores_plot.png" alt="Per Sequence Quality Scores" /></p>
      </li>
      <li>
        <p>For both forward and reverse <span class="notranslate">reads</span>, the percentage of A, T, C, G over sequence length is biased. As for any <span class="notranslate">RNA</span>-seq data or more generally libraries produced by priming using random hexamers, the first 10-12 bases have an intrinsic bias.</p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_base_sequence_content_forward.png" alt="Per Base Sequence Content (forward)" /></p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_base_sequence_content_reverse.png" alt="Per Base Sequence Content (reverse)" /></p>

        <p>We could also see that after these first bases the distinction between C-G and A-T groups is not clear as expected. It explains the error raised by FastQC.</p>
      </li>
      <li>
        <p>With sequences from random position of a genome, we expect a normal distribution of the %GC of <span class="notranslate">reads</span> around the mean %GC of the genome. Here, we have <span class="notranslate">RNA</span> <span class="notranslate">reads</span> from various genomes. We do not expect a normal distribution of the %GC. Indeed, for the forward <span class="notranslate">reads</span>, the distribution shows with several peaks: maybe corresponding to mean %GC of different organisms.</p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_sequence_gc_content_plot.png" alt="Per Sequence GC Content" /></p>
      </li>
      <li>
        <p>Almost no N were found in the <span class="notranslate">reads</span>: so almost no unindentified bases</p>

        <p><img src="../../images/metatranscriptomics/fastqc_per_base_n_content_plot.png" alt="Per base N content" /></p>
      </li>
      <li>
        <p>The forward <span class="notranslate">reads</span> seem to have more duplicated <span class="notranslate">reads</span> than the reverse <span class="notranslate">reads</span> with a rate of duplication up to 60% and some <span class="notranslate">reads</span> identified over 10 times.</p>

        <p><img src="../../images/metatranscriptomics/fastqc_sequence_counts_plot.png" alt="Sequence Counts" />
<img src="../../images/metatranscriptomics/fastqc_sequence_duplication_levels_plot.png" alt="Sequence Duplication Levels" /></p>

        <p>In data from <span class="notranslate">RNA</span> (metatranscriptomics data), duplicated <span class="notranslate">reads</span> are expected. The low rate of duplication in reverse <span class="notranslate">reads</span> could be due to bad quality: some nucleotides may have been wrongly identified, altering the <span class="notranslate">reads</span> and reducing the duplication.</p>
      </li>
      <li>
        <p>The high rate of overrepresented sequences in the forward <span class="notranslate">reads</span> is linked to the high rate of duplication.</p>

        <p><img src="../../images/metatranscriptomics/fastqc_overrepresented_sequencesi_plot.png" alt=" Overrepresented sequences" /></p>
      </li>
      <li>
        <p>Illumina universal adapters are still present in the <span class="notranslate">reads</span>, especially at the 3’ end.</p>

        <p><img src="../../images/metatranscriptomics/fastqc_adapter_content_plot.png" alt=" Adapter Content" /></p>
      </li>
      <li>After checking what is wrong, we should think about the errors reported by FastQC: they may come from the type of sequencing or what we sequenced (check the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control” training</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> for more details): some like the duplication rate or the base content biases are due to the <span class="notranslate">RNA</span> sequencing. However, despite these challenges, we can still get slightly better sequences for the <span class="notranslate">downstream</span> analyses.</li>
    </ol>
  </blockquote>
</blockquote>

<p>Even though our data is already of pretty high quality, we can improve it even more by:</p>
<ol>
  <li>Trimming <span class="notranslate">reads</span> to remove bases that were sequenced with low certainty (= low-quality bases) at the ends of the <span class="notranslate">reads</span></li>
  <li>Removing <span class="notranslate">reads</span> of overall bad quality.</li>
  <li>Removing <span class="notranslate">reads</span> that are too short to be informative in <span class="notranslate">downstream</span> analysis</li>
</ol>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What are the possible tools to perform such functions?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>There are many tools such as <strong>Cutadapt</strong>, <strong>Trimmomatic</strong>, <strong>Trim Galore</strong>, <strong>Clip</strong>, <strong>trim putative adapter</strong> sequences. etc. We choose here <strong>Cutadapt</strong> because it is error tolerant, it is fast and the version is pretty stable.</p>
  </blockquote>

</blockquote>

<p>There are several tools out there that can perform these steps, but in this analysis we use <strong>Cutadapt</strong> (<span class="citation"><a href="#martin2011cutadapt">Martin 2011</a></span>).</p>

<p><strong>Cutadapt</strong> also helps find and remove adapter sequences, primers, poly-A tails and/or other unwanted sequences from the input FASTQ files. It trims the input <span class="notranslate">reads</span> by finding the adapter or primer sequences in an error-tolerant way. Additional features include modifying and filtering <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-trimming-and-filtering"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read trimming and filtering</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5</span></span>  with the following parameters to trim low quality sequences:
      <ul>
        <li><em>“Single-end or Paired-end <span class="notranslate">reads</span>?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FASTQ/A file #1”</em>: <code class="language-plaintext highlighter-rouge">T1A_forward</code></li>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FASTQ/A file #2”</em>: <code class="language-plaintext highlighter-rouge">T1A_reverse</code></li>
          </ul>

          <p>The order is important here!</p>
        </li>
        <li>In <em>“Filter Options”</em>
          <ul>
            <li><em>“Minimum length”</em>: <code class="language-plaintext highlighter-rouge">150</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>
          <ul>
            <li><em>“Quality cutoff”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>

          <blockquote class="question">

            <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

            <p>Why do we run the trimming tool only once on a paired-end dataset and not twice, once for each dataset?</p>

            <blockquote class="solution">
              <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

              <p>The tool can remove sequences if they become too short during the trimming process. For paired-end files it removes entire sequence pairs if one (or both) of the two <span class="notranslate">reads</span> became shorter than the set length cutoff. <span class="notranslate">Reads</span> of a read-pair that are longer than a given threshold but for which the partner read has become too short can optionally be written out to single-end files. This ensures that the information of a read pair is not lost entirely if only one read is of good quality.</p>

            </blockquote>
          </blockquote>
        </li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span>
      <ul>
        <li><code class="language-plaintext highlighter-rouge">Read 1 output</code> to <code class="language-plaintext highlighter-rouge">QC controlled forward <span class="notranslate">reads</span></code></li>
        <li><code class="language-plaintext highlighter-rouge">Read 2 output</code> to <code class="language-plaintext highlighter-rouge">QC controlled reverse <span class="notranslate">reads</span></code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs a report file containing some information about the trimming and filtering it performed.</p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Inspect the output report from <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

  <ol>
    <li>How many basepairs has been removed from the forwards <span class="notranslate">reads</span> because of bad quality? And from the reverse <span class="notranslate">reads</span>?</li>
    <li>How many sequence pairs have been removed because at least one read was shorter than the length cutoff?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>203,654 bp has been trimmed for the forward read (read 1) and 569,653 bp bp on the reverse (read 2). It is not a surprise: we saw that at the end of the sequences the quality was dropping more for the reverse <span class="notranslate">reads</span> than for the forward <span class="notranslate">reads</span>.</li>
      <li>27,677 (10.6%) <span class="notranslate">reads</span> were too short after trimming and then filtered.</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="ribosomal-rna-fragments-filtering">Ribosomal <span class="notranslate">RNA</span> fragments filtering</h2>

<p>Metatranscriptomics sequencing targets any <span class="notranslate">RNA</span> in a pool of micro-organisms. The highest proportion of the <span class="notranslate">RNA</span> sequences in any organism will be ribosomal <span class="notranslate">RNA</span>s.</p>

<p>These r<span class="notranslate">RNA</span>s are useful for the taxonomic assignment (i.e. which organisms are found) but they do not provide any functional information, (i.e. which genes are expressed) To make the <span class="notranslate">downstream</span> functional annotation faster, we will sort the r<span class="notranslate">RNA</span> sequences using <strong>SortMe<span class="notranslate">RNA</span></strong> (<span class="citation"><a href="#kopylova2012sortmerna">Kopylova <i>et al.</i> 2012</a></span>). It can handle large <span class="notranslate">RNA</span> databases and sort out all fragments matching to the database with high accuracy and specificity:</p>

<p><img src="../../images/metatranscriptomics/sortmerna.png" alt="SortMeRNA" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ribosomal-rna-fragments-filtering"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Ribosomal <span class="notranslate">RNA</span> fragments filtering</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/sortmerna/bg_sortmerna/2.1b.6" title="Tested with toolshed.g2.bx.psu.edu/repos/rnateam/sortmerna/bg_sortmerna/2.1b.6"><strong>Filter with SortMe<span class="notranslate">RNA</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">rna</span>team/sortme<span class="notranslate">rna</span>/bg_sortme<span class="notranslate">rna</span>/2.1b.6</span></span>  with the following parameters:
      <ul>
        <li><em>“Sequencing type”</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> are paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Forward <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">QC controlled forward <span class="notranslate">reads</span></code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reverse <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">QC controlled reverse <span class="notranslate">reads</span></code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“If one of the paired-end <span class="notranslate">reads</span> aligns and the other one does not”</em>: <code class="language-plaintext highlighter-rouge">Output both <span class="notranslate">reads</span> to rejected file (--paired_out)</code></li>
          </ul>
        </li>
        <li><em>“Databases to query”</em>: <code class="language-plaintext highlighter-rouge">Public pre-indexed ribosomal databases</code>
          <ul>
            <li><em>“r<span class="notranslate">RNA</span> databases”</em>: <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> Select all
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">rfam-5s-database-id98</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-arc-23s-id98</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-euk-28s-id98</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-bac-23s-id98</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-euk-18s-id95</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-bac-16s-id90</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">rfam-5.8s-database-id98</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">silva-arc-16s-id95</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Include aligned <span class="notranslate">reads</span> in FASTA/FASTQ format?”</em>: <code class="language-plaintext highlighter-rouge">Yes (--fastx)</code>
          <ul>
            <li><em>“Include rejected <span class="notranslate">reads</span> file?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Generate statistics file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Expand the aligned and unaligned forward <span class="notranslate">reads</span> datasets in the history</p>

      <blockquote class="question">
        <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How many sequences have been identified as r<span class="notranslate">RNA</span> and non r<span class="notranslate">RNA</span>?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>Aligned forward read file has 1,947 sequences and the unaligned 2,858 sequences. Then 1,947 <span class="notranslate">reads</span> have been identified as r<span class="notranslate">RNA</span> and 2,858 as non r<span class="notranslate">RNA</span>. The numbers are the same for the reverse <span class="notranslate">reads</span>.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p><strong>SortMe<span class="notranslate">RNA</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> removes any <span class="notranslate">reads</span> identified as r<span class="notranslate">RNA</span> from our dataset, and outputs a log file with more information about this filtering.</p>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Inspect the log output from <strong>SortMe<span class="notranslate">RNA</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, and scroll down to the <code class="language-plaintext highlighter-rouge">Results</code> section.</p>

  <ol>
    <li>How many <span class="notranslate">reads</span> have been processed?</li>
    <li>How many <span class="notranslate">reads</span> have been identified as r<span class="notranslate">RNA</span> given the log file?</li>
    <li>Which type of r<span class="notranslate">RNA</span> are identified? Which organisms are we then expected to identify?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>
        <p>465,754 <span class="notranslate">reads</span> are processed: 232,877 for forward and 232,877 for reverse (given the <strong>Cutadapt</strong> report)</p>
      </li>
      <li>
        <p>Out of the 465,754 <span class="notranslate">reads</span>, 119,646 (26%) have passed the e-value threshold and are identified as r<span class="notranslate">RNA</span>.</p>

        <p>The proportion of r<span class="notranslate">RNA</span> sequences is then quite high (around 40%), compared to metagenomics data where usually they represent &lt; 1% of the sequences. Indeed there are only few copies of r<span class="notranslate">RNA</span> genes in genomes, but they are expressed a lot for the cells.</p>

        <p>Some of the aligned <span class="notranslate">reads</span> are forward (resp. reverse) <span class="notranslate">reads</span> but the corresponding reverse (resp. forward) <span class="notranslate">reads</span> are not aligned. As we choose <em>“If one of the paired-end <span class="notranslate">reads</span> aligns and the other one does not”</em>: <code class="language-plaintext highlighter-rouge">Output both <span class="notranslate">reads</span> to rejected file (--paired_out)</code>, if one read in a pair does not align, both go to unaligned.</p>
      </li>
      <li>
        <p>The 20.56% r<span class="notranslate">RNA</span> <span class="notranslate">reads</span> are 23S bacterial r<span class="notranslate">RNA</span>, 2.34% 16S bacterial r<span class="notranslate">RNA</span> and 1.74% 18S eukaryotic r<span class="notranslate">RNA</span>. We then expect to identify mostly bacteria but also probably some archae (18S eukaryotic r<span class="notranslate">RNA</span>).</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="interlace-forward-and-reverse-reads">Interlace forward and reverse <span class="notranslate">reads</span></h2>

<p>The tool for functional annotations needs a single file as input, even with paired-end data.</p>

<p>We need to join the two separate files (forward and reverse) to create a single interleaced file, using <strong>FASTQ interlacer</strong>, in which the forward <span class="notranslate">reads</span> have <code class="language-plaintext highlighter-rouge">/1</code> in their id and reverse <span class="notranslate">reads</span> <code class="language-plaintext highlighter-rouge">/2</code>. The join is performed using sequence identifiers (headers), allowing the two files to contain differing ordering. If a sequence identifier does not appear in both files, it is output in a separate file named <code class="language-plaintext highlighter-rouge">singles</code>.</p>

<p>We use <strong>FASTQ interlacer</strong> on the unaligned (non-r<span class="notranslate">RNA</span>) <span class="notranslate">reads</span> from <strong>SortMe<span class="notranslate">RNA</span></strong> to prepare for the functional analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-interlace-fastq-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Interlace FastQ files</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer/1.2.0.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer/1.2.0.1+galaxy0"><strong>FASTQ interlacer</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastq_paired_end_interlacer/fastq_paired_end_interlacer/1.2.0.1+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>“Type of paired-end datasets”</em>: <code class="language-plaintext highlighter-rouge">2 separate datasets</code></li>
      </ul>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Left-hand mates”</em>: <code class="language-plaintext highlighter-rouge">Unaligned forward <span class="notranslate">reads</span></code> (output of <strong>SortMe<span class="notranslate">RNA</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Right-hand mates”</em>: <code class="language-plaintext highlighter-rouge">Unaligned reverse <span class="notranslate">reads</span></code> (output of <strong>SortMe<span class="notranslate">RNA</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the pair output to <code class="language-plaintext highlighter-rouge">Interlaced non r<span class="notranslate">RNA</span> <span class="notranslate">reads</span></code></li>
  </ol>

</blockquote>

<h1 id="extraction-of-the-community-profile">Extraction of the community profile</h1>

<p><a href="../metatranscriptomics-short/tutorial.html#extraction-of-the-community-profile"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">exchange</span> Switch to short tutorial</a></p>

<p>The first important information to get from microbiome data is the community structure: which organisms are present and in which abundance. This is called <strong>taxonomic profiling</strong>.</p>

<p>Different approaches can be used:</p>

<ul>
  <li>
    <p>Identification and classification of OTUs, as used in amplicon data</p>

    <p>Such an approach first requires sequence sorting to extract only the 16S and 18S sequences (e.g. using the aligned <span class="notranslate">reads</span> from <strong>SortMe<span class="notranslate">RNA</span></strong>), then again using the same tools as for amplicon data (as explained in tutorials like <a href="/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html">16S Microbial Analysis with mothur</a> or <a href="/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html">16S Microbial analysis with Nanopore data</a>).</p>

    <p>However, because r<span class="notranslate">RNA</span> sequences represent less than 50% of the raw sequences, this approach is not the most statistically supported.</p>
  </li>
  <li>
    <p>Assignment of taxonomy on the whole sequences using databases with marker genes</p>
  </li>
</ul>

<p>In this tutorial, we follow second approach using <strong>MetaPhlAn</strong> (<span class="citation"><a href="#truong2015metaphlan2">Truong <i>et al.</i> 2015</a></span>). This tool uses a database of ~1M unique clade-specific marker genes (not only the r<span class="notranslate">RNA</span> genes) identified from ~17,000 reference (bacterial, archeal, viral and eukaryotic) genomes.</p>

<p>As r<span class="notranslate">RNA</span>s <span class="notranslate">reads</span> are good marker genes, we will use directly the quality controlled files (output of <strong>Cutadapt</strong>) with all <span class="notranslate">reads</span> (not only the non r<span class="notranslate">RNA</span>s).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-the-community-structure"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract the community structure</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/metaphlan/metaphlan/3.0.8+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/metaphlan/metaphlan/3.0.8+galaxy0"><strong>MetaPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/metaphlan/metaphlan/3.0.8+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li>In <em>“Input(s)”</em>
          <ul>
            <li><em>“Input(s)”</em>: <code class="language-plaintext highlighter-rouge">Fasta/FastQ file(s) with metagenomic <span class="notranslate">reads</span></code>
              <ul>
                <li><em>“Fasta/FastQ file(s) with metagenomic <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Paired-end files</code>
                  <ul>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Forward paired-end Fasta/FastQ file with metagenomic <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">QC controlled forward <span class="notranslate">reads</span></code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reverse paired-end Fasta/FastQ file with metagenomic <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">QC controlled reverse <span class="notranslate">reads</span></code> (outputs of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Database with clade-specific marker genes”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
              <ul>
                <li><em>“Cached database with clade-specific marker genes”</em>: <code class="language-plaintext highlighter-rouge">MetaPhlAn clade-specific marker genes (mpa_v30_CHOCOPhlAn_201901)</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Analysis”</em>
          <ul>
            <li><em>“Type of analysis to perform”</em>: <code class="language-plaintext highlighter-rouge">rel_ab: Profiling a metagenomes in terms of relative abundances</code>
              <ul>
                <li><em>“Taxonomic level for the relative abundance output”</em>: <code class="language-plaintext highlighter-rouge">All taxonomic levels</code></li>
                <li>*“Generate a report for each taxonomic level?”: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
            <li><em>“Quantile value for the robust average”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
            <li><em>“Organisms to profile”</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Profile viral organisms (add_viruses)</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Output”</em>
          <ul>
            <li><em>“Output for Krona”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>This step may take a couple of minutes as each sequence is compare to the full database with ~1 million reference sequences.</p>

<p>5 files and a collection are generated by <strong>MetaPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:</p>

<ul>
  <li>
    <p>The main output: A <strong>tabular file</strong> called <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances</code> with the <em>*community profile</em></p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  #mpa_v30_CHOCOPhlAn_201901
  # ....
  #SampleID	Metaphlan_Analysis
  #clade_name	NCBI_tax_id	relative_abundance	additional_species
  k__Bacteria	2	99.40284
  k__Archaea	2157	0.59716
  k__Bacteria|p__Firmicutes	2|1239	94.67418
  k__Bacteria|p__Coprothermobacterota	2|2138240	4.72866
  k__Archaea|p__Euryarchaeota	2157|28890	0.59716
  k__Bacteria|p__Firmicutes|c__Clostridia	2|1239|186801	94.67418
  k__Bacteria|p__Coprothermobacterota|c__Coprothermobacteria	2|2138240|2138243	4.72866
  k__Archaea|p__Euryarchaeota|c__Methanobacteria
</code></pre></div>    </div>

    <p>Each line contains 4 columns:</p>
    <ol>
      <li>the lineage with different taxonomic levels</li>
      <li>the previous lineage with NCBI taxon id</li>
      <li>the relative abundance found for our sample for the lineage</li>
      <li>any additional species</li>
    </ol>

    <p>The file starts with high level taxa (kingdom: <code class="language-plaintext highlighter-rouge">k__</code>) and go to more precise taxa.</p>

    <blockquote class="question">
      <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <p>Inspect the <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances</code> file output by <strong>MetaPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <ol>
        <li>How many taxons have been identified?</li>
        <li>What are the different taxonomic levels we have access to with MetaPhlAn?</li>
        <li>What genus and species are found in our sample?</li>
        <li>Has only bacteria been identified in our sample?</li>
      </ol>

      <blockquote class="solution">
        <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
        <ol>
          <li>The file has 20 lines, including an header. Therefore, 17 taxons of different levels have been identified</li>
          <li>We have access: kingdom (<code class="language-plaintext highlighter-rouge">k__</code>), phylum (<code class="language-plaintext highlighter-rouge">p__</code>), class (<code class="language-plaintext highlighter-rouge">c__</code>), order (<code class="language-plaintext highlighter-rouge">o__</code>), family (<code class="language-plaintext highlighter-rouge">f__</code>), genus (<code class="language-plaintext highlighter-rouge">g__</code>), species (<code class="language-plaintext highlighter-rouge">s__</code>), strain (<code class="language-plaintext highlighter-rouge">t__</code>)</li>
          <li>In our sample, we identified:
            <ul>
              <li>3 genera: Coprothermobacter, Methanothermobacter, Hungateiclostridium</li>
              <li>3 species: Coprothermobacter proteolyticus, Methanothermobacter thermautotrophicus, Hungateiclostridium thermocellum</li>
            </ul>
          </li>
          <li>As expected from the r<span class="notranslate">RNA</span> sorting, we have some archaea, Methanobacteria, in our sample.</li>
        </ol>
      </blockquote>
    </blockquote>
  </li>
  <li>A <strong>collection</strong> with the same information as in the tabular file but splitted into different files, one per taxonomic level</li>
  <li>A <strong>tabular file</strong> called <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances for Krona</code> with the same information as the previous file but formatted for visualization using Krona. We will use this file later</li>
  <li>
    <p>A <strong>BIOM file</strong> with the same information as the previous file but in BIOM format</p>

    <p>BIOM format is quite common in microbiomics. This is standard, for example, as the input for tools like mothur or QIIME.</p>
  </li>
  <li>A <strong>SAM file</strong> with the results of the sequence <span class="notranslate">mapping</span> on the reference database.</li>
  <li>A <strong>tabular file</strong> called <code class="language-plaintext highlighter-rouge">Bowtie2 output</code> with similar information as the one in the <strong>SAM file</strong></li>
</ul>

<blockquote class="comment">
  <h3 id="comment-note-analyzing-an-isolated-metatranscriptome"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: Analyzing an isolated metatranscriptome</h3>

  <p>We are analyzing our <span class="notranslate">RNA</span> <span class="notranslate">reads</span> as we would do for <span class="notranslate">DNA</span> <span class="notranslate">reads</span>. This approach has one main caveat. In <strong>MetaPhlAn</strong>, the species are quantified based on the recruitment of <span class="notranslate">reads</span> to species-specific marker genes. In metagenomic data, each genome copy is assumed to donate ~1 copy of each marker. But the same assumption cannot be made for <span class="notranslate">RNA</span> data: markers may be transcribed more or less within a given species in this sample compared to the average transcription rate. A species will still be detected in the metatranscriptomic data as long as a non-trivial fraction of the species’ markers is expressed.</p>

  <p>We should then carefully interpret the species relative abundance. These values reflect species’ relative contributions to the pool of species-specific transcripts and not the overall transcript pool.</p>
</blockquote>

<p>Some <span class="notranslate">downstream</span> tools need the MetaPhlAn table with predicted taxon abundance with only 2 columns: the lineage and the abundance.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-format-taxon-relative-abundances-table"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Format taxon relative abundances table</h3>
  <ol>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c3</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances</code> (output of <strong>MetaPhlAn</strong>)</li>
      </ul>
    </li>
    <li>Rename <code class="language-plaintext highlighter-rouge">Cut predicted taxon relative abundances table</code></li>
  </ol>

</blockquote>

<h2 id="community-structure-visualization">Community structure visualization</h2>

<p>Even if the output of <strong>MetaPhlAn</strong> can be easy to parse, we want to visualize and explore the community structure. 2 tools can be used there:</p>

<ul>
  <li><strong>Krona</strong> for an interactive HTML output</li>
  <li><strong>Graphlan</strong> for a publication ready visualization</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-interactive-community-structure-visualization-with-krona"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Interactive community structure visualization with KRONA</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+galaxy0"><strong>Krona pie chart</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1+<span class="notranslate">galaxy</span>0</span></span>  with
      <ul>
        <li><em>“What is the type of your input data”</em>: <code class="language-plaintext highlighter-rouge">Tabular</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances for Krona</code> (output of <strong>MetaPhlAn</strong>)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><strong>Krona</strong> <span class="citation"><a href="#ondov2011interactive">Ondov <i>et al.</i> 2011</a></span> renders results of a metagenomic profiling as a zoomable pie chart. It allows hierarchical data, here taxonomic levels, to be explored with zooming, multi-layered pie charts</p>

<blockquote class="question">
  <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Inspect the output from <strong>Krona</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. (The interactive plot is also shown below)</p>

  <ol>
    <li>What are the abundances of 2 kingdoms identified here?</li>
    <li>When zooming on Bacteria, what are the 2 subclasses identified?</li>
  </ol>

  <p><br /><br /></p>
  <center><iframe id="krona" src="krona.html" frameborder="0" width="90%" height="600px"> ![Krona at bacteria level](../../images/metatranscriptomics/krona_bacteria.png) </iframe></center>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>Archaea represents 0.6% so bacteria are 99.4% of the organisms identified in our sample</p>

        <p><img src="../../images/metatranscriptomics/krona.png" alt="Krona" width="75%" /></p>
      </li>
      <li>
        <p>5% of bacteria are Coprothermobacter proteolyticus and the rest Hungateiclostridium thermocellum .</p>

        <p><img src="../../images/metatranscriptomics/krona_bacteria.png" alt="Krona at bacteria level" width="75%" /></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p><strong>GraPhlAn</strong> is another software tool for producing high-quality circular representations of taxonomic and phylogenetic trees.</p>

<p>It takes a taxonomic tree file as the input. We first need to convert the <strong>MetaPhlAn</strong> output using <strong>Export to GraPhlAn</strong>. This conversion software tool produces both annotation and tree file for GraPhlAn.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-publication-ready-community-structure-visualization-with-graphlan"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Publication-ready community structure visualization with GraPhlAn</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/export2graphlan/export2graphlan/0.20+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/export2graphlan/export2graphlan/0.20+galaxy0"><strong>Export to GraPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/export2graphlan/export2graphlan/0.20+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">Cut predicted taxon relative abundances table</code></li>
        <li><em>“List which levels should be annotated in the tree”</em>: <code class="language-plaintext highlighter-rouge">1,2</code></li>
        <li><em>“List which levels should use the exte<span class="notranslate">rna</span>l legend for the annotation”</em>: <code class="language-plaintext highlighter-rouge">3,4,5</code></li>
        <li><em>“List which levels should be highlight with a shaded background”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/graphlan_annotate/graphlan_annotate/1.0.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/graphlan_annotate/graphlan_annotate/1.0.0.0"><strong>Generation, personalization and annotation of tree</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/graphlan_annotate/graphlan_annotate/1.0.0.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tree”</em>: <code class="language-plaintext highlighter-rouge">Tree</code> (output of <strong>Export to GraPhlAn</strong>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotation file”</em>: <code class="language-plaintext highlighter-rouge">Annotation</code> (output of <strong>Export to GraPhlAn</strong>)</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/graphlan/graphlan/1.0.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/graphlan/graphlan/1.0.0.0"><strong>GraPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/graphlan/graphlan/1.0.0.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tree”</em>: <code class="language-plaintext highlighter-rouge">Tree in PhyloXML</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">PNG</code></li>
      </ul>
    </li>
    <li>Inspect <strong>GraPhlAn</strong> output</li>
  </ol>

</blockquote>

<p><img src="../../images/metatranscriptomics/graphlan.png" alt="GraPhlAn" /></p>

<h1 id="extract-the-functional-information">Extract the functional information</h1>
<p><a href="../metatranscriptomics-short/tutorial.html#extract-the-functional-information"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">exchange</span> Switch to short tutorial</a></p>

<p>We would now like to answer the question “What are the micro-organisms doing?” or “Which functions are performed by the micro-organisms in the environment?”.</p>

<p>In the metatranscriptomics data, we have access to the genes that are expressed by the community. We can use that to identify genes, their functions, and build pathways, etc., to investigate their contribution to the community using <strong>HUMAnN</strong> (<span class="citation"><a href="#franzosa2018species">Franzosa <i>et al.</i> 2018</a></span>). <strong>HUMAnN</strong> is a pipeline developed for efficiently and accurately profiling the presence/absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data.</p>

<p>To identify the functions made by the community, we do not need the r<span class="notranslate">RNA</span> sequences, specially because they had noise and will slow the run. We will then use the output of <strong>SortMe<span class="notranslate">RNA</span></strong>, but also the identified community profile from <strong>MetaPhlAn</strong>. This will help <strong>HUMAnN</strong> to focus on the know sequences for the identified organisms.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-the-functional-information"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract the functional information</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann/humann/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann/humann/3.0.0+galaxy1"><strong>HUMAnN</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann/humann/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>“Input(s)”</em>: <code class="language-plaintext highlighter-rouge">Quality-controlled shotgun sequencing <span class="notranslate">reads</span> (metagenome (<span class="notranslate">DNA</span> <span class="notranslate">reads</span>) or metatranscriptome (<span class="notranslate">RNA</span> <span class="notranslate">reads</span>))</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Quality-controlled shotgun sequencing <span class="notranslate">reads</span> (metagenome (<span class="notranslate">DNA</span> <span class="notranslate">reads</span>) or metatranscriptome (<span class="notranslate">RNA</span> <span class="notranslate">reads</span>))”</em>: <code class="language-plaintext highlighter-rouge">Interlaced non r<span class="notranslate">RNA</span> <span class="notranslate">reads</span></code></li>
          </ul>
        </li>
        <li><em>“Steps”</em>: <code class="language-plaintext highlighter-rouge">Bypass the taxonomic profiling step and creates a custom ChocoPhlAn database of the species provided afterwards</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span><em>“Taxonomic profile file”</em>: <code class="language-plaintext highlighter-rouge">Predicted taxon relative abundances</code> (output of <strong>MetaPhlAn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li>In <em>“Nucleotide search / <span class="notranslate">Mapping</span> <span class="notranslate">reads</span> to community pangenomes”</em>
              <ul>
                <li><em>“Nucleotide database”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
                  <ul>
                    <li><em>“Nucleotide database”</em>: <code class="language-plaintext highlighter-rouge">Full ChocoPhlAn for HUMAnN</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>In <em>“Translated search / Aligning unmapped <span class="notranslate">reads</span> to a protein database”</em>
              <ul>
                <li><em>“Protein database”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
                  <ul>
                    <li><em>“Protein database”</em>: <code class="language-plaintext highlighter-rouge">Full UniRef90 for HUMAnN</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-running-low-on-time-import-the-humann-outputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Running low on time? Import the HUMAnN outputs</h3>

  <ol>
    <li>
      <p>Import the 2 files:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4776250/files/T1A_HUMAnN_Gene_families_and_their_abundance.tabular
https://zenodo.org/record/4776250/files/T1A_HUMAnN_Pathways_and_their_abundance.tabular
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><strong>HUMAnN</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> generates 4 files:</p>

<ol>
  <li>
    <p>A tabular file with the <strong>gene families and their abundance</strong>:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> # Gene Family	humann_Abundance-RPKs
 UNMAPPED	98294.0000000000
 UniRef90_A3DCI4	42213.2661898446
 UniRef90_A3DCI4|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	42205.5707425926
 UniRef90_A3DCI4|unclassified	7.6954472520
 UniRef90_A3DCB9	39273.2031556915
 UniRef90_A3DCB9|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	39273.2031556915
</code></pre></div>    </div>

    <p>This file details the abundance of each gene family in the community. Gene families are groups of evolutionarily-related protein-coding sequences that often perform similar functions. Here we used <a href="https://www.uniprot.org/help/uniref">UniRef90 gene families</a>: sequences in a gene families have at least 50% sequence identity.</p>

    <p>Gene family abundance at the community level is stratified to show the contributions from known and unknown species. Individual species’ abundance contributions sum to the community total abundance.</p>

    <p>Gene family abundance is reported in RPK (<span class="notranslate">reads</span> per kilobase) units to normalize for gene length. It reflects the relative gene (or transcript) copy number in the community.</p>

    <p>The “UNMAPPED” value is the total number of <span class="notranslate">reads</span> which remain unmapped after both alignment steps (nucleotide and translated search). Since other gene features in the table are quantified in RPK units, “UNMAPPED” can be interpreted as a single unknown gene of length 1 kilobase recruiting all <span class="notranslate">reads</span> that failed to map to known sequences.</p>

    <blockquote class="question">
      <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <p><strong>Inspect</strong> the <code class="language-plaintext highlighter-rouge">Gene families and their abundances</code> file from <strong>HUMAnN</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <ol>
        <li>What is the most abundant family?</li>
        <li>Which species is involved in production of this family?</li>
        <li>How many gene families have been identified?</li>
      </ol>

      <blockquote class="solution">
        <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
        <ol>
          <li>The most abundant family is the first one in the family: UniRef90_A3DCI4. We can use the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+galaxy1"><strong>Rename features of a HUMAnN generated table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  to add extra information about the gene family.</li>
          <li>Beta-lactamase TEM seems mostly produced here by <em>Hungateiclostridium thermocellum</em>.</li>
          <li>
            <p>There is 6,374 lines in gene family file. But some of the gene families have multiple lines when the involved species are known.</p>

            <p>To know the number of gene families, we need to remove all lines with the species information, i.e. lines with <code class="language-plaintext highlighter-rouge">|</code> in them using the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1"><strong>Split a HUMAnN table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span></p>

            <p>The tool generates 2 output file:</p>
            <ul>
              <li>a stratified table with all lines with <code class="language-plaintext highlighter-rouge">|</code> in them</li>
              <li>a unstratied table with all lines without <code class="language-plaintext highlighter-rouge">|</code> in them</li>
            </ul>

            <p>In the unstratified table, there are 3,175 lines, so 3,174 gene families.</p>
          </li>
        </ol>
      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>A tabular file with the <strong>pathways and their abundance</strong>:</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  # Pathway	humann_Abundance
 UNMAPPED	21383.0328532606
 UNINTEGRATED	123278.2617863865
 UNINTEGRATED|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	114029.8324087679
 UNINTEGRATED|unclassified	6301.1699240597
 UNINTEGRATED|g__Coprothermobacter.s__Coprothermobacter_proteolyticus	5407.2826151090
 PWY-6609: adenine and adenosine salvage III	285.7955913866
 PWY-6609: adenine and adenosine salvage III|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	66.3688569288
 PWY-1042: glycolysis IV	194.7465938041
</code></pre></div>    </div>

    <p>This file shows each pathway and their abundance. Here, we used  the <a href="https://metacyc.org/">MetaCyc Metabolic Pathway Database</a>, a curated database of experimentally elucidated metabolic pathways from all domains of life.</p>

    <p>The abundance of a pathway in the sample is computed as a function of the abundances of the pathway’s component reactions, with each reaction’s abundance computed as the sum over abundances of genes catalyzing the reaction. The abundance is proportional to the number of complete “copies” of the pathway in the community. Indeed, for a simple linear pathway <code class="language-plaintext highlighter-rouge">RXN1 --&gt; RXN2 --&gt; RXN3 --&gt; RXN4</code>, if RXN1 is 10 times as abundant as RXNs 2-4, the pathway abundance will be driven by the abundances of RXNs 2-4.</p>

    <p>The pathway abundance is computed once for all species (community level) and again for each species  using species gene abundances along the components of the pathway. Unlike gene abundance, a pathway’s  abundance at community-level is not necessarily the sum of the abundance values of each species. For example, for the same pathway example as above, if the abundances of RXNs 1-4 are [5, 5, 10, 10] in Species A and [10, 10, 5, 5] in Species B, the pathway abundance would be 5 for Species A and Species B, but 15 at the community level as the reaction totals are [15, 15, 15, 15].</p>

    <blockquote class="question">
      <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <p><strong>View</strong> the <code class="language-plaintext highlighter-rouge">Pathways and their abundance</code> output from <strong>HUMAnN</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <ol>
        <li>What is the most abundant pathway?</li>
        <li>Which species is involved in production of this pathway?</li>
        <li>How many gene families have been identified?</li>
        <li>What is the “UNINTEGRATED” abundance?</li>
      </ol>

      <blockquote class="solution">
        <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
        <ol>
          <li>The most abundant pathway is PWY-6609. It produces the adenine and adenosine salvage III.</li>
          <li>Like the gene family, this pathway is mostly achieved by <em>Hungateiclostridium thermocellum</em>.</li>
          <li>
            <p>There are 118 lines in the pathway file, including the lines with species information. To compute the number of gene families, we need to apply a similar approach as for the gene families by removing the lines with <code class="language-plaintext highlighter-rouge">|</code> in them using the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1"><strong>Split a HUMAnN table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span> .
  The unstratified output file has 64 lines, including the header, UNMAPPED and UNINTEGRATED. Therefore, 61 MetaCyc pathways have been identified for our sample.</p>
          </li>
          <li>The “UNINTEGRATED” abundance corresponds to the total abundance of genes in the different levels that do not contribute to any pathways.</li>
        </ol>
      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>A file with the <strong>pathways and their coverage</strong>:</p>

    <p>Pathway coverage provides an alte<span class="notranslate">rna</span>tive description of the presence (1) and absence (0) of pathways in a community, independent of their quantitative abundance.</p>
  </li>
  <li>
    <p>A log file</p>
  </li>
</ol>

<blockquote class="comment">
  <h3 id="comment-note-analyzing-an-isolated-metatranscriptome-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: Analyzing an isolated metatranscriptome</h3>

  <p>As we already mentioned above, we are analyzing our <span class="notranslate">RNA</span> <span class="notranslate">reads</span> as we would do for <span class="notranslate">DNA</span> <span class="notranslate">reads</span> and therefore we should be careful when interpreting the results. We already mentioned the analysis of the species’ relative abundance from <strong>MetaPhlAn</strong>, but there is another aspect we should be careful about.</p>

  <p>From a lone metatranscriptomic dataset, the transcript abundance can be confounded with the underlying gene copy number. For example, transcript X may be more abundant in sample A relative to sample B because the underlying X gene (same number in both samples) is more highly expressed in sample A relative to sample B; or there are more copies of gene X in sample A relative to sample B (all of which are equally expressed). This is a general challenge in analyzing isolated metatranscriptomes.</p>

  <p>The best approach would be to combine the metatranscriptomic analysis with a metagenomic analysis. In this case, rather than running <strong>MetaPhlAn</strong> on the metatranscriptomic data, we run it on the metagenomic data and use the taxonomic profile as input to <strong>HUMAnN</strong>. <span class="notranslate">RNA</span> <span class="notranslate">reads</span> are then mapped to any species’ pangenomes detected in the metagenome. Then we run <strong>HUMAnN</strong> on both metagenomics and metatranscriptomic data. We can use both outputs to normalize the <span class="notranslate">RNA</span>-level outputs (e.g. transcript family abundance) by corresponding <span class="notranslate">DNA</span>-level outputs to the quantification of microbial <span class="notranslate">expression</span> independent of gene copy number.</p>

  <p>Here we do not have a metagenomic dataset to combine with and need to be careful in our interpretation</p>

</blockquote>

<h2 id="normalize-the-abundances">Normalize the abundances</h2>

<p>Gene family and pathway abundances are in RPKs (<span class="notranslate">reads</span> per kilobase), accounting for gene length but not sample sequencing depth. While there are some applications, e.g. strain profiling, where RPK units are superior to depth-normalized units, most of the time we need to renormalize our samples prior to <span class="notranslate">downstream</span> analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-normalize-the-gene-family-abundances"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Normalize the gene family abundances</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+galaxy1"><strong>Renormalize a HUMAnN generated table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  with
      <ul>
        <li><em>“Gene/pathway table”</em>: <code class="language-plaintext highlighter-rouge">Gene families and their abundance</code> (output of <strong>HUMAnN</strong>)</li>
        <li><em>“Normalization scheme”</em>: <code class="language-plaintext highlighter-rouge">Relative abundance</code></li>
        <li><em>“Normalization level”</em>: <code class="language-plaintext highlighter-rouge">Normalization of all levels by community total</code></li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the generated file <code class="language-plaintext highlighter-rouge">Normalized gene families</code></li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p><strong>Inspect</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the <code class="language-plaintext highlighter-rouge">Normalized gene families</code> file</p>

  <ol>
    <li>What percentage of sequences has not been assigned to a gene family?</li>
    <li>What is the relative abundance of the most abundant gene family?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>14% (<code class="language-plaintext highlighter-rouge">0.14438 x 100</code>) of the sequences have not be assigned to a gene family</li>
      <li>The UniRef90_A3DCI4 family represents 6% of the <span class="notranslate">reads</span>.</li>
    </ol>
  </blockquote>
</blockquote>

<p>Let’s do the same for the pathway abundances.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-normalize-the-pathway-abundances"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Normalize the pathway abundances</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+galaxy1"><strong>Renormalize a HUMAnN generated table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_renorm_table/humann_renorm_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  with
      <ul>
        <li><em>“Gene/pathway table”</em>: <code class="language-plaintext highlighter-rouge">Pathways and their abundance</code> (output of <strong>HUMAnN</strong>)</li>
        <li><em>“Normalization scheme”</em>: <code class="language-plaintext highlighter-rouge">Relative abundance</code></li>
        <li><em>“Normalization level”</em>: <code class="language-plaintext highlighter-rouge">Normalization of all levels by community total</code></li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the generated file <code class="language-plaintext highlighter-rouge">Normalized pathways</code></li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-12"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p><strong>Inspect</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the <code class="language-plaintext highlighter-rouge">Normalized pathways</code> file.</p>
  <ol>
    <li>What is the UNMAPPED percentage?</li>
    <li>What percentage of <span class="notranslate">reads</span> assigned to a gene family has not be assigned to a pathway?</li>
    <li>What is the relative abundance of the most abundant gene family?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-12"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>UNMAPPED, here 14% of the <span class="notranslate">reads</span>, corresponds to the percentage of <span class="notranslate">reads</span> not assigned to gene families. It is the same value as in the normalized gene family file.</li>
      <li>83% (UNINTEGRATED) of <span class="notranslate">reads</span> assigned to a gene family have not be assigned to a pathway</li>
      <li>The PWY-6609 pathway represents 0.2% of the <span class="notranslate">reads</span>.</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="identify-the-gene-families-involved-in-the-pathways">Identify the gene families involved in the pathways</h2>

<p>We would like to know which gene families are involved in our most abundant pathways and which species.
For this, we use the tool <strong>Unpack pathway abundances to show genes included</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from <strong>HUMAnN</strong> tool suites.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-normalize-the-gene-family-abundances-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Normalize the gene family abundances</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_unpack_pathways/humann_unpack_pathways/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_unpack_pathways/humann_unpack_pathways/3.0.0+galaxy1"><strong>Unpack pathway abundances to show genes included</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_unpack_pathways/humann_unpack_pathways/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  with
      <ul>
        <li><em>“Gene family or EC abundance file”</em>: <code class="language-plaintext highlighter-rouge">Normalized gene families</code></li>
        <li><em>“Pathway abundance file”</em>: <code class="language-plaintext highlighter-rouge">Normalized pathways</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This tool unpacks the pathways to show the genes for each. It adds another level of stratification to the pathway abundance table by including the gene family abundances:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># Pathway	humann_Abundance-RELAB
ANAGLYCOLYSIS-PWY: glycolysis III (from glucose)	0.000312308
BRANCHED-CHAIN-AA-SYN-PWY: superpathway of branched chain amino acid biosynthesis	0.00045009
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	0.00045009
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DIY4	9.67756e-06
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DIE1	0.000123259
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DID9	9.73261e-05
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DDR1	7.97357e-05
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DF94	1.4837e-05
BRANCHED-CHAIN-AA-SYN-PWY|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DIY3	0.000117313
</code></pre></div></div>

<blockquote class="question">
  <h3 id="question-questions-13"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p><strong>Inspect</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the output from <strong>Unpack pathway abundances to show genes included</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

  <p>Which gene families are involved in the PWY-6609 pathway (the most abundant one)? And which species?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-13"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>If we search the generated file for (using <kbd>CTR</kbd><kbd>F</kbd> or <kbd>CMD</kbd><kbd>F</kbd>):</p>

    <pre><code class="language-`">PWY-6609: adenine and adenosine salvage III	0.00192972
PWY-6609|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	0.000448128
PWY-6609|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DD28	4.47176e-05
PWY-6609|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DHM7	0.000235689
PWY-6609|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum|UniRef90_A3DEQ4	3.68298e-05
</code></pre>

    <p>The gene families UniRef90_A3DD28, UniRef90_A3DHM7 and UniRef90_A3DEQ4 are identified, for <em>Hungateiclostridium thermocellum</em>.</p>
  </blockquote>
</blockquote>

<h2 id="group-gene-families-into-go-terms">Group gene families into GO terms</h2>

<p>The gene families can be a long list of ids and going through the gene families one by one to identify the interesting ones can be cumbersome. To help constuct a big picture, we could identify and use categories of genes using the gene families.</p>

<p><a href="http://www.geneontology.org/">Gene Ontology (GO)</a> analysis is widely used to reduce complexity and highlight biological processes in genome-wide <span class="notranslate">expression</span> studies. There is a dedicated tool which groups and converts UniRef50 gene family abundances generated with <strong>HUMAnN</strong> into GO terms.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-group-abundances-into-go-terms"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Group abundances into GO terms</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_regroup_table/humann_regroup_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_regroup_table/humann_regroup_table/3.0.0+galaxy1"><strong>Regroup HUMAnN table features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_regroup_table/humann_regroup_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene families table”</em>: <code class="language-plaintext highlighter-rouge">Gene families and their abundance</code> (output of <strong>HUMAnN</strong>)</li>
        <li><em>“How to combine grouped features?”</em>: <code class="language-plaintext highlighter-rouge">Sum</code></li>
        <li><em>“Grouping”</em>: <code class="language-plaintext highlighter-rouge">Grouping with larger <span class="notranslate">mapping</span></code>
          <ul>
            <li><em>“<span class="notranslate">Mapping</span> to use for the grouping”</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Mapping</span> (full) for Gene Ontology (GO) from UniRef90</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output is a table with the GO terms, their abundance and the involved species:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># Gene Family	humann_Abundance-RPKs
UNMAPPED	98294.0
UNGROUPED	174431.144
UNGROUPED|g__Coprothermobacter.s__Coprothermobacter_proteolyticus	6941.859
UNGROUPED|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	151964.371
UNGROUPED|unclassified	15524.915
GO:0000015	250.882
GO:0000015|g__Coprothermobacter.s__Coprothermobacter_proteolyticus	11.313
GO:0000015|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	239.568
</code></pre></div></div>

<blockquote class="question">
  <h3 id="question-questions-14"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How many GO term have been identified?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-14"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>Using the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1"><strong>Split a HUMAnN table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span> , we see that the unstratified table has 1,171 lines (including the UNMAPPED, UNGROUPED and the header). So 1,168 GO terms have been identified.</p>
  </blockquote>
</blockquote>

<p>The GO term with their id are quite cryptic. We can rename them and then split them in 3 groups  (molecular functions [MF], biological processes [BP] and cellular components [CC])</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-rename-go-terms"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Rename GO terms</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+galaxy0"><strong>Rename features of a HUMAnN generated table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_rename_table/humann_rename_table/3.0.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene families table”</em>: output of <strong>Regroup HUMAnN table features</strong></li>
        <li><em>“Type of feature renaming”</em>: <code class="language-plaintext highlighter-rouge">Advanced feature renaming</code>
          <ul>
            <li><em>“Features to be renamed”</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Mapping</span> (full) between Gene Ontology (GO) ids and names</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: output of last <strong>Rename features of a HUMAnN generated table</strong></li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">\[CC\]</code></li>
      </ul>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the generated file <code class="language-plaintext highlighter-rouge">[CC] GO terms and their abundance</code></p>
    </li>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: output of last <strong>Rename features of a HUMAnN generated table</strong></li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">\[MF\]</code></li>
      </ul>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the generated file <code class="language-plaintext highlighter-rouge">[MF] GO terms and their abundance</code></p>
    </li>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: output of last <strong>Rename features of a HUMAnN generated table</strong></li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">\[BP\]</code></li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the generated file <code class="language-plaintext highlighter-rouge">[BP] GO terms and their abundance</code></li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-15"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p><strong>Inspect</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the 3 outputs of these steps</p>

  <ol>
    <li>How many GO terms have been identified</li>
    <li>Which of the GO terms related to molecular functions is the most abundant?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-15"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>After running <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+galaxy1"><strong>Split a HUMAnN table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/humann_split_stratified_table/humann_split_stratified_table/3.0.0+<span class="notranslate">galaxy</span>1</span></span>  on the 3 outputs, we found:
        <ul>
          <li>414 BP GO terms</li>
          <li>689 MF GO terms</li>
          <li>59 CC GO terms</li>
        </ul>
      </li>
      <li>
        <p>The GO terms in the <code class="language-plaintext highlighter-rouge">[MF] GO terms and their abundance</code> file are not sorted by abundance:</p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GO:0000030: [MF] mannosyltransferase activity	16.321
GO:0000030: [MF] mannosyltransferase activity|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	16.321
GO:0000036: [MF] acyl carrier activity	726.249
GO:0000036: [MF] acyl carrier activity|g__Coprothermobacter.s__Coprothermobacter_proteolyticus	10.101
GO:0000036: [MF] acyl carrier activity|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	572.281
GO:0000036: [MF] acyl carrier activity|unclassified	143.868
GO:0000049: [MF] t<span class="notranslate">RNA</span> binding	4818.299
</code></pre></div>        </div>

        <p>So to identify the most abundant GO terms, we first need to sort the file using the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1"><strong>Sort data in ascending or descending order</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1</span></span>  tool (on column 2, in descending order):</p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GO:0015035: [MF] protein disulfide oxidoreductase activity	42908.123
GO:0003735: [MF] structural constituent of ribosome	42815.337
GO:0015035: [MF] protein disulfide oxidoreductase activity|g__Hungateiclostridium.s__Hungateiclostridium_thermocellum	40533.997
GO:0005524: [MF] ATP binding	37028.271
GO:0046872: [MF] metal ion binding	31068.144
</code></pre></div>        </div>

        <p>The most abundant GO terms related to molecular functions seem to be linked to protein disulfide oxidoreductase activity, but also to structural constituent of ribosome and ATP and metal ion binding.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="combine-taxonomic-and-functional-information">Combine taxonomic and functional information</h1>

<p>With <strong>MetaPhlAn</strong> and <strong>HUMAnN</strong>, we investigated “Which micro-organims are present in my sample?” and “What functions are performed by the micro-organisms in my sample?”. We can go further in these analyses, for example using a combination of functional and taxonomic results. Although we did not detail that in this tutorial you can find more methods of analysis in our tutorials on shotgun metagenomic data analysis.</p>

<p>Although gene families and pathways, and their abundance may be related to a species, in the <strong>HUMAnN</strong> output, relative abundance of the species is not indicated. Therefore, for each gene family/pathway and the corresponding taxonomic stratification, we will now extract the relative abundance of this gene family/pathway and the relative abundance of the corresponding species and genus.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-combine-taxonomic-and-functional-information"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Combine taxonomic and functional information</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bebatut/combine_metaphlan2_humann2/combine_metaphlan2_humann2/0.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bebatut/combine_metaphlan2_humann2/combine_metaphlan2_humann2/0.1.0"><strong>Combine MetaPhlAn2 and HUMAnN2 outputs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bebatut/combine_metaphlan2_humann2/combine_metaphlan2_humann2/0.1.0</span></span>   with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file corresponding to MetaPhlAN output”</em>: <code class="language-plaintext highlighter-rouge">Cut predicted taxon relative abundances table</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file corresponding HUMAnN output”</em>: <code class="language-plaintext highlighter-rouge">Normalized gene families</code></li>
        <li><em>“Type of characteristics in HUMAnN file”</em>: <code class="language-plaintext highlighter-rouge">Gene families</code></li>
      </ul>
    </li>
    <li>Inspect the generated file</li>
  </ol>
</blockquote>

<p>The generated file is a table with 7 columns:</p>
<ol>
  <li>genus</li>
  <li>abundance of the genus (percentage)</li>
  <li>species</li>
  <li>abundance of the species (percentage)</li>
  <li>gene family id</li>
  <li>gene family name</li>
  <li>gene family abundance (percentage)</li>
</ol>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>genus	genus_abundance	species	species_abundance	gene_families_id	gene_families_name	gene_families_abundance
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_A3DCI4		6.199410892039433
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_A3DCB9		5.768680830061254
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_A3DC67		4.872720701140655
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_A3DBR3		3.3930004882222335
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_A3DI60		2.924280420777634
Hungateiclostridium	94.67418	Hungateiclostridium_thermocellum	94.67418	UniRef90_G2JC59		2.639170379752865
</code></pre></div></div>

<blockquote class="question">
  <h3 id="question-questions-16"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Are there gene families associated with each genus identified with <strong>MetaPhlAn</strong>?</li>
    <li>How many gene families are associated to each genus?</li>
    <li>Are there gene families associated to each species identified with <strong>MetaPhlAn</strong>?</li>
    <li>How many gene families are associated to each species?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-16"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>To answer the questions, we need to group the contents of the output of <strong>Combine MetaPhlAn2 and HUMAnN2 outputs</strong> by 1st column and count the number of occurrences of gene families. We do that using <strong>Group data by a column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:</p>

        <blockquote class="notranslate hands_on">
          <h3 id="hands_on-hands-on-group-by-genus-and-count-gene-families"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Group by genus and count gene families</h3>
          <ol>
            <li><span class="tool" data-tool="Grouping1" title="Tested with Grouping1"><strong>Group data by a column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grouping1</span></span>
              <ul>
                <li><em>“Select data”</em>: output of <strong>Combine MetaPhlAn2 and HUMAnN2 outputs</strong></li>
                <li><em>“Group by column”</em>: <code class="language-plaintext highlighter-rouge">Column:1</code></li>
                <li><em>“Operation”</em>:
                  <ul>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation”</em>
                      <ul>
                        <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Count</code></li>
                        <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">Column:5</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ol>
        </blockquote>

        <p>With <strong>MetaPhlAn</strong>, we identified 3 genus (Coprothermobacter, Methanothermobacter, Hungateiclostridium). But in the output of <strong>Combine MetaPhlAn2 and HUMAnN2 outputs</strong>, we have only gene families for Coprothermobacter and Hungateiclostridium. The abundance of Methanothermobacter is probably too low to correctly identify correctly some gene families.</p>
      </li>
      <li>
        <p>1,889 gene families are associated to Hungateiclostridium and 528 to Coprothermobacter and 202 to Methanothermobacter.</p>
      </li>
      <li>
        <p>For this question, we should group on the 3rd column:</p>

        <blockquote class="notranslate hands_on">
          <h3 id="hands_on-hands-on-group-by-species-and-count-gene-families"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Group by species and count gene families</h3>
          <ol>
            <li><span class="tool" data-tool="Grouping1" title="Tested with Grouping1"><strong>Group data by a column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grouping1</span></span>
              <ul>
                <li><em>“Select data”</em>: output of <strong>Combine MetaPhlAn2 and HUMAnN2 outputs</strong></li>
                <li><em>“Group by column”</em>: <code class="language-plaintext highlighter-rouge">Column:3</code></li>
                <li><em>“Operation”</em>:
                  <ul>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Operation”</em>
                      <ul>
                        <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">Count</code></li>
                        <li><em>“On column”</em>: <code class="language-plaintext highlighter-rouge">Column:5</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ol>
        </blockquote>

        <p>Only 2 species (Coprothermobacter_proteolyticus and Hungateiclostridium thermocellum) identified by <strong>MetaPhlAn</strong> are associated to gene families.</p>
      </li>
      <li>
        <p>As the species found derived directly from the genus (not 2 species for the same genus here), the number of gene families identified are the sames: 528 for Coprothermobacter proteolyticus and 1,889 for Hungateiclostridium thermocellum.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>We could now apply the same tool to the pathways and run similar analysis.</p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we analyzed one metatranscriptomics sample from raw sequences to community structure, functional profiling. To do that, we:</p>

<ol>
  <li>preprocessed the raw data: quality control, trimming and filtering, sequence sorting and formatting</li>
  <li>
    <p>extracted and analyzed the community structure (taxonomic information)</p>

    <p>We identified bacteria to the level of strains, but also some archaea.</p>
  </li>
  <li>
    <p>extracted and analyzed the community functions (functional information)</p>

    <p>We extracted gene families, pathways, but also the gene families involved in pathways and aggregated the gene families into GO terms</p>
  </li>
  <li>combined taxonomic and functional information to offer insights into taxonomic contribution to a function or functions expressed by a particular taxonomy</li>
</ol>

<p>The <span class="notranslate">workflow</span> can be represented this way:</p>

<p><img src="../../images/asaim-wf.png" alt="ASaiM diagram" /></p>

<p>The dataset used here was extracted from a time-series analysis of a microbial community inside a bioreactor (<span class="citation"><a href="#kunath2018proteins">Kunath <i>et al.</i> 2018</a></span>) in which there are 3 replicates over 7 time points. We analyzed here only one single time point for one replicate.</p>



                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Metatranscriptomics data have the same QC profile that RNA-seq data</p>
</li>
                        
                        <li><p>A lot of metatranscriptomics sequences are identified as rRNA sequences</p>
</li>
                        
                        <li><p>With shotgun data, we can extract information about the studied community structure and also the functions realised by the community</p>
</li>
                        
                        <li><p>Metatranscriptomics data analyses are complex and must be careful done, specially when they are done without combination to metagenomics data analyses</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="martin2011cutadapt">Martin, M., 2011 <b>Cutadapt removes adapter sequences from high-throughput sequencing reads</b>. EMBnet. journal 17: 10–12.</li>
<li id="ondov2011interactive">Ondov, B. D., N. H. Bergman, and A. M. Phillippy, 2011 <b>Interactive metagenomic visualization in a Web browser</b>. BMC bioinformatics 12: 385.</li>
<li id="kopylova2012sortmerna">Kopylova, E., L. Noé, and H. Touzet, 2012 <b>SortMeRNA: fast and accurate filtering of ribosomal RNAs in metatranscriptomic data</b>. Bioinformatics 28: 3211–3217.</li>
<li id="truong2015metaphlan2">Truong, D. T., E. A. Franzosa, T. L. Tickle, M. Scholz, G. Weingart <i>et al.</i>, 2015 <b>MetaPhlAn2 for enhanced metagenomic taxonomic profiling</b>. Nature methods 12: 902.</li>
<li id="batut2018asaim">Batut, B., K. Gravouil, C. Defois, S. Hiltemann, J.-F. Brugère <i>et al.</i>, 2018 <b>ASaiM: a Galaxy-based framework to analyze microbiota data</b>. GigaScience 7: giy057.</li>
<li id="franzosa2018species">Franzosa, E. A., L. J. McIver, G. Rahnavard, L. R. Thompson, M. Schirmer <i>et al.</i>, 2018 <b>Species-level functional profiling of metagenomes and metatranscriptomes</b>. Nature methods 15: 962.</li>
<li id="kunath2018proteins">Kunath, B. J., F. Delogu, A. E. Naas, M. Ø. Arntzen, V. G. H. Eijsink <i>et al.</i>, 2018 <b>From proteins to polysaccharides: lifestyle and genetic evolution of Coprothermobacter proteolyticus</b>. The ISME journal 1.</li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Metatranscriptomics analysis using microbiome RNA-seq data (Metagenomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Pratik Jagtap, Subina Mehta, Ray Sajulga, Bérénice Batut, Emma Leith, Praveen Kumar, Saskia Hiltemann, 2021 <b>Metatranscriptomics analysis using microbiome RNA-seq data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{metagenomics-metatranscriptomics,
author = "Pratik Jagtap and Subina Mehta and Ray Sajulga and Bérénice Batut and Emma Leith and Praveen Kumar and Saskia Hiltemann",
title = "Metatranscriptomics analysis using microbiome RNA-seq data (Galaxy Training Materials)",
year = "2021",
month = "05",
day = "20"
url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
