<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Reference-based RNA-Seq data analysis</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Reference-based RNA-Seq data analysis" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/ref-based/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/ref-based/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Reference-based RNA-Seq data analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / ref-based / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Reference-based RNA-Seq data analysis' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html",
  "timeRequired": "PT8H",
  "keywords": "bulk, rna-seq",
  "description": "The questions this  addresses are:\n - What are the steps to process RNA-Seq data?\n - How to identify differentially expressed genes across multiple experimental conditions?\n - What are the biological functions impacted by the differential expression of genes?\n\n\\nThe objectives are:\n - Check a sequence quality report generated by FastQC for RNA-Seq data\n - Explain the principle and specificity of mapping of RNA-Seq data to an eukaryotic reference genome\n - Select and run a state of the art mapping tool for RNA-Seq data\n - Evaluate the quality of mapping results\n - Describe the process to estimate the library strandness\n - Estimate the number of reads per genes\n - Explain the count normalization to perform before sample comparison\n - Construct and run a differential gene expression analysis\n - Analyze the DESeq2 output to identify, annotate and visualize differentially expressed genes\n - Perform a gene ontology enrichment analysis\n - Perform and visualize an enrichment analysis for KEGG pathways\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    },
    {
      "@type": "Person",
      "name": "Mo Heydarian"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Pavankumar Videm"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Peter van Heusden"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    },
    {
      "@type": "Person",
      "name": "Mo Heydarian"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Pavankumar Videm"
    },
    {
      "@type": "Person",
      "name": "Clemens Blank"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Peter van Heusden"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
</script>

<section class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>Reference-based RNA-Seq data analysis</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="https://avatars.githubusercontent.com/bebatut?s=27" alt="Avatar">Bérénice Batut</a>


<a href="/training-material/hall-of-fame/malloryfreeberg/" class="contributor-badge contributor-malloryfreeberg"><img src="https://avatars.githubusercontent.com/malloryfreeberg?s=27" alt="Avatar">Mallory Freeberg</a>


<a href="/training-material/hall-of-fame/moheydarian/" class="contributor-badge contributor-moheydarian"><img src="https://avatars.githubusercontent.com/moheydarian?s=27" alt="Avatar">Mo Heydarian</a>


<a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge contributor-erxleben"><img src="https://avatars.githubusercontent.com/erxleben?s=27" alt="Avatar">Anika Erxleben</a>


<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="https://avatars.githubusercontent.com/pavanvidem?s=27" alt="Avatar">Pavankumar Videm</a>


<a href="/training-material/hall-of-fame/blankclemens/" class="contributor-badge contributor-blankclemens"><img src="https://avatars.githubusercontent.com/blankclemens?s=27" alt="Avatar">Clemens Blank</a>


<a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="https://avatars.githubusercontent.com/mblue9?s=27" alt="Avatar">Maria Doyle</a>


<a href="/training-material/hall-of-fame/nsoranzo/" class="contributor-badge contributor-nsoranzo"><img src="https://avatars.githubusercontent.com/nsoranzo?s=27" alt="Avatar">Nicola Soranzo</a>


<a href="/training-material/hall-of-fame/pvanheus/" class="contributor-badge contributor-pvanheus"><img src="https://avatars.githubusercontent.com/pvanheus?s=27" alt="Avatar">Peter van Heusden</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the steps to process RNA-Seq data?</p>
</li>
        
        <li><p>How to identify differentially expressed genes across multiple experimental conditions?</p>
</li>
        
        <li><p>What are the biological functions impacted by the differential expression of genes?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Check a sequence quality report generated by FastQC for RNA-Seq data</p>
</li>
        
        <li><p>Explain the principle and specificity of mapping of RNA-Seq data to an eukaryotic reference genome</p>
</li>
        
        <li><p>Select and run a state of the art mapping tool for RNA-Seq data</p>
</li>
        
        <li><p>Evaluate the quality of mapping results</p>
</li>
        
        <li><p>Describe the process to estimate the library strandness</p>
</li>
        
        <li><p>Estimate the number of reads per genes</p>
</li>
        
        <li><p>Explain the count normalization to perform before sample comparison</p>
</li>
        
        <li><p>Construct and run a differential gene expression analysis</p>
</li>
        
        <li><p>Analyze the DESeq2 output to identify, annotate and visualize differentially expressed genes</p>
</li>
        
        <li><p>Perform a gene ontology enrichment analysis</p>
</li>
        
        <li><p>Perform and visualize an enrichment analysis for KEGG pathways</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 8 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4541751">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/ref-based/workflows/" title="Workflows" alt="Reference-based RNA-Seq data analysis workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            
                <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/tutorials/ref-based/tours/" title="Galaxy interactive tour" >
    <i class="fas fa-magic" aria-hidden="true"></i> Tours
</a>

</li>
            

            

            
            
            
                 <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/" title="Frequently Asked Questions" alt="Tutorial FAQs">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
  </li>


            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/transcriptomics/ref-based/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jul 9, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In recent years, <span class="notranslate">RNA</span> sequencing (in short <span class="notranslate">RNA</span>-Seq) has become a very widely used technology to analyze the continuously changing cellular transcriptome, i.e. the set of all <span class="notranslate">RNA</span> molecules in one cell or a population of cells. One of the most common aims of <span class="notranslate">RNA</span>-Seq is the profiling of gene <span class="notranslate">expression</span> by identifying genes or molecular pathways that are differentially expressed (DE) between two or more biological conditions. This tutorial demonstrates a computational <span class="notranslate">workflow</span> for the detection of DE genes and pathways from <span class="notranslate">RNA</span>-Seq data by providing a complete analysis of an <span class="notranslate">RNA</span>-Seq experiment profiling <em>Drosophila</em> cells after the depletion of a regulatory gene.</p>

<p>In the study of <span class="citation"><a href="#brooks2011conservation">Brooks <i>et al.</i> 2011</a></span>, the authors identified genes and pathways regulated by the <em>Pasilla</em> gene (the <em>Drosophila</em> homologue of the mammalian splicing regulators Nova-1 and Nova-2 proteins) using <span class="notranslate">RNA</span>-Seq data. They depleted the <em>Pasilla</em> (<em>PS</em>) gene in <em>Drosophila melanogaster</em> by <span class="notranslate">RNA</span> interference (<span class="notranslate">RNA</span>i). Total <span class="notranslate">RNA</span> was then isolated and used to prepare both single-end and paired-end <span class="notranslate">RNA</span>-Seq libraries for treated (PS depleted) and untreated samples. These libraries were sequenced to obtain <span class="notranslate">RNA</span>-Seq <span class="notranslate">reads</span> for each sample. The <span class="notranslate">RNA</span>-Seq data for the treated and the untreated samples can be compared to identify the effects of <em>Pasilla</em> gene depletion on gene <span class="notranslate">expression</span>.</p>

<p>In this tutorial, we illustrate the analysis of the gene <span class="notranslate">expression</span> data step by step using 7 of the original datasets:</p>

<ul>
  <li>4 untreated samples: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461176">GSM461176</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461177">GSM461177</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461178">GSM461178</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461182">GSM461182</a></li>
  <li>3 treated samples (<em>Pasilla</em> gene depleted by <span class="notranslate">RNA</span>i): <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461179">GSM461179</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461180">GSM461180</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM461181">GSM461181</a></li>
</ul>

<p>Each sample constitutes a separate biological replicate of the corresponding condition (treated or untreated). Moreover, two of the treated and two of the untreated samples are from a paired-end sequencing assay, while the remaining samples are from a single-end sequencing experiment.</p>

<blockquote class="comment">
  <h3 id="comment-full-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Full data</h3>
  <p>The original data are available at NCBI Gene <span class="notranslate">Expression</span> Omnibus (GEO) under accession number <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE18508">GSE18508</a>. The raw <span class="notranslate">RNA</span>-Seq <span class="notranslate">reads</span> have been extracted from the Sequence Read Archive (SRA) files and converted into FASTQ files.</p>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
  <li><a href="#mapping" id="markdown-toc-mapping"><span class="notranslate">Mapping</span></a>    <ol>
      <li><a href="#mapping-1" id="markdown-toc-mapping-1"><span class="notranslate">Mapping</span></a></li>
      <li><a href="#inspection-of-the-mapping-results" id="markdown-toc-inspection-of-the-mapping-results">Inspection of the <span class="notranslate">mapping</span> results</a></li>
    </ol>
  </li>
  <li><a href="#counting-the-number-of-reads-per-annotated-gene" id="markdown-toc-counting-the-number-of-reads-per-annotated-gene">Counting the number of <span class="notranslate">reads</span> per annotated gene</a>    <ol>
      <li><a href="#estimation-of-the-strandness" id="markdown-toc-estimation-of-the-strandness">Estimation of the strandness</a></li>
      <li><a href="#counting-reads-per-genes" id="markdown-toc-counting-reads-per-genes">Counting <span class="notranslate">reads</span> per genes</a></li>
    </ol>
  </li>
  <li><a href="#analysis-of-the-differential-gene-expression" id="markdown-toc-analysis-of-the-differential-gene-expression">Analysis of the differential gene <span class="notranslate">expression</span></a>    <ol>
      <li><a href="#identification-of-the-differentially-expressed-features" id="markdown-toc-identification-of-the-differentially-expressed-features">Identification of the differentially expressed features</a></li>
      <li><a href="#extraction-and-annotation-of-differentially-expressed-genes" id="markdown-toc-extraction-and-annotation-of-differentially-expressed-genes">Extraction and annotation of differentially expressed genes</a></li>
      <li><a href="#visualization-of-the-expression-of-the-differentially-expressed-genes" id="markdown-toc-visualization-of-the-expression-of-the-differentially-expressed-genes">Visualization of the <span class="notranslate">expression</span> of the differentially expressed genes</a></li>
    </ol>
  </li>
  <li><a href="#functional-enrichment-analysis-of-the-de-genes" id="markdown-toc-functional-enrichment-analysis-of-the-de-genes">Functional enrichment analysis of the DE genes</a>    <ol>
      <li><a href="#gene-ontology-analysis" id="markdown-toc-gene-ontology-analysis">Gene Ontology analysis</a></li>
      <li><a href="#kegg-pathways-analysis" id="markdown-toc-kegg-pathways-analysis">KEGG pathways analysis</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<p>In the first part of this tutorial we will use the files for 2 out of the 7 samples to demonstrate how to calculate read counts (a measure of the gene <span class="notranslate">expression</span>) from FASTQ files (quality control, <span class="notranslate">mapping</span>, read counting). We provide the FASTQ files for the other 5 samples if you want to reproduce the whole analysis later.</p>

<p>In the second part of the tutorial, read counts of all 7 samples are used to identify and visualize the DE genes, gene families and molecular pathways due to the depletion of the <em>PS</em> gene.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this <span class="notranslate">RNA</span>-Seq exercise</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Import the FASTQ file pairs from <a href="https://zenodo.org/record/4541751">Zenodo</a> or a data library:
      <ul>
        <li><code class="language-plaintext highlighter-rouge">GSM461177</code> (untreated): <code class="language-plaintext highlighter-rouge">GSM461177_1</code> and <code class="language-plaintext highlighter-rouge">GSM461177_2</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461180</code> (treated): <code class="language-plaintext highlighter-rouge">GSM461180_1</code> and <code class="language-plaintext highlighter-rouge">GSM461180_2</code></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/GSM461177_1.fastqsanger
https://zenodo.org/record/4541751/files/GSM461177_2.fastqsanger
https://zenodo.org/record/4541751/files/GSM461180_1.fastqsanger
https://zenodo.org/record/4541751/files/GSM461180_2.fastqsanger
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>Note that these are the full files for the samples and ~1.5Gb each so it may take some minutes to import.</p>

        <p>For a quicker run-through of the FASTQ steps a small subset of each FASTQ file (~5Mb) can be found here on <a href="https://zenodo.org/record/4541751">Zenodo</a>:</p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/GSM461177_1_subsampled.fastqsanger
https://zenodo.org/record/4541751/files/GSM461177_2_subsampled.fastqsanger
https://zenodo.org/record/4541751/files/GSM461180_1_subsampled.fastqsanger
https://zenodo.org/record/4541751/files/GSM461180_2_subsampled.fastqsanger
</code></pre></div>        </div>
      </blockquote>
    </li>
    <li>
      <p>Rename each dataset according to the sample id (e.g. <code class="language-plaintext highlighter-rouge">GSM461177_1</code>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">fastqsanger</code> (e.g. <strong>not</strong> <code class="language-plaintext highlighter-rouge">fastq</code>). If it is not, please change the datatype to <code class="language-plaintext highlighter-rouge">fastqsanger</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p>Add to each dataset a tag corresponding to the name of the sample (<code class="language-plaintext highlighter-rouge">#GSM461177</code> or <code class="language-plaintext highlighter-rouge">#GSM461180</code>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How are the <span class="notranslate">DNA</span> sequences stored?</li>
    <li>What are the other entries of the file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>The <span class="notranslate">DNA</span> sequences are stored in a FASTQ file, in the second line of every 4-line group.</li>
      <li>This file format is called <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a>. It stores sequence information and quality information. Each sequence is represented by a group of 4 lines with the 1st line being the sequence id, the second the sequence of nucleotides, the third a transition line and the last one a sequence of quality score for each nucleotide.</li>
    </ol>
  </blockquote>
</blockquote>

<p>The <span class="notranslate">reads</span> are raw data from the sequencing machine without any pretreatments. They need to be assessed for their quality.</p>

<h1 id="quality-control">Quality control</h1>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data. Adapters may also be present if the <span class="notranslate">reads</span> are longer than the fragments sequenced and trimming these may improve the number of <span class="notranslate">reads</span> mapped.</p>

<p>Sequence quality control is therefore an essential first step in your analysis. We will use similar tools as described in the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control” tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> to create a report of sequence quality, <a href="https://multiqc.info/">MultiQC</a> (<span class="citation"><a href="#ewels2016multiqc">Ewels <i>et al.</i> 2016</a></span>) to aggregate generated reports and <a href="https://cutadapt.readthedocs.io/en/stable/guide.html">Cutadapt</a> (<span class="citation"><a href="#marcel2011cutadapt">Marcel 2011</a></span>) to improve the quality of sequences via trimming and filtering.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: input datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>
    </li>
    <li>
      <p>Inspect the webpage output of <strong>FastQC</strong> for the <code class="language-plaintext highlighter-rouge">GSM461177</code> sample</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>What is the read length?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The read length is 37 bp.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  with the following parameters to aggregate the FastQC reports:
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code> files (output of <strong>FastQC</strong>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the webpage output from MultiQC for each FASTQ</p>

      <blockquote class="question">
        <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What do you think of the quality of the sequences?</li>
          <li>What should we do?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>
              <p>Everything seems good for 3 of the files. The <code class="language-plaintext highlighter-rouge">GSM461177</code> have 10.3 millions of sequences and <code class="language-plaintext highlighter-rouge">GSM461180</code> 12.3 millions of sequences. But in <code class="language-plaintext highlighter-rouge">GSM461180_2</code> (reverse <span class="notranslate">reads</span> of GSM461180) the quality decreases quite a lot at the end of the sequences.</p>

              <p>All files except <code class="language-plaintext highlighter-rouge">GSM461180_2</code> have a high proportion of duplicated <span class="notranslate">reads</span> (expected in <span class="notranslate">RNA</span>-Seq data).</p>

              <figure id="figure-1"><img src="../../images/ref-based/fastqc_sequence_counts_plot.png" alt="Sequence Counts. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Sequence Counts</figcaption></figure>

              <p>The “Per base sequence quality” is globally good with a slight decrease at the end of the sequences. For <code class="language-plaintext highlighter-rouge">GSM461180_2</code>, the decrease is quite large.</p>

              <figure id="figure-2"><img src="../../images/ref-based/fastqc_per_base_sequence_quality_plot.png" alt="Sequence Quality. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Sequence Quality</figcaption></figure>

              <p>The mean quality score over the <span class="notranslate">reads</span> is quite high, but the distribution is slightly different for <code class="language-plaintext highlighter-rouge">GSM461180_2</code>.</p>

              <figure id="figure-3"><img src="../../images/ref-based/fastqc_per_sequence_quality_scores_plot.png" alt="Per Sequence Quality Scores. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Per Sequence Quality Scores</figcaption></figure>

              <p><span class="notranslate">Reads</span> do not really follow a normal distribution of GC content, except for <code class="language-plaintext highlighter-rouge">GSM461180_2</code>.</p>

              <figure id="figure-4"><img src="../../images/ref-based/fastqc_per_sequence_gc_content_plot.png" alt="Per Sequence GC Content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Per Sequence GC Content</figcaption></figure>

              <p>The proportion of N in the <span class="notranslate">reads</span> (bases which could not be called) is low.</p>

              <figure id="figure-5"><img src="../../images/ref-based/fastqc_per_base_n_content_plot.png" alt="Per base N content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Per base N content</figcaption></figure>

              <p>Duplicated sequences: &gt;10 to &gt;500</p>

              <figure id="figure-6"><img src="../../images/ref-based/fastqc_sequence_duplication_levels_plot.png" alt="Sequence Duplication Levels. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Sequence Duplication Levels</figcaption></figure>

              <p>There are almost no known adapters and overrepresented sequences.</p>
            </li>
            <li>
              <p>If the quality of the <span class="notranslate">reads</span> is poor, we should:</p>
              <ol>
                <li>Check what is wrong and think about possible reasons for the poor read quality: it may come from the type of sequencing or what we sequenced (high quantity of overrepresented sequences in transcriptomics data, biased percentage of bases in HiC data)</li>
                <li>Ask the sequencing facility about it</li>
                <li>Perform some quality treatment (taking care not to lose too much information) with some trimming or removal of bad <span class="notranslate">reads</span></li>
              </ol>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We should trim the <span class="notranslate">reads</span> to get rid of bases that were sequenced with high uncertainty (i.e. low quality bases) at the read ends, and also remove the <span class="notranslate">reads</span> of overall bad quality.</p>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What is the relation between <code class="language-plaintext highlighter-rouge">GSM461177_1</code> and <code class="language-plaintext highlighter-rouge">GSM461177_2</code> ?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The data has been sequenced using paired-end sequencing.</p>

    <p>The paired-end sequencing is based on the idea that the initial <span class="notranslate">DNA</span> fragments (longer than the actual read length) is sequenced from both sides. This approach results in two <span class="notranslate">reads</span> per fragment, with the first read in forward orientation and the second read in reverse-complement orientation. The distance between both <span class="notranslate">reads</span> is known. Thus, it can be used as an additional piece of information to improve the read <span class="notranslate">mapping</span>.</p>

    <p>With paired-end sequencing, each fragment is more covered than with single-end sequencing (only forward orientation sequenced):</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    ------&gt;
      ------&gt;
        ------&gt;
          ------&gt;

    ----------------------------- [fragment]

    ------&gt;         &lt;------
      ------&gt;         &lt;------
        ------&gt;         &lt;------
          ------&gt;         &lt;------
</code></pre></div>    </div>

    <p>The paired-end sequencing generates then 2 files:</p>
    <ul>
      <li>One file with the sequences corresponding to forward orientation of all the fragments</li>
      <li>One file with the sequences corresponding to reverse orientation of all the fragments</li>
    </ul>

    <p>Here <code class="language-plaintext highlighter-rouge">GSM461177_1</code> corresponds to the forward <span class="notranslate">reads</span> and <code class="language-plaintext highlighter-rouge">GSM461177_2</code> to the reverse <span class="notranslate">reads</span>.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5</span></span>  with the following parameters to trim low quality sequences:
      <ul>
        <li><em>“Single-end or Paired-end <span class="notranslate">reads</span>?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FASTQ/A file #1”</em>: both <code class="language-plaintext highlighter-rouge">_1</code> fastqsanger datasets (multiple datasets)</li>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FASTQ/A file #2”</em>: both <code class="language-plaintext highlighter-rouge">_2</code> fastqsanger datasets (multiple datasets)</li>
          </ul>

          <p>The order is important here!</p>
        </li>
        <li>In <em>“Filter Options”</em>
          <ul>
            <li><em>“Minimum length”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>
          <ul>
            <li><em>“Quality cutoff”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>

          <blockquote class="question">

            <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

            <p>Why do we run the trimming tool only once on a paired-end dataset and not twice, once for each dataset?</p>

            <blockquote class="solution">
              <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

              <p>The tool can remove sequences if they become too short during the trimming process. For paired-end files it removes entire sequence pairs if one (or both) of the two <span class="notranslate">reads</span> became shorter than the set length cutoff. <span class="notranslate">Reads</span> of a read-pair that are longer than a given threshold but for which the partner read has become too short can optionally be written out to single-end files. This ensures that the information of a read pair is not lost entirely if only one read is of good quality.</p>

            </blockquote>
          </blockquote>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt files (<code class="language-plaintext highlighter-rouge">Report</code>)</p>

      <blockquote class="question">
        <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many basepairs have been removed from the forward <span class="notranslate">reads</span> because of bad quality? And from the reverse <span class="notranslate">reads</span>?</li>
          <li>How many sequence pairs have been removed because at least one read was shorter than the length cutoff?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>For <code class="language-plaintext highlighter-rouge">GSM461177</code>, 5,072,810 bp have been trimmed from the forward <span class="notranslate">reads</span> (read 1) and 8,648,619 bp from the reverse <span class="notranslate">reads</span> (read 2) because of quality. For <code class="language-plaintext highlighter-rouge">GSM461180</code>, 10,224,537 bp from the forward <span class="notranslate">reads</span> and 51,746,850 bp from the reverse <span class="notranslate">reads</span>. This is not a surprise; we saw that at the end of the <span class="notranslate">reads</span> the quality was dropping more for the reverse <span class="notranslate">reads</span> than for the forward <span class="notranslate">reads</span>, especially for <code class="language-plaintext highlighter-rouge">GSM461180</code>.</li>
            <li>147,810 (1.4%) <span class="notranslate">reads</span> were too short for <code class="language-plaintext highlighter-rouge">GSM461177</code> and 1,101,875 (9%) for <code class="language-plaintext highlighter-rouge">GSM461180</code>.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="mapping"><span class="notranslate">Mapping</span></h1>

<p>To make sense of the <span class="notranslate">reads</span>, we need to first figure out where the sequences originated from in the genome, so we can then determine to which genes they belong. When a reference genome for the organism is available, this process is known as aligning or “<span class="notranslate">mapping</span>” the <span class="notranslate">reads</span> to the reference. This is equivalent to solving a jigsaw puzzle, but unfortunately, not all pieces are unique.</p>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Do you want to learn more about the principles behind <span class="notranslate">mapping</span>? Follow our <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">training</a>.</p>
</blockquote>

<p>In this study, the authors used <em>Drosophila melanogaster</em> cells. We should therefore map the quality-controlled sequences to the reference genome of <em>Drosophila melanogaster</em>.</p>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is a reference genome?</li>
    <li>For each model organism, several possible reference genomes may be available (e.g. <code class="language-plaintext highlighter-rouge">hg19</code> and <code class="language-plaintext highlighter-rouge">hg38</code> for human). What do they correspond to?</li>
    <li>Which reference genome should we use?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>A reference genome (or reference assembly) is a set of nucleic acid sequences assembled as a representative example of a species’ genetic material. As they are often assembled from the sequencing of different individuals, they do not accurately represent the set of genes of any single organism, but a mosaic of different nucleic acid sequences from each individual.</li>
      <li>As the cost of <span class="notranslate">DNA</span> sequencing falls, and new full genome sequencing technologies emerge, more genome sequences continue to be generated. Using these new sequences, new alignments are built and the reference genomes improved (fewer gaps, fixed misrepresentations in the sequence, etc). The different reference genomes correspond to the different released versions (called “builds”).</li>
      <li>The genome of <em>Drosophila melanogaster</em> is known and assembled and it can be used as the reference genome in this analysis. Note that new versions of reference genomes may be released if the assembly improves, for this tutorial we are going to use the release 6 of the <em>Drosophila melanogaster</em> reference genome assembly <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4383921/">(dm6)</a>.</li>
    </ol>
  </blockquote>
</blockquote>

<p>With eukaryotic transcriptomes most <span class="notranslate">reads</span> originate from processed m<span class="notranslate">RNA</span>s lacking introns:</p>

<figure id="figure-7"><img src="../../images/ref-based/rna-seq-reads.png" alt="Types of RNA-Seq reads. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> The types of <span class="notranslate">RNA</span>-seq <span class="notranslate">reads</span> (adaption of the Figure 1a from <span class="citation"><a href="#kim2015hisat">Kim <i>et al.</i> 2015</a></span>): <span class="notranslate">reads</span> that mapped entirely within an exon (in red), <span class="notranslate">reads</span> spanning over 2 exons (in blue), read spanning over more than 2 exons (in purple)</figcaption></figure>

<p>Therefore they cannot be simply mapped back to the genome as we normally do for <span class="notranslate">DNA</span> data. Spliced-awared mappers have been developed to efficiently map transcript-derived <span class="notranslate">reads</span> against a reference genome:</p>

<figure id="figure-8"><img src="../../images/splice_aware_alignment.png" alt="Splice-aware alignment. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Principle of spliced mappers: (1) identification of the <span class="notranslate">reads</span> spanning a single exon, (2) identification of the splicing junctions on the unmapped <span class="notranslate">reads</span></figcaption></figure>

<blockquote class="details">
  <h3 id="details-more-details-on-the-different-spliced-mappers"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More details on the different spliced mappers</h3>

  <p>Several spliced mappers have been developed over the past years to process the explosion of <span class="notranslate">RNA</span>-Seq data.</p>

  <p><a href="https://ccb.jhu.edu/software/tophat/index.shtml"><strong>TopHat</strong></a> (<span class="citation"><a href="#trapnell2009tophat">Trapnell <i>et al.</i> 2009</a></span>) was one of the first tools designed specifically to address this problem. In <strong>TopHat</strong> <span class="notranslate">reads</span> are mapped against the genome and are separated into two categories: (1) those that map, and (2) those that are initially unmapped (IUM). “Piles” of <span class="notranslate">reads</span> representing potential exons are extended in search of potential donor/acceptor splice sites and potential splice junctions are reconstructed. IUMs are then mapped to these junctions.</p>

  <figure id="figure-9"><img src="../../images/tophat.png" alt="TopHat. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> TopHat (Figure 1 from <span class="citation"><a href="#trapnell2009tophat">Trapnell <i>et al.</i> 2009</a></span>)</figcaption></figure>

  <p><strong>TopHat</strong> has been subsequently improved with the development of <strong>TopHat2</strong> (<span class="citation"><a href="#kim2013tophat2">Kim <i>et al.</i> 2013</a></span>):</p>

  <figure id="figure-10"><img src="../../images/13059_2012_Article_3053_Fig6_HTML.jpg" alt="TopHat2. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> TopHat2 (Figure 6 from <span class="citation"><a href="#kim2013tophat2">Kim <i>et al.</i> 2013</a></span>)</figcaption></figure>

  <p>To further optimize and speed up spliced read alignment, <a href="https://ccb.jhu.edu/software/hisat2/index.shtml"><strong>HISAT2</strong></a> (<span class="citation"><a href="#kim2019graph">Kim <i>et al.</i> 2019</a></span>) was developed. It uses a hierarchical graph <a href="https://en.wikipedia.org/wiki/FM-index">FM</a> (HGFM) index, representing the entire genome and eventual variants, together with overlapping local indexes (each spanning ~57 kb) that collectively cover the genome and its variants. This allows to find initial seed locations for potential read alignments in the genome using global index and to rapidly refine these alignments using a corresponding local index:</p>

  <figure id="figure-11"><img src="../../images/hisat.png" alt="Hierarchical Graph FM index in HISAT/HISAT2. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> Hierarchical Graph FM index in HISAT/HISAT2 (Figure S8 from <span class="citation"><a href="#kim2015hisat">Kim <i>et al.</i> 2015</a></span>)</figcaption></figure>

  <p>A part of the read (blue arrow) is first mapped to the genome using the global FM index. <strong>HISAT2</strong> then tries to extend the alignment directly utilizing the genome sequence (violet arrow). In (<strong>a</strong>) it succeeds and this read is aligned as it completely resides within an exon. In (<strong>b</strong>) the extension hits a mismatch. Now <strong>HISAT2</strong> takes advantage of the local FM index overlapping this location to find the appropriate <span class="notranslate">mapping</span> for the remainder of this read (green arrow). The (<strong>c</strong>) shows a combination these two strategies: the beginning of the read is mapped using global FM index (blue arrow), extended until it reaches the end of the exon (violet arrow), mapped using local FM index (green arrow) and extended again (violet arrow).</p>

  <p><a href="https://github.com/alexdobin/STAR"><strong>STAR</strong> aligner</a> (<span class="citation"><a href="#dobin2013star">Dobin <i>et al.</i> 2013</a></span>) is a fast alte<span class="notranslate">rna</span>tive for <span class="notranslate">mapping</span> <span class="notranslate">RNA</span>-Seq <span class="notranslate">reads</span> against a reference genome utilizing an uncompressed <a href="https://en.wikipedia.org/wiki/Suffix_array">suffix array</a>. It operates in two stages. In the first stage it performs a seed search:</p>

  <figure id="figure-12"><img src="../../images/star.png" alt="STAR's seed search. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> STAR's seed search (Figure 1 from <span class="citation"><a href="#dobin2013star">Dobin <i>et al.</i> 2013</a></span>)</figcaption></figure>

  <p>Here a read is split between two consecutive exons. <strong>STAR</strong> starts to look for a maximum mappable prefix (MMP) from the beginning of the read until it can no longer match continuously. After this point it start to look for a MMP for the unmatched portion of the read (<strong>a</strong>). In the case of mismatches (<strong>b</strong>) and unalignable regions (<strong>c</strong>) MMPs serve as anchors from which to extend alignments.</p>

  <p>At the second stage <strong>STAR</strong> stitches MMPs to generate read-level alignments that (contrary to MMPs) can contain mismatches and indels. A scoring scheme is used to evaluate and prioritize stitching combinations and to evaluate <span class="notranslate">reads</span> that map to multiple locations. <strong>STAR</strong> is extremely fast but requires a substantial amount of RAM to run efficiently.</p>

</blockquote>

<h2 id="mapping-1"><span class="notranslate">Mapping</span></h2>

<p>We will map our <span class="notranslate">reads</span> to the <em>Drosophila melanogaster</em> genome using <strong>STAR</strong> (<span class="citation"><a href="#dobin2013star">Dobin <i>et al.</i> 2013</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-spliced-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Spliced <span class="notranslate">mapping</span></h3>

  <ol>
    <li>
      <p>Import the Ensembl gene annotation for <em>Drosophila melanogaster</em> (<code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>) from the Shared Data library if available or from <a href="https://zenodo.org/record/4541751/files/Drosophila_melanogaster.BDGP6.87.gtf">Zenodo</a> into your current <span class="notranslate">Galaxy</span> history</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/Drosophila_melanogaster.BDGP6.87.gtf
</code></pre></div>      </div>

      <ol>
        <li>Rename the dataset if necessary</li>
        <li>Verify that the datatype is <code class="language-plaintext highlighter-rouge">gtf</code> and not <code class="language-plaintext highlighter-rouge">gff</code>, and that the database is <code class="language-plaintext highlighter-rouge">dm6</code></li>
      </ol>

      <blockquote class="comment">
        <h3 id="comment-how-to-get-annotation-file"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> How to get annotation file?</h3>

        <p>Annotation files from model organisms may be available on the Shared Data library (the path to them will change from one <span class="notranslate">Galaxy</span> server to the other). You could also retrieve the annotation file from UCSC (using <strong>UCSC Main</strong> tool).</p>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a"><strong><span class="notranslate">RNA</span> STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/rg<span class="notranslate">rna</span>star/<span class="notranslate">rna</span>_star/2.7.8a</span></span>  with the following parameters to map your <span class="notranslate">reads</span> on the reference genome:
      <ul>
        <li><em>“Single-end or paired-end <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Paired-end (as individual datasets)</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“<span class="notranslate">RNA</span>-Seq FASTQ/FASTA file, forward <span class="notranslate">reads</span>”</em>: the <code class="language-plaintext highlighter-rouge">Read 1 Output</code> (outputs of <strong>Cutadapt</strong>)</li>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“<span class="notranslate">RNA</span>-Seq FASTQ/FASTA file, reverse <span class="notranslate">reads</span>”</em>: the <code class="language-plaintext highlighter-rouge">Read 2 Output</code> (outputs of <strong>Cutadapt</strong>)</li>
          </ul>
        </li>
        <li><em>“Custom or built-in reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in index</code>
          <ul>
            <li><em>“Reference genome with or without an annotation”</em>: <code class="language-plaintext highlighter-rouge">use genome reference without builtin gene-model</code>
              <ul>
                <li><em>“Select reference genome”</em>: <code class="language-plaintext highlighter-rouge">Fly (Drosophila Melanogaster): dm6 Full</code></li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene model (gff3,gtf) file for splice junctions”</em>: the imported <code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code></li>
                <li>
                  <p><em>“Length of the genomic sequence around annotated junctions”</em>: <code class="language-plaintext highlighter-rouge">36</code></p>

                  <p>This parameter should be length of <span class="notranslate">reads</span> - 1</p>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the STAR logs:
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">STAR</code></li>
            <li>In <em>“STAR output”</em>
              <ul>
                <li><em>“Type of STAR output?”</em>: <code class="language-plaintext highlighter-rouge">Log</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“STAR output”</em>: <code class="language-plaintext highlighter-rouge">log</code> files (output of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <ol>
          <li>What percentage of <span class="notranslate">reads</span> are mapped exactly once for both samples?</li>
          <li>What are the other available statistics?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>More than 83% for GSM461177 and more than 79% for GSM461180. We can proceed with the analysis since only percentages below 70% should be investigated for potential contamination.</li>
            <li>
              <p>We also have access to the number and percentage of <span class="notranslate">reads</span> that are mapped at several location, mapped at too many different location, not mapped because too short.</p>

              <figure id="figure-13"><img src="../../images/ref-based/star_alignment_plot.png" alt="STAR Alignment Scores. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Alignment scores</figcaption></figure>

              <p>We could have been probably more strict in the minimal read length to avoid these unmapped <span class="notranslate">reads</span> because of length.</p>
            </li>
          </ol>
        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<p>According to the <strong>MultiQC</strong> report, more than 80% of <span class="notranslate">reads</span> for both samples are mapped exactly once to the reference genome. We can proceed with the analysis since only percentages below 70% should be investigated for potential contamination. Both samples have a low (less than 10%) percentage of <span class="notranslate">reads</span> that mapped to multiple locations on the reference genome. This is in the normal range for Illumina short-read sequencing, but may be lower for newer long-read sequencing datasets that can span larger repeated regions in the reference genome.</p>

<p>The main output of <strong>STAR</strong> is a BAM file.</p>

<p>A BAM (<a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Binary Alignment Map</a>) file is a compressed binary file storing the read sequences, whether they have been aligned to a reference sequence (e.g. a chromosome), and if so, the position on the reference sequence at which they have been aligned.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-a-bamsam-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect a BAM/SAM file</h3>

  <ol>
    <li>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong><span class="notranslate">RNA</span> STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>

</blockquote>

<p>A BAM file (or a SAM file, the non-compressed version) consists of:</p>

<ul>
  <li>A header section (the lines starting with <code class="language-plaintext highlighter-rouge">@</code>) containing metadata particularly the chromosome names and lengths (lines starting with the <code class="language-plaintext highlighter-rouge">@SQ</code> symbol)</li>
  <li>
    <p>An alignment section consisting of a table with 11 mandatory fields, as well as a variable number of optional fields:</p>

    <table>
      <thead>
        <tr>
          <th>Col</th>
          <th>Field</th>
          <th>Type</th>
          <th>Brief Description</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td>QNAME</td>
          <td>String</td>
          <td>Query template NAME</td>
        </tr>
        <tr>
          <td>2</td>
          <td>FLAG</td>
          <td>Integer</td>
          <td>Bitwise FLAG</td>
        </tr>
        <tr>
          <td>3</td>
          <td><span class="notranslate">RNA</span>ME</td>
          <td>String</td>
          <td>References sequence NAME</td>
        </tr>
        <tr>
          <td>4</td>
          <td>POS</td>
          <td>Integer</td>
          <td>1- based leftmost <span class="notranslate">mapping</span> POSition</td>
        </tr>
        <tr>
          <td>5</td>
          <td>MAPQ</td>
          <td>Integer</td>
          <td><span class="notranslate">MAPping</span> Quality</td>
        </tr>
        <tr>
          <td>6</td>
          <td>CIGAR</td>
          <td>String</td>
          <td>CIGAR String</td>
        </tr>
        <tr>
          <td>7</td>
          <td>RNEXT</td>
          <td>String</td>
          <td>Ref. name of the mate/next read</td>
        </tr>
        <tr>
          <td>8</td>
          <td>PNEXT</td>
          <td>Integer</td>
          <td>Position of the mate/next read</td>
        </tr>
        <tr>
          <td>9</td>
          <td>TLEN</td>
          <td>Integer</td>
          <td>Observed Template LENgth</td>
        </tr>
        <tr>
          <td>10</td>
          <td>SEQ</td>
          <td>String</td>
          <td>Segment SEQuence</td>
        </tr>
        <tr>
          <td>11</td>
          <td>QUAL</td>
          <td>String</td>
          <td>ASCII of Phred-scaled base QUALity+33</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<blockquote class="question">
  <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which information do you find in a SAM/BAM file?</li>
    <li>What is the additional information compared to a FASTQ file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Sequences and quality information, like a FASTQ</li>
      <li><span class="notranslate">Mapping</span> information, Location of the read on the chromosome, <span class="notranslate">Mapping</span> quality, etc</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="inspection-of-the-mapping-results">Inspection of the <span class="notranslate">mapping</span> results</h2>

<p>The BAM file contains information for all our <span class="notranslate">reads</span>, making it difficult to inspect and explore in text format. A powerful tool to visualize the content of BAM files is the Integrative Genomics Viewer (<strong>IGV</strong>, <span class="citation"><a href="#robinson2011integrative">Robinson <i>et al.</i> 2011</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspection-of-mapping-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspection of <span class="notranslate">mapping</span> results</h3>

  <ol>
    <li>Install <a href="https://software.broadinstitute.org/software/igv/download"><strong>IGV</strong></a> (if not already installed)</li>
    <li>Start IGV locally</li>
    <li>Expand the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">mapped.bam</code> file (output of <strong><span class="notranslate">RNA</span> STAR</strong>) for <code class="language-plaintext highlighter-rouge">GSM461177</code></li>
    <li>
      <p>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV local D. melanogaster (dm6)</code> to load the <span class="notranslate">reads</span> into the IGV browser</p>

      <blockquote class="comment">
        <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>

        <p>In order for this step to work, you will need to have either IGV or <a href="https://www.java.com/en/download/faq/java_webstart.xml">Java Web Start</a>
installed on your machine. However, the questions in this section can also be answered by inspecting the IGV screenshots below.</p>

        <p>Check the <a href="https://software.broadinstitute.org/software/igv/AlignmentData">IGV documentation</a> for more information.</p>

      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Zoom to <code class="language-plaintext highlighter-rouge">chr4:540,000-560,000</code> (Chromosome 4 between 540 kb to 560 kb)</p>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <figure id="figure-14"><img src="../../images/junction_igv_screenshot.png" alt="Screenshot of the IGV view on Chromosome 4. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> Screenshot of IGV on Chromosome 4</figcaption></figure>

        <ol>
          <li>What information appears at the top as grey peaks?</li>
          <li>What do the connecting lines between some of the aligned <span class="notranslate">reads</span> indicate?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>The coverage plot: the sum of mapped <span class="notranslate">reads</span> at each position</li>
            <li>They indicate junction events (or splice sites), <em>i.e.</em> <span class="notranslate">reads</span> that are mapped across an intron</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Inspect the splice junctions using a <strong>Sashimi plot</strong></p>

      <blockquote class="comment">
        <h3 id="comment-creation-of-a-sashimi-plot"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Creation of a Sashimi plot</h3>

        <ul>
          <li>Right click on the BAM file (in IGV)</li>
          <li>Select <strong>Sashimi Plot</strong> from the menu</li>
        </ul>
      </blockquote>

      <blockquote class="question">
        <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <figure id="figure-15"><img src="../../images/star_igv_sashimi.png" alt="Screenshot of a Sashimi plot of Chromosome 4. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> Screenshot of a Sashimi plot of Chromosome 4</figcaption></figure>

        <ol>
          <li>What does the vertical red bar graph represent? What about the arcs with numbers?</li>
          <li>What do the numbers on the arcs mean?</li>
          <li>Why do we observe different stacked groups of blue linked boxes at the bottom?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>The coverage for each alignment track is plotted as a red bar graph. Arcs represent observed splice junctions, <em>i.e.</em>, <span class="notranslate">reads</span> spanning introns.</li>
            <li>The numbers refer to the number of observed junction <span class="notranslate">reads</span>.</li>
            <li>The different groups of linked boxes on the bottom represent the different transcripts from the genes at this location which are present in the GTF file.</li>
          </ol>

        </blockquote>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Check the <a href="https://software.broadinstitute.org/software/igv/Sashimi">IGV documentation on Sashimi plots</a> to find some clues</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <h3 id="details-further-check-for-the-quality-of-the-data"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Further check for the quality of the data</h3>

  <p>The quality of the data and <span class="notranslate">mapping</span> can be checked further, e.g. by inspecting read duplication level, number of <span class="notranslate">reads</span> mapped to each chromosome, gene body coverage, and read distribution across features.</p>

  <p><em>These steps have been inspired by the ones provided in the <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">great “<span class="notranslate">RNA</span>-Seq <span class="notranslate">reads</span> to counts” tutorial</a> and adapted to our datasets.</em></p>

  <h4 id="duplicate-reads">Duplicate <span class="notranslate">reads</span></h4>

  <p>In the FastQC report, we saw that some <span class="notranslate">reads</span> are duplicated:</p>

  <figure id="figure-16"><img src="../../images/ref-based/fastqc_sequence_duplication_levels_plot.png" alt="Sequence Duplication Levels. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 16:</span> Sequence Duplication Levels</figcaption></figure>

  <p>Duplicate <span class="notranslate">reads</span> can come from highly-expressed genes, therefore they are usually kept in <span class="notranslate">RNA</span>-Seq differential <span class="notranslate">expression</span> analysis. But a high percentage of duplicates may indicate an issue, e.g. over amplification during PCR of low complexity library.</p>

  <p><strong>MarkDuplicates</strong> from <a href="http://broadinstitute.github.io/picard/">Picard suite</a> examines aligned records from a BAM file to locate duplicate <span class="notranslate">reads</span>, i.e. <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the same location (based on the start position of the <span class="notranslate">mapping</span>).</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-check-duplicate-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check duplicate <span class="notranslate">reads</span></h3>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2"><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2</span></span> :
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select SAM/BAM dataset or dataset collection”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the MarkDuplicates logs:
        <ul>
          <li>In <em>“Results”</em>
            <ul>
              <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">Picard</code></li>
              <li>In <em>“Picard output”</em>
                <ul>
                  <li><em>“Type of Picard output?”</em>: <code class="language-plaintext highlighter-rouge">Markdups</code></li>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Picard output”</em>: <code class="language-plaintext highlighter-rouge">MarkDuplicate metrics</code> files (output of <strong>MarkDuplicates</strong>)</li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

          <p>What are the percentages of duplicate <span class="notranslate">reads</span> for each sample?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
            <p>The sample <code class="language-plaintext highlighter-rouge">GSM461177</code> has 27.8% of duplicated <span class="notranslate">reads</span> while <code class="language-plaintext highlighter-rouge">GSM461180</code> has 25.9%.</p>
          </blockquote>
        </blockquote>
      </li>
    </ol>
  </blockquote>

  <p>In general, obtaining up to 50% duplicated <span class="notranslate">reads</span> is considered normal. So both our samples are fine.</p>

  <h4 id="number-of-reads-mapped-to-each-chromosome">Number of <span class="notranslate">reads</span> mapped to each chromosome</h4>

  <p>To assess the sample quality (e.g. excess of mitochondrial contamination), we can check the sex of samples, or to see if any chromosomes have highly expressed genes, we can check the numbers of <span class="notranslate">reads</span> mapped to each chromosome using <strong>IdxStats</strong> from the <strong>Samtools</strong> suite.</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-check-the-number-of-reads-mapped-to-each-chromosome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check the number of <span class="notranslate">reads</span> mapped to each chromosome</h3>

    <ol>
      <li><strong>Samtools idxstats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the idxstats logs:
        <ul>
          <li>In <em>“Results”</em>
            <ul>
              <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">Samtools</code></li>
              <li>In <em>“Samtools output”</em>
                <ul>
                  <li><em>“Type of Samtools output?”</em>: <code class="language-plaintext highlighter-rouge">idxstats</code></li>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Samtools output”</em>: <code class="language-plaintext highlighter-rouge">Samtools idxstats</code> files</li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <p><img src="../../images/ref-based/samtools-idxstats-mapped-reads-plot.png" alt="Samtools idxstats. " loading="lazy" /></p>

          <ol>
            <li>How many chromosomes does the <em>Drosophila</em> genome have?</li>
            <li>Where did the <span class="notranslate">reads</span> mostly map?</li>
            <li>Can we determine the sex of the sample?</li>
          </ol>

          <blockquote class="solution">
            <h3 id="solution-solution-12"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
            <ol>
              <li>The genome of <em>Drosophila</em> has 4 pairs of chromosomes: X/Y, 2, 3, and 4.</li>
              <li>The <span class="notranslate">reads</span> mapped mostly to chromosome 2 (chr2L and chr2R), 3 (chr3L and chr3R) and X. Only a few <span class="notranslate">reads</span> mapped to chromosome 4, which is expected given that this chromosome is very small.</li>
              <li>
                <p>Judging from the percentage of X+Y <span class="notranslate">reads</span>, most of the <span class="notranslate">reads</span> map to X and only a few to Y. This indicates there are probably not many genes on Y, so the samples are probably both female.</p>

                <p><img src="../../images/ref-based/samtools-idxstats-xy-plot.png" alt="Samtools idxstats. " loading="lazy" /></p>
              </li>
            </ol>
          </blockquote>
        </blockquote>
      </li>
    </ol>
  </blockquote>

  <h4 id="gene-body-coverage">Gene body coverage</h4>

  <p>The gene body is the different regions of a gene. It is important to check if read coverage is uniform over gene body or if there is any 5’/3’ bias. For example, a bias towards the 5’ end of genes could indicate degradation of the <span class="notranslate">RNA</span>. Alte<span class="notranslate">rna</span>tively, a 3’ bias could indicate that the data is from a 3’ assay. To assess this, we can use the <strong>Gene Body Coverage</strong> tool from the RSeQC (<span class="citation"><a href="#wang2012rseqc">Wang <i>et al.</i> 2012</a></span>) tool suite. This tool scales all transcripts to 100 nucleotides (using a provided annotation file) and calculates the number of <span class="notranslate">reads</span> covering each nucleotide position.</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-check-gene-body-coverage"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check gene body coverage</h3>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357"><strong>Convert GTF to <span class="notranslate">BED</span>12</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gtftobed12/gtftobed12/357</span></span>  to convert the GTF file to <span class="notranslate">BED</span>:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF File to convert”</em>: <code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/2.6.4.3" title="Tested with toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/2.6.4.3"><strong>Gene Body Coverage (BAM)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_geneBody_coverage/2.6.4.3</span></span> :
        <ul>
          <li><em>“Run each sample separately, or combine mutiple samples into one plot”</em>: <code class="language-plaintext highlighter-rouge">Run each sample separately</code>
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input .bam file”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
            </ul>
          </li>
          <li><em>“Reference gene model”</em>: <span class="notranslate">BED</span>12 file (output of <strong>Convert GTF to <span class="notranslate">BED</span>12</strong>)</li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the idxstats logs:
        <ul>
          <li>In <em>“Results”</em>
            <ul>
              <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">RSeQC</code></li>
              <li>In <em>“RSeQC output”</em>
                <ul>
                  <li><em>“Type of RSeQC output?”</em>: <code class="language-plaintext highlighter-rouge">gene_body_coverage</code></li>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“RSeQC gene_body_coverage output”</em>: <code class="language-plaintext highlighter-rouge">(text)</code> files (outputs of <strong>Gene Body Coverage (BAM)</strong>)</li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-question-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

          <p><img src="../../images/ref-based/rseqc_gene_body_coverage_plot.png" alt="Gene body coverage. " loading="lazy" /></p>

          <p>How are the coverage across gene bodies? Are the samples biased in 3’ or 5’?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-13"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p>For both samples there is a pretty even coverage from 5’ to 3’ ends (despite some noise in the middle). So no obvious bias in both samples.</p>
          </blockquote>
        </blockquote>
      </li>
    </ol>
  </blockquote>

  <h4 id="read-distribution-across-features">Read distribution across features</h4>

  <p>With <span class="notranslate">RNA</span>-Seq data, we expect most <span class="notranslate">reads</span> to map to exons rather than introns or intergenic regions. Before going further in counting and differential <span class="notranslate">expression</span> analysis, it may be interesting to check the distribution of <span class="notranslate">reads</span> across known gene features (exons, CDS, 5’ UTR, 3’ UTR, introns, intergenic regions). For example, a high number of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to intergenic regions may indicate the presence of <span class="notranslate">DNA</span> contamination.</p>

  <p>Here we will use the <strong>Read Distribution</strong> tool from the RSeQC (<span class="citation"><a href="#wang2012rseqc">Wang <i>et al.</i> 2012</a></span>) tool suite, which uses the annotation file to identify the position of the different gene features.</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-check-the-number-of-reads-mapped-to-each-chromosome-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check the number of <span class="notranslate">reads</span> mapped to each chromosome</h3>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/2.6.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/2.6.4.1"><strong>Read Distribution</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_read_distribution/2.6.4.1</span></span> :
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input .bam/.sam file”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
          <li><em>“Reference gene model”</em>: <span class="notranslate">BED</span>12 file (output <strong>Convert GTF to <span class="notranslate">BED</span>12</strong>)</li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the idxstats logs:
        <ul>
          <li>In <em>“Results”</em>
            <ul>
              <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">RSeQC</code></li>
              <li>In <em>“RSeQC output”</em>
                <ul>
                  <li><em>“Type of RSeQC output?”</em>: <code class="language-plaintext highlighter-rouge">read_distribution</code></li>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“RSeQC read_distribution output”</em>: outputs of <strong>Read Distribution</strong></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-question-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

          <p><img src="../../images/ref-based/rseqc_read_distribution_plot.png" alt="Samtools idxstats. " loading="lazy" /></p>

          <p>What do you think of the read distribution?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-14"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p>Most of the <span class="notranslate">reads</span> are mapped to exons (&gt;80%), only ~2% to introns and ~5% to intergenic regions, which is what we expect. It confirms that our data are <span class="notranslate">RNA</span>-Seq data and that <span class="notranslate">mapping</span> was successful.</p>
          </blockquote>
        </blockquote>
      </li>
    </ol>
  </blockquote>

  <p>Now that we have checked the results of the read <span class="notranslate">mapping</span>, we can proceed to the next phase of the analysis.</p>
</blockquote>

<p>After the <span class="notranslate">mapping</span>, we now have the information on where the <span class="notranslate">reads</span> are located on the reference genome and how well they were mapped. The next step in <span class="notranslate">RNA</span>-Seq data analysis is quantification of the number of <span class="notranslate">reads</span> mapped to genomic features (genes, transcripts, exons, …).</p>

<blockquote class="comment">
  <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>The quantification depends on both the reference genome (the FASTA file) and its associated annotations (the GTF file). It is extremely important to use an annotation file that corresponds to the same version of the reference genome you used for the <span class="notranslate">mapping</span> (e.g. <code class="language-plaintext highlighter-rouge">dm6</code> here), as the chromosomal coordinates of genes are usually different amongst different reference genome versions.</p>
</blockquote>

<p>Here we will focus on the genes, as we would like to identify the ones that are differentially expressed because of the Pasilla gene knockdown.</p>

<h1 id="counting-the-number-of-reads-per-annotated-gene">Counting the number of <span class="notranslate">reads</span> per annotated gene</h1>

<p>To compare the <span class="notranslate">expression</span> of single genes between different conditions (<em>e.g.</em> with or without PS depletion), an essential first step is to quantify the number of <span class="notranslate">reads</span> per gene, or more specifically the number of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the exons of each gene.</p>

<figure id="figure-17"><img src="../../images/gene_counting.png" alt="Counting the number of reads per annotated gene. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 17:</span> Counting the number of <span class="notranslate">reads</span> per annotated gene</figcaption></figure>

<blockquote class="question">
  <h3 id="question-question-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>In the previous image,</p>

  <ol>
    <li>How many <span class="notranslate">reads</span> are found for the different exons?</li>
    <li>How many <span class="notranslate">reads</span> are found for the different genes?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-15"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>Number of <span class="notranslate">reads</span> per exons</p>

        <table>
          <thead>
            <tr>
              <th>Exon</th>
              <th>Number of <span class="notranslate">reads</span></th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>gene1 - exon1</td>
              <td>3</td>
            </tr>
            <tr>
              <td>gene1 - exon2</td>
              <td>2</td>
            </tr>
            <tr>
              <td>gene2 - exon1</td>
              <td>3</td>
            </tr>
            <tr>
              <td>gene2 - exon2</td>
              <td>4</td>
            </tr>
            <tr>
              <td>gene2 - exon3</td>
              <td>3</td>
            </tr>
          </tbody>
        </table>
      </li>
      <li>
        <p>gene1 has 4 <span class="notranslate">reads</span>, not 5, because of the splicing of the last read (gene1 - exon1 + gene1 - exon2). gene2 has 6 <span class="notranslate">reads</span>, 3 of which are spliced.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<p>Two main tools are available for read counting: <a href="http://htseq.readthedocs.io/en/release_0.9.1/count.html"><strong>HTSeq-count</strong></a> (<span class="citation"><a href="#anders2015htseq">Anders <i>et al.</i> 2015</a></span>) or <strong>featureCounts</strong> (<span class="citation"><a href="#liao2013featurecounts">Liao <i>et al.</i> 2013</a></span>). <strong>featureCounts</strong> is considerably faster and requires far less computational resources, so we will use it here.</p>

<p>In principle, the counting of <span class="notranslate">reads</span> overlapping with genomic features is a fairly simple task. But there are some details that need to be given to <strong>featureCounts</strong>, e.g. the strandness.</p>

<h2 id="estimation-of-the-strandness">Estimation of the strandness</h2>

<p><span class="notranslate">RNA</span>s that are typically targeted in <span class="notranslate">RNA</span>-Seq experiments are single stranded (<em>e.g.</em>, m<span class="notranslate">RNA</span>s) and thus have polarity (5’ and 3’ ends that are functionally distinct). During a typical <span class="notranslate">RNA</span>-Seq experiment the information about strandness is lost after both strands of c<span class="notranslate">DNA</span> are synthesized, size selected, and converted into a sequencing library. However, this information can be quite useful for the read counting step, especially for <span class="notranslate">reads</span> located on the overlap of 2 genes that are on different strands.</p>

<figure id="figure-18"><img src="../../images/ref-based/strandness_why.png" alt="Why strandness?" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 18:</span> Read1 will be assigned to gene1 located on the forward strand but Read2 could be assigned to gene1 (forward strand) or gene2 (reverse strand) depending if the strandness information is conserved.</figcaption></figure>

<p>Some library preparation protocols create so-called <em>stranded</em> <span class="notranslate">RNA</span>-Seq libraries that preserve the strand information (<span class="citation"><a href="#levin2010comprehensive">Levin <i>et al.</i> 2010</a></span> provides an excellent overview). In practice, with Illumina paired-end <span class="notranslate">RNA</span>-Seq protocols you are unlikely to encounter many of these possibilities. You will most likely deal with either:</p>

<ul>
  <li>Unstranded <span class="notranslate">RNA</span>-Seq data</li>
  <li>Stranded <span class="notranslate">RNA</span>-Seq data generated by the use of specialized <span class="notranslate">RNA</span> isolation kits during sample preparation</li>
</ul>

<blockquote class="details">
  <h3 id="details-more-details-about-strandness"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More details about strandness</h3>

  <figure id="figure-19"><img src="../../images/dna_rna.png" alt="Relationship between DNA and RNA orientation. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 19:</span> Relationship between <span class="notranslate">DNA</span> and <span class="notranslate">RNA</span> orientation</figcaption></figure>

  <p>The implication of stranded <span class="notranslate">RNA</span>-Seq is that you can distinguish whether the <span class="notranslate">reads</span> are derived from forward or reverse-encoded transcripts:</p>

  <figure id="figure-20"><img src="../../images/stranded_result.png" alt="Stranded RNA-Seq data look like this. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 20:</span> Non-stranded vs. strand-specific <span class="notranslate">RNA</span>-Seq read alignment (Figure from GATC Biotech)</figcaption></figure>

  <p>Depending on the approach, and whether one performs single-end or paired-end sequencing, there are multiple possibilities on how to interpret the results of the <span class="notranslate">mapping</span> of these <span class="notranslate">reads</span> to the genome:</p>

  <figure id="figure-21"><img src="../../images/rnaseq_library_type.png" alt="Effects of RNA-Seq library types. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 21:</span> Effects of <span class="notranslate">RNA</span>-Seq library types (Figure adapted from Sailfish documentation)</figcaption></figure>
</blockquote>

<p>This information should be provided with your FASTQ files, ask your sequencing facility! If not, try to find it on the site where you downloaded the data or in the corresponding publication.</p>

<p>Another option is to estimate these parameters with a tool called <strong>Infer Experiment</strong> from the RSeQC (<span class="citation"><a href="#wang2012rseqc">Wang <i>et al.</i> 2012</a></span>) tool suite. This tool takes the BAM files from the <span class="notranslate">mapping</span>, selects a subsample of the <span class="notranslate">reads</span> and compares their genome coordinates and strands with those of the reference gene model (from an annotation file). Based on the strand of the genes, it can gauge whether sequencing is strand-specific, and if so, how <span class="notranslate">reads</span> are stranded (forward or reverse):</p>

<figure id="figure-22"><img src="../../images/ref-based/strandness_cases.png" alt="How to estimate the strandness?" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 22:</span> In a stranded forward library, <span class="notranslate">reads</span> map mostly on the genes located on forward strand (here gene1). With stranded reverse library, <span class="notranslate">reads</span> map mostly on genes on the reverse strand (here gene2). With unstranded library, <span class="notranslate">reads</span> maps on genes on both strands.</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-determining-the-library-strandness"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Determining the library strandness</h3>

  <p>We use the <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span>12</code> file which we already converted from the <code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code> dataset earlier.</p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1"><strong>Infer Experiment</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1</span></span>  to determine the library strandness with:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input .bam file”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference gene model”</em>: <span class="notranslate">BED</span>12 file (output of the <strong>Convert GTF to <span class="notranslate">BED</span>12</strong> tool)</li>
        <li><em>“Number of <span class="notranslate">reads</span> sampled from SAM/BAM file (default = 200000)”</em>: <code class="language-plaintext highlighter-rouge">200000</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1"><strong>Infer Experiment</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nilesh/rseqc/rseqc_infer_experiment/2.6.4.1</span></span>  tool generates one file with information on:</p>

<ul>
  <li>Paired-end or single-end library</li>
  <li>Fraction of <span class="notranslate">reads</span> failed to determine</li>
  <li>2 lines
    <ul>
      <li>For single-end
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Fraction of <span class="notranslate">reads</span> explained by "++,--"</code>: the fraction of <span class="notranslate">reads</span> that assigned to forward strand</li>
          <li><code class="language-plaintext highlighter-rouge">Fraction of <span class="notranslate">reads</span> explained by "+-,-+"</code>: the fraction of <span class="notranslate">reads</span> that assigned to reverse strand</li>
        </ul>
      </li>
      <li>For paired-end
        <ul>
          <li><code class="language-plaintext highlighter-rouge">Fraction of <span class="notranslate">reads</span> explained by "1++,1--,2+-,2-+"</code>: the fraction of <span class="notranslate">reads</span> that assigned to forward strand</li>
          <li><code class="language-plaintext highlighter-rouge">Fraction of <span class="notranslate">reads</span> explained by "1+-,1-+,2++,2--"</code>: the fraction of <span class="notranslate">reads</span> that assigned to reverse strand</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>If the two “Fraction of <span class="notranslate">reads</span> explained by” numbers are close to each other, we conclude that the library is not a strand-specific dataset (or unstranded).</p>

<blockquote class="question">
  <h3 id="question-question-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>What are the “Fraction of the <span class="notranslate">reads</span> explained by” results for <code class="language-plaintext highlighter-rouge">GSM461177</code>?</li>
    <li>Do you think the library type of the 2 samples is stranded or unstranded?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-16"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>Results for <code class="language-plaintext highlighter-rouge">GSM46177</code>:</p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>This is PairEnd Data
Fraction of <span class="notranslate">reads</span> failed to determine: 0.0963
Fraction of <span class="notranslate">reads</span> explained by "1++,1--,2+-,2-+": 0.4649
Fraction of <span class="notranslate">reads</span> explained by "1+-,1-+,2++,2--": 0.4388
</code></pre></div>        </div>

        <p>so 46.46% of the <span class="notranslate">reads</span> are assigned to the forward strand and 43.88% to the reverse strand.</p>
      </li>
      <li>
        <p>Similar statistics are found for <code class="language-plaintext highlighter-rouge">GSM461180</code>, so the library seems to be of the type unstranded for both samples.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="details">
  <h3 id="details-strandness-and-software-settings"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Strandness and software settings</h3>
  <p>As it is sometimes quite difficult to find out which settings correspond to those of other programs, the following table might be helpful to identify the library type:</p>

  <table>
    <thead>
      <tr>
        <th>Library type</th>
        <th><strong>Infer Experiment</strong></th>
        <th><strong>TopHat</strong></th>
        <th><strong>HISAT2</strong></th>
        <th><strong>HTSeq-count</strong></th>
        <th><strong>featureCounts</strong></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Paired-End (PE) - SF</td>
        <td>1++,1–,2+-,2-+</td>
        <td>FR Second Strand</td>
        <td>Second Strand F/FR</td>
        <td>yes</td>
        <td>Forward (1)</td>
      </tr>
      <tr>
        <td>PE - SR</td>
        <td>1+-,1-+,2++,2–</td>
        <td>FR First Strand</td>
        <td>First Strand R/RF</td>
        <td>reverse</td>
        <td>Reverse (2)</td>
      </tr>
      <tr>
        <td>Single-End (SE) - SF</td>
        <td>++,–</td>
        <td>FR Second Strand</td>
        <td>Second Strand F/FR</td>
        <td>yes</td>
        <td>Forward (1)</td>
      </tr>
      <tr>
        <td>SE - SR</td>
        <td>+-,-+</td>
        <td>FR First Strand</td>
        <td>First Strand R/RF</td>
        <td>reverse</td>
        <td>Reverse (2)</td>
      </tr>
      <tr>
        <td>PE, SE - U</td>
        <td>undecided</td>
        <td>FR Unstranded</td>
        <td>default</td>
        <td>no</td>
        <td>Unstranded (0)</td>
      </tr>
    </tbody>
  </table>

</blockquote>

<h2 id="counting-reads-per-genes">Counting <span class="notranslate">reads</span> per genes</h2>

<p>We now run <strong>featureCounts</strong> to count the number of <span class="notranslate">reads</span> per annotated gene.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-counting-the-number-of-reads-per-annotated-gene"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Counting the number of <span class="notranslate">reads</span> per annotated gene</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.1"><strong>featureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.1</span></span>  to count the number of <span class="notranslate">reads</span> per gene:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Alignment file”</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> files (outputs of <strong><span class="notranslate">RNA</span> STAR</strong>)</li>
        <li><em>“Specify strand information”</em>: <code class="language-plaintext highlighter-rouge">Unstranded</code></li>
        <li><em>“Gene annotation file”</em>: <code class="language-plaintext highlighter-rouge">in your history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene annotation file”</em>: <code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code></li>
          </ul>
        </li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">Gene-ID "\t" read-count (MultiQC/DESeq2/edgeR/limma-voom compatible)</code></li>
        <li><em>“Create gene-length file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>In <em>“Options for paired-end <span class="notranslate">reads</span>”</em>:
          <ul>
            <li><em>“Count fragments instead of <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Enabled; fragments (or templates) will be counted instead of <span class="notranslate">reads</span></code></li>
          </ul>
        </li>
        <li>In <em>“Advanced options”</em>:
          <ul>
            <li><em>“GFF feature type filter”</em>: <code class="language-plaintext highlighter-rouge">exon</code></li>
            <li><em>“GFF gene identifier”</em>: <code class="language-plaintext highlighter-rouge">gene_id</code></li>
            <li><em>“Allow <span class="notranslate">reads</span> to map to multiple features”</em>: <code class="language-plaintext highlighter-rouge">Disabled; <span class="notranslate">reads</span> that align to multiple features or overlapping features are excluded</code></li>
            <li><em>“Minimum <span class="notranslate">mapping</span> quality per read”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.7</span></span>  to aggregate the report:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">featureCounts</code>
              <ul>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Output of FeatureCounts”</em>: <code class="language-plaintext highlighter-rouge">summary</code> files (outputs of <strong>featureCounts</strong>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <ol>
          <li>How many <span class="notranslate">reads</span> have been assigned to a gene?</li>
          <li>When should we be worried about the assignment rate? What should we do?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-17"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>
              <p>Around 70% of the <span class="notranslate">reads</span> have been assigned to genes: this quantity is good enough.</p>

              <figure id="figure-23"><img src="../../images/ref-based/featureCounts_assignment_plot.png" alt="featureCounts assignment. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 23:</span> Assignments with featureCounts</figcaption></figure>

              <p>Some <span class="notranslate">reads</span> are not assigned because they were multi-mapped; others were assigned to no features or to ambiguous ones.</p>
            </li>
            <li>
              <p>If the percentage is below 50%, you should investigate where your <span class="notranslate">reads</span> are <span class="notranslate">mapping</span> (inside genes or not, with IGV) and check that the annotation corresponds to the correct reference genome version.</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The main output of <strong>featureCounts</strong> is a table with the counts, i.e. the number of <span class="notranslate">reads</span> (or fragments in the case of paired-end <span class="notranslate">reads</span>) mapped to each gene (in rows, with their ID in the first column) in the provided annotation. <strong>FeatureCount</strong> generates also the <strong>feature length</strong> output datasets. We will need this file later on when we will run the <strong>goseq</strong> tool.</p>

<blockquote class="question">
  <h3 id="question-question-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Which feature has the most counts for both samples? (Hint: Use the Sort tool)</p>

  <blockquote class="solution">
    <h3 id="solution-solution-18"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>To display the most abundantly detected feature, we need to sort the table of counts. This can be done using the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1"><strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1</span></span>  tool:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort Query”</em>: count file, output of by <strong>featureCounts</strong></li>
      <li><em>“Number of header”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      <li>In <em>“1: Column selections”</em>:
        <ul>
          <li>
            <p><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">2</code></p>

            <p>This column contains the number of <span class="notranslate">reads</span> = counts</p>
          </li>
          <li>
            <p><em>“in”</em>: <code class="language-plaintext highlighter-rouge">Descending order</code></p>
          </li>
        </ul>
      </li>
    </ul>

    <p>The result of sorting the table on column 2 reveals that FBgn0000556 is the feature with the most counts (around 128,741 in <code class="language-plaintext highlighter-rouge">GSM461177</code> and 127,416 in <code class="language-plaintext highlighter-rouge">GSM461180</code>).</p>

    <p>Comparing different output files is easier if we can view more than one dataset simultaneously. The Scratchbook function allows us to build up a collection of datasets that will be shown on the screen together.</p>

    <blockquote class="notranslate hands_on">
      <h3 id="hands_on-optional-view-the-sorted-featurecounts-using-the-scratchbook"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> (Optional) View the sorted featureCounts using the Scratchbook</h3>

      <ol>
        <li>
          <p>The <strong>Scratchbook</strong> is enabled by clicking the nine-blocks icon seen on the right of the <span class="notranslate">Galaxy</span> top menubar:</p>

          <figure id="figure-24"><img src="../../images/ref-based/menubarWithScratchbook.png" alt="scratchbook icon. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 24:</span> Menu bar with Scratchbook icon</figcaption></figure>
        </li>
        <li>
          <p>When the Scratchbook is <strong>enabled</strong> datasets being viewed (by clicking the eye-icon) are added to the Scratchbook view:</p>

          <figure id="figure-25"><img src="../../images/ref-based/menubarWithScratchbookEnabled.png" alt="Scratchbook icon enabled. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 25:</span> Menu bar with Scratchbook icon enabled</figcaption></figure>
        </li>
        <li>
          <p>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon to view one of the <strong>sorted featureCounts</strong> files. Instead of occupying the entire middle bar the dataset view is now shown an overlay:</p>

          <figure id="figure-26"><img src="../../images/ref-based/scratchbookOneDataset.png" alt="Scratchbook one dataset shown. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 26:</span> Scratchbook showing one dataset overlay</figcaption></figure>
        </li>
        <li>
          <p>When a dataset is shown <strong>click in the main interface</strong> to prepare to select another dataset. The interface now shows that there is one saved view in the Scratchbook:</p>

          <figure id="figure-27"><img src="../../images/ref-based/scratchbookOneSavedView.png" alt="scratchbook one saved view. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 27:</span> Menu bar with one saved dataset view in Scratchbook</figcaption></figure>
        </li>
        <li>
          <p>Next click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon on the <strong>second sorted featureCounts</strong> file. The two datasets can now be seen side by side:</p>

          <figure id="figure-28"><img src="../../images/ref-based/scratchbookTwoDatasetsShown.png" alt="Scratchbook two datasets shown. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 28:</span> Scratchbook showing two side by side datasets</figcaption></figure>
        </li>
        <li>
          <p>To <strong>leave</strong> Scratchbook selection mode, click on the <strong>Scratchbook icon</strong> again. Your saved view will still remain for future viewing:</p>

          <figure id="figure-29"><img src="../../images/ref-based/scratchbookDisabledWithSavedDatasets.png" alt="Scratchbook disabled datasets saved. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 29:</span> Scratchbook disabled with two datasets saved</figcaption></figure>
        </li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<p>Here we counted <span class="notranslate">reads</span> mapped to genes for two samples. It is really interesting to redo the same procedure on the other datasets, especially to check how parameters differ given the different type of data (single-end versus paired-end).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-re-run-on-the-other-datasets"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> (Optional) Hands-on: Re-run on the other datasets</h3>

  <p>You can do the same process on the other sequence files available on <a href="https://zenodo.org/record/4541751">Zenodo</a> and on the data library.</p>

  <ul>
    <li>Paired-end data
      <ul>
        <li><code class="language-plaintext highlighter-rouge">GSM461178_1</code> and <code class="language-plaintext highlighter-rouge">GSM461178_2</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461181_1</code> and <code class="language-plaintext highlighter-rouge">GSM461181_2</code></li>
      </ul>
    </li>
    <li>Single-end data
      <ul>
        <li><code class="language-plaintext highlighter-rouge">GSM461176</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461179</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461182</code></li>
      </ul>
    </li>
  </ul>

  <p>The links to these files are below:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/GSM461178_1.fastqsanger
https://zenodo.org/record/4541751/files/GSM461178_2.fastqsanger
https://zenodo.org/record/4541751/files/GSM461181_1.fastqsanger
https://zenodo.org/record/4541751/files/GSM461181_2.fastqsanger
https://zenodo.org/record/4541751/files/GSM461176.fastqsanger
https://zenodo.org/record/4541751/files/GSM461179.fastqsanger
https://zenodo.org/record/4541751/files/GSM461182.fastqsanger
</code></pre></div>  </div>

</blockquote>

<h1 id="analysis-of-the-differential-gene-expression">Analysis of the differential gene <span class="notranslate">expression</span></h1>

<h2 id="identification-of-the-differentially-expressed-features">Identification of the differentially expressed features</h2>

<p>To be able to identify differential gene <span class="notranslate">expression</span> induced by PS depletion, all datasets (3 treated and 4 untreated) must be analyzed following the same procedure. To save time, we have run the previous steps for you. We then obtain 7 files with the counts for each gene of <em>Drosophila</em> for each sample.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-import-all-count-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Import all count files</h3>

  <ol>
    <li>Create a new empty history</li>
    <li>
      <p>Import the seven count files from <a href="https://zenodo.org/record/4541751">Zenodo</a> or the Shared Data library:</p>

      <ul>
        <li><code class="language-plaintext highlighter-rouge">GSM461176_untreat_single.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461177_untreat_paired.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461179_treat_single.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461180_treat_paired.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461181_treat_paired.counts</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM461182_untreat_single.counts</code></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/GSM461176_untreat_single.counts
https://zenodo.org/record/4541751/files/GSM461177_untreat_paired.counts
https://zenodo.org/record/4541751/files/GSM461178_untreat_paired.counts
https://zenodo.org/record/4541751/files/GSM461179_treat_single.counts
https://zenodo.org/record/4541751/files/GSM461180_treat_paired.counts
https://zenodo.org/record/4541751/files/GSM461181_treat_paired.counts
https://zenodo.org/record/4541751/files/GSM461182_untreat_single.counts
</code></pre></div>      </div>
    </li>
    <li>As before, rename the datasets according to the name of the sample, e.g. <code class="language-plaintext highlighter-rouge">GSM461182_untreat_single</code>, check the datatype (<code class="language-plaintext highlighter-rouge">tabular</code>) and add a different tag for each sample (e.g. <code class="language-plaintext highlighter-rouge">GSM461182_untreat_single</code>).</li>
  </ol>
</blockquote>

<p>You might think We can just compare the count values in the files directly and calculate the extent of differential gene <span class="notranslate">expression</span>. However, it is not that simple.</p>

<p>Let’s imagine we have <span class="notranslate">RNA</span>-Seq counts from 3 samples for a genome with 4 genes:</p>

<table>
  <thead>
    <tr>
      <th>Gene</th>
      <th> Sample 1 Counts</th>
      <th>Sample 2 Counts</th>
      <th>Sample 3 Counts</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>A (2kb)</td>
      <td>10</td>
      <td>12</td>
      <td>30</td>
    </tr>
    <tr>
      <td>B (4kb)</td>
      <td>20</td>
      <td>25</td>
      <td>60</td>
    </tr>
    <tr>
      <td>C (1kb)</td>
      <td>5</td>
      <td>8</td>
      <td>15</td>
    </tr>
    <tr>
      <td>D (10kb)</td>
      <td>0</td>
      <td>0</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

<p>Sample 3 has more <span class="notranslate">reads</span> than the other replicates, regardless of the gene. It has a higher sequencing depth than the other replicates. Gene B is twice as long as gene A: it might explain why it has twice as many <span class="notranslate">reads</span>, regardless of replicates.</p>

<p>The number of sequenced <span class="notranslate">reads</span> mapped to a gene therefore depends on:</p>

<ul>
  <li>
    <p>The <strong>sequencing depth</strong> of the samples</p>

    <p>Samples sequenced with more depth will have more <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to each genes</p>
  </li>
  <li>
    <p>The <strong>length of the gene</strong></p>

    <p>Longer genes will have more <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to them</p>
  </li>
</ul>

<p>To compare samples or gene <span class="notranslate">expression</span>s, the gene counts need to be normalized. We could use TPM (Transcripts Per Kilobase Million).</p>

<blockquote class="details">
  <h3 id="details-rpkm-fpkm-and-tpm"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> RPKM, FPKM and TPM?</h3>

  <p>These three metrics are used to normalize count tables for:</p>
  <ul>
    <li>sequencing depth (the “Million” part)</li>
    <li>gene length (the “Kilobase” part)</li>
  </ul>

  <p>Let’s use the previous example to explain RPK, FPKM and TPM.</p>

  <p>For <strong>RPKM</strong> (<span class="notranslate">Reads</span> Per Kilobase Million),</p>
  <ol>
    <li>
      <p>Compute the “per million” scaling factor: sum up the total <span class="notranslate">reads</span> in a sample and divide that number by 1,000,000.</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 Counts</th>
            <th>Sample 2 Counts</th>
            <th>Sample 3 Counts</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>10</td>
            <td>12</td>
            <td>30</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>20</td>
            <td>25</td>
            <td>60</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>5</td>
            <td>8</td>
            <td>15</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>1</td>
          </tr>
          <tr>
            <td><strong>Total <span class="notranslate">reads</span></strong></td>
            <td>35</td>
            <td>45</td>
            <td>106</td>
          </tr>
          <tr>
            <td><strong>Scaling factor</strong></td>
            <td>3.5</td>
            <td>4.5</td>
            <td>10.6</td>
          </tr>
        </tbody>
      </table>

      <p><em>Because of the small values in the example, we are scoring using a factor of 10.</em> <!-- confusing --></p>
    </li>
    <li>
      <p>Divide the read counts by the “per million” scaling factor</p>

      <p>This normalizes for sequencing depth, giving <span class="notranslate">reads</span> per million (RPM)</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 RPM</th>
            <th>Sample 2 RPM</th>
            <th>Sample 3 RPM</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>2.86</td>
            <td>2.67</td>
            <td>2.83</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>5.71</td>
            <td>5.56</td>
            <td>5.66</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>1.43</td>
            <td>1.78</td>
            <td>1.43</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>0.09</td>
          </tr>
        </tbody>
      </table>
    </li>
    <li>
      <p>Divide the RPM values by the length of the gene, in kilobases.</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 RPKM</th>
            <th>Sample 2 RPKM</th>
            <th>Sample 3 RPKM</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>1.43</td>
            <td>1.33</td>
            <td>1.42</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>1.43</td>
            <td>1.39</td>
            <td>1.42</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>1.43</td>
            <td>1.78</td>
            <td>1.42</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>0.009</td>
          </tr>
        </tbody>
      </table>
    </li>
  </ol>

  <p><strong>FPKM</strong> (Fragments Per Kilobase Million) is very similar to RPKM. RPKM is used for single-end <span class="notranslate">RNA</span>-seq, while FPKM is used for paired-end <span class="notranslate">RNA</span>-seq. With single-end, every read corresponds to a single fragment that was sequenced. With paired-end <span class="notranslate">RNA</span>-seq, two <span class="notranslate">reads</span> of a pair are mapped from a single fragment, or if one read in the pair did not map, one read can correspond to a single fragment (in case we decided to keep these). FPKM keeps tracks of fragments so that one fragment with 2 <span class="notranslate">reads</span> is counted only once.</p>

  <p><strong>TPM</strong> (Transcripts Per Kilobase Million) is very similar to RPKM and FPKM, except the order of the operation</p>

  <ol>
    <li>
      <p>Divide the read counts by the length of each gene in kilobases</p>

      <p>This gives the <span class="notranslate">reads</span> per kilobase (RPK).</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 RPK</th>
            <th>Sample 2 RPK</th>
            <th>Sample 3 RPK</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>5</td>
            <td>6</td>
            <td>15</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>5</td>
            <td>6.25</td>
            <td>15</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>5</td>
            <td>8</td>
            <td>15</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>0.1</td>
          </tr>
        </tbody>
      </table>
    </li>
    <li>
      <p>Compute the “per million” scaling factor: sum up all the RPK values in a sample and divide this number by 1,000,000</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 RPK</th>
            <th>Sample 2 RPK</th>
            <th>Sample 3 RPK</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>5</td>
            <td>6</td>
            <td>15</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>5</td>
            <td>6.25</td>
            <td>15</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>5</td>
            <td>8</td>
            <td>15</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>0.1</td>
          </tr>
          <tr>
            <td><strong>Total RPK</strong></td>
            <td>15</td>
            <td>20.25</td>
            <td>45.1</td>
          </tr>
          <tr>
            <td><strong>Scaling factor</strong></td>
            <td>1.5</td>
            <td>2.03</td>
            <td>4.51</td>
          </tr>
        </tbody>
      </table>

      <p><em>As above, because of the small values in the example, we are scoring using a factor of 10.</em></p>
    </li>
    <li>
      <p>Divide the RPK values by the “per million” scaling factor</p>

      <table>
        <thead>
          <tr>
            <th>Gene</th>
            <th> Sample 1 TPM</th>
            <th>Sample 2 TPM</th>
            <th>Sample 3 TPM</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>A (2kb)</td>
            <td>3.33</td>
            <td>2.96</td>
            <td>3.33</td>
          </tr>
          <tr>
            <td>B (4kb)</td>
            <td>3.33</td>
            <td>3.09</td>
            <td>3.33</td>
          </tr>
          <tr>
            <td>C (1kb)</td>
            <td>3.33</td>
            <td>3.95</td>
            <td>3.33</td>
          </tr>
          <tr>
            <td>D (10kb)</td>
            <td>0</td>
            <td>0</td>
            <td>0.1</td>
          </tr>
        </tbody>
      </table>
    </li>
  </ol>

  <p>Unlike RPKM and FPKM, when calculating TPM, we normalize for gene length first, and then normalize for sequencing depth second. However, the effects of this difference are quite profound, as we already saw with the example.</p>

  <p>The sums of each column are very different:</p>

  <table>
    <thead>
      <tr>
        <th>Gene</th>
        <th> Sample 1 RPKM</th>
        <th>Sample 2 RPKM</th>
        <th>Sample 3 RPKM</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>A (2kb)</td>
        <td>1.43</td>
        <td>1.33</td>
        <td>1.42</td>
      </tr>
      <tr>
        <td>B (4kb)</td>
        <td>1.43</td>
        <td>1.39</td>
        <td>1.42</td>
      </tr>
      <tr>
        <td>C (1kb)</td>
        <td>1.43</td>
        <td>1.78</td>
        <td>1.42</td>
      </tr>
      <tr>
        <td>D (10kb)</td>
        <td>0</td>
        <td>0</td>
        <td>0.009</td>
      </tr>
      <tr>
        <td><strong>Total</strong></td>
        <td>4.29</td>
        <td>4.5</td>
        <td>4.25</td>
      </tr>
    </tbody>
  </table>

  <table>
    <thead>
      <tr>
        <th>Gene</th>
        <th> Sample 1 TPM</th>
        <th>Sample 2 TPM</th>
        <th>Sample 3 TPM</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>A (2kb)</td>
        <td>3.33</td>
        <td>2.96</td>
        <td>3.33</td>
      </tr>
      <tr>
        <td>B (4kb)</td>
        <td>3.33</td>
        <td>3.09</td>
        <td>3.33</td>
      </tr>
      <tr>
        <td>C (1kb)</td>
        <td>3.33</td>
        <td>3.95</td>
        <td>3.33</td>
      </tr>
      <tr>
        <td>D (10kb)</td>
        <td>0</td>
        <td>0</td>
        <td>0.1</td>
      </tr>
      <tr>
        <td><strong>Total</strong></td>
        <td>10</td>
        <td>10</td>
        <td>10</td>
      </tr>
    </tbody>
  </table>

  <p>The sum of all TPMs in each sample are the same. This makes it easier to compare the proportion of <span class="notranslate">reads</span> that mapped to a gene in each sample. In contrast, with RPKM and FPKM, the sum of the normalized <span class="notranslate">reads</span> in each sample may be different, and this makes it harder to compare samples directly.</p>

  <p>In the example, TPM for gene A in Sample 1 is 3.33 and in sample 2 is 3.33. The same proportion of total <span class="notranslate">reads</span> maps then to gene A in both samples (0.33 here). Indeed, the sum of the TPMs in both samples adds up to the same number (10 here), the denominator required to calculate the proportions is then the same regardless of the sample, and so the proportion of <span class="notranslate">reads</span> for gene A (3.33/10 = 0.33) for both sample.</p>

  <p>With RPKM or FPKM, it is harder to compare the proportion of total <span class="notranslate">reads</span> because the sum of normalized <span class="notranslate">reads</span> in each sample can be different (4.29 for Sample 1 and 4.25 for Sample 2). Thus, if RPKM for gene A in Sample 1 is 1.43 and in Sample B is 1.43, we do not know if the same proportion of <span class="notranslate">reads</span> in Sample 1 mapped to gene A as in Sample 2.</p>

  <p>Since <span class="notranslate">RNA</span>-Seq is all about comparing relative proportion of <span class="notranslate">reads</span>, TPM seems more appropriate than RPKM/FPKM.</p>
</blockquote>

<p><span class="notranslate">RNA</span>-Seq is often used to compare one tissue type to another, for example, muscle vs. epithelial tissue. And it could be that there are a lot of muscle specific genes transcribed in muscle but not in the epithelial tissue. We call this a <strong>difference in library composition</strong>.</p>

<p>It is also possible to see a difference in library composition in the same tissue type after the knock out of a transcription factor.</p>

<p>Let’s imagine we have <span class="notranslate">RNA</span>-Seq counts from 2 samples (same library size: 635 <span class="notranslate">reads</span>), for a genome with 6 genes. The genes have the same <span class="notranslate">expression</span> in both samples, except one: only Sample 1 transcribes gene D, at a high level (563 <span class="notranslate">reads</span>). As the library size is the same for both samples, sample 2 has 563 extra <span class="notranslate">reads</span> to be distributed over genes A, B, C, E and F.</p>

<table>
  <thead>
    <tr>
      <th>Gene</th>
      <th> Sample 1</th>
      <th>Sample 2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>A</td>
      <td>30</td>
      <td>235</td>
    </tr>
    <tr>
      <td>B</td>
      <td>24</td>
      <td>188</td>
    </tr>
    <tr>
      <td>C</td>
      <td>0</td>
      <td>0</td>
    </tr>
    <tr>
      <td>D</td>
      <td>563</td>
      <td>0</td>
    </tr>
    <tr>
      <td>E</td>
      <td>5</td>
      <td>39</td>
    </tr>
    <tr>
      <td>F</td>
      <td>13</td>
      <td>102</td>
    </tr>
    <tr>
      <td><strong>Total</strong></td>
      <td>635</td>
      <td>635</td>
    </tr>
  </tbody>
</table>

<p>As a result, the read count for all genes except for genes C and D is really high in Sample 2. Nonetheless, the only differentially expressed gene is gene D.</p>

<p>TPM, RPKM or FPKM do not deal with these differences in library composition during normalization, but more complex tools, like DESeq2, do.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong></a> (<span class="citation"><a href="#love2014moderated">Love <i>et al.</i> 2014</a></span>) is a great tool for dealing with <span class="notranslate">RNA</span>-seq data and running Differential Gene <span class="notranslate">Expression</span> (DGE) analysis. It takes read count files from different samples, combines them into a big table (with genes in the rows and samples in the columns) and applies normalization for <strong>sequencing depth</strong> and <strong>library composition</strong>. We do not need to account for gene length normalization does because we are comparing the counts between sample groups for the same gene.</p>

<blockquote class="details">
  <h3 id="details-normalization-in-deseq2"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Normalization in DESeq2</h3>

  <p>Let’s take an example to illustrate how DESeq2 scales the different samples:</p>

  <table>
    <tbody>
      <tr>
        <td>Gene</td>
        <td>Sample 1</td>
        <td>Sample 2</td>
        <td>Sample 3</td>
      </tr>
      <tr>
        <td>A</td>
        <td>0</td>
        <td>10</td>
        <td>4</td>
      </tr>
      <tr>
        <td>B</td>
        <td>2</td>
        <td>6</td>
        <td>12</td>
      </tr>
      <tr>
        <td>C</td>
        <td>33</td>
        <td>55</td>
        <td>200</td>
      </tr>
    </tbody>
  </table>

  <p>The goal is to calculate a scaling factor for each sample, which takes read depth and library composition into account.</p>

  <ol>
    <li>
      <p>Take the log\(_e\) of all the values:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>log(Sample 1)</td>
            <td>log(Sample 2)</td>
            <td>log(Sample 3)</td>
          </tr>
          <tr>
            <td>A</td>
            <td>-Inf</td>
            <td>2.3</td>
            <td>1.4</td>
          </tr>
          <tr>
            <td>B</td>
            <td>0.7</td>
            <td>1.8</td>
            <td>2.5</td>
          </tr>
          <tr>
            <td>C</td>
            <td>3.5</td>
            <td>4.0</td>
            <td>5.3</td>
          </tr>
        </tbody>
      </table>
    </li>
    <li>
      <p>Average each row:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>Average of log values</td>
          </tr>
          <tr>
            <td>A</td>
            <td>-Inf</td>
          </tr>
          <tr>
            <td>B</td>
            <td>1.7</td>
          </tr>
          <tr>
            <td>C</td>
            <td>4.3</td>
          </tr>
        </tbody>
      </table>

      <p>The average of the log values (also known as the geometric average) is used here because it is not easily impacted by outliers (e.g. gene C with its outlier for Sample 3).</p>
    </li>
    <li>
      <p>Filter out genes with which have infinity as a value.</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>Average of log values</td>
          </tr>
          <tr>
            <td> </td>
            <td> </td>
          </tr>
          <tr>
            <td>B</td>
            <td>1.7</td>
          </tr>
          <tr>
            <td>C</td>
            <td>4.3</td>
          </tr>
        </tbody>
      </table>

      <p>Here we filter out genes with no read counts in at least 1 sample, e.g. genes only transcribed in one tissue like gene D in the previous example. This helps to focus the scaling factors on genes transcribed at similar levels, regardless of the condition.</p>
    </li>
    <li>
      <p>Subtract the average log value from the log counts:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>log(Sample 1)</td>
            <td>log(Sample 2)</td>
            <td>log(Sample 3)</td>
          </tr>
          <tr>
            <td> </td>
            <td> </td>
            <td> </td>
            <td> </td>
          </tr>
          <tr>
            <td>B</td>
            <td>-1.0</td>
            <td>0.1</td>
            <td>0.8</td>
          </tr>
          <tr>
            <td>C</td>
            <td>-0.8</td>
            <td>-0.3</td>
            <td>1.0</td>
          </tr>
        </tbody>
      </table>

\[log(\textrm{counts for gene X}) - average(\textrm{log values for counts for gene X}) = log(\frac{\textrm{counts for gene X}}{\textrm{average for gene X}})\]

      <p>This step compares the ratio of the counts in each sample to the average across all samples.</p>
    </li>
    <li>
      <p>Calculate the median of the ratios for each sample:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>log(Sample 1)</td>
            <td>log(Sample 2)</td>
            <td>log(Sample 3)</td>
          </tr>
          <tr>
            <td> </td>
            <td> </td>
            <td> </td>
            <td> </td>
          </tr>
          <tr>
            <td>B</td>
            <td>-1.0</td>
            <td>0.1</td>
            <td>0.8</td>
          </tr>
          <tr>
            <td>C</td>
            <td>-0.8</td>
            <td>-0.3</td>
            <td>1.0</td>
          </tr>
          <tr>
            <td><strong>Median</strong></td>
            <td> -0.9</td>
            <td>-0.1</td>
            <td>0.9</td>
          </tr>
        </tbody>
      </table>

      <p>The median is used here to avoid extreme genes (most likely rare ones) from swaying the value too much in one direction. It helps to put more emphasis on moderately expressed genes.</p>
    </li>
    <li>
      <p>Compute the scaling factor by taking the exponential of the medians:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>Sample 1</td>
            <td>Sample 2</td>
            <td>Sample 3</td>
          </tr>
          <tr>
            <td><strong>Median</strong></td>
            <td> -0.9</td>
            <td>-0.1</td>
            <td>0.9</td>
          </tr>
          <tr>
            <td><strong>Scaling factors</strong></td>
            <td> 0.4</td>
            <td>0.9</td>
            <td>2.5</td>
          </tr>
        </tbody>
      </table>
    </li>
    <li>
      <p>Compute the normalized counts: divide the original counts by the scaling factors:</p>

      <table>
        <tbody>
          <tr>
            <td>Gene</td>
            <td>Sample 1</td>
            <td>Sample 2</td>
            <td>Sample 3</td>
          </tr>
          <tr>
            <td>A</td>
            <td>0</td>
            <td>11.11</td>
            <td>1.6</td>
          </tr>
          <tr>
            <td>B</td>
            <td>5</td>
            <td>6.67</td>
            <td>4.8</td>
          </tr>
          <tr>
            <td>C</td>
            <td>83</td>
            <td>61.11</td>
            <td>80</td>
          </tr>
        </tbody>
      </table>
    </li>
  </ol>

  <p><em>This explanation is a transcription and adaptation of the <a href="https://www.youtube.com/watch?v=UFB993xufUU&amp;t=35s">StatQuest video explaining Library Normalization in DESEq2</a></em>.</p>

</blockquote>

<p>DESeq2 also runs the Differential Gene <span class="notranslate">Expression</span> (DGE) analysis, which has two basic tasks:</p>

<ul>
  <li>Estimate the biological variance using the replicates for each condition</li>
  <li>Estimate the significance of <span class="notranslate">expression</span> differences between any two conditions</li>
</ul>

<p>This <span class="notranslate">expression</span> analysis is estimated from read counts and attempts are made to correct for variability in measurements using replicates, that are absolutely essential for accurate results. For your own analysis, we advise you to use at least 3, but preferably 5 biological replicates per condition. It is possible to have different numbers of replicates per condition.</p>

<blockquote class="details">
  <h3 id="details-technical-vs-biological-replicates"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Technical vs biological replicates</h3>

  <p>A technical replicate is an experiment which is performed once but measured several times (e.g. multiple sequencing of the same library). A biological replicate is an experiment performed (and also measured) several times.</p>

  <p>In our data, we have 4 biological replicates (here called samples) without treatment and 3 biological replicates with treatment (<em>Pasilla</em> gene depleted by <span class="notranslate">RNA</span>i).</p>

  <p>We recommend to combine the count tables for different technical replicates (but not for biological replicates) before a differential <span class="notranslate">expression</span> analysis (see <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#collapsing-technical-replicates">DESeq2 documentation</a>)</p>
</blockquote>

<p>Multiple factors with several levels can then be incorporated in the analysis describing known sources of variation (e.g. treatment, tissue type, gender, batches), with two or more levels representing the conditions for each factor. After normalization we can compare the response of the <span class="notranslate">expression</span> of any gene to the presence of different levels of a factor in a statistically reliable way.</p>

<p>In our example, we have samples with two varying factors that can contribute to differences in gene <span class="notranslate">expression</span>:</p>

<ul>
  <li>Treatment (either treated or untreated)</li>
  <li>Sequencing type (paired-end or single-end)</li>
</ul>

<p>Here, treatment is the primary factor that we are interested in. The sequencing type is further information we know about the data that might affect the analysis. Multi-factor analysis allows us to assess the effect of the treatment, while taking the sequencing type into account too.</p>

<blockquote class="comment">
  <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>We recommend that you add all factors you think may affect gene <span class="notranslate">expression</span> in your experiment. It can be the sequencing type like here, but it can also be the manipulation (if different persons are involved in the library preparation), other batch effects, etc…</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-determine-differentially-expressed-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Determine differentially expressed features</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.6+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.6+galaxy1"><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.6+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>“how”</em>: <code class="language-plaintext highlighter-rouge">Select datasets per level</code>
          <ul>
            <li>In “1: Factor”
              <ul>
                <li><em>“Specify a factor name”</em>: <code class="language-plaintext highlighter-rouge">Treatment</code></li>
                <li>In <em>“1: Factor level”</em>:
                  <ul>
                    <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">treated</code></li>
                    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: the 3 gene count files with <code class="language-plaintext highlighter-rouge">treat</code> in their name</li>
                  </ul>
                </li>
                <li>In <em>“2: Factor level”</em>:
                  <ul>
                    <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">untreated</code></li>
                    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: the 4 gene count files with <code class="language-plaintext highlighter-rouge">untreat</code> in their name</li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor”</em> (not on “Insert Factor level”)</li>
            <li>In “2: Factor”
              <ul>
                <li>“Specify a factor name” to <code class="language-plaintext highlighter-rouge">Sequencing</code></li>
                <li>In <em>“1: Factor level”</em>:
                  <ul>
                    <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">PE</code></li>
                    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: the 4 gene count files with <code class="language-plaintext highlighter-rouge">paired</code> in their name</li>
                  </ul>
                </li>
                <li>In <em>“2: Factor level”</em>:
                  <ul>
                    <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">SE</code></li>
                    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: the 3 gene count files with <code class="language-plaintext highlighter-rouge">single</code> in their name</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Files have header?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Visualising the analysis results”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Output normalized counts table”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-using-group-tags-for-large-sample-sets"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Using group tags for large sample sets</h3>

        <p>If you have a large number of samples, or a complex experimental design, manually selecting files for each factor level
may be a lot of work. In these situations, <strong>group tags</strong> can make your life a lot easier.</p>

        <p>For more information about setting and using group tags, please see <a href="/training-material/topics/galaxy-interface/tutorials/group-tags/tutorial.html">this tutorial</a>.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p><strong>DESeq2</strong> generated 3 outputs:</p>

<ul>
  <li>A table with the normalized counts for each gene (rows) in the samples (columns)</li>
  <li>
    <p>A graphical summary of the results, useful to evaluate the quality of the experiment:</p>

    <ol>
      <li>
        <p>A plot of the first 2 dimensions from a principal component analysis (<a href="https://en.wikipedia.org/wiki/Principal_component_analysis">PCA</a>), run on the normalized counts of the samples</p>

        <blockquote class="details">
          <h3 id="details-what-is-a-pca"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> What is a PCA?</h3>

          <p>Let’s imagine we have some beer bottles standing here on the table. We can describe each beer by its colour, its foam, by how strong it is, and so on. We can compose a whole list of different characteristics of each beer in a beer shop. But many of them will measure related properties and so will be redundant. If so, we should be able to summarize each beer with fewer characteristics. This is what PCA or principal component analysis does.</p>

          <p>With PCA, we do not just select some interesting characteristics and discard the others. Instead, we construct some new characteristics that summarize our list of beers well. These new characteristics are constructed using the old ones. For example, a new characteristic might be computed, e.g. foam size minus beer pH. They are linear combinations.</p>

          <p>In fact, PCA finds the best possible characteristics, the ones that summarize the list of beers. These characteristics can be used to find similarities between beers and group them.</p>

          <p>Going back to read counts, the PCA is run on the normalized counts for all the samples. Here, we would like to describe the samples based on the <span class="notranslate">expression</span> of the genes. So the characteristics are the number of <span class="notranslate">reads</span> mapped on each genes. We use them and linear combinations of them to represent the samples and their similarities.</p>

          <p><em>The beer analogy has been adapted from <a href="https://stats.stackexchange.com/questions/2691/making-sense-of-principal-component-analysis-eigenvectors-eigenvalues">an answer on StackExchange</a></em>.</p>

        </blockquote>

        <p>It shows the samples in the 2D plane spanned by their first two principal components. Each replicate is plotted as an individual data point. This type of plot is useful for visualizing the overall effect of experimental covariates and batch effects.</p>

        <blockquote class="question">
          <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <figure id="figure-30"><img src="../../images/ref-based/deseq2_pca.png" alt="DESeq PCA. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 30:</span> Principal component plot of the samples</figcaption></figure>

          <ol>
            <li>What is the first dimension (PC1) separating?</li>
            <li>And the second dimension (PC2)?</li>
            <li>What can we conclude about the DESeq design (factors, levels) we choose?</li>
          </ol>

          <blockquote class="solution">
            <h3 id="solution-solution-19"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <ol>
              <li>The first dimension is separating the treated samples from the untreated sample.</li>
              <li>The second dimension is separating the single-end datasets from the paired-end datasets.</li>
              <li>The datasets are grouped following the levels of the two factors. No hidden effect seems to be present on the data. If there is unwanted variation present in the data (e.g. batch effects), it is always recommended to correct for this, which can be achieved in DESeq2 by including in the design any known batch variables.</li>
            </ol>
          </blockquote>
        </blockquote>
      </li>
      <li>
        <p>Heatmap of the sample-to-sample distance matrix (with clustering) based on the normalized counts.</p>

        <p>The heatmap gives an overview of similarities and dissimilarities between samples: the color represents the distance between the samples. Dark blue means shorter distance, i.e. closer samples given the normalized counts.</p>

        <blockquote class="question">
          <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <figure id="figure-31"><img src="../../images/ref-based/deseq2_sample_sample_distance_heatmap.png" alt="Heatmap of the sample-to-sample distances. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 31:</span> Heatmap of the sample-to-sample distances</figcaption></figure>

          <p>How are the samples grouped?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-20"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
            <p>They are first grouped by the treatment (the first factor) and secondly by the sequencing type (the second factor), as in the PCA plot.</p>

          </blockquote>
        </blockquote>
      </li>
      <li>
        <p>Dispersion estimates: gene-wise estimates (black), the fitted values (red), and the final maximum a posteriori estimates used in testing (blue)</p>

        <p>This dispersion plot is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value. The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.</p>
      </li>
      <li>
        <p>Histogram of <em>p</em>-values for the genes in the comparison between the 2 levels of the 1st factor</p>
      </li>
      <li>
        <p>An <a href="https://en.wikipedia.org/wiki/MA_plot">MA plot</a>:</p>

        <p>This displays the global view of the relationship between the <span class="notranslate">expression</span> change of conditions (log ratios, M), the average <span class="notranslate">expression</span> strength of the genes (average mean, A), and the ability of the algorithm to detect differential gene <span class="notranslate">expression</span>. The genes that passed the significance threshold (adjusted p-value &lt; 0.1) are colored in red.</p>
      </li>
    </ol>
  </li>
  <li>
    <p>A summary file with the following values for each gene:</p>

    <ol>
      <li>Gene identifiers</li>
      <li>Mean normalized counts, averaged over all samples from both conditions</li>
      <li>
        <p>Fold change in log2 (logarithm base 2)</p>

        <p>The log2 fold changes are based on the primary factor level 1 vs factor level 2, hence the input order of factor levels is important. Here, DESeq2 computes fold changes of ‘treated’ samples against ‘untreated’ from the first factor ‘Treatment’, <em>i.e.</em> the values correspond to up- or downregulation of genes in treated samples.</p>
      </li>
      <li>Standard error estimate for the log2 fold change estimate</li>
      <li><a href="https://en.wikipedia.org/wiki/Wald_test">Wald</a> statistic</li>
      <li><em>p</em>-value for the statistical significance of this change</li>
      <li><em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure, which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>)</li>
    </ol>
  </li>
</ul>

<p>For more information about <strong>DESeq2</strong> and its outputs, you can have a look at the <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><strong>DESeq2</strong> documentation</a>.</p>

<blockquote class="question">
  <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Is the FBgn0003360 gene differentially expressed because of the treatment? If yes, how much?</li>
    <li>We could also hypothetically be interested in the effect of the sequencing (or other secondary factors in other cases). How would we know the differentially expressed genes because of sequencing type?</li>
    <li>We would like to analyze the interaction between the treatment and the sequencing.</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-21"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>FBgn0003360 is differentially expressed because of the treatment: it has a significant adjusted p-value (\(4.0 \cdot 10^{-178} &lt;&lt; 0.05\)). It is less expressed (<code class="language-plaintext highlighter-rouge">-</code> in the log2FC column) in treated samples compared to untreated samples, by a factor ~8 (\(2^{log2FC} = 2^{2.99977727873544}\)).</li>
      <li>DESeq2 in <span class="notranslate">Galaxy</span> returns the comparison between the different levels for the 1st factor, after
correction for the variability due to the 2nd factor. In our current case, treated against untreated for any sequencing type. To compare sequencing types, we should run DESeq2 again switching factors: factor 1 (treatment) becomes factor 2 and factor 2 (sequencing) becomes factor 1.</li>
      <li>To add the interaction between two factors (e.g. treated for paired-end data vs untreated for single-end), we should run DESeq2 another time but with only one factor with the following 4 levels:
        <ul>
          <li>treated-PE</li>
          <li>untreated-PE</li>
          <li>treated-SE</li>
          <li>untreated-SE</li>
        </ul>

        <p>By selection <em>“Output all levels vs all levels of primary factor (use when you have &gt;2 levels for primary factor)”</em> to <code class="language-plaintext highlighter-rouge">Yes</code>, we can then compare treated-PE vs untreated-SE.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-what-are-p-values-and-what-are-they-used-for"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: What are p-values and what are they used for?</h3>
  <p>The p-value is a measure often used to determine whether or not a particular observation possesses statistical significance. Strictly speaking, the p-value is the probability that the data could have arisen randomly, assuming that the null hypothesis is correct. In the concrete case of <span class="notranslate">RNA</span>-Seq, the null hypothesis is that there is no differential gene <span class="notranslate">expression</span>. So a p-value of 0.13 for a particular gene indicates that, for that gene, assuming it is not differentially expressed, there is a 13% chance that any apparent differential <span class="notranslate">expression</span> could simply be produced by random variation in the experimental data.</p>

  <p>13% is still quite high, so we cannot really be confident differential gene <span class="notranslate">expression</span> is taking place. The most common way that scientists use p-values is to set a threshold (commonly 0.05, sometimes other values such as 0.01) and reject the null hypothesis only for p-values below this value. Thus, for genes with p-values less than 0.05, we can feel safe stating that differential gene <span class="notranslate">expression</span> plays a role. It should be noted that any such threshold is arbitrary and there is no meaningful difference between a p-value of 0.049 and 0.051, even if we only reject the null hypothesis in the first case.</p>

  <p>Unfortunately, p-values are often heavily misused in scientific research, enough so that Wikipedia provides a <a href="https://en.wikipedia.org/wiki/Misuse_of_p-values">dedicated article</a> on the subject. See also <a href="https://fivethirtyeight.com/features/not-even-scientists-can-easily-explain-p-values/">this article</a> (aimed at a general, non-scientific audience).</p>
</blockquote>

<h2 id="extraction-and-annotation-of-differentially-expressed-genes">Extraction and annotation of differentially expressed genes</h2>

<p>Now we would like to extract the most differentially expressed genes due to the treatment and with an absolute fold change &gt; 2.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-the-most-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract the most differentially expressed genes</h3>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter data on any column using simple <span class="notranslate">expression</span>s</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  to extract genes with a significant change in gene <span class="notranslate">expression</span> (adjusted <em>p</em>-value below 0.05) between treated and untreated samples:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: the <code class="language-plaintext highlighter-rouge">DESeq2 result file</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value</code>.</p>

      <blockquote class="question">
        <h3 id="question-question-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>How many genes have a significant change in gene <span class="notranslate">expression</span> between these conditions?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-22"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>We get 1,091 genes (6.21%) with a significant change in gene <span class="notranslate">expression</span> between treated and untreated samples.</p>

        </blockquote>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>The file with the independently filtered results can be used for further <span class="notranslate">downstream</span> analysis as it excludes genes with only a few read counts, as these genes will not be considered as significantly differentially expressed.</p>
      </blockquote>

      <p>We will now select only the genes with an absolute fold change (FC) &gt; 2. Note that the DESeq2 output file contains \(log_{2} FC\), rather than FC itself, so we filter for \(abs(log_{2} FC) &gt; 1\) (which implies FC &gt; 2).</p>
    </li>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  to extract genes with an \(abs(log_{2} FC) &gt; 1\):
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">abs(c3)&gt;1</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>How many genes have been conserved?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-23"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>130, or 11.92% of the significantly differentially expressed genes.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We now have a table with 130 lines corresponding to the most differentially expressed genes. For each gene, we have its ID, its mean normalized counts (averaged over all samples from both conditions), its \(log_{2} FC\) and other information.</p>

<p>The ID for each gene is something like FBgn0003360, which is an ID from the corresponding database, here Flybase (<span class="citation"><a href="#thurmond2018flybase">Thurmond <i>et al.</i> 2018</a></span>). These IDs are unique but sometimes we prefer to have the gene names, even if they may not reference an unique gene (e.g. duplicated after re-annotation). But gene names may hint already to a function or they help you to search for desired candidates. We would also like to display the location of these genes within the genome. We can extract such information from the annotation file which we used for <span class="notranslate">mapping</span> and counting.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-annotation-of-the-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Annotation of the differentially expressed genes</h3>

  <ol>
    <li>
      <p>Import the Ensembl gene annotation for <em>Drosophila melanogaster</em> (<code class="language-plaintext highlighter-rouge">Drosophila_melanogaster.BDGP6.87.gtf</code>) from the previous history, or from the Shared Data library or from Zenodo:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4541751/files/Drosophila_melanogaster.BDGP6.87.gtf
</code></pre></div>      </div>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/deg_annotate/deg_annotate/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/deg_annotate/deg_annotate/1.1.0"><strong>Annotate DESeq2/DEXSeq output tables</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/deg_annotate/deg_annotate/1.1.0</span></span>  with:</p>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular output of DESeq2/edgeR/limma/DEXSeq”</em>: output of the last <strong>Filter</strong></li>
        <li><em>“Input file type”</em>: <code class="language-plaintext highlighter-rouge">DESeq2/edgeR/limma</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference annotation in GFF/GTF format”</em>: imported GTF file</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The generated output is an extension of the previous file:</p>

<ol>
  <li>Gene identifiers</li>
  <li>Mean normalized counts over all samples</li>
  <li>Log2 fold change</li>
  <li>Standard error estimate for the log2 fold change estimate</li>
  <li>Wald statistic</li>
  <li><em>p</em>-value for the Wald statistic</li>
  <li><em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure for the Wald statistic</li>
  <li>Chromosome</li>
  <li>Start</li>
  <li>End</li>
  <li>Strand</li>
  <li>Feature</li>
  <li>Gene name</li>
</ol>

<blockquote class="question">
  <h3 id="question-questions-12"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Where is the most over-expressed gene located?</li>
    <li>What is the name of the gene?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-24"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>FBgn0025111 (the top-ranked gene with the highest positive log2FC value) is located on the reverse strand of chromosome X, between 10,778,953 bp and 10,786,907 bp.</li>
      <li>From the table, we got the gene symbol: Ant2. After some search on the <a href="https://www.ncbi.nlm.nih.gov/gene/32008">online biological databases</a>, we find that Ant2 corresponds to adenine nucleotide translocase 2.</li>
    </ol>
  </blockquote>
</blockquote>

<p>The annotated table contains no column names, which makes it difficult to read. We would like to add them before going further.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-column-names"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add column names</h3>

  <ol>
    <li>
      <p>Create a new file from the following (header line of the DESeq2 output)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GeneID	Base mean	log2(FC)	StdErr	Wald-Stats	P-value	P-adj	Chromosome	Start	End	Strand	Feature	Gene name
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new file</h3>  <ul>  <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the file contents into the text field* Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code></p>  </li>  <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li></ul></blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0"><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0</span></span>  to add this header line to the <strong>Annotate</strong> output:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: the <code class="language-plaintext highlighter-rouge">Pasted entry</code> dataset</li>
        <li><em>“Dataset”</em>
          <ul>
            <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Dataset”</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“select”</em>: output of <strong>Annotate</strong></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value &amp; abs(FC) &gt; 2</code></li>
  </ol>
</blockquote>

<h2 id="visualization-of-the-expression-of-the-differentially-expressed-genes">Visualization of the <span class="notranslate">expression</span> of the differentially expressed genes</h2>

<p>We could plot the \(log_{2} FC\) for the extracted genes, but here we would like to look at a heatmap of <span class="notranslate">expression</span> for these genes in the different samples. So we need to extract the normalized counts for these genes.</p>

<p>We proceed in several steps:</p>

<ul>
  <li>Extract and plot the normalized counts for these genes for each sample with a heatmap, using the normalized count file generated by DESeq2</li>
  <li>Compute, extract and plot the Z-score of the normalized counts</li>
</ul>

<blockquote class="comment">
  <h3 id="comment-advanced-tutorials-on-visualization"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Advanced tutorials on visualization</h3>

  <p>In this tutorial, we will quickly explain some possible visualization. For more details, please have a look in the extra tutorials on visualization of <span class="notranslate">RNA</span>-Seq results:</p>
  <ul>
    <li><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">Visualization of <span class="notranslate">RNA</span>-Seq results with heatmap2</a></li>
    <li><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html">Visualization of <span class="notranslate">RNA</span>-Seq results with Volcano Plot</a></li>
  </ul>
</blockquote>

<h3 id="visualization-of-the-normalized-counts">Visualization of the normalized counts</h3>

<p>To extract the normalized counts for the interesting genes, we join the normalized count table generated by DESeq2 with the table we just generated. We will then keep only the lines corresponding to the most differentially expressed genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-the-normalized-counts-of-the-most-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract the normalized counts of the most differentially expressed genes</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/join/gops_join_1/1.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/join/gops_join_1/1.0.0"><strong>Join two Datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/join/gops_join_1/1.0.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Join”</em>: the <code class="language-plaintext highlighter-rouge">Normalized counts</code> file (output of <strong>DESeq2</strong>)</li>
        <li><em>“using column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“with”</em>: <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value &amp; abs(FC) &gt; 2</code></li>
        <li><em>“and column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><em>“Keep lines of first input that do not join with second input”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Keep the header lines”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>The generated file has more columns than we need for the heatmap: mean normalized counts, \(log_{2} FC\) and other annotation information. We need to remove the extra columns.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  to extract the columns with the gene IDs and normalized counts:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1-c8</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: the joined dataset (output of <strong>Join two Datasets</strong>)</li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">Normalized counts for the most differentially expressed genes</code>.</li>
  </ol>
</blockquote>

<p>We now have a table with 130 lines (the most differentially expressed genes) and the normalized counts for these genes across the 7 samples.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-heatmap-of-the-normalized-counts-of-these-genes-for-the-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the heatmap of the normalized counts of these genes for the samples</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1"><strong>heatmap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1</span></span>  to plot the heatmap:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input should have column headers”</em>: <code class="language-plaintext highlighter-rouge">Normalized counts for the most differentially expressed genes</code></li>
        <li><em>“Advanced - log transformation”</em>/<em>“Data transformation”</em>: <code class="language-plaintext highlighter-rouge">Log2(value) transform my data</code></li>
        <li><em>“Enable data clustering”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Labeling columns and rows”</em>: <code class="language-plaintext highlighter-rouge">Label columns and not rows</code></li>
        <li><em>“Coloring groups”</em>: <code class="language-plaintext highlighter-rouge">Blue to white to red</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>You should obtain something similar to:</p>

<figure id="figure-32"><img src="../../images/ref-based/heatmap2_normalized_counts.png" alt="Heatmap with the normalized counts for the most differentially expressed genes. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 32:</span> Normalized counts for the most differentially expressed genes</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-13"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What does the X-axis of the heatmap represent? What about the Y axis?</li>
    <li>Do you observe anything in the clustering of the samples and the genes?</li>
    <li>What changes if you regenerate the heatmap, this time selecting <code class="language-plaintext highlighter-rouge">Plot the data as it is</code> in <em>“Advanced - log transformation”</em>/<em>“Data transformation”</em>?</li>
    <li>How could you generate a heatmap of normalized counts for all up-regulated genes with absolute fold change &gt; 2?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-25"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The X-axis shows the 7 samples, together with a dendrogram representing the similarity between their patterns of gene <span class="notranslate">expression</span>. The Y-axis shows the 130 differentially expressed genes, likewise with a dendrogram representing the similarity between the patterns of gene <span class="notranslate">expression</span>. <!-- hope this is correct --></li>
      <li>The samples are clustering by treatment.</li>
      <li>The scale changes and the differences between the genes are not visible anymore.</li>
      <li>Extract the genes with \(log_{2} FC\) &gt; 1 (filter for genes with <code class="language-plaintext highlighter-rouge">c3&gt;1</code> on the summary of the differentially expressed genes) and run <strong>heatmap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the generated table.</li>
    </ol>
  </blockquote>
</blockquote>

<h3 id="computation-and-visualization-of-the-z-score">Computation and visualization of the Z-score</h3>

<p>To compare the gene <span class="notranslate">expression</span> over samples, we could also use the Z-score, which is often represented in publications.</p>

<p>The Z-score gives the number of standard-deviations that a value is away from the mean of all the values in the same group, here the same gene. A Z-score of -2 for the gene X in sample A means that this value is 2 standard-deviations lower than the mean of the values for gene X in all the samples (A, B, C, etc).</p>

<p>The Z-score \(z_{i,j}\) for a gene \(i\) in a sample \(j\) given the normalized count \(x_{i,j}\) is computed as \(z_{i,j} = \frac{x_{i,j}- \overline{x_i}}{s_i}\) with \(\overline{x_i}\) the mean and \(s_i\) the standard deviation of the normalized counts for the gene \(i\) over all samples.</p>

<blockquote class="comment">
  <h3 id="comment-compute-the-z-score-for-all-genes"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Compute the Z-score for all genes</h3>

  <p>To save time in this tutorial, we will compute the Z-score directly on the normalized count of the most differentially expressed genes. In a standard analysis, we would compute the Z-score on all the genes (<code class="language-plaintext highlighter-rouge">Normalized count</code> file from <strong>DESeq</strong>) and then filter for the interesting genes as we did in the previous step.</p>
</blockquote>

<p>To compute the Z-score, we break the process into 2 steps:</p>

<ol>
  <li>Substract each value by the mean of values in the row (i.e. \(x_{i,j}- \overline{x_i}\)) using the normalized count table</li>
  <li>Divide the previous values by the standard deviation of values of row, using 2 tables (the normalized counts and the table computed in the previous step)</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compute-and-extract-the-z-score-of-the-most-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Compute and extract the Z-score of the most differentially expressed genes</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0"><strong>Table Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters to first substract the mean values per row
      <ul>
        <li><em>“Input Single or Multiple Tables”</em>: <code class="language-plaintext highlighter-rouge">Single Table</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Table”</em>: <code class="language-plaintext highlighter-rouge">Normalized counts for the most differentially expressed genes</code></li>
            <li><em>“Type of table operation”</em>: <code class="language-plaintext highlighter-rouge">Perform a full table operation</code>
              <ul>
                <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Custom</code>
                  <ul>
                    <li>
                      <p><em>“Custom <span class="notranslate">expression</span> on ‘table’, along ‘axis’ (0 or 1)”</em>: <code class="language-plaintext highlighter-rouge">table.sub(table.mean(1), 0)</code></p>

                      <p>The <code class="language-plaintext highlighter-rouge">table.mean(1)</code> <span class="notranslate">expression</span> computes the mean for each row (here the genes) and <code class="language-plaintext highlighter-rouge">table.sub(table.mean(1), 0)</code> substracts each value by the mean of the row (computed with <code class="language-plaintext highlighter-rouge">table.mean(1)</code>)</p>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+galaxy0"><strong>Table Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/table_compute/table_compute/1.2.4+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>“Input Single or Multiple Tables”</em>: <code class="language-plaintext highlighter-rouge">Multiple Table</code>
          <ul>
            <li>In <em>“1: Tables”</em>:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Table”</em>: <code class="language-plaintext highlighter-rouge">Normalized counts for the most differentially expressed genes</code></li>
              </ul>
            </li>
            <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Tables”</em></li>
            <li>In <em>“2: Tables”</em>:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Table”</em>: output of the first <strong>Table Compute</strong></li>
              </ul>
            </li>
            <li>
              <p><em>“Custom <span class="notranslate">expression</span> on ‘tableN’“</em>: <code class="language-plaintext highlighter-rouge">table2.div(table1.std(1),0)</code></p>

              <p>The <code class="language-plaintext highlighter-rouge">table1.std(1)</code> <span class="notranslate">expression</span> computes the standard deviations of each row on the 1st table (normalized counts) and <code class="language-plaintext highlighter-rouge">table2.div</code> divides the values of 2nd table (previously computed) by these standard deviations.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">Z-scores for the most differentially expressed genes</code></li>
    <li>Inspect the output file</li>
  </ol>
</blockquote>

<p>We now have a table with the Z-score for the most differentially expressed genes in the 7 samples.</p>

<blockquote class="question">
  <h3 id="question-questions-14"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is the range for the Z-score?</li>
    <li>What can we say about the Z-scores for the differentially expressed genes?</li>
    <li>Can we use the Z-score to estimate the strength of the differential <span class="notranslate">expression</span> of a gene?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-26"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The Z-score ranges from -3 standard deviations up to +3 standard deviations. It can be placed on a normal distribution curve: -3 being the far left of the normal distribution curve and +3 the far right of the normal distribution curve</li>
      <li>When a gene is differentially expressed between two groups (here treated and untreated), the Z-scores for this gene will be (mostly) positive for the samples in one group and (mostly) negative for the samples in the other group.</li>
      <li>The Z-score is a signal-to-noise ratio. Large absolute Z-scores, i.e. large positive or negative values, is not a direct estimate of the effect, i.e. the strength of the differential <span class="notranslate">expression</span>. A same large Z-score can have different meanings, depending on the noise:
        <ul>
          <li>with large noise: a very large effect</li>
          <li>with some noise: a rather large effect</li>
          <li>with only little noise: a rather small effect</li>
          <li>with almost no noise: a tiny effect</li>
        </ul>

        <p>The problem is that “noise” here is not only noise from the measure. It can also be linked to the “tightness” of the gene regulation control. Not tightly controlled genes, i.e. whose <span class="notranslate">expression</span> may vary in a wide range over samples, can be considerably induced or repressed. Their absolute Z-score will be small as the variations over samples is big. In contrast, genes that are tightly controlled may have only very small changes in their <span class="notranslate">expression</span>, without any biological impact. The absolute Z-score will be large for these genes.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>We would like now to plot a heatmap for the Z-scores:</p>

<figure id="figure-33"><img src="../../images/ref-based/z-score-heatmap.png" alt="Heatmap with the Z-score counts for the most differentially expressed genes. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 33:</span> Z-scores for the most differentially expressed genes</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-z-score-of-the-most-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the Z-score of the most differentially expressed genes</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1"><strong>heatmap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap2/ggplot2_heatmap2/3.0.1</span></span>  to plot the heatmap:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input should have column headers”</em>: <code class="language-plaintext highlighter-rouge">Z-scores for the most differentially expressed genes</code></li>
        <li><em>“Advanced - log transformation”</em>/<em>“Data transformation”</em>: <code class="language-plaintext highlighter-rouge">Plot the data as it is</code></li>
        <li><em>“Enable data clustering”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Labeling columns and rows”</em>: <code class="language-plaintext highlighter-rouge">Label columns and not rows</code></li>
        <li><em>“Coloring groups”</em>: <code class="language-plaintext highlighter-rouge">Blue to white to red</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h1 id="functional-enrichment-analysis-of-the-de-genes">Functional enrichment analysis of the DE genes</h1>

<p>We have extracted genes that are differentially expressed in treated (PS gene-depleted) samples compared to untreated samples. Now, we would like to know if the differentially expressed genes are enriched transcripts of genes which belong to more common or specific categories in order to identify biological functions that might be impacted.</p>

<h2 id="gene-ontology-analysis">Gene Ontology analysis</h2>

<p><a href="http://www.geneontology.org/">Gene Ontology (GO)</a> analysis is widely used to reduce complexity and highlight biological processes in genome-wide <span class="notranslate">expression</span> studies. However, standard methods give biased results on <span class="notranslate">RNA</span>-Seq data due to over-detection of differential <span class="notranslate">expression</span> for long and highly-expressed transcripts.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/goseq/inst/doc/goseq.pdf"><strong>goseq</strong></a> (<span class="citation"><a href="#young2010gene">Young <i>et al.</i> 2010</a></span>) provides methods for performing GO analysis of <span class="notranslate">RNA</span>-Seq data while taking length bias into account. <strong>goseq</strong> could also be applied to other category-based tests of <span class="notranslate">RNA</span>-Seq data, such as KEGG pathway analysis, as discussed in a further section.</p>

<p><strong>goseq</strong> needs 2 files as inputs:</p>

<ul>
  <li>A tabular file with the differentially expressed genes from all genes assayed in the <span class="notranslate">RNA</span>-Seq experiment with 2 columns:
    <ul>
      <li>the Gene IDs (unique within the file), in uppercase letters</li>
      <li>a boolean indicating whether the gene is differentially expressed or not (<code class="language-plaintext highlighter-rouge">True</code> if differentially expressed or <code class="language-plaintext highlighter-rouge">False</code> if not)</li>
    </ul>
  </li>
  <li>A file with information about the length of a gene to correct for potential length bias in differentially expressed genes</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-prepare-the-datasets-for-goseq"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Prepare the datasets for goseq</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6"><strong>Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6</span></span>  with
      <ul>
        <li><em>“Add <span class="notranslate">expression</span>”</em>: <code class="language-plaintext highlighter-rouge">bool(c7&lt;0.05)</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“as a new column to”</em>: the <code class="language-plaintext highlighter-rouge">DESeq2 result file</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  with
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c8</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: the output of the <strong>Compute</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="ChangeCase" title="Tested with ChangeCase"><strong>Change Case</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: ChangeCase</span></span>  with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: the output of the previous <strong>Cut</strong></li>
        <li><em>“Change case of columns”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><em>“To”</em>: <code class="language-plaintext highlighter-rouge">Upper case</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output to <code class="language-plaintext highlighter-rouge">Gene IDs and differential <span class="notranslate">expression</span></code>.</p>

      <p>We just generated the first input for <strong>goseq</strong>. As second input for <strong>goseq</strong> we need the gene lengths. We can use here the gene lengths generated by <strong>featureCounts</strong> and format the gene IDs.</p>
    </li>
    <li>
      <p>Drag and drop one of the feature length datasets generated by <strong>featureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> into this history using the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-columns</span> <strong>View all histories</strong>.</p>
    </li>
    <li><span class="tool" data-tool="ChangeCase" title="Tested with ChangeCase"><strong>Change Case</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: ChangeCase</span></span>  with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: the <code class="language-plaintext highlighter-rouge">feature lengths</code> (output of <strong>featureCounts</strong>)</li>
        <li><em>“Change case of columns”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><em>“To”</em>: <code class="language-plaintext highlighter-rouge">Upper case</code></li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">Gene IDs and length</code></li>
  </ol>
</blockquote>

<p>We have now the two required input files for goseq.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-perform-go-analysis"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Perform GO analysis</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+galaxy0"><strong>goseq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+<span class="notranslate">galaxy</span>0</span></span>  with
      <ul>
        <li><em>“Differentially expressed genes file”</em>: <code class="language-plaintext highlighter-rouge">Gene IDs and differential <span class="notranslate">expression</span></code></li>
        <li><em>“Gene lengths file”</em>: <code class="language-plaintext highlighter-rouge">Gene IDs and length</code></li>
        <li><em>“Gene categories”</em>: <code class="language-plaintext highlighter-rouge">Get categories</code>
          <ul>
            <li><em>“Select a genome to use”</em>: <code class="language-plaintext highlighter-rouge">Fruit fly (dm6)</code></li>
            <li><em>“Select Gene ID format”</em>: <code class="language-plaintext highlighter-rouge">Ensembl Gene ID</code></li>
            <li><em>“Select one or more categories”</em>: <code class="language-plaintext highlighter-rouge">GO: Cellular Component</code>, <code class="language-plaintext highlighter-rouge">GO: Biological Process</code>, <code class="language-plaintext highlighter-rouge">GO: Molecular Function</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Output Top GO terms plot?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Extract the DE genes for the categories (GO/KEGG terms)?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><strong>goseq</strong> generates with these parameters 3 outputs:</p>

<ol>
  <li>
    <p>A table (<code class="language-plaintext highlighter-rouge">Ranked category list - Wallenius method</code>) with the following columns for each GO term:</p>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">category</code>: GO category</li>
      <li><code class="language-plaintext highlighter-rouge">over_rep_pval</code>: <em>p</em>-value for over-representation of the term in the differentially expressed genes</li>
      <li><code class="language-plaintext highlighter-rouge">under_rep_pval</code>: <em>p</em>-value for under-representation of the term in the differentially expressed genes</li>
      <li><code class="language-plaintext highlighter-rouge">numDEInCat</code>: number of differentially expressed genes in this category</li>
      <li><code class="language-plaintext highlighter-rouge">numInCat</code>: number of genes in this category</li>
      <li><code class="language-plaintext highlighter-rouge">term</code>: detail of the term</li>
      <li><code class="language-plaintext highlighter-rouge">ontology</code>: MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)</li>
      <li><code class="language-plaintext highlighter-rouge">p.adjust.over_represented</code>: <em>p</em>-value for over-representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</li>
      <li><code class="language-plaintext highlighter-rouge">p.adjust.under_represented</code>: <em>p</em>-value for under-representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure</li>
    </ol>

    <p>To identify categories significantly enriched/unenriched below some p-value cutoff, it is necessary to use the adjusted <em>p</em>-value.</p>

    <blockquote class="question">
      <h3 id="question-questions-15"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <ol>
        <li>How many GO terms are over-represented with an adjusted P-value &lt; 0.05? How many are under-represented?</li>
        <li>How are the over-represented GO terms divided into MF, CC and BP? And for under-represented GO terms?</li>
      </ol>

      <blockquote class="solution">
        <h3 id="solution-solution-27"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

        <ol>
          <li>
            <p>31 GO terms (0.27%) are over-represented and 83 (0.72%) under-represented.</p>

            <p><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  on c8 (adjusted p-value for over-represented GO terms) and c9 (adjusted p-value for under-represented GO terms)</p>
          </li>
          <li>
            <p>For over-represented, 20 BP, 3 CC and 8 MF and for under-represented, 49 BP, 29 CC and 5 MF</p>

            <p><span class="tool" data-tool="Grouping1" title="Tested with Grouping1"><strong>Group data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grouping1</span></span>  on column 7 (category) and count on column 1 (IDs)</p>
          </li>
        </ol>

      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>A graph with the top 10 over-represented GO terms</p>

    <blockquote class="question">
      <h3 id="question-questions-16"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <p><img src="../../images/ref-based/top_over-represented_go_terms.png" alt="Top over-represented GO terms. " loading="lazy" /></p>

      <p>What is the x-axis? How is it computed?</p>

      <blockquote class="solution">
        <h3 id="solution-solution-28"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

        <p>The x-axis is the percentage of genes in the category that have been identified as differentially expressed: \(100 x \frac{numDEInCat}{numInCat}\)</p>

      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>A table with the differentially expressed genes (from the list we provided) associated to the GO terms (<code class="language-plaintext highlighter-rouge">DE genes for categories (GO/KEGG terms)</code>)</p>
  </li>
</ol>

<blockquote class="comment">
  <h3 id="comment-advanced-tutorial-on-enrichment-analysis"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Advanced tutorial on enrichment analysis</h3>

  <p>In this tutorial, we covered GO enrichment analysis with <strong>goseq</strong>. To learn other methods and tools for gene set enrichment analysis, please have a look at the <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">“<span class="notranslate">RNA</span>-Seq genes to pathways”</a> tutorial.</p>
</blockquote>

<h2 id="kegg-pathways-analysis">KEGG pathways analysis</h2>

<p><strong>goseq</strong> can also be used to identify interesting KEGG pathways. The KEGG pathway database is a collection of pathway maps representing current knowledge of molecular interaction, reaction and relation networks. A map can integrate many entities including genes, proteins, <span class="notranslate">RNA</span>s, chemical compounds, glycans, and chemical reactions, as well as disease genes and drug targets.</p>

<p>For example, the pathway <code class="language-plaintext highlighter-rouge">dme00010</code> represents the glycolysis process (conversion of glucose into pyruvate with generation of small amounts of ATP and NADH) for Drosophila melanogaster:</p>

<p><img src="../../images/ref-based/dme00010_empty.png" alt="dme00010 KEGG pathway. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-perform-go-analysis-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Perform GO analysis</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+galaxy0"><strong>goseq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/goseq/goseq/1.36.0+<span class="notranslate">galaxy</span>0</span></span>  with
      <ul>
        <li><em>“Differentially expressed genes file”</em>: <code class="language-plaintext highlighter-rouge">Gene IDs and differential <span class="notranslate">expression</span></code></li>
        <li><em>“Gene lengths file”</em>: <code class="language-plaintext highlighter-rouge">Gene IDs and length</code></li>
        <li><em>“Gene categories”</em>: <code class="language-plaintext highlighter-rouge">Get categories</code>
          <ul>
            <li><em>“Select a genome to use”</em>: <code class="language-plaintext highlighter-rouge">Fruit fly (dm6)</code></li>
            <li><em>“Select Gene ID format”</em>: <code class="language-plaintext highlighter-rouge">Ensembl Gene ID</code></li>
            <li><em>“Select one or more categories”</em>: <code class="language-plaintext highlighter-rouge">KEGG</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Output Top GO terms plot?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Extract the DE genes for the categories (GO/KEGG terms)?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>As for the GO terms, 2 files are generated:</p>

<ol>
  <li>
    <p>A large table with the KEGG terms and some statistics</p>

    <blockquote class="question">
      <h3 id="question-questions-17"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

      <ol>
        <li>How many KEGG pathways terms have been identified?</li>
        <li>How many KEGG pathways terms are over-represented with an adjusted P value &lt; 0.05?</li>
        <li>What are the over-represented KEGG pathways terms?</li>
        <li>How many KEGG pathways terms are under-represented with an adjusted P value &lt; 0.05?</li>
      </ol>

      <blockquote class="solution">
        <h3 id="solution-solution-29"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

        <ol>
          <li>The file has 127 lines including an header, so 126 KEGG pathways have been identified.</li>
          <li>2 KEGG pathways (1.57%) are over-represented, using <strong>Filter</strong> on c6 (adjusted p-value for over-represented KEGG pathways)</li>
          <li>The first 2 KEGG pathways are <code class="language-plaintext highlighter-rouge">01100</code> and <code class="language-plaintext highlighter-rouge">00010</code>. By searching on the <a href="https://www.genome.jp/kegg/kegg2.html">KEGG database</a> for them, we can find more information about these pathways: <code class="language-plaintext highlighter-rouge">01100</code> corresponds to all metabolic pathways and <code class="language-plaintext highlighter-rouge">00010</code> to pathway for Glycolysis / Gluconeogenesis.</li>
          <li>1 KEGG pathways (0.79%) is under-represented, using <strong>Filter</strong> on c7 (adjusted p-value for under-represented KEGG pathways): <code class="language-plaintext highlighter-rouge">03040</code>, Spliceosome</li>
        </ol>
      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>A table with the differentially expressed genes (from the list we provided) associated with the KEGG pathways (<code class="language-plaintext highlighter-rouge">DE genes for categories (GO/KEGG terms)</code>)</p>
  </li>
</ol>

<p>We could investigate which genes are involved in which pathways by looking at the second file generated by <strong>goseq</strong>. However, this can be cumbersome and we would like to see the pathways as represented in the previous image. <strong>Pathview</strong> (<span class="citation"><a href="#luo2013pathview">Luo and Brouwer 2013</a></span>) can help to automatically generate similar images to the previous one, while also adding extra information about the genes (e.g. <span class="notranslate">expression</span>) in our study.</p>

<p>This tool needs 2 main inputs:</p>

<ul>
  <li>Pathway ID(s) to plot, either as just one ID or as a file with one column with the pathway IDs</li>
  <li>A tabular file with the genes in the <span class="notranslate">RNA</span>-Seq experiment with 2 (or more) columns:
    <ul>
      <li>the gene IDs (unique within the file)</li>
      <li>
        <p>some information about the genes</p>

        <p>This can be for example a p-value or a fold change. This information will be added to the pathway plot: the node of the corresponding gene will be colored given the value. If there are different columns, the different information will be plotted side by side on the node.</p>
      </li>
    </ul>
  </li>
</ul>

<p>Here we would like to visualize the 2 KEGG pathways: the over-represented <code class="language-plaintext highlighter-rouge">00010</code> (Glycolysis / Gluconeogenesis) and the under-represented <code class="language-plaintext highlighter-rouge">03040</code> (Spliceosome). We would like the gene nodes to be colored by Log2 Fold Change for the differentially expressed genes because of the treatment.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-perform-kegg-enrichment-analysis"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Perform KEGG enrichment analysis</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  with
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c3</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value</code></li>
      </ul>
    </li>
    <li>
      <p>Rename to <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value and their Log2 FC</code></p>

      <p>We extracted the ID and Log2 Fold Change for the genes that have a significant adjusted p-value.</p>
    </li>
    <li>
      <p>Create a new tabular file from the following (IDs of the pathways to plot)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>00010
03040
</code></pre></div>      </div>
    </li>
    <li>
      <p>Rename to <code class="language-plaintext highlighter-rouge">KEGG pathways to plot</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pathview/pathview/1.24.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pathview/pathview/1.24.0+galaxy0"><strong>Pathview</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pathview/pathview/1.24.0+<span class="notranslate">galaxy</span>0</span></span>  with
      <ul>
        <li><em>“Number of pathways to plot”</em>: <code class="language-plaintext highlighter-rouge">Multiple</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“KEGG pathways”</em>: <code class="language-plaintext highlighter-rouge">KEGG pathways to plot</code></li>
            <li><em>“Does the file have header (a first line with column names)?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li><em>“Species to use”</em>: <code class="language-plaintext highlighter-rouge">Fly</code></li>
        <li><em>“Provide a gene data file?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene data”</em>: <code class="language-plaintext highlighter-rouge">Genes with significant adj p-value and their Log2 FC</code></li>
            <li><em>“Does the file have header (a first line with column names)?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Format for gene data”</em>: <code class="language-plaintext highlighter-rouge">Ensembl Gene ID</code></li>
          </ul>
        </li>
        <li><em>“Provide a compound data file?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>In <em>“Output Options”</em>
          <ul>
            <li><em>“Output for pathway”</em>: <code class="language-plaintext highlighter-rouge">KEGG native</code>
              <ul>
                <li><em>“Plot on same layer?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><strong>Pathview</strong> generates a collection with the KEGG visualization: one file per pathway.</p>

<blockquote class="question">
  <h3 id="question-questions-18"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p><code class="language-plaintext highlighter-rouge">dme00010</code> KEGG pathway from <strong>Pathview</strong></p>

  <p><img src="../../images/ref-based/dme00010.png" alt="KEGG pathway. " loading="lazy" /></p>

  <ol>
    <li>What are the colored boxes?</li>
    <li>What is the color code?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-30"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The colored boxes are genes in the pathway that are differentially expressed</li>
      <li>Pay attention that the color code is counterintuitive: green is for value below 0, it means for genes with a log2FC &lt; 0 and red for genes with a log2FC &gt; 0.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real <span class="notranslate">RNA</span> sequencing data to extract useful information, such as which genes are up or downregulated by depletion of the <em>Pasilla</em> gene, but also which GO terms or KEGG pathways they are involved in. To answer these questions, we analyzed <span class="notranslate">RNA</span> sequence datasets using a reference-based <span class="notranslate">RNA</span>-Seq data analysis approach. This approach can be summarized with the following scheme:</p>

<figure id="figure-34"><img src="../../images/ref-based/tutorial-scheme.png" alt="Summary of the analysis pipeline used. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 34:</span> Summary of the analysis pipeline used</figcaption></figure>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>A spliced mapping tool should be used on eukaryotic RNA-Seq data</p>
</li>
                        
                        <li><p>Numerous factors should be taken into account when running a differential gene expression analysis</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="trapnell2009tophat">Trapnell, C., L. Pachter, and S. L. Salzberg, 2009 <b>TopHat: discovering splice junctions with RNA-Seq</b>. Bioinformatics 25: 1105–1111. <a href="https://academic.oup.com/bioinformatics/article/25/9/1105/203994">https://academic.oup.com/bioinformatics/article/25/9/1105/203994</a></li>
<li id="levin2010comprehensive">Levin, J. Z., M. Yassour, X. Adiconis, C. Nusbaum, D. A. Thompson <i>et al.</i>, 2010 <b>Comprehensive comparative analysis of strand-specific RNA sequencing methods</b>. Nature Methods 7: 709. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/</a></li>
<li id="young2010gene">Young, M. D., M. J. Wakefield, G. K. Smyth, and A. Oshlack, 2010 <b>Gene ontology analysis for RNA-seq: accounting for selection bias</b>. Genome Biology 11: R14. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-2-r14">https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-2-r14</a></li>
<li id="brooks2011conservation">Brooks, A. N., L. Yang, M. O. Duff, K. D. Hansen, J. W. Park <i>et al.</i>, 2011 <b>Conservation of an RNA regulatory map between Drosophila and mammals</b>. Genome Research 21: 193–202. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3032923/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3032923/</a></li>
<li id="marcel2011cutadapt">Marcel, M., 2011 <b>Cutadapt removes adapter sequences from high-throughput sequencing reads</b>. EMBnet.journal 17: 10–12. <a href="http://journal.embnet.org/index.php/embnetjournal/article/view/200">http://journal.embnet.org/index.php/embnetjournal/article/view/200</a></li>
<li id="robinson2011integrative">Robinson, J. T., H. Thorvaldsdóttir, W. Winckler, M. Guttman, E. S. Lander <i>et al.</i>, 2011 <b>Integrative genomics viewer</b>. Nature Biotechnology 29: 24. <a href="https://www.nature.com/nbt/journal/v29/n1/abs/nbt.1754.html">https://www.nature.com/nbt/journal/v29/n1/abs/nbt.1754.html</a></li>
<li id="wang2012rseqc">Wang, L., S. Wang, and W. Li, 2012 <b>RSeQC: quality control of RNA-seq experiments</b>. Bioinformatics 28: 2184–2185. <a href="https://www.ncbi.nlm.nih.gov/pubmed/22743226">https://www.ncbi.nlm.nih.gov/pubmed/22743226</a></li>
<li id="dobin2013star">Dobin, A., C. A. Davis, F. Schlesinger, J. Drenkow, C. Zaleski <i>et al.</i>, 2013 <b>STAR: ultrafast universal RNA-seq aligner</b>. Bioinformatics 29: 15–21. <a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">https://academic.oup.com/bioinformatics/article/29/1/15/272537</a></li>
<li id="kim2013tophat2">Kim, D., G. Pertea, C. Trapnell, H. Pimentel, R. Kelley <i>et al.</i>, 2013 <b>TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions</b>. Genome Biology 14: R36. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36</a></li>
<li id="liao2013featurecounts">Liao, Y., G. K. Smyth, and W. Shi, 2013 <b>featureCounts: an efficient general purpose program for assigning sequence reads to genomic features</b>. Bioinformatics 30: 923–930. <a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">https://academic.oup.com/bioinformatics/article/31/2/166/2366196</a></li>
<li id="luo2013pathview">Luo, W., and C. Brouwer, 2013 <b>Pathview: an R/Bioconductor package for pathway-based data integration and visualization</b>. Bioinformatics 29: 1830–1831. <a href="https://academic.oup.com/bioinformatics/article-abstract/29/14/1830/232698">https://academic.oup.com/bioinformatics/article-abstract/29/14/1830/232698</a></li>
<li id="love2014moderated">Love, M. I., W. Huber, and S. Anders, 2014 <b>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</b>. Genome Biology 15: 550. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8</a></li>
<li id="anders2015htseq">Anders, S., P. T. Pyl, and W. Huber, 2015 <b>HTSeq—a Python framework to work with high-throughput sequencing data</b>. Bioinformatics 31: 166–169. <a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">https://academic.oup.com/bioinformatics/article/31/2/166/2366196</a></li>
<li id="kim2015hisat">Kim, D., B. Langmead, and S. L. Salzberg, 2015 <b>HISAT: a fast spliced aligner with low memory requirements</b>. Nature Methods 12: 357. <a href="https://www.nature.com/articles/nmeth.3317">https://www.nature.com/articles/nmeth.3317</a></li>
<li id="ewels2016multiqc">Ewels, P., M. Magnusson, S. Lundin, and M. Käller, 2016 <b>MultiQC: summarize analysis results for multiple tools and samples in a single report</b>. Bioinformatics 32: 3047–3048. <a href="https://academic.oup.com/bioinformatics/article/32/19/3047/2196507">https://academic.oup.com/bioinformatics/article/32/19/3047/2196507</a></li>
<li id="thurmond2018flybase">Thurmond, J., J. L. Goodman, V. B. Strelets, H. Attrill, L. S. Gramates <i>et al.</i>, 2018 <b>FlyBase 2.0: the next generation</b>. Nucleic Acids Research 47: D759–D765. <a href="https://academic.oup.com/nar/article-abstract/47/D1/D759/5144957">https://academic.oup.com/nar/article-abstract/47/D1/D759/5144957</a></li>
<li id="kim2019graph">Kim, D., J. M. Paggi, C. Park, C. Bennett, and S. L. Salzberg, 2019 <b>Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype</b>. Nature Biotechnology 37: 907–915. <a href="https://www.nature.com/articles/s41587-019-0201-4">https://www.nature.com/articles/s41587-019-0201-4</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Reference-based RNA-Seq data analysis (Transcriptomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Bérénice Batut, Mallory Freeberg, Mo Heydarian, Anika Erxleben, Pavankumar Videm, Clemens Blank, Maria Doyle, Nicola Soranzo, Peter van Heusden, 2021 <b>Reference-based RNA-Seq data analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-ref-based,
author = "Bérénice Batut and Mallory Freeberg and Mo Heydarian and Anika Erxleben and Pavankumar Videm and Clemens Blank and Maria Doyle and Nicola Soranzo and Peter van Heusden",
title = "Reference-based RNA-Seq data analysis (Galaxy Training Materials)",
year = "2021",
month = "07",
day = "09"
url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                
                <blockquote class="agenda follow-up">
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Visualization of RNA-Seq results with heatmap2:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Visualization of RNA-Seq results with Volcano Plot:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> 3: RNA-seq genes to pathways:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
