<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>1: RNA-Seq reads to counts</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: 1: RNA-Seq reads to counts" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "1: RNA-Seq reads to counts",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / rna-seq-reads-to-counts / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for '1: RNA-Seq reads to counts' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html",
  "timeRequired": "PT3H",
  "keywords": "collections, mouse, QC",
  "description": "The questions this  addresses are:\n - How to convert RNA-seq reads into counts?\n - How to perform quality control (QC) of RNA-seq reads?\n - How to do this analysis efficiently in Galaxy?\n\n\\nThe objectives are:\n - Learn how RNA-seq reads are converted into counts\n - Understand QC steps that can be performed on RNA-seq reads\n - Generate interactive reports to summarise QC information with MultiQC\n - Use the Galaxy Rule-based Uploader to import FASTQs from URLs\n - Make use of Galaxy Collections for a tidy analysis\n - Create a Galaxy Workflow that converts RNA-seq reads into counts\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html",
      "name": "Reference-based RNA-Seq data analysis",
      "description": "Hands-on for 'Reference-based RNA-Seq data analysis' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/tutorials/collections/tutorial.html",
      "name": "Using dataset collections",
      "description": "Hands-on for 'Using dataset collections' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html",
      "name": "Rule Based Uploader",
      "description": "Hands-on for 'Rule Based Uploader' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Belinda Phipson"
    },
    {
      "@type": "Person",
      "name": "Harriet Dashnow"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Belinda Phipson"
    },
    {
      "@type": "Person",
      "name": "Harriet Dashnow"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
</script>

<section class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>1: RNA-Seq reads to counts</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="https://avatars.githubusercontent.com/mblue9?s=27" alt="Avatar">Maria Doyle</a>


<a href="/training-material/hall-of-fame/bphipson/" class="contributor-badge contributor-bphipson"><img src="https://avatars.githubusercontent.com/bphipson?s=27" alt="Avatar">Belinda Phipson</a>


<a href="/training-material/hall-of-fame/hdashnow/" class="contributor-badge contributor-hdashnow"><img src="https://avatars.githubusercontent.com/hdashnow?s=27" alt="Avatar">Harriet Dashnow</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to convert RNA-seq reads into counts?</p>
</li>
        
        <li><p>How to perform quality control (QC) of RNA-seq reads?</p>
</li>
        
        <li><p>How to do this analysis efficiently in Galaxy?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn how RNA-seq reads are converted into counts</p>
</li>
        
        <li><p>Understand QC steps that can be performed on RNA-seq reads</p>
</li>
        
        <li><p>Generate interactive reports to summarise QC information with MultiQC</p>
</li>
        
        <li><p>Use the Galaxy Rule-based Uploader to import FASTQs from URLs</p>
</li>
        
        <li><p>Make use of Galaxy Collections for a tidy analysis</p>
</li>
        
        <li><p>Create a Galaxy Workflow that converts RNA-seq reads into counts</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Reference-based RNA-Seq data analysis:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/galaxy-interface">Using Galaxy and Managing your Data</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Using dataset collections:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Rule Based Uploader:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4249555">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/workflows/" title="Workflows" alt="1: RNA-Seq reads to counts workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 18, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Measuring gene <span class="notranslate">expression</span> on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene <span class="notranslate">expression</span> in often complex systems. As sequencing costs have decreased, using <span class="notranslate">RNA</span>-Seq to simultaneously measure the <span class="notranslate">expression</span> of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.</p>

<p>There are many steps involved in analysing an <span class="notranslate">RNA</span>-Seq experiment. The analysis begins with sequencing <span class="notranslate">reads</span> (FASTQ files). These are usually aligned to a reference genome, if available. Then the number of <span class="notranslate">reads</span> mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on to determine differentially expressed genes and pathways. The purpose of this tutorial is to demonstrate how to do read alignment and counting, prior to performing differential <span class="notranslate">expression</span>. Differential <span class="notranslate">expression</span> analysis with limma-voom is covered in an accompanying tutorial <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><span class="notranslate">RNA</span>-seq counts to genes</a>. The tutorial here shows how to start from FASTQ data and perform the <span class="notranslate">mapping</span> and counting steps, along with associated Quality Control.</p>

<p><strong>Mouse mammary gland dataset</strong></p>

<p>The data for this tutorial comes from a Nature Cell Biology paper by <span class="citation"><a href="#Fu2015">Fu <i>et al.</i> 2015</a></span>. Both the raw data (sequence <span class="notranslate">reads</span>) and processed data (counts) can be downloaded from Gene <span class="notranslate">Expression</span> Omnibus database (GEO) under accession number <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60450">GSE60450</a>.</p>

<p>This study examined the <span class="notranslate">expression</span> profiles of basal and luminal cells in the mammary gland of virgin, pregnant and lactating mice. Six groups are present, with one for each combination of cell type and mouse status. Note that two biological replicates are used here, two independent sorts of cells from the mammary glands of virgin, pregnant or lactating mice, however three replicates is usually recommended as a minimum requirement for <span class="notranslate">RNA</span>-seq.</p>

<p>This is a <span class="notranslate">Galaxy</span> tutorial based on material from the <a href="http://combine-australia.github.io/RNAseq-R/07-rnaseq-day2.html">COMBINE R <span class="notranslate">RNA</span>seq workshop</a>, first taught <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">here</a>.</p>

<figure id="figure-1"><img src="../../images/rna-seq-reads-to-counts/mouse_exp.png" alt="Tutorial Dataset. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Tutorial Dataset</figcaption></figure>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#preparing-the-reads" id="markdown-toc-preparing-the-reads">Preparing the <span class="notranslate">reads</span></a>    <ol>
      <li><a href="#import-data-from-urls" id="markdown-toc-import-data-from-urls">Import data from URLs</a></li>
      <li><a href="#raw-reads-qc" id="markdown-toc-raw-reads-qc">Raw <span class="notranslate">reads</span> QC</a></li>
      <li><a href="#trim-reads" id="markdown-toc-trim-reads">Trim <span class="notranslate">reads</span></a></li>
      <li><a href="#trimmed-reads-qc" id="markdown-toc-trimmed-reads-qc">Trimmed <span class="notranslate">reads</span> QC</a></li>
    </ol>
  </li>
  <li><a href="#mapping" id="markdown-toc-mapping"><span class="notranslate">Mapping</span></a>    <ol>
      <li><a href="#map-reads-to-reference-genome" id="markdown-toc-map-reads-to-reference-genome">Map <span class="notranslate">reads</span> to reference genome</a></li>
    </ol>
  </li>
  <li><a href="#counting" id="markdown-toc-counting">Counting</a>    <ol>
      <li><a href="#count-reads-mapped-to-genes" id="markdown-toc-count-reads-mapped-to-genes">Count <span class="notranslate">reads</span> mapped to genes</a></li>
      <li><a href="#create-count-matrix" id="markdown-toc-create-count-matrix">Create count matrix</a></li>
    </ol>
  </li>
  <li><a href="#generating-a-qc-summary-report" id="markdown-toc-generating-a-qc-summary-report">Generating a QC summary report</a>    <ol>
      <li><a href="#strandness" id="markdown-toc-strandness">Strandness</a></li>
      <li><a href="#duplicate-reads" id="markdown-toc-duplicate-reads">Duplicate <span class="notranslate">reads</span></a></li>
      <li><a href="#reads-mapped-to-chromosomes" id="markdown-toc-reads-mapped-to-chromosomes"><span class="notranslate">Reads</span> mapped to chromosomes</a></li>
      <li><a href="#gene-body-coverage-5-3" id="markdown-toc-gene-body-coverage-5-3">Gene body coverage (5’-3’)</a></li>
      <li><a href="#read-distribution-across-features-exons-introns-intergenic" id="markdown-toc-read-distribution-across-features-exons-introns-intergenic">Read distribution across features (exons, introns, intergenic..)</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>

<h1 id="preparing-the-reads">Preparing the <span class="notranslate">reads</span></h1>

<h2 id="import-data-from-urls">Import data from URLs</h2>

<p>Read sequences are usually stored in compressed (gzipped) FASTQ files. Before the differential <span class="notranslate">expression</span> analysis can proceed, these <span class="notranslate">reads</span> must be aligned to the reference genome and counted into annotated genes. <span class="notranslate">Mapping</span> <span class="notranslate">reads</span> to the genome is a very important task, and many different aligners are available, such as HISAT2 (<span class="citation"><a href="#kim2015hisat">Kim <i>et al.</i> 2015</a></span>), STAR (<span class="citation"><a href="#dobin2013star">Dobin <i>et al.</i> 2013</a></span>) and Subread (<span class="citation"><a href="#Liao2013">Liao <i>et al.</i> 2013</a></span>). Most <span class="notranslate">mapping</span> tasks require larger computers than an average laptop, so usually read <span class="notranslate">mapping</span> is done on a server in a linux-like environment, requiring some programming knowledge. However, <span class="notranslate">Galaxy</span> enables you to do this <span class="notranslate">mapping</span> without needing to know programming and if you don’t have access to a server you can try to use one of the publically available Galaxies e.g. <a href="https://usegalaxy.org">use<span class="notranslate">galaxy</span>.org</a>, <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a>, <a href="https://usegalaxy.org.au/">use<span class="notranslate">galaxy</span>.org.au</a>.</p>

<p>The raw <span class="notranslate">reads</span> used in this tutorial were obtained from SRA from the link in GEO for the the mouse mammary gland dataset (e.g <code class="language-plaintext highlighter-rouge">ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/<span class="notranslate">reads</span>/ByStudy/sra/SRP%2FSRP045%2FSRP045534</code>). For the purpose of this tutorial we are going to be working with a small part of the FASTQ files. We are only going to be <span class="notranslate">mapping</span> 1000 <span class="notranslate">reads</span> from each sample to enable running through all the steps quickly. If working with your own data you would use the full data and some results for the full mouse dataset will be shown for comparison. The small FASTQ files are available in <a href="https://zenodo.org/record/4249555">Zenodo</a> and the links to the FASTQ files are provided below.</p>

<p>If you are sequencing your own data, the sequencing facility will almost always provide compressed FASTQ files which you can upload into <span class="notranslate">Galaxy</span>. For sequence data available through URLs, The <a href="/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"><span class="notranslate">Galaxy</span> Rule-based Uploader</a> can be used to import the files. It is much quicker than downloading FASTQs to your computer and uploading into <span class="notranslate">Galaxy</span> and also enables importing as a <strong>Collection</strong>. When you have more than a few files, using <span class="notranslate">Galaxy</span> Collections helps keep the datasets organised and tidy in the history. Collections also make it easier to maintain the sample names through tools and workflows. If you are not familiar with collections, you can take a look at the <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"><span class="notranslate">Galaxy</span> Collections tutorial</a> for more details. The screenshots below show a comparison of what the FASTQ datasets for this tutorial would look like in the history if we imported them as datasets versus as a collection with the Rule-based Uploader.</p>

<table>
  <thead>
    <tr>
      <th>Datasets</th>
      <th>Collection</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/rna-seq-reads-to-counts/imported-as-datasets.png" alt="Samples imported as datasets. " loading="lazy" /></td>
      <td><img src="../../images/rna-seq-reads-to-counts/imported-as-collection.png" alt="Samples imported as collection with Rule-based Uploader. " loading="lazy" /></td>
    </tr>
  </tbody>
</table>

<blockquote class="details">
  <h3 id="details-collections-and-sample-names"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Collections and sample names</h3>

  <p>Collections can also help to maintain the original sample names on the files throughout the tools used. The screenshots below show what we would see in one of the MultiQC reports that we will generate if we used datasets versus a collection.</p>

  <table>
    <thead>
      <tr>
        <th>Datasets</th>
        <th>Collection</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td><img src="../../images/rna-seq-reads-to-counts/samplesnames_without_collection.png" alt="Sample names without collection. " loading="lazy" /></td>
        <td><img src="../../images/rna-seq-reads-to-counts/samplesnames_with_collection.png" alt="Sample names with collection. " loading="lazy" /></td>
      </tr>
    </tbody>
  </table>

</blockquote>

<p>The information we need to import the samples for this tutorial (sample ID, Group, and link to the FASTQ file (URL) are in the grey box below.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SampleID	Group	URL
MCL1-DL	basallactate	https://zenodo.org/record/4249555/files/SRR1552455.fastq.gz
MCL1-DK	basallactate	https://zenodo.org/record/4249555/files/SRR1552454.fastq.gz
MCL1-DJ	basalpregnant	https://zenodo.org/record/4249555/files/SRR1552453.fastq.gz
MCL1-DI	basalpregnant	https://zenodo.org/record/4249555/files/SRR1552452.fastq.gz
MCL1-DH	basalvirgin	https://zenodo.org/record/4249555/files/SRR1552451.fastq.gz
MCL1-DG	basalvirgin	https://zenodo.org/record/4249555/files/SRR1552450.fastq.gz
MCL1-LF	luminalvirgin	https://zenodo.org/record/4249555/files/SRR1552449.fastq.gz
MCL1-LE	luminalvirgin	https://zenodo.org/record/4249555/files/SRR1552448.fastq.gz
MCL1-LD	luminalpregnant	https://zenodo.org/record/4249555/files/SRR1552447.fastq.gz
MCL1-LC	luminalpregnant	https://zenodo.org/record/4249555/files/SRR1552446.fastq.gz
MCL1-LB	luminalvirgin	https://zenodo.org/record/4249555/files/SRR1552445.fastq.gz
MCL1-LA	luminalvirgin	https://zenodo.org/record/4249555/files/SRR1552444.fastq.gz
</code></pre></div></div>

<p>In order to get these files into <span class="notranslate">Galaxy</span>, we will want to do a few things:</p>

<ul>
  <li>Strip the <em>header</em> out of the sample information (it doesn’t contain a URL <span class="notranslate">Galaxy</span> can download).</li>
  <li>Define the file <strong>Identifier</strong> column (<code class="language-plaintext highlighter-rouge">SampleID</code>).</li>
  <li>Define the <strong>URL</strong> column (<code class="language-plaintext highlighter-rouge">URL</code>) (this is the location <span class="notranslate">Galaxy</span> can download the data from).</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial e.g. <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>-seq <span class="notranslate">reads</span> to counts</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
    <li>
      <p>Import the files from Zenodo using <span class="notranslate">Galaxy</span>’s Rule-based Uploader.</p>
      <ul>
        <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
        <li>Click the tab <strong>Rule-based</strong>
          <ul>
            <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
            <li><em>“Load tabular data from”</em>: <code class="language-plaintext highlighter-rouge">Pasted Table</code></li>
          </ul>
        </li>
        <li>Paste the table from the grey box above. <em>(You should now see below)</em></li>
        <li>
          <p>Click <strong>Build</strong></p>

          <figure id="figure-2"><img src="../../images/rna-seq-reads-to-counts/rule_uploader.png" alt="Rule-based Uploader. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Rule-based Uploader</figcaption></figure>
        </li>
        <li>
          <p>In the <code class="language-plaintext highlighter-rouge">rules editor</code> that pops up:</p>

          <ul>
            <li><strong>Remove the header</strong>. From the <strong>Filter</strong> menu select <code class="language-plaintext highlighter-rouge">First or Last N Rows</code>
              <ul>
                <li><em>“Filter which rows?”</em>: <code class="language-plaintext highlighter-rouge">first</code></li>
                <li><em>“Filter how many rows?”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li>Click <code class="language-plaintext highlighter-rouge">Apply</code></li>
              </ul>
            </li>
            <li><strong>Define the Identifier and URL columns</strong>. From the <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select List Identifier(s)
                  <ul>
                    <li><em>“List Identifier(s)”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                  </ul>
                </li>
                <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button again and select URL instead
                  <ul>
                    <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">C</code></li>
                  </ul>
                </li>
                <li>Click <code class="language-plaintext highlighter-rouge">Apply</code>, and you should see your new column definitions listed</li>
              </ul>
            </li>
            <li><strong>Name the collection</strong>. For <em>“Name”</em> enter: <code class="language-plaintext highlighter-rouge">fastqs</code> <em>(You should now see below)</em></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Upload</code></li>
          </ul>

          <figure id="figure-3"><img src="../../images/rna-seq-reads-to-counts/rule_editor.png" alt="Rules Editor. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Rules Editor</figcaption></figure>

          <p>You should see a collection (list) called <code class="language-plaintext highlighter-rouge">fastqs</code> in your history containing all 12 FASTQ files.</p>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>If your data is not accessible by URL, for example, if your FASTQ files are located on your laptop and are not too large, you can upload into a collection as below. If they are large you could use FTP. You can take a look at the <a href="/training-material/topics/galaxy-interface/tutorials/get-data/slides.html">Getting data into <span class="notranslate">Galaxy</span> slides</a> for more information.</p>

<blockquote class="tip">
  <h3 id="tip-tip-upload-local-files-into-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Upload local files into a collection</h3>

  <ul>
    <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
    <li>Click the tab <strong>Collection</strong></li>
    <li>Click <strong>Choose Local Files</strong> and locate the files you want to upload
      <ul>
        <li><em>“Collection Type”</em>: <code class="language-plaintext highlighter-rouge">List</code></li>
      </ul>
    </li>
    <li>In the pop up that appears:
      <ul>
        <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">fastqs</code></li>
        <li>Click <code class="language-plaintext highlighter-rouge">Create list</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>If your FASTQ files are located in Shared Data, you can import them into your history as a collection as below.</p>

<blockquote class="tip">
  <h3 id="tip-tip-import-files-from-shared-data-into-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Import files from Shared Data into a collection</h3>

  <ul>
    <li>In the Menu at the top go to Shared Data &gt; Data Libraries</li>
    <li>Locate your FASTQ files</li>
    <li>Tick the checkboxes to select the files</li>
    <li>From the <strong>To History</strong> menu select <code class="language-plaintext highlighter-rouge">as a Collection</code></li>
    <li>In the pop up that appears:
      <ul>
        <li><em>“Which datasets?”</em>: <code class="language-plaintext highlighter-rouge">current selection</code></li>
        <li><em>“Collection type”</em>: <code class="language-plaintext highlighter-rouge">List</code></li>
        <li><em>“Select history”</em>: <code class="language-plaintext highlighter-rouge">select your History</code></li>
        <li>Click <code class="language-plaintext highlighter-rouge">Continue</code></li>
      </ul>
    </li>
    <li>In the pop up that appears:
      <ul>
        <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">fastqs</code></li>
        <li>Click <code class="language-plaintext highlighter-rouge">Create list</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>Take a look at one of the FASTQ files to see what it contains.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-take-a-look-at-fastq-format"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Take a look at FASTQ format</h3>

  <ol>
    <li>Click on the collection name (<code class="language-plaintext highlighter-rouge">fastqs</code>)</li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of one of the FASTQ files to have a look at what it contains</li>
  </ol>
</blockquote>

<blockquote class="details">
  <h3 id="details-fastq-format"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> FASTQ format</h3>
  <p>If you are not familiar with FASTQ format, see the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control tutorial</a></p>
</blockquote>

<h2 id="raw-reads-qc">Raw <span class="notranslate">reads</span> QC</h2>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data. Every base sequence gets a quality score from the sequencer and this information is present in the FASTQ file. A quality score of 30 corresponds to a 1 in 1000 chance of an incorrect base call (a quality score of 10 is a 1 in 10 chance of an incorrect base call). To look at the overall distribution of quality scores across the <span class="notranslate">reads</span>, we can use FastQC.</p>

<p>Sequence quality control is therefore an essential first step in your analysis. We will use similar tools as described in the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control” tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and Cutadapt (<span class="citation"><a href="#marcel2011cutadapt">Marcel 2011</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-raw-reads-with-fastqc"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check raw <span class="notranslate">reads</span> with <strong>FastQC</strong></h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">fastqs</code> (Input dataset collection)</li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <code class="language-plaintext highlighter-rouge">MCL1-DL</code> sample by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon</li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <ol>
    <li>What is the read length?</li>
    <li>What base quality score encoding is used?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The read length is 100 bp.</li>
      <li>Sanger quality score encoding is used.
This information can be seen at the top of the FastQC Webpage as below.</li>
    </ol>

    <figure id="figure-4"><img src="../../images/rna-seq-reads-to-counts/fastqc_webpage.png" alt="FastQC Webpage. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> FastQC Webpage</figcaption></figure>

  </blockquote>

</blockquote>

<p>The FastQC report contains a lot of information and we can look at the report for each sample. However, that is quite a few reports, 12 for this dataset. If you had more samples it could be a lot more. Luckily, there is a very useful tool called MultiQC (<span class="citation"><a href="#ewels2016multiqc">Ewels <i>et al.</i> 2016</a></span>) that can summarise QC information for multiple samples into a single report. We’ll generate a few MultiQC outputs in this tutorial so we’ll add name tags so we can differentiate them.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-aggregate-fastqc-reports-with-multiqc"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Aggregate FastQC reports with <strong>MultiQC</strong></h3>

  <ol>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters to aggregate the FastQC reports
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FastQC output”</em>: <code class="language-plaintext highlighter-rouge">RawData</code> files (output of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on trimmed <span class="notranslate">reads</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#fastqc-raw</code> to the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC and inspect the webpage</li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>

</blockquote>

<p>Note that these are the results for just 1000 <span class="notranslate">reads</span>. The FastQC results for the full dataset are shown below. The 1000 <span class="notranslate">reads</span> are the first <span class="notranslate">reads</span> from the FASTQ files, and the first <span class="notranslate">reads</span> usually originate from the flowcell edges, so we can expect that they may have lower quality and the patterns may be a bit different from the distribution in the full dataset.</p>

<p>You should see that most of the plots in the small FASTQs look similar to the full dataset. However, in the small FASTQs, there is less duplication, some Ns in the <span class="notranslate">reads</span> and some overrepresented sequences.</p>

<figure id="figure-5"><img src="../../images/rna-seq-reads-to-counts/fastqc_table.png" alt="General Statistics. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> General Statistics</figcaption></figure>
<figure id="figure-6"><img src="../../images/rna-seq-reads-to-counts/fastqc_sequence_counts_plot.png" alt="Sequence Counts. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Sequence Counts</figcaption></figure>
<figure id="figure-7"><img src="../../images/rna-seq-reads-to-counts/fastqc_per_base_sequence_quality_plot.png" alt="Sequence Quality. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Sequence Quality</figcaption></figure>
<figure id="figure-8"><img src="../../images/rna-seq-reads-to-counts/fastqc_per_sequence_quality_scores_plot.png" alt="Per Sequence Quality Scores. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Per Sequence Quality Scores</figcaption></figure>
<figure id="figure-9"><img src="../../images/rna-seq-reads-to-counts/fastqc_per_sequence_gc_content_plot.png" alt="Per Sequence GC Content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Per Sequence GC Content</figcaption></figure>
<figure id="figure-10"><img src="../../images/rna-seq-reads-to-counts/fastqc_per_base_n_content_plot.png" alt="Per base N content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> Per base N content</figcaption></figure>
<figure id="figure-11"><img src="../../images/rna-seq-reads-to-counts/fastqc_sequence_duplication_levels_plot.png" alt="Sequence Duplication Levels. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> Sequence Duplication Levels</figcaption></figure>
<figure id="figure-12"><img src="../../images/rna-seq-reads-to-counts/fastqc_adapter_content_plot.png" alt="Adapter Content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> Adapter Content</figcaption></figure>

<p>See the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control tutorial</a> for more information on FastQC plots.</p>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What do you think of the overall quality of the sequences?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>Overall, the samples look pretty good. The main things to note here are:</p>
    <ul>
      <li>The <strong>base quality</strong> is high in all samples.</li>
      <li>Some Illumina <strong>adapter</strong> has been detected.</li>
      <li>Some duplication in <span class="notranslate">RNA</span>-seq can be normal due to the presence of highly expressed genes. However, for some reason <code class="language-plaintext highlighter-rouge">MCL1-LE</code> and <code class="language-plaintext highlighter-rouge">MCL1-LF</code> have higher numbers of duplicates detected than the other samples.</li>
    </ul>

  </blockquote>
</blockquote>

<p>We will use Cutadapt to trim the <span class="notranslate">reads</span> to remove the Illumina adapter and any low quality bases at the ends (quality score &lt; 20). We will discard any sequences that are too short (&lt; 20bp) after trimming. We will also output the Cutadapt report for summarising with MultiQC.</p>

<p>The Cutadapt tool Help section provides the sequence we can use to trim this standard Illumina adapter <code class="language-plaintext highlighter-rouge">AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</code>, as given on the <a href="https://cutadapt.readthedocs.io/en/stable/guide.html#illumina-truseq">Cutadapt website</a>. For trimming paired-end data see the Cutadapt Help section. Other Illumina adapter sequences (e.g. Nextera) can be found at the <a href="http://sapac.support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html">Illumina website</a>. Note that Cutadapt requires at least three bases to match between adapter and read to reduce the number of falsely trimmed bases, which can be changed in the Cutadapt options if desired.</p>

<h2 id="trim-reads">Trim <span class="notranslate">reads</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-trim-reads-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Trim <span class="notranslate">reads</span> with <strong>Cutadapt</strong></h3>

  <ol>
    <li><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Single-end or Paired-end <span class="notranslate">reads</span>?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: <code class="language-plaintext highlighter-rouge">fastqs</code> (Input dataset collection)</li>
            <li>In <em>“Read 1 Options”</em>:
              <ul>
                <li>In <em>“3’ (End) Adapters”</em>:
                  <ul>
                    <li>Click on <em>“Insert 3’ (End) Adapters”</em>:</li>
                    <li>In <em>“1: 3’ (End) Adapters”</em>:
                      <ul>
                        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Enter custom 3’ adapter name (Optional)”</em>: <code class="language-plaintext highlighter-rouge">Illumina</code></li>
                            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Enter custom 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Minimum length”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Quality cutoff”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>“Output Options”</em>:
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We can take a look at the <span class="notranslate">reads</span> again now that they’ve been trimmed.</p>

<h2 id="trimmed-reads-qc">Trimmed <span class="notranslate">reads</span> QC</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-qc-of-trimmed-reads-with-fastqc"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: QC of trimmed <span class="notranslate">reads</span> with <strong>FastQC</strong></h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">RawData</code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters to aggregate the FastQC reports
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FastQC output”</em>: <code class="language-plaintext highlighter-rouge">RawData</code> files (output of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#fastqc-trimmed</code> to the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC and inspect the webpage</li>
  </ol>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-13"><img src="../../images/rna-seq-reads-to-counts/post_cutadapt_adapter_content.png" alt="Adapter Content post-trimming. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Adapter Content post-trimming</figcaption></figure>
<figure id="figure-14"><img src="../../images/rna-seq-reads-to-counts/post_cutadapt_fastqc_sequence_length_distribution_plot.png" alt="Sequence Length post-trimming. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> Sequence Length post-trimming</figcaption></figure>

<p>After trimming we can see that:</p>

<ul>
  <li>No adapter is detected now.</li>
  <li>The <span class="notranslate">reads</span> are no longer all the same length, we now have sequences of different lengths detected.</li>
</ul>

<h1 id="mapping"><span class="notranslate">Mapping</span></h1>

<p>Now that we have prepared our <span class="notranslate">reads</span>, we can align the <span class="notranslate">reads</span> for our 12 samples. There is an existing reference genome for mouse and we will map the <span class="notranslate">reads</span> to that. The current most widely used version of the mouse reference genome is <code class="language-plaintext highlighter-rouge">mm10/GRCm38</code> (although note that there is a new version <code class="language-plaintext highlighter-rouge">mm39</code> released June 2020). Here we will use <a href="https://ccb.jhu.edu/software/hisat2/index.shtml"><strong>HISAT2</strong></a> to align the <span class="notranslate">reads</span>. HISAT2 is the descendent of TopHat, one of the first widely-used aligners, but alte<span class="notranslate">rna</span>tive mappers could be used, such as STAR. See the <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#mapping"><span class="notranslate">RNA</span>-seq ref-based tutorial</a> for more information on <span class="notranslate">RNA</span>-seq mappers. There are often numerous <span class="notranslate">mapping</span> parameters that we can specify, but usually the default <span class="notranslate">mapping</span> parameters are fine. However, library type (paired-end vs single-end) and library strandness (stranded vs unstranded) require some different settings when <span class="notranslate">mapping</span> and counting, so they are two important pieces of information to know about samples. The mouse data comprises unstranded, single-end <span class="notranslate">reads</span> so we will specify that where necessary. HISAT2 can output a <span class="notranslate">mapping</span> summary file that tells what proportion of <span class="notranslate">reads</span> mapped to the reference genome. Summary files for multiple samples can be summarised with MultiQC. As we’re only using a subset of 1000 <span class="notranslate">reads</span> per sample, aligning should just take a minute or so. To run the full samples from this dataset would take longer.</p>

<h2 id="map-reads-to-reference-genome">Map <span class="notranslate">reads</span> to reference genome</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-map-reads-to-reference-with-hisat2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Map <span class="notranslate">reads</span> to reference with <strong>HISAT2</strong></h3>

  <ol>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select a reference genome”</em>: <code class="language-plaintext highlighter-rouge">mm10</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Is this a single or paired library?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTA/Q file”</em>: <code class="language-plaintext highlighter-rouge">Read 1 Output</code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li>In <em>“Summary Options”</em>:
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Output alignment summary in a more machine-friendly style.”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Print alignment summary to a file.”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters to aggregate the HISAT2 summary files
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">HISAT2</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Output of HISAT2”</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Mapping</span> summary</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#hisat</code> to the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC and inspect the webpage</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-settings-for-paired-end-or-stranded-reads"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Settings for Paired-end or Stranded <span class="notranslate">reads</span></h3>

  <ul>
    <li>If you have <strong>paired-end</strong> <span class="notranslate">reads</span>
      <ul>
        <li>Select <em>“Is this a single or paired library”</em> <code class="language-plaintext highlighter-rouge">Paired-end</code> or <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code> or <code class="language-plaintext highlighter-rouge">Paired-end data from single interleaved dataset</code></li>
      </ul>
    </li>
    <li>If you have <strong>stranded</strong> <span class="notranslate">reads</span>
      <ul>
        <li>Select <em>“Specify strand information”</em>: <code class="language-plaintext highlighter-rouge">Forward (FR)</code> or <code class="language-plaintext highlighter-rouge">Reverse (RF)</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-15"><img src="../../images/rna-seq-reads-to-counts/hisat2_se_plot.png" alt="HISAT2 mapping. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> HISAT2 <span class="notranslate">mapping</span></figcaption></figure>

<p>An important metric to check is the percentage of <span class="notranslate">reads</span> mapped to the reference genome. A low percentage can indicate issues with the data or analysis. Over 90% of <span class="notranslate">reads</span> have mapped in all samples, which is a good <span class="notranslate">mapping</span> rate, and the vast majority of <span class="notranslate">reads</span> have mapped uniquely, they haven’t mapped to multiple locations in the reference genome.</p>

<p class="hands_on">It is also good practice to visualise the read alignments in the BAM file, for example using IGV, see the <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#inspection-of-the-mapping-results"><span class="notranslate">RNA</span>-seq ref-based tutorial</a>.</p>

<p><strong>HISAT2</strong> generates a BAM file with mapped <span class="notranslate">reads</span>.</p>

<p>A BAM (<a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Binary Alignment Map</a>) file is a compressed binary file storing the read sequences, whether they have been aligned to a reference sequence (e.g. a chromosome), and if so, the position on the reference sequence at which they have been aligned.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-a-bamsam-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect a BAM/SAM file</h3>

  <ol>
    <li>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>

</blockquote>

<p>A BAM file (or a SAM file, the non-compressed version) consists of:</p>

<ul>
  <li>A header section (the lines starting with <code class="language-plaintext highlighter-rouge">@</code>) containing metadata particularly the chromosome names and lengths (lines starting with the <code class="language-plaintext highlighter-rouge">@SQ</code> symbol)</li>
  <li>
    <p>An alignment section consisting of a table with 11 mandatory fields, as well as a variable number of optional fields:</p>

    <table>
      <thead>
        <tr>
          <th>Col</th>
          <th>Field</th>
          <th>Type</th>
          <th>Brief Description</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td>QNAME</td>
          <td>String</td>
          <td>Query template NAME</td>
        </tr>
        <tr>
          <td>2</td>
          <td>FLAG</td>
          <td>Integer</td>
          <td>Bitwise FLAG</td>
        </tr>
        <tr>
          <td>3</td>
          <td><span class="notranslate">RNA</span>ME</td>
          <td>String</td>
          <td>References sequence NAME</td>
        </tr>
        <tr>
          <td>4</td>
          <td>POS</td>
          <td>Integer</td>
          <td>1- based leftmost <span class="notranslate">mapping</span> POSition</td>
        </tr>
        <tr>
          <td>5</td>
          <td>MAPQ</td>
          <td>Integer</td>
          <td><span class="notranslate">MAPping</span> Quality</td>
        </tr>
        <tr>
          <td>6</td>
          <td>CIGAR</td>
          <td>String</td>
          <td>CIGAR String</td>
        </tr>
        <tr>
          <td>7</td>
          <td>RNEXT</td>
          <td>String</td>
          <td>Ref. name of the mate/next read</td>
        </tr>
        <tr>
          <td>8</td>
          <td>PNEXT</td>
          <td>Integer</td>
          <td>Position of the mate/next read</td>
        </tr>
        <tr>
          <td>9</td>
          <td>TLEN</td>
          <td>Integer</td>
          <td>Observed Template LENgth</td>
        </tr>
        <tr>
          <td>10</td>
          <td>SEQ</td>
          <td>String</td>
          <td>Segment SEQuence</td>
        </tr>
        <tr>
          <td>11</td>
          <td>QUAL</td>
          <td>String</td>
          <td>ASCII of Phred-scaled base QUALity+33</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which information do you find in a SAM/BAM file?</li>
    <li>What is the additional information compared to a FASTQ file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Sequences and quality information, like a FASTQ</li>
      <li><span class="notranslate">Mapping</span> information, Location of the read on the chromosome, <span class="notranslate">Mapping</span> quality, etc</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-downloading-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Downloading a collection</h3>

  <p>To download a collection of datasets (e.g. the collection of BAM files) click on the floppy disk icon within the collection. This will download a tar file containing all the datasets in the collection. Note that for BAM files the .bai indexes (required for IGV) will be included automatically in the download.</p>

</blockquote>

<h1 id="counting">Counting</h1>

<p>The alignment produces a set of BAM files, where each file contains the read alignments for each sample. In the BAM file, there is a chromosomal location for every read that mapped. Now that we have figured out where each read comes from in the genome, we need to summarise the information across genes or exons. The mapped <span class="notranslate">reads</span> can be counted across mouse genes by using a tool called featureCounts (<span class="citation"><a href="#liao2013featurecounts">Liao <i>et al.</i> 2013</a></span>). featureCounts requires gene annotation specifying the genomic start and end position of each exon of each gene. For convenience, featureCounts contains built-in annotation for mouse (<code class="language-plaintext highlighter-rouge">mm10</code>, <code class="language-plaintext highlighter-rouge">mm9</code>) and human (<code class="language-plaintext highlighter-rouge">hg38</code>, <code class="language-plaintext highlighter-rouge">hg19</code>) genome assemblies, where exon intervals are defined from the NCBI RefSeq annotation of the reference genome. <span class="notranslate">Reads</span> that map to exons of genes are added together to obtain the count for each gene, with some care taken with <span class="notranslate">reads</span> that span exon-exon boundaries. The output is a count for each Entrez Gene ID, which are numbers such as <code class="language-plaintext highlighter-rouge">100008567</code>. For other species, users will need to read in a data frame in GTF format to define the genes and exons. Users can also specify a custom annotation file in SAF format. See the tool help in <span class="notranslate">Galaxy</span>, which has an example of what an SAF file should like like, or the Rsubread users guide for more information.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>In this example we have kept many of the default settings, which are typically optimised to work well under a variety of situations. For example, the default setting for featureCounts is that it only keeps <span class="notranslate">reads</span> that uniquely map to the reference genome. For testing differential <span class="notranslate">expression</span> of genes, this is preferred, as the <span class="notranslate">reads</span> are unambigously assigned to one place in the genome, allowing for easier interpretation of the results. Understanding all the different parameters you can change involves doing a lot of reading about the tool that you are using, and can take a lot of time to understand! We won’t be going into the details of the parameters you can change here, but you can get more information from looking at the tool help.</p>
</blockquote>

<h2 id="count-reads-mapped-to-genes">Count <span class="notranslate">reads</span> mapped to genes</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-count-reads-mapped-to-genes-with-featurecounts"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Count <span class="notranslate">reads</span> mapped to genes with <strong>featureCounts</strong></h3>

  <ol>
    <li><strong>featureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Alignment file”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Gene annotation file”</em>: <code class="language-plaintext highlighter-rouge">featureCounts built-in</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select built-in genome”</em>: <code class="language-plaintext highlighter-rouge">mm10</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">featureCounts</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Output of FeatureCounts”</em>: <code class="language-plaintext highlighter-rouge">featureCounts summary</code> (output of <strong>featureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#featurecounts</code> to the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC and inspect the webpage</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-settings-for-paired-end-or-stranded-reads-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Settings for Paired-end or Stranded <span class="notranslate">reads</span></h3>

  <ul>
    <li>If you have <strong>paired-end</strong> <span class="notranslate">reads</span>
      <ul>
        <li>Click <em>“Options for paired-end <span class="notranslate">reads</span>”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Count fragments instead of <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Enabled; fragments (or templates) will be counted instead of <span class="notranslate">reads</span></code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>If you have <strong>stranded</strong> <span class="notranslate">reads</span>
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> Select <em>“Specify strand information”</em>: <code class="language-plaintext highlighter-rouge">Stranded (Forward)</code> or <code class="language-plaintext highlighter-rouge">Stranded (Reverse)</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-16"><img src="../../images/rna-seq-reads-to-counts/featureCounts_assignment_plot.png" alt="featureCounts assignments. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 16:</span> featureCounts assignments</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What % <span class="notranslate">reads</span> are assigned to exons?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>~60-70% of <span class="notranslate">reads</span> are assigned to exons. This is a fairly typical number for <span class="notranslate">RNA</span>-seq.</p>

  </blockquote>

</blockquote>

<p>The counts for the samples are output as tabular files. Take a look at one. The numbers in the first column of the counts file represent the Entrez gene identifiers for each gene, while the second column contains the counts for each gene for the sample.</p>

<h2 id="create-count-matrix">Create count matrix</h2>

<p>The counts files are currently in the format of one file per sample. However, it is often convenient to have a count matrix. A count matrix is a single table containing the counts for all samples, with the genes in rows and the samples in columns. The counts files are all within a collection so we can use the <span class="notranslate">Galaxy</span> <strong>Column Join on multiple datasets</strong> tool to easily create a count matrix from the single counts files.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-count-matrix-with-column-join-on-multiple-datasets"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create count matrix with <strong>Column Join on multiple datasets</strong></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3"><strong>Column Join on multiple datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Tabular files”</em>: <code class="language-plaintext highlighter-rouge">Counts</code> (output of <strong>featureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Identifier column”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Number of header lines in each input file”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Add column name to header”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Take a look at the output. The tutorial uses a small subset of the data ~ 1000 <span class="notranslate">reads</span> per sample to save on processing time. Most rows in that matrix will contain all zeros, there will be ~600 non-zero rows. The output for the full dataset is shown below.</p>

<figure id="figure-17"><img src="../../images/rna-seq-reads-to-counts/count_matrix.png" alt="Count matrix. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 17:</span> Count matrix</figcaption></figure>

<p>Now it is easier to see the counts for a gene across all samples. The accompanying tutorial, <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><span class="notranslate">RNA</span>-seq counts to genes</a>, shows how gene information (symbols etc) can be added to a count matrix.</p>

<h1 id="generating-a-qc-summary-report">Generating a QC summary report</h1>

<p>There are several additional QCs we can perform to better understand the data, to see if it’s good quality. These can also help determine if changes could be made in the lab to improve the quality of future datasets.</p>

<p>We’ll use a prepared <span class="notranslate">workflow</span> to run the first few of the QCs below. This will also demonstrate how you can make use of <span class="notranslate">Galaxy</span> workflows to easily run and reuse multiple analysis steps. The <span class="notranslate">workflow</span> will run the first three tools: <strong>Infer Experiment</strong>, <strong>MarkDuplicates</strong> and <strong>IdxStats</strong> and generate a <strong>MultiQC</strong> report. You can then edit the <span class="notranslate">workflow</span> if you’d like to add other steps.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-qc-report-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run QC report <span class="notranslate">workflow</span></h3>

  <ol>
    <li><strong>Import the <span class="notranslate">workflow</span></strong> into <span class="notranslate">Galaxy</span>
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/workflows/qc_report.ga">this <span class="notranslate">workflow</span></a> or download it to your computer.</li>
        <li>Import the <span class="notranslate">workflow</span> into <span class="notranslate">Galaxy</span></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>“Archived <span class="notranslate">Workflow</span> URL”</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>“Archived <span class="notranslate">Workflow</span> File”</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>
    </li>
    <li>Import this file as type <span class="notranslate">BED</span> file:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://sourceforge.net/projects/rseqc/files/<span class="notranslate">BED</span>/Mouse_Mus_musculus/mm10_RefSeq.<span class="notranslate">bed</span>.gz/download
</code></pre></div>      </div>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
    </li>
    <li>Run <strong><span class="notranslate">Workflow</span> QC Report</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span></span> using the following parameters:
      <ul>
        <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1: Reference genes”</em>: the imported RefSeq <span class="notranslate">BED</span> file</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“2: BAM files”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>
</blockquote>

<p><strong>You do not need to run the hands-on steps below.</strong> They are just to show how you could run the tools individually and what parameters to set.</p>

<h2 id="strandness">Strandness</h2>

<p>As far as we know this data is unstranded, but as a sanity check you can check the strandness. You can use RSeQC Infer Experiment tool to “guess” the strandness, as explained in the <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"><span class="notranslate">RNA</span>-seq ref-based tutorial</a>. This is done through comparing the “strandness of <span class="notranslate">reads</span>” with the “strandness of transcripts”. For this tool, and many of the other RSeQC (<span class="citation"><a href="#wang2012rseqc">Wang <i>et al.</i> 2012</a></span>) tools, a reference <span class="notranslate">bed</span> file of genes (<code class="language-plaintext highlighter-rouge">reference genes</code>) is required. RSeQC provides some reference <span class="notranslate">BED</span> files for model organisms. You can import the RSeQC mm10 RefSeq <span class="notranslate">BED</span> file from the link <code class="language-plaintext highlighter-rouge">https://sourceforge.net/projects/rseqc/files/<span class="notranslate">BED</span>/Mouse_Mus_musculus/mm10_RefSeq.<span class="notranslate">bed</span>.gz/download</code> (and rename to <code class="language-plaintext highlighter-rouge">reference genes</code>) or import a file from Shared data if provided. Alte<span class="notranslate">rna</span>tively, you can provide your own <span class="notranslate">BED</span> file of reference genes, for example from UCSC (see the <a href="/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">Peaks to Genes tutorial</a>. Or the <strong>Convert GTF to <span class="notranslate">BED</span>12</strong> tool can be used to convert a GTF into a <span class="notranslate">BED</span> file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-strandness-with-infer-experiment"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check strandness with <strong>Infer Experiment</strong></h3>

  <ol>
    <li><strong>Infer Experiment</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input .bam file”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference gene model”</em>: <code class="language-plaintext highlighter-rouge">reference genes</code> (Reference <span class="notranslate">BED</span> file)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“1: Results”</em>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">RSeQC</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of RSeQC output?”</em>: <code class="language-plaintext highlighter-rouge">infer_experiment</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“RSeQC infer_experiment output”</em>: <code class="language-plaintext highlighter-rouge">Infer Experiment output</code> (output of <strong>Infer Experiment</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-18"><img src="../../images/rna-seq-reads-to-counts/rseqc_infer_experiment_plot.png" alt="Infer Experiment. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 18:</span> Infer Experiment</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Do you think the data is stranded or unstranded?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>It is unstranded as approximately equal numbers of <span class="notranslate">reads</span> have aligned to the sense and antisense strands.</p>

  </blockquote>

</blockquote>

<h2 id="duplicate-reads">Duplicate <span class="notranslate">reads</span></h2>

<p>Duplicate <span class="notranslate">reads</span> are usually kept in <span class="notranslate">RNA</span>-seq differential <span class="notranslate">expression</span> analysis as they can come from highly-expressed genes but it is still a good metric to check. A high percentage of duplicates can indicate a problem with the sample, for example, PCR amplification of a low complexity library (not many transcripts) due to not enough <span class="notranslate">RNA</span> used as input. FastQC gives us an idea of duplicates in the <span class="notranslate">reads</span> before <span class="notranslate">mapping</span> (note that it just takes a sample of the data). We can assess the numbers of duplicates in all mapped <span class="notranslate">reads</span> using the <strong>Picard MarkDuplicates</strong> tool. Picard considers duplicates to be <span class="notranslate">reads</span> that map to the same location, based on the start position of where the read maps. In general, we consider normal to obtain up to 50% of duplication.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-duplicate-reads-with-markduplicates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check duplicate <span class="notranslate">reads</span> with <strong>MarkDuplicates</strong></h3>

  <ol>
    <li><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select SAM/BAM dataset or dataset collection”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“1: Results”</em>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">Picard</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of Picard output?”</em>: <code class="language-plaintext highlighter-rouge">Markdups</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Picard output”</em>: <code class="language-plaintext highlighter-rouge">MarkDuplicate metrics</code> (output of <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-19"><img src="../../images/rna-seq-reads-to-counts/picard_deduplication.png" alt="MarkDups metrics. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 19:</span> MarkDups metrics</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Which two samples have the most duplicates detected?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p><code class="language-plaintext highlighter-rouge">MCL1-LE</code> and <code class="language-plaintext highlighter-rouge">MCL1-LF</code> have the highest number of duplicates in mapped <span class="notranslate">reads</span> compared to the other samples, similar to what we saw in the raw <span class="notranslate">reads</span> with FastQC.</p>

  </blockquote>

</blockquote>

<h2 id="reads-mapped-to-chromosomes"><span class="notranslate">Reads</span> mapped to chromosomes</h2>

<p>You can check the numbers of <span class="notranslate">reads</span> mapped to each chromosome with the <strong>Samtools IdxStats</strong> tool. This can help assess the sample quality, for example, if there is an excess of mitochondrial contamination. It could also help to check the sex of the sample through the numbers of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to X/Y or to see if any chromosomes have highly expressed genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-count-reads-mapping-to-each-chromosome-with-idxstats"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Count <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to each chromosome with <strong>IdxStats</strong></h3>

  <ol>
    <li><strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM file”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“1: Results”</em>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">Samtools</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of Samtools output?”</em>: <code class="language-plaintext highlighter-rouge">idxstats</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Samtools idxstats output”</em>: <code class="language-plaintext highlighter-rouge">IdxStats output</code> (output of <strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>
</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-20"><img src="../../images/rna-seq-reads-to-counts/samtools-idxstats-mapped-reads-plot.png" alt="IdxStats chromsome mappings. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 20:</span> IdxStats Chromosome <span class="notranslate">Mapping</span>s</figcaption></figure>
<figure id="figure-21"><img src="../../images/rna-seq-reads-to-counts/samtools-idxstats-xy-plot.png" alt="IdxStats XY . " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 21:</span> IdxStats X/Y <span class="notranslate">Mapping</span>s</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What do you think of the chromosome <span class="notranslate">mapping</span>s?</li>
    <li>Are the samples male or female? <em>(If a sample is not in the XY plot it means no <span class="notranslate">reads</span> mapped to Y)</em></li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Some of the samples have very high <span class="notranslate">mapping</span> on chromosome 5. What is going on there?</li>
      <li>The samples appear to be all female as there are few <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the Y chromosome. As this is a experiment studying virgin, pregnant and lactating mice if we saw large numbers of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the Y chromosome in a sample it would be unexpected and a probable cause for concern.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="gene-body-coverage-5-3">Gene body coverage (5’-3’)</h2>

<p>The coverage of <span class="notranslate">reads</span> along gene bodies can be assessed to check if there is any bias in coverage. For example, a bias towards the 3’ end of genes could indicate degradation of the <span class="notranslate">RNA</span>. Alte<span class="notranslate">rna</span>tively, a 3’ bias could indicate that the data is from a 3’ assay (e.g. oligodT-primed, 3’<span class="notranslate">RNA</span>-seq). You can use the RSeQC <strong>Gene Body Coverage (BAM)</strong> tool to assess gene body coverage in the BAM files.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-coverage-of-genes-with-gene-body-coverage-bam"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check coverage of genes with <strong>Gene Body Coverage (BAM)</strong></h3>

  <ol>
    <li><strong>Gene Body Coverage (BAM)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Run each sample separately, or combine mutiple samples into one plot”</em>: <code class="language-plaintext highlighter-rouge">Run each sample separately</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input .bam file”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference gene model”</em>: <code class="language-plaintext highlighter-rouge">reference genes</code> (Input dataset)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“1: Results”</em>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">RSeQC</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of RSeQC output?”</em>: <code class="language-plaintext highlighter-rouge">gene_body_coverage</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“RSeQC gene_body_coverage output”</em>: <code class="language-plaintext highlighter-rouge">Gene Body Coverage (BAM) (text)</code> (output of <strong>Gene Body Coverage (
BAM)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>

</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-22"><img src="../../images/rna-seq-reads-to-counts/rseqc_gene_body_coverage_plot.png" alt="Gene Body Coverage. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 22:</span> Gene Body Coverage</figcaption></figure>

<p>The plot below from the RSeQC website shows what samples with 3’biased coverage would look like.</p>
<figure id="figure-23"><img src="../../images/rna-seq-reads-to-counts/genebodycoverage.png" alt="Gene Body Coverage comparison. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 23:</span> Gene Body Coverage comparison</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What do you think of the coverage across gene bodies in these samples?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>It looks good. This plot looks a bit noisy in the small FASTQs but it still shows there’s pretty even coverage from 5’ to 3’ ends with no obvious bias in all the samples.</p>

  </blockquote>

</blockquote>

<h2 id="read-distribution-across-features-exons-introns-intergenic">Read distribution across features (exons, introns, intergenic..)</h2>

<p>We can also check the distribution of <span class="notranslate">reads</span> across known gene features, such as exons (CDS, 5’UTR, 3’UTR), introns and intergenic regions. In <span class="notranslate">RNA</span>-seq we expect most <span class="notranslate">reads</span> to map to exons rather than introns or intergenic regions. It is also the <span class="notranslate">reads</span> mapped to exons that will be counted so it is good to check what proportions of <span class="notranslate">reads</span> have mapped to those. High numbers of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to intergenic regions could indicate the presence of <span class="notranslate">DNA</span> contamination.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-distribution-of-reads-with-read-distribution"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check distribution of <span class="notranslate">reads</span> with <strong>Read Distribution</strong></h3>

  <ol>
    <li><strong>Read Distribution</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input .bam/.sam file”</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span> (BAM)</code> (output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference gene model”</em>: <code class="language-plaintext highlighter-rouge">reference genes</code> (Input dataset)</li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“1: Results”</em>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">RSeQC</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of RSeQC output?”</em>: <code class="language-plaintext highlighter-rouge">read_distribution</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“RSeQC read_distribution output”</em>: <code class="language-plaintext highlighter-rouge">Read Distribution output</code> (output of <strong>Read Distribution</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Webpage</code> output from MultiQC</li>
  </ol>

</blockquote>

<p>The MultiQC plot below shows the result from the full dataset for comparison.</p>

<figure id="figure-24"><img src="../../images/rna-seq-reads-to-counts/rseqc_read_distribution_plot.png" alt="Read Distribution. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 24:</span> Read Distribution</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What do you think of the read distribution?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>It looks good, most of the <span class="notranslate">reads</span> have mapped to exons and not many to introns or intergenic regions. The samples have pretty consistent read distribution, albeit with slightly higher numbers of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to CDS exons for <code class="language-plaintext highlighter-rouge">MCL1-LC</code> and <code class="language-plaintext highlighter-rouge">MCL1-LD</code>, and <code class="language-plaintext highlighter-rouge">MCL1-LE</code> and <code class="language-plaintext highlighter-rouge">MCL1-LF</code> have more <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to CDS exons than the other samples.</p>

  </blockquote>

</blockquote>

<p>The MultiQC report can be downloaded by clicking on the floppy disk icon on the dataset in the history.</p>

<blockquote class="question">
  <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Can you think of any other QCs that could be performed on <span class="notranslate">RNA</span>-seq <span class="notranslate">reads</span>?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The <span class="notranslate">reads</span> could be checked for:</p>
    <ul>
      <li>Ribosomal contamination</li>
      <li>Contamination with other species e.g. bacteria</li>
      <li>GC bias of the mapped <span class="notranslate">reads</span></li>
      <li>This is single-end data but paired-end mapped <span class="notranslate">reads</span> could be checked for fragment size (distance between the read pairs).</li>
    </ul>

  </blockquote>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we have seen how <span class="notranslate">reads</span> (FASTQ files) can be converted into counts. We have also seen QC steps that can be performed to help assess the quality of the data. A follow-on tutorial, <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><span class="notranslate">RNA</span>-seq counts to genes</a>, shows how to perform differential <span class="notranslate">expression</span> and QC on the counts for this dataset.</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>In RNA-seq, reads (FASTQs) are mapped to a reference genome with a spliced aligner (e.g HISAT2, STAR)</p>
</li>
                        
                        <li><p>The aligned reads (BAMs) can then be converted to counts</p>
</li>
                        
                        <li><p>Many QC steps can be performed to help check the quality of the data</p>
</li>
                        
                        <li><p>MultiQC can be used to create a nice summary report of QC information</p>
</li>
                        
                        <li><p>The Galaxy Rule-based Uploader, Collections and Workflows can help make analysis more efficient and easier</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="marcel2011cutadapt">Marcel, M., 2011 <b>Cutadapt removes adapter sequences from high-throughput sequencing reads</b>. EMBnet.journal 17: 10–12. <a href="http://journal.embnet.org/index.php/embnetjournal/article/view/200">http://journal.embnet.org/index.php/embnetjournal/article/view/200</a></li>
<li id="wang2012rseqc">Wang, L., S. Wang, and W. Li, 2012 <b>RSeQC: quality control of RNA-seq experiments</b>. Bioinformatics 28: 2184–2185. <a href="https://www.ncbi.nlm.nih.gov/pubmed/22743226">https://www.ncbi.nlm.nih.gov/pubmed/22743226</a></li>
<li id="dobin2013star">Dobin, A., C. A. Davis, F. Schlesinger, J. Drenkow, C. Zaleski <i>et al.</i>, 2013 <b>STAR: ultrafast universal RNA-seq aligner</b>. Bioinformatics 29: 15–21. <a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537">https://academic.oup.com/bioinformatics/article/29/1/15/272537</a></li>
<li id="liao2013featurecounts">Liao, Y., G. K. Smyth, and W. Shi, 2013 <b>featureCounts: an efficient general purpose program for assigning sequence reads to genomic features</b>. Bioinformatics 30: 923–930. <a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">https://academic.oup.com/bioinformatics/article/31/2/166/2366196</a></li>
<li id="Liao2013">Liao, Y., G. K. Smyth, and W. Shi, 2013 <b>The Subread aligner: fast,  accurate and scalable read mapping by seed-and-vote</b>. Nucleic Acids Research 41: e108–e108. <a href="https://doi.org/10.1093/nar/gkt214">10.1093/nar/gkt214</a></li>
<li id="Fu2015">Fu, N. Y., A. C. Rios, B. Pal, R. Soetanto, A. T. L. Lun <i>et al.</i>, 2015 <b>EGF-mediated induction of Mcl-1 at the switch to lactation is essential for alveolar cell survival</b>. Nature Cell Biology 17: 365–375. <a href="https://doi.org/10.1038/ncb3117">10.1038/ncb3117</a></li>
<li id="kim2015hisat">Kim, D., B. Langmead, and S. L. Salzberg, 2015 <b>HISAT: a fast spliced aligner with low memory requirements</b>. Nature Methods 12: 357. <a href="https://www.nature.com/articles/nmeth.3317">https://www.nature.com/articles/nmeth.3317</a></li>
<li id="ewels2016multiqc">Ewels, P., M. Magnusson, S. Lundin, and M. Käller, 2016 <b>MultiQC: summarize analysis results for multiple tools and samples in a single report</b>. Bioinformatics 32: 3047–3048. <a href="https://academic.oup.com/bioinformatics/article/32/19/3047/2196507">https://academic.oup.com/bioinformatics/article/32/19/3047/2196507</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=1: RNA-Seq reads to counts (Transcriptomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Maria Doyle, Belinda Phipson, Harriet Dashnow, 2021 <b>1: RNA-Seq reads to counts (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-rna-seq-reads-to-counts,
author = "Maria Doyle and Belinda Phipson and Harriet Dashnow",
title = "1: RNA-Seq reads to counts (Galaxy Training Materials)",
year = "2021",
month = "11",
day = "18"
url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                
                <blockquote class="agenda follow-up">
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> 2: RNA-seq counts to genes:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
