<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Introduction to proteomics, protein identification, quantification and statistical modelling</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/introduction/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1705797054' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1705797054' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Introduction to proteomics, protein identification, quantification and statistical modelling">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-01-21 00:30:56 +0000">
<meta name="DC.creator" content="Melanie FÃ¶ll">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Introduction to proteomics, protein identification, quant...">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1705797054'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/proteomics/tutorials/introduction/slides.html" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/introduction/slides-plain.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/proteomics/tutorials/introduction/slides.html">View markdown source on GitHub</a>
	<h1>Introduction to proteomics, protein identification, quantification and statistical modelling</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/foellmelanie/" class="contributor-badge contributor-foellmelanie"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/foellmelanie?s=36" alt="Avatar" width="36" height="36">Melanie FÃ¶ll</a>
	</div>

</div>


		

		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 21, 2024 </div>

	<hr />


	<h1 id="proteomics">Proteomics</h1>

<p><em>âA Proteome is the entire complement of proteins that is or can be expressed by a cell, tissue, or organism at a given time.â</em> - Marc Wilkins 1996</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/proteomics_schematic.png" alt="Schematic overview of proteomics, dna is transcribed to rna which is trnalsated to protein. The complexity is shown increasing at every step from minimal with dna, alternative splicing showing in the RNA, and then finall post translational modifications at the protein." /></p>

<p>.footnote[Image credit: <a href="https://commons.wikimedia.org/wiki/File:Genetic_code.svg">Madprime</a>]</p>

<p><span>Speaker Notes</span></p>

<p>Proteins are macromolecules that have many important functions in a cell.
Protein coding genes are transcribed into mRNA, which is translated into amino acids. The amino acid chain forms secondary, tertiary and quartary structures to obtain a functional protein. One gene may generate different proteins due to alternative splicing and post-translational modifications. Therefore, the proteome level shows a higher molecular complexity. The proteome is defined as the entirety of proteins expressed by a genome or by a cell or tissue at a given time. The study of proteins is important as their identity and abundance is only partially predictable from DNA and mRNA information. This is due to alternative splicing, post-translational modifications, protein turnover and subcellular localization.</p>

<hr />

<h1 id="mass-spectometry-ms-based-proteomics">Mass Spectometry (MS)-based Proteomics</h1>

<p>.force-right[</p>
<ul>
  <li>Bottom-up approach: measurement of peptides after
enzymatic digestion of proteins</li>
  <li>Measures the peptides mass-to-charge (m/z) ratio and stores
m/z â intensity pairs in mass spectra</li>
  <li>High sensitivity and throughput</li>
  <li>Standard method for the analysis of complex samples
]</li>
</ul>

<p><br /><br />
<img src="/training-material/topics/proteomics/images/intro-slides/ms-proteomics.png" alt="schematic of mass spec with nano hplc producing tens of thousands of mass spectra, these show up as intensity vs m/z plots with skinny peaks." /></p>

<p><span>Speaker Notes</span></p>

<p>Mass spectrometry is the standard method for proteomic analyses of complex samples. In the classical bottom-up approach, proteins are enzymatically digested into peptides. Peptides can be analyzed with high sensitivity and throughput in a mass spectrometer. The peptide mass is measured as mass-to-charge ratio. Only charged peptides can be measured in the mass spectrometer. Tens of thousands of mass spectra are generated per sample. Each spectrum consists of many mass-to-charge and intensity pairs.</p>

<hr />

<h1 id="ms-based-proteomic-techniques">MS-based Proteomic Techniques</h1>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">.image-90[<img src="/training-material/topics/proteomics/images/intro-slides/explorative-proteomics.png" alt="Cartoon of various blobs" />]</td>
      <td style="text-align: center">.image-90[<img src="/training-material/topics/proteomics/images/intro-slides/targeted-proteomics.png" alt="Target symbol" />]</td>
      <td style="text-align: center">.image-90[<img src="/training-material/topics/proteomics/images/intro-slides/ms-imaging.png" alt="Colourful blob with a scale next to it indicating different regions." />]</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>Explorative Proteomics</strong></td>
      <td style="text-align: center"><strong>Targeted Proteomics</strong></td>
      <td style="text-align: center"><strong>Mass Spectometry</strong></td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>(DDA/DIA)</strong></td>
      <td style="text-align: center"><strong>(SRM/PRM)</strong></td>
      <td style="text-align: center"><strong>Imaging</strong></td>
    </tr>
    <tr>
      <td style="text-align: center">All proteins of a cell/organ(ism)</td>
      <td style="text-align: center">Focus on subset of proteins</td>
      <td style="text-align: center">Focus on location of proteins</td>
    </tr>
  </tbody>
</table>

<p><span>Speaker Notes</span></p>

<p>Different mass spectrometry based proteomic techniques exist. The most common one is explorative or shotgun proteomics. It aims to identify as many proteins as possible from a sample. It comes in two flavours: data dependent acquisition (DDA) and data independent acquisition (DIA). A second technique is targeted proteomics. It measures only a predefined set of proteins with very accurate quantification. A third technique is mass spectrometry imaging. It measures the spatial distribution of peptides or proteins in thin tissue sections.</p>

<hr />

<h1 id="proteomics-tools-in-galaxy">Proteomics tools in Galaxy</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/tools-overview.png" alt="screenshot of a list of tools with a large number of logos include proteowizard, trans proteomic pipeline, cardinal, openms, openswatch, DIA umpire, encyclopeDIA, compomics, uniprot, ms, mq." /></p>

<p><span>Speaker Notes</span></p>

<p>Plenty of software for DDA, D I A and mass spectrometry imaging are available in Galaxy. Here is an overview of all Proteomics tools installed on the European Galaxy Server. Other public Galaxy servers offer a similar or complementary proteomic tool kit. The Galaxy proteomics tools and training materials are constantly expanding and improving.</p>

<hr />

<h1 id="explorative-proteomics">Explorative Proteomics</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/explorative-proteomics.png" alt="Colourful blobs next to what might be a liver" /></p>

<p><strong>(DDA/DIA)</strong></p>

<p>All proteins of a cell / organ / organism</p>

<p><span>Speaker Notes</span></p>

<p>This presentation will focus on explorative proteomics via the traditional DDA approach.</p>

<hr />

<h1 id="explorative-proteomics-workflow">Explorative Proteomics Workflow</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/explorative-proteomics-workflow.png" alt="Flow chart from sample prep to mass spec to data analysis." /></p>

<p><span>Speaker Notes</span></p>

<h2 id="proteomics-experiments-consist-of-three-main-steps-first-the-sample-is-prepared-for-the-analysis-in-the-mass-spectrometer-then-the-sample-is-measured-in-the-mass-spectrometer-last-the-obtained-data-is-analyzed">Proteomics experiments consist of three main steps. First, the sample is prepared for the analysis in the mass spectrometer. Then the sample is measured in the mass spectrometer. Last, the obtained data is analyzed.</h2>

<p>class: top</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/wf1-sample-prep.png" alt="sample prep, MS, data analysis flow chart, but Sample Preparation is highlighted." /></p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/sample-prep.png" alt="liver followed by protein extraction, reduction and alkylation, tryptic digestion, and deslating drying to produce a peptide pellet. Below is a schematic of protein crosslinking with S-S binding." /></p>

<p><span>Speaker Notes</span></p>

<p>Typical sample preparation steps include protein extraction, reduction and alkylation, tryptic digestion and desalting. Before tryptic digestion, disulfide bridges are reduced and cysteins alkylated. This ensures that tryptic peptides are separated from each other and allows their mass based identification. Trypsin cleaves the amino acid sequences C-terminal of arginin and lysine. Desalting is a clean up step to protect the instrument from contamination and clogging.</p>

<hr />

<p>class: top</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/wf2-mass-spec.png" alt="sample prep, MS, data analysis flow chart, but Mass spectrometry is highlighted." /></p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/mass-spec-schematic.png" alt="peptide pellet from previous slide is solved in acidic buffer followed by: liquid chromotography, ion source, mass analyzer, detecter." /></p>

<p><span>Speaker Notes</span></p>

<p>Sample measurement in a mass spectrometer consists of different steps. A high performance liquid chromatography (HPLC) system is attached in front of the mass spectrometer. It separates the injected peptide mixture according to their hydrophobicity. The peptides elute from the LC column into the mass spectrometer within several minutes to hours. This reduces sample complexity and gives the mass spectrometer more time for the measurement. The acidic LC buffer charges the peptides positively at their N-terminus and the basic lysin or arginin amino acids on the C-terminus. The LC column is directly connected with the ion source needle. There, high voltage and heat are applied to evaporate the ionized peptides into the gas phase. This process is called electrospray ionization. Inside the mass spectrometer the mass analyzer separates peptides based on their mass-to-charge ratio. The detector detects the peptide ions.</p>

<hr />

<p>class: top</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/wf2-mass-spec.png" alt="sample prep, MS, data analysis flow chart, but Mass spectrometry is highlighted." /></p>

<p>Liquid chromatography tandem mass spectrometry (LC-MS/MS)</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/mass-spec-schematic2.png" alt="an LC+Ion source goes into ms1: mass spectra which are filtered for most abundant m/z, then fragmentation of filtered m/z produces an MS2: Mass spectra. This process is repeated for the 2nd and 3rd most abundant m/z and so on." /></p>

<p><span>Speaker Notes</span></p>

<p>Typically explorative proteomics is performed via liquid chromatography tandem mass spectrometry (LC-MSMS).  While the sample elutes from the LC column, thousands of mass spectra are acquired. First, a mass spectrum of all peptides at this time point is measured. These mass spectra are called MS1 spectra. From these spectra the N most abundant peptide peaks are determined. These topN peptides get fragmented. N is typically between 3 and 20. This example shows a Top3 method. The filter unit of the mass spectrometer (a quadrupole) allows only these peptides to pass. One after the other is selected in the filter unit and then fragmented by collision with neutral gas molecules. This fragmentation breaks the peptide bonds and generates peptide fragments. The peptide fragments are measured again via the mass analyzer and detector. These spectra are called MS2 spectra. After all topN peptides were fragmented and measured, another full MS1 mass spectra is acquired. MS1 and MS2 spectra are acquired in this way during the elution of the sample from the LC.</p>

<hr />

<p>class: top</p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/wf3-data-analysis.png" alt="sample prep, MS, data analysis flow chart, but data analysis is highlighted." /></p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/data-analysis-schematic.png" alt="the MS1 and MS2 mass spectra go into this workflow. The ms1 goes through peptide quantification. the MS2 goes through peptide identification and then protein identification. everything goes to protein quantification and statitstical analysis." /></p>

<p><span>Speaker Notes</span></p>

<p>The analysis of the acquired mass spectra comprises several steps. First peptides are identified via their MS2 fragmentation spectra. From these peptide identities the corresponding proteins are assembled. The MS1 spectra are used for peptide quantification. Peptide quantities are summarized into protein quantities. The information about protein identity and quantity allows following statistical analyses.</p>

<hr />

<h1 id="typical-explorative-tandem-mass-spectrometry-workflow">Typical explorative tandem mass spectrometry workflow</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/all-schematic.png" alt="proteins in tubes, mass spec in the center, and a computer on the right." /></p>

<p><img src="/training-material/topics/proteomics/images/intro-slides/wf4-all.png" alt="sample prep, ms, data analysis flow chart." /></p>

<p><span>Speaker Notes</span></p>

<p>This was the overview of a typical explorative tandem mass spectrometry workflow. Now we will dive into more details of the data analysis part.</p>

<hr />

<h1 id="peptide-identification-with-ms2-fragment-spectra">Peptide identification with MS2 fragment spectra</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/fragment-spectra.png" alt="MS1 spectra with precursor ions are fragmented, an example sequence is given. On the right we see different fragments for each potential cut site in the sequence. Below the MS2 spectra show numerous lines which sum up to the m/z of the entire sequence and have individual peaks for each of the individual subsequences of the protein." /></p>

<p><span>Speaker Notes</span></p>

<p>Many tryptic peptides of an organism have same or similar masses. Therefore, MS1 spectra donât allow reliable peptide sequence identifications. MS2 spectra allow peptide identification via the generated peptide fragments. The N-terminal fragments are called b-ions and the C-terminal fragments y-ions. The differences between the fragment masses correspond to the mass of an amino acid. This allows manual interpretation of the spectra. However, this is a tricky procedure because in reality the MS2 spectra contain more noise and side product peaks than shown here. Also, in explorative proteomics tens of thousands of spectra are acquired and make manual interpretation unfeasible. The manual interpretation process is automatized with so called âde novo sequencingâ software. These algorithms have improved in the last years. No information about potential protein sequences in the sample are needed. The default software for peptide identification are so called âsearch enginesâ. They require information about all protein sequences of the analyzed organism as FASTA database. From this they generate in silico spectra which are then matched to the measured mass spectra.</p>

<hr />

<h1 id="peptide-spectrum-matching">Peptide spectrum matching</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/insilico-identification1.png" alt="Flowchart again, proteins go to peptides go to MS1 to MS2 to a Peak List. Below a sequence database entry goes through in-silico digestion and theoretical fragmentation, producing a theoretical peaklist which is matched against the previous peaklist." /></p>

<p><span>Speaker Notes</span></p>

<p>This process is often called âpeptide spectrum matchingâ. It starts with a protein sequence database of all protein sequences of the analyzed organism. Analogous to the procedure in the sample, the protein sequences are in silico digested. This means that the sequences are cut after each lysine and arginin. These in silico tryptic peptide sequences are then in silico fragmented. All amino acid bonds may potentially break and generate peptide fragments. Therefore, all possible fragments are generated in silico. From each in silico peptide fragment, the mass is calculated. In case of amino acid modifications the mass of the modification is added accordingly. Fixed modifications are added to each occurrence of the amino acid on which they occur. Variable modifications may not occur on every amino acid and therefore two masses, with and without the modification, are calculated. These in silico generated mass values are matched to the mass values from each measured MS2 spectra. A matching score allows to find the best identifications for each MS2 spectrum.</p>

<hr />

<h1 id="peptide-spectrum-matching-1">Peptide spectrum matching</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/insilico-identification2.png" alt="The previous process has added decoy sequences to the sequence database. The rest is the same. A ranked score chart on the right shows a combination of real and decoy colours. 1% FDR cutoff is written next to a point." /></p>

<p><span>Speaker Notes</span></p>

<p>Potentially false matches may occur, therefore the false positive rate is controlled. This is done by adding decoy sequences to the protein sequence database. These sequences are generated by reversing or shuffling the real sequences and will not exist in the sample. In case such sequences are considered a good match to an MS2 spectra, this is a false match. One option to control the number of false positive matches is via a false discovery cutoff, that includes the best matching scores with only 1 % wrong decoy matches included.</p>

<hr />

<h1 id="fasta-file-format">FASTA File Format</h1>

<ul>
  <li>Text based file to store DNA, RNA, protein sequences in a single letter code</li>
  <li>Each entry has
    <ul>
      <li>One header line: starting with <code class="language-plaintext highlighter-rouge">&gt;</code> and containing a unique identifier</li>
      <li>A (nucleotide/amino acid) sequence</li>
    </ul>
  </li>
</ul>

<p><img src="/training-material/topics/proteomics/images/intro-slides/fasta-proteomics.png" alt="a fasta file is shown with interleaved headers and sequences." /></p>

<p><span>Speaker Notes</span></p>

<p>The protein sequence database is stored in a FASTA file. This is a text based file to store DNA, RNA or protein sequences in a single letter code. Each entry contains a header line and the sequence. The header line starts with a &gt; and is follwed by a unique identifier.</p>

<hr />

<h1 id="fasta-files-for-proteomics">FASTA Files for Proteomics</h1>

<ul>
  <li>
    <p>A proteome is the set of proteins thought to be expressed by an organism.</p>
  </li>
  <li>
    <p>Only proteins that are present in the FASTA file can be identified</p>
  </li>
  <li>The more proteins are present in the FASTA file the higher the changes for false identifications
    <ul>
      <li>Choose the right FASTA file</li>
    </ul>
  </li>
  <li>Source for proteome FASTA files:
    <ul>
      <li>Uniprot, neXtProt, NCBI, own/public DNA or RNA sequencing data</li>
    </ul>
  </li>
</ul>

<p>.footnote[
Tutorial: <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Protein FASTA database handling</a>
]</p>

<p><span>Speaker Notes</span></p>

<p>A proteome is the set of proteins thought to be expressed by an organism. Only proteins that are present in the FASTA file can be identified. But the more proteins are present in the FASTA file the higher the chances for false identifications and the longer the computation time.
Sources for proteome FASTA files include uniprot, nextprot, NCBI or DNA and RNA sequencing data.</p>

<hr />

<h1 id="protein-identification-via-protein-inference">Protein Identification via Protein Inference</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/protein-interference.png" alt="another fasta file is shown with regions of two sequences highlighted. In the first a uniue region is. A shared/razor region is highlighted in both." /></p>

<p><span>Speaker Notes</span></p>

<p>After having identified peptides, they need to be reconstructed into proteins. This step is called protein inference and is not trivial. Unique or proteotypic peptide sequences belong to one protein. But other peptide sequences may belong to several proteins. These peptides are called shared or razor peptides. Most protein inference algorithms assign them to the protein that has the most other peptides. In the depicted example peptide two would be matched to protein one which is for sure present in the sample because it has one unique peptide.</p>

<hr />

<h1 id="quantification-methods-in-proteomics">Quantification methods in proteomics</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/proteomic_quant_overview.png" alt="Flowchart guiding in the choice of method" /></p>

<p>.footnote[ Tutorial: <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html">Label-free versus Labelled: How to Choose Your Quantitation Method</a>]</p>

<p><span>Speaker Notes</span></p>

<p>Different quantification approaches exist in mass spectrometry based proteomics. In explorative proteomic approaches relative quantification methods are applied. They compare the amount of proteins between different samples. Label-free and label based methods exist. Labels add specific mass tags to the peptides of different samples via metabolic or chemical ways.</p>

<hr />

<h1 id="quantification-methods-in-proteomics-1">Quantification Methods in Proteomics</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/quantification-methods.png" alt="A chart, on the left is label free methods (peak integration). Each sample goes in its own tube. On the right is label-based with iTRAQ and SILAC mentioned. Each of 4 samples go in a tube before a label is added and they are mixed together. On the right is SILAC where all samples are mixed together. Below in the graphic the peaks are shown. For peak integration two spectra are produced. In iTRAQ and SILAC both produce a single plot with many peaks inside." /></p>

<p>.footnote[Image credit: <a href="https://doi.org/10.1371/journal.pcbi.1002277">KÃ¤ll and Vitek 2011</a>]</p>

<p><span>Speaker Notes</span></p>

<p>In label-free approaches every sample is measured separately. Afterwards, the protein amounts are compared between measurements. Chemical labeling techniques add a mass label to the digested peptide. Afterwards the samples are mixed and measured in one run. The different masses of the added labels allow distinguishing the origin of the proteins during data analysis. Depending on the labeling technique, up to 16 different labels exist. In metabolic labeling amino acids with heavy isotopes are added to the cell culture medium of one condition. During cell growth these amino acids get incorporated into proteins. Thus, proteins of the heavy condition can be distinguished by their normal counterparts via a fixed mass shift.</p>

<hr />

<h1 id="label-free-peptide-quantification">Label-free Peptide Quantification</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/label-free-quantification.png" alt="Here there are two xyz plots with a single m/z (542.5) and a plot with a different peak/retention time behaviour in sample 1 and 2." /></p>

<p><span>Speaker Notes</span></p>

<p>For label-free quantification all peak areas in the MS1 spectra are integrated.</p>

<hr />

<h1 id="protein-quantification">Protein Quantification</h1>

<ul>
  <li>Include only unique or also ambiguous razor peptides?</li>
  <li>Include peptides with modifications?</li>
  <li>How to summarize peptide abundances into protein abundances?</li>
</ul>

<p><img src="/training-material/topics/proteomics/images/intro-slides/protein-quantification-table.png" alt="table with peptides 1-4. Some are unique, some are 'razor'. Some have modificatinos. Their intensities are listed. This table has an arrow below for 'summarization by mean, sum, ...' and quantificatino value for Protein A." /></p>

<p><span>Speaker Notes</span></p>

<p>Peptide abundances are summarized into protein quantifications. This requires decisions about which peptides to include in the summarization. Only unique peptides? Only proteins with or without modifications? Last, the protein abundance may be computed by taking the median, mean, weighted mean or sum of all its peptides.</p>

<hr />

<h1 id="maxquant-software">MaxQuant Software</h1>

<p>.left-column50[</p>

<p>.image-100[
  <img src="/training-material/topics/proteomics/images/intro-slides/maxquant-paper.png" alt="Screenshot of the header of a paper" />
]</p>

<ul>
  <li>Freeware, âBlack Boxâ</li>
  <li>Popular non-commercial proteomics software
]</li>
</ul>

<p>.right-column50[</p>

<ul>
  <li><strong>Raw data import</strong></li>
  <li><strong>Protein Identification:</strong>
    <ul>
      <li>Andromeda Search Engine</li>
    </ul>
  </li>
  <li><strong>Protein Quantification:</strong>
    <ul>
      <li>Label-free</li>
      <li>Label-based
        <ul>
          <li>SILAC, Dimethyl, â¦</li>
        </ul>
      </li>
      <li>Reporter ion MS2
        <ul>
          <li>TMT, iTRAQ
]</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>.footnote[
Watch: <a href="https://www.youtube.com/channel/UCKYzYTm1cnmc0CFAMhxDO8w">MaxQuant videos on YouTube</a>
]</p>

<p><span>Speaker Notes</span></p>

<p>MaxQuant is the most popular non-commercial software for quantitative proteomics experiments. It performs protein identification via its Andromeda search engine. Protein quantification of label-free and many label-based methods is supported as well. MaxQuant accepts raw data in vendor specific formats.</p>

<hr />

<h1 id="follow--up--statistical-analysis">Follow  up / Statistical Analysis</h1>

<p>.pull-left[
<strong>Typical follow up procedures:</strong></p>

<ul>
  <li>Visualization
    <ul>
      <li>E.g. venn-diagramm, volcano plot, heatmap, boxplots, histogram</li>
    </ul>
  </li>
  <li>
    <p>Protein-Protein interaction network analysis</p>
  </li>
  <li>
    <p>GO annotation and enrichment analysis</p>
  </li>
  <li>Differentially abundant protein analysis</li>
</ul>

<p>]</p>

<p>.pull-right[
.image-100[
 <img src="/training-material/topics/proteomics/images/intro-slides/statistical-analysis.png" alt="Graphic with several plot types." />
]
]</p>

<p>.footnote[
Exemplary visualizations taken from MÃ¼ller, 2018, Neoplasia
]</p>

<p><span>Speaker Notes</span></p>

<p>Typical follow up analyses include visualization, network and GO enrichment analysis. Finding differentially abundant proteins between different groups requires statistical analyses.</p>

<hr />

<h1 id="msstats-software">MSstats Software</h1>

<p>.left-column50[</p>

<p>.image-100[
  <img src="/training-material/topics/proteomics/images/intro-slides/msstats-paper.png" alt="screenshot of msstats paper" />
]</p>

<ul>
  <li>Two Bioconductor R Packages
    <ul>
      <li>MSstats &amp; MSstatsTMT</li>
    </ul>
  </li>
  <li>Popular open-source statistical proteomics software
]</li>
</ul>

<p>.right-column50[</p>

<ul>
  <li><strong>Import and pre-process</strong> results from common proteomics software:
    <ul>
      <li>MaxQuant, Skyline, OpenMS</li>
    </ul>
  </li>
  <li><strong>data processing</strong></li>
  <li><strong>Statistical modelling</strong>
    <ul>
      <li>linear models to detect differentially abundant proteins in label-free and isobaric labeled experiments</li>
    </ul>
  </li>
</ul>

<p>]</p>

<p>.footnote[
Watch: <a href="https://www.youtube.com/c/MayInstituteNEU/search?query=msstats">MSstats videos on YouTube</a>
]</p>

<p><span>Speaker Notes</span></p>

<p>MSstats is an open-source software for statistical modelling of quantitative proteomics data. It is compatible with complex designs of label-free and isobaric labeled quantification experiments. First several processing steps are performed. Then MSstats applies flexible linear models to detect differentially abundant proteins.</p>

<hr />

<h1 id="statistical-analysis-with-msstats">Statistical Analysis with MSstats</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/msstats_overview.png" alt="msstats takes output of quant. proteomics software and annotation files and converts them to msstats format. Removes/keeps features, peptides, and proteins and summarizes. This is passed to 'data processing' which has intensity transformation/normalisation, filtering of peptides/features, missing value imputation, and run level summarisation. This is passed to statistical modelling along with a comparison matrix and does group comparison." /></p>

<p><span>Speaker Notes</span></p>

<p>MSstats takes identified and quantified spectral peaks from common proteomics software such as MaxQuant as input. For DDA data MSstats starts with peptide level data. It applies several feature selection and processing steps in order to account for proteomics specific data properties. Afterwards MSstats calculates new protein abundances and performs statistical modelling on them.</p>

<hr />

<h1 id="annotation-file-with-experimental-design">Annotation file with experimental design</h1>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
      <th style="text-align: center">Â </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>Raw.file</strong></td>
      <td style="text-align: center"><strong>Condition</strong></td>
      <td style="text-align: center"><strong>BioReplicate</strong></td>
      <td style="text-align: center"><strong>Run</strong></td>
      <td style="text-align: center"><strong>IsotopeLabelType</strong></td>
    </tr>
    <tr>
      <td style="text-align: center">File1.raw.thermo</td>
      <td style="text-align: center">Cond1</td>
      <td style="text-align: center">1</td>
      <td style="text-align: center">1</td>
      <td style="text-align: center">L</td>
    </tr>
    <tr>
      <td style="text-align: center">File2.raw.thermo</td>
      <td style="text-align: center">Cond1</td>
      <td style="text-align: center">2</td>
      <td style="text-align: center">2</td>
      <td style="text-align: center">L</td>
    </tr>
    <tr>
      <td style="text-align: center">File3.raw.thermo</td>
      <td style="text-align: center">Cond1</td>
      <td style="text-align: center">3</td>
      <td style="text-align: center">3</td>
      <td style="text-align: center">L</td>
    </tr>
    <tr>
      <td style="text-align: center">File4.raw.thermo</td>
      <td style="text-align: center">Cond2</td>
      <td style="text-align: center">4</td>
      <td style="text-align: center">4</td>
      <td style="text-align: center">L</td>
    </tr>
    <tr>
      <td style="text-align: center">File5.raw.thermo</td>
      <td style="text-align: center">Cond2</td>
      <td style="text-align: center">5</td>
      <td style="text-align: center">5</td>
      <td style="text-align: center">L</td>
    </tr>
  </tbody>
</table>

<p><span>Speaker Notes</span></p>

<p>In addition to the results of the proteomics software an annotation file is needed as input. In this file the experimental design is described. It specifies conditions and biological and technical replicates. In case MaxQuant results are used as MSstats input, an additional column with the label type is needed. In a DDA experiment the value is L for all conditions.</p>

<hr />

<h1 id="conversion-from-maxquant-to-msstats-format">Conversion from MaxQuant to MSstats format</h1>

<p><img src="/training-material/topics/proteomics/images/intro-slides/conversion_msstats.png" alt="three files, max quant protein groups.txt, evidence.txt, and annotation file go into a large table with proteinA and several different run intensities." /></p>

<p><span>Speaker Notes</span></p>

<p>First, the input data is converted into an MSstats compatible table. For this step several parameters to filter and adjust the input data can be selected.</p>

<hr />

<h1 id="intensity-manipulations">Intensity manipulations</h1>

<ul>
  <li><strong>Log-transformation</strong>: log2 or log10</li>
  <li><strong>Intensity normalization</strong>: equalize medians or quantile normalization</li>
</ul>

<p><img src="/training-material/topics/proteomics/images/intro-slides/qc_plot.png" alt="box and whisker plot with ms runs along x, and log2 intensities along y. they are segmented into C1-C6." /></p>

<p><span>Speaker Notes</span></p>

<p>Log transformation brings the peptide intensities into a close to normal distribution. Normalization aims to make the intensities of different runs more comparable to each other. The default normalization method is called âequalize mediansâ. It assumes that the majority of proteins do not change across runs. It shifts all intensities of a run by a constant to obtain equal median intensities across runs.</p>

<hr />

<h1 id="peptide-filtering-and-missing-value-imputation">Peptide filtering and missing value imputation</h1>

<ul>
  <li><strong>Feature selection</strong>: which peptides should be kept for protein quantification</li>
  <li><strong>Missing value imputation</strong>: Imputation of NA and very low abundant intensities</li>
  <li><strong>Protein summarization</strong>: calculates new protein abundances after data processing via TMP model</li>
</ul>

<p><img src="/training-material/topics/proteomics/images/intro-slides/protein_table.png" alt="Heatmap of Protein1 (with 5 sub-peptides) under 2 conditions and three replicates. Squares are different colours." /></p>

<p><span>Speaker Notes</span></p>

<p>Feature selection allows the use of either all or only the most abundant peptides for protein summarization. The table represents the intensity values for the peptides of one protein. The dark grey fields represent missing intensity values of peptides. Missing values and noisy peptides with outliers are typical in label-free DDA datasets but influence protein summarization. For a reliable and robust statistical analysis, missing value imputation is recommended. Missing values in MaxQuant mean that they are missing because they were below the limit of detection. This means the values are not missing for random but for the reason of low abundance. Therefore, the values are only partially known and  called âcensored valuesâ. This may also be the case for very low intensity values, which might not be reliable and can be imputed. Censored intensity values are imputed via an accelerated failure time model. Alternatively, they may be replaced with a value obtained from the other measured intensities for the peptide and/or the run. Protein summarization is by default performed via Tukeyâs median polish for robust parameter estimation with median across rows and columns.</p>

<hr />

<h1 id="statistical-modelling">Statistical modelling</h1>

<ul>
  <li>Uses run-level summarized data for hypothesis testing</li>
  <li>Needs comparison matrix to specify comparisons</li>
  <li>Adjusts the linear model according to information from annotation file</li>
</ul>

<p>.pull-left[
|   |   |   |   |   |
|:-:|:-:|:-:|:-:|:-:|
| <strong>name</strong> | <strong>Cond1</strong> | <strong>Cond2</strong> | <strong>Cond3</strong> | <strong>Cond4</strong> |
| cond1-cond3 | 1 | 0 | -1 | 0 |
| cond1-cond4 | 1 | 0 | 0 | -1 |
| cond2-cond3 | 0 | 1 | -1 | 0 |
| cond2-cond4 | 0 | 1 | 0 | -1 |
]</p>

<p>.pull-right[
.image-100[
 <img src="/training-material/topics/proteomics/images/intro-slides/volcano_plot.png" alt="Volcano plot with log2 fold change plotted against -log10 adjusted p-value." />
]
]</p>

<p><span>Speaker Notes</span></p>

<p>The calculated run-level protein summaries are used for statistical group comparison. Any two conditions can be compared to find differentially abundant proteins between them. MSstats uses a family of linear mixed models for this. The model is automatically adjusted for the comparison type according to the information in the annotation file. This means MSstats accounts for technical replicates, paired designs or time course experiments automatically.</p>


	<hr />








<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</section>





        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/proteomics/tutorials/introduction/slides.html" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1705797054'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
