<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Antibiotic resistance detection</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:title" content="Galaxy Training: Antibiotic resistance detection" />
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:24 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Antibiotic resistance detection",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
  },
  "courseCode": "metagenomics / plasmid-metagenomics-nanopore / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Antibiotic resistance detection' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html",
  "timeRequired": "PT3H",
  "keywords": "nanopore, plasmids, assembly, amr",
  "description": "The questions this  addresses are:\n - How do I assemble a genome with Nanopore data?\n - How do I get more information about the structure of the genomes?\n - How do I get more information about the antimicrobial resistance genes?\n\n\\nThe objectives are:\n - Perform Quality control on your reads\n - Assemble a genome with Minimap2/Miniasm/Racon\n - Determine the structure of the genome(s)\n - Scan for antimicrobial resistance genes with Staramr\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Willem de Koning"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Willem de Koning"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ]
}
</script>

<section class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Antibiotic resistance detection</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/willemdek11/" class="contributor-badge contributor-willemdek11"><img src="https://avatars.githubusercontent.com/willemdek11?s=27" alt="Avatar">Willem de Koning</a>


<a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="https://avatars.githubusercontent.com/shiltemann?s=27" alt="Avatar">Saskia Hiltemann</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do I assemble a genome with Nanopore data?</p>
</li>
        
        <li><p>How do I get more information about the structure of the genomes?</p>
</li>
        
        <li><p>How do I get more information about the antimicrobial resistance genes?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Perform Quality control on your reads</p>
</li>
        
        <li><p>Assemble a genome with Minimap2/Miniasm/Racon</p>
</li>
        
        <li><p>Determine the structure of the genome(s)</p>
</li>
        
        <li><p>Scan for antimicrobial resistance genes with Staramr</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/metagenomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.3247504">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/workflows/" title="Workflows" alt="Antibiotic resistance detection workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            
                <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/main/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tours/" title="Galaxy interactive tour" >
    <i class="fas fa-magic" aria-hidden="true"></i> Tours
</a>

</li>
            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/metagenomics/plasmid-metagenomics-nanopore/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/metagenomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://nanopore.usegalaxy.eu/" title="NanoGalaxy">
            NanoGalaxy
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://plants.usegalaxy.eu/" title="Plant Anaylsis Workbench">
            Plant Anaylsis Workbench
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 24, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="overview">Overview</h1>

<p>Pervasive use (and misuse) of antibiotics for human disease treatment, as well as for various agricultural purposes, has resulted in the evolution of multi-drug resistant (MDR) pathogenic bacteria. The <a href="https://www.cdc.gov/drugresistance/">Center for Disease Control estimates</a> that in the U.S. alone, every year at least 2 million people get an antibiotic-resistant infection, and at least 23,000 people die. Antibiotic resistance poses a major public health challenge, and its causes and mitigations are widely studied.</p>

<p>Plasmids are small <span class="notranslate">DNA</span> molecules within a cell which are physically separated from chromosomal <span class="notranslate">DNA</span> and can replicate independently. They are most commonly found as small circular, double-stranded <span class="notranslate">DNA</span> molecules in bacteria.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/plasmids.png" alt="Depiction of plasmids. " loading="lazy" /></p>

<p>Plasmids are considered a major vector facilitating the transmission of drug resistant genes among bacteria via <a href="https://en.wikipedia.org/wiki/Horizontal_gene_transfer">horizontal transfer</a> (<span class="citation"><a href="#Beatson2014">Beatson and Walker 2014</a></span>, <span class="citation"><a href="#Smillie2010">Smillie <i>et al.</i> 2010</a></span>). Careful characterization of plasmids and other MDR mobile genetic elements is vital for understanding their evolution and transmission and adaptation to new hosts.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/bacterial_transformation.png" alt="Illustration of transformation of a bacteria to drug resistance. " loading="lazy" /></p>

<p>Due to the high prevalence of repeat sequences and inserts in plasmids, using traditional NGS short-read sequencing to assemble plasmid sequences is difficult and time-consuming. With the advent of third-generation single-molecule long-read sequencing technologies, full assembly of plasmid sequences is now possible.</p>

<p>In this tutorial we will recreate the analysis described in the paper by <span class="citation"><a href="#LiXie2018">Li <i>et al.</i> 2018</a></span> entitled <em>Efficient generation of complete sequences of MDR-encoding plasmids by rapid assembly of MinION barcoding sequencing data</em>. We will use data sequenced by the <a href="https://nanoporetech.com/">Nanopore</a> MinION sequencer.</p>

<p>The assembly will be performed with <strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Li2018">Li 2018</a></span>),
<strong>Miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Li2016">Li <i>et al.</i> 2015</a></span>), <strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Vaser2017">Vaser <i>et al.</i> 2017</a></span>) and <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Wick2017">Wick <i>et al.</i> 2017</a></span>).
The <span class="notranslate">downstream</span> analysis will use <strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#DeCoster2018">Coster <i>et al.</i> 2018</a></span>),
<strong>Bandage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Wick2015">Wick <i>et al.</i> 2015</a></span>), <strong>PlasFlow</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Krawczyk2018">Krawczyk <i>et al.</i> 2018</a></span>) and <strong>starmr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<a href="https://github.com/phac-nml/staramr">GitHub</a>).</p>

<p>A schematic view of the <span class="notranslate">workflow</span> we will perform in this tutorial is given below:</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Workflow.png" alt="Workflow representation of this tutorial. " loading="lazy" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#obtaining-and-preparing-data" id="markdown-toc-obtaining-and-preparing-data">Obtaining and preparing data</a>    <ol>
      <li><a href="#importing-the-data-into-galaxy" id="markdown-toc-importing-the-data-into-galaxy">Importing the data into <span class="notranslate">Galaxy</span></a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a>    <ol>
      <li><a href="#nanoplot-to-explore-data" id="markdown-toc-nanoplot-to-explore-data">NanoPlot to explore data</a></li>
    </ol>
  </li>
  <li><a href="#de-novo-assembly" id="markdown-toc-de-novo-assembly">De-novo Assembly</a>    <ol>
      <li><a href="#pairwise-alignment-using-minimap2" id="markdown-toc-pairwise-alignment-using-minimap2">Pairwise alignment using Minimap2</a></li>
      <li><a href="#ultrafast-de-novo-assembly-using-miniasm" id="markdown-toc-ultrafast-de-novo-assembly-using-miniasm">Ultrafast de novo assembly using Miniasm</a></li>
      <li><a href="#remapping-using-minimap2" id="markdown-toc-remapping-using-minimap2">Re<span class="notranslate">mapping</span> using Minimap2</a></li>
      <li><a href="#ultrafast-consensus-module-using-racon" id="markdown-toc-ultrafast-consensus-module-using-racon">Ultrafast consensus module using Racon</a></li>
      <li><a href="#visualize-assemblies-using-bandage" id="markdown-toc-visualize-assemblies-using-bandage">Visualize assemblies using Bandage</a></li>
      <li><a href="#optimizing-assemblies-using-unicycler" id="markdown-toc-optimizing-assemblies-using-unicycler">Optimizing assemblies using Unicycler</a></li>
    </ol>
  </li>
  <li><a href="#species-and-plasmids" id="markdown-toc-species-and-plasmids">Species and plasmids</a>    <ol>
      <li><a href="#prediction-of-plasmid-sequences-and-classes-using-plasflow" id="markdown-toc-prediction-of-plasmid-sequences-and-classes-using-plasflow">Prediction of plasmid sequences and classes using PlasFlow</a></li>
    </ol>
  </li>
  <li><a href="#antibiotic-resistance" id="markdown-toc-antibiotic-resistance">Antibiotic Resistance</a>    <ol>
      <li><a href="#scan-genome-contigs-for-antimicrobial-resistance-genes" id="markdown-toc-scan-genome-contigs-for-antimicrobial-resistance-genes">Scan genome contigs for antimicrobial resistance genes</a></li>
      <li><a href="#card-database" id="markdown-toc-card-database">CARD database</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>

<p>In this tutorial we use metagenomic Nanopore data, but similar pipelines can be used for other types of datasets or other long-read sequencing platforms.</p>

<h1 id="obtaining-and-preparing-data">Obtaining and preparing data</h1>

<p>We are interested in the reconstruction of full plasmid sequences and determining the presence of any antimicrobial resistance genes.
We will use the plasmid dataset created by <span class="citation"><a href="#LiXie2018">Li <i>et al.</i> 2018</a></span> for their evaluation of the efficiency of MDR plasmid sequencing by MinION platform. In the experiment, 12 MDR plasmid-bearing bacterial strains were selected for plasmid extraction, including <em>E. coli</em>, <em>S. typhimurium</em>, <em>V. parahaemolyticus</em>, and <em>K. pneumoniae</em>.</p>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Nanopore sequencing</h3>  <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>  <ol>  <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>  <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>  <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li></ol>  <p><img src="/training-material/topics/metagenomics/images/plasmid-metagenomics-nanopore/sequence_method.jpg" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://nanoporetech.com/sites/default/files/s3/white-papers/WGS_Assembly_white_paper.pdf?submissionGuid=40a7546b-9e51-42e7-bde9-b5ddef3c3512&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>  <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change inelectrical current is measured over the membrane of a flow cell. Whennucleotides pass the pores in the flow cell the current change is translated(basecalled) to nucleotides by a basecaller. A schematic overview is given inthe picture above.</p>  <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software ispresent on the devices. With basecalling the electrical signals are translatedto bases (A,T,G,C) with a quality score per base. The sequenced <span class="notranslate">DNA</span> strand willbe basecalled and this will form one read. Multiple <span class="notranslate">reads</span> will be stored in afastq file.</p></blockquote>

<h2 id="importing-the-data-into-galaxy">Importing the data into <span class="notranslate">Galaxy</span></h2>

<p>For this tutorial, in order to speed up the analysis time, we will use 6 of the 12 samples from the original study.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-obtaining-our-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Obtaining our data</h3>

  <ol>
    <li>
      <p>Make sure you have an empty analysis history. Give it a name.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li><strong>Import Sample Data</strong> <a href="https://doi.org/10.5281/zenodo.3247504"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.3247504.svg" alt="DOI. " loading="lazy" /></a>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3247504/files/RB01.fasta
https://zenodo.org/record/3247504/files/RB02.fasta
https://zenodo.org/record/3247504/files/RB04.fasta
https://zenodo.org/record/3247504/files/RB05.fasta
https://zenodo.org/record/3247504/files/RB10.fasta
https://zenodo.org/record/3247504/files/RB12.fasta
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
    </li>
    <li>
      <p><strong>Build a list collection</strong> containing all fasta files. Name it <code class="language-plaintext highlighter-rouge">Plasmids</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a dataset collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>    <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>  </li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="quality-control">Quality Control</h1>

<h2 id="nanoplot-to-explore-data">NanoPlot to explore data</h2>

<p>The first thing we want to do is to get a feeling for our input data and its quality. This is done
using the NanoPlot tool. This will create several plots, a statisical report and an HTML
report page.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/NanoPlot.png" alt="NanoPlot example. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-scripts-for-long-read-sequencing-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plotting scripts for long read sequencing data</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1"><strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select multifile mode”</em>: <code class="language-plaintext highlighter-rouge">batch</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Type of the file(s) to work on”</em>: <code class="language-plaintext highlighter-rouge">fasta</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“files”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection you just created</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">HTML report</code> gives an overview of various QC metrics for each sample. For example, it will
plot the read length distribution of each sample:
<img src="../../images/plasmid-metagenomics-nanopore/NanoPlot_output.png" alt="NanoPlot Output. " loading="lazy" /></p>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>What was the mean read length for this (RB01) sample?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>4906.3</p>

    <p>This can be determined by looking at the NanoStats or HTML output of NanoPlot RB01.</p>
  </blockquote>
</blockquote>

<p>For more information on the topic of quality control, please see our training materials
<a href="/training-material/topics/sequence-analysis/">here</a></p>

<h1 id="de-novo-assembly">De-novo Assembly</h1>

<h2 id="pairwise-alignment-using-minimap2">Pairwise alignment using Minimap2</h2>

<p>In this experiment we used Nanopore sequencing; this means sequencing results with long <span class="notranslate">reads</span>, and significant overlaps between those <span class="notranslate">reads</span>.
To find this overlap, Minimap2 is used. Minimap2 is a sequence alignment program that can be used for different
purposes, but in this case we’ll use it to find overlaps between long <span class="notranslate">reads</span> with an error rate up to ~15%.
Typical other use cases for Minimap2 include:</p>

<ol>
  <li><span class="notranslate">mapping</span> PacBio or Oxford Nanopore genomic <span class="notranslate">reads</span> to the human genome</li>
  <li>splice-aware alignment of PacBio Iso-Seq or Nanopore c<span class="notranslate">DNA</span> or Direct <span class="notranslate">RNA</span> <span class="notranslate">reads</span> against a reference genome</li>
  <li>aligning Illumina single- or paired-end <span class="notranslate">reads</span></li>
  <li>assembly-to-assembly alignment</li>
  <li>full-genome alignment between two closely related species with divergence below ~15%.</li>
</ol>

<p>Minimap2 is faster and more accurate than mainstream long-read mappers such as BLASR, BWA-MEM, NGMLR and GMAP and
therefore widely used for Nanopore alignment. Detailed evaluations of Minimap2 are available in
the Minimap2 publication (<span class="citation"><a href="#Li2018">Li 2018</a></span>).</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Minimap2.png" alt="Pairwise alignment. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-pairwise-sequence-alignment"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Pairwise sequence alignment</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Use the following data collection as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection we just created</li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Single or Paired-end <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select fastq dataset”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection</li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Oxford Nanopore all-vs-all overlap <span class="notranslate">mapping</span></code></li>
          </ul>
        </li>
        <li>In the section <strong>Set advanced output options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<p>This step maps the Nanopore sequence <span class="notranslate">reads</span> against itself to find overlaps. The result is a PAF file.
PAF is a text format describing the approximate <span class="notranslate">mapping</span> positions between two
set of sequences. PAF is TAB-delimited with each line consisting of the
following predefined fields:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: right">Col</th>
      <th style="text-align: center">Type</th>
      <th style="text-align: left">Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: right">1</td>
      <td style="text-align: center">string</td>
      <td style="text-align: left">Query sequence name</td>
    </tr>
    <tr>
      <td style="text-align: right">2</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Query sequence length</td>
    </tr>
    <tr>
      <td style="text-align: right">3</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Query start (0-based)</td>
    </tr>
    <tr>
      <td style="text-align: right">4</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Query end (0-based)</td>
    </tr>
    <tr>
      <td style="text-align: right">5</td>
      <td style="text-align: center">char</td>
      <td style="text-align: left">Relative strand: “+” or “-“</td>
    </tr>
    <tr>
      <td style="text-align: right">6</td>
      <td style="text-align: center">string</td>
      <td style="text-align: left">Target sequence name</td>
    </tr>
    <tr>
      <td style="text-align: right">7</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Target sequence length</td>
    </tr>
    <tr>
      <td style="text-align: right">8</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Target start on original strand (0-based)</td>
    </tr>
    <tr>
      <td style="text-align: right">9</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Target end on original strand (0-based)</td>
    </tr>
    <tr>
      <td style="text-align: right">10</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Number of residue matches</td>
    </tr>
    <tr>
      <td style="text-align: right">11</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left">Alignment block length</td>
    </tr>
    <tr>
      <td style="text-align: right">12</td>
      <td style="text-align: center">int</td>
      <td style="text-align: left"><span class="notranslate">Mapping</span> quality (0-255; 255 for missing)</td>
    </tr>
  </tbody>
</table>

<p>View the output of <strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of the collection against RB12, it should look something like this:</p>
<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>channel_100_69f2ea89-01c5-45f4-8e1b-55a09acdb3f5_template	4518	114	2613	+	channel_139_250c7e7b-f063-4313-8564-d3efbfa7e38d_template	3657	206	2732	273	2605	0	tp:A:S	cm:i:29	s1:i:240	dv:f:0.2016	rl:i:1516
channel_100_69f2ea89-01c5-45f4-8e1b-55a09acdb3f5_template	4518	148	1212	+	channel_313_35f447cb-7e4b-4c3d-977e-dc0de2717a4d_template	3776	2433	3450	218	1064	0	tp:A:S	cm:i:31	s1:i:210	dv:f:0.1291	rl:i:1516
channel_100_69f2ea89-01c5-45f4-8e1b-55a09acdb3f5_template	4518	251	1328	+	channel_313_a83f7257-52db-46e4-8e2a-1776500c7363_template	3699	2327	3382	208	1082	0	tp:A:S	cm:i:29	s1:i:203	dv:f:0.1363	rl:i:1516
</code></pre></div></div>

<h2 id="ultrafast-de-novo-assembly-using-miniasm">Ultrafast de novo assembly using Miniasm</h2>

<p>The mapped <span class="notranslate">reads</span> are ready to be assembled with <strong>Miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Li2016">Li <i>et al.</i> 2015</a></span>). Miniasm is a very fast Overlap Layout Consensus based de-novo assembler for noisy long <span class="notranslate">reads</span>.
It takes all-vs-all read self-<span class="notranslate">mapping</span>s (typically by <strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>) as input and outputs an assembly graph in the GFA format.</p>

<p>Different from mainstream assemblers, miniasm does not have a consensus step.
It simply concatenates pieces of read sequences to generate the final sequences.
The optimal case would be to recreate a complete chromosome or plasmid.
Thus the per-base error rate is similar to the raw input <span class="notranslate">reads</span>.</p>

<figure id="figure-1"><img src="../../images/plasmid-metagenomics-nanopore/Miniasm.png" alt="Pairwise alignment. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Sequencing and Assembly Schematic. A longer sequence is constructed from many smaller, overlapping fragments. These are aligned and reconstructed during assembly to produce the best guess assembly.</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-de-novo-assembly"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: De novo assembly</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/miniasm/miniasm/0.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/miniasm/miniasm/0.3+galaxy0"><strong>miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/miniasm/miniasm/0.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequence <span class="notranslate">Reads</span>”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“PAF file”</em>: <code class="language-plaintext highlighter-rouge">Output Minimap dataset collection</code> created by <strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">Assembly Graph</code> output file gives information about the steps taken in the assembly.</p>

<p>The output should look like:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>S	utg000001l	GAAATCATCAGGCGTTTTTCACGATATGGACGGGAAGATGCGGAAATAGGCAGGAGGACATAGAA [..]
a	utg000001l	0	channel_364_204a2254-2b6f-4f10-9ec5-6d40f0b870e4_template:101-4457	+	4357
</code></pre></div></div>

<h2 id="remapping-using-minimap2">Re<span class="notranslate">mapping</span> using Minimap2</h2>

<p>Re<span class="notranslate">mapping</span> is done with the original <span class="notranslate">reads</span>, using the Miniasm assembly as a reference, in order to improve the consensus base call per position. This is used by <strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for consensus construction. This is done as some <span class="notranslate">reads</span> which might not have mapped well during the consensus calling, will now map to your scaffold.</p>

<p>The Assembly graph created can be used for <span class="notranslate">mapping</span> again with Minimap2, but first the graph should be transformed to FASTA format.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-pairwise-sequence-alignment-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Pairwise sequence alignment</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.1"><strong>GFA to Fasta</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.1</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input GFA file”</em>: the <code class="language-plaintext highlighter-rouge">Assembly Graph</code> (collection) created by <strong>Miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>How many contigs do we have for the RB05 sample after de novo assembly?
<br /><br />
Hint: run <strong>Nanoplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output of <strong>GFA to Fasta</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>25</p>

          <p>This can be determined by looking at the NanoStats output of NanoPlot.</p>
        </blockquote>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy2"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">FASTA file</code> output from <strong>GFA to Fasta</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (collection)</li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Single or Paired-end <span class="notranslate">reads</span>”</em>: <code class="language-plaintext highlighter-rouge">single</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select fastq dataset”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> collection</li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select a profile of preset options”</em>:: <code class="language-plaintext highlighter-rouge">PacBio/Oxford Nanopore read to reference <span class="notranslate">mapping</span> (-Hk19)</code></li>
          </ul>
        </li>
        <li>In the section <strong>Set advanced output options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-selecting-a-datast-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a datast collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="ultrafast-consensus-module-using-racon">Ultrafast consensus module using Racon</h2>

<p>The mapped <span class="notranslate">reads</span> can be improved even more using <strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Vaser2017">Vaser <i>et al.</i> 2017</a></span>) to find a consensus sequence.
Racon is a standalone consensus module to correct raw contigs generated by rapid assembly methods which do not include a consensus step.
It supports data produced by both Pacific Biosciences and Oxford Nanopore Technologies.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Racon.png" alt="Consensus Module. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-consensus-module"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Consensus module</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.4.13" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/racon/racon/1.4.13"><strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/racon/racon/1.4.13</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequences”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Overlaps”</em>: the latest <code class="language-plaintext highlighter-rouge">PAF file</code> collection created by <strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Target sequences”</em>: the <code class="language-plaintext highlighter-rouge">FASTA file</code> collection created by <strong>GFA to Fasta</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The <strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output file gives the final contigs.</p>

<p>The output of RB04 should look something like:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;utg000001c LN:i:4653 RC:i:11 XC:f:0.888889
AATGCAGCTATGGCGCGTGCGGTGCCAAGAAAGCCCGCAGATATTCCGCTTCCTCGCTCATT [..]
</code></pre></div></div>

<h2 id="visualize-assemblies-using-bandage">Visualize assemblies using Bandage</h2>

<p>To get a sense of how well our data was assembled, and to determine whether the contigs are chomosomal or plasmid <span class="notranslate">DNA</span> (the former being linear sequences while plasmids are circular molecules), <strong>Bandage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> can give a clear view of the assembly.</p>

<p><strong>Bandage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Wick2015">Wick <i>et al.</i> 2015</a></span>) (a <strong>B</strong>ioinformatics <strong>A</strong>pplication for <strong>N</strong>avigating <strong>D</strong>e novo <strong>A</strong>ssembly <strong>G</strong>raphs <strong>E</strong>asily), is a program that creates visualisations of assembly graphs.
Sequence assembler programs (such as <strong>Miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Li2016">Li <i>et al.</i> 2015</a></span>), <strong>Velvet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Zerbino2008">Zerbino and Birney 2008</a></span>), <strong>SPAdes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Bankevich2012">Bankevich <i>et al.</i> 2012</a></span>), <strong>Trinity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Grabherr2011">Grabherr <i>et al.</i> 2011</a></span>) and <strong>MEGAHIT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <span class="citation"><a href="#Li2015">Li <i>et al.</i> 2015</a></span>) carry out assembly by building a graph, from which contigs are generated.</p>

<p>By visualizing these assembly graphs, Bandage allows users to better understand, troubleshoot, and improve the assemblies.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Bandage.png" alt="Bandage GUI. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualising-de-novo-assembly-graphs"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualising de novo assembly graphs</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2"><strong>Bandage image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Graphical Fragment Assembly”</em>: the <code class="language-plaintext highlighter-rouge">Assembly graph</code> collection created by <strong>Miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>Explore <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the output images</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

    <p>In how many samples were the full plasmid sequences assembled?</p>

    <p>Hint: what shape do you expect plasmid molecules to be?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>Ideally, we want to see circular assemblies, indicating the full plasmid sequence was resolved.
This is not the case for most of the samples, but we will improve our assemblies in the next section!</p>
    </blockquote>
  </blockquote>

</blockquote>

<p>For example, the assembly for sample RB01 looks something like this (your assembly will look a bit different due to randomness in several of the tools):</p>

<figure id="figure-2"><img src="../../images/plasmid-metagenomics-nanopore/bandage_pre_unicycler.png" alt="Bandage output for sample RB01. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Bandage output for sample RB01. Large fragments were assembled, but not the full circular plasmid molecules.</figcaption></figure>

<p>As you can see from these Bandage outputs, we were able to assemble our data into fairly large fragments, but were not quite successful in assembling the full (circular) plasmid sequences.</p>

<p>However, all the tools we used to do the assembly have many different parameters that we did not explore, and multiple rounds of <span class="notranslate">mapping</span> and cleaning could improve our data as well. Choosing these parameters carefully could potentially improve our assembly, but this is also a lot of work and not an easy task. This is where <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Wick2017">Wick <i>et al.</i> 2017</a></span>) can help us out.</p>

<h2 id="optimizing-assemblies-using-unicycler">Optimizing assemblies using Unicycler</h2>

<p>The assembly tools we used in this tutorial are all implemented in <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, which will repeatedly run these tools on your data using different parameter settings, in order to find the optimal assembly.</p>

<p><strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> has a couple of advantages over running the tools separately:</p>

<ol>
  <li>The first modification is to <strong>help circular replicons assemble</strong> into circular string graphs.</li>
  <li><strong>Racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> polishing is carried out in <strong>multiple rounds to improve the sequence accuracy</strong>. It will polish until the assembly stops improving, as measured by the agreement between the <span class="notranslate">reads</span> and the assembly.</li>
  <li>Circular replicons are ‘rotated’ (have their starting position shifted) between rounds of polishing to ensure that <strong>no part of the sequence is left unpolished</strong>.</li>
</ol>

<p>Let’s try it on our data!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-unicycler-assembly"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Unicycler assembly</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/unicycler/unicycler/0.4.8.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/unicycler/unicycler/0.4.8.0"><strong>Create assemblies with Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/unicycler/unicycler/0.4.8.0</span></span>  with the following parameters
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Paired or Single end data”</em>: <code class="language-plaintext highlighter-rouge">None</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select long <span class="notranslate">reads</span>. If there are no long <span class="notranslate">reads</span>, leave this empty”</em>: The <code class="language-plaintext highlighter-rouge">Plasmids</code> dataset collection</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2"><strong>Bandage image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Graphical Fragment Assembly”</em>: the <code class="language-plaintext highlighter-rouge">Final Assembly Graph</code> collection created by <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>
      <p>Examine <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the output images again</p>
    </li>
    <li>Use the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> to compare the two assemblies for sample <code class="language-plaintext highlighter-rouge">RB01</code>
      <ul>
        <li>Compare the <strong>Bandage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> images for our two assemblies:
          <ol>
            <li>The assembly we got from running <strong>minimap2, miniasm, racon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (first time we ran bandage)</li>
            <li>The assembly obtained with <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ol>
        </li>
        <li>Tip: Search your history for the term <code class="language-plaintext highlighter-rouge">bandage</code> to easily find the outputs from our two bandage runs</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-using-the-scratchbook-to-view-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using the Scratchbook to view multiple datasets</h3>  <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in <span class="notranslate">Galaxy</span>:</p>  <ol>  <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> on the top menu bar.    <ul>      <li>You should see a little checkmark on the icon now</li>    </ul>  </li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> to view the output    <ul>      <li>You should see the output in a window overlayed over <span class="notranslate">Galaxy</span></li>      <li>You can resize this window by dragging the bottom-right corner</li>    </ul>  </li>  <li><strong>Click</strong> outside the file to exit the Scratchbook</li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a second dataset from your history    <ul>      <li>You should now see a second window with the new dataset</li>      <li>This makes it easier to compare the two outputs</li>    </ul>  </li>  <li>Repeat this for as many files as you would like to compare</li>  <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> by clicking on the icon again</li></ol></blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-searching-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Searching your history</h3>  <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>  <ol>  <li>    <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>    <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>  </li>  <li><strong>Type</strong> a search term in this box    <ul>      <li>For example a tool name, or sample name</li>    </ul>  </li>  <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-clear</span> next to the search box</li></ol></blockquote>
    </li>
    <li>
      <p>Repeat this comparison for the other samples.</p>

      <blockquote class="question">
        <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>For which samples has the plasmid assembly improved?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>Exploring the outputs for all the samples reveals that many now display circular assemblies, indicating the full plasmids sequence was resolved.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The Assembly graph image of the RB01 assembly with <strong>miniasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> shows one unclear hypothetical plasmid, where the output of <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> shows two clear plasmids, as also shown by <span class="citation"><a href="#LiXie2018">Li <i>et al.</i> 2018</a></span>.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Bandage_output.png" alt="Bandage output. " loading="lazy" /></p>

<h1 id="species-and-plasmids">Species and plasmids</h1>

<h2 id="prediction-of-plasmid-sequences-and-classes-using-plasflow">Prediction of plasmid sequences and classes using PlasFlow</h2>

<p>To automatically determine whether the contigs represent chromosomal or plasmid <span class="notranslate">DNA</span>, <strong>PlasFlow</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="citation"><a href="#Krawczyk2018">Krawczyk <i>et al.</i> 2018</a></span>) can be used, also in the case where a full circular plasmid sequence was not assembled. Furthermore, it assigns the contigs to a bacterial class.</p>

<p><strong>PlasFlow</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is a set of scripts used for prediction of plasmid sequences in metagenomic contigs.
It relies on the neural network models trained on full genome and plasmid sequences and is able to differentiate between plasmids and chromosomes with accuracy reaching 96%.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/PlasFlow.png" alt="Pairwise alignment. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-prediction-of-plasmid-sequences"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Prediction of plasmid sequences</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/plasflow/PlasFlow/1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/plasflow/PlasFlow/1.0"><strong>PlasFlow</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/plasflow/PlasFlow/1.0</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequence <span class="notranslate">Reads</span>”</em>: the <code class="language-plaintext highlighter-rouge">Final Assembly</code> collection created by <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

    <p>What is the classification of contig_id 0 in RB10? (Hint: Check the probability table created by PlasFlow)</p>

    <blockquote class="solution">
      <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>plasmid.Proteobacteria</p>

      <p>This can be determined by looking at the 5th column of the probability table.</p>
    </blockquote>
  </blockquote>
</blockquote>

<p>The most important output of <strong>PlasFlow</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is a tabular file containing all predictions, consisting of several columns including:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>contig_id 	contig_name 	contig_length 	id 	label 	...
</code></pre></div></div>

<p>where:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">contig_id</code>is an inte<span class="notranslate">rna</span>l id of sequence used for the classification</li>
  <li><code class="language-plaintext highlighter-rouge">contig_name</code> is a name of contig used in the classification</li>
  <li><code class="language-plaintext highlighter-rouge">contig_length</code> shows the length of a classified sequence</li>
  <li><code class="language-plaintext highlighter-rouge">id</code> is an inte<span class="notranslate">rna</span>l id of a produced label (classification)</li>
  <li><code class="language-plaintext highlighter-rouge">label</code> is the actual classification</li>
  <li><code class="language-plaintext highlighter-rouge">...</code> represents additional columns showing probabilities of assignment to each possible class</li>
</ul>

<p>Additionally, PlasFlow produces FASTA files containing input sequences binned to plasmids, chromosomes and unclassified.</p>

<h1 id="antibiotic-resistance">Antibiotic Resistance</h1>

<h2 id="scan-genome-contigs-for-antimicrobial-resistance-genes">Scan genome contigs for antimicrobial resistance genes</h2>

<p>To determine whether the contigs contain antimircobial resistance genes (AMR) <a href="https://github.com/phac-nml/staramr">staramr</a> can be used.
<strong>Staramr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> scans bacterial genome contigs against both the <strong>ResFinder</strong> (<span class="citation"><a href="#Zankari2012">Zankari <i>et al.</i> 2012</a></span>), <strong>PointFinder</strong> (<span class="citation"><a href="#Zankari2017">Zankari <i>et al.</i> 2017</a></span>), and <strong>PlasmidFinder</strong> (<span class="citation"><a href="#Carattoli2014">Carattoli <i>et al.</i> 2014</a></span>) databases (used by the ResFinder webservice)
and compiles a summary report of detected antimicrobial resistance genes.</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/StarAmr.png" alt="Pairwise alignment. " loading="lazy" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-prediction-of-amr-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Prediction of AMR genes</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+galaxy2"><strong>staramr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.7.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“genomes”</em>: the <code class="language-plaintext highlighter-rouge">Final Assembly</code> collection created by <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

    <p>Which samples contained the resistance gene: dfrA17?</p>

    <p>Hint: Check the resfinder.tsv created by staramr</p>

    <blockquote class="solution">
      <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>RB01, RB02, and RB10</p>

      <p>This can be determined by looking at the 2nd column of the resfinder.tsv output (and the first column for the sample names).</p>
    </blockquote>
  </blockquote>
</blockquote>

<p>There are 5 different output files produced by <strong>staramr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:</p>

<ol>
  <li><code class="language-plaintext highlighter-rouge">summary.tsv</code>:  A summary of all detected AMR genes/mutations in each genome, one genome per line.</li>
  <li><code class="language-plaintext highlighter-rouge">resfinder.tsv</code>: A tabular file of each AMR gene and additional BLAST information from the <strong>ResFinder</strong> database, one gene per line.</li>
  <li><code class="language-plaintext highlighter-rouge">pointfinder.tsv</code>: A tabular file of each AMR point mutation and additional BLAST information from the <strong>PointFinder</strong> database, one gene per line.</li>
  <li><code class="language-plaintext highlighter-rouge">settings.txt</code>: The command-line, database versions, and other settings used to run <code class="language-plaintext highlighter-rouge">staramr</code>.</li>
  <li><code class="language-plaintext highlighter-rouge">results.xlsx</code>: An Excel sp<span class="notranslate">reads</span>heet containing the previous 4 files as separate worksheets.</li>
</ol>

<p>The summary file is most important and provides all the resistance genes found.</p>

<h2 id="card-database">CARD database</h2>
<p>To get more information about these antibiotic resistant genes, you can check the <a href="https://card.mcmaster.ca">CARD database</a> (<strong>C</strong>omprehensive <strong>A</strong>ntibiotic <strong>R</strong>esistance <strong>D</strong>atabase) (<span class="citation"><a href="#Jia2016">Jia <i>et al.</i> 2016</a></span>)</p>

<figure id="figure-3"><img src="../../images/plasmid-metagenomics-nanopore/card_database.png" alt="Screenshot of the CARD database. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Screenshot of the CARD database interface. CARD gives information about the antibiotic resistance genes, as well as links to relevant publications.</figcaption></figure>

<blockquote class="question">
  <h3 id="question-question-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>What is the resistance mechanism of the <em>dfrA17</em> gene?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>antibiotic target replacement</p>

    <p>This can be determined by searching for <a href="https://card.mcmaster.ca/ontology/39294">the gene on the CARD database</a>
<br /><br />
For more information about antibiotic resistance mechanisms, see <span class="citation"><a href="#Munita">Munita and Arias</a></span></p>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>
<p>You have now seen how to perform an assembly on Nanopore sequencing data, and classify the type and species of the sequences, as well as determined the presence of potential antibiotic resistance genes.</p>

<p>As for any analysis, there are many different tools that can do the job, and the tools presented here are just
one possible pipeline. Which tools are best for your specific data and research question depends on a number of
factors. For more information and comparisons between various tools, review papers such as <span class="citation"><a href="#DeMaio2019">Maio <i>et al.</i> 2019</a></span> and <span class="citation"><a href="#Jayakumar2017">Jayakumar and Sakakibara 2017</a></span>
may provide further insight.</p>

<p>You have worked your way through the following pipeline:</p>

<p><img src="../../images/plasmid-metagenomics-nanopore/Workflow.png" alt="Workflow representation of this tutorial. " loading="lazy" /></p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Minimap2, Miniasm, and Racon can be used for quickly assembling Nanopore data</p>
</li>
                        
                        <li><p>Unicycler can be used to optimize settings of assembly tools</p>
</li>
                        
                        <li><p>Nanopore sequencing is useful for reconstruction of genomes</p>
</li>
                        
                        <li><p>Antimicrobial resistance genes are detectable after fast assembly</p>
</li>
                        
                        <li><p>The CARD database is a useful resource describing antibiotic resistance genes</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Zerbino2008">Zerbino, D. R., and E. Birney, 2008 <b>Velvet: Algorithms for de novo short read assembly using de Bruijn graphs</b>. Genome Research 18: 821–829. <a href="https://doi.org/10.1101/gr.074492.107">10.1101/gr.074492.107</a></li>
<li id="Smillie2010">Smillie, C., M. P. Garcillan-Barcia, M. V. Francia, E. P. C. Rocha, and F. de la Cruz, 2010 <b>Mobility of Plasmids</b>. Microbiology and Molecular Biology Reviews 74: 434–452. <a href="https://doi.org/10.1128/mmbr.00020-10">10.1128/mmbr.00020-10</a></li>
<li id="Grabherr2011">Grabherr, M. G., B. J. Haas, M. Yassour, J. Z. Levin, D. A. Thompson <i>et al.</i>, 2011 <b>Full-length transcriptome assembly from RNA-Seq data without a reference genome</b>. Nature Biotechnology 29: 644–652. <a href="https://doi.org/10.1038/nbt.1883">10.1038/nbt.1883</a></li>
<li id="Bankevich2012">Bankevich, A., S. Nurk, D. Antipov, A. A. Gurevich, M. Dvorkin <i>et al.</i>, 2012 <b>SPAdes: A New Genome Assembly Algorithm and Its Applications to Single-Cell Sequencing</b>. Journal of Computational Biology 19: 455–477. <a href="https://doi.org/10.1089/cmb.2012.0021">10.1089/cmb.2012.0021</a></li>
<li id="Zankari2012">Zankari, E., H. Hasman, S. Cosentino, M. Vestergaard, S. Rasmussen <i>et al.</i>, 2012 <b>Identification of acquired antimicrobial resistance genes</b>. Journal of Antimicrobial Chemotherapy 67: 2640–2644. <a href="https://doi.org/10.1093/jac/dks261">10.1093/jac/dks261</a></li>
<li id="Beatson2014">Beatson, S. A., and M. J. Walker, 2014 <b>Tracking antibiotic resistance</b>. Science 345: 1454–1455. <a href="https://doi.org/10.1126/science.1260471">10.1126/science.1260471</a></li>
<li id="Carattoli2014">Carattoli, A., E. Zankari, Garcı́a-Fernández Aurora, M. V. Larsen, O. Lund <i>et al.</i>, 2014 <b>In SilicoDetection and Typing of Plasmids using PlasmidFinder and Plasmid Multilocus Sequence Typing</b>. Antimicrobial Agents and Chemotherapy 58: 3895–3903. <a href="https://doi.org/10.1128/aac.02412-14">10.1128/aac.02412-14</a></li>
<li id="Li2016">Li, D., C.-M. Liu, R. Luo, K. Sadakane, and T.-W. Lam, 2015 <b>MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</b>. Bioinformatics 31: 1674–1676. <a href="https://doi.org/10.1093/bioinformatics/btv033">10.1093/bioinformatics/btv033</a></li>
<li id="Li2015">Li, W., J. Köster, H. Xu, C.-H. Chen, T. Xiao <i>et al.</i>, 2015 <b>Quality control,  modeling,  and visualization of CRISPR screens with MAGeCK-VISPR</b>. 16: <a href="https://doi.org/10.1186/s13059-015-0843-6">10.1186/s13059-015-0843-6</a></li>
<li id="Wick2015">Wick, R. R., M. B. Schultz, J. Zobel, and K. E. Holt, 2015 <b>Bandage: interactive visualization ofde novogenome assemblies: Fig. 1.</b> Bioinformatics 31: 3350–3352. <a href="https://doi.org/10.1093/bioinformatics/btv383">10.1093/bioinformatics/btv383</a></li>
<li id="Jia2016">Jia, B., A. R. Raphenya, B. Alcock, N. Waglechner, P. Guo <i>et al.</i>, 2016 <b>CARD 2017: expansion and model-centric curation of the comprehensive antibiotic resistance database</b>. Nucleic Acids Research 45: D566–D573. <a href="https://doi.org/10.1093/nar/gkw1004">10.1093/nar/gkw1004</a></li>
<li id="Jayakumar2017">Jayakumar, V., and Y. Sakakibara, 2017 <b>Comprehensive evaluation of non-hybrid genome assembly tools for third-generation PacBio long-read sequence data</b>. Briefings in Bioinformatics 20: 866–876. <a href="https://doi.org/10.1093/bib/bbx147">10.1093/bib/bbx147</a></li>
<li id="Vaser2017">Vaser, R., I. Sović, N. Nagarajan, and M. Šikić, 2017 <b>Fast and accurate de novo genome assembly from long uncorrected reads</b>. Genome Research 27: 737–746. <a href="https://doi.org/10.1101/gr.214270.116">10.1101/gr.214270.116</a></li>
<li id="Wick2017">Wick, R. R., L. M. Judd, C. L. Gorrie, and K. E. Holt, 2017 <b>Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads</b>. PLOS Computational Biology 13: 1–22. <a href="https://doi.org/10.1371/journal.pcbi.1005595">10.1371/journal.pcbi.1005595</a></li>
<li id="Zankari2017">Zankari, E., R. Allesøe, K. G. Joensen, L. M. Cavaco, O. Lund <i>et al.</i>, 2017 <b>PointFinder: a novel web tool for WGS-based detection of antimicrobial resistance associated with chromosomal point mutations in bacterial pathogens</b>. Journal of Antimicrobial Chemotherapy 72: 2764–2768. <a href="https://doi.org/10.1093/jac/dkx217">10.1093/jac/dkx217</a></li>
<li id="DeCoster2018">Coster, W. D., S. D’Hert, D. T. Schultz, M. Cruts, and C. V. Broeckhoven, 2018 <b>NanoPack: visualizing and processing long-read sequencing data</b> (B. Berger, Ed.). Bioinformatics 34: 2666–2669. <a href="https://doi.org/10.1093/bioinformatics/bty149">10.1093/bioinformatics/bty149</a></li>
<li id="Krawczyk2018">Krawczyk, P. S., L. Lipinski, and A. Dziembowski, 2018 <b>PlasFlow: predicting plasmid sequences in metagenomic data using genome signatures</b>. Nucleic Acids Research 46: e35–e35. <a href="https://doi.org/10.1093/nar/gkx1321">10.1093/nar/gkx1321</a></li>
<li id="Li2018">Li, H., 2018 <b>Minimap2: pairwise alignment for nucleotide sequences</b> (I. Birol, Ed.). Bioinformatics 34: 3094–3100. <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a></li>
<li id="LiXie2018">Li, R., M. Xie, N. Dong, D. Lin, X. Yang <i>et al.</i>, 2018 <b>Efficient generation of complete sequences of MDR-encoding plasmids by rapid assembly of MinION barcoding sequencing data</b>. GigaScience 7: <a href="https://doi.org/10.1093/gigascience/gix132">10.1093/gigascience/gix132</a></li>
<li id="DeMaio2019">Maio, N. D., L. P. Shaw, A. Hubbard, S. George, N. Sanderson <i>et al.</i>, 2019 <b>Comparison of long-read sequencing technologies in the hybrid assembly of complex bacterial genomes</b>. <a href="https://doi.org/10.1101/530824">10.1101/530824</a></li>
<li id="Munita">Munita, J. M., and C. A. Arias <b>Mechanisms of Antibiotic Resistance</b>, pp. 481–511 <span style="font-style: normal">in</span> <i>Virulence Mechanisms of Bacterial Pathogens,  Fifth Edition</i>, <i>American Society of Microbiology</i>. <a href="https://doi.org/10.1128/microbiolspec.vmbf-0016-2015">10.1128/microbiolspec.vmbf-0016-2015</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Antibiotic resistance detection (Metagenomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Willem de Koning, Saskia Hiltemann, 2021 <b>Antibiotic resistance detection (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{metagenomics-plasmid-metagenomics-nanopore,
author = "Willem de Koning and Saskia Hiltemann",
title = "Antibiotic resistance detection (Galaxy Training Materials)",
year = "2021",
month = "03",
day = "24"
url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
