<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Identification of the binding sites of the Estrogen receptor</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1698452657' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1698452657' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Identification of the binding sites of the Estrogen receptor">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-02-17 10:40:02 +0000">
<meta name="DC.creator" content="Friederike Dündar">
<meta name="DC.creator" content="Anika Erxleben">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Vivek Bhardwaj">
<meta name="DC.creator" content="Fidel Ramirez">

        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Identification of the binding sites of the Estrogen receptor">
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-02-17 10:40:02 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Identification of the binding sites of the Estrogen receptor",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Under development",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Identification of the binding sites of the Estrogen receptor' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html",
  "timeRequired": "PT3H",
  "teaches": "Inspect read quality with FastQC\n - Map reads with Bowtie2\n - Assess the quality of an ChIP-seq experiments\n - Extract coverage files\n - Call enriched regions or peaks",
  "keywords": "epigenetics, ChIP-seq, work-in-progress",
  "description": "The questions this  addresses are:\n - How is raw ChIP-seq data processed and analyzed?\n - What are the binding sites of the Estrogen receptor?\n\n\nThe objectives are:\n - Inspect read quality with FastQC\n - Map reads with Bowtie2\n - Assess the quality of an ChIP-seq experiments\n - Extract coverage files\n - Call enriched regions or peaks\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/friedue/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/friedue/",
      "name": "Friederike Dündar",
      "image": "https://avatars.githubusercontent.com/friedue",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "name": "Anika Erxleben",
      "image": "https://avatars.githubusercontent.com/erxleben",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7427-6478",
      "orcid": "https://orcid.org/0000-0002-7427-6478"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "name": "Vivek Bhardwaj",
      "image": "https://avatars.githubusercontent.com/vivekbhr",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "name": "Fidel Ramirez",
      "image": "https://avatars.githubusercontent.com/fidelram",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "ChIP-seq"
    },
    {
      "@type": "Thing",
      "name": "work-in-progress"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1698452657'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Epigenetics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/epigenetics/faqs/" title="Check our FAQs for the Epigenetics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-epigenetics">
    <h1 data-toc-skip>Identification of the binding sites of the Estrogen receptor</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/friedue/" class="contributor-badge contributor-friedue"><img src="https://avatars.githubusercontent.com/friedue?s=36" alt="Avatar" width="36" height="36">Friederike Dündar</a><a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge contributor-erxleben"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/erxleben?s=36" alt="Avatar" width="36" height="36">Anika Erxleben</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Avatar" width="36" height="36">Bérénice Batut</a><a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge contributor-vivekbhr"><img src="https://avatars.githubusercontent.com/vivekbhr?s=36" alt="Avatar" width="36" height="36">Vivek Bhardwaj</a><a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge contributor-fidelram"><img src="https://avatars.githubusercontent.com/fidelram?s=36" alt="Avatar" width="36" height="36">Fidel Ramirez</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How is raw ChIP-seq data processed and analyzed?</p>
</li>
        
        <li><p>What are the binding sites of the Estrogen receptor?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Inspect read quality with FastQC</p>
</li>
        
        <li><p>Map reads with Bowtie2</p>
</li>
        
        <li><p>Assess the quality of an ChIP-seq experiments</p>
</li>
        
        <li><p>Extract coverage files</p>
</li>
        
        <li><p>Call enriched regions or peaks</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.892432">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/workflows/" title="Identification of the binding sites of the Estrogen receptor workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://galaxy.mesocentre.uca.fr" title="">
			Galaxy@AuBi
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 17, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00138">https://gxy.io/GTN:T00138</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>This exercise uses the dataset from the Nature publication by <a href="https://www.ncbi.nlm.nih.gov/pubmed/22217937">Ross-Inness et al., 2012</a>.
The goal of this article was to identify the binding sites of the Estrogen receptor, a transcription factor known to be associated with different types of breast cancer.</p>

<p>To this end, ChIP-seq was performed in breast cancer cells from 4 patients of different outcomes (good and poor). For each ChIP-seq experiment there is a matching technical control, i.e. there are 8 samples in total:</p>

<table>
  <thead>
    <tr>
      <th>Patient</th>
      <th>Outcome</th>
      <th>Treatment</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Patient 1</td>
      <td>Good</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 1</td>
      <td>Good</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 2</td>
      <td>Good</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 2</td>
      <td>Good</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 3</td>
      <td>Poor</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 3</td>
      <td>Poor</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 4</td>
      <td>Poor</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 4</td>
      <td>Poor</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
  </tbody>
</table>

<p>Half of which are the so-called ‘input’ samples for which the same treatment as the ChIP-seq samples was done except for the immunoprecipitation step.
The input files are used to identify potential sequencing bias, like open chromatin or GC bias.</p>

<p>Because of the long processing time for the large original files, we have downsampled the original data and provide already processed data for subsequent steps.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#step-1-quality-control-and-treatment-of-the-sequences" id="markdown-toc-step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</a></li>
  <li><a href="#step-2-mapping-of-the-reads" id="markdown-toc-step-2-mapping-of-the-reads">Step 2: Mapping of the reads</a>    <ol>
      <li><a href="#visualization-using-a-genome-browser" id="markdown-toc-visualization-using-a-genome-browser">Visualization using a Genome Browser</a></li>
      <li><a href="#inspection-of-the-sam-format" id="markdown-toc-inspection-of-the-sam-format">Inspection of the SAM format</a></li>
    </ol>
  </li>
  <li><a href="#step-3-chip-seq-quality-control" id="markdown-toc-step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</a>    <ol>
      <li><a href="#correlation-between-samples" id="markdown-toc-correlation-between-samples">Correlation between samples</a></li>
      <li><a href="#ip-strength-estimation" id="markdown-toc-ip-strength-estimation">IP strength estimation</a></li>
    </ol>
  </li>
  <li><a href="#step-4-normalization" id="markdown-toc-step-4-normalization">Step 4: Normalization</a>    <ol>
      <li><a href="#generation-of-coverage-files-normalized-by-sequencing-depth" id="markdown-toc-generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</a></li>
      <li><a href="#generation-of-input-normalized-coverage-files-and-their-visualization" id="markdown-toc-generation-of-input-normalized-coverage-files-and-their-visualization">Generation of input-normalized coverage files and their visualization</a></li>
    </ol>
  </li>
  <li><a href="#step-5-detecting-enriched-regions-peak-calling" id="markdown-toc-step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</a></li>
  <li><a href="#step-6-plot-the-signal-on-the-peaks-between-samples" id="markdown-toc-step-6-plot-the-signal-on-the-peaks-between-samples">Step 6: Plot the signal on the peaks between samples</a>    <ol>
      <li><a href="#computematrix" id="markdown-toc-computematrix">computeMatrix</a></li>
      <li><a href="#plotheatmap" id="markdown-toc-plotheatmap">plotHeatmap</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
  <li><a href="#additional-exercise-if-you-have-finished-all-above" id="markdown-toc-additional-exercise-if-you-have-finished-all-above">Additional exercise (if you have finished all above)</a>    <ol>
      <li><a href="#additional-quality-control--gc-bias-assessment" id="markdown-toc-additional-quality-control--gc-bias-assessment">Additional Quality control : GC bias assessment</a></li>
      <li><a href="#plotting-heatmap-from-multiple-samples-with-clustering" id="markdown-toc-plotting-heatmap-from-multiple-samples-with-clustering">Plotting heatmap from multiple samples with clustering</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</h1>

<p>The first step of any ChIP-Seq data analysis is quality control of the raw sequencing data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> from <a href="https://zenodo.org/record/892432/files/patient1_input_good_outcome.fastq">Zenodo</a> or from the data library into the history</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data via links<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
          <li>Click on the <code class="language-plaintext highlighter-rouge">pencil</code> icon once the file is imported</li>
          <li>Click on <strong>Datatype</strong> in the central panel</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code> as <strong>New Type</strong></li>
        </ul>
      </blockquote>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Go into “Shared data” (top panel) then “Data libraries”</li>
          <li>Click on “Training data” and then “Analyses of ChIP-Seq data”</li>
          <li>Select interesting file</li>
          <li>Click on “Import selected datasets into history”</li>
          <li>Import in the history</li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Inspect the file by clicking on the <code class="language-plaintext highlighter-rouge">eye</code> icon</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How are the DNA sequences stored?</li>
          <li>What are the other entries?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>The DNA sequences are stored in the second line of every 4-line group</li>
            <li>This file is called a <a href="https://en.wikipedia.org/wiki/FASTQ_format">FastQ file</a>. It stores sequence information and quality information. Each sequence is represented by a group of 4 lines with the 1st line being the sequence id, the second the sequence of nucleotides, the third a transition line and the last one a sequence of quality score for each nucleotide.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with</p>
      <ul>
        <li>“Short read data from your current history” to the imported file</li>
      </ul>

      <p>Inspect the generated files</p>
    </li>
  </ol>
</blockquote>

<p>It is often necessary to trim sequenced read, for example, to get rid of bases that were sequenced with high uncertainty (= low quality bases).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control</div>

  <ol>
    <li>Run <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Is this library paired- or single-end?” to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>“Reads in FASTQ format” to the imported file</li>
        <li>“Trim Galore! advanced settings” to <code class="language-plaintext highlighter-rouge">Full parameter list</code></li>
        <li>“Trim low-quality ends from reads” to <code class="language-plaintext highlighter-rouge">15</code></li>
        <li>“Overlap with adapter sequence required to trim a sequence” to <code class="language-plaintext highlighter-rouge">3</code></li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-from-a-data-library-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>If your FASTQ files cannot be selected, you might check whether their format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger</code>). You can edit the data type by clicking on the <code class="language-plaintext highlighter-rouge">pencil</code> symbol.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="step-2-mapping-of-the-reads">Step 2: Mapping of the reads</h1>

<p>In order to figure where the sequenced DNA fragments originated from in the genome, the short reads must be aligned to the reference genome. This is equivalent to solving a jigsaw puzzles, but unfortunately, not all pieces are unique. In principle, you could do a BLAST analysis to figure out where the sequenced pieces fit best in the known genome. Aligning millions of short sequences this way may, however, take a couple of weeks.</p>

<h3 id="running-bowtie2">Running Bowtie2</h3>

<p>Nowadays, there are many read alignment programs for shotgun sequenced DNA, Bowtie2 being one of them.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping</div>

  <ol>
    <li><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Is this single or paired library” to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>“FASTA/Q file” to the Trim Galore! output with the trimmed reads</li>
        <li>“Will you select a reference genome from your history or use a built-in index?” to <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code></li>
        <li>“Select reference genome” to <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg18</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How many reads where mapped?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>This information can be accessed by clicking on the resulting history entry. You can see some basic mapping statistics once the alignment is completed. 16676 (66.96%) were mapped exactly 1 time and 7919 (31.80%) aligned &gt;1 times. The overall alignment rate is then 98.76%.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The read alignment step with bowtie2 resulted in a compressed, binary file (BAM) that is not human-readable. It’s like the zipped version of a text file.</p>

<p>We will show you two ways to inspect the file:</p>

<ol>
  <li>Visualization using a Genome Browser</li>
  <li>Converting the binary format into its text file equivalent</li>
</ol>

<h2 id="visualization-using-a-genome-browser">Visualization using a Genome Browser</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualization-of-the-reads-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualization of the reads in IGV</div>

  <ol>
    <li>Click on the <code class="language-plaintext highlighter-rouge">display with IGV local</code> to load the reads into the IGV browser</li>
    <li>Zoom at the start of chromosome 11 (or <code class="language-plaintext highlighter-rouge">chr11:1,562,200-1,591,483</code>)</li>
  </ol>
</blockquote>

<p>The reads have a direction: they are mapped to the forward or reverse strand, respectively. When hovering over a read, extra information is displayed</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Some reads have colored lines included. What is this?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>Try to zoom in in one of those lines and you will see the answer!</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Because the number of reads over a region can be quite large, the IGV browser by default only allows to see the reads that fall into a small window. This behaviour can be changed in the IGV preferences panel.</p>
</blockquote>

<h2 id="inspection-of-the-sam-format">Inspection of the SAM format</h2>

<p>As mentioned above, you can convert the binary BAM file into a simple (but large!) text file, which is called a <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">SAM</a> (Sequence Alignment Map) file.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-conversion-into-a-sam-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Conversion into a SAM file</div>

  <ol>
    <li><strong>BAM-to-SAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“BAM File to Convert” to the file generated by Bowtie2</li>
        <li>“Header options” to <code class="language-plaintext highlighter-rouge">Include header in SAM output</code></li>
      </ul>
    </li>
    <li>Inspect the file by clicking on <code class="language-plaintext highlighter-rouge">eye</code> icon</li>
  </ol>

</blockquote>

<p>A SAM file is a file with</p>

<ul>
  <li>A header with the chromosome names and lengths</li>
  <li>A file content as a tabular file with the location and other information of each read found in the FASTQ file and the mapping information</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Which information do you find in a SAM/BAM file? What is the additional information compared to a FASTQ file.</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>Sequences and Quality information, like FASTQ</li>
      <li>Mapping information; Location of the read on the chromosome; Mapping quality …</li>
      <li>For more information please see <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">https://en.wikipedia.org/wiki/SAM_(file_format)</a></li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</h1>

<p>We already checked the quality of the raw sequencing reads in the first step.
Now we would like to test the quality of the ChIP-seq preparation, to know if your ChIP-seq samples are more enriched than the control (input) samples.</p>

<h2 id="correlation-between-samples">Correlation between samples</h2>

<p>To assess the similarity between the replicates of the ChIP and the input, respectively, it is a common technique to calculate the correlation of
read counts on different regions for all different samples.
We expect that the replicates of the ChIP-seq experiments should be clustered more closely to each other than the replicates of the input sample.
That is, because the input samples should not have enriched regions included - remember the immuno-precipitation step was skiped during the sample preparation.</p>

<p>To compute the correlation between the samples we are going to to use the QC modules of deepTools (http://deeptools.readthedocs.io/), a software package for the QC, processing and analysis of NGS data. Before computing the correlation a time consuming step is required, which is to compute the read coverage over a large number of regions from each BAM file. For this we will use the tool <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<p>To do that, we need at first to catch up for all our samples and re-run the previous steps (quality control and mapping) on each sample.
To save time, we already did that and we can now work directly on the BAM files of the 8 samples</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-correlation-between-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Correlation between samples</div>

  <ol>
    <li>Create a new history</li>
    <li>Import the 8 BAM files from <a href="https://doi.org/10.5281/zenodo.892432">Zenodo</a> or from the data library into the history
      <ul>
        <li><a href="https://zenodo.org/record/892432/files/patient1_ChIP_ER_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient1_input_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient1_input_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient2_ChIP_ER_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient2_input_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient2_input_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient3_ChIP_ER_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient3_ChIP_ER_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient3_input_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient3_input_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient4_ChIP_ER_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient4_ChIP_ER_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient4_input_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient4_input_poor_outcome.bam</code></a></li>
      </ul>
    </li>
    <li><strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Sample order matters” to <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>“Bam file” to the 8 imported BAM files</li>
        <li>“Choose computation mode” to <code class="language-plaintext highlighter-rouge">Bins</code></li>
        <li>
          <p>“Bin size in bp” to <code class="language-plaintext highlighter-rouge">100</code></p>

          <p>This corresponds to the length of the fragments that were sequenced; it is not the read length!</p>
        </li>
        <li>“Distance between bins” to <code class="language-plaintext highlighter-rouge">500000</code> (to reduce the computation time for the tutorial)</li>
        <li>“Region of the genome to limit the operation to” to <code class="language-plaintext highlighter-rouge">chr1</code> (to reduce the computation time for the tutorial)</li>
      </ul>

      <p>Using these parameters, the tool will take bins of 100 bp separated by 500,000. For each bin the overlapping reads in each sample will be computed</p>
    </li>
    <li><strong>plotCorrelation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Matrix file from the multiBamSummary tool” to the generated multiBamSummary output</li>
      </ul>

      <p>To compute and visualize the sample correlation we use plotCorrelation from deepTools. This is a fast process that allows the user to quickly try different color combinations and outputs. Feel free to try different parameters.</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p><a href="../../images/estrogen-receptor-binding-site-identification/plotCorrelation_output.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/estrogen-receptor-binding-site-identification/plotCorrelation_output.png" alt="Output for plotCorrelation with the correlation scores between the 8 samples. " width="1100" height="950" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Correlation scores between the 8 samples&lt;/figcaption&gt;</figure></a></p>

  <ol>
    <li>How are your samples clustered? Does that correspond to your expectations?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>Badly, this is due to the subsampling of our data. You should expect better results from your real data.</li>
      <li>Expected is that your inputs and your ChIP samples cluster together.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>More information on these two tools can be found at the <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation page</a>.</p>
</blockquote>

<h2 id="ip-strength-estimation">IP strength estimation</h2>

<p>To evaluate the quality of the immuno-precipitation step, we can compute the IP strength. It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample. To do that we take the data for one Patient and compare the input sample and the ChIP-seq sample.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>For more information on how the IP strength is estimated, you can check the <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation page</a>.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-ip-strength-estimation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: IP strength estimation</div>

  <ol>
    <li><strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Sample order matters” to <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>“Bam file” to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>“Region of the genome to limit the operation to” to <code class="language-plaintext highlighter-rouge">chr1</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The plotFingerprint tool generates a fingerprint plot. You need to intepret it to know the IP strength. The <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation</a> explains it clearly:</p>

<p><a href="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" alt="A guide to interpret a fingerprint plot. " width="1436" height="722" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> How to interpret a fingerprint plot? Image extracted from the deepTools documentation&lt;/figcaption&gt;</figure></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p><a href="../../images/estrogen-receptor-binding-site-identification/plotFingerprint_output.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/estrogen-receptor-binding-site-identification/plotFingerprint_output.png" alt="Output for plotFingerprint with the fingerprint plot to estimate the IP strength. " width="640" height="480" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Fingerprint plot for the Patient 1 to estimate the IP strength&lt;/figcaption&gt;</figure></a></p>

  <ol>
    <li>What do you think about the quality of the IP for this experiment?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The difference between input and ChIP signal is not totally clear. 20% of chromosome 1 are not sequenced
 at all and the ChIP signal is only slightly more enrichted then the input.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optional-ip-strength-estimation-other-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: (Optional) IP strength estimation (other samples)</div>

  <ol>
    <li>Run the same analysis on data of the 3 other patients</li>
  </ol>
</blockquote>

<h1 id="step-4-normalization">Step 4: Normalization</h1>

<p>We would like to know where the binding sites of the estrogen receptor are located. For this we need
to extract which parts of the genome have been enriched (more reads mapped) within the samples that underwent immunoprecipitation.</p>

<p>For the normalization we have two options.</p>

<ol>
  <li>Normalization by sequencing depth</li>
  <li>Normalization by input file</li>
</ol>

<h2 id="generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</h2>

<p>We first need to make the samples comparable. Indeed, the different samples have usually a different sequencing depth, i.e. a different number of reads.
These differences can bias the interpretation of the number of reads mapped to a specific genome region.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-coverage-file-normalization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Coverage file normalization</div>

  <ol>
    <li><strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“BAM file” to “Multiple datasets”: <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What is the output of this tool?</li>
          <li>How many reads has been mapped on chr2 for the input and for the ChIP-seq samples?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>This tool estimates how many reads mapped to which chromosome. Furthermore, it tells the chromosome lengths and naming convention (with or without ‘chr’ in the beginning)</li>
            <li>1,089,370 for ChIP-seq samples and 1,467,480 for the input</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“BAM file” to “Multiple datasets”: <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>“Bin size in bases” to <code class="language-plaintext highlighter-rouge">25</code></li>
        <li>“Scaling/Normalization method” to <code class="language-plaintext highlighter-rouge">Normalize coverage to 1x</code></li>
        <li>“Effective genome size” to <code class="language-plaintext highlighter-rouge">hg19 (2451960000)</code></li>
        <li>“Coverage file format” to <code class="language-plaintext highlighter-rouge">bedgraph</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What are the different columns of a <code class="language-plaintext highlighter-rouge">bedgraph</code> file?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>chrom, chromStart, chromEnd and a data value</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but to generate a <code class="language-plaintext highlighter-rouge">bigWig</code> output file</li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect both signal coverages (input and ChIP samples) in IGV</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What is a bigWig file?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>A bigWig file is a compressed bedgraph file. Similar in relation as BAm to SAM, but this time just for coverage data. This means bigWig and bedgraph
files are much smaller than BAM or SAM files.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="generation-of-input-normalized-coverage-files-and-their-visualization">Generation of input-normalized coverage files and their visualization</h2>

<p>To extract only the information induced by the immunoprecipitation, we normalize for each patient the coverage file for the sample that underwent immunoprecipitation by the coverage file for the input sample. Here we use the tool bamCompare which compare 2 BAM files while caring for sequencing depth normalization.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generation-of-input-normalized-coverage-files"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generation of input-normalized coverage files</div>

  <ol>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“First BAM file (e.g. treated sample)” to <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>“Second BAM file (e.g. control sample)” to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>“Bin size in bases” to <code class="language-plaintext highlighter-rouge">50</code></li>
        <li>“How to compare the two files” to <code class="language-plaintext highlighter-rouge">Compute log2 of the number of reads ratio</code></li>
        <li>“Coverage file format” to <code class="language-plaintext highlighter-rouge">bedgraph</code></li>
        <li>
          <p>“Region of the genome to limit the operation to” to <code class="language-plaintext highlighter-rouge">chr11</code> (to reduce the computation time for the tutorial)</p>

          <blockquote class="question">
            <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

            <ol>
              <li>What does mean a positive or a negative value in the 4th column?</li>
            </ol>

            <blockquote class="solution">
              <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
              <ol>
                <li>The 4th column contains the log2 of the number of reads ratio between the ChIP-seq sample and the input sample. A positive value means that the coverage on the portion is more important in the ChIP-seq sample than in the input sample</li>
              </ol>
            </blockquote>
          </blockquote>
        </li>
      </ul>
    </li>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but to generate a <code class="language-plaintext highlighter-rouge">bigWig</code> output file</li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect the log2 ratio</p>

      <p>Remember that the bigWig file contains only the signal on chromosome 11!</p>
    </li>
  </ol>

</blockquote>

<h1 id="step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</h1>

<p>We can also call the enriched regions, or peaks, found in the ChIP-seq samples.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-peak-calling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Peak calling</div>

  <ol>
    <li><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“ChIP-Seq Treatment File” to <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>“ChIP-Seq Control File” to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>“Effective genome size” to <code class="language-plaintext highlighter-rouge">H. sapiens (2.7e9)</code></li>
        <li>“Outputs” to <code class="language-plaintext highlighter-rouge">Summary page (html)</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>The advanced options may be adjusted, depending of the samples.
If your ChIP-seq experiment targets regions of broad enrichment, <em>e.g.</em> non-punctuate histone modifications, select calling of broad regions.
If your sample has a low duplication rate (<em>e.g.</em> below 10%), you might keep all duplicate reads (tags). Otherwise, you might use the ‘auto’ option to estimate the maximal allowed number of duplicated reads per genomic location.</p>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect with the signal coverage and log2 ratio tracks</li>
  </ol>

</blockquote>

<p>The called peak regions can be filtered by, <em>e.g.</em> fold change, FDR and region length for further downstream analysis.</p>

<h1 id="step-6-plot-the-signal-on-the-peaks-between-samples">Step 6: Plot the signal on the peaks between samples</h1>

<p>Plotting your region of interest will involve using two tools from the <strong>deepTools</strong> suite.</p>
<ul>
  <li>computeMatrix : Computes the signal on given regions, using the bigwig coverage files from different samples.</li>
  <li>plotHeatmap : Plots heatMap of the signals using the computeMatrix output.</li>
</ul>

<p>ptionally, you can also use <code class="language-plaintext highlighter-rouge">plotProfile</code>to create a profile plot using to computeMatrix output.</p>

<h2 id="computematrix">computeMatrix</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualization-of-the-coverage"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualization of the coverage</div>

  <ol>
    <li><strong>UCSC Main</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“assembly” to <code class="language-plaintext highlighter-rouge">hg18</code></li>
        <li>“track” to <code class="language-plaintext highlighter-rouge">RefSeq genes</code></li>
        <li>“region” to <code class="language-plaintext highlighter-rouge">position</code> with <code class="language-plaintext highlighter-rouge">chr11</code></li>
        <li>“output format” to <code class="language-plaintext highlighter-rouge">BED</code></li>
        <li>“Send output to” to <code class="language-plaintext highlighter-rouge">Galaxy</code></li>
      </ul>
    </li>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Regions to plot” to the imported UCSC file</li>
        <li>“Score file” to the bigwig file generated by bamCompare</li>
        <li>
          <p>“computeMatrix has two main output options” to <code class="language-plaintext highlighter-rouge">scale-regions</code></p>

          <p>This option stretches or shrinks all regions in the BED file (here: genes) to the same length (bp) as indicated by the user</p>
        </li>
        <li>“Show advanced options” to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>“Convert missing values to 0?” to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This tool prepares a file with scores per genomic region, which is required as input for the next tool.</p>
    </li>
  </ol>

</blockquote>

<h2 id="plotheatmap">plotHeatmap</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualization-of-the-coverage-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualization of the coverage</div>
  <ol>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Matrix file from the computeMatrix tool” to the generated matrix</li>
        <li>“Show advanced options” to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>“Did you compute the matrix with more than one groups of regions?” to the correct setting</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p><a href="../../images/estrogen-receptor-binding-site-identification/tutorial-scheme.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/estrogen-receptor-binding-site-identification/tutorial-scheme.png" alt="Summary of the different steps of the tutorial and the generated files. " width="3793" height="3212" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Different steps of the tutorials with the generated files&lt;/figcaption&gt;</figure></a></p>

<h1 id="additional-exercise-if-you-have-finished-all-above">Additional exercise (if you have finished all above)</h1>

<h2 id="additional-quality-control--gc-bias-assessment">Additional Quality control : GC bias assessment</h2>

<p>A common problem of PCR-based protocols is the observation that GC-rich regions tend to be amplified more readily than GC-poor regions.
We need to check that our samples do not have more reads from regions of the genome with high GC.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>GC bias was for many years a big problem, but recent advances in sample preparation have solved this problem to a degree that you can skip this step more often.</p>
</blockquote>

<p>For practical reasons, we will focus here only on one of our BAM files. With real data you should repeat these steps for all your samples.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Can you guess why it makes more sense to check the input files for GC bias?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We only want to assess the bias induced by the PCR-based protocols. This is not possible with the ChIP samples, as the enriched regions (binding sites) can have a potential GC enrichment on their own.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-gc-bias-assessment"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: GC bias assessment</div>

  <ol>
    <li><strong>computeGCbias</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“BAM file” to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>“Reference genome” to <code class="language-plaintext highlighter-rouge">locally cached</code></li>
        <li>“Using reference genome” to <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg18</code></li>
        <li>“Effective genome size” to <code class="language-plaintext highlighter-rouge">hg19 (2451960000)</code></li>
        <li>“Fragment length used for the sequencing” to <code class="language-plaintext highlighter-rouge">300</code></li>
        <li>“Region of the genome to limit the operation to” to <code class="language-plaintext highlighter-rouge">chr1</code> (to reduce the computation time for the tutorial)</li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p><a href="../../images/estrogen-receptor-binding-site-identification/computeGCBias_output.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/estrogen-receptor-binding-site-identification/computeGCBias_output.png" alt="Output for computeGCbias with the GC bias estimation. " width="590" height="790" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Estimation of the GC bias for the input sample for the Patient 1&lt;/figcaption&gt;</figure></a></p>

        <ol>
          <li>Does this dataset have a GC bias?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>There is no significantly more reads in the GC-rich regions.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="plotting-heatmap-from-multiple-samples-with-clustering">Plotting heatmap from multiple samples with clustering</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-multiple-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: plotting multiple samples</div>

  <ol>
    <li>Run <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with  same parameters as above, for all 4 patients:
      <ul>
        <li>“First BAM file (e.g. treated sample)” to <code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient3_ChIP_ER_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient4_ChIP_ER_good_outcome</code></li>
        <li>“Second BAM file (e.g. control sample)” to <code class="language-plaintext highlighter-rouge">patient2_input_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient3_input_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient4_input_good_outcome</code></li>
      </ul>
    </li>
    <li>
      <p>Perform peak calling again using treatment file : <code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome</code> and control <code class="language-plaintext highlighter-rouge">patient2_input_good_outcome</code>, using macs2 parameters same as above.</p>
    </li>
    <li>
      <p>Concatenate the outputs (summits in BED) from <code class="language-plaintext highlighter-rouge">patient1</code> and <code class="language-plaintext highlighter-rouge">patient2</code> using <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> –&gt; <code class="language-plaintext highlighter-rouge">Concatenate</code></p>
    </li>
    <li>
      <p>Sort the output <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> –&gt; <code class="language-plaintext highlighter-rouge">sortBED</code></p>
    </li>
    <li>
      <p>Merge the overlapping intervals using <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> –&gt; <code class="language-plaintext highlighter-rouge">MergeBED</code></p>
    </li>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but:
      <ul>
        <li>Regions to plot : select the merged bed from above</li>
        <li>Output option : <code class="language-plaintext highlighter-rouge">reference-point</code></li>
        <li>The reference point for the plotting: <code class="language-plaintext highlighter-rouge">center of region</code></li>
        <li>Distance upstream of the start site of the regions defined in the region file : <code class="language-plaintext highlighter-rouge">3000</code></li>
        <li>Distance downstream of the end site of the given regions: <code class="language-plaintext highlighter-rouge">3000</code></li>
      </ul>

      <p>With this option, it considers only those genomic positions before (downstream) and/or after (upstream) a reference point (<em>e.g.</em> TSS, which corresponds to the annotated gene start in our case)</p>
    </li>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>“Matrix file from the computeMatrix tool” to the generated matrix</li>
        <li>“Show advanced options” to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>“Did you compute the matrix with more than one groups of regions?” to <code class="language-plaintext highlighter-rouge">No, I used only one group</code></li>
        <li>“Clustering algorithm” to <code class="language-plaintext highlighter-rouge">Kmeans clustering</code>
Inspect the output</li>
      </ul>
    </li>
  </ol>

</blockquote>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>ChIP-seq data requires multiple methods of quality assessment to ensure that the data is of high quality.</p>
</li>
                        
                        <li><p>Multiple normalization methods exists depending on the availability of input data.</p>
</li>
                        
                        <li><p>Heatmaps containing all genes of an organism can be easily plotted given a BED file and a coverage file.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/epigenetics/faqs/">FAQ page for the Epigenetics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Identification of the binding sites of the Estrogen receptor (Epigenetics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Friederike Dündar, Anika Erxleben, Bérénice Batut, Vivek Bhardwaj, Fidel Ramirez,  <b>Identification of the binding sites of the Estrogen receptor (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{epigenetics-estrogen-receptor-binding-site-identification,
author = "Friederike Dündar and Anika Erxleben and Bérénice Batut and Vivek Bhardwaj and Fidel Ramirez",
	title = "Identification of the binding sites of the Estrogen receptor (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: deeptools_bam_compare
  owner: bgruening
  revisions: a1ca9bccc7f2
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_bam_coverage
  owner: bgruening
  revisions: 5d11599b8a7d
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_compute_gc_bias
  owner: bgruening
  revisions: 29fcfd36e90a
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_compute_matrix
  owner: bgruening
  revisions: fd1275e01605
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_multi_bam_summary
  owner: bgruening
  revisions: 2d466b530754
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_correlation
  owner: bgruening
  revisions: f264afa9fa6f
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_fingerprint
  owner: bgruening
  revisions: f8fc4e5aff9d
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_heatmap
  owner: bgruening
  revisions: 512155688d98
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trim_galore
  owner: bgruening
  revisions: 1bf4789584dc
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bam_to_sam
  owner: devteam
  revisions: 88eedb4abea0
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bowtie2
  owner: devteam
  revisions: 66f992977578
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: f2e8552cf1d0
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_idxstats
  owner: devteam
  revisions: 88b8c2916784
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: macs2
  owner: iuc
  revisions: f5d67c722d67
  tool_panel_section_label: Epigenetics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00138">https://gxy.io/GTN:T00138</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1698452657'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>