<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Avian influenza viral strain analysis from gene segment sequencing data</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1702773026' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1702773026' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Avian influenza viral strain analysis from gene segment sequencing data">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-06-21 13:37:34 +0000">
<meta name="DC.creator" content="Wolfgang Maier">

        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Avian influenza viral strain analysis from gene segment s...">
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-06-21 13:37:34 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Avian influenza viral strain analysis from gene segment sequencing data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Avian influenza viral strain analysis from gene segment sequencing data' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html",
  "timeRequired": "PT4H",
  "teaches": "Determine how reassortment impacts reference-based mapping approaches\n - Use a collection of per-segment reference sequences to construct a hybrid reference genome that is sufficiently close to a sequenced sample to be useful as a reference for mapping\n - Construct a sample consensus genome from mapped reads\n - Generate per-segment phylogenetic trees of AIV consensus sequences",
  "keywords": "variant-analysis, virology, one-health",
  "description": "The questions this  addresses are:\n - With reassortment of gene segments being a common event in avian influenza virus (AIV) evolution, does it make sense to use a reference-based mapping approach for constructing consensus genome sequences for AIV samples?\n - Is it possible to reuse existing tools and workflows developed for the analysis of sequencing data from other viruses?\n - How can we obtain meaningful phylogenetic insight from AIV consensus sequences?\n\n\nThe objectives are:\n - Determine how reassortment impacts reference-based mapping approaches\n - Use a collection of per-segment reference sequences to construct a hybrid reference genome that is sufficiently close to a sequenced sample to be useful as a reference for mapping\n - Construct a sample consensus genome from mapped reads\n - Generate per-segment phylogenetic trees of AIV consensus sequences\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "name": "virology"
    },
    {
      "@type": "Thing",
      "name": "one-health"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1702773026'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/aiv-analysis/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Avian influenza viral strain analysis from gene segment sequencing data</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Avatar" width="36" height="36">Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>With reassortment of gene segments being a common event in avian influenza virus (AIV) evolution, does it make sense to use a reference-based mapping approach for constructing consensus genome sequences for AIV samples?</p>
</li>
        
        <li><p>Is it possible to reuse existing tools and workflows developed for the analysis of sequencing data from other viruses?</p>
</li>
        
        <li><p>How can we obtain meaningful phylogenetic insight from AIV consensus sequences?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Determine how reassortment impacts reference-based mapping approaches</p>
</li>
        
        <li><p>Use a collection of per-segment reference sequences to construct a hybrid reference genome that is sufficiently close to a sequenced sample to be useful as a reference for mapping</p>
</li>
        
        <li><p>Construct a sample consensus genome from mapped reads</p>
</li>
        
        <li><p>Generate per-segment phylogenetic trees of AIV consensus sequences</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/aiv-analysis/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 21, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00308">https://gxy.io/GTN:T00308</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Of the four species of influenza viruses (Influenza A-D), <em>Influenza A</em> is the most virulent in human hosts and subtypes of it have been responsible for all historic flu pandemics.</p>

<p>The different influenza species infect distinct animal hosts (though all of them can infect humans and pigs) and the most important natural reservoir for <em>Influenza A</em> are birds (in particular, wild aquatic birds), in which it causes <strong>Avian influenza</strong>.</p>

<p>Importantly, all flu pandemics of the last century have been triggered by <strong>reassortment</strong> events between avian and human <em>Influenza A</em> strains (although at least the 2009 swine flu pandemic involved an additional mixing event with an <em>Influenza A</em> strain from pigs).</p>

<p>Reassortment events are a key trait of <em>Influenza A</em> made possible by its wide range of natural hosts combined with two molecular characteristics of the species:</p>

<ol>
  <li>the linear negative-sense single-stranded RNA genomes of influenza viruses (and of all viruses of the <em>Orthomyxoviridae</em> family) are segmented, i.e. consist of several (eight in the case of <em>Influenza A</em> and <em>B</em>) distinct pieces of RNA, which typically encode just one, sometimes two viral proteins.</li>
  <li>the mutation rate in the genome is high in influenza viruses, and particularly high in <em>Influenza A</em>, since their RNA polymerase lacks exonuclease activity necessary for proof-reading.</li>
</ol>

<p>Together these characteristics enable <em>Influenza A</em> to evolve relatively rapidly in non-human hosts, then re-adapt to humans through exchange of segments (reassortment) in a host co-infected with a human and, <em>e.g.</em>, an avian strain. If such a reassortment introduces new variants of the two most antigenic proteins of influenza, HA (hemagglutinin) and NA (neuraminidase), this antigenic shift provides the reassorted strain with immune escape potential, which, if it happens in a genetic background compatible with efficient transmission in humans, can trigger an unusually strong wave of influenza or even a new pandemic.</p>

<p>In order to estimate the likelihood of an epidemic or pandemic influenza event in the near future it is, therefor, important to monitor closely the genome composition of circulating Avian Influenza strains in wild and domestic birds and the huge technological advances over the last decade make it possible to use next-generation sequencing for this purpose.</p>

<p>At the same time, the segmented nature of their genomes combined with high genetic variability, especially in the HA and NA genes, requires rather specialized bioinformatics workflows for the analysis of data from influenza viruses compared to other viruses with similar genome size (like, <em>e.g.</em> SARS-CoV-2):</p>

<p>The viral surface proteins HA and (to a lesser extent) NA are the main targets of the host antibody response and are, thus, under constant selection pressure to mutate into forms capable of evading an existing host immune response. As a consequence, these segments have evolved into a much richer panel of sequence variants than the other segments, to the point that sequences of the HA segment can, at present, be classified into 18 distinct subfamilies, H1-H18, while there are 11 recognized subfamilies of NA, and <em>Influenza A</em> strains are subtyped (as, for example, H5N1, H3N2, etc.) according to the combination of HA- and NA-encoding segments they are carrying.</p>

<p>Importantly, the sequence diversity of HA and (again to a lesser extent) of NA segments is big enough to prevent a naive approach of mapping sequenced reads to one specific agreed-upon <em>Influenza A</em> reference sequence. While this would work for the other six segments, mapping software would regularly fail to find enough plausible mappings for sequenced reads of HA and NA origin to continue analysis with. This is why, in this tutorial we are going to explore an alternative approach, which is also mapping-based but chooses a suitable reference for each segment dynamically based on the input sequencing data.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#prepare-analysis-history-and-data" id="markdown-toc-prepare-analysis-history-and-data">Prepare analysis history and data</a></li>
  <li><a href="#get-a-history-with-reference-data" id="markdown-toc-get-a-history-with-reference-data">Get a history with reference data</a>    <ol>
      <li><a href="#get-the-sequencing-data" id="markdown-toc-get-the-sequencing-data">Get the sequencing data</a></li>
      <li><a href="#inspect-the-reference-data" id="markdown-toc-inspect-the-reference-data">Inspect the reference data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
  <li><a href="#per-segment-subtyping-and-hybrid-reference-construction" id="markdown-toc-per-segment-subtyping-and-hybrid-reference-construction">Per-segment subtyping and hybrid reference construction</a></li>
  <li><a href="#mapping-to-a-hybrid-reference" id="markdown-toc-mapping-to-a-hybrid-reference">Mapping to a hybrid reference</a></li>
  <li><a href="#consensus-sequence-construction" id="markdown-toc-consensus-sequence-construction">Consensus sequence construction</a></li>
  <li><a href="#placing-segments-on-a-phylogenetic-tree" id="markdown-toc-placing-segments-on-a-phylogenetic-tree">Placing segments on a phylogenetic tree</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="prepare-analysis-history-and-data">Prepare analysis history and data</h1>

<p>In this tutorial you are going to work on a single avian influenza sample sequenced in paired-end mode on the Illumina platform, <em>i.e.</em> we are going to download two datasets of sequenced reads for that sample.</p>

<p>In addition, we are going to base the analysis on a small collection of multiple reference sequences for each influenza gene segment. We prepared this collection for you from public <a href="https://insaflu.insa.pt/dashboard/">INSAFlu</a> <a href="https://insaflu.readthedocs.io/en/latest/uploading_data.html?highlight=references#uploading-reference-data">data</a>.</p>

<p>If you have your own curated collection of reference sequences, you should be able to use it to follow this tutorial without any problem. Note, however, that the analysis results referred to in many of the questions will be different if you are exchanging the reference collection.
If you want to learn how you can create a Galaxy collection from your own references that is structured like the default we are suggesting here, you may want to follow the dedicated tutorial on <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">Using dataset collections</a>.</p>

<h1 id="get-a-history-with-reference-data">Get a history with reference data</h1>

<p>Any analysis should get its own Galaxy history, but in this tutorial we <em>won’t</em> start with creating one.</p>

<p>Instead, to save some time, you can copy an existing shared history on Galaxy Europe.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-prepare-the-galaxy-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Prepare the Galaxy history</div>

  <ol>
    <li>
      <p>Open the <a href="https://usegalaxy.eu/u/wolfgang-maier/h/influenza-resources">shared history</a> with pre-prepared INSAFlu reference data</p>
    </li>
    <li>
      <p>Rename the history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name: <code class="language-plaintext highlighter-rouge">Avian influenza tutorial</code></li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name: <code class="language-plaintext highlighter-rouge">Avian influenza tutorial</code></li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h2 id="get-the-sequencing-data">Get the sequencing data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get the data</div>

  <ol>
    <li>
      <p>Import the forward and reverse reads of the sequenced sample into your history and organize them into a <em>Paired Collection</em>.</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://usegalaxy.eu/api/datasets/4838ba20a6d86765171e4f201205515c/display?to_ext=fastqsanger.gz
https://usegalaxy.eu/api/datasets/4838ba20a6d867657d09280baae4e1ec/display?to_ext=fastqsanger.gz
</code></pre></div>      </div>

      <ol>
        <li>Copy the links above</li>
        <li>Open the <span class="tool" data-tool="upload1" title="Upload tool" aria-role="link"><strong>Upload</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: upload1</span></span> Manager</li>
        <li>In the top row of tabs select <strong>Collection</strong></li>
        <li>Configure the drop-down select boxes on that tab like this:
          <ul>
            <li><em>“Collection Type”</em>: <code class="language-plaintext highlighter-rouge">Pair</code></li>
            <li><em>“File Type”</em>: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
          </ul>
        </li>
        <li>Click on <strong>Paste/Fetch data</strong> and paste the links you copied into the empty text box</li>
        <li>Press <strong>Start</strong></li>
        <li>Wait for the <strong>Build</strong> button to become enabled, click it, and, in the next dialogue, give a suitable <strong>Name</strong>, like <code class="language-plaintext highlighter-rouge">Sequencing data</code>, to the new collection.</li>
        <li>Click on <strong>Create collection</strong></li>
      </ol>
    </li>
  </ol>

</blockquote>

<h2 id="inspect-the-reference-data">Inspect the reference data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-what-is-in-the-reference-collection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: What is in the reference collection?</div>

  <ol>
    <li>Step inside the imported reference collection by clicking on it.</li>
    <li>Expand individual elements of the collection by clicking on them to reveal more details about them.</li>
    <li>
      <p>View the content of any element by clicking its <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>.</p>

      <p>You can then scroll through the data and use your browser’s search function to look for particular sequences.</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-questions"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Questions</div>

        <ol>
          <li>How many elements are in the collection? What do they represent?</li>
          <li>How many sequences are stored in each element?</li>
          <li>Are all subtypes of HA and NA represented in the sequences?</li>
          <li>Are there sequences of non-A Influenza species?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answers"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answers" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answers<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The collection consists of eight elements and each element provides reference sequences for one genome segment.</li>
            <li>Each element has data for 56 sequences (Galaxy knows how to interpret the fasta format of the sequencing data and counts sequences for you; you can see the count for any of the elements by expanding it).</li>
            <li>The collection has reference sequence data for H1-H18 (with the exception of H11) and for N1-N11. For many of those subtypes, however, there is only one reference, i.e., within-subtype variation is not captured well by the collection.</li>
            <li>There are 6 <em>Influenza B</em> references in the collection. Since <em>Influenza B</em> is not normally seen in birds and we are going to analyze an avian influenza sample here, these can be considered controls: if we get an <em>Influenza B</em> assignment for the sample at any step in our analysis, we know something is very suspicious. This also means that we only have 50 <em>informative</em> references in the collection.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> There are lots of text processing and filtering tools for Galaxy, of which this tutorial is going to introduce just a small subset. You could use them to extract statistics about which subtypes are found how many times in the collection of references instead of relying on scrolling and searching. If you are interested and have the time, you may want to try the <a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">Data Manipulation Olympics</a> to learn more about available such tools and how to combine them.</p>

<h1 id="quality-control">Quality control</h1>

<p>As a very first step, we would like to make sure that we base our analysis only on the high-quality parts of the sequenced reads.</p>

<p>NGS reads often have lower base calling quality near their ends, so here we are going to trim low-quality stretches of bases from both ends. In addition, we are going to discard reads shorter than 30 bases after trimming. This is reasonable since our next step will involve extracting all possible 21-mers from the reads and there are very few possibilities for a read shorter than 30 bases.</p>

<p>The tool <strong>fastp</strong> lets us perform these tasks and obtain a nice quality report for our reads before and after processing in one go, but many other options exist to perform sequenced reads quality control and trimming/filtering in Galaxy and the dedicated tutorial on <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">quality control</a> introduces more of them.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-qc-and-read-trimming-filtering-with-fastp"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: QC and read trimming/filtering with fastp</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0" title="fastp tool" aria-role="link"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.23.2+galaxy0</span></span>
      <ul>
        <li><em>“Single-end or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code></li>
        <li><em>“Select paired collection(s)”</em>: the uploaded paired collection of sequenced reads</li>
        <li>In <em>“Filter Options”</em> under <em>“Length filtering options”</em>:
          <ul>
            <li><em>“Length required”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
        <li>In <em>“Read Modification Options”</em> under <em>“Per read cutting by quality options”</em>:
          <ul>
            <li><em>“Cut by quality in front (5’)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Cut by quality in tail (3’)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Cutting mean quality”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="details">
        <div class="box-title details-title" id="details-fastp-default-settings"><button class="gtn-boxify-button details" type="button" aria-controls="details-fastp-default-settings" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: fastp default settings<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>fastp comes with basic default settings for some of its many parameters.
In particular, with no explicit parameter specifications, the tool will still remove reads that</p>
        <ul>
          <li>have base qualities &lt; 15 for more than 40% of their bases</li>
          <li>have more than 5 Ns in their sequence</li>
          <li>
            <p>have a length &lt; 15 bases (after removing poly-G tails from Illumina reads)</p>

            <p>Because our input data has far longer MiSeq-generated reads, we can increase this length threshold to 30 bases as done above.</p>
          </li>
        </ul>

        <p>These are very relaxed criteria that will only remove the least usable reads, but still assure minimal
quality standards that are good enough for many (including training) purposes.</p>
      </blockquote>

      <p>Running the tool will result in two outputs. One is a new paired collection with the processed reads, the other one is a report of initial quality, the processing actions performed and their effect on key quality metrics.</p>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <strong>Inspect</strong> the report and try to answer the following questions:</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Questions</div>

        <ol>
          <li>Which set of reads (forward or reverse reads) did profit most quality-wise from our low-quality base trimming?</li>
          <li>What percentage of reads got discarded completely with our settings, and what percentage of bases?</li>
          <li>Why, according to the Filtering result, have some reads been discarded because of too many Ns?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answers-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answers-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answers<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The forward reads were of worse quality in particular near their 3’-ends than the reverse reads and, consequently, were affected more strongly by trimming (see the different “quality” plots in the report).</li>
            <li>Less than 2% of reads got discarded completely (the “Filtering result” section of the report says 98.03% of all reads were retained). In contrast, around 10% of all bases got discarded (compare total bases before and after filtering) so most reads got trimmed but not discarded.</li>
            <li>As explained in the “fastp default settings” box above, <strong>fastp</strong> is a tool with many hidden default actions in its various sections.
Inside the <em>Filter Options</em> inspect <em>Quality filtering options</em> and the various defaults mentioned in the parameters help therein. One of these defaults is to discard reads with more than 5 Ns. These defaults are often reasonable, but it’s good to be aware of them.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="per-segment-subtyping-and-hybrid-reference-construction">Per-segment subtyping and hybrid reference construction</h1>

<p>Our goal at this step is to find best matches between our sequencing data and the reference sequences of each genome segment. This will:</p>
<ul>
  <li>give us preliminary subtyping results with regard to the HA and NA segments for the sequenced sample</li>
  <li>suggest the best reference to map the sequencing data to for each segment.</li>
</ul>

<p>We are then going to combine the best reference of each segment into a hybrid reference genome to use for mapping our sequenced reads against.</p>

<p>To identify the best matching reference segments, we are going to run the tool <strong>VAPOR</strong> asking it to report the stats for any hits it identifies.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-exploring-best-matching-reference-segment-scores"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Exploring best-matching reference segment scores</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3" title="VAPOR tool" aria-role="link"><strong>VAPOR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/vapor/vapor/1.0.2+galaxy3</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Reference sequences”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li><em>“Type of sequencing data”</em>: <code class="language-plaintext highlighter-rouge">Paired-end as collection</code></li>
        <li><em>“Paired collection of sequenced reads”</em>: quality-trimmed reads; output of <strong>fastp</strong></li>
        <li><em>“Desired output”</em>: <code class="language-plaintext highlighter-rouge">Return scores of best matches</code></li>
        <li><em>“Limit number of reported matches to”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li>In <em>“Optional arguments”</em>:
          <ul>
            <li><em>“Read kmer filtering threshold”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
            <li><em>“Minimum k-mer proportion”</em>: <code class="language-plaintext highlighter-rouge">0.0</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Explore the output collection produced by the tool</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Questions</div>

        <ol>
          <li>Is there a difference between the results for segment 4 (encoding HA), segment 6 (encoding NA) and those for the remaining segments?</li>
          <li>According to the tool, what is the likely subtype with regard to HA and NA of the sample?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answers-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answers-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answers<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The values for <code class="language-plaintext highlighter-rouge">% of query bases in reads</code> and for <code class="language-plaintext highlighter-rouge">Total score</code> are drastically lower for the HA and somewhat lower for the NA segment than for any of the other segments. This could be due to the structure of the collection of references we have used, but also fits very well with the higher selection pressure on NA and, in particular on HA, as the main antigenic proteins of the virus, which leads to higher variability in these segments than in the others.</li>
            <li>
              <p>The best match (assigned an almost 2x higher score than the second-best match) found for the HA segment is from an H4N6 strain. The top two matches with regard to the NA segment (again assigned ~ 2x higher scores than the next best matches) are from strains with the N6 subtype of NA.</p>

              <p>The most likely subtype of the sample, thus, appears to be H4N6.</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Now that we have established that things <em>may</em> make sense, we can use the output of <strong>VAPOR</strong> to extract the actual sequence of the top hit for each reference segment. We then concatenate these best matches into a hybrid reference genome for mapping.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtaining-sequences-of-top-hits-identified-by-vapor"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtaining sequences of top hits identified by VAPOR</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="link"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span> to extract just the name of the sequence from each line of VAPOR’s output
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to process”</em>: collection of score outputs of <strong>VAPOR</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^.+\t&gt;(.+)$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">$1</code></li>
            <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Show beginning1" title="Select first lines from a dataset tool" aria-role="link"><strong>Select first lines from a dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Show beginning1</span></span> to get the first line, i.e. the best match from VAPOR’s output
      <ul>
        <li><em>“Select first”</em>: <code class="language-plaintext highlighter-rouge">1</code> lines</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“from”</em>: output of <strong>Replace</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.3.1" title="seqtk_subseq tool" aria-role="link"><strong>seqtk_subseq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_subseq/1.3.1</span></span> to extract the reference sequences based on their names reported by VAPOR
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input FASTA/Q file”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li><em>“Select source of sequence choices”</em>: <code class="language-plaintext highlighter-rouge">FASTA/Q ID list</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input ID list”</em>: the collection output of <strong>Select first</strong></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0" title="Collapse Collection tool" aria-role="link"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</span></span> to combine the best-matching sequence of each segment into one dataset
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>: the selected sequences collection produced by <strong>seqtk_subseq</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>At this point, you may want to <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> inspect the output of the last step to see if it is an eight-segments reference genome as expected, and if the segments correspond to the top hits found by VAPOR.</p>

<h1 id="mapping-to-a-hybrid-reference">Mapping to a hybrid reference</h1>

<p>If things went well, the hybrid reference we just obtained should be close enough across all segments to our sample to allow successful mapping of reads. Before we start the mapping we may want to truncate the segment names in our hybrid reference genome though because currently these names still reflect the full origin of the segment sequences, but from now on we are fine with just the segment ID. We can use the <strong>Replace</strong> tool from before again to truncate the names.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-shortening-sequence-titles"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Shortening sequence titles</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="link"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: the hybrid reference genome; output of <strong>Collapse Collection</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;([^|]+).+$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">&gt;$1</code></li>
            <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Add a tag to the output dataset</p>

      <p>This is the hybrid reference genome that we are going to use for mapping now.
It is an important intermediate result, to which we will need to return later, and to make that easier we are going to add a tag to it now for extra visibility in the history.</p>

      <ol>
        <li>Click on the dataset to expand it</li>
        <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <strong>Add Tags</strong></li>
        <li>Enter a tag like <code class="language-plaintext highlighter-rouge">HybridRef</code> (tags have to consist of a single word)</li>
        <li>Confirm your choice by pressing Enter on the keyboard or clicking on the new name.</li>
      </ol>
    </li>
  </ol>

</blockquote>

<p>Having polished the titles of the segments in our hybrid reference genome we are finally ready for mapping, which we will carry out with <strong>BWA-MEM</strong>, clean up a bit with <strong>Samtools view</strong> and produce a quality report for with <strong>QualiMap BamQC</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-mapping-and-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read mapping and quality control</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Map with BWA-MEM tool" aria-role="link"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: hybrid reference genome with shortened names; output of <strong>Replace</strong></li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code></li>
        <li><em>“Select a paired collection”</em>: quality-trimmed reads; output of <strong>fastp</strong> in QC and Trimming</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0" title="Samtools view tool" aria-role="link"><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SAM/BAM/CRAM data set”</em>: mapped reads BAM dataset; output of <strong>BWA-MEM</strong></li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
        <li>In <em>“Configure filters”</em>:
          <ul>
            <li><em>“Filter by quality”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>“Require that these flags are set”</em>: <code class="language-plaintext highlighter-rouge">Read is paired</code> and <code class="language-plaintext highlighter-rouge">Read is mapped in a proper pair</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3" title="QualiMap BamQC tool" aria-role="link"><strong>QualiMap BamQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/qualimap_bamqc/qualimap_bamqc/2.2.2d+galaxy3</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mapped reads input dataset”</em>: filtered mapped reads BAM dataset; output of <strong>Samtools view</strong></li>
        <li><em>“Skip duplicate reads”</em>: <code class="language-plaintext highlighter-rouge">Unselect all</code></li>
        <li>In <em>“Settings affecting specific plots”</em>:
          <ul>
            <li><em>“Number of bins to use in across-reference plots”</em>: <code class="language-plaintext highlighter-rouge">40</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Study the report generated with QualiMap</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Questions</div>

        <ol>
          <li>What is the coverage of each segment by the sequenced reads, and is it uniform?</li>
          <li>Look for a plot showing read mapping quality across the reference. What can you conclude?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answers-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answers-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answers<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>Coverage is rather different for the different segments. Looking at “Mean coverage” and its “Standard deviation” in the “Chromosome stats” table and at the “Coverage across reference” plot, coverage of the HA segment seems to be most critical since it approaches zero in some regions.</li>
            <li>Mapping quality is almost constant at or very near 60 (which happens to be the maximum mapping quality value emitted by BWA-MEM) across all segments with the exception of HA.</li>
          </ol>

          <p>These two observations combined with the VAPOR statistics for HA show again that even the best matching reference for this segment is not exactly close to the sequence of our sample. The read mapper had difficulties placing the sequenced reads on that sub-optimal reference and it looks as if we might have very little sequence information for some HA regions.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="consensus-sequence-construction">Consensus sequence construction</h1>

<p>From the polished mapping of reads to our custom reference we can now construct the consensus sequence of our sample.</p>

<p>Unfortunately, the tool we are going to use for this, <strong>ivar consensus</strong>, is not capable of working with more than one reference name at a time, but because this is influenza data we have mappings to 8 different segments described in our data. So we need to take a little detour and split the mapped reads data into a collection of datasets each containing the mappings for just one segment first again, then perform the consensus construction for all of them in parallel.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-splitting-mapped-reads-by-genome-segment"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Splitting mapped reads by genome segment</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.2+galaxy1" title="Split BAM by Reference tool" aria-role="link"><strong>Split BAM by Reference</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/bamtools_split_ref/bamtools_split_ref/2.5.2+galaxy1</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM dataset to split by reference”</em>: filtered mapped reads; output of <strong>Samtools view</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output from this step has the desired collection structure, but the names of the collection elements are not the nicest. Ideally, we would just reuse the segment names, which are already provided in our mapping reference genome. So lets extract these names again and use them as new element labels</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-relabeling-collection-elements"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Relabeling collection elements</div>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Select lines that match an expression tool" aria-role="link"><strong>Select lines that match an expression</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>
      <ul>
        <li>
          <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: the mapping reference; output of <strong>Replace</strong> in Mapping to a hybrid reference</p>

          <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> This is the dataset you should have <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> tagged before. Lets hope you did!</p>
        </li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;.+</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="link"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output of <strong>Select</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1. Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;(.+)$</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">$1</code></li>
            <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="__RELABEL_FROM_FILE__" title="Relabel identifiers tool" aria-role="link"><strong>Relabel identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <strong>RELABEL_FROM_FILE</strong></span></span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input collection”</em>: the split mappings; output of <strong>Split BAM by Reference</strong></li>
        <li><em>“How should the new labels be specified?”</em>: <code class="language-plaintext highlighter-rouge">Using lines in a simple text file.</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“New Identifiers”</em>: output of <strong>Replace</strong></li>
        <li><em>“Ensure strict mapping”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>And with that we are ready for consensus sequence generation!</p>

<p>To accept any base suggested by the mapped sequenced reads as the consensus base for the corresponding genome position, we ask for the following requirements to be fulfilled:</p>

<ul>
  <li>at least ten sequenced reads have to provide information about the base in question</li>
  <li>at a minimum, 70% of these reads have to agree on the base at this position.</li>
</ul>

<p>To avoid getting misled too much by sequencing errors, we are also going to ignore bases with a base calling quality less than 20 in the above counts (i.e., we are going to base our decisions only on bases in sequenced reads that the basecaller of the sequencer was reasonably sure about.</p>

<p>Now what if we cannot obtain a consensus base for a position with the above criteria? In such cases of uncertainty we want to insert an N (i.e. an unknown base) to express that we either did not have enough information about the position or that this information was ambiguous.</p>

<p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> All of the above limits for consensus base calling are arbitrary to some degree, and depend somewhat on the quality of the sequencing data. With very high overall coverage, for example, it is possible to increase the coverage threshold, but if you increase that threshold too much, you may end up with a consensus sequence consisting mostly of Ns.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-per-segment-consensus-construction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Per-segment consensus construction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0" title="ivar consensus tool" aria-role="link"><strong>ivar consensus</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.4.2+galaxy0</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Bam file”</em>: the relabeled collection of mapped reads; output of <strong>Relabel identifiers</strong></li>
        <li><em>“Minimum quality score threshold to count base”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
        <li><em>“Minimum frequency threshold”</em>: <code class="language-plaintext highlighter-rouge">0.7</code></li>
        <li><em>“Minimum indel frequency threshold”</em>: <code class="language-plaintext highlighter-rouge">0.8</code></li>
        <li><em>“Minimum depth to call consensus”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
        <li><em>“How to represent positions with coverage less than the minimum depth threshold”</em>: <code class="language-plaintext highlighter-rouge">Represent as N (-n N)</code></li>
      </ul>

      <p>The output is a consensus sequence in FASTA format, one per segment, with
the names just providing a bit too much detail for our purpose.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="link"><strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to process”</em>: consensus sequences; output of <strong>ivar consensus</strong></li>
        <li>In <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“1: Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^&gt;Consensus_(.*)_threshold_.*</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">&gt;$1</code></li>
            <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect each consensus sequence generated for the different segments</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Does everything look ok?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answer"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answer" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answer<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>As expected from the findings so far, the consensus sequence for the HA segment has stretches of Ns in it, which likely reflect the mapping issues and associated loss of coverage caused by our insufficiently sized collection of references.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="placing-segments-on-a-phylogenetic-tree">Placing segments on a phylogenetic tree</h1>

<p>The next logical step after obtaining the consensus sequences of segments of our sample is to explore how those sequences are related to the sequences in our reference collection.
To do so, we are going to combine the reference sequences of all segments with their corresponding consensus sequence into one multi-sequence fasta dataset per segment. Then we build a multiple sequence alignment (MSA) from the sequences of each segment, and use these alignments to generate phylogenetic trees, again one per segment. We are going to use two rather standard tools, <strong>MAFFT</strong> and <strong>IQTree</strong>, for generating MSAs and trees, respectively.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-exploring-phylogeny"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Exploring phylogeny</div>

  <ol>
    <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="link"><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: cat1</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Concatenate Dataset”</em>: <code class="language-plaintext highlighter-rouge">References per segment (INSAFlu)</code></li>
        <li>In <em>“Dataset”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Dataset”</em>
              <ul>
                <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select”</em>: collection of renamed consensus sequences; output of <strong>Replace</strong> on consensus sequences</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-pick-the-right-concatenate-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-pick-the-right-concatenate-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Pick the right Concatenate tool<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>On most Galaxy servers you will find two <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Concatenate datasets</strong> tools installed:</p>   <ol>   <li><strong>Concatenate datasets</strong> tail-to-head</li>   <li><strong>Concatenate datasets</strong> tail-to-head (cat)</li> </ol>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> The tool you want to use at this step is <code class="language-plaintext highlighter-rouge">#1</code>, but if you are interested in the details, here is why:</p>   <p>The two tools have nearly identical interfaces, but behave differently in certain situations, specifically:</p>   <ul>   <li>     <p>The second tool, the one with “(cat)” in its name, simply concatenates everything you give to it into a single output dataset.</p>     <p>Whether you give it multiple datasets or a collection as the first parameter, or some datasets as the first and some others as the second parameter, it will always concatenate them all. In fact, the only reason for having multiple parameters for this tool is that by providing inputs through multiple parameters, you can make sure they are concatenated in the order you pass them in.</p>   </li>   <li>     <p>The first tool, on the other hand, will only ever concatenate inputs provided through <em>different</em> parameters.</p>     <p>This tool allows you to specify an arbitrary number of <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, but if you also want to use <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets or <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> a collection for some of the Dataset parameters, then all of these need to be of the same type (multiple datasets <em>or</em> collections) and have the same number of inputs.</p>     <p>Now depending on the inputs, one of the following behaviors will occur:</p>     <ul>       <li>If all the different inputs are <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the tool will concatenate them all and produce a single output dataset.</li>       <li>If all the different inputs are specified <em>either</em> as <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets <em>or</em> as <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>, and all have the same number of datasets, then the tool will concatenate the first datasets of each input parameter, the second datasets of each input parameter, the third, etc., and produce an output collection with as many elements as there are inputs per Dataset parameter.</li>       <li>In extension of the above, if <em>some</em> additional inputs are provided as <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the content of these will be recycled and be reused in the concatenation of all the nth elements of the other parameters.</li>     </ul>   </li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>The tool should produce a collection of eight multi-sequence fasta datasets, each of which has the generated consensus sequence for one segment concatenated to the INSAFlu reference sequences of that segment.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.508+galaxy0" title="MAFFT tool" aria-role="link"><strong>MAFFT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.508+galaxy0</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Sequences to align”</em>: collection of concatenated sequences; output of <strong>Concatenate datasets</strong></li>
        <li><em>“Data type”</em>: <code class="language-plaintext highlighter-rouge">Nucleic Acids</code></li>
        <li><em>“Matrix selection”</em>: <code class="language-plaintext highlighter-rouge">No matrix</code></li>
      </ul>

      <p>The result is a collection of MSAs, each aligning all reference sequences of one genome segment plus the consensus sequence we have obtained for that segment against each other.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.1.2+galaxy2" title="IQ-TREE tool" aria-role="link"><strong>IQ-TREE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/iqtree/iqtree/2.1.2+galaxy2</span></span>
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Specify input alignment file in PHYLIP, FASTA, NEXUS, CLUSTAL or MSF format.”</em>: output of <strong>MAFFT</strong></li>
        <li><em>“Specify sequence type …“</em>: <code class="language-plaintext highlighter-rouge">DNA</code></li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Explore each of the final trees produced by IQTree for the different segments</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>What are your conclusions about the sample in general and its HA and NA segments in particular?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-answers-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-answers-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution: Answers<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ul>
            <li>For most of its segments the sample resembles relatively recent (from the last decade) Eurasian reference sequences.</li>
            <li>For HA and NA the sample clusters with the few available samples of the corresponding subtype.</li>
            <li>None of the references closest to the sample with respect to HA and NA are close to the recent Eurasian reference cluster for their remaining segments.</li>
            <li>A plausible explanation is that the H4 and N6 segments of the sample have been brought into the recent Eurasian background through a reassortment event. Caveat: interpretations like this can be heavily influenced by the size of the reference collection!</li>
          </ul>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Analysis workflows for influenza whole-genome sequencing data need to take into account the specific characteristics of the viral genome. Due to their higher natural variability this is especially true for avian influenza samples and for the HA- and NA-encoding segments of the genome.</p>

<p>Nevertheless, it looks possible, with carefully chosen reference segment sequences and bioinformatic tools, to avoid a computationally expensive de-novo assembly approach and to use mapping against a dynamically compiled reference genome instead.</p>

<p>The rather small reference segment collection suggested for this tutorial consists of 56 different samples, of which only a single one has the H4N6 subtype of the sample analyzed here. Still it allowed us to perform subtyping of the sample, to construct complete consensus sequences for 7 segments including NA, and to draw valuable conclusions about the origin of the sample. It is conceivable that a larger collection of references chosen to capture several strains from each HA subtype could solve the remaining issue of the incomplete HA consensus sequence.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Reassortment of gene segments makes reference-based mapping of influenza sequencing data challenging</p>
</li>
                        
                        <li><p>An alternative to <em>de-novo</em> assembly can be mapping to a dynamically chosen reference genome</p>
</li>
                        
                        <li><p>Variant calling and consensus genome construction can follow workflows used also for other viral sequence data</p>
</li>
                        
                        <li><p>Standard phylogenetic tools can be used to find relationships between influenza samples but should be used on a per-segment basis</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Avian influenza viral strain analysis from gene segment sequencing data (Variant Analysis)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier,  <b>Avian influenza viral strain analysis from gene segment sequencing data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-aiv-analysis,
author = "Wolfgang Maier",
	title = "Avian influenza viral strain analysis from gene segment sequencing data (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/variant-analysis/tutorials/aiv-analysis/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa
  owner: devteam
  revisions: e188dc7a68e6
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bamtools_split_ref
  owner: iuc
  revisions: 9b520009db81
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastp
  owner: iuc
  revisions: 65b93b623c77
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: iqtree
  owner: iuc
  revisions: 9b892d7f82cc
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ivar_consensus
  owner: iuc
  revisions: 17f911830a8c
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qualimap_bamqc
  owner: iuc
  revisions: 19ece8afbaab
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_view
  owner: iuc
  revisions: 5826298f6a73
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seqtk
  owner: iuc
  revisions: 3da72230c066
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: vapor
  owner: iuc
  revisions: 244812f5bd1f
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collapse_collections
  owner: nml
  revisions: 90981f86000f
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mafft
  owner: rnateam
  revisions: a65296ffb4a0
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00308">https://gxy.io/GTN:T00308</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/aiv-analysis/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1702773026'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>