<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>metaQuantome 1: Data creation</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: metaQuantome 1: Data creation" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Proteomics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url">
                            Français
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url">
                            日本語
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url">
                            Español
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url">
                            Português
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url">
                            العربية
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/proteomics/faqs/" title="Check our FAQs for the Proteomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">🏳️‍🌈
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> 🎃
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> 🇦🇺
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:10 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "metaQuantome 1: Data creation",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
  },
  "courseCode": "proteomics / metaquantome-data-creation / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'metaQuantome 1: Data creation' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html",
  "timeRequired": "PT1H",
  "keywords": "microbiome",
  "description": "The questions this  addresses are:\n - How do I perform functional and taxonomy analysis on metaproteomics data?\n - How can I perform quantitation on metaproteomics data?\n - How do I create inputs that can be used in metaquantome to examine differentially expressed proteins?\n\n\\nThe objectives are:\n - A taxonomy, functional and quantitational analysis of metaproteomic mass spectrometry data.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Emma Leith"
    },
    {
      "@type": "Person",
      "name": "Marie Crane"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Emma Leith"
    },
    {
      "@type": "Person",
      "name": "Marie Crane"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "skillLevel": "Intermediate"
}
</script>

<section class="tutorial topic-proteomics">
    <h1 data-toc-skip>metaQuantome 1: Data creation</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/subinamehta/" class="contributor-badge contributor-subinamehta"><img src="https://avatars.githubusercontent.com/subinamehta?s=27" alt="Avatar">Subina Mehta</a>


<a href="/training-material/hall-of-fame/timothygriffin/" class="contributor-badge contributor-timothygriffin"><img src="https://avatars.githubusercontent.com/timothygriffin?s=27" alt="Avatar">Timothy J. Griffin</a>


<a href="/training-material/hall-of-fame/pratikdjagtap/" class="contributor-badge contributor-pratikdjagtap"><img src="https://avatars.githubusercontent.com/pratikdjagtap?s=27" alt="Avatar">Pratik Jagtap</a>


<a href="/training-material/hall-of-fame/emmaleith/" class="contributor-badge contributor-emmaleith"><img src="https://avatars.githubusercontent.com/emmaleith?s=27" alt="Avatar">Emma Leith</a>


<a href="/training-material/hall-of-fame/mariecrane/" class="contributor-badge contributor-mariecrane"><img src="https://avatars.githubusercontent.com/mariecrane?s=27" alt="Avatar">Marie Crane</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do I perform functional and taxonomy analysis on metaproteomics data?</p>
</li>
        
        <li><p>How can I perform quantitation on metaproteomics data?</p>
</li>
        
        <li><p>How do I create inputs that can be used in metaquantome to examine differentially expressed proteins?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>A taxonomy, functional and quantitational analysis of metaproteomic mass spectrometry data.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/proteomics/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.4037137">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/workflows/" title="Workflows" alt="metaQuantome 1: Data creation workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 10, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Metaproteomics <span class="citation"><a href="#Metaproteomics_video">“<b>Metaproteomics video</b>”</a></span> involves characterization of community level <span class="notranslate">expression</span> of microbial proteins from an environmental
or clinical sample. Metaproteomics data <span class="citation"><a href="#Metaproteomics_community_effort">“<b>Metaproteomics community effort</b>”</a></span> <span class="citation"><a href="#Jagtap2015">Jagtap <i>et al.</i> 2015</a></span> is primarily used to determine the functional status of the microbiome under study along with its taxonomic composition. The <a href="http://galaxyp.org/"><span class="notranslate">Galaxy</span>-P</a> <span class="citation"><a href="#Galaxy-P_Metaproteomics">Metaproteomics</a></span> team published a software suite named metaQuantome { % cite Easterly2019 %} to enable quantitative and statistical analysis and visualization of functional,
taxonomic <span class="notranslate">expression</span> as well as functional and taxonomy interaction. metaQuantome leverages peptide level quantitative information to analyze the taxonomic, functional <span class="notranslate">expression</span> within the microbial community in different conditions.</p>

<p><img src="../../images/microbiome.png" alt="Microbiome. " width="75%" loading="lazy" /></p>

<p>metaQuantome offers differential abundance analysis, principal components analysis, and clustered heat map visualizations,
across multiple experimental conditions. metaQuantome, an open source tool, is available via command line and also
accessible via <span class="notranslate">Galaxy</span> platform for reproducible analysis. As a first step for metaQuantome analysis, metaproteomics
data needs to be made compatible for subsequent analysis. With this in mind, we have developed a metaQuantome data
generation <span class="notranslate">workflow</span> tutorial that will help users generate inputs for metaQuantome analysis.</p>

<p><img src="../../images/metaquantomeworkflow.png" alt="Workflow. " width="75%" loading="lazy" /></p>

<p>To demonstrate the use of the data creation <span class="notranslate">workflow</span>, we have used a thermophilic biogas reactor dataset wherein municipal
food waste and manure is digested to generate methane gas. After one round in the reactor, the microbial community was
simplified and enriched via serial dilution. This inoculum was then transferred to a solution of cellulose from Norwegian
Spruce and incubated at 65°C. Triplicate samples were taken in a time series from 0 to 43 hours after inoculation and mass
spectrometry data was acquired on a Q-Exactive (Thermo) mass spectrometer. For this training, we have chosen two time points-8 hour and 33 hour.</p>

<p><img src="../../images/biogasdataset.png" alt="Dataset. " width="75%" loading="lazy" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments"><strong>Pretreatments</strong></a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data"><em>Get data</em></a></li>
    </ol>
  </li>
  <li><a href="#match-peptide-sequences" id="markdown-toc-match-peptide-sequences"><strong>Match peptide sequences</strong></a>    <ol>
      <li><a href="#search-gui" id="markdown-toc-search-gui"><em>Search GUI</em></a></li>
      <li><a href="#peptide-shaker" id="markdown-toc-peptide-shaker"><em>Peptide Shaker</em></a></li>
      <li><a href="#removing-contaminants" id="markdown-toc-removing-contaminants"><em>Removing Contaminants</em></a></li>
      <li><a href="#removing-file-extensions-for-quantification" id="markdown-toc-removing-file-extensions-for-quantification"><em>Removing file extensions for Quantification</em></a></li>
      <li><a href="#extracting-peptide-list" id="markdown-toc-extracting-peptide-list"><em>Extracting Peptide list</em></a></li>
    </ol>
  </li>
  <li><a href="#peptide-quantification" id="markdown-toc-peptide-quantification"><strong>Peptide Quantification</strong></a>    <ol>
      <li><a href="#filtering-peptides-that-are-less-than-50-amino-acids" id="markdown-toc-filtering-peptides-that-are-less-than-50-amino-acids"><em>Filtering peptides that are less than 50 amino acids</em></a></li>
      <li><a href="#manipulating-text-for-metaquantome" id="markdown-toc-manipulating-text-for-metaquantome"><em>Manipulating text for metaQuantome</em></a></li>
    </ol>
  </li>
  <li><a href="#functional-and-taxonomy-annotation" id="markdown-toc-functional-and-taxonomy-annotation"><strong>Functional and Taxonomy annotation</strong></a>    <ol>
      <li><a href="#unipept-for-taxonomy-annotation" id="markdown-toc-unipept-for-taxonomy-annotation"><em>Unipept</em> for taxonomy annotation</a></li>
      <li><a href="#unipept-for-functional-annotation" id="markdown-toc-unipept-for-functional-annotation"><em>Unipept</em> for Functional annotation</a></li>
      <li><a href="#extracting-ec-values" id="markdown-toc-extracting-ec-values"><em>Extracting EC values</em></a></li>
      <li><a href="#filtering-confident-peptides" id="markdown-toc-filtering-confident-peptides"><em>Filtering confident peptides</em></a></li>
      <li><a href="#removing-hashtag-from-output" id="markdown-toc-removing-hashtag-from-output"><em>Removing Hashtag from output</em></a></li>
      <li><a href="#filter---ec-values" id="markdown-toc-filter---ec-values"><em>Filter</em> - EC values</a></li>
      <li><a href="#filter---biological-functions" id="markdown-toc-filter---biological-functions"><em>Filter</em> - Biological Functions</a></li>
      <li><a href="#filter---cellular-components" id="markdown-toc-filter---cellular-components"><em>Filter</em> - Cellular components</a></li>
      <li><a href="#filter---molecular-function" id="markdown-toc-filter---molecular-function"><em>Filter</em> - Molecular Function</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="pretreatments"><strong>Pretreatments</strong></h1>

<p>The first step in a tutorial is to get the data from the zenodo link provided and making sure that it is in the correct format.</p>

<h2 id="get-data"><em>Get data</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
    <li>
      <p>Import the files: 6 MZML files, a Protein FASTA file, and an Experimental Design file from <a href="https://doi.org/10.5281/zenodo.4037137">Zenodo</a>
or from the shared data library (<code class="language-plaintext highlighter-rouge">GTN - Material</code> -&gt; <code class="language-plaintext highlighter-rouge">proteomics</code> -&gt; <code class="language-plaintext highlighter-rouge">metaQuantome 1: Data creation</code>)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4037137/files/ExperimentalDesign.tsv
https://zenodo.org/record/4037137/files/ProteinDB_cRAP.fasta
https://zenodo.org/record/4037137/files/T2_A1.mzml
https://zenodo.org/record/4037137/files/T2_B1.mzml
https://zenodo.org/record/4037137/files/T7A_1.mzml
https://zenodo.org/record/4037137/files/T7B_1.mzml
</code></pre></div>      </div>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>Rename the datasets (If needed)</li>
    <li>
      <p>Check that the datatype ( Make sure they are in the correct formats).
6 MZML files (format=<code class="language-plaintext highlighter-rouge">mzml</code>, a Protein FASTA file (format=<code class="language-plaintext highlighter-rouge">fasta</code>), and an Experimental Design file (format=<code class="language-plaintext highlighter-rouge">tabular</code>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">datatypes</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>Add to each database a tag corresponding to the name of the input data (optional).</li>
    <li>Build a <strong>Dataset list</strong> for the four mzml files.
      <ul>
        <li>Click the <strong>Operations on multiple datasets</strong> check box at the top of the history panel</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a dataset collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>    <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>  </li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>
    </li>
    <li>Rename the dataset collection as <code class="language-plaintext highlighter-rouge">MZML dataset collection</code>.</li>
  </ol>

</blockquote>

<p>We have a choice to run all these steps using a single <span class="notranslate">workflow</span>, then discuss each step and the results in more detail.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-pretreatments"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Pretreatments</h3>

  <ol>
    <li><strong>Import the <span class="notranslate">workflow</span></strong> into <span class="notranslate">Galaxy</span>
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/proteomics/tutorials/metaquantome-data-creation/workflows/main_workflow.ga">this <span class="notranslate">workflow</span></a> or download it to your computer.</li>
        <li>Import the <span class="notranslate">workflow</span> into <span class="notranslate">Galaxy</span></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>“Archived <span class="notranslate">Workflow</span> URL”</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>“Archived <span class="notranslate">Workflow</span> File”</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>
    </li>
    <li>Run <strong><span class="notranslate">Workflow</span></strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span></span> using the following parameters:
      <ul>
        <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1: Input Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">MZML dataset collection</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“2: Experimental Design”</em>: <code class="language-plaintext highlighter-rouge">Experimental Design.tsv</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“3: FASTA_db”</em>: <code class="language-plaintext highlighter-rouge">ProteinDB_cRAP.fasta</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“4: In FlashLFQ - control condition for Bayesian protein fold-change analysis”</em>: <code class="language-plaintext highlighter-rouge">S1</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="match-peptide-sequences"><strong>Match peptide sequences</strong></h1>

<p>For this, the sequence database-searching program called <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> will be used.
The created dataset collection of the four <em>MZML files</em> in the history has to be first converted to MGF to be used as the MS/MS input.</p>

<h3 id="convert-mzml-to-mgf-with-msconvert"><em>Convert mzml to MGF with msconvert</em></h3>

<p>msconvert is used in order to convert the input file type, a mzml data collection, to a mgf file type.
The mgf file type can then be used as the Input Peak Lists when running SearchGUI.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mzml-to-mgf"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: mzml to MGF</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.19052.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.19052.0"><strong>msconvert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/msconvert/msconvert/3.0.19052.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input unrefined MS data”</em>: <code class="language-plaintext highlighter-rouge">MZML dataset collection</code></li>
        <li><em>“Do you agree to the vendor licenses?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Output Type”</em>: <code class="language-plaintext highlighter-rouge">mgf</code></li>
        <li>In <em>“Data Processing Filters”</em>:
          <ul>
            <li><em>“Apply peak picking?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Apply m/z refinement with identification data?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“(Re-)calculate charge states?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Filter m/z Window”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Filter out ETD precursor peaks?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“De-noise MS2 with moving window filter”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li>In <em>“Scan Inclusion/Exclusion Filters”</em>:
          <ul>
            <li><em>“Filter MS Levels”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li>In <em>“General Options”</em>:
          <ul>
            <li><em>“Sum adjacent scans”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Output multiple runs per file”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>This is a critical step for running this <span class="notranslate">workflow</span>.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Why do we need to convert the files to MGF?</li>
    <li>Can we use any other input format?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The files have to be converted to MGF for this <span class="notranslate">workflow</span> because we use SearchGUI as the searching tool and it can only read MGF files.</li>
      <li>Yes, we can also use RAW files as input and just convert RAW files to MGF.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="search-gui"><em>Search GUI</em></h2>
<p>SearchGUI is a tool that searches sequence databases on any number of MGF files. In this case, the previously made collection of three MGF files (entitles MGF files) will be used as the MS/MS input. This tool will produce an output file, called a SearchGUI archive file. This file will serve as in input for the next tool used, PeptideShaker.</p>
<blockquote class="notranslate hands_on">

  <h3 id="hands_on-hands-on-search-sequence-databases"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Search sequence databases</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/3.3.10.1" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/3.3.10.1"><strong>Search GUI</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/peptideshaker/search_gui/3.3.10.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein Database”</em>: <code class="language-plaintext highlighter-rouge">ProteinDB_cRAP.fasta</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input Peak Lists (mgf)”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>msconvert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Search Engine Options”</em>:
          <ul>
            <li><em>“DB-Search Engines”</em>: <code class="language-plaintext highlighter-rouge">X!tandem</code></li>
          </ul>
        </li>
        <li>In <em>“Protein Digestion Options”</em>:
          <ul>
            <li><em>“Digestion”</em>: <code class="language-plaintext highlighter-rouge">Trypsin</code></li>
            <li><em>“Maximum Missed Cleavages”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
          </ul>
        </li>
        <li>In <em>“Precursor Options”</em>:
          <ul>
            <li><em>“Fragment Tolerance”</em>: <code class="language-plaintext highlighter-rouge">0.2</code></li>
            <li><em>“Maximum Charge”</em>: <code class="language-plaintext highlighter-rouge">6</code></li>
          </ul>
        </li>
        <li>In <em>“Protein Modification Options”</em>:
          <ul>
            <li><em>“Fixed Modifications”</em>: <code class="language-plaintext highlighter-rouge">Carbamidomethylation of C</code></li>
            <li><em>“Variable Modifications”</em>: <code class="language-plaintext highlighter-rouge">Oxidation of M, Acetylation of Protein N-term</code></li>
          </ul>
        </li>
        <li>In <em>“Andvanced Options”</em>:
          <ul>
            <li><em>“SearchGUI Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“X!Tandem Options”</em>: <code class="language-plaintext highlighter-rouge">Advanced</code>
              <ul>
                <li><em>“X!Tandem: Quick Acetyl”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“X!Tandem: Quick Pyrolidone”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“X!Tandem: Maximum Valid Expectation Value”</em>: <code class="language-plaintext highlighter-rouge">100.0</code></li>
                <li><em>“X!Tandem peptide model refinement”</em>: <code class="language-plaintext highlighter-rouge">Don't refine</code></li>
              </ul>
            </li>
            <li><em>“OMSSA Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“MSGF Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“MS Amanda Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“TIDE Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“MyriMatch Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“Comet Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“DirectTag Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
            <li><em>“Novor Options”</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Note that sequence databases used for metaproteomics are usually much larger than the excerpt used in this tutorial. When using large databases, the peptide identification step can take much more time for computation. In metaproteomics, choosing the optimal database is a crucial step of your <span class="notranslate">workflow</span>, for further reading see <a href="https://www.ncbi.nlm.nih.gov/pubmed/27824341">Timmins-Schiffman et al (2017)</a>. To learn more about database construction in general, like integrating contaminant databases or using a decoy strategy for FDR searching, please consult our tutorial on <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a>.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many Search Engines can be used?</li>
    <li>Can the parameters be manipulated?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>There are 8 database search algorithms, you can use as many as you want. Ideally, 4 database algorithms gives the best results.</li>
      <li>Yes, The parameters can be manipulated according to the experimental design of the datasets.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="peptide-shaker"><em>Peptide Shaker</em></h2>

<p><a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a> is a post-processing software tool that processes data from the SearchGUI software tool. PeptideShaker is a search engine for interpretation of proteomics identification results from multiple search engines, currently supporting X!Tandem, MS-GF+, MS Amanda, OMSSA, MyriMatch, Comet, Tide, Mascot, Andromeda and mzIdentML. More specifically, PeptideShaker processes data from  the SearchGUI tool through the organization of Peptide-Spectral Matches (PSMs) generated. In addition to organization, it provides an assessment of confidence of the data and generates outputs that can be visualized by users to interpret the results.</p>
<blockquote class="notranslate hands_on">

  <h3 id="hands_on-hands-on-interpretation-of-searchgui"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Interpretation of SearchGUI</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/1.16.36.3" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/1.16.36.3"><strong>Peptide Shaker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/peptideshaker/peptide_shaker/1.16.36.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Compressed SearchGUI results”</em>: <code class="language-plaintext highlighter-rouge">searchgui_results</code> (output of <strong>Search GUI</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Specify Advanced PeptideShaker Processing Options”</em>: <code class="language-plaintext highlighter-rouge">Advanced Processing Options</code>
          <ul>
            <li><em>“The PTM probabilistic score to use for PTM localization”</em>: <code class="language-plaintext highlighter-rouge">A-score</code></li>
          </ul>
        </li>
        <li><em>“Specify Advanced Filtering Options”</em>: <code class="language-plaintext highlighter-rouge">Advanced Filtering Options</code>
          <ul>
            <li><em>“Maximum Peptide Length”</em>: <code class="language-plaintext highlighter-rouge">60</code></li>
          </ul>
        </li>
        <li><em>“Specify Contact Information for mzIdendML”</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Galaxy</span>P Project contact (Not suitable for PRIDE submission)</code></li>
        <li>In <em>“Exporting options”</em>:
          <ul>
            <li><em>“Creates a mzIdentML file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Compress results into a single zip file”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Reports to be generated”</em>: <code class="language-plaintext highlighter-rouge">PSM Report, Peptide Report, Protein Report, Certificate of Analysis,</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>There are a number of choices for different data files that can be generated using
 PeptideShaker. A compressed file can be made containing all information needed to view the
results in the standalone PeptideShaker viewer. A <code class="language-plaintext highlighter-rouge">mzidentML</code> file can be created that contains
all peptide sequence matching information and can be utilized by compatible <span class="notranslate">downstream</span>
software. Other outputs are focused on the inferred proteins identified from the PSMs, as well
as phosphorylation reports, relevant if a phosphoproteomics experiment has been undertaken.
More detailed information on peptide inference using SearchGUI and PeptideShaker can be found in
our tutorial on <a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">Peptide and Protein ID</a>.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="removing-contaminants"><em>Removing Contaminants</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-contaminants-from-psm-report"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove contaminants from PSM report</h3>
  <p>This Select tool is used to remove all the contaminants from the Peptide Spectral Match (PSM) search results.</p>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">output_psm</code> (PSM report output of <strong>Peptide Shaker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">NOT Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">con_</code></li>
      </ul>
    </li>
    <li>
      <p>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> to output file to <code class="language-plaintext highlighter-rouge">PSM_Report_no_contaminants</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-removing-contaminants-from-peptide-report"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Removing contaminants from Peptide report</h3>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">output_peptides</code> (Peptide Report output of <strong>Peptide Shaker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">NOT Matching</code></li>
        <li><em>“the pattern”</em>: <code class="language-plaintext highlighter-rouge">con_</code></li>
      </ul>
    </li>
    <li>
      <p>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the output file to <code class="language-plaintext highlighter-rouge">Peptide_Report_no_contaminants</code></p>

      <blockquote class="comment">
        <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>In Proteomics, contamination is generally detected as peaks in spectra that did not originate
from the samples and can be introduced in the sample from a variety of environmental sources or human error. Identification of these
contaminants is critical to enable their removal before data analysis, mainly, to maintain the validity of conclusions
drawn from statistical analyses. Thus, this selection tool helps us remove the contaminants that were identified in the spectral data.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Why is removing contaminants important?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Ideally, we would like to remove known contaminants from our samples just to maintain discovering novel proteoforms in our sample.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="removing-file-extensions-for-quantification"><em>Removing file extensions for Quantification</em></h2>
<p>This is a data manipulation step to make the data compatible with other <span class="notranslate">downstream</span> processing tools. The Replace text tool replaces the .mgf extension from the PSM report so that it can be used as an input for FlashLFQ.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-removing-file-extensions"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Removing file extensions</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3"><strong>Replace Text in a specific column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.1.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: <code class="language-plaintext highlighter-rouge">PSM_Report_no_contaminants</code> (output of <strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Replacement”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Replacement”</em>
              <ul>
                <li><em>“in column”</em>: <code class="language-plaintext highlighter-rouge">Column: 10</code></li>
                <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">.mzml.mgf</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename Input_for_FlashLFQ.</p>

      <blockquote class="comment">
        <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Replace Text searches given columns and finds and replaces patterns provided by the user.
This tool is removing the extensions (.raw,.mzml,.mgf) in the spectral file column provided by the PeptideShaker tool. This step is critical for FlashLFQ to work.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="extracting-peptide-list"><em>Extracting Peptide list</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-selecting-peptide-list"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Selecting peptide list</h3>
  <p>This step selects the peptide column from the Select output ( where we have removed the contaminants)</p>

  <ol>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">Peptide_Report_no_contaminants</code> (output of <strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>Rename file as peptide_list.</li>
  </ol>
</blockquote>

<h1 id="peptide-quantification"><strong>Peptide Quantification</strong></h1>

<p>In this tutorial, we are using FlashLFQ as the quantitation tool. The user can choose to work with other quantitation tools, e.g. moFF and MaxQuant are available in <span class="notranslate">Galaxy</span>.</p>

<h3 id="flashlfq"><em>FlashLFQ</em></h3>
<p><a href="https://github.com/smith-chem-wisc/FlashLFQ">FlashLFQ</a> can quantify MS peaks in order to find the abundances of peptides. FlashLFQ is a fast label-free quantification algorithm. Additionally, the abundances of peptides within the sample can be compared between samples as further analysis beyond this <span class="notranslate">workflow</span>.</p>
<blockquote class="notranslate hands_on">

  <h3 id="hands_on-hands-on-quantification"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quantification</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/flashlfq/flashlfq/1.0.3.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/flashlfq/flashlfq/1.0.3.0"><strong>FlashLFQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/flashlfq/flashlfq/1.0.3.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“identification file”</em>: <code class="language-plaintext highlighter-rouge">Input_for_FlashLFQ</code> (output of <strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“spectrum files”</em>: <code class="language-plaintext highlighter-rouge">MZML dataset collection</code></li>
        <li><em>“match between runs”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Use experimental design for normalization or protein fold-change analysis”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“ExperimentalDesign.tsv”</em>: <code class="language-plaintext highlighter-rouge">ExperimentalDesign.tsv</code></li>
            <li><em>“Perform Bayesian protein fold-change analysis”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>“control condition for Bayesian protein fold-change analysis”</em>: <code class="language-plaintext highlighter-rouge">S1</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p><a href="https://github.com/smith-chem-wisc/FlashLFQ">FlashLFQ</a> is a label-free quantification tool for mass-spectrometry proteomics. It supports both .mzML and Thermo .raw file formats.
To run FlashLFQ on <span class="notranslate">Galaxy</span>, there are three main input files:</p>
        <ul>
          <li>PSM report from Peptide Shaker (<code class="language-plaintext highlighter-rouge">Input_for_FlashLFQ</code>)</li>
          <li>MZML/ RAW spectrum files (<code class="language-plaintext highlighter-rouge">MZML dataset collection</code>)</li>
          <li>Experimental Design File ( <code class="language-plaintext highlighter-rouge">ExperimentalDesign.tsv</code>)
The Experimental Design file should be a tabular file with a “File”, “Condition”, “Sample”, “Fraction”, and “Replicate” column. The “File” column should match your mzml spectrum file name.</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>
      <p>Can FlashLFQ be used with fractionated data?</p>
    </li>
    <li>
      <p>Does FlashLFQ perform peptide and protein level quantification?</p>
    </li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>Yes, FlashLFQ can be used with fractionated datasets and multiple conditions</p>
      </li>
      <li>
        <p>FlashLFQ performed both peptide level and protein level quantification. For protein level, FlashLFQ used Bayesian Fold change analysis.</p>
      </li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="filtering-peptides-that-are-less-than-50-amino-acids"><em>Filtering peptides that are less than 50 amino acids</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-peptides50-amino-acids"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting peptides&lt;50 amino acids</h3>
  <p>This is a data manipulation tool. Here, we select those peptides with less than 50 amino acids in length.</p>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter data on any column using simple <span class="notranslate">expression</span>s </strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">peptide_list</code> (output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">len(c1)&lt;=50</code></li>
        <li><em>“Number of header lines to skip”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li>
      <p>Rename as Unipept_peptide_list.</p>

      <blockquote class="comment">
        <h3 id="comment-comment-6"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>Unipept fails with peptides more than 50 amino acids in length, thus we decided to work with peptides that are less than 50 amino acids.</p>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="manipulating-text-for-metaquantome"><em>Manipulating text for metaQuantome</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-text-manipulation-for-metaquantome-intensity-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Text manipulation for metaQuantome intensity file</h3>
  <p>Regex Find And Replace goes line by line through the input file and will remove any patterns specified by the user and replace them with <span class="notranslate">expression</span>s also specified by the user. In this case, Regex Find And Replace is being used on a FlashLFQ output file and manipulating the header to make it compatible with metaQuantome along with completely removing the N-terminus and C-terminus tag in the peptide sequences.</p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.0"><strong>Regex Find And Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/regex_find_replace/regex1/1.0.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">QuantifiedPeptides</code> (output of <strong>FlashLFQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">Base Sequence</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">peptide</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">Intensity_</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>”Find Regex”</em>: <code class="language-plaintext highlighter-rouge">NH2-</code></li>
                <li><em>”Replacement”</em>: <code class="language-plaintext highlighter-rouge">leave it blank</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>”Find Regex”</em>: <code class="language-plaintext highlighter-rouge">-COOH</code></li>
                <li><em>”Replacement”</em>: <code class="language-plaintext highlighter-rouge">leave it blank</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the file as <code class="language-plaintext highlighter-rouge">Intensity</code></li>
  </ol>

</blockquote>

<h1 id="functional-and-taxonomy-annotation"><strong>Functional and Taxonomy annotation</strong></h1>

<h2 id="unipept-for-taxonomy-annotation"><em>Unipept</em> for taxonomy annotation</h2>

<p>Unipept <span class="citation"><a href="#Mesuere2018">Mesuere <i>et al.</i> 2018</a></span> is used again to match tryptic peptides and find the taxonomy and lowest common ancestor of each peptide.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-taxonomy-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Taxonomy annotation</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.0.0"><strong>Unipept</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/unipept/unipept/4.0.0</span></span>  with the following parameters:
      <ul>
        <li><em>“Unipept application”</em>: <code class="language-plaintext highlighter-rouge">pept2lca: lowest common ancestor</code>
          <ul>
            <li><em>“Equate isoleucine and leucine”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“allfields”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Peptides input format”</em>: <code class="language-plaintext highlighter-rouge">tabular</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Input Containing Peptide column”</em>: <code class="language-plaintext highlighter-rouge">Unipept_peptide_list</code> (output of <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Select column with peptides”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
        <li><em>“Choose outputs”</em>: <code class="language-plaintext highlighter-rouge">Select all</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-7"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>There are two Unipept in this <span class="notranslate">workflow</span>, One for taxonomy and other for function.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/taxa.png" alt="Taxa. " width="75%" loading="lazy" /></p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Can any other taxonomy and functional tool be used apart from Unipept?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Yes, any tool can be used for taxonomy and functional output. Please make sure the output has the information that includes peptide,taxon_name, taxon_id, genus, species etc.</li>
    </ol>

  </blockquote>

</blockquote>

<p>The JSON output from the Taxonomy can be visualized using the visualize option and Select the Unipept Taxonomyviewer.</p>

<p><img src="../../images/UnipeptJSON.png" alt="Unipept-JSON. " width="30%" loading="lazy" /></p>

<p><img src="../../images/unipept_taxonomy_viewer.png" alt="Taxa-viewer. " width="30%" loading="lazy" /></p>

<p><img src="../../images/UnipeptJSONoutput.png" alt="Output. " width="75%" loading="lazy" /></p>

<h2 id="unipept-for-functional-annotation"><em>Unipept</em> for Functional annotation</h2>

<p>Unipept is used to match tryptic peptides and find the taxonomy and Functional annotation of the peptides. Unipept is used to match sample tryptic peptides to proteins using a fast-matching algorithm. Although Unipept can be accessed and used through the web page, the use of Unipept on <span class="notranslate">Galaxy</span> allows the production of output datasets including the peptide information to be used in sequential steps. Unipept requires a list containing the peptide sequences which was generated by Query Tabular.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-functional-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Functional annotation</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.3.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.3.0"><strong>Unipept</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/unipept/unipept/4.3.0</span></span>  with the following parameters:
      <ul>
        <li><em>“Unipept application”</em>: <code class="language-plaintext highlighter-rouge">peptinfo: Tryptic peptides and associated EC and GO terms and lowest common ancestor taxonomy</code>
          <ul>
            <li><em>“Equate isoleucine and leucine”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“retrieve extra information”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“group responses by GO namespace (biological process, molecular function, cellular component)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“allfields”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Peptides input format”</em>: <code class="language-plaintext highlighter-rouge">tabular</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Input Containing Peptide column”</em>: <code class="language-plaintext highlighter-rouge">Unipept_peptide_list</code> (output of <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Select column with peptides”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
        <li><em>“Choose outputs”</em>: <code class="language-plaintext highlighter-rouge">Select all</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-8"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>There are two Unipept in this <span class="notranslate">workflow</span>, One for taxonomy and other for function. Please select all the output options from Unipept.</p>
      </blockquote>
    </li>
  </ol>

  <p>The JSON output from the Taxonomy can be visualized using the visualize option and Select the Unipept Taxonomyviewer.</p>

</blockquote>

<h2 id="extracting-ec-values"><em>Extracting EC values</em></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-ec-numbers"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract EC numbers</h3>
  <p>The cut tool cuts out specific columns from the dataset. In this case, the cut tool is being used to extract columns 1 (peptide) and 3 (EC number) from the dataset peptinfo EC.tsv output. This is a manipulation tool for metaQuantome’s convenience.</p>

  <ol>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c3</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">output_ec_tsv</code> (output of <strong>Unipept</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>Rename EC_table.</li>
  </ol>
</blockquote>

<h2 id="filtering-confident-peptides"><em>Filtering confident peptides</em></h2>

<p>Query Tabular is a tool that can load tabular data into a SQLite database. This step precedes UniPept, as a list containing the peptide sequences must be generated. In this step a list of gene ontology (GO) terms is being generated.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-confident-peptides"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering confident peptides</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0"><strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0</span></span>  with the following parameters:
      <ul>
        <li>In <em>“Database Table”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Database Table”</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Dataset for Table”</em>: <code class="language-plaintext highlighter-rouge">output_go_tsv</code> (output of <strong>Unipept</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                <li>In <em>“Table Options”</em>:
                  <ul>
                    <li><em>“Specify Name for Table”</em>: <code class="language-plaintext highlighter-rouge">Goterm</code></li>
                    <li><em>“Specify Column Names (comma-separated list)”</em>: <code class="language-plaintext highlighter-rouge">peptide,total_protein_count,go_term,protein_count,go_name,go_funct</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT Goterm.*
FROM Goterm
WHERE ((1.0*Goterm.protein_count)/(1.0*Goterm.total_protein_count)) &gt;= 0.05
</code></pre></div>          </div>
        </li>
        <li><em>“include query result column headers”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Rename as Unipept_Function.</p>

      <blockquote class="comment">
        <h3 id="comment-comment-9"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>In the Unipept API output, the threshold is set to 0.5% of the overall number of peptides unambiguously assigned to a taxon at a particular taxonomic rank level. Here in the <span class="notranslate">Galaxy</span> platform, we are using Query tabular to perform this filtering.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="removing-hashtag-from-output"><em>Removing Hashtag from output</em></h2>

<p>This step is to remove the hashtag from the Peptide header in the Unipept output.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove--from-peptide-header"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove # from peptide header</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2"><strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: <code class="language-plaintext highlighter-rouge">Unipept_Function</code> (output of <strong>Unipept</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Replacement”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Replacement”</em>
              <ul>
                <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">#peptide</code></li>
                <li><em>“Replace with:”</em>: <code class="language-plaintext highlighter-rouge">peptide</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename file as All_functions</li>
  </ol>
</blockquote>

<h2 id="filter---ec-values"><em>Filter</em> - EC values</h2>

<p>We are using this Query tabular to rename the output that we obtained from the Cut column tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-ec-for-metaquantome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting EC for metaQuantome</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0"><strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0</span></span>  with the following parameters:
      <ul>
        <li>In <em>“Database Table”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Database Table”</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Tabular Dataset for Table”</em>: <code class="language-plaintext highlighter-rouge">EC_table</code> (output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                <li>In <em>“Table Options”</em>:
                  <ul>
                    <li><em>“Specify Name for Table”</em>: <code class="language-plaintext highlighter-rouge">ec</code></li>
                    <li><em>“Specify Column Names (comma-separated list)”</em>: <code class="language-plaintext highlighter-rouge">peptide,go_ec</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“SQL Query to generate tabular output”</em>:
<code class="language-plaintext highlighter-rouge">SELECT *
FROM ec</code></li>
        <li><em>“include query result column headers”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Rename file as go_ec</li>
  </ol>
</blockquote>

<h2 id="filter---biological-functions"><em>Filter</em> - Biological Functions</h2>

<p>The filter tool allows restriction of the dataset using simple conditional statements. This step is used to filter out the GO terms with biological processes and the corresponding number of peptides associated with these terms.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-biological-processes-for-metaquantome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting biological processes for metaQuantome</h3>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Unipept_Function</code> (output of <strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c5=='biological process'</code></li>
        <li><em>“Number of header lines to skip”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li>Rename file as <code class="language-plaintext highlighter-rouge">go_bp</code></li>
  </ol>
</blockquote>

<p><img src="../../images/biologicalprocess.png" alt="Biological Processes. " width="60%" loading="lazy" /></p>

<h2 id="filter---cellular-components"><em>Filter</em> - Cellular components</h2>

<p>This step is used to filter out the GO terms with cellular components and the corresponding number of peptides associated with these terms.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-cellular-component-for-metaquantome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting cellular component for metaQuantome</h3>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Unipept_Function</code> (output of <strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c5=='cellular component'</code></li>
        <li><em>“Number of header lines to skip”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li>Rename file as <code class="language-plaintext highlighter-rouge">go_cc</code></li>
  </ol>
</blockquote>

<p><img src="../../images/cellularcomponent.png" alt="Cellular-Component. " width="60%" loading="lazy" /></p>

<h2 id="filter---molecular-function"><em>Filter</em> - Molecular Function</h2>

<p>This step is used to filter out the GO terms with molecular function and the corresponding number of peptides associated with these terms.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-molecular-function-for-metaquantome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting molecular function for metaQuantome</h3>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Unipept_Function</code> (output of <strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c5=='molecular function'</code></li>
        <li><em>“Number of header lines to skip”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li>Rename file as <code class="language-plaintext highlighter-rouge">go_mf</code></li>
  </ol>
</blockquote>

<p><img src="../../images/molecularfunction.png" alt="Molecular-Function. " width="60%" loading="lazy" /></p>

<h1 class="no_toc" id="conclusion"><strong>Conclusion</strong></h1>

<p>This completes the walkthrough of the metaQuantome data creation <span class="notranslate">workflow</span>. This tutorial is a guide to have datasets that are metaQuantome ready/compatible and can be used for metaproteomics research. We have incorporated only two conditions in this <span class="notranslate">workflow</span> but users can use as many as they want. Researchers can use this <span class="notranslate">workflow</span> with their data also, please note that the tool parameters and the <span class="notranslate">workflow</span> will be needed to be modified accordingly.</p>

<p>This <span class="notranslate">workflow</span> was developed by the <span class="notranslate">Galaxy</span>-P team at the University of Minnesota. For more information about <span class="notranslate">Galaxy</span>-P or our ongoing work, please visit us at <a href="https://galaxyp.org"><span class="notranslate">galaxy</span>p.org</a></p>



                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Use dataset collections</p>
</li>
                        
                        <li><p>With SearchGUI and PeptideShaker you can gain access to multiple search engines</p>
</li>
                        
                        <li><p>Learning the basics of SQL queries can pay off</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/proteomics/faqs/">FAQ page for the Proteomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Metaproteomics_video"><b>Metaproteomics video</b>. <a href="http://z.umn.edu/mpvideo2018">http://z.umn.edu/mpvideo2018</a></li>
<li id="Metaproteomics_community_effort"><b>Metaproteomics community effort</b>. <a href="https://z.umn.edu/gcc2017mporal">https://z.umn.edu/gcc2017mporal</a></li>
<li id="Jagtap2015">Jagtap, P. D., A. Blakely, K. Murray, S. Stewart, J. Kooren <i>et al.</i>, 2015 <b>Metaproteomic analysis using the Galaxy framework</b>. PROTEOMICS 15: 3553–3565. <a href="https://doi.org/10.1002/pmic.201500074">10.1002/pmic.201500074</a></li>
<li id="Mesuere2018">Mesuere, B., F. V. der Jeugt, T. Willems, T. Naessens, B. Devreese <i>et al.</i>, 2018 <b>High-throughput metaproteomics data analysis with Unipept: A tutorial</b>. Journal of Proteomics 171: 11–22. <a href="https://doi.org/10.1016/j.jprot.2017.05.022">10.1016/j.jprot.2017.05.022</a></li>
<li id="Galaxy-P_Metaproteomics">Metaproteomics, G.-P. <b>Galaxy-P Metaproteomics instance</b>. <a href="https://proteomics.usegalaxy.eu/">https://proteomics.usegalaxy.eu/</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=metaQuantome 1: Data creation (Proteomics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane, 2021 <b>metaQuantome 1: Data creation (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{proteomics-metaquantome-data-creation,
author = "Subina Mehta and Timothy J. Griffin and Pratik Jagtap and Emma Leith and Marie Crane",
title = "metaQuantome 1: Data creation (Galaxy Training Materials)",
year = "2021",
month = "11",
day = "10"
url = "\url{https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                
                <blockquote class="agenda follow-up">
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Metaproteomics tutorial:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
