<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Introduction to Genomics and Galaxy</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1706056126' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1706056126' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Introduction to Genomics and Galaxy">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-23 15:01:04 +0000">
<meta name="DC.creator" content="Dave Clements">
<meta name="DC.creator" content="CristÃ³bal Gallardo">

        
        
        
        
        <meta name="description" content="Galaxy is a scientific workflow, data integration, and da...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Introduction to Genomics and Galaxy">
        <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-23 15:01:04 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Introduction to Genomics and Galaxy",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Introduction to Galaxy Analyses",
    "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
    "url": "https://training.galaxyproject.org/training-material/topics/introduction/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Introduction to Genomics and Galaxy' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html",
  "timeRequired": "PT2H30M",
  "teaches": "Learn some of the basic concepts of genomics and bioinformatics\n - Familiarize yourself with the basics of Galaxy",
  "keywords": "introduction",
  "description": "The questions this  addresses are:\n - Do genes on opposite strands ever overlap?  If so, how often?\n\n\nThe objectives are:\n - Learn some of the basic concepts of genomics and bioinformatics\n - Familiarize yourself with the basics of Galaxy\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/tnabtaf/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/tnabtaf/",
      "name": "Dave Clements",
      "image": "https://avatars.githubusercontent.com/tnabtaf",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9970-6657",
      "orcid": "https://orcid.org/0000-0002-9970-6657"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "name": "CristÃ³bal Gallardo",
      "image": "https://avatars.githubusercontent.com/gallardoalba",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5752-2155",
      "orcid": "https://orcid.org/0000-0002-5752-2155"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [

  ],
  "abstract": "# Introduction to Galaxy"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1706056126'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/introduction" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Introduction to Galaxy Analyses
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/introduction/faqs/" title="Check our FAQs for the Introduction to Galaxy Analyses topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/introduction/tutorials/galaxy-intro-strands/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-introduction">
    <h1 data-toc-skip>Introduction to Genomics and Galaxy</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/tnabtaf/" class="contributor-badge contributor-tnabtaf"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/tnabtaf?s=36" alt="Avatar" width="36" height="36">Dave Clements</a><a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/gallardoalba?s=36" alt="Avatar" width="36" height="36">CristÃ³bal Gallardo</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Do genes on opposite strands ever overlap?  If so, how often?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn some of the basic concepts of genomics and bioinformatics</p>
</li>
        
        <li><p>Familiarize yourself with the basics of Galaxy</p>
</li>
        
        </ul>

        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours 30 minutes</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/workflows/" title="Introduction to Genomics and Galaxy workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        
        
        

        <li>

            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/introduction/galaxy-intro-strands" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture & Tutorial (May 2023)
            </a>
        </li>
        
        
        
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">â</abbr> â­ï¸
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 23, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00191">https://gxy.io/GTN:T00191</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>

<p>This practical aims to familiarize you with the Galaxy user interface. It will teach you how to perform basic tasks such as importing data, running tools, working with histories, creating workflows, and sharing your work.</p>

<blockquote class="comment">
  <h3 id="comment-audience"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Audience</h3>
  <p>This tutorial is for those who are new to Galaxy, genomics, and bioinformatics.  If you arenât new to bioinformatics you can just do the items listed in the Hands-On boxes (<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span>), or you can try one of the <a href="/training-material/topics/introduction/index.html">other introductory tutorials</a>.</p>

  <p>This tutorial teaches the same basic content as <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html">Galaxy 101</a>, but requires less knowledge of biology to understand the questions this tutorial addresses.</p>
</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will:</p>

<ol id="markdown-toc">
  <li><a href="#introduction-to-galaxy" id="markdown-toc-introduction-to-galaxy">Introduction to Galaxy</a></li>
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#get-human-data" id="markdown-toc-get-human-data">Get human data</a>    <ol>
      <li><a href="#get-data-into-galaxy" id="markdown-toc-get-data-into-galaxy">Get data into Galaxy</a></li>
      <li><a href="#examine-the-data" id="markdown-toc-examine-the-data">Examine the data</a></li>
      <li><a href="#naming" id="markdown-toc-naming">Naming</a></li>
    </ol>
  </li>
  <li><a href="#weve-got-the-data---whats-our-plan-for-answering-the-question" id="markdown-toc-weve-got-the-data---whats-our-plan-for-answering-the-question">Weâve got the data - whatâs our plan for answering the question?</a>    <ol>
      <li><a href="#split-the-genes-into-forward-and-reverse-datasets" id="markdown-toc-split-the-genes-into-forward-and-reverse-datasets">Split the genes into forward and reverse datasets</a></li>
      <li><a href="#check-for-overlaps" id="markdown-toc-check-for-overlaps">Check for overlaps</a></li>
    </ol>
  </li>
  <li><a href="#results-and-final-steps" id="markdown-toc-results-and-final-steps">Results and final steps.</a>    <ol>
      <li><a href="#combine-forward-and-reverse-overlapping-genes-into-one-dataset" id="markdown-toc-combine-forward-and-reverse-overlapping-genes-into-one-dataset">Combine forward and reverse overlapping genes into one dataset.</a></li>
      <li><a href="#visualize-the-overlapping-genes" id="markdown-toc-visualize-the-overlapping-genes">Visualize the overlapping genes</a></li>
      <li><a href="#do-we-have-a-problem" id="markdown-toc-do-we-have-a-problem">Do we have a problem?</a></li>
      <li><a href="#walk-through-the-workflow-tutorial" id="markdown-toc-walk-through-the-workflow-tutorial">Walk through the workflow tutorial</a></li>
    </ol>
  </li>
  <li><a href="#rerun-analysis-with-exon-data" id="markdown-toc-rerun-analysis-with-exon-data">Rerun analysis with exon data</a>    <ol>
      <li><a href="#get-the-exon-data" id="markdown-toc-get-the-exon-data">Get the exon data</a></li>
      <li><a href="#rerun-the-analysis-this-time-on-exons" id="markdown-toc-rerun-the-analysis-this-time-on-exons">Rerun the analysis, this time on exons.</a></li>
      <li><a href="#are-we-done" id="markdown-toc-are-we-done">Are we done?</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="introduction">Introduction</h1>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-requirements"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Requirements</div>

  <p>To run this practical you will need</p>

  <ol>
    <li>An internet-connected computer.  Galaxy can run on your laptop without an internet connection, but this practical requires access to resources on the web.</li>
    <li>A web browser. <a href="https://www.mozilla.org/firefox">Firefox</a> and <a href="https://www.google.com/chrome/">Google Chrome</a> work well, as does <a href="https://www.apple.com/safari/">Safari</a>.  Internet Explorer is known to have issues with Galaxy so avoid using that.</li>
    <li>Access to a Galaxy instance.  Galaxy is available in many ways. If you are doing this practical as part of a workshop, the instructor will tell you which instance to use. If you are doing this on your own, you can use <a href="https://usegalaxy.org">usegalaxy.org</a>.</li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-our-motivating-question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Our Motivating Question</div>
  <p><em>I wonder if genes on opposite strands ever overlap with each other, and if so, how common is that?</em></p>
</blockquote>

<p>To explore this question we need a basic understanding of <em>genomes, chromosomes, strands,</em> and  <em>genes.</em></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-definitions-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Definitions 1</div>

  <ul>
    <li>
      <p><strong>Genome</strong></p>

      <p>The genome is the collection of all DNA native to an organism. For humans, the genome is all of a personâs chromosomes.</p>
    </li>
    <li>
      <p><strong>Chromosome</strong></p>

      <p>The largest unit of DNA organization in an organism.  Humans have two copies of 23 chromosomes.  Chromosomes are <em>linear</em> in humans, and all animals and plants.  (Bacteria have <em>circular</em> chromosomes.)</p>
    </li>
    <li>
      <p><strong>Strand</strong></p>

      <p>Chromosomes are <em>double-stranded</em>.  One is the forward strand, is typically drawn on top, and moves from left to right. The other, reverse strand, is typically drawn on the bottom and moves from right to left.  Genes can occur on either strand.  A single gene will have parts on only one stand.</p>
    </li>
    <li>
      <p><strong>Gene</strong></p>

      <p>âWhat is a gene?â is actually a hotly debated question.  For our purposes, a gene is a section of DNA on chromosome strand that creates a molecule used by an organism.</p>
    </li>
  </ul>

  <p>Graphically, the human genome can be shown as the chromosomes that are in it:</p>

  <p><a href="../../images/human_genome_chroms.png" rel="noopener noreferrer"><img src="../../images/human_genome_chroms.png" alt="Human genome consists of 23 (pairs of) chromosomes. " width="794" height="329" loading="lazy" /></a></p>

  <p>And hereâs a representation of a chromosome, strands, and genes on the chromosome.</p>

  <p><a href="../../images/chromosome_strands_genes.png" rel="noopener noreferrer"><img src="../../images/chromosome_strands_genes.png" alt="Chromosomes are double stranded and have forward and reverse strands; strands have genes on them. " width="794" height="487" loading="lazy" /></a></p>
</blockquote>

<p>The question we are asking looks like:</p>

<p><a href="../../images/do_genes_on_opposite_strands_ever_overlap.png" rel="noopener noreferrer"><img src="../../images/do_genes_on_opposite_strands_ever_overlap.png" alt="Does a situation like this exist?" width="794" height="233" loading="lazy" /></a></p>

<p><em>Non-overlapping genes are common.  How common are overlapping genes?</em></p>

<h1 id="get-human-data">Get human data</h1>

<p>To answer this question we need to know where genes start and stop on human chromosomes.  That seems like a simple question, but if you are new to bioinformatics itâs actually a hard question to answer.  Web searches will land you at any number of useful places on the web, but without a lot of background knowledge itâs hard to know what you want:  <em>Whatâs the difference between sequence and annotation?  What are FASTA, BED, GTF, GFF3, and VCF?  What are GRCh37, GRCh38, hg19, and hg38 (and what happened to hg20 through hg37 - are they okay)?</em></p>

<p>It turns out that for this particular question (and for many others), most <strong>Galaxy</strong> instances can help us find this information.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-log-in-to-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Log in to Galaxy</div>
  <p>In your web browser, <em>go to</em> <strong>your Galaxy instance</strong> and <em>log in or register</em>.</p>
</blockquote>

<p>The Galaxy interface consists of three main parts. The available tools are listed on the left, your analysis history is recorded on the right, and the central panel will show the home page, tool forms, and dataset content.</p>

<p><a href="../../images/galaxy_interface.png" rel="noopener noreferrer"><img src="../../images/galaxy_interface.png" alt="Galaxy interface showing history panel on the right, tools panel on the left and main panel at the center. " width="1777" height="1290" loading="lazy" /></a></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-start-with-an-empty-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Start with an empty history</div>

  <ul>
    <li>
      <p>At the start of the tutorial you should ensure that you are working in a new history, i.e. one without any datasets</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ul>

</blockquote>

<h2 id="get-data-into-galaxy">Get data into Galaxy</h2>

<p>There are <a href="/training-material/topics/galaxy-interface/tutorials/get-data/slides.html">many ways to get data into a Galaxy instance</a>. We are going to use the <strong>Get Data</strong> toolbox in the <strong>Tools</strong> panel on the left.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-open-get-data-toolbox"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Open Get Data toolbox</div>

  <ol>
    <li>
      <p><em>Click</em> on the <strong>Get Data</strong> toolbox to expand it.</p>

      <p><a href="../../images/getdata.PNG" rel="noopener noreferrer"><img src="../../images/getdata.PNG" alt="The Get Data toolbox. " width="289" height="397" loading="lazy" /></a></p>
    </li>
  </ol>

</blockquote>

<p>The <strong>Get Data</strong> toolbox contains a list of data sources that this Galaxy instance can get data directly from.  <strong>Upload file</strong> is quite useful for getting data from your computer or from the web (see the <a href="/training-material/topics/galaxy-interface/tutorials/get-data/slides.html">Getting data into Galaxy slides</a>).  Today we are going to use the <strong>UCSC Main table browser</strong>.</p>

<h3 id="get-genes">Get genes</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-go-to-ucsc"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Go to UCSC</div>

  <ol>
    <li>
      <p><em>Click</em> on the tool <span><strong>UCSC Main table browser</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> to go to UCSC.</p>

      <p><a href="../../images/101_01.png" rel="noopener noreferrer"><img src="../../images/101_01.png" alt="Click on UCSC Main table browser. " width="346" height="160" loading="lazy" /></a></p>
    </li>
  </ol>

</blockquote>

<p>This will take you to the UCSC Table Browser:</p>

<p><a href="../../images/ucsc_table_browser_w_default_settings.png" rel="noopener noreferrer"><img src="../../images/ucsc_table_browser_w_default_settings.png" alt="UCSC table browser tool, first screen for genes. " width="909" height="606" loading="lazy" /></a></p>

<p>The <a href="https://genome.ucsc.edu/cgi-bin/hgTables">UCSC Table Browser</a> provides access to all the data that is shown in the <a href="https://genome.ucsc.edu/cgi-bin/hgGateway">UCSC Genome Browser</a> (see box below). If you are working on a species that UCSC supports (like human) then the Table Browser is a great place to get genomic data.</p>

<p>The Table Browser has a daunting number of options. Fortunately, they are all set to commonly used defaults, greatly simplifying things, and most of the options are already set to what we want:</p>

<ul>
  <li><strong>clade:</strong> <code class="language-plaintext highlighter-rouge">Mammal</code></li>
  <li><strong>genome:</strong> <code class="language-plaintext highlighter-rouge">Human</code></li>
  <li><strong>assembly:</strong> <code class="language-plaintext highlighter-rouge">Dec. 2013 (GRCh38/hg38)</code></li>
  <li><strong>group:</strong> <code class="language-plaintext highlighter-rouge">Genes and Gene Predictions</code></li>
  <li><strong>track:</strong> <code class="language-plaintext highlighter-rouge">GENCODE v41</code></li>
</ul>

<p><strong>clade</strong> and <strong>genome</strong> seem pretty clear.  <strong>assembly</strong> asks which version/definition of the human genome we want.  (Any will do for our question, but UCSC is suggesting <code class="language-plaintext highlighter-rouge">hg38</code>, which is also the most recent.)  <strong>group</strong> is set to <code class="language-plaintext highlighter-rouge">Genes and Gene Predictions</code> which sounds like what we want. So far so good.</p>

<p><strong>track</strong> has a bewildering list of options. UCSC suggests <code class="language-plaintext highlighter-rouge">GENCODE v41</code>.  A web search leads us to the <a href="https://www.gencodegenes.org/">GENCODE web site</a> which prominently states:</p>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-all-gencode-is-different-from-gencode"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: ALL GENCODE is different from GENCODE</div>
  <p><strong>Warning</strong> The Table browser only provides the most recent release of GENCODE which is updated several times per year. ALL GENCODE does not contain the same data as GENCODE and you should select the GENCODE track even if the version number is wrong.</p>
</blockquote>

<blockquote class="quote">
  <p>The goal of the GENCODE project is to identify and classify all gene features in the human and mouse genomes with high accuracy based on biological evidenceâ¦</p>
</blockquote>

<p>Time for a few more definitions.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-definitions-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Definitions 2</div>

  <ul>
    <li>
      <p><strong>Reference genome</strong></p>

      <p>A reference genome is the <em>genome of a single individual</em> that has been thoroughly studied, to the point that we know exactly what most of that individualâs DNA is.  In practice a reference genome is used as shared map by researchers working on that organism. Reference genomes are updated periodically as techniques improve.</p>
    </li>
    <li>
      <p><strong>Sequence</strong></p>

      <p>A genomeâs sequence describes the DNA in that genome, down to the A, C, T, and G (single nucleotide) level including the exact location where each is.  Given a reference genome, you can ask questions like, âWhatâs the DNA on chromosome 2 between positions 1,678,901 and 1.688,322?â</p>
    </li>
    <li>
      <p><strong>Genome/Gene annotation</strong></p>

      <p>The sequence tells us what DNA is where, but it doesnât tell us anything about the function of that DNA.  <em>Annotation</em> is additional information about particular regions of the genome like where genes, repeats, promotors, and centromeres are, or how active a particular gene is.</p>
    </li>
  </ul>
</blockquote>

<p>The <strong>track</strong> option asks us which set of annotations do we want to get?  There are so many choices because annotation is the result of analysis and interpretation, and there are many ways to do this. (And in this case, many of the options arenât even genes or gene predictions.)</p>

<p>GENCODE is âhigh-qualityâ and  âgene annotation.â That sounds like a good thing to use.  Lets stay with the default: <code class="language-plaintext highlighter-rouge">GENCODE V41</code>.</p>

<p>So far we havenât changed <em>anything</em> from the defaults.  Lets change something.  The default  <strong>region</strong> is the whole genome, which can be done, but itâs a lot of information. For this exercise lets use just one (small) chromosome.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-limit-the-region-and-get-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Limit the region and get the data.</div>

  <ol>
    <li>Say that we just want chromosome 22
      <ul>
        <li>For <strong>region</strong> select <code class="language-plaintext highlighter-rouge">position</code>.</li>
        <li>In the text box next to <code class="language-plaintext highlighter-rouge">position</code> enter <code class="language-plaintext highlighter-rouge">chr22</code> (case matters).</li>
      </ul>

      <p><a href="../../images/ucsc_tb_set_region_chr22.png" rel="noopener noreferrer"><img src="../../images/ucsc_tb_set_region_chr22.png" alt="Change the region. " width="530" height="83" loading="lazy" /></a></p>
    </li>
    <li>
      <p><em>Click</em> the <strong>get output</strong> button.</p>

      <p>And, that doesnât actually get us the output.  It sends us to a second UCSC page that asks us exactly what we want.</p>

      <p><a href="../../images/ucsc_tb_2nd_page_whole_gene.png" rel="noopener noreferrer"><img src="../../images/ucsc_tb_2nd_page_whole_gene.png" alt="UCSC Table Browser 2nd page. " width="1157" height="647" loading="lazy" /></a></p>
    </li>
    <li>
      <p>Under <strong>Create one BED record per</strong> make sure that <strong>Whole Gene</strong> is selected.</p>
    </li>
    <li><em>Click</em> the <strong>Send query to Galaxy</strong> button at the bottom of the form.</li>
  </ol>
</blockquote>

<p>This returns us to Galaxy, first displaying a big green box (thatâs good!) and then returning us to the view we started with.  Except that we now have an item in our history, the dataset from UCSC.</p>

<h3 id="history-item-status">History Item Status</h3>

<p>Watch your new history item.  It will go through three statuses before itâs done.</p>

<table>
  <thead>
    <tr>
      <th>Color</th>
      <th>Icon</th>
      <th>Meaning</th>
      <th>Â </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><strong>Grey</strong></td>
      <td>Clock</td>
      <td>Item is waiting to start (waiting for data transfer to start)</td>
      <td><a href="../../images/status_queued.png" rel="noopener noreferrer"><img src="../../images/status_queued.png" alt="Status: Queued. " width="250" height="38" loading="lazy" /></a></td>
    </tr>
    <tr>
      <td><strong>Yellow</strong></td>
      <td>Spinner</td>
      <td>Item is running (data is actively being transferred).</td>
      <td><a href="../../images/status_running.png" rel="noopener noreferrer"><img src="../../images/status_running.png" alt="Status: Running. " width="250" height="38" loading="lazy" /></a></td>
    </tr>
    <tr>
      <td><strong>Green</strong></td>
      <td>None</td>
      <td>Item has finished successfully (data transfer complete).</td>
      <td><a href="../../images/status_finished_success.png" rel="noopener noreferrer"><img src="../../images/status_finished_success.png" alt="Status: Successfully finished. " width="250" height="38" loading="lazy" /></a></td>
    </tr>
    <tr>
      <td><strong>Red</strong></td>
      <td>Cross</td>
      <td>The job has failed. There can be <a href="https://galaxyproject.org/support/tool-error/">many reasons</a>.</td>
      <td><a href="../../images/status_failed.png" rel="noopener noreferrer"><img src="../../images/status_failed.png" alt="Status: Failed. " width="250" height="38" loading="lazy" /></a></td>
    </tr>
  </tbody>
</table>

<p>You can find more information in the <a href="/training-material/topics/galaxy-interface/tutorials/history/tutorial.html">Undestanding Galaxy history system</a> training.</p>

<h2 id="examine-the-data">Examine the data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-look-at-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Look at the data.</div>
  <p>Once the dataset is green, <em>click</em> on the dataset name (something like <strong>UCSC Main on Humanâ¦</strong>)</p>
</blockquote>

<p>This expands the dataset and shows you information about it, and a preview of its contents.</p>

<p><a href="../../images/genes_human_chr22_preview.png" rel="noopener noreferrer"><img src="../../images/genes_human_chr22_preview.png" alt="Preview of the data from UCSC Table Browser. " width="545" height="351" loading="lazy" /></a></p>

<p>From this preview we can obtain a huge amount of useful information:</p>
<ol>
  <li>The dataset has over 4000 regions, meaning that there are over 4000 genes on chromosome 22.</li>
  <li>The dataset is in <strong>BED</strong> format, one of several standards for representing genome annotation.  We got BED format because BED was preselected as the output format in the UCSC table browser.</li>
  <li>The datasetâs âdatabaseâ is <strong>hg38</strong>.  This says which revision of the reference genome this data maps too.  hg38 is the latest human reference genome.  hg38 was also selected by default in UCSC.</li>
  <li>Finally, it shows us the first 5 rows in the dataset.</li>
</ol>

<p>The dataset preview is informative, but you canât see much of the actual dataset.  Lets use one of the dataset icons to see the whole dataset:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-look-at-all-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Look at all the data.</div>
  <ol>
    <li>
      <p>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon to view the contents of the dataset. This displays all of the data.</p>

      <p><a href="../../images/genes_human_chr22_dataset_view.png" rel="noopener noreferrer"><img src="../../images/genes_human_chr22_dataset_view.png" alt="Full dataset in central panel. " width="991" height="594" loading="lazy" /></a></p>
    </li>
    <li>
      <p>Use the side scroll bars to move through the file.</p>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>How many columns include the BED file?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>Our BED file includes 12 columns, on which the information about the genes distribution along the genome in the chromosome 12 is summarized.</p>
    </blockquote>

  </blockquote>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-additional-information-about-bed-files"><button class="gtn-boxify-button details" type="button" aria-controls="details-additional-information-about-bed-files" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Additional information about BED files<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a> is one of several well-established tabular formats for genomic data.  Other formats include GFF3 and GTF.  For the type of analysis we are doing today, BED format is easiest to work with.  BED was created to power the UCSC Genome Browser.  BED files contain between 3 and 15 columns.  Our example BED file describes genes and contains 12 columns.</p>

  <p>We care about columns 1, 2, 3, and 6:</p>

  <table>
    <thead>
      <tr>
        <th>#</th>
        <th>Column Name</th>
        <th>Meaning</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>1</td>
        <td>Chromosome</td>
        <td>The name of the chromosome this gene is on.</td>
      </tr>
      <tr>
        <td>2</td>
        <td>Start</td>
        <td>Where on the chromosome the gene starts.</td>
      </tr>
      <tr>
        <td>3</td>
        <td>End</td>
        <td>Where on the chromosome the gene ends.</td>
      </tr>
      <tr>
        <td>6</td>
        <td>Strand</td>
        <td>Which strand the gene is on.  <code class="language-plaintext highlighter-rouge">+</code> means forward (top, left to right), <code class="language-plaintext highlighter-rouge">-</code> means reverse (bottom, right to left)</td>
      </tr>
    </tbody>
  </table>

  <p>See the <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED format description at UCSC</a> for a full description of all the columns.</p>

</blockquote>

<h2 id="naming">Naming</h2>

<p>Galaxy allows you to name your analyses (your histories) and your datasets.  We only have one history (âUnnamed historyâ) and one dataset (âUCSC Main on Human:â¦â) so far, but itâs a good idea to</p>

<ol>
  <li>Always name your histories</li>
  <li>Name your input and final output datasets, and any significant intermediate datasets.</li>
</ol>

<p>You donât have to do this.  Galaxy is quite happy for you to have an infinite number of âUnnamed historyâ histories, and to have all your datasets be obscurely named.  However, once youâve run your first 5 unnamed analyses, all with obscurely named datasets, youâll might wish you would have named everything.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-name-your-stuff"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Name your stuff</div>

  <ol>
    <li><strong>Name your history</strong> to be meaningful and easy to find.
      <ul>
        <li><em>Click</em> on the title of the history and enter something like <strong>Intro - Strands</strong> as the name.  Hit the <code class="language-plaintext highlighter-rouge">enter</code> key on your keyboard to save it.
<a href="../../../../shared/images/rename_history.png" rel="noopener noreferrer"><img src="../../../../shared/images/rename_history.png" alt="Rename the history. " width="270" height="320" loading="lazy" /></a></li>
      </ul>
    </li>
    <li><strong>Rename your dataset</strong>
      <ul>
        <li><em>Click</em> on the <strong>pencil icon</strong> to edit the dataset attributes.</li>
        <li>In the next screen change the name of the dataset to something like <code class="language-plaintext highlighter-rouge">Genes</code> or <code class="language-plaintext highlighter-rouge">Genes chr22</code>.</li>
        <li><em>Click</em> the <strong>Save</strong> button at the bottom of the screen.</li>
      </ul>

      <p>Your history should now look something like this:</p>

      <p><a href="../../images/genes_human_chr22_rename.png" rel="noopener noreferrer"><img src="../../images/genes_human_chr22_rename.png" alt="Rename dataset. " width="903" height="355" loading="lazy" /></a></p>
    </li>
  </ol>
</blockquote>

<h1 id="weve-got-the-data---whats-our-plan-for-answering-the-question">Weâve got the data - whatâs our plan for answering the question?</h1>

<p>You have to know whatâs possible, before you can build a plan.  If you donât have experience with data analysis then you might not have any idea how you would answer our question.  Before we dive in using a particular solution, think about how you might solve this.  If you donât have any experience with tools, then think about how you might solve it manually, using pencil and paper (it may help to assume you have an infinite supply of helpers to do the pencil and paper work).</p>

<p>Hereâs how weâll answer this question:</p>

<ol>
  <li>Split the genes dataset in two: one for genes on the forward strand, and one for genes on the reverse strand.</li>
  <li>Compare the two datasets to see which ones, if any, overlap.</li>
  <li>Check how many (or what percentage) of our genes overlap with another gene.</li>
</ol>

<p>It turns out that all of these steps are easy in Galaxy!</p>

<h2 id="split-the-genes-into-forward-and-reverse-datasets">Split the genes into forward and reverse datasets</h2>

<p>How might we do this?  Column 6 contains the strand information.  Can we split genes into two datasets based on the value of Column 6.  How?  Lets take a look at our available tools.  And <em>whoa! There are over 40 toolboxes, and several hundred tools.</em> How are we going to find a tool that can do the split?</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-searching-for-and-launching-a-tool"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Searching for and launching a tool</div>

  <ol>
    <li>Try the <strong>tools search box</strong>.  Think of terms that might describe what we want to do and type them in the search box.  Do you see anything promising?  Explore a little.</li>
    <li>If you havenât already searched with it, <em>enter</em> <code class="language-plaintext highlighter-rouge">filter</code> in the <strong>tool search box</strong>.  Among the results you should see
      <ul>
        <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="link"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span> data on any column using simple expressions.</li>
        <li>That might work.</li>
      </ul>
    </li>
    <li><em>Click</em> on <span class="tool" data-tool="Filter1" title="Filter tool" aria-role="link"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span> to open the Filter tool in the central panel.</li>
    <li>Take a look at the <strong>Syntax</strong> and <strong>Example</strong> sections to understand what the tool does.</li>
  </ol>

  <p><a href="../../images/filter_tool_defaults.png" rel="noopener noreferrer"><img src="../../images/filter_tool_defaults.png" alt="The Filter Tool form. " width="883" height="417" loading="lazy" /></a></p>
</blockquote>

<p>It doesnât say anything about <strong>Filter</strong> being able to split a file into multiple files.  It does look like we can use Filter to get only genes on the forward strand, or only genes on the reverse strand.  We would have to run Filter twice, once for forward strand genes, and once for reverse strand genes. Letâs do that.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-filter-tool-to-get-genes-on-the-forward-strand"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run the Filter tool to get genes on the forward strand.</div>

  <ul>
    <li>
      <p>The filter tool has 3 fields:</p>

      <ol>
        <li><strong>Dataset</strong>: This pulldown will list any dataset from your history that this tool can work on.  In your case thatâs probably only one dataset. <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>âfilterâ</em>: <code class="language-plaintext highlighter-rouge">Genes chr22</code> .</li>
        <li><strong>Condition</strong>: this free text field is where we specify which records we want in the output dataset. Put in <em>âConditionâ</em>:  <code class="language-plaintext highlighter-rouge">c6 == "+" </code> .
 This specifies that column 6 (the strand) must be equal to (<code class="language-plaintext highlighter-rouge">==</code> is Python for <em>is equal to</em>) a plus sign.</li>
        <li>
          <p><em>âHeader lines to skipâ</em>:  <code class="language-plaintext highlighter-rouge">0 </code></p>

          <p>Our dataset does not have any header lines.</p>
        </li>
      </ol>
    </li>
    <li>
      <p>Finally, <em>click</em> the <strong>Execute</strong> button.</p>

      <p><a href="../../images/filter_tool_forward_strand_genes_only.png" rel="noopener noreferrer"><img src="../../images/filter_tool_forward_strand_genes_only.png" alt="Run the Filter tool to get only the forward strand genes. " width="724" height="337" loading="lazy" /></a></p>
    </li>
  </ul>

</blockquote>

<p>This adds another dataset to your history. This one should contain only genes on the forward strand.  Once the dataset is green, click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon to confirm this. We also recommend that you rename this dataset to something like <code class="language-plaintext highlighter-rouge">Genes, forward strand</code> (remember how?).</p>

<p>Now we want to get the genes on the reverse strand.  There are actually many ways to get this.  Here are two of them.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-genes-on-the-reverse-strand"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get genes on the reverse strand</div>

  <p><strong>Method 1</strong></p>

  <ol>
    <li>Open the dataset preview by <em>clicking</em> on the name of the <code class="language-plaintext highlighter-rouge">Genes, forward strand</code> dataset.  This shows an icon than the uploaded <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset did not: a looping arrow.</li>
    <li><em>Click</em> the <strong>looping arrow</strong> (âRun this job againâ) icon.  This wonât actually run the job again.  What it will do is bring up the Filter tool form with <em>the exact same settings that were used to produce this dataset.</em></li>
    <li>Rerun <span class="tool" data-tool="Filter1" title="Filter tool" aria-role="link"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span> but with
      <ul>
        <li><em>âConditionâ</em>:  <code class="language-plaintext highlighter-rouge">c6 == "-" </code></li>
      </ul>
    </li>
    <li><em>Click</em> the <strong>Run Tool</strong> button.</li>
  </ol>

  <p><strong>Method 2</strong></p>
  <ol>
    <li><em>Click</em> on <span class="tool" data-tool="Filter1" title="Filter tool" aria-role="link"><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span> in the tool panel to open the Filter tool in the central panel.</li>
    <li><em>Fill</em> the form as before, <em>except</em>:
      <ul>
        <li>Make sure the <strong>Dataset</strong> pulldown is set to the <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset.</li>
        <li><em>Set</em> <strong>Condition</strong> to <code class="language-plaintext highlighter-rouge">c6 == "-"</code>.</li>
      </ul>
    </li>
    <li><em>Click</em> the <strong>Run Tool</strong> button.</li>
  </ol>

  <p><strong>Both Methods</strong></p>
  <ol>
    <li><em>Rename</em> your new dataset to something like <code class="language-plaintext highlighter-rouge">Genes, reverse strand</code></li>
  </ol>
</blockquote>

<p>The rerun button can be a huge help as you run more complex tools.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-empty-result"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Empty result?</div>

  <p>If you used Method 2 and didnât explicitly set the dataset, then you ran Filter on the <code class="language-plaintext highlighter-rouge">Genes, forward strand</code> dataset. None of the genes in the forward strand dataset have â-â in column 6 so all of them were filtered out from the result.</p>

  <p>Try again and set the dataset to your <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset.</p>
</blockquote>

<p>Your history should now have (at least) 3 datasets in it, with names like:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">Genes chr22</code></li>
  <li><code class="language-plaintext highlighter-rouge">Genes, forward strand</code></li>
  <li><code class="language-plaintext highlighter-rouge">Genes, reverse strand</code></li>
</ul>

<p>The number of genes in the <code class="language-plaintext highlighter-rouge">forward</code> plus <code class="language-plaintext highlighter-rouge">reverse</code> datasets should be the same as in the <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset.  If they arenât can you figure out why?</p>

<h2 id="check-for-overlaps">Check for overlaps</h2>

<p>Genes are an example of a <em>genomic interval</em>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-definitions-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Definitions 3</div>

  <ul>
    <li><strong>Genomic interval</strong>
In Galaxy, a <em>genomic interval</em> is a something that spans part of a chromosome (or some other linear frame of reference like a contig).  Genes are a common example of a genomic interval.  Even a chromosome is a genomic interval, albeit a very long one.</li>
  </ul>
</blockquote>

<p>Galaxy excels at answering questions about genomic intervals and different sets of genomic intervals relate to each other.  Lets take a look.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-genomic-interval-tools"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Genomic Interval Tools</div>

  <ul>
    <li>The next step is finding overlapping intervals, so type <code class="language-plaintext highlighter-rouge">interval</code> in the tool search box. There are many results from this search, so you might want to click the <em>Show sections</em> button (available on Galaxy servers) to see which sections the tools are organised into.</li>
    <li><em>Explore</em> the tools in this toolbox, looking for something that we can use to see which genes on opposite strands overlap.</li>
  </ul>
</blockquote>

<p>Of the tools in the <strong>Operate on Genomic Intervals</strong> toolbox, <strong>Join</strong> and particularly <strong>Intersect</strong> have the most promise.  Letâs try <strong>Intersect</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-genomic-interval-tools-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Genomic Interval Tools</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/intersect/gops_intersect_1/0.0.1" title="Intersect tool" aria-role="link"><strong>Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/intersect/gops_intersect_1/0.0.1</span></span> with the following parameters:
      <ul>
        <li><em>âReturnâ</em>:  <code class="language-plaintext highlighter-rouge">Overlapping Intervals</code>.
This looks like it might return whole genes, while <code class="language-plaintext highlighter-rouge">Overlapping pieces</code> may return only the parts that overlap.  We suspect that whole genes might be more useful.</li>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span><em>âofâ</em>:  <code class="language-plaintext highlighter-rouge">Genes, forward strand</code> (the first dataset)</li>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>âthat intersectâ</em> : <code class="language-plaintext highlighter-rouge">Genes, reverse strand</code> (the second dataset)</li>
        <li>
          <p><em>âfor at leastâ</em>: <code class="language-plaintext highlighter-rouge">1</code></p>

          <p>This will return genes with even just one position overlapping.</p>
        </li>
        <li><em>Click</em> <strong>Run Tool</strong>.</li>
      </ul>

      <p><a href="../../images/genes_human_intersect_strands.png" rel="noopener noreferrer"><img src="../../images/genes_human_intersect_strands.png" alt="Run Intersect. " width="1195" height="515" loading="lazy" /></a></p>
    </li>
    <li>
      <p>Now repeat the intersect, but make the first dataset be the reverse genes, and the second be the forward genes.</p>
    </li>
    <li>Finally give both of the new datasets meaningful names, like <code class="language-plaintext highlighter-rouge">Overlapping forward genes</code> and <code class="language-plaintext highlighter-rouge">Overlapping reverse genes</code></li>
  </ol>
</blockquote>

<h1 id="results-and-final-steps">Results and final steps.</h1>

<p>At this point we <em>could</em> say that we have answered our question. Using dataset previews in the history panel, we can compare the number of genes in the <code class="language-plaintext highlighter-rouge">Overlapping forward</code> and <code class="language-plaintext highlighter-rouge">Overlapping reverse</code> datasets with the number of genes in the full <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset, and <em>conclude that overlapping genes on opposite strands are actually pretty common.</em></p>

<p>However, before we rush off to publish our conclusions, letâs</p>

<ol>
  <li>Get both the forward and reverse overlapping genes into a single dataset (one link will look better in our publication), and</li>
  <li><em>Visualize</em> our new dataset, just to double-check our results.</li>
</ol>

<h2 id="combine-forward-and-reverse-overlapping-genes-into-one-dataset">Combine forward and reverse overlapping genes into one dataset.</h2>

<p>What tool can we use to combine the two datasets into one?  Try <em>searching</em> for <code class="language-plaintext highlighter-rouge">combine</code> or <code class="language-plaintext highlighter-rouge">join</code> or <code class="language-plaintext highlighter-rouge">stack</code> in the tool search box.  Youâll find lots of tools, but none of them do what we want to do. <em>Some times you just have to manually look through toolboxes to find what you need.</em>  Where should we look?  Itâs probably not <strong>Get Data</strong> or <strong>Send Data</strong>, but it could easily be in any of the next 4 toolboxes: <strong>Lift-Over, Collection Operations, Text Manipulation, or Datamash</strong>.</p>

<p>It turns out that <strong>Lift-Over</strong> and <strong>Collection Operations</strong> are not what we want.  (But, take a look at them: if you are going to work with genomic data it wonât take long before youâll need both.)</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-concatenate-two-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Concatenate two datasets</div>

  <ol>
    <li><em>Open</em> the <strong>Text Manipulation</strong> toolbox.</li>
    <li>Near the top of the toolbox is <strong>Concatenate datasets tail-to-head</strong>. <em>Click</em> on it.  Lets try that tool.</li>
    <li><span class="tool" data-tool="cat1" title="Concatenate tool" aria-role="link"><strong>Concatenate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: cat1</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>âConcatenate Datasetâ</em>: <code class="language-plaintext highlighter-rouge">Overlapping reverse genes</code>.</li>
        <li><em>âDataset</em>â</li>
      </ul>
      <ul>
        <li>
          <p>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>âInsert Datasetâ</em></p>

          <p>This adds a second dataset pull-down menu to the form.</p>
        </li>
        <li>
          <p>In <em>â1: Datasetâ</em></p>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>âSelectâ</em>: <code class="language-plaintext highlighter-rouge">Overlapping forward genes</code> as the second dataset.</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><em>Click</em> <strong>Run Tool</strong></li>
    <li><em>Rename</em> the resulting dataset something informative like <code class="language-plaintext highlighter-rouge">Overlapping genes</code></li>
  </ol>
</blockquote>

<p>Once the concatenate operation is finished, preview the dataset in your history panel.  Does it have the expected number of genes in it?  If not, see if you can figure out what happened.</p>

<h2 id="visualize-the-overlapping-genes">Visualize the overlapping genes</h2>

<p>Galaxy knows about several visualization options for lots of different dataset types, including BED.  Whenever you preview a dataset in the history panel, Galaxy provides links to these visualizations.  For BED files (which is the format we have), options include <strong>IGB, IGV,</strong> and <strong>UCSC main.</strong>  IGB and IGV are widely used desktop applications and eventually you may want to install one or both of them.  For now, letâs visualize the data at UCSC, using the UCSC <em>Genome</em> Browser.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-display-data-in-genome-browser"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Display data in Genome Browser</div>

  <ol>
    <li>Click on your <code class="language-plaintext highlighter-rouge">Overlapping genes</code> dataset in your history panel. This will show the dataset preview in the history panel.</li>
    <li>Click to expand the dataset, if it isnât already, so that you can see the dataset metadata and additional actions like Visualize.</li>
    <li>Click on the <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> (<strong>Visualize</strong>) icon</li>
    <li>Click on the <strong>display at UCSC (main)</strong> link that appears in the blue box at the top of the screen.
<a href="/training-material/topics/introduction/images/101_displayucsc.png" rel="noopener noreferrer"><img src="/training-material/topics/introduction/images/101_displayucsc.png" alt="visualisation options are shown in Galaxy's middle panel. " width="457" height="370" loading="lazy" /></a></li>
  </ol>

  <p>This will launch a new window, showing UCSCâs Genome Browser with our dataset shown right at the top.  UCSC figures out that our first overlapping gene is ~11 million bases into chromosome 22, and it has landed us there.</p>

  <p><a href="../../images/ucsc_gb_genes_strands_overlapping.png" rel="noopener noreferrer"><img src="../../images/ucsc_gb_genes_strands_overlapping.png" alt="UCSC Genome Browser. " width="982" height="297" loading="lazy" /></a></p>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-background-ucsc-genome-browser"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Background: UCSC Genome Browser</div>

  <ul>
    <li>
      <p><em>Genome browsers</em> are software for viewing genomic information graphically.  The <a href="https://genome.ucsc.edu/cgi-bin/hgGateway">UCSC Genome Browser</a> (and most genome browsers) typically display different types of <em>annotation</em> about a region of a genome.  This is displayed as a stack of <em>tracks</em> and each track contains a different type of information.</p>
    </li>
    <li>
      <p>Genome browsers are useful for seeing information in context and for seeing (and discovering) correlations between different types of information.  (They are also useful for visually checking results, which is what we are doing now.)</p>
    </li>
    <li>
      <p>The UCSC Genome Browser has information on over 100 animals, and their <a href="http://archaea.ucsc.edu/cgi-bin/hgGateway?db=pyrFur2">Archaeal Genome Browser</a> has genomic information on well over 100 microbial species.</p>
    </li>
  </ul>
</blockquote>

<p>Now, take a look at one of our results.  (Any pair of overlapping genes will do.)  Our data is in the second to top track (<strong>User Track / User Supplied Track</strong>). That track shows a line of small black boxes, sometimes connected with a line.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-zoom-in-on-an-area-of-the-chromosome-that-shows-a-set-of-linked-black-boxes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Zoom in on an area of the chromosome that shows a set of linked black boxes</div>
  <p>To zoom in,</p>
  <ul>
    <li><em>Click</em> on the <strong>Scale</strong> track (the top track) just to the left of the start of the black boxes.</li>
    <li>Now <em>drag</em> the mouse across the Scale track to just to the right of the  black boxes and let go.
<a href="../../images/ucsc_gb_zoom_by_drag.png" rel="noopener noreferrer"><img src="../../images/ucsc_gb_zoom_by_drag.png" alt="Zoom by clicking and dragging on the Scale track. " width="967" height="66" loading="lazy" /></a></li>
    <li>A window pops up describing several ways to interact with the browser.  Just <em>click</em> the <strong>Zoom In</strong> button at the bottom.</li>
    <li>This redraws the window, this time zoomed in to what you highlighted.</li>
    <li>Continue to zoom in until you have the set of linked black boxes you picked centered on the screen.</li>
    <li>Once you are as zoomed as you want to be, click on one of the linked boxes.  This will expand the track:
<a href="../../images/ucsc_gb_zoom_to_1_gene.png" rel="noopener noreferrer"><img src="../../images/ucsc_gb_zoom_to_1_gene.png" alt="One gene, fully zoomed and expanded. " width="957" height="250" loading="lazy" /></a></li>
  </ul>
</blockquote>

<p>The black boxes connected by lines represent genes, and each set of connected boxes is a single gene (actually, a single transcript of a gene).  Take a close look at the top several tracks.</p>

<ul>
  <li>It looks like we preserved the gene definitions just fine.</li>
  <li>It looks like, sure enough, there are overlapping genes here, and they are on opposite sides.  The arrows on the connecting lines tell us which strand the gene is on.</li>
</ul>

<p>Um, <em>whatâs up with the boxes and the lines connecting them?</em></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-definitions-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Definitions 4</div>

  <ul>
    <li><strong>Exon</strong>
In humans (and in all plants and animals) the molecules that are built from genes are often only built from a part of the DNA in the gene. The sections of DNA that can produce the molecules are called <em>exons</em>.</li>
  </ul>
</blockquote>

<p>As you may have guessed (or already knew): The black boxes are exons.  <em>Genes</em> are defined as <em>covering the entire area from the first black box to the last connected black box.</em></p>

<h2 id="do-we-have-a-problem">Do we have a problem?</h2>

<p><em>Maybe.</em></p>

<p>Our <em>ad hoc</em> review of identified overlapping genes in the UCSC Genome Browser has <em>(or should have!)</em> confirmed that every gene we said has an overlapping gene on the opposite strand does in fact have that.  So, our conclusion appears solid: A significant percentage of genes <em>do</em> overlap with other genes on the opposite strand.</p>

<p>But, our conclusion may not be as <em>significant</em> as we had hoped. If only parts of genes, the exons, make stuff in our bodies, then should we have run this analysis on just the exons rather then the entire genes?  Probably.</p>

<p>Letâs refine our question slightly</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-our-revised-motivating-question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Our Revised Motivating Question</div>
  <p><em>I wonder if <strong>exons</strong> on opposite strands ever overlap with each other, and if so, how common is that?</em></p>
</blockquote>

<h3 id="so-i-have-to-do-this-all-over-again-but-with-exons">So, I have to do this all over again, but with exons?</h3>

<p><em>Well, yes and no.</em>  We will have to run the analysis again, this time on exons instead of whole genes. But we wonât have to manually recreate every step of our analysis. Galaxy enables us to create a reusable <em>workflow</em> from the analysis we just did, and then we can rerun the analysis, as one step, anytime we want to, and on any dataset.</p>

<h2 id="walk-through-the-workflow-tutorial">Walk through the workflow tutorial</h2>

<p>Run the <a href="/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html">Create a reusable workflow from a history</a> tutorial for how to do this, <em>and then come back here to run your newly created workflow with the exon data.</em></p>

<h1 id="rerun-analysis-with-exon-data">Rerun analysis with exon data</h1>

<p>We want to run the same analysis, but this time only look for overlaps that happen in <em>exons</em>, the parts of genes that produce stuff our body uses.  Before we start looking at exons, letâs start a new history, one that contains only the genes file we got from UCSC. We could go back to UCSC and refetch the file, but there is an easier way.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-a-new-history-that-contains-some-data-from-current-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create a new history that contains some data from current history</div>

  <ol>
    <li><em>Click</em> on the <strong>gear icon</strong> at the top of the current history.</li>
    <li><em>Select</em> the <strong>Copy Datasets</strong> option from the pull down menu.  This launches the copy datasets form.</li>
    <li>Under <strong>Source History:</strong>, <em>select</em> the dataset(s) you want to copy to the new history.
      <ul>
        <li>In our case, we only want the <code class="language-plaintext highlighter-rouge">Genes chr22</code> dataset.</li>
      </ul>
    </li>
    <li>Under <strong>Destination History</strong> enter an informative history name in the <strong>New history named:</strong> box.
      <ul>
        <li>For example, <code class="language-plaintext highlighter-rouge">Exon overlaps on opposite strands</code>
<a href="../../images/copy_datasets_to_new_history.png" rel="noopener noreferrer"><img src="../../images/copy_datasets_to_new_history.png" alt="Copy only the Genes chr22  dataset to the new history. " width="962" height="350" loading="lazy" /></a></li>
      </ul>
    </li>
    <li><em>Click</em> the <strong>Copy History Items</strong> button to create your new history.
      <ul>
        <li>This creates a new history (with the copied dataset) and throws up a green box saying:
â1 dataset copied to 1 history: <code class="language-plaintext highlighter-rouge">name you gave your new history</code>.â</li>
      </ul>
    </li>
    <li>The history name is a link.  <em>Click</em> on it.</li>
  </ol>
</blockquote>

<h2 id="get-the-exon-data">Get the exon data</h2>

<p>And your new history appears in the history panel with the copied <em>genes</em> dataset.  What we need is <em>exons.</em> How can we get the exon information?  There are two relatively easy ways to get this information, one of which will be very familiar.</p>

<p>The first method involves going back to the UCSC Table Browser.  Everything on the first form would stay the same: We still want human, hg38, GENCODE v41, and just <code class="language-plaintext highlighter-rouge">chr22</code>. The second form is what changes.  Instead of getting the <strong>Whole Gene</strong>, we need to retrieve the <strong>Coding Exons</strong> only.</p>

<p>The second method is to use the <strong>Gene BED To Exon/Intron/Codon BED expander</strong> tool in the <strong>Operate on Genomic Intervals</strong> toolbox to extract the exon information from the genes BED file we already have. (<em>Of course!</em> you say.  Umm, there is no way that you should have known that you already had this information in the genes file, or that this tool existed.)</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-the-exon-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get the exon data</div>

  <ol>
    <li>Get the exon information, either by revisiting UCSC, or by using the <span class="tool" data-tool="gene2exon1" title="Gene BED To Exon/Intron/Codon BED expander tool" aria-role="link"><strong>Gene BED To Exon/Intron/Codon BED expander</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: gene2exon1</span></span> tool.  If you use the expander tool select <strong>Coding Exons only</strong> from the <strong>Extract</strong> pull-down.</li>
    <li>Give the resulting dataset a meaningful name.</li>
  </ol>
</blockquote>

<p>If you got the data from UCSC it will look something like this:
<a href="../../images/101_exons.png" rel="noopener noreferrer"><img src="../../images/101_exons.png" alt="Contents of the exons dataset from UCSC. " width="632" height="278" loading="lazy" /></a></p>

<p>Your history should now have two datasets: one describing entire genes, and one describing just the exons.</p>

<h2 id="rerun-the-analysis-this-time-on-exons">Rerun the analysis, this time on exons.</h2>

<p>When you did the <em>History to Workflow</em> tutorial you created a new workflow that was then added to your list of defined workflows.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-workflow-on-the-exon-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run the workflow on the exon data.</div>

  <ol>
    <li>
      <p><em>Click</em> the <strong>Workflow</strong> tab in the menu at the top of the Galaxy page.</p>

      <p>This lists all of your defined workflows, including the one you just created.</p>
    </li>
    <li>
      <p><em>Click</em> on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (Run workflow) button next to your workflow.</p>

      <p>This launches the workflow run form (which you also saw when testing this workflow)</p>
    </li>
    <li>
      <p><em>Select</em> <strong>No</strong> under <strong>Send results to a new history</strong></p>

      <p>If you do not see this option displayed, <em>click</em> on <strong>Expand to full workflow form</strong> to reveal it.</p>
    </li>
    <li><em>Set</em> the first (and only) input dataset to the <code class="language-plaintext highlighter-rouge">Exons</code> dataset, using the pull-down menu.</li>
    <li><em>Click</em> the <strong>Run workflow</strong> button.</li>
  </ol>
</blockquote>

<p>And Galaxy launches the workflow and says (in a nice big green box) something like:</p>

<blockquote class="quote">
  <p>You can check the status of queued jobs and view the resulting data by refreshing the History pane.</p>
</blockquote>

<p>Which in this case <em>is the truth</em>.  You can refresh the history panel by either reloading the whole page, of by clicking the looping arrow icon at the top of the history panel.  What youâll see is a stack of history steps that will go from queued to running to done as you watch them.</p>

<p>All steps in the history will be green when the workflow is done. Once that happens compare your output dataset with your input dataset?  What percentage of exons overlap with other exons on the opposite strand, and is it common or rare?  Is it a smaller percentage than we saw for whole genes?</p>

<h2 id="are-we-done">Are we done?</h2>

<p>Probably.  Note that we can no longer say what percentage of <em>genes</em> overlap.  We <em>can</em> say what percentage of exons overlap, and that is probably close enough for our goals.  If it isnât and we actually need to say what percentage of genes overlap, then we will have to do some extra work.  This can be done is several ways, but the Galaxy 101 tutorial may give you some ideas on how to follow this question all the way to genes.</p>

<h1 id="conclusion">Conclusion</h1>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Well done! You have just performed your first analysis in Galaxy!</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-why-not-use-excel"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-not-use-excel" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Why not use Excel?<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>   <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel wonât help you here. It doesnât track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>   <p>Galaxy, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable workflow from your analysis that others (or yourself) can use on other datasets.</p>   <p>Another challenge with spreadsheet programs is that they donât scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but youâll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p> </blockquote>
<p><!--END_SNIPPET--></p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Galaxy provides an easy-to-use graphical user interface for often complex command-line tools</p>
</li>
                        
                        <li><p>Galaxy keeps a full record of your analysis in a history</p>
</li>
                        
                        <li><p>Workflows enable you to repeat your analysis on different data</p>
</li>
                        
                        <li><p>Galaxy can connect to external sources for data import and visualization purposes</p>
</li>
                        
                        <li><p>Galaxy provides ways to share your results and methods with others</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/introduction/faqs/">FAQ page for the Introduction to Galaxy Analyses topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Introduction to Genomics and Galaxy (Introduction to Galaxy Analyses)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Dave Clements, CristÃ³bal Gallardo,  <b>Introduction to Genomics and Galaxy (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html">https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{introduction-galaxy-intro-strands,
author = "Dave Clements and CristÃ³bal Gallardo",
	title = "Introduction to Genomics and Galaxy (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine FÃ¶ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and BjÃ¶rn GrÃ¼ning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/introduction/tutorials/galaxy-intro-strands/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: ddf54b12c295
  tool_panel_section_label: Introduction to Galaxy Analyses
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: intersect
  owner: devteam
  revisions: 5b3c6135a982
  tool_panel_section_label: Introduction to Galaxy Analyses
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: intersect
  owner: devteam
  revisions: 69c10b56f46d
  tool_panel_section_label: Introduction to Galaxy Analyses
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00191">https://gxy.io/GTN:T00191</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/introduction/tutorials/galaxy-intro-strands/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1706056126'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>