<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>VGP assembly pipeline</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="DNA sequence data has become an indispensable tool for Mo..." />
        <meta property="og:title" content="Galaxy Training: VGP assembly pipeline" />
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Mo..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Assembly
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/assembly/faqs/" title="Check our FAQs for the Assembly topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/vgp_genome_assembly/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:42 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "VGP assembly pipeline",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
  },
  "courseCode": "assembly / vgp_genome_assembly / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'VGP assembly pipeline' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - what combination of tools can produce the highest quality assembly of vertebrate genomes?\n - How can we evaluate how good it is?\n\n\\nThe objectives are:\n - Learn the tools necessary to perform a de novo assembly of a vertebrate genome\n - Evaluate the quality of the assembly\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Delphine Lariviere"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    },
    {
      "@type": "Person",
      "name": "Crist√≥bal Gallardo"
    },
    {
      "@type": "Person",
      "name": "Anna Syme"
    },
    {
      "@type": "Person",
      "name": "Linelle Abueg"
    },
    {
      "@type": "Person",
      "name": "Brandon Pickett"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Delphine Lariviere"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    },
    {
      "@type": "Person",
      "name": "Crist√≥bal Gallardo"
    },
    {
      "@type": "Person",
      "name": "Anna Syme"
    },
    {
      "@type": "Person",
      "name": "Linelle Abueg"
    },
    {
      "@type": "Person",
      "name": "Brandon Pickett"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
    }
  ]
}
</script>

<section class="tutorial topic-assembly">
    <h1 data-toc-skip>VGP assembly pipeline</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/delphine-l/" class="contributor-badge contributor-delphine-l"><img src="https://avatars.githubusercontent.com/delphine-l?s=27" alt="Avatar">Delphine Lariviere</a>


<a href="/training-material/hall-of-fame/astrovsky01/" class="contributor-badge contributor-astrovsky01"><img src="https://avatars.githubusercontent.com/astrovsky01?s=27" alt="Avatar">Alex Ostrovsky</a>


<a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="https://avatars.githubusercontent.com/gallardoalba?s=27" alt="Avatar">Crist√≥bal Gallardo</a>


<a href="/training-material/hall-of-fame/annasyme/" class="contributor-badge contributor-annasyme"><img src="https://avatars.githubusercontent.com/annasyme?s=27" alt="Avatar">Anna Syme</a>


<a href="/training-material/hall-of-fame/abueg/" class="contributor-badge contributor-abueg"><img src="https://avatars.githubusercontent.com/abueg?s=27" alt="Avatar">Linelle Abueg</a>


<a href="/training-material/hall-of-fame/pickettbd/" class="contributor-badge contributor-pickettbd"><img src="https://avatars.githubusercontent.com/pickettbd?s=27" alt="Avatar">Brandon Pickett</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>what combination of tools can produce the highest quality assembly of vertebrate genomes?</p>
</li>
        
        <li><p>How can we evaluate how good it is?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn the tools necessary to perform a de novo assembly of a vertebrate genome</p>
</li>
        
        <li><p>Evaluate the quality of the assembly</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/5887339">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/workflows/" title="Workflows" alt="VGP assembly pipeline workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/assembly/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://metagenomics.usegalaxy.eu/" title="Metagenomics">
            Metagenomics
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://plants.usegalaxy.eu/" title="Plant Anaylsis Workbench">
            Plant Anaylsis Workbench
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jan 25, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Advances in sequencing technologies over the last few decades have revolutionised the field of genomics, allowing for a reduction in both the time and resources required to <em>de novo</em> genome assembly. Until recently, second-generation sequencing technologies (also known as Next Generation Sequencing or NGS) allowed to produce highly accurate but short (up to 800bp), whose extension was not long enough to cope with the difficulties associated with repetitive regions. Today, so-called third-generation sequencing (TGS) technologies, usually known as single-molecule real-time (SMRT) sequencing, have become dominant in <em>de novo</em> assembly of large genomes. TGS can use native <span class="notranslate">DNA</span> without amplification, reducing sequencing error and bias (<span class="citation"><a href="#Hon2020">Hon <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>). Very recently, Pacific Biosciences introduced High-Fidelity (HiFi) sequencing, which produces <span class="notranslate">reads</span> 10-25 kpb in length with a minimum accuracy of 99% (Q20). In this tutorial you will use HiFi <span class="notranslate">reads</span> in combination with data from additional sequencing technologies to generate a high-quality genome assembly.</p>

<p>Deciphering the structural organisation of complex vertebrate genomes is currently one of the largest challenges in genomics (<span class="citation"><a href="#Frenkel2012">Frenkel <i>et al.</i> 2012</a></span>). Despite the significant progress made in recent years, a key question remains: what combination of data and tools can produce the highest quality assembly? In order to adequately answer this question, it is necessary to analyse two of the main factors that determine the difficulty of genome assembly processes: repetitive content and heterozygosity.</p>

<p>Repetitive elements can be grouped into two categories: interspersed repeats, such as transposable elements (TE) that occur at multiple loci throughout the genome, and tandem repeats (TR) that occur at a single locus (<span class="citation"><a href="#Trresen2019">T√∏rresen <i>et al.</i> 2019</a></span>). Repetitive elements are an important component of eukaryotic genomes, constituting over a third of the genome in the case of mammals (<span class="citation"><a href="#SoteroCaio2017">Sotero-Caio <i>et al.</i> 2017</a></span>, <span class="citation"><a href="#Chalopin2015">Chalopin <i>et al.</i> 2015</a></span>). In the case of tandem repeats, various estimates suggest that they are present in at least one third of human protein sequences (<span class="citation"><a href="#Marcotte1999">Marcotte <i>et al.</i> 1999</a></span>). TE content is among the main factors contributing to the lack of continuity in the reconstruction of genomes, especially in the case of large ones, as TE content is highly correlated with genome size (<span class="citation"><a href="#SoteroCaio2017">Sotero-Caio <i>et al.</i> 2017</a></span>). On the other hand, TR usually lead to local genome assembly collapse, especially when their length is close to that of the <span class="notranslate">reads</span> (<span class="citation"><a href="#Trresen2019">T√∏rresen <i>et al.</i> 2019</a></span>).</p>

<p>Heterozygosity is also an important factor in genome assembly. Haplotype phasing, that is, the identification of alleles that are co-located on the same chromosome, has become a fundamental problem in heterozygous and polyploid genome assemblies (<span class="citation"><a href="#Zhang2020">Zhang <i>et al.</i> 2020</a></span>). When no reference sequence is available, the <em>state-of-the-art</em> strategy consists of constructing a string graph with vertexes representing <span class="notranslate">reads</span> and edges representing consistent overlaps. In this kind of graph, after transitive reduction, heterozygous alleles in the string graph are represented by bubbles. When combined with Hi-C data, this approach allows complete diploid reconstruction (<span class="citation"><a href="#DominguezDelAngel2018">Angel <i>et al.</i> 2018</a></span>, <span class="citation"><a href="#Zhang2020">Zhang <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Dida2021">Dida and Yi 2021</a></span>).</p>

<p>The G10K consortium launched the Vertebrate Genomes Project (VGP), whose goal is generating high-quality, near-error-free, gap-free, chromosome-level, haplotype-phased, annotated reference genome assemblies for every vertebrate species (<span class="citation"><a href="#Rhie2021">Rhie <i>et al.</i> 2021</a></span>). This tutorial will guide you step by step to assemble a high-quality genome using the VGP assembly pipeline.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#vgp-assembly-pipeline-overview" id="markdown-toc-vgp-assembly-pipeline-overview">VGP assembly pipeline overview</a>    <ol>
      <li><a href="#run-the-vgp-workflows-automatically" id="markdown-toc-run-the-vgp-workflows-automatically">Run the VGP workflows automatically</a></li>
    </ol>
  </li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#genome-profile-analysis" id="markdown-toc-genome-profile-analysis">Genome profile analysis</a>    <ol>
      <li><a href="#generation-of-k-mer-spectra-with-meryl" id="markdown-toc-generation-of-k-mer-spectra-with-meryl">Generation of k-mer spectra with <strong>Meryl</strong></a></li>
      <li><a href="#genome-profiling-with-genomescope2" id="markdown-toc-genome-profiling-with-genomescope2">Genome profiling with <strong>GenomeScope2</strong></a></li>
    </ol>
  </li>
  <li><a href="#hifi-phased-assembly-with-hifiasm" id="markdown-toc-hifi-phased-assembly-with-hifiasm">HiFi phased assembly with hifiasm</a>    <ol>
      <li><a href="#genome-assembly-with-hifiasm" id="markdown-toc-genome-assembly-with-hifiasm">Genome assembly with <strong>hifiasm</strong></a></li>
      <li><a href="#initial-assembly-evaluation" id="markdown-toc-initial-assembly-evaluation">Initial assembly evaluation</a></li>
    </ol>
  </li>
  <li><a href="#post-assembly-processing" id="markdown-toc-post-assembly-processing">Post-assembly processing</a>    <ol>
      <li><a href="#remove-haplotypic-duplication-with-purge_dups" id="markdown-toc-remove-haplotypic-duplication-with-purge_dups">Remove haplotypic duplication with <strong>purge_dups</strong></a></li>
      <li><a href="#second-round-of-assembly-evaluation" id="markdown-toc-second-round-of-assembly-evaluation">Second round of assembly evaluation</a></li>
    </ol>
  </li>
  <li><a href="#hybrid-scaffolding" id="markdown-toc-hybrid-scaffolding">Hybrid scaffolding</a>    <ol>
      <li><a href="#hybrid-scaffolding-using-bionano-data" id="markdown-toc-hybrid-scaffolding-using-bionano-data">Hybrid scaffolding using Bionano data</a></li>
      <li><a href="#hybrid-scaffolding-based-on-hi-c-mapping-data" id="markdown-toc-hybrid-scaffolding-based-on-hi-c-mapping-data">Hybrid scaffolding based on Hi-C <span class="notranslate">mapping</span> data</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="vgp-assembly-pipeline-overview">VGP assembly pipeline overview</h1>

<p>The figure 1 represents the VGP assembly pipeline.</p>

<figure id="figure-1"><img src="../../images/vgp_assembly/VGP_Pipeline.png" alt="Figure 1:VGP pipeline. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> VPG Pipeline 2.0. The pipeline starts with assembly of the HiFi <span class="notranslate">reads</span> into contigs. These contigs undergo purge_dups to yield a purged primary assembly and an alte<span class="notranslate">rna</span>te assembly. The Bionano and HiC data are then used to generate a scaffolded primary assembly. This primary and the alte<span class="notranslate">rna</span>te assembly are now ready for curation. Quality control outputs are displayed next to the intermediate steps where they are generated.</figcaption></figure>

<p>The tutorial is structured in four main sections:</p>

<ul>
  <li>Genome profile analysis</li>
  <li>HiFi phased assembly with hifiasm</li>
  <li>Post-assembly processing</li>
  <li>Hybrid scaffolding</li>
</ul>

<h2 id="run-the-vgp-workflows-automatically">Run the VGP workflows automatically</h2>

<p>The pipeline presented in this training is an adaption of the <a href="https://github.com/Delphine-L/iwc/tree/VGP/workflows/VGP-assembly-v2">current <span class="notranslate">workflow</span> versions</a>. This tutorial will manually take you through the steps in the VGP pipeline, with a goal of explaining the pipeline step by step. If you desire to run the <em>state-of-the-art</em> VGP pipeline as workflows (<em>i.e.</em>, not manually), just follow the following instructions:</p>

<blockquote>
  <h3 id="hands_on-hands-on-run-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run <span class="notranslate">workflow</span></h3>

  <ol>
    <li>Download the <span class="notranslate">workflow</span> files (whose extension is .<em>ga</em>) from this <a href="https://github.com/Delphine-L/iwc/tree/VGP/workflows/VGP-assembly-v2">GitHub repository</a>. The <span class="notranslate">workflow</span> files are inside their corresponding folders (for example, the VGP-Hifiasm <span class="notranslate">workflow</span> file is inside the VGP-Hifiasm folder).</li>
    <li>Click on <span class="notranslate">Workflow</span> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>
    <li>Click on the import button <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen.</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <figure id="figure-2"><img src="../../images/vgp_assembly/import_workflow.png" alt="Figure 2: Import Workflow. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Import <span class="notranslate">workflow</span> from a file or URL into <span class="notranslate">Galaxy</span>.</figcaption></figure>
  <ol>
    <li>Provide your <span class="notranslate">workflow</span>:
      <ul>
        <li>Option 1: Upload the <span class="notranslate">workflow</span> file in the box labelled ‚ÄúArchived <span class="notranslate">Workflow</span> File‚Äù</li>
        <li>Option 2: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled ‚ÄúArchived <span class="notranslate">Workflow</span> URL‚Äù</li>
      </ul>
    </li>
    <li>Click the <strong>import <span class="notranslate">workflow</span></strong> button.</li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>The <span class="notranslate">Galaxy</span> workflows include additional steps (<em>e.g.</em>, parse parameter value) required for running it automatically, but are not necessary when we run the pipeline step by step manually.</p>
</blockquote>

<h1 id="get-data">Get data</h1>

<p>To reduce compute times, we will assemble samples from the yeast <em>Saccharomyces cerevisiae</em>, one of the most intensively studied eukaryotic model organisms in molecular and cell biology. Yeast can be haploid or diploid, depending on the stage of its life cycle. Both cell types are stable and can reproduce asexually by mitosis. For this tutorial, we generated an artificial diploid genome by mixing two different lineages: CEN.PK and S288c.</p>

<p>The first step is to get the datasets from Zenodo. The VGP assembly pipeline uses data generated by a variety of technologies, including PacBio HiFi <span class="notranslate">reads</span>, Bionano optical maps, and Hi-C chromatin interaction maps.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/record/5887339">Zenodo</a></p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Rule-based</strong> tab</li>
        <li><em>‚ÄúUpload data as‚Äù</em>: <code class="language-plaintext highlighter-rouge">Datasets</code></li>
        <li>
          <p>Copy the tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Hi-C_dataset_F   https://zenodo.org/record/5550653/files/SRR7126301_1.fastq.gz?download=1   fastqsanger.gz    Hi-C
Hi-C_dataset_R   https://zenodo.org/record/5550653/files/SRR7126301_2.fastq.gz?download=1   fastqsanger.gz    Hi-C
Bionano_dataset    https://zenodo.org/record/5550653/files/bionano.cmap?download=1   cmap    Bionano
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name</code>: column <code class="language-plaintext highlighter-rouge">A</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Type</code>: column <code class="language-plaintext highlighter-rouge">C</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name Tag</code>: column <code class="language-plaintext highlighter-rouge">D</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
    <li>
      <p>Import the remaining datasets from <a href="https://zenodo.org/record/5887339">Zenodo</a></p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Rule-based</strong> tab</li>
        <li><em>‚ÄúUpload data as‚Äù</em>: <code class="language-plaintext highlighter-rouge">Collections</code></li>
        <li>
          <p>Copy the tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR13577846_1    https://zenodo.org/record/5887339/files/SRR13577846_1.fastq.gz?download=1  fastqsanger.gz    HiFi  HiFi_collection
SRR13577846_2    https://zenodo.org/record/5887339/files/SRR13577846_2.fastq.gz?download=1  fastqsanger.gz    HiFi  HiFi_collection
SRR13577847_1    https://zenodo.org/record/5887339/files/SRR13577847_1.fastq.gz?download=1  fastqsanger.gz    HiFi  HiFi_collection
SRR13577847_2    https://zenodo.org/record/5887339/files/SRR13577847_2.fastq.gz?download=1  fastqsanger.gz    HiFi  HiFi_collection
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>: column <code class="language-plaintext highlighter-rouge">A</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Type</code>: column <code class="language-plaintext highlighter-rouge">C</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Group Tag</code>: column <code class="language-plaintext highlighter-rouge">D</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Collection Name</code>: column <code class="language-plaintext highlighter-rouge">E</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="hifi-reads-preprocessing-with-cutadapt">HiFi <span class="notranslate">reads</span> preprocessing with <strong>Cutadapt</strong></h3>

<p>Once we have retrieved the data, we need to do adapter trimming using Cutadapt, an open-source tool written mainly in Python language. Adapter trimming usually means trimming the adapter sequence off the ends of <span class="notranslate">reads</span>, which is where the adapter sequence is usually located in NGS <span class="notranslate">reads</span>. For these HiFi <span class="notranslate">reads</span>, the adapter does not have a specific, predictable location in the read, due to the nature of SMRT sequencing. Additionally, the <span class="notranslate">reads</span> containing adapter sequence could be of generally lower quality compared to the rest of the <span class="notranslate">reads</span>. Thus, we will use Cutadapt not to trim, but to remove the entire read if a read is found to have an adapter inside of it. For this purpose, we are going to use Cutadapt, an open-source tool written mainly in Python language.</p>

<blockquote class="comment">
  <h3 id="comment-background-on-pacbio-hifi-reads"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Background on PacBio HiFi <span class="notranslate">reads</span></h3>

  <p>PacBio HiFi <span class="notranslate">reads</span> rely on the Single Molecule Real-Time (SMRT) sequencing technology. SMRT is based on real-time imaging of fluorescently tagged nucleotides as they are added to a newly synthesized <span class="notranslate">DNA</span> strand. HiFi further combine multiple sub<span class="notranslate">reads</span> from the same circular template to produce one highly accurate consensus sequence (fig. 3). This technology allows to generate long-read sequencing data with read lengths in the range of 10-25 kb and minimum read consensus accuracy  greater than 99% (Q20).</p>

  <figure id="figure-3"><img src="../../images/vgp_assembly/pacbio_hifi.png" alt="Figure 3: PacBio sequencing technolgoy. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> PacBio HiFi sequencing. Adapted from Wenger et al., 2019</figcaption></figure>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-primer-removal-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Primer removal with Cutadapt</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or Paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ/A file‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection</code></li>
            <li>In <em>‚ÄúRead 1 Options‚Äù</em>:
              <ul>
                <li>In <em>‚Äú5‚Äô or 3‚Äô (Anywhere) Adapters‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert 5‚Äô or 3‚Äô (Anywhere) Adapters‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSource‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>‚ÄúEnter custom 5‚Äô or 3‚Äô adapter name‚Äù</em>: <code class="language-plaintext highlighter-rouge">First adapter</code></li>
                            <li><em>‚ÄúEnter custom 5‚Äô or 3‚Äô adapter sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert 5‚Äô or 3‚Äô (Anywhere) Adapters‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSource‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>‚ÄúEnter custom 5‚Äô or 3‚Äô adapter name‚Äù</em>: <code class="language-plaintext highlighter-rouge">Second adapter</code></li>
                            <li><em>‚ÄúEnter custom 5‚Äô or 3‚Äô adapter sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">ATCTCTCTCTTTTCCTCCTCCTCCGTTGTTGTTGTTGAGAGAGAT</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdapter Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúMatch times‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>‚ÄúMaximum error rate‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
            <li><em>‚ÄúMinimum overlap length‚Äù</em>: <code class="language-plaintext highlighter-rouge">35</code></li>
            <li><em>‚ÄúLook for adapters in the reverse complement‚Äù</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúFilter Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúDiscard Trimmed <span class="notranslate">Reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output file as <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code>. To rename an output file, click on the result, and then click again on the title to change it. After closing, you may need to refresh the history to see the name change.</li>
  </ol>

</blockquote>

<h1 id="genome-profile-analysis">Genome profile analysis</h1>

<p>Before starting a <em>de novo</em> genome assembly project, it is useful to collect metrics on the properties of the genome under consideration, such as the expected genome size. Traditionally, <span class="notranslate">DNA</span> flow cytometry was considered the golden standard for estimating the genome size. Nowadays, experimental methods have been replaced by computational approaches (<span class="citation"><a href="#wang2020estimation">Wang <i>et al.</i> 2020</a></span>). One of the widely used genome profiling methods is based on the analysis of k-mer frequencies. It allows one to provide information not only about the genomic complexity, such as the genome size and levels of heterozygosity and repeat content, but also about the data quality.</p>

<blockquote class="comment">
  <h3 id="comment-k-mer-size-sequencing-coverage-and-genome-size"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> K-mer size, sequencing coverage and genome size</h3>

  <p><em>K</em>-mers are unique substrings of length k contained within a <span class="notranslate">DNA</span> sequence. For example, the <span class="notranslate">DNA</span> sequence <em>TCGATCACA</em> can be decomposed into six unique <em>k</em>-mers that have five bases long: <em>TCGAT</em>, <em>CGATC</em>, <em>GATCA</em>, <em>ATCAC</em> and <em>TCACA</em>. A sequence of length L will have  L-k+1 <em>k</em>-mers. On the other hand, the number of possible <em>k</em>-mers can be calculated as  n<sup>k</sup>, where n is number of possible monomers and k is the k-mer size.</p>

  <table>
    <thead>
      <tr>
        <th>Bases</th>
        <th>K-mer size</th>
        <th>Total possible k-mers</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>4</td>
        <td>1</td>
        <td>4</td>
      </tr>
      <tr>
        <td>4</td>
        <td>2</td>
        <td>16</td>
      </tr>
      <tr>
        <td>4</td>
        <td>3</td>
        <td>64</td>
      </tr>
      <tr>
        <td>4</td>
        <td>4</td>
        <td>256</td>
      </tr>
      <tr>
        <td>4</td>
        <td>‚Ä¶</td>
        <td>‚Ä¶</td>
      </tr>
      <tr>
        <td>4</td>
        <td>10</td>
        <td>1.048.576</td>
      </tr>
    </tbody>
  </table>

  <p>Thus, the k-mer size is a key parameter, which must be large enough to map  uniquely to the genome, but not too large, since it can lead to wasting computational resources. In the case of the human genome, <em>k</em>-mers of 31 bases in length lead to 96.96% of unique <em>k</em>-mers.</p>

  <p>Each unique k-mer can be assigned a value for coverage based on the number of times it occurs in a sequence, whose distribution will approximate a Poisson distribution, with the peak corresponding to the average genome sequencing depth. From the genome coverage, the genome size can be easily computed.</p>
</blockquote>

<p>In this section we will use two basic tools to computationally estimate the genome features: Meryl and GenomeScope.</p>

<h2 id="generation-of-k-mer-spectra-with-meryl">Generation of k-mer spectra with <strong>Meryl</strong></h2>

<p><strong>Meryl</strong> will allow us to generate the <em>k</em>-mer profile by decomposing the sequencing data into <em>k</em>-length substrings, counting the occurrence of each <em>k</em>-mer and determining its frequency. The original version of Meryl was developed for the Celera Assembler. The current Meryl version comprises three main modules: one for generating <em>k</em>-mer databases, one for filtering and combining databases, and one for searching databases. <em>K</em>-mers are stored in lexicographical order in the database, similar to words in a dictionary (<span class="citation"><a href="#Rhie2020">Rhie <i>et al.</i> 2020</a></span>).</p>

<blockquote class="comment">
  <h3 id="comment-k-mer-size-estimation"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> <em>k</em>-mer size estimation</h3>

  <p>Given an estimated genome size (G) and a tolerable collision rate (p), an appropriate k can be computed as k = log4 (G(1 ‚àí p)/p).</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-k-mers-count-distribution"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate <em>k</em>-mers count distribution</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy2"><strong>Meryl</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúOperation type selector‚Äù</em>: <code class="language-plaintext highlighter-rouge">Count operations</code>
          <ul>
            <li><em>‚ÄúCount operations‚Äù</em>: <code class="language-plaintext highlighter-rouge">Count: count the ocurrences of canonical k-mers</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúInput sequences‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code></li>
            <li><em>‚Äúk-mer size selector‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set a k-mer size</code>
              <ul>
                <li>‚Äú<em>k-mer size</em>‚Äù: <code class="language-plaintext highlighter-rouge">21</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-selection-of-k-mer-size"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Selection of <em>k</em>-mer size</h3>

        <p>We used 21 as <em>k</em>-mer size, as this length has demonstrated to be sufficiently long that most <em>k</em>-mers are not repetitive and is short enough to be more robust to sequencing errors. For very large (haploid size &gt; 10 Gb) and/or very repetitive genomes, larger <em>k</em>-mer length is recommended to increase the number of unique <em>k</em>-mers.</p>
      </blockquote>
    </li>
    <li>
      <p>Rename it <code class="language-plaintext highlighter-rouge">Collection meryldb</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy1"><strong>Meryl</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúOperation type selector‚Äù</em>: <code class="language-plaintext highlighter-rouge">Operations on sets of k-mers</code>
          <ul>
            <li><em>‚ÄúOperations on sets of k-mers‚Äù</em>: <code class="language-plaintext highlighter-rouge">Union-sum: return k-mers that occur in any input, set the count to the sum of the counts</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput meryldb‚Äù</em>: <code class="language-plaintext highlighter-rouge">Collection meryldb</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename it as <code class="language-plaintext highlighter-rouge">Merged meryldb</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy0"><strong>Meryl</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúOperation type selector‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generate histogram dataset</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput meryldb‚Äù</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Finally, rename it as <code class="language-plaintext highlighter-rouge">Meryldb histogram</code>.</li>
  </ol>

</blockquote>

<h2 id="genome-profiling-with-genomescope2">Genome profiling with <strong>GenomeScope2</strong></h2>

<p>The next step is to infer the genome properties from the <em>k</em>-mer histogram generated by Meryl, for which we will use <strong>GenomeScope2</strong>. Genomescope2 relies on a nonlinear least-squares optimization to fit a mixture of negative binomial distributions, generating estimated values for genome size, repetitiveness, and heterozygosity rates (<span class="citation"><a href="#RanalloBenavidez2020">Ranallo-Benavidez <i>et al.</i> 2020</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-estimate-genome-properties"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Estimate genome properties</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0"><strong>GenomeScope</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput histogram file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Meryldb histogram</code></li>
        <li><em>‚Äúk-mer length used to calculate k-mer spectra‚Äù</em>: <code class="language-plaintext highlighter-rouge">21</code></li>
      </ul>
    </li>
  </ol>

  <ul>
    <li>In ‚Äú<em>Output options</em>‚Äù: mark <code class="language-plaintext highlighter-rouge">Summary of the analysis</code></li>
    <li>In ‚Äú<em>Advanced options</em>‚Äù:
      <ul>
        <li><em>‚ÄúCreate testing.tsv file with model parameters‚Äù</em>: <code class="language-plaintext highlighter-rouge">true</code></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>Genomescope will generate six outputs:</p>

<ul>
  <li>Plots
    <ul>
      <li><em>Linear plot</em>: <em>k</em>-mer spectra and fitted models: frequency (y-axis) versus coverage.</li>
      <li><em>Log plot</em>: logarithmic transformation of the previous plot.</li>
      <li>Transformed linear plot: <em>k</em>-mer spectra and fitted models: frequency times coverage (y-axis) versus coverage (x-axis). This transformation increases the heights of higher-order peaks, overcoming the effect of high heterozygosity.</li>
      <li>Transformed log plot: logarithmic transformation of the previous plot.</li>
    </ul>
  </li>
  <li>Model: this file includes a detailed report about the model fitting.</li>
  <li>Summary: it includes the properties inferred from the model, such as genome haploid length and the percentage of heterozygosity.</li>
</ul>

<p>Now, let‚Äôs analyze the <em>k</em>-mer profiles, fitted models and estimated parameters:</p>

<figure id="figure-4"><img src="../../images/vgp_assembly/genomescope_plot.png" alt="fig3:Genomescope plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Genomescope2 plot.</figcaption></figure>

<p>This distribution is the result of the Poisson process underlying the generation of sequencing <span class="notranslate">reads</span>. As we can see, there is a unique peak centered around 28x, the modal <em>k</em>-mer coverage. The absence of a secondary peak at half diploid coverage is suggestive of the haploid nature of this genome, although it could also result from very low heterozygosity. Low frequency <em>k</em>-mers are the result of sequencing errors.</p>

<p>Before proceeding to the next section, we need to carry out some operations on the output generated by GenomeScope2. The goal is to extract some parameters which at a later stage will be used by <strong>purge_dups</strong>. The first relevant parameter is the <code class="language-plaintext highlighter-rouge">estimated genome size</code>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-estimated-genome-size"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get estimated genome size</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to process‚Äù</em>: <code class="language-plaintext highlighter-rouge">summary</code> (output of <strong>GenomeScope</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">bp</code></li>
        <li><em>‚ÄúReplace all occurrences of the pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFind and Replace text in‚Äù</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to process‚Äù</em>: output file of <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">,</code></li>
        <li><em>‚ÄúReplace all occurrences of the pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFind and Replace text in‚Äù</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1"><strong>Search in textfiles (grep)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect lines from‚Äù</em>: output file of the previous step.</li>
        <li><em>‚ÄúType of regex‚Äù</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
        <li><em>‚ÄúRegular <span class="notranslate">Expression</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Haploid</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Convert characters1" title="Tested with Convert characters1"><strong>Convert delimiters to TAB</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Convert characters1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúin Dataset‚Äù</em>: output of <strong>Search in textfiles</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to cut‚Äù</em>: output of <strong>Convert delimiters to TAB</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúCut by‚Äù</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><em>‚ÄúList of Fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 5</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Estimated genome size</code>.</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>What is the estimated genome size?</p>

    <blockquote class="solution">
      <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <p>The estimated genome size is 12792715 bp.</p>

    </blockquote>

  </blockquote>

</blockquote>

<p>Now let‚Äôs parse the <code class="language-plaintext highlighter-rouge">upper bound for the read depth estimation</code> parameter.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-maximum-read-depth"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get maximum read depth</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6"><strong>Compute an <span class="notranslate">expression</span> on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúAdd <span class="notranslate">expression</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.5*c3</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúas a new column to‚Äù</em>: <code class="language-plaintext highlighter-rouge">model_params</code> (output of <strong>GenomeScope</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúRound result?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúInput has a header line with column names?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6"><strong>Compute an <span class="notranslate">expression</span> on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúAdd <span class="notranslate">expression</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">3*c7</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúas a new column to‚Äù</em>: output of the previous step.</li>
        <li><em>‚ÄúRound result?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúInput has a header line with column names?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>
      <p>Rename it as <code class="language-plaintext highlighter-rouge">Parsing temporal output</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to cut‚Äù</em>: <code class="language-plaintext highlighter-rouge">Parsing temporal output</code></li>
        <li><em>‚ÄúCut by‚Äù</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><em>‚ÄúList of Fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 8</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Maximum depth</code></li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>What is the estimated maximum depth?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <p>The estimated maximum depth is  63 <span class="notranslate">reads</span>.</p>

    </blockquote>

  </blockquote>

</blockquote>

<p>Finally, let‚Äôs parse the <code class="language-plaintext highlighter-rouge">transition between haploid and diploid coverage depths</code> parameter.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-transition-parameter"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get transition parameter</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to cut‚Äù</em>: <code class="language-plaintext highlighter-rouge">Parsing temporal output</code></li>
        <li><em>‚ÄúCut by‚Äù</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><em>‚ÄúList of Fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 7</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Transition parameter</code></li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>What is the estimated transition parameter?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <p>The estimated transition parameter is  21 <span class="notranslate">reads</span>.</p>

    </blockquote>

  </blockquote>

</blockquote>

<h1 id="hifi-phased-assembly-with-hifiasm">HiFi phased assembly with hifiasm</h1>

<p>Once we have done genome profiling stage, we can start the genome assembly with <strong>hifiasm</strong>,  a fast open-source <em>de novo</em> assembler specifically developed for PacBio HiFi <span class="notranslate">reads</span>.</p>

<h2 id="genome-assembly-with-hifiasm">Genome assembly with <strong>hifiasm</strong></h2>

<p>One of the key advantages of hifiasm is that it allows us to resolve near-identical, but not exactly identical sequences, such as repeats and segmental duplications (<span class="citation"><a href="#Cheng2021">Cheng <i>et al.</i> 2021</a></span>).</p>

<blockquote class="comment">
  <h3 id="comment-hifiasm-algorithm-details"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Hifiasm algorithm details</h3>

  <p>By default hifiasm performs three rounds of haplotype-aware error correction to correct sequence errors but keeping heterozygous alleles. A position on the target read to be corrected is considered informative if there are two different nucleotides at that position in the alignment, and each allele is supported by at least three <span class="notranslate">reads</span>.</p>

  <figure id="figure-5"><img src="../../images/vgp_assembly/hifiasm_algorithm.png" alt="fig4:Hifiasm algorithm overview. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Hifiasm algorithm overview. Orange and blue bars represent the <span class="notranslate">reads</span> with heterozygous alleles carrying local phasing information, while green bars come from the homozygous regions without any heterozygous alleles.</figcaption></figure>

  <p>Then, hifiasm builds a phased assembly string graph with local phasing information from the corrected <span class="notranslate">reads</span>. Only the <span class="notranslate">reads</span> coming from the same haplotype are connected in the phased assembly graph. After transitive reduction, a pair of heterozygous alleles is represented by a <em>bubble</em> in the string graph. If there is no additional data, hifiasm arbitrarily selects one side of each bubble and outputs a primary assembly. In the case of a heterozygous genome, the primary assembly generated at this step may still retain haplotigs from the alte<span class="notranslate">rna</span>te allele.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-phased-assembly-with-hifiasm"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Phased assembly with <strong>hifiasm</strong></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.14+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.14+galaxy0"><strong>Hifiasm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.14+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúAssembly mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>‚ÄúOptions for purging duplicates‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specify</code>
          <ul>
            <li><em>‚ÄúCoverage upper bound‚Äù</em>: <code class="language-plaintext highlighter-rouge">63</code> (maximum depth previously obtained)</li>
          </ul>
        </li>
        <li><em>‚ÄúOptions for Hi-C partition‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specify</code>
          <ul>
            <li><em>‚ÄúHi-C R1 <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
            <li><em>‚ÄúHi-C R2 <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_R</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the <code class="language-plaintext highlighter-rouge">Hi-C hap1 contig graph</code> as <code class="language-plaintext highlighter-rouge">Primary contigs graph</code> and add a <code class="language-plaintext highlighter-rouge">#primary</code> tag</li>
    <li>Rename the <code class="language-plaintext highlighter-rouge">Hi-C hap2 contig graph</code> as <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs graph</code> and  add a <code class="language-plaintext highlighter-rouge">#alte<span class="notranslate">rna</span>te</code> tag</li>
  </ol>

</blockquote>

<p>Hifiasm generates four outputs in Graphical Fragment Assembly (GFA) format; this format is designed to represent genome variation, splice graphs in genes, and even overlaps between <span class="notranslate">reads</span>.</p>

<p>We have obtained the fully phased contig graphs of the primary and alte<span class="notranslate">rna</span>te haplotypes, but the output format of <strong>hifiasm</strong> must be converted to FASTA format for the subsequent steps.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-gfa-to-fasta"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: convert GFA to FASTA</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.2"><strong>GFA to FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gfa_to_fa/gfa_to_fa/0.1.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúInput GFA file‚Äù</em>: select <code class="language-plaintext highlighter-rouge">Primary contigs graph</code> and the <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs graph</code> datasets</li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code> and <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code></li>
  </ol>

</blockquote>

<h2 id="initial-assembly-evaluation">Initial assembly evaluation</h2>

<p>The VGP assembly pipeline contains several built-in QC steps, including QUAST, BUSCO, Merqury, and Pretext. QUAST will generate summary statistics, BUSCO will search for universal single-copy ortholog genes, Merqury will evaluate assembly copy-numbers using <em>k</em>-mers, and Pretext will be used to evaluate the assembly contiguity.</p>

<blockquote class="comment">
  <h3 id="comment-quast-statistics"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> QUAST statistics</h3>

  <p>QUAST will provide us with the following statistics:</p>

  <ul>
    <li>No. of contigs: The total number of contigs in the assembly.</li>
    <li>Largest contig: The length of the largest contig in the assembly.</li>
    <li>Total length: The total number of bases in the assembly.</li>
    <li>Nx: The largest contig length, <em>L</em>, such that contigs of length &gt;= <em>L</em> account for at least <em>x</em>% of the bases of the assembly.</li>
    <li>NGx: The contig length such that using equal or longer length contigs produces <em>x</em>% of the length of the reference genome, rather than <em>x</em>% of the assembly length.</li>
    <li>GC content: the percentage of nitrogenous bases which are either guanine or cytosine.</li>
  </ul>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assembly-evaluation-with-quast"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: assembly evaluation with QUAST</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúUse customized names for the input files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, specify custom names</code></li>
        <li>In <em>‚Äú1. Contigs/scaffolds‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
            <li><em>‚ÄúName‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly</code></li>
          </ul>
        </li>
        <li>Click in <em>‚ÄúInsert Contigs/scaffolds‚Äù</em></li>
        <li>In <em>‚Äú2. Contigs/scaffolds‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code></li>
            <li><em>‚ÄúName‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te assembly</code></li>
          </ul>
        </li>
        <li><em>‚Äú<span class="notranslate">Reads</span> options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Pacbio SMRT <span class="notranslate">reads</span></code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ file‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi collection (trim)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúType of assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome</code>
          <ul>
            <li><em>‚ÄúUse a reference genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúEstimated reference genome size (in bp) for computing NGx statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">12792715</code> (previously estimated)</li>
              </ul>
            </li>
            <li><em>‚ÄúType of organism‚Äù</em>: <code class="language-plaintext highlighter-rouge">Eukaryote: use of GeneMark-ES for gene finding, Bar<span class="notranslate">rna</span>p for ribosomal <span class="notranslate">RNA</span> genes prediction, BUSCO for conserved orthologs finding (--eukaryote)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs genome large (&gt;100Mpb)?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Remember that for this training we are using <em>S. cerevisiae</em>, a reduced genome. In the case of assembling a vertebrate genome, you must select <code class="language-plaintext highlighter-rouge">yes</code> in the previous option.</p>
      </blockquote>
    </li>
    <li>Rename the HTML report as <code class="language-plaintext highlighter-rouge">QUAST initial report</code></li>
  </ol>

</blockquote>

<p>Let‚Äôs have a look at the report generated by QUAST.</p>

<figure id="figure-6"><img src="../../images/vgp_assembly/QUAST_initial.png" alt="fig5:QUAST plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> QUAST initial report.</figcaption></figure>

<p>According with the plot, the primary assembly includes 40 contigs, whose accumulative length is around 12 Mbp. On the other hand, the accumulative length of the secondary assembly is much smaller, despite including more than 120 contigs. This is expected, since we are working with a haploid organism.</p>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is the longest contig in the primary assembly? And in the alte<span class="notranslate">rna</span>te one?</li>
    <li>What is the N50 of the primary assembly?</li>
    <li>Which percentage of <span class="notranslate">reads</span> mapped to each assembly?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The longest contig in the primary assembly is 914.549 bp, and 15.845 bp in the alte<span class="notranslate">rna</span>te assembly.</li>
      <li>The N50 of the primary assembly is 425.706 bp.</li>
      <li>According the report, 100% of <span class="notranslate">reads</span> mapped to the primary assembly, but only around 57% mapped to the alte<span class="notranslate">rna</span>te assembly.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Next, we will use BUSCO, which will provide quantitative assessment of the completeness of a genome assembly in terms of expected gene content. It relies in the analysis of genes that should be present only once in a complete assembly or gene set, while allowing for rare gene duplications or losses (<span class="citation"><a href="#Simo2015">Sim√£o <i>et al.</i> 2015</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assessing-assembly-completeness-with-busco"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: assessing assembly completeness with BUSCO</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0"><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSequences to analyse‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code> and <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code></li>
        <li><em>‚ÄúMode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (<span class="notranslate">DNA</span>)</code>
          <ul>
            <li><em>‚ÄúUse Augustus instead of Metaeuk‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
          </ul>
        </li>
        <li><em>‚ÄúAuto-detect or select lineage?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
          <ul>
            <li><em>‚ÄúLineage‚Äù</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
          </ul>
        </li>
        <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: <code class="language-plaintext highlighter-rouge">short summary text</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Remember to modify the lineage option if you are working with vertebrate genomes.</p>
      </blockquote>
    </li>
    <li>Rename the summary as <code class="language-plaintext highlighter-rouge">BUSCO initial report</code></li>
  </ol>

</blockquote>

<p>BUSCO generates two outputs by default, a tabular file which contains the complete results (<em>e.g.</em>, scores, descriptions, lengths and coordinates of matches) for each BUSCO gene (fig. 8), and a short summary.</p>

<figure id="figure-7"><img src="../../images/vgp_assembly/BUSCO_full_table.png" alt="fig8:BUSCO plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> BUSCO full table</figcaption></figure>

<p>As we can see in the report, the results are simplified into four categories: <em>complete and single-copy</em>, <em>complete and duplicated</em>, <em>fragmented</em> and  <em>Missing BUSCOs</em>.</p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many complete BUSCO genes have been identified?</li>
    <li>How many BUSCOs gene are absent?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>According the report, our assembly contains the complete sequence of  2121 complete BUSCO genes.</li>
      <li>8 BUSCO genes are missing.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Despite BUSCO being robust for species that have been widely studied, it can be inaccurate when the newly assembled genome belongs to a taxonomic group that is not well represented in <a href="https://www.orthodb.org/">OrthoDB</a>. Merqury provides a complementary approach for assessing genome assembly quality metrics in a reference-free manner via <em>k</em>-mer copy number analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-k-mer-based-evaluation-with-merqury"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <em>k</em>-mer based evaluation with Merqury</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3"><strong>Merqury</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúEvaluation mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default mode</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúk-mer counts database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
            <li><em>‚ÄúNumber of assemblies‚Äù</em>: `Two assemblies
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFirst genome assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSecond genome assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>By default, Merqury generates three collections as output: stats, plots and QV stats. The ‚Äústats‚Äù collection contains the completeness statistics, while the ‚ÄúQV stats‚Äù collection contains the quality value statistics.</p>

<p>The copy number spectum plot, known as the <em>spectra-cn</em> plot (fig. 9a), tracks the multiplicity of each k-mer found in the Hi-Fi read set and colors it by the number of times it is found in a given assembly. Merqury connects the midpoint of each histogram bin with a line, giving the illusion of a smooth curve. Remember that the histogram dataset was generated previously by Meryl.</p>

<figure id="figure-8"><img src="../../images/vgp_assembly/merqury_cn_plot.png" alt="fi9:Merqury plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Merqury CN plot</figcaption></figure>

<p>The black region in the left side corresponds to <em>k</em>-mers found only in the read set; it is usually indicative of sequencing error in the read set, although it can also be indicative of missing sequences in the assembly. The read area represents one-copy <em>k</em>-mers in the genome, while blue area represents two-copy <em>k</em>-mers originating from homozygous sequence or haplotype-specific duplications. From that figure we can state that the sequencing coverage is around 30x.</p>

<p>The figure 1b corresponds to the assembly spectrum plot (<em>spectra-asm</em> in the outputs). This plot is helpful for measuring diploid assembly completeness as it shows the fraction of k-mers specific to both the primary and alte<span class="notranslate">rna</span>te assemblies, as well as k-mers shared between them (<span class="citation"><a href="#Rhie2020">Rhie <i>et al.</i> 2020</a></span>).</p>

<h1 id="post-assembly-processing">Post-assembly processing</h1>

<p>An ideal haploid representation would consist of one allelic copy of all heterozygous regions in the two haplomes, as well as all hemizygous regions from both haplomes (<span class="citation"><a href="#Guan2019">Guan <i>et al.</i> 2019</a></span>). However, in highly heterozygous genomes, assembly algorithms are frequently not able to identify the highly divergent allelic sequences as belonging to the same region, resulting in the assembly of those regions as separate contigs. This can lead to issues in <span class="notranslate">downstream</span> analysis, such as scaffolding, gene annotation and read <span class="notranslate">mapping</span> in general (<span class="citation"><a href="#Small2007">Small <i>et al.</i> 2007</a></span>, <span class="citation"><a href="#Guan2019">Guan <i>et al.</i> 2019</a></span>, <span class="citation"><a href="#Roach2018">Roach <i>et al.</i> 2018</a></span>). In order to solve this problem, we are going to use purge_dups; this tool will allow us to identify and reassign allelic contigs.</p>

<h2 id="remove-haplotypic-duplication-with-purge_dups">Remove haplotypic duplication with <strong>purge_dups</strong></h2>

<p>This stage consists of three substages: coverage related calculation, generation of all versus all self-alignment and resolution of haplotigs and overlaps (fig. 10).</p>

<figure id="figure-9"><img src="../../images/vgp_assembly/purge_dupspipeline.png" alt="fig4:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Purge_dups pipeline. Adapted from github.com/dfguan/purge_dups.</figcaption></figure>

<h3 id="read-depth-analysis">Read-depth analysis</h3>

<p>Initially, we need to collapse our HiFi trimmed <span class="notranslate">reads</span> collection into a single dataset.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-collapse-the-collection"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Collapse the collection</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCollection of files to collapse into single dataset‚Äù</em>:<code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">HiFi <span class="notranslate">reads</span> collapsed</code></li>
  </ol>
</blockquote>

<p>Now, we will map the <span class="notranslate">reads</span> against the primary assembly by using Minimap2 (<span class="citation"><a href="#Li2018">Li 2018</a></span>), an alignment program designed to map long sequences.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-map-the-reads-to-contigs-with-minimap2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Map the <span class="notranslate">reads</span> to contigs with <strong>Minimap2</strong></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>4</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi <span class="notranslate">reads</span> collapsed</code></li>
            <li><em>‚ÄúSelect a profile of preset options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Long assembly to reference <span class="notranslate">mapping</span> (-k19 -w19 -A1 -B19 -O39,81 -E3,1 -s200 -z200 --min-occ-floor=100). Typically, the alignment will not extend to regions with 5% or higher sequence divergence. Only use this preset if the average divergence is far below 5%. (asm5)</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúSet advanced output options‚Äù</em>:
          <ul>
            <li><em>‚ÄúSelect an output format‚Äù</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> mapped to contigs</code></li>
  </ol>
</blockquote>

<p>Finally, we will use the <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> mapped to contigs</code> pairwise <span class="notranslate">mapping</span> format (PAF) file for calculating some statistics required in a later stage. In this step, purge_dups initially produces a read-depth histogram from base-level coverages. This information is used for estimating the coverage cutoffs, taking into account that collapsed haplotype contigs will lead to <span class="notranslate">reads</span> from both alleles <span class="notranslate">mapping</span> to those contigs, whereas if the alleles have assembled as separate contigs, then the <span class="notranslate">reads</span> will be split over the two contigs, resulting in half the read-depth (<span class="citation"><a href="#Roach2018">Roach <i>et al.</i> 2018</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-depth-analsys"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read-depth analsys</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>3</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúFunction mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Calculate coverage cutoff, base-level read depth and create read depth histogram for PacBio data (calcuts+pbcstats)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPAF input file‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> mapped to contigs</code></li>
            <li>In <em>‚ÄúCalcuts options‚Äù</em>:
              <ul>
                <li><em>‚ÄúUpper bound for read depth‚Äù</em>: <code class="language-plaintext highlighter-rouge">63</code> (the previously estimated maximum depth)</li>
                <li><em>‚ÄúPloidity‚Äù</em>: <code class="language-plaintext highlighter-rouge">Haploid</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>In case you are working with a diploid organism, you should select <code class="language-plaintext highlighter-rouge">diploid</code> in the ploidy option.</p>
      </blockquote>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage primary</code>, <code class="language-plaintext highlighter-rouge">Histogram plot primary</code> and <code class="language-plaintext highlighter-rouge">Calcuts cutoff primary</code>.</li>
  </ol>
</blockquote>

<p>Purge_dups generates three outputs:</p>

<ul>
  <li>PBCSTAT base coverage: it contains the base-level coverage information.</li>
  <li>Calcuts-cutoff: it includes the thresholds calculated by purge_dups.</li>
  <li>Histogram plot.</li>
</ul>

<p>As we can see in the histogram plot, the read depth distribution follows a symmetric and unimodal distribution (fig. 11). This result is expected, since we are assembling a haploid genome.</p>

<figure id="figure-10"><img src="../../images/vgp_assembly/purge_dups_plot.png" alt="fi9:purge_dups plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> purge_dups histogram plot</figcaption></figure>

<blockquote class="details">
  <h3 id="details-expected-results-in-diploid-organisms"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Expected results in diploid organisms</h3>

  <p>Diploid organisms frequently result in bimodal distributions when allelic sequences from the same genomic region are assembled separately.</p>

  <figure id="figure-11"><img src="../../images/vgp_assembly/purge_dups_plot2.png" alt="fi9:purge_dups plot. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> purge_dups histogram generated from a diploid genome.</figcaption></figure>

</blockquote>

<h3 id="generation-of-all-versus-all-self-alignmnet">Generation of all versus all self-alignmnet</h3>

<p>Now, we will segment the draft assembly into contigs by cutting at blocks of <em>N</em>s, and use minimap2 to generate an all by all self-alignment.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-purge_dups-pipeline"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: purge_dups pipeline</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúFunction mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">split assembly FASTA file by 'N's (split_fa)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAssembly FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">Split FASTA</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>4</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Split FASTA</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Split FASTA</code></li>
            <li><em>‚ÄúSelect a profile of preset options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Construct a self-homology map - use the same genome as query and reference (-DP -k19 -w 19 -m200) (self-homology)</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúSet advanced output options‚Äù</em>:
          <ul>
            <li><em>‚ÄúSelect an output format‚Äù</em>: <code class="language-plaintext highlighter-rouge">PAF</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Self-homology map primary</code></li>
  </ol>
</blockquote>

<h3 id="resolution-of-haplotigs-and-overlaps">Resolution of haplotigs and overlaps</h3>

<p>During the final step of the purge_dups pipeline, it will use the self alignments and the cutoffs for identifying the haplotypic duplications.</p>

<blockquote class="comment">
  <h3 id="comment-purge_dups-algorithm-details"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Purge_dups algorithm details</h3>

  <p>In order to identify the haplotypic duplications, purge_dups uses the  base-level coverage information to flag the contigs according the following criteria:</p>
  <ul>
    <li>If more than 80% bases of a contig are above the high read depth cutoff or below the noise cutoff, it is discarded.</li>
    <li>If more than 80% bases are in the diploid depth interval, it is labelled as a primary contig, otherwise it is considered further as a possible haplotig.</li>
  </ul>

  <p>Contigs that were flagged for further analysis according to read-depth are then evaluated to attempt to identify synteny with its allelic companion contig. In this step, purge_dups uses the information contained in the self alignments:</p>
  <ul>
    <li>
      <p>If the alignment score is larger than the cutoff <em>s</em> (default 70), the contig is marked for reassignment as haplotig. Contigs marked for reassignment with a maximum match score greater than the cutoff <em>m</em> (default 200) are further flagged as repetitive regions.</p>
    </li>
    <li>
      <p>Otherwise contigs are considered as a candidate primary contig.</p>
    </li>
  </ul>

  <p>Once all matches associated with haplotigs have been removed from the self-alignment set, purge_dups ties consistent matches between the remaining candidates to find collinear matches, filtering all the matches whose score is less than the minimum chaining score <em>l</em>.</p>

  <p>Finally, purge_dups calculates the average coverage of the matching intervals for each overlap, and mark an unambiguous overlap as heterozygous when the average coverage on both contigs is less than the read-depth cutoff, removing the sequences corresponding to the matching interval in the shorter contig.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-resolution-of-haplotigs-and-overlaps"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Resolution of haplotigs and overlaps</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>5</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSelect the purge_dups function‚Äù</em>: <code class="language-plaintext highlighter-rouge">Purge haplotigs and overlaps for an assembly (purge_dups)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPAF input file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Self-homology map primary</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBase-level coverage file‚Äù</em>: <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage primary</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúCutoffs file‚Äù</em>: <code class="language-plaintext highlighter-rouge">calcuts cutoff primary</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSelect the purge_dups function‚Äù</em>: <code class="language-plaintext highlighter-rouge">Obtain sequences after purging (get_seqs)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAssembly FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">BED</span> input file‚Äù</em>: <code class="language-plaintext highlighter-rouge">purge_dups <span class="notranslate">BED</span></code> (output of the previous step)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output <code class="language-plaintext highlighter-rouge">get_seq purged sequences</code> as <code class="language-plaintext highlighter-rouge">Primary contigs purged</code> and the <code class="language-plaintext highlighter-rouge">get_seq haplotype</code> file as <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te haplotype contigs</code>.</li>
  </ol>

</blockquote>

<h3 id="process-the-alternative-assembly">Process the alte<span class="notranslate">rna</span>tive assembly</h3>

<p>Now we should repeat the same procedure with the alte<span class="notranslate">rna</span>te contigs generated by hifiasm.  In that case, we should start by merging the <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te haplotype contigs</code> generated in the previous step and the <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code> file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-merge-the-purged-sequences-and-the-alternate-contigs"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Merge the purged sequences and the Alte<span class="notranslate">rna</span>te contigs</h3>

  <ol>
    <li><span class="tool" data-tool="cat1" title="Tested with cat1"><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: cat1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConcatenate Dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs FASTA</code></li>
        <li>In <em>‚ÄúDataset‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Dataset‚Äù</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect‚Äù</em>: <code class="language-plaintext highlighter-rouge">Purged sequences</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Remember that the <code class="language-plaintext highlighter-rouge">Purged sequences</code> file contains those contigs that were considered to be haplotypic duplications of the primary contigs.</p>
      </blockquote>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs full</code></li>
  </ol>

</blockquote>

<p>Once we have merged the files, we should run the purge_dups pipeline again, but using the <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs full</code> file as input.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-process-the-alternate-assembly-with-purge_dups"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Process the alte<span class="notranslate">rna</span>te assembly with <em>purge_dups</em></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>4</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs full</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi <span class="notranslate">reads</span> collapsed</code></li>
            <li><em>‚ÄúSelect a profile of preset options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Long assembly to reference <span class="notranslate">mapping</span> (-k19 -w19 -A1 -B19 -O39,81 -E3,1 -s200 -z200 --min-occ-floor=100). Typically, the alignment will not extend to regions with 5% or higher sequence divergence. Only use this preset if the average divergence is far below 5%. (asm5)</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúSet advanced output options‚Äù</em>:
          <ul>
            <li><em>‚ÄúSelect an output format‚Äù</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> mapped to contigs alte<span class="notranslate">rna</span>te</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>3</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúFunction mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Calculate coverage cutoff, base-level read depth and create read depth histogram for PacBio data (calcuts+pbcstats)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPAF input file‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> mapped to contigs alte<span class="notranslate">rna</span>te</code></li>
            <li>In <em>‚ÄúCalcuts options‚Äù</em>:
              <ul>
                <li><em>‚ÄúUpper bound for read depth‚Äù</em>: <code class="language-plaintext highlighter-rouge">63</code></li>
                <li><em>‚ÄúPloidity‚Äù</em>: <code class="language-plaintext highlighter-rouge">Haploid</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>In case you are working with a diploid organism, you should select <code class="language-plaintext highlighter-rouge">diploid</code>.</p>
      </blockquote>
    </li>
    <li>
      <p>Rename the outputs as ` PBCSTAT base coverage alte<span class="notranslate">rna</span>te<code class="language-plaintext highlighter-rouge">, </code>Histogram plot alte<span class="notranslate">rna</span>te<code class="language-plaintext highlighter-rouge"> and </code>Calcuts cutoff alte<span class="notranslate">rna</span>te`.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúFunction mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">split assembly FASTA file by 'N's (split_fa)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAssembly FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs full</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">Split FASTA alte<span class="notranslate">rna</span>te</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+<span class="notranslate">galaxy</span>4</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Split FASTA alte<span class="notranslate">rna</span>te</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Split FASTA alte<span class="notranslate">rna</span>te</code></li>
            <li><em>‚ÄúSelect a profile of preset options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Construct a self-homology map - use the same genome as query and reference (-DP -k19 -w 19 -m200) (self-homology)</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúSet advanced output options‚Äù</em>:
          <ul>
            <li><em>‚ÄúSelect an output format‚Äù</em>: <code class="language-plaintext highlighter-rouge">PAF</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">Self-homology map alte<span class="notranslate">rna</span>te</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>5</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSelect the purge_dups function‚Äù</em>: <code class="language-plaintext highlighter-rouge">Purge haplotigs and overlaps for an assembly (purge_dups)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPAF input file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Self-homology map alte<span class="notranslate">rna</span>te</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBase-level coverage file‚Äù</em>: <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage alte<span class="notranslate">rna</span>te</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúCutoffs file‚Äù</em>: <code class="language-plaintext highlighter-rouge">calcuts cutoff alte<span class="notranslate">rna</span>te</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2"><strong>purge_dups</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSelect the purge_dups function‚Äù</em>: <code class="language-plaintext highlighter-rouge">Obtain sequences after purging (get_seqs)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAssembly FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs full</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">BED</span> input file‚Äù</em>: <code class="language-plaintext highlighter-rouge">purge_dups <span class="notranslate">BED</span> alte<span class="notranslate">rna</span>te</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs purged</code> and <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te haplotype contigs</code>.</li>
  </ol>

</blockquote>

<h2 id="second-round-of-assembly-evaluation">Second round of assembly evaluation</h2>

<p>Once we have run purge_dups, we can evaluate the assembly again, and compare the results before and after purging. In this case, we will only evaluate the primary assembly.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assembly-evaluation-with-quast-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: assembly evaluation with QUAST</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúUse customized names for the input files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, specify custom names</code></li>
        <li>In <em>‚Äú1. Contigs/scaffolds‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs purged</code></li>
            <li><em>‚ÄúName‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly</code></li>
          </ul>
        </li>
        <li>Click in <em>‚ÄúInsert Contigs/scaffolds‚Äù</em></li>
        <li>In <em>‚Äú2. Contigs/scaffolds‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs purged</code></li>
            <li><em>‚ÄúName‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te assembly</code></li>
          </ul>
        </li>
        <li><em>‚Äú<span class="notranslate">Reads</span> options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Pacbio SMRT <span class="notranslate">reads</span></code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ file‚Äù</em>: <code class="language-plaintext highlighter-rouge">HiFi <span class="notranslate">reads</span> collapsed</code></li>
          </ul>
        </li>
        <li><em>‚ÄúType of assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome</code>
          <ul>
            <li><em>‚ÄúUse a reference genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúEstimated reference genome size (in bp) for computing NGx statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">12792715</code> (previously estimated)</li>
              </ul>
            </li>
            <li><em>‚ÄúType of organism‚Äù</em>: <code class="language-plaintext highlighter-rouge">Eukaryote: use of GeneMark-ES for gene finding, Bar<span class="notranslate">rna</span>p for ribosomal <span class="notranslate">RNA</span> genes prediction, BUSCO for conserved orthologs finding (--eukaryote)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs genome large (&gt;100Mpb)?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>Rename the HTML report as <code class="language-plaintext highlighter-rouge">QUAST second report</code></li>
  </ol>

</blockquote>

<p>In the figure 13 we can see the cumulative length plot corresponding to the primary assembly after being processed.</p>

<figure id="figure-12"><img src="../../images/vgp_assembly/QUAST_cummulative.png" alt="fig12:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> QUAST cumulative plot</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many contigs does the primary assembly have after being processed with purge_dups? What about before purge_dups?</li>
    <li>Which percentage of <span class="notranslate">reads</span> mapped to the primary assembly?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The primary assembly includes 37 contigs after being processed. The original assembly included 40 contigs.</li>
      <li>According the report, 100% of <span class="notranslate">reads</span> mapped to the primary assembly.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Now, let‚Äôs evaluate the assembly with BUSCO.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assessing-assembly-completeness-with-busco-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: assessing assembly completeness with BUSCO</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0"><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSequences to analyse‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary contigs purged</code> and <code class="language-plaintext highlighter-rouge">Alte<span class="notranslate">rna</span>te contigs purged</code></li>
        <li><em>‚ÄúMode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (<span class="notranslate">DNA</span>)</code>
          <ul>
            <li><em>‚ÄúUse Augustus instead of Metaeuk‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
          </ul>
        </li>
        <li><em>‚ÄúAuto-detect or select lineage?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
          <ul>
            <li><em>‚ÄúLineage‚Äù</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdvanced Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: <code class="language-plaintext highlighter-rouge">short summary text</code> and <code class="language-plaintext highlighter-rouge">summary image</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the summary as <code class="language-plaintext highlighter-rouge">BUSCO second report</code></li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How many complete BUSCO genes have been identified?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>According the report, the primary assembly contains 2021 complete BUSCO genes.</p>

  </blockquote>

</blockquote>

<h1 id="hybrid-scaffolding">Hybrid scaffolding</h1>

<p>Once we have obtained the primary and alte<span class="notranslate">rna</span>te assemblies, the next step is to reduce the fragmentation of the assembly. This will be done by combining the assembled contigs with Bionano optical maps to generate hybrid maps. The resulting contigs will be further scaffolded using Hi-C data.</p>

<h2 id="hybrid-scaffolding-using-bionano-data">Hybrid scaffolding using Bionano data</h2>

<p>In this step, the linkage information provided by optical maps is integrated with primary assembly sequences, and the overlaps are used to orient and order the contigs, resolve chimeric joins, and estimate the length of gaps between adjacent contigs. One of the advantage of optical maps is that can easily span genomic regions that are difficult to resolve using <span class="notranslate">DNA</span> sequencing technologies (<span class="citation"><a href="#Savara2021">Savara <i>et al.</i> 2021</a></span>, <span class="citation"><a href="#Yuan2020">Yuan <i>et al.</i> 2020</a></span>).</p>

<blockquote class="comment">
  <h3 id="comment-background-on-bionano-optical-maps"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Background on Bionano optical maps</h3>

  <p>Bionano technology relies on the isolation of kilobase-long <span class="notranslate">DNA</span> fragments, which are labeled at specific sequence motifs with a fluorescent dye, resulting in a unique fluorescent pattern for each genome. <span class="notranslate">DNA</span> molecules are stretched into nanoscale channels and imaged with a high-resolution camera, allowing us to build optical maps that include the physical locations of labels rather than base-level information (<span class="citation"><a href="#Lam2012">Lam <i>et al.</i> 2012</a></span>, <span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Savara2021">Savara <i>et al.</i> 2021</a></span>).</p>

  <figure id="figure-13"><img src="../../images/vgp_assembly/bionano.png" alt="Bionano optical maps. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Bionano optical maps. Source: https://bionanogenomics.com</figcaption></figure>

  <p>The average optical map molecule length, around 225 kbp, is substantially larger than the PacBio HiFi <span class="notranslate">reads</span>, with read lengths averaging 10-25 kbp.</p>

</blockquote>

<p>The <em>Bionano Hybrid Scaffold</em> tool automates the scaffolding process, which includes five main steps:</p>

<ol>
  <li>Generate <em>in silico</em> maps for sequence assembly.</li>
  <li>Align <em>in silico</em> sequence maps against Bionano genome maps to identify and resolve potential conflicts.</li>
  <li>Merge the non-conflicting maps into hybrid scaffolds.</li>
  <li>Align sequence maps to the hybrid scaffolds</li>
  <li>Generate AGP and FASTA files for the scaffolds.</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-bionano-hybrid-scaffolding"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Bionano hybrid scaffolding</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.6.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.6.1+galaxy2"><strong>Bionano Hybrid Scaffold</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.6.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúNGS FASTA‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBioNano CMAP‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
        <li><em>‚ÄúConfiguration mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">VGP mode</code></li>
        <li><em>‚ÄúGenome maps conflict filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cut contig at conflict</code></li>
        <li><em>‚ÄúSequences conflict filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cut contig at conflict</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>If your data are not associated with VGP, make sure that the configuration mode fits with your samples.</p>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="cat1" title="Tested with cat1"><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: cat1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConcatenate Dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">NGScontigs scaffold NCBI trimmed</code> (output of <strong>Bionano Hybrid Scaffold</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚ÄúDataset‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Dataset‚Äù</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect‚Äù</em>: <code class="language-plaintext highlighter-rouge">NGScontigs not scaffolded trimmed</code> (output of <strong>Bionano Hybrid Scaffold</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
  </ol>
</blockquote>

<p>Now, let‚Äôs evaluate the assembly:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-bionano-assembly-evaluation-with-quast"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Bionano assembly evaluation with QUAST</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúUse customized names for the input files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, use dataset names</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
          </ul>
        </li>
        <li><em>‚ÄúType of assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome</code>
          <ul>
            <li><em>‚ÄúUse a reference genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúEstimated reference genome size (in bp) for computing NGx statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">12792715</code> (previously estimated)</li>
              </ul>
            </li>
            <li><em>‚ÄúType of organism‚Äù</em>: <code class="language-plaintext highlighter-rouge">Eukaryote (--eukaryote): use of GeneMark-ES for gene finding, Bar<span class="notranslate">rna</span>p for ribosomal <span class="notranslate">RNA</span> genes prediction, BUSCO for conserved orthologs finding</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúGenes‚Äù</em>:
          <ul>
            <li><em>‚ÄúTool for gene prediction‚Äù</em>: <code class="language-plaintext highlighter-rouge">Don't predict genes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many scaffolds are in the primary assembly after the hybrid scaffolding? How does this compare to the number of contigs before scaffolding?</li>
    <li>What is the size of the largest scaffold? And before?</li>
    <li>Has the N50 increased after the hybrid scaffolding?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The number of contigs have been reduced from 37 to 20 scaffolds after the hybrid scaffolding.</li>
      <li>The largest contig is 1.506.138 bp long. Previously it was 914.549 bp.</li>
      <li>N50 has increased from 425.706 up to 926.785.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="hybrid-scaffolding-based-on-hi-c-mapping-data">Hybrid scaffolding based on Hi-C <span class="notranslate">mapping</span> data</h2>

<p>Hi-C is a sequencing-based molecular assay designed to identify regions of frequent physical interaction in the genome by measuring the contact frequency between all pairs of loci, allowing us to provide an insight into the three-dimensional organisation of a genome  (<span class="citation"><a href="#Dixon2012">Dixon <i>et al.</i> 2012</a></span>, <span class="citation"><a href="#LiebermanAiden2009">Lieberman-Aiden <i>et al.</i> 2009</a></span>). In this final stage, we will exploit the fact that the contact frequency between a pair of loci strongly correlates with the one-dimensional distance between them with the objective of linking the Bionano scaffolds to a chromosome scale.</p>

<blockquote class="comment">
  <h3 id="comment-background-about-hi-c-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Background about Hi-C data</h3>

  <p>The high-throughput chromosome conformation capture (Hi-C) technology is based on the capture of the chromatin in three-dimensional space. During Hi-C library preparation, <span class="notranslate">DNA</span> is crosslinked in its 3D conformation. Then, the <span class="notranslate">DNA</span> is digested using restriction enzymes, and the digested ends are filled with biotinylated nucleotides (fig. 14). The biotinylated nucleotides enable the specific purification of the ligation junctions, preventing the sequencing of <span class="notranslate">DNA</span> molecules that do not contain such junctions which are thus mostly uninformative (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <figure id="figure-14"><img src="../../images/vgp_assembly/hi-c_protocol.png" alt="fig4:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> Hi-C protocol. Adapted from Rao et al. 2014</figcaption></figure>

  <p>Next, the blunt ends of spatially proximal digested end are ligated. Each <span class="notranslate">DNA</span> fragment is then sequenced from each end of this artificial junction, generating read pairs. This provides contact information that can be used to reconstruct the proximity of genomic sequences belonging to the same chromosome (<span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>). Hi-C data are in the form of two-dimensional matrices (contact maps) whose entries quantify the intensity of the physical interaction between genome regions.</p>

</blockquote>

<h3 id="pre-processing-hi-c-data">Pre-processing Hi-C data</h3>

<p>Despite Hi-C generating paired-end <span class="notranslate">reads</span>, we need to map each read separately. This is because most aligners assume that the distance between paired-end <span class="notranslate">reads</span> fit a known distribution, but in Hi-C data the insert size of the ligation product can vary between one base pair to hundreds of megabases (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping-hi-c-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span> Hi-C <span class="notranslate">reads</span></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>‚ÄúBAM sorting mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM forward</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: Primary assembly bionano`</li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>‚ÄúBAM sorting mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM reverse</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0"><strong>Filter and merge</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFirst set of <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM forward</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSecond set of <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM  reverse</code></li>
      </ul>
    </li>
    <li>Rename it as <code class="language-plaintext highlighter-rouge">BAM Hi-C <span class="notranslate">reads</span></code></li>
  </ol>
</blockquote>

<p>Finally, we need to convert the BAM file to <span class="notranslate">BED</span> format and sort it.</p>

<h3 id="generate-initial-hi-c-contact-map">Generate initial Hi-C contact map</h3>

<p>After <span class="notranslate">mapping</span> the Hi-C <span class="notranslate">reads</span>, the next step is to generate an initial Hi-C contact map, which will allow us to compare the Hi-C contact maps before and after using the Hi-C for scaffolding.</p>

<blockquote class="comment">
  <h3 id="comment-biological-basis-of-hi-c-contact-maps"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Biological basis of Hi-C contact maps</h3>

  <p>Hi-C contact maps reflect the interaction frequency between genomic loci. In order to understand the Hi-C contact maps, it is necessary to take into account two factors: the higher interaction frequency between loci that reside in the same chromosome (<em>i.e.</em>, in cis), and the distance-dependent decay of interaction frequency (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <p>The higher interaction between cis regions can be explained, at least in part, by the territorial organization of chromosomes in interphase (chromosome territories), and in a genome-wide contact map, this pattern appears as blocks of high interaction centered along the diagonal and matching individual chromosomes (fig. 15) (<span class="citation"><a href="#Cremer2010">Cremer and Cremer 2010</a></span>, <span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <figure id="figure-15"><img src="../../images/vgp_assembly/hic_map.png" alt="fig4:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> An example of a Hi-C map. Genomic regions are arranged along the x and y axes, and contacts are colored on the matrix like a heat map; here darker color indicates greater interaction frequency.</figcaption></figure>

  <p>On the other hand, the distance-dependent decay may be due to random movement of the chromosomes, and in the contact map appears as a gradual decrease of the interaction frequency the farther away from the diagonal it moves (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-a-contact-map-with-pretextmap-and-pretext-snapshot"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate a contact map with <strong>PretextMap</strong> and <strong>Pretext Snapshot</strong></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0"><strong>PretextMap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput dataset in SAM or BAM format‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C <span class="notranslate">reads</span></code></li>
        <li><em>‚ÄúSort by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Don't sort</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">PretextMap output</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0"><strong>Pretext Snapshot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput Pretext map file‚Äù</em>: <code class="language-plaintext highlighter-rouge">PretextMap output</code></li>
        <li><em>‚ÄúOutput image format‚Äù</em>: <code class="language-plaintext highlighter-rouge">png</code></li>
        <li><em>‚ÄúShow grid?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Let‚Äôs have a look at the Hi-C contact maps generated by Pretext Snapshot.</p>

<figure id="figure-16"><img src="../../images/vgp_assembly/hic_map_pretext.png" alt="fig4:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 16:</span> Hi-C map generated by Pretext. Full map (a) and Super-Scaffold_100003 (b)</figcaption></figure>

<h3 id="salsa2-scaffolding">SALSA2 scaffolding</h3>

<p>SALSA2 is an open source software that makes use of Hi-C to linearly orient and order assembled contigs along entire chromosomes.</p>

<blockquote class="comment">
  <h3 id="comment-salsa2-algorithm-overview"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> SALSA2 algorithm overview</h3>

  <p>Initially SALSA2 uses the physical coverage of Hi-C pairs to identify suspicious regions and break the sequence at the likely point of mis-assembly. Then, a hybrid scaffold graph is constructed using edges from the Hi-C <span class="notranslate">reads</span>, scoring the edges according to a <em>best buddy</em> scheme (fig. 16a).</p>

  <figure id="figure-17"><img src="../../images/vgp_assembly/salsa2_algorithm.png" alt="fig4:Post-processing step. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 17:</span> Overview of the SALSA2 algorithm. Solid edges indicate the linkages between different contigs and dotted edges indicate the links between the ends of the same contig. B and E denote the start and end of contigs, respectively. Adapted from Ghurye et al. 2019.</figcaption></figure>

  <p>From this graph scaffolds are iteratively constructed using a greedy weighted maximum matching. After each iteration, a mis-join detection step is performed to check if any of the joins made during this round are incorrect. Incorrect joins are broken and the edges blacklisted during subsequent iterations. This process continues until the majority of joins made in the prior iteration are incorrect. This provides a natural stopping condition, when accurate Hi-C links have been exhausted (<span class="citation"><a href="#Ghurye2019">Ghurye <i>et al.</i> 2019</a></span>).</p>

</blockquote>

<p>Before launching SALSA2, we need to carry out some modifications on our datasets.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-bam-to-bed-conversion"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: BAM to <span class="notranslate">BED</span> conversion</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy1"><strong><span class="notranslate">bed</span>tools BAM to <span class="notranslate">BED</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/<span class="notranslate">bed</span>tools/<span class="notranslate">bed</span>tools_bamtobed/2.30.0+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConvert the following BAM file to <span class="notranslate">BED</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C <span class="notranslate">reads</span></code></li>
        <li><em>‚ÄúWhat type of <span class="notranslate">BED</span> output would you like‚Äù</em>: <code class="language-plaintext highlighter-rouge">Create a full, 12-column "blocked" <span class="notranslate">BED</span> file</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span> unsorted</code></p>
    </li>
    <li><span class="tool" data-tool="sort1" title="Tested with sort1"><strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: sort1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSort Dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span> unsorted</code></li>
        <li><em>‚Äúon column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
        <li><em>‚Äúwith flavor‚Äù</em>: <code class="language-plaintext highlighter-rouge">Alphabetical sort</code></li>
        <li><em>‚Äúeverything in‚Äù</em>: <code class="language-plaintext highlighter-rouge">Ascending order</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span> sorted</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to process‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
        <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">:</code></li>
        <li><em>‚ÄúReplace all occurrences of the pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFind and Replace text in‚Äù</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Primary assembly bionano edited</code></li>
  </ol>
</blockquote>

<p>Now we can launch SALSA2 in order to generate the hybrid scaffolding based on the Hi-C data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-salsa-scaffolding"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Salsa scaffolding</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/salsa/salsa/2.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/salsa/salsa/2.3+galaxy0"><strong>SALSA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/salsa/salsa/2.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInitial assembly file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano edited</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Bed</span> alignment‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span> sorted</code></li>
        <li><em>‚ÄúRestriction enzyme sequence(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTTAAG</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code> and <code class="language-plaintext highlighter-rouge">SALSA2 scaffold AGP</code></li>
  </ol>

</blockquote>

<p>Finally, let‚Äôs evaluate the hybrid scaffolded assembly generated by SALSA2.</p>

<h3 id="evaluate-the-salsa-scaffolding-results">Evaluate the Salsa scaffolding results</h3>

<p>In order to evaluate the results, we will use QUAST and Pretext.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-evaluation-with-busco"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Evaluation with BUSCO</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.2.2+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.2.2+galaxy0"><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/busco/busco/5.2.2+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSequences to analyse‚Äù</em>: <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code></li>
        <li><em>‚ÄúMode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (<span class="notranslate">DNA</span>)</code>
          <ul>
            <li><em>‚ÄúUse Augustus instead of Metaeuk‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
          </ul>
        </li>
        <li><em>‚ÄúLineage‚Äù</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
        <li>In <em>‚ÄúAdvanced Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: <code class="language-plaintext highlighter-rouge">short summary text</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BUSCO final report</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+galaxy1"><strong>Quast</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/quast/quast/5.0.2+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúUse customized names for the input files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, use dataset names</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúContigs/scaffolds file‚Äù</em>: <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code></li>
          </ul>
        </li>
        <li><em>‚ÄúType of assembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genome</code>
          <ul>
            <li><em>‚ÄúUse a reference genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúEstimated reference genome size (in bp) for computing NGx statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">12792715</code> (previously estimated)</li>
              </ul>
            </li>
            <li><em>‚ÄúType of organism‚Äù</em>: <code class="language-plaintext highlighter-rouge">Eukaryote (--eukaryote): use of GeneMark-ES for gene finding, Bar<span class="notranslate">rna</span>p for ribosomal <span class="notranslate">RNA</span> genes prediction, BUSCO for conserved orthologs finding</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs genome large (&gt; 100 Mbp)?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>Rename the HTML report as <code class="language-plaintext highlighter-rouge">QUAST final report</code></li>
  </ol>
</blockquote>

<p>We will analyse the results generated by BUSCO and QUAST to assess the extent to which SALSA2 has improved our genome assembly.</p>

<blockquote class="question">
  <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is the size of the largest scaffold?</li>
    <li>How many scaffolds are in the final assembly?</li>
    <li>What is the N50 statistic?</li>
    <li>How many complete BUSCO genes are in the final assembly?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The largest contig has 2.592.144 bp.</li>
      <li>The final assembly includes 19 scaffolds.</li>
      <li>The N50 value is 801.160.</li>
      <li>The final assembly contains 2121 complete BUSCO genes.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Finally, we should repeat the procedure described previously for generating the contact maps, but in that case, we will use the scaffold generated by SALSA2.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping-reads-against-the-scaffold"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span> <span class="notranslate">reads</span> against the scaffold</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">scaffolds_fasta</code> (output of <strong>SALSA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>‚ÄúBAM sorting mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM forward SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect fastq dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_R</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>‚ÄúBAM sorting mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM reverse SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0"><strong>Filter and merge</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFirst set of <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM forward SALSA2</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSecond set of <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM reverse SALSA2</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM Hi-C <span class="notranslate">reads</span> SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0"><strong>PretextMap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput dataset in SAM or BAM format‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C <span class="notranslate">reads</span> SALSA2</code></li>
        <li><em>‚ÄúSort by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Don't sort</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">PretextMap output SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0"><strong>Pretext Snapshot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput Pretext map file‚Äù</em>: <code class="language-plaintext highlighter-rouge">PretextMap output SALSA2</code></li>
        <li><em>‚ÄúOutput image format‚Äù</em>: <code class="language-plaintext highlighter-rouge">png</code></li>
        <li><em>‚ÄúShow grid?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<figure id="figure-18"><img src="../../images/vgp_assembly/hi-c_pretext_final.png" alt="fig4:Post-processing step. " width="80%" loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 18:</span> Hi-C map generated by Pretext after the hybrid scaffolding based on Hi-C data.</figcaption></figure>



                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>The VGP pipeline allows to generate error-free, near gapless reference-quality genome assemblies</p>
</li>
                        
                        <li><p>The assembly can be divided in four main stages: genome profile analysis, HiFi long read phased assembly with hifiasm, Bionano hybrid scaffolding and Hi-C hybrid scaffolding</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/assembly/faqs/">FAQ page for the Assembly topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Marcotte1999">Marcotte, E. M., M. Pellegrini, T. O. Yeates, and D. Eisenberg, 1999 <b>A census of protein repeats</b>. Journal of Molecular Biology 293: 151‚Äì160. <a href="https://doi.org/10.1006/jmbi.1999.3136">10.1006/jmbi.1999.3136</a></li>
<li id="Small2007">Small, K. S., M. Brudno, M. M. Hill, and A. Sidow, 2007 <b>A haplome alignment and reference sequence of the highly polymorphic Ciona savignyi genome</b>. Genome Biology 8: R41. <a href="https://doi.org/10.1186/gb-2007-8-3-r41">10.1186/gb-2007-8-3-r41</a></li>
<li id="LiebermanAiden2009">Lieberman-Aiden, E., N. L. van Berkum, L. Williams, M. Imakaev, T. Ragoczy <i>et al.</i>, 2009 <b>Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome</b>. Science 326: 289‚Äì293. <a href="https://doi.org/10.1126/science.1181369">10.1126/science.1181369</a></li>
<li id="Cremer2010">Cremer, T., and M. Cremer, 2010 <b>Chromosome Territories</b>. Cold Spring Harbor Perspectives in Biology 2: a003889‚Äìa003889. <a href="https://doi.org/10.1101/cshperspect.a003889">10.1101/cshperspect.a003889</a></li>
<li id="Dixon2012">Dixon, J. R., S. Selvaraj, F. Yue, A. Kim, Y. Li <i>et al.</i>, 2012 <b>Topological domains in mammalian genomes identified by analysis of chromatin interactions</b>. Nature 485: 376‚Äì380. <a href="https://doi.org/10.1038/nature11082">10.1038/nature11082</a></li>
<li id="Frenkel2012">Frenkel, S., V. Kirzhner, and A. Korol, 2012 <b>Organizational Heterogeneity of Vertebrate Genomes</b> (V. Laudet, Ed.). PLoS ONE 7: e32076. <a href="https://doi.org/10.1371/journal.pone.0032076">10.1371/journal.pone.0032076</a></li>
<li id="Lam2012">Lam, E. T., A. Hastie, C. Lin, D. Ehrlich, S. K. Das <i>et al.</i>, 2012 <b>Genome mapping on nanochannel arrays for structural variation analysis and sequence assembly</b>. Nature Biotechnology 30: 771‚Äì776. <a href="https://doi.org/10.1038/nbt.2303">10.1038/nbt.2303</a></li>
<li id="Chalopin2015">Chalopin, D., M. Naville, F. Plard, D. Galiana, and J.-N. Volff, 2015 <b>Comparative Analysis of Transposable Elements Highlights Mobilome Diversity and Evolution in Vertebrates</b>. Genome Biology and Evolution 7: 567‚Äì580. <a href="https://doi.org/10.1093/gbe/evv005">10.1093/gbe/evv005</a></li>
<li id="Lajoie2015">Lajoie, B. R., J. Dekker, and N. Kaplan, 2015 <b>The Hitchhiker‚Äôs guide to Hi-C analysis: Practical guidelines</b>. Methods 72: 65‚Äì75. <a href="https://doi.org/10.1016/j.ymeth.2014.10.031">10.1016/j.ymeth.2014.10.031</a></li>
<li id="Simo2015">Sim√£o, F. A., R. M. Waterhouse, P. Ioannidis, E. V. Kriventseva, and E. M. Zdobnov, 2015 <b>BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs</b>. Bioinformatics 31: 3210‚Äì3212. <a href="https://doi.org/10.1093/bioinformatics/btv351">10.1093/bioinformatics/btv351</a></li>
<li id="SoteroCaio2017">Sotero-Caio, C. G., R. N. Platt, A. Suh, and D. A. Ray, 2017 <b>Evolution and Diversity of Transposable Elements in Vertebrate Genomes</b>. Genome Biology and Evolution 9: 161‚Äì177. <a href="https://doi.org/10.1093/gbe/evw264">10.1093/gbe/evw264</a></li>
<li id="DominguezDelAngel2018">Angel, V. D. D., E. Hjerde, L. Sterck, S. Capella-Gutierrez, C. Notredame <i>et al.</i>, 2018 <b>Ten steps to get started in Genome Assembly and Annotation</b>. F1000Research 7: 148. <a href="https://doi.org/10.12688/f1000research.13598.1">10.12688/f1000research.13598.1</a></li>
<li id="Li2018">Li, H., 2018 <b>Minimap2: pairwise alignment for nucleotide sequences</b> (I. Birol, Ed.). Bioinformatics 34: 3094‚Äì3100. <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a></li>
<li id="Roach2018">Roach, M. J., S. A. Schmidt, and A. R. Borneman, 2018 <b>Purge Haplotigs: allelic contig reassignment for third-gen diploid genome assemblies</b>. BMC Bioinformatics 19: <a href="https://doi.org/10.1186/s12859-018-2485-7">10.1186/s12859-018-2485-7</a></li>
<li id="Ghurye2019">Ghurye, J., A. Rhie, B. P. Walenz, A. Schmitt, S. Selvaraj <i>et al.</i>, 2019 <b>Integrating Hi-C links with assembly graphs for chromosome-scale assembly</b> (I. Ioshikhes, Ed.). PLOS Computational Biology 15: e1007273. <a href="https://doi.org/10.1371/journal.pcbi.1007273">10.1371/journal.pcbi.1007273</a></li>
<li id="Guan2019">Guan, D., S. A. McCarthy, J. Wood, K. Howe, Y. Wang <i>et al.</i>, 2019 <b>Identifying and removing haplotypic duplication in primary genome assemblies</b>. <a href="https://doi.org/10.1101/729962">10.1101/729962</a></li>
<li id="Trresen2019">T√∏rresen, O. K., B. Star, P. Mier, M. A. Andrade-Navarro, A. Bateman <i>et al.</i>, 2019 <b>Tandem repeats lead to sequence assembly errors and impose multi-level challenges for genome and protein databases</b>. Nucleic Acids Research 47: 10994‚Äì11006. <a href="https://doi.org/10.1093/nar/gkz841">10.1093/nar/gkz841</a></li>
<li id="Giani2020">Giani, A. M., G. R. Gallo, L. Gianfranceschi, and G. Formenti, 2020 <b>Long walk to genomics: History and current approaches to genome sequencing and assembly</b>. Computational and Structural Biotechnology Journal 18: 9‚Äì19. <a href="https://doi.org/10.1016/j.csbj.2019.11.002">10.1016/j.csbj.2019.11.002</a></li>
<li id="Hon2020">Hon, T., K. Mars, G. Young, Y.-C. Tsai, J. W. Karalius <i>et al.</i>, 2020 <b>Highly accurate long-read HiFi sequencing data for five complex genomes</b>. Scientific Data 7: <a href="https://doi.org/10.1038/s41597-020-00743-4">10.1038/s41597-020-00743-4</a></li>
<li id="RanalloBenavidez2020">Ranallo-Benavidez, T. R., K. S. Jaron, and M. C. Schatz, 2020 <b>GenomeScope 2.0 and Smudgeplot for reference-free profiling of polyploid genomes</b>. Nature Communications 11: <a href="https://doi.org/10.1038/s41467-020-14998-3">10.1038/s41467-020-14998-3</a></li>
<li id="Rhie2020">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="wang2020estimation">Wang, H., B. Liu, Y. Zhang, F. Jiang, Y. Ren <i>et al.</i>, 2020 <b>Estimation of genome size using k-mer frequencies from corrected long reads</b>. arXiv preprint arXiv:2003.11817.</li>
<li id="Yuan2020">Yuan, Y., C. Y.-L. Chung, and T.-F. Chan, 2020 <b>Advances in optical mapping for genomic research</b>. Computational and Structural Biotechnology Journal 18: 2051‚Äì2062. <a href="https://doi.org/10.1016/j.csbj.2020.07.018">10.1016/j.csbj.2020.07.018</a></li>
<li id="Zhang2020">Zhang, X., R. Wu, Y. Wang, J. Yu, and H. Tang, 2020 <b>Unzipping haplotypes in diploid and polyploid genomes</b>. Computational and Structural Biotechnology Journal 18: 66‚Äì72. <a href="https://doi.org/10.1016/j.csbj.2019.11.011">10.1016/j.csbj.2019.11.011</a></li>
<li id="Cheng2021">Cheng, H., G. T. Concepcion, X. Feng, H. Zhang, and H. Li, 2021 <b>Haplotype-resolved de novo assembly using phased assembly graphs with hifiasm</b>. Nature Methods 18: 170‚Äì175. <a href="https://doi.org/10.1038/s41592-020-01056-5">10.1038/s41592-020-01056-5</a></li>
<li id="Dida2021">Dida, F., and G. Yi, 2021 <b>Empirical evaluation of methods for de novo genome assembly</b>. PeerJ Computer Science 7: e636. <a href="https://doi.org/10.7717/peerj-cs.636">10.7717/peerj-cs.636</a></li>
<li id="Rhie2021">Rhie, A., S. A. McCarthy, O. Fedrigo, J. Damas, G. Formenti <i>et al.</i>, 2021 <b>Towards complete and error-free genome assemblies of all vertebrate species</b>. Nature 592: 737‚Äì746. <a href="https://doi.org/10.1038/s41586-021-03451-0">10.1038/s41586-021-03451-0</a></li>
<li id="Savara2021">Savara, J., T. Novos√°d, P. Gajdo≈°, and E. Kriegov√°, 2021 <b>Comparison of structural variants detected by optical mapping with long-read next-generation sequencing</b> (C. Kendziorski, Ed.). Bioinformatics 37: 3398‚Äì3404. <a href="https://doi.org/10.1093/bioinformatics/btab359">10.1093/bioinformatics/btab359</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=VGP assembly pipeline (Assembly)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Delphine Lariviere, Alex Ostrovsky, Crist√≥bal Gallardo, Anna Syme, Linelle Abueg, Brandon Pickett, 2022 <b>VGP assembly pipeline (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{assembly-vgp_genome_assembly,
author = "Delphine Lariviere and Alex Ostrovsky and Crist√≥bal Gallardo and Anna Syme and Linelle Abueg and Brandon Pickett",
title = "VGP assembly pipeline (Galaxy Training Materials)",
year = "2022",
month = "01",
day = "25"
url = "\url{https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
