<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Infinium Human Methylation BeadChip</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703550391' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703550391' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Infinium Human Methylation BeadChip">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-25 05:30:06 +0000">
<meta name="DC.creator" content="Katarzyna Kamieniecka">
<meta name="DC.creator" content="Krzysztof Poterlowicz">

        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Infinium Human Methylation BeadChip">
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-25 05:30:06 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Infinium Human Methylation BeadChip",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Infinium Human Methylation BeadChip' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html",
  "timeRequired": "PT1H",
  "teaches": "Learn how to perform reproducible Infinium Human Methylation BeadChip analysis\n - Visualise differentially methylated positions using UCSC browser",
  "keywords": "epigenetics",
  "description": "The questions this  addresses are:\n - Which DNA regions and positions are diffrentialy methylated in pre MAPKi treatment and post MAPKi resistance Melanomas GSE65183?\n - How to analyse and visualise Infinium Human Methylation BeadChip's?\n\n\nThe objectives are:\n - Learn how to perform reproducible Infinium Human Methylation BeadChip analysis\n - Visualise differentially methylated positions using UCSC browser\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "name": "Katarzyna Kamieniecka",
      "image": "https://avatars.githubusercontent.com/kkamieniecka",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0009-0004-2454-5950",
      "orcid": "https://orcid.org/0009-0004-2454-5950"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "name": "Krzysztof Poterlowicz",
      "image": "https://avatars.githubusercontent.com/poterlowicz-lab",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6173-5674",
      "orcid": "https://orcid.org/0000-0001-6173-5674"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "> <div class="box-title agenda-title" id="agenda">Agenda</div>"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703550391'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Epigenetics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/epigenetics/faqs/" title="Check our FAQs for the Epigenetics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/ewas-suite/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-epigenetics">
    <h1 data-toc-skip>Infinium Human Methylation BeadChip</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/kkamieniecka/" class="contributor-badge contributor-kkamieniecka"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/kkamieniecka?s=36" alt="Avatar" width="36" height="36">Katarzyna Kamieniecka</a><a href="/training-material/hall-of-fame/poterlowicz-lab/" class="contributor-badge contributor-poterlowicz-lab"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/poterlowicz-lab?s=36" alt="Avatar" width="36" height="36">Krzysztof Poterlowicz</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which DNA regions and positions are diffrentialy methylated in pre MAPKi treatment and post MAPKi resistance Melanomas GSE65183?</p>
</li>
        
        <li><p>How to analyse and visualise Infinium Human Methylation BeadChip’s?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn how to perform reproducible Infinium Human Methylation BeadChip analysis</p>
</li>
        
        <li><p>Visualise differentially methylated positions using UCSC browser</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/epigenetics/tutorials/ewas-suite/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/ewas-suite/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 25, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00139">https://gxy.io/GTN:T00139</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda-1">Agenda</div>
  <p>In this tutorial we will do:</p>
<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#raw-intensity-data-loading" id="markdown-toc-raw-intensity-data-loading">Raw intensity data loading</a></li>
  <li><a href="#idat-preprocessing" id="markdown-toc-idat-preprocessing">.idat preprocessing</a></li>
  <li><a href="#differentially-methylated-regions-and-positions-analysis" id="markdown-toc-differentially-methylated-regions-and-positions-analysis">Differentially methylated regions and positions analysis</a></li>
  <li><a href="#annotation-and-visualization" id="markdown-toc-annotation-and-visualization">Annotation and visualization</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>
  <p>We will use a small subset of the original data. If we run the tutorial on the orginal dataset, analysis will be time consuming and not reproducible Infinium Human Methylation BeadChip computation on the orginal data can be found at <a href="https://github.com/kpbioteam/case_study">case study</a>.</p>

</blockquote>

<p>This tutorial is based on Hugo W, Shi H, Sun L, Piva M et al.: Non-genomic and Immune Evolution of Melanoma Acquiring MAPKi Resistance <span class="citation"><a href="#Hugo2015">Hugo <i>et al.</i> 2015</a></span>.</p>

<p>The data we use in this tutorial are available at <a href="https://zenodo.org/record/1251211">Zenodo</a>.</p>

<hr />

<h1 id="introduction">Introduction</h1>

<p>The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provided training based on Galaxy Infinium Human Methylation BeadChip tool that represents many popular algorithms for detecting somatic genetic alterations from genome and exome data.</p>

<p><a href="../../images/Epigenetic_mechanisms.jpg" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/Epigenetic_mechanisms.jpg" alt="epimechanism. " width="647" height="436" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> How epigenetics mechanism can effect health (adapted from https://commonfund.nih.gov/epigenomics/figure)&lt;/figcaption&gt;</figure></a></p>

<p>This exercise uses datasets from the Cell publication by <span class="citation"><a href="#Hugo2015">Hugo <i>et al.</i> 2015</a></span>. with the goal being the identification of differentially methylated regions and positions associated with treatment resistant melanomas.
Datasets include the Infinium Human Methylation BeadChip array performed in melanoma tumors in a sample of patients pre and post MAPKi and BRAFi treatment with different outcomes (sensitive and resistant). For each sample there is raw green (methylated) and red (unmethylated) colour arrays containing the summarised bead information generated by the Infinium Human Methylation BeadChip scanner.</p>

<p>The Infinium Human Methylation BeadChip uses two different bead types to detect changes in DNA methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. In our study unmethylated and methylated bead signals are reported as green and red colors respectively.</p>

<p><a href="../../images/methylationassay.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/methylationassay.png" alt="methassay. " width="806" height="288" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Infinium Methylation Assay Overview (adapted from <span class="citation"><a href="#Illumina">“<b>Illumina Infinium Methylation Assay Overview</b>”</a></span>)&lt;/figcaption&gt;</figure></a></p>

<table>
  <thead>
    <tr>
      <th>Accession</th>
      <th>Sensitivity</th>
      <th>Treatment</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GSM1588704</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588705</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588706</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
    <tr>
      <td>GSM1588707</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
  </tbody>
</table>

<p>The workflow combines 5 main steps, starting with raw intensity data loading (.idat) and then optional preprocessing and normalisation of the data. The next quality control step performs an additional sample check to remove low-quality data, which normalisation cannot detect. The workflow gives the user the opportunity to perform any of these preparation and data cleaning steps, including a highly recommended genetic variation annotation step resulting in single nucleotide polymorphism identification and removal. Finally, the dataset generated through all of these steps can be used to hunt (find) differentially-methylated positions (DMP) and regions (DMR) with respect to a phenotype covariate.</p>

<h1 id="raw-intensity-data-loading">Raw intensity data loading</h1>

<p>The first step of the Infinium Human Methylation BeadChip array analysis is raw methylation data loading (intensity information files for each two colour micro array)</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-loading"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data Loading</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Import the following IDAT files from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> or from the data library (ask your instructor)
      <ul>
        <li><code class="language-plaintext highlighter-rouge">GSM1588704_8795207135_R01C02_Red.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588705_8795207119_R05C02_Red.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588706_8795207135_R02C02_Red.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588707_8795207119_R06C02_Red.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588704_8795207135_R01C02_Grn.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588705_8795207119_R05C02_Grn.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588706_8795207135_R02C02_Grn.idat</code></li>
        <li><code class="language-plaintext highlighter-rouge">GSM1588707_8795207119_R06C02_Grn.idat</code></li>
      </ul>

      <blockquote class="details">
        <div class="box-title details-title" id="details-list-of-zenodo-urls"><button class="gtn-boxify-button details" type="button" aria-controls="details-list-of-zenodo-urls" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: List of Zenodo URLs<span class="fold-unfold fa fa-minus-square"></span></button></div>
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Grn.idat
</code></pre></div>        </div>
      </blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Run <strong>Infinium Human Methylation BeadChip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“red channel files”</em>: all files ending in <code class="language-plaintext highlighter-rouge">_Red</code></li>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“green channel files”</em>: all files ending in <code class="language-plaintext highlighter-rouge">Grn</code>
<a href="../../images/1Raw_intensity_data_loading.png" rel="noopener noreferrer"><img src="../../images/1Raw_intensity_data_loading.png" alt="Raw_intensity_data_loading. " width="809" height="499" loading="lazy" /></a></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h1 id="idat-preprocessing">.idat preprocessing</h1>

<p>Preprocessing and data quality assurance is an important step in Infinium Methylation Assay analysis. Idat dataset represents two colour data with a green and a red channel and can be converted into methylated and unmethylated signals or into Beta values. The Infinium Human Methylation BeadChip tool extracts and plots the quality control data frame with two columns mMed and uMed which are the medians of methylation signals (Meth and Unmeth). Comparing them against one another allows users to detect and remove low-quality samples.</p>

<p><a href="../../images/quality_control.png" rel="noopener noreferrer"><img src="../../images/quality_control.png" alt="quality_control. " width="480" height="480" loading="lazy" /></a></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-preprocessing"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Preprocessing</div>
  <p>Ilumina methylation array data can be mapped to the genome with or without additional preprocessing methods. Incomplete annotation of genetic variations such as single nucleotide polymorphism (SNP) may affect DNA measurements and disrupt downstream analysis of results.  <span class="citation"><a href="#Hansen">Aryee <i>et al.</i> 2014</a></span> It is highly recommended to remove the probes that contain either an SNP at the methylated loci interrogation or at the single nucleotide extension. In this tutorial we will remove probes affected by genetic variation by selecting <strong>(Optional) Preprocessing Method</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.
<a href="../../images/2idat_preprocessing.png" rel="noopener noreferrer"><img src="../../images/2idat_preprocessing.png" alt="idat_preprocessing. " width="802" height="212" loading="lazy" /></a></p>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-normalisation-of-the-data"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Normalisation of the data</div>
  <p>If your files require normalisation, you might prefer to use one of the other preprocessing tools provided in Infinium Human Methylation BeadChip tool i.e. Preprocess Funnorm or Preprocess Quantile look for recommendation at <span class="citation"><a href="#Hansen">Aryee <i>et al.</i> 2014</a></span>.</p>

</blockquote>

<h1 id="differentially-methylated-regions-and-positions-analysis">Differentially methylated regions and positions analysis</h1>

<p>The main goal of the <strong>Infinium Human Methylation BeadChip</strong> analysis is to simplify the way differentially methylated loci sites are detected. The <strong>Infinium Human Methylation BeadChip</strong> pipeline contains differentially methylated positions (DMPs) detection with respect to a phenotype covariate, and more complex solutions for finding differentially methylated regions (DMRs). Genomic regions that are differentially methylated between two conditions can be tracked using a bumphunting algorithm. The algorithm first implements a t-statistic at each methylated loci location, with optional smoothing, then groups probes into clusters with a maximum location gap and a cutoff size to refer the lowest possible value of genomic profile hunted by our tool.</p>
<blockquote class="comment">
  <div class="box-title comment-title" id="comment-phenotype-table"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Phenotype table</div>
  <p>Phenotype table can be in different sizes with different arguments, however the second column is required to contain phenotype covariate information for each sample.</p>
</blockquote>

<p>However, for the purpose of this tutorial we would like you to upload phenotype table from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> repository.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-phenotypetable-txt-from-zenodo-https-zenodo-org-record-1251211-wwreq1mvz-y-or-data-library"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import `phenotypeTable.txt` from [Zenodo](https://zenodo.org/record/1251211#.WwREQ1Mvz-Y) or data library:</div>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>   https://zenodo.org/record/1251211/files/phenotypeTable.txt
</code></pre></div>  </div>
  <p><a href="../../images/3parameters.png" rel="noopener noreferrer"><img src="../../images/3parameters.png" alt="pheno table. " width="804" height="349" loading="lazy" /></a>
Set the following parameters:</p>
  <ul>
    <li><em>“maxGap Size”</em>:<code class="language-plaintext highlighter-rouge">250</code>
   We will use the default gap of 250 base pairs (bps), i.e. any two points more than 250 bps away are put in a new cluster.</li>
    <li><em>“Cutoff Size”</em>:<code class="language-plaintext highlighter-rouge">0.1</code>
   In order to find segments that are positive, near zero, and negative. We need a cutoff which is one number in which case “near zero” default 0.1</li>
  </ul>

  <p><a href="../../images/4parameters.png" rel="noopener noreferrer"><img src="../../images/4parameters.png" alt="params. " width="800" height="310" loading="lazy" /></a></p>
  <ul>
    <li><em>“Number of Resamples”</em>:<code class="language-plaintext highlighter-rouge">0</code>
Default value 0 for permutation method apply selection of randomized cases with replacement from the original data while using ‘bootstrap’ method.</li>
    <li><em>“nullMethod”</em>:<code class="language-plaintext highlighter-rouge">permutation</code>
Method used to generate null candidate regions, must be one of ‘bootstrap’ or
‘permutation’ (defaults to ‘permutation’).</li>
    <li><em>“Phenotype Type”</em>:<code class="language-plaintext highlighter-rouge">categorical</code>
Identify regions where methylation is associated with a continuous or categorical phenotype.</li>
  </ul>

  <p>Search for <code class="language-plaintext highlighter-rouge">UCSC Main</code> in the tool search bar (top left)
<a href="../../images/5parameters.png" rel="noopener noreferrer"><img src="../../images/5parameters.png" alt="params. " width="801" height="356" loading="lazy" /></a></p>
  <ul>
    <li><em>“qCutoff Size”</em>:<code class="language-plaintext highlighter-rouge">0.5</code>
Diffrentialy methylated positions with an FDR q-value greater than this value will not be returned.</li>
    <li><em>“Variance Shrinkage”</em>:<code class="language-plaintext highlighter-rouge"> TRUE</code>
Default TRUE as it is recommended when sample sizes are small &lt;10</li>
    <li><em>“Genome Table”</em>: <code class="language-plaintext highlighter-rouge">wgEncodeHaibMethyl450 ... </code>
Click on <code class="language-plaintext highlighter-rouge">UCSC Main</code> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. You will be taken to the <strong>UCSC table browser</strong>
<a href="../../images/5UCSC.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/5UCSC.png" alt="UCSC. " width="939" height="414" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> UCSC&lt;/figcaption&gt;</figure></a>
Set the following options:
    - <em>“clade”</em>: <code class="language-plaintext highlighter-rouge">Mammal</code>
    - <em>“genome”</em>: <code class="language-plaintext highlighter-rouge">Human</code>
    - <em>“assembly”</em>: <code class="language-plaintext highlighter-rouge">Feb. 2009 (GRCh37/hg19)</code>
    - <em>“group”</em>: <code class="language-plaintext highlighter-rouge">Regulation</code>
    - <em>“track”</em>: <code class="language-plaintext highlighter-rouge">HAIB Methyl450</code>
    - <em>“table”</em>: <code class="language-plaintext highlighter-rouge">GM12878 (wgEncodeHaibMethyl450Gm12878SitesRep1)</code>
    - <em>“region”</em>: <code class="language-plaintext highlighter-rouge">genome</code>
    - <em>“output format”</em>: <code class="language-plaintext highlighter-rouge">GTF - gene transfer (limited)</code>
    - <em>“Send output to”</em>: <code class="language-plaintext highlighter-rouge">Galaxy</code> (only)
Click on the <strong>get output</strong> button at the bottom of the screen
On the next page, click on the <strong>Send Query to Galaxy</strong> button
Wait for the upload to finish
We will now map the imported datasets against phenotype covariate and reference genome obtained from UCSC.
Click on the <strong>Differentially_Methylated_Positions.bed</strong> output in your history to expand it.
Set the database build of your dataset to <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code>(if it is not set automatically)</li>
  </ul>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing database/build (dbkey)<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">hg19</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
  <p><!--END_SNIPPET--></p>

  <p>Click on <code class="language-plaintext highlighter-rouge">display at UCSC</code> towards the bottom of the history item.
This will launch UCSC Genome Browser with your Custom Track</p>
</blockquote>

<p><a href="../../images/ucsc.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/ucsc.png" alt="Display at UCSC. " width="1440" height="803" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> UCSC genome track showing differentialy methylated regions located on chromosome 6&lt;/figcaption&gt;</figure></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>How do we define phenotype covariate?</p>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>Phenotype covariate is the set of observable characteristics of an individual resulting from the gene-environment interactions</p>
  </blockquote>
</blockquote>

<h1 id="annotation-and-visualization">Annotation and visualization</h1>

<p>In addition to downstream analysis users can annotate the differentially methylated loci at the promoter regions of genes with gene function descriptions, and relationships between these concepts.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-annotate-differentially-methylated-position"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Annotate Differentially Methylated Position</div>
  <ol>
    <li>Run <strong>chipeakanno annopeaks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>on the output of minfi_dmp with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Differentialy methylated data”</em>: output of <strong>minfi dmp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“bindingType”</em>: <code class="language-plaintext highlighter-rouge">StartSite</code></li>
        <li><em>“bindingRegionStart”</em>:<code class="language-plaintext highlighter-rouge">-5000</code></li>
        <li><em>“bindingRegionEnd”</em>:<code class="language-plaintext highlighter-rouge">3000</code></li>
        <li>
          <p><em>“Additional Column of Score”</em>:<code class="language-plaintext highlighter-rouge">8</code></p>

          <p>Position of column of score optional value if it is required</p>
        </li>
      </ul>
    </li>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to cut “gene_name” column from table of annotated peaks and then get a list of genes
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c16</code>hhh</li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: output of <strong>chipeakanno annopeaks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of <code class="language-plaintext highlighter-rouge">Gene List</code> with the following parameters
      <ul>
        <li><em>“Remove first”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from”</em>: output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>Run  <strong>clusterProfiler bitr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to convert the list of genes to list of entrez ID
      <ul>
        <li><em>“Input Type Gene ID”</em>: <code class="language-plaintext highlighter-rouge">SYMBOL</code></li>
        <li><em>“Output Type Gene ID”</em>: <code class="language-plaintext highlighter-rouge">ENTREZID</code></li>
      </ul>
    </li>
    <li>Use the output of the  clusterProfiler bitr <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to run a GO Enrichment Analysis using <strong>clusterProfiler go</strong></li>
  </ol>
</blockquote>

<p><a href="../../images/funcann.jpg" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/funcann.jpg" alt="Functional annotations. " width="1398" height="1398" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Results of GO enrichments analysis for DMPs&lt;/figcaption&gt;</figure></a></p>

<table>
  <thead>
    <tr>
      <th>ID</th>
      <th>Description</th>
      <th>pvalue</th>
      <th>qvalue</th>
      <th>geneID</th>
      <th>Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GO:0048732</td>
      <td>gland development</td>
      <td>1.38E-58</td>
      <td>4.23E-55</td>
      <td>PTGS2 / KCNC1 / FZD1 /SLC22A18 /SLC22A3 (…)</td>
      <td>372</td>
    </tr>
    <tr>
      <td>GO:1901652</td>
      <td>response to peptide</td>
      <td>3.99E-57</td>
      <td>8.13E-54</td>
      <td>SULF1/ LAMA5/ MED1 /CFLAR/ MSX2 (…)</td>
      <td>359</td>
    </tr>
    <tr>
      <td>GO:0048545</td>
      <td>response to steroid hormone</td>
      <td>1.38EE-54</td>
      <td>2.11E-51</td>
      <td>HDAC9/ RAB10/ CFLAR/ WDTC1 (…)</td>
      <td>394</td>
    </tr>
  </tbody>
</table>

<h1 id="conclusion">Conclusion</h1>

<p>Epigenetic aberrations which involve DNA modifications give researchers an interest in identifying novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and time consuming computational analysis. Here, the EWAS suite becomes a solution for this inconvenience and provides a simplified downstream analysis available as a ready to run pipline in supplementary materials.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Infinium Human Methylation BeadChip is an array based technology to generate DNA methylation profiling at individual CpG loci in the human genome based on Illumina’s bead technology.</p>
</li>
                        
                        <li><p>Time and cost efficiency followed by high sample output, and overall quantitative accuracy and reproducibility made Infinium Human Methylation BeadChip one of the most widely used arrays on the market.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/epigenetics/faqs/">FAQ page for the Epigenetics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Illumina"><b>Illumina Infinium Methylation Assay Overview</b>. Accessed: 2020-02-22.</li>
<li id="Hansen">Aryee, M. J., A. E. Jaffe, H. Corrada-Bravo, C. Ladd-Acosta, A. P. Feinberg <i>et al.</i>, 2014 <b>Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays</b>. Bioinformatics 30: 1363–1369. <a href="https://doi.org/10.1093/bioinformatics/btu049">10.1093/bioinformatics/btu049</a></li>
<li id="Hugo2015">Hugo, W., H. Shi, L. Sun, M. Piva, C. Song <i>et al.</i>, 2015 <b>Non-genomic and Immune Evolution of Melanoma Acquiring MAPKi Resistance</b>. Cell 162: 1271–1285. <a href="https://doi.org/10.1016/j.cell.2015.07.061">10.1016/j.cell.2015.07.061</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Infinium Human Methylation BeadChip (Epigenetics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Katarzyna Kamieniecka, Krzysztof Poterlowicz,  <b>Infinium Human Methylation BeadChip (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{epigenetics-ewas-suite,
author = "Katarzyna Kamieniecka and Krzysztof Poterlowicz",
	title = "Infinium Human Methylation BeadChip (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/epigenetics/tutorials/ewas-suite/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00139">https://gxy.io/GTN:T00139</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/ewas-suite/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703550391'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>