<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>ATAC-Seq data analysis</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: ATAC-Seq data analysis" />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Epigenetics
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/epigenetics/tutorials/atac-seq/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/epigenetics/faqs/" title="Check our FAQs for the Epigenetics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/atac-seq/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:00:22 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "ATAC-Seq data analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
  },
  "courseCode": "epigenetics / atac-seq / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'ATAC-Seq data analysis' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html",
  "timeRequired": "PT3H",
  "description": "The questions this  addresses are:\n - Which DNA regions are accessible in the human lymphoblastoid cell line GM12878?\n - How to analyse and visualise ATAC-Seq data?\n\n\\nThe objectives are:\n - Apply appropriate analysis and quality control steps for ATAC-Seq\n - Generate a heatmap of transcription start site accessibility\n - Visualise peaks for specific regions\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Lucille Delisle"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Florian Heyl"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Lucille Delisle"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    },
    {
      "@type": "Person",
      "name": "Florian Heyl"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
</script>

<section class="tutorial topic-epigenetics">
    <h1 data-toc-skip>ATAC-Seq data analysis</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/lldelisle/" class="contributor-badge contributor-lldelisle"><img src="https://avatars.githubusercontent.com/lldelisle?s=27" alt="Avatar">Lucille Delisle</a>


<a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="https://avatars.githubusercontent.com/mblue9?s=27" alt="Avatar">Maria Doyle</a>


<a href="/training-material/hall-of-fame/heylf/" class="contributor-badge contributor-heylf"><img src="https://avatars.githubusercontent.com/heylf?s=27" alt="Avatar">Florian Heyl</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which DNA regions are accessible in the human lymphoblastoid cell line GM12878?</p>
</li>
        
        <li><p>How to analyse and visualise ATAC-Seq data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Apply appropriate analysis and quality control steps for ATAC-Seq</p>
</li>
        
        <li><p>Generate a heatmap of transcription start site accessibility</p>
</li>
        
        <li><p>Visualise peaks for specific regions</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/epigenetics/tutorials/atac-seq/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3862793">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/atac-seq/workflows/" title="Workflows" alt="ATAC-Seq data analysis workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/epigenetics/atac-seq/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
          
          
          

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/epigenetics/atac-seq/slides" title="View GTN Video Library for these Slides">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Lecture (February 2021)
            </a>
        </li>
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://hicexplorer.usegalaxy.eu" title="HiCExplorer">
            HiCExplorer
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://plants.usegalaxy.eu/" title="Plant Anaylsis Workbench">
            Plant Anaylsis Workbench
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 22, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In many eukaryotic organisms, such as humans, the genome is tightly packed and organized with the help of nucleosomes (chromatin). A nucleosome is a complex formed by eight histone proteins that is wrapped with ~147bp of <span class="notranslate">DNA</span>. When the <span class="notranslate">DNA</span> is being actively transcribed into <span class="notranslate">RNA</span>, the <span class="notranslate">DNA</span> will be opened and loosened from the nucleosome complex. Many factors, such as the chromatin structure, the position of the nucleosomes, and histone modifications, play an important role in the organization and accessibility of the <span class="notranslate">DNA</span>. Consequently, these factors are also important for the activation and inactivation of genes. <strong>A</strong>ssay for <strong>T</strong>ransposase-<strong>A</strong>ccessible <strong>C</strong>hromatin using <strong>seq</strong>uencing (<a href="https://en.wikipedia.org/wiki/ATAC-seq">ATAC-Seq</a>) is a method to investigate the accessibility of chromatin and thus a method to determine regulatory mechanisms of gene <span class="notranslate">expression</span>. The method can help identify promoter regions and potential enhancers and silencers. A promoter is the <span class="notranslate">DNA</span> region close to the transcription start site (TSS). It contains binding sites for transcription factors that will recruit the <span class="notranslate">RNA</span> polymerase. An enhancer is a <span class="notranslate">DNA</span> region that can be located up to 1 Mb <span class="notranslate">downstream</span> or <span class="notranslate">upstream</span> of the promoter. When transcription factors bind an enhancer and contact a promoter region, the transcription of the gene is increased. In contrast, a silencer decreases or inhibits the gene‚Äôs <span class="notranslate">expression</span>. ATAC-Seq has become popular for identifying accessible regions of the genome as it‚Äôs easier, faster and requires less cells than alte<span class="notranslate">rna</span>tive techniques, such as FAIRE-Seq and <span class="notranslate">DNa</span>se-Seq.</p>

<figure id="figure-1"><img src="../../images/atac-seq/atac-seq.jpeg" alt="ATAC-Seq. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Buenrostro et al. 2013 Nat Methods</figcaption></figure>

<p>With ATAC-Seq, to find accessible (open) chromatin regions, the genome is treated with a hyperactive derivative of the Tn5 transposase. A <a href="https://en.wikipedia.org/wiki/Transposase">transposase</a> can bind to a <a href="https://en.wikipedia.org/wiki/Transposable_element">transposable element</a>, which is a <span class="notranslate">DNA</span> sequence that can change its position (jump) within a genome (read the two links to get a deeper insight). During ATAC-Seq, the modified Tn5 inserts <span class="notranslate">DNA</span> sequences corresponding to truncated Nextera adapters into open regions of the genome and concurrently, the <span class="notranslate">DNA</span> is sheared by the transposase activity. The read library is then prepared for sequencing, including PCR amplification with full Nextera adapters and purification steps. Paired-end <span class="notranslate">reads</span> are recommended for ATAC-Seq for the reasons described <a href="https://informatics.fas.harvard.edu/atac-seq-guidelines.html">here</a>.</p>

<p>In this tutorial we will use data from the study of <span class="citation"><a href="#Buenrostro2013">Buenrostro <i>et al.</i> 2013</a></span>, the first paper on the ATAC-Seq method. The data is from a human cell line of purified CD4+ T cells, called GM12878. The original dataset had 2 x 200 million <span class="notranslate">reads</span> and would be too big to process in a training session, so we downsampled the original dataset to 200,000 randomly selected <span class="notranslate">reads</span>. We also added about 200,000 <span class="notranslate">reads</span> pairs that will map to chromosome 22 to have a good profile on this chromosome, similar to what you might get with a typical ATAC-Seq sample (2 x 20 million <span class="notranslate">reads</span> in original FASTQ). Furthermore, we want to compare the predicted open chromatin regions to the known binding sites of CTCF, a <span class="notranslate">DNA</span>-binding protein implicated in 3D structure: <a href="https://en.wikipedia.org/wiki/CTCF">CTCF</a>. CTCF is known to bind to thousands of sites in the genome and thus it can be used as a positive control for assessing if the ATAC-Seq experiment is good quality. Good ATAC-Seq data would have accessible regions both within and outside of TSS, for example, at some CTCF binding sites. For that reason, we will download binding sites of CTCF identified by ChIP in the same cell line from ENCODE (ENCSR000AKB, dataset ENCFF933NTR).</p>

<h3 class="no_toc" id="when-working-with-real-data">When working with real data</h3>
<p>When you use your own data we suggest you to use <a href="https://usegalaxy.eu/u/ldelisle/w/atac-seq-gtm-with-control">this <span class="notranslate">workflow</span></a> which includes the same steps but is compatible with replicates. If you do not have any control data you can import and edit this <span class="notranslate">workflow</span>, removing all steps with the controls. Controls for the ATAC-Seq procedure are not commonly performed, as discussed <a href="https://informatics.fas.harvard.edu/atac-seq-guidelines.html">here</a>, but could be ATAC-Seq of purified <span class="notranslate">DNA</span>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#preprocessing" id="markdown-toc-preprocessing">Preprocessing</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get Data</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality Control</a></li>
      <li><a href="#trimming-reads" id="markdown-toc-trimming-reads">Trimming <span class="notranslate">Reads</span></a></li>
    </ol>
  </li>
  <li><a href="#mapping" id="markdown-toc-mapping"><span class="notranslate">Mapping</span></a>    <ol>
      <li><a href="#mapping-reads-to-reference-genome" id="markdown-toc-mapping-reads-to-reference-genome"><span class="notranslate">Mapping</span> <span class="notranslate">Reads</span> to Reference Genome</a></li>
    </ol>
  </li>
  <li><a href="#filtering-mapped-reads" id="markdown-toc-filtering-mapped-reads">Filtering Mapped <span class="notranslate">Reads</span></a>    <ol>
      <li><a href="#filter-uninformative-reads" id="markdown-toc-filter-uninformative-reads">Filter Uninformative <span class="notranslate">Reads</span></a></li>
      <li><a href="#filter-duplicate-reads" id="markdown-toc-filter-duplicate-reads">Filter Duplicate <span class="notranslate">Reads</span></a></li>
      <li><a href="#check-insert-sizes" id="markdown-toc-check-insert-sizes">Check Insert Sizes</a></li>
    </ol>
  </li>
  <li><a href="#peak-calling" id="markdown-toc-peak-calling">Peak calling</a>    <ol>
      <li><a href="#call-peaks" id="markdown-toc-call-peaks">Call Peaks</a></li>
    </ol>
  </li>
  <li><a href="#visualisation-of-coverage" id="markdown-toc-visualisation-of-coverage">Visualisation of Coverage</a>    <ol>
      <li><a href="#prepare-the-datasets" id="markdown-toc-prepare-the-datasets">Prepare the Datasets</a></li>
      <li><a href="#create-heatmap-of-coverage-at-tss-with-deeptools" id="markdown-toc-create-heatmap-of-coverage-at-tss-with-deeptools">Create heatmap of coverage at TSS with deepTools</a></li>
      <li><a href="#visualise-regions-with-pygenometracks" id="markdown-toc-visualise-regions-with-pygenometracks">Visualise Regions with pyGenomeTracks</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<!--SNIPPET-->
<blockquote class="comment">  <h3 data-toc-skip="" id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>

<h1 id="preprocessing">Preprocessing</h1>

<h2 id="get-data">Get Data</h2>

<p>We first need to download the sequenced <span class="notranslate">reads</span> (FASTQs) as well as other annotation files. Then, to increase the number of <span class="notranslate">reads</span> that will map to the reference genome (here human genome version 38, GRCh38/hg38), we need to preprocess the <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>
      <p>Import the files from <a href="https://doi.org/10.5281/zenodo.3862792">Zenodo</a> or from
the shared data library (<code class="language-plaintext highlighter-rouge">GTN - Material</code> -&gt; <code class="language-plaintext highlighter-rouge">epigenetics</code>
 -&gt; <code class="language-plaintext highlighter-rouge">ATAC-Seq data analysis</code>):</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3862793/files/ENCFF933NTR.<span class="notranslate">bed</span>.gz
https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R1.fastq.gz
https://zenodo.org/record/3862793/files/SRR891268_chr22_enriched_R2.fastq.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>
      <p>Add a tag called <code class="language-plaintext highlighter-rouge">#SRR891268_R1</code> to the R1 file and a tag called <code class="language-plaintext highlighter-rouge">#SRR891268_R2</code> to the R2 file.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
    <li>
      <p>Check that the datatype of the 2 FASTQ files is <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> and the peak file (ENCFF933NTR.<span class="notranslate">bed</span>.gz) is <code class="language-plaintext highlighter-rouge">encodepeak</code>. If they are not then change the datatype as described below.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">datatypes</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-fastq-format"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> FASTQ format</h3>
  <p>If you are not familiar with FASTQ format, see the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control tutorial</a></p>
</blockquote>
<blockquote class="comment">

  <h3 id="comment-bed--encode-narrowpeak-format"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> <span class="notranslate">BED</span> / encode narrowPeak format</h3>
  <p>If you are not familiar with <span class="notranslate">BED</span> format or encode narrowPeak format, see the <a href="https://genome.ucsc.edu/FAQ/FAQformat.html"><span class="notranslate">BED</span> Format</a></p>
</blockquote>

<p>We will visualise regions later in the analysis and obtain the gene information now. We will get information for chromosome 22 genes (names of transcripts and genomic positions) using the UCSC tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-obtain-annotation-for-hg38-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Obtain Annotation for hg38 genes</h3>

  <ol>
    <li><span class="tool" data-tool="ucsc_table_direct1" title="Tested with ucsc_table_direct1"><strong>UCSC Main table browser</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: ucsc_table_direct1</span></span>  with the following parameters:
      <ul>
        <li><em>‚Äúclade‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mammal</code></li>
        <li><em>‚Äúgenome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human</code></li>
        <li><em>‚Äúassembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dec. 2013 (GRCh38/hg38)</code></li>
        <li><em>‚Äúgroup‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genes and Gene Prediction</code></li>
        <li><em>‚Äútrack‚Äù</em>: <code class="language-plaintext highlighter-rouge">All GENCODE V37</code></li>
        <li><em>‚Äútable‚Äù</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
        <li><em>‚Äúregion‚Äù</em>: <code class="language-plaintext highlighter-rouge">position</code> <code class="language-plaintext highlighter-rouge">chr22</code></li>
        <li><em>‚Äúoutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">all fields from selected table</code></li>
        <li><em>‚ÄúSend output to‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Galaxy</span></code></li>
      </ul>
    </li>
    <li>Click <strong>get output</strong></li>
    <li>
      <p>Click <strong>Send query to <span class="notranslate">Galaxy</span></strong></p>

      <p>This table contains all the information but is not in a <span class="notranslate">BED</span> format. To transform it into <span class="notranslate">BED</span> format we will cut out the required columns and rearrange:</p>
    </li>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c3,c5,c6,c13,c12,c4</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFrom‚Äù</em>: <code class="language-plaintext highlighter-rouge">UCSC Main on Human: wgEncodeGencodeBasicV37 (chr22:1-50,818,468)</code></li>
      </ul>
    </li>
    <li>
      <p>Check the contents of your file, is this as you expect it to be?</p>

      <blockquote class="question">
        <h3 id="question-question-expected-output"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question: Expected output</h3>

        <p>Our goal here was to convert the data to <span class="notranslate">BED</span> format.</p>

        <ol>
          <li>Which columns do you expect in your file? (Tip: read about <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1"><span class="notranslate">BED</span> format</a>)</li>
          <li>Does your file look like a valid <span class="notranslate">BED</span> format?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>We expect at least 3 columns, <code class="language-plaintext highlighter-rouge">chromosome - start - end</code>, and possibly more optional columns</li>
            <li>Your file should look something like this:
              <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>chr22	10736170	10736283	U2	0	-
chr22	11066417	11068174	CU104787.1	0	+
chr22	11249808	11249959	5_8S_r<span class="notranslate">RNA</span>	0	-
chr22	15273854	15273961	U6	0	+
[..]
</code></pre></div>              </div>
            </li>
          </ol>

          <ul>
            <li><strong>Troubleshooting:</strong> Is your second column the <code class="language-plaintext highlighter-rouge">Strand</code> column?
              <ul>
                <li>Make sure you used the correct <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (the one that matches the tool name mentioned in the previous step <em>exactly</em>)</li>
                <li>There is another tool with <code class="language-plaintext highlighter-rouge">(cut)</code> behind the title, we do NOT want to use this tool in this step.</li>
              </ul>
            </li>
            <li><strong>Tip:</strong> Always check your output files to make sure they match your expectations!</li>
          </ul>

        </blockquote>

      </blockquote>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the dataset as <code class="language-plaintext highlighter-rouge">chr22 genes</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>
      <p><strong>Change</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> its datatype to <span class="notranslate">BED</span></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span></code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the file. It should have now column names and they should match the content.</li>
  </ol>

</blockquote>

<blockquote class="comment">

  <h3 id="comment-gene-file"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Gene file</h3>
  <p>The chr22 genes <span class="notranslate">BED</span> we produced only contains the start, the end, the name, and the strand of each transcript. It does not contain exon information.
To be able to have the exon information, you could use a GTF file which can be downloaded from the <a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_37/gencode.v37.annotation.gtf.gz">gencode website</a> but this file would include the information for the whole genome and would slow the analysis.</p>
</blockquote>

<h2 id="quality-control">Quality Control</h2>

<p>The first step is to check the quality of the <span class="notranslate">reads</span> and the presence of the Nextera adapters. When we perform ATAC-Seq, we can get <span class="notranslate">DNA</span> fragments of about 40 bp if two adjacent Tn5 transposases cut the <span class="notranslate">DNA</span> <span class="citation"><a href="#Adey2010">Adey <i>et al.</i> 2010</a></span>. This can be smaller than the sequencing length so we expect to have Nextera adapters at the end of those <span class="notranslate">reads</span>. We can assess the <span class="notranslate">reads</span> with <strong>FastQC</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúShort read data from your current history‚Äù</em>: Choose here either only the <code class="language-plaintext highlighter-rouge">SRR891268_R1</code> file with <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> or use <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong> to choose both <code class="language-plaintext highlighter-rouge">SRR891268_R1</code> and <code class="language-plaintext highlighter-rouge">SRR891268_R2</code>.</li>
      </ul>
    </li>
    <li>Inspect the web page output of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <code class="language-plaintext highlighter-rouge">SRR891268_R1</code> sample. Check what adapters are found at the end of the <span class="notranslate">reads</span>.</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>How many <span class="notranslate">reads</span> are in the FASTQ?</li>
      <li>Which sections have a warning?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <ol>
        <li>There are 285247 <span class="notranslate">reads</span>.</li>
        <li>
          <p>The 3 steps below have warnings:</p>

          <ol>
            <li>
              <p><strong>Per base sequence content</strong></p>

              <p>It is well known that the Tn5 has a strong sequence bias at the insertion site. You can read more about it in <span class="citation"><a href="#Green2012">Green <i>et al.</i> 2012</a></span>.</p>
            </li>
            <li>
              <p><strong>Sequence Duplication Levels</strong></p>

              <p>The read library quite often has PCR duplicates that are introduced
simply by the PCR itself. We will remove these duplicates later on.</p>
            </li>
            <li>
              <p><strong>Overrepresented sequences</strong></p>

              <p>One sequence is over represented: 
you have 306 <span class="notranslate">reads</span> which are exactly the sequence of the Nextera adapter.
They correspond to adapters amplified head-to-head.
306 is really low (only 0.1% of <span class="notranslate">reads</span>).</p>
            </li>
          </ol>
        </li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-fastqc-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> FastQC Results</h3>
  <p>This is what you should expect from the <strong>Adapter Content</strong> section:</p>
  <figure id="figure-2"><img src="../../images/atac-seq/Screenshot_fastqcBeforecutadapt.png" alt="FastQC screenshot of the Adapter Content section. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> FastQC screenshot on the Adapter Content section</figcaption></figure>
</blockquote>

<p>The FastQC web page <strong>Adapter Content</strong> section shows the presence of Nextera Transposase Sequence in the <span class="notranslate">reads</span>. We will remove the adapters with Cutadapt.</p>

<h2 id="trimming-reads">Trimming <span class="notranslate">Reads</span></h2>

<p>To trim the adapters we provide the Nextera adapter sequences to <strong>Cutadapt</strong>. These adapters are shown in the image below.</p>

<figure id="figure-3"><img src="../../images/atac-seq/nexteraLibraryPicture.svg.png" alt="Nextera library with the sequence of adapters. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Nextera library with the sequence of adapters</figcaption></figure>

<p>The forward and reverse adapters are slightly different. We will also trim low quality bases at the ends of the <span class="notranslate">reads</span> (quality less than 20). We will only keep <span class="notranslate">reads</span> that are at least 20 bases long. We remove short <span class="notranslate">reads</span> (&lt; 20bp) as they are not useful, they will either be thrown out by the <span class="notranslate">mapping</span> or may interfere with our results at the end.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/1.16.5</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or Paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #1‚Äù</em>: select <code class="language-plaintext highlighter-rouge">SRR891268_R1</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #2‚Äù</em>: select <code class="language-plaintext highlighter-rouge">SRR891268_R2</code></li>
            <li>In <em>‚ÄúRead 1 Options‚Äù</em>:
              <ul>
                <li>In <em>‚Äú3‚Äô (End) Adapters‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert 3‚Äô (End) Adapters‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSource‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>‚ÄúEnter custom 3‚Äô adapter name (Optional if Multiple output is ‚ÄòNo‚Äô)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nextera R1</code></li>
                            <li><em>‚ÄúEnter custom 3‚Äô adapter sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCTCCGAGCCCACGAGAC</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>In <em>‚ÄúRead 2 Options‚Äù</em>:
              <ul>
                <li>In <em>‚Äú3‚Äô (End) Adapters‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert 3‚Äô (End) Adapters‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSource‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>‚ÄúEnter custom 3‚Äô adapter name (Optional)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nextera R2</code></li>
                            <li><em>‚ÄúEnter custom 3‚Äô adapter sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCTGACGCTGCCGACGA</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúFilter Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúMinimum length‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúRead Modification Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúQuality cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúOutput Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúReport‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the report and read the first lines.</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-cutadapt-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Cutadapt Results</h3>
  <p>You should get similar output to this from Cutadapt:</p>
  <figure id="figure-4"><img src="../../images/atac-seq/Screenshot_cutadaptSummary.png" alt="Summary of cutadapt. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Summary of cutadapt</figcaption></figure>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What percentage of <span class="notranslate">reads</span> contain adapters?</li>
    <li>What percentage of <span class="notranslate">reads</span> are still longer than 20bp after the trimming?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>~14%</li>
      <li>~99%</li>
    </ol>

  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-adapter-removal-with-fastqc"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check Adapter Removal with FastQC</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúShort read data from your current history‚Äù</em>: select the output of <strong>Cutadapt</strong> <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong> to choose both <code class="language-plaintext highlighter-rouge">Read 1 Output</code> and <code class="language-plaintext highlighter-rouge">Read 2 Output</code>.</li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the report and read the first lines.</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-fastqc-results-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> FastQC Results</h3>
  <p>Now, you should see under <strong>Overrepresented sequences</strong> that there is no more overrepresented sequences and under <strong>Adapter Content</strong> that the Nextera adapters are no longer present.</p>
  <figure id="figure-5"><img src="../../images/atac-seq/Screenshot_fastqcAftercutadapt.png" alt="FastQC screenshot on the adapter content section after cutadapt. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> FastQC screenshot on the adapter content section after cutadapt</figcaption></figure>
  <p>However, you may have noticed that you have a new section with warning: <strong>Sequence Length Distribution</strong>. This is expected as you trimmed part of the <span class="notranslate">reads</span>.</p>
</blockquote>

<h1 id="mapping"><span class="notranslate">Mapping</span></h1>

<h2 id="mapping-reads-to-reference-genome"><span class="notranslate">Mapping</span> <span class="notranslate">Reads</span> to Reference Genome</h2>

<p>Next we map the trimmed <span class="notranslate">reads</span> to the human reference genome. Here we will use <strong>Bowtie2</strong>. We will extend the maximum fragment length (distance between read pairs) from 500 to 1000 because we know some valid read pairs are from this fragment length. We will use the <code class="language-plaintext highlighter-rouge">--very-sensitive</code> parameter to have more chance to get the best match even if it takes a bit longer to run. We will run the <strong>end-to-end</strong> mode because we trimmed the adapters so we expect the whole read to map, no clipping of ends is needed. Regarding the genome to choose. The hg38 version of the human genome contains <a href="https://www.ncbi.nlm.nih.gov/grc/help/definitions/#ALTERNATE">alte<span class="notranslate">rna</span>te loci</a>. This means that some region of the genome are present both in the canonical chromosome and on its alte<span class="notranslate">rna</span>te loci. The <span class="notranslate">reads</span> that map to these regions would map twice. To be able to filter <span class="notranslate">reads</span> falling into repetitive regions but keep <span class="notranslate">reads</span> falling into regions present in alte<span class="notranslate">rna</span>te loci, we will map on the Canonical version of hg38 (only the chromosome with numbers, chrX, chrY, and chrM).</p>

<blockquote class="comment">
  <h3 id="comment-dovetailing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Dovetailing</h3>
  <p>We will allow dovetailing of read pairs with Bowtie2. This is because adapters are removed by Cutadapt only when at least 3 bases match the adapter sequence, so it is possible that after trimming a read can contain 1-2 bases of adapter and go beyond it‚Äôs mate start site. For example, if the first mate in the read pair is: <code class="language-plaintext highlighter-rouge">GCTATGAAGAATAGGGCGAAGGGGCCTGCGGCGTATTCGATGTTGAAGCT</code> and the second mate is <code class="language-plaintext highlighter-rouge">CTTCAACATCGAATACGCCGCAGGCCCCTTCGCCCTATTCTTCATAGCCT</code>, where both contain 2 bases of adapter sequence, they will not be trimmed by Cutadapt and will map this way:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&lt;--------------------Mate 1-----------------------
AGCTTCAACATCGAATACGCCGCAGGCCCCTTCGCCCTATTCTTCATAGC
  CTTCAACATCGAATACGCCGCAGGCCCCTTCGCCCTATTCTTCATAGCCT
  ----------------------Mate 2---------------------&gt;
</code></pre></div>  </div>
  <p>This is what we call dovetailing and we want to consider this pair as a valid concordant alignment.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping-reads-to-reference-genome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span> <span class="notranslate">reads</span> to reference genome</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.2+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.2+galaxy0"><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.4.2+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúIs this single or paired library‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #1‚Äù</em>: select the output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚ÄúRead 1 Output‚Äù</em></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #2‚Äù</em>: select the output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚ÄúRead 2 Output‚Äù</em></li>
            <li><em>‚ÄúDo you want to set paired-end options?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúSet the maximum fragment length for valid paired-end alignments‚Äù</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
                <li><em>‚ÄúAllow mate dovetailing‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúSelect reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg38 Canonical</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1: Default setting only</code>
          <ul>
            <li><em>‚ÄúDo you want to use presets?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Very sensitive end-to-end (--very-sensitive)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúDo you want to tweak SAM/BAM Options?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúSave the bowtie2 <span class="notranslate">mapping</span> statistics to the history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the <span class="notranslate">mapping</span> stats.</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-bowtie2-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Bowtie2 Results</h3>
  <p>You should get similar results to this from Bowtie2:</p>
  <figure id="figure-6"><img src="../../images/atac-seq/Screenshot_bowtie2MappingStats.png" alt="Mapping statistics of bowtie2. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> <span class="notranslate">Mapping</span> statistics of bowtie2</figcaption></figure>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What percentage of read pairs mapped concordantly?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>54.8+42.87=97.67%</p>

  </blockquote>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment-on-the-number-of-uniquely-mapped"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment on the number of uniquely mapped.</h3>

  <p>You might be surprised by the number of uniquely mapped compared to the number of multi-mapped <span class="notranslate">reads</span> (<span class="notranslate">reads</span> <span class="notranslate">mapping</span> to more than one location in the genome).
One of the reasons is that we have used the parameter <code class="language-plaintext highlighter-rouge">--very-sensitive</code>. Bowtie2 considers a read as multi-mapped even if the second hit has a much lower quality than the first one.
Another reason is that we have <span class="notranslate">reads</span> that map to the mitochondrial genome. The mitochondrial genome has a lot of regions with similar sequence.</p>

</blockquote>

<h1 id="filtering-mapped-reads">Filtering Mapped <span class="notranslate">Reads</span></h1>

<h2 id="filter-uninformative-reads">Filter Uninformative <span class="notranslate">Reads</span></h2>

<p>We apply some filters to the <span class="notranslate">reads</span> after the <span class="notranslate">mapping</span>. ATAC-Seq datasets can have a lot of <span class="notranslate">reads</span> that map to the mitchondrial genome because it is nucleosome-free and thus very accessible to Tn5 insertion. The mitchondrial genome is uninteresting for ATAC-Seq so we remove these <span class="notranslate">reads</span>. We also remove <span class="notranslate">reads</span> with low <span class="notranslate">mapping</span> quality and <span class="notranslate">reads</span> that are not properly paired.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-of-uninformative-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering of uninformative <span class="notranslate">reads</span></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1"><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM dataset(s) to filter‚Äù</em>: Select the output of  <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚Äúalignments‚Äù</em></li>
        <li>In <em>‚ÄúCondition‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Condition‚Äù</em>
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Filter‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code>
                          <ul>
                            <li><em>‚ÄúFilter on read <span class="notranslate">mapping</span> quality (phred scale)‚Äù</em>: <code class="language-plaintext highlighter-rouge">&gt;=30</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Filter‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">isProperPair</code>
                          <ul>
                            <li><em>‚ÄúSelect properly paired <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Filter‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference</code>
                          <ul>
                            <li><em>‚ÄúFilter on the reference name for the read‚Äù</em>: <code class="language-plaintext highlighter-rouge">!chrM</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúWould you like to set rules?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>Click on the input and the output BAM files of the filtering step. Check the size of the files.</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Based on the file size, what proportion of alignments was removed (approximately)?</li>
    <li>Which parameter should be modified if you are interested in repetitive regions?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>The original BAM file is 28.1 MB, the filtered one is 15.2 MB. Approximately half of the alignments were removed.</p>
      </li>
      <li>
        <p>You should modify the mapQuality criteria and decrease the threshold.</p>
      </li>
    </ol>

  </blockquote>

</blockquote>

<p>High numbers of mitochondrial <span class="notranslate">reads</span> can be a problem in ATAC-Seq. Some ATAC-Seq samples have been reported to be 80% mitochondrial <span class="notranslate">reads</span> and so wet-lab methods have been developed to deal with this issue <span class="citation"><a href="#Corces2017">Corces <i>et al.</i> 2017</a></span> and <span class="citation"><a href="#Litzenburger2017">Litzenburger <i>et al.</i> 2017</a></span>. It can be a useful QC to assess the number of mitochondrial <span class="notranslate">reads</span>.
However, it does not predict the quality of the rest of the data. It is just that sequencing <span class="notranslate">reads</span> have been wasted.</p>

<blockquote class="tip">
  <h3 id="tip-tip-getting-the-number-of-mitochondrial-reads"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Getting the number of mitochondrial <span class="notranslate">reads</span></h3>

  <p>To get the number of <span class="notranslate">reads</span> that mapped to the mitochondrial genome (chrM) you can run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.3" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.3"><strong>Samtools idxstats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.3</span></span>  on the output of  <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚Äúalignments‚Äù</em>.
The columns of the output are: chromosome name, chromosome length, number of <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the chromosome, number of unaligned mate whose mate is <span class="notranslate">mapping</span> to the chromosome.
The first 2 lines of the result would be (after using <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1"><strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1</span></span> ):</p>

  <figure id="figure-7"><img src="../../images/atac-seq/Screenshot_samtoolsIdxStatsChrM.png" alt="Samtools idxstats result. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Samtools idxstats result</figcaption></figure>

  <p>There are 220 000 <span class="notranslate">reads</span> which map to chrM and 165 000 which map to chr22.</p>
</blockquote>

<h2 id="filter-duplicate-reads">Filter Duplicate <span class="notranslate">Reads</span></h2>

<p>Because of the PCR amplification, there might be read duplicates (different <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to exactly the same genomic region) from overamplification of some regions. As the Tn5 insertion is random within an accessible region, we do not expect to see fragments with the same coordinates. We consider such fragments to be PCR duplicates. We will remove them with <strong>Picard MarkDuplicates</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove duplicates</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2"><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect SAM/BAM dataset or dataset collection‚Äù</em>: Select the output of  <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚ÄúBAM‚Äù</em></li>
        <li><em>‚ÄúIf true do not write duplicates to the output file instead of writing them with appropriate flags set‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-default-of--markduplicates-tool"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Default of  <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></h3>

        <p>By default, the tool will only ‚ÄúMark‚Äù the duplicates. This means that it will change the Flag of the duplicated <span class="notranslate">reads</span> to enable them to be filtered afterwards. We use the parameter <em>‚ÄúIf true do not write duplicates to the output file instead of writing them with appropriate flags set‚Äù</em> to directly remove the duplicates.</p>
      </blockquote>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the MarkDuplicate metrics.</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-markduplicates-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> MarkDuplicates Results</h3>
  <p>You should get similar output to this from MarkDuplicates:</p>
  <figure id="figure-8"><img src="../../images/atac-seq/Screenshot_picardRemoveDup.png" alt="Metrics of MarkDuplicates. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Metrics of MarkDuplicates</figcaption></figure>
</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-formatting-the-markduplicate-metrics-for-readability"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Formatting the MarkDuplicate metrics for readability</h3>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect lines from‚Äù</em>: Select the output of  <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚Äúthat</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>‚Äúthe pattern</em>: <code class="language-plaintext highlighter-rouge">(Library|LIBRARY)</code></li>
      </ul>
    </li>
    <li>Check that the datatype is tabular. If not, change it.
<!--SNIPPET-->&lt;blockquote class="tip"&gt;  &lt;h3 data-toc-skip id="-tip-changing-the-datatype"&gt;<i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype&lt;/h3&gt;  &lt;ul&gt;  &lt;li&gt;Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes&lt;/li&gt;  &lt;li&gt;In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top&lt;/li&gt;  &lt;li&gt;Select <code class="language-plaintext highlighter-rouge">tabular</code>&lt;/li&gt;  &lt;li&gt;Click the <strong>Save</strong> button&lt;/li&gt;&lt;/ul&gt;&lt;/blockquote&gt;</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0"><strong>Transpose rows/columns in a tabular file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0</span></span> :
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect lines from‚Äù</em>: Select the output of <strong>Select</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

  <figure id="figure-9"><img src="../../images/atac-seq/Screenshot_picardRemoveDupAfterTranspose.png" alt="Metrics of MarkDuplicates. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Metrics of MarkDuplicates</figcaption></figure>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many pairs were in the input?</li>
    <li>How many pairs are duplicates?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>135813</li>
      <li>3584</li>
    </ol>

  </blockquote>

</blockquote>

<p>Once again, if you have a high number of replicates it does not mean that your data are not good, it just means that you sequenced too much compared to the diversity of the library you generated. Consequently, libraries with a high portion of duplicates should not be resequenced as this would not increase the amount of data.</p>

<h2 id="check-insert-sizes">Check Insert Sizes</h2>

<p>We will check the insert sizes with <strong>Paired-end histogram</strong> of insert size frequency. The insert size is the distance between the R1 and R2 read pairs. This tells us the size of the <span class="notranslate">DNA</span> fragment the read pairs came from. The fragment length distribution of a sample gives a very good indication of the quality of the ATAC-Seq.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-distribution-of-fragment-sizes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the distribution of fragment sizes.</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1"><strong>Paired-end histogram</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pe_histogram/pe_histogram/1.0.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM file‚Äù</em>: Select the output of  <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <em>‚ÄúBAM output‚Äù</em></li>
        <li><em>‚ÄúLower bp limit (optional)‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li><em>‚ÄúUpper bp limit (optional)‚Äù</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the lower one of the 2 outputs (the png file).</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-collectinsertsizemetrics-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> CollectInsertSizeMetrics Results</h3>
  <p>This is what you get from CollectInsertSizeMetrics:</p>
  <figure id="figure-10"><img src="../../images/atac-seq/Screenshot_sizeDistribution.png" alt="Fragment size distribution. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> Fragment size distribution</figcaption></figure>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Could you guess what the peaks at approximately 50bp, 200bp, 400bp and 600bp correspond to?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The first peak (50bp) corresponds to where the Tn5 transposase inserted into nucleosome-free regions. The second peak (a bit less than 200bp) corresponds to where Tn5 inserted around a single nucleosome. The third one (around 400bp) is where Tn5 inserted around two adjacent nucleosomes and the fourth one (around 600bp) is where Tn5 inserted around three adjacent nucleosomes.</p>

  </blockquote>

</blockquote>

<p>This fragment size distribution is a good indication if your experiment worked or not.
In absence of chromatin (without nucleosome), this is the profile you would get:</p>

<figure id="figure-11"><img src="../../images/atac-seq/Screenshot_sizeDistribution_Naked.png" alt="Fragment size distribution of a purified DNA. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> Fragment size distribution of a purified <span class="notranslate">DNA</span></figcaption></figure>

<p>Here are examples of Fragment size distributions of ATAC-Seq which were very noisy:</p>

<figure id="figure-12"><img src="../../images/atac-seq/Screenshot_sizeDistribution_Failed.png" alt="Fragment size distribution of a failed ATAC-Seq. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> Fragment size distribution of a failed ATAC-Seq</figcaption></figure>

<figure id="figure-13"><img src="../../images/atac-seq/Screenshot_sizeDistribution_Failed2.png" alt="Fragment size distribution of another failed ATAC-Seq. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Fragment size distribution of another very noisy ATAC-Seq</figcaption></figure>

<p>A final example of a Fragment size distribution of a very good ATAC-Seq, even if we cannot see the third nucleosome ‚Äúpeak‚Äù.</p>
<figure id="figure-14"><img src="../../images/atac-seq/Screenshot_sizeDistribution_Good.png" alt="Fragment size distribution of a good ATAC-Seq. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> Fragment size distribution of a good ATAC-Seq</figcaption></figure>

<h1 id="peak-calling">Peak calling</h1>

<h2 id="call-peaks">Call Peaks</h2>

<p>We have now finished the data preprocessing. Next, in order to find regions corresponding to potential open chromatin regions, we want to identify regions where <span class="notranslate">reads</span> have piled up (peaks) greater than the background read coverage. The tools which are currently used are <a href="https://github.com/jsh58/Genrich">Genrich</a> and <a href="https://github.com/taoliu/MACS">MACS2</a>. MACS2 is more widely used. Genrich has a mode dedicated to ATAC-Seq but is still not published and the more <span class="notranslate">reads</span> you have, the less peaks you get (see the issue <a href="https://github.com/jsh58/Genrich/issues/33">here</a>). That‚Äôs why we will not use Genrich in this tutorial.</p>

<p>At this step, two approaches exists:</p>

<ul>
  <li>The first one is to select only paired whose fragment length is below 100bp corresponding to nucleosome-free regions and to use a peak calling like you would do for a ChIP-seq, joining signal between mates. The disadvantages of this approach is that you can only use it if you have paired-end data and you will miss small open regions where only one Tn5 bound.</li>
  <li>The second one chosen here is to use all <span class="notranslate">reads</span> to be more exhaustive. In this approach, it is very important to re-center the signal of each <span class="notranslate">reads</span> on the 5‚Äô extremity (read start site) as this is where Tn5 cuts. Indeed, you want your peaks around the nucleosomes and not directly on the nucleosome:</li>
</ul>
<figure id="figure-15"><img src="../../images/atac-seq/schemeWithLegend.jpg" alt="Scheme of ATAC-Seq reads relative to nucleosomes. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> Scheme of ATAC-Seq <span class="notranslate">reads</span> relative to nucleosomes</figcaption></figure>

<blockquote class="comment">
  <h3 id="comment-comment-on-tn5-insertion"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment on Tn5 insertion</h3>

  <p>When Tn5 cuts an accessible chromatin locus it inserts adapters separated by 9bp (<span class="citation"><a href="#Kia2017">Kia <i>et al.</i> 2017</a></span>):</p>
  <figure id="figure-16"><img src="../../images/atac-seq/NexteraLibraryConstruction.jpg" alt="Nextera Library Construction. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 16:</span> Nextera Library Construction</figcaption></figure>

  <p>This means in order to have the read start site reflecting the centre of where Tn5 bound, the <span class="notranslate">reads</span> on the positive strand should be shifted 4 bp to the right and <span class="notranslate">reads</span> on the negative strands should be shifted 5 bp to the left as in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3959825">Buenrostro et al. 2013</a>. <strong>Genrich</strong> can apply these shifts when ATAC-seq mode is selected. In most cases, we do not have 9bp resolution so we don‚Äôt take it into account but if you are interested in the footprint, this is important.</p>
</blockquote>

<p>If we only assess the coverage of the 5‚Äô extremity of the <span class="notranslate">reads</span>, the data would be too sparse and it would be impossible to call peaks. Thus, we will extend the start sites of the <span class="notranslate">reads</span> by 200bp (100bp in each direction) to assess coverage.</p>

<h3 id="using-macs2">Using MACS2</h3>

<p>We convert the BAM file to <span class="notranslate">BED</span> format because when we set the extension size in MACS2, it will only consider one read of the pair while here we would like to use the information from both.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-the-bam-to-bed"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Convert the BAM to <span class="notranslate">BED</span></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0"><strong><span class="notranslate">bed</span>tools BAM to <span class="notranslate">BED</span> converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/<span class="notranslate">bed</span>tools/<span class="notranslate">bed</span>tools_bamtobed/2.30.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConvert the following BAM file to <span class="notranslate">BED</span>‚Äù</em>: Select the output of <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We call peaks with MACS2. In order to get the coverage centered on the 5‚Äô extended 100bp each side we will use <code class="language-plaintext highlighter-rouge">--shift -100</code> and <code class="language-plaintext highlighter-rouge">--extend 200</code>:</p>
<figure id="figure-17"><img src="../../images/atac-seq/macs2Options.jpg" alt="MACS2 options to get 100bp each side. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 17:</span> MACS2 options to get 100bp each side</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-call-peaks-with-macs2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Call peaks with MACS2</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.1.1.20160309.6" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.1.1.20160309.6"><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.1.1.20160309.6</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúAre you pooling Treatment Files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> Select the output of <strong><span class="notranslate">bed</span>tools BAM to <span class="notranslate">BED</span></strong> converter <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúDo you have a Control File?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúFormat of Input Files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end <span class="notranslate">BED</span></code></li>
        <li><em>‚ÄúEffective genome size‚Äù</em>: <code class="language-plaintext highlighter-rouge">H. sapiens (2.7e9)</code></li>
        <li><em>‚ÄúBuild Model‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not build the shifting model (--nomodel)</code>
          <ul>
            <li><em>‚ÄúSet extension size‚Äù</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
            <li><em>‚ÄúSet shift size‚Äù</em>: <code class="language-plaintext highlighter-rouge">-100</code>. It needs to be - half the extension size to be centered on the 5‚Äô.</li>
          </ul>
        </li>
        <li><em>‚ÄúAdditional Outputs‚Äù</em>:
          <ul>
            <li>Check <code class="language-plaintext highlighter-rouge">Peaks as tabular file (compatible with MultiQC)</code></li>
            <li>Check <code class="language-plaintext highlighter-rouge">Peak summits</code></li>
            <li>Check <code class="language-plaintext highlighter-rouge">Scores in <span class="notranslate">bed</span>Graph files</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdvanced Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúComposite broad regions‚Äù</em>: <code class="language-plaintext highlighter-rouge">No broad regions</code>
              <ul>
                <li><em>‚ÄúUse a more sophisticated signal processing approach to find subpeak summits in each enriched peak region‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
            <li><em>‚ÄúHow many duplicate tags at the exact same location are allowed?‚Äù</em>: <code class="language-plaintext highlighter-rouge">all</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-why-keeping-all-duplicates-is-important"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Why keeping all duplicates is important</h3>

        <p>We previously removed duplicates using <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> using paired-end information. If two pairs had identical R1 but different R2, we knew it was not a PCR duplicate. Because we converted the BAM to <span class="notranslate">BED</span> we lost the pair information. If we keep the default (removing duplicates) one of the 2 identical R1 would be filtered out as duplicate.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="visualisation-of-coverage">Visualisation of Coverage</h1>

<h2 id="prepare-the-datasets">Prepare the Datasets</h2>

<h3 id="extract-ctcf-peaks-on-chr22-in-intergenic-regions">Extract CTCF peaks on chr22 in intergenic regions</h3>
<p>As our training dataset is focused on chromosome 22 we will only use the CTCF peaks from chr22. We expect to have ATAC-seq coverage at TSS but only good ATAC-seq have coverage on intergenic CTCF. Indeed, the CTCF protein is able to position nucleosomes and creates a region depleted of nucleosome of around 120bp <span class="citation"><a href="#fu_insulator_2008">Fu <i>et al.</i> 2008</a></span>. This is smaller than the 200bp nucleosome-free region around TSS and also probably not present in all cells. Thus it is more difficult to get enrichment. In order to get the list of intergenic CTCF peaks of chr22, we will first select the peaks on chr22 and then exclude the one which overlap with genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-ctcf-peaks-from-chr22-in-intergenic-regions"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select CTCF peaks from chr22 in intergenic regions:</h3>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter data on any column using simple <span class="notranslate">expression</span>s</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFilter‚Äù</em>: Select the first dataset: <code class="language-plaintext highlighter-rouge">ENCFF933NTR.<span class="notranslate">bed</span>.gz</code></li>
        <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1=='chr22'</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0"><strong><span class="notranslate">bed</span>tools Intersect intervals find overlapping intervals in various ways</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/<span class="notranslate">bed</span>tools/<span class="notranslate">bed</span>tools_intersectbed/2.30.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile A to intersect with B‚Äù</em>: Select the output of <strong>Filter</strong> data on any column using simple <span class="notranslate">expression</span>s <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúCombined or separate output files‚Äù</em>: <code class="language-plaintext highlighter-rouge">One output file per 'input B' file</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile B to intersect with A‚Äù</em>:  Select the dataset <code class="language-plaintext highlighter-rouge">chr22 genes</code></li>
          </ul>
        </li>
        <li><em>‚ÄúWhat should be written to the output file?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Write the original entry in A for each overlap (-wa)</code></li>
        <li><em>‚ÄúRequired overlap‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default: 1bp</code></li>
        <li><em>‚ÄúReport only those alignments that <strong>do not</strong> overlap with file(s) B‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Rename the datasets <code class="language-plaintext highlighter-rouge">intergenic CTCF peaks chr22</code>.</li>
  </ol>
</blockquote>

<h3 id="convert-bedgraph-from-macs2-to-bigwig">Convert <span class="notranslate">bed</span>graph from <strong>MACS2</strong> to bigwig</h3>
<p>The <span class="notranslate">bed</span>graph format is easily readable for human but it can be very large and visualising a specific region is quite slow. We will change it to bigwig format which is a binary format, so we can visualise any region of the genome very quickly.</p>

<blockquote class="tip">
  <h3 id="tip-tip-speed-up-the-bedgraph-to-bigwig-conversion"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Speed-up the <span class="notranslate">bed</span>graph to bigwig conversion</h3>

  <p>In this tutorial we focus on chr22, thus we could restrict our <span class="notranslate">bed</span>graph to chr22 before doing the conversion. This will both speed-up the conversion and decrease the amount of memory needed.</p>

  <p>To do so, we will use <span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter data on any column using simple <span class="notranslate">expression</span>s</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFilter‚Äù</em>: Select the output of <strong>MACS2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="notranslate">Bed</span>graph Treatment).</li>
    <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1=='chr22'</code></li>
  </ul>

  <p>This will decrease by half the size of the file.
In the next step, choose the output of <strong>Filter</strong> instead of the output of <strong>MACS2</strong>.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-bedgraphs-to-bigwig"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Convert <span class="notranslate">bed</span>graphs to bigWig.</h3>

  <ol>
    <li><span class="tool" data-tool="wig_to_bigWig" title="Tested with wig_to_bigWig"><strong>Wig/<span class="notranslate">Bed</span>Graph-to-bigWig</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: wig_to_bigWig</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConvert‚Äù</em>: Select the output of <strong>MACS2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="notranslate">Bed</span>graph Treatment).</li>
        <li><em>‚ÄúConverter settings to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default</code></li>
      </ul>
    </li>
    <li>Rename the datasets <code class="language-plaintext highlighter-rouge">MACS2 bigwig</code>.</li>
  </ol>
</blockquote>

<h2 id="create-heatmap-of-coverage-at-tss-with-deeptools">Create heatmap of coverage at TSS with deepTools</h2>

<p>You might be interested in checking the coverage on specific regions. For this, you can compute a heatmap. We will use the <strong>deepTools plotHeatmap</strong>. As an example, we will here make a heatmap centered on the transcription start sites (TSS) and another one centered on intergenic CTCF peaks. First, on the TSS:</p>

<h3 id="generate-computematrix">Generate computeMatrix</h3>

<p>The input of <strong>plotHeatmap</strong> is a matrix in a hdf5 format. To generate it we use the tool <strong>computeMatrix</strong> that will evaluate the coverage at each locus we are interested in.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-the-matrix"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate the matrix</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0"><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0</span></span>  with the following parameters:
      <ul>
        <li>In <em>‚ÄúSelect regions‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Select regions‚Äù</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRegions to plot‚Äù</em>: Select the dataset <code class="language-plaintext highlighter-rouge">chr22 genes</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúScore file‚Äù</em>: Select the output of <strong>Wig/<span class="notranslate">Bed</span>Graph-to-bigWig</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> that should be named <code class="language-plaintext highlighter-rouge">MACS2 bigwig</code>.</li>
          </ul>
        </li>
        <li><em>‚ÄúcomputeMatrix has two main output options‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference-point</code></li>
        <li><em>‚ÄúThe reference point for the plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">beginning of region (e.g. TSS)</code></li>
        <li><em>‚ÄúShow advanced output settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>‚ÄúConvert missing values to 0?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="plot-with-plotheatmap">Plot with <strong>plotHeatmap</strong></h3>

<p>We will now generate a heatmap. Each line will be a transcript. The coverage will be summarized with a color code from red (no coverage) to blue (maximum coverage). All TSS will be aligned in the middle of the figure and only the 2 kb around the TSS will be displayed. Another plot, on top of the heatmap, will show the mean signal at the TSS. There will be one heatmap per bigwig.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-the-heatmap"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate the heatmap</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1"><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMatrix file from the computeMatrix tool‚Äù</em>: Select the output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</li>
        <li><em>‚ÄúShow advanced output settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-plotheatmap-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> plotHeatmap Results</h3>
  <p>This is what you get from plotHeatmap:</p>
  <figure id="figure-18"><img src="../../images/atac-seq/plotHeatmapOutput.png" alt="plotHeatmap output. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 18:</span> plotHeatmap output</figcaption></figure>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Is the coverage symmetric?</li>
    <li>What is the mean value in genes?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>No, it is higher on the left which is expected as usually the promoter of active genes is accessible.</li>
      <li>Around 5.5.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Now we will repeat the procedure for CTCF peaks of chr22 in intergenic regions:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-the-matrix-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate the matrix</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0"><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.3.2.0.0</span></span>  with the following parameters:
      <ul>
        <li>In <em>‚ÄúSelect regions‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Select regions‚Äù</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRegions to plot‚Äù</em>: Select the dataset <code class="language-plaintext highlighter-rouge">intergenic CTCF peaks chr22</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúScore file‚Äù</em>: Select the output of <strong>Wig/<span class="notranslate">Bed</span>Graph-to-bigWig</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> that should be named <code class="language-plaintext highlighter-rouge">MACS2 bigwig</code>.</li>
          </ul>
        </li>
        <li><em>‚ÄúWould you like custom sample labels?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, use sample names in the history</code></li>
        <li><em>‚ÄúcomputeMatrix has two main output options‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference-point</code>
          <ul>
            <li><em>‚ÄúThe reference point for the plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">center of region</code></li>
          </ul>
        </li>
        <li><em>‚ÄúShow advanced output settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>‚ÄúConvert missing values to 0?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1"><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.3.2.0.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMatrix file from the computeMatrix tool‚Äù</em>:  Select the output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</li>
        <li><em>‚ÄúShow advanced output settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li>In <em>‚ÄúColormap to use for each sample‚Äù</em>:
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Colormap to use for each sample‚Äù</em>
                  <ul>
                    <li><em>‚ÄúColor map to use for the heatmap‚Äù</em>: <code class="language-plaintext highlighter-rouge">Blues</code> # Or what you want</li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúThe x-axis label‚Äù</em>: <code class="language-plaintext highlighter-rouge">distance from peak center (bp)</code></li>
            <li><em>‚ÄúThe y-axis label for the top panel‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTCF peaks</code></li>
            <li><em>‚ÄúReference point label‚Äù</em>: <code class="language-plaintext highlighter-rouge">peak center</code></li>
            <li><em>‚ÄúLabels for the regions plotted in the heatmap‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTCF_peaks</code></li>
            <li><em>‚ÄúDid you compute the matrix with more than one groups of regions?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, I used multiple groups of regions</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-plotheatmap-results-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> plotHeatmap Results</h3>
  <p>This is what you get from plotHeatmap, this is much more symetric:</p>
  <figure id="figure-19"><img src="../../images/atac-seq/plotHeatmapOutput_CTCF.png" alt="plotHeatmap output on CTCF. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 19:</span> plotHeatmap output on CTCF</figcaption></figure>
</blockquote>

<h2 id="visualise-regions-with-pygenometracks">Visualise Regions with pyGenomeTracks</h2>

<p>In order to visualise a specific region (e.g. the gene <em>RAC2</em>), we can either use a genome browser like <strong>IGV</strong> or <strong>UCSC browser</strong>, or use <strong>pyGenomeTracks</strong> to make publishable figures. We will use <strong>pyGenomeTracks</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.6" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.6"><strong>pyGenomeTracks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/pygenometracks/pygenomeTracks/3.6</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúRegion of the genome to limit the operation‚Äù</em>: <code class="language-plaintext highlighter-rouge">chr22:37,193,000-37,252,000</code></li>
        <li>In <em>‚ÄúInclude tracks in your plot‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Include tracks in your plot‚Äù</em>
              <ul>
                <li><em>‚ÄúChoose style of the track‚Äù</em>: <code class="language-plaintext highlighter-rouge">Bigwig track</code>
                  <ul>
                    <li><em>‚ÄúPlot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">Coverage from MACS2 (extended +/-100bp)</code></li>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTrack file(s) bigwig format‚Äù</em>: Select the output of <strong>Wig/<span class="notranslate">Bed</span>Graph-to-bigWig</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> called <code class="language-plaintext highlighter-rouge">MACS2 bigwig</code>.</li>
                    <li><em>‚ÄúColor of track‚Äù</em>: Select the color of your choice</li>
                    <li><em>‚ÄúMinimum value‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                    <li><em>‚Äúheight‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                    <li><em>‚ÄúShow visualization of data range‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Include tracks in your plot‚Äù</em>
              <ul>
                <li><em>‚ÄúChoose style of the track‚Äù</em>: <code class="language-plaintext highlighter-rouge">NarrowPeak track</code>
                  <ul>
                    <li><em>‚ÄúPlot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">Peaks from MACS2 (extended +/-100bp)</code></li>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTrack file(s) encodepeak or <span class="notranslate">bed</span> format‚Äù</em>: Select the output of <strong>MACS2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (narrow Peaks).</li>
                    <li><em>‚ÄúColor of track‚Äù</em>: Select the color of your choice</li>
                    <li><em>‚Äúdisplay to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">box: Draw a box</code></li>
                    <li><em>‚ÄúPlot labels (name, p-val, q-val)‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Include tracks in your plot‚Äù</em>
              <ul>
                <li><em>‚ÄúChoose style of the track‚Äù</em>: <code class="language-plaintext highlighter-rouge">Gene track / <span class="notranslate">Bed</span> track</code>
                  <ul>
                    <li><em>‚ÄúPlot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genes</code></li>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTrack file(s) <span class="notranslate">bed</span> or gtf format‚Äù</em>: <code class="language-plaintext highlighter-rouge">chr22 genes</code></li>
                    <li><em>‚ÄúColor of track‚Äù</em>: Select the color of your choice</li>
                    <li><em>‚Äúheight‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                    <li><em>‚ÄúPlot labels‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
                      <ul>
                        <li><em>‚ÄúPut all labels inside the plotted region‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                        <li><em>‚ÄúAllow to put labels in the right margin‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Include tracks in your plot‚Äù</em>
              <ul>
                <li><em>‚ÄúChoose style of the track‚Äù</em>: <code class="language-plaintext highlighter-rouge">NarrowPeak track</code>
                  <ul>
                    <li><em>‚ÄúPlot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTCF peaks</code></li>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTrack file(s) encodepeak or <span class="notranslate">bed</span> format‚Äù</em>: Select the first dataset: <code class="language-plaintext highlighter-rouge">ENCFF933NTR.<span class="notranslate">bed</span>.gz</code></li>
                    <li><em>‚ÄúColor of track‚Äù</em>: Select the color of your choice</li>
                    <li><em>‚Äúdisplay to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">box: Draw a box</code></li>
                    <li><em>‚ÄúPlot labels (name, p-val, q-val)‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Include tracks in your plot‚Äù</em>
              <ul>
                <li><em>‚ÄúChoose style of the track‚Äù</em>: <code class="language-plaintext highlighter-rouge">X-axis</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon of the output.</li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-pygenometracks-results"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> pyGenomeTracks Results</h3>
  <p>You should get similar to results to this from pyGenomeTracks:</p>
  <figure id="figure-20"><img src="../../images/atac-seq/pyGenomeTracksOutput.png" alt="pyGenomeTracks output. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 20:</span> pyGenomeTracks output</figcaption></figure>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>In the ATAC-Seq sample in this selected region we see four peaks detected by MACS2.</p>

  <ol>
    <li>How many TSS are accessible in the sample in the displayed region?</li>
    <li>How many CTCF binding loci are accessible?</li>
    <li>Can you spot peaks with no TSS and no CTCF peak?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>In total, we can see 3 TSS for 6 transcripts for 2 genes. The TSS of RAC2 corresponds to an ATAC-Seq peak whereas there is no significant coverage on both TSS of SSTR3.</p>
      </li>
      <li>
        <p>Only the first peak on the left overlaps with a CTCF binding site.</p>
      </li>
      <li>
        <p>Amongst the 4 peaks in this region, the 2 in the middle do not correspond to CTCF peaks or TSS.</p>
      </li>
    </ol>

  </blockquote>

</blockquote>

<p>As CTCF creates accessible regions, a region containing a peak with no corresponding CTCF peak or TSS could be a putative enhancer. In the pyGenomeTracks plot we see a region like this located in the intron of a gene and another one between genes. However, it is impossible to guess from the position which would be the gene controlled by this region. And of course, more analyses are needed to assess if it is a real enhancer, for example, histone ChIP-seq, 3D structure, transgenic assay, etc.</p>

<h1 id="conclusion">Conclusion</h1>

<p>In this training you have learned the general principles of ATAC-Seq data analysis. ATAC-Seq
is a method to investigate the chromatin accessibility and the genome is treated with
a transposase (enzyme) called Tn5. It marks open chromatin regions by cutting and
inserting adapters for sequencing. The training material gave you an insight into how to quality control the data. You should look for low quality bases, adapter contamination, correct insert size and PCR duplicates (duplication level). We showed you how to remove adapters and PCR duplicates, if <strong>FastQC</strong>, shows a warning in these areas. We mapped the <span class="notranslate">reads</span>
with <strong>Bowtie2</strong>, filtered our <span class="notranslate">reads</span> for properly paired, good quality and <span class="notranslate">reads</span> that do not
map to the mitochondrial genome. We found open chromatin regions with <strong>MACS2</strong>, a tool to find regions of genomic enrichment (peaks). We investigated the read coverage around TSS with the help of <strong>computeMatrix</strong> and <strong>plotHeatmap</strong>. Last but not least, we visualised the peaks and other informative tracks, such as CTCF binding regions and hg38 genes, with the help of <strong>pyGenomeTracks</strong>. At the end, we found open chromatin regions that did not overlap with CTCF sites or TSS, which could be potential putative enhancer regions detected by the ATAC-Seq experiment.</p>

<figure id="figure-21"><img src="../../images/atac-seq/ATACWF.svg" alt="ATAC workflow. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 21:</span> ATAC <span class="notranslate">workflow</span></figcaption></figure>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>ATAC-Seq can be used to identify accessible gene promoters and enhancers</p>
</li>
                        
                        <li><p>Several filters are applied to the reads, such as removing those mapped to mitochondria</p>
</li>
                        
                        <li><p>Fragment distribution can help determine whether an ATAC-Seq experiment has worked well</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/epigenetics/faqs/">FAQ page for the Epigenetics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="fu_insulator_2008">Fu, Y., M. Sinha, C. L. Peterson, and Z. Weng, 2008 <b>The insulator binding protein CTCF positions 20 nucleosomes around its binding sites across the human genome</b>. PLoS genetics 4: e1000138. <a href="https://doi.org/10.1371/journal.pgen.1000138">10.1371/journal.pgen.1000138</a></li>
<li id="Adey2010">Adey, A., H. G. Morrison, A. (no last name), X. Xun, J. O. Kitzman <i>et al.</i>, 2010 <b>Rapid,  low-input,  low-bias construction of shotgun fragment libraries by high-density in vitro transposition</b>. Genome Biology 11: R119. <a href="https://doi.org/10.1186/gb-2010-11-12-r119">10.1186/gb-2010-11-12-r119</a></li>
<li id="Green2012">Green, B., C. Bouchier, C. Fairhead, N. L. Craig, and B. P. Cormack, 2012 <b>Insertion site preference of Mu,  Tn5,  and Tn7 transposons</b>. Mobile DNA 3: 3. <a href="https://doi.org/10.1186/1759-8753-3-3">10.1186/1759-8753-3-3</a></li>
<li id="Buenrostro2013">Buenrostro, J. D., P. G. Giresi, L. C. Zaba, H. Y. Chang, and W. J. Greenleaf, 2013 <b>Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin,  DNA-binding proteins and nucleosome position</b>. Nature Methods 10: 1213‚Äì1218. <a href="https://doi.org/10.1038/nmeth.2688">10.1038/nmeth.2688</a></li>
<li id="Corces2017">Corces, M. R., A. E. Trevino, E. G. Hamilton, P. G. Greenside, N. A. Sinnott-Armstrong <i>et al.</i>, 2017 <b>An improved ATAC-seq protocol reduces background and enables interrogation of frozen tissues</b>. Nature Methods 14: 959‚Äì962. <a href="https://doi.org/10.1038/nmeth.4396">10.1038/nmeth.4396</a></li>
<li id="Kia2017">Kia, A., C. Gloeckner, T. Osothprarop, N. Gormley, E. Bomati <i>et al.</i>, 2017 <b>Improved genome sequencing using an engineered transposase</b>. BMC Biotechnology 17: <a href="https://doi.org/10.1186/s12896-016-0326-1">10.1186/s12896-016-0326-1</a></li>
<li id="Litzenburger2017">Litzenburger, U. M., J. D. Buenrostro, B. Wu, Y. Shen, N. C. Sheffield <i>et al.</i>, 2017 <b>Single-cell epigenomic variability reveals functional cancer heterogeneity</b>. Genome Biology 18: <a href="https://doi.org/10.1186/s13059-016-1133-7">10.1186/s13059-016-1133-7</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=ATAC-Seq data analysis (Epigenetics)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Lucille Delisle, Maria Doyle, Florian Heyl, 2021 <b>ATAC-Seq data analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html">https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{epigenetics-atac-seq,
author = "Lucille Delisle and Maria Doyle and Florian Heyl",
title = "ATAC-Seq data analysis (Galaxy Training Materials)",
year = "2021",
month = "06",
day = "22"
url = "\url{https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
