<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1706833679' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1706833679' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-07-03 15:49:13 +0000">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1706833679'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/index.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        



<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item "><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
     
     
     
     
     <li class="breadcrumb-item "><a href="/training-material/topics/metagenomics/faqs">Topic FAQs</a></li>
     

     
     <li class="breadcrumb-item active" aria-current="page">Tutorial FAQs</li>
     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
<!--<div class="contributors-line">Authors: 
    </div>-->
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="tutorial-faqs"> <a href="#tutorial-faqs"> Tutorial Questions </a></h3>

<hr/>



<h4 class="faq-area"> After sequencing with MinKNOW software, we get many fastq files, do these files need to be combined into one file before uploading or is it possible to upload them all at once? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/compining_fastq_files_of_one_sample.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-after-sequencing-with-minknow-software-we-get-many-fastq-files-do-these-files-need-to-be-combined-into-one-file-before-uploading-or-is-it-possible-to-upload-them-all-at-once"><i class="far fa-question-circle" aria-hidden="true"></i> Question: After sequencing with MinKNOW software, we get many fastq files, do these files need to be combined into one file before uploading or is it possible to upload them all at once?</div>   <p>After sequencing with MinKNOW software, it is a good approach to combine the files from the same run before processing them. You could create a collection per run with all fastq files and then use the collection operation to concatenate all files in a collection.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we also use this workflow on Illumina raw reads? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/worflow_and_sequencing_techniques.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-we-also-use-this-workflow-on-illumina-raw-reads"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can we also use this workflow on Illumina raw reads?</div>   <p>Yes, some tools would need to be changed or removed:</p>   <ul>   <li>For the Preprocessing workflow, plotting with Nanoplot shall be removed and keep only FastQC, MultiQC and Fastp.</li>   <li>For the mapping in the SNP based pathogen detection workflow, instead of Minimap2, Bowtie can be used.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we use snippy pipeline instead for the phylogenetic analysis? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/variant_calling_quality.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-we-use-snippy-pipeline-instead-for-the-phylogenetic-analysis"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can we use snippy pipeline instead for the phylogenetic analysis?</div>   <p>On principle yes. We did not try yet. Snippy is available in Galaxy</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do the pipelines work with both isolates and direct from raw meat? or only isolate? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/workflow_and_isolations.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-the-pipelines-work-with-both-isolates-and-direct-from-raw-meat-or-only-isolate"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do the pipelines work with both isolates and direct from raw meat? or only isolate?</div>   <p>The workflow can work with both isolates and raw meat. The workflow is designed to remove hosts before detecting any pathogen, so both isolates and raw meat samples are pre-processed equaliy before the analysis starts.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> For preprocessing part with host removal: Where do you find the abbreviations for each host species available (e.g. bos is cow, homo is human..)? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_removal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-for-preprocessing-part-with-host-removal-where-do-you-find-the-abbreviations-for-each-host-species-available-e-g-bos-is-cow-homo-is-human"><i class="far fa-question-circle" aria-hidden="true"></i> Question: For preprocessing part with host removal: Where do you find the abbreviations for each host species available (e.g. bos is cow, homo is human..)?</div>   <p>The abbreviation (i.e. the genus) is the first word in the <a href="https://github.com/lskatz/Kalamari/blob/master/src/Kalamari_v3.9.1.tsv">list of possible hosts</a>. The names are the scientific names for species, which would be shown on the taxonomy tree if you would look up the common name (i.e. bovine) on Wikipedia.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> From where can I import other genomes? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/importing_genomes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-from-where-can-i-import-other-genomes"><i class="far fa-question-circle" aria-hidden="true"></i> Question: From where can I import other genomes?</div>   <p>In this tutorial, we used kalamari DB with the <a href="https://github.com/lskatz/Kalamari/blob/master/src/Kalamari_v3.9.1.tsv">full list of possible host sequences</a> that can be removed. Reads are either tagged to map one of those species or are left unassigned. If the task at hand in the real world cannot be covered by those, you can also try another DB for Kraken2 that includes your species (or maybe retain unmapped reads from a read aligner such as Bowtie2, Minimap2…).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is there a way to filter on the Kalimari database? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/kalamri_database_filtering.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-there-a-way-to-filter-on-the-kalimari-database"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is there a way to filter on the Kalimari database?</div>   <p>To filter the Kalamari database, e.g. leaving out milk bacteria only to detect spoilers or contaminants, but the Kalimera list contains a lot more than that, you can:</p>   <ol>   <li>Look at a publication etc. to find a list of bacteria to remove.</li>   <li>Change the regex <code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos.*$</code>  to <code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos|Bacterium1|Bacterium2...|BacteriumN.*$</code></li> </ol>   <p>Milk pathogens are somewhat known, <em>Salmonella</em>, <em>Escherichia</em>… It might be easier to retain reads only mapping to pathogens instead</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Isn't it awkward to find so many humans sequences there, since we filter for them before? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/remaining_human_sequences.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-isn-t-it-awkward-to-find-so-many-humans-sequences-there-since-we-filter-for-them-before"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Isn't it awkward to find so many humans sequences there, since we filter for them before?</div>   <p>We see a lot that Kraken tends to assign many reads to human, despite they do not map to human genome. Due to resemblance between organisms and the limited species coverage of Kraken databases sometimes does happen that reads corresponding to higher organisms get mapped to humans. It was a very severe problem for the standard databases, because yeast genes were mis-assigned to human.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What does `^.*Gallus|Homo|Bos.*$` mean? <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_filtering_filter.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-does-gallus-homo-bos-mean"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What does `^.*Gallus|Homo|Bos.*$` mean?</div>   <p><code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos.*$</code> is a <a href="https://www.regular-expressions.info/">regular expression</a> that matches a string containing the words Gallus <strong>OR</strong> Homo <strong>OR</strong> Bos.</p> </blockquote> <!--END_SNIPPET-->




<br>
<h3 id="general-faqs"> <a href="#general-faqs"> General Questions </a></h3>
<hr>
<!-- Add the "help my tool is missing FAQ to all tutorial faq pages -->
<h4 class="faq-area"> Can't find one of the tools for this tutorial? <a href="/training-material/faqs/galaxy/tools_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-if-a-tool-is-missing"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-if-a-tool-is-missing" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: If a Tool is Missing<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To use the tools installed and available on the Galaxy server:</p>   <ol>   <li>At the top of the left tool panel, type in a tool name or datatype into the tool search box.</li>   <li>Shorter keywords find more choices.</li>   <li>Tools can also be directly browsed by category in the tool panel.</li> </ol>   <p>If you can’t find a tool you need for a tutorial on Galaxy, please:</p>   <ol>   <li><strong>Check that you are using a compatible Galaxy server</strong>     <ul>       <li>Navigate to the overview box at the top of the tutorial</li>       <li>Find the <strong>“Supporting Materials”</strong> section</li>       <li>Check <strong>“Available on these Galaxies”</strong></li>       <li>If your server is not listed here, the tutorial is not supported on your Galaxy server</li>       <li>You can create an account on one of the supporting Galaxies   <img src="/training-material/faqs/galaxy/images/supporting_galaxies.png" alt="screenshot of overview box with available Galaxies section" /></li>     </ul>   </li>   <li>Use the <strong>Tutorial mode</strong> feature     <ul>       <li>Open your Galaxy server</li>       <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>       <li>Navigate to your tutorial</li>       <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>       <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>     </ul>   </li>   <li><strong>Still not finding the tool?</strong>     <ul>       <li>Ask help in <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a>.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->
<h4 class="faq-area"> Running into an error? <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Troubleshooting errors<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When something goes wrong in Galaxy, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your Galaxy errors.</p>   <ol>   <li><strong>Expand</strong> the red history dataset by clicking on it.     <ul>       <li>Sometimes you can already see an error message here</li>     </ul>   </li>   <li>     <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></p>   </li>   <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:     <ul>       <li>Expand the history item</li>       <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>       <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:         <ul>           <li>Tool Standard Output</li>           <li>Tool Standard Error</li>         </ul>       </li>       <li>For more information about specific tool errors, please see the <a href="/training-material/faqs/galaxy/index.html#troubleshooting">Troubleshooting section</a></li>     </ul>   </li>   <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.     <ul>       <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></li>       <li>Write down any information you think might help solve the problem         <ul>           <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>         </ul>       </li>       <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> Report</strong> button</li>     </ul>   </li>   <li><strong>Ask for help!</strong>     <ul>       <li>Where?         <ul>           <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a></li>           <li>In the <a href="https://gitter.im/galaxyproject/Lobby">Galaxy Gitter Channel</a></li>           <li>Browse the <a href="https://help.galaxyproject.org/">Galaxy Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>         </ul>       </li>       <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->









<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div>Galaxy Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/index.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1706833679'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>
