<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Inferring Trajectories using Monocle3 (R)</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1705797054' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1705797054' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Inferring Trajectories using Monocle3 (R)">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-09-21 15:52:41 +0000">
<meta name="DC.creator" content="Julia Jakiela">

        
        
        
        
        <meta name="description" content="Training material and practicals for all kinds of single ...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Inferring Trajectories using Monocle3 (R)">
        <meta property="og:description" content="Training material and practicals for all kinds of single ...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-09-21 15:52:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Inferring Trajectories using Monocle3 (R)",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Inferring Trajectories using Monocle3 (R)' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html",
  "timeRequired": "PT3H",
  "teaches": "Identify which operations are necessary to transform an AnnData object into the files needed for Monocle\n - Describe the Monocle3 functions in R\n - Recognise steps that can be performed in R, but not with current Galaxy tools\n - Repeat the Monocle3 workflow and choose appropriate parameter values\n - Compare the outputs from Scanpy, Monocle in Galaxy and Monocle in R\n - Describe differential expression analysis methods",
  "keywords": "single-cell, single-cell, trajectory-analysis, paper-replication, rmarkdown-notebook, jupyter-notebook",
  "description": "The questions this  addresses are:\n - How do I perform a trajectory analysis using Monocle3 in R?\n - What should I do when Galaxy's Monocle tools are not enough?\n - How do I assign cell types to the clusters?\n - How do I infer lineage relationships between clusters, without a time series?\n - How do I perform batch correction and differential expression analysis in Monocle?\n\n\nThe objectives are:\n - Identify which operations are necessary to transform an AnnData object into the files needed for Monocle\n - Describe the Monocle3 functions in R\n - Recognise steps that can be performed in R, but not with current Galaxy tools\n - Repeat the Monocle3 workflow and choose appropriate parameter values\n - Compare the outputs from Scanpy, Monocle in Galaxy and Monocle in R\n - Describe differential expression analysis methods\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html",
      "name": "Generating a single cell matrix using Alevin",
      "description": "Hands-on for 'Generating a single cell matrix using Alevin' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html",
      "name": "Combining datasets after pre-processing",
      "description": "Hands-on for 'Combining datasets after pre-processing' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html",
      "name": "Filter, Plot and Explore Single-cell RNA-seq Data",
      "description": "Hands-on for 'Filter, Plot and Explore Single-cell RNA-seq Data' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "name": "Julia Jakiela",
      "image": "https://avatars.githubusercontent.com/wee-snufkin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "single-cell"
    },
    {
      "@type": "Thing",
      "name": "trajectory-analysis"
    },
    {
      "@type": "Thing",
      "name": "paper-replication"
    },
    {
      "@type": "Thing",
      "name": "rmarkdown-notebook"
    },
    {
      "@type": "Thing",
      "name": "jupyter-notebook"
    }
  ],
  "abstract": "## Setting up the environment and file upload"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1705797054'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Inferring Trajectories using Monocle3 (R)</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin"><img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Avatar" width="36" height="36">Julia Jakiela</a>
		</td>
	</tr>
	


	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hexylena?s=36" alt="Avatar" width="36" height="36">Helena Rasche</a><a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Avatar" width="36" height="36">Pavankumar Videm</a><a href="/training-material/hall-of-fame/nomadscientist/" class="contributor-badge contributor-nomadscientist"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/nomadscientist?s=36" alt="Avatar" width="36" height="36">Wendi Bacon</a></td>
	</tr>
	

	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do I perform a trajectory analysis using Monocle3 in R?</p>
</li>
        
        <li><p>What should I do when Galaxy’s Monocle tools are not enough?</p>
</li>
        
        <li><p>How do I assign cell types to the clusters?</p>
</li>
        
        <li><p>How do I infer lineage relationships between clusters, without a time series?</p>
</li>
        
        <li><p>How do I perform batch correction and differential expression analysis in Monocle?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Identify which operations are necessary to transform an AnnData object into the files needed for Monocle</p>
</li>
        
        <li><p>Describe the Monocle3 functions in R</p>
</li>
        
        <li><p>Recognise steps that can be performed in R, but not with current Galaxy tools</p>
</li>
        
        <li><p>Repeat the Monocle3 workflow and choose appropriate parameter values</p>
</li>
        
        <li><p>Compare the outputs from Scanpy, Monocle in Galaxy and Monocle in R</p>
</li>
        
        <li><p>Describe differential expression analysis methods</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/single-cell">Single Cell</a>
        
            <ul>
                
                    
                        
                            
                            <li> Generating a single cell matrix using Alevin:
                            
                            
                                
                                     <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Combining datasets after pre-processing:
                            
                            
                                
                                     <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Filter, Plot and Explore Single-cell RNA-seq Data:
                            
                            
                                
                                     <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/7455590">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            

            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/single-cell-scrna-case_monocle3-rstudio.ipynb" title="Jupyter Notebook" alt="Inferring Trajectories using Monocle3 (R) Notebook">
        <i class="far fa-file-code" aria-hidden="true"></i> Jupyter Notebook
    </a>
    
        <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/single-cell-scrna-case_monocle3-rstudio.Rmd" title="R markdown notebook" alt="Inferring Trajectories using Monocle3 (R) Notebook">
            <i class="far fa-file-code" aria-hidden="true"></i> RMarkdown Notebook
        </a>
    

</li>
            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Sep 21, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00336">https://gxy.io/GTN:T00336</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 
                <section aria-label="Tutorial Notebook access information" id="tutorial-notebook">
                     
                     
                     
                     
                    
                        
                        <!--SNIPPET--><h1 id="introduction">Introduction</h1> <p>This tutorial is the next one in the <a href="/training-material/topics/single-cell/index.html">Single-cell RNA-seq: Case Study</a> series. This tutorial focuses on trajectory analysis using <a href="https://cole-trapnell-lab.github.io/monocle3/">monocle3</a>, similar to the <a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">Monocle3 in Galaxy tutorial</a>. However, in this tutorial we will use the R programming language that hides behind the user-friendly Galaxy tools. Sometimes you might encounter limitations when working with Galaxy tools, or you might want to make a wee modification that has to be done manually. It is therefore useful to be able to switch to R. If you do not feel confident using R, <a href="/training-material/topics/data-science/tutorials/r-basics/tutorial.html">this tutorial</a> is a good place to start. However, our tutorial is quite straightforward to follow and at the end you will feel like a programmer! On the other hand, if you are not confident with the biological or statistical theory behind trajectory analysis, check out the <a href="/training-material/topics/single-cell/tutorials/scrna-trajectories/slides.html">slide deck</a>. With those resources (including the previous case study tutorials) you are well-equipped to go through this tutorial with ease. Let’s get started!</p> <blockquote class="comment">   <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>   <p>This tutorial is significantly based on the <a href="https://cole-trapnell-lab.github.io/monocle3/docs/introduction/">Monocle3 documentation</a>.</p> </blockquote> <h2 id="optional-get-data-into-a-galaxy-history">Optional: Get data into a Galaxy history</h2> <p>In the <a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">Monocle3 in Galaxy tutorial</a>, we showed that Monocle3 works great with annotated data, but what if your data is not annotated yet? Is it still possible to use Monocle? The answer is yes, Monocle can annotate cells according to their type.</p> <p>First, we need to retrieve the appropriate data. We will continue to work on the case study data from a mouse model of fetal growth restriction <span class="citation"><a href="#Bacon2018">Bacon <i>et al.</i> 2018</a></span> (see <a href="https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-6945/results/tsne">the study in Single Cell Expression Atlas</a> and <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6945/">the project submission</a>). We will use the filtered AnnData object, before normalisation and annotation, generated in the <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">filtering tutorial</a> as step <code class="language-plaintext highlighter-rouge">20: Filtered Object</code>. However for this tutorial, you don’t have to download any files on your computer or even import files to Galaxy! We will show you the whole analysis in R, starting from AnnData object. However, if you’d like to examine the datasets in a history, the instructions are below.</p> <blockquote class="hands_on">   <div class="box-title hands-on-title" id="hands-on-optional-data-upload-into-galaxy-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optional data upload into Galaxy history</div>   <p>You have three options for importing the input data into a Galaxy history.</p>   <ol>     <li>       <p>You can import a history from: <a href="https://usegalaxy.eu/u/wendi.bacon.training/h/cs4trajectories--monocle3--rstudio---input">input history</a>; Import the files from <a href="https://zenodo.org/record/7455590">Zenodo</a>; or Import the files from the shared data library (<code class="language-plaintext highlighter-rouge">GTN - Material</code> -&gt; <code class="language-plaintext highlighter-rouge">single-cell</code>  -&gt; <code class="language-plaintext highlighter-rouge">Inferring Trajectories using Monocle3 (R)</code>):</p>       <!--SNIPPET-->       <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>Import history</em> button on the top right</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>       <p><!--END_SNIPPET--></p>       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7455590/files/AnnData_filtered.h5ad<br></code></pre></div>      </div>       <!--SNIPPET-->       <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>       <p><!--END_SNIPPET--></p>       <!--SNIPPET-->       <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>       <p><!--END_SNIPPET--></p>     </li>     <li>       <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">h5ad</code></p>       <!--SNIPPET-->       <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Changing the datatype<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the lower part <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong>, select <code class="language-plaintext highlighter-rouge">h5ad</code>     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>       <p><!--END_SNIPPET--></p>     </li>     <li>       <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the history to your name of choice.</p>     </li>   </ol> </blockquote> <h2 id="launching-jupyterlab">Launching JupyterLab</h2> <p><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Please note: this is only currently available on the <a href="https://usegalaxy.eu">usegalaxy.eu</a> site.</p> <blockquote class="warning">   <div class="box-title warning-title" id="warning-data-uploads-amp-jupyterlab"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Data uploads &amp; JupyterLab</div>   <p>There are a few ways of importing and uploading data into JupyterLab. You might find yourself accidentally doing this differently than the tutorial, and that’s ok. There are a few key steps where you will call files from a location - if these don’t work from you, check that the file location is correct and change accordingly!</p> </blockquote> <p>JupyterLab is a bit like RStudio but for other coding languages. What, you’ve never heard of <a href="https://www.rstudio.com/products/rstudio/features/">RStudio</a>? Then don’t worry, just follow the instructions! Saying that, for those that are confident with the R programming language and want to run this locally in the RStudio programming environment, we have included some extra details throughout that helped us run the code locally in RStudio. This tutorial is built to run in JupyterLab, however.</p> <blockquote>   <!--SNIPPET-->   <blockquote class="hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch JupyterLab</div>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-jupyterlab-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-jupyterlab-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Launch JupyterLab in Galaxy<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently JupyterLab in Galaxy is available on <a href="https://live.usegalaxy.eu">Live.useGalaxy.eu</a>, <a href="https://usegalaxy.org">usegalaxy.org</a> and <a href="https://usegalaxy.eu">usegalaxy.eu</a>.</p>   <blockquote class="hands_on">     <div class="box-title hands-on-title" id="hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run JupyterLab</div>     <ol>       <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive Jupyter Notebook tool" aria-role="link"><strong>Interactive Jupyter Notebook</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_jupyter_notebook</span></span>. Note that on some Galaxies this is called <span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive JupyTool and notebook tool" aria-role="link"><strong>Interactive JupyTool and notebook</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_jupyter_notebook</span></span>:</li>       <li>Click Run Tool</li>       <li>         <p>The tool will start running and will stay running permanently</p>         <p>This may take a moment, but once the <code class="language-plaintext highlighter-rouge">Executed notebook</code> in your history is <strong>orange</strong>, you are up and running!</p>       </li>       <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>       <li>Click on your JupyterLab instance</li>     </ol>   </blockquote> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-try-jupyterlab-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-try-jupyterlab-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Launch Try JupyterLab if not available on Galaxy<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If JupyterLab is not available on the Galaxy instance:</p>   <ol>     <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>   </ol> </blockquote> </blockquote>   <p><!--END_SNIPPET--></p> </blockquote> <p>Welcome to JupyterLab!</p> <blockquote class="warning">   <div class="box-title warning-title" id="warning-danger-you-can-lose-data"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Danger: You can lose data!</div>   <p>Do NOT delete or close this notebook dataset in your history. YOU WILL LOSE IT!</p> </blockquote> <h2 id="open-the-notebook">Open the notebook</h2> <p>You have two options for how to proceed with this JupyterLab tutorial - you can run the tutorial from a pre-populated notebook, or you can copy and paste the code for each step into a fresh notebook and run it. The initial instructions for both options are below.</p> <blockquote class="hands_on">   <div class="box-title hands-on-title" id="hands-on-option-1-open-the-notebook-directly-in-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 1: Open the notebook directly in JupyterLab</div>   <ol>     <li>Open a <code class="language-plaintext highlighter-rouge">Terminal</code> in JupyterLab with File -&gt; New -&gt; Terminal</li>   </ol>   <p><img src="../../images/scrna-casestudy-monocle/terminal_choose.jpg" alt="Screenshot of the Launcher tab with an arrow indicating where to find Terminal." title="This is how the Launcher tab looks like and where you can find Terminal." /></p>   <ol>     <li>Run       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/single-cell-scrna-case_monocle3-rstudio.ipynb<br></code></pre></div>      </div>     </li>     <li>Select the notebook that appears in the list of files on the left.</li>   </ol>   <p>Remember that you can also download this <i class="far fa-file-code" aria-hidden="true"></i><span class="visually-hidden">notebook</span> <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/single-cell-scrna-case_monocle3-rstudio.ipynb">Jupyter Notebook</a> from the <i class="fas fa-external-link-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy_instance</span> Supporting Materials in the Overview box at the beginning of this tutorial.</p> </blockquote> <blockquote class="hands_on">   <div class="box-title hands-on-title" id="hands-on-option-2-creating-a-notebook"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 2: Creating a notebook</div>   <ol>     <li>Select the <strong>R</strong> icon under <strong>Notebook</strong></li>   </ol>   <p><img src="../../images/scrna-casestudy-monocle/R_logo.png" alt="R icon" title="R Notebook Button" /></p>   <ol>     <li>       <p>Save your file (<strong>File</strong>: <strong>Save</strong>, or click the <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> Save icon at the top left)</p>     </li>     <li>       <p>If you right click on the file in the folder window at the left, you can rename your file <code class="language-plaintext highlighter-rouge">whateveryoulike.ipynb</code></p>     </li>   </ol> </blockquote> <blockquote class="warning">   <div class="box-title warning-title" id="warning-you-should-b-save-b-frequently"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: You should <b>Save</b> frequently!</div>   <p>This is both for good practice and to protect you in case you accidentally close the browser. Your environment will still run, so it will contain the last saved notebook you have. You might eventually stop your environment after this tutorial, but ONLY once you have saved and exported your notebook (more on that at the end!) Note that you can have multiple notebooks going at the same time within this JupyterLab, so if you do, you will need to save and export each individual notebook. You can also download them at any time.</p> </blockquote> <p>Let’s crack on!</p> <h2 id="installation">Installation</h2> <p>If you followed the <i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> tip above, you should already have your JupyterLab instance open. Before we start working on the tutorial notebook, we need to install required packages.</p> <blockquote class="hands_on">   <div class="box-title hands-on-title" id="hands-on-installing-the-packages"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Installing the packages</div>   <ol>     <li>Navigate back to the <code class="language-plaintext highlighter-rouge">Terminal</code></li>     <li>In the Terminal tab open, write the following, one line at a time:       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda install -y -c conda-forge -c bioconda r-monocle3<br></code></pre></div>      </div>       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda install -y -c conda-forge -c bioconda anndata<br></code></pre></div>      </div>       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda install -y -c conda-forge r-viridislite<br></code></pre></div>      </div>       <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>conda install -y -c conda-forge bioconductor-biomart<br></code></pre></div>      </div>     </li>     <li>If you are asked at any point <code class="language-plaintext highlighter-rouge">Proceed ([y]/n)?</code>, type <code class="language-plaintext highlighter-rouge">y</code> - surely we want to proceed!</li>   </ol> </blockquote> <p>Installation will take a while, so in the meantime, when it’s running, you can open the notebook and follow the rest of this tutorial there!</p> <blockquote class="tip">   <div class="box-title tip-title" id="tip-installation-for-rstudio-users"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-installation-for-rstudio-users" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Installation for RStudio users<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Monocle 3 runs in the R statistical computing environment. You will need R version 4.1.0 or higher, Bioconductor version 3.14, and monocle3 1.2.7 or higher to have access to the latest features. Here is the original code that you should run to install BiocManager and monocle3.</p>   <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Install Bioconductor and some of its dependencies</span><span class="w"><br></span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"><br></span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w"><br></span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="n">version</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"3.14"</span><span class="p">)</span><span class="w"><br></span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s1">'BiocGenerics'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DelayedArray'</span><span class="p">,</span><span class="w"> </span><span class="s1">'DelayedMatrixStats'</span><span class="p">,</span><span class="w"><br>                       </span><span class="s1">'limma'</span><span class="p">,</span><span class="w"> </span><span class="s1">'lme4'</span><span class="p">,</span><span class="w"> </span><span class="s1">'S4Vectors'</span><span class="p">,</span><span class="w"> </span><span class="s1">'SingleCellExperiment'</span><span class="p">,</span><span class="w"><br>                       </span><span class="s1">'SummarizedExperiment'</span><span class="p">,</span><span class="w"> </span><span class="s1">'batchelor'</span><span class="p">,</span><span class="w"> </span><span class="s1">'HDF5Array'</span><span class="p">,</span><span class="w"><br>                       </span><span class="s1">'terra'</span><span class="p">,</span><span class="w"> </span><span class="s1">'ggrastr'</span><span class="p">))</span><span class="w"><br><br></span><span class="c1"># Install monocle3 through the cole-trapnell-lab GitHub:</span><span class="w"><br></span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"devtools"</span><span class="p">)</span><span class="w"><br></span><span class="n">devtools</span><span class="o">::</span><span class="n">install_github</span><span class="p">(</span><span class="s1">'cole-trapnell-lab/monocle3'</span><span class="p">)</span><span class="w"><br></span><span class="n">library</span><span class="p">(</span><span class="n">monocle3</span><span class="p">)</span><span class="w"><br></span></code></pre></div>  </div>   <blockquote class="warning">     <div class="box-title warning-title" id="warning-installation-errors"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Installation errors</div>     <p>It may happen that you will encounter some problems with installation of monocle3 when using RStudio Galaxy instance or RStudio Cloud. It might be due to using older versions of R or required packages or lack of required dependencies. If it happens, you would need to carefully read the error messages and follow the suggestions. If you are facing any difficulties with installation process, it is recommended that you consult additional online resources. It is more likely that RStudio Cloud or Galaxy tool would fail rather than local RStudio. To make your analysis stress-free, you can follow the Jupyter Notebook instead, which should not give you installation issues.</p>   </blockquote> </blockquote> <!--END_SNIPPET-->
                    
                </section>
                

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h2 id="setting-up-the-environment-and-file-upload">Setting up the environment and file upload</h2>
<p>Once the installation is done, we should load the needed packages into our notebook. Navigate back to your <code class="language-plaintext highlighter-rouge">notebook</code>. If you are using our prepopulated notebook, you can follow the tutorial from there. Otherwise, input the following into your fresh notebook.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">install.packages</span><span class="p">(</span><span class="s2">"Rcpp"</span><span class="p">)</span><span class="w">    </span><span class="c1"># needed for reduce_dimension to avoid AnnoyAngular error; library(Rcpp) might work as well depending on the version</span><span class="w">
</span></code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">monocle3</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">magrittr</span><span class="p">)</span><span class="w">     </span><span class="c1"># needed for %&gt;%</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">viridisLite</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">install.packages</span><span class="p">(</span><span class="s2">"anndata"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Alright, the libraries are here - so now let’s get our AnnData file.</p>

<h2 id="upload-view-and-modify-the-files">Upload, view and modify the files</h2>

<p>We can download the file into Jupyter notebook also using a Zendo link. The first argument is the download link and the second one is the name under which the downloaded file is saved:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">download.file</span><span class="p">(</span><span class="s1">'https://zenodo.org/record/7455590/files/AnnData_filtered.h5ad'</span><span class="p">,</span><span class="w"> </span><span class="s1">'AnnData.h5ad'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Then we have to read in this h5ad file:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ann</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">anndata</span><span class="o">::</span><span class="n">read_h5ad</span><span class="p">(</span><span class="s1">'AnnData.h5ad'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>Now we store all the information we need in AnnData object. However, Monocle uses <em>cell_data_set class</em> to hold expression data, which requires three input files: <code class="language-plaintext highlighter-rouge">expression_matrix</code>, <code class="language-plaintext highlighter-rouge">cell_metadata</code> and <code class="language-plaintext highlighter-rouge">gene_metadata</code>. Therefore, we have to somehow ‘transfer’ that information from our AnnData object to Monocle’s cell_data_set (cds). AnnData stores a data matrix <code class="language-plaintext highlighter-rouge">X</code> together with annotations of observations <code class="language-plaintext highlighter-rouge">obs</code> and variables <code class="language-plaintext highlighter-rouge">var</code>, so we can extract those parameters and use them for further analysis.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">expression_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ann</span><span class="o">$</span><span class="n">X</span><span class="w">
</span><span class="n">cell_metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ann</span><span class="o">$</span><span class="n">obs</span><span class="w">
</span><span class="n">gene_metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ann</span><span class="o">$</span><span class="n">var</span><span class="w">
</span></code></pre></div></div>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-uploading-files-from-your-computer-is-also-an-option-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-uploading-files-from-your-computer-is-also-an-option-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Uploading files from your computer is also an option!<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you already have files containing the expression matrix, genes and cells metadata, you can upload them to JupyLab and generate a cds file from them instead. For example, if you first downloaded the files from Galaxy, your files will have the <code class="language-plaintext highlighter-rouge">.tabular</code> extension. In this case, we will use the <code class="language-plaintext highlighter-rouge">read.delim()</code> function to read them in. In this function, the first argument is the file path - in our case, the files are in the same folder as the notebook, so the file path is the same as the file name. You can always check that by right-clicking on the file and choosing <code class="language-plaintext highlighter-rouge">Copy path</code>. The second argument, <code class="language-plaintext highlighter-rouge">row.names=1</code> takes the column number of the data file from which to take the row names.</p>
  <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># read in the files</span><span class="w">
</span><span class="n">cell_metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="s1">'cells.tabular'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w">
</span><span class="n">gene_metadata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="s1">'genes.tabular'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w">
</span><span class="n">expression_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="s1">'expression.tabular'</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w">
</span></code></pre></div>  </div>

  <blockquote class="question">
    <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>Why should we set <code class="language-plaintext highlighter-rouge">row.names=1</code>?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>This allows us to ensure that the expression value matrix has the same number of columns as the <code class="language-plaintext highlighter-rouge">cell_metadata</code> has rows and the same number of rows as the <code class="language-plaintext highlighter-rouge">gene_metadata</code> has rows. Importantly, the row names of the <code class="language-plaintext highlighter-rouge">cell_metadata</code> object should match the column names of the expression matrix and the row names of the <code class="language-plaintext highlighter-rouge">gene_metadata</code> object should match the row names of the expression matrix.</p>

    </blockquote>
  </blockquote>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-reading-in-the-files-for-rstudio-users-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-reading-in-the-files-for-rstudio-users-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Reading in the files for RStudio users<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you are working in RStudio Cloud, you have to click on ‘Upload’ button in the right bottom window toolbar and choose already downloaded data files to upload. You should now see all three files in this window. You might want to rename the files to make their names shorter.</p>

  <p><a href="../../images/scrna-casestudy-monocle/r_files_tab.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/r_files_tab.png" alt="Screenshot of Files tab in RStudio, highlighting 'Upload' and 'Rename' buttons and listing three uploaded and renamed files: 'cell_metadata', 'gene_metadata', 'expression_matrix'." width="665" height="228" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> The view of the Files tab with uploaded files and highlighted relevant buttons.&lt;/figcaption&gt;</figure></a></p>

  <p>If you are using RStudio Galaxy tool, you can get data directly from your history by running:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># get the files from Galaxy history
file.copy(gx_get(2), "cell_metadata")
file.copy(gx_get(3), "gene_metadata")
file.copy(gx_get(4), "expression_matrix")
</code></pre></div>  </div>
  <p>The number in the brackets corresponds to the dataset number in your history, so make sure you put the right number for the corresponding file. We can specify the name of the fetched files in the quotation marks, as shown above. All three files should appear in the Files tab window.</p>

  <p>If you are using RStudio locally, then you don’t have to bother about uploading the files – just run <code class="language-plaintext highlighter-rouge">file.choose()</code> and choose the corresponding file which you want to get path to:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># get the file path
cells_path &lt;- file.choose()
genes_path &lt;- file.choose()
expression_path &lt;- file.choose()
</code></pre></div>  </div>
  <p>You should now see the new variables in the Environment tab window.</p>

</blockquote>

<p>According to the <a href="https://cole-trapnell-lab.github.io/monocle3/docs/starting/">Monocle3 documentation</a>, the <code class="language-plaintext highlighter-rouge">expression_matrix</code> should have genes as rows and cells as columns. Let’s check if that’s the case here.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">head</span><span class="p">(</span><span class="n">expression_matrix</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="m">5</span><span class="p">))</span><span class="w">           </span><span class="c1"># preview the content of the file by calling its the first 5 rows by 5 columns</span><span class="w">
</span></code></pre></div></div>
<p>We can see that in our matrix rows are cells and genes are columns, so we have to transpose the matrix simply using function <code class="language-plaintext highlighter-rouge">t()</code>. But before doing so, we will change its type from dataframe to matrix - this is Monocle’s requirement to generate cell_data_set afterwards.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">expression_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">expression_matrix</span><span class="p">)</span><span class="w">   </span><span class="c1"># change the type to matrix</span><span class="w">
</span><span class="n">expression_matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">expression_matrix</span><span class="p">)</span><span class="w">           </span><span class="c1"># transpose the matrix</span><span class="w">
</span></code></pre></div></div>
<p>Another condition we have to satisfy is that one of the columns of the <code class="language-plaintext highlighter-rouge">gene_metadata</code> should be named “gene_short_name”, which represents the gene symbol for each gene. Some functions won’t work without that. Do we have such a column? Let’s check.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">head</span><span class="p">(</span><span class="n">gene_metadata</span><span class="p">)</span><span class="w">             </span><span class="c1"># preview the top ten rows of the file to check the name of the column containing gene symbols</span><span class="w">
</span></code></pre></div></div>

<p>The second column indeed contains gene symbols, but is called “Symbol” instead of “gene_short_name”. That can be easily changed by a simple assignment, as long as we know the number of the column that we want to modify. In our case the gene symbols are stored in column 2. We can access the column names by <code class="language-plaintext highlighter-rouge">colnames()</code>.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">colnames</span><span class="p">(</span><span class="n">gene_metadata</span><span class="p">)[</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'gene_short_name'</span><span class="w">     </span><span class="c1"># change the column name</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">gene_metadata</span><span class="p">)</span><span class="w">                             </span><span class="c1"># see the changes</span><span class="w">
</span></code></pre></div></div>

<h2 id="generating-cds-object">Generating CDS object</h2>
<p>Now let’s store our files in one object – the <code class="language-plaintext highlighter-rouge">cell_data_set</code>. This is the main class used by Monocle to hold single cell expression data. The class is derived from the Bioconductor SingleCellExperiment class. Similar to Python’s AnnData, the cell_data_set stores a data matrix together with annotations of observations and variables. There are three ways of creating CDS object in monocle:</p>
<ul>
  <li>Using <code class="language-plaintext highlighter-rouge">new_cell_data_set()</code> function with three data frames as arguments (not their paths!): expression matrix (can also be a sparseMatrix), cell metadata and gene metadata</li>
  <li>Using <code class="language-plaintext highlighter-rouge">load_cellranger_data()</code> function providing the path to the folder containing 10X Genomics Cell Ranger output files. This function takes an argument <code class="language-plaintext highlighter-rouge">umi_cutoff</code> that determines how many reads a cell must have to be included</li>
  <li>Using <code class="language-plaintext highlighter-rouge">load_mm_data()</code> function providing the paths to matrix file and two metadata files (features and cell information).</li>
</ul>

<p>In this tutorial we will use the first option:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create CDS object</span><span class="w">
</span><span class="n">cds</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">new_cell_data_set</span><span class="p">(</span><span class="n">expression_matrix</span><span class="p">,</span><span class="w"> </span><span class="n">cell_metadata</span><span class="p">,</span><span class="w"> </span><span class="n">gene_metadata</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>We are now ready to process our data!</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-format-conversion-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-format-conversion-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Format conversion<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Since Monocle’s CDS object is analogous to Python’s AnnData, why don’t we use some kind of conversion between those two formats? There is indeed a package called <code class="language-plaintext highlighter-rouge">sceasy</code> that helps easy conversion of different single-cell data formats to each other. However, when we tested this conversion on our dataset and then used Monocle to plot the expression of genes, the plots were not correct – the expression was shown to be identical throughout the sample. For comparison, Seurat did well when plotting gene expression of the same converted object! Although conversion functions are very handy, you have to be aware that their output might be interpreted differently by certain packages. Therefore, to make sure that the analysis is reliable, we decided to generate CDS object directly using Monocle’s function.
<a href="../../images/scrna-casestudy-monocle/monocle_seurat.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/monocle_seurat.png" alt="Comparison between plots of gene expression generated by Monocle and Seurat using the CDS object that was converted from AnnData using SCEasy tool. Gene expression in Monocle plots is identical throughout the sample, while the expression of the same genes plotted by Seurat is noticeable only in specific clusters." width="866" height="839" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Comparison between plots of gene expression generated by Monocle and Seurat using the CDS object that was converted from AnnData using SCEasy tool.&lt;/figcaption&gt;</figure></a></p>

</blockquote>

<h1 id="additional-step-adding-genes-symbols-based-on-their-ids">Additional step: adding genes symbols based on their IDs</h1>
<blockquote class="warning">
  <div class="box-title warning-title" id="warning-additional-step-2"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Additional step</div>
  <p>This step is not necessary for the dataset we are working on but some users might find it helpful when analysing their own data.</p>

</blockquote>

<p>If you remember the very first tutorial in the case study, we started with gene IDs and added gene symbols based on the Ensembl GTF file.<br />
But what if we didn’t have the genes symbols in our CDS object and wanted to add them now? Of course - it’s possible! We will also base this annotation on Ensembl - the genome database – with the use of the library BioMart. We will use the same archive as in the Alevin tutorial (Genome assembly GRCm38) to get the gene names. Please note that the updated version (GRCm39) is available, but some of the gene IDs are not in that EnsEMBL database. The code below is written in a way that it will work for the updated dataset too, but will produce ‘NA’ where the corresponding gene name couldn’t be found.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">cds_extra</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds</span><span class="w">                                        </span><span class="c1"># assign our CDS to a new object for the demonstration purpose</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">fData</span><span class="p">(</span><span class="n">cds_extra</span><span class="p">)))</span><span class="w">                        </span><span class="c1"># preview of the gene IDs as rownames</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># get relevant gene names</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="s2">"biomaRt"</span><span class="p">)</span><span class="w">                                      </span><span class="c1"># load the BioMart library</span><span class="w">
</span><span class="n">ensembl.ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">fData</span><span class="p">(</span><span class="n">cds_extra</span><span class="p">))</span><span class="w">               </span><span class="c1"># fData() allows to access cds rowData table</span><span class="w">
</span><span class="n">mart</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">useEnsembl</span><span class="p">(</span><span class="n">biomart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ENSEMBL_MART_ENSEMBL"</span><span class="p">)</span><span class="w">    </span><span class="c1"># connect to a specified BioMart database and dataset hosted by Ensembl</span><span class="w">
</span><span class="n">ensembl_m</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">useMart</span><span class="p">(</span><span class="s2">"ensembl"</span><span class="p">,</span><span class="w"> </span><span class="n">dataset</span><span class="o">=</span><span class="s2">"mmusculus_gene_ensembl"</span><span class="p">,</span><span class="w"> </span><span class="n">host</span><span class="o">=</span><span class="s1">'https://nov2020.archive.ensembl.org'</span><span class="p">)</span><span class="w"> 	

</span><span class="c1"># The line above connects to a specified BioMart database and dataset within this database.</span><span class="w">
</span><span class="c1"># In our case we choose the mus musculus database and to get the desired Genome assembly GRCm38,</span><span class="w">
</span><span class="c1"># we specify the host with this archive. If you want to use the most recent version of the dataset, just run:</span><span class="w">
</span><span class="c1"># ensembl_m = useMart("ensembl", dataset="mmusculus_gene_ensembl")</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'ensembl_gene_id'</span><span class="p">,</span><span class="s1">'external_gene_name'</span><span class="p">),</span><span class="w">
               </span><span class="n">filters</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'ensembl_gene_id'</span><span class="p">,</span><span class="w">
               </span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl.ids</span><span class="p">,</span><span class="w">
               </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl_m</span><span class="p">)</span><span class="w">

</span><span class="c1"># The line above retrieves the specified attributes from the connected BioMart database;</span><span class="w">
</span><span class="c1"># 'ensembl_gene_id' are genes IDs,</span><span class="w">
</span><span class="c1"># 'external_gene_name' are the genes symbols that we want to get for our values stored in ‘ensembl.ids’.</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the resulting data</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span><span class="w">                          
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># replace IDs for gene names</span><span class="w">
</span><span class="n">gene_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ensembl.ids</span><span class="w">	 
</span><span class="n">count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="w"> 	 
</span><span class="k">for</span><span class="w"> </span><span class="p">(</span><span class="n">geneID</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="n">gene_names</span><span class="p">)</span><span class="w">
</span><span class="p">{</span><span class="w">
 </span><span class="n">index</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">genes</span><span class="o">==</span><span class="n">geneID</span><span class="p">)</span><span class="w">    </span><span class="c1"># finds an index of geneID in the genes object created by getBM()</span><span class="w">
 </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">index</span><span class="p">)</span><span class="o">==</span><span class="m">0</span><span class="p">)</span><span class="w">            </span><span class="c1"># condition in case if there is no corresponding gene name in the chosen dataset</span><span class="w">
  </span><span class="p">{</span><span class="w">
    </span><span class="n">gene_names</span><span class="p">[</span><span class="n">count</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'NA'</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="k">else</span><span class="w">
  </span><span class="p">{</span><span class="w">
    </span><span class="n">gene_names</span><span class="p">[</span><span class="n">count</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genes</span><span class="o">$</span><span class="n">external_gene_name</span><span class="p">[</span><span class="n">index</span><span class="p">]</span><span class="w"> 	</span><span class="c1"># replaces gene ID by the corresponding gene name based on the found geneID’s index</span><span class="w">
  </span><span class="p">}</span><span class="w">
 </span><span class="n">count</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">count</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1</span><span class="w">                </span><span class="c1"># increased count so that every element in gene_names is replaced</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># store the gene names in our CDS object in a new column gene_short_name_extra</span><span class="w">
</span><span class="n">fData</span><span class="p">(</span><span class="n">cds_extra</span><span class="p">)</span><span class="o">$</span><span class="n">gene_short_name_extra</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gene_names</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the changes</span><span class="w">
</span><span class="n">fData</span><span class="p">(</span><span class="n">cds_extra</span><span class="p">)</span><span class="w">                    
</span></code></pre></div></div>

<p>If you are working on your own data and it’s not mouse data, you can check available datasets for other species and just use relevant dataset in <code class="language-plaintext highlighter-rouge">useMart()</code> function.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">listDatasets</span><span class="p">(</span><span class="n">mart</span><span class="p">)</span><span class="w">                </span><span class="c1"># available datasets</span><span class="w">
</span></code></pre></div></div>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-ensembl-connection-problems-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Ensembl connection problems</div>
  <p>Sometimes you may encounter some connection issues with Ensembl. To improve performance Ensembl provides several mirrors of their site distributed around the globe. When you use the default settings for useEnsembl() your queries will be directed to your closest mirror geographically. In theory this should give you the best performance, however this is not always the case in practice. For example, if the nearest mirror is experiencing many queries from other users it may perform poorly for you. In such cases, the other mirrors should be chosen automatically.</p>

</blockquote>

<h1 id="monocle-workflow">Monocle workflow</h1>
<p>Do you remember the Monocle workflow introduced in the previous tutorial? Here is a recap:
<a href="../../images/scrna-casestudy-monocle/monocle3_new_workflow.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/monocle3_new_workflow.png" alt="Monocle workflow: scRNA-seq dataset, pre-process data (normalise, remove batch effects), non-linear dimensionality reduction (t-SNE, UMAP), cluster cells, compare clusters (identify top markers, targeted contrasts), trajectory analysis. " width="8000" height="1374" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Workflow provided by Monocle3 documentation&lt;/figcaption&gt;</figure></a></p>

<h2 id="pre-processing">Pre-processing</h2>
<p>Let’s start with normalisation and pre-processing that can be performed using the function <code class="language-plaintext highlighter-rouge">preprocess_cds()</code>. The argument <code class="language-plaintext highlighter-rouge">num_dim</code> is the number of principal components that will be used. You can check that you’re using enough PCs to capture most of the variation in gene expression across all the cells in the data set. Note that “PCA” is the default method of pre-processing in Monocle3, so although we can specify this in our function, we don’t have to.</p>

<p>Note that this step can take awhile - around 5 or so minutes due to the high number of PCs calculated. Feel free to use a <code class="language-plaintext highlighter-rouge">num_dim</code> value around 100 to accelerate preprocessing and compare the results.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># PCA pre-processing with 210 principal components</span><span class="w">
</span><span class="n">cds_preprocessing</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">preprocess_cds</span><span class="p">(</span><span class="n">cds</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"PCA"</span><span class="p">,</span><span class="w"> </span><span class="n">num_dim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">210</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="c1"># plot the variation in gene expression vs PCA components</span><span class="w">
</span><span class="n">plot_pc_variance_explained</span><span class="p">(</span><span class="n">cds_preprocessing</span><span class="p">)</span><span class="w">                              		                            
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/pca_plot.jpg" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/pca_plot.jpg" alt="Plot of variation in gene expression vs PCA components, decreasing exponentially." width="859" height="543" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span>  Plot of variation in gene expression vs PCA components.&lt;/figcaption&gt;</figure></a></p>

<p>The plot shows that actually using more than ~100 PCs captures only a small amount of additional variation. However, if we look at how the cells are plotted on 2D graph when using different values of PCs, it is easier to visualise how the <code class="language-plaintext highlighter-rouge">num_dim</code> actually affects the output. We will use the value of 210, which, compared to the results from the previous tutorial, makes the most sense for our dataset.</p>

<p><a href="../../images/scrna-casestudy-monocle/num_dim.jpg" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/num_dim.jpg" alt="Six plots showing only the shaded shape of how the cells are clustered depending on the num_dim argument. The general trend is maintained though." width="1768" height="1784" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> The “shape” of the plot showing how the cells are clustered depending on the ‘num_dim’ argument.&lt;/figcaption&gt;</figure></a></p>

<h2 id="batch-correction-and-dimensionality-reduction">Batch correction and Dimensionality reduction</h2>
<p>Our dataset actually comprises data from 7 samples, so there is a risk that batch effects will impact analysis. Batch effects are systematic differences in the transcriptome of cells measured in different experimental batches. However, we can use Monocle to deal with that!
First, let’s check how our dataset looks in terms of batch effects. We can do that by colouring the cells by batch.  This information is stored in our CDS object from the <code class="language-plaintext highlighter-rouge">cell_metadata</code> file. Before asking Monocle to plot anything, let’s check the exact column name of the batch information column.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">colnames</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">cds_preprocessing</span><span class="p">))</span><span class="w"> 	          </span><span class="c1"># check column names</span><span class="w">
</span></code></pre></div></div>
<p>In our case it’s indeed ‘batch’, but your data might have another name (eg. “plate”, etc.), so make sure you put the correct argument value.</p>

<p>In order to plot our cells on a 2D graph, we need to reduce the numbers of dimensions. The <a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">previous tutorial</a> introduced the methods of dimensionality reduction in Monocle. You can replicate what we did in Galaxy to compare the output of dimensionality reduction using different methods, simply by changing the <code class="language-plaintext highlighter-rouge">reduction_method</code> argument. Options currently supported by Monocle are “UMAP”, “tSNE”, “PCA”, “LSI”, and “Aligned”. However, as for now, let’s just recall that UMAP gave the best results, so we will use UMAP here as well.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># reduce dimension first</span><span class="w">
</span><span class="n">cds_preprocessing_UMAP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reduce_dimension</span><span class="p">(</span><span class="n">cds_preprocessing</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">preprocess_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"PCA"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># then produce a plot to check for batch effects</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_preprocessing_UMAP</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"batch"</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>We can see that upper and lower right branches mostly consist of N705 and N706, so indeed batch correction might be helpful. Let’s run this.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># perform batch correction</span><span class="w">
</span><span class="n">cds_batch</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">align_cds</span><span class="p">(</span><span class="n">cds_preprocessing_UMAP</span><span class="p">,</span><span class="w"> </span><span class="n">preprocess_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"PCA"</span><span class="p">,</span><span class="w"> </span><span class="n">alignment_group</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"batch"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>To see the changes, we have to run UMAP again, but this time on the aligned dataset. We will specify that <code class="language-plaintext highlighter-rouge">preprocess_method </code> as “Aligned” and not “PCA”. Monocle would use the “Aligned” argument automatically if no <code class="language-plaintext highlighter-rouge">preprocess_method</code> was specified.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># dimensionality reduction after alignment</span><span class="w">
</span><span class="n">cds_red_dim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reduce_dimension</span><span class="p">(</span><span class="n">cds_batch</span><span class="p">,</span><span class="w"> </span><span class="n">preprocess_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Aligned"</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">)</span><span class="w">  
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the batch correction effect on a plot</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_red_dim</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"batch"</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">		
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/batch_correction.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/batch_correction.png" alt="Left image showing dataset before batch correction: upper and lower right branches mostly consist of N705 and N706. Right image showing the dataset after batch correction: the cells from all the samples are evenly spread throughout the whole dataset." width="1383" height="590" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Comparison of the dataset before and after batch correction.&lt;/figcaption&gt;</figure></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Does your plot look the same as the one in the Figure?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Your plot might be slightly different to the one shown in the Figure but this is fine, as long as you see analogical patterns. Some libraries that you’re using might have been updated, giving non-identical output. However, the principle behind the analysis is still the same, so you can peacefully follow the tutorial. Just keep your eyes open and… THINK!</p>

  </blockquote>

</blockquote>

<p>Do you see this? It’s amazing! Batch correction did a great job here! Now the dataset is nicely aligned, and the cells from all the samples are evenly spread throughout the whole dataset. It is worth mentioning that removing batch effects was done using mutual nearest neighbor alignment, a technique introduced by John Marioni’s lab (<span class="citation"><a href="#Haghverdi_2018">Haghverdi <i>et al.</i> 2018</a></span>) and supported by Aaron Lun’s package <a href="https://bioconductor.org/packages/release/bioc/html/batchelor.html">batchelor</a>. Also, due to the machine learning elements of the code of this technique - as well as the fact that packages are updated regularly - your plots may not look identical to the ones pictured here. Nevertheless, the interpretation should be the same - the batch corrected plot should show better batch distribution than the uncorrected one.</p>

<p>Now we can move to the next step and perform dimensionality reduction.</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-plotting-labels-vs-legend-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-plotting-labels-vs-legend-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Plotting: labels vs legend<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>When creating graphs, we sometimes use labels and sometimes just a legend. You can choose whichever you think makes the data clear and readable.</p>

  <p>If you want to use a legend, then specify an argument <code class="language-plaintext highlighter-rouge">label_cell_groups=FALSE</code> in the function <code class="language-plaintext highlighter-rouge">plot_cells()</code>.</p>

  <p>The labels are set automatically, but if you want to change their size (default labels are tiny), use the argument <code class="language-plaintext highlighter-rouge">group_label_size</code>.</p>
</blockquote>

<h2 id="clustering-clusters">Clustering: clusters</h2>
<p>We want to get some information about cell types, don’t we? In order to do so, we have to cluster our cells first.
Monocle uses a technique called “community detection” (<span class="citation"><a href="#Traag_2019">Traag <i>et al.</i> 2019</a></span>) to group cells. This approach was introduced by <span class="citation"><a href="#Levine_2015">Levine <i>et al.</i> 2015</a></span> as part of the phenoGraph algorithm.
Monocle also divides the cells into larger, more well separated groups called partitions, using a statistical test from <span class="citation"><a href="#Wolf_2019">Wolf <i>et al.</i> 2019</a></span>, introduced as part of their <a href="https://github.com/theislab/paga">PAGA</a> algorithm.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-clusters-vs-partitions-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-clusters-vs-partitions-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Clusters vs partitions<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Clusters are particularly useful while trying to assign cells to a certain type, because they are based on the similarity in gene expression. The relationships between different clusters are analysed to identify possible trajectories.</p>

  <p>Partitions, meanwhile, are larger groups of cells that usually contain several clusters. Trajectory inference is performed only within one partition, so it is essential that all the cells that we want to analyse in pseudotime belong to the same partition.</p>

</blockquote>

<p>Therefore, let’s perform clustering and visualise the resulting clusters.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># clustering</span><span class="w">
</span><span class="n">cds_auto_cluster</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cluster_cells</span><span class="p">(</span><span class="n">cds_red_dim</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the clusters</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_auto_cluster</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cluster'</span><span class="p">,</span><span class="w"> </span><span class="n">group_label_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w">     
</span></code></pre></div></div>

<p>When using standard igraph louvain clustering, the value of resolution parameter is by default set to NULL, which means that it is determined automatically. Although the resulting clusters are OK, it would be nice to get some more granularity to identify cell types more specifically. The higher the resolution value, the more clusters we get. We will set the resolution value to 0.0002, but you are very welcome to try different values to see the changes.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># clustering with changed resolution value</span><span class="w">
</span><span class="n">cds_clustered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cluster_cells</span><span class="p">(</span><span class="n">cds_red_dim</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">resolution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.0002</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the new clusters</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cluster'</span><span class="p">,</span><span class="w"> </span><span class="n">group_label_size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span><span class="w"> 	
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/clusters_compare.png" rel="noopener noreferrer"><figure id="figure-7" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/clusters_compare.png" alt="Left image showing 6 clusters formed using automatic standard igraph louvain clustering. Right image showing the dataset with clusters formed using resolution argument set to 0.0002: now there are 7 clusters, as one automatically formed cluster could be divided into two smaller distinct clusters." width="1383" height="590" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Comparison of the clusters formed using standard igraph louvain clustering and using resolution argument set to 0.0002.&lt;/figcaption&gt;</figure></a></p>

<h2 id="clustering-partitions">Clustering: partitions</h2>
<p>OK, what about partitions? They were also created during the clustering step and it’s important to check them before calculating the trajectory because it is performed only within one partition. It is essential that all the cells that we want to analyse in pseudotime belong to the same partition.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the partitions</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'partition'</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>While your plot might be slightly different due to package updates, we can see that there are 3 partitions identified in <code class="language-plaintext highlighter-rouge">cds_clustered</code> object. Ideally, we would like to combine partitions 1 and 2 to draw a trajectory through all those cells (we can ignore cells in outlier partition). Sometimes using the default values might result in multiple partitions while you only need one. Then you would have to change the q-value cutoff in <code class="language-plaintext highlighter-rouge">partition_qval</code>. The default is 0.05 and by increasing this value you can increase the span of partitions, meaning that you would get fewer partitions. When trying different values of q-value, you also have to check if the clusters didn’t change. It’s all about finding a balance between the value of <code class="language-plaintext highlighter-rouge">resolution</code> and <code class="language-plaintext highlighter-rouge">partition_qval</code> so that both clusters and partitions are satisfactory enough for downstream analysis. Let’s try that on our dataset.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># changing the partition q-value</span><span class="w">
</span><span class="n">cds_clustered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cluster_cells</span><span class="p">(</span><span class="n">cds_red_dim</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">resolution</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.0002</span><span class="p">,</span><span class="w"> </span><span class="n">partition_qval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the partitions with the changed q-value</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'partition'</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># check if clusters didn't change</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cluster'</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Have the clusters change after changing the partition q-value?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>It might be the case that after changing partition q-value, you will notice that additional clusters appeared. In that situation, you might either play around the <code class="language-plaintext highlighter-rouge">resolution</code> and <code class="language-plaintext highlighter-rouge">partition_qval</code> values, go forward with the current clustering (adjusting the parameters accordingly), or check the other method of assigning cells to one partition given below.</p>

  </blockquote>

</blockquote>

<p>Now we have cells of interest in one partition, we still have reasonable clusters, so now we can learn the trajectory. However, in some cases even this method might not be enough. Then, there is a last resort… assigning cells to a partition manually.</p>

<h2 id="additional-step-assigning-cells-to-one-partition">Additional step: assigning cells to one partition</h2>
<blockquote class="warning">
  <div class="box-title warning-title" id="warning-additional-step-3"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Additional step</div>
  <p>This step is not necessary for the dataset we are working on but some users might find it helpful when analysing their own data.</p>

</blockquote>

<p>Let’s assume we have 4 partitions that cannot be extended to one big partition using <code class="language-plaintext highlighter-rouge">partition_qval</code> (it might sound unreasonable but it does happen!) and we are desperate to have all the cells in one partition to draw a trajectory through all of them. We can simulate the initial dataset by setting <code class="language-plaintext highlighter-rouge">partition_qval</code> value to 0.0001.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># simulate the dataset</span><span class="w">
</span><span class="n">cds_partitions_extra</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cluster_cells</span><span class="p">(</span><span class="n">cds_red_dim</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">partition_qval</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.0001</span><span class="p">)</span><span class="w">		
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the simulated partitions</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_partitions_extra</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'partition'</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># store ‘1’ the number of times equal to the number of cells</span><span class="w">
</span><span class="n">big_partition</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">cds_partitions_extra</span><span class="o">@</span><span class="n">colData</span><span class="o">@</span><span class="n">rownames</span><span class="p">)))</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># take the barcode of each cell and assign ‘1’ to each of them (now the ‘ones’ are named)</span><span class="w">
</span><span class="nf">names</span><span class="p">(</span><span class="n">big_partition</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds_partitions_extra</span><span class="o">@</span><span class="n">colData</span><span class="o">@</span><span class="n">rownames</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># convert from numeric to factor</span><span class="w">
</span><span class="n">big_partition</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.factor</span><span class="p">(</span><span class="n">big_partition</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># assign the created barcodes-partition list to the location where information about partitions is stored in CDS object</span><span class="w">
</span><span class="n">cds_partitions_extra</span><span class="o">@</span><span class="n">clusters</span><span class="o">$</span><span class="n">UMAP</span><span class="o">$</span><span class="n">partitions</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">big_partition</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the new partition assignment</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_partitions_extra</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'partition'</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/partitions_compare.png" rel="noopener noreferrer"><figure id="figure-8" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/partitions_compare.png" alt="Left image showing the dataset divided into 4 partitions. Right image showing all cells assigned to one partition." width="1383" height="590" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 8</strong>:</span> The result of the manual assignment of all the cells to one partition.&lt;/figcaption&gt;</figure></a></p>

<blockquote class="details">
  <div class="box-title details-title" id="details-and-operators"><button class="gtn-boxify-button details" type="button" aria-controls="details-and-operators" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: @ and $ operators<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>As you saw above, we used <code class="language-plaintext highlighter-rouge">@</code> and <code class="language-plaintext highlighter-rouge">$</code> operators to navigate in CDS object. What is the difference between them? <code class="language-plaintext highlighter-rouge">$</code> operator is used to access one variable/column, while <code class="language-plaintext highlighter-rouge">@</code> extracts the contents of a slot in an object with a formal (S4) class structure. If you use <code class="language-plaintext highlighter-rouge">View()</code> function on our CDS object, you will see the elements of type “S4” that you can access using <code class="language-plaintext highlighter-rouge">@</code> operator. The suggestion list of elements that can be accessed should also pop up as you type the name of the object followed by the corresponding operator.</p>

</blockquote>

<h1 id="assigning-cell-types">Assigning cell types</h1>
<p>There are two main approaches to assigning cell types to clusters that we’ve just identified – supervised and unsupervised, both based on gene expression in each cluster.</p>

<h2 id="supervised-approach">Supervised approach</h2>
<p>The supervised approach relies on prior knowledge of which cell types to expect. We can simply check the expression of marker genes specific to the expected cell types. Let’s then check the markers mentioned in the original paper <span class="citation"><a href="#Bacon2018">Bacon <i>et al.</i> 2018</a></span>.</p>

<table>
  <thead>
    <tr>
      <th>Marker</th>
      <th>Cell type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Il2ra</td>
      <td>Double negative (early T-cell)</td>
    </tr>
    <tr>
      <td>Cd8b1, Cd8a, Cd4</td>
      <td>Double positive (middle T-cell)</td>
    </tr>
    <tr>
      <td>Cd8b1, Cd8a, Cd4 - high</td>
      <td>Double positive (late middle T-cell)</td>
    </tr>
    <tr>
      <td>Itm2a</td>
      <td>Mature T-cell</td>
    </tr>
    <tr>
      <td>Aif1</td>
      <td>Macrophages</td>
    </tr>
    <tr>
      <td>Hba-a1</td>
      <td>RBC</td>
    </tr>
  </tbody>
</table>

<p>To plot the expression of all those genes across our dataset in one go, we will pass a vector with the names of those markers into a parameter <code class="language-plaintext highlighter-rouge">genes</code> in the <code class="language-plaintext highlighter-rouge">plot_cells</code> function.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># expression of marker genes across the sample</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">genes</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'Il2ra'</span><span class="p">,</span><span class="s1">'Cd8b1'</span><span class="p">,</span><span class="s1">'Cd8a'</span><span class="p">,</span><span class="s1">'Cd4'</span><span class="p">,</span><span class="s1">'Itm2a'</span><span class="p">,</span><span class="s1">'Aif1'</span><span class="p">,</span><span class="s1">'Hba-a1'</span><span class="p">),</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/genes.png" rel="noopener noreferrer"><figure id="figure-9" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/genes.png" alt="7 graphs, each showing expression of one of the genes in the corresponding clusters: 'Il2ra' – cluster 4; 'Cd8b1' and 'Cd8a' – clusters 1,6,2; 'Cd4' – high expression in cluster 5; 'Itm2a' – cluster 3; 'Aif1' – barely anything, no specific cluster assigned; 'Hba-a1' - throughout the entire sample in low numbers." width="700" height="540" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 9</strong>:</span> Expression of the reference marker genes across analysed sample.&lt;/figcaption&gt;</figure></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-genes-cell-types-and-clusters-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Genes, cell types and clusters</div>
  <p>Based on the gene expression graph that we just generated, the table above and your knowledge from the previous tutorials, how would you assign the clusters?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Keep in mind that these results refer to our numbered clusters, while yours might be slightly different.</p>
    <ul>
      <li><code class="language-plaintext highlighter-rouge">Il2ra</code> (DN): mostly expressed in cluster 4</li>
      <li><code class="language-plaintext highlighter-rouge">Cd8b1, Cd8a</code> (DP middle): expressed in clusters 1, 6, and highly in cluster 2</li>
      <li><code class="language-plaintext highlighter-rouge">Cd4</code> (DP late): average expression in clusters 1, 6, 2 and high expression in cluster 5</li>
      <li><code class="language-plaintext highlighter-rouge">Itm2a</code> (T-mat): expressed in cluster 3</li>
      <li><code class="language-plaintext highlighter-rouge">Aif1</code> (macrophages): barely anything here, minimal expression spread across the sample with some more cells in cluster 4 and 3 – not enough to form a distinct cluster though). In theory, we shouldn’t have any macrophages in our sample. If you remember from the previous tutorials, we actually filtered out macrophages from the sample during the processing step, because we worked on annotated data. When analysing unannotated data, we could only assign macrophages and then filter them out, provided that Monocle clusters them into a separate group. As you can see, it’s not the case here, so we will just carry on with the analysis, interpreting this as a contamination.</li>
      <li><code class="language-plaintext highlighter-rouge">Hba-a1</code> (RBC): appears throughout the entire sample in low numbers suggesting some background contamination of red blood cell debris in the cell samples during library generation, but also shows higher expression in a distinct tiny bit of cluster 3, at the border between clusters 1 and 5. However, it’s too small to be clustered into a separate group and filtered out in this case.
If you remember, this gene was found to be expressed in the previous Scanpy tutorial also in low numbers across the sample, and in the other Monocle tutorial (using Galaxy tools and annotated data) algorithms allowed us to gather the cells expressing that gene into a distinct group. Our result now sits somewhere in between.
<a href="../../images/scrna-casestudy-monocle/hb_all.png" rel="noopener noreferrer"><figure id="figure-10" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/hb_all.png" alt="In Scanpy graph the marker gene appears throughout the entire sample in low numbers, in Monocle in Galaxy cells expressing hemoglobin gene were grouped into a small branch of DP-M4, allowing to group those cells. Monocle in RStudio graph is somewhere in between showing mostly low expression across the sample, but also having a tiny bit of grouped cells, less distinct than in Galaxy though." width="1548" height="591" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 10</strong>:</span> Hemoglobin across clusters - comparison between Scanpy, Monocle using Galaxy tools and Monocle run in RStudio.&lt;/figcaption&gt;</figure></a></li>
    </ul>

  </blockquote>
</blockquote>

<p>Having identified which cluster corresponds to a specific cell type, we can finally run some code to add the annotation to our CDS object. First, we will create a new column called <code class="language-plaintext highlighter-rouge">cell_type</code>  in <code class="language-plaintext highlighter-rouge">colData()</code> - this is where the information about the cells is stored (eg. batch, genotype, sex, etc) - and initialize it with the values of clusters. Then, we will get the <code class="language-plaintext highlighter-rouge">dplyr</code> package which will be used for cluster annotation.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># just to keep the objects tidy and not overwrite them so that you can come back to any point of the analysis</span><span class="w">
</span><span class="n">cds_annotated</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds_clustered</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create a new column ‘cell_type’ and initialise it with clusters values</span><span class="w">
</span><span class="n">colData</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">)</span><span class="o">$</span><span class="n">cell_type</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">clusters</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">))</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># annotate clusters</span><span class="w">
</span><span class="n">colData</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">)</span><span class="o">$</span><span class="n">cell_type</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">recode</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">)</span><span class="o">$</span><span class="n">cell_type</span><span class="p">,</span><span class="w">
                                                       </span><span class="s1">'1'</span><span class="o">=</span><span class="s1">'DP-M1'</span><span class="p">,</span><span class="w">   </span><span class="c1"># double positive – middle T-cell (1st cluster)</span><span class="w">
                                                       </span><span class="s1">'2'</span><span class="o">=</span><span class="s1">'DP-M2'</span><span class="p">,</span><span class="w">	  </span><span class="c1"># double positive – middle T-cell (2nd cluster)</span><span class="w">
                                                       </span><span class="s1">'3'</span><span class="o">=</span><span class="s1">'T-mat'</span><span class="p">,</span><span class="w">	  </span><span class="c1"># mature T-cell</span><span class="w">
                                                       </span><span class="s1">'4'</span><span class="o">=</span><span class="s1">'DN'</span><span class="p">,</span><span class="w">		  </span><span class="c1"># double negative – early T-cell</span><span class="w">
                                                       </span><span class="s1">'5'</span><span class="o">=</span><span class="s1">'DP-L'</span><span class="p">,</span><span class="w">	  </span><span class="c1"># double positive – late middle T-cell</span><span class="w">
                                                       </span><span class="s1">'6'</span><span class="o">=</span><span class="s1">'DP-M3'</span><span class="p">,</span><span class="w">	  </span><span class="c1"># double positive – middle T-cell (3rd cluster)</span><span class="w">
                                                       </span><span class="s1">'7'</span><span class="o">=</span><span class="s1">'Unknown'</span><span class="p">)</span><span class="w"> </span><span class="c1"># no info for now, so call it ‘Unknown’</span><span class="w">
                                                       </span><span class="s1">'8'</span><span class="o">=</span><span class="s1">'Unknown'</span><span class="p">)</span><span class="w"> </span><span class="c1"># no info for now, so call it ‘Unknown’</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># check the annotation</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w"> </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">    
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/annotated.png" rel="noopener noreferrer"><figure id="figure-11" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/annotated.png" alt="A plot showing the identified clusters, now each coloured by the assigned cell type." width="600" height="540" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 11</strong>:</span> Our annotated dataset.&lt;/figcaption&gt;</figure></a></p>

<h2 id="unsupervised-approach">Unsupervised approach</h2>
<p>But what if we don’t have any reference that we can use to assign our clusters? In that case, we will turn to the mentioned unsupervised approach - we will check what are the specifically expressed genes for each cluster. Then we can identify the cell types by looking up what cell types contain those genes. That’s a more tedious process, but sometimes can lead to exciting and unexpected results.
We will use Monocle’s function <code class="language-plaintext highlighter-rouge">top_markers()</code> and store the information about specifically expressed genes for each cluster in the data frame <code class="language-plaintext highlighter-rouge">marker_test</code>.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># find top marker genes in each cluster</span><span class="w">
</span><span class="n">marker_test</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">top_markers</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">group_cells_by</span><span class="o">=</span><span class="s2">"cluster"</span><span class="p">,</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">,</span><span class="w"> </span><span class="n">reference_cells</span><span class="o">=</span><span class="m">1000</span><span class="p">,</span><span class="w"> </span><span class="n">cores</span><span class="o">=</span><span class="m">8</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>You can group the cells by any categorical variable in <code class="language-plaintext highlighter-rouge">colData(cds_clustered)</code>. The parameter <code class="language-plaintext highlighter-rouge">reference_cells</code> is used to accelerate the marker significance test at some cost in sensitivity. It works by randomly selecting a specified number of cells and performing the marker significance test against a chosen set of cells. If your dataset is not massively big, you might skip this parameter as it wouldn’t help much.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What are the variables stored in <code class="language-plaintext highlighter-rouge">marker_test</code> data frame and what do they mean?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Those are 10 variables that you can check easily either using colnames(marker_test) or <code class="language-plaintext highlighter-rouge">View(marker_test)</code> which will also display all the corresponding values.</p>
    <ul>
      <li><code class="language-plaintext highlighter-rouge">gene_id</code> - Ensembl gene ID</li>
      <li><code class="language-plaintext highlighter-rouge">gene_short_name</code> - short name of the gene corresponding to its ID</li>
      <li><code class="language-plaintext highlighter-rouge">cell_group</code> - a group to which the cell belongs, specified in the <code class="language-plaintext highlighter-rouge">group_cells_by</code> argument</li>
      <li><code class="language-plaintext highlighter-rouge">marker_score</code> - numeric vector of marker scores as the fraction expressing scaled by the specificity. The value ranges from 0 to 1</li>
      <li><code class="language-plaintext highlighter-rouge">mean_expression</code> -  numeric vector of mean normalized expression of the gene in the cell group</li>
      <li><code class="language-plaintext highlighter-rouge">fraction_expressing</code> - numeric vector of fraction of cells expressing the gene within the cell group</li>
      <li><code class="language-plaintext highlighter-rouge">specificity</code> - numeric vector of a measure of how specific the gene’s expression is to the cell group based on the Jensen-Shannon divergence. The value ranges from 0 to 1.</li>
      <li><code class="language-plaintext highlighter-rouge">pseudo_R2</code> - numeric vector of pseudo R-squared values, a measure of how well the gene expression model fits the categorical data relative to the null model. The value ranges from 0 to 1.</li>
      <li><code class="language-plaintext highlighter-rouge">marker_test_p_value</code> - numeric vector of likelihood ratio p-values; p-value is an area in the tail of a distribution that tells you the odds of a result happening by chance</li>
      <li><code class="language-plaintext highlighter-rouge">marker_test_q_value</code> -  numeric vector of likelihood ratio q-values; q-value is a p-value that has been adjusted for the False Discovery Rate (FDR)</li>
    </ul>

  </blockquote>

</blockquote>

<p>We can now use data in <code class="language-plaintext highlighter-rouge">marker_test</code> to rank the cells based on one of the specificity metrics and take the top gene(s) for each cluster. We will filter the expressing cells that constitute more than 10% of the cell group and we will take one gene in each cluster with the highest <code class="language-plaintext highlighter-rouge">pseudo_R2</code> value (you can of course modify this value and choose more genes to be selected).</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># filter the ‘marker_test’ data frame</span><span class="w">
</span><span class="n">top_specific_markers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">marker_test</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                            </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">fraction_expressing</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.10</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">       </span><span class="c1"># set the fraction of expressing cells</span><span class="w">
                            </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">cell_group</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">                      </span><span class="c1"># set a group to which the cells belong</span><span class="w">
                            </span><span class="n">dplyr</span><span class="o">::</span><span class="n">top_n</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">pseudo_R2</span><span class="p">)</span><span class="w">                           </span><span class="c1"># set the number of top genes and the variable from 'marker_test' to rank by</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># store the names of the marker genes</span><span class="w">
</span><span class="c1"># you can also use IDs, the conversion to gene names should happen automatically when plotting</span><span class="w">
</span><span class="n">top_marker_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">top_specific_markers</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">pull</span><span class="p">(</span><span class="n">gene_short_name</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>

<p>Now we have all elements to plot the expression and fraction of cells that express found markers in each group.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plot_genes_by_group</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w">                    </span><span class="c1"># our CDS object</span><span class="w">
                    </span><span class="n">top_marker_names</span><span class="p">,</span><span class="w">                 </span><span class="c1"># a list of gene names to show in the plot</span><span class="w">
                    </span><span class="n">group_cells_by</span><span class="o">=</span><span class="s2">"cluster"</span><span class="p">,</span><span class="w">         </span><span class="c1"># how to group cells when labeling</span><span class="w">
                    </span><span class="n">ordering_type</span><span class="o">=</span><span class="s2">"maximal_on_diag"</span><span class="p">)</span><span class="w">  </span><span class="c1"># how to order the genes / groups on the dot plot</span><span class="w">
</span></code></pre></div></div>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-unexpectedly-too-many-genes-on-y-axis-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-unexpectedly-too-many-genes-on-y-axis-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Unexpectedly too many genes on y-axis?<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you notice that on your dot plot any cluster has more genes than you specified (here we set one gene per cluster <code class="language-plaintext highlighter-rouge">top_n(1, pseudo_R2)</code>), go back to the code and create a new cell, right after assigning <code class="language-plaintext highlighter-rouge">top_specific_markers</code> object and just before <code class="language-plaintext highlighter-rouge">top_marker_names</code>. In the new cell, paste and execute the following:</p>

  <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">top_specific_markers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">top_specific_markers</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                           </span><span class="n">dplyr</span><span class="o">::</span><span class="n">distinct</span><span class="p">(</span><span class="n">pseudo_R2</span><span class="p">,</span><span class="w"> </span><span class="n">.keep_all</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">     </span><span class="c1"># select only one row if there are multiple rows with the same value in 'pseudo_R2' column</span><span class="w">
</span></code></pre></div>  </div>
  <p>Then you can execute <code class="language-plaintext highlighter-rouge">top_marker_names</code> again and see how the plot looks like now. Better, right?
This problem may arise when several genes in one cluster have the same values of specific variable from ‘marker_test’ (in our case we chose <code class="language-plaintext highlighter-rouge">pseudo_R2</code>). It might likely happen in small and quite unsignificant clusters.</p>
</blockquote>

<p><a href="../../images/scrna-casestudy-monocle/gene_group.png" rel="noopener noreferrer"><figure id="figure-12" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/gene_group.png" alt="A dot plot showing the expression of genes and fraction of cells that express found markers in each group. On the diagonal there are two genes corresponding to each cluster with the highest pseudo_R2 score in their group." width="700" height="550" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 12</strong>:</span> A plot of the expression and fraction of cells that express found markers in each group. Here, the two top genes from each cluster were plotted on diagonal beased on the highest pseudo_R2 score. Shown is the result of changing the value from 1 to 2 in “top_n(1, pseudo_R2)” in the code above.&lt;/figcaption&gt;</figure></a></p>

<p>Look at this – we have identified some more marker genes specific to each cluster! However, sometimes it happens that the found genes are not as specific as one would expect, and they appear across the whole sample. Therefore, it is a good idea to plot all those marker genes and check how they appear in the bigger picture.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># plot the identified genes to check their expression</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">,</span><span class="w"> </span><span class="n">genes</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'Pcdhgc4'</span><span class="p">,</span><span class="s1">'Pcdhga5'</span><span class="p">,</span><span class="s1">'Gm5559'</span><span class="p">,</span><span class="s1">'Gm10359'</span><span class="p">,</span><span class="s1">'Ccr9'</span><span class="p">,</span><span class="s1">'Cd8b1'</span><span class="p">,</span><span class="s1">'Plac8'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Il2ra'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Cd52'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Tmsb10'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Mki67'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Hmgb2'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Pclaf'</span><span class="p">,</span><span class="w"> </span><span class="s1">'Npm1'</span><span class="p">),</span><span class="w"> </span><span class="n">reduction_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"UMAP"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/markers_plot.png" rel="noopener noreferrer"><figure id="figure-13" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/markers_plot.png" alt="Plots showing the expression of fourteen found marker genes across the whole sample. Those genes are 'Pcdhgc4','Pcdhga5','Gm5559','Gm10359','Ccr9','Cd8b1','Plac8', 'Il2ra', 'Cd52', 'Tmsb10', 'Mki67', 'Hmgb2', 'Pclaf', 'Npm1'." width="900" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 13</strong>:</span> Plots of the expression of marker genes across the sample.&lt;/figcaption&gt;</figure></a></p>

<p>Further steps from now would include reviewing literature and checking what cell types correspond to the genes expressed in each cluster. Then you can annotate your clusters in the same way as shown above. Once you have your clusters annotated, Monocle can generate a file of marker genes for the identified cell types. This file can be then used with <a href="https://cole-trapnell-lab.github.io/garnett/">Garnett</a>, a software toolkit for automatically annotating cells. We will not go through this in the current tutorial, but we will generate the file of marker genes.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># use ‘top_markers()’ again, now grouping cells by the assigned cell type</span><span class="w">
</span><span class="n">assigned_marker_test</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">top_markers</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">,</span><span class="w">
                                             </span><span class="n">group_cells_by</span><span class="o">=</span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w">
                                             </span><span class="n">reference_cells</span><span class="o">=</span><span class="m">1000</span><span class="p">,</span><span class="w">
                                             </span><span class="n">cores</span><span class="o">=</span><span class="m">8</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># filter these markers according to how stringent you want to be</span><span class="w">
</span><span class="n">garnett_markers</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">assigned_marker_test</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                        </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">marker_test_q_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">specificity</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.25</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                        </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">cell_group</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                        </span><span class="n">dplyr</span><span class="o">::</span><span class="n">top_n</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="n">marker_score</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># exclude genes that are good markers for more than one cell type:</span><span class="w">
</span><span class="n">garnett_markers_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">garnett_markers</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                        </span><span class="n">dplyr</span><span class="o">::</span><span class="n">group_by</span><span class="p">(</span><span class="n">gene_short_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
                        </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">dplyr</span><span class="o">::</span><span class="n">n</span><span class="p">()</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># generate a file of marker genes</span><span class="w">
</span><span class="n">generate_garnett_marker_file</span><span class="p">(</span><span class="n">garnett_markers_filtered</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s2">"./marker_file.txt"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>A new file should appear in the ‘Files’ window. If you click on it, you will see the cell types and their corresponding marker genes, satisfying your chosen conditions.</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-marker-genes-file-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-marker-genes-file-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Marker genes file<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Note that you can use the above block of code to generate file with the marker genes for unannotated CDS object to help you identify and check specifically expressed genes – you’d only have to change <code class="language-plaintext highlighter-rouge">group_cells_by</code> parameter from “cell_type” to “cluster”.</p>
</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-only-for-users-of-local-rstudio-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-only-for-users-of-local-rstudio-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Only for users of local RStudio<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you are working in RStudio locally, you might want to try a great function <code class="language-plaintext highlighter-rouge">choose_cells()</code>, which allows you to make a subset of the CDS object containing only cells of interest to investigate certain clusters more in-depth. It only works in interactive mode, so can be only used locally - when you call it, then a pop-up window will appear where you can choose cells to subset.</p>
  <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create a CDS subset</span><span class="w">
</span><span class="n">cds_subset</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">choose_cells</span><span class="p">(</span><span class="n">cds_clustered</span><span class="p">)</span><span class="w">
</span></code></pre></div>  </div>
  <p>Now the chosen cluster is stored as a separate CDS object and you can analyse it independently, using the methods described above.</p>

</blockquote>

<h1 id="trajectory-inference">Trajectory inference</h1>
<p>It’s time to perform trajectory analysis! First, let’s learn the trajectory graph. With an argument <code class="language-plaintext highlighter-rouge">use_partition</code> we can specify if we want to learn a disjoint graph (value TRUE - default) in each partition, or to learn a single graph (value FALSE) across all partitions. The thing is, we can visualise the cells in pseudotime only if they belong to one partition. This is why it is important to make sure that all the cells that you want to analyse in pseudotime belong to one partition.</p>

<p>In one of the previous sections we’ve already prepared the partitions for trajectory inference - we assigned all the cells of interest into one partition by changing the q-value. As a result, we got two partitions – one containing only the cells that we labeled as ‘Unknown’ and another including all other cells. That is a perfect assignment – we’d like to focus only on maturation of T-cells, so we don’t want to connect those two partitions (therefore we specify <code class="language-plaintext highlighter-rouge">use_partition=TRUE</code> to get a disjoint graph) and we will work downstream only on the graph going through annotated T-cells.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># learn trajectory graph</span><span class="w">
</span><span class="n">cds_trajectory</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">learn_graph</span><span class="p">(</span><span class="n">cds_annotated</span><span class="p">,</span><span class="w"> </span><span class="n">use_partition</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># visualise the learned trajectory</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w">
           </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w">
           </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_groups_by_cluster</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_leaves</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_branch_points</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/learned_trajectory_r.png" rel="noopener noreferrer"><figure id="figure-14" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/learned_trajectory_r.png" alt="Learned trajectory graph connecting all the annotated clusters except the ‘Unknown’ group of cells." width="650" height="540" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 14</strong>:</span> Learned trajectory graph.&lt;/figcaption&gt;</figure></a></p>

<p>We have to tell Monocle where to start ordering the cells, ie. when we expect the analysed biological process to begin. Thanks to our biological knowledge, we know that the beginning of the trajectory should be at DN cluster.
There are a couple of ways to specify the root cells:</p>

<h2 id="option-1-root-nodes">Option 1: Root nodes</h2>
<p>Here, you will use <code class="language-plaintext highlighter-rouge">root_pr_nodes</code> argument in <code class="language-plaintext highlighter-rouge">order_cells()</code> function.</p>

<p>To find the names of the principal points, you have to plot the learned trajectory again, specifying <code class="language-plaintext highlighter-rouge">label_principal_points = TRUE</code></p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># specifying root cells: `root_pr_nodes` argument - check the principal points</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w">
          </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w">
          </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
          </span><span class="n">label_groups_by_cluster</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
          </span><span class="n">label_leaves</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
          </span><span class="n">label_branch_points</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
          </span><span class="n">label_principal_points</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w">       </span><span class="c1"># set this to TRUE</span><span class="w">
          </span><span class="n">graph_label_size</span><span class="o">=</span><span class="m">3</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>You can see now the principal points and their labels in the form <code class="language-plaintext highlighter-rouge">Y_number</code>. Pick the principal point in the cluster that you expect to be the beginning of the trajectory and type its name in the <code class="language-plaintext highlighter-rouge">root_pr_nodes</code> argument when calling <code class="language-plaintext highlighter-rouge">order_cells()</code> function.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># specifying root cells: `root_pr_nodes` argument - use the relevant principal point</span><span class="w">
</span><span class="n">cds_order_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">order_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w"> </span><span class="n">root_pr_nodes</span><span class="o">=</span><span class="s1">'Y_14'</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>There is also a helper function to identify the root principal points based on the annotated cell types. This function uses <code class="language-plaintext highlighter-rouge">pr_graph_cell_proj_closest_vertex</code> which is just a matrix with a single column that stores for each cell, the ID of the principal graph node it’s closest to.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># a helper function to identify the root principal points</span><span class="w">
</span><span class="n">get_correct_root_state</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">cds</span><span class="p">,</span><span class="w"> </span><span class="n">cell_phenotype</span><span class="p">,</span><span class="w"> </span><span class="n">root_type</span><span class="p">){</span><span class="w">
      </span><span class="n">cell_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">pData</span><span class="p">(</span><span class="n">cds</span><span class="p">)[,</span><span class="w"> </span><span class="n">cell_phenotype</span><span class="p">]</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">root_type</span><span class="p">)</span><span class="w">

      </span><span class="n">closest_vertex</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds</span><span class="o">@</span><span class="n">principal_graph_aux</span><span class="p">[[</span><span class="s2">"UMAP"</span><span class="p">]]</span><span class="o">$</span><span class="n">pr_graph_cell_proj_closest_vertex</span><span class="w">
      </span><span class="n">closest_vertex</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.matrix</span><span class="p">(</span><span class="n">closest_vertex</span><span class="p">[</span><span class="n">colnames</span><span class="p">(</span><span class="n">cds</span><span class="p">),</span><span class="w"> </span><span class="p">])</span><span class="w">
      </span><span class="n">root_pr_nodes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">igraph</span><span class="o">::</span><span class="n">V</span><span class="p">(</span><span class="n">principal_graph</span><span class="p">(</span><span class="n">cds</span><span class="p">)[[</span><span class="s2">"UMAP"</span><span class="p">]])</span><span class="o">$</span><span class="n">name</span><span class="p">[</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">names</span><span class="w">
                                                                    </span><span class="p">(</span><span class="n">which.max</span><span class="p">(</span><span class="n">table</span><span class="p">(</span><span class="n">closest_vertex</span><span class="p">[</span><span class="n">cell_ids</span><span class="p">,]))))]</span><span class="w">       
      </span><span class="n">root_pr_nodes</span><span class="w">
      </span><span class="p">}</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># call the function to automatically find the node in the principal graph where our DN cells reside</span><span class="w">
</span><span class="n">DN_node_id</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">get_correct_root_state</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w"> </span><span class="n">cell_phenotype</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'cell_type'</span><span class="p">,</span><span class="w"> </span><span class="s2">"DN"</span><span class="p">)</span><span class="w">

</span><span class="n">DN_node_id</span><span class="w">      </span><span class="c1"># check the node found</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># order cells using the helper function output</span><span class="w">
</span><span class="n">cds_order_1_helper</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">order_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w"> </span><span class="n">root_pr_nodes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">DN_node_id</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h2 id="option-2-root-cells">Option 2: Root cells</h2>
<p>Here, you will use <code class="language-plaintext highlighter-rouge">root_cells</code> argument in <code class="language-plaintext highlighter-rouge">order_cells()</code> function.</p>

<p>Specify a vector of starting cell IDs. You can provide only one cell as well as all cells of a given type.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># find the names of all cells belonging to a certain type, identified as a beginning of a trajectory</span><span class="w">
</span><span class="n">starting_cell_type</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s1">'DN'</span><span class="w">
</span><span class="n">index</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="o">@</span><span class="n">colData</span><span class="o">$</span><span class="n">cell_type</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">starting_cell_type</span><span class="p">)</span><span class="w">
</span><span class="n">DN_cells</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">)[</span><span class="n">index</span><span class="p">]</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># alternatively, if you work on unannotated data, you can use the number of the cluster that should be used as the beginning of the trajectory and pass it in the ‘root_cells’ argument</span><span class="w">
</span><span class="n">starting_cluster</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">[,</span><span class="n">clusters</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">4</span><span class="p">])</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># order cells</span><span class="w">
</span><span class="n">cds_order_2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">order_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">,</span><span class="w"> </span><span class="n">root_cells</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">DN_cells</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-only-for-users-of-local-rstudio-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-only-for-users-of-local-rstudio-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Only for users of local RStudio<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If you are working in RStudio locally, you can use <code class="language-plaintext highlighter-rouge">order_cells()</code> function in the interactive mode. The pop-up window should appear and then you can simply select root nodes in the cluster with the cell type identified as the beginning of the trajectory.</p>

  <div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># specifying root cells: pop up window</span><span class="w">
</span><span class="n">cds_order_1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">order_cells</span><span class="p">(</span><span class="n">cds_trajectory</span><span class="p">)</span><span class="w">
</span></code></pre></div>  </div>

</blockquote>

<h1 id="plotting-in-pseudotime">Plotting in pseudotime</h1>

<p>You can use any <code class="language-plaintext highlighter-rouge">cds_order</code> object for the downstream analysis. Let’s pick one and assign it to an object with a shorter and more general name.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">cds_order</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds_order_1_helper</span><span class="w">
</span></code></pre></div></div>

<p>The function <code class="language-plaintext highlighter-rouge">order_cells()</code> calculates pseudotime for each cell, so we can now visualise the T-cell maturation process in pseudotime! Additionally, we can access it and store in our CDS object for further analysis.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># plot cells in pseudotime</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w">
           </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"pseudotime"</span><span class="p">,</span><span class="w">
           </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_leaves</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_branch_points</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># access pseudotime calculated for each cell and store it alongside cell metadata</span><span class="w">
</span><span class="n">pseudotime</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pseudotime</span><span class="p">(</span><span class="n">cds_order</span><span class="p">)</span><span class="w"> 	
</span><span class="n">cds_order</span><span class="o">@</span><span class="n">colData</span><span class="o">$</span><span class="n">pseudotime</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pseudotime</span><span class="w"> 		 
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/pseudotime_r.png" rel="noopener noreferrer"><figure id="figure-15" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/pseudotime_r.png" alt="Plot of the cells ordered in pseudotime. The trajectory starts at double negative cells (ie. pseudotime = 0) which are coloured blue, then goes through double positive (middle T-cells), gradually changing colour from purple to pink, then double positive (late T-cells) are coloured orange and the trajectory ends at T-mature cells, which are coloured yellow." width="650" height="540" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Cells ordered in pseudotime.&lt;/figcaption&gt;</figure></a></p>

<p>We can now see how our hard work has come together to give a final pseudotime trajectory analysis, which starts at double negative cells, then gently switches to double positives: from middle to late T-cells, and ends up on mature T-cells.</p>

<h1 id="differential-expression-analysis">Differential expression analysis</h1>
<p>There are two approaches for more advanced differential analysis in Monocle:</p>
<ul>
  <li>Regression analysis: using <code class="language-plaintext highlighter-rouge">fit_models()</code>, you can evaluate whether each gene depends on variables such as time, treatments, etc.</li>
  <li>Graph-autocorrelation analysis: using <code class="language-plaintext highlighter-rouge">graph_test()</code>, you can find genes that vary over a trajectory or between clusters.</li>
</ul>

<p>In this section, for some examples, we will work on the subset of our CDS object because the computation time for the whole dataset would take quite some time. Let’s make the subset CDS containing the information about the genes listed in the table in the previous section, so instead having 15395 elements, it will have only 7.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># make the subset CDS</span><span class="w">
</span><span class="n">test_genes</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s1">'Il2ra'</span><span class="p">,</span><span class="s1">'Cd8b1'</span><span class="p">,</span><span class="s1">'Cd8a'</span><span class="p">,</span><span class="s1">'Cd4'</span><span class="p">,</span><span class="s1">'Itm2a'</span><span class="p">,</span><span class="s1">'Aif1'</span><span class="p">,</span><span class="s1">'Hba-a1'</span><span class="p">)</span><span class="w">
</span><span class="n">cds_subset</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">cds_order</span><span class="p">[</span><span class="n">rowData</span><span class="p">(</span><span class="n">cds_order</span><span class="p">)</span><span class="o">$</span><span class="n">gene_short_name</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">test_genes</span><span class="p">,]</span><span class="w">
</span></code></pre></div></div>

<h2 id="some-more-plotting">Some more plotting</h2>
<p>Monocle also provides some easy ways to plot the expression of a small set of genes grouped by the factors you use during differential analysis.
For example <code class="language-plaintext highlighter-rouge">plot_genes_violin()</code> allows us to create violin plots which are quite common in the field. Therefore let’s visualise how the gene expression changes between the cell types.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># produce violin plots</span><span class="w">
</span><span class="n">plot_genes_violin</span><span class="p">(</span><span class="n">cds_subset</span><span class="p">,</span><span class="w"> </span><span class="n">group_cells_by</span><span class="o">=</span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/violin_plots.png" rel="noopener noreferrer"><figure id="figure-16" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/violin_plots.png" alt=" Violin plots of the expression of the specified genes in each assigned cell type. The trends are consistent with the previous biological interpretation and assignment." width="859" height="543" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 16</strong>:</span>  Violin plots of the expression of the specified genes in each assigned cell type.&lt;/figcaption&gt;</figure></a></p>

<p>When analysing the above violin plots, we will realise that the results are consistent with the detailed analysis in the previous tutorials, as well as findings in current tutorial.</p>

<p>Another great function <code class="language-plaintext highlighter-rouge">plot_genes_in_pseudotime()</code> takes a small set of genes and shows their dynamics as a function of pseudotime.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plot_genes_in_pseudotime</span><span class="p">(</span><span class="n">cds_subset</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"cell_type"</span><span class="p">,</span><span class="w"> </span><span class="n">min_expr</span><span class="o">=</span><span class="m">0.5</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><a href="../../images/scrna-casestudy-monocle/genes_in_pseudotime.png" rel="noopener noreferrer"><figure id="figure-17" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/genes_in_pseudotime.png" alt="On the left hand side scatter plots of expression of the specified genes in pseudotime. On the right hand side also the expression of specified genes as a function of pseudotime but shown with the cell type data superimposed. Again the trends are consistent with the previous biological interpretation and assignment. " width="488" height="550" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 17</strong>:</span>  A visualisation of how specified genes change in the pseudotime.&lt;/figcaption&gt;</figure></a></p>

<h2 id="regression-analysis---advanced">Regression analysis - advanced</h2>
<p>We will use a function which fits a generalized linear model for each gene in a CDS object. We have to specify the model formula which is any term that exists as a column in <code class="language-plaintext highlighter-rouge">colData()</code>. We want to test genes that differ between cell types and batches, so we will use “~cell_type + batch” argument. Then, we extract a table of coefficients from each model. <code class="language-plaintext highlighter-rouge">coefficient_table()</code> tests whether each coefficient differs significantly from zero under the <a href="https://en.wikipedia.org/wiki/Wald_test">Wald test</a>.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># fit a generalized linear model for each gene</span><span class="w">
</span><span class="n">gene_fits</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fit_models</span><span class="p">(</span><span class="n">cds_subset</span><span class="p">,</span><span class="w"> </span><span class="n">model_formula_str</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"~cell_type + batch"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># extract a table of coefficients</span><span class="w">
</span><span class="n">fit_coefs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">coefficient_table</span><span class="p">(</span><span class="n">gene_fits</span><span class="p">)</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># preview the content of 'fit_coefs'</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">fit_coefs</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p>If you inspect the <code class="language-plaintext highlighter-rouge">fit_coefs</code> object, you will notice that the table includes one row for each term of each gene’s model. We generally don’t care about the intercept term, so we can filter it out. In this way, we will be able to control for the chosen factors. To focus on only one variable, you have to check the <code class="language-plaintext highlighter-rouge">term</code> column in the <code class="language-plaintext highlighter-rouge">fit_coefs</code> and pass this as an argument for filtering. Then, you should also filter the results with q_value &lt; 0.05 to control the false discovery rate.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># filter out Intercept term</span><span class="w">
</span><span class="n">no_intercept_coefs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fit_coefs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">term</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s2">"(Intercept)"</span><span class="p">)</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># extract results for DP-M1 cells only</span><span class="w">
</span><span class="n">DP_M1_coefs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">fit_coefs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">term</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"cell_typeDP-M1"</span><span class="p">)</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># control the false discovery rate and choose only several variables to store</span><span class="w">
</span><span class="n">DP_M1_coefs_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DP_M1_coefs</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="w"> </span><span class="p">(</span><span class="n">q_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
  </span><span class="n">dplyr</span><span class="o">::</span><span class="n">select</span><span class="p">(</span><span class="n">gene_short_name</span><span class="p">,</span><span class="w"> </span><span class="n">term</span><span class="p">,</span><span class="w"> </span><span class="n">q_value</span><span class="p">,</span><span class="w"> </span><span class="n">estimate</span><span class="p">)</span><span class="w">		
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># view the resulting table</span><span class="w">
</span><span class="n">DP_M1_coefs_filtered</span><span class="w">
</span></code></pre></div></div>
<p>The resulting table shows the genes that differ depending on the chosen term. Maybe this function is not very helpful in the case of our dataset, but may be useful when analysing unannotated data or choosing another term from <code class="language-plaintext highlighter-rouge">colData()</code>.</p>

<h2 id="graph-autocorrelation-analysis---advanced">Graph-autocorrelation analysis - advanced</h2>
<p>Alongside regression analysis, Monocle also provides another way of finding genes that vary between groups of cells. The function <code class="language-plaintext highlighter-rouge">graph_test()</code> uses a statistic from spatial autocorrelation analysis called <a href="https://en.wikipedia.org/wiki/Moran%27s_I">Moran’s I</a>, which <span class="citation"><a href="#Cao_2019">Cao <i>et al.</i> 2018</a></span> showed to be effective in finding genes that vary in single-cell RNA-seq datasets. Let’s try to perform this step on our full dataset (be patient!).</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># run autocorrelation test</span><span class="w">
</span><span class="n">graph_test_res</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">graph_test</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w"> </span><span class="n">neighbor_graph</span><span class="o">=</span><span class="s2">"knn"</span><span class="p">,</span><span class="w"> </span><span class="n">cores</span><span class="o">=</span><span class="m">8</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>The output data frame has a bunch of statistical values that you can use to rank the genes by, for example <code class="language-plaintext highlighter-rouge">morans_I</code> column, which ranges from -1 to +1. A value of 0 indicates no effect, while +1 indicates perfect positive autocorrelation and suggests that nearby cells have very similar values of a gene’s expression.</p>

<p>We will now try to associate genes with clusters by grouping genes into modules that have similar patterns of expression. You can call <code class="language-plaintext highlighter-rouge">find_gene_modules()</code>, which essentially runs UMAP on the genes (as opposed to the cells) and then groups them into modules using Louvain community analysis.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># get gene IDs for which q-value &lt; 0.05</span><span class="w">
</span><span class="n">pr_deg_ids</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">row.names</span><span class="p">(</span><span class="n">subset</span><span class="p">(</span><span class="n">graph_test_res</span><span class="p">,</span><span class="w"> </span><span class="n">q_value</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">))</span><span class="w"> 	
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># group genes into modules</span><span class="w">
</span><span class="n">gene_module_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">find_gene_modules</span><span class="p">(</span><span class="n">cds_order</span><span class="p">[</span><span class="n">pr_deg_ids</span><span class="p">,],</span><span class="w"> </span><span class="n">resolution</span><span class="o">=</span><span class="m">1e-2</span><span class="p">)</span><span class="w"> 	
</span></code></pre></div></div>
<p>Now we can show the aggregate expression of all genes in each module across all the clusters. Monocle provides a simple utility function called <code class="language-plaintext highlighter-rouge">aggregate_gene_expression</code> for this purpose:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># aggregate expression of genes in each module</span><span class="w">
</span><span class="n">cell_group_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tibble</span><span class="o">::</span><span class="n">tibble</span><span class="p">(</span><span class="n">cell</span><span class="o">=</span><span class="n">row.names</span><span class="p">(</span><span class="n">colData</span><span class="p">(</span><span class="n">cds_order</span><span class="p">)),</span><span class="w">
                                </span><span class="n">cell_group</span><span class="o">=</span><span class="n">clusters</span><span class="p">(</span><span class="n">cds_order</span><span class="p">)[</span><span class="n">colnames</span><span class="p">(</span><span class="n">cds_order</span><span class="p">)])</span><span class="w">
</span><span class="n">agg_mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">aggregate_gene_expression</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w"> </span><span class="n">gene_module_df</span><span class="p">,</span><span class="w"> </span><span class="n">cell_group_df</span><span class="p">)</span><span class="w">
</span><span class="n">row.names</span><span class="p">(</span><span class="n">agg_mat</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">stringr</span><span class="o">::</span><span class="n">str_c</span><span class="p">(</span><span class="s2">"Module "</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="p">(</span><span class="n">agg_mat</span><span class="p">))</span><span class="w">
</span><span class="n">colnames</span><span class="p">(</span><span class="n">agg_mat</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">stringr</span><span class="o">::</span><span class="n">str_c</span><span class="p">(</span><span class="s2">"Cluster "</span><span class="p">,</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">agg_mat</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>
<p>We can now use this data to create a heatmap. Don’t worry if yours look a little bit different that the one shown in this tutorial. The general features should be maintained though.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create a heatmap</span><span class="w">
</span><span class="n">pheatmap</span><span class="o">::</span><span class="n">pheatmap</span><span class="p">(</span><span class="n">agg_mat</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_rows</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">cluster_cols</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w">
                   </span><span class="n">scale</span><span class="o">=</span><span class="s2">"column"</span><span class="p">,</span><span class="w"> </span><span class="n">clustering_method</span><span class="o">=</span><span class="s2">"ward.D2"</span><span class="p">,</span><span class="w">
                   </span><span class="n">fontsize</span><span class="o">=</span><span class="m">6</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="../../images/scrna-casestudy-monocle/heatmap.png" rel="noopener noreferrer"><figure id="figure-18" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/heatmap.png" alt="A heatmap showing modules of co-regulated genes across the clusters. Modules listed vertically while clusters horizontally. Some modules are highly specific to certain partitions of cells, while others are shared across multiple partitions." width="500" height="700" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 18</strong>:</span> Heatmap showing modules of co-regulated genes across the clusters.&lt;/figcaption&gt;</figure></a></p>

<p>You can also visualise the modules using <code class="language-plaintext highlighter-rouge">plot_cells()</code> function. We’ve chosen some modules to see how the differences on a heatmap correlate with the expression shown on our starting plot.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Which modules to plot?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>This is totally up to you! It might be the case that you got different numbering of modules, so then using the numbers specified in the code below won’t make much sense. Just look at your heatmap, compare the differences between modules and think which ones would be the most interesing to visualise.</p>

  </blockquote>

</blockquote>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the chosen modules across the whole sample</span><span class="w">
</span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w">
           </span><span class="n">genes</span><span class="o">=</span><span class="n">gene_module_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="n">filter</span><span class="p">(</span><span class="n">module</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">40</span><span class="p">,</span><span class="w"> </span><span class="m">39</span><span class="p">,</span><span class="w"> </span><span class="m">36</span><span class="p">,</span><span class="w"> </span><span class="m">17</span><span class="p">)),</span><span class="w">
           </span><span class="n">group_cells_by</span><span class="o">=</span><span class="s2">"cluster"</span><span class="p">,</span><span class="w">
           </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"cluster"</span><span class="p">,</span><span class="w">
           </span><span class="n">show_trajectory_graph</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><a href="../../images/scrna-casestudy-monocle/modules_plot.png" rel="noopener noreferrer"><figure id="figure-19" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/modules_plot.png" alt="Plots showing expression of the genes belonging to specified modules across whole sample." width="650" height="550" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 19</strong>:</span> Plots of expression of the genes belonging to specified modules across whole sample.&lt;/figcaption&gt;</figure></a></p>

<p>With the visualisation methods above, you can now come back to the generated data frame <code class="language-plaintext highlighter-rouge">gene_module_df</code>, filter genes that belong to the module of interest and check their functions to get some more evidence for the correct biological interpretation.</p>

<h1 id="3d-plotting">3D plotting</h1>
<p>Let’s have some fun at the end! That was quite a long and insightful analysis – you definitely deserve to look at some nice, rotatable, cool plot now!
Essentially the workflow is the same as we followed in two dimensions. The crucial part is to specify the dimensionality of the reduced space with the <code class="language-plaintext highlighter-rouge">max_components</code> parameter in <code class="language-plaintext highlighter-rouge">reduce_dimension()</code> function. The default is 2, but if we want to see our data in 3D, we will change that value to 3. From there, you can just repeat the next steps in 3D… or just reward yourself for completing this tutorial by toggling this beautiful 3D plot!</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># reduce dimension to 3D</span><span class="w">
</span><span class="n">cds_3d</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">reduce_dimension</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w"> </span><span class="n">preprocess_method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s1">'Aligned'</span><span class="p">,</span><span class="w"> </span><span class="n">max_components</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># see the resulting 3D plot</span><span class="w">
</span><span class="n">plot_cells_3d</span><span class="p">(</span><span class="n">cds_3d</span><span class="p">,</span><span class="w"> </span><span class="n">color_cells_by</span><span class="o">=</span><span class="s2">"cell_type"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h1 id="export-your-data-figures-and-notebook">Export your data, figures, and notebook</h1>

<p>Don’t forget to save and export your data! First, we will get Jupyter to see those as files.</p>

<h2 id="export-plots">Export plots</h2>
<p>If you want to export your plot, you have to make sure that you assigned it to an object. For example, if you want to save the plot of cells in pseudotime, simply assign the function you used to generate this plot to an object. Here we call this object <code class="language-plaintext highlighter-rouge">plot_pseudotime</code>, like so:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plot_pseudotime</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">plot_cells</span><span class="p">(</span><span class="n">cds_order</span><span class="p">,</span><span class="w">
           </span><span class="n">color_cells_by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"pseudotime"</span><span class="p">,</span><span class="w">
           </span><span class="n">label_cell_groups</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_leaves</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w">
           </span><span class="n">label_branch_points</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>Then, if you want to save the plot as PDF:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pdf</span><span class="p">(</span><span class="s2">"plot_pseudotime.pdf"</span><span class="p">)</span><span class="w">             </span><span class="c1"># open the graphical device and specify the directory and the name of the output pdf file</span><span class="w">
</span><span class="n">plot_pseudotime</span><span class="w">                        </span><span class="c1"># specify the object that your plot is assigned to</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">                              </span><span class="c1"># close the graphical device</span><span class="w">
</span></code></pre></div></div>

<p>The procedure is very similar if you want to export the file as PNG (or analogically JPEG – just replace png with jpeg):</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">png</span><span class="p">(</span><span class="s2">"plot_pseudotime.png"</span><span class="p">,</span><span class="w">            </span><span class="c1"># open the graphical device and specify the directory and the name of the output png file</span><span class="w">
</span><span class="n">width</span><span class="o">=</span><span class="m">600</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">400</span><span class="p">)</span><span class="w">                </span><span class="c1"># optionally you can specify the width and height of the final plot</span><span class="w">
</span><span class="n">plot_pseudotime</span><span class="w">                       </span><span class="c1"># specify the object that your plot is assigned to</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">                             </span><span class="c1"># close the graphical device</span><span class="w">
</span></code></pre></div></div>

<p>However, it often happens that the quality of the exported PNG and JPEG files is not perfect. For best results, we recommend exporting to SVG:</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">svg</span><span class="p">(</span><span class="s2">"plot_pseudotime.svg"</span><span class="p">)</span><span class="w">             </span><span class="c1"># open the graphical device and specify the directory and the name of the output svg file</span><span class="w">
</span><span class="n">plot_pseudotime</span><span class="w">                        </span><span class="c1"># specify the object that your plot is assigned to</span><span class="w">
</span><span class="n">dev.off</span><span class="p">()</span><span class="w">                              </span><span class="c1"># close the graphical device</span><span class="w">
</span></code></pre></div></div>

<p>You can do the same with any plot that you want to save! You will find the saved figures in the left panel of your JupyterLab. You can right-click on them and download directly onto your computer. You can also push them into your Galaxy history. To do so, you have to change Kernel to Python3 (either click on <code class="language-plaintext highlighter-rouge">Kernel</code> -&gt; <code class="language-plaintext highlighter-rouge">Change Kernel...</code> in the upper left corner of your JupyterLab or click on the displayed current kernel in the upper right corner and change it).
<a href="../../images/scrna-casestudy-monocle/switch_kernel.jpg" rel="noopener noreferrer"><figure id="figure-20" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy-monocle/switch_kernel.jpg" alt="Figure showing the JupyterLab interface with an arrow pointing to the left corner, showing the option `Kernel` -&gt; `Change Kernel...` and another arrow pointing to the right corner, showing the icon of the current kernel. The pop-up window asks which kernel should be chosen instead." width="1760" height="580" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 20</strong>:</span> Two ways of switching kernel.&lt;/figcaption&gt;</figure></a></p>

<p>Check in the upper right corner that selected kernel is Python3, and run the following:</p>
<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">put</span><span class="p">(</span><span class="sh">"</span><span class="s">marker_file.txt</span><span class="sh">"</span><span class="p">)</span>
<span class="nf">put</span><span class="p">(</span><span class="sh">"</span><span class="s">plot_pseudotime.pdf</span><span class="sh">"</span><span class="p">)</span>
<span class="nf">put</span><span class="p">(</span><span class="sh">"</span><span class="s">plot_pseudotime.png</span><span class="sh">"</span><span class="p">)</span>
</code></pre></div></div>

<p>In this way you can push all the files you’ve saved into your Galaxy history. You can also do the same with this notebook. The cell below will only work if you haven’t changed the name of the notebook. If you renamed it, simply type its new name in the parenthesis.</p>
<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">put</span><span class="p">(</span><span class="sh">"</span><span class="s">single-cell-scrna-case_monocle3-rstudio.ipynb</span><span class="sh">"</span><span class="p">)</span>
</code></pre></div></div>

<p>Now you can go check your Galaxy history to make sure your files have all made it back into your Galaxy history.</p>

<h1 id="after-jupyter">After Jupyter</h1>

<p>Congratulations! You’ve made it through Jupyter!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-closing-jupyterlab-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Closing JupyterLab</div>
  <ol>
    <li>Click <strong>User</strong>: <strong>Active Interactive Tools</strong></li>
    <li>Tick the box of your Jupyter Interactive Tool, and click <strong>Stop</strong></li>
  </ol>
</blockquote>

<p>If you want to run this notebook again, or share it with others, it now exists in your history. You can use this ‘finished’ version just the same way as you downloaded the directions file and uploaded it into the Jupyter environment.</p>

<h1 id="conclusion">Conclusion</h1>

<p>If you’re following the Case Study tutorials from the beginning, you have already experienced what it’s like to analyse and question a dataset, potentially without clear cut-offs or clear answers. The <a href="/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html">Monocle in Galaxy tutorial</a> was focused more on explaining each step of the trajectory analysis and interpreting the results in the biological context. The current tutorial aims at showing the variety of methods that can be used when Galaxy’s Monocle tools are not enough. It shows the potential of batch correction, differential expression analysis and flexibility when using different functions. It’s also a guide for those who would like to understand what is happening ‘behind the scenes’ when clicking on Galaxy buttons.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Monocle3 in R gives more flexibility when it comes to differential expression analysis and plotting, but Galaxy offers great reproducibility and ease of analysis.</p>
</li>
                        
                        <li><p>Comparing the output of several different methods applied on the same dataset might be useful to confirm the results, to ensure that the findings are reliable and even sometimes to find a new piece of information.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Levine_2015">Levine, J. H., E. F. Simonds, S. C. Bendall, K. L. Davis, E.-ad D. Amir <i>et al.</i>, 2015 <b>Data-Driven Phenotypic Dissection of AML Reveals Progenitor-like Cells that Correlate with Prognosis</b>. Cell 162: 184–197. <a href="https://doi.org/10.1016/j.cell.2015.05.047">10.1016/j.cell.2015.05.047</a></li>
<li id="Bacon2018">Bacon, W. A., R. S. Hamilton, Z. Yu, J. Kieckbusch, D. Hawkes <i>et al.</i>, 2018 <b>Single-Cell Analysis Identifies Thymic Maturation Delay in Growth-Restricted Neonatal Mice</b>. Frontiers in Immunology 9: <a href="https://doi.org/10.3389/fimmu.2018.02523">10.3389/fimmu.2018.02523</a></li>
<li id="Cao_2019">Cao, J., M. Spielmann, X. Qiu, X. Huang, D. M. Ibrahim <i>et al.</i>, 2018 <b>The single-cell transcriptional landscape of mammalian organogenesis</b>. Nature 566: 496–502. <a href="https://doi.org/10.1038/s41586-019-0969-x">10.1038/s41586-019-0969-x</a> <a href="https://10.1038/s41586-019-0969-x">https://10.1038/s41586-019-0969-x</a></li>
<li id="Haghverdi_2018">Haghverdi, L., A. T. L. Lun, M. D. Morgan, and J. C. Marioni, 2018 <b>Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors</b>. Nature Biotechnology 36: 421–427. <a href="https://doi.org/10.1038/nbt.4091">10.1038/nbt.4091</a></li>
<li id="Traag_2019">Traag, V. A., L. Waltman, and N. J. van Eck, 2019 <b>From Louvain to Leiden: guaranteeing well-connected communities</b>. Sci Rep 9: <a href="https://doi.org/10.1038/s41598-019-41695-z">10.1038/s41598-019-41695-z</a></li>
<li id="Wolf_2019">Wolf, F. A., F. K. Hamey, M. Plass, J. Solana, J. S. Dahlin <i>et al.</i>, 2019 <b>PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells</b>. Genome Biol 20: <a href="https://doi.org/10.1186/s13059-019-1663-x">10.1186/s13059-019-1663-x</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Inferring Trajectories using Monocle3 (R) (Single Cell)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Julia Jakiela,  <b>Inferring Trajectories using Monocle3 (R) (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scrna-case_monocle3-rstudio,
author = "Julia Jakiela",
	title = "Inferring Trajectories using Monocle3 (R) (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                
<div markdown="1">
<h2 id="funding">Funding</h2>
<p>These individuals or organisations provided funding support for the development of this resource</p>

<div class="row">

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://www.eosc-life.eu/wp-content/themes/eosc-life-v2/assets/images/eosclogo.png" alt="Logo">
		
		<a href="/training-material/hall-of-fame/eosc-life/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		<p>EOSC-Life has received funding from the European Union’s Horizon 2020 programme under grant agreement number 824087</p>

	</div>

</div>
</div>


                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00336">https://gxy.io/GTN:T00336</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1705797054'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>