<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Mapping and molecular identification of phenotype-causing mutations</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1702254526' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1702254526' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Mapping and molecular identification of phenotype-causing mutations">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-03-05 09:19:48 +0000">
<meta name="DC.creator" content="Wolfgang Maier">

        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Mapping and molecular identification of phenotype-causing...">
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-03-05 09:19:48 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Mapping and molecular identification of phenotype-causing mutations",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Mapping and molecular identification of phenotype-causing mutations' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html",
  "timeRequired": "PT2H",
  "teaches": "Use joint variant calling and extraction to facilitate variant comparison across samples\n - Perform variant linkage analyses for phenotypically selected recombinant progeny\n - Filter, annotate and report lists of variants",
  "keywords": "variant-analysis",
  "description": "The questions this  addresses are:\n - What is mapping-by-sequencing?\n - How can it help you identify the causative mutation in phenotypic mutants isolated from genetic screens?\n\n\nThe objectives are:\n - Use joint variant calling and extraction to facilitate variant comparison across samples\n - Perform variant linkage analyses for phenotypically selected recombinant progeny\n - Filter, annotate and report lists of variants\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1702254526'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Mapping and molecular identification of phenotype-causing mutations</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Avatar" width="36" height="36">Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What is mapping-by-sequencing?</p>
</li>
        
        <li><p>How can it help you identify the causative mutation in phenotypic mutants isolated from genetic screens?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Use joint variant calling and extraction to facilitate variant comparison across samples</p>
</li>
        
        <li><p>Perform variant linkage analyses for phenotypically selected recombinant progeny</p>
</li>
        
        <li><p>Filter, annotate and report lists of variants</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.1098033">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/workflows/" title="Mapping and molecular identification of phenotype-causing mutations workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://galaxytrakr.org/" title="">
			GalaxyTrakr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 5, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00312">https://gxy.io/GTN:T00312</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>In order to map and identify phenotype-causing mutations efficiently from a
single experiment, modern genetic research aims to combine classical genetic
mapping concepts with the power of next-generation-sequencing.</p>

<p>For example, after having obtained a mutant strain of an organism with a particular phenotype from a forward genetic screen, the classical approach towards
identification of the underlying causative mutation would be to:</p>

<ol>
  <li>perform mapping crosses to probe for linkage between the unknown mutation
and selected markers with known location on the genome, in order to determine
the approximate genomic region that the mutation resides in, then</li>
  <li>sequence candidate DNA stretches in this region to identify the precise
nature of the mutation.</li>
</ol>

<p>Modern approaches, in contrast, use just one set of genome-wide sequencing data
obtained from mapping-cross progeny and suitable parent strains to
simultaneously:</p>

<ul>
  <li>discover available marker mutations for linkage analysis,</li>
  <li>map the causative mutation using these markers, and</li>
  <li>identify candidate mutations.</li>
</ul>

<p>Since it uses sequencing data already at the mapping step, this process is
called <strong>mapping-by-sequencing</strong>.</p>

<p>Since genome-wide sequencing will often reveal a few hundreds to many thousands
of marker variants that should be considered together in the linkage analysis,
mapping-by-sequencing data is typically analyzed with dedicated computational
tools. For Galaxy, the <strong>MiModD</strong> suite of tools offers efficient and flexible
analysis workflows compatible with a variety of mapping-by-sequencing
approaches.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-further-reading"><button class="gtn-boxify-button details" type="button" aria-controls="details-further-reading" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: Further reading<span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>The MiModD documentation has its own chapter on supported
<a href="http://mimodd.readthedocs.io/en/latest/nacreousmap.html">mapping-by-sequencing schemes</a>
not covered here.</p>
</blockquote>

<p>In this tutorial we are going to use several of these tools to map and identify
a point mutation in an <em>Arabidopsis thaliana</em> strain from whole-genome
sequencing data. This will demonstrate one particular kind of linkage analysis
supported by <strong>MiModD</strong> called <em>Variant Allele Frequency Mapping</em>.</p>

<h3 id="variant-allele-frequency-mapping">Variant Allele Frequency Mapping</h3>

<p><a href="../../images/mbs_scheme.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/mbs_scheme.png" alt="Principal of variant allele frequency mapping. " width="972" height="1121" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Principle of variant allele frequency mapping&lt;/figcaption&gt;</figure></a></p>

<p>As illustrated in Figure 1, variant allele frequency mapping analyzes marker
variant inheritance patterns in pooled F2 recombinants selected for the
mutation to be mapped.</p>

<p>In the figure we assume a diploid
organism with just two pairs (dark and light grey) of chromosomes.</p>

<p>A mutant line established for any organism will harbor
two types of variants (diamonds):</p>
<ul>
  <li>variants induced during mutagenesis (orange), among them the causative
variant responsible for the mutant phenotype (yellow)</li>
  <li>variants (with respect to the species’ reference genome) that were
present in the premutagenesis strain (green open diamonds)</li>
</ul>

<p>For mapping the causative mutation, additional variants (blue open diamonds)
are introduced through a cross to a mapping strain and F2 progeny is selected
for the mutant phenotype. While each individual in the F2 generation carries
its uniquely recombined variant pattern (3 possible outcomes for the diploid
genome are shown in the figure), this phenotypic selection will work against
mapping strain variants in the vicinity (yellow boxes) of the causative
variant. Hence, when the recombined DNA from phenotypic F2 progeny gets pooled
and sequenced, the causative variant can be found by looking for a region for
which all sequenced reads support variant alleles found in the original mutant
line, rather than mapping strain variant alleles.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a></li>
  <li><a href="#joint-variant-calling-and-extraction" id="markdown-toc-joint-variant-calling-and-extraction">Joint Variant Calling and Extraction</a></li>
  <li><a href="#linkage-analysis" id="markdown-toc-linkage-analysis">Linkage Analysis</a></li>
  <li><a href="#identifying-candidate-mutations" id="markdown-toc-identifying-candidate-mutations">Identifying Candidate Mutations</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>In this tutorial we are going to (re-)identify the recessive causative mutation
in a late-flowering mutant line of <em>Arabidopsis thaliana</em>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p><strong>MiModD</strong> was developed as a successor of the older and now unmaintained
<em>CloudMap</em> suite of tools, but also uses some key concepts first implemented
in the command line package <em>SHOREmap</em>.
The data used in this tutorial represent a subsample of the
<a href="http://bioinfo.mpipz.mpg.de/shoremap/examples.html">SHOREmap proof-of-principle data</a>
(specifically of the <code class="language-plaintext highlighter-rouge">OCF2</code> and <code class="language-plaintext highlighter-rouge">Ler</code> datasets).</p>
</blockquote>

<p>The mutant line was obtained from a mutagenesis screen on the <strong>Col</strong>umbia
reference background of <em>A. thaliana</em>. For mapping-by-sequencing, the line was
outcrossed to the highly polymorphic <strong>L</strong>andsberg <em><strong>er</strong>ecta</em> background as a
source of marker variants. From this outcross, 119 F2 siblings showing the
original late-flowering phenotype were recovered and their genomic DNA was
pooled and subjected to whole-genome sequencing.
Together with whole-genome sequencing data of a pure L<em>er</em> background strain,
this outcrossed pool data forms the starting point of our analysis.
The L<em>er</em> strain sample, thus, can inform us about the markers that the
outcrossed F2 pool <em>could</em> have inherited and exclusion of L<em>er</em> markers from
a particular genomic region from the F2 pool provides evidence that the
phenotype causing variant lies close to this region. With respect to
<a href="#figure-1">Figure 1</a>, this means that we will only use “blue diamond” marker
variants introduced through the mapping cross for the linkage analysis and
ignore potentially informative (green and orange) variants found in the
original mutant line.</p>

<p>For your convenience, the sequenced reads from both datasets have already been
mapped (using <strong>Bowtie2</strong>) to the <em>A. thaliana</em> Col reference genome (version
TAIR10) so the two datasets take the form of BAM files with the mapping results
(see the tutorial on
<a href="../../../sequence-analysis/tutorials/mapping/tutorial.html">NGS data mapping</a>
if you do not know what this means).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-and-preprocessing"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload and preprocessing</div>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip"><button class="gtn-boxify-button tip" type="button" aria-controls="tip" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>This section assumes that you already know</p>

    <ul>
      <li>how to upload data to Galaxy via links</li>
      <li>edit dataset attributes</li>
      <li>do some basic text manipulation on datasets</li>
    </ul>

    <p>An excellent introduction into these topics can be found in the
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">From peaks to genes</a>
tutorial (specifically, in the sections on
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html#data-upload">Data upload</a>
and
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html#file-preparation">File preparation</a>).</p>
  </blockquote>

  <ol>
    <li>Import the two BAM datasets representing the:
      <ul>
        <li><a href="https://zenodo.org/record/1098034/files/Ler_mapping_strain.bam">L<em>er</em> mapping strain</a></li>
        <li><a href="https://zenodo.org/record/1098034/files/outcrossed_F2.bam">outcrossed F2 pool</a></li>
      </ul>
    </li>
    <li>Specify the genome that was used at the reads mapping step
      <ol>
        <li>Click on <strong>Edit attributes</strong> (the pencil icon displayed with the dataset)</li>
        <li>As the <strong>Database/Build</strong> select <code class="language-plaintext highlighter-rouge">Arabidopsis thaliana TAIR10</code></li>
        <li><strong>Save</strong> the edited attributes</li>
      </ol>
    </li>
    <li>
      <p>Import the reference genome</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-tair10-version"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: TAIR10 Version</div>
        <p>This step is only necessary if the TAIR10 version of the <em>A. thaliana</em>
genome is not available as a built-in genome on your Galaxy server.</p>
      </blockquote>

      <ol>
        <li>Upload the reference genome using the URL:
<code class="language-plaintext highlighter-rouge">https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_chromosome_files/TAIR10_chr_all.fas</code></li>
        <li>Use <strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to truncate the title
lines in the fasta dataset to just the bare chromosome names. Set:
          <ul>
            <li>
              <p>the <strong>Find pattern</strong> to <code class="language-plaintext highlighter-rouge">^&gt;(\w+).*</code></p>

              <p>This will find entire fasta title lines, or more specifically, any
lines starting with <code class="language-plaintext highlighter-rouge">&gt;</code> followed by at least one letter, digit or
underscore.</p>
            </li>
            <li>
              <p>the <strong>Replace with</strong> pattern to <code class="language-plaintext highlighter-rouge">&gt;chr$1</code></p>

              <p>The <code class="language-plaintext highlighter-rouge">$1</code> in it refers to the parenthesized part of the find pattern,
<em>i.e.</em>, just the substring of letters/digits/underscores directly
following the <code class="language-plaintext highlighter-rouge">&gt;</code>.</p>
            </li>
            <li>
              <p><strong>Find-Pattern is a regular expression</strong> to <code class="language-plaintext highlighter-rouge">Yes</code></p>
            </li>
          </ul>
        </li>
        <li>Make sure the datatype of the resulting dataset is set to <code class="language-plaintext highlighter-rouge">fasta</code> and
give the dataset a nice name (<em>e.g.</em>, TAIR10 reference).</li>
        <li>At this point you can delete the originally downloaded dataset.</li>
      </ol>
    </li>
  </ol>
</blockquote>

<h1 id="joint-variant-calling-and-extraction">Joint Variant Calling and Extraction</h1>

<p>A hallmark of all mapping-by-sequencing analyses is that one would like to
follow the fate of variants through genetic crosses. In this tutorial, for
example, we would like to determine, for every marker variant provided by the
L<em>er</em> background, at what frequency it got inherited by the outcrossed F2 pool.
For such analyses to work reliably, it is best to call variants jointly for all
relevant samples because this allows the variant caller to give us per-sample
variant call statistics for all sites at which <strong>any</strong> sample is found to have
a non-wt genotype. In our case, this means that the variant caller will report
any variants it finds for the L<em>er</em> background sample along with the call
statistics for the outcrossed F2 pool at that same site - even if the variant
allele contributes very little to the pool and would, thus, have been
disregarded in a standard single-sample analysis.</p>

<p>Here, we are using the <strong>MiModD Variant Calling</strong> tool for joint variant
calling for our two samples. This tool outputs variant call statistics across
all sites and all samples in the compact binary BCF format. We can then use
the <strong>MiModD Extract Variant Sites</strong> tool to extract sites, for which at least
one sample appears to have a non-wt genotype, and report them in VCF format.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-variant-calling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Variant Calling</div>

  <ol>
    <li>Use <strong>MiModD Variant Calling</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to generate combined
variant call statistics:
      <ul>
        <li>
          <p>Select the <strong>reference genome</strong> to call variants against.</p>

          <p>If your Galaxy server offers version TAIR10 of the <em>A. thaliana</em>
reference as a <strong>built-in genome</strong>, you can go ahead and use it.
Otherwise, download and preprocess the reference as described under
Data Preparation and use it as a <strong>genome from my history</strong>.</p>
        </li>
        <li>As the <strong>Aligned reads input datasets</strong> for the analysis select the two
BAM datasets representing the two samples (use the <code class="language-plaintext highlighter-rouge">Control</code> key on the
keyboard to select multiple datasets from the list.</li>
        <li><strong>Execute</strong> the job</li>
      </ul>
    </li>
    <li>Run <strong>MiModD Extract Variant Sites</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><strong>BCF input file</strong> set to the BCF dataset generated in the previous
step.</li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>Approximately how many variants were extracted?</li>
      <li>Does this number surprise you?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>You can click on the extracted variants dataset in your history to
get more details about it. You should see that the file has an
estimated 130,000 lines. Since one variant gets reported per line in VCF format (apart from a number
of header lines), this roughly
equals the number of variants in the dataset.</li>
        <li>
          <p>The huge number of variants reflects the fact that the L<em>er</em>
mapping strain is really quite diverged from the Col reference
background that the mutant was generated in and, thus, offers many
marker variants for mapping.</p>

          <p>In total, the L<em>er</em> genome harbors &gt;400,000 single nucleotide
variants compared to the Col reference, but to keep the tutorial
datasets small we only provided reads mapping to 2 out of 5
chromosomes. In addition, we provided really low-coverage data
even for those two chromosomes so our analysis will miss many true
variants and contain many questionable variant calls that may represent
false-positives. Regardless, an analysis based on such a large overall number
of variants is robust enough to provide useful results.</p>
        </li>
      </ol>
    </blockquote>

  </blockquote>
</blockquote>

<h1 id="linkage-analysis">Linkage Analysis</h1>

<p>The extracted variants dataset obtained in the previous step contains
information about thousands of sites that can be used as markers to map the
causative mutation in the F2 pool. We will now analyze the data with the
<strong>MiModD NacreousMap</strong> tool in <strong>Variant Allele Frequency</strong> mapping mode to
identify these markers and look for linkage between them and the causative
mutation.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-linkage-analysis-with-one-set-of-parentally-contributed-markers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Linkage Analysis with one set of parentally contributed markers</div>

  <p>In the <strong>MiModD NacreousMap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> interface, set</p>
  <ul>
    <li><strong>type of mapping analysis to perform</strong> to <code class="language-plaintext highlighter-rouge">Variant Allele Frequency
Mapping</code> and</li>
    <li><strong>data source to use</strong> to <code class="language-plaintext highlighter-rouge">VCF file of variants</code>.</li>
    <li>As the <strong>input file with variants to analyze</strong> select your extracted
variants VCF dataset obtained in the previous step.</li>
    <li>As the <strong>mapping sample name</strong> use <code class="language-plaintext highlighter-rouge">outcrossed F2</code>.</li>
    <li>Leave the <strong>name of the related parent sample</strong> input field empty because
in this two-sample there is no such sample.</li>
    <li>As the <strong>name of the unrelated parent sample</strong> specify <code class="language-plaintext highlighter-rouge">Ler</code>.
      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-sample-roles"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Sample roles</div>
        <p>In <code class="language-plaintext highlighter-rouge">Variant Allele Frequency Mapping</code> mode the tool requires:</p>
        <ul>
          <li>a <strong>mapping sample</strong>, which corresponds to the recombinant pool, from
which allele frequencies should be calculated</li>
          <li>either a <strong>related</strong> or an <strong>unrelated parent sample</strong> (or both),
from which the marker variant sites to use in the linkage analysis
will be determined. A related parent sample is related (or identical)
to the original mutant line by descent and provides information about
(some of) the variants present in that line, while an unrelated parent
sample provides information about the mapping strain variants
introduced in the mapping cross. If both a related and an unrelated
parent sample are available, the tool can also use the marker sets
from both sides for the linkage analysis.</li>
        </ul>

      </blockquote>
      <p>With these settings, we base the linkage analysis on crossed-in variants
only, <em>i.e.</em>, the tool will look for variants, for which the L<em>er</em> sample
appears homozygous mutant, and determine the contribution of each such
variant allele to the recombinant pool highlighting regions of low
contribution, <em>i.e.</em>, with suspected linkage to the causative variant, in its
output.</p>
    </li>
  </ul>

  <p>Running this job will generate two datasets, one with linkage tables for each
chromosome in the <em>A. thaliana</em> genome, one with graphical representations
of the data.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>Have a look at the linkage plots dataset generated by the tool run. In
which interval on which chromosome would you expect the causative
mutation?</li>
      <li>Are the per-marker scatter plots and the histogram plots in agreement?</li>
      <li>What about those <code class="language-plaintext highlighter-rouge">No markers found!</code> messages (you saw them, right?)
that the tool printed?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>The plots suggest that the causative variant is likely to be found
in the region from 18,000,000 to 19,000,000 on chr2.</li>
        <li>
          <p>Look at the y axis labels! The scatter plots show the rates at
which marker sites appear to recombine with the causative mutation
in the pool of F2 recombinants, while the histograms show linkage
evidence. Hence, a dip of the scatter plot regression line towards zero
corresponds to a peak in the histograms and the two kinds of plots are,
thus, in very good agreement for this analysis.</p>

          <p>In general, the histogram plots are easier to digest and often
provide more precise mapping intervals, and are often all you need for
mapping a causative variant. The per-marker scatter plots, on the other
hand, are closer to the raw data and can sometimes reveal additional
irregularities in the data that may indicate problems with the mapping
cross or unexpected factors influencing the inheritance pattern.</p>
        </li>
        <li>Since we included only reads mapping to chromosome 1 or 2 in
the original BAM files, the tool will not be able to detect any markers
for the other chromosomes.</li>
      </ol>
    </blockquote>

  </blockquote>
</blockquote>

<h1 id="identifying-candidate-mutations">Identifying Candidate Mutations</h1>

<p>With the mapping results obtained above we can now try to get a list of
candidates for the causative mutation. Here is what we know about this mutation
so far:</p>

<ul>
  <li>
    <p>The F2 recombinant pool should appear homozygous mutant for it.</p>

    <p>This is because the F2 recombinants got selected based on the phenotype of
the recessive mutation so we can assume that all individuals in the pool were
homozygous for it.</p>
  </li>
  <li>
    <p>The L<em>er</em> strain should not carry it.</p>
  </li>
  <li>
    <p>It should reside in the genomic region identified in the previous step.</p>
  </li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-variant-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Variant Filtering</div>

  <p>With the above information, we can run <strong>MiModD VCF Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
to obtain a new VCF dataset with only those variants from the originally
extracted ones that fulfill all three criteria. In the tool interface,</p>

  <ul>
    <li>select as <strong>VCF input file</strong> the extracted variants dataset from your
history</li>
    <li>add a <strong>Sample-specific Filter</strong> and set its
      <ul>
        <li><strong>sample</strong> to <code class="language-plaintext highlighter-rouge">outcrossed F2</code></li>
        <li><strong>genotype pattern(s) for the inclusion of variants</strong> to <code class="language-plaintext highlighter-rouge">1/1</code></li>
      </ul>

      <p>These settings will make the filter retain only variants for which the F2
pool has a <code class="language-plaintext highlighter-rouge">1/1</code>, <em>i.e.</em>, homozygous mutant genotype.</p>
    </li>
    <li>add another <strong>Sample-specific Filter</strong> and set its
      <ul>
        <li><strong>sample</strong> to <code class="language-plaintext highlighter-rouge">Ler</code></li>
        <li><strong>genotype pattern(s) for the inclusion of variants</strong> to <code class="language-plaintext highlighter-rouge">0/0</code></li>
      </ul>

      <p>This filter will retain only variants for which the mapping strain sample
has a homozygous wt genotype.</p>
    </li>
    <li>add a <strong>Region Filter</strong> and set its
      <ul>
        <li><strong>Chromosome</strong> to <code class="language-plaintext highlighter-rouge">chr2</code></li>
        <li><strong>Region Start</strong> to 18000000</li>
        <li><strong>Region End</strong> to 19000000</li>
      </ul>

      <p>to retain only variants falling into the previously determined mapping
interval.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-sample-and-chromosome-names"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Sample and chromosome names</div>
        <p>This is the second time in this tutorial that we ask you to type sample
or chromosome names exactly as they appear in a dataset, and so you
may have started wondering how you are supposed to keep track of which
names are used where in your own analyses.</p>

        <p>Because this is a valid concern, the MiModD tool suite features the
<strong>MiModD File Information</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool that can report the
sample and sequence names stored in almost any dataset that you may
encounter in a mapping-by-sequencing analysis (specifically, the
tool can handle datasets in vcf, bcf, sam, bam and fasta format).</p>

      </blockquote>
    </li>
  </ul>

  <p>Running the job yields a new VCF dataset with variants that passed all
filters.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How many variants passed the filters?</li>
      <li>How many variants would you have to consider as candidates if you did
not have any mapping information?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>If you used the suggested mapping interval of 1Mb on chr2, you
should have retained 5 variants. Filtering is a really powerful way to
reduce the number of variants to consider!</li>
        <li>Re-running the job with the region filter removed shows that across
the two chromosomes, for which the input datasets provided sequencing
reads, there are 50 variants for which the pool appears homozygous
mutant and for which the L<em>er</em> strain is homozygous wt.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<p>Not all of the retained variants are equally likely candidates for being the
causative mutation. Since the input datasets consist of reads from whole-genome
sequencing, as opposed to exome sequencing, some of the variants may fall in
intergenic regions or introns where they are unlikely to cause a phenotype.
Other variants may fall in coding regions, but may be silent on the protein
level. Hence, we can prioritize our candidate variants after annotating them
with predicted functional effects.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-variant-annotation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Variant Annotation</div>

  <p>To annotate our variants we are going to use <strong>SnpEff</strong> with its
<code class="language-plaintext highlighter-rouge">Arabidopsis_thaliana</code> genome database as the source of functional genomic
annotations, then generate a human-friendly variant report with the
<strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool.</p>

  <ol>
    <li>
      <p>Use <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to download genome database
<code class="language-plaintext highlighter-rouge">Arabidopsis_thaliana</code>.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-snpeff-versions"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: SnpEff versions</div>
        <p>SnpEff genome databases can only be used with the specific <code class="language-plaintext highlighter-rouge">major.minor</code>
version of SnpEff they were built for.
Make sure you are using matching versions (e.g., <code class="language-plaintext highlighter-rouge">4.3</code> - differences
past the second digit are fine and denote subversions with compatible
database schemes) of <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in this and of
<strong>SnpEff Variant effect and annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the next
step.</p>

        <p>Please also note that before SnpEff version 4.3 the TAIR10 version of
the <em>A. thaliana</em> genome database was named <code class="language-plaintext highlighter-rouge">athalianaTair10</code>.
If your Galaxy server offers only older versions of the SnpEff tools,
you will have to use that name instead of <code class="language-plaintext highlighter-rouge">Arabidopsis_thaliana</code>.</p>
      </blockquote>
    </li>
    <li>In the <strong>SnpEff Variant effect and annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> interface,
set:
      <ul>
        <li><strong>Sequence changes (SNPs, MNPs, InDels)</strong> to your filtered VCF dataset</li>
        <li><strong>Genome source</strong> to <code class="language-plaintext highlighter-rouge">Downloaded snpEff database in your history</code></li>
        <li><strong>SnpEff4.3 Genome Data</strong> to your just downloaded genome dataset</li>
        <li>
          <p>under <strong>Annotation options</strong> check <code class="language-plaintext highlighter-rouge">Use 'EFF' field compatible with
older versions (instead of 'ANN')</code></p>

          <p>This last setting is required for compatibility of the resulting dataset
with the current version of <strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>
        </li>
      </ul>

      <p>Running this job will produce another VCF dataset with the functional
annotation data added to its INFO column. Inspecting this dataset is not
particularly convenient, but you can turn it into a more user-friendly
report by</p>
    </li>
    <li>using <strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and setting:
      <ul>
        <li><strong>The VCF input with the variants to be reported</strong> to the annotated
variants dataset generated by <strong>SnpEff</strong></li>
        <li>the <strong>Format to use for the report</strong> to <code class="language-plaintext highlighter-rouge">HTML</code></li>
        <li>
          <p><strong>Species</strong> to <code class="language-plaintext highlighter-rouge">A. thaliana</code>.</p>

          <p>The species information can be used by the tool to enrich the output
with species-specific database and genome browser links.</p>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>Variant reports in html format are meant to be used on relatively small
lists of candidate variants like in this example. If the input VCF
contains &gt; 1000 variants, the relatively large size of the reports and
slow loading in the browser may become annoying. In this case, or if you
experience formatting problems with the generated html in your browser,
it may be better to choose <code class="language-plaintext highlighter-rouge">Tab-separated plain text</code> as the report
format although the output will then lack those convenient hyperlinks.</p>
      </blockquote>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>The published confirmed mutations in the mutant line are <em>chr2:18,774,111
C→T</em> and <em>chr2:18,808,927 C→T</em>. The latter was shown through follow-up
experiments to be the causative variant.</p>

    <ol>
      <li>What are the functional effects of these two mutations according to the
report?</li>
      <li>What biological functions are described for the two genes? How do these
relate to the phenotype of our mutant line?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>The C→T transition at 18,774,111 results in a premature
translational stop for the transcript AT2G45550.1. The C→T transition
at 18,808,927 affects a splice donor site and, thus, is likely to
interfere with correct splicing of the transcript AT2G45660.1.</li>
        <li>From the linked gene pages at <em>TAIR</em> you can learn that the
gene AT2G45550 encodes a cytochrome P450, while AT2G45660 encodes a
protein called AGL20/SOC1, which is known to control flowering. Thus, there
is a direct connection between the late-flowering phenotype of the
mutant line studied here and the known role of the gene affected by
chr2:18,808,927 C→T.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Mapping-by-sequencing can greatly speed up and facilitate the molecular
identification of mutations recovered from mutagenesis screens. The method
makes extensive use of cross-sample comparison of variants and benefits strongly
from joint variant calling to generate reliable and manageable variant
information. A combination of variant linkage analysis and filtering often
results in very small lists of candidate variants that can then be confirmed
through further experimental work. Importantly, the selection of biological
samples for sequencing determines the meaningful variant comparisons that can
be made in the bioinformatics analysis so it is important to understand the
essence of the analysis method <strong>before</strong> the preparation of any biological
samples.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Mapping-by-sequencing is a powerful method for the molecular identification of mutations</p>
</li>
                        
                        <li><p>The MiModD suite of tools bundles most of the functionality required to perform mapping-by-sequencing analyses with Galaxy</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Mapping and molecular identification of phenotype-causing mutations (Variant Analysis)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier,  <b>Mapping and molecular identification of phenotype-causing mutations (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-mapping-by-sequencing,
author = "Wolfgang Maier",
	title = "Mapping and molecular identification of phenotype-causing mutations (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: a6f147a050a2
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff
  owner: iuc
  revisions: 20f0429a4bfe
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff
  owner: iuc
  revisions: 20f0429a4bfe
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mimodd_main
  owner: wolma
  revisions: f0f2795de2c7
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mimodd_main
  owner: wolma
  revisions: f0f2795de2c7
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mimodd_main
  owner: wolma
  revisions: f0f2795de2c7
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mimodd_main
  owner: wolma
  revisions: f0f2795de2c7
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mimodd_main
  owner: wolma
  revisions: f0f2795de2c7
  tool_panel_section_label: Variant Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00312">https://gxy.io/GTN:T00312</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1702254526'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>