<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1703464120' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1703464120' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-07-27 12:30:19 +0000">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Engy Nasr">
<meta name="DC.creator" content="Paul Zierep">

        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Pathogen detection from (direct Nanopore) sequencing data...">
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-07-27 12:30:19 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html",
  "timeRequired": "PT4H",
  "teaches": "Check quality reports generated by FastQC and NanoPlot for metagenomics Nanopore data\n - Preprocess the sequencing data to remove adapters, poor quality base content and host/contaminating reads\n - Perform taxonomy profiling indicating and visualizing up to species level in the samples\n - Identify pathogens based on the found virulence factor gene products via assembly, identify strains and indicate all antimicrobial resistance genes in samples\n - Identify pathogens via SNP calling and build the consensus gemone of the samples\n - Relate all samples' pathogenic genes for tracking pathogens via phylogenetic trees and heatmaps",
  "keywords": "metagenomics, Nanopore data analysis, Pathogens detection, Phylogenetic tree, Heatmap, cyoa",
  "description": "The questions this  addresses are:\n - What are the preprocessing steps to prepare ONT sequencing data for further analysis?\n - How to identify pathogens using sequencing data?\n - How to track the found pathogens through all your samples datasets?\n\n\nThe objectives are:\n - Check quality reports generated by FastQC and NanoPlot for metagenomics Nanopore data\n - Preprocess the sequencing data to remove adapters, poor quality base content and host/contaminating reads\n - Perform taxonomy profiling indicating and visualizing up to species level in the samples\n - Identify pathogens based on the found virulence factor gene products via assembly, identify strains and indicate all antimicrobial resistance genes in samples\n - Identify pathogens via SNP calling and build the consensus gemone of the samples\n - Relate all samples' pathogenic genes for tracking pathogens via phylogenetic trees and heatmaps\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/EngyNasr/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/EngyNasr/",
      "name": "Engy Nasr",
      "image": "https://avatars.githubusercontent.com/EngyNasr",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9047-4215",
      "orcid": "https://orcid.org/0000-0001-9047-4215"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/paulzierep/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/paulzierep/",
      "name": "Paul Zierep",
      "image": "https://avatars.githubusercontent.com/paulzierep",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-2982-388X",
      "orcid": "https://orcid.org/0000-0003-2982-388X"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org/training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "name": "Nanopore data analysis"
    },
    {
      "@type": "Thing",
      "name": "Pathogens detection"
    },
    {
      "@type": "Thing",
      "name": "Phylogenetic tree"
    },
    {
      "@type": "Thing",
      "name": "Heatmap"
    },
    {
      "@type": "Thing",
      "name": "cyoa"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1703464120'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metagenomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/metagenomics/faqs/" title="Check our FAQs for the Metagenomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-metagenomics">
    <h1 data-toc-skip>Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Avatar" width="36" height="36">Bérénice Batut</a><a href="/training-material/hall-of-fame/EngyNasr/" class="contributor-badge contributor-EngyNasr"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/EngyNasr?s=36" alt="Avatar" width="36" height="36">Engy Nasr</a><a href="/training-material/hall-of-fame/paulzierep/" class="contributor-badge contributor-paulzierep"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/paulzierep?s=36" alt="Avatar" width="36" height="36">Paul Zierep</a>
		</td>
	</tr>
	


	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/hrhotz/" class="contributor-badge contributor-hrhotz"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hrhotz?s=36" alt="Avatar" width="36" height="36">Hans-Rudolf Hotz</a><a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Avatar" width="36" height="36">Wolfgang Maier</a></td>
	</tr>
	

	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the preprocessing steps to prepare ONT sequencing data for further analysis?</p>
</li>
        
        <li><p>How to identify pathogens using sequencing data?</p>
</li>
        
        <li><p>How to track the found pathogens through all your samples datasets?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Check quality reports generated by FastQC and NanoPlot for metagenomics Nanopore data</p>
</li>
        
        <li><p>Preprocess the sequencing data to remove adapters, poor quality base content and host/contaminating reads</p>
</li>
        
        <li><p>Perform taxonomy profiling indicating and visualizing up to species level in the samples</p>
</li>
        
        <li><p>Identify pathogens based on the found virulence factor gene products via assembly, identify strains and indicate all antimicrobial resistance genes in samples</p>
</li>
        
        <li><p>Identify pathogens via SNP calling and build the consensus gemone of the samples</p>
</li>
        
        <li><p>Relate all samples’ pathogenic genes for tracking pathogens via phylogenetic trees and heatmaps</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/7593928">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/" title="Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/metagenomics/pathogen-detection-from-nanopore-foodborne-data/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (May 2023)
            </a>
        </li>
        
        
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jul 27, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00208">https://gxy.io/GTN:T00208</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Food contamination with pathogens is a major burden on our society. In the year 2019, foodborne pathogens caused 137 hospitalisations in Germany <a href="https://www.bvl.bund.de/SharedDocs/Berichte/10_BELA_lebensmittelbed_Krankheitsausbruechen_Dtl/Jahresbericht2019.pdf?__blob=publicationFile&amp;v=4">(BVL 2019)</a>. Globally, they affect an estimated 600 million people a year and impact socioeconomic development at different levels. These outbreaks are mainly due to <em>Salmonella spp.</em> followed by <em>Campylobacter spp.</em> and Noroviruses, as studied by the <a href="https://www.who.int/publications/i/item/9789241565165"><strong>Food safety - World Health Organization (WHO)</strong></a>.</p>

<p>During the investigation of a foodborne outbreak, a microbiological analysis of the potentially responsible food vehicle is performed in order to detect the responsible pathogens and identify the contamination source. By default, the <a href="https://eur-lex.europa.eu/LexUriServ/LexUriServ.do?uri=OJ:L:2005:338:0001:0026:EN:PDF"><strong>European Regulation (EC)</strong></a> follows ISO standards to detect bacterial pathogens in food: pathogens are detected and identified by <strong>stepwise cultures</strong> on selective media and/or <strong>targeting specific genes with real-time PCRs</strong>. The current gold standard is Pulsed-field Gel Electrophoresis (PFGE) or Multiple-Locus Variable Number Tandem Repeat Analysis (MLVA) to characterize the detected strains. These techniques have some disadvantages.</p>

<p><strong>Whole Genome Sequencing</strong> (WGS) has been proposed as an alternative. With just one sequencing run, we can:</p>
<ul>
  <li>detect all genes</li>
  <li>run phylogenetic analysis to link cases</li>
  <li>get information on antimicrobial resistance genes, virulence, serotype, resistance to sanitizers, root cause, and other critical factors in one assay, including historical reference to pathogen emergence.</li>
</ul>

<p>WGS is more than a surveillance tool and was recommended by the European Centre for Disease Prevention and Control (ECDC) and the European Food Safety Authority (EFSA) for surveillance and outbreak investigation. WGS still requires isolation of the targeted pathogen, which is a time-consuming process, the execution is not always straightforward, nor the success is guaranteed. <strong>Sequencing methods without prior isolation could solve this issue</strong>.</p>

<p>The evolution of sequencing techniques in the last decades has made the development of shotgun metagenomic sequencing possible, <em>i.e.</em> the <strong>direct sequencing of all DNA present in a sample</strong>. This approach gives an overview of the genomic composition of all cells in the sample, including the food source itself, the microbial community, and any possible pathogens and their complete genetic information without the need for prior isolation. Several studies have demonstrated the potential of shotgun metagenomics to identify and characterize pathogens and their functional characteristics (<em>e.g.</em> virulence genes) in naturally contaminated or purposefully spiked food samples.</p>

<p>The currently available studies used Illumina sequencing, generating short reads. Longer read lengths, generated by third-generation sequencing platforms such as Pacific Biosciences (PacBio) and Oxford Nanopore Technologies (ONT), make it <strong>easier and more practical to identify strains with fewer reads</strong>. MinION (from Oxford Nanopore) is a portable, real-time device for ONT sequencing. Several proof-of-principle studies have shown the <strong>utility of ONT long-read sequencing from metagenomic samples for pathogen identification</strong> (<span class="citation"><a href="#CIUFFREDA20211497">Ciuffreda <i>et al.</i> 2021</a></span>).</p>

<!--SNIPPET-->
<blockquote class="comment">   <div class="box-title comment-title" id="comment-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Nanopore sequencing</div>   <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>   <ol>   <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>   <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>   <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li> </ol>   <p><img src="/training-material/topics/metagenomics/images/plasmid-metagenomics-nanopore/nanopore_sequencing.png" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://doi.org/10.7875/togopic.2020.01&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>   <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change in electrical current is measured over the membrane of a flow cell. When nucleotides pass the pores in the flow cell the current change is translated (basecalled) to nucleotides by a basecaller. A schematic overview is given in the picture above.</p>   <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software is present on the devices. With basecalling the electrical signals are translated to bases (A,T,G,C) with a quality score per base. The sequenced DNA strand will be basecalled and this will form one read. Multiple reads will be stored in a fastq file.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<p>To identify and track foodborne pathogens using long-read metagenomic sequencing, different samples of potentially contaminated food (at different time points or different locations) are prepared, DNA is extracted and sequenced using MinION (ONT). The generated sequencing data then need to be processed using bioinformatics tools.</p>

<p>In this tutorial, we will be presenting a series of Galaxy workflows whose main goals are to:</p>
<ol>
  <li><strong>agnostically detect pathogens</strong> (What exactly is this pathogen and what virulence factors does it carry?) from data extracted directly (without prior cultivation) from a potentially contaminated sample (e.g. food like chicken, beef, etc.) and sequenced using Nanopore</li>
  <li><strong>compare different samples to trace</strong> the possible source of contamination</li>
</ol>

<p>To illustrate how to process such data, we will use datasets generated by <a href="https://www.biolytix.ch/">Biolytix</a> with the following approach:</p>

<p><a href="./images/Biolytix_Workflow_Sequencing.png" rel="noopener noreferrer"><img src="./images/Biolytix_Workflow_Sequencing.png" alt="From left to right: Biolytix logo. Chicken + milk. An arrow going to the right toward Chicken + milk and a syringe with &quot;Contamination with known pathogens&quot; written below. An arrow going to the right toward an Eppendorf tube with &quot;DNA extraction&quot; written below,  An arrow going to the right toward a qPCR machine, and another arrow over the qPCR toward a Nanopore sequencing. . " width="1589" height="387" loading="lazy" /></a></p>

<p>Food samples, here chicken, are spiked with known pathogens, here:</p>
<ul>
  <li><em>Salmonella enterica subsp. enterica</em>  in the sample named <code class="language-plaintext highlighter-rouge">Barcode 10 Spike 2</code></li>
  <li><em>Salmonella enterica subsp. houtenae</em> in the sample named <code class="language-plaintext highlighter-rouge">Barcode 11 Spike 2b</code></li>
</ul>

<p>DNA in the samples is extracted, analyzed with qPCR, and sequenced via Nanopore. We start the tutorial from raw data generated by Nanopore.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#prepare-galaxy-and-data" id="markdown-toc-prepare-galaxy-and-data">Prepare Galaxy and data</a></li>
  <li><a href="#pre-processing" id="markdown-toc-pre-processing">Pre-Processing</a>    <ol>
      <li><a href="#quality-control-and-preprocessing" id="markdown-toc-quality-control-and-preprocessing">Quality Control and preprocessing</a></li>
      <li><a href="#host-read-filtering" id="markdown-toc-host-read-filtering">Host read filtering</a></li>
    </ol>
  </li>
  <li><a href="#taxonomy-profiling" id="markdown-toc-taxonomy-profiling">Taxonomy Profiling</a></li>
  <li><a href="#gene-based-pathogenic-identification" id="markdown-toc-gene-based-pathogenic-identification">Gene based pathogenic identification</a>    <ol>
      <li><a href="#assembly" id="markdown-toc-assembly">Assembly</a></li>
      <li><a href="#multilocus-sequence-typing" id="markdown-toc-multilocus-sequence-typing">Multilocus Sequence Typing</a></li>
      <li><a href="#antimicrobial-resistance-genes" id="markdown-toc-antimicrobial-resistance-genes">Antimicrobial Resistance Genes</a></li>
      <li><a href="#virulence-factor-identification" id="markdown-toc-virulence-factor-identification">Virulence Factor identification</a></li>
    </ol>
  </li>
  <li><a href="#snp-based-pathogenic-identification" id="markdown-toc-snp-based-pathogenic-identification">SNP based pathogenic identification</a>    <ol>
      <li><a href="#variant-calling-or-snp-calling" id="markdown-toc-variant-calling-or-snp-calling">Variant Calling or SNP Calling</a></li>
      <li><a href="#consensus-genome-building" id="markdown-toc-consensus-genome-building">Consensus Genome Building</a></li>
    </ol>
  </li>
  <li><a href="#pathogen-tracking-among-all-samples" id="markdown-toc-pathogen-tracking-among-all-samples">Pathogen Tracking among all samples</a>    <ol>
      <li><a href="#heatmap" id="markdown-toc-heatmap">Heatmap</a></li>
      <li><a href="#phylogenetic-tree-building" id="markdown-toc-phylogenetic-tree-building">Phylogenetic Tree building</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="prepare-galaxy-and-data">Prepare Galaxy and data</h1>
<p>Any analysis should get its own Galaxy history. So let’s start by creating a new one:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this analysis</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>Before we can begin any Galaxy analysis, we need to upload the input data: FASTQ files with the sequenced samples.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import datasets</div>

  <ol>
    <li>
      <p>Import the following samples via link from <a href="https://zenodo.org/record/7593928">Zenodo</a> or Galaxy shared data libraries:</p>

      <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7593928/files/Barcode10_Spike2.fastq.gz
https://zenodo.org/record/7593928/files/Barcode11_Spike2b.fastq.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the files to <code class="language-plaintext highlighter-rouge">Barcode10</code> and <code class="language-plaintext highlighter-rouge">Barcode11</code> respectively</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a dataset<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Create a collection named <code class="language-plaintext highlighter-rouge">Samples</code> that includes both datasets (<code class="language-plaintext highlighter-rouge">Barcode10</code> and <code class="language-plaintext highlighter-rouge">Barcode11</code>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a dataset collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>In this tutorial, we can offer 2 versions:</p>
<ul>
  <li>A short version, running prebuilt workflows</li>
  <li>A long version, going step-by-step</li>
</ul>

<blockquote class="notranslate hands_on gtn-cyoa" id="gtn-cyoa">
	<div class="box-title" aria-label="hands-on box: choose your own tutorial"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Choose Your Own Tutorial</div>
	<p>This is a "Choose Your Own Tutorial" section, where you can select between multiple paths. Click one of the buttons below to select how you want to follow the tutorial</p>
	
	
		<input id="cyoa-opt1" type="radio" name="cyoa" value="Short-Version" onclick="cyoaChoice('Short-Version')" />
		<label for="cyoa-opt1" class="select btn btn-secondary">Short Version</label>
	
	
		<input id="cyoa-opt2" type="radio" name="cyoa" value="Long-Version" onclick="cyoaChoice('Long-Version')" />
		<label for="cyoa-opt2" class="select btn btn-secondary">Long Version</label>
	
	
	

	
	<script>
		function ready(fn) {
			if (document.readyState != 'loading'){
				fn();
			} else {
				document.addEventListener('DOMContentLoaded', fn);
			}
		}
		ready(function() {
			cyoaDefault("Short-Version");
		})
	</script>
	
</blockquote>

<h1 id="pre-processing">Pre-Processing</h1>

<p>Before starting any analysis, it is always a good idea to assess the quality of your input data and to discard poor quality base content by trimming and filtering reads.</p>

<div class="Short-Version">

  <p>In this section we will run a Galaxy workflow that performs the following tasks with the following tools:</p>
  <ol>
    <li>Assess the reads quality before and after preprocessing it using <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><strong>FastQC</strong></a>, <a href="https://github.com/wdecoster/NanoPlot"><strong>NanoPlot</strong></a> and  <a href="https://multiqc.info/"><strong>MultiQC</strong></a> (<span class="citation"><a href="#Ewels2016">Ewels <i>et al.</i> 2016</a></span>)</li>
    <li>Trimming and filtering reads by length and quality using <a href="https://github.com/rrwick/Porechop"><strong>Porechop</strong></a> and <strong>Fastp</strong> (<span class="citation"><a href="#Chen2018">Chen <i>et al.</i> 2018</a></span>)</li>
    <li>Remove all possible hosts sequences e.g. chicken, cow, etc. using <a href="https://ccb.jhu.edu/software/kraken2/"><strong>Kraken2</strong></a> (<span class="citation"><a href="#Wood2014">Wood and Salzberg 2014</a></span>) with the <a href="https://github.com/lskatz/Kalamari"><strong>Kalamari</strong></a> database, and <a href="https://github.com/jenniferlu717/KrakenTools/blob/master/extract_kraken_reads.py"><strong>Krakentools: Extract Kraken Reads By ID</strong></a> to remove all the hosts sequences before moving on to the next section with only the non-host sequences.</li>
  </ol>

  <p>We will run all these steps using a single workflow, then discuss each step and the results in more detail.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-pre-processing"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Pre-Processing</div>

    <ol>
      <li><strong>Import the workflow</strong> into Galaxy
        <ul>
          <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/nanopore_preprocessing.ga">this workflow</a> or download it to your computer.</li>
          <li>Import the workflow into Galaxy</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>
        <p>Run <strong>Workflow 1:  Nanopore Datasets - Pre-Processing</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>

        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“1: Collection of all samples”</em>: <code class="language-plaintext highlighter-rouge">Samples</code> collection created from the imported Fastq.qz files</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>The workflow will take a little while to complete. Once tools have completed, the results will be available in your history for viewing. Note that only the most important outputs will be visible; intermediate files are hidden by default.</p>

  <p>While you are waiting for the workflow to complete, please continue reading in the next section(s) where we will go into a bit more detail about what happens at each step of the workflow we launched and examine the results.</p>
</div>

<h2 id="quality-control-and-preprocessing">Quality Control and preprocessing</h2>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data. <strong>Sequence quality control is therefore an essential first step in your analysis.</strong></p>

<p>In this tutorial we use similar tools as described in the tutorial <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control”</a>, but more specific to Nanopore data:</p>

<ul>
  <li>Quality control with
    <ul>
      <li><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"><strong>FastQC</strong></a> generates a web report that will aid you in assessing the quality of your data</li>
      <li><a href="https://github.com/wdecoster/NanoPlot"><strong>NanoPlot</strong></a> plotting tool for long read sequencing data and alignments</li>
    </ul>

    <div class="Long-Version">

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-initial-quality-assessment"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Initial quality assessment</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Raw read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">Samples</code> collection created from the imported Fastq.qz files</li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1" title="NanoPlot tool" aria-role="link"><strong>NanoPlot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1</span></span> with the following parameters:
            <ul>
              <li><em>“Select multifile mode”</em>: <code class="language-plaintext highlighter-rouge">batch</code>
                <ul>
                  <li><em>“Type of the file(s) to work on”</em>: <code class="language-plaintext highlighter-rouge">fastq</code>
                    <ul>
                      <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Data input files”</em>: <code class="language-plaintext highlighter-rouge">Samples</code> collection created from the imported Fastq.qz files</li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>

            <blockquote class="comment">
              <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
              <p>This step, as it does not require the results of FastQC to run, can be launched even if FastQC is not ready</p>
            </blockquote>
          </li>
        </ol>

      </blockquote>

    </div>
  </li>
  <li>
    <p>Read trimming and filtering with <a href="https://github.com/rrwick/Porechop"><strong>Porechop</strong></a> and <strong>Fastp</strong> (<span class="citation"><a href="#Chen2018">Chen <i>et al.</i> 2018</a></span>)</p>

    <div class="Long-Version">

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-read-trimming-and-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Read trimming and filtering</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.4" title="Porechop tool" aria-role="link"><strong>Porechop</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.4</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input FASTA/FASTQ”</em>: <code class="language-plaintext highlighter-rouge">Samples</code> collection created from the imported Fastq.qz files</li>
              <li><em>“Output format for the reads”</em>: <code class="language-plaintext highlighter-rouge">fastq.gz</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0" title="fastp tool" aria-role="link"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0</span></span> with the following parameters:
            <ul>
              <li><em>“Single-end or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
                <ul>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input 1”</em>: outputs of <strong>Porechop</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                </ul>
              </li>
              <li>In <em>Output Options</em>
                <ul>
                  <li><em>“Output JSON report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                </ul>
              </li>
            </ul>

            <blockquote class="comment">
              <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
              <p>This step can be launched even if <strong>Porechop</strong> is not done. It will be scheduled and wait until <strong>Porechop</strong> is done to start.</p>
            </blockquote>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
  <li>
    <p>Quality recheck after read trimming and filtering with <strong>FastQC</strong> and <strong>Nanoplot</strong> and report aggregation with <a href="https://multiqc.info/"><strong>MultiQC</strong></a></p>

    <div class="Long-Version">

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-final-quality-checks"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Final quality checks</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FastQC tool" aria-role="link"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Raw read data from your current history”</em>: outputs of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1" title="NanoPlot tool" aria-role="link"><strong>NanoPlot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/nanoplot/nanoplot/1.28.2+galaxy1</span></span> with the following parameters:
            <ul>
              <li><em>“Select multifile mode”</em>: <code class="language-plaintext highlighter-rouge">batch</code>
                <ul>
                  <li><em>“Type of the file(s) to work on”</em>: <code class="language-plaintext highlighter-rouge">fastq</code>
                    <ul>
                      <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“files”</em>: outputs of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0" title="MultiQC tool" aria-role="link"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy0</span></span> with the following parameters:
            <ul>
              <li>In <em>“Results”</em>:
                <ul>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Results”</em>
                    <ul>
                      <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code>
                        <ul>
                          <li>In <em>“FastQC output”</em>:
                            <ul>
                              <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert FastQC output”</em>
                                <ul>
                                  <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: 4 <code class="language-plaintext highlighter-rouge">Raw data</code> outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                                </ul>
                              </li>
                            </ul>
                          </li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Results”</em>
                    <ul>
                      <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">fastp</code>
                        <ul>
                          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Output of fastp”</em>: <code class="language-plaintext highlighter-rouge">JSON report</code> outputs of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect the HTML output of <strong>MultiQC</strong> for <code class="language-plaintext highlighter-rouge">Barcode10</code></p>

  <ol>
    <li>How many sequences does <code class="language-plaintext highlighter-rouge">Barcode10</code> contain before and after trimming?</li>
    <li>What is the quality score over the reads before and after trimming? And the mean score?</li>
    <li>What is the importance of <strong>FastQC</strong>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Before trimming the file has 114,344 sequences and After trimming the file has 91,434 sequences</li>
      <li>
        <p>The “Per base sequence quality” is globally medium: the quality score stays above 20 over the entire length of reads after trimming, while quality below 20 could be seen before trimming specially at the beginning and the end of the reads.</p>

        <p><a href="./images/fastqc_per_base_sequence_quality_plot_barcode10.png" rel="noopener noreferrer"><img src="./images/fastqc_per_base_sequence_quality_plot_barcode10.png" alt="Sequence Quality. " width="1200" height="800" loading="lazy" /></a></p>
      </li>
      <li>After checking what is wrong, e.g. before trimming, we should think about the errors reported by <strong>FastQC</strong>: they may come from the type of sequencing or what we sequenced (check the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control” training</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> for more details). However, despite these challenges, we can already see sequences getting slightly better after the trimming and filtering, so now we can proceed with our analyses.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>For more information about how to interpret the plots generated by <strong>FastQC</strong> and <strong>MultiQC</strong>, please see our dedicated <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">“Quality control”</a> Tutorial.</p>
</blockquote>

<h2 id="host-read-filtering">Host read filtering</h2>

<p>Generally, we are not interested in the food (host) sequences, rather only those originating from the pathogen itself. It is an important to get rid of all host sequences and to only retain sequences that might include a pathogen, both in order to speed up further steps and to avoid host sequences compromising the analysis.</p>

<p>In this tutorial, we know the samples come from <strong>chicken</strong> meat spiked with <strong><em>Salmonella</em></strong> so we already know what will we get as the host and the main pathogen.</p>

<p>In this tutorial we use:</p>
<ol>
  <li>
    <p>Assign reads to taxa using <strong>Kraken2</strong> (<span class="citation"><a href="#Wood2014">Wood and Salzberg 2014</a></span>) and <strong>Kalamari</strong>, a database of completed assemblies for metagenomics-related tasks used widely in contamination and host filtering</p>

    <div class="Long-Version">

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-read-taxonomic-classification-for-host-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read taxonomic classification for host filtering</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1" title="Kraken2 tool" aria-role="link"><strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1</span></span> with the following parameters:
            <ul>
              <li><em>“Single or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
                <ul>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input sequences”</em>: collection output of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                </ul>
              </li>
              <li><em>“Print scientific names instead of just taxids”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li>In <em>“Create Report”</em>:
                <ul>
                  <li><em>“Print a report with aggregrate counts/clade to file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  <li><em>“Format report output like Kraken 1’s kraken-mpa-report”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                  <li><em>“Report counts for ALL taxa, even if counts are zero”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  <li><em>“Report minimizer data”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                </ul>
              </li>
              <li><em>“Select a Kraken2 database”</em>: <code class="language-plaintext highlighter-rouge">kalamari</code></li>
            </ul>
          </li>
        </ol>

      </blockquote>

    </div>

    <blockquote class="question">
      <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

      <p>Inspect the report of <strong>Kraken2</strong> collection for <code class="language-plaintext highlighter-rouge">Barcode10</code></p>

      <ol>
        <li>What is the species of the host?</li>
        <li>How many sequences of this host was found?</li>
      </ol>

      <blockquote class="solution">
        <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li><em>Gallus gallus</em> (taxid 9031), which is chicken</li>
          <li>836</li>
        </ol>

      </blockquote>
    </blockquote>
  </li>
  <li>
    <p>Filter host assigned reads based on <strong>Kraken2</strong> assignments</p>
    <ol>
      <li>Manipulate <strong>Kraken2</strong> classification to extract the sequence ids of all hosts sequences identified with <strong>Kraken2</strong></li>
      <li>Filter the FASTQ files to get 1 ouput with the host-assigned sequences and 1 output without the host-assigned reads</li>
    </ol>

    <div class="Long-Version">

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-host-read-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Host read filtering</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads/1.2+galaxy1" title="Krakentools: Extract Kraken Reads By ID tool" aria-role="link"><strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/krakentools_extract_kraken_reads/krakentools_extract_kraken_reads/1.2+galaxy1</span></span> with the following parameters:
            <ul>
              <li><em>“Single or paired reads?”</em>: <code class="language-plaintext highlighter-rouge">Single</code></li>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Results”</em>: <code class="language-plaintext highlighter-rouge">Kraken2 with Kalamri database Results</code> outputs of <strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Report”</em>: <code class="language-plaintext highlighter-rouge">Kraken2 with Kalamri database Report</code> outputs of <strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li>
                <p><em>“Taxonomix ID(s) to match”</em>:<code class="language-plaintext highlighter-rouge">9031 9606 9913</code></p>

                <p>We specify here the taxonomic ID of the hosts so we can filter reads assigned to these hosts. Kraken2 uses taxonomic IDs from NCBI, the IDs for a specific taxa can be found at <a href="https://www.ncbi.nlm.nih.gov/taxonomy">ncbi</a>. To be generic, we remove here:</p>
                <ul>
                  <li>Human (<code class="language-plaintext highlighter-rouge">9606</code>)</li>
                  <li>Chicken (<code class="language-plaintext highlighter-rouge">9031</code>)</li>
                  <li>Beef (<code class="language-plaintext highlighter-rouge">9913</code>)</li>
                </ul>

                <p>If the contaminated food comes from and may include other animals, you can change the value here.</p>
              </li>
              <li><em>“Invert output”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Output as FASTQ”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Include parents”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Include children”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
</ol>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>We will need the outputs from this section in the next one.  If yours is still running or you get an error you can go on and upload it so you can start the next workflow, the next hands-on is optional.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-optional-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optional Data upload</div>

    <ol>
      <li>
        <p>Import the quality processed samples fastqsanger files via link from <a href="https://zenodo.org/record/7593928">Zenodo</a> or the Shared Data library:</p>

        <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7593928/files/Nanopore_processed_sequenced_reads_Barcode10_Spike2.fastqsanger
https://zenodo.org/record/7593928/files/Nanopore_processed_sequenced_reads_Barcode11_Spike2b.fastqsanger
</code></pre></div>        </div>
      </li>
      <li>
        <p>Rename datasets to <code class="language-plaintext highlighter-rouge">Barcode10</code> and <code class="language-plaintext highlighter-rouge">Barcode11</code> respectively</p>
      </li>
      <li>
        <p>Create a collection named <code class="language-plaintext highlighter-rouge">Nanopore processed sequenced reads</code> from the two imported datasets</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="taxonomy-profiling">Taxonomy Profiling</h1>

<p>In this section we would like to identify the different organisms found in our samples by assigning taxonomy levels to the reads starting from the kingdom level down to the species level and visualize the result. It’s important to check what might be the species of a possible pathogen to be found, it gets us closer to the investigation as well as discovering possible multiple food infections if any existed.</p>

<!--SNIPPET-->
<blockquote class="details">   <div class="box-title details-title" id="details-what-is-taxonomy"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-taxonomy" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> Details: What is Taxonomy?<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Taxonomy is the method used to naming, defining (circumscribing) and classifying groups of biological organisms based on shared characteristics such as morphological characteristics, phylogenetic characteristics, DNA data, etc. It is founded on the concept that the similarities descend from a common evolutionary ancestor.</p>   <p>Defined groups of organisms are known as <strong>taxa</strong>. Taxa are given a taxonomic rank and are aggregated into super groups of higher rank to create a taxonomic hierarchy. The taxonomic hierarchy includes eight levels: <strong>Domain, Kingdom, Phylum, Class, Order, Family, Genus and Species.</strong></p>   <p><img src="/training-material/topics/metagenomics/faqs/images/taxonomy.png" alt="Example of taxonomy. It starts, top to bottom, with Kingdom &quot;Animalia&quot;, Phylum &quot;Chordata&quot;, Class &quot;Mammalia&quot;, and Order &quot;Carnivora&quot;. Then it splits in 3. On the left, Family &quot;Felidae&quot;, with 2 genus &quot;Felis&quot; and &quot;Panthera&quot; and below 3 species &quot;F. catus&quot; and &quot;F. pardalis&quot; below &quot;Felis&quot;, &quot;P. pardus&quot; below &quot;Panthera&quot;. In the middle, Family &quot;Canidae&quot;, genus &quot;Canis&quot; and 2 species &quot;C. familiaris&quot; and &quot;C. lupus&quot;. On the right, Family &quot;Ursidae&quot;, Genus &quot;Ursus&quot; and 2 species &quot;U. arctos&quot; and &quot;U. horribilus&quot;. Below each species is a illustration of the species" /></p>   <p>The classification system begins with <strong>3 domains</strong> that encompass all living and extinct forms of life</p>   <ul>   <li>The <strong>Bacteria</strong> and <strong>Archae</strong> are mostly microscopic, but quite widespread.</li>   <li>Domain <strong>Eukarya</strong> contains more complex organisms</li> </ul>   <p>When new species are found, they are assigned into taxa in the taxonomic hierarchy. For example for the cat:</p>   <table>   <thead>     <tr>       <th>Level</th>       <th>Classification</th>     </tr>   </thead>   <tbody>     <tr>       <td>Domain</td>       <td>Eukaryota</td>     </tr>     <tr>       <td>Kingdom</td>       <td>Animalia</td>     </tr>     <tr>       <td>Phylum</td>       <td>Chordata</td>     </tr>     <tr>       <td>Class</td>       <td>Mammalia</td>     </tr>     <tr>       <td>Order</td>       <td>Carnivora</td>     </tr>     <tr>       <td>Family</td>       <td>Felidae</td>     </tr>     <tr>       <td>Genus</td>       <td><em>Felis</em></td>     </tr>     <tr>       <td>Species</td>       <td><em>F. catus</em></td>     </tr>   </tbody> </table>   <p>From this classification, one can generate a tree of life, also known as a phylogenetic tree. It is a rooted tree that describes the relationship of all life on earth. At the root sits the “last universal common ancestor” and the three main branches (in taxonomy also called domains) are bacteria, archaea and eukaryotes. Most important for this is the idea that all life on earth is derived from a common ancestor and therefore when comparing two species, you will -sooner or later- find a common ancestor for all of them.</p>   <p>Let’s explore taxonomy in the Tree of Life, using <a href="https://lifemap.univ-lyon1.fr/">Lifemap</a></p>   <iframe id="Lifemap" src="https://lifemap.univ-lyon1.fr/explore.html" frameborder="0" width="100%" height="600px"></iframe> </blockquote>
<p><!--END_SNIPPET--></p>

<p>In the previous section we ran <strong>Kraken2</strong> along with the <strong>Kalamari</strong> database, which is also a kind of taxonomy profiling but the database used is designed to include all possible host sequences. In the following part, we run <strong>Kraken2</strong> again; but this time with one of its built-in databases, <strong>Standard PlusPF</strong>, which can give us more insight into pathogen candidate species than <strong>Kalamari</strong>. You can test this yourself by comparing reports of both <strong>Kraken2</strong> runs.</p>

<!--SNIPPET-->
<blockquote class="details">   <div class="box-title details-title" id="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification"><button class="gtn-boxify-button details" type="button" aria-controls="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> Details: Kraken2 and the k-mer approach for taxonomy classification<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In the \(k\)-mer approach for taxonomy classification, we use a database containing DNA sequences of genomes whose taxonomy we already know. On a computer, the genome sequences are broken into short pieces of length \(k\) (called \(k\)-mers), usually 30bp.</p>   <p><strong>Kraken</strong> examines the \(k\)-mers within the query sequence, searches for them in the database, looks for where these are placed within the taxonomy tree inside the database, makes the classification with the most probable position, then maps \(k\)-mers to the lowest common ancestor (LCA) of all genomes known to contain the given \(k\)-mer.</p>   <p><img src="/training-material/topics/metagenomics/faqs/images/kmers-kraken.jpg" alt="Kraken2" title="Kraken sequence classification algorithm. To classify a sequence, each k-mer in the sequence is mapped to the lowest common ancestor (LCA, i.e. the lowest node) of the genomes that contain that k-mer in the database. The taxa associated with the sequence's k-mers, as well as the taxa's ancestors, form a pruned subtree of the general taxonomy tree, which is used for classification. In the classification tree, each node has a weight equal to the number of k-mers in the sequence associated with the node's taxon. Each root-to-leaf (RTL) path in the classification tree is scored by adding all weights in the path, and the maximal RTL path in the classification tree is the classification path (nodes highlighted in yellow). The leaf of this classification path (the orange, leftmost leaf in the classification tree) is the classification used for the query sequence. Source: &lt;span class=&quot;citation&quot;&gt;&lt;a href=&quot;#Wood2014&quot;&gt;Wood and Salzberg 2014&lt;/a&gt;&lt;/span&gt;" /></p>   <p><strong>Kraken2</strong> uses a compact hash table, a probabilistic data structure that allows for faster queries and lower memory requirements. It applies a spaced seed mask of <em>s</em> spaces to the minimizer and calculates a compact hash code, which is then used as a search query in its compact hash table; the lowest common ancestor (LCA) taxon associated with the compact hash code is then assigned to the k-mer.</p>   <p>You can find more information about the <strong>Kraken2</strong> algorithm in the paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0"><em>Improved metagenomic analysis with Kraken 2</em></a>.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<div class="Short-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-taxonomy-profiling-and-visualisation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Taxonomy Profiling and visualisation</div>

    <ol>
      <li><strong>Import the workflow</strong> into Galaxy
        <ul>
          <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/nanopore_taxonomy_profiling_and_visualization.ga">this workflow</a> or download it to your computer.</li>
          <li>Import the workflow into Galaxy</li>
        </ul>
      </li>
      <li>Run <strong>Workflow 2: Nanopore Datasets - Taxonomy Profiling and Visualization</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
        <ul>
          <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Nanopore Sequenced Reads Collection”</em>: <code class="language-plaintext highlighter-rouge">Nanopore processed sequenced reads</code> collection, output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preproceesing workflow</li>
          <li><em>“Sample Metadata”</em>: Leave empty</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

</div>

<p>To assign reads to taxons, we use <strong>Kraken2</strong> with <strong>Standard PlusPF</strong> database.</p>

<div class="Long-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-taxonomy-profiling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Taxonomy Profiling</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1" title="Kraken2 tool" aria-role="link"><strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.1.1+galaxy1</span></span> with the following parameters:
        <ul>
          <li><em>“Single or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input sequences”</em>: collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing section</li>
            </ul>
          </li>
          <li>In <em>“Create Report”</em>:
            <ul>
              <li><em>“Print a report with aggregrate counts/clade to file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            </ul>
          </li>
          <li><em>“Select a Kraken2 database”</em>: <code class="language-plaintext highlighter-rouge">Prebuilt Refseq indexes:  PlusPF (Standard plus protozoa and fungi) (Version:  2022-06-07 - Downloaded: 2022-09-04T165121Z)</code></li>
        </ul>
      </li>
    </ol>
  </blockquote>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect the <strong>Kraken2</strong> report for <code class="language-plaintext highlighter-rouge">Barcode10</code></p>

  <ol>
    <li>What is the most commonly found species?</li>
    <li>What is the second most commonly found species?</li>
    <li>How many sequences are classified and how many are unclassified?</li>
    <li>What are the differences between <strong>Kraken2</strong> tool’s report with <strong>Kalamari</strong> database and <strong>Kraken2</strong> tool’s report with <strong>Standard PlusPF</strong> database regarding the previous 3 questions?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li><em>Escherichia coli</em> with 10,243 sequences</li>
      <li><em>Salmonella enterica</em> with 7,457 sequences</li>
      <li>40,113 sequences are classified and 50,455 are unclassified</li>
      <li>With <strong>Kalamari</strong> database the most found species is <em>Escherichia coli</em> with 12,577 sequences and the second most found species is <em>Salmonella enterica</em> with 10,632 sequences. The number of classified sequences are 32,020 sequences and the unclassified sequences are 59,414. In conclusion, both databases are able to show the same results of the most common species. However, the number of the classified sequences with <strong>Standard PlusPF</strong> database is higher than <strong>Kalamari</strong> database and it would be even higher since all chicken sequences were removed before testing the <strong>Standard PlusPF</strong> database.</li>
    </ol>
  </blockquote>
</blockquote>

<p>In order to view the taxonomy profiling produced by <strong>Kraken2</strong> tool, there are a lot of tools to be used afterwards such as <strong>Krona pie chart</strong>, however too many species were detected to be shown by this tool. For that reason, we have chosen the <strong>Phinch visualization</strong> interactive tool as it contains multiple visualization plots, it is interactive alowing you to choose between different parameters, you can visualize each taxonomic level on its own, you can have the metadata of the samples represented along with the taxonomic visualization, download all plots for publications and a lot of other benefits. For later, you can check out <a href="https://academic.oup.com/bioinformatics/article/36/4/1303/5573755"><strong>Pavian</strong></a> tool as well it can replace <strong>Phinch visualization</strong> with similar outputs.</p>

<p><strong>Phinch visualization</strong> needs a <strong>BIOM</strong> file format as an input, and for that we need the <a href="https://github.com/smdabdoub/kraken-biom"><strong>Kraken-Biom</strong></a>  tool to convert the <strong>Kraken2</strong> tabular output into a Biom file.</p>

<div class="Long-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-biom-generation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  BIOM generation</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/kraken_biom/kraken_biom/1.2.0+galaxy1" title="Kraken-biom tool" aria-role="link"><strong>Kraken-biom</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/kraken_biom/kraken_biom/1.2.0+galaxy1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input files to Kraken-biom: Kraken report output file(s)”</em>: Report collection output of <strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sample metadata file”</em>: Leave empty</li>
          <li><em>“Do you want to create an OTU IDs file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          <li><em>“Output Format”</em>: <code class="language-plaintext highlighter-rouge">JSON</code></li>
        </ul>
      </li>
    </ol>

  </blockquote>

</div>

<p>Once the BIOM file has been generated, we launch the interactive visualisation tool called <a href="https://www.phinch.org/"><strong>Phinch visualization</strong></a>:</p>

<div class="Long-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-visualisation-with-phinch"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Visualisation with Phinch</div>

    <ol>
      <li><span class="tool" data-tool="interactive_tool_phinch" title="Phinch Visualisation tool" aria-role="link"><strong>Phinch Visualisation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_phinch</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Biom1 dataset”</em>: output of <strong>Kraken-biom</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
    </ol>

  </blockquote>

</div>

<p>Now let’s explore the <strong>Phinch visulization</strong> tool running for <code class="language-plaintext highlighter-rouge">Barcode11</code></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-explore-data-interactively"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Explore data interactively</div>

  <ol>
    <li>
      <p>Open Phinch interactive tool</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-open-interactive-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-open-interactive-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Open interactive tool<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to User &gt; Active InteractiveTools</li>   <li>Wait for the Phinch visualization to be running (Job Info)</li>   <li>Click on Phinch visualization</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Click on <strong>Proceed to Gallery</strong> button on the top right of the opened webpage to see all the plots</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What is the most commonly found species?</li>
    <li>What is the second most commonly found species?</li>
    <li>What’s your favorite visualization plot?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li><em>Salmonella enterica</em> with 17,309 sequences</li>
      <li><em>Pseudomonas lundensis</em> with 13,227 sequences</li>
      <li>
        <p>All of them are good visualization of the data but for us to answer these questions, we used the <strong>Taxonomy Bar Chart</strong></p>

        <p><a href="./images/phinch_taxonomy_bar_chart.png" rel="noopener noreferrer"><img src="./images/phinch_taxonomy_bar_chart.png" alt="Taxonomy Bar Chart. " width="1920" height="585" loading="lazy" /></a></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>While these steps are running, you can move on to the next section <strong>Gene based pathogenic identification</strong> and run the steps there, as well. Both analyses can execute in parallel.</p>
</blockquote>

<p>You may have noticed some sequences have been assigned to the Human Genome (Homosapians) species, when we run <strong>Kraken2</strong> using the <strong>Standard PlusPF</strong> in this section. However, in the <strong>pre-processing</strong> section when we ran <strong>Kraken2</strong> with <strong>Kalamari</strong> no Human Genomes were found. The lab (data producers) has confirmed that these sequences assigned to human by <strong>Standard PlusPF</strong> database are not human and there should be no human sequences in the samples as <strong>Kalamari</strong> database result’s confirmed. So these sequences were wrongly assigned to human by <strong>Standard PlusPF</strong>. That is due to resemblance between organisms and the limited species coverage of <strong>Kraken2</strong> databases sometimes do happen that reads corresponding to higher organisms get mapped to humans. It was a very severe problem for the <strong>Standard PlusPF</strong>, because yeast genes were mis-assigned to human.</p>

<p>We decide to keep these sequences since we do not know what are they via the <strong>taxonomy profiling</strong> step, which could mean that they might be identified as pathogens in the coming steps, and if we delete them we are possibly losing important information and losing the main goal of the workflow to detect pathogens and track them.</p>

<h1 id="gene-based-pathogenic-identification">Gene based pathogenic identification</h1>

<p>With taxonomy profiling, we identified some bacterial species. But we want to be sure they are pathogenic, by <strong>looking for genes known to be linked to pathogenicity or to the pathogenecity character</strong> of the organim:</p>

<ul>
  <li><a href="https://www.sciencedirect.com/topics/immunology-and-microbiology/virulence-factor"><strong>Virulence Factor (VF)</strong></a>: gene products, usually proteins, involved in pathogenicity. By identifiying them we can call a pathogen and its severity level</li>
  <li>
    <p><a href="https://www.sciencedirect.com/topics/engineering/antibiotic-resistance-genes"><strong>Antimicrobial Resistance genes (AMR)</strong></a>.</p>

    <p>These type of genes have three fundamental mechanisms of antimicrobial resistance that are enzymatic degradation of antibacterial drugs, alteration of bacterial proteins that are antimicrobial targets, and changes in membrane permeability to antibiotics, which will lead to not altering the target site and spread throughput the pathogenic bacteria decreasing the overall fitness of the host.</p>
  </li>
</ul>

<p>To look for these genes and determine the strain of the bacteria we are testing for pathogenicity we use <strong>Multilocus Sequence Typing</strong> approach and dedicated <a href="https://pubmlst.org/">pubMLST datases</a> database:</p>

<ol>
  <li>Genome assembly to get contigs, i.e. longer sequences, using <strong>metaflye</strong> (<span class="citation"><a href="#flye">Kolmogorov <i>et al.</i> 2020</a></span>)  then assembly polishing using <a href="https://github.com/nanoporetech/medaka"><strong>medaka consensus pipeline</strong></a> and visualizing the assembly graph using <strong>Bandage Image</strong> (<span class="citation"><a href="#Wick2015">Wick <i>et al.</i> 2015</a></span>)</li>
  <li>Generate an <strong>MLST</strong> report with <strong>MLST</strong> tool that scans genomes against PubMLST schemes</li>
  <li>Generate reports with <strong>AMR</strong> genes and <strong>VF</strong> using <a href="https://github.com/tseemann/abricate"><strong>ABRicate</strong></a></li>
</ol>

<p>As outputs, we will get our <strong>FASTA</strong> and <strong>Tabular</strong> files to track genes and visualize our pathogenic identification. For that we will need one more file to create a report and we can upload it directly:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Import a tabular file via link from <a href="https://zenodo.org/record/7593928">Zenodo</a> or shared data libraries</p>

      <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7593928/files/MLST_Report_Header.tabular
</code></pre></div>      </div>
    </li>
    <li>
      <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">Tabular</code></p>
    </li>
  </ol>
</blockquote>

<div class="Short-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-gene-based-pathogenic-identification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Gene based Pathogenic Identification</div>

    <ol>
      <li><strong>Import the workflow</strong> into Galaxy
        <ul>
          <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/Nanopore_Datasets_Gene_based_pathogenic_Identification.ga">this workflow</a> or download it to your computer.</li>
          <li>Import the workflow into Galaxy</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>Run <strong>Workflow 3: Nanopore Datasets - Gene based Pathogenic Identification</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Nanopore Sequenced Reads Collection”</em>: <code class="language-plaintext highlighter-rouge">Nanopore processed sequenced reads</code> collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing workflow</li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“MLST Report Header”</em>: <code class="language-plaintext highlighter-rouge">MLST Report with Header</code></li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

</div>

<h2 id="assembly">Assembly</h2>

<p>To identify VF or AMR genes, it is better to assemble reads into longer seuqences or contigs, that can be then used to search databases for the presence of any pathogenic gene:</p>

<ul>
  <li>
    <p>Assembly of long-read metagenomic data using <strong>metaflye</strong> or <strong>Flye</strong>.</p>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-assembly-with-flye"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Assembly with Flye</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9+galaxy0" title="Flye tool" aria-role="link"><strong>Flye</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/flye/flye/2.9+galaxy0</span></span> with the following parameters:
            <ul>
              <li>
                <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input reads”</em>: collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing section</p>

                <blockquote class="comment">
                  <div class="box-title comment-title" id="comment-5"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
                  <p>We need to run <strong>Flye</strong> individually on each sample otherwise <strong>Flye</strong> runs by default a co-assembly mode, <em>i.e.</em> it combines reads of both samples together before running the assembly.</p>
                </blockquote>
              </li>
              <li><em>“Mode”</em>: <code class="language-plaintext highlighter-rouge">Nanopore HQ (--nano-hq)</code></li>
              <li><em>“Perform metagenomic assembly”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Reduced contig assembly coverage”</em>: <code class="language-plaintext highlighter-rouge">Disable reduced coverage for initial disjointing assembly</code></li>
            </ul>
          </li>
        </ol>

      </blockquote>

    </div>
  </li>
  <li>
    <p>For the visualization of the assembly graph output from <strong>Flye</strong> we have chosen <strong>Bandage Image</strong>.</p>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-visualization-of-the-assembly-grap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Visualization of the assembly grap</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2" title="Bandage Image tool" aria-role="link"><strong>Bandage Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/bandage/bandage_image/0.8.1+galaxy2</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Graphical Fragment Assembly”</em>: <code class="language-plaintext highlighter-rouge">assembly_graph</code> Assembly graph outputs of <strong>Flye</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
  <li>
    <p>Contig polishing using <strong>medaka</strong> to correct the long, error-prone Nanopore reads</p>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-contig-polishing"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Contig polishing</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline/1.7.2+galaxy0" title="medaka consensus pipeline tool" aria-role="link"><strong>medaka consensus pipeline</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/medaka_consensus_pipeline/medaka_consensus_pipeline/1.7.2+galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select basecalls”</em>: collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing section</li>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select assembly”</em>: <code class="language-plaintext highlighter-rouge">consensus</code> collection output of <strong>Flye</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Select model”</em>: <code class="language-plaintext highlighter-rouge">r941_min_hac_g507</code></li>
              <li><em>“Select output file(s)”</em>: <code class="language-plaintext highlighter-rouge">select all</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>

      <p>To keep information about the provenance of the contigs, we extract the sample names from the <strong>Nanopore processed sequences reads</strong> collection and add it to the contigs files.</p>

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-contig-renaming-to-add-sample-names"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Contig renaming to add sample names</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1" title="FASTA-to-Tabular tool" aria-role="link"><strong>FASTA-to-Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Convert these sequences”</em>: collection output of <strong>medaka consensus pipeline</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2" title="Extract element identifiers tool" aria-role="link"><strong>Extract element identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Dataset collection”</em>: collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing section</li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0" title="Split file tool" aria-role="link"><strong>Split file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Text file to split”</em>: output from <strong>Extract element identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="param_value_from_file" title="Parse parameter value tool" aria-role="link"><strong>Parse parameter value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: param_value_from_file</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file containing parameter to parse out of”</em>: output from <strong>Split file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace tool" aria-role="link"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: collection output of <strong>FASTA-to-Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li>In <em>“Find and Replace”</em>:
                <ul>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
                    <ul>
                      <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">^(.+)$</code></li>
                      <li><em>“Replace with”</em>: output from <strong>Parse parameter value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
                      <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Replace all occurences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">specific column</code>
                        <ul>
                          <li><em>“in column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
                        </ul>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li>Rename the output collection <code class="language-plaintext highlighter-rouge">Contigs</code></li>
        </ol>
      </blockquote>

    </div>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect <strong>Flye</strong> and <strong>Medaka consensus pipeline</strong> output results for <code class="language-plaintext highlighter-rouge">Barcode10</code></p>

  <ol>
    <li>How many different contigs did you get after <strong>Flye</strong>?</li>
    <li>How many were left after <strong>Medaka consensus pipeline</strong>, and what does that mean?</li>
    <li>What is the result of your <strong>Bandage Image</strong>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>After <strong>Flye</strong> we have got 130 contigs</li>
      <li>After <strong>Medaka consensus pipeline</strong> all 130 contigs were kept, which means that the quality of the <strong>Flye</strong> run was high, and as a result the polishing did not remove any of the contigs.</li>
      <li>
        <p>The graph looks like:</p>

        <p><a href="./images/bandage_image_flye_graph.png" rel="noopener noreferrer"><img src="./images/bandage_image_flye_graph.png" alt="Bandage Image Barcode 10 Assembly Graph. " width="1115" height="848" loading="lazy" /></a></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="multilocus-sequence-typing">Multilocus Sequence Typing</h2>

<p><strong>MLST</strong> tool is used to scan the <a href="https://pubmlst.org/">pubMLST</a> database against PubMLST typing schemes. It’s one of the analyses that you can perform on your dataset to determine the allele IDs, you can also detect novel alleles. This step is not essential to identify pathogens and track them in the remainder of this tutorial, however we wanted to show some of the analysis that one can use Galaxy in to understand more about the dataset, as well as identifying the strain that might be a pathogen or not.</p>

<div class="Long-Version">

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-multilocus-sequence-typing"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Multilocus Sequence Typing</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/mlst/mlst/2.22.0" title="MLST tool" aria-role="link"><strong>MLST</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/mlst/mlst/2.22.0</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“input_files”</em>:  collection output of <strong>medaka consensus pipeline</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> FASTA files with contigs</li>
          <li><em>“Specify advanced parameters”</em>: <code class="language-plaintext highlighter-rouge">Yes, see full parameter list</code>
            <ul>
              <li><em>“Output novel alleles”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Automatically set MLST scheme”</em>: <code class="language-plaintext highlighter-rouge">Automatic MLST scheme detection</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

</div>

<p>The output file of the <strong>MLST</strong> tool is a tab-separated output file which contains:</p>
<ul>
  <li>the filename</li>
  <li>the closest PubMLST scheme name</li>
  <li>the ST (sequence type)</li>
  <li>the allele IDs</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect <strong>MLST</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> results</p>

  <ol>
    <li>What is the the closest PubMLST typing scheme name detected by the tool for  <code class="language-plaintext highlighter-rouge">Barcode11</code>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>senterica_achtman_2 (<span class="citation"><a href="#sentericaachtman2">“<b>Multilocus Sequence Typing as a Replacement for Serotyping in Salmonella enterica</b>” 2012</a></span>)</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="antimicrobial-resistance-genes">Antimicrobial Resistance Genes</h2>

<p>Now, to search <strong>AMR</strong> genes among our samples’ contigs, we run <strong>ABRicate</strong> and choose the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6811410/"><strong>NCBI Bacterial Antimicrobial Resistance Gene Database (AMRFinderPlus)</strong></a> from the advanced options of the tool. The tool checks if there is an AMR found or not, if found then in which contig it is, its location on the contig, what the name of the exact product is, what substance it provides resistance against and a lot of other information regarding the found <strong>AMR</strong>.</p>

<div class="Long-Version">

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-antimicrobial-resistance-genes-identification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Antimicrobial Resistance Genes Identification</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1" title="ABRicate tool" aria-role="link"><strong>ABRicate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file (Fasta, Genbank or EMBL file)”</em>: collection output of <strong>medaka consensus pipeline</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> FASTA files with contigs</li>
          <li>In <em>“Advanced options”</em>:
            <ul>
              <li><em>“Database to use - default is ‘resfinder’“</em>: <code class="language-plaintext highlighter-rouge">NCBI Bacterial Antimicrobial Resistance Reference Gene Database</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>
  </blockquote>

</div>

<p>The outputs of <strong>ABRicate</strong> is a tabular file with different columns:</p>
<ol>
  <li><code class="language-plaintext highlighter-rouge">FILE</code>: The filename this hit came from</li>
  <li><code class="language-plaintext highlighter-rouge">SEQUENCE</code>: The sequence in the filename</li>
  <li><code class="language-plaintext highlighter-rouge">START</code>: Start coordinate in the sequence</li>
  <li><code class="language-plaintext highlighter-rouge">END</code>: End coordinate</li>
  <li><code class="language-plaintext highlighter-rouge">STRAND</code>: AMR gene</li>
  <li><code class="language-plaintext highlighter-rouge">GENE</code>: AMR gene</li>
  <li><code class="language-plaintext highlighter-rouge">COVERAGE</code>: What proportion of the gene is in our sequence</li>
  <li><code class="language-plaintext highlighter-rouge">COVERAGE_MA</code>: A visual represenation</li>
  <li><code class="language-plaintext highlighter-rouge">GAPS</code>: Was there any gaps in the alignment - possible pseudogene?</li>
  <li><code class="language-plaintext highlighter-rouge">%COVERAGE</code>: Proportion of gene covered</li>
  <li><code class="language-plaintext highlighter-rouge">%IDENTITY</code>: Proportion of exact nucleotide matches</li>
  <li><code class="language-plaintext highlighter-rouge">DATABASE</code>: The database this sequence comes from</li>
  <li><code class="language-plaintext highlighter-rouge">ACCESSION</code>: The genomic source of the sequence</li>
  <li><code class="language-plaintext highlighter-rouge">PRODUCT</code></li>
  <li><code class="language-plaintext highlighter-rouge">RESISTANCE</code></li>
</ol>

<!--<div class="Long-Version" markdown="1">

> <div class="box-title hands-on-title" id="hands-on-antimicrobial-resistance-genes-identification-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Antimicrobial Resistance Genes Identification</div>
> 2. <span class="tool" data-tool="Cut1" title="Cut tool" aria-role="link"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span> with the following parameters:
>    - *"Cut columns"*: `c13`
>    - <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> *"From"*: `report` (output of the **ABRicate** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run)
>
> 3. <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0" title="Add line to file tool" aria-role="link"><strong>Add line to file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0</span></span> with the following parameters:
>    - *"text to add"*: `Spike2Barcode10` or `Spike2bBarcode11` respectively to the tag chosen for the first parameter (the sample ID)
>    - <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> *"input file"*: `out_file1` (output of the **Cut** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)
{: .hands-on}

</div>-->

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect <strong>ABRicate</strong> output files from <code class="language-plaintext highlighter-rouge">Barcode10</code>and <code class="language-plaintext highlighter-rouge">Barcode11</code> tags</p>

  <ol>
    <li>How many <strong>AMR</strong> genes found in <code class="language-plaintext highlighter-rouge">Barcode10</code> sample, what are they? Give more details about them.</li>
    <li>How many <strong>AMR</strong> genes found in <code class="language-plaintext highlighter-rouge">Barcode11</code> sample, what are they? Give more details about them.</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>5 <strong>AMR</strong> genes were found:
        <ol>
          <li>Tet(C), which resists <a href="https://medlineplus.gov/druginfo/meds/a682098.html">TETRACYCLINE</a>. It was found in contig 158 from the position 1633 till 2808, with 100% coverage, so 100% of gene is covered in this contig.</li>
          <li>2 genes with sulfonamide-resistant dihydropteroate synthase Sul1 products</li>
          <li>2 genes with oxacillin-hydrolyzing class D beta-lactamase OXA-2 products</li>
        </ol>
      </li>
      <li>No <strong>AMR</strong> genes were found by the database in <code class="language-plaintext highlighter-rouge">Barcode11</code> sample.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="virulence-factor-identification">Virulence Factor identification</h2>

<p>In this step we return back to the main goal of the tutorial where we want to identify the pathogens: <strong>identify if the bacteria found in our samples are pathogenic bacteria or not</strong>. One of the ways to do that is to identify if the sequences include genes with a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2646308/">Virulence Facor</a> or not, such that if the samples include gene(s) with a Virulence Factor then it for sure is a pathogen.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-definitions"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Definitions</div>

  <p><strong>Bacterial Pathogen</strong>: A bacterial pathogen is usually defined as any bacterium that has the capacity to cause disease. Its ability to cause disease is called pathogenicity.</p>

  <p><strong>Virulence</strong>: Virulence provides a quantitative measure of the pathogenicity or the likelihood of causing a disease.</p>

  <p><strong>Virulence Factors</strong>: Virulence factors refer to the properties (i.e., gene products) that enable a microorganism to establish itself on or within a host of a particular species and enhance its potential to cause disease. Virulence factors include bacterial toxins, cell surface proteins that mediate bacterial attachment, cell surface carbohydrates and proteins that protect a bacterium, and hydrolytic enzymes that may contribute to the pathogenicity of the bacterium.</p>

</blockquote>

<p>To identifly VFs, we use again <strong>ABRicate</strong> but this time with the <a href="https://pubmed.ncbi.nlm.nih.gov/26578559/"><strong>VFDB</strong></a> from the advanced options of the tool.</p>

<div class="Long-Version">

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-virulence-factor-identification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Virulence Factor identification</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1" title="ABRicate tool" aria-role="link"><strong>ABRicate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/abricate/abricate/1.0.1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file (Fasta, Genbank or EMBL file)”</em>: collection output of <strong>medaka consensus pipeline</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> FASTA files with contigs</li>
          <li>In <em>“Advanced options”</em>:
            <ul>
              <li><em>“Database to use - default is ‘resfinder’“</em>: <code class="language-plaintext highlighter-rouge">VFDB</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename <strong>ABRicate</strong> the output collection <code class="language-plaintext highlighter-rouge">VFs</code></li>
    </ol>
  </blockquote>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect <strong>VFs of genes Identified by VFDB</strong> output file from<code class="language-plaintext highlighter-rouge">Barcode10</code>and <code class="language-plaintext highlighter-rouge">Barcode11</code></p>

  <ol>
    <li>How many different <strong>VFs</strong> gene products were found in <code class="language-plaintext highlighter-rouge">Barcode10</code> sample?</li>
    <li>How many different <strong>VFs</strong> gene products were found in <code class="language-plaintext highlighter-rouge">Barcode11</code> sample?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>107</li>
      <li>97</li>
    </ol>

  </blockquote>
</blockquote>

<p>To prepare the <strong>ABRicate</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output tabulars of both samples for further analysis in the <strong>Pathogen Tracking among all samples</strong> section, tabular manipulation tools such as <strong>Cut</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>Add line to file</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> are used. We mainly use them to filter and keep only the columns of interest, e.g. the Accession IDs of the found genes with a Virulence Factor and the sample ID along with which contig at which exact location.</p>

<div class="Long-Version">

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-formatting"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Formatting</div>
    <ol>
      <li><span class="tool" data-tool="Cut1" title="Cut tool" aria-role="link"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span> with the following parameters:
        <ul>
          <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c13</code></li>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“From”</em>: collection output of <strong>ABRicate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
      <li>
        <p>Rename the outputs <code class="language-plaintext highlighter-rouge">VFs accessions</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0" title="Add line to file tool" aria-role="link"><strong>Add line to file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0</span></span> with the following parameters:
        <ul>
          <li><em>“text to add”</em>: collection output from <strong>Parse parameter value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <code class="language-plaintext highlighter-rouge">VFs accessions</code></li>
        </ul>
      </li>
      <li>Rename the output collection <code class="language-plaintext highlighter-rouge">VFs accessions with SampleID</code></li>
    </ol>
  </blockquote>

</div>

<h1 id="snp-based-pathogenic-identification">SNP based pathogenic identification</h1>

<p>Now we would like to identify pathogens with a third approach based on variant and <a href="https://genome.sph.umich.edu/w/images/e/e6/Seqshop_may_2015_day2_snp_lecture_v2.pdf">single nucleotide polymorphisms (SNPs) calling</a>: comparison of reads to a targeted reference genome, and call the differences between sample reads and reference genomes to identify variants.</p>

<p>For example, if we want to check whether our samples include <a href="https://www.cdc.gov/campylobacter/index.html"><em>Campylobacter</em> pathogenic strains</a> or not, we will map our samples against the reference genome of the <em>Campylobacter</em> species. Variants in specific positions on the genome are queried to judge if these variations would indicate a pathogen or not.</p>

<p>This approach also allows identification of novel alleles and possible <strong>new variants of the pathogen</strong>.</p>

<p>Using this approach, we also build the consensus genome of each sample, so we can later build a phylogenetic tree of all samples’ full genomes and have an insight into events that occurred during the evolution of same or different species in the samples.</p>

<p>In this training, we are testing <em>Salmonella enterica</em>, with different strains of which our samples were spiked. So we will now upload to our history the reference genome of <a href="https://www.ncbi.nlm.nih.gov/genome/?term=txid99287[Organism:exp]"><em>S. enterica</em></a> we originally obtained from the <a href="https://www.ncbi.nlm.nih.gov/">National Center for Biotechnology Information (NCBI) database</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Import a reference genome FASTA file via link from <a href="https://zenodo.org/record/7593928">Zenodo</a> or Galaxy shared data libraries</p>

      <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7593928/files/Salmonella_Ref_genome.fna.gz
</code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<div class="Short-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-snp-based-pathogenic-identification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SNP based Pathogenic Identification</div>

    <ol>
      <li><strong>Import the workflow</strong> into Galaxy
        <ul>
          <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/nanopore_snp_based_pathogenetic_identification.ga">this workflow</a> or download it to your computer.</li>
          <li>Import the workflow into Galaxy</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>Run <strong>Workflow 4: Nanopore Datasets - SNP based Pathogenic Identification</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
        <ul>
          <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Nanopore Sequenced Reads Collection”</em>: <code class="language-plaintext highlighter-rouge">Nanopore processed sequenced reads</code> collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing workflow</li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference Genome of Tested Strain”</em>: <code class="language-plaintext highlighter-rouge">Salmonella_Ref_genome.fna.gz</code></li>
        </ul>
      </li>
    </ol>

  </blockquote>

</div>

<h2 id="variant-calling-or-snp-calling">Variant Calling or SNP Calling</h2>

<p>To identify variants, we</p>

<ol>
  <li>
    <p><strong>Map reads to the reference genome</strong> of the species of the pathogen we want to test our samples against using <strong>Minimap2</strong> (<span class="citation"><a href="#Li2017Minimap2">Li 2017</a></span>) as our datasets are from a Nanopore:</p>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-mapping-to-reference-genome"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Mapping to reference genome</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.24+galaxy0" title="Map with minimap2 tool" aria-role="link"><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.24+galaxy0</span></span> with the following parameters:
            <ul>
              <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Salmonella_Ref_genome.fna.gz</code></li>
                </ul>
              </li>
              <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
                <ul>
                  <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select fastq dataset”</em>: collection output from <strong>Krakentools: Extract Kraken Reads By ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the preprocessing section</li>
                </ul>
              </li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
  <li>
    <p><strong>Identify variants and single nucleotide variants</strong> using <strong>Clair3</strong> (<span class="citation"><a href="#Zheng2021">Zheng <i>et al.</i> 2021</a></span>), which is designed specifically for Nanopore datasets and giving better results than other variant calling tools, as well as being new and up-to-date.</p>

    <blockquote class="comment">
      <div class="box-title comment-title" id="comment-6"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
      <p><a href="https://github.com/nanoporetech/medaka"><strong>Medaka consensus tool</strong> and <strong>medaka variant tool</strong></a> can be also used instead of <strong>Clair3</strong>, they give similar results but they are much slower then <strong>Clair3</strong> and offer fewer options.</p>
    </blockquote>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-variant-or-snp-calling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Variant or SNP Calling</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/0.1.12+galaxy0" title="Clair3 tool" aria-role="link"><strong>Clair3</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/clair3/clair3/0.1.12+galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM/CRAM file input”</em>: collection output of <strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Reference genome source”</em>: <code class="language-plaintext highlighter-rouge">History</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: <code class="language-plaintext highlighter-rouge">Salmonella_Ref_genome.fna.gz</code></li>
                </ul>
              </li>
              <li><em>“Clair3 model”</em>: <code class="language-plaintext highlighter-rouge">Built-in</code>
                <ul>
                  <li><em>“Built-in model”</em>: <code class="language-plaintext highlighter-rouge">r941_prom_hac_g360+g422</code></li>
                </ul>
              </li>
              <li>In <em>“Advanced parameters”</em>:
                <ul>
                  <li><em>“Call with the following ploidy model”</em>: <code class="language-plaintext highlighter-rouge">haploid precise (--haploid_precise)</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
  <li>
    <p><strong>Left-align and normalize indels</strong> using <strong>bcftools norm</strong> (<span class="citation"><a href="#Li20092">Li <i>et al.</i> 2009</a></span>)</p>

    <p>This step:</p>
    <ul>
      <li>checks REF alleles in the output match the reference;</li>
      <li>splits multiallelic sites into multiple rows;</li>
      <li>recovers multiallelics from multiple rows</li>
    </ul>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-left-align-and-normalize-indels"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Left-align and normalize indels</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.9+galaxy1" title="bcftools norm tool" aria-role="link"><strong>bcftools norm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.9+galaxy1</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“VCF/BCF Data”</em>: collection output of <strong>Clair3</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: <code class="language-plaintext highlighter-rouge">Salmonella_Ref_genome.fna.gz</code></li>
                </ul>
              </li>
              <li><em>“output_type”</em>: <code class="language-plaintext highlighter-rouge">uncompressed VCF</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
  <li>
    <p><strong>Filter variants</strong> to keep only the pass and good quality variants using <strong>SnpSift Filter</strong> (<span class="citation"><a href="#Cingolani2012">Cingolani <i>et al.</i> 2012</a></span>)</p>

    <blockquote class="comment">
      <div class="box-title comment-title" id="comment-7"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
      <p><a href="https://csb5.github.io/lofreq/"><strong>LoFreq filter</strong></a> can be also used instead, both tools performs equal and fast results.</p>
    </blockquote>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-filter-variants"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter variants</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1" title="SnpSift Filter tool" aria-role="link"><strong>SnpSift Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_filter/4.3+t.galaxy1</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input variant list in VCF format”</em>: collection output of <strong>bcftools norm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Type of filter expression”</em>: <code class="language-plaintext highlighter-rouge">Simple expression</code>
                <ul>
                  <li><em>“Filter criteria”</em>: <code class="language-plaintext highlighter-rouge">(QUAL &gt; 2)</code></li>
                </ul>
              </li>
              <li><em>“Filter mode”</em>: <code class="language-plaintext highlighter-rouge">Retain selected variants, remove others</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>

    <p>The output is a VCF file. VCF is the standard file format for storing variation data, with different columns:</p>
    <ul>
      <li><code class="language-plaintext highlighter-rouge">#CHROM</code>: Chromosome</li>
      <li><code class="language-plaintext highlighter-rouge">POS</code>: Co-ordinate - The start coordinate of the variant.</li>
      <li><code class="language-plaintext highlighter-rouge">ID</code>: Identifier</li>
      <li><code class="language-plaintext highlighter-rouge">REF</code>: Reference allele - The reference allele is whatever is found in the reference genome. It is not necessarily the major allele.</li>
      <li><code class="language-plaintext highlighter-rouge">ALT</code>: Alternative allele - The alternative allele is the allele found in the sample you are studying.</li>
      <li><code class="language-plaintext highlighter-rouge">QUAL</code>: Score - Quality score out of 100.</li>
      <li><code class="language-plaintext highlighter-rouge">FILTER</code>: Pass/fail - If it passed quality filters.</li>
      <li><code class="language-plaintext highlighter-rouge">INFO</code>: Further information - Allows you to provide further information on the variants. Keys in the INFO field can be defined in header lines above thetable.</li>
      <li><code class="language-plaintext highlighter-rouge">FORMAT</code>: Information about the following columns - The GT in the FORMAT column tells us to expect genotypes in the following columns.</li>
      <li><code class="language-plaintext highlighter-rouge">Individual identifier (optional)</code> - The previous column told us to expect to see genotypes here. The genotype is in the form “0|1”, where 0 indicates the reference allele and 1 indicates the alternative allele, i.e it is heterozygous. The vertical pipe “|” indicates that the genotype is phased, and is used to indicate which chromosome the alleles are on. If this is a slash (/) rather than a vertical pipe, it means we don’t know which chromosome they are on.</li>
    </ul>
  </li>
  <li>
    <p><strong>Extract a tabular report</strong> with Chromosome, Position, Identifier, Reference allele, Alternative allele and Filter from the VCF files using <a href="http://pcingola.github.io/SnpEff/"><strong>SnpSift Extract Fields</strong></a></p>

    <div class="Long-Version">

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-extract-a-tabular-report"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Extract a tabular report</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0" title="SnpSift Extract Fields tool" aria-role="link"><strong>SnpSift Extract Fields</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Variant input file in VCF format”</em>: collection output of <strong>SnpSift Filter</strong></li>
              <li><em>“Fields to extract”</em>: <code class="language-plaintext highlighter-rouge">CHROM POS ID REF ALT FILTER</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>

    </div>
  </li>
</ol>

<blockquote class="question">
  <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Now let’s inspect the outputs for <code class="language-plaintext highlighter-rouge">Barcode10</code>:</p>

  <ol>
    <li>How many variants were found by <strong>Clair3</strong>?</li>
    <li>How many variants were found after quality filtering?</li>
    <li>What is the Strain of the NCBI reference gemome used for identifying the SNPs in our samples?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Before filtering: 2,652</li>
      <li>After filtering 2,489</li>
      <li>Strain <a href="https://bacdive.dsmz.de/strain/5117">LT2</a>, can be inferred searching the Chroms. NCBI Reference Sequence ID: NC_003197.2</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="consensus-genome-building">Consensus Genome Building</h2>

<p>For further anaylsis we have included one more step in this section, where we build the full genome of our samples.</p>

<p>This consensus genome can be used later to compare and relate samples together based on their full genome. In cases such as SARS-CoV2, it is important to do so in order to discover new outbreaks. In this example of the training, it is not really important to draw a tree of the full genomes of the samples as <em>Salmonella</em> does not have such a speedy outbreak as SARS-CoV2 does. However, we decided to include it in the workflow for any further analysis of the users, if needed.</p>

<p>For this step we run <a href="https://samtools.github.io/bcftools/bcftools.html"><strong>bcftools consensus</strong></a> (<span class="citation"><a href="#Li20092">Li <i>et al.</i> 2009</a></span>).</p>

<div class="Long-Version">

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-consensus-genome-building"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Consensus Genome Building</div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3" title="bcftools consensus tool" aria-role="link"><strong>bcftools consensus</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/bcftools_consensus/bcftools_consensus/1.15.1+galaxy3</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“VCF/BCF Data”</em>: collection output of <strong>SnpSift Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reference genome”</em>: <code class="language-plaintext highlighter-rouge">Salmonella_Ref_genome.fna.gz</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>
  </blockquote>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Inspect the <strong>bcftools consensus</strong> output for <code class="language-plaintext highlighter-rouge">Barcode11</code></p>

  <ol>
    <li>How many sequences did we get for the sample? What are they?</li>
    <li>Why?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>We got 2 sequences: the complete genome and the complete <a href="https://www.genome.gov/genetics-glossary/Plasmid">plasmid</a> genome.</li>
      <li>The tool uses the reference genome and the variants found to build the consensus genome of the sample, and the reference genome FASTA file we use includes two sequences a complete one and a plasmid complete one. So the tool constructed both sequences for us to choose from, based on our further analysis.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="pathogen-tracking-among-all-samples">Pathogen Tracking among all samples</h1>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-8"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>If you did not get your <strong>Gene based pathogenic identification</strong> section output files needed yet or you got an error for some reason, you can go on and download them all or the ones missing from Zenodo so you can start this workflow, please don’t forget to create the collections for them as explained in the pervious hands-on.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-optional-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optional Data upload</div>

    <ol>
      <li>
        <p>Import all tabular and FASTA files needed for this section via link from <a href="https://zenodo.org/record/7593928">Zenodo</a> to the new created history:</p>

        <div class="language-text highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7593928/files/VFs_Barcode10.tabular
https://zenodo.org/record/7593928/files/VFs_Barcode11.tabular
https://zenodo.org/record/7593928/files/VFs_accessions_with_SampleID_Barcode10.tabular
https://zenodo.org/record/7593928/files/VFs_accessions_with_SampleID_Barcode11.tabular
https://zenodo.org/record/7593928/files/VFs_accessions_Barcode10.tabular
https://zenodo.org/record/7593928/files/VFs_accessions_Barcode11.tabular
https://zenodo.org/record/7593928/files/Contigs_Barcode10.fasta
https://zenodo.org/record/7593928/files/Contigs_Barcode11.fasta
</code></pre></div>        </div>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>In this last section, we would like to show how to aggregate results and use the results to help tracking pathogenes among samples by:</p>

<ol>
  <li>Drawing a presence-absence heatmap of the identified <strong>VF</strong> genes within all samples to visualize in which samples these genes can be found.</li>
  <li>Drawing a <a href="https://www.sciencedirect.com/topics/medicine-and-dentistry/phylogenetic-tree">phylogenetic tree</a> for each pathogenic gene detected, where we will relate the contigs of the samples together where this gene is found.</li>
</ol>

<p>With these two types of visualizations we can have an overview of all samples and the genes, but also how samples are related to each other i.e. which common pathogenic genes they share. Given the time of the sampling and the location one can easily identify using these graphs, where and when the contamination has occurred among the different samples.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-organize-imported-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Organize imported data</div>

  <p>Follow these steps only if you imported the datasets, but if your <strong>Gene based Pathogentic Identification</strong> part is already finished correctly then skip the following 4 steps.</p>

  <ol>
    <li>
      <p>Create a collection named <code class="language-plaintext highlighter-rouge">VFs</code> with <code class="language-plaintext highlighter-rouge">VFs</code> files</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a dataset collection<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Create a collection named <code class="language-plaintext highlighter-rouge">VFs accessions</code> with <code class="language-plaintext highlighter-rouge">VFs accessions</code> files</p>
    </li>
    <li>
      <p>Create a collection named <code class="language-plaintext highlighter-rouge">VFs accessions with SampleID</code> with <code class="language-plaintext highlighter-rouge">VFs accessions with SampleID</code> files</p>
    </li>
    <li>
      <p>Create a collection named <code class="language-plaintext highlighter-rouge">Contigs</code> with <code class="language-plaintext highlighter-rouge">Contigs</code> files</p>
    </li>
  </ol>

</blockquote>

<div class="Short-Version">

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-all-samples-analysis"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: All Samples Analysis</div>

    <ol>
      <li><strong>Import the workflow</strong> into Galaxy
        <ul>
          <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/Nanopore_Datasets_Pathogen_Tracking_among_all_samples.ga">this workflow</a> or download it to your computer.</li>
          <li>Import the workflow into Galaxy</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>Run <strong>Workflow 5: Nanopore Datasets - Reports of All Samples along with Full genomes and VF genes Phylogenetic trees</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
        <ul>
          <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Contigs”</em>: collection <code class="language-plaintext highlighter-rouge">Contigs</code> output from the <strong>Gene based Pathogentic Identification</strong> workflow</li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“VFs”</em>: collection <code class="language-plaintext highlighter-rouge">VFs</code> output from the <strong>Gene based Pathogentic Identification</strong> workflow</li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“VFs accessions with SampleID”</em>: collection <code class="language-plaintext highlighter-rouge">VFs accessions with SampleID</code> output from the <strong>Gene based Pathogentic Identification</strong> workflow</li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“VFs accessions”</em>: collection <code class="language-plaintext highlighter-rouge">VFs accessions</code> output from the <strong>Gene based Pathogentic Identification</strong> workflow</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

</div>

<h2 id="heatmap">Heatmap</h2>

<p>A heatmap is one of the visualization techniques that can give you a complete overview of all the samples together and whether or not a certain value exists. In this tutorial, we use the heatmap to visualize all samples aside and check which common bacteria pathogen genes are found in samples and which is only found in one of them.</p>

<p>We use <strong>Heatmap w ggplot</strong> tool along with other tabular manipulating tools to prepare the tabular files.</p>

<div class="Long-Version">

  <ol>
    <li>
      <p>Combine VFs accessions for samples into a table and get 0 or 1 for absence / presence</p>

      <blockquote class="notranslate hands_on">
        <div class="box-title hands-on-title" id="hands-on-heatmap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Heatmap</div>

        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0" title="Collapse Collection tool" aria-role="link"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>: <code class="language-plaintext highlighter-rouge">VFs accessions</code> collection output of <strong>cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the <strong>Gene based Pathogentic Identification</strong> section</li>
              <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0" title="Add line to file tool" aria-role="link"><strong>Add line to file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/add_line_to_file/add_line_to_file/0.1.0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input file”</em>: <code class="language-plaintext highlighter-rouge">VFs accessions</code> output from <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“text to add”</em>: <code class="language-plaintext highlighter-rouge">All_VFs</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/1.1.1" title="Multi-Join tool" aria-role="link"><strong>Multi-Join</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/1.1.1</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to join”</em>: <code class="language-plaintext highlighter-rouge">VFs accessions</code> output from <strong>Add line to file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“add additional file”</em>: <code class="language-plaintext highlighter-rouge">VFs accessions with SampleID</code> collection output of <strong>Add line to file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from the <strong>Gene based Pathogentic Identification</strong> section</li>
              <li><em>“Common key column”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              <li><em>“Column with values to preserve”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
              <li><em>“Add header line to the output file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Input files contain a header line (as first line)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“Ignore duplicated keys”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace tool" aria-role="link"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output of <strong>Multi-Join</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li>In <em>“Find and Replace”</em>:
                <ul>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
                    <ul>
                      <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">dataset_(.*?)_</code></li>
                      <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">Sample_</code></li>
                      <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Replace all occurences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Ignore first line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                      <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
                    </ul>
                  </li>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
                    <ul>
                      <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">(\S+)</code></li>
                      <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">Acc_$1</code></li>
                      <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Case-Insensitive search”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Replace all occurences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Ignore first line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace tool" aria-role="link"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output of <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li>In <em>“Find and Replace”</em>:
                <ul>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
                    <ul>
                      <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">Acc_0</code></li>
                      <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                      <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                      <li><em>“Case-Insensitive search”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Replace all occurences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Ignore first line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace tool" aria-role="link"><strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output of <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li>In <em>“Find and Replace”</em>:
                <ul>
                  <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
                    <ul>
                      <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">Acc_\S*</code></li>
                      <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                      <li><em>“Find-Pattern is a regular expression”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Replace all occurences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Ignore first line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      <li><em>“Case-Insensitive search”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                      <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Advanced Cut tool" aria-role="link"><strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: output of <strong>Multi-Join</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Operation”</em>:<code class="language-plaintext highlighter-rouge">Keep</code></li>
              <li><em>“Delimited by”</em>:<code class="language-plaintext highlighter-rouge">Tab</code></li>
              <li><em>“Cut by”</em>:<code class="language-plaintext highlighter-rouge">fields</code></li>
              <li><em>“List of Fields”</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Advanced Cut tool" aria-role="link"><strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: output of the last <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Operation”</em>:<code class="language-plaintext highlighter-rouge">Discard</code></li>
              <li><em>“Delimited by”</em>:<code class="language-plaintext highlighter-rouge">Tab</code></li>
              <li><em>“Cut by”</em>:<code class="language-plaintext highlighter-rouge">fields</code></li>
              <li><em>“List of Fields”</em>:<code class="language-plaintext highlighter-rouge">1,2</code></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="Paste1" title="Paste tool" aria-role="link"><strong>Paste</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Paste1</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Paste”</em>: output of <strong>Advanced Cut</strong> (Tool N. 7) <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“and”</em>: output of <strong>Advanced Cut</strong> (Tool N. 8) <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Delimit by”</em>:<code class="language-plaintext highlighter-rouge">Tab</code></li>
            </ul>
          </li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Draw heatmap</p>

      <blockquote class="hands-on">
        <div class="box-title hands-on-title" id="hands-on-heatmap-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Heatmap</div>
        <ol>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0+galaxy2" title="Transpose tool" aria-role="link"><strong>Transpose</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0+galaxy2</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input tabular dataset”</em>: output of <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            </ul>
          </li>
          <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0" title="Heatmap w ggplot tool" aria-role="link"><strong>Heatmap w ggplot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0</span></span> with the following parameters:
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select table”</em>: output of <strong>Transpose</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Select input dataset options”</em>: <code class="language-plaintext highlighter-rouge">Dataset with header and row names</code>
                <ul>
                  <li><em>“Select column, for row names”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
                  <li><em>“Sample names orientation”</em>: <code class="language-plaintext highlighter-rouge">horizontal</code></li>
                </ul>
              </li>
              <li><em>“Plot title”</em>: <code class="language-plaintext highlighter-rouge">All Samples Common VFs Heatmap</code></li>
              <li>In <em>“Output Options”</em>:
                <ul>
                  <li><em>“width of output”</em>: <code class="language-plaintext highlighter-rouge">15.0</code></li>
                  <li><em>“height of output”</em>: <code class="language-plaintext highlighter-rouge">17.0</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ol>
      </blockquote>
    </li>
  </ol>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Now let’s see how your heatmap looks like, you can zoom-in and out in your Galaxy history.</p>

  <p><a href="./images/heatmap.png" rel="noopener noreferrer"><img src="./images/heatmap.png" alt="Heatmap. " width="926" height="922" loading="lazy" /></a></p>

  <ol>
    <li>Mention three of the common bacterial pathogen genes found in both samples.</li>
    <li>How can the differences in the found <strong>VF</strong> bacteria pathogen genes between the two samples be interpreted?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>A lot of bacteria pathogen <strong>VF</strong> gene products identified by the <strong>VFDB</strong> are common in both samples, three of them are with the following accession number: <strong>NP_461810</strong>, <strong>NP_461809</strong> and <strong>NP_459541</strong></li>
      <li><strong>AAG03023</strong> is only found in <code class="language-plaintext highlighter-rouge">Barcode10</code> sample and <strong>NP_460360</strong> is only found in <code class="language-plaintext highlighter-rouge">Barcode11</code> sample</li>
      <li>
        <p>Both samples were spiked with the same pathogen species, <em>S. enterica</em>, but not the same strain:</p>

        <ul>
          <li><code class="language-plaintext highlighter-rouge">Barcode10</code> sample is spiked with <em>S. enterica subsp. enterica</em> strain</li>
          <li><code class="language-plaintext highlighter-rouge">Barcode11</code> sample is spiked with <em>S. enterica subsp. houtenae</em> strain.</li>
        </ul>

        <p>This can be the main cause of the big similarities and the few differences of the bacteria pathogen <strong>VF</strong> gene products found between both of the two samples.
Other factors such as the <strong>time</strong> and <strong>location</strong> of the sampling may cause other differences. By knowing the metadata of the samples inputted for the workflows in real life we can understand what actually happened. We can have samples with no pathogen found then we start detecting genes from the 7th or 8th sample, then we can identify where and when the pathogen entered the host, and stop the cause of that</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="phylogenetic-tree-building">Phylogenetic Tree building</h2>

<p>Phylogenetic trees can be used to track the evolution of the pathogen between the samples. Therefore, the VFs are used as a marker gene for the pathogen, similar to 16S marker genes for species profiling. We use the VFs since we know they are associated to the pathogenicity of the sample. By observing the created trees one can identify groups of related pathogens. If additional meta data of the samples would be available one could further identify groups that are associated to specific traits such as increase pathogenicity or faster transmission. Consequently, the tree could be used for phylogenetic placement of unknwon samples.</p>

<p>For the phylogenetic trees, for each bacteria pathogen gene found in the samples we use <strong>ClustalW</strong> (<span class="citation"><a href="#Larkin2007">Larkin <i>et al.</i> 2007</a></span>) for <a href="https://www.sciencedirect.com/topics/medicine-and-dentistry/multiple-sequence-alignment">Multiple Sequence Alignment (MSA)</a> needed before constructing a phylogenetic tree, for the tree itself we use <strong>FASTTREE</strong> and <strong>Newick Display</strong> (<span class="citation"><a href="#Dress2008">Dress <i>et al.</i> 2008</a></span>) to visualize it.</p>

<div class="Long-Version">

  <p>To get the sequence to align, we need to extract the sequences of the VFs in the contigs:</p>

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-extract-the-sequences-of-the-vfs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Extract the sequences of the VFs</div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0" title="Collapse Collection tool" aria-role="link"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>: <code class="language-plaintext highlighter-rouge">Contigs</code></li>
          <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0" title="Collapse Collection tool" aria-role="link"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>: <code class="language-plaintext highlighter-rouge">VFs</code></li>
          <li><em>“Keep one header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6" title="Split by group tool" aria-role="link"><strong>Split by group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to split”</em>: output of 2nd <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 13</code></li>
          <li><em>“Include header in splits?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="Remove beginning1" title="Remove beginning tool" aria-role="link"><strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Remove beginning1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from”</em>: output of <strong>Split by group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7" title="Filter sequences by ID tool" aria-role="link"><strong>Filter sequences by ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sequence file to be filtered”</em>: output of 1st <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Filter using the ID list from”</em>: <code class="language-plaintext highlighter-rouge">tabular file</code>
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Tabular file containing sequence identifiers”</em>: output of <strong>Split by group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Column(s) containing sequence identifiers”</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
            </ul>
          </li>
          <li><em>“Output positive matches, negative matches, or both?”</em>: <code class="language-plaintext highlighter-rouge">Just positive matches (ID on list), as a single file</code></li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>We can now run multiple sequence alignment, build the trees for each VF and display them.</p>

  <blockquote class="hands-on">
    <div class="box-title hands-on-title" id="hands-on-phylogenetic-tree-building"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Phylogenetic Tree building</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1" title="ClustalW tool" aria-role="link"><strong>ClustalW</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTA file”</em>: output of <strong>Filter sequences by ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Data type”</em>: <code class="language-plaintext highlighter-rouge">DNA nucleotide sequences</code></li>
          <li><em>“Output alignment format”</em>: <code class="language-plaintext highlighter-rouge">FASTA format</code></li>
          <li><em>“Output complete alignment (or specify part to output)”</em>: <code class="language-plaintext highlighter-rouge">Complete alignment</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1" title="FASTTREE tool" aria-role="link"><strong>FASTTREE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1</span></span> with the following parameters:
        <ul>
          <li><em>“Aligned sequences file (FASTA or Phylip format)”</em>: <code class="language-plaintext highlighter-rouge">fasta</code>
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTA file”</em>: output of <strong>ClustalW</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              <li><em>“Set starting tree(s)”</em>: <code class="language-plaintext highlighter-rouge">No starting trees</code></li>
            </ul>
          </li>
          <li><em>“Protein or nucleotide alignment”</em>: <code class="language-plaintext highlighter-rouge">Nucleotide</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1" title="Newick Display tool" aria-role="link"><strong>Newick Display</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1</span></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Newick file”</em>: output of <strong>FASTTREE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Branch support”</em>: <code class="language-plaintext highlighter-rouge">Hide branch support</code></li>
          <li><em>“Branch length”</em>: <code class="language-plaintext highlighter-rouge">Hide branch length</code></li>
          <li><em>“Image width”</em>: <code class="language-plaintext highlighter-rouge">2000</code></li>
        </ul>
      </li>
    </ol>
  </blockquote>

</div>

<blockquote class="question">
  <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Now let’s see how your trees for the bacteria pathogen gene with accession IDs: <strong>AAF37887</strong> and <strong>NP_459543</strong> look like. To access that go to the output of Newick</p>

  <ol>
    <li>In which samples and contigs is gene <strong>AAF37887</strong> found?</li>
    <li>In which samples and contigs is gene <strong>NP_459543</strong> found?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <p>In the <code class="language-plaintext highlighter-rouge">Barcode10</code>: Contig 119 and <code class="language-plaintext highlighter-rouge">Barcode11</code>: Contig 1</p>

        <p><a href="./images/AAF37887_tree.png" rel="noopener noreferrer"><img src="./images/AAF37887_tree.png" alt="AAF37887_tree. " width="640" height="142" loading="lazy" /></a></p>
      </li>
      <li>
        <p>In the <code class="language-plaintext highlighter-rouge">Barcode10</code>: Contig 90 and <code class="language-plaintext highlighter-rouge">Barcode11</code>: Contig 1</p>

        <p><a href="./images/NP_459543_tree.png" rel="noopener noreferrer"><img src="./images/NP_459543_tree.png" alt="NP_459543_tree. " width="610" height="145" loading="lazy" /></a></p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial, we have tried the workflow designed to detect and track pathogens in our food and drinks. Through out the full workflow we used our Nanopore sequenced datasets from Biolytix and analyzed it, found the pathogens and tracked it. This approach can be summarized with the following scheme:</p>

<p><a href="./images/FoodBorne-Workflow-updated.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="./images/FoodBorne-Workflow-updated.png" alt="Foodborne full workflow big picture. " width="18000" height="6196" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> The complete picture of the workflow used in this training highlighting, not all, but the most important steps done in 5 sub-workflows explained in the training&lt;/figcaption&gt;</figure></a></p>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/metagenomics/faqs/">FAQ page for the Metagenomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Larkin2007">Larkin, M. A., G. Blackshields, N. P. Brown, R. Chenna, P. A. McGettigan <i>et al.</i>, 2007 <b>Clustal W and Clustal X version 2.0</b>. Bioinformatics 23: 2947–2948. <a href="https://doi.org/10.1093/bioinformatics/btm404">10.1093/bioinformatics/btm404</a></li>
<li id="Dress2008">Dress, A. W. M., C. Flamm, G. Fritzsch, S. Gr\~ A1\textfractionsolidus4newald, M. Kruspe <i>et al.</i>, 2008 <b>Noisy: Identification of problematic columns in multiple sequence alignments</b>. Algorithms for Molecular Biology 3: 7. <a href="https://doi.org/10.1186/1748-7188-3-7">10.1186/1748-7188-3-7</a></li>
<li id="Li20092">Li, H., B. Handsaker, A. Wysoker, T. Fennell, J. Ruan <i>et al.</i>, 2009 <b>The Sequence Alignment/Map format and SAMtools</b>. Bioinformatics 25: 2078–2079. <a href="https://doi.org/10.1093/bioinformatics/btp352">10.1093/bioinformatics/btp352</a></li>
<li id="sentericaachtman2"><b>Multilocus Sequence Typing as a Replacement for Serotyping in Salmonella enterica</b>, 2012 PLOS Pathogens 8: e1002776. https://journals.plos.org/plospathogens/article/file?id=10.1371/journal.ppat.1002776&type=printable. <a href="https://doi.org/10.1371/journal.ppat.1002776">10.1371/journal.ppat.1002776</a> <a href="https://app.dimensions.ai/details/publication/pub.1037561963">https://app.dimensions.ai/details/publication/pub.1037561963</a></li>
<li id="Cingolani2012">Cingolani, P., V. M. Patel, M. Coon, T. Nguyen, S. J. Land <i>et al.</i>, 2012 <b>Using Drosophila melanogaster as a Model for Genotoxic Chemical Mutational Studies with a New Program, SnpSift</b>. Frontiers in Genetics 3: <a href="https://doi.org/10.3389/fgene.2012.00035">10.3389/fgene.2012.00035</a></li>
<li id="Wood2014">Wood, D. E., and S. L. Salzberg, 2014 <b>Kraken: ultrafast metagenomic sequence classification using exact alignments</b>. Genome Biology 15: R46. <a href="https://doi.org/10.1186/gb-2014-15-3-r46">10.1186/gb-2014-15-3-r46</a></li>
<li id="Wick2015">Wick, R. R., M. B. Schultz, J. Zobel, and K. E. Holt, 2015 <b>Bandage: interactive visualization of \lessi\greaterde novo\less/i\greater genome assemblies</b>. Bioinformatics 31: 3350–3352. <a href="https://doi.org/10.1093/bioinformatics/btv383">10.1093/bioinformatics/btv383</a></li>
<li id="Ewels2016">Ewels, P., M. Magnusson, M. Käller, and S. Lundin, 2016 <b>MultiQC: summarize analysis results for multiple tools and samples in a single report</b>. Bioinformatics 32: 3047–3048. <a href="https://doi.org/10.1093/bioinformatics/btw354">10.1093/bioinformatics/btw354</a></li>
<li id="Li2017Minimap2">Li, H., 2017 <b>Minimap2: fast pairwise alignment for long nucleotide sequences</b>. <i>https://arxiv.org/abs/1708.01492</i>. <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a></li>
<li id="Chen2018">Chen, C.-H., T. Xiao, H. Xu, P. Jiang, C. A. Meyer <i>et al.</i>, 2018 <b>Improved design and analysis of CRISPR knockout screens</b> (J. Wren, Ed.). Bioinformatics 34: 4095–4101. <a href="https://doi.org/10.1093/bioinformatics/bty450">10.1093/bioinformatics/bty450</a></li>
<li id="flye">Kolmogorov, M., D. Bickhart, B. Behsaz, A. Gurevich, M. Rayko <i>et al.</i>, 2020 <b>metaFlye: scalable long-read metagenome assembly using repeat graphs</b>. Nature Methods 17: 1–8. <a href="https://doi.org/10.1038/s41592-020-00971-x">10.1038/s41592-020-00971-x</a></li>
<li id="CIUFFREDA20211497">Ciuffreda, L., H. Rodríguez-Pérez, and C. Flores, 2021 <b>Nanopore sequencing and its application to the study of microbial communities</b>. Computational and Structural Biotechnology Journal 19: 1497–1511. <a href="https://doi.org/https://doi.org/10.1016/j.csbj.2021.02.020">https://doi.org/10.1016/j.csbj.2021.02.020</a> <a href="https://www.sciencedirect.com/science/article/pii/S2001037021000714">https://www.sciencedirect.com/science/article/pii/S2001037021000714</a></li>
<li id="Zheng2021">Zheng, Z., S. Li, J. Su, A. W.-S. Leung, T.-W. Lam <i>et al.</i>, 2021 <b>Symphonizing pileup and full-alignment for deep learning-based long-read variant calling</b>. <a href="https://doi.org/10.1038/s43588-022-00387-x">10.1038/s43588-022-00387-x</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition (Metagenomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Bérénice Batut, Engy Nasr, Paul Zierep,  <b>Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html">https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metagenomics-pathogen-detection-from-nanopore-foodborne-data,
author = "Bérénice Batut and Engy Nasr and Paul Zierep",
	title = "Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                
<div markdown="1">
<h2 id="funding">Funding</h2>
<p>These individuals or organisations provided funding support for the development of this resource</p>

<div class="row">

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://www.erasmusplus.nl/assets/images/logo.png" alt="Logo">
		
		<a href="/training-material/hall-of-fame/erasmusplus/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		<p>This project (<a href="https://ec.europa.eu/programmes/erasmus-plus/projects/eplus-project-details/#project/2020-1-NL01-KA203-064717"><code class="language-plaintext highlighter-rouge">2020-1-NL01-KA203-064717</code></a>) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.
<img src="https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg" alt="eu flag with the text: with the support of the erasmus programme of the european union" /></p>

	</div>

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://www.eosc-life.eu/wp-content/themes/eosc-life-v2/assets/images/eosclogo.png" alt="Logo">
		
		<a href="/training-material/hall-of-fame/eosc-life/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		<p>EOSC-Life has received funding from the European Union’s Horizon 2020 programme under grant agreement number 824087</p>

	</div>

</div>
</div>


                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: add_line_to_file
  owner: bgruening
  revisions: 8e251055b1a9
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: flye
  owner: bgruening
  revisions: 276f5d8712d5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: flye
  owner: bgruening
  revisions: cb8dfd28c16f
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: split_file_on_column
  owner: bgruening
  revisions: ff2a81aa3f08
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: split_file_to_collection
  owner: bgruening
  revisions: 6cbe2f30c2d7
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: clustalw
  owner: devteam
  revisions: 5b097fde95dd
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fasta_to_tabular
  owner: devteam
  revisions: e7ed3c310b74
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 3d0c7bdf12f5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 5ec9f6bceaee
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: tabular_to_fasta
  owner: devteam
  revisions: 0a7799698fe5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: abricate
  owner: iuc
  revisions: c2ef298da409
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bandage
  owner: iuc
  revisions: b2860df42e16
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bcftools_consensus
  owner: iuc
  revisions: 147de996e34f
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bcftools_consensus
  owner: iuc
  revisions: f867422772d6
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bcftools_norm
  owner: iuc
  revisions: da6fc9f4a367
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bedtools
  owner: iuc
  revisions: a1a923cd89e8
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: clair3
  owner: iuc
  revisions: 63e02ef6e153
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collection_element_identifiers
  owner: iuc
  revisions: d3c07d270a50
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: compose_text_param
  owner: iuc
  revisions: e188c9826e0f
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: datamash_transpose
  owner: iuc
  revisions: ac092723240d
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastp
  owner: iuc
  revisions: dbf9c561ef29
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fasttree
  owner: iuc
  revisions: e005e659ae21
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ggplot2_heatmap
  owner: iuc
  revisions: 3d059e9c2fba
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ggplot2_heatmap
  owner: iuc
  revisions: 10515715c940
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: kraken2
  owner: iuc
  revisions: e674066930b2
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: kraken_biom
  owner: iuc
  revisions: 65eb9962d272
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: krakentools_extract_kraken_reads
  owner: iuc
  revisions: f329328da134
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: medaka_consensus_pipeline
  owner: iuc
  revisions: 79e9a869f40e
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: minimap2
  owner: iuc
  revisions: 11a0d50a54e6
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mlst
  owner: iuc
  revisions: 31812e7a1315
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: 9a913cdee30e
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: nanoplot
  owner: iuc
  revisions: edbb6c5028f5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: newick_utils
  owner: iuc
  revisions: b4163d2f64ab
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: porechop
  owner: iuc
  revisions: 5f76b20953e5
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpsift
  owner: iuc
  revisions: 5fab4f81391d
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpsift
  owner: iuc
  revisions: 5fab4f81391d
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collapse_collections
  owner: nml
  revisions: 90981f86000f
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seq_filter_by_id
  owner: peterjc
  revisions: 141612f8c3e3
  tool_panel_section_label: Metagenomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00208">https://gxy.io/GTN:T00208</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1703464120'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>