<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Generating a single cell matrix using Alevin</title>
        
            <!--
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
-->
<!-- We let users opt-out, so if they've opted in, we need to append the plausible script to the body -->

<script>
var scriptElement = document.createElement("script");
scriptElement.async = true;
scriptElement.defer = true;
scriptElement.setAttribute("data-domain", "training.galaxyproject.org");
scriptElement.src = "https://plausible.galaxyproject.eu/js/plausible.js";

// Appending the script element to the body
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Plausible: opt-in");
	// Wait for the document to be available
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
}
</script>

<!-- JavaScript Error Monitoring, and performance tracking. -->
<!--
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
-->
<script type="text/javascript">
var scriptElement = document.createElement("script");
scriptElement.src = "https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js";
scriptElement.integrity = "sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43";
scriptElement.crossOrigin = "anonymous";

if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	// Appending the script element to the body
	document.addEventListener("DOMContentLoaded", function(event) {
		document.body.appendChild(scriptElement);
	});
		
	Sentry.init({
	  dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
	  release: "galaxy-training-network@ee51c939937f9893c63a0f9494eaef0bfbfa1c6d",
	  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
	  sampleRate: 0.1,
	  tracesSampleRate: 0.1,
	  // Capture Replay for no sessions by default
	  replaysSessionSampleRate: 0.01,
	  // plus for 1% of sessions with an error
	  replaysOnErrorSampleRate: 0.01,
	  // PII OFF
	  sendDefaultPii: false, // Off by default but just in case.
	  environment: "production",
	});
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.js?v=1698193548' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.js?v=1698193548' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>


        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Generating a single cell matrix using Alevin">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-03 07:40:26 +0000">
<meta name="DC.creator" content="Wendi Bacon">
<meta name="DC.creator" content="Jonathan Manning">

        
        
        
        
        <meta name="description" content="Training material and practicals for all kinds of single ...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Generating a single cell matrix using Alevin">
        <meta property="og:description" content="Training material and practicals for all kinds of single ...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-05-03 07:40:26 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Generating a single cell matrix using Alevin",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Generating a single cell matrix using Alevin' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html",
  "timeRequired": "PT2H",
  "teaches": "Generate a cellxgene matrix for droplet-based single cell sequencing data\n - Interpret quality control (QC) plots to make informed decisions on cell thresholds\n - Find relevant information in GTF files for the particulars of their study, and include this in data matrix metadata",
  "keywords": "single-cell, single-cell, 10x, paper-replication, espa√±ol, transcriptomics",
  "description": "The questions this  addresses are:\n - I have some single cell FASTQ files I want to analyse. Where do I start?\n\n\nThe objectives are:\n - Generate a cellxgene matrix for droplet-based single cell sequencing data\n - Interpret quality control (QC) plots to make informed decisions on cell thresholds\n - Find relevant information in GTF files for the particulars of their study, and include this in data matrix metadata\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-intro/slides.html",
      "name": "An introduction to scRNA-seq data analysis",
      "description": "Slides for 'An introduction to scRNA-seq data analysis' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html",
      "name": "Understanding Barcodes",
      "description": "Hands-on for 'Understanding Barcodes' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/nomadscientist/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/nomadscientist/",
      "name": "Wendi Bacon",
      "image": "https://avatars.githubusercontent.com/nomadscientist",
      "description": "Lecturer at The Open University",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-8170-8806",
      "orcid": "https://orcid.org/0000-0002-8170-8806"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pinin4fjords/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pinin4fjords/",
      "name": "Jonathan Manning",
      "image": "https://avatars.githubusercontent.com/pinin4fjords",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "single-cell"
    },
    {
      "@type": "Thing",
      "name": "10x"
    },
    {
      "@type": "Thing",
      "name": "paper-replication"
    },
    {
      "@type": "Thing",
      "name": "espa√±ol"
    },
    {
      "@type": "Thing",
      "name": "transcriptomics"
    }
  ],
  "abstract": "# Introduction"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.theme.js?v=1698193548'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_alevin/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Themes have moved <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Generating a single cell matrix using Alevin</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/nomadscientist/" class="contributor-badge contributor-nomadscientist"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/nomadscientist?s=36" alt="Avatar" width="36" height="36">Wendi Bacon</a><a href="/training-material/hall-of-fame/pinin4fjords/" class="contributor-badge contributor-pinin4fjords"><img src="https://avatars.githubusercontent.com/pinin4fjords?s=36" alt="Avatar" width="36" height="36">Jonathan Manning</a>
		</td>
	</tr>
	


	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hexylena?s=36" alt="Avatar" width="36" height="36">Helena Rasche</a></td>
	</tr>
	

	
	<tr>
		<td><abbr title="These people tested the tutorial to ensure it works correctly for students, or reported issues with the tutorial.">Tester(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin"><img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Avatar" width="36" height="36">Julia Jakiela</a></td>
	</tr>
	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>I have some single cell FASTQ files I want to analyse. Where do I start?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Generate a cellxgene matrix for droplet-based single cell sequencing data</p>
</li>
        
        <li><p>Interpret quality control (QC) plots to make informed decisions on cell thresholds</p>
</li>
        
        <li><p>Find relevant information in GTF files for the particulars of their study, and include this in data matrix metadata</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/single-cell">Single Cell</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> An introduction to scRNA-seq data analysis:
                            
                                <a href="/training-material/topics/single-cell/tutorials/scrna-intro/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Understanding Barcodes:
                            
                            
                                
                                     <a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4574153">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/workflows/" title="Generating a single cell matrix using Alevin workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/single-cell/scrna-case_alevin/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021)
            </a>
        </li>
        
        
        
        
    </ul>
  
</li>
  
   



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">‚úÖ</abbr> ‚≠êÔ∏è
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">‚úÖ</abbr> ‚≠êÔ∏è
		</a>
	</li>
    
    
    
    

    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 3, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>.
             The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00245">https://gxy.io/GTN:T00245</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p>This tutorial will take you from raw FASTQ files to a cell x gene data matrix in AnnData format. What‚Äôs a data matrix, and what‚Äôs AnnData format? Well you‚Äôll find out! Importantly, this is the first step in processing single cell data in order to start analysing it. Currently you have a bunch of strings of <code class="language-plaintext highlighter-rouge">ATGGGCTT</code> etc. in your sequencing files, and what you need to know is how many cells you have and what genes appear in those cells. These steps are the most computationally heavy in the single cell world, as you‚Äôre starting with 100s of millions of reads, each with 4 lines of text. Later on in analysis, this data becomes simple gene counts such as ‚ÄòCell A has 4 GAPDHs‚Äô, which is a lot easier to store! Because of this data overload, we have downsampled the FASTQ files to speed up the analysis a bit. Saying that, you‚Äôre still having to map loads of reads to the massive murine genome, so get yourself a cup of coffee and prepare to analyse!</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#generating-a-matrix" id="markdown-toc-generating-a-matrix">Generating a matrix</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get Data</a></li>
    </ol>
  </li>
  <li><a href="#important-tips-for-easier-analysis" id="markdown-toc-important-tips-for-easier-analysis">Important tips for easier analysis</a>    <ol>
      <li><a href="#generate-a-transcript-to-gene-map" id="markdown-toc-generate-a-transcript-to-gene-map">Generate a transcript to gene map</a></li>
      <li><a href="#generate-a-transcriptome-index--quantify" id="markdown-toc-generate-a-transcriptome-index--quantify">Generate a transcriptome index &amp; quantify!</a></li>
    </ol>
  </li>
  <li><a href="#basic-qc" id="markdown-toc-basic-qc">Basic QC</a>    <ol>
      <li><a href="#generate-an-unprocessed-matrix-in-a-usable-format" id="markdown-toc-generate-an-unprocessed-matrix-in-a-usable-format">Generate an unprocessed matrix in a usable format</a></li>
      <li><a href="#adding-in-gene-metadata" id="markdown-toc-adding-in-gene-metadata">Adding in Gene metadata</a></li>
    </ol>
  </li>
  <li><a href="#emptydrops" id="markdown-toc-emptydrops">emptyDrops</a></li>
  <li><a href="#analysing-multiple-fastq-files" id="markdown-toc-analysing-multiple-fastq-files">Analysing multiple FASTQ files</a></li>
  <li><a href="#mitochondrial-flagging" id="markdown-toc-mitochondrial-flagging">Mitochondrial flagging</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="generating-a-matrix">Generating a matrix</h1>

<p>In this section, we will show you the principles of the initial phase of single-cell RNA-seq analysis: generating expression measures in a matrix. We‚Äôll concentrate on droplet-based (rather than plate-based) methodology, since this is the process with most differences with respect to conventional approaches developed for bulk RNA-seq.</p>

<p>Droplet-based data consists of three components: cell barcodes, unique molecular identifiers (UMIs) and cDNA reads. To generate cell-wise quantifications we need to:</p>

<ul>
  <li>Process cell barcodes, working out which ones correspond to ‚Äòreal‚Äô cells, which to sequencing artefacts, and possibly correct any barcodes likely to be the product of sequencing errors by comparison to more frequent sequences.</li>
  <li>Map biological sequences to the reference genome or transcriptome.</li>
  <li>‚ÄòDe-duplicate‚Äô using the UMIs.</li>
</ul>

<p>This used to be a complex process involving multiple algorithms, or was performed with technology-specific methods (such as 10X‚Äôs ‚ÄòCellranger‚Äô tool)  but is now much simpler thanks to the advent of a few new methods. When selecting methodology for your own work you should consider:</p>

<ul>
  <li><a href="https://github.com/alexdobin/STAR">STARsolo</a> - a droplet-based scRNA-seq-specific variant of the popular genome alignment method STAR. Produces results very close to those of Cellranger (which itself uses STAR under the hood).</li>
  <li><a href="https://www.kallistobus.tools/">Kallisto/ bustools</a> - developed by the originators of the transcriptome quantification method, Kallisto.</li>
  <li><a href="https://salmon.readthedocs.io/en/latest/alevin.html">Alevin</a> - another transcriptome analysis method developed by the authors of the Salmon tool.</li>
</ul>

<p>We‚Äôre going to use Alevin <span class="citation"><a href="#article-Alevin">Srivastava <i>et al.</i> 2019</a></span> for demonstration purposes, but we do not endorse one method over another.</p>

<h2 id="get-data">Get Data</h2>

<p>We‚Äôve provided you with some example data to play with, a small subset of the reads in a mouse dataset of fetal growth restriction <span class="citation"><a href="#Bacon2018">Bacon <i>et al.</i> 2018</a></span> (see the <a href="https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-6945/results/tsne">study in Single Cell Expression Atlas</a> and the <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6945/">project submission</a>). This is a study using the Drop-seq chemistry, however this tutorial is almost identical to a 10x chemistry. We will point out the one tool parameter change you will need to run 10x samples. This data is not carefully curated, standard tutorial data - it‚Äôs real, it‚Äôs messy, it desperately needs filtering, it has background RNA running around, and most of all it will give you a chance to practice your analysis as if this data were yours.</p>

<p>Down-sampled reads and some associated annotation can be imported below. How did I downsample these FASTQ files? Check out <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/pre-processing-with-alevin---part-1---how-to-downsample">this history</a> to find out!</p>

<p>Additionally, to map your reads, you will need a transcriptome to align against (a FASTA) as well as the gene information for each transcript (a gtf) file. You can download these for your species of interest <a href="https://www.ensembl.org/info/data/ftp/index.html">from Ensembl</a>. These files are included in the data import step below. Keep in mind, these are big files, so the fastest way to get these into your Galaxy account is through importing them by history.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-option-1-data-upload-import-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 1: Data upload - Import history</div>

  <ol>
    <li>
      <p>Import history from: <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1pre-processing-with-alevin---input-1">example input history</a></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>Import history</em> button on the top right</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the the history to your name of choice.</p>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-option-2-data-upload-add-to-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 2: Data upload - Add to history</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the Experimental Design table, sequencing reads 1 &amp; 2, the GTF and fasta files from <a href="https://zenodo.org/record/4574153">Zenodo</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4574153/files/Experimental_Design.tabular
https://zenodo.org/record/4574153/files/Mus_musculus.GRCm38.100.gtf.gff
https://zenodo.org/record/4574153/files/Mus_musculus.GRCm38.cdna.all.fa.fasta
https://zenodo.org/record/4574153/files/SLX-7632.TAAGGCGA.N701.s_1.r_1.fq-400k.fastq
https://zenodo.org/record/4574153/files/SLX-7632.TAAGGCGA.N701.s_1.r_2.fq-400k.fastq
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the datasets</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Have a look at the files you now have in your history.</p>
  <ol>
    <li>Which of the FASTQ files do you think contains the barcode sequences?</li>
    <li>Given the chemistry this study should have, are the barcode/UMI reads the correct length?</li>
    <li>What is the ‚ÄòN701‚Äô referring to?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Read 1 (SLX-7632.TAAGGCGA.N701.s_1.r_1.fq-400k) contains the cell barcode and UMI because it is significantly shorter (indeed, 20 bp!) compared to the longer, r_2 transcript read. For ease, rename these files N701-Read1 and N701-Read2.</li>
      <li>You can see Read 1 is only 20 bp long, which for original Drop-Seq is 12 bp for cell barcode and 8 bp for UMI. This is correct! Be warned - 10x Chromium (and many technologies) change their chemistry over time, so particularly when you are accessing public data, you want to check and make sure you have your numbers correct!</li>
      <li>‚ÄòN701‚Äô is referring to an index read. This sample was run alongside 6 other samples, each denoted by an Illumina Nextera Index (N70X). Later, this will tell you batch information. If you look at the ‚ÄòExperimental Design‚Äô file, you‚Äôll see that the N701 sample was from a male wildtype neonatal thymus.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="important-tips-for-easier-analysis">Important tips for easier analysis</h1>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Using tutorial mode<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step‚Ä¶ this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We‚Äôve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren‚Äôt seeing the button.</li>   </ul> </blockquote> </blockquote>
<p><!--END_SNIPPET--></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <ul>
    <li>The Galaxy tool search panel sometimes doesn‚Äôt find the tools we need from the thousands available.</li>
    <li>You‚Äôll have a much easier time selecting tools from the panel (if you aren‚Äôt using tutorial mode!) if you are on the <a href="https://humancellatlas.usegalaxy.eu">https://humancellatlas.usegalaxy.eu</a></li>
  </ul>
</blockquote>

<h2 id="generate-a-transcript-to-gene-map">Generate a transcript to gene map</h2>

<p>Gene-level, rather than transcript-level, quantification is standard in scRNA-seq, which means that the expression level of alternatively spliced RNA molecules are combined to create gene-level values. Droplet-based scRNA-seq techniques only sample one end each transcript, so lack the full-molecule coverage that would be required to accurately quantify different transcript isoforms.</p>

<p>To generate gene-level quantifications based on transcriptome quantification, Alevin and similar tools require a conversion between transcript and gene identifiers. We can derive a transcript-gene conversion from the gene annotations available in genome resources such as Ensembl. The transcripts in such a list need to match the ones we will use later to build a binary transcriptome index. If you were using spike-ins, you‚Äôd need to add these to the transcriptome and the transcript-gene mapping.</p>

<p>In your example data you will see the murine reference annotation as retrieved from Ensembl in GTF format. This annotation contains gene, exon, transcript and all sorts of other information on the sequences. We will use these to generate the transcript/ gene mapping by passing that information to a tool that extracts just the transcript identifiers we need.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Which of the ‚Äòattributes‚Äô in the last column of the GTF files contains the transcript and gene identifiers?</p>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-hint"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-hint" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Hint<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The file is organised such that the last column (headed ‚ÄòGroup‚Äô) contains a wealth of information in the format: attribute1 ‚Äúinformation associated with attribute 1‚Äù;attribute2 ‚Äúinformation associated with attribute 2‚Äù etc.</p>
  </blockquote>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p><em>gene_id</em> and <em>transcript_id</em> are each followed by ‚Äúensembl gene_id‚Äù and ‚Äúensembl transcript_id‚Äù</p>
  </blockquote>
</blockquote>

<p>It‚Äôs now time to parse the GTF file using the <a href="https://bioconductor.org/packages/release/bioc/html/rtracklayer.html">rtracklayer</a> package in R. This parsing will give us a conversion table with a list of transcript identifiers and their corresponding gene identifiers for counting. Additionally, because we will be generating our own binary index (more later!), we also need to input our FASTA so that it can be filtered to only contain transcriptome information found in the GTF.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-a-filtered-fasta-and-transcript-gene-map"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate a filtered FASTA and transcript-gene map</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.52.0+galaxy0" title="GTF2GeneList tool" aria-role="link"><strong>GTF2GeneList</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.52.0+galaxy0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúEnsembl GTF file‚Äù</em>: <code class="language-plaintext highlighter-rouge">GTF file in the history</code> <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-history</span></li>
        <li><em>‚ÄúFeature type for which to derive annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">transcript</code> (Your sequences are transcript sequencing, so this is your starting point)</li>
        <li><em>‚ÄúField to place first in output table‚Äù</em>: <code class="language-plaintext highlighter-rouge">transcript_id</code> (This is accessing the column you identified above!)</li>
        <li><em>‚ÄúSuppress header line in output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (The next tool (Alevin) does not expect a header)</li>
        <li><em>‚ÄúComma-separated list of field names to extract from the GTF (default: use all fields)‚Äù</em>: <code class="language-plaintext highlighter-rouge">transcript_id,gene_id</code> (This calls the first column to be the transcript_id, and the second the gene_id. Thus, your key can turn transcripts into genes)</li>
        <li><em>‚ÄúAppend version to transcript identifiers?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (The Ensembl FASTA files usually have these, and since we need the FASTA transcriptome and the GTF gene information to work together, we need to append these!)</li>
        <li><em>‚ÄúFlag mitochondrial features?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúProvide a cDNA file for extracting annotations and/ or possible filtering?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA-format cDNA/transcript file‚Äù</em>: <code class="language-plaintext highlighter-rouge">FASTA file in your history</code> <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-history</span></li>
        <li><em>‚ÄúAnnotation field to match with sequences‚Äù</em>: <code class="language-plaintext highlighter-rouge">transcript_id</code></li>
        <li><em>‚ÄúFilter the cDNA file to match the annotations?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the annotation table to <code class="language-plaintext highlighter-rouge">Map</code></p>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the uncompressed filtered FASTA file to <code class="language-plaintext highlighter-rouge">Filtered FASTA</code></li>
  </ol>
</blockquote>

<h2 id="generate-a-transcriptome-index--quantify">Generate a transcriptome index &amp; quantify!</h2>

<p>Alevin collapses the steps involved in dealing with dscRNA-seq into a single process. Such tools need to compare the sequences in your sample to a reference containing all the likely transcript sequences (a ‚Äòtranscriptome‚Äô). This will contain the biological transcript sequences known for a given species, and perhaps also technical sequences such as ‚Äòspike ins‚Äô if you have those.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-how-does-alevin-work"><button class="gtn-boxify-button details" type="button" aria-controls="details-how-does-alevin-work" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> Details: How does Alevin work?<span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>To be able to search a transcriptome quickly, Alevin needs to convert the text (FASTA) format sequences into something it can search quickly, called an ‚Äòindex‚Äô. The index is in a binary rather than human-readable format, but allows fast lookup by Alevin. Because the types of biological and technical sequences we need to include in the index can vary between experiments, and because we often want to use the most up-to-date reference sequences from Ensembl or NCBI, we can end up re-making the indices quite often. Making these indices is time-consuming! Have a look at the uncompressed FASTA to see what it starts with.</p>

</blockquote>

<p>We now have:</p>

<ul>
  <li>Barcode/ UMI reads</li>
  <li>cDNA reads</li>
  <li>transcript/ gene mapping</li>
  <li>filtered FASTA</li>
</ul>

<p>We can now run Alevin. In some public instances, Alevin won‚Äôt show up if you search for it. Instead, you may have to click the Single Cell tab at the left and scroll down to the Alevin tool. Alternatively, use Tutorial Mode as described above and you‚Äôll easily navigate to all the tools, and their versions will all be the tried and tested ones of this tutorial. It‚Äôs often a good idea to check your tool versions. To identify which version of a tool you are using, select <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> ‚ÄòVersions‚Äô and choose the appropriate version. In this case the tutorial was built with Alevin Galaxy Version 1.9.0+galaxy2.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-running-alevin"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Running Alevin</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/alevin/alevin/1.9.0+galaxy2" title="Alevin tool" aria-role="link"><strong>Alevin</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/alevin/alevin/1.9.0+galaxy2</span></span></p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Try to fill in the parameters of Alevin using what you know!</p>

        <blockquote class="tip">
          <div class="box-title tip-title" id="tip-strandedness"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-strandedness" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Strandedness?<span class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>The Salmon documentation on ‚ÄòFragment Library Types‚Äô and running the Alevin command is here: <a href="https://salmon.readthedocs.io/en/latest/library_type.html">salmon.readthedocs.io/en/latest/library_type.html</a> and <a href="https://salmon.readthedocs.io/en/latest/alevin.html">salmon.readthedocs.io/en/latest/alevin.html</a>. These links will help here, although keep in mind the image there is drawn with the RNA 5‚Äô on top, whereas in this scRNA-seq protocol, the polyA is captured by its 3‚Äô tail and thus effectively the bottom or reverse strand‚Ä¶)</p>
        </blockquote>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ul>
            <li><em>‚ÄúSelect a reference transcriptome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use one from the history</code>
              <ul>
                <li>You are going to generate the binary index using your filtered FASTA!</li>
              </ul>
            </li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTranscripts FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filtered FASTA</code></li>
            <li><em>‚ÄúSingle or paired-end reads?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMate pair 1‚Äù</em>: <code class="language-plaintext highlighter-rouge">N701-Read1</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMate pair 2‚Äù</em>: <code class="language-plaintext highlighter-rouge">N701-Read2</code></li>
            <li><em>‚ÄúSpecify the strandedness of the reads‚Äù</em>: <code class="language-plaintext highlighter-rouge">Infer automatically (A)</code></li>
            <li><em>‚ÄúType of single-cell protocol‚Äù</em>: <code class="language-plaintext highlighter-rouge">DropSeq Single Cell protocol</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTranscript to gene map file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Map</code></li>
            <li>In <em>‚ÄúExtra output files‚Äù</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Salmon Quant log file</code></li>
                <li>
                  <p><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Features used by the CB classification and their counts at each cell level (--dumpFeatures)</code></p>
                </li>
                <li>Of course you are welcome to select more options and explore the output files (<i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> warning: <em>‚ÄúPer cell level parsimonious Umi graph (‚ÄìdumpUmiGraph)‚Äù</em> will generate over 2 thousand single files), but for this tutorial you will only need to select those specified.</li>
              </ul>
            </li>
            <li>In <em>‚ÄúAdvanced options‚Äù</em>:
              <ul>
                <li><em>‚ÄúDump cell v transcripts count matrix in MTX format‚Äù</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-if-i-m-running-a-10x-sample"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What if I'm running a 10x sample?</div>

  <p>The main parameter that needs changing for a 10X Chromium sample is the ‚ÄòProtocol‚Äô parameter of Alevin. Just select the correct 10x Chemistry there instead.</p>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-alevin-file-names"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Alevin file names</div>

  <p>You will notice that the names of the output files of Alevin are written in a certain convention, mentioning which tool was used and on which files, for example: <em>‚ÄúAlevin on data X, data Y, and others: whitelist‚Äù</em>. Remember that you can always rename the files if you wish! For simplicity, when we refer to those files in the tutorial, we skip the information about tool and only use the second part of the name - in this case it would be simply <em>‚Äúwhitelist‚Äù</em>.</p>
</blockquote>

<p>This tool will take a while to run. Alevin produces many file outputs, not all of which we‚Äôll use. You can refer to the <a href="https://salmon.readthedocs.io/en/latest/alevin.html">Alevin documentation</a> if you‚Äôre curious what they all are, but we‚Äôre most interested in is:</p>

<ul>
  <li>the matrix itself (<em>per-cell gene-count matrix (MTX)</em> - the count by gene and cell)</li>
  <li>the row (cell/ barcode) identifiers (<em>row index (CB-ids)</em>) and</li>
  <li>the column (gene) labels (<em>column headers (gene-ids)</em>).</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>After you‚Äôve run Alevin, <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> look through all the different files. Can you find:</p>
  <ol>
    <li>The Mapping Rate?</li>
    <li>How many cells are present in the matrix output?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> the file <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Salmon log file</em>. You can see the mapping rate is a paltry <code class="language-plaintext highlighter-rouge">25.45%</code>. This is a terrible mapping rate. Why might this be? Remember this was downsampled, and specifically by taking only the last 400,000 reads of the FASTQ file. The overall mapping rate of the file is more like 50%, which is still quite poor, but for early Drop-Seq samples and single-cell data in general, you might expect a slightly poorer mapping rate. 10x samples are much better these days! This is real data, not test data, after all!</li>
      <li>Inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> the file <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>row index (CB-ids)</em>, and you can see it has <code class="language-plaintext highlighter-rouge">2163</code> lines. The rows refer to the cells in the cell x gene matrix. According to this (rough) estimate, your sample has 2163 cells in it!</li>
    </ol>

  </blockquote>

</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-choose-the-appropriate-input-going-forward"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Choose the appropriate input going forward!</div>
  <p>Make certain to use <strong>per-cell gene-count matrix (MTX)</strong> file going forward.</p>
</blockquote>

<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations - you‚Äôve made an expression matrix! We could almost stop here. But it‚Äôs sensible to do some basic QC, and one of the things we can do is look at a barcode rank plot.</p>

<h1 id="basic-qc">Basic QC</h1>

<p>The question we‚Äôre looking to answer here, is: ‚Äúdo we mostly have a single cell per droplet‚Äù? That‚Äôs what experimenters are normally aiming for, but it‚Äôs not entirely straightforward to get exactly one cell per droplet. Sometimes almost no cells make it into droplets, other times we have too many cells in each droplet. At a minimum, we should easily be able to distinguish droplets with cells from those without.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-a-raw-barcode-qc-plot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate a raw barcode QC plot</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot/1.6.1+galaxy2" title="Droplet barcode rank plot tool" aria-role="link"><strong>Droplet barcode rank plot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot/1.6.1+galaxy2</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúInput MTX-format matrix?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúA two-column tab-delimited file, with barcodes in the first column and frequencies in the second‚Äù</em>: Output of Alevin <code class="language-plaintext highlighter-rouge">raw CB classification frequencies</code></li>
        <li><em>‚ÄúLabel to place in plot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcode rank plot (raw barcode frequencies)</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the image output <code class="language-plaintext highlighter-rouge">Barcode Plot - raw barcode frequencies</code></li>
  </ol>
</blockquote>

<p><a href="../../images/scrna-casestudy/wab-raw_barcodes-400k.png" rel="noopener noreferrer"><figure id="figure-1" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-raw_barcodes-400k.png" alt="raw droplet barcode plots-400k. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 1</strong>:</span> 400k subsample raw&lt;/figcaption&gt;</figure></a></p>

<p>Now, the image generated here (400k) isn‚Äôt the most informative - but you are dealing with a fraction of the reads! If you run the total sample (so identical steps above, but with significantly more time!) you‚Äôd get the image below.</p>

<p><a href="../../images/scrna-casestudy/wab-raw_barcodes-total.png" rel="noopener noreferrer"><figure id="figure-2" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-raw_barcodes-total.png" alt="raw droplet barcode plots-total. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Total sample - 32,579,453 reads - raw&lt;/figcaption&gt;</figure></a></p>

<p>This is our own formulation of the barcode plot based on a <a href="https://github.com/COMBINE-lab/salmon/issues/362#issuecomment-490160480">discussion</a> we had with community members. The left hand plots with the smooth lines are the main plots, showing the UMI counts for individual cell barcodes ranked from high to low. We expect a sharp drop-off between cell-containing droplets and ones that are empty or contain only cell debris. Now, this data is not an ideal dataset, so for perspective, in an ideal world with a very clean 10x run, data will look a bit more like the following taken from the lung atlas (see the <a href="https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-6653/results/tsne">study in Single Cell Expression Atlas</a> and the <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-6653/">project submission</a>).</p>

<p><a href="../../images/scrna-casestudy/wab-lung-atlas-barcodes-raw.png" rel="noopener noreferrer"><figure id="figure-3" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-lung-atlas-barcodes-raw.png" alt="raw droplet barcode plots - lung atlas. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Pretty data - raw&lt;/figcaption&gt;</figure></a></p>

<p>In that plot, you can see the clearer ‚Äòknee‚Äô bend, showing the cut-off between empty droplets and cell-containing droplets.</p>

<p>The right hand plots are density plots from the first one, and the thresholds are generated either using <a href="https://bioconductor.org/packages/release/bioc/html/DropletUtils.html">dropletUtils</a> or by the method described in the discussion mentioned above. We could use any of these thresholds to select cells, assuming that anything with fewer counts is not a valid cell. By default, Alevin does something similar, and we can learn something about that by plotting just the barcodes Alevin retains.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-alevin-s-barcode-qc-plot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate Alevin's barcode QC plot</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot/1.6.1+galaxy2" title="Droplet barcode rank plot tool" aria-role="link"><strong>Droplet barcode rank plot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/droplet_barcode_plot/_dropletBarcodePlot/1.6.1+galaxy2</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúInput MTX-format matrix?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúMatrix-market format matrix file, with cells by column (overrides ‚Äìbarcode-frequencies if supplied)‚Äù</em>: <code class="language-plaintext highlighter-rouge">per-cell gene-count matrix (MTX)</code></li>
        <li><em>‚ÄúFor use with ‚Äìmtx-matrix: force interpretation of matrix to assume cells are by row, rather than by column (default)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúLabel to place in plot title‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcode rank plot (Alevin-processed)</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the image output <code class="language-plaintext highlighter-rouge">Barcode Plot - Alevin processed barcodes</code></li>
  </ol>
</blockquote>

<p><a href="../../images/scrna-casestudy/wab-alevin-barcodes-400k.png" rel="noopener noreferrer"><figure id="figure-4" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-alevin-barcodes-400k.png" alt="raw droplet barcode plots - 400k. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 4</strong>:</span> 400k subsample - Alevin processed&lt;/figcaption&gt;</figure></a></p>

<p>And the full sample looks like:</p>

<p><a href="../../images/scrna-casestudy/wab-alevin-barcodes-total.png" rel="noopener noreferrer"><figure id="figure-5" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-alevin-barcodes-total.png" alt="raw droplet barcode plots - total. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Total sample - 32,579,453 reads - Alevin processed&lt;/figcaption&gt;</figure></a></p>

<p>And to round this off, here‚Äôs the lung atlas plot.</p>

<p><a href="../../images/scrna-casestudy/wab-alevin-barcodes-lung.png" rel="noopener noreferrer"><figure id="figure-6" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-alevin-barcodes-lung.png" alt="raw droplet barcode plots - total. " width="1000" height="600" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Pretty data - Alevin processed&lt;/figcaption&gt;</figure></a></p>

<p>You should see a completely vertical drop-off where Alevin has trunctated the distribution (after excluding any cell barcode that had &lt;10 UMI, Alevin then chose a threshold based off the curve and removed all barcodes with fewer UMIs).</p>

<p>In experiments with relatively simple characteristics, this ‚Äòknee detection‚Äô method works relatively well. But some populations (such as our sample!) present difficulties. For instance, sub-populations of small cells may not be distinguished from empty droplets based purely on counts by barcode. Some libraries produce multiple ‚Äòknees‚Äô for multiple sub-populations. The <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1662-y">emptyDrops</a> method has become a popular way of dealing with this. emptyDrops still retains barcodes with very high counts, but also adds in barcodes that can be statistically distinguished from the ambient profiles, even if total counts are similar. In order to ultimately run emptyDrops (or indeed, whatever tool you like that accomplishes biologically relevant thresholding), we first need to re-run Alevin, but prevent it from applying its own less than ideal thresholds.</p>

<p>To use emptyDrops effectively, we need to go back and re-run Alevin, stopping it from applying it‚Äôs own thresholds. Click the re-run icon <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> on any Alevin output in your history, because almost every parameter is the same as before, except you need to change the following:</p>

<h2 id="generate-an-unprocessed-matrix-in-a-usable-format">Generate an unprocessed matrix in a usable format</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-stopping-alevin-from-thresholding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Stopping Alevin from thresholding</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/alevin/alevin/1.9.0+galaxy2" title="Alevin tool" aria-role="link"><strong>Alevin</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/alevin/alevin/1.9.0+galaxy2</span></span> (Click re-run on the last Alevin output)
      <ul>
        <li><em>‚ÄúAdvanced options‚Äù</em></li>
        <li><em>‚ÄúFraction of cellular barcodes to keep‚Äù</em>: ‚Äò1‚Äô - i.e. keep them all!</li>
        <li><em>‚ÄúMinimum frequency for a barcode to be considered‚Äù</em>: ‚Äò3‚Äô - This will only remove cell barcodes with a frequency of less than 3, a low bar to pass but useful way of avoiding processing a bunch of almost certainly empty barcodes</li>
      </ul>
    </li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Re-running a tool<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote>
  <p><!--END_SNIPPET--></p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many cells are in the output now?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">22952</code> cells are in the quants_mat_rows now! Far more than the Alevin-filtered <code class="language-plaintext highlighter-rouge">2163</code>. This needs some serious filtering with EmptyDrops!</li>
    </ol>

  </blockquote>

</blockquote>

<p>Alevin outputs MTX format, which we can pass to the dropletUtils package and run emptyDrops. Unfortunately the matrix is in the wrong orientation for tools expecting files like those produced by 10X software (which dropletUtils does). We need to ‚Äòtransform‚Äô the matrix such that cells are in columns and genes are in rows.</p>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-be-careful"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Be careful!</div>
  <p>Don‚Äôt mix up files from the different Alevin runs! Use the later run, which has higher numbers in the history!</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-transform-matrix"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Transform matrix</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/salmon_kallisto_mtx_to_10x/_salmon_kallisto_mtx_to_10x/0.0.1+galaxy6" title="salmonKallistoMtxTo10x tool" aria-role="link"><strong>salmonKallistoMtxTo10x</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/salmon_kallisto_mtx_to_10x/_salmon_kallisto_mtx_to_10x/0.0.1+galaxy6</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú.mtx-format matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">per-cell gene-count matrix (MTX)</code> (output of <strong>Alevin</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTab-delimited genes file‚Äù</em>: <code class="language-plaintext highlighter-rouge">column headers (gene-ids)</code> (output of <strong>Alevin</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTab-delimited barcodes file‚Äù</em>: <code class="language-plaintext highlighter-rouge">row index (CB-ids)</code> (output of <strong>Alevin</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> ‚ÄòsalmonKallistoMtxTo10x‚Ä¶.:genes‚Äô to <code class="language-plaintext highlighter-rouge">Gene table</code></li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> ‚ÄòsalmonKallistoMtxTo10x‚Ä¶.:barcodes‚Äô to <code class="language-plaintext highlighter-rouge">Barcode table</code></li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> ‚ÄòsalmonKallistoMtxTo10x‚Ä¶.:matrix‚Äô to <code class="language-plaintext highlighter-rouge">Matrix table</code></li>
  </ol>
</blockquote>

<p>The output is a matrix in the correct orientation for the rest of our tools. However, our matrix is looking a bit sparse - for instance, click on <code class="language-plaintext highlighter-rouge">Gene table</code>. I don‚Äôt know about you, but I‚Äôd struggle to have a good biological discussion using only Ensembl gene_ids! What I‚Äôd really like is the more understandable ‚ÄòGAPDH‚Äô or other gene acronym, as well as information on mitochondrial genes so that I can assess if my cells were stressed out or not. In order to prepare our data for emptyDrops, we‚Äôre going to combine this information into an object, and it‚Äôs easiest to add in that information now.</p>

<h2 id="adding-in-gene-metadata">Adding in Gene metadata</h2>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Where can we find this gene information?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Our old friend the GTF file!</p>

  </blockquote>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Which of the ‚Äòattributes‚Äô in the last column of that file contains the gene acronym?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>gene_name</p>

  </blockquote>

</blockquote>

<p>We‚Äôre now going to re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool that extracts information from our GTF file.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-gene-information"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate gene information</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.52.0+galaxy0" title="GTF2GeneList tool" aria-role="link"><strong>GTF2GeneList</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/gtf2gene_list/_ensembl_gtf2gene_list/1.52.0+galaxy0</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúFeature type for which to derive annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene</code></li>
        <li><em>‚ÄúField to place first in output table‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene_id</code></li>
        <li><em>‚ÄúSuppress header line in output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúComma-separated list of field names to extract from the GTF (default: use all fields)‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene_id,gene_name,mito</code></li>
        <li><em>‚ÄúAppend version to transcript identifiers?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFlag mitochondrial features?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> - note, this will auto-fill a bunch of acronyms for searching in the GTF for mitochondrial associated genes. This is good!</li>
        <li><em>‚ÄúFilter a FASTA-format cDNA file to match annotations?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code> - we don‚Äôt need to, we‚Äôre done with the FASTA!</li>
      </ul>
    </li>
    <li>Check that the output file type is <code class="language-plaintext highlighter-rouge">tabular</code>. If not, change the file type by clicking the ‚ÄòEdit attributes‚Äô<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> on the dataset in the history (as if you were renaming the file.) Then click <code class="language-plaintext highlighter-rouge">Datatypes</code> and type in <code class="language-plaintext highlighter-rouge">tabular</code>. Click <code class="language-plaintext highlighter-rouge">Change datatype</code>.)</li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the annotation table to <code class="language-plaintext highlighter-rouge">Gene Information</code></li>
  </ol>
</blockquote>

<p>Inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> the <strong>Gene Information</strong> object in the history. Now you have made a new key for gene_id, with gene name and a column of mitochondrial information (false = not mitochondrial, true = mitochondrial). We need to add this information into the salmonKallistoMtxTo10x output ‚ÄòGene table‚Äô. But we need to keep ‚ÄòGene table‚Äô in the same order, since it is referenced in the ‚ÄòMatrix table‚Äô by row.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-combine-mtx-gene-table-with-gene-information"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Combine MTX Gene Table with Gene Information</div>

  <ol>
    <li><span class="tool" data-tool="join1" title="Join two Datasets tool" aria-role="link"><strong>Join two Datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: join1</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúJoin‚Äù</em>: <code class="language-plaintext highlighter-rouge">Gene Table</code></li>
        <li><em>‚ÄúUsing column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><em>‚Äúwith‚Äù</em>: <code class="language-plaintext highlighter-rouge">Gene Information</code></li>
        <li><em>‚Äúand column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><em>‚ÄúKeep lines of first input that do not join with second input‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúKeep lines of first input that are incomplete‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFill empty columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúKeep the header lines‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <p>If you inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> the object, you‚Äôll see we have joined these tables and now have quite a few gene_id repeats. Let‚Äôs take those out, while keeping the order of the original ‚ÄòGene Table‚Äô.</p>
    </li>
    <li><span class="tool" data-tool="Cut1" title="Cut columns from a table tool" aria-role="link"><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1,c4,c5</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><em>‚ÄúFrom‚Äù</em>: output of <strong>Join two Datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>Rename output <code class="language-plaintext highlighter-rouge">Annotated Gene Table</code></li>
  </ol>
</blockquote>

<p>Inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> your <code class="language-plaintext highlighter-rouge">Annotated Gene Table</code>. That‚Äôs more like it! You now have <code class="language-plaintext highlighter-rouge">gene_id</code>, <code class="language-plaintext highlighter-rouge">gene_name</code>, and <code class="language-plaintext highlighter-rouge">mito</code>. Now let‚Äôs get back to your journey to emptyDrops and sophisticated thresholding of empty droplets!</p>

<h1 id="emptydrops">emptyDrops</h1>

<p>emptyDrops <span class="citation"><a href="#article-emptyDrops">Lun <i>et al.</i> 2019</a></span> works with a specific form of R object called a SingleCellExperiment. We need to convert our transformed MTX files into that form, using the DropletUtils Read10x tool:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-converting-to-singlecellexperiment-format"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Converting to SingleCellExperiment format</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/dropletutils_read_10x/dropletutils_read_10x/1.0.4+galaxy0" title="DropletUtils Read10x tool" aria-role="link"><strong>DropletUtils Read10x</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/dropletutils_read_10x/dropletutils_read_10x/1.0.4+galaxy0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúExpression matrix in sparse matrix format (.mtx)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Matrix table</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGene Table‚Äù</em>: <code class="language-plaintext highlighter-rouge">Annotated Gene Table</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBarcode/cell table‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcode table</code></li>
        <li><em>‚ÄúShould metadata file be added?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output: <code class="language-plaintext highlighter-rouge">SCE Object</code></li>
  </ol>
</blockquote>

<p>Fantastic! Now that our matrix is combined into an object, specifically the SingleCellExperiment format, we can now run emptyDrops! Let‚Äôs get rid of those background droplets containing no cells!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-emptydrops"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Emptydrops</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/dropletutils_empty_drops/dropletutils_empty_drops/1.0.4+galaxy0" title="DropletUtils emptyDrops tool" aria-role="link"><strong>DropletUtils emptyDrops</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/dropletutils_empty_drops/dropletutils_empty_drops/1.0.4+galaxy0</span></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSingleCellExperiment rdata object‚Äù</em>: <code class="language-plaintext highlighter-rouge">SCE Object</code></li>
        <li><em>‚ÄúShould barcodes estimated to have no cells be removed from the output object?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <code class="language-plaintext highlighter-rouge">serialised SingleCellExperiment</code> output as <code class="language-plaintext highlighter-rouge">Emptied-Object</code></p>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <code class="language-plaintext highlighter-rouge">tabular output</code> as <code class="language-plaintext highlighter-rouge">Emptied-Tabular Output</code></li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many cell barcodes remain after the emptyDrops treatment? Why might that be?</p>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-hint-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-hint-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> Tip: Hint<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>If you click on the <code class="language-plaintext highlighter-rouge">Emptied-Object</code> in the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-history</span> history, the text in that window says <code class="language-plaintext highlighter-rouge">37 barcodes</code> or something similar to that - there is an element of random in the algorithm, so yours might differ slightly. Why is this so low?? And why might the number be different?
Consider‚Ä¶is this a complete set of data?</p>
  </blockquote>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Remember this is a subsampled dataset. If you look carefully at the parameters of emptyDrops, you‚Äôll see it set a minimum threshold at 100 UMI. If you look at the barcode plots above for the 400k read sample, you‚Äôll see this is far too stringent for this subsampled data! To satisfy your curiosity, this minimum threshold would yield <code class="language-plaintext highlighter-rouge">4332</code> barcodes for the total sample. Also, the number may vary slightly as the output depends on a large number of random iterations.</p>

  </blockquote>

</blockquote>

<p>We will nevertheless proceed with your majestic annotated expression matrix of 37 cells, ready to go for further processing and analysis! However, the next tutorials we will link to use a tool suite called Scanpy <span class="citation"><a href="#Wolf2018">Wolf <i>et al.</i> 2018</a></span>. You need to convert this SingleCellExperiment object into a format called <code class="language-plaintext highlighter-rouge">annData</code>, which is a variant of a file format called <code class="language-plaintext highlighter-rouge">hdf5</code>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-converting-to-anndata-format"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Converting to AnnData format</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/sceasy_convert/sceasy_convert/0.0.5+galaxy1" title="SCEasy convert tool" aria-role="link"><strong>SCEasy convert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/ebi-gxa/sceasy_convert/sceasy_convert/0.0.5+galaxy1</span></span> with the following parameters:
      <ul>
        <li><em>‚ÄúDirection of conversion‚Äù</em>: <code class="language-plaintext highlighter-rouge">SingleCellExperiment to AnnData</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput object in SingleCellExperiment RDS format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Emptied-Object</code></li>
        <li><em>‚ÄúName of the assay to be transferred as main layer‚Äù</em>: <code class="language-plaintext highlighter-rouge">counts</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">N701-400k-AnnData</code></li>
  </ol>

</blockquote>

<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congrats! Your object is ready to for the scanpy pipeline! You can can check your work against the <a href="https://humancellatlas.usegalaxy.eu/u/j.jakiela/h/generating-a-single-cell-matrix-using-alevin-3">example history</a> (if you used Alevin Galaxy Version 1.5.1+galaxy0, you can check the <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1pre-processing-with-alevin---answer-key">example history</a> run with that version of the tool). You can also compare how the subsampled datasets you‚Äôve generated compare with the <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1pre-processing-with-alevin---total-n701-answer-key">total sample</a> (using Alevin Galaxy Version 1.5.1+galaxy0).</p>

<p>However, it may be that you want to combine this object with others like it, for instance, maybe you ran 5 samples, and you are starting with 10 FASTQ files‚Ä¶</p>

<h1 id="analysing-multiple-fastq-files">Analysing multiple FASTQ files</h1>

<p>This sample was originally one of seven. So to run the other <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/alevin-tutorial---all-samples---400k">12 downsampled FASTQ files</a>, you can use a <a href="https://humancellatlas.usegalaxy.eu/u/j.jakiela/w/copy-of-alevin-15">workflow</a> and run them all at once! All these samples are going to take a while, so go and have several cups of tea‚Ä¶ Or, better yet, I have <a href="https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1generating-a-single-cell-matrix-using-alevin---all-downsample-fastq-processed-to-anndata">run them myself</a>. To combine the resultant files into a single matrix, you can look at the next tutorial in this case study: <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html">Combining datasets after pre-processing</a></p>

<h1 id="mitochondrial-flagging">Mitochondrial flagging</h1>

<p>We have assumed you will be combining multiple files - but if that‚Äôs not the case, you‚Äôll need to perform this step to turn your column of <code class="language-plaintext highlighter-rouge">true</code> and <code class="language-plaintext highlighter-rouge">false</code> labelling the mitochondrial genes into some metrics telling you the % of mitochondrial genes in each cell. You can follow that step here: <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html#mitochondrial-reads">Mitochondrial calculations</a>.</p>

<h1 id="conclusion">Conclusion</h1>

<p><a href="../../images/scrna-casestudy/wab-alevin-part1workflow.png" rel="noopener noreferrer"><figure id="figure-7" style="max-width: 90%; margin:auto;"><img src="../../images/scrna-casestudy/wab-alevin-part1workflow.png" alt="Workflow Part 1. " width="1070" height="884" loading="lazy" />&lt;figcaption&gt;<span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Workflow  - Steps 1-3&lt;/figcaption&gt;</figure></a></p>

<p>We have:</p>

<ul>
  <li>Examined raw read data, annotations and necessary input files for quantification.</li>
  <li>Run Alevin in two different parameterisations, both allowing Alevin to make its own calls on what constitutes empty droplets, and applying emptyDrops instead.</li>
  <li>Deployed barcode rank plots as a way of quickly assessing the signal present in droplet datasets.</li>
  <li>Applied the necessary conversion to pass these data to downstream processes.</li>
</ul>

<p>To discuss with like-minded scientists, join our Gitter channel for all things Galaxy-single cell!
 <a href="https://gitter.im/Galaxy-Training-Network/galaxy-single-cell?utm_source=badge&amp;utm_medium=badge&amp;utm_campaign=pr-badge"><a href="https://badges.gitter.im/Galaxy-Training-Network/galaxy-single-cell.svg" rel="noopener noreferrer"><img src="https://badges.gitter.im/Galaxy-Training-Network/galaxy-single-cell.svg" alt="Gitter. " loading="lazy" /></a></a></p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Create a scanpy-accessible AnnData object from FASTQ files, including relevant gene metadata</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Bacon2018">Bacon, W. A., R. S. Hamilton, Z. Yu, J. Kieckbusch, D. Hawkes <i>et al.</i>, 2018 <b>Single-Cell Analysis Identifies Thymic Maturation Delay in Growth-Restricted Neonatal Mice</b>. Frontiers in Immunology 9: <a href="https://doi.org/10.3389/fimmu.2018.02523">10.3389/fimmu.2018.02523</a></li>
<li id="Wolf2018">Wolf, F. A., P. Angerer, and F. J. Theis, 2018 <b>SCANPY: large-scale single-cell gene expression data analysis</b>. Genome Biology 19: <a href="https://doi.org/10.1186/s13059-017-1382-0">10.1186/s13059-017-1382-0</a></li>
<li id="article-emptyDrops">Lun, A., S. Riesenfeld, T. Andrews, T. Dao, T. Gomes <i>et al.</i>, 2019 <b>EmptyDrops: Distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data</b>. Genome Biology 20: 63. <a href="https://doi.org/10.1186/s13059-019-1662-y">10.1186/s13059-019-1662-y</a></li>
<li id="article-Alevin">Srivastava, A., L. Malik, T. Smith, I. Sudbery, and R. Patro, 2019 <b>Alevin efficiently estimates accurate gene abundances from dscRNA-seq data</b>. Genome Biology 20: <a href="https://doi.org/10.1186/s13059-019-1670-y">10.1186/s13059-019-1670-y</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Generating a single cell matrix using Alevin (Single Cell)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wendi Bacon, Jonathan Manning,  <b>Generating a single cell matrix using Alevin (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scrna-case_alevin,
author = "Wendi Bacon and Jonathan Manning",
	title = "Generating a single cell matrix using Alevin (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine F√∂ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Bj√∂rn Gr√ºning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/single-cell">Single Cell</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Combining datasets after pre-processing:
                            
                            
                                
                                     <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scrna-case_alevin/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: alevin
  owner: bgruening
  revisions: e4c01dcece8b
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: alevin
  owner: bgruening
  revisions: f2000bc73102
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: droplet_barcode_plot
  owner: ebi-gxa
  revisions: 8f605643ccfd
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: dropletutils_empty_drops
  owner: ebi-gxa
  revisions: fa662ca7ae7f
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: dropletutils_read_10x
  owner: ebi-gxa
  revisions: ca62f52e0d7f
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gtf2gene_list
  owner: ebi-gxa
  revisions: 14b3f2a4523b
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: salmon_kallisto_mtx_to_10x
  owner: ebi-gxa
  revisions: 88f5a0a6a818
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sceasy_convert
  owner: ebi-gxa
  revisions: f62bc418173f
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/ee51c939937f9893c63a0f9494eaef0bfbfa1c6d">ee51c93</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00245">https://gxy.io/GTN:T00245</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_alevin/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


    </div>
</footer>


        <script src='/training-material/assets/js/bundle.main.js?v=1698193548'></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </body>
</html>